# Changelog

## [0.4.3] 2022-02-04

### Automatic update
- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0`
- `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2`
- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13`
- `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.3`
- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy2`
- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1`
- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.3.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.3.1+galaxy2`
- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0`

## [0.4.2] 2021-12-13

### Added
- Added GitHub Actions workflow. No functional changes.

## [0.4.1] 2021-07-23

### Added

Added RO-Crate metadata file. No functional changes.

## [0.4] - 2021-06-16

### Changed

- Upgrade multiqc to 1.9+galaxy1

## [0.3] - 2021-05-19

### Changed

This version brings a number of tweaks to the ivar-dependent steps of the
workflow. Together, these are expected to make variant allele frequency
calculations more precise, in general, and robust in the face of an increasing
number of variants at primer binding sites:

- Upgrade ivar from version 1.2.2 to 1.3.1
  This affects ivar trim and ivar removereads
- Use the newly introduced -f option of ivar trim to exclude read pairs from
  further analysis that extend beyond amplicon boundaries.
  This change should be benefitial for accurate AF calculations in general,
  but in particular for corrected AF values after removal of biased amplicons,
  where aberrant read pairs often represent a larger fraction of the remaining
  reads.
- Run ivar trim only after realignment and addition of indel qualities by
  lofeq. This should make sure that indels close to primer sequences are
  seen as read-internal events.
- Turn the lower and upper thresholds for variant AF that triggers readremoval
  into workflow input parameters and adjust their defaults to trigger read
  removal only in more obvious cases of non-fixed variants.
- Require a minimum depth of coverage for recalled variants after read removal
  of 20 to ensure reliable AF values.
  This change also prevents situations where variants are recalled successfully
  after read removal, but are later excluded from variant reports generated by
  the reporting workflow due to that workflow's min_dp_alt >= 10 filter.

## [0.2]

### Changed

- Turn the AmpliconRemoval variant FILTER into an AmpliconBias INFO flag
- Apply the strand-bias filter only after variant annotation with snpEff. By
  producing fully annotated VCFs with and without filtering, downstream
  workflows can easily be switched between filtered/unfiltered input data

### Fixed

- Make sure the header information about the added flag gets propagated to the
  final VCF

## [0.1]

- Initial version of COVID-19: variation analysis on ARTIC PE data workflow
