<!DOCTYPE html><html><head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<style id="nuxt-ui-colors">:root {
--color-primary-50: 242 236 191;
--color-primary-100: 239 231 174;
--color-primary-200: 232 221 141;
--color-primary-300: 225 211 107;
--color-primary-400: 218 202 73;
--color-primary-500: 208 189 42;
--color-primary-600: 161 147 33;
--color-primary-700: 115 104 23;
--color-primary-800: 68 62 14;
--color-primary-900: 21 19 4;
--color-primary-950: 0 0 0;
--color-primary-DEFAULT: 208 189 42;
--color-primary-DEFAULT: var(--color-primary-500);

--color-gray-50: 223 226 234;
--color-gray-100: 211 214 226;
--color-gray-200: 187 192 210;
--color-gray-300: 162 169 194;
--color-gray-400: 137 146 178;
--color-gray-500: 113 123 162;
--color-gray-600: 93 103 141;
--color-gray-700: 76 85 116;
--color-gray-800: 60 67 92;
--color-gray-900: 44 49 67;
--color-gray-950: 33 37 50;
--color-gray-DEFAULT: 44 49 67;
}

.dark {
  --color-primary-DEFAULT: var(--color-primary-400);
}
</style>
<link rel="icon" type="image/png" href="/iwc_logo_white.png">
<link rel="preload" as="fetch" crossorigin="anonymous" href="/_payload.json?827a8048-e294-42c6-bf63-c21b2e56230e">
<style>*,::backdrop,:after,:before{--tw-border-spacing-x:0;--tw-border-spacing-y:0;--tw-translate-x:0;--tw-translate-y:0;--tw-rotate:0;--tw-skew-x:0;--tw-skew-y:0;--tw-scale-x:1;--tw-scale-y:1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness:proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:#3b82f680;--tw-ring-offset-shadow:0 0 #0000;--tw-ring-shadow:0 0 #0000;--tw-shadow:0 0 #0000;--tw-shadow-colored:0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }/*! tailwindcss v3.4.14 | MIT License | https://tailwindcss.com*/*,:after,:before{border-color:rgb(var(--color-gray-200)/1);border-style:solid;border-width:0;box-sizing:border-box}:after,:before{--tw-content:""}:host,html{line-height:1.5;-webkit-text-size-adjust:100%;font-family:ui-sans-serif,system-ui,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;font-feature-settings:normal;font-variation-settings:normal;tab-size:4;-webkit-tap-highlight-color:transparent}body{line-height:inherit;margin:0}hr{border-top-width:1px;color:inherit;height:0}abbr:where([title]){-webkit-text-decoration:underline dotted;text-decoration:underline dotted}h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit}a{color:inherit;text-decoration:inherit}b,strong{font-weight:bolder}code,kbd,pre,samp{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;font-feature-settings:normal;font-size:1em;font-variation-settings:normal}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:initial}sub{bottom:-.25em}sup{top:-.5em}table{border-collapse:collapse;border-color:inherit;text-indent:0}button,input,optgroup,select,textarea{color:inherit;font-family:inherit;font-feature-settings:inherit;font-size:100%;font-variation-settings:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;margin:0;padding:0}button,select{text-transform:none}button,input:where([type=button]),input:where([type=reset]),input:where([type=submit]){-webkit-appearance:button;background-color:initial;background-image:none}:-moz-focusring{outline:auto}:-moz-ui-invalid{box-shadow:none}progress{vertical-align:initial}::-webkit-inner-spin-button,::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}summary{display:list-item}blockquote,dd,dl,figure,h1,h2,h3,h4,h5,h6,hr,p,pre{margin:0}fieldset{margin:0}fieldset,legend{padding:0}menu,ol,ul{list-style:none;margin:0;padding:0}dialog{padding:0}textarea{resize:vertical}input::placeholder,textarea::placeholder{color:rgb(var(--color-gray-400)/1);opacity:1}[role=button],button{cursor:pointer}:disabled{cursor:default}audio,canvas,embed,iframe,img,object,svg,video{display:block;vertical-align:middle}img,video{height:auto;max-width:100%}[hidden]:where(:not([hidden=until-found])){display:none}.container{width:100%}@media (min-width:640px){.container{max-width:640px}}@media (min-width:768px){.container{max-width:768px}}@media (min-width:1024px){.container{max-width:1024px}}@media (min-width:1280px){.container{max-width:1280px}}@media (min-width:1536px){.container{max-width:1536px}}.form-input,.form-multiselect,.form-select,.form-textarea{appearance:none;background-color:#fff;border-color:rgb(var(--color-gray-500)/var(--tw-border-opacity,1));border-radius:0;border-width:1px;font-size:1rem;line-height:1.5rem;padding:.5rem .75rem;--tw-shadow:0 0 #0000}.form-input:focus,.form-multiselect:focus,.form-select:focus,.form-textarea:focus{outline:2px solid #0000;outline-offset:2px;--tw-ring-inset:var(--tw-empty,/*!*/ /*!*/);--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:#2563eb;--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);border-color:#2563eb;box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow)}.form-input::placeholder,.form-textarea::placeholder{color:rgb(var(--color-gray-500)/var(--tw-text-opacity,1));opacity:1}.form-input::-webkit-datetime-edit-fields-wrapper{padding:0}.form-input::-webkit-date-and-time-value{min-height:1.5em;text-align:inherit}.form-input::-webkit-datetime-edit{display:inline-flex}.form-input::-webkit-datetime-edit,.form-input::-webkit-datetime-edit-day-field,.form-input::-webkit-datetime-edit-hour-field,.form-input::-webkit-datetime-edit-meridiem-field,.form-input::-webkit-datetime-edit-millisecond-field,.form-input::-webkit-datetime-edit-minute-field,.form-input::-webkit-datetime-edit-month-field,.form-input::-webkit-datetime-edit-second-field,.form-input::-webkit-datetime-edit-year-field{padding-bottom:0;padding-top:0}.form-select{background-image:url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 20 20'%3E%3Cpath stroke='%236b7280' stroke-linecap='round' stroke-linejoin='round' stroke-width='1.5' d='m6 8 4 4 4-4'/%3E%3C/svg%3E");background-position:right .5rem center;background-repeat:no-repeat;background-size:1.5em 1.5em;padding-right:2.5rem;-webkit-print-color-adjust:exact;print-color-adjust:exact}.form-select:where([size]:not([size="1"])){background-image:none;background-position:0 0;background-repeat:unset;background-size:initial;padding-right:.75rem;-webkit-print-color-adjust:unset;print-color-adjust:unset}.form-checkbox,.form-radio{appearance:none;background-color:#fff;background-origin:border-box;border-color:rgb(var(--color-gray-500)/var(--tw-border-opacity,1));border-width:1px;color:#2563eb;display:inline-block;flex-shrink:0;height:1rem;padding:0;-webkit-print-color-adjust:exact;print-color-adjust:exact;-webkit-user-select:none;user-select:none;vertical-align:middle;width:1rem;--tw-shadow:0 0 #0000}.form-checkbox{border-radius:0}.form-radio{border-radius:100%}.form-checkbox:focus,.form-radio:focus{outline:2px solid #0000;outline-offset:2px;--tw-ring-inset:var(--tw-empty,/*!*/ /*!*/);--tw-ring-offset-width:2px;--tw-ring-offset-color:#fff;--tw-ring-color:#2563eb;--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow)}.form-checkbox:checked,.form-radio:checked{background-color:currentColor;background-position:50%;background-repeat:no-repeat;background-size:100% 100%;border-color:#0000}.form-checkbox:checked{background-image:url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23fff' viewBox='0 0 16 16'%3E%3Cpath d='M12.207 4.793a1 1 0 0 1 0 1.414l-5 5a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L6.5 9.086l4.293-4.293a1 1 0 0 1 1.414 0'/%3E%3C/svg%3E")}@media (forced-colors:active) {.form-checkbox:checked{appearance:auto}}.form-radio:checked{background-image:url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23fff' viewBox='0 0 16 16'%3E%3Ccircle cx='8' cy='8' r='3'/%3E%3C/svg%3E")}@media (forced-colors:active) {.form-radio:checked{appearance:auto}}.form-checkbox:checked:focus,.form-checkbox:checked:hover,.form-checkbox:indeterminate,.form-radio:checked:focus,.form-radio:checked:hover{background-color:currentColor;border-color:#0000}.form-checkbox:indeterminate{background-image:url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 16 16'%3E%3Cpath stroke='%23fff' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M4 8h8'/%3E%3C/svg%3E");background-position:50%;background-repeat:no-repeat;background-size:100% 100%}@media (forced-colors:active) {.form-checkbox:indeterminate{appearance:auto}}.form-checkbox:indeterminate:focus,.form-checkbox:indeterminate:hover{background-color:currentColor;border-color:#0000}.prose{color:var(--tw-prose-body);max-width:65ch}.prose :where(p):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:1.25em;margin-top:1.25em}.prose :where([class~=lead]):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-lead);font-size:1.25em;line-height:1.6;margin-bottom:1.2em;margin-top:1.2em}.prose :where(a):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-links);font-weight:500;text-decoration:underline}.prose :where(strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-bold);font-weight:600}.prose :where(a strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(blockquote strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(thead th strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(ol):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:decimal;margin-bottom:1.25em;margin-top:1.25em;padding-inline-start:1.625em}.prose :where(ol[type=A]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:upper-alpha}.prose :where(ol[type=a]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:lower-alpha}.prose :where(ol[type=A s]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:upper-alpha}.prose :where(ol[type=a s]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:lower-alpha}.prose :where(ol[type=I]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:upper-roman}.prose :where(ol[type=i]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:lower-roman}.prose :where(ol[type=I s]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:upper-roman}.prose :where(ol[type=i s]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:lower-roman}.prose :where(ol[type="1"]):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:decimal}.prose :where(ul):not(:where([class~=not-prose],[class~=not-prose] *)){list-style-type:disc;margin-bottom:1.25em;margin-top:1.25em;padding-inline-start:1.625em}.prose :where(ol>li):not(:where([class~=not-prose],[class~=not-prose] *))::marker{color:var(--tw-prose-counters);font-weight:400}.prose :where(ul>li):not(:where([class~=not-prose],[class~=not-prose] *))::marker{color:var(--tw-prose-bullets)}.prose :where(dt):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-weight:600;margin-top:1.25em}.prose :where(hr):not(:where([class~=not-prose],[class~=not-prose] *)){border-color:var(--tw-prose-hr);border-top-width:1px;margin-bottom:3em;margin-top:3em}.prose :where(blockquote):not(:where([class~=not-prose],[class~=not-prose] *)){border-inline-start-color:var(--tw-prose-quote-borders);border-inline-start-width:.25rem;color:var(--tw-prose-quotes);font-style:italic;font-weight:500;margin-bottom:1.6em;margin-top:1.6em;padding-inline-start:1em;quotes:"\201C""\201D""\2018""\2019"}.prose :where(blockquote p:first-of-type):not(:where([class~=not-prose],[class~=not-prose] *)):before{content:open-quote}.prose :where(blockquote p:last-of-type):not(:where([class~=not-prose],[class~=not-prose] *)):after{content:close-quote}.prose :where(h1):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-size:2.25em;font-weight:800;line-height:1.1111111;margin-bottom:.8888889em;margin-top:0}.prose :where(h1 strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-weight:900}.prose :where(h2):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-size:1.5em;font-weight:700;line-height:1.3333333;margin-bottom:1em;margin-top:2em}.prose :where(h2 strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-weight:800}.prose :where(h3):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-size:1.25em;font-weight:600;line-height:1.6;margin-bottom:.6em;margin-top:1.6em}.prose :where(h3 strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-weight:700}.prose :where(h4):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-weight:600;line-height:1.5;margin-bottom:.5em;margin-top:1.5em}.prose :where(h4 strong):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-weight:700}.prose :where(img):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:2em;margin-top:2em}.prose :where(picture):not(:where([class~=not-prose],[class~=not-prose] *)){display:block;margin-bottom:2em;margin-top:2em}.prose :where(video):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:2em;margin-top:2em}.prose :where(kbd):not(:where([class~=not-prose],[class~=not-prose] *)){border-radius:.3125rem;box-shadow:0 0 0 1px rgb(var(--tw-prose-kbd-shadows)/10%),0 3px 0 rgb(var(--tw-prose-kbd-shadows)/10%);color:var(--tw-prose-kbd);font-family:inherit;font-size:.875em;font-weight:500;padding-inline-end:.375em;padding-bottom:.1875em;padding-top:.1875em;padding-inline-start:.375em}.prose :where(code):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-code);font-size:.875em;font-weight:600}.prose :where(code):not(:where([class~=not-prose],[class~=not-prose] *)):before{content:"`"}.prose :where(code):not(:where([class~=not-prose],[class~=not-prose] *)):after{content:"`"}.prose :where(a code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(h1 code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(h2 code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-size:.875em}.prose :where(h3 code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit;font-size:.9em}.prose :where(h4 code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(blockquote code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(thead th code):not(:where([class~=not-prose],[class~=not-prose] *)){color:inherit}.prose :where(pre):not(:where([class~=not-prose],[class~=not-prose] *)){background-color:var(--tw-prose-pre-bg);border-radius:.375rem;color:var(--tw-prose-pre-code);font-size:.875em;font-weight:400;line-height:1.7142857;margin-bottom:1.7142857em;margin-top:1.7142857em;overflow-x:auto;padding-inline-end:1.1428571em;padding-bottom:.8571429em;padding-top:.8571429em;padding-inline-start:1.1428571em}.prose :where(pre code):not(:where([class~=not-prose],[class~=not-prose] *)){background-color:initial;border-radius:0;border-width:0;color:inherit;font-family:inherit;font-size:inherit;font-weight:inherit;line-height:inherit;padding:0}.prose :where(pre code):not(:where([class~=not-prose],[class~=not-prose] *)):before{content:none}.prose :where(pre code):not(:where([class~=not-prose],[class~=not-prose] *)):after{content:none}.prose :where(table):not(:where([class~=not-prose],[class~=not-prose] *)){font-size:.875em;line-height:1.7142857;margin-bottom:2em;margin-top:2em;table-layout:auto;width:100%}.prose :where(thead):not(:where([class~=not-prose],[class~=not-prose] *)){border-bottom-color:var(--tw-prose-th-borders);border-bottom-width:1px}.prose :where(thead th):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-headings);font-weight:600;padding-inline-end:.5714286em;padding-bottom:.5714286em;padding-inline-start:.5714286em;vertical-align:bottom}.prose :where(tbody tr):not(:where([class~=not-prose],[class~=not-prose] *)){border-bottom-color:var(--tw-prose-td-borders);border-bottom-width:1px}.prose :where(tbody tr:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){border-bottom-width:0}.prose :where(tbody td):not(:where([class~=not-prose],[class~=not-prose] *)){vertical-align:initial}.prose :where(tfoot):not(:where([class~=not-prose],[class~=not-prose] *)){border-top-color:var(--tw-prose-th-borders);border-top-width:1px}.prose :where(tfoot td):not(:where([class~=not-prose],[class~=not-prose] *)){vertical-align:top}.prose :where(th,td):not(:where([class~=not-prose],[class~=not-prose] *)){text-align:start}.prose :where(figure>*):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:0;margin-top:0}.prose :where(figcaption):not(:where([class~=not-prose],[class~=not-prose] *)){color:var(--tw-prose-captions);font-size:.875em;line-height:1.4285714;margin-top:.8571429em}.prose{--tw-prose-body:#374151;--tw-prose-headings:#111827;--tw-prose-lead:#4b5563;--tw-prose-links:#111827;--tw-prose-bold:#111827;--tw-prose-counters:#6b7280;--tw-prose-bullets:#d1d5db;--tw-prose-hr:#e5e7eb;--tw-prose-quotes:#111827;--tw-prose-quote-borders:#e5e7eb;--tw-prose-captions:#6b7280;--tw-prose-kbd:#111827;--tw-prose-kbd-shadows:17 24 39;--tw-prose-code:#111827;--tw-prose-pre-code:#e5e7eb;--tw-prose-pre-bg:#1f2937;--tw-prose-th-borders:#d1d5db;--tw-prose-td-borders:#e5e7eb;--tw-prose-invert-body:#d1d5db;--tw-prose-invert-headings:#fff;--tw-prose-invert-lead:#9ca3af;--tw-prose-invert-links:#fff;--tw-prose-invert-bold:#fff;--tw-prose-invert-counters:#9ca3af;--tw-prose-invert-bullets:#4b5563;--tw-prose-invert-hr:#374151;--tw-prose-invert-quotes:#f3f4f6;--tw-prose-invert-quote-borders:#374151;--tw-prose-invert-captions:#9ca3af;--tw-prose-invert-kbd:#fff;--tw-prose-invert-kbd-shadows:255 255 255;--tw-prose-invert-code:#fff;--tw-prose-invert-pre-code:#d1d5db;--tw-prose-invert-pre-bg:#00000080;--tw-prose-invert-th-borders:#4b5563;--tw-prose-invert-td-borders:#374151;font-size:1rem;line-height:1.75}.prose :where(picture>img):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:0;margin-top:0}.prose :where(li):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:.5em;margin-top:.5em}.prose :where(ol>li):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-start:.375em}.prose :where(ul>li):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-start:.375em}.prose :where(.prose>ul>li p):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:.75em;margin-top:.75em}.prose :where(.prose>ul>li>p:first-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:1.25em}.prose :where(.prose>ul>li>p:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:1.25em}.prose :where(.prose>ol>li>p:first-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:1.25em}.prose :where(.prose>ol>li>p:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:1.25em}.prose :where(ul ul,ul ol,ol ul,ol ol):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:.75em;margin-top:.75em}.prose :where(dl):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:1.25em;margin-top:1.25em}.prose :where(dd):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:.5em;padding-inline-start:1.625em}.prose :where(hr+*):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:0}.prose :where(h2+*):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:0}.prose :where(h3+*):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:0}.prose :where(h4+*):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:0}.prose :where(thead th:first-child):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-start:0}.prose :where(thead th:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-end:0}.prose :where(tbody td,tfoot td):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-end:.5714286em;padding-bottom:.5714286em;padding-top:.5714286em;padding-inline-start:.5714286em}.prose :where(tbody td:first-child,tfoot td:first-child):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-start:0}.prose :where(tbody td:last-child,tfoot td:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){padding-inline-end:0}.prose :where(figure):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:2em;margin-top:2em}.prose :where(.prose>:first-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-top:0}.prose :where(.prose>:last-child):not(:where([class~=not-prose],[class~=not-prose] *)){margin-bottom:0}.sr-only{height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px;clip:rect(0,0,0,0);border-width:0;white-space:nowrap}.pointer-events-none{pointer-events:none}.pointer-events-auto{pointer-events:auto}.invisible{visibility:hidden}.static{position:static}.fixed{position:fixed}.absolute{position:absolute}.relative{position:relative}.sticky{position:sticky}.inset-0{inset:0}.inset-x-0{left:0;right:0}.inset-y-0{bottom:0;top:0}.-bottom-1{bottom:-.25rem}.-bottom-\[0\.5px\]{bottom:-.5px}.-left-1{left:-.25rem}.-right-1{right:-.25rem}.-top-1{top:-.25rem}.bottom-0{bottom:0}.bottom-4{bottom:1rem}.end-0{inset-inline-end:0}.end-4{inset-inline-end:1rem}.left-0{left:0}.left-\[4px\]{left:4px}.right-0{right:0}.start-0{inset-inline-start:0}.start-4{inset-inline-start:1rem}.top-0{top:0}.top-1\/2{top:50%}.top-4{top:1rem}.top-\[4px\]{top:4px}.z-10{z-index:10}.z-20{z-index:20}.z-50{z-index:50}.z-\[1\]{z-index:1}.z-\[55\]{z-index:55}.col-start-1{grid-column-start:1}.row-start-1{grid-row-start:1}.-m-1\.5{margin:-.375rem}.-mx-1{margin-left:-.25rem;margin-right:-.25rem}.-my-1\.5{margin-bottom:-.375rem;margin-top:-.375rem}.mx-1{margin-left:.25rem;margin-right:.25rem}.mx-3{margin-left:.75rem;margin-right:.75rem}.mx-auto{margin-left:auto;margin-right:auto}.my-2{margin-bottom:.5rem;margin-top:.5rem}.-me-1\.5{margin-inline-end:-.375rem}.-mt-1{margin-top:-.25rem}.mb-1{margin-bottom:.25rem}.mb-1\.5{margin-bottom:.375rem}.mb-10{margin-bottom:2.5rem}.mb-16{margin-bottom:4rem}.mb-2{margin-bottom:.5rem}.mb-4{margin-bottom:1rem}.mb-6{margin-bottom:1.5rem}.mb-8{margin-bottom:2rem}.me-2{margin-inline-end:.5rem}.ml-2{margin-left:.5rem}.mr-2{margin-right:.5rem}.ms-3{margin-inline-start:.75rem}.ms-auto{margin-inline-start:auto}.mt-0{margin-top:0}.mt-1{margin-top:.25rem}.mt-10{margin-top:2.5rem}.mt-16{margin-top:4rem}.mt-2{margin-top:.5rem}.mt-3{margin-top:.75rem}.mt-4{margin-top:1rem}.mt-6{margin-top:1.5rem}.mt-8{margin-top:2rem}.line-clamp-1{display:-webkit-box;overflow:hidden;-webkit-box-orient:vertical;-webkit-line-clamp:1}.block{display:block}.inline-block{display:inline-block}.inline{display:inline}.flex{display:flex}.inline-flex{display:inline-flex}.table{display:table}.grid{display:grid}.inline-grid{display:inline-grid}.hidden{display:none}.h-0\.5{height:.125rem}.h-1{height:.25rem}.h-1\.5{height:.375rem}.h-10{height:2.5rem}.h-12{height:3rem}.h-14{height:3.5rem}.h-16{height:4rem}.h-2{height:.5rem}.h-2\.5{height:.625rem}.h-20{height:5rem}.h-3{height:.75rem}.h-3\.5{height:.875rem}.h-4{height:1rem}.h-5{height:1.25rem}.h-6{height:1.5rem}.h-7{height:1.75rem}.h-8{height:2rem}.h-\[4px\]{height:4px}.h-\[5px\]{height:5px}.h-dvh{height:100dvh}.h-full{height:100%}.h-px{height:1px}.h-screen{height:100vh}.max-h-60{max-height:15rem}.max-h-96{max-height:24rem}.min-h-0{min-height:0}.min-h-full{min-height:100%}.w-0{width:0}.w-1\/4{width:25%}.w-10{width:2.5rem}.w-11{width:2.75rem}.w-12{width:3rem}.w-14{width:3.5rem}.w-16{width:4rem}.w-2{width:.5rem}.w-2\.5{width:.625rem}.w-20{width:5rem}.w-3{width:.75rem}.w-3\/4{width:75%}.w-4{width:1rem}.w-48{width:12rem}.w-5{width:1.25rem}.w-6{width:1.5rem}.w-7{width:1.75rem}.w-8{width:2rem}.w-9{width:2.25rem}.w-\[3\.25rem\]{width:3.25rem}.w-\[3\.75rem\]{width:3.75rem}.w-\[calc\(100\%\+0\.5rem\)\]{width:calc(100% + .5rem)}.w-fit{width:fit-content}.w-full{width:100%}.w-px{width:1px}.w-screen{width:100vw}.min-w-0{min-width:0}.min-w-\[0\.375rem\]{min-width:.375rem}.min-w-\[0\.5rem\]{min-width:.5rem}.min-w-\[0\.625rem\]{min-width:.625rem}.min-w-\[0\.75rem\]{min-width:.75rem}.min-w-\[0\.875rem\]{min-width:.875rem}.min-w-\[1\.25rem\]{min-width:1.25rem}.min-w-\[16px\]{min-width:16px}.min-w-\[1rem\]{min-width:1rem}.min-w-\[20px\]{min-width:20px}.min-w-\[24px\]{min-width:24px}.min-w-\[4px\]{min-width:4px}.min-w-\[5px\]{min-width:5px}.min-w-fit{min-width:fit-content}.min-w-full{min-width:100%}.\!max-w-none{max-width:none!important}.max-w-3xl{max-width:48rem}.max-w-6xl{max-width:72rem}.max-w-7xl{max-width:80rem}.max-w-md{max-width:28rem}.max-w-xs{max-width:20rem}.flex-1{flex:1 1 0%}.flex-none{flex:none}.flex-shrink{flex-shrink:1}.flex-shrink-0{flex-shrink:0}.flex-grow{flex-grow:1}.table-fixed{table-layout:fixed}.-translate-x-1\/2{--tw-translate-x:-50%}.-translate-x-1\/2,.-translate-x-full{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.-translate-x-full{--tw-translate-x:-100%}.-translate-y-1\/2{--tw-translate-y:-50%}.-translate-y-1\/2,.-translate-y-full{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.-translate-y-full{--tw-translate-y:-100%}.translate-x-0{--tw-translate-x:0px}.translate-x-0,.translate-x-1\/2{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-x-1\/2{--tw-translate-x:50%}.translate-x-2{--tw-translate-x:0.5rem}.translate-x-2,.translate-x-2\.5{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-x-2\.5{--tw-translate-x:0.625rem}.translate-x-3{--tw-translate-x:0.75rem}.translate-x-3,.translate-x-4{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-x-4{--tw-translate-x:1rem}.translate-x-5{--tw-translate-x:1.25rem}.translate-x-5,.translate-x-6{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-x-6{--tw-translate-x:1.5rem}.translate-x-7{--tw-translate-x:1.75rem}.translate-x-7,.translate-x-full{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-x-full{--tw-translate-x:100%}.translate-y-0{--tw-translate-y:0px}.translate-y-0,.translate-y-1{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-y-1{--tw-translate-y:0.25rem}.translate-y-1\/2{--tw-translate-y:50%}.translate-y-1\/2,.translate-y-2{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-y-2{--tw-translate-y:0.5rem}.translate-y-4{--tw-translate-y:1rem}.translate-y-4,.translate-y-full{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.translate-y-full{--tw-translate-y:100%}.-rotate-180{--tw-rotate:-180deg}.-rotate-180,.rotate-180{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rotate-180{--tw-rotate:180deg}.scale-100{--tw-scale-x:1;--tw-scale-y:1}.scale-100,.scale-95{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.scale-95{--tw-scale-x:.95;--tw-scale-y:.95}.transform{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}@keyframes pulse{50%{opacity:.5}}.animate-pulse{animation:pulse 2s cubic-bezier(.4,0,.6,1) infinite}@keyframes spin{to{transform:rotate(1turn)}}.animate-spin{animation:spin 1s linear infinite}.cursor-default{cursor:default}.cursor-not-allowed{cursor:not-allowed}.cursor-pointer{cursor:pointer}.select-none{-webkit-user-select:none;user-select:none}.resize-none{resize:none}.resize{resize:both}.snap-x{scroll-snap-type:x var(--tw-scroll-snap-strictness)}.snap-mandatory{--tw-scroll-snap-strictness:mandatory}.snap-center{scroll-snap-align:center}.scroll-py-1{scroll-padding-bottom:.25rem;scroll-padding-top:.25rem}.scroll-py-2{scroll-padding-bottom:.5rem;scroll-padding-top:.5rem}.list-inside{list-style-position:inside}.list-disc{list-style-type:disc}.appearance-none{appearance:none}.grid-cols-1{grid-template-columns:repeat(1,minmax(0,1fr))}.flex-row{flex-direction:row}.flex-row-reverse{flex-direction:row-reverse}.flex-col{flex-direction:column}.flex-nowrap{flex-wrap:nowrap}.content-center{align-content:center}.items-start{align-items:flex-start}.items-end{align-items:flex-end}.items-center{align-items:center}.justify-end{justify-content:flex-end}.justify-center{justify-content:center}.justify-between{justify-content:space-between}.gap-0\.5{gap:.125rem}.gap-1{gap:.25rem}.gap-1\.5{gap:.375rem}.gap-2{gap:.5rem}.gap-3{gap:.75rem}.gap-6{gap:1.5rem}.gap-8{gap:2rem}.gap-x-1{column-gap:.25rem}.gap-x-1\.5{column-gap:.375rem}.gap-x-2{column-gap:.5rem}.gap-x-2\.5{column-gap:.625rem}.-space-x-px>:not([hidden])~:not([hidden]){--tw-space-x-reverse:0;margin-left:calc(-1px*(1 - var(--tw-space-x-reverse)));margin-right:calc(-1px*var(--tw-space-x-reverse))}.-space-y-px>:not([hidden])~:not([hidden]){--tw-space-y-reverse:0;margin-bottom:calc(-1px*var(--tw-space-y-reverse));margin-top:calc(-1px*(1 - var(--tw-space-y-reverse)))}.space-x-2>:not([hidden])~:not([hidden]){--tw-space-x-reverse:0;margin-left:calc(.5rem*(1 - var(--tw-space-x-reverse)));margin-right:calc(.5rem*var(--tw-space-x-reverse))}.space-x-8>:not([hidden])~:not([hidden]){--tw-space-x-reverse:0;margin-left:calc(2rem*(1 - var(--tw-space-x-reverse)));margin-right:calc(2rem*var(--tw-space-x-reverse))}.space-y-2>:not([hidden])~:not([hidden]){--tw-space-y-reverse:0;margin-bottom:calc(.5rem*var(--tw-space-y-reverse));margin-top:calc(.5rem*(1 - var(--tw-space-y-reverse)))}.space-y-3>:not([hidden])~:not([hidden]){--tw-space-y-reverse:0;margin-bottom:calc(.75rem*var(--tw-space-y-reverse));margin-top:calc(.75rem*(1 - var(--tw-space-y-reverse)))}.space-y-6>:not([hidden])~:not([hidden]){--tw-space-y-reverse:0;margin-bottom:calc(1.5rem*var(--tw-space-y-reverse));margin-top:calc(1.5rem*(1 - var(--tw-space-y-reverse)))}.divide-y>:not([hidden])~:not([hidden]){--tw-divide-y-reverse:0;border-bottom-width:calc(1px*var(--tw-divide-y-reverse));border-top-width:calc(1px*(1 - var(--tw-divide-y-reverse)))}.divide-gray-100>:not([hidden])~:not([hidden]){--tw-divide-opacity:1;border-color:rgb(var(--color-gray-100)/var(--tw-divide-opacity))}.divide-gray-200>:not([hidden])~:not([hidden]){--tw-divide-opacity:1;border-color:rgb(var(--color-gray-200)/var(--tw-divide-opacity))}.divide-gray-300>:not([hidden])~:not([hidden]){--tw-divide-opacity:1;border-color:rgb(var(--color-gray-300)/var(--tw-divide-opacity))}.self-center{align-self:center}.overflow-auto{overflow:auto}.overflow-hidden{overflow:hidden}.overflow-x-auto{overflow-x:auto}.overflow-y-auto{overflow-y:auto}.overflow-y-hidden{overflow-y:hidden}.scroll-smooth{scroll-behavior:smooth}.truncate{overflow:hidden;text-overflow:ellipsis}.truncate,.whitespace-nowrap{white-space:nowrap}.break-all{word-break:break-all}.rounded{border-radius:.25rem}.rounded-2xl{border-radius:1rem}.rounded-3xl{border-radius:1.5rem}.rounded-full{border-radius:9999px}.rounded-lg{border-radius:.5rem}.rounded-md{border-radius:.375rem}.rounded-none{border-radius:0}.rounded-sm{border-radius:.125rem}.rounded-xl{border-radius:.75rem}.rounded-b{border-bottom-left-radius:.25rem;border-bottom-right-radius:.25rem}.rounded-b-2xl{border-bottom-left-radius:1rem;border-bottom-right-radius:1rem}.rounded-b-3xl{border-bottom-left-radius:1.5rem;border-bottom-right-radius:1.5rem}.rounded-b-full{border-bottom-left-radius:9999px;border-bottom-right-radius:9999px}.rounded-b-lg{border-bottom-left-radius:.5rem;border-bottom-right-radius:.5rem}.rounded-b-md{border-bottom-left-radius:.375rem;border-bottom-right-radius:.375rem}.rounded-b-none{border-bottom-left-radius:0;border-bottom-right-radius:0}.rounded-b-sm{border-bottom-left-radius:.125rem;border-bottom-right-radius:.125rem}.rounded-b-xl{border-bottom-left-radius:.75rem;border-bottom-right-radius:.75rem}.rounded-e{border-end-end-radius:.25rem;border-start-end-radius:.25rem}.rounded-e-2xl{border-end-end-radius:1rem;border-start-end-radius:1rem}.rounded-e-3xl{border-end-end-radius:1.5rem;border-start-end-radius:1.5rem}.rounded-e-full{border-end-end-radius:9999px;border-start-end-radius:9999px}.rounded-e-lg{border-end-end-radius:.5rem;border-start-end-radius:.5rem}.rounded-e-md{border-end-end-radius:.375rem;border-start-end-radius:.375rem}.rounded-e-none{border-end-end-radius:0;border-start-end-radius:0}.rounded-e-sm{border-end-end-radius:.125rem;border-start-end-radius:.125rem}.rounded-e-xl{border-end-end-radius:.75rem;border-start-end-radius:.75rem}.rounded-s{border-end-start-radius:.25rem;border-start-start-radius:.25rem}.rounded-s-2xl{border-end-start-radius:1rem;border-start-start-radius:1rem}.rounded-s-3xl{border-end-start-radius:1.5rem;border-start-start-radius:1.5rem}.rounded-s-full{border-end-start-radius:9999px;border-start-start-radius:9999px}.rounded-s-lg{border-end-start-radius:.5rem;border-start-start-radius:.5rem}.rounded-s-md{border-end-start-radius:.375rem;border-start-start-radius:.375rem}.rounded-s-none{border-end-start-radius:0;border-start-start-radius:0}.rounded-s-sm{border-end-start-radius:.125rem;border-start-start-radius:.125rem}.rounded-s-xl{border-end-start-radius:.75rem;border-start-start-radius:.75rem}.rounded-t{border-top-left-radius:.25rem;border-top-right-radius:.25rem}.rounded-t-2xl{border-top-left-radius:1rem;border-top-right-radius:1rem}.rounded-t-3xl{border-top-left-radius:1.5rem;border-top-right-radius:1.5rem}.rounded-t-full{border-top-left-radius:9999px;border-top-right-radius:9999px}.rounded-t-lg{border-top-left-radius:.5rem;border-top-right-radius:.5rem}.rounded-t-md{border-top-left-radius:.375rem;border-top-right-radius:.375rem}.rounded-t-none{border-top-left-radius:0;border-top-right-radius:0}.rounded-t-sm{border-top-left-radius:.125rem;border-top-right-radius:.125rem}.rounded-t-xl{border-top-left-radius:.75rem;border-top-right-radius:.75rem}.border{border-width:1px}.border-0{border-width:0}.border-2{border-width:2px}.border-b{border-bottom-width:1px}.border-l-4{border-left-width:4px}.border-s{border-inline-start-width:1px}.border-s-\[2px\]{border-inline-start-width:2px}.border-s-\[3px\]{border-inline-start-width:3px}.border-s-\[4px\]{border-inline-start-width:4px}.border-s-\[5px\]{border-inline-start-width:5px}.border-s-\[6px\]{border-inline-start-width:6px}.border-t{border-top-width:1px}.border-t-8{border-top-width:8px}.border-t-\[2px\]{border-top-width:2px}.border-t-\[3px\]{border-top-width:3px}.border-t-\[4px\]{border-top-width:4px}.border-t-\[5px\]{border-top-width:5px}.border-t-\[6px\]{border-top-width:6px}.border-solid{border-style:solid}.border-dashed{border-style:dashed}.border-dotted{border-style:dotted}.border-none{border-style:none}.border-gray-200{--tw-border-opacity:1;border-color:rgb(var(--color-gray-200)/var(--tw-border-opacity))}.border-gray-300{--tw-border-opacity:1;border-color:rgb(var(--color-gray-300)/var(--tw-border-opacity))}.border-hokey-pokey{--tw-border-opacity:1;border-color:rgb(208 189 42/var(--tw-border-opacity))}.border-orange-500{--tw-border-opacity:1;border-color:rgb(249 115 22/var(--tw-border-opacity))}.border-transparent{border-color:#0000}.bg-amber-400{--tw-bg-opacity:1;background-color:rgb(251 191 36/var(--tw-bg-opacity))}.bg-amber-500{--tw-bg-opacity:1;background-color:rgb(245 158 11/var(--tw-bg-opacity))}.bg-bay-of-many-400{--tw-bg-opacity:1;background-color:rgb(143 185 221/var(--tw-bg-opacity))}.bg-bay-of-many-500{--tw-bg-opacity:1;background-color:rgb(111 165 212/var(--tw-bg-opacity))}.bg-blue-100{--tw-bg-opacity:1;background-color:rgb(219 234 254/var(--tw-bg-opacity))}.bg-blue-400{--tw-bg-opacity:1;background-color:rgb(96 165 250/var(--tw-bg-opacity))}.bg-blue-500{--tw-bg-opacity:1;background-color:rgb(59 130 246/var(--tw-bg-opacity))}.bg-chicago-400{--tw-bg-opacity:1;background-color:rgb(135 135 137/var(--tw-bg-opacity))}.bg-chicago-50{--tw-bg-opacity:1;background-color:rgb(245 245 246/var(--tw-bg-opacity))}.bg-chicago-500{--tw-bg-opacity:1;background-color:rgb(108 108 110/var(--tw-bg-opacity))}.bg-current{background-color:currentColor}.bg-cyan-400{--tw-bg-opacity:1;background-color:rgb(34 211 238/var(--tw-bg-opacity))}.bg-cyan-500{--tw-bg-opacity:1;background-color:rgb(6 182 212/var(--tw-bg-opacity))}.bg-ebony-clay{--tw-bg-opacity:1;background-color:rgb(44 49 67/var(--tw-bg-opacity))}.bg-ebony-clay-400{--tw-bg-opacity:1;background-color:rgb(137 146 178/var(--tw-bg-opacity))}.bg-ebony-clay-500{--tw-bg-opacity:1;background-color:rgb(113 123 162/var(--tw-bg-opacity))}.bg-ebony-clay-800{--tw-bg-opacity:1;background-color:rgb(60 67 92/var(--tw-bg-opacity))}.bg-emerald-400{--tw-bg-opacity:1;background-color:rgb(52 211 153/var(--tw-bg-opacity))}.bg-emerald-500{--tw-bg-opacity:1;background-color:rgb(16 185 129/var(--tw-bg-opacity))}.bg-fuchsia-400{--tw-bg-opacity:1;background-color:rgb(232 121 249/var(--tw-bg-opacity))}.bg-fuchsia-500{--tw-bg-opacity:1;background-color:rgb(217 70 239/var(--tw-bg-opacity))}.bg-gold-400{--tw-bg-opacity:1;background-color:rgb(255 230 13/var(--tw-bg-opacity))}.bg-gold-500{--tw-bg-opacity:1;background-color:rgb(255 215 0/var(--tw-bg-opacity))}.bg-gray-100{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-100)/var(--tw-bg-opacity))}.bg-gray-200{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.bg-gray-200\/75{background-color:rgb(var(--color-gray-200)/.75)}.bg-gray-50{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-50)/var(--tw-bg-opacity))}.bg-gray-500{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-500)/var(--tw-bg-opacity))}.bg-gray-900{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.bg-green-400{--tw-bg-opacity:1;background-color:rgb(74 222 128/var(--tw-bg-opacity))}.bg-green-500{--tw-bg-opacity:1;background-color:rgb(34 197 94/var(--tw-bg-opacity))}.bg-hokey-pokey-400{--tw-bg-opacity:1;background-color:rgb(218 202 73/var(--tw-bg-opacity))}.bg-hokey-pokey-500{--tw-bg-opacity:1;background-color:rgb(208 189 42/var(--tw-bg-opacity))}.bg-indigo-400{--tw-bg-opacity:1;background-color:rgb(129 140 248/var(--tw-bg-opacity))}.bg-indigo-500{--tw-bg-opacity:1;background-color:rgb(99 102 241/var(--tw-bg-opacity))}.bg-lime-400{--tw-bg-opacity:1;background-color:rgb(163 230 53/var(--tw-bg-opacity))}.bg-lime-500{--tw-bg-opacity:1;background-color:rgb(132 204 22/var(--tw-bg-opacity))}.bg-orange-400{--tw-bg-opacity:1;background-color:rgb(251 146 60/var(--tw-bg-opacity))}.bg-orange-500{--tw-bg-opacity:1;background-color:rgb(249 115 22/var(--tw-bg-opacity))}.bg-pink-400{--tw-bg-opacity:1;background-color:rgb(244 114 182/var(--tw-bg-opacity))}.bg-pink-500{--tw-bg-opacity:1;background-color:rgb(236 72 153/var(--tw-bg-opacity))}.bg-primary-100{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-100)/var(--tw-bg-opacity))}.bg-primary-400{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-400)/var(--tw-bg-opacity))}.bg-primary-50{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-50)/var(--tw-bg-opacity))}.bg-primary-500{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-500)/var(--tw-bg-opacity))}.bg-primary-600{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-600)/var(--tw-bg-opacity))}.bg-primary-900{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-900)/var(--tw-bg-opacity))}.bg-primary-950{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-950)/var(--tw-bg-opacity))}.bg-purple-400{--tw-bg-opacity:1;background-color:rgb(192 132 252/var(--tw-bg-opacity))}.bg-purple-500{--tw-bg-opacity:1;background-color:rgb(168 85 247/var(--tw-bg-opacity))}.bg-red-400{--tw-bg-opacity:1;background-color:rgb(248 113 113/var(--tw-bg-opacity))}.bg-red-500{--tw-bg-opacity:1;background-color:rgb(239 68 68/var(--tw-bg-opacity))}.bg-rose-400{--tw-bg-opacity:1;background-color:rgb(251 113 133/var(--tw-bg-opacity))}.bg-rose-500{--tw-bg-opacity:1;background-color:rgb(244 63 94/var(--tw-bg-opacity))}.bg-sky-400{--tw-bg-opacity:1;background-color:rgb(56 189 248/var(--tw-bg-opacity))}.bg-sky-500{--tw-bg-opacity:1;background-color:rgb(14 165 233/var(--tw-bg-opacity))}.bg-teal-400{--tw-bg-opacity:1;background-color:rgb(45 212 191/var(--tw-bg-opacity))}.bg-teal-500{--tw-bg-opacity:1;background-color:rgb(20 184 166/var(--tw-bg-opacity))}.bg-transparent{background-color:initial}.bg-violet-400{--tw-bg-opacity:1;background-color:rgb(167 139 250/var(--tw-bg-opacity))}.bg-violet-500{--tw-bg-opacity:1;background-color:rgb(139 92 246/var(--tw-bg-opacity))}.bg-white{--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.bg-yellow-400{--tw-bg-opacity:1;background-color:rgb(250 204 21/var(--tw-bg-opacity))}.bg-yellow-500{--tw-bg-opacity:1;background-color:rgb(234 179 8/var(--tw-bg-opacity))}.bg-none{background-image:none}.p-0{padding:0}.p-0\.5{padding:.125rem}.p-1{padding:.25rem}.p-1\.5{padding:.375rem}.p-2{padding:.5rem}.p-2\.5{padding:.625rem}.p-4{padding:1rem}.p-6{padding:1.5rem}.p-8{padding:2rem}.p-px{padding:1px}.px-1{padding-left:.25rem;padding-right:.25rem}.px-1\.5{padding-left:.375rem;padding-right:.375rem}.px-2{padding-left:.5rem;padding-right:.5rem}.px-2\.5{padding-left:.625rem;padding-right:.625rem}.px-3{padding-left:.75rem;padding-right:.75rem}.px-3\.5{padding-left:.875rem;padding-right:.875rem}.px-4{padding-left:1rem;padding-right:1rem}.px-6{padding-left:1.5rem;padding-right:1.5rem}.py-0\.5{padding-bottom:.125rem;padding-top:.125rem}.py-1{padding-bottom:.25rem;padding-top:.25rem}.py-1\.5{padding-bottom:.375rem;padding-top:.375rem}.py-14{padding-bottom:3.5rem;padding-top:3.5rem}.py-16{padding-bottom:4rem;padding-top:4rem}.py-2{padding-bottom:.5rem;padding-top:.5rem}.py-2\.5{padding-bottom:.625rem;padding-top:.625rem}.py-3{padding-bottom:.75rem;padding-top:.75rem}.py-3\.5{padding-bottom:.875rem;padding-top:.875rem}.py-4{padding-bottom:1rem;padding-top:1rem}.py-5{padding-bottom:1.25rem;padding-top:1.25rem}.py-6{padding-bottom:1.5rem;padding-top:1.5rem}.py-8{padding-bottom:2rem;padding-top:2rem}.pb-3{padding-bottom:.75rem}.pe-10{padding-inline-end:2.5rem}.pe-11{padding-inline-end:2.75rem}.pe-12{padding-inline-end:3rem}.pe-2{padding-inline-end:.5rem}.pe-7{padding-inline-end:1.75rem}.pe-8{padding-inline-end:2rem}.pe-9{padding-inline-end:2.25rem}.ps-10{padding-inline-start:2.5rem}.ps-11{padding-inline-start:2.75rem}.ps-12{padding-inline-start:3rem}.ps-4{padding-inline-start:1rem}.ps-7{padding-inline-start:1.75rem}.ps-8{padding-inline-start:2rem}.ps-9{padding-inline-start:2.25rem}.pt-1\.5{padding-top:.375rem}.text-left{text-align:left}.text-center{text-align:center}.text-end{text-align:end}.align-middle{vertical-align:middle}.font-mono{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace}.font-sans{font-family:ui-sans-serif,system-ui,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji}.text-2xl{font-size:1.5rem;line-height:2rem}.text-3xl{font-size:1.875rem;line-height:2.25rem}.text-4xl{font-size:2.25rem;line-height:2.5rem}.text-\[10px\]{font-size:10px}.text-\[11px\]{font-size:11px}.text-\[12px\]{font-size:12px}.text-\[14px\]{font-size:14px}.text-\[4px\]{font-size:4px}.text-\[5px\]{font-size:5px}.text-\[6px\]{font-size:6px}.text-\[7px\]{font-size:7px}.text-\[8px\]{font-size:8px}.text-base{font-size:1rem;line-height:1.5rem}.text-lg{font-size:1.125rem;line-height:1.75rem}.text-sm{font-size:.875rem;line-height:1.25rem}.text-xl{font-size:1.25rem;line-height:1.75rem}.text-xs{font-size:.75rem;line-height:1rem}.font-bold{font-weight:700}.font-medium{font-weight:500}.font-normal{font-weight:400}.font-semibold{font-weight:600}.leading-4{line-height:1rem}.leading-5{line-height:1.25rem}.leading-6{line-height:1.5rem}.leading-none{line-height:1}.leading-relaxed{line-height:1.625}.leading-tight{line-height:1.25}.text-amber-400{--tw-text-opacity:1;color:rgb(251 191 36/var(--tw-text-opacity))}.text-amber-500{--tw-text-opacity:1;color:rgb(245 158 11/var(--tw-text-opacity))}.text-bay-of-many-400{--tw-text-opacity:1;color:rgb(143 185 221/var(--tw-text-opacity))}.text-bay-of-many-500{--tw-text-opacity:1;color:rgb(111 165 212/var(--tw-text-opacity))}.text-black{--tw-text-opacity:1;color:rgb(0 0 0/var(--tw-text-opacity))}.text-blue-400{--tw-text-opacity:1;color:rgb(96 165 250/var(--tw-text-opacity))}.text-blue-500{--tw-text-opacity:1;color:rgb(59 130 246/var(--tw-text-opacity))}.text-blue-600{--tw-text-opacity:1;color:rgb(37 99 235/var(--tw-text-opacity))}.text-chicago-400{--tw-text-opacity:1;color:rgb(135 135 137/var(--tw-text-opacity))}.text-chicago-500{--tw-text-opacity:1;color:rgb(108 108 110/var(--tw-text-opacity))}.text-cyan-400{--tw-text-opacity:1;color:rgb(34 211 238/var(--tw-text-opacity))}.text-cyan-500{--tw-text-opacity:1;color:rgb(6 182 212/var(--tw-text-opacity))}.text-ebony-clay-400{--tw-text-opacity:1;color:rgb(137 146 178/var(--tw-text-opacity))}.text-ebony-clay-500{--tw-text-opacity:1;color:rgb(113 123 162/var(--tw-text-opacity))}.text-emerald-400{--tw-text-opacity:1;color:rgb(52 211 153/var(--tw-text-opacity))}.text-emerald-500{--tw-text-opacity:1;color:rgb(16 185 129/var(--tw-text-opacity))}.text-fuchsia-400{--tw-text-opacity:1;color:rgb(232 121 249/var(--tw-text-opacity))}.text-fuchsia-500{--tw-text-opacity:1;color:rgb(217 70 239/var(--tw-text-opacity))}.text-gold-400{--tw-text-opacity:1;color:rgb(255 230 13/var(--tw-text-opacity))}.text-gold-500{--tw-text-opacity:1;color:rgb(255 215 0/var(--tw-text-opacity))}.text-gray-400{--tw-text-opacity:1;color:rgb(var(--color-gray-400)/var(--tw-text-opacity))}.text-gray-500{--tw-text-opacity:1;color:rgb(var(--color-gray-500)/var(--tw-text-opacity))}.text-gray-700{--tw-text-opacity:1;color:rgb(var(--color-gray-700)/var(--tw-text-opacity))}.text-gray-900{--tw-text-opacity:1;color:rgb(var(--color-gray-900)/var(--tw-text-opacity))}.text-green-400{--tw-text-opacity:1;color:rgb(74 222 128/var(--tw-text-opacity))}.text-green-500{--tw-text-opacity:1;color:rgb(34 197 94/var(--tw-text-opacity))}.text-hokey-pokey-400{--tw-text-opacity:1;color:rgb(218 202 73/var(--tw-text-opacity))}.text-hokey-pokey-500{--tw-text-opacity:1;color:rgb(208 189 42/var(--tw-text-opacity))}.text-indigo-400{--tw-text-opacity:1;color:rgb(129 140 248/var(--tw-text-opacity))}.text-indigo-500{--tw-text-opacity:1;color:rgb(99 102 241/var(--tw-text-opacity))}.text-lime-400{--tw-text-opacity:1;color:rgb(163 230 53/var(--tw-text-opacity))}.text-lime-500{--tw-text-opacity:1;color:rgb(132 204 22/var(--tw-text-opacity))}.text-orange-400{--tw-text-opacity:1;color:rgb(251 146 60/var(--tw-text-opacity))}.text-orange-500{--tw-text-opacity:1;color:rgb(249 115 22/var(--tw-text-opacity))}.text-pink-400{--tw-text-opacity:1;color:rgb(244 114 182/var(--tw-text-opacity))}.text-pink-500{--tw-text-opacity:1;color:rgb(236 72 153/var(--tw-text-opacity))}.text-primary-400{--tw-text-opacity:1;color:rgb(var(--color-primary-400)/var(--tw-text-opacity))}.text-primary-500{--tw-text-opacity:1;color:rgb(var(--color-primary-500)/var(--tw-text-opacity))}.text-primary-600{--tw-text-opacity:1;color:rgb(var(--color-primary-600)/var(--tw-text-opacity))}.text-purple-400{--tw-text-opacity:1;color:rgb(192 132 252/var(--tw-text-opacity))}.text-purple-500{--tw-text-opacity:1;color:rgb(168 85 247/var(--tw-text-opacity))}.text-red-400{--tw-text-opacity:1;color:rgb(248 113 113/var(--tw-text-opacity))}.text-red-500{--tw-text-opacity:1;color:rgb(239 68 68/var(--tw-text-opacity))}.text-rose-400{--tw-text-opacity:1;color:rgb(251 113 133/var(--tw-text-opacity))}.text-rose-500{--tw-text-opacity:1;color:rgb(244 63 94/var(--tw-text-opacity))}.text-sky-400{--tw-text-opacity:1;color:rgb(56 189 248/var(--tw-text-opacity))}.text-sky-500{--tw-text-opacity:1;color:rgb(14 165 233/var(--tw-text-opacity))}.text-teal-400{--tw-text-opacity:1;color:rgb(45 212 191/var(--tw-text-opacity))}.text-teal-500{--tw-text-opacity:1;color:rgb(20 184 166/var(--tw-text-opacity))}.text-violet-400{--tw-text-opacity:1;color:rgb(167 139 250/var(--tw-text-opacity))}.text-violet-500{--tw-text-opacity:1;color:rgb(139 92 246/var(--tw-text-opacity))}.text-white{--tw-text-opacity:1;color:rgb(255 255 255/var(--tw-text-opacity))}.text-yellow-400{--tw-text-opacity:1;color:rgb(250 204 21/var(--tw-text-opacity))}.text-yellow-500{--tw-text-opacity:1;color:rgb(234 179 8/var(--tw-text-opacity))}.underline-offset-4{text-underline-offset:4px}.placeholder-gray-400::placeholder{--tw-placeholder-opacity:1;color:rgb(var(--color-gray-400)/var(--tw-placeholder-opacity))}.opacity-0{opacity:0}.opacity-100{opacity:1}.opacity-50{opacity:.5}.opacity-90{opacity:.9}.shadow{--tw-shadow:0 1px 3px 0 #0000001a,0 1px 2px -1px #0000001a;--tw-shadow-colored:0 1px 3px 0 var(--tw-shadow-color),0 1px 2px -1px var(--tw-shadow-color)}.shadow,.shadow-lg{box-shadow:var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow)}.shadow-lg{--tw-shadow:0 10px 15px -3px #0000001a,0 4px 6px -4px #0000001a;--tw-shadow-colored:0 10px 15px -3px var(--tw-shadow-color),0 4px 6px -4px var(--tw-shadow-color)}.shadow-sm{--tw-shadow:0 1px 2px 0 #0000000d;--tw-shadow-colored:0 1px 2px 0 var(--tw-shadow-color)}.shadow-sm,.shadow-xl{box-shadow:var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow)}.shadow-xl{--tw-shadow:0 20px 25px -5px #0000001a,0 8px 10px -6px #0000001a;--tw-shadow-colored:0 20px 25px -5px var(--tw-shadow-color),0 8px 10px -6px var(--tw-shadow-color)}.outline{outline-style:solid}.outline-primary-400{outline-color:rgb(var(--color-primary-400)/1)}.outline-primary-500{outline-color:rgb(var(--color-primary-500)/1)}.ring{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color)}.ring,.ring-0{box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.ring-0{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(var(--tw-ring-offset-width)) var(--tw-ring-color)}.ring-1{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color)}.ring-1,.ring-2{box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.ring-2{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color)}.ring-inset{--tw-ring-inset:inset}.ring-current{--tw-ring-color:currentColor}.ring-gray-200{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-200)/var(--tw-ring-opacity))}.ring-gray-300{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-300)/var(--tw-ring-opacity))}.ring-primary-400{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}.ring-primary-500{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-500)/var(--tw-ring-opacity))}.ring-red-400{--tw-ring-opacity:1;--tw-ring-color:rgb(248 113 113/var(--tw-ring-opacity))}.ring-red-500{--tw-ring-opacity:1;--tw-ring-color:rgb(239 68 68/var(--tw-ring-opacity))}.ring-white{--tw-ring-opacity:1;--tw-ring-color:rgb(255 255 255/var(--tw-ring-opacity))}.ring-opacity-25{--tw-ring-opacity:0.25}.blur{--tw-blur:blur(8px)}.blur,.filter{filter:var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)}.transition{transition-duration:.15s;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,-webkit-backdrop-filter;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter,-webkit-backdrop-filter;transition-timing-function:cubic-bezier(.4,0,.2,1)}.transition-\[height\]{transition-duration:.15s;transition-property:height;transition-timing-function:cubic-bezier(.4,0,.2,1)}.transition-all{transition-duration:.15s;transition-property:all;transition-timing-function:cubic-bezier(.4,0,.2,1)}.transition-colors{transition-duration:.15s;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke;transition-timing-function:cubic-bezier(.4,0,.2,1)}.transition-opacity{transition-duration:.15s;transition-property:opacity;transition-timing-function:cubic-bezier(.4,0,.2,1)}.transition-transform{transition-duration:.15s;transition-property:transform;transition-timing-function:cubic-bezier(.4,0,.2,1)}.duration-100{transition-duration:.1s}.duration-150{transition-duration:.15s}.duration-200{transition-duration:.2s}.duration-300{transition-duration:.3s}.duration-500{transition-duration:.5s}.duration-75{transition-duration:75ms}.ease-in{transition-timing-function:cubic-bezier(.4,0,1,1)}.ease-in-out{transition-timing-function:cubic-bezier(.4,0,.2,1)}.ease-out{transition-timing-function:cubic-bezier(0,0,.2,1)}.content-\[\"\"\]{--tw-content:"";content:var(--tw-content)}.dark\:prose-invert:is(.dark *){--tw-prose-body:var(--tw-prose-invert-body);--tw-prose-headings:var(--tw-prose-invert-headings);--tw-prose-lead:var(--tw-prose-invert-lead);--tw-prose-links:var(--tw-prose-invert-links);--tw-prose-bold:var(--tw-prose-invert-bold);--tw-prose-counters:var(--tw-prose-invert-counters);--tw-prose-bullets:var(--tw-prose-invert-bullets);--tw-prose-hr:var(--tw-prose-invert-hr);--tw-prose-quotes:var(--tw-prose-invert-quotes);--tw-prose-quote-borders:var(--tw-prose-invert-quote-borders);--tw-prose-captions:var(--tw-prose-invert-captions);--tw-prose-kbd:var(--tw-prose-invert-kbd);--tw-prose-kbd-shadows:var(--tw-prose-invert-kbd-shadows);--tw-prose-code:var(--tw-prose-invert-code);--tw-prose-pre-code:var(--tw-prose-invert-pre-code);--tw-prose-pre-bg:var(--tw-prose-invert-pre-bg);--tw-prose-th-borders:var(--tw-prose-invert-th-borders);--tw-prose-td-borders:var(--tw-prose-invert-td-borders)}.file\:mr-1\.5::file-selector-button{margin-right:.375rem}.file\:border-0::file-selector-button{border-width:0}.file\:bg-transparent::file-selector-button{background-color:initial}.file\:p-0::file-selector-button{padding:0}.file\:font-medium::file-selector-button{font-weight:500}.file\:text-gray-500::file-selector-button{--tw-text-opacity:1;color:rgb(var(--color-gray-500)/var(--tw-text-opacity))}.file\:outline-none::file-selector-button{outline:2px solid #0000;outline-offset:2px}.before\:visible:before{content:var(--tw-content);visibility:visible}.before\:absolute:before{content:var(--tw-content);position:absolute}.before\:inset-px:before{content:var(--tw-content);inset:1px}.before\:inset-x-0:before{content:var(--tw-content);left:0;right:0}.before\:inset-y-2:before{bottom:.5rem;content:var(--tw-content);top:.5rem}.before\:z-\[-1\]:before{content:var(--tw-content);z-index:-1}.before\:block:before{content:var(--tw-content);display:block}.before\:h-2:before{content:var(--tw-content);height:.5rem}.before\:w-2:before{content:var(--tw-content);width:.5rem}.before\:rotate-45:before{content:var(--tw-content);--tw-rotate:45deg;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.before\:rounded-md:before{border-radius:.375rem;content:var(--tw-content)}.before\:rounded-sm:before{border-radius:.125rem;content:var(--tw-content)}.before\:bg-gray-100:before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-gray-100)/var(--tw-bg-opacity))}.before\:bg-gray-200:before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.before\:bg-white:before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.before\:shadow:before{content:var(--tw-content);--tw-shadow:0 1px 3px 0 #0000001a,0 1px 2px -1px #0000001a;--tw-shadow-colored:0 1px 3px 0 var(--tw-shadow-color),0 1px 2px -1px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow)}.before\:ring-1:before{content:var(--tw-content);--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.before\:ring-gray-200:before{content:var(--tw-content);--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-200)/var(--tw-ring-opacity))}.after\:absolute:after{content:var(--tw-content);position:absolute}.after\:inset-x-2\.5:after{content:var(--tw-content);left:.625rem;right:.625rem}.after\:bottom-0:after{bottom:0;content:var(--tw-content)}.after\:ms-0\.5:after{content:var(--tw-content);margin-inline-start:.125rem}.after\:mt-2:after{content:var(--tw-content);margin-top:.5rem}.after\:block:after{content:var(--tw-content);display:block}.after\:h-\[2px\]:after{content:var(--tw-content);height:2px}.after\:rounded-full:after{border-radius:9999px;content:var(--tw-content)}.after\:bg-primary-500:after{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-primary-500)/var(--tw-bg-opacity))}.after\:text-red-500:after{content:var(--tw-content);--tw-text-opacity:1;color:rgb(239 68 68/var(--tw-text-opacity))}.after\:content-\[\'\*\'\]:after{--tw-content:"*";content:var(--tw-content)}.first\:me-0:first-child{margin-inline-end:0}.first\:rounded-s-md:first-child{border-end-start-radius:.375rem;border-start-start-radius:.375rem}.last\:rounded-e-md:last-child{border-end-end-radius:.375rem;border-start-end-radius:.375rem}.indeterminate\:relative:indeterminate{position:relative}.indeterminate\:after\:rounded-full:indeterminate:after{border-radius:9999px;content:var(--tw-content)}.hover\:bg-bay-of-many:hover{--tw-bg-opacity:1;background-color:rgb(37 83 123/var(--tw-bg-opacity))}.hover\:bg-chicago-50:hover{--tw-bg-opacity:1;background-color:rgb(245 245 246/var(--tw-bg-opacity))}.hover\:bg-gray-100:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-100)/var(--tw-bg-opacity))}.hover\:bg-gray-50:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-50)/var(--tw-bg-opacity))}.hover\:bg-gray-800:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.hover\:bg-primary-100:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-100)/var(--tw-bg-opacity))}.hover\:bg-primary-50:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-50)/var(--tw-bg-opacity))}.hover\:bg-primary-600:hover{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-600)/var(--tw-bg-opacity))}.hover\:bg-white:hover{--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.hover\:text-gray-700:hover{--tw-text-opacity:1;color:rgb(var(--color-gray-700)/var(--tw-text-opacity))}.hover\:text-gray-900:hover{--tw-text-opacity:1;color:rgb(var(--color-gray-900)/var(--tw-text-opacity))}.hover\:text-hokey-pokey:hover{--tw-text-opacity:1;color:rgb(208 189 42/var(--tw-text-opacity))}.hover\:text-primary-600:hover{--tw-text-opacity:1;color:rgb(var(--color-primary-600)/var(--tw-text-opacity))}.hover\:underline:hover{text-decoration-line:underline}.hover\:before\:bg-gray-50:hover:before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-gray-50)/var(--tw-bg-opacity))}.focus\:shadow-none:focus{--tw-shadow:0 0 #0000;--tw-shadow-colored:0 0 #0000;box-shadow:var(--tw-ring-offset-shadow,0 0 #0000),var(--tw-ring-shadow,0 0 #0000),var(--tw-shadow)}.focus\:outline-none:focus{outline:2px solid #0000;outline-offset:2px}.focus\:ring-0:focus{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(var(--tw-ring-offset-width)) var(--tw-ring-color)}.focus\:ring-0:focus,.focus\:ring-2:focus{box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.focus\:ring-2:focus{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color)}.focus\:ring-primary-500:focus{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-500)/var(--tw-ring-opacity))}.focus\:ring-red-500:focus{--tw-ring-opacity:1;--tw-ring-color:rgb(239 68 68/var(--tw-ring-opacity))}.focus\:ring-transparent:focus{--tw-ring-color:#0000}.focus\:ring-offset-transparent:focus{--tw-ring-offset-color:#0000}.focus-visible\:outline-none:focus-visible{outline:2px solid #0000;outline-offset:2px}.focus-visible\:outline:focus-visible{outline-style:solid}.focus-visible\:outline-0:focus-visible{outline-width:0}.focus-visible\:outline-2:focus-visible{outline-width:2px}.focus-visible\:outline-offset-2:focus-visible{outline-offset:2px}.focus-visible\:outline-primary-500:focus-visible{outline-color:rgb(var(--color-primary-500)/1)}.focus-visible\:ring-2:focus-visible{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.focus-visible\:ring-inset:focus-visible{--tw-ring-inset:inset}.focus-visible\:ring-primary-500:focus-visible{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-500)/var(--tw-ring-opacity))}.focus-visible\:ring-red-500:focus-visible{--tw-ring-opacity:1;--tw-ring-color:rgb(239 68 68/var(--tw-ring-opacity))}.focus-visible\:ring-offset-2:focus-visible{--tw-ring-offset-width:2px}.focus-visible\:ring-offset-white:focus-visible{--tw-ring-offset-color:#fff}.focus-visible\:before\:ring-1:focus-visible:before{content:var(--tw-content);--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.focus-visible\:before\:ring-inset:focus-visible:before{content:var(--tw-content);--tw-ring-inset:inset}.focus-visible\:before\:ring-primary-500:focus-visible:before{content:var(--tw-content);--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-500)/var(--tw-ring-opacity))}.disabled\:cursor-not-allowed:disabled{cursor:not-allowed}.disabled\:bg-gray-50:disabled{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-50)/var(--tw-bg-opacity))}.disabled\:bg-gray-900:disabled{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.disabled\:bg-primary-50:disabled{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-50)/var(--tw-bg-opacity))}.disabled\:bg-primary-500:disabled{--tw-bg-opacity:1;background-color:rgb(var(--color-primary-500)/var(--tw-bg-opacity))}.disabled\:bg-transparent:disabled{background-color:initial}.disabled\:bg-white:disabled{--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.disabled\:bg-opacity-50:disabled{--tw-bg-opacity:0.5}.disabled\:text-primary-500:disabled{--tw-text-opacity:1;color:rgb(var(--color-primary-500)/var(--tw-text-opacity))}.disabled\:opacity-50:disabled{opacity:.5}.disabled\:opacity-75:disabled{opacity:.75}.group:hover .group-hover\:text-gray-700{--tw-text-opacity:1;color:rgb(var(--color-gray-700)/var(--tw-text-opacity))}.peer:disabled~.peer-disabled\:bg-opacity-50{--tw-bg-opacity:0.5}.aria-disabled\:cursor-not-allowed[aria-disabled=true]{cursor:not-allowed}.aria-disabled\:bg-gray-50[aria-disabled=true]{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-50)/var(--tw-bg-opacity))}.aria-disabled\:bg-gray-900[aria-disabled=true]{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.aria-disabled\:bg-transparent[aria-disabled=true]{background-color:initial}.aria-disabled\:bg-white[aria-disabled=true]{--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.aria-disabled\:opacity-75[aria-disabled=true]{opacity:.75}.group[data-popper-placement*=bottom] .group-data-\[popper-placement\*\=\'bottom\'\]\:-top-1{top:-.25rem}.group[data-popper-placement*=left] .group-data-\[popper-placement\*\=\'left\'\]\:-right-1{right:-.25rem}.group[data-popper-placement*=right] .group-data-\[popper-placement\*\=\'right\'\]\:-left-1{left:-.25rem}.group[data-popper-placement*=top] .group-data-\[popper-placement\*\=\'top\'\]\:-bottom-1{bottom:-.25rem}:where([data-headlessui-focus-visible]) .ui-focus-visible\:outline-0:focus{outline-width:0}:where([data-headlessui-focus-visible]) .ui-focus-visible\:ring-2:focus{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}:where([data-headlessui-focus-visible]) .ui-focus-visible\:ring-primary-500:focus{--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-500)/var(--tw-ring-opacity))}.ui-not-focus-visible\:outline-none:focus:where(:not([data-headlessui-focus-visible] .ui-not-focus-visible\:outline-none)){outline:2px solid #0000;outline-offset:2px}.dark\:divide-gray-700:is(.dark *)>:not([hidden])~:not([hidden]){--tw-divide-opacity:1;border-color:rgb(var(--color-gray-700)/var(--tw-divide-opacity))}.dark\:divide-gray-800:is(.dark *)>:not([hidden])~:not([hidden]){--tw-divide-opacity:1;border-color:rgb(var(--color-gray-800)/var(--tw-divide-opacity))}.dark\:border-gray-700:is(.dark *){--tw-border-opacity:1;border-color:rgb(var(--color-gray-700)/var(--tw-border-opacity))}.dark\:border-gray-800:is(.dark *){--tw-border-opacity:1;border-color:rgb(var(--color-gray-800)/var(--tw-border-opacity))}.dark\:bg-amber-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(251 191 36/var(--tw-bg-opacity))}.dark\:bg-bay-of-many-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(143 185 221/var(--tw-bg-opacity))}.dark\:bg-blue-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(96 165 250/var(--tw-bg-opacity))}.dark\:bg-chicago-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(135 135 137/var(--tw-bg-opacity))}.dark\:bg-cyan-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(34 211 238/var(--tw-bg-opacity))}.dark\:bg-ebony-clay-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(137 146 178/var(--tw-bg-opacity))}.dark\:bg-emerald-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(52 211 153/var(--tw-bg-opacity))}.dark\:bg-fuchsia-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(232 121 249/var(--tw-bg-opacity))}.dark\:bg-gold-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(255 230 13/var(--tw-bg-opacity))}.dark\:bg-gray-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-400)/var(--tw-bg-opacity))}.dark\:bg-gray-700:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}.dark\:bg-gray-800:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.dark\:bg-gray-800\/50:is(.dark *){background-color:rgb(var(--color-gray-800)/.5)}.dark\:bg-gray-800\/75:is(.dark *){background-color:rgb(var(--color-gray-800)/.75)}.dark\:bg-gray-900:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.dark\:bg-green-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(74 222 128/var(--tw-bg-opacity))}.dark\:bg-hokey-pokey-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(218 202 73/var(--tw-bg-opacity))}.dark\:bg-indigo-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(129 140 248/var(--tw-bg-opacity))}.dark\:bg-lime-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(163 230 53/var(--tw-bg-opacity))}.dark\:bg-orange-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(251 146 60/var(--tw-bg-opacity))}.dark\:bg-pink-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(244 114 182/var(--tw-bg-opacity))}.dark\:bg-primary-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-400)/var(--tw-bg-opacity))}.dark\:bg-primary-950:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-950)/var(--tw-bg-opacity))}.dark\:bg-purple-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(192 132 252/var(--tw-bg-opacity))}.dark\:bg-red-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(248 113 113/var(--tw-bg-opacity))}.dark\:bg-rose-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(251 113 133/var(--tw-bg-opacity))}.dark\:bg-sky-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(56 189 248/var(--tw-bg-opacity))}.dark\:bg-teal-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(45 212 191/var(--tw-bg-opacity))}.dark\:bg-violet-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(167 139 250/var(--tw-bg-opacity))}.dark\:bg-white:is(.dark *){--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.dark\:bg-yellow-400:is(.dark *){--tw-bg-opacity:1;background-color:rgb(250 204 21/var(--tw-bg-opacity))}.dark\:bg-opacity-10:is(.dark *){--tw-bg-opacity:0.1}.dark\:text-amber-400:is(.dark *){--tw-text-opacity:1;color:rgb(251 191 36/var(--tw-text-opacity))}.dark\:text-bay-of-many-400:is(.dark *){--tw-text-opacity:1;color:rgb(143 185 221/var(--tw-text-opacity))}.dark\:text-black:is(.dark *){--tw-text-opacity:1;color:rgb(0 0 0/var(--tw-text-opacity))}.dark\:text-blue-400:is(.dark *){--tw-text-opacity:1;color:rgb(96 165 250/var(--tw-text-opacity))}.dark\:text-chicago-400:is(.dark *){--tw-text-opacity:1;color:rgb(135 135 137/var(--tw-text-opacity))}.dark\:text-cyan-400:is(.dark *){--tw-text-opacity:1;color:rgb(34 211 238/var(--tw-text-opacity))}.dark\:text-ebony-clay-400:is(.dark *){--tw-text-opacity:1;color:rgb(137 146 178/var(--tw-text-opacity))}.dark\:text-emerald-400:is(.dark *){--tw-text-opacity:1;color:rgb(52 211 153/var(--tw-text-opacity))}.dark\:text-fuchsia-400:is(.dark *){--tw-text-opacity:1;color:rgb(232 121 249/var(--tw-text-opacity))}.dark\:text-gold-400:is(.dark *){--tw-text-opacity:1;color:rgb(255 230 13/var(--tw-text-opacity))}.dark\:text-gray-200:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-200)/var(--tw-text-opacity))}.dark\:text-gray-400:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-400)/var(--tw-text-opacity))}.dark\:text-gray-500:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-500)/var(--tw-text-opacity))}.dark\:text-gray-900:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-900)/var(--tw-text-opacity))}.dark\:text-green-400:is(.dark *){--tw-text-opacity:1;color:rgb(74 222 128/var(--tw-text-opacity))}.dark\:text-hokey-pokey-400:is(.dark *){--tw-text-opacity:1;color:rgb(218 202 73/var(--tw-text-opacity))}.dark\:text-indigo-400:is(.dark *){--tw-text-opacity:1;color:rgb(129 140 248/var(--tw-text-opacity))}.dark\:text-lime-400:is(.dark *){--tw-text-opacity:1;color:rgb(163 230 53/var(--tw-text-opacity))}.dark\:text-orange-400:is(.dark *){--tw-text-opacity:1;color:rgb(251 146 60/var(--tw-text-opacity))}.dark\:text-pink-400:is(.dark *){--tw-text-opacity:1;color:rgb(244 114 182/var(--tw-text-opacity))}.dark\:text-primary-400:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-primary-400)/var(--tw-text-opacity))}.dark\:text-purple-400:is(.dark *){--tw-text-opacity:1;color:rgb(192 132 252/var(--tw-text-opacity))}.dark\:text-red-400:is(.dark *){--tw-text-opacity:1;color:rgb(248 113 113/var(--tw-text-opacity))}.dark\:text-rose-400:is(.dark *){--tw-text-opacity:1;color:rgb(251 113 133/var(--tw-text-opacity))}.dark\:text-sky-400:is(.dark *){--tw-text-opacity:1;color:rgb(56 189 248/var(--tw-text-opacity))}.dark\:text-teal-400:is(.dark *){--tw-text-opacity:1;color:rgb(45 212 191/var(--tw-text-opacity))}.dark\:text-violet-400:is(.dark *){--tw-text-opacity:1;color:rgb(167 139 250/var(--tw-text-opacity))}.dark\:text-white:is(.dark *){--tw-text-opacity:1;color:rgb(255 255 255/var(--tw-text-opacity))}.dark\:text-yellow-400:is(.dark *){--tw-text-opacity:1;color:rgb(250 204 21/var(--tw-text-opacity))}.dark\:placeholder-gray-500:is(.dark *)::placeholder{--tw-placeholder-opacity:1;color:rgb(var(--color-gray-500)/var(--tw-placeholder-opacity))}.dark\:ring-gray-700:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-700)/var(--tw-ring-opacity))}.dark\:ring-gray-800:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-800)/var(--tw-ring-opacity))}.dark\:ring-gray-900:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-900)/var(--tw-ring-opacity))}.dark\:ring-primary-400:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}.dark\:ring-red-400:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(248 113 113/var(--tw-ring-opacity))}.dark\:ring-opacity-25:is(.dark *){--tw-ring-opacity:0.25}.dark\:file\:text-gray-400:is(.dark *)::file-selector-button{--tw-text-opacity:1;color:rgb(var(--color-gray-400)/var(--tw-text-opacity))}.dark\:before\:bg-gray-700:is(.dark *):before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}.dark\:before\:bg-gray-800:is(.dark *):before{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.dark\:before\:ring-gray-700:is(.dark *):before{content:var(--tw-content);--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-700)/var(--tw-ring-opacity))}.dark\:before\:ring-gray-800:is(.dark *):before{content:var(--tw-content);--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-gray-800)/var(--tw-ring-opacity))}.dark\:after\:bg-primary-400:is(.dark *):after{content:var(--tw-content);--tw-bg-opacity:1;background-color:rgb(var(--color-primary-400)/var(--tw-bg-opacity))}.dark\:after\:text-red-400:is(.dark *):after{content:var(--tw-content);--tw-text-opacity:1;color:rgb(248 113 113/var(--tw-text-opacity))}.dark\:checked\:border-transparent:checked:is(.dark *){border-color:#0000}.dark\:checked\:bg-current:checked:is(.dark *){background-color:currentColor}.dark\:indeterminate\:border-transparent:indeterminate:is(.dark *){border-color:#0000}.dark\:indeterminate\:bg-current:indeterminate:is(.dark *){background-color:currentColor}.dark\:hover\:bg-gray-100:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-100)/var(--tw-bg-opacity))}.dark\:hover\:bg-gray-700\/50:hover:is(.dark *){background-color:rgb(var(--color-gray-700)/.5)}.dark\:hover\:bg-gray-800:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.dark\:hover\:bg-gray-800\/50:hover:is(.dark *){background-color:rgb(var(--color-gray-800)/.5)}.dark\:hover\:bg-gray-900:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.dark\:hover\:bg-primary-500:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-500)/var(--tw-bg-opacity))}.dark\:hover\:bg-primary-900:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-900)/var(--tw-bg-opacity))}.dark\:hover\:bg-primary-950:hover:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-950)/var(--tw-bg-opacity))}.dark\:hover\:text-gray-200:hover:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-200)/var(--tw-text-opacity))}.dark\:hover\:text-primary-500:hover:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-primary-500)/var(--tw-text-opacity))}.dark\:hover\:text-white:hover:is(.dark *){--tw-text-opacity:1;color:rgb(255 255 255/var(--tw-text-opacity))}.dark\:hover\:before\:bg-gray-800\/50:hover:is(.dark *):before{background-color:rgb(var(--color-gray-800)/.5);content:var(--tw-content)}.dark\:focus\:ring-primary-400:focus:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}.dark\:focus\:ring-red-400:focus:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(248 113 113/var(--tw-ring-opacity))}.dark\:focus-visible\:outline-none:focus-visible:is(.dark *){outline:2px solid #0000;outline-offset:2px}.dark\:focus-visible\:outline-primary-400:focus-visible:is(.dark *){outline-color:rgb(var(--color-primary-400)/1)}.dark\:focus-visible\:ring-primary-400:focus-visible:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}.dark\:focus-visible\:ring-red-400:focus-visible:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(248 113 113/var(--tw-ring-opacity))}.dark\:focus-visible\:ring-offset-gray-900:focus-visible:is(.dark *){--tw-ring-offset-color:rgb(var(--color-gray-900)/1)}.dark\:focus-visible\:before\:ring-primary-400:focus-visible:is(.dark *):before{content:var(--tw-content);--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}.dark\:disabled\:bg-gray-800:disabled:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.dark\:disabled\:bg-gray-900:disabled:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.dark\:disabled\:bg-primary-400:disabled:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-400)/var(--tw-bg-opacity))}.dark\:disabled\:bg-primary-950:disabled:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-primary-950)/var(--tw-bg-opacity))}.dark\:disabled\:bg-transparent:disabled:is(.dark *){background-color:initial}.dark\:disabled\:bg-white:disabled:is(.dark *){--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.dark\:disabled\:text-primary-400:disabled:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-primary-400)/var(--tw-text-opacity))}.group:hover .dark\:group-hover\:text-gray-200:is(.dark *){--tw-text-opacity:1;color:rgb(var(--color-gray-200)/var(--tw-text-opacity))}.dark\:aria-disabled\:bg-gray-800[aria-disabled=true]:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-800)/var(--tw-bg-opacity))}.dark\:aria-disabled\:bg-gray-900[aria-disabled=true]:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.dark\:aria-disabled\:bg-transparent[aria-disabled=true]:is(.dark *){background-color:initial}.dark\:aria-disabled\:bg-white[aria-disabled=true]:is(.dark *){--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}:where([data-headlessui-focus-visible]) .dark\:ui-focus-visible\:ring-primary-400:focus:is(.dark *){--tw-ring-opacity:1;--tw-ring-color:rgb(var(--color-primary-400)/var(--tw-ring-opacity))}@media (min-width:640px){.sm\:my-8{margin-bottom:2rem;margin-top:2rem}.sm\:w-96{width:24rem}.sm\:max-w-lg{max-width:32rem}.sm\:translate-x-0{--tw-translate-x:0px}.sm\:translate-x-0,.sm\:translate-x-2{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.sm\:translate-x-2{--tw-translate-x:0.5rem}.sm\:translate-y-0{--tw-translate-y:0px}.sm\:scale-100,.sm\:translate-y-0{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.sm\:scale-100{--tw-scale-x:1;--tw-scale-y:1}.sm\:scale-95{--tw-scale-x:.95;--tw-scale-y:.95;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.sm\:items-center{align-items:center}.sm\:p-0{padding:0}.sm\:p-6{padding:1.5rem}.sm\:px-14{padding-left:3.5rem;padding-right:3.5rem}.sm\:px-6{padding-left:1.5rem;padding-right:1.5rem}.sm\:text-sm{font-size:.875rem;line-height:1.25rem}}@media (min-width:768px){.md\:inline-flex{display:inline-flex}.md\:w-1\/2{width:50%}.md\:grid-cols-2{grid-template-columns:repeat(2,minmax(0,1fr))}.md\:flex-row{flex-direction:row}.md\:text-4xl{font-size:2.25rem;line-height:2.5rem}.md\:text-5xl{font-size:3rem;line-height:1}}@media (min-width:1024px){.lg\:grid-cols-3{grid-template-columns:repeat(3,minmax(0,1fr))}.lg\:px-8{padding-left:2rem;padding-right:2rem}}.rtl\:-translate-x-0:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-0px;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-2:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-0.5rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-2\.5:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-0.625rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-3:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-0.75rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-4:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-1rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-5:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-1.25rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-6:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-1.5rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-7:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-1.75rem;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:-translate-x-full:where([dir=rtl],[dir=rtl] *){--tw-translate-x:-100%;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:translate-x-full:where([dir=rtl],[dir=rtl] *){--tw-translate-x:100%;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:rotate-180:where([dir=rtl],[dir=rtl] *){--tw-rotate:180deg;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:text-right:where([dir=rtl],[dir=rtl] *){text-align:right}.\[\&\:\:-moz-meter-bar\]\:h-0\.5::-moz-meter-bar{height:.125rem}.\[\&\:\:-moz-meter-bar\]\:h-1::-moz-meter-bar{height:.25rem}.\[\&\:\:-moz-meter-bar\]\:h-2::-moz-meter-bar{height:.5rem}.\[\&\:\:-moz-meter-bar\]\:h-3::-moz-meter-bar{height:.75rem}.\[\&\:\:-moz-meter-bar\]\:h-4::-moz-meter-bar{height:1rem}.\[\&\:\:-moz-meter-bar\]\:h-5::-moz-meter-bar{height:1.25rem}.\[\&\:\:-moz-meter-bar\]\:h-px::-moz-meter-bar{height:1px}.\[\&\:\:-moz-meter-bar\]\:rounded-full::-moz-meter-bar{border-radius:9999px}.\[\&\:\:-moz-meter-bar\]\:border-none::-moz-meter-bar{border-style:none}.\[\&\:\:-moz-meter-bar\]\:bg-current::-moz-meter-bar{background-color:currentColor}.\[\&\:\:-moz-meter-bar\]\:bg-none::-moz-meter-bar{background-image:none}.\[\&\:\:-moz-meter-bar\]\:transition-all::-moz-meter-bar{transition-duration:.15s;-moz-transition-property:all;transition-property:all;transition-timing-function:cubic-bezier(.4,0,.2,1)}.\[\&\:\:-moz-progress-bar\]\:rounded-full::-moz-progress-bar{border-radius:9999px}.\[\&\:\:-moz-progress-bar\]\:bg-current::-moz-progress-bar{background-color:currentColor}.\[\&\:\:-moz-range-thumb\]\:relative::-moz-range-thumb{position:relative}.\[\&\:\:-moz-range-thumb\]\:z-\[1\]::-moz-range-thumb{z-index:1}.\[\&\:\:-moz-range-thumb\]\:-mt-1::-moz-range-thumb{margin-top:-.25rem}.\[\&\:\:-moz-range-thumb\]\:mt-\[-2\.5px\]::-moz-range-thumb{margin-top:-2.5px}.\[\&\:\:-moz-range-thumb\]\:mt-\[-3px\]::-moz-range-thumb{margin-top:-3px}.\[\&\:\:-moz-range-thumb\]\:h-1\.5::-moz-range-thumb{height:.375rem}.\[\&\:\:-moz-range-thumb\]\:h-2::-moz-range-thumb{height:.5rem}.\[\&\:\:-moz-range-thumb\]\:h-3::-moz-range-thumb{height:.75rem}.\[\&\:\:-moz-range-thumb\]\:h-4::-moz-range-thumb{height:1rem}.\[\&\:\:-moz-range-thumb\]\:h-5::-moz-range-thumb{height:1.25rem}.\[\&\:\:-moz-range-thumb\]\:h-6::-moz-range-thumb{height:1.5rem}.\[\&\:\:-moz-range-thumb\]\:h-7::-moz-range-thumb{height:1.75rem}.\[\&\:\:-moz-range-thumb\]\:w-1\.5::-moz-range-thumb{width:.375rem}.\[\&\:\:-moz-range-thumb\]\:w-2::-moz-range-thumb{width:.5rem}.\[\&\:\:-moz-range-thumb\]\:w-3::-moz-range-thumb{width:.75rem}.\[\&\:\:-moz-range-thumb\]\:w-4::-moz-range-thumb{width:1rem}.\[\&\:\:-moz-range-thumb\]\:w-5::-moz-range-thumb{width:1.25rem}.\[\&\:\:-moz-range-thumb\]\:w-6::-moz-range-thumb{width:1.5rem}.\[\&\:\:-moz-range-thumb\]\:w-7::-moz-range-thumb{width:1.75rem}.\[\&\:\:-moz-range-thumb\]\:appearance-none::-moz-range-thumb{appearance:none}.\[\&\:\:-moz-range-thumb\]\:rounded-full::-moz-range-thumb{border-radius:9999px}.\[\&\:\:-moz-range-thumb\]\:border-0::-moz-range-thumb{border-width:0}.\[\&\:\:-moz-range-thumb\]\:bg-current::-moz-range-thumb{background-color:currentColor}.\[\&\:\:-moz-range-track\]\:h-0\.5::-moz-range-track{height:.125rem}.\[\&\:\:-moz-range-track\]\:h-1::-moz-range-track{height:.25rem}.\[\&\:\:-moz-range-track\]\:h-2::-moz-range-track{height:.5rem}.\[\&\:\:-moz-range-track\]\:h-3::-moz-range-track{height:.75rem}.\[\&\:\:-moz-range-track\]\:h-4::-moz-range-track{height:1rem}.\[\&\:\:-moz-range-track\]\:h-5::-moz-range-track{height:1.25rem}.\[\&\:\:-moz-range-track\]\:h-px::-moz-range-track{height:1px}.\[\&\:\:-moz-range-track\]\:rounded-lg::-moz-range-track{border-radius:.5rem}.\[\&\:\:-moz-range-track\]\:bg-gray-200::-moz-range-track{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.group:disabled .\[\&\:\:-moz-range-track\]\:group-disabled\:bg-opacity-50::-moz-range-track{--tw-bg-opacity:0.5}.\[\&\:\:-moz-range-track\]\:dark\:bg-gray-700:is(.dark *)::-moz-range-track{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}.\[\&\:\:-webkit-meter-bar\]\:border-none::-webkit-meter-bar{border-style:none}.\[\&\:\:-webkit-meter-bar\]\:bg-transparent::-webkit-meter-bar{background-color:initial}.\[\&\:\:-webkit-meter-bar\]\:bg-none::-webkit-meter-bar{background-image:none}.\[\&\:\:-webkit-meter-inner-element\]\:relative::-webkit-meter-inner-element{position:relative}.\[\&\:\:-webkit-meter-inner-element\]\:block::-webkit-meter-inner-element{display:block}.\[\&\:\:-webkit-meter-inner-element\]\:border-none::-webkit-meter-inner-element{border-style:none}.\[\&\:\:-webkit-meter-inner-element\]\:bg-transparent::-webkit-meter-inner-element{background-color:initial}.\[\&\:\:-webkit-meter-inner-element\]\:bg-none::-webkit-meter-inner-element{background-image:none}.\[\&\:\:-webkit-meter-optimum-value\]\:h-0\.5::-webkit-meter-optimum-value{height:.125rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-1::-webkit-meter-optimum-value{height:.25rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-2::-webkit-meter-optimum-value{height:.5rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-3::-webkit-meter-optimum-value{height:.75rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-4::-webkit-meter-optimum-value{height:1rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-5::-webkit-meter-optimum-value{height:1.25rem}.\[\&\:\:-webkit-meter-optimum-value\]\:h-px::-webkit-meter-optimum-value{height:1px}.\[\&\:\:-webkit-meter-optimum-value\]\:rounded-full::-webkit-meter-optimum-value{border-radius:9999px}.\[\&\:\:-webkit-meter-optimum-value\]\:border-none::-webkit-meter-optimum-value{border-style:none}.\[\&\:\:-webkit-meter-optimum-value\]\:bg-current::-webkit-meter-optimum-value{background-color:currentColor}.\[\&\:\:-webkit-meter-optimum-value\]\:bg-none::-webkit-meter-optimum-value{background-image:none}.\[\&\:\:-webkit-meter-optimum-value\]\:transition-all::-webkit-meter-optimum-value{transition-duration:.15s;-webkit-transition-property:all;transition-property:all;transition-timing-function:cubic-bezier(.4,0,.2,1)}.\[\&\:\:-webkit-progress-bar\]\:w-full::-webkit-progress-bar{width:100%}.\[\&\:\:-webkit-progress-bar\]\:rounded-full::-webkit-progress-bar{border-radius:9999px}.\[\&\:\:-webkit-progress-bar\]\:bg-gray-200::-webkit-progress-bar{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.\[\&\:\:-webkit-progress-bar\]\:dark\:bg-gray-700:is(.dark *)::-webkit-progress-bar{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}.\[\&\:\:-webkit-progress-value\]\:rounded-full::-webkit-progress-value{border-radius:9999px}.\[\&\:\:-webkit-progress-value\]\:bg-current::-webkit-progress-value{background-color:currentColor}.\[\&\:\:-webkit-progress-value\]\:transition-all::-webkit-progress-value{transition-duration:.15s;-webkit-transition-property:all;transition-property:all;transition-timing-function:cubic-bezier(.4,0,.2,1)}.\[\&\:\:-webkit-progress-value\]\:ease-in-out::-webkit-progress-value{transition-timing-function:cubic-bezier(.4,0,.2,1)}.\[\&\:\:-webkit-slider-runnable-track\]\:h-0\.5::-webkit-slider-runnable-track{height:.125rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-1::-webkit-slider-runnable-track{height:.25rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-2::-webkit-slider-runnable-track{height:.5rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-3::-webkit-slider-runnable-track{height:.75rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-4::-webkit-slider-runnable-track{height:1rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-5::-webkit-slider-runnable-track{height:1.25rem}.\[\&\:\:-webkit-slider-runnable-track\]\:h-px::-webkit-slider-runnable-track{height:1px}.\[\&\:\:-webkit-slider-runnable-track\]\:rounded-lg::-webkit-slider-runnable-track{border-radius:.5rem}.\[\&\:\:-webkit-slider-runnable-track\]\:bg-gray-200::-webkit-slider-runnable-track{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.group:disabled .\[\&\:\:-webkit-slider-runnable-track\]\:group-disabled\:bg-opacity-50::-webkit-slider-runnable-track{--tw-bg-opacity:0.5}.\[\&\:\:-webkit-slider-runnable-track\]\:dark\:bg-gray-700:is(.dark *)::-webkit-slider-runnable-track{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}.\[\&\:\:-webkit-slider-thumb\]\:relative::-webkit-slider-thumb{position:relative}.\[\&\:\:-webkit-slider-thumb\]\:z-\[1\]::-webkit-slider-thumb{z-index:1}.\[\&\:\:-webkit-slider-thumb\]\:-mt-1::-webkit-slider-thumb{margin-top:-.25rem}.\[\&\:\:-webkit-slider-thumb\]\:mt-\[-2\.5px\]::-webkit-slider-thumb{margin-top:-2.5px}.\[\&\:\:-webkit-slider-thumb\]\:mt-\[-3px\]::-webkit-slider-thumb{margin-top:-3px}.\[\&\:\:-webkit-slider-thumb\]\:h-1\.5::-webkit-slider-thumb{height:.375rem}.\[\&\:\:-webkit-slider-thumb\]\:h-2::-webkit-slider-thumb{height:.5rem}.\[\&\:\:-webkit-slider-thumb\]\:h-3::-webkit-slider-thumb{height:.75rem}.\[\&\:\:-webkit-slider-thumb\]\:h-4::-webkit-slider-thumb{height:1rem}.\[\&\:\:-webkit-slider-thumb\]\:h-5::-webkit-slider-thumb{height:1.25rem}.\[\&\:\:-webkit-slider-thumb\]\:h-6::-webkit-slider-thumb{height:1.5rem}.\[\&\:\:-webkit-slider-thumb\]\:h-7::-webkit-slider-thumb{height:1.75rem}.\[\&\:\:-webkit-slider-thumb\]\:w-1\.5::-webkit-slider-thumb{width:.375rem}.\[\&\:\:-webkit-slider-thumb\]\:w-2::-webkit-slider-thumb{width:.5rem}.\[\&\:\:-webkit-slider-thumb\]\:w-3::-webkit-slider-thumb{width:.75rem}.\[\&\:\:-webkit-slider-thumb\]\:w-4::-webkit-slider-thumb{width:1rem}.\[\&\:\:-webkit-slider-thumb\]\:w-5::-webkit-slider-thumb{width:1.25rem}.\[\&\:\:-webkit-slider-thumb\]\:w-6::-webkit-slider-thumb{width:1.5rem}.\[\&\:\:-webkit-slider-thumb\]\:w-7::-webkit-slider-thumb{width:1.75rem}.\[\&\:\:-webkit-slider-thumb\]\:appearance-none::-webkit-slider-thumb{appearance:none}.\[\&\:\:-webkit-slider-thumb\]\:rounded-full::-webkit-slider-thumb{border-radius:9999px}.\[\&\:\:-webkit-slider-thumb\]\:bg-white::-webkit-slider-thumb{--tw-bg-opacity:1;background-color:rgb(255 255 255/var(--tw-bg-opacity))}.\[\&\:\:-webkit-slider-thumb\]\:ring-2::-webkit-slider-thumb{--tw-ring-offset-shadow:var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow:var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000)}.\[\&\:\:-webkit-slider-thumb\]\:ring-current::-webkit-slider-thumb{--tw-ring-color:currentColor}.\[\&\:\:-webkit-slider-thumb\]\:dark\:bg-gray-900:is(.dark *)::-webkit-slider-thumb{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-900)/var(--tw-bg-opacity))}.\[\&\:indeterminate\:\:-moz-progress-bar\]\:rounded-full:indeterminate::-moz-progress-bar{border-radius:9999px}.\[\&\:indeterminate\:\:-webkit-progress-value\]\:rounded-full:indeterminate::-webkit-progress-value{border-radius:9999px}.rtl\:\[\&_span\:first-child\]\:rotate-180 span:first-child:where([dir=rtl],[dir=rtl] *){--tw-rotate:180deg;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.rtl\:\[\&_span\:last-child\]\:rotate-180 span:last-child:where([dir=rtl],[dir=rtl] *){--tw-rotate:180deg;transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}@media(pointer:coarse){.\[\@media\(pointer\:coarse\)\]\:hidden{display:none}}@supports(selector(&::-moz-progress-bar)){.\[\@supports\(selector\(\&\:\:-moz-progress-bar\)\)\]\:bg-gray-200{--tw-bg-opacity:1;background-color:rgb(var(--color-gray-200)/var(--tw-bg-opacity))}.\[\@supports\(selector\(\&\:\:-moz-progress-bar\)\)\]\:dark\:bg-gray-700:is(.dark *){--tw-bg-opacity:1;background-color:rgb(var(--color-gray-700)/var(--tw-bg-opacity))}}</style>
<style>.dark{color-scheme:dark}a:focus-visible{outline-color:rgb(var(--color-primary-DEFAULT)/1)}::-moz-selection{background-color:rgb(var(--color-primary-DEFAULT)/.4)}::selection{background-color:rgb(var(--color-primary-DEFAULT)/.4)}</style>
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/CBfqzfVe.js">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/EKpfgqNd.js">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/HFPQ6t7W.js">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/IShi1APO.js">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/kPkxnBls.js">
<link rel="prefetch" as="style" href="/_nuxt/_id_.CpkGk6rS.css">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/oIdYgKnD.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/Cpj98o6Y.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/kqN0fDuk.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/DpILShx0.js">
<script type="module" src="/_nuxt/CBfqzfVe.js" crossorigin></script>
<style>:where(.i-uil\:github){display:inline-block;width:1em;height:1em;background-color:currentColor;-webkit-mask-image:var(--svg);mask-image:var(--svg);-webkit-mask-repeat:no-repeat;mask-repeat:no-repeat;-webkit-mask-size:100% 100%;mask-size:100% 100%;--svg:url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='24' height='24'%3E%3Cpath fill='black' d='M12 2.247a10 10 0 0 0-3.162 19.487c.5.088.687-.212.687-.475c0-.237-.012-1.025-.012-1.862c-2.513.462-3.163-.613-3.363-1.175a3.64 3.64 0 0 0-1.025-1.413c-.35-.187-.85-.65-.013-.662a2 2 0 0 1 1.538 1.025a2.137 2.137 0 0 0 2.912.825a2.1 2.1 0 0 1 .638-1.338c-2.225-.25-4.55-1.112-4.55-4.937a3.9 3.9 0 0 1 1.025-2.688a3.6 3.6 0 0 1 .1-2.65s.837-.262 2.75 1.025a9.43 9.43 0 0 1 5 0c1.912-1.3 2.75-1.025 2.75-1.025a3.6 3.6 0 0 1 .1 2.65a3.87 3.87 0 0 1 1.025 2.688c0 3.837-2.338 4.687-4.562 4.937a2.37 2.37 0 0 1 .674 1.85c0 1.338-.012 2.413-.012 2.75c0 .263.187.575.687.475A10.005 10.005 0 0 0 12 2.247'/%3E%3C/svg%3E")}</style><script>"use strict";(()=>{const t=window,e=document.documentElement,c=["dark","light"],n=getStorageValue("localStorage","nuxt-color-mode")||"light";let i=n==="system"?u():n;const r=e.getAttribute("data-color-mode-forced");r&&(i=r),l(i),t["__NUXT_COLOR_MODE__"]={preference:n,value:i,getColorScheme:u,addColorScheme:l,removeColorScheme:d};function l(o){const s=""+o+"",a="";e.classList?e.classList.add(s):e.className+=" "+s,a&&e.setAttribute("data-"+a,o)}function d(o){const s=""+o+"",a="";e.classList?e.classList.remove(s):e.className=e.className.replace(new RegExp(s,"g"),""),a&&e.removeAttribute("data-"+a)}function f(o){return t.matchMedia("(prefers-color-scheme"+o+")")}function u(){if(t.matchMedia&&f("").media!=="not all"){for(const o of c)if(f(":"+o).matches)return o}return"light"}})();function getStorageValue(t,e){switch(t){case"localStorage":return window.localStorage.getItem(e);case"sessionStorage":return window.sessionStorage.getItem(e);case"cookie":return getCookie(e);default:return null}}function getCookie(t){const c=("; "+window.document.cookie).split("; "+t+"=");if(c.length===2)return c.pop()?.split(";").shift()}</script></head><body><div id="__nuxt"><div class="h-dvh flex flex-col"><header class="p-8 bg-ebony-clay-800 border-b text-white"><div class="flex"><div class="flex flex-grow space-x-2"><a aria-current="page" href="/" class="router-link-active router-link-exact-active flex items-center space-x-2 hover:text-hokey-pokey"><img src="/iwc_logo_white.png" alt="IWC Logo" width="128" height="128"><span class="text-4xl font-mono font-semibold px-4"> Galaxy IWC - Workflow Library </span></a></div><div class="flex items-center space-x-8 font-mono"><a href="https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#adding-workflows" rel="noopener noreferrer" class="hover:text-hokey-pokey"> Contribute </a><a href="https://github.com/galaxyproject/iwc/" rel="noopener noreferrer" class="hover:text-hokey-pokey"> GitHub <span class="iconify i-uil:github" aria-hidden="true" style=""></span></a></div></div></header><main class="flex overflow-hidden"><div class="flex w-full"><div class="w-1/4 p-4 overflow-y-auto"><!--[--><div class="sticky top-4 h-16"><input value="" type="text" placeholder="Search workflows" class="w-full mb-4 p-2 border rounded"></div><ul class="mt-6"><!--[--><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Taxonomic abundance summary tables for a specified taxonomic rank</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify&#39;s amplicon pipeline v5.0 - ITS</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify&#39;s amplicon pipeline v5.0</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify&#39;s amplicon pipeline v5.0 - rRNA prediction</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MAPseq to ampvis2</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify amplicon summary tables</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify&#39;s amplicon pipeline v5.0 - Quality control SE</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MGnify&#39;s amplicon pipeline v5.0 - Quality control PE</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">sra_manifest_to_concatenated_fastqs_parallel</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Parallel Accession Download</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Velocyto-on10X-from-bundled</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Velocyto-on10X-filtered-barcodes</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Differential gene expression for single-cell data using pseudo-bulk counts with edgeR</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">baredSC_1d_logNorm</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">baredSC_2d_logNorm</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">scRNA-seq_preprocessing_10X_cellPlex</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">scRNA-seq_preprocessing_10X_v3_Bundle</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Preprocessing and Clustering of single-cell RNA-seq data with Scanpy</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: variation analysis on WGS SE data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: variation analysis reporting</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">SARS-CoV-2 Illumina Amplicon pipeline - iVar based</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: consensus construction</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: variation analysis on WGS PE data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: variation analysis of ARTIC ONT data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">COVID-19: variation analysis on ARTIC PE data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">dada2 amplicon analysis pipeline - for paired end data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 Ia: multiplexed data (single-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 Ib: multiplexed data (paired-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 Ic: Demultiplexed data (single-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 Id: Demultiplexed data (paired-end)</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">QIIME2 VI: Diversity metrics and estimations</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Assembly-Hifi-only-VGP3</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Assembly-decontamination-VGP9</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Scaffolding with Hi-C data VGP8</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Assembly-Hifi-Trio-phasing-VGP5</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Purge duplicate contigs from a diploid assembly VGP6</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">kmer-profiling-hifi-VGP1</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Scaffolding-BioNano-VGP7</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">kmer-profiling-hifi-trio-VGP2</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Assembly-Hifi-HiC-phasing-VGP4</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Generate Nx and Size plots for multiple assemblies</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Purging duplicates in one haplotype VGP6b</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Mitogenome-Assembly-VGP0</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Segmentation and counting of cell nuclei in fluorescence microscopy images</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">ATACseq</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">ChIPseq_SR</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Get Confident Peaks From ChIP_SR replicates</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Get Confident Peaks From ChIP_PE replicates</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Get Confident Peaks From ATAC or CUTandRUN replicates</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Hi-C_fastqToCool_hicup_cooler</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">cHi-C_fastqToCool_hicup_cooler</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Hi-C_juicermediumtabixToCool_cooler</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Hi-C_fastqToPairs_hicup</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">ChIPseq_PE</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">CUTandRUN</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Average Bigwig between replicates</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MMGBSA calculations with GROMACS</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Create GRO and TOP complex files</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">dcTMD calculations with GROMACS</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Pox Virus Illumina Amplicon Workflow from half-genomes</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Influenza A isolate subtyping and consensus sequence generation</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Generic variation analysis on WGS PE data</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Paired end variant calling in haploid system</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Generic variation analysis reporting</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">RNA-seq for Paired-end fastqs</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">RNA-seq for Single-read fastqs</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">BREW3R</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">RNAseq_DE_filtering_plotting</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Goseq GO-KEGG Enrichment Analysis</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Genome annotation with Helixer</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Functional annotation of protein sequences</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">MetaProSIP OpenMS 2.8</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Clinical Metaproteomics Discovery Workflow</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Generate a Clinical Metaproteomics Database</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Clinical Metaproteomics Verification Workflow</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Clinical Metaproteomics Data Interpretation</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Clinical Metaproteomics Quantitation</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">amr_gene_detection</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">bacterial_genome_annotation</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Repeat masking with RepeatModeler and RepeatMasker</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Mass spectrometry: GCMS with metaMS</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Mass spectrometry: LC-MS preprocessing with XCMS</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Bacterial Genome Assembly using Shovill</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Assembly polishing with long reads</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Genome assembly with Flye</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Quality and Contamination Control For Genome Assembly</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Gene-based Pathogen Identification</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Nanopore Preprocessing</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Allele-based Pathogen Identification</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation</li><li class="cursor-pointer hover:bg-gray-100 p-2 rounded">Taxonomy Profiling and Visualization with Krona</li><!--]--></ul><!--]--></div><div class="w-3/4 p-4 overflow-y-auto"><!--[--><!--[--><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-31b19088-dad2-41a0-b086-94037bce4784"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Taxonomic abundance summary tables for a specified taxonomic rank</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow creates taxonomic summary tables for a specified taxonomic rank out of MAPseq&#39;s OTU tables output collection.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ftaxonomic-rank-abundance-summary-table%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-04fa2e92-8fcd-4a20-8af1-063c7f514f12"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify&#39;s amplicon pipeline v5.0 - ITS</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Classification and visualization of ITS regions.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-pipeline-v5-its%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-cdf82f4c-734e-4441-941a-2626aa7d8dae"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify&#39;s amplicon pipeline v5.0</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">MGnify&#39;s amplicon pipeline v5.0. Including the Quality control for single-end and paired-end reads, rRNA-prediction, and ITS sub-WFs.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-pipeline-v5-complete%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-53c5bc12-4d7c-4799-b053-ced394665d61"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify&#39;s amplicon pipeline v5.0 - rRNA prediction</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Classification and visualization of SSU, LSU sequences.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-pipeline-v5-rrna-prediction%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-e24060a0-1cfe-40f1-9251-48e96f7e1121"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MAPseq to ampvis2</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">The MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmapseq-to-ampvis2%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-dda70895-1e39-42a3-8b4f-b6493cf15e8f"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify amplicon summary tables</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow creates taxonomic summary tables out of the amplicon pipeline results. </p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-taxonomic-summary-tables%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-68bf2a73-d68a-4e25-ba52-5586c057df77"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify&#39;s amplicon pipeline v5.0 - Quality control SE</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Quality control subworkflow for single-end reads.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-pipeline-v5-quality-control-single-end%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a9df547d-8f99-4740-b884-73d1e83a89b5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MGnify&#39;s amplicon pipeline v5.0 - Quality control PE</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Quality control subworkflow for paired-end reads. </p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MGnify_Amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->amplicon<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metagenomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microgalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fmgnify-amplicon-pipeline-v5-quality-control-paired-end%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-363898af-e598-4f0e-abd9-e6ded395ce66"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">sra_manifest_to_concatenated_fastqs_parallel</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a SRA_manifest from SRA Run Selector and will generate one fastq file or fastq pair of file for each experiment (concatenated multiple runs if necessary). Output will be relabelled to match the column specified by the user.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsra-manifest-to-concatenated-fastqs%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-15c3229d-8786-457f-a020-7aa64c0cf6e8"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Parallel Accession Download</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Downloads fastq files for sequencing run accessions provided in a text file using fasterq-dump. Creates one job per listed run accession.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fparallel-accession-download%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-0608c7f4-767c-47b3-ae45-4e98c7c83bec"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Velocyto-on10X-from-bundled</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Run velocyto to get loom with counts of spliced and unspliced. It will extract the &#39;barcodes&#39; from the bundled outputs.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:single-cell<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fvelocyto%2FVelocyto-on10X-from-bundled/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-33862923-af05-48ba-aec5-14393981cee2"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Velocyto-on10X-filtered-barcodes</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Run velocyto to get loom with counts of spliced and unspliced</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:single-cell<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fvelocyto%2FVelocyto-on10X-filtered-barcodes/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-e4e4055c-7e3f-4014-a8e6-343199fd9b01"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Differential gene expression for single-cell data using pseudo-bulk counts with edgeR</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow uses the decoupler tool in Galaxy to generate pseudobulk counts from an annotated AnnData file obtained from scRNA-seq analysis. Following the pseudobulk step, differential expression genes (DEG) are calculated using the edgeR tool. The workflow also includes data sanitation steps to ensure smooth operation of edgeR and minimizing potential issues. Additionally, a Volcano plot tool is used to visualize the results after the DEG analysis.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fpseudobulk-worflow-decoupler-edger%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-1ea0b92d-d863-435a-96ac-1e547fdbc994"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">baredSC_1d_logNorm</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Run baredSC in 1 dimension in logNorm for 1 to N gaussians and combine models.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fbaredsc%2FbaredSC-1d-logNorm/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-1936816b-958a-4140-b839-58be4cdae1a5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">baredSC_2d_logNorm</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Run baredSC in 2 dimensions in logNorm for 1 to N gaussians and combine models.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fbaredsc%2FbaredSC-2d-logNorm/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-470536e7-ac58-4859-94d6-2085be9c2e17"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">scRNA-seq_preprocessing_10X_cellPlex</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow processes the CMO fastqs with CITE-seq-Count and include the translation step required for cellPlex processing. In parallel it processes the Gene Expresion fastqs with STARsolo, filter cells with DropletUtils and reformat all outputs to be easily used by the function &#39;Read10X&#39; from Seurat.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->#single-cell<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ffastq-to-matrix-10x%2Fscrna-seq-fastq-to-matrix-10x-cellplex/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-87f25ddb-a663-4582-9ec6-d0b4bef4109c"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">scRNA-seq_preprocessing_10X_v3_Bundle</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow processes the Gene Expresion fastqs with STARsolo, filter cells with DropletUtils and reformat all outputs to be easily used by the function &#39;Read10X&#39; from Seurat.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->#single-cell<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ffastq-to-matrix-10x%2Fscrna-seq-fastq-to-matrix-10x-v3/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-d6b2d1bb-6aaf-47cc-8956-5ae80b775de5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Preprocessing and Clustering of single-cell RNA-seq data with Scanpy</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Single-cell RNA-seq workflow with Scanpy and Anndata. Based on the 3k PBMC clustering tutorial from Scanpy. It takes count matrix, barcodes and feature files as input and creates an Anndata object out of them. It then performs QC and filters for lowly expressed genes and cells. Then the data is normalized and scaled. Then PCs are computed to further cluster using louvain algorithm. It also generated various plots of clustering colored with highly ranked genes.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:scRNA-seq<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:single-cell<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fscanpy-clustering%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-5171bc59-590c-4e97-8db8-004f73928627"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: variation analysis on WGS SE data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflows performs single end read mapping with bowtie2 followed by sensitive variant calling across a wide range of AFs with lofreq</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-se-illumina-wgs-variant-calling%2FCOVID-19-SE-WGS-ILLUMINA/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-b08c744d-7c61-4b58-ac5f-4b5886c3c643"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: variation analysis reporting</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes a VCF dataset of variants produced by any of the *-variant-calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-variation-reporting%2FCOVID-19-VARIATION-REPORTING/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-3d86e5fc-cad1-4e21-8ecc-d0e6637899bc"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">SARS-CoV-2 Illumina Amplicon pipeline - iVar based</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Find and annotate variants in ampliconic SARS-CoV-2 Illumina sequencing data and classify samples with pangolin and nextclade</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ARTIC<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->iwc<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-pe-illumina-artic-ivar-analysis%2FSARS-COV-2-ILLUMINA-AMPLICON-IVAR-PANGOLIN-NEXTCLADE/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-06dc40a7-99f2-4b3c-ae21-b3dcb239306f"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: consensus construction</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Build a consensus sequence from FILTER PASS variants with intrasample allele-frequency above a configurable consensus threshold.
Hard-mask regions with low coverage (but not consensus variants within them) and ambiguous sites.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-consensus-from-variation%2FCOVID-19-CONSENSUS-CONSTRUCTION/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-6695eb16-30a5-42ea-8591-56f2630cf8bf"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: variation analysis on WGS PE data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflows performs paired end read mapping with bwa-mem followed by sensitive variant calling across a wide range of AFs with lofreq</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->iwc<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->emergen_validated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-pe-illumina-wgs-variant-calling%2FCOVID-19-PE-WGS-ILLUMINA/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a1ceaf35-f3d7-452d-b5ed-53a4f429d1cf"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: variation analysis of ARTIC ONT data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipelines minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ARTIC<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ONT<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-ont-artic-variant-calling%2FCOVID-19-ARTIC-ONT/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-864427b0-d25a-499c-b437-f83f2d6b83f2"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">COVID-19: variation analysis on ARTIC PE data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">The workflow for Illumina-sequenced ARTIC data builds on the RNASeq workflow for paired-end data using the same steps for mapping and variant calling, but adds extra logic for trimming ARTIC primer sequences off reads with the ivar package. In addition, this workflow uses ivar also to identify amplicons affected by ARTIC primer-binding site mutations and tries to exclude reads derived from such tainted amplicons when calculating allele-frequencies of other variants.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->COVID-19<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ARTIC<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->covid19.galaxyproject.org<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fsars-cov-2-pe-illumina-artic-variant-calling%2FCOVID-19-PE-ARTIC-ILLUMINA/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-11d717f8-92ef-4dd7-bda7-c193175fece2"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">dada2 amplicon analysis pipeline - for paired end data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">dada2 amplicon analysis for paired end data

The workflow has three main outputs: 
- the sequence table (output of makeSequenceTable)
- the taxonomy (output of assignTaxonomy)
- the counts which allow to track the number of sequences in the samples through the steps (output of sequence counts)</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:amplicon<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fdada2%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-718df5b9-9cbb-4792-be50-604e18ef0548"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-II-denoising%2FIIa-denoising-se/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-8b8c4f4c-7cc0-422d-9a6f-47357faa0082"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-II-denoising%2FIIb-denoising-pe/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-951c418e-f57e-491d-af38-f2c44c6089ce"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 Ia: multiplexed data (single-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Importing single-end multiplexed data (not demultiplexed yet)</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-I-import%2FIa-import-multiplexed-se/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-b09b5420-bb7c-4f27-9c90-7e2031889379"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 Ib: multiplexed data (paired-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Importing paired-end multiplexed data (not demultiplexed yet)</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-I-import%2FIb-import-multiplexed-pe/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-5a57f873-52f8-4c51-9e8d-e747d32d183a"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 Ic: Demultiplexed data (single-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Importing demultiplexed data (single-end)</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-I-import%2FIc-import-demultiplexed-se/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f4887024-6f68-4d53-a889-5635366dd18d"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 Id: Demultiplexed data (paired-end)</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Importing demultiplexed data (paired-end)</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-I-import%2FId-import-demultiplexed-pe/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-3163175d-7673-42d9-8aae-505afbd74946"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow 
- Reconstruct phylogeny (insert fragments in a reference)
- Alpha rarefaction analysis
- Taxonomic analysis</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-III-VI-downsteam%2FQIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-300b5a7d-f763-486a-a8ea-87ea84571f19"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">QIIME2 VI: Diversity metrics and estimations</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">The first step in hypothesis testing in microbial ecology is typically to look at within- (alpha) and between-sample (beta) diversity. We can calculate diversity metrics, apply appropriate statistical tests, and visualize the data using the q2-diversity plugin.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fqiime2-III-VI-downsteam%2FQIIME2-VI-diversity-metrics-and-estimations/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-380b07b6-6444-4aaf-b2f8-788eec425986"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Assembly-Hifi-only-VGP3</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FAssembly-Hifi-only-VGP3%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-aa4960d9-198d-4990-a393-a98efcd3070a"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Assembly-decontamination-VGP9</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP_curated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FAssembly-decontamination-VGP9%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-2b0cba18-2060-483d-96d0-76d976c2b817"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Scaffolding with Hi-C data VGP8</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Scaffolding using HiC data with YAHS.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP_curated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FScaffolding-HiC-VGP8%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-bd09a130-801c-4cb8-b848-ab90b3161322"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Assembly-Hifi-Trio-phasing-VGP5</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FAssembly-Hifi-Trio-phasing-VGP5%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-4893d49b-6e4e-41f3-abf6-6dac5a62a4f1"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Purge duplicate contigs from a diploid assembly VGP6</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Purge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication). This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5)</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP_curated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FPurge-duplicate-contigs-VGP6%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-58adb6d9-0345-44ca-82fd-e394eda5a18b"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">kmer-profiling-hifi-VGP1</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Performs k-mer profiling on PacBio data and generates GenomeScope plots and summary for genome characteristics assessment.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fkmer-profiling-hifi-VGP1%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-9fd3c0de-a497-47dc-96e3-0f3e1998173d"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Scaffolding-BioNano-VGP7</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP_curated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FScaffolding-Bionano-VGP7%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-d146054e-ba07-4d0f-b6af-fef24101d9bc"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">kmer-profiling-hifi-trio-VGP2</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fkmer-profiling-hifi-trio-VGP2%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a4513aba-433a-46a0-84cc-f675a29b205b"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Assembly-Hifi-HiC-phasing-VGP4</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Assemble Genome using PacBio HiFi and HiC data for phasing. Prerequisite: Run k-mer profiling workflow (VGP1). </p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FAssembly-Hifi-HiC-phasing-VGP4%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-40c0a173-1805-46c6-9005-5ead67bc9f48"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Generate Nx and Size plots for multiple assemblies</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FPlot-Nx-Size%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-b4825ad9-fd20-4e21-856b-0b49a67a75f5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Purging duplicates in one haplotype VGP6b</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Purge duplicates from one haplotype. Prerequisites: run after a k-mer profiling workflow (VGP 1 or 2) and a contiging workflow (VGP 3,4 or 5).</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP_curated<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FPurge-duplicates-one-haplotype-VGP6b%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-57920bf9-1741-4170-badf-39fdfcdaa2e9"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Mitogenome-Assembly-VGP0</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Reviewed<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VGP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2FMitogenome-assembly-VGP0%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-8c2d765b-2764-4b66-8542-adc2109c24f2"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Segmentation and counting of cell nuclei in fluorescence microscopy images</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ffluorescence-nuclei-segmentation-and-counting%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-747f2432-7493-44b8-a5ee-82d104407cad"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">ATACseq</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a collection of paired fastq. It will remove bad quality and adapters with cutadapt. Map with Bowtie2 end-to-end. Will remove reads on MT and unconcordant pairs and pairs with mapping quality below 30 and PCR duplicates. Will compute the pile-up on 5&#39; +- 100bp. Will call peaks and count the number of reads falling in the 1kb region centered on the summit. Will compute 2 normalization for coverage: normalized by million reads and normalized by million reads in peaks. Will plot the number of reads for each fragment length.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ATACseq<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fatacseq%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-fe6bda9f-1fc2-4a86-bf8d-e779c79466fc"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">ChIPseq_SR</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a collection of fastqs (single reads). Remove adapters with cutadapt, map with bowtie2. Keep MAPQ30. MACS2 for bam with fixed extension or model.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ChIP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fchipseq-sr%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f71eee25-8402-41da-8284-5298656ab379"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Get Confident Peaks From ChIP_SR replicates</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input SR BAM from ChIP-seq. It calls peaks on each replicate and intersect them. In parallel, each BAM is subsetted to smallest number of reads. Peaks are called using all subsets combined. Only peaks called using a combination of all subsets which have summits intersecting the intersection of at least x replicates will be kept.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ATAC-seq<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fconsensus-peaks%2Fconsensus-peaks-chip-sr/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a2f66906-e9e3-42f9-8629-390017ed2db0"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Get Confident Peaks From ChIP_PE replicates</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input PE BAM from ChIP-seq. It calls peaks on each replicate and intersect them. In parallel, each BAM is subsetted to smallest number of reads. Peaks are called using all subsets combined. Only peaks called using a combination of all subsets which have summits intersecting the intersection of at least x replicates will be kept.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ATAC-seq<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fconsensus-peaks%2Fconsensus-peaks-chip-pe/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f58371cd-2e2d-41da-b1ca-ef10f7330047"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Get Confident Peaks From ATAC or CUTandRUN replicates</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input BAM from ATAC-seq or CUT&amp;amp;RUN. It calls peaks on each replicate and intersect them. In parallel, each BAM is subsetted to smallest number of reads. Peaks are called using all subsets combined. Only peaks called using a combination of all subsets which have summits intersecting the intersection of at least x replicates will be kept.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ATAC-seq<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fconsensus-peaks%2Fconsensus-peaks-atac-cutandrun/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-2196fbff-317e-437d-a022-696a1f95a850"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Hi-C_fastqToCool_hicup_cooler</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a collection of paired fastq. It uses HiCUP to go from fastq to validPair file using the middle of the fragment as coordinates. The pairs are filtered for MAPQ and sorted by cooler to generate a tabix dataset. Cooler is used to generate a balanced cool file to the desired resolution.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Hi-C<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fhic-hicup-cooler%2Fhic-fastq-to-cool-hicup-cooler/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-2a89cbe4-a69f-4bff-9465-2e04b0217e35"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">cHi-C_fastqToCool_hicup_cooler</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow take as input a collection of paired fastq. It uses HiCUP to go from fastq to validPair file. The pairs are filtered for MAPQ and for the region captured. Then, they are sorted by cooler to generate a tabix dataset. Cooler is used to generate a balanced cool file to the desired resolution.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Hi-C<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fhic-hicup-cooler%2Fchic-fastq-to-cool-hicup-cooler/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-48733b55-4cde-45d4-a69e-78faff0db3f5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Hi-C_juicermediumtabixToCool_cooler</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow uses as input a collection of juicer medium tabix files and a genome name. It builds balanced cool file to the desired resolution.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Hi-C<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fhic-hicup-cooler%2Fhic-juicermediumtabix-to-cool-cooler/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-48f579ea-eaac-49ee-b38f-93c82f8b7d76"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Hi-C_fastqToPairs_hicup</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a collection of paired fastq. It uses HiCUP to go from fastq to validPair file. First truncate the fastq using the cutting sequence to guess the fill-in. Then map the truncated fastq. Then asign to fragment and filter the self-ligated and dandling ends or internal (it can also filter for the size). Then it removes the duplicates. Convert the output to be compatible with juicebox or cooler using the middle of the fragment as coordinates. Finally filter for mapping quality</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Hi-C<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fhic-hicup-cooler%2Fhic-fastq-to-pairs-hicup/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-1834d09c-c683-4424-bc8d-57df156c2fc8"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">ChIPseq_PE</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a collection of paired fastqs. Remove adapters with cutadapt, map pairs with bowtie2. Keep MAPQ30 and concordant pairs. MACS2 for paired bam.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ChIP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fchipseq-pe%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-98e9e34b-b203-43db-9a23-5fa72963fa6d"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">CUTandRUN</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow take as input a collection of paired fastq. Remove adapters with cutadapt, map pairs with bowtie2 allowing dovetail. Keep MAPQ30 and concordant pairs. BAM to BED. MACS2 with &quot;ATAC&quot; parameters.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->CUTnRUN<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fcutandrun%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-c6ed4b0e-d719-4997-9860-ba99488ec332"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Average Bigwig between replicates</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">We assume the identifiers of the input list are like:
sample_name_replicateID.
The identifiers of the output list will be:
sample_name</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Faverage-bigwig-between-replicates%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-bb767af6-3641-4a5e-843d-78508fd7577a"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Virtual screening of the SARS-CoV-2 main protease with rDock and pose scoring</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ffragment-based-docking-scoring%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-16455667-cdef-4979-9603-f338e09aa2a1"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MMGBSA calculations with GROMACS</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">MMGBSA simulation and calculation</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgromacs-mmgbsa%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f1aea424-b536-496f-a0e6-fe46df8e6e18"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Create GRO and TOP complex files</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fprotein-ligand-complex-parameterization%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-7950b716-2bc9-4cb6-8472-6506c702adea"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">dcTMD calculations with GROMACS</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Perform dcTMD free energy simulations and calculations</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgromacs-dctmd%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f8552617-4af2-4fa6-883f-0d605225939e"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Pox Virus Illumina Amplicon Workflow from half-genomes</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">A workflow for the analysis of pox virus genomes sequenced as half-genomes (for ITR resolution) in a tiled-amplicon approach</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->pox<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->virology<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fpox-virus-amplicon%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-26114133-2c7d-4f90-8f62-0ed5837b5948"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Influenza A isolate subtyping and consensus sequence generation</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow performs subtyping and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->AIV<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->virology<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Finfluenza-isolates-consensus-and-subtyping%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-be0d055d-25ac-422a-ae9c-b0efdfb5bf86"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Generic variation analysis on WGS PE data</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Workflow for variant analysis against  a reference genome in GenBank format</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->mpxv<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->generic<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgeneric-variant-calling-wgs-pe%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-d5b4eace-d876-4999-bad3-14c1cb02b0c9"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Paired end variant calling in haploid system</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Workflow for variant analysis against a reference genome in GenBank format</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->generic<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->VeuPath<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Haploid<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fhaploid-variant-calling-wgs-pe%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-9e94da53-d8ab-4713-8aac-7c6674fa1d39"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Generic variation analysis reporting</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes a VCF dataset of variants produced by any of the variant calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->mpvx<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->generic<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fvariation-reporting%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-5ab1d724-01d4-4673-a3c3-ae2bc944458f"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">RNA-seq for Paired-end fastqs</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a list of paired-end fastqs. Adapters and bad quality bases are removed with fastp. Reads are mapped with STAR with ENCODE parameters and genes are counted simultaneously as well as normalized coverage (per million mapped reads) on uniquely mapped reads. The counts are reprocessed to be similar to HTSeq-count output. Alternatively, featureCounts can be used to count the reads/fragments per gene. FPKM are computed with cufflinks and/or with StringTie. The unstranded normalized coverage is computed with bedtools.
</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->RNAseq<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->transcriptomics<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Frnaseq-pe%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-885ec907-9f61-4a11-9d23-2243389e7335"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">RNA-seq for Single-read fastqs</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes as input a list of single-end fastqs. Adapters and bad quality bases are removed with fastp. Reads are mapped with STAR with ENCODE parameters and genes are counted simultaneously as well as normalized coverage (per million mapped reads) on uniquely mapped reads. The counts are reprocessed to be similar to HTSeq-count output. Alternatively, featureCounts can be used to count the reads/fragments per gene. FPKM are computed with cufflinks and/or with StringTie. The unstranded normalized coverage is computed with bedtools.
</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->RNAseq<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->transcriptomics<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Frnaseq-sr%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-ac8ef6cb-1621-4b73-9362-8a460ef3c3a2"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">BREW3R</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow takes a collection of BAM (output of STAR) and a gtf. It extends the input gtf using de novo annotation.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fbrew3r%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-15d188a7-b028-4cf7-bb5d-13858e1f6e78"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">RNAseq_DE_filtering_plotting</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow can only work on an experimental setup with exactly 2 conditions. It takes two collections of count tables as input and performs differential expression analysis. Additionally it filters for DE genes based on adjusted p-value  and log2 fold changes thresholds. It also generates informative plots.
</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->transcriptomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->RNAseq<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Frnaseq-de%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-84a63bc6-a96a-4263-a1e5-bc140f412e42"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Goseq GO-KEGG Enrichment Analysis</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow is used for GO and KEGG enrichment analysis using GOseq tools.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgoseq%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-24ab5ff5-6b5b-4caf-8cf0-c2f2ccc139e8"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Genome annotation with Helixer</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. </p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fannotation-helixer%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-4cbba315-c9bc-4895-aeeb-57dadef3542a"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Functional annotation of protein sequences</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow uses eggNOG mapper and InterProScan for functional annotation of protein sequences.</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ffunctional-annotation-protein-sequences%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-b03da1a7-fc6f-419f-aa2c-431a670f9a96"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">MetaProSIP OpenMS 2.8</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics </p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fopenms-metaprosip%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-94138387-134e-4622-96a9-fdaee428ee1b"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Clinical Metaproteomics Discovery Workflow</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Workflow for clinical metaproteomics database searching</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:clinicalMP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fclinicalmp-discovery%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-758f86d1-b81e-438d-9666-bc7dc31b4627"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Generate a Clinical Metaproteomics Database</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">The workflow begins with the Database Generation process. The Galaxy-P team has developed a workflow that collects protein sequences from known disease-causing microorganisms to build a comprehensive database. This extensive database is then refined into a smaller, more relevant dataset using the Metanovo tool.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:clinicalMP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fclinicalmp-database-generation%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-22302833-4954-4414-8725-d8bf74d6ffd0"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Clinical Metaproteomics Verification Workflow</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">In proteomics research, verifying detected peptides is essential for ensuring data accuracy and biological relevance. This tutorial continues from the clinical metaproteomics discovery workflow, focusing on verifying identified microbial peptides using the PepQuery tool.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:clinicalMP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fclinicalmp-verification%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-3a5d60ec-2b30-47cf-a351-7b85df8af9e3"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Clinical Metaproteomics Data Interpretation</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow will perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:clinicalMP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fclinicalmp-data-interpretation%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-e21d954c-bb46-46f3-8081-0095b6b0de1f"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Clinical Metaproteomics Quantitation</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Clinical Metaproteomics 4: Quantitation </p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:clinicalMP<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fclinicalmp-quantitation%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-dfcedd6a-9dc1-4dcd-8b15-dbd21482f463"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">amr_gene_detection</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Antimicrobial resistance gene detection from assembled bacterial genomes</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->fasta<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ABRomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->antibiotic-resistance<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->antimicrobial-resistance-genes<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->antimicrobial resistance<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->bacterial-genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->AMR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->AMR-detection<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Famr_gene_detection%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-d55a9e46-a205-4143-b92f-3783499063ed"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">bacterial_genome_annotation</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Annotation of an assembled bacterial genomes to detect genes, potential plasmids, integrons and Insertion sequence (IS) elements.</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->fasta<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ABRomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->bacterial-genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Annotation<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->genome-annotation<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fbacterial_genome_annotation%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-f25be8fa-7823-456f-9707-a497703f48d7"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Repeat masking with RepeatModeler and RepeatMasker</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4"></p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Frepeatmasking%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a2b0d4f7-164c-40d3-ac34-f248db18bbef"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Mass spectrometry: GCMS with metaMS</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract and the metaMS R package (Wehrens, R 2014) for the field of untargeted metabolomics. 

https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metabolomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MS<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->workflow4metabolomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->GC-MS<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->GTN<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metaMS<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgcms-metams%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-4b4dda98-7422-4207-b9bd-62c11ce8e0f0"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Mass spectrometry: LC-MS preprocessing with XCMS</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package (Kuhl, C 2012).


https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html </p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->metabolomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->MS<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->LC-MS<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->workflow4metabolomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->xcms<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->GTN<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Flcms-preprocessing%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-bfe7f341-07ef-4bfe-9b50-2bc7628d6c55"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Bacterial Genome Assembly using Shovill</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Assembly of bacterial paired-end short read data with generation of quality metrics and reports</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->fastq<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->bacterial-genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->paired-end<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->assembly<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->quality<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ABRomics<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fbacterial-genome-assembly%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-68d6270a-bfd1-452a-abc9-4a314e13af85"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Assembly polishing with long reads</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Racon polish with long reads, x4</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fpolish-with-long-reads%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-83ef6fdc-b8a9-41af-a824-0225f0199e63"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Genome assembly with Flye</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Assemble long reads with Flye, then view assembly statistics and assembly graph</p><!----><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fassembly-with-flye%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-eb1c667d-639d-4403-a2b4-1cb6683e0fa5"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Quality and Contamination Control For Genome Assembly</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Short paired-end read analysis to provide quality analysis, read cleaning and taxonomy assignation</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->Genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->fastq<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->bacterial-genomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->taxonomy-assignment<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->paired-end<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->quality<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->ABRomics<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->trimming<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fquality-and-contamination-control%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-d6795753-1562-4ccc-8f9c-f598acd5404e"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Gene-based Pathogen Identification</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Nanopore datasets analysis - Phylogenetic Identification - antibiotic resistance genes detection and contigs building</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Collection<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:PathoGFAIR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:IWC<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microGalaxy<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fgene-based-pathogen-identification%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-0b976704-e1e0-4cc3-9e75-38bda67c01ff"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Nanopore Preprocessing</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Microbiome - QC and Contamination Filtering</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Collection<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microGalaxy<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:PathoGFAIR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Nanopore<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:IWC<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fnanopore-pre-processing%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-435875a1-72c4-4fda-8943-3a78f25c6574"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Allele-based Pathogen Identification</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Microbiome - Variant calling and Consensus Building</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Collection<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microGalaxy<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:PathoGFAIR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:IWC<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fallele-based-pathogen-identification%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-cb26018e-9d2f-4513-a372-122474df703e"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Pathogens of all samples report generation and visualization</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Collection<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microGalaxy<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:PathoGFAIR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:IWC<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Fpathogen-detection-pathogfair-samples-aggregation-and-visualisation%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><div class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 mb-4 p-6" id="workflow-a55428c8-9934-4941-93c4-3920b4ba3366"><div class="px-4 py-5 sm:px-6"><!--[--><h2 class="text-xl font-bold mb-2">Taxonomy Profiling and Visualization with Krona</h2><!--]--></div><div class="px-4 py-5 sm:p-6"><!--[--><p class="mb-4">Microbiome - Taxonomy Profiling</p><div class="mb-4"><!--[--><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:Collection<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:microGalaxy<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:PathoGFAIR<!--]--></span><span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 text-primary-500 dark:text-primary-400 mr-2 mb-2"><!--[-->name:IWC<!--]--></span><!--]--></div><!--[--><a class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center" href="/workflow/%23workflow%2Fgithub.com%2Fiwc-workflows%2Ftaxonomy-profiling-and-visualization-with-krona%2Fmain/"><!--[--><!--[--><!----><!--]--><!--[-->Details<!--]--><!--[--><!----><!--]--><!--]--></a><!--]--><!--]--></div><!----></div><!--]--><!--]--></div></div></main></div></div><div id="teleports"></div><script type="application/json" id="__NUXT_DATA__" data-ssr="true" data-src="/_payload.json?827a8048-e294-42c6-bf63-c21b2e56230e">[{"state":1,"once":7,"_errors":9,"serverRendered":5,"path":11,"pinia":12,"prerenderedAt":17},["Reactive",2],{"$scolor-mode":3},{"preference":4,"value":4,"unknown":5,"forced":6},"light",true,false,["Reactive",8],["Set"],["Reactive",10],{},"/",["Reactive",13],{"workflow":14},{"workflow":15},["EmptyRef",16],"null",1740507588405]</script>
<script>window.__NUXT__={};window.__NUXT__.config={public:{mdc:{components:{prose:true,map:{p:"prose-p",a:"prose-a",blockquote:"prose-blockquote","code-inline":"prose-code-inline",code:"ProseCodeInline",em:"prose-em",h1:"prose-h1",h2:"prose-h2",h3:"prose-h3",h4:"prose-h4",h5:"prose-h5",h6:"prose-h6",hr:"prose-hr",img:"prose-img",ul:"prose-ul",ol:"prose-ol",li:"prose-li",strong:"prose-strong",table:"prose-table",thead:"prose-thead",tbody:"prose-tbody",td:"prose-td",th:"prose-th",tr:"prose-tr"}},headings:{anchorLinks:{h1:false,h2:true,h3:true,h4:true,h5:false,h6:false}}},content:{locales:[],defaultLocale:"",integrity:1740507553532,experimental:{stripQueryParameters:false,advanceQuery:false,clientDB:false},respectPathCase:false,api:{baseURL:"/api/_content"},navigation:{fields:[]},tags:{p:"prose-p",a:"prose-a",blockquote:"prose-blockquote","code-inline":"prose-code-inline",code:"ProseCodeInline",em:"prose-em",h1:"prose-h1",h2:"prose-h2",h3:"prose-h3",h4:"prose-h4",h5:"prose-h5",h6:"prose-h6",hr:"prose-hr",img:"prose-img",ul:"prose-ul",ol:"prose-ol",li:"prose-li",strong:"prose-strong",table:"prose-table",thead:"prose-thead",tbody:"prose-tbody",td:"prose-td",th:"prose-th",tr:"prose-tr"},highlight:false,wsUrl:"",documentDriven:false,host:"",trailingSlash:false,search:"",contentHead:true,anchorLinks:{depth:4,exclude:[1]}}},app:{baseURL:"/",buildAssetsDir:"/_nuxt/",cdnURL:""}}</script></body></html>