<!DOCTYPE html><html lang="en"> <head><meta charset="UTF-8"><meta name="viewport" content="width=device-width"><meta name="description" content="Ready-to-use, open-source pipelines with sample data and training materials to make progress quickly and reliably"><title>Intergalactic Workflow Commission</title><!-- Open Graph --><meta property="og:title" content="Intergalactic Workflow Commission"><meta property="og:description" content="Discover and run vetted analysis pipelines on Galaxy"><meta property="og:image" content="https://iwc.galaxyproject.org/iwc_logo.png"><meta property="og:type" content="website"><!-- Twitter Card --><meta name="twitter:card" content="summary"><meta name="twitter:title" content="Intergalactic Workflow Commission"><meta name="twitter:description" content="Discover and run vetted analysis pipelines on Galaxy"><meta name="twitter:image" content="https://iwc.galaxyproject.org/iwc_logo.png"><link rel="preload" as="image" href="/iwc_logo_white.png" fetchpriority="high"><link rel="icon" type="image/png" href="/iwc_logo_white.png"><link rel="stylesheet" href="/_astro/about.R_rN8b3q.css">
<style>.filter-wrapper:last-child .divider[data-v-761c6766]{visibility:hidden}
</style></head> <body class="h-dvh flex flex-col bg-chicago-50 font-sans"> <!-- Header component will go here --> <header> <header class="py-2 px-6 bg-ebony-clay text-white"> <div class="flex"> <div class="flex flex-grow space-x-2"> <a href="/" class="flex items-center space-x-2 hover:text-hokey-pokey"> <img src="/iwc_logo_white.png" alt="IWC Logo" width="64" height="64" loading="eager" fetchpriority="high"> <span class="text-xl font-semibold px-4">Galaxy IWC - Workflow Library</span> </a> </div> <nav class="flex items-center space-x-8"> <a href="/about" class="hover:text-hokey-pokey">About</a> <a href="https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#adding-workflows" class="hover:text-hokey-pokey" target="_blank">Contribute</a> <a href="https://github.com/galaxyproject/iwc/" class="hover:text-hokey-pokey flex items-center gap-1" target="_blank">
GitHub
<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="currentColor" class="inline-block"> <path d="M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z"></path> </svg> </a> </nav> </div> </header> </header> <main class="flex overflow-y-auto"> <div class="w-full overflow-y-auto"> <!-- Hero section if provided --> <div> <div class="flex bg-ebony-clay p-4 items-center flex-col"> <h1 class="text-5xl my-8 text-white text-center font-bold">
Discover and run vetted analysis pipelines on Galaxy
</h1> <h2 class="text-xl text-white text-center font-semibold mb-16">
Ready-to-use, open-source pipelines with sample data and training materials to make progress quickly and reliably.
</h2> <style>astro-island,astro-slot,astro-static-slot{display:contents}</style><script>(()=>{var e=async t=>{await(await t())()};(self.Astro||(self.Astro={})).load=e;window.dispatchEvent(new Event("astro:load"));})();</script><script>(()=>{var A=Object.defineProperty;var g=(i,o,a)=>o in i?A(i,o,{enumerable:!0,configurable:!0,writable:!0,value:a}):i[o]=a;var d=(i,o,a)=>g(i,typeof o!="symbol"?o+"":o,a);{let i={0:t=>m(t),1:t=>a(t),2:t=>new RegExp(t),3:t=>new Date(t),4:t=>new Map(a(t)),5:t=>new Set(a(t)),6:t=>BigInt(t),7:t=>new URL(t),8:t=>new Uint8Array(t),9:t=>new Uint16Array(t),10:t=>new Uint32Array(t),11:t=>1/0*t},o=t=>{let[l,e]=t;return l in i?i[l](e):void 0},a=t=>t.map(o),m=t=>typeof t!="object"||t===null?t:Object.fromEntries(Object.entries(t).map(([l,e])=>[l,o(e)]));class y extends HTMLElement{constructor(){super(...arguments);d(this,"Component");d(this,"hydrator");d(this,"hydrate",async()=>{var b;if(!this.hydrator||!this.isConnected)return;let e=(b=this.parentElement)==null?void 0:b.closest("astro-island[ssr]");if(e){e.addEventListener("astro:hydrate",this.hydrate,{once:!0});return}let c=this.querySelectorAll("astro-slot"),n={},h=this.querySelectorAll("template[data-astro-template]");for(let r of h){let s=r.closest(this.tagName);s!=null&&s.isSameNode(this)&&(n[r.getAttribute("data-astro-template")||"default"]=r.innerHTML,r.remove())}for(let r of c){let s=r.closest(this.tagName);s!=null&&s.isSameNode(this)&&(n[r.getAttribute("name")||"default"]=r.innerHTML)}let p;try{p=this.hasAttribute("props")?m(JSON.parse(this.getAttribute("props"))):{}}catch(r){let s=this.getAttribute("component-url")||"<unknown>",v=this.getAttribute("component-export");throw v&&(s+=` (export ${v})`),console.error(`[hydrate] Error parsing props for component ${s}`,this.getAttribute("props"),r),r}let u;await this.hydrator(this)(this.Component,p,n,{client:this.getAttribute("client")}),this.removeAttribute("ssr"),this.dispatchEvent(new CustomEvent("astro:hydrate"))});d(this,"unmount",()=>{this.isConnected||this.dispatchEvent(new CustomEvent("astro:unmount"))})}disconnectedCallback(){document.removeEventListener("astro:after-swap",this.unmount),document.addEventListener("astro:after-swap",this.unmount,{once:!0})}connectedCallback(){if(!this.hasAttribute("await-children")||document.readyState==="interactive"||document.readyState==="complete")this.childrenConnectedCallback();else{let e=()=>{document.removeEventListener("DOMContentLoaded",e),c.disconnect(),this.childrenConnectedCallback()},c=new MutationObserver(()=>{var n;((n=this.lastChild)==null?void 0:n.nodeType)===Node.COMMENT_NODE&&this.lastChild.nodeValue==="astro:end"&&(this.lastChild.remove(),e())});c.observe(this,{childList:!0}),document.addEventListener("DOMContentLoaded",e)}}async childrenConnectedCallback(){let e=this.getAttribute("before-hydration-url");e&&await import(e),this.start()}async start(){let e=JSON.parse(this.getAttribute("opts")),c=this.getAttribute("client");if(Astro[c]===void 0){window.addEventListener(`astro:${c}`,()=>this.start(),{once:!0});return}try{await Astro[c](async()=>{let n=this.getAttribute("renderer-url"),[h,{default:p}]=await Promise.all([import(this.getAttribute("component-url")),n?import(n):()=>()=>{}]),u=this.getAttribute("component-export")||"default";if(!u.includes("."))this.Component=h[u];else{this.Component=h;for(let f of u.split("."))this.Component=this.Component[f]}return this.hydrator=p,this.hydrate},e,this)}catch(n){console.error(`[astro-island] Error hydrating ${this.getAttribute("component-url")}`,n)}}attributeChangedCallback(){this.hydrate()}}d(y,"observedAttributes",["props"]),customElements.get("astro-island")||customElements.define("astro-island",y)}})();</script><astro-island uid="ZzIgEy" prefix="s1" component-url="/_astro/PopularWorkflows.yFoYqo-M.js" component-export="default" renderer-url="/_astro/client.BrhupUx7.js" props="{&quot;workflowTrsIds&quot;:[1,[[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-pe/main&quot;],[0,&quot;#workflow/github.com/iwc-workflows/chipseq-pe/main&quot;],[0,&quot;#workflow/github.com/iwc-workflows/atacseq/main&quot;]]],&quot;workflows&quot;:[1,[[0,{&quot;iwcID&quot;:[0,&quot;protein-ligand-complex-parameterization-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/protein-ligand-complex-parameterization/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f1aea424-b536-496f-a0e6-fe46df8e6e18&quot;],&quot;name&quot;:[0,&quot;Create GRO and TOP complex files&quot;],&quot;annotation&quot;:[0,&quot;&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fragment-based-docking-scoring-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fragment-based-docking-scoring/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;bb767af6-3641-4a5e-843d-78508fd7577a&quot;],&quot;name&quot;:[0,&quot;Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring&quot;],&quot;annotation&quot;:[0,&quot;Virtual screening of the SARS-CoV-2 main protease with rDock and pose scoring&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gromacs-mmgbsa-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gromacs-mmgbsa/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;16455667-cdef-4979-9603-f338e09aa2a1&quot;],&quot;name&quot;:[0,&quot;MMGBSA calculations with GROMACS&quot;],&quot;annotation&quot;:[0,&quot;MMGBSA simulation and calculation&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gromacs-dctmd-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gromacs-dctmd/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7950b716-2bc9-4cb6-8472-6506c702adea&quot;],&quot;name&quot;:[0,&quot;dcTMD calculations with GROMACS&quot;],&quot;annotation&quot;:[0,&quot;Perform dcTMD free energy simulations and calculations&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial-quality-and-contamination-control-post-assembly-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial-quality-and-contamination-control-post-assembly/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8911f5cd-67b1-4d57-be62-abe84ca07f23&quot;],&quot;name&quot;:[0,&quot;Post-Assembly Quality Control and Contamination Check for Bacterial Genomes&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs quality and contamination control analysis on assembled contigs to assess bacterial genome quality and taxonomic assignment&quot;],&quot;release&quot;:[0,&quot;1.0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;taxonomy-assignment&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;],[0,&quot;fasta&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;amr_gene_detection-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/amr_gene_detection/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-11T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;284cc748-7532-4749-8f68-d19d41c048ef&quot;],&quot;name&quot;:[0,&quot;amr_gene_detection&quot;],&quot;annotation&quot;:[0,&quot;Antimicrobial resistance gene detection from assembled bacterial genomes&quot;],&quot;release&quot;:[0,&quot;1.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;fasta&quot;],[0,&quot;Genomics&quot;],[0,&quot;ABRomics&quot;],[0,&quot;antibiotic-resistance&quot;],[0,&quot;antimicrobial-resistance-genes&quot;],[0,&quot;antimicrobialresistance&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;AMR&quot;],[0,&quot;AMR-detection&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial_genome_annotation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial_genome_annotation/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;47a7aa1d-8cb6-4eab-a6d8-86909bca0d43&quot;],&quot;name&quot;:[0,&quot;bacterial_genome_annotation&quot;],&quot;annotation&quot;:[0,&quot;Annotation of an assembled bacterial genomes to detect genes, potential plasmids, integrons and Insertion sequence (IS) elements.&quot;],&quot;release&quot;:[0,&quot;1.1.11&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;fasta&quot;],[0,&quot;ABRomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;Annotation&quot;],[0,&quot;genome-annotation&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;repeatmasking-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/repeatmasking/main&quot;],&quot;updated&quot;:[0,&quot;2023-09-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f25be8fa-7823-456f-9707-a497703f48d7&quot;],&quot;name&quot;:[0,&quot;Repeat masking with RepeatModeler and RepeatMasker&quot;],&quot;annotation&quot;:[0,&quot;&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;parallel-accession-download-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/parallel-accession-download/main&quot;],&quot;updated&quot;:[0,&quot;2024-05-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Data Fetching&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;15c3229d-8786-457f-a020-7aa64c0cf6e8&quot;],&quot;name&quot;:[0,&quot;Parallel Accession Download&quot;],&quot;annotation&quot;:[0,&quot;Downloads fastq files for sequencing run accessions provided in a text file using fasterq-dump. Creates one job per listed run accession.&quot;],&quot;release&quot;:[0,&quot;0.1.14&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sra-manifest-to-concatenated-fastqs-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sra-manifest-to-concatenated-fastqs/main&quot;],&quot;updated&quot;:[0,&quot;2024-06-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Data Fetching&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;363898af-e598-4f0e-abd9-e6ded395ce66&quot;],&quot;name&quot;:[0,&quot;sra_manifest_to_concatenated_fastqs_parallel&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes as input a SRA_manifest from SRA Run Selector and will generate one fastq file or fastq pair of file for each experiment (concatenated multiple runs if necessary). Output will be relabelled to match the column specified by the user.&quot;],&quot;release&quot;:[0,&quot;0.7&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mfassignr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mfassignr/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-16T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3be0019d-8058-4e57-922a-417b9e3578a2&quot;],&quot;name&quot;:[0,&quot;Molecular formula assignment and recalibration with MFAssignR package.&quot;],&quot;annotation&quot;:[0,&quot;This workflow can be used to assign multi-element molecular formulas to ultrahigh resolution mass spectra.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;recalibration&quot;],[0,&quot;noiseestimation&quot;],[0,&quot;formulaassignment&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;lcms-preprocessing-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/lcms-preprocessing/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;af2457b4-9013-4d8c-9e76-9948c1f96fcc&quot;],&quot;name&quot;:[0,&quot;Mass spectrometry: LC-MS preprocessing with XCMS&quot;],&quot;annotation&quot;:[0,&quot;This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package (Kuhl, C 2012).\n\n\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html &quot;],&quot;release&quot;:[0,&quot;1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;MS&quot;],[0,&quot;LC-MS&quot;],[0,&quot;workflow4metabolomics&quot;],[0,&quot;xcms&quot;],[0,&quot;GTN&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qcxms-sdf-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qcxms-sdf/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;067945f1-8716-4558-918e-4a9c9d42f604&quot;],&quot;name&quot;:[0,&quot;QCxMS Spectra Prediction from SDF&quot;],&quot;annotation&quot;:[0,&quot;Workflow to predict EI mass spectra using QCxMS starting from a single SDF file, containing the 3D coordinates of all atoms in the molecule. These files can typically be obtained from PubChem.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gcms-metams-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gcms-metams/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7abd10fb-d083-4732-b11f-56581e23905d&quot;],&quot;name&quot;:[0,&quot;Mass spectrometry: GCMS with metaMS&quot;],&quot;annotation&quot;:[0,&quot;This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract and the metaMS R package (Wehrens, R 2014) for the field of untargeted metabolomics. \n\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;MS&quot;],[0,&quot;workflow4metabolomics&quot;],[0,&quot;GC-MS&quot;],[0,&quot;GTN&quot;],[0,&quot;metaMS&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mags-building-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mags-building/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e4e70fc0-b584-4e86-8b8f-c6bcdda2e11a&quot;],&quot;name&quot;:[0,&quot;Metagenome-Assembled Genomes (MAGs) generation&quot;],&quot;annotation&quot;:[0,&quot;This workflow constructs Metagenome-Assembled Genomes (MAGs) using SPAdes or MEGAHIT as assemblers, followed by binning with four different tools and refinement using Binette. The resulting MAGs are dereplicated across the entire input sample set, then annotated and evaluated for quality.\nYou can provide pooled reads (for co-assembly/binning), individual read sets, or a combination of both. The input samples must consist of the original reads, which are used for abundance estimation. In all cases, reads should be trimmed, adapters removed, and cleaned of host or other contaminants before processing.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:FAIRyMAGs&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gene-based-pathogen-identification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gene-based-pathogen-identification/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d6795753-1562-4ccc-8f9c-f598acd5404e&quot;],&quot;name&quot;:[0,&quot;Gene-based Pathogen Identification&quot;],&quot;annotation&quot;:[0,&quot;Nanopore datasets analysis - Phylogenetic Identification - antibiotic resistance genes detection and contigs building&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;],[0,&quot;name:microGalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;taxonomy-profiling-and-visualization-with-krona-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/taxonomy-profiling-and-visualization-with-krona/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a55428c8-9934-4941-93c4-3920b4ba3366&quot;],&quot;name&quot;:[0,&quot;Taxonomy Profiling and Visualization with Krona&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - Taxonomy Profiling&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;nanopore-pre-processing-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/nanopore-pre-processing/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;0b976704-e1e0-4cc3-9e75-38bda67c01ff&quot;],&quot;name&quot;:[0,&quot;Nanopore Preprocessing&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - QC and Contamination Filtering&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:Nanopore&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pathogen-detection-pathogfair-samples-aggregation-and-visualisation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pathogen-detection-pathogfair-samples-aggregation-and-visualisation/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;cb26018e-9d2f-4513-a372-122474df703e&quot;],&quot;name&quot;:[0,&quot;Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation&quot;],&quot;annotation&quot;:[0,&quot;Pathogens of all samples report generation and visualization&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;allele-based-pathogen-identification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/allele-based-pathogen-identification/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;435875a1-72c4-4fda-8943-3a78f25c6574&quot;],&quot;name&quot;:[0,&quot;Allele-based Pathogen Identification&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - Variant calling and Consensus Building&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-artic-ivar-analysis-sars-cov-2-illumina-amplicon-ivar-pangolin-nextclade&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-ivar-analysis/SARS-COV-2-ILLUMINA-AMPLICON-IVAR-PANGOLIN-NEXTCLADE&quot;],&quot;updated&quot;:[0,&quot;2025-04-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3d86e5fc-cad1-4e21-8ecc-d0e6637899bc&quot;],&quot;name&quot;:[0,&quot;SARS-CoV-2 Illumina Amplicon pipeline - iVar based&quot;],&quot;annotation&quot;:[0,&quot;Find and annotate variants in ampliconic SARS-CoV-2 Illumina sequencing data and classify samples with pangolin and nextclade&quot;],&quot;release&quot;:[0,&quot;0.3.1&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;iwc&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-consensus-from-variation-covid-19-consensus-construction&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-consensus-from-variation/COVID-19-CONSENSUS-CONSTRUCTION&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;06dc40a7-99f2-4b3c-ae21-b3dcb239306f&quot;],&quot;name&quot;:[0,&quot;COVID-19: consensus construction&quot;],&quot;annotation&quot;:[0,&quot;Build a consensus sequence from FILTER PASS variants with intrasample allele-frequency above a configurable consensus threshold.\nHard-mask regions with low coverage (but not consensus variants within them) and ambiguous sites.&quot;],&quot;release&quot;:[0,&quot;0.4.3&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-artic-variant-calling-covid-19-pe-artic-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2025-03-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;864427b0-d25a-499c-b437-f83f2d6b83f2&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on ARTIC PE data&quot;],&quot;annotation&quot;:[0,&quot;The workflow for Illumina-sequenced ARTIC data builds on the RNASeq workflow for paired-end data using the same steps for mapping and variant calling, but adds extra logic for trimming ARTIC primer sequences off reads with the ivar package. In addition, this workflow uses ivar also to identify amplicons affected by ARTIC primer-binding site mutations and tries to exclude reads derived from such tainted amplicons when calculating allele-frequencies of other variants.&quot;],&quot;release&quot;:[0,&quot;0.5.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-variation-reporting-covid-19-variation-reporting&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-variation-reporting/COVID-19-VARIATION-REPORTING&quot;],&quot;updated&quot;:[0,&quot;2024-09-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b08c744d-7c61-4b58-ac5f-4b5886c3c643&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis reporting&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes a VCF dataset of variants produced by any of the *-variant-calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.&quot;],&quot;release&quot;:[0,&quot;0.3.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-se-illumina-wgs-variant-calling-covid-19-se-wgs-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-se-illumina-wgs-variant-calling/COVID-19-SE-WGS-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5171bc59-590c-4e97-8db8-004f73928627&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on WGS SE data&quot;],&quot;annotation&quot;:[0,&quot;This workflows performs single end read mapping with bowtie2 followed by sensitive variant calling across a wide range of AFs with lofreq&quot;],&quot;release&quot;:[0,&quot;0.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-wgs-variant-calling-covid-19-pe-wgs-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-wgs-variant-calling/COVID-19-PE-WGS-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;6695eb16-30a5-42ea-8591-56f2630cf8bf&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on WGS PE data&quot;],&quot;annotation&quot;:[0,&quot;This workflows performs paired end read mapping with bwa-mem followed by sensitive variant calling across a wide range of AFs with lofreq&quot;],&quot;release&quot;:[0,&quot;0.2.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;],[0,&quot;iwc&quot;],[0,&quot;emergen_validated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-ont-artic-variant-calling-covid-19-artic-ont&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-ont-artic-variant-calling/COVID-19-ARTIC-ONT&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a1ceaf35-f3d7-452d-b5ed-53a4f429d1cf&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis of ARTIC ONT data&quot;],&quot;annotation&quot;:[0,&quot;This workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.&quot;],&quot;release&quot;:[0,&quot;0.3.2&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;ONT&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;brew3r-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/brew3r/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-07T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;ac8ef6cb-1621-4b73-9362-8a460ef3c3a2&quot;],&quot;name&quot;:[0,&quot;BREW3R&quot;],&quot;annotation&quot;:[0,&quot;Extends 3&#39; ends of gene annotations using BAM files (from STAR alignments) and a reference GTF. Specifically designed for 3&#39;-biased sequencing techniques like 10X scRNA-seq or BRB-seq that primarily capture transcript 3&#39; ends. The BREW3R tool enhances annotations by using evidence from RNA-seq data to improve 3&#39; UTR definitions, which is particularly important for accurate quantification in single-cell and bulk RNA-seq experiments.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;goseq-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/goseq/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-03T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;84a63bc6-a96a-4263-a1e5-bc140f412e42&quot;],&quot;name&quot;:[0,&quot;Gene Ontology and KEGG Pathway Enrichment Analysis&quot;],&quot;annotation&quot;:[0,&quot;Performs functional enrichment analysis of gene sets using GOseq, identifying over-represented Gene Ontology terms and KEGG pathways. This workflow accounts for gene length bias in RNA-seq data when calculating enrichment statistics. Input requires differentially expressed genes and gene lengths. Generates comprehensive tables and visualizations of enriched GO terms across all three ontologies (Biological Process, Molecular Function, Cellular Component) as well as KEGG pathway enrichment results.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-de-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-de/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;15d188a7-b028-4cf7-bb5d-13858e1f6e78&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Differential Expression Analysis with Visualization&quot;],&quot;annotation&quot;:[0,&quot;Identifies differentially expressed genes between exactly two experimental conditions from count tables. The workflow performs statistical testing, applies filters based on adjusted p-value and log2 fold change thresholds, and generates publication-quality visualizations including volcano plots, MA plots, and heatmaps. Takes two collections of count tables as input and produces filtered gene lists and interactive plots for interpreting expression differences. Optimal for simple two-condition experimental designs.&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;transcriptomics&quot;],[0,&quot;RNAseq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-pe/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5ab1d724-01d4-4673-a3c3-ae2bc944458f&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Analysis: Paired-End Read Processing and Quantification&quot;],&quot;annotation&quot;:[0,&quot;Complete RNA-Seq analysis for paired-end data: Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;1.2&quot;],&quot;tags&quot;:[1,[[0,&quot;RNAseq&quot;],[0,&quot;transcriptomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-sr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-sr/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;885ec907-9f61-4a11-9d23-2243389e7335&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Analysis: Single-End Read Processing and Quantification&quot;],&quot;annotation&quot;:[0,&quot;Complete RNA-Seq analysis for single-end data:  Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;1.2&quot;],&quot;tags&quot;:[1,[[0,&quot;RNAseq&quot;],[0,&quot;transcriptomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;atacseq-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/atacseq/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;747f2432-7493-44b8-a5ee-82d104407cad&quot;],&quot;name&quot;:[0,&quot;ATAC-seq Analysis: Chromatin Accessibility Profiling&quot;],&quot;annotation&quot;:[0,&quot;Complete ATAC-seq analysis pipeline for paired-end reads. Processes raw FASTQ data through adapter and bad quality removal (cutadapt), alignment (Bowtie2 end-to-end), and filtering (removes MT reads, discordant pairs, low mapping quality &lt;30, PCR duplicates). Generates 5&#39; cut site pileups (±100bp), performs peak calling, and quantifies reads in 1kb summit-centered regions. Produces two normalized coverage tracks (per million mapped reads and per million reads in peaks) and fragment length distribution plots for quality assessment.&quot;],&quot;release&quot;:[0,&quot;1.0&quot;],&quot;tags&quot;:[1,[[0,&quot;ATACseq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-chip-sr&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-sr&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f71eee25-8402-41da-8284-5298656ab379&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ChIP-seq Single-End Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ChIP-seq single-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-chip-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-pe&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a2f66906-e9e3-42f9-8629-390017ed2db0&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ChIP-seq Paired-End Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ChIP-seq paired-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-atac-cutandrun&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-atac-cutandrun&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f58371cd-2e2d-41da-b1ca-ef10f7330047&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ATAC-seq and CUT&amp;RUN Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ATAC-seq or CUT&amp;RUN replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-fastq-to-cool-hicup-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-cool-hicup-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;2196fbff-317e-437d-a022-696a1f95a850&quot;],&quot;name&quot;:[0,&quot;Hi-C Processing: FASTQ to Balanced Cool Files&quot;],&quot;annotation&quot;:[0,&quot;Comprehensive Hi-C data processing workflow that converts paired-end FASTQ files to balanced contact matrices. Uses HiCUP for pre-processing (adapter trimming, mapping, filtering) with fragment midpoint coordinate assignment, filters reads by mapping quality, and generates sorted tabix files. The final output is a balanced cool file at user-specified resolution created with cooler, ready for multi-resolution contact matrix analysis.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-chic-fastq-to-cool-hicup-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/chic-fastq-to-cool-hicup-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;2a89cbe4-a69f-4bff-9465-2e04b0217e35&quot;],&quot;name&quot;:[0,&quot;Capture Hi-C Processing: FASTQ to Balanced Cool Files&quot;],&quot;annotation&quot;:[0,&quot;End-to-end processing of Capture Hi-C paired-end sequencing data. Transforms raw FASTQ files into balanced contact matrices using HiCUP for processing Hi-C reads (adapter trimming, mapping, filtering for valid pairs) with additional filtering for MAPQ and captured regions. The workflow generates sorted tabix files via cooler and produces balanced cool files at user-specified resolution for downstream analysis and visualization.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-juicermediumtabix-to-cool-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-juicermediumtabix-to-cool-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;48733b55-4cde-45d4-a69e-78faff0db3f5&quot;],&quot;name&quot;:[0,&quot;Hi-C Format Conversion: Juicer Medium to Cooler Files&quot;],&quot;annotation&quot;:[0,&quot;Converts Hi-C interaction data from Juicer medium format (tabix files) to balanced cooler format. This streamlined workflow takes a collection of Juicer medium tabix files and a reference genome name as input and generates balanced cool files at user-specified resolution. Enables seamless transition between different Hi-C analysis ecosystems while maintaining data integrity.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-fastq-to-pairs-hicup&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-pairs-hicup&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;48f579ea-eaac-49ee-b38f-93c82f8b7d76&quot;],&quot;name&quot;:[0,&quot;Hi-C Data Processing: FASTQ to Valid Interaction Pairs&quot;],&quot;annotation&quot;:[0,&quot;Processes Hi-C paired-end FASTQ files to generate validated interaction pairs using HiCUP. The workflow truncates reads at ligation junctions, maps to reference genome, assigns to restriction fragments, and filters out experimental artifacts (self-ligated, dangling ends, internal fragments, and size outliers). Removes duplicates and converts outputs to formats compatible with Juicebox/cooler using fragment midpoints as coordinates. Final filtering by mapping quality ensures high-confidence interaction data.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;chipseq-sr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/chipseq-sr/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b57e88a9-fe46-4995-812c-16c28efeac21&quot;],&quot;name&quot;:[0,&quot;ChIP-seq Analysis: Single-End Read Processing&quot;],&quot;annotation&quot;:[0,&quot;Complete ChIP-seq analysis for single-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and quality filtering (MAPQ &amp;gt;= 30). Peak calling with MACS2 uses either a fixed extension parameter or built-in model to identify protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;0.15&quot;],&quot;tags&quot;:[1,[[0,&quot;ChIP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;cutandrun-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/cutandrun/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;98e9e34b-b203-43db-9a23-5fa72963fa6d&quot;],&quot;name&quot;:[0,&quot;CUT&amp;RUN/CUT&amp;TAG Analysis: Protein-DNA Interaction Mapping&quot;],&quot;annotation&quot;:[0,&quot;Complete CUT&amp;RUN/CUT&amp;TAG analysis workflow for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt) and alignment (Bowtie2 with dovetail option enabled). Applies quality filtering (MAPQ ≥ 30, concordant pairs only), converts BAM to BED format, and performs peak calling using MACS2 with parameters optimized for the punctate signal profile characteristic of CUT&amp;RUN/CUT&amp;TAG experiments.&quot;],&quot;release&quot;:[0,&quot;0.14&quot;],&quot;tags&quot;:[1,[[0,&quot;CUTnRUN&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;chipseq-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/chipseq-pe/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;25c8f8e3-7ed9-4523-8862-8496fa1777cd&quot;],&quot;name&quot;:[0,&quot;ChIP-seq Analysis: Paired-End Read Processing&quot;],&quot;annotation&quot;:[0,&quot;Complete ChIP-seq analysis for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and stringent quality filtering (MAPQ &amp;gt;= 30, concordant pairs only). Peak calling with MACS2 optimized for paired-end reads identifies protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;0.15&quot;],&quot;tags&quot;:[1,[[0,&quot;ChIP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;average-bigwig-between-replicates-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/average-bigwig-between-replicates/main&quot;],&quot;updated&quot;:[0,&quot;2023-09-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;],[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c6ed4b0e-d719-4997-9860-ba99488ec332&quot;],&quot;name&quot;:[0,&quot;BigWig Replicates Averaging Workflow&quot;],&quot;annotation&quot;:[0,&quot;Calculates average signal values across replicate BigWig files to generate a single consolidated track. Requires input file identifiers formatted as \&quot;sample_name_replicateID\&quot;. The tool automatically groups files by sample name, computes the mean signal at each position, and outputs consolidated BigWig files named after the common sample prefix (without the replicate ID suffix).&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fluorescence-nuclei-segmentation-and-counting-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fluorescence-nuclei-segmentation-and-counting/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-07T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8c2d765b-2764-4b66-8542-adc2109c24f2&quot;],&quot;name&quot;:[0,&quot;Segmentation and counting of cell nuclei in fluorescence microscopy images&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;tissue-microarray-analysis-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/tissue-microarray-analysis/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c880e149-6237-46b6-aac6-7e1e3bb58ede&quot;],&quot;name&quot;:[0,&quot;End-to-End Tissue Microarray Analysis&quot;],&quot;annotation&quot;:[0,&quot;Complete multiplex tissue image (MTI) analysis pipeline for tissue microarray (TMA) data imaged using cyclic immunofluorescence: Performs illumination correction, stitching and registration, and tissue microarray segmentation. Tissue-segmented images undergo nuclear segmentation, cell/nuclei feature quantification (mean marker intensities, cell coordinates, and morphological features), and cell phenotyping. Produces outputs that are compatible with downstream single-cell/spatial analysis and interactive image viewers including: Pyramidal OME-TIFF images, nuclear segmentation masks (TIFF), quantified feature tables (CSV, h5ad) with cell type annotations, and an interactive Vitessce dashboard that combines image viewing with linked single-cell data visualizations. &quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;imaging&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;multiplex-tissue-microarray-analysis-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/multiplex-tissue-microarray-analysis/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-22T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;eeedc263-2a29-4651-9b2b-007bf5d3cf4b&quot;],&quot;name&quot;:[0,&quot;Multiplex Tissue Microarray Analysis&quot;],&quot;annotation&quot;:[0,&quot;Perform background subtraction, nuclear segmentation, feature quantification, cellular phenotyping, spatial analysis, and interactive visualization of registered TMA core multiplex tissue images&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-quality-control-single-end-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-single-end/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;68bf2a73-d68a-4e25-ba52-5586c057df77&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - Quality control SE&quot;],&quot;annotation&quot;:[0,&quot;Quality control subworkflow for single-end reads.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-complete-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-complete/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;fad227f2-8b1e-4782-a7a5-a1743b9643bf&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0&quot;],&quot;annotation&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0. Including the Quality control for single-end and paired-end reads, rRNA-prediction, and ITS sub-WFs.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;amplicon&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mapseq-to-ampvis2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mapseq-to-ampvis2/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e24060a0-1cfe-40f1-9251-48e96f7e1121&quot;],&quot;name&quot;:[0,&quot;MAPseq to ampvis2&quot;],&quot;annotation&quot;:[0,&quot;The MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;taxonomic-rank-abundance-summary-table-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/taxonomic-rank-abundance-summary-table/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;31b19088-dad2-41a0-b086-94037bce4784&quot;],&quot;name&quot;:[0,&quot;Taxonomic abundance summary tables for a specified taxonomic rank&quot;],&quot;annotation&quot;:[0,&quot;This workflow creates taxonomic summary tables for a specified taxonomic rank out of MAPseq&#39;s OTU tables output collection.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-rrna-prediction-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-rrna-prediction/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;53c5bc12-4d7c-4799-b053-ced394665d61&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - rRNA prediction&quot;],&quot;annotation&quot;:[0,&quot;Classification and visualization of SSU, LSU sequences.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;amplicon&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-quality-control-paired-end-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-paired-end/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;467d4418-a41e-44eb-9959-7c4a242d2082&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - Quality control PE&quot;],&quot;annotation&quot;:[0,&quot;Quality control subworkflow for paired-end reads. &quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-taxonomic-summary-tables-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-taxonomic-summary-tables/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;dda70895-1e39-42a3-8b4f-b6493cf15e8f&quot;],&quot;name&quot;:[0,&quot;MGnify amplicon summary tables&quot;],&quot;annotation&quot;:[0,&quot;This workflow creates taxonomic summary tables out of the amplicon pipeline results. &quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-its-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-its/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;04fa2e92-8fcd-4a20-8af1-063c7f514f12&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - ITS&quot;],&quot;annotation&quot;:[0,&quot;Classification and visualization of ITS regions.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ia-import-multiplexed-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ia-import-multiplexed-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;951c418e-f57e-491d-af38-f2c44c6089ce&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ia: multiplexed data (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing single-end multiplexed data (not demultiplexed yet)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ib-import-multiplexed-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ib-import-multiplexed-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b09b5420-bb7c-4f27-9c90-7e2031889379&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ib: multiplexed data (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing paired-end multiplexed data (not demultiplexed yet)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ic-import-demultiplexed-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ic-import-demultiplexed-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5a57f873-52f8-4c51-9e8d-e747d32d183a&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ic: Demultiplexed data (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing demultiplexed data (single-end)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-id-import-demultiplexed-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Id-import-demultiplexed-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f4887024-6f68-4d53-a889-5635366dd18d&quot;],&quot;name&quot;:[0,&quot;QIIME2 Id: Demultiplexed data (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing demultiplexed data (paired-end)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-ii-denoising-iia-denoising-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIa-denoising-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;718df5b9-9cbb-4792-be50-604e18ef0548&quot;],&quot;name&quot;:[0,&quot;QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-ii-denoising-iib-denoising-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIb-denoising-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8b8c4f4c-7cc0-422d-9a6f-47357faa0082&quot;],&quot;name&quot;:[0,&quot;QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-iii-vi-downsteam-qiime2-iii-v-phylogeny-rarefaction-taxonomic-analysis&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3163175d-7673-42d9-8aae-505afbd74946&quot;],&quot;name&quot;:[0,&quot;QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis&quot;],&quot;annotation&quot;:[0,&quot;This workflow \n- Reconstruct phylogeny (insert fragments in a reference)\n- Alpha rarefaction analysis\n- Taxonomic analysis&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-iii-vi-downsteam-qiime2-vi-diversity-metrics-and-estimations&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-VI-diversity-metrics-and-estimations&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;300b5a7d-f763-486a-a8ea-87ea84571f19&quot;],&quot;name&quot;:[0,&quot;QIIME2 VI: Diversity metrics and estimations&quot;],&quot;annotation&quot;:[0,&quot;The first step in hypothesis testing in microbial ecology is typically to look at within- (alpha) and between-sample (beta) diversity. We can calculate diversity metrics, apply appropriate statistical tests, and visualize the data using the q2-diversity plugin.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;dada2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/dada2/main&quot;],&quot;updated&quot;:[0,&quot;2025-02-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;11d717f8-92ef-4dd7-bda7-c193175fece2&quot;],&quot;name&quot;:[0,&quot;dada2 amplicon analysis pipeline - for paired end data&quot;],&quot;annotation&quot;:[0,&quot;dada2 amplicon analysis for paired end data\n\nThe workflow has three main outputs: \n- the sequence table (output of makeSequenceTable)\n- the taxonomy (output of assignTaxonomy)\n- the counts which allow to track the number of sequences in the samples through the steps (output of sequence counts)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:amplicon&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;haploid-variant-calling-wgs-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/haploid-variant-calling-wgs-pe/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d5b4eace-d876-4999-bad3-14c1cb02b0c9&quot;],&quot;name&quot;:[0,&quot;Paired end variant calling in haploid system&quot;],&quot;annotation&quot;:[0,&quot;Workflow for variant analysis against a reference genome in GenBank format&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;generic&quot;],[0,&quot;VeuPath&quot;],[0,&quot;Haploid&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;variation-reporting-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/variation-reporting/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9e94da53-d8ab-4713-8aac-7c6674fa1d39&quot;],&quot;name&quot;:[0,&quot;Generic variation analysis reporting&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes a VCF dataset of variants produced by any of the variant calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;mpvx&quot;],[0,&quot;generic&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;generic-variant-calling-wgs-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/generic-variant-calling-wgs-pe/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;be0d055d-25ac-422a-ae9c-b0efdfb5bf86&quot;],&quot;name&quot;:[0,&quot;Generic variation analysis on WGS PE data&quot;],&quot;annotation&quot;:[0,&quot;Workflow for variant analysis against  a reference genome in GenBank format&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;mpxv&quot;],[0,&quot;generic&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-quantitation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-quantitation/main&quot;],&quot;updated&quot;:[0,&quot;2024-08-14T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e21d954c-bb46-46f3-8081-0095b6b0de1f&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Quantitation&quot;],&quot;annotation&quot;:[0,&quot;Clinical Metaproteomics 4: Quantitation &quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-discovery-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-discovery/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;94138387-134e-4622-96a9-fdaee428ee1b&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Discovery Workflow&quot;],&quot;annotation&quot;:[0,&quot;Workflow for clinical metaproteomics database searching&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-data-interpretation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-data-interpretation/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-19T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3a5d60ec-2b30-47cf-a351-7b85df8af9e3&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Data Interpretation&quot;],&quot;annotation&quot;:[0,&quot;This workflow will perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-verification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-verification/main&quot;],&quot;updated&quot;:[0,&quot;2024-12-16T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;22302833-4954-4414-8725-d8bf74d6ffd0&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Verification Workflow&quot;],&quot;annotation&quot;:[0,&quot;In proteomics research, verifying detected peptides is essential for ensuring data accuracy and biological relevance. This tutorial continues from the clinical metaproteomics discovery workflow, focusing on verifying identified microbial peptides using the PepQuery tool.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-database-generation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-database-generation/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;758f86d1-b81e-438d-9666-bc7dc31b4627&quot;],&quot;name&quot;:[0,&quot;Generate a Clinical Metaproteomics Database&quot;],&quot;annotation&quot;:[0,&quot;The workflow begins with the Database Generation process. The Galaxy-P team has developed a workflow that collects protein sequences from known disease-causing microorganisms to build a comprehensive database. This extensive database is then refined into a smaller, more relevant dataset using the Metanovo tool.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;openms-metaprosip-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/openms-metaprosip/main&quot;],&quot;updated&quot;:[0,&quot;2024-06-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b03da1a7-fc6f-419f-aa2c-431a670f9a96&quot;],&quot;name&quot;:[0,&quot;MetaProSIP OpenMS 2.8&quot;],&quot;annotation&quot;:[0,&quot;Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics &quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pox-virus-amplicon-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pox-virus-amplicon/main&quot;],&quot;updated&quot;:[0,&quot;2025-10-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b7574fe6-4a44-4d3b-93e9-e6d4065d2dfd&quot;],&quot;name&quot;:[0,&quot;Pox Virus Illumina Amplicon Workflow from half-genomes&quot;],&quot;annotation&quot;:[0,&quot;A workflow for the analysis of pox virus genomes sequenced as half-genomes (for ITR resolution) in a tiled-amplicon approach&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;pox&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;influenza-isolates-consensus-and-subtyping-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/influenza-isolates-consensus-and-subtyping/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-02T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;fb161003-8ad7-4a72-8835-bfd2c9088092&quot;],&quot;name&quot;:[0,&quot;Influenza A isolate subtyping and consensus sequence generation&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs subtyping and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;AIV&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;generic-non-segmented-viral-variant-calling-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/generic-non-segmented-viral-variant-calling/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;],[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;967c16f1-0cd3-42c3-a6ad-9b392d597613&quot;],&quot;name&quot;:[0,&quot;Variant calling and consensus construction from paired end short read data of non-segmented viral genomes&quot;],&quot;annotation&quot;:[0,&quot;Variant calling and consensus sequence generation for batches of Illumina PE sequenced viruses with uncomplicated and stable genome structure (like e.g. Morbilliviruses).&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;ARTIC&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mitogenome-assembly-vgp0-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Mitogenome-assembly-VGP0/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7de562f8-2ef4-4277-a88b-c0d12f22f0b6&quot;],&quot;name&quot;:[0,&quot;Mitogenome Assembly VGP0&quot;],&quot;annotation&quot;:[0,&quot;Generate mitochondrial assembly based on PacBio HiFi reads. Part of the VGP suite, it can be run at any time independently of the other workflows. This workflow uses MitoHiFi and a mitochondrial reference to assemble the mitochondrial genome from PacBio reads. You do not need to provide the reference yourself, only the Latin name of the species.&quot;],&quot;release&quot;:[0,&quot;0.2.2&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;purge-duplicate-contigs-vgp6-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Purge-duplicate-contigs-VGP6/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c4a1ca2e-13a0-4d0b-bdec-d5bbac5b9a34&quot;],&quot;name&quot;:[0,&quot;Purge duplicate contigs from a diploid assembly VGP6&quot;],&quot;annotation&quot;:[0,&quot;Purge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication). The contigs are purged from the first assembly (hap1, pri...), added to the second assembly (hp2, alt... ), then the 2nd assembly is purged as well. If you think only one of the assemblies needs purging, use the VGP6b workflow.  This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).&quot;],&quot;release&quot;:[0,&quot;0.10.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;kmer-profiling-hifi-trio-vgp2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/kmer-profiling-hifi-trio-VGP2/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-30T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a8867bbf-2a39-4cf1-b530-eb5650961c78&quot;],&quot;name&quot;:[0,&quot;kmer-profiling-hifi-trio-VGP2&quot;],&quot;annotation&quot;:[0,&quot;Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.&quot;],&quot;release&quot;:[0,&quot;0.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-trio-phasing-vgp5-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-Trio-phasing-VGP5/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9fc9042f-6171-451a-bdd7-010abdb5b683&quot;],&quot;name&quot;:[0,&quot;Genome Assembly with Pacbio Hifi reads and Trio data for phasing - VGP5&quot;],&quot;annotation&quot;:[0,&quot;Generate phased assembly based on PacBio HiFi reads and parental Illumina data for phasing. Part of the VGP workflow suite, it needs to be run after the Trio k-mer Profiling workflow VGP2. This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury plots, and the genome assembly contigs in fasta and GFA format. &quot;],&quot;release&quot;:[0,&quot;0.9.9&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scaffolding-bionano-vgp7-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Scaffolding-Bionano-VGP7/main&quot;],&quot;updated&quot;:[0,&quot;2024-08-13T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9fd3c0de-a497-47dc-96e3-0f3e1998173d&quot;],&quot;name&quot;:[0,&quot;Scaffolding-BioNano-VGP7&quot;],&quot;annotation&quot;:[0,&quot;Scaffold a genome assembly using Bionano optical maps data. Part of the VGP suite, it can be run on the assembly in GFA format generated by the contigging step in workflow 3,4, or 5. If you used a different method to generate your assembly, you can use the tools gfastats in Galaxy to generate a GFA from your assembly fasta, and create a simple text file containing the estimated genome size.&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hi-c-contact-map-for-assembly-manual-curation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hi-c-contact-map-for-assembly-manual-curation/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;34226708-cba3-421c-809f-6ba105b3812c&quot;],&quot;name&quot;:[0,&quot;PretextMap Generation from 1 or 2 haplotypes&quot;],&quot;annotation&quot;:[0,&quot;This workflow generates Hi-C contact maps for genome assemblies in the Pretext format. It is compatible with one or 2 haplotypes. It includes tracks for PacBio read coverage, Gaps, and telomeres. The Pretext files can be open in PretextView for the manual curation of genome assemblies.&quot;],&quot;release&quot;:[0,&quot;1.0beta6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-decontamination-vgp9-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-decontamination-VGP9/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;dc6237a5-4fe1-41bf-bf08-b98852503132&quot;],&quot;name&quot;:[0,&quot;Assembly decontamination VGP9 &quot;],&quot;annotation&quot;:[0,&quot;Decontamination (foreign contaminants and mitochondrial sequences) of a genome assembly after the final scaffolding step. Uses NCBI FCS GX  to identify foreign contaminants and Blast to identify mitochondrial sequences. Part of the VGP Suite.&quot;],&quot;release&quot;:[0,&quot;1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;purge-duplicates-one-haplotype-vgp6b-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Purge-duplicates-one-haplotype-VGP6b/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;882b18ac-4a8c-4cf1-8c63-a85260bc37b9&quot;],&quot;name&quot;:[0,&quot;Purging duplicates in one haplotype VGP6b&quot;],&quot;annotation&quot;:[0,&quot;Purge contigs marked as duplicates by purge_dups in a single haplotype (could be haplotypic duplication or overlap duplication). If you think the purged contigs might belong to the other haplotype, use the workflow VGP6 instead. This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).&quot;],&quot;release&quot;:[0,&quot;0.8.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;kmer-profiling-hifi-vgp1-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/kmer-profiling-hifi-VGP1/main&quot;],&quot;updated&quot;:[0,&quot;2025-10-01T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f76c7eb0-71c7-4697-8fab-011eb7e4bcda&quot;],&quot;name&quot;:[0,&quot;K-mer profiling and reads statistics VGP1&quot;],&quot;annotation&quot;:[0,&quot;Evaluation of Pacbio Hifi Reads and genome profiling. Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.&quot;],&quot;release&quot;:[0,&quot;0.5&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scaffolding-hic-vgp8-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Scaffolding-HiC-VGP8/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-26T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;97db8bb9-1842-439d-b363-2a06b905a098&quot;],&quot;name&quot;:[0,&quot;Scaffolding with Hi-C data VGP8&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs the scaffolding of a genome assembly using HiC data with YAHS. Can be used on any assembly with Hi-C data, and the assembly in the gfa format.&quot;],&quot;release&quot;:[0,&quot;3.1&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;plot-nx-size-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Plot-Nx-Size/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7a7ad2da-596c-40d1-b67d-0b3b6a412c8b&quot;],&quot;name&quot;:[0,&quot;Generate Nx and Size plots for multiple assemblies&quot;],&quot;annotation&quot;:[0,&quot;Generate Nx and size plots for multiple assemblies to compare the evolution of assembly quality through the scaffolding process. Inputs are the fasta files for each assembly to compare.&quot;],&quot;release&quot;:[0,&quot;0.1.8&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-only-vgp3-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-only-VGP3/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;40ef3ef9-6781-42d6-baf0-70fe10cc9622&quot;],&quot;name&quot;:[0,&quot;Genome Assembly from Hifi reads - VGP3&quot;],&quot;annotation&quot;:[0,&quot;Generate a genome assembly based on PacBio HiFi reads. Part of the VGP suite, it needs to be run after the VGP1 k-mer profiling workflow. The assembly contigs are built using HiFiasm, and the workflow generates assembly statistics, BUSCO reports, Merqury plots, and the contigs in fasta and GFA formats.&quot;],&quot;release&quot;:[0,&quot;0.3.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-hic-phasing-vgp4-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-HiC-phasing-VGP4/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-30T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f0ae78ab-8dc1-4807-b9c6-1efd8929ed6f&quot;],&quot;name&quot;:[0,&quot;Genome Assembly from Hifi reads with HiC phasing - VGP4&quot;],&quot;annotation&quot;:[0,&quot;Assemble Genome using PacBio HiFi and HiC data from the same individual for phasing. Prerequisite: Run k-mer profiling workflow (VGP1). This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury histograms, and the genome assembly contigs in fasta and GFA format.&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;annotation-helixer-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/annotation-helixer/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;02f1b56b-f9e2-4da5-9d52-83e1f059fae6&quot;],&quot;name&quot;:[0,&quot;Genome annotation with Helixer&quot;],&quot;annotation&quot;:[0,&quot;This workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. &quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;genomeannotation&quot;],[0,&quot;busco&quot;],[0,&quot;jbrowse&quot;],[0,&quot;helixer&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;annotation-maker-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/annotation-maker/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-06T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;87be9418-f744-42b6-8ac8-01dae7998e4f&quot;],&quot;name&quot;:[0,&quot;Genome annotation with Maker&quot;],&quot;annotation&quot;:[0,&quot;This workflow allows for genome annotation using Maker and evaluates the quality of the annotation.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;maker&quot;],[0,&quot;genomeannotation&quot;],[0,&quot;busco&quot;],[0,&quot;jbrowse&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;functional-annotation-protein-sequences-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/functional-annotation-protein-sequences/main&quot;],&quot;updated&quot;:[0,&quot;2024-12-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;4cbba315-c9bc-4895-aeeb-57dadef3542a&quot;],&quot;name&quot;:[0,&quot;Functional annotation of protein sequences&quot;],&quot;annotation&quot;:[0,&quot;This workflow uses eggNOG mapper and InterProScan for functional annotation of protein sequences.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;lncrnas-annotation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/lncRNAs-annotation/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f6d6b63f-6fcc-42bb-bafb-12d9dd98983e&quot;],&quot;name&quot;:[0,&quot;lncRNAs annotation workflow&quot;],&quot;annotation&quot;:[0,&quot;This workflow runs the FEELnc tool to annotate long non-coding RNAs. Before annotating these long non-coding RNAs, StringTie will be used to assemble the RNA-seq alignments into potential trancriptions. The gffread tool provides a genome annotation file in GTF format.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;quality-and-contamination-control-raw-reads-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/quality-and-contamination-control-raw-reads/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-21T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8467a350-2e23-41f0-bd5c-d767569528c8&quot;],&quot;name&quot;:[0,&quot;Raw Read Quality and Contamination Control For Genome Assembly&quot;],&quot;annotation&quot;:[0,&quot;Short paired-end read analysis to provide quality analysis, read cleaning and taxonomy assignation directly from raw reads&quot;],&quot;release&quot;:[0,&quot;1.1.10&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;fastq&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;taxonomy-assignment&quot;],[0,&quot;paired-end&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;],[0,&quot;trimming&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial-genome-assembly-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial-genome-assembly/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-11T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;14fc2baa-a9fb-4e90-a8ec-f542faf1d8c2&quot;],&quot;name&quot;:[0,&quot;Bacterial Genome Assembly using Shovill&quot;],&quot;annotation&quot;:[0,&quot;Assembly of bacterial paired-end short read data&quot;],&quot;release&quot;:[0,&quot;2.0&quot;],&quot;tags&quot;:[1,[[0,&quot;fastq&quot;],[0,&quot;Genomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;paired-end&quot;],[0,&quot;assembly&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;polish-with-long-reads-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/polish-with-long-reads/main&quot;],&quot;updated&quot;:[0,&quot;2023-07-15T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;68d6270a-bfd1-452a-abc9-4a314e13af85&quot;],&quot;name&quot;:[0,&quot;Assembly polishing with long reads&quot;],&quot;annotation&quot;:[0,&quot;Racon polish with long reads, x4&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-with-flye-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/assembly-with-flye/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-19T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;21643946-a984-4a3f-be71-d480590522c3&quot;],&quot;name&quot;:[0,&quot;Genome assembly with Flye&quot;],&quot;annotation&quot;:[0,&quot;Assemble long reads with Flye, then view assembly statistics and assembly graph&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-cellplex&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-cellplex&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c54841b3-902d-4949-93ca-f6882c12c4dd&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Preprocessing: 10X Genomics CellPlex Multiplexed Samples&quot;],&quot;annotation&quot;:[0,&quot;Comprehensive preprocessing for 10X Genomics CellPlex multiplexed single-cell RNA-seq data. Processes Cell Multiplexing Oligo (CMO) FASTQ files with CITE-seq-Count including required CellPlex-specific translation steps. Simultaneously processes gene expression FASTQ files with STARsolo and DropletUtils for alignment and cell filtering, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).&quot;],&quot;release&quot;:[0,&quot;0.6.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-v3&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-v3&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;677cbe1a-f944-4c6f-8a7e-118030b1d567&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat and Scanpy Compatible Format&quot;],&quot;annotation&quot;:[0,&quot;Complete preprocessing pipeline for 10X Genomics v3 single-cell RNA-seq data. Aligns raw FASTQ files using STARsolo, performs cell calling and quality filtering with DropletUtils, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).&quot;],&quot;release&quot;:[0,&quot;0.6.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;baredsc-baredsc-1d-lognorm&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/baredsc/baredSC-1d-logNorm&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;1ea0b92d-d863-435a-96ac-1e547fdbc994&quot;],&quot;name&quot;:[0,&quot;Single-Cell Mixture Analysis: baredSC 1D Log-Normalized Models&quot;],&quot;annotation&quot;:[0,&quot;Applies baredSC algorithm to fit and combine one-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on expression of genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.&quot;],&quot;release&quot;:[0,&quot;0.6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;baredsc-baredsc-2d-lognorm&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/baredsc/baredSC-2d-logNorm&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;1936816b-958a-4140-b839-58be4cdae1a5&quot;],&quot;name&quot;:[0,&quot;Single-Cell Mixture Analysis: baredSC 2D Log-Normalized Models&quot;],&quot;annotation&quot;:[0,&quot;Applies baredSC algorithm to fit and combine two-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on co-expression patterns of two genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.&quot;],&quot;release&quot;:[0,&quot;0.6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pseudobulk-worflow-decoupler-edger-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pseudobulk-worflow-decoupler-edger/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e4e4055c-7e3f-4014-a8e6-343199fd9b01&quot;],&quot;name&quot;:[0,&quot;Single-Cell Pseudobulk Differential Expression Analysis with edgeR&quot;],&quot;annotation&quot;:[0,&quot;Performs differential gene expression analysis on single-cell data using a pseudobulk approach. Aggregates cell-level counts from an annotated AnnData object by cell type or other metadata using Decoupler, then applies edgeR for robust statistical testing between conditions. Includes data preprocessing steps for compatibility with edgeR and generates interactive volcano plots to visualize significantly differentially expressed genes.&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scanpy-clustering-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/scanpy-clustering/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d6b2d1bb-6aaf-47cc-8956-5ae80b775de5&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering&quot;],&quot;annotation&quot;:[0,&quot;End-to-end analysis of single-cell RNA-seq data using the Scanpy/AnnData ecosystem. Imports count matrices, applies quality control filtering of low-quality cells and genes, normalizes and scales data, performs dimension reduction (PCA), and identifies cell clusters using the Louvain algorithm. Generates publication-ready visualizations including UMAP plots colored by clusters and marker genes.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:scRNA-seq&quot;],[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;velocyto-velocyto-on10x-from-bundled&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-from-bundled&quot;],&quot;updated&quot;:[0,&quot;2024-02-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;0608c7f4-767c-47b3-ae45-4e98c7c83bec&quot;],&quot;name&quot;:[0,&quot;RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output&quot;],&quot;annotation&quot;:[0,&quot;Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Automatically extracts cell barcodes from standard 10X bundled output and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;velocyto-velocyto-on10x-filtered-barcodes&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-filtered-barcodes&quot;],&quot;updated&quot;:[0,&quot;2024-02-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;33862923-af05-48ba-aec5-14393981cee2&quot;],&quot;name&quot;:[0,&quot;RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes&quot;],&quot;annotation&quot;:[0,&quot;Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Takes pre-filtered cell barcodes as input and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis in tools like scVelo.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}]]]}" ssr client="load" opts="{&quot;name&quot;:&quot;PopularWorkflows&quot;,&quot;value&quot;:true}" await-children><div class="w-full max-w-6xl mx-auto"><h2 class="text-xl my-6 text-white text-center font-semibold"> Get started with one of our most popular workflows, or browse the full library below. </h2><div class="grid grid-cols-3 gap-4 mx-auto px-4"><!--[--><div class="rounded-lg border border-gray-200 text-gray-950 shadow flex flex-col h-[18rem] hover:shadow-lg transition-shadow bg-white" id="workflow-5ab1d724-01d4-4673-a3c3-ae2bc944458f"><!--[--><div class="flex flex-col space-y-1.5 p-3"><!--[--><a href="/workflow/rnaseq-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-lg">RNA-Seq Analysis: Paired-End Read Processing and Quantification</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-3 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow text-sm line-clamp-6 mb-2">Complete RNA-Seq analysis for paired-end data: Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.</p><div class="flex flex-wrap gap-1"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 text-xs cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--><!----></div></div><!--]--></div><div class="p-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 text-gray-950 shadow flex flex-col h-[18rem] hover:shadow-lg transition-shadow bg-white" id="workflow-25c8f8e3-7ed9-4523-8862-8496fa1777cd"><!--[--><div class="flex flex-col space-y-1.5 p-3"><!--[--><a href="/workflow/chipseq-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-lg">ChIP-seq Analysis: Paired-End Read Processing</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-3 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow text-sm line-clamp-6 mb-2">Complete ChIP-seq analysis for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and stringent quality filtering (MAPQ &amp;gt;= 30, concordant pairs only). Peak calling with MACS2 optimized for paired-end reads identifies protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.</p><div class="flex flex-wrap gap-1"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 text-xs cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--><!----></div></div><!--]--></div><div class="p-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.15</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 text-gray-950 shadow flex flex-col h-[18rem] hover:shadow-lg transition-shadow bg-white" id="workflow-747f2432-7493-44b8-a5ee-82d104407cad"><!--[--><div class="flex flex-col space-y-1.5 p-3"><!--[--><a href="/workflow/atacseq-main/"><h2 class="font-bold hover:underline text-ebony-clay text-lg">ATAC-seq Analysis: Chromatin Accessibility Profiling</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-3 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow text-sm line-clamp-6 mb-2">Complete ATAC-seq analysis pipeline for paired-end reads. Processes raw FASTQ data through adapter and bad quality removal (cutadapt), alignment (Bowtie2 end-to-end), and filtering (removes MT reads, discordant pairs, low mapping quality &lt;30, PCR duplicates). Generates 5&#39; cut site pileups (±100bp), performs peak calling, and quantifies reads in 1kb summit-centered regions. Produces two normalized coverage tracks (per million mapped reads and per million reads in peaks) and fragment length distribution plots for quality assessment.</p><div class="flex flex-wrap gap-1"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 text-xs cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--><!----></div></div><!--]--></div><div class="p-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.0</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 28, 2024</p></div></div><!--]--></div><!--]--></div></div><!--astro:end--></astro-island> <div class="max-w-6xl w-full mt-12 p-4"> <input type="text" id="search-input" placeholder="Search workflows" class="w-full p-2 mb-2 border border-gray-300 bg-white rounded-lg"> </div> <astro-island uid="1grJoz" prefix="s2" component-url="/_astro/Filters.DipEqHfD.js" component-export="default" renderer-url="/_astro/client.BrhupUx7.js" props="{&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Data Fetching&quot;],[0,&quot;Epigenetics&quot;],[0,&quot;Genome Annotation&quot;],[0,&quot;Genome assembly&quot;],[0,&quot;Imaging&quot;],[0,&quot;Metabarcoding&quot;],[0,&quot;Metabolomics&quot;],[0,&quot;Metaproteomics&quot;],[0,&quot;Microbiome&quot;],[0,&quot;Proteomics&quot;],[0,&quot;SARS-COV-2&quot;],[0,&quot;Single Cell&quot;],[0,&quot;Transcriptomics&quot;],[0,&quot;Variant Calling&quot;],[0,&quot;Vertebrate Genome Project&quot;],[0,&quot;Virology&quot;]]]}" ssr client="load" opts="{&quot;name&quot;:&quot;Filters&quot;,&quot;value&quot;:true}" await-children><div id="filters" class="w-full" data-v-761c6766><div class="flex flex-wrap justify-center items-center mb-2" data-v-761c6766><!--[--><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Computational Chemistry</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Data Fetching</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Epigenetics</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Genome Annotation</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Genome assembly</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Imaging</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Metabarcoding</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Metabolomics</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Metaproteomics</span><!----></div><!--]--></div><div class="flex flex-wrap justify-center items-center" data-v-761c6766><!--[--><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Microbiome</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Proteomics</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>SARS-COV-2</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Single Cell</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Transcriptomics</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Variant Calling</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Vertebrate Genome Project</span><span class="divider text-gray-200 mx-1" data-v-761c6766>|</span></div><div class="filter-wrapper flex items-center" data-v-761c6766><span class="text-gold border-transparent hover:bg-gold hover:text-white hover:bg-opacity-5 py-1 px-2 cursor-pointer transition-all duration-200 ease-in-out rounded border-b-2 border-t-2 whitespace-nowrap" data-v-761c6766>Virology</span><!----></div><!--]--></div></div><!--astro:end--></astro-island> </div> </div> <div class="flex"> <!-- Left sidebar if provided -->  <!-- Main content --> <div class="flex-1 p-4 overflow-y-auto"> <div> <div id="category-description" class="w-full my-4 p-4 bg-white rounded-lg shadow-md" style="display:none;"> <h2 id="category-title" class="text-xl font-semibold mb-4"></h2> <div id="category-content" class="prose !max-w-none"></div> </div> <astro-island uid="J9iSt" prefix="s0" component-url="/_astro/WorkflowGrid.C--Hk2G1.js" component-export="default" renderer-url="/_astro/client.BrhupUx7.js" props="{&quot;workflows&quot;:[1,[[0,{&quot;iwcID&quot;:[0,&quot;protein-ligand-complex-parameterization-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/protein-ligand-complex-parameterization/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f1aea424-b536-496f-a0e6-fe46df8e6e18&quot;],&quot;name&quot;:[0,&quot;Create GRO and TOP complex files&quot;],&quot;annotation&quot;:[0,&quot;&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fragment-based-docking-scoring-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fragment-based-docking-scoring/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;bb767af6-3641-4a5e-843d-78508fd7577a&quot;],&quot;name&quot;:[0,&quot;Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring&quot;],&quot;annotation&quot;:[0,&quot;Virtual screening of the SARS-CoV-2 main protease with rDock and pose scoring&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gromacs-mmgbsa-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gromacs-mmgbsa/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;16455667-cdef-4979-9603-f338e09aa2a1&quot;],&quot;name&quot;:[0,&quot;MMGBSA calculations with GROMACS&quot;],&quot;annotation&quot;:[0,&quot;MMGBSA simulation and calculation&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gromacs-dctmd-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gromacs-dctmd/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7950b716-2bc9-4cb6-8472-6506c702adea&quot;],&quot;name&quot;:[0,&quot;dcTMD calculations with GROMACS&quot;],&quot;annotation&quot;:[0,&quot;Perform dcTMD free energy simulations and calculations&quot;],&quot;release&quot;:[0,&quot;0.1.5&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial-quality-and-contamination-control-post-assembly-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial-quality-and-contamination-control-post-assembly/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8911f5cd-67b1-4d57-be62-abe84ca07f23&quot;],&quot;name&quot;:[0,&quot;Post-Assembly Quality Control and Contamination Check for Bacterial Genomes&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs quality and contamination control analysis on assembled contigs to assess bacterial genome quality and taxonomic assignment&quot;],&quot;release&quot;:[0,&quot;1.0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;taxonomy-assignment&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;],[0,&quot;fasta&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;amr_gene_detection-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/amr_gene_detection/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-11T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;284cc748-7532-4749-8f68-d19d41c048ef&quot;],&quot;name&quot;:[0,&quot;amr_gene_detection&quot;],&quot;annotation&quot;:[0,&quot;Antimicrobial resistance gene detection from assembled bacterial genomes&quot;],&quot;release&quot;:[0,&quot;1.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;fasta&quot;],[0,&quot;Genomics&quot;],[0,&quot;ABRomics&quot;],[0,&quot;antibiotic-resistance&quot;],[0,&quot;antimicrobial-resistance-genes&quot;],[0,&quot;antimicrobialresistance&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;AMR&quot;],[0,&quot;AMR-detection&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial_genome_annotation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial_genome_annotation/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;47a7aa1d-8cb6-4eab-a6d8-86909bca0d43&quot;],&quot;name&quot;:[0,&quot;bacterial_genome_annotation&quot;],&quot;annotation&quot;:[0,&quot;Annotation of an assembled bacterial genomes to detect genes, potential plasmids, integrons and Insertion sequence (IS) elements.&quot;],&quot;release&quot;:[0,&quot;1.1.11&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;fasta&quot;],[0,&quot;ABRomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;Annotation&quot;],[0,&quot;genome-annotation&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;repeatmasking-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/repeatmasking/main&quot;],&quot;updated&quot;:[0,&quot;2023-09-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f25be8fa-7823-456f-9707-a497703f48d7&quot;],&quot;name&quot;:[0,&quot;Repeat masking with RepeatModeler and RepeatMasker&quot;],&quot;annotation&quot;:[0,&quot;&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;parallel-accession-download-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/parallel-accession-download/main&quot;],&quot;updated&quot;:[0,&quot;2024-05-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Data Fetching&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;15c3229d-8786-457f-a020-7aa64c0cf6e8&quot;],&quot;name&quot;:[0,&quot;Parallel Accession Download&quot;],&quot;annotation&quot;:[0,&quot;Downloads fastq files for sequencing run accessions provided in a text file using fasterq-dump. Creates one job per listed run accession.&quot;],&quot;release&quot;:[0,&quot;0.1.14&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sra-manifest-to-concatenated-fastqs-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sra-manifest-to-concatenated-fastqs/main&quot;],&quot;updated&quot;:[0,&quot;2024-06-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Data Fetching&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;363898af-e598-4f0e-abd9-e6ded395ce66&quot;],&quot;name&quot;:[0,&quot;sra_manifest_to_concatenated_fastqs_parallel&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes as input a SRA_manifest from SRA Run Selector and will generate one fastq file or fastq pair of file for each experiment (concatenated multiple runs if necessary). Output will be relabelled to match the column specified by the user.&quot;],&quot;release&quot;:[0,&quot;0.7&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mfassignr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mfassignr/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-16T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3be0019d-8058-4e57-922a-417b9e3578a2&quot;],&quot;name&quot;:[0,&quot;Molecular formula assignment and recalibration with MFAssignR package.&quot;],&quot;annotation&quot;:[0,&quot;This workflow can be used to assign multi-element molecular formulas to ultrahigh resolution mass spectra.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;recalibration&quot;],[0,&quot;noiseestimation&quot;],[0,&quot;formulaassignment&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;lcms-preprocessing-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/lcms-preprocessing/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;af2457b4-9013-4d8c-9e76-9948c1f96fcc&quot;],&quot;name&quot;:[0,&quot;Mass spectrometry: LC-MS preprocessing with XCMS&quot;],&quot;annotation&quot;:[0,&quot;This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package (Kuhl, C 2012).\n\n\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html &quot;],&quot;release&quot;:[0,&quot;1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;MS&quot;],[0,&quot;LC-MS&quot;],[0,&quot;workflow4metabolomics&quot;],[0,&quot;xcms&quot;],[0,&quot;GTN&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qcxms-sdf-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qcxms-sdf/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;067945f1-8716-4558-918e-4a9c9d42f604&quot;],&quot;name&quot;:[0,&quot;QCxMS Spectra Prediction from SDF&quot;],&quot;annotation&quot;:[0,&quot;Workflow to predict EI mass spectra using QCxMS starting from a single SDF file, containing the 3D coordinates of all atoms in the molecule. These files can typically be obtained from PubChem.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gcms-metams-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gcms-metams/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Computational Chemistry&quot;],[0,&quot;Metabolomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7abd10fb-d083-4732-b11f-56581e23905d&quot;],&quot;name&quot;:[0,&quot;Mass spectrometry: GCMS with metaMS&quot;],&quot;annotation&quot;:[0,&quot;This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract and the metaMS R package (Wehrens, R 2014) for the field of untargeted metabolomics. \n\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;metabolomics&quot;],[0,&quot;MS&quot;],[0,&quot;workflow4metabolomics&quot;],[0,&quot;GC-MS&quot;],[0,&quot;GTN&quot;],[0,&quot;metaMS&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mags-building-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mags-building/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e4e70fc0-b584-4e86-8b8f-c6bcdda2e11a&quot;],&quot;name&quot;:[0,&quot;Metagenome-Assembled Genomes (MAGs) generation&quot;],&quot;annotation&quot;:[0,&quot;This workflow constructs Metagenome-Assembled Genomes (MAGs) using SPAdes or MEGAHIT as assemblers, followed by binning with four different tools and refinement using Binette. The resulting MAGs are dereplicated across the entire input sample set, then annotated and evaluated for quality.\nYou can provide pooled reads (for co-assembly/binning), individual read sets, or a combination of both. The input samples must consist of the original reads, which are used for abundance estimation. In all cases, reads should be trimmed, adapters removed, and cleaned of host or other contaminants before processing.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:FAIRyMAGs&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;gene-based-pathogen-identification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/gene-based-pathogen-identification/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d6795753-1562-4ccc-8f9c-f598acd5404e&quot;],&quot;name&quot;:[0,&quot;Gene-based Pathogen Identification&quot;],&quot;annotation&quot;:[0,&quot;Nanopore datasets analysis - Phylogenetic Identification - antibiotic resistance genes detection and contigs building&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;],[0,&quot;name:microGalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;taxonomy-profiling-and-visualization-with-krona-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/taxonomy-profiling-and-visualization-with-krona/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a55428c8-9934-4941-93c4-3920b4ba3366&quot;],&quot;name&quot;:[0,&quot;Taxonomy Profiling and Visualization with Krona&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - Taxonomy Profiling&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;nanopore-pre-processing-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/nanopore-pre-processing/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;0b976704-e1e0-4cc3-9e75-38bda67c01ff&quot;],&quot;name&quot;:[0,&quot;Nanopore Preprocessing&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - QC and Contamination Filtering&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:Nanopore&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pathogen-detection-pathogfair-samples-aggregation-and-visualisation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pathogen-detection-pathogfair-samples-aggregation-and-visualisation/main&quot;],&quot;updated&quot;:[0,&quot;2024-04-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;cb26018e-9d2f-4513-a372-122474df703e&quot;],&quot;name&quot;:[0,&quot;Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation&quot;],&quot;annotation&quot;:[0,&quot;Pathogens of all samples report generation and visualization&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;allele-based-pathogen-identification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/allele-based-pathogen-identification/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;435875a1-72c4-4fda-8943-3a78f25c6574&quot;],&quot;name&quot;:[0,&quot;Allele-based Pathogen Identification&quot;],&quot;annotation&quot;:[0,&quot;Microbiome - Variant calling and Consensus Building&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[[0,&quot;name:Collection&quot;],[0,&quot;name:microGalaxy&quot;],[0,&quot;name:PathoGFAIR&quot;],[0,&quot;name:IWC&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-artic-ivar-analysis-sars-cov-2-illumina-amplicon-ivar-pangolin-nextclade&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-ivar-analysis/SARS-COV-2-ILLUMINA-AMPLICON-IVAR-PANGOLIN-NEXTCLADE&quot;],&quot;updated&quot;:[0,&quot;2025-04-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3d86e5fc-cad1-4e21-8ecc-d0e6637899bc&quot;],&quot;name&quot;:[0,&quot;SARS-CoV-2 Illumina Amplicon pipeline - iVar based&quot;],&quot;annotation&quot;:[0,&quot;Find and annotate variants in ampliconic SARS-CoV-2 Illumina sequencing data and classify samples with pangolin and nextclade&quot;],&quot;release&quot;:[0,&quot;0.3.1&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;iwc&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-consensus-from-variation-covid-19-consensus-construction&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-consensus-from-variation/COVID-19-CONSENSUS-CONSTRUCTION&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;06dc40a7-99f2-4b3c-ae21-b3dcb239306f&quot;],&quot;name&quot;:[0,&quot;COVID-19: consensus construction&quot;],&quot;annotation&quot;:[0,&quot;Build a consensus sequence from FILTER PASS variants with intrasample allele-frequency above a configurable consensus threshold.\nHard-mask regions with low coverage (but not consensus variants within them) and ambiguous sites.&quot;],&quot;release&quot;:[0,&quot;0.4.3&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-artic-variant-calling-covid-19-pe-artic-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2025-03-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;864427b0-d25a-499c-b437-f83f2d6b83f2&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on ARTIC PE data&quot;],&quot;annotation&quot;:[0,&quot;The workflow for Illumina-sequenced ARTIC data builds on the RNASeq workflow for paired-end data using the same steps for mapping and variant calling, but adds extra logic for trimming ARTIC primer sequences off reads with the ivar package. In addition, this workflow uses ivar also to identify amplicons affected by ARTIC primer-binding site mutations and tries to exclude reads derived from such tainted amplicons when calculating allele-frequencies of other variants.&quot;],&quot;release&quot;:[0,&quot;0.5.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-variation-reporting-covid-19-variation-reporting&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-variation-reporting/COVID-19-VARIATION-REPORTING&quot;],&quot;updated&quot;:[0,&quot;2024-09-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b08c744d-7c61-4b58-ac5f-4b5886c3c643&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis reporting&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes a VCF dataset of variants produced by any of the *-variant-calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.&quot;],&quot;release&quot;:[0,&quot;0.3.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-se-illumina-wgs-variant-calling-covid-19-se-wgs-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-se-illumina-wgs-variant-calling/COVID-19-SE-WGS-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5171bc59-590c-4e97-8db8-004f73928627&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on WGS SE data&quot;],&quot;annotation&quot;:[0,&quot;This workflows performs single end read mapping with bowtie2 followed by sensitive variant calling across a wide range of AFs with lofreq&quot;],&quot;release&quot;:[0,&quot;0.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-pe-illumina-wgs-variant-calling-covid-19-pe-wgs-illumina&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-wgs-variant-calling/COVID-19-PE-WGS-ILLUMINA&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;6695eb16-30a5-42ea-8591-56f2630cf8bf&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis on WGS PE data&quot;],&quot;annotation&quot;:[0,&quot;This workflows performs paired end read mapping with bwa-mem followed by sensitive variant calling across a wide range of AFs with lofreq&quot;],&quot;release&quot;:[0,&quot;0.2.4&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;covid19.galaxyproject.org&quot;],[0,&quot;iwc&quot;],[0,&quot;emergen_validated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;sars-cov-2-ont-artic-variant-calling-covid-19-artic-ont&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/sars-cov-2-ont-artic-variant-calling/COVID-19-ARTIC-ONT&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;SARS-COV-2&quot;],[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a1ceaf35-f3d7-452d-b5ed-53a4f429d1cf&quot;],&quot;name&quot;:[0,&quot;COVID-19: variation analysis of ARTIC ONT data&quot;],&quot;annotation&quot;:[0,&quot;This workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.&quot;],&quot;release&quot;:[0,&quot;0.3.2&quot;],&quot;tags&quot;:[1,[[0,&quot;COVID-19&quot;],[0,&quot;ARTIC&quot;],[0,&quot;ONT&quot;],[0,&quot;covid19.galaxyproject.org&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;brew3r-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/brew3r/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-07T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;ac8ef6cb-1621-4b73-9362-8a460ef3c3a2&quot;],&quot;name&quot;:[0,&quot;BREW3R&quot;],&quot;annotation&quot;:[0,&quot;Extends 3&#39; ends of gene annotations using BAM files (from STAR alignments) and a reference GTF. Specifically designed for 3&#39;-biased sequencing techniques like 10X scRNA-seq or BRB-seq that primarily capture transcript 3&#39; ends. The BREW3R tool enhances annotations by using evidence from RNA-seq data to improve 3&#39; UTR definitions, which is particularly important for accurate quantification in single-cell and bulk RNA-seq experiments.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;goseq-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/goseq/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-03T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;84a63bc6-a96a-4263-a1e5-bc140f412e42&quot;],&quot;name&quot;:[0,&quot;Gene Ontology and KEGG Pathway Enrichment Analysis&quot;],&quot;annotation&quot;:[0,&quot;Performs functional enrichment analysis of gene sets using GOseq, identifying over-represented Gene Ontology terms and KEGG pathways. This workflow accounts for gene length bias in RNA-seq data when calculating enrichment statistics. Input requires differentially expressed genes and gene lengths. Generates comprehensive tables and visualizations of enriched GO terms across all three ontologies (Biological Process, Molecular Function, Cellular Component) as well as KEGG pathway enrichment results.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-de-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-de/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-24T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;15d188a7-b028-4cf7-bb5d-13858e1f6e78&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Differential Expression Analysis with Visualization&quot;],&quot;annotation&quot;:[0,&quot;Identifies differentially expressed genes between exactly two experimental conditions from count tables. The workflow performs statistical testing, applies filters based on adjusted p-value and log2 fold change thresholds, and generates publication-quality visualizations including volcano plots, MA plots, and heatmaps. Takes two collections of count tables as input and produces filtered gene lists and interactive plots for interpreting expression differences. Optimal for simple two-condition experimental designs.&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;transcriptomics&quot;],[0,&quot;RNAseq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-pe/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5ab1d724-01d4-4673-a3c3-ae2bc944458f&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Analysis: Paired-End Read Processing and Quantification&quot;],&quot;annotation&quot;:[0,&quot;Complete RNA-Seq analysis for paired-end data: Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;1.2&quot;],&quot;tags&quot;:[1,[[0,&quot;RNAseq&quot;],[0,&quot;transcriptomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;rnaseq-sr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/rnaseq-sr/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;885ec907-9f61-4a11-9d23-2243389e7335&quot;],&quot;name&quot;:[0,&quot;RNA-Seq Analysis: Single-End Read Processing and Quantification&quot;],&quot;annotation&quot;:[0,&quot;Complete RNA-Seq analysis for single-end data:  Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;1.2&quot;],&quot;tags&quot;:[1,[[0,&quot;RNAseq&quot;],[0,&quot;transcriptomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;atacseq-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/atacseq/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-28T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;747f2432-7493-44b8-a5ee-82d104407cad&quot;],&quot;name&quot;:[0,&quot;ATAC-seq Analysis: Chromatin Accessibility Profiling&quot;],&quot;annotation&quot;:[0,&quot;Complete ATAC-seq analysis pipeline for paired-end reads. Processes raw FASTQ data through adapter and bad quality removal (cutadapt), alignment (Bowtie2 end-to-end), and filtering (removes MT reads, discordant pairs, low mapping quality &lt;30, PCR duplicates). Generates 5&#39; cut site pileups (±100bp), performs peak calling, and quantifies reads in 1kb summit-centered regions. Produces two normalized coverage tracks (per million mapped reads and per million reads in peaks) and fragment length distribution plots for quality assessment.&quot;],&quot;release&quot;:[0,&quot;1.0&quot;],&quot;tags&quot;:[1,[[0,&quot;ATACseq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-chip-sr&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-sr&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f71eee25-8402-41da-8284-5298656ab379&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ChIP-seq Single-End Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ChIP-seq single-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-chip-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-pe&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a2f66906-e9e3-42f9-8629-390017ed2db0&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ChIP-seq Paired-End Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ChIP-seq paired-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;consensus-peaks-consensus-peaks-atac-cutandrun&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-atac-cutandrun&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f58371cd-2e2d-41da-b1ca-ef10f7330047&quot;],&quot;name&quot;:[0,&quot;Consensus Peak Calling for ATAC-seq and CUT&amp;RUN Replicates&quot;],&quot;annotation&quot;:[0,&quot;Identifies high-confidence consensus peaks from ATAC-seq or CUT&amp;RUN replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.&quot;],&quot;release&quot;:[0,&quot;1.3&quot;],&quot;tags&quot;:[1,[[0,&quot;ATAC-seq&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-fastq-to-cool-hicup-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-cool-hicup-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;2196fbff-317e-437d-a022-696a1f95a850&quot;],&quot;name&quot;:[0,&quot;Hi-C Processing: FASTQ to Balanced Cool Files&quot;],&quot;annotation&quot;:[0,&quot;Comprehensive Hi-C data processing workflow that converts paired-end FASTQ files to balanced contact matrices. Uses HiCUP for pre-processing (adapter trimming, mapping, filtering) with fragment midpoint coordinate assignment, filters reads by mapping quality, and generates sorted tabix files. The final output is a balanced cool file at user-specified resolution created with cooler, ready for multi-resolution contact matrix analysis.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-chic-fastq-to-cool-hicup-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/chic-fastq-to-cool-hicup-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;2a89cbe4-a69f-4bff-9465-2e04b0217e35&quot;],&quot;name&quot;:[0,&quot;Capture Hi-C Processing: FASTQ to Balanced Cool Files&quot;],&quot;annotation&quot;:[0,&quot;End-to-end processing of Capture Hi-C paired-end sequencing data. Transforms raw FASTQ files into balanced contact matrices using HiCUP for processing Hi-C reads (adapter trimming, mapping, filtering for valid pairs) with additional filtering for MAPQ and captured regions. The workflow generates sorted tabix files via cooler and produces balanced cool files at user-specified resolution for downstream analysis and visualization.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-juicermediumtabix-to-cool-cooler&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-juicermediumtabix-to-cool-cooler&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;48733b55-4cde-45d4-a69e-78faff0db3f5&quot;],&quot;name&quot;:[0,&quot;Hi-C Format Conversion: Juicer Medium to Cooler Files&quot;],&quot;annotation&quot;:[0,&quot;Converts Hi-C interaction data from Juicer medium format (tabix files) to balanced cooler format. This streamlined workflow takes a collection of Juicer medium tabix files and a reference genome name as input and generates balanced cool files at user-specified resolution. Enables seamless transition between different Hi-C analysis ecosystems while maintaining data integrity.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hic-hicup-cooler-hic-fastq-to-pairs-hicup&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-pairs-hicup&quot;],&quot;updated&quot;:[0,&quot;2023-09-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;48f579ea-eaac-49ee-b38f-93c82f8b7d76&quot;],&quot;name&quot;:[0,&quot;Hi-C Data Processing: FASTQ to Valid Interaction Pairs&quot;],&quot;annotation&quot;:[0,&quot;Processes Hi-C paired-end FASTQ files to generate validated interaction pairs using HiCUP. The workflow truncates reads at ligation junctions, maps to reference genome, assigns to restriction fragments, and filters out experimental artifacts (self-ligated, dangling ends, internal fragments, and size outliers). Removes duplicates and converts outputs to formats compatible with Juicebox/cooler using fragment midpoints as coordinates. Final filtering by mapping quality ensures high-confidence interaction data.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;Hi-C&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;chipseq-sr-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/chipseq-sr/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b57e88a9-fe46-4995-812c-16c28efeac21&quot;],&quot;name&quot;:[0,&quot;ChIP-seq Analysis: Single-End Read Processing&quot;],&quot;annotation&quot;:[0,&quot;Complete ChIP-seq analysis for single-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and quality filtering (MAPQ &amp;gt;= 30). Peak calling with MACS2 uses either a fixed extension parameter or built-in model to identify protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;0.15&quot;],&quot;tags&quot;:[1,[[0,&quot;ChIP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;cutandrun-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/cutandrun/main&quot;],&quot;updated&quot;:[0,&quot;2025-01-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;98e9e34b-b203-43db-9a23-5fa72963fa6d&quot;],&quot;name&quot;:[0,&quot;CUT&amp;RUN/CUT&amp;TAG Analysis: Protein-DNA Interaction Mapping&quot;],&quot;annotation&quot;:[0,&quot;Complete CUT&amp;RUN/CUT&amp;TAG analysis workflow for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt) and alignment (Bowtie2 with dovetail option enabled). Applies quality filtering (MAPQ ≥ 30, concordant pairs only), converts BAM to BED format, and performs peak calling using MACS2 with parameters optimized for the punctate signal profile characteristic of CUT&amp;RUN/CUT&amp;TAG experiments.&quot;],&quot;release&quot;:[0,&quot;0.14&quot;],&quot;tags&quot;:[1,[[0,&quot;CUTnRUN&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;chipseq-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/chipseq-pe/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;25c8f8e3-7ed9-4523-8862-8496fa1777cd&quot;],&quot;name&quot;:[0,&quot;ChIP-seq Analysis: Paired-End Read Processing&quot;],&quot;annotation&quot;:[0,&quot;Complete ChIP-seq analysis for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and stringent quality filtering (MAPQ &amp;gt;= 30, concordant pairs only). Peak calling with MACS2 optimized for paired-end reads identifies protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.&quot;],&quot;release&quot;:[0,&quot;0.15&quot;],&quot;tags&quot;:[1,[[0,&quot;ChIP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;average-bigwig-between-replicates-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/average-bigwig-between-replicates/main&quot;],&quot;updated&quot;:[0,&quot;2023-09-27T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Epigenetics&quot;],[0,&quot;Transcriptomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c6ed4b0e-d719-4997-9860-ba99488ec332&quot;],&quot;name&quot;:[0,&quot;BigWig Replicates Averaging Workflow&quot;],&quot;annotation&quot;:[0,&quot;Calculates average signal values across replicate BigWig files to generate a single consolidated track. Requires input file identifiers formatted as \&quot;sample_name_replicateID\&quot;. The tool automatically groups files by sample name, computes the mean signal at each position, and outputs consolidated BigWig files named after the common sample prefix (without the replicate ID suffix).&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fluorescence-nuclei-segmentation-and-counting-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fluorescence-nuclei-segmentation-and-counting/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-07T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8c2d765b-2764-4b66-8542-adc2109c24f2&quot;],&quot;name&quot;:[0,&quot;Segmentation and counting of cell nuclei in fluorescence microscopy images&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;tissue-microarray-analysis-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/tissue-microarray-analysis/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-21T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c880e149-6237-46b6-aac6-7e1e3bb58ede&quot;],&quot;name&quot;:[0,&quot;End-to-End Tissue Microarray Analysis&quot;],&quot;annotation&quot;:[0,&quot;Complete multiplex tissue image (MTI) analysis pipeline for tissue microarray (TMA) data imaged using cyclic immunofluorescence: Performs illumination correction, stitching and registration, and tissue microarray segmentation. Tissue-segmented images undergo nuclear segmentation, cell/nuclei feature quantification (mean marker intensities, cell coordinates, and morphological features), and cell phenotyping. Produces outputs that are compatible with downstream single-cell/spatial analysis and interactive image viewers including: Pyramidal OME-TIFF images, nuclear segmentation masks (TIFF), quantified feature tables (CSV, h5ad) with cell type annotations, and an interactive Vitessce dashboard that combines image viewing with linked single-cell data visualizations. &quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;imaging&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;multiplex-tissue-microarray-analysis-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/multiplex-tissue-microarray-analysis/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-22T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Imaging&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;eeedc263-2a29-4651-9b2b-007bf5d3cf4b&quot;],&quot;name&quot;:[0,&quot;Multiplex Tissue Microarray Analysis&quot;],&quot;annotation&quot;:[0,&quot;Perform background subtraction, nuclear segmentation, feature quantification, cellular phenotyping, spatial analysis, and interactive visualization of registered TMA core multiplex tissue images&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-quality-control-single-end-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-single-end/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;68bf2a73-d68a-4e25-ba52-5586c057df77&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - Quality control SE&quot;],&quot;annotation&quot;:[0,&quot;Quality control subworkflow for single-end reads.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-complete-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-complete/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;fad227f2-8b1e-4782-a7a5-a1743b9643bf&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0&quot;],&quot;annotation&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0. Including the Quality control for single-end and paired-end reads, rRNA-prediction, and ITS sub-WFs.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;amplicon&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mapseq-to-ampvis2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mapseq-to-ampvis2/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e24060a0-1cfe-40f1-9251-48e96f7e1121&quot;],&quot;name&quot;:[0,&quot;MAPseq to ampvis2&quot;],&quot;annotation&quot;:[0,&quot;The MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;taxonomic-rank-abundance-summary-table-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/taxonomic-rank-abundance-summary-table/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;31b19088-dad2-41a0-b086-94037bce4784&quot;],&quot;name&quot;:[0,&quot;Taxonomic abundance summary tables for a specified taxonomic rank&quot;],&quot;annotation&quot;:[0,&quot;This workflow creates taxonomic summary tables for a specified taxonomic rank out of MAPseq&#39;s OTU tables output collection.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-rrna-prediction-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-rrna-prediction/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;53c5bc12-4d7c-4799-b053-ced394665d61&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - rRNA prediction&quot;],&quot;annotation&quot;:[0,&quot;Classification and visualization of SSU, LSU sequences.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;amplicon&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-quality-control-paired-end-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-paired-end/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;467d4418-a41e-44eb-9959-7c4a242d2082&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - Quality control PE&quot;],&quot;annotation&quot;:[0,&quot;Quality control subworkflow for paired-end reads. &quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-taxonomic-summary-tables-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-taxonomic-summary-tables/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;dda70895-1e39-42a3-8b4f-b6493cf15e8f&quot;],&quot;name&quot;:[0,&quot;MGnify amplicon summary tables&quot;],&quot;annotation&quot;:[0,&quot;This workflow creates taxonomic summary tables out of the amplicon pipeline results. &quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mgnify-amplicon-pipeline-v5-its-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-its/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Microbiome&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;04fa2e92-8fcd-4a20-8af1-063c7f514f12&quot;],&quot;name&quot;:[0,&quot;MGnify&#39;s amplicon pipeline v5.0 - ITS&quot;],&quot;annotation&quot;:[0,&quot;Classification and visualization of ITS regions.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;MGnify_Amplicon&quot;],[0,&quot;amplicon&quot;],[0,&quot;metagenomics&quot;],[0,&quot;name:microgalaxy&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ia-import-multiplexed-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ia-import-multiplexed-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;951c418e-f57e-491d-af38-f2c44c6089ce&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ia: multiplexed data (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing single-end multiplexed data (not demultiplexed yet)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ib-import-multiplexed-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ib-import-multiplexed-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b09b5420-bb7c-4f27-9c90-7e2031889379&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ib: multiplexed data (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing paired-end multiplexed data (not demultiplexed yet)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-ic-import-demultiplexed-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Ic-import-demultiplexed-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;5a57f873-52f8-4c51-9e8d-e747d32d183a&quot;],&quot;name&quot;:[0,&quot;QIIME2 Ic: Demultiplexed data (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing demultiplexed data (single-end)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-i-import-id-import-demultiplexed-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-I-import/Id-import-demultiplexed-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f4887024-6f68-4d53-a889-5635366dd18d&quot;],&quot;name&quot;:[0,&quot;QIIME2 Id: Demultiplexed data (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Importing demultiplexed data (paired-end)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-ii-denoising-iia-denoising-se&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIa-denoising-se&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;718df5b9-9cbb-4792-be50-604e18ef0548&quot;],&quot;name&quot;:[0,&quot;QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)&quot;],&quot;annotation&quot;:[0,&quot;Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-ii-denoising-iib-denoising-pe&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIb-denoising-pe&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8b8c4f4c-7cc0-422d-9a6f-47357faa0082&quot;],&quot;name&quot;:[0,&quot;QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)&quot;],&quot;annotation&quot;:[0,&quot;Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-iii-vi-downsteam-qiime2-iii-v-phylogeny-rarefaction-taxonomic-analysis&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3163175d-7673-42d9-8aae-505afbd74946&quot;],&quot;name&quot;:[0,&quot;QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis&quot;],&quot;annotation&quot;:[0,&quot;This workflow \n- Reconstruct phylogeny (insert fragments in a reference)\n- Alpha rarefaction analysis\n- Taxonomic analysis&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;qiime2-iii-vi-downsteam-qiime2-vi-diversity-metrics-and-estimations&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-VI-diversity-metrics-and-estimations&quot;],&quot;updated&quot;:[0,&quot;2024-11-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;300b5a7d-f763-486a-a8ea-87ea84571f19&quot;],&quot;name&quot;:[0,&quot;QIIME2 VI: Diversity metrics and estimations&quot;],&quot;annotation&quot;:[0,&quot;The first step in hypothesis testing in microbial ecology is typically to look at within- (alpha) and between-sample (beta) diversity. We can calculate diversity metrics, apply appropriate statistical tests, and visualize the data using the q2-diversity plugin.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;dada2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/dada2/main&quot;],&quot;updated&quot;:[0,&quot;2025-02-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Metabarcoding&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;11d717f8-92ef-4dd7-bda7-c193175fece2&quot;],&quot;name&quot;:[0,&quot;dada2 amplicon analysis pipeline - for paired end data&quot;],&quot;annotation&quot;:[0,&quot;dada2 amplicon analysis for paired end data\n\nThe workflow has three main outputs: \n- the sequence table (output of makeSequenceTable)\n- the taxonomy (output of assignTaxonomy)\n- the counts which allow to track the number of sequences in the samples through the steps (output of sequence counts)&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:amplicon&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;haploid-variant-calling-wgs-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/haploid-variant-calling-wgs-pe/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d5b4eace-d876-4999-bad3-14c1cb02b0c9&quot;],&quot;name&quot;:[0,&quot;Paired end variant calling in haploid system&quot;],&quot;annotation&quot;:[0,&quot;Workflow for variant analysis against a reference genome in GenBank format&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;generic&quot;],[0,&quot;VeuPath&quot;],[0,&quot;Haploid&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;variation-reporting-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/variation-reporting/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9e94da53-d8ab-4713-8aac-7c6674fa1d39&quot;],&quot;name&quot;:[0,&quot;Generic variation analysis reporting&quot;],&quot;annotation&quot;:[0,&quot;This workflow takes a VCF dataset of variants produced by any of the variant calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;mpvx&quot;],[0,&quot;generic&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;generic-variant-calling-wgs-pe-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/generic-variant-calling-wgs-pe/main&quot;],&quot;updated&quot;:[0,&quot;2023-11-20T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;be0d055d-25ac-422a-ae9c-b0efdfb5bf86&quot;],&quot;name&quot;:[0,&quot;Generic variation analysis on WGS PE data&quot;],&quot;annotation&quot;:[0,&quot;Workflow for variant analysis against  a reference genome in GenBank format&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;mpxv&quot;],[0,&quot;generic&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-quantitation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-quantitation/main&quot;],&quot;updated&quot;:[0,&quot;2024-08-14T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e21d954c-bb46-46f3-8081-0095b6b0de1f&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Quantitation&quot;],&quot;annotation&quot;:[0,&quot;Clinical Metaproteomics 4: Quantitation &quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-discovery-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-discovery/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;94138387-134e-4622-96a9-fdaee428ee1b&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Discovery Workflow&quot;],&quot;annotation&quot;:[0,&quot;Workflow for clinical metaproteomics database searching&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-data-interpretation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-data-interpretation/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-19T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;3a5d60ec-2b30-47cf-a351-7b85df8af9e3&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Data Interpretation&quot;],&quot;annotation&quot;:[0,&quot;This workflow will perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-verification-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-verification/main&quot;],&quot;updated&quot;:[0,&quot;2024-12-16T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;22302833-4954-4414-8725-d8bf74d6ffd0&quot;],&quot;name&quot;:[0,&quot;Clinical Metaproteomics Verification Workflow&quot;],&quot;annotation&quot;:[0,&quot;In proteomics research, verifying detected peptides is essential for ensuring data accuracy and biological relevance. This tutorial continues from the clinical metaproteomics discovery workflow, focusing on verifying identified microbial peptides using the PepQuery tool.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;clinicalmp-database-generation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/clinicalmp-database-generation/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;],[0,&quot;Metaproteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;758f86d1-b81e-438d-9666-bc7dc31b4627&quot;],&quot;name&quot;:[0,&quot;Generate a Clinical Metaproteomics Database&quot;],&quot;annotation&quot;:[0,&quot;The workflow begins with the Database Generation process. The Galaxy-P team has developed a workflow that collects protein sequences from known disease-causing microorganisms to build a comprehensive database. This extensive database is then refined into a smaller, more relevant dataset using the Metanovo tool.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:clinicalMP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;openms-metaprosip-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/openms-metaprosip/main&quot;],&quot;updated&quot;:[0,&quot;2024-06-17T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Proteomics&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b03da1a7-fc6f-419f-aa2c-431a670f9a96&quot;],&quot;name&quot;:[0,&quot;MetaProSIP OpenMS 2.8&quot;],&quot;annotation&quot;:[0,&quot;Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics &quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pox-virus-amplicon-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pox-virus-amplicon/main&quot;],&quot;updated&quot;:[0,&quot;2025-10-08T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;b7574fe6-4a44-4d3b-93e9-e6d4065d2dfd&quot;],&quot;name&quot;:[0,&quot;Pox Virus Illumina Amplicon Workflow from half-genomes&quot;],&quot;annotation&quot;:[0,&quot;A workflow for the analysis of pox virus genomes sequenced as half-genomes (for ITR resolution) in a tiled-amplicon approach&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;pox&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;influenza-isolates-consensus-and-subtyping-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/influenza-isolates-consensus-and-subtyping/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-02T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;fb161003-8ad7-4a72-8835-bfd2c9088092&quot;],&quot;name&quot;:[0,&quot;Influenza A isolate subtyping and consensus sequence generation&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs subtyping and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;AIV&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;generic-non-segmented-viral-variant-calling-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/generic-non-segmented-viral-variant-calling/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Variant Calling&quot;],[0,&quot;Virology&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;967c16f1-0cd3-42c3-a6ad-9b392d597613&quot;],&quot;name&quot;:[0,&quot;Variant calling and consensus construction from paired end short read data of non-segmented viral genomes&quot;],&quot;annotation&quot;:[0,&quot;Variant calling and consensus sequence generation for batches of Illumina PE sequenced viruses with uncomplicated and stable genome structure (like e.g. Morbilliviruses).&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;ARTIC&quot;],[0,&quot;virology&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;mitogenome-assembly-vgp0-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Mitogenome-assembly-VGP0/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7de562f8-2ef4-4277-a88b-c0d12f22f0b6&quot;],&quot;name&quot;:[0,&quot;Mitogenome Assembly VGP0&quot;],&quot;annotation&quot;:[0,&quot;Generate mitochondrial assembly based on PacBio HiFi reads. Part of the VGP suite, it can be run at any time independently of the other workflows. This workflow uses MitoHiFi and a mitochondrial reference to assemble the mitochondrial genome from PacBio reads. You do not need to provide the reference yourself, only the Latin name of the species.&quot;],&quot;release&quot;:[0,&quot;0.2.2&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;purge-duplicate-contigs-vgp6-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Purge-duplicate-contigs-VGP6/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c4a1ca2e-13a0-4d0b-bdec-d5bbac5b9a34&quot;],&quot;name&quot;:[0,&quot;Purge duplicate contigs from a diploid assembly VGP6&quot;],&quot;annotation&quot;:[0,&quot;Purge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication). The contigs are purged from the first assembly (hap1, pri...), added to the second assembly (hp2, alt... ), then the 2nd assembly is purged as well. If you think only one of the assemblies needs purging, use the VGP6b workflow.  This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).&quot;],&quot;release&quot;:[0,&quot;0.10.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;kmer-profiling-hifi-trio-vgp2-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/kmer-profiling-hifi-trio-VGP2/main&quot;],&quot;updated&quot;:[0,&quot;2025-04-30T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;a8867bbf-2a39-4cf1-b530-eb5650961c78&quot;],&quot;name&quot;:[0,&quot;kmer-profiling-hifi-trio-VGP2&quot;],&quot;annotation&quot;:[0,&quot;Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.&quot;],&quot;release&quot;:[0,&quot;0.1.6&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-trio-phasing-vgp5-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-Trio-phasing-VGP5/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9fc9042f-6171-451a-bdd7-010abdb5b683&quot;],&quot;name&quot;:[0,&quot;Genome Assembly with Pacbio Hifi reads and Trio data for phasing - VGP5&quot;],&quot;annotation&quot;:[0,&quot;Generate phased assembly based on PacBio HiFi reads and parental Illumina data for phasing. Part of the VGP workflow suite, it needs to be run after the Trio k-mer Profiling workflow VGP2. This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury plots, and the genome assembly contigs in fasta and GFA format. &quot;],&quot;release&quot;:[0,&quot;0.9.9&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scaffolding-bionano-vgp7-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Scaffolding-Bionano-VGP7/main&quot;],&quot;updated&quot;:[0,&quot;2024-08-13T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;9fd3c0de-a497-47dc-96e3-0f3e1998173d&quot;],&quot;name&quot;:[0,&quot;Scaffolding-BioNano-VGP7&quot;],&quot;annotation&quot;:[0,&quot;Scaffold a genome assembly using Bionano optical maps data. Part of the VGP suite, it can be run on the assembly in GFA format generated by the contigging step in workflow 3,4, or 5. If you used a different method to generate your assembly, you can use the tools gfastats in Galaxy to generate a GFA from your assembly fasta, and create a simple text file containing the estimated genome size.&quot;],&quot;release&quot;:[0,&quot;0.1.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;hi-c-contact-map-for-assembly-manual-curation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/hi-c-contact-map-for-assembly-manual-curation/main&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;34226708-cba3-421c-809f-6ba105b3812c&quot;],&quot;name&quot;:[0,&quot;PretextMap Generation from 1 or 2 haplotypes&quot;],&quot;annotation&quot;:[0,&quot;This workflow generates Hi-C contact maps for genome assemblies in the Pretext format. It is compatible with one or 2 haplotypes. It includes tracks for PacBio read coverage, Gaps, and telomeres. The Pretext files can be open in PretextView for the manual curation of genome assemblies.&quot;],&quot;release&quot;:[0,&quot;1.0beta6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-decontamination-vgp9-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-decontamination-VGP9/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;dc6237a5-4fe1-41bf-bf08-b98852503132&quot;],&quot;name&quot;:[0,&quot;Assembly decontamination VGP9 &quot;],&quot;annotation&quot;:[0,&quot;Decontamination (foreign contaminants and mitochondrial sequences) of a genome assembly after the final scaffolding step. Uses NCBI FCS GX  to identify foreign contaminants and Blast to identify mitochondrial sequences. Part of the VGP Suite.&quot;],&quot;release&quot;:[0,&quot;1.1&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;purge-duplicates-one-haplotype-vgp6b-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Purge-duplicates-one-haplotype-VGP6b/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;882b18ac-4a8c-4cf1-8c63-a85260bc37b9&quot;],&quot;name&quot;:[0,&quot;Purging duplicates in one haplotype VGP6b&quot;],&quot;annotation&quot;:[0,&quot;Purge contigs marked as duplicates by purge_dups in a single haplotype (could be haplotypic duplication or overlap duplication). If you think the purged contigs might belong to the other haplotype, use the workflow VGP6 instead. This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).&quot;],&quot;release&quot;:[0,&quot;0.8.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;kmer-profiling-hifi-vgp1-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/kmer-profiling-hifi-VGP1/main&quot;],&quot;updated&quot;:[0,&quot;2025-10-01T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f76c7eb0-71c7-4697-8fab-011eb7e4bcda&quot;],&quot;name&quot;:[0,&quot;K-mer profiling and reads statistics VGP1&quot;],&quot;annotation&quot;:[0,&quot;Evaluation of Pacbio Hifi Reads and genome profiling. Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.&quot;],&quot;release&quot;:[0,&quot;0.5&quot;],&quot;tags&quot;:[1,[[0,&quot;Reviewed&quot;],[0,&quot;VGP&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scaffolding-hic-vgp8-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Scaffolding-HiC-VGP8/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-26T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;97db8bb9-1842-439d-b363-2a06b905a098&quot;],&quot;name&quot;:[0,&quot;Scaffolding with Hi-C data VGP8&quot;],&quot;annotation&quot;:[0,&quot;This workflow performs the scaffolding of a genome assembly using HiC data with YAHS. Can be used on any assembly with Hi-C data, and the assembly in the gfa format.&quot;],&quot;release&quot;:[0,&quot;3.1&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP_curated&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;plot-nx-size-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Plot-Nx-Size/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-25T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;7a7ad2da-596c-40d1-b67d-0b3b6a412c8b&quot;],&quot;name&quot;:[0,&quot;Generate Nx and Size plots for multiple assemblies&quot;],&quot;annotation&quot;:[0,&quot;Generate Nx and size plots for multiple assemblies to compare the evolution of assembly quality through the scaffolding process. Inputs are the fasta files for each assembly to compare.&quot;],&quot;release&quot;:[0,&quot;0.1.8&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-only-vgp3-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-only-VGP3/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-29T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;40ef3ef9-6781-42d6-baf0-70fe10cc9622&quot;],&quot;name&quot;:[0,&quot;Genome Assembly from Hifi reads - VGP3&quot;],&quot;annotation&quot;:[0,&quot;Generate a genome assembly based on PacBio HiFi reads. Part of the VGP suite, it needs to be run after the VGP1 k-mer profiling workflow. The assembly contigs are built using HiFiasm, and the workflow generates assembly statistics, BUSCO reports, Merqury plots, and the contigs in fasta and GFA formats.&quot;],&quot;release&quot;:[0,&quot;0.3.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-hifi-hic-phasing-vgp4-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/Assembly-Hifi-HiC-phasing-VGP4/main&quot;],&quot;updated&quot;:[0,&quot;2025-09-30T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Vertebrate Genome Project&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f0ae78ab-8dc1-4807-b9c6-1efd8929ed6f&quot;],&quot;name&quot;:[0,&quot;Genome Assembly from Hifi reads with HiC phasing - VGP4&quot;],&quot;annotation&quot;:[0,&quot;Assemble Genome using PacBio HiFi and HiC data from the same individual for phasing. Prerequisite: Run k-mer profiling workflow (VGP1). This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury histograms, and the genome assembly contigs in fasta and GFA format.&quot;],&quot;release&quot;:[0,&quot;0.4&quot;],&quot;tags&quot;:[1,[[0,&quot;VGP&quot;],[0,&quot;Reviewed&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;annotation-helixer-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/annotation-helixer/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;02f1b56b-f9e2-4da5-9d52-83e1f059fae6&quot;],&quot;name&quot;:[0,&quot;Genome annotation with Helixer&quot;],&quot;annotation&quot;:[0,&quot;This workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. &quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[[0,&quot;genomeannotation&quot;],[0,&quot;busco&quot;],[0,&quot;jbrowse&quot;],[0,&quot;helixer&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;annotation-maker-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/annotation-maker/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-06T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;87be9418-f744-42b6-8ac8-01dae7998e4f&quot;],&quot;name&quot;:[0,&quot;Genome annotation with Maker&quot;],&quot;annotation&quot;:[0,&quot;This workflow allows for genome annotation using Maker and evaluates the quality of the annotation.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;maker&quot;],[0,&quot;genomeannotation&quot;],[0,&quot;busco&quot;],[0,&quot;jbrowse&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;functional-annotation-protein-sequences-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/functional-annotation-protein-sequences/main&quot;],&quot;updated&quot;:[0,&quot;2024-12-04T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;4cbba315-c9bc-4895-aeeb-57dadef3542a&quot;],&quot;name&quot;:[0,&quot;Functional annotation of protein sequences&quot;],&quot;annotation&quot;:[0,&quot;This workflow uses eggNOG mapper and InterProScan for functional annotation of protein sequences.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;lncrnas-annotation-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/lncRNAs-annotation/main&quot;],&quot;updated&quot;:[0,&quot;2025-03-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome Annotation&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;f6d6b63f-6fcc-42bb-bafb-12d9dd98983e&quot;],&quot;name&quot;:[0,&quot;lncRNAs annotation workflow&quot;],&quot;annotation&quot;:[0,&quot;This workflow runs the FEELnc tool to annotate long non-coding RNAs. Before annotating these long non-coding RNAs, StringTie will be used to assemble the RNA-seq alignments into potential trancriptions. The gffread tool provides a genome annotation file in GTF format.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;quality-and-contamination-control-raw-reads-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/quality-and-contamination-control-raw-reads/main&quot;],&quot;updated&quot;:[0,&quot;2025-07-21T00:00:00&quot;],&quot;collections&quot;:[1,[]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;8467a350-2e23-41f0-bd5c-d767569528c8&quot;],&quot;name&quot;:[0,&quot;Raw Read Quality and Contamination Control For Genome Assembly&quot;],&quot;annotation&quot;:[0,&quot;Short paired-end read analysis to provide quality analysis, read cleaning and taxonomy assignation directly from raw reads&quot;],&quot;release&quot;:[0,&quot;1.1.10&quot;],&quot;tags&quot;:[1,[[0,&quot;Genomics&quot;],[0,&quot;fastq&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;taxonomy-assignment&quot;],[0,&quot;paired-end&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;],[0,&quot;trimming&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;bacterial-genome-assembly-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/bacterial-genome-assembly/main&quot;],&quot;updated&quot;:[0,&quot;2025-08-11T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;14fc2baa-a9fb-4e90-a8ec-f542faf1d8c2&quot;],&quot;name&quot;:[0,&quot;Bacterial Genome Assembly using Shovill&quot;],&quot;annotation&quot;:[0,&quot;Assembly of bacterial paired-end short read data&quot;],&quot;release&quot;:[0,&quot;2.0&quot;],&quot;tags&quot;:[1,[[0,&quot;fastq&quot;],[0,&quot;Genomics&quot;],[0,&quot;bacterial-genomics&quot;],[0,&quot;paired-end&quot;],[0,&quot;assembly&quot;],[0,&quot;quality&quot;],[0,&quot;ABRomics&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;polish-with-long-reads-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/polish-with-long-reads/main&quot;],&quot;updated&quot;:[0,&quot;2023-07-15T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;68d6270a-bfd1-452a-abc9-4a314e13af85&quot;],&quot;name&quot;:[0,&quot;Assembly polishing with long reads&quot;],&quot;annotation&quot;:[0,&quot;Racon polish with long reads, x4&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;assembly-with-flye-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/assembly-with-flye/main&quot;],&quot;updated&quot;:[0,&quot;2025-05-19T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Genome assembly&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;21643946-a984-4a3f-be71-d480590522c3&quot;],&quot;name&quot;:[0,&quot;Genome assembly with Flye&quot;],&quot;annotation&quot;:[0,&quot;Assemble long reads with Flye, then view assembly statistics and assembly graph&quot;],&quot;release&quot;:[0,&quot;0.3&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-cellplex&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-cellplex&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;c54841b3-902d-4949-93ca-f6882c12c4dd&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Preprocessing: 10X Genomics CellPlex Multiplexed Samples&quot;],&quot;annotation&quot;:[0,&quot;Comprehensive preprocessing for 10X Genomics CellPlex multiplexed single-cell RNA-seq data. Processes Cell Multiplexing Oligo (CMO) FASTQ files with CITE-seq-Count including required CellPlex-specific translation steps. Simultaneously processes gene expression FASTQ files with STARsolo and DropletUtils for alignment and cell filtering, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).&quot;],&quot;release&quot;:[0,&quot;0.6.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-v3&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-v3&quot;],&quot;updated&quot;:[0,&quot;2025-06-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;677cbe1a-f944-4c6f-8a7e-118030b1d567&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat and Scanpy Compatible Format&quot;],&quot;annotation&quot;:[0,&quot;Complete preprocessing pipeline for 10X Genomics v3 single-cell RNA-seq data. Aligns raw FASTQ files using STARsolo, performs cell calling and quality filtering with DropletUtils, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).&quot;],&quot;release&quot;:[0,&quot;0.6.3&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;baredsc-baredsc-1d-lognorm&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/baredsc/baredSC-1d-logNorm&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;1ea0b92d-d863-435a-96ac-1e547fdbc994&quot;],&quot;name&quot;:[0,&quot;Single-Cell Mixture Analysis: baredSC 1D Log-Normalized Models&quot;],&quot;annotation&quot;:[0,&quot;Applies baredSC algorithm to fit and combine one-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on expression of genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.&quot;],&quot;release&quot;:[0,&quot;0.6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;baredsc-baredsc-2d-lognorm&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/baredsc/baredSC-2d-logNorm&quot;],&quot;updated&quot;:[0,&quot;2025-03-10T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;1936816b-958a-4140-b839-58be4cdae1a5&quot;],&quot;name&quot;:[0,&quot;Single-Cell Mixture Analysis: baredSC 2D Log-Normalized Models&quot;],&quot;annotation&quot;:[0,&quot;Applies baredSC algorithm to fit and combine two-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on co-expression patterns of two genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.&quot;],&quot;release&quot;:[0,&quot;0.6&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;pseudobulk-worflow-decoupler-edger-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/pseudobulk-worflow-decoupler-edger/main&quot;],&quot;updated&quot;:[0,&quot;2024-11-18T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Transcriptomics&quot;],[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;e4e4055c-7e3f-4014-a8e6-343199fd9b01&quot;],&quot;name&quot;:[0,&quot;Single-Cell Pseudobulk Differential Expression Analysis with edgeR&quot;],&quot;annotation&quot;:[0,&quot;Performs differential gene expression analysis on single-cell data using a pseudobulk approach. Aggregates cell-level counts from an annotated AnnData object by cell type or other metadata using Decoupler, then applies edgeR for robust statistical testing between conditions. Includes data preprocessing steps for compatibility with edgeR and generates interactive volcano plots to visualize significantly differentially expressed genes.&quot;],&quot;release&quot;:[0,&quot;0.1.1&quot;],&quot;tags&quot;:[1,[]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;scanpy-clustering-main&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/scanpy-clustering/main&quot;],&quot;updated&quot;:[0,&quot;2024-10-09T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;d6b2d1bb-6aaf-47cc-8956-5ae80b775de5&quot;],&quot;name&quot;:[0,&quot;Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering&quot;],&quot;annotation&quot;:[0,&quot;End-to-end analysis of single-cell RNA-seq data using the Scanpy/AnnData ecosystem. Imports count matrices, applies quality control filtering of low-quality cells and genes, normalizes and scales data, performs dimension reduction (PCA), and identifies cell clusters using the Louvain algorithm. Generates publication-ready visualizations including UMAP plots colored by clusters and marker genes.&quot;],&quot;release&quot;:[0,&quot;0.1&quot;],&quot;tags&quot;:[1,[[0,&quot;name:scRNA-seq&quot;],[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;velocyto-velocyto-on10x-from-bundled&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-from-bundled&quot;],&quot;updated&quot;:[0,&quot;2024-02-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;0608c7f4-767c-47b3-ae45-4e98c7c83bec&quot;],&quot;name&quot;:[0,&quot;RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output&quot;],&quot;annotation&quot;:[0,&quot;Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Automatically extracts cell barcodes from standard 10X bundled output and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}],[0,{&quot;iwcID&quot;:[0,&quot;velocyto-velocyto-on10x-filtered-barcodes&quot;],&quot;trsID&quot;:[0,&quot;#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-filtered-barcodes&quot;],&quot;updated&quot;:[0,&quot;2024-02-05T00:00:00&quot;],&quot;collections&quot;:[1,[[0,&quot;Single Cell&quot;]]],&quot;definition&quot;:[0,{&quot;uuid&quot;:[0,&quot;33862923-af05-48ba-aec5-14393981cee2&quot;],&quot;name&quot;:[0,&quot;RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes&quot;],&quot;annotation&quot;:[0,&quot;Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Takes pre-filtered cell barcodes as input and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis in tools like scVelo.&quot;],&quot;release&quot;:[0,&quot;0.2&quot;],&quot;tags&quot;:[1,[[0,&quot;name:single-cell&quot;]]]}]}]]]}" ssr client="load" opts="{&quot;name&quot;:&quot;WorkflowGrid&quot;,&quot;value&quot;:true}" await-children><div class="w-full"><div id="workflows" class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4"><!--[--><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-b7574fe6-4a44-4d3b-93e9-e6d4065d2dfd"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/pox-virus-amplicon-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Pox Virus Illumina Amplicon Workflow from half-genomes</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">A workflow for the analysis of pox virus genomes sequenced as half-genomes (for ITR resolution) in a tiled-amplicon approach</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Virology<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 8, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f76c7eb0-71c7-4697-8fab-011eb7e4bcda"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/kmer-profiling-hifi-vgp1-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">K-mer profiling and reads statistics VGP1</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Evaluation of Pacbio Hifi Reads and genome profiling. Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.5</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 1, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f0ae78ab-8dc1-4807-b9c6-1efd8929ed6f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/assembly-hifi-hic-phasing-vgp4-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome Assembly from Hifi reads with HiC phasing - VGP4</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Assemble Genome using PacBio HiFi and HiC data from the same individual for phasing. Prerequisite: Run k-mer profiling workflow (VGP1). This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury histograms, and the genome assembly contigs in fasta and GFA format.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 30, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-c4a1ca2e-13a0-4d0b-bdec-d5bbac5b9a34"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/purge-duplicate-contigs-vgp6-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Purge duplicate contigs from a diploid assembly VGP6</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Purge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication). The contigs are purged from the first assembly (hap1, pri...), added to the second assembly (hp2, alt... ), then the 2nd assembly is purged as well. If you think only one of the assemblies needs purging, use the VGP6b workflow.  This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.10.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 29, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-9fc9042f-6171-451a-bdd7-010abdb5b683"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/assembly-hifi-trio-phasing-vgp5-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome Assembly with Pacbio Hifi reads and Trio data for phasing - VGP5</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Generate phased assembly based on PacBio HiFi reads and parental Illumina data for phasing. Part of the VGP workflow suite, it needs to be run after the Trio k-mer Profiling workflow VGP2. This workflow uses HiFiasm for contigging, and generates assembly statistics, BUSCO reports, Merqury plots, and the genome assembly contigs in fasta and GFA format. </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.9.9</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 29, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-882b18ac-4a8c-4cf1-8c63-a85260bc37b9"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/purge-duplicates-one-haplotype-vgp6b-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Purging duplicates in one haplotype VGP6b</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Purge contigs marked as duplicates by purge_dups in a single haplotype (could be haplotypic duplication or overlap duplication). If you think the purged contigs might belong to the other haplotype, use the workflow VGP6 instead. This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.8.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 29, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-40ef3ef9-6781-42d6-baf0-70fe10cc9622"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/assembly-hifi-only-vgp3-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome Assembly from Hifi reads - VGP3</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Generate a genome assembly based on PacBio HiFi reads. Part of the VGP suite, it needs to be run after the VGP1 k-mer profiling workflow. The assembly contigs are built using HiFiasm, and the workflow generates assembly statistics, BUSCO reports, Merqury plots, and the contigs in fasta and GFA formats.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 29, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-97db8bb9-1842-439d-b363-2a06b905a098"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/scaffolding-hic-vgp8-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Scaffolding with Hi-C data VGP8</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow performs the scaffolding of a genome assembly using HiC data with YAHS. Can be used on any assembly with Hi-C data, and the assembly in the gfa format.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 3.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 26, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-dc6237a5-4fe1-41bf-bf08-b98852503132"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/assembly-decontamination-vgp9-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Assembly decontamination VGP9 </h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Decontamination (foreign contaminants and mitochondrial sequences) of a genome assembly after the final scaffolding step. Uses NCBI FCS GX  to identify foreign contaminants and Blast to identify mitochondrial sequences. Part of the VGP Suite.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 29, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-8911f5cd-67b1-4d57-be62-abe84ca07f23"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/bacterial-quality-and-contamination-control-post-assembly-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Post-Assembly Quality Control and Contamination Check for Bacterial Genomes</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow performs quality and contamination control analysis on assembled contigs to assess bacterial genome quality and taxonomic assignment</p><!----></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 25, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-7de562f8-2ef4-4277-a88b-c0d12f22f0b6"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mitogenome-assembly-vgp0-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Mitogenome Assembly VGP0</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Generate mitochondrial assembly based on PacBio HiFi reads. Part of the VGP suite, it can be run at any time independently of the other workflows. This workflow uses MitoHiFi and a mitochondrial reference to assemble the mitochondrial genome from PacBio reads. You do not need to provide the reference yourself, only the Latin name of the species.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 25, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-7a7ad2da-596c-40d1-b67d-0b3b6a412c8b"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/plot-nx-size-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Generate Nx and Size plots for multiple assemblies</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Generate Nx and size plots for multiple assemblies to compare the evolution of assembly quality through the scaffolding process. Inputs are the fasta files for each assembly to compare.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.8</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 25, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-967c16f1-0cd3-42c3-a6ad-9b392d597613"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/generic-non-segmented-viral-variant-calling-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Variant calling and consensus construction from paired end short read data of non-segmented viral genomes</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Variant calling and consensus sequence generation for batches of Illumina PE sequenced viruses with uncomplicated and stable genome structure (like e.g. Morbilliviruses).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Virology<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 18, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-14fc2baa-a9fb-4e90-a8ec-f542faf1d8c2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/bacterial-genome-assembly-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Bacterial Genome Assembly using Shovill</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Assembly of bacterial paired-end short read data</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome assembly<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 2.0</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 11, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-47a7aa1d-8cb6-4eab-a6d8-86909bca0d43"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/bacterial_genome_annotation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">bacterial_genome_annotation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Annotation of an assembled bacterial genomes to detect genes, potential plasmids, integrons and Insertion sequence (IS) elements.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.1.11</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jul 21, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-8467a350-2e23-41f0-bd5c-d767569528c8"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/quality-and-contamination-control-raw-reads-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Raw Read Quality and Contamination Control For Genome Assembly</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Short paired-end read analysis to provide quality analysis, read cleaning and taxonomy assignation directly from raw reads</p><!----></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.1.10</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jul 21, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-02f1b56b-f9e2-4da5-9d52-83e1f059fae6"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/annotation-helixer-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome annotation with Helixer</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jul 18, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-eeedc263-2a29-4651-9b2b-007bf5d3cf4b"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/multiplex-tissue-microarray-analysis-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Multiplex Tissue Microarray Analysis</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Perform background subtraction, nuclear segmentation, feature quantification, cellular phenotyping, spatial analysis, and interactive visualization of registered TMA core multiplex tissue images</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Imaging<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 22, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-c880e149-6237-46b6-aac6-7e1e3bb58ede"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/tissue-microarray-analysis-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">End-to-End Tissue Microarray Analysis</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete multiplex tissue image (MTI) analysis pipeline for tissue microarray (TMA) data imaged using cyclic immunofluorescence: Performs illumination correction, stitching and registration, and tissue microarray segmentation. Tissue-segmented images undergo nuclear segmentation, cell/nuclei feature quantification (mean marker intensities, cell coordinates, and morphological features), and cell phenotyping. Produces outputs that are compatible with downstream single-cell/spatial analysis and interactive image viewers including: Pyramidal OME-TIFF images, nuclear segmentation masks (TIFF), quantified feature tables (CSV, h5ad) with cell type annotations, and an interactive Vitessce dashboard that combines image viewing with linked single-cell data visualizations. </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Imaging<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 21, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-e4e70fc0-b584-4e86-8b8f-c6bcdda2e11a"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mags-building-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Metagenome-Assembled Genomes (MAGs) generation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow constructs Metagenome-Assembled Genomes (MAGs) using SPAdes or MEGAHIT as assemblers, followed by binning with four different tools and refinement using Binette. The resulting MAGs are dereplicated across the entire input sample set, then annotated and evaluated for quality.
You can provide pooled reads (for co-assembly/binning), individual read sets, or a combination of both. The input samples must consist of the original reads, which are used for abundance estimation. In all cases, reads should be trimmed, adapters removed, and cleaned of host or other contaminants before processing.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-b57e88a9-fe46-4995-812c-16c28efeac21"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/chipseq-sr-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">ChIP-seq Analysis: Single-End Read Processing</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete ChIP-seq analysis for single-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and quality filtering (MAPQ &amp;gt;= 30). Peak calling with MACS2 uses either a fixed extension parameter or built-in model to identify protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.15</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-25c8f8e3-7ed9-4523-8862-8496fa1777cd"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/chipseq-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">ChIP-seq Analysis: Paired-End Read Processing</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete ChIP-seq analysis for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and stringent quality filtering (MAPQ &amp;gt;= 30, concordant pairs only). Peak calling with MACS2 optimized for paired-end reads identifies protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.15</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-34226708-cba3-421c-809f-6ba105b3812c"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/hi-c-contact-map-for-assembly-manual-curation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">PretextMap Generation from 1 or 2 haplotypes</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow generates Hi-C contact maps for genome assemblies in the Pretext format. It is compatible with one or 2 haplotypes. It includes tracks for PacBio read coverage, Gaps, and telomeres. The Pretext files can be open in PretextView for the manual curation of genome assemblies.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.0beta6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-c54841b3-902d-4949-93ca-f6882c12c4dd"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-cellplex/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell RNA-seq Preprocessing: 10X Genomics CellPlex Multiplexed Samples</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Comprehensive preprocessing for 10X Genomics CellPlex multiplexed single-cell RNA-seq data. Processes Cell Multiplexing Oligo (CMO) FASTQ files with CITE-seq-Count including required CellPlex-specific translation steps. Simultaneously processes gene expression FASTQ files with STARsolo and DropletUtils for alignment and cell filtering, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.6.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-677cbe1a-f944-4c6f-8a7e-118030b1d567"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/fastq-to-matrix-10x-scrna-seq-fastq-to-matrix-10x-v3/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat and Scanpy Compatible Format</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete preprocessing pipeline for 10X Genomics v3 single-cell RNA-seq data. Aligns raw FASTQ files using STARsolo, performs cell calling and quality filtering with DropletUtils, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.6.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 9, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-21643946-a984-4a3f-be71-d480590522c3"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/assembly-with-flye-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome assembly with Flye</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Assemble long reads with Flye, then view assembly statistics and assembly graph</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome assembly<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 19, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-af2457b4-9013-4d8c-9e76-9948c1f96fcc"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/lcms-preprocessing-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Mass spectrometry: LC-MS preprocessing with XCMS</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package (Kuhl, C 2012).


https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabolomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 5, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-7abd10fb-d083-4732-b11f-56581e23905d"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/gcms-metams-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Mass spectrometry: GCMS with metaMS</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract and the metaMS R package (Wehrens, R 2014) for the field of untargeted metabolomics. 

https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabolomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 5, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-fad227f2-8b1e-4782-a7a5-a1743b9643bf"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-pipeline-v5-complete-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify&#39;s amplicon pipeline v5.0</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">MGnify&#39;s amplicon pipeline v5.0. Including the Quality control for single-end and paired-end reads, rRNA-prediction, and ITS sub-WFs.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 5, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-467d4418-a41e-44eb-9959-7c4a242d2082"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-pipeline-v5-quality-control-paired-end-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify&#39;s amplicon pipeline v5.0 - Quality control PE</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Quality control subworkflow for paired-end reads. </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 5, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-fb161003-8ad7-4a72-8835-bfd2c9088092"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/influenza-isolates-consensus-and-subtyping-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Influenza A isolate subtyping and consensus sequence generation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow performs subtyping and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Virology<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 2, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-a8867bbf-2a39-4cf1-b530-eb5650961c78"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/kmer-profiling-hifi-trio-vgp2-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">kmer-profiling-hifi-trio-VGP2</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 30, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-067945f1-8716-4558-918e-4a9c9d42f604"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qcxms-sdf-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QCxMS Spectra Prediction from SDF</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Workflow to predict EI mass spectra using QCxMS starting from a single SDF file, containing the 3D coordinates of all atoms in the molecule. These files can typically be obtained from PubChem.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabolomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 28, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-3d86e5fc-cad1-4e21-8ecc-d0e6637899bc"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-pe-illumina-artic-ivar-analysis-sars-cov-2-illumina-amplicon-ivar-pangolin-nextclade/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">SARS-CoV-2 Illumina Amplicon pipeline - iVar based</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Find and annotate variants in ampliconic SARS-CoV-2 Illumina sequencing data and classify samples with pangolin and nextclade</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 28, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-3be0019d-8058-4e57-922a-417b9e3578a2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mfassignr-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Molecular formula assignment and recalibration with MFAssignR package.</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow can be used to assign multi-element molecular formulas to ultrahigh resolution mass spectra.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabolomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 16, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-15d188a7-b028-4cf7-bb5d-13858e1f6e78"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/rnaseq-de-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">RNA-Seq Differential Expression Analysis with Visualization</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Identifies differentially expressed genes between exactly two experimental conditions from count tables. The workflow performs statistical testing, applies filters based on adjusted p-value and log2 fold change thresholds, and generates publication-quality visualizations including volcano plots, MA plots, and heatmaps. Takes two collections of count tables as input and produces filtered gene lists and interactive plots for interpreting expression differences. Optimal for simple two-condition experimental designs.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 24, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-864427b0-d25a-499c-b437-f83f2d6b83f2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-pe-illumina-artic-variant-calling-covid-19-pe-artic-illumina/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: variation analysis on ARTIC PE data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">The workflow for Illumina-sequenced ARTIC data builds on the RNASeq workflow for paired-end data using the same steps for mapping and variant calling, but adds extra logic for trimming ARTIC primer sequences off reads with the ivar package. In addition, this workflow uses ivar also to identify amplicons affected by ARTIC primer-binding site mutations and tries to exclude reads derived from such tainted amplicons when calculating allele-frequencies of other variants.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.5.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 17, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-435875a1-72c4-4fda-8943-3a78f25c6574"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/allele-based-pathogen-identification-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Allele-based Pathogen Identification</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Microbiome - Variant calling and Consensus Building</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-06dc40a7-99f2-4b3c-ae21-b3dcb239306f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-consensus-from-variation-covid-19-consensus-construction/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: consensus construction</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Build a consensus sequence from FILTER PASS variants with intrasample allele-frequency above a configurable consensus threshold.
Hard-mask regions with low coverage (but not consensus variants within them) and ambiguous sites.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.4.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-5171bc59-590c-4e97-8db8-004f73928627"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-se-illumina-wgs-variant-calling-covid-19-se-wgs-illumina/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: variation analysis on WGS SE data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflows performs single end read mapping with bowtie2 followed by sensitive variant calling across a wide range of AFs with lofreq</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-68bf2a73-d68a-4e25-ba52-5586c057df77"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-pipeline-v5-quality-control-single-end-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify&#39;s amplicon pipeline v5.0 - Quality control SE</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Quality control subworkflow for single-end reads.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-e24060a0-1cfe-40f1-9251-48e96f7e1121"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mapseq-to-ampvis2-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MAPseq to ampvis2</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">The MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.</p><!----></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-31b19088-dad2-41a0-b086-94037bce4784"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/taxonomic-rank-abundance-summary-table-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Taxonomic abundance summary tables for a specified taxonomic rank</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow creates taxonomic summary tables for a specified taxonomic rank out of MAPseq&#39;s OTU tables output collection.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-53c5bc12-4d7c-4799-b053-ced394665d61"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-pipeline-v5-rrna-prediction-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify&#39;s amplicon pipeline v5.0 - rRNA prediction</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Classification and visualization of SSU, LSU sequences.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-dda70895-1e39-42a3-8b4f-b6493cf15e8f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-taxonomic-summary-tables-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify amplicon summary tables</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow creates taxonomic summary tables out of the amplicon pipeline results. </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-04fa2e92-8fcd-4a20-8af1-063c7f514f12"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/mgnify-amplicon-pipeline-v5-its-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MGnify&#39;s amplicon pipeline v5.0 - ITS</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Classification and visualization of ITS regions.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-1ea0b92d-d863-435a-96ac-1e547fdbc994"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/baredsc-baredsc-1d-lognorm/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell Mixture Analysis: baredSC 1D Log-Normalized Models</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Applies baredSC algorithm to fit and combine one-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on expression of genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-1936816b-958a-4140-b839-58be4cdae1a5"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/baredsc-baredsc-2d-lognorm/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell Mixture Analysis: baredSC 2D Log-Normalized Models</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Applies baredSC algorithm to fit and combine two-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on co-expression patterns of two genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 10, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-87be9418-f744-42b6-8ac8-01dae7998e4f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/annotation-maker-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Genome annotation with Maker</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow allows for genome annotation using Maker and evaluates the quality of the annotation.</p><!----></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 6, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f6d6b63f-6fcc-42bb-bafb-12d9dd98983e"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/lncrnas-annotation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">lncRNAs annotation workflow</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow runs the FEELnc tool to annotate long non-coding RNAs. Before annotating these long non-coding RNAs, StringTie will be used to assemble the RNA-seq alignments into potential trancriptions. The gffread tool provides a genome annotation file in GTF format.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Mar 5, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-11d717f8-92ef-4dd7-bda7-c193175fece2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/dada2-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">dada2 amplicon analysis pipeline - for paired end data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">dada2 amplicon analysis for paired end data

The workflow has three main outputs: 
- the sequence table (output of makeSequenceTable)
- the taxonomy (output of assignTaxonomy)
- the counts which allow to track the number of sequences in the samples through the steps (output of sequence counts)</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Feb 17, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-5ab1d724-01d4-4673-a3c3-ae2bc944458f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/rnaseq-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">RNA-Seq Analysis: Paired-End Read Processing and Quantification</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete RNA-Seq analysis for paired-end data: Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-885ec907-9f61-4a11-9d23-2243389e7335"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/rnaseq-sr-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">RNA-Seq Analysis: Single-End Read Processing and Quantification</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete RNA-Seq analysis for single-end data:  Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f71eee25-8402-41da-8284-5298656ab379"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/consensus-peaks-consensus-peaks-chip-sr/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Consensus Peak Calling for ChIP-seq Single-End Replicates</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Identifies high-confidence consensus peaks from ChIP-seq single-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-a2f66906-e9e3-42f9-8629-390017ed2db0"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/consensus-peaks-consensus-peaks-chip-pe/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Consensus Peak Calling for ChIP-seq Paired-End Replicates</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Identifies high-confidence consensus peaks from ChIP-seq paired-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f58371cd-2e2d-41da-b1ca-ef10f7330047"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/consensus-peaks-consensus-peaks-atac-cutandrun/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Consensus Peak Calling for ATAC-seq and CUT&amp;RUN Replicates</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Identifies high-confidence consensus peaks from ATAC-seq or CUT&amp;RUN replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-98e9e34b-b203-43db-9a23-5fa72963fa6d"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/cutandrun-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">CUT&amp;RUN/CUT&amp;TAG Analysis: Protein-DNA Interaction Mapping</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete CUT&amp;RUN/CUT&amp;TAG analysis workflow for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt) and alignment (Bowtie2 with dovetail option enabled). Applies quality filtering (MAPQ ≥ 30, concordant pairs only), converts BAM to BED format, and performs peak calling using MACS2 with parameters optimized for the punctate signal profile characteristic of CUT&amp;RUN/CUT&amp;TAG experiments.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.14</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jan 27, 2025</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-22302833-4954-4414-8725-d8bf74d6ffd0"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/clinicalmp-verification-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Clinical Metaproteomics Verification Workflow</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">In proteomics research, verifying detected peptides is essential for ensuring data accuracy and biological relevance. This tutorial continues from the clinical metaproteomics discovery workflow, focusing on verifying identified microbial peptides using the PepQuery tool.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metaproteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Dec 16, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-4cbba315-c9bc-4895-aeeb-57dadef3542a"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/functional-annotation-protein-sequences-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Functional annotation of protein sequences</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow uses eggNOG mapper and InterProScan for functional annotation of protein sequences.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Dec 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-747f2432-7493-44b8-a5ee-82d104407cad"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/atacseq-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">ATAC-seq Analysis: Chromatin Accessibility Profiling</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Complete ATAC-seq analysis pipeline for paired-end reads. Processes raw FASTQ data through adapter and bad quality removal (cutadapt), alignment (Bowtie2 end-to-end), and filtering (removes MT reads, discordant pairs, low mapping quality &lt;30, PCR duplicates). Generates 5&#39; cut site pileups (±100bp), performs peak calling, and quantifies reads in 1kb summit-centered regions. Produces two normalized coverage tracks (per million mapped reads and per million reads in peaks) and fragment length distribution plots for quality assessment.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.0</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 28, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-3a5d60ec-2b30-47cf-a351-7b85df8af9e3"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/clinicalmp-data-interpretation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Clinical Metaproteomics Data Interpretation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow will perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metaproteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 19, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-94138387-134e-4622-96a9-fdaee428ee1b"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/clinicalmp-discovery-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Clinical Metaproteomics Discovery Workflow</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Workflow for clinical metaproteomics database searching</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metaproteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 18, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-758f86d1-b81e-438d-9666-bc7dc31b4627"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/clinicalmp-database-generation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Generate a Clinical Metaproteomics Database</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">The workflow begins with the Database Generation process. The Galaxy-P team has developed a workflow that collects protein sequences from known disease-causing microorganisms to build a comprehensive database. This extensive database is then refined into a smaller, more relevant dataset using the Metanovo tool.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metaproteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 18, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-e4e4055c-7e3f-4014-a8e6-343199fd9b01"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/pseudobulk-worflow-decoupler-edger-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell Pseudobulk Differential Expression Analysis with edgeR</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Performs differential gene expression analysis on single-cell data using a pseudobulk approach. Aggregates cell-level counts from an annotated AnnData object by cell type or other metadata using Decoupler, then applies edgeR for robust statistical testing between conditions. Includes data preprocessing steps for compatibility with edgeR and generates interactive volcano plots to visualize significantly differentially expressed genes.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 18, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-8c2d765b-2764-4b66-8542-adc2109c24f2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/fluorescence-nuclei-segmentation-and-counting-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Segmentation and counting of cell nuclei in fluorescence microscopy images</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Imaging<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 7, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-951c418e-f57e-491d-af38-f2c44c6089ce"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-i-import-ia-import-multiplexed-se/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 Ia: multiplexed data (single-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Importing single-end multiplexed data (not demultiplexed yet)</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-b09b5420-bb7c-4f27-9c90-7e2031889379"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-i-import-ib-import-multiplexed-pe/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 Ib: multiplexed data (paired-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Importing paired-end multiplexed data (not demultiplexed yet)</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-5a57f873-52f8-4c51-9e8d-e747d32d183a"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-i-import-ic-import-demultiplexed-se/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 Ic: Demultiplexed data (single-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Importing demultiplexed data (single-end)</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f4887024-6f68-4d53-a889-5635366dd18d"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-i-import-id-import-demultiplexed-pe/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 Id: Demultiplexed data (paired-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Importing demultiplexed data (paired-end)</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-718df5b9-9cbb-4792-be50-604e18ef0548"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-ii-denoising-iia-denoising-se/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-8b8c4f4c-7cc0-422d-9a6f-47357faa0082"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-ii-denoising-iib-denoising-pe/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-3163175d-7673-42d9-8aae-505afbd74946"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-iii-vi-downsteam-qiime2-iii-v-phylogeny-rarefaction-taxonomic-analysis/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow 
- Reconstruct phylogeny (insert fragments in a reference)
- Alpha rarefaction analysis
- Taxonomic analysis</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-300b5a7d-f763-486a-a8ea-87ea84571f19"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/qiime2-iii-vi-downsteam-qiime2-vi-diversity-metrics-and-estimations/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">QIIME2 VI: Diversity metrics and estimations</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">The first step in hypothesis testing in microbial ecology is typically to look at within- (alpha) and between-sample (beta) diversity. We can calculate diversity metrics, apply appropriate statistical tests, and visualize the data using the q2-diversity plugin.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metabarcoding<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 4, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-84a63bc6-a96a-4263-a1e5-bc140f412e42"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/goseq-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Gene Ontology and KEGG Pathway Enrichment Analysis</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Performs functional enrichment analysis of gene sets using GOseq, identifying over-represented Gene Ontology terms and KEGG pathways. This workflow accounts for gene length bias in RNA-seq data when calculating enrichment statistics. Input requires differentially expressed genes and gene lengths. Generates comprehensive tables and visualizations of enriched GO terms across all three ontologies (Biological Process, Molecular Function, Cellular Component) as well as KEGG pathway enrichment results.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 3, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-d5b4eace-d876-4999-bad3-14c1cb02b0c9"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/haploid-variant-calling-wgs-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Paired end variant calling in haploid system</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Workflow for variant analysis against a reference genome in GenBank format</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 29, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-284cc748-7532-4749-8f68-d19d41c048ef"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/amr_gene_detection-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">amr_gene_detection</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Antimicrobial resistance gene detection from assembled bacterial genomes</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 1.1.6</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 11, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-d6b2d1bb-6aaf-47cc-8956-5ae80b775de5"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/scanpy-clustering-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">End-to-end analysis of single-cell RNA-seq data using the Scanpy/AnnData ecosystem. Imports count matrices, applies quality control filtering of low-quality cells and genes, normalizes and scales data, performs dimension reduction (PCA), and identifies cell clusters using the Louvain algorithm. Generates publication-ready visualizations including UMAP plots colored by clusters and marker genes.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 9, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-ac8ef6cb-1621-4b73-9362-8a460ef3c3a2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/brew3r-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">BREW3R</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Extends 3&#39; ends of gene annotations using BAM files (from STAR alignments) and a reference GTF. Specifically designed for 3&#39;-biased sequencing techniques like 10X scRNA-seq or BRB-seq that primarily capture transcript 3&#39; ends. The BREW3R tool enhances annotations by using evidence from RNA-seq data to improve 3&#39; UTR definitions, which is particularly important for accurate quantification in single-cell and bulk RNA-seq experiments.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Oct 7, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-b08c744d-7c61-4b58-ac5f-4b5886c3c643"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-variation-reporting-covid-19-variation-reporting/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: variation analysis reporting</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow takes a VCF dataset of variants produced by any of the *-variant-calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 24, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-e21d954c-bb46-46f3-8081-0095b6b0de1f"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/clinicalmp-quantitation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Clinical Metaproteomics Quantitation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Clinical Metaproteomics 4: Quantitation </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Metaproteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 14, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-9fd3c0de-a497-47dc-96e3-0f3e1998173d"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/scaffolding-bionano-vgp7-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Scaffolding-BioNano-VGP7</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Scaffold a genome assembly using Bionano optical maps data. Part of the VGP suite, it can be run on the assembly in GFA format generated by the contigging step in workflow 3,4, or 5. If you used a different method to generate your assembly, you can use the tools gfastats in Galaxy to generate a GFA from your assembly fasta, and create a simple text file containing the estimated genome size.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Vertebrate Genome Project<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Aug 13, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-363898af-e598-4f0e-abd9-e6ded395ce66"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sra-manifest-to-concatenated-fastqs-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">sra_manifest_to_concatenated_fastqs_parallel</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow takes as input a SRA_manifest from SRA Run Selector and will generate one fastq file or fastq pair of file for each experiment (concatenated multiple runs if necessary). Output will be relabelled to match the column specified by the user.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Data Fetching<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.7</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 17, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-b03da1a7-fc6f-419f-aa2c-431a670f9a96"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/openms-metaprosip-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MetaProSIP OpenMS 2.8</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics </p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Proteomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jun 17, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-15c3229d-8786-457f-a020-7aa64c0cf6e8"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/parallel-accession-download-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Parallel Accession Download</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Downloads fastq files for sequencing run accessions provided in a text file using fasterq-dump. Creates one job per listed run accession.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Data Fetching<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.14</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on May 27, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-a55428c8-9934-4941-93c4-3920b4ba3366"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/taxonomy-profiling-and-visualization-with-krona-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Taxonomy Profiling and Visualization with Krona</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Microbiome - Taxonomy Profiling</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 25, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-0b976704-e1e0-4cc3-9e75-38bda67c01ff"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/nanopore-pre-processing-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Nanopore Preprocessing</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Microbiome - QC and Contamination Filtering</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 25, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-cb26018e-9d2f-4513-a372-122474df703e"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/pathogen-detection-pathogfair-samples-aggregation-and-visualisation-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Pathogens of all samples report generation and visualization</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 24, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-d6795753-1562-4ccc-8f9c-f598acd5404e"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/gene-based-pathogen-identification-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Gene-based Pathogen Identification</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Nanopore datasets analysis - Phylogenetic Identification - antibiotic resistance genes detection and contigs building</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Microbiome<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Apr 18, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-0608c7f4-767c-47b3-ae45-4e98c7c83bec"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/velocyto-velocyto-on10x-from-bundled/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Automatically extracts cell barcodes from standard 10X bundled output and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Feb 5, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-33862923-af05-48ba-aec5-14393981cee2"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/velocyto-velocyto-on10x-filtered-barcodes/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Takes pre-filtered cell barcodes as input and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis in tools like scVelo.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Single Cell<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Feb 5, 2024</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-16455667-cdef-4979-9603-f338e09aa2a1"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/gromacs-mmgbsa-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">MMGBSA calculations with GROMACS</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">MMGBSA simulation and calculation</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.5</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 27, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f1aea424-b536-496f-a0e6-fe46df8e6e18"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/protein-ligand-complex-parameterization-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Create GRO and TOP complex files</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4"></p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-bb767af6-3641-4a5e-843d-78508fd7577a"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/fragment-based-docking-scoring-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Virtual screening of the SARS-CoV-2 main protease with rDock and pose scoring</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.5</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-7950b716-2bc9-4cb6-8472-6506c702adea"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/gromacs-dctmd-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">dcTMD calculations with GROMACS</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Perform dcTMD free energy simulations and calculations</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Computational Chemistry<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.5</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-6695eb16-30a5-42ea-8591-56f2630cf8bf"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-pe-illumina-wgs-variant-calling-covid-19-pe-wgs-illumina/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: variation analysis on WGS PE data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflows performs paired end read mapping with bwa-mem followed by sensitive variant calling across a wide range of AFs with lofreq</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2.4</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-a1ceaf35-f3d7-452d-b5ed-53a4f429d1cf"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/sars-cov-2-ont-artic-variant-calling-covid-19-artic-ont/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">COVID-19: variation analysis of ARTIC ONT data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->SARS-COV-2<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-9e94da53-d8ab-4713-8aac-7c6674fa1d39"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/variation-reporting-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Generic variation analysis reporting</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">This workflow takes a VCF dataset of variants produced by any of the variant calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-be0d055d-25ac-422a-ae9c-b0efdfb5bf86"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/generic-variant-calling-wgs-pe-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Generic variation analysis on WGS PE data</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Workflow for variant analysis against  a reference genome in GenBank format</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Variant Calling<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Nov 20, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-c6ed4b0e-d719-4997-9860-ba99488ec332"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/average-bigwig-between-replicates-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">BigWig Replicates Averaging Workflow</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Calculates average signal values across replicate BigWig files to generate a single consolidated track. Requires input file identifiers formatted as &quot;sample_name_replicateID&quot;. The tool automatically groups files by sample name, computes the mean signal at each position, and outputs consolidated BigWig files named after the common sample prefix (without the replicate ID suffix).</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Transcriptomics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.2</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 27, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-f25be8fa-7823-456f-9707-a497703f48d7"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/repeatmasking-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Repeat masking with RepeatModeler and RepeatMasker</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4"></p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome Annotation<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 21, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-2196fbff-317e-437d-a022-696a1f95a850"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/hic-hicup-cooler-hic-fastq-to-cool-hicup-cooler/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Hi-C Processing: FASTQ to Balanced Cool Files</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Comprehensive Hi-C data processing workflow that converts paired-end FASTQ files to balanced contact matrices. Uses HiCUP for pre-processing (adapter trimming, mapping, filtering) with fragment midpoint coordinate assignment, filters reads by mapping quality, and generates sorted tabix files. The final output is a balanced cool file at user-specified resolution created with cooler, ready for multi-resolution contact matrix analysis.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 8, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-2a89cbe4-a69f-4bff-9465-2e04b0217e35"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/hic-hicup-cooler-chic-fastq-to-cool-hicup-cooler/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Capture Hi-C Processing: FASTQ to Balanced Cool Files</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">End-to-end processing of Capture Hi-C paired-end sequencing data. Transforms raw FASTQ files into balanced contact matrices using HiCUP for processing Hi-C reads (adapter trimming, mapping, filtering for valid pairs) with additional filtering for MAPQ and captured regions. The workflow generates sorted tabix files via cooler and produces balanced cool files at user-specified resolution for downstream analysis and visualization.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 8, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-48733b55-4cde-45d4-a69e-78faff0db3f5"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/hic-hicup-cooler-hic-juicermediumtabix-to-cool-cooler/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Hi-C Format Conversion: Juicer Medium to Cooler Files</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Converts Hi-C interaction data from Juicer medium format (tabix files) to balanced cooler format. This streamlined workflow takes a collection of Juicer medium tabix files and a reference genome name as input and generates balanced cool files at user-specified resolution. Enables seamless transition between different Hi-C analysis ecosystems while maintaining data integrity.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 8, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-48f579ea-eaac-49ee-b38f-93c82f8b7d76"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/hic-hicup-cooler-hic-fastq-to-pairs-hicup/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Hi-C Data Processing: FASTQ to Valid Interaction Pairs</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Processes Hi-C paired-end FASTQ files to generate validated interaction pairs using HiCUP. The workflow truncates reads at ligation junctions, maps to reference genome, assigns to restriction fragments, and filters out experimental artifacts (self-ligated, dangling ends, internal fragments, and size outliers). Removes duplicates and converts outputs to formats compatible with Juicebox/cooler using fragment midpoints as coordinates. Final filtering by mapping quality ensures high-confidence interaction data.</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Epigenetics<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.3</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Sep 8, 2023</p></div></div><!--]--></div><div class="rounded-lg border border-gray-200 bg-white text-gray-950 shadow flex flex-col mb-6 hover:shadow-lg transition-shadow" id="workflow-68d6270a-bfd1-452a-abc9-4a314e13af85"><!--[--><div class="flex flex-col space-y-1.5 p-6 px-6 py-4 pb-2"><!--[--><a href="/workflow/polish-with-long-reads-main/"><h2 class="font-bold hover:underline text-ebony-clay text-xl">Assembly polishing with long reads</h2></a><!--]--></div><div class="p-6 flex-1 flex flex-col px-6 py-2"><!--[--><div class="flex flex-col flex-grow"><p class="flex-grow mb-4">Racon polish with long reads, x4</p><div class="flex flex-wrap gap-2"><!--[--><div class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 border-transparent bg-gray-100 text-gray-900 hover:bg-gray-200 cursor-pointer hover:bg-hokey-pokey hover:text-white transition-colors"><!--[-->Genome assembly<!--]--></div><!--]--></div></div><!--]--></div><div class="px-6 py-3 border-t border-gray-200"><div class="flex justify-between text-xs text-gray-500"><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 7h.01M7 3h5c.512 0 1.024.195 1.414.586l7 7a2 2 0 010 2.828l-7 7a2 2 0 01-2.828 0l-7-7A1.994 1.994 0 013 12V7a4 4 0 014-4z"></path></svg> 0.1</p><p class="flex items-center"><svg class="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"></path></svg> Updated on Jul 15, 2023</p></div></div><!--]--></div><!--]--></div><!----></div><!--astro:end--></astro-island> </div> </div> <!-- Right sidebar if provided -->  </div> </div> </main> </body></html> <script type="module">const e=document.getElementById("search-input");e&&e.addEventListener("input",t=>{const n=new CustomEvent("search-change",{detail:t.target.value});window.dispatchEvent(n)});</script>