# JBrowse2 instance generation

This workflow generates a JBrowse 2 instance for a genome assembly. It is meant to be run after the Pre-curation workflow generating the PretextMaps files. . It includes tracks for PacBio read coverage, Gaps, telomeres, and syntenic dotplots for related species. 


## Inputs

1. **Genome assembly to report on** [fasta] The concatenated haplotypes Generated by the Pre-curation PretextMap workflow
2. **Window size for bed to bigwig - 50 default** The window size use to display the coverage and gaps density
3. Telomere bed file [bed] Output of seqtk_telo in the  Pre-curation PretextMap workflow
4. Coverage bigwig [bigwig] Output of bamcoverage in the  Pre-curation PretextMap workflow
5. Gaps bed [bed] Output of gfastats in the  Pre-curation PretextMap workflow 
6. **Will you provide a collection of related species for sequence similarity mapping?** If *yes*, provide the next input
7. **Collection of related species** [fasta] Collection containing the genomic sequences of closely related species


## Outputs

1. JBrowse2 For Viewing : click on the eye to display the instance in Galaxy
2. JBrowse2 for export : zip file for download of export.