# VGP Workflow #2

This workflow performs a HiC-phased assembly using PacBio HiFi long read data and HiC data. This results in two phased assemblies: **hap1 and hap2**. These phased assemblies do not usually require purging afterwards.  

## VGP Assembly Pipeline

The collaboration between the [VGP](https://vertebrategenomesproject.org/) has led to the creation of a set of workflows dedicated to high-quality genome assembly with long reads. The pipeline includes the following workflows:
-   Creation of k-mer count databases and histograms with Meryl
-   HiFi long read assembly with Hifiasm (three modes available: pseudohaplotype, HiC-phased, and trio)
-   Scaffolding with Bionano optical map data (optional)
-   Scaffolding with HiC data

### Inputs

-   PacBio HiFi long reads (as a collection)
-   HiC forward (R1) reads
    -   NOTE: if you have multiple lanes of forward reads, you must concatenate them into one file    
-   HiC reverses (R2) reads
    -   NOTE: if you have multiple lanes of reverse reads, you must concatenate them into one file, in the same order as the concatenated R1 reads
-   Meryl database, generated by the workflow `VGP #1 : Meryl database creation`

### Outputs

-   Cutadapt trimmed reads (as a collection)
-   Hifiasm assembly:
    -   GFAs:
        -   hap1 contig graph
        -   hap2 contig graph
        -   Raw unitig graph
        -   Processed unitig graph
    -   Each assembly:
        -   hap1/hap2 assembly in fasta format
        -   BUSCO 
            -   Short text summary
            -   Missing Buscos table
            -   Full table
            -   Summary image
-   Parameter file containing the estimated genome size (for use in downstream workflows)
-   QUAST statistics for both assemblies:
        -   Tabular report
        -   HTML report
        -   PDF report
        -   Log
-   Merqury analysis of Hifiasm assemblies
    -   Stats (completeness stats)
    -   Size files
    -   QV files
    -   Images (spectra-cn & spectra-asm plots)
    -   BED files
    -   WIG files
-   gfastats
    -   gfastats on hap1 contig graph
    -   gfastats on hap2 contig graph
