# VGP Workflow #2b

This workflow is meant for purging the pseudohaplotype primary/alternate assemblies generated by running `VGP Workflow #2: Long read assembly with hifiasm` using only PacBio HiFi reads. Because that workflow disables hifiasm's internal purging, there might still be false duplicates present in the assemblies. Determining purge_dups cutoffs via the Canu paper methology, this workflow first purges the primary alternate, adds those removed haplotigs to the alternate assembly, and then purges this (alternate+haplotig) assembly to create the purged primary assembly and the purged alternate assembly. This purged alternate assembly is done, while the purged primary assembly will continue to further scaffolding. 

## VGP Assembly Pipeline

The collaboration between the [VGP](https://vertebrategenomesproject.org/) has led to the creation of a set of workflows dedicated to high-quality genome assembly with long reads. The pipeline includes the following workflows:
-   Creation of k-mer count databases and histograms with Meryl
-   HiFi long read assembly with Hifiasm (three modes available: pseudohaplotype, HiC-phased, and trio)
-   Scaffolding with Bionano optical map data (optional)
-   Scaffolding with HiC data


### Inputs

-   Primary assembly as a fasta file
-   Alternate assembly as a fasta file
-   Cutadapt trimmed reads (as a collection), generated by the workflow `VGP #2: Long read assembly with hifiasm`
-   Estimated genome size parameter, generated by the workflow `VGP #2: Long read assembly with hifiasm`
-   GenomeScope model parameters, generated by the workflow `VGP #1: Meryl database creation`

### Outputs

-   Cutadapt trimmed reads (as a collection)
-   Purged primary assembly as a fasta (tagged `p1`)
    -   additionally, the sequences which were purged from the primary (tagged `seq_purged_p1`)
-   Purged alternate assembly as a fasta (tagged `p2`)
    -   additionally, the sequences which were purged from the alternate (tagged `seq_purged_p2`)
-   Purge_dups intermediates (for primary and alternate)
    -   coverage (tagged `depths`)
    -   PBCSTAT base coverage  (tagged `calcuts_basecov`)
    -   PBCSTAT base coverage (as .WIG) (tagged `calcuts_basecov_wig`)
    -   histogram plot (tagged `calcuts_hist`)
    -   calcuts log (tagged `calcuts_log`)
    -   calcuts cutoff (tagged `calcuts_cutoffs`)
    -   purge_dups regions (as .BED) (tagged `purgeoverlap_bed`)

