# Microbiome Workflows

In this directory, you will find a collection of workflows designed for microbiome data analysis, pathogen detection, and tracking. These workflows are ready to use and can be adapted for various sequencing techniques using Galaxy's customizable and automatable API.

## Avaiable Workflows

- **Nanopore Preprocessing**

- **Taxonomy Profiling and Visualisation with Krona**

- **Gene-based Pathogen Identification**

- **Allele-based Pathogen Identification**

- **Pathogen Detection: PathoGFAIR Samples Aggregation and Visualisation**

## Getting Started

To learn more about these workflows and to try them with real datasets, please visit our Microbiome tutorials on the Galaxy Training Network (GTN):

[Microbiome Tutorials on GTN](https://training.galaxyproject.org/training-material/topics/microbiome/)


## Dedicated Training Material

The workflows for **Nanopore Preprocessing**, **Taxonomy Profiling and Visualization with Krona**, **Gene-based Pathogen Identification**, **Allele-based Pathogen Identification**, and **Pathogen Detection: PathoGFAIR Samples Aggregation and Visualization** can all be tried out in a dedicated training material on GTN for foodborne pathogen detection and tracking:

[GTN Tutorial for Foodborne Pathogen Detection and Tracking](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)
