import{_ as z,f,s as w,o as v,w as R,r as S,G as A,H as k,v as g,C as K,a as X,P as G,e as Y,q as J,I as Z,J as ee,h as p,K as x,x as te,L as oe,M as ne,N as C,O as E,Q as ae,R as se,j as b,S as T,U as ie,V as N,W as _e,X as le,Y as ue,Z as h,$ as re,a0 as pe,m as j,a1 as y,a2 as ce,a3 as de,a4 as me}from"./DODj-B7H.js";const M=Object.freeze({ignoreUnknown:!1,respectType:!1,respectFunctionNames:!1,respectFunctionProperties:!1,unorderedObjects:!0,unorderedArrays:!1,unorderedSets:!1,excludeKeys:void 0,excludeValues:void 0,replacer:void 0});function D(t,s){s?s={...M,...s}:s=M;const a=L(s);return a.dispatch(t),a.toString()}const fe=Object.freeze(["prototype","__proto__","constructor"]);function L(t){let s="",a=new Map;const o=e=>{s+=e};return{toString(){return s},getContext(){return a},dispatch(e){return t.replacer&&(e=t.replacer(e)),this[e===null?"null":typeof e](e)},object(e){if(e&&typeof e.toJSON=="function")return this.object(e.toJSON());const n=Object.prototype.toString.call(e);let i="";const r=n.length;r<10?i="unknown:["+n+"]":i=n.slice(8,r-1),i=i.toLowerCase();let l=null;if((l=a.get(e))===void 0)a.set(e,a.size);else return this.dispatch("[CIRCULAR:"+l+"]");if(typeof Buffer<"u"&&Buffer.isBuffer&&Buffer.isBuffer(e))return o("buffer:"),o(e.toString("utf8"));if(i!=="object"&&i!=="function"&&i!=="asyncfunction")this[i]?this[i](e):t.ignoreUnknown||this.unkown(e,i);else{let u=Object.keys(e);t.unorderedObjects&&(u=u.sort());let c=[];t.respectType!==!1&&!I(e)&&(c=fe),t.excludeKeys&&(u=u.filter(_=>!t.excludeKeys(_)),c=c.filter(_=>!t.excludeKeys(_))),o("object:"+(u.length+c.length)+":");const d=_=>{this.dispatch(_),o(":"),t.excludeValues||this.dispatch(e[_]),o(",")};for(const _ of u)d(_);for(const _ of c)d(_)}},array(e,n){if(n=n===void 0?t.unorderedArrays!==!1:n,o("array:"+e.length+":"),!n||e.length<=1){for(const l of e)this.dispatch(l);return}const i=new Map,r=e.map(l=>{const u=L(t);u.dispatch(l);for(const[c,d]of u.getContext())i.set(c,d);return u.toString()});return a=i,r.sort(),this.array(r,!1)},date(e){return o("date:"+e.toJSON())},symbol(e){return o("symbol:"+e.toString())},unkown(e,n){if(o(n),!!e&&(o(":"),e&&typeof e.entries=="function"))return this.array(Array.from(e.entries()),!0)},error(e){return o("error:"+e.toString())},boolean(e){return o("bool:"+e)},string(e){o("string:"+e.length+":"),o(e)},function(e){o("fn:"),I(e)?this.dispatch("[native]"):this.dispatch(e.toString()),t.respectFunctionNames!==!1&&this.dispatch("function-name:"+String(e.name)),t.respectFunctionProperties&&this.object(e)},number(e){return o("number:"+e)},xml(e){return o("xml:"+e.toString())},null(){return o("Null")},undefined(){return o("Undefined")},regexp(e){return o("regex:"+e.toString())},uint8array(e){return o("uint8array:"),this.dispatch(Array.prototype.slice.call(e))},uint8clampedarray(e){return o("uint8clampedarray:"),this.dispatch(Array.prototype.slice.call(e))},int8array(e){return o("int8array:"),this.dispatch(Array.prototype.slice.call(e))},uint16array(e){return o("uint16array:"),this.dispatch(Array.prototype.slice.call(e))},int16array(e){return o("int16array:"),this.dispatch(Array.prototype.slice.call(e))},uint32array(e){return o("uint32array:"),this.dispatch(Array.prototype.slice.call(e))},int32array(e){return o("int32array:"),this.dispatch(Array.prototype.slice.call(e))},float32array(e){return o("float32array:"),this.dispatch(Array.prototype.slice.call(e))},float64array(e){return o("float64array:"),this.dispatch(Array.prototype.slice.call(e))},arraybuffer(e){return o("arraybuffer:"),this.dispatch(new Uint8Array(e))},url(e){return o("url:"+e.toString())},map(e){o("map:");const n=[...e];return this.array(n,t.unorderedSets!==!1)},set(e){o("set:");const n=[...e];return this.array(n,t.unorderedSets!==!1)},file(e){return o("file:"),this.dispatch([e.name,e.size,e.type,e.lastModfied])},blob(){if(t.ignoreUnknown)return o("[blob]");throw new Error(`Hashing Blob objects is currently not supported
Use "options.replacer" or "options.ignoreUnknown"
`)},domwindow(){return o("domwindow")},bigint(e){return o("bigint:"+e.toString())},process(){return o("process")},timer(){return o("timer")},pipe(){return o("pipe")},tcp(){return o("tcp")},udp(){return o("udp")},tty(){return o("tty")},statwatcher(){return o("statwatcher")},securecontext(){return o("securecontext")},connection(){return o("connection")},zlib(){return o("zlib")},context(){return o("context")},nodescript(){return o("nodescript")},httpparser(){return o("httpparser")},dataview(){return o("dataview")},signal(){return o("signal")},fsevent(){return o("fsevent")},tlswrap(){return o("tlswrap")}}}const B="[native code] }",ge=B.length;function I(t){return typeof t!="function"?!1:Function.prototype.toString.call(t).slice(-ge)===B}function P(t,s,a={}){return t===s||D(t,a)===D(s,a)}function be(t,s,a={}){const o=q(t,a),e=q(s,a);return O(o,e,a)}function O(t,s,a={}){var n,i;const o=[],e=new Set([...Object.keys(t.props||{}),...Object.keys(s.props||{})]);if(t.props&&s.props)for(const r of e){const l=t.props[r],u=s.props[r];l&&u?o.push(...O((n=t.props)==null?void 0:n[r],(i=s.props)==null?void 0:i[r],a)):(l||u)&&o.push(new V((u||l).key,l?"removed":"added",u,l))}return e.size===0&&t.hash!==s.hash&&o.push(new V((s||t).key,"changed",s,t)),o}function q(t,s,a=""){if(t&&typeof t!="object")return new F(a,t,D(t,s));const o={},e=[];for(const n in t)o[n]=q(t[n],s,a?`${a}.${n}`:n),e.push(o[n].hash);return new F(a,t,`{${e.join(":")}}`,o)}class V{constructor(s,a,o,e){this.key=s,this.type=a,this.newValue=o,this.oldValue=e}toString(){return this.toJSON()}toJSON(){var s;switch(this.type){case"added":return`Added   \`${this.key}\``;case"removed":return`Removed \`${this.key}\``;case"changed":return`Changed \`${this.key}\` from \`${((s=this.oldValue)==null?void 0:s.toString())||"-"}\` to \`${this.newValue.toString()}\``}}}class F{constructor(s,a,o,e){this.key=s,this.value=a,this.hash=o,this.props=e}toString(){return this.props?`{${Object.keys(this.props).join(",")}}`:JSON.stringify(this.value)}toJSON(){const s=this.key||".";return this.props?`${s}({${Object.keys(this.props).join(",")}})`:`${s}(${this.value})`}}const U={to:{type:[String,Object],default:void 0,required:!1},href:{type:[String,Object],default:void 0,required:!1},target:{type:String,default:void 0,required:!1},rel:{type:String,default:void 0,required:!1},noRel:{type:Boolean,default:void 0,required:!1},prefetch:{type:Boolean,default:void 0,required:!1},noPrefetch:{type:Boolean,default:void 0,required:!1},activeClass:{type:String,default:void 0,required:!1},exactActiveClass:{type:String,default:void 0,required:!1},prefetchedClass:{type:String,default:void 0,required:!1},replace:{type:Boolean,default:void 0,required:!1},ariaCurrentValue:{type:String,default:void 0,required:!1},external:{type:Boolean,default:void 0,required:!1}},Te=t=>Object.keys(U).reduce((a,o)=>(t[o]!==void 0&&(a[o]=t[o]),a),{}),he=f({inheritAttrs:!1,props:{...U,as:{type:String,default:"button"},type:{type:String,default:"button"},disabled:{type:Boolean,default:null},active:{type:Boolean,default:void 0},exact:{type:Boolean,default:!1},exactQuery:{type:[Boolean,String],default:!1},exactHash:{type:Boolean,default:!1},inactiveClass:{type:String,default:void 0}},setup(t){function s(o,e){const n=be(o,e).reduce((i,r)=>(r.type==="added"&&i.push(r.key),i),[]);return P(o,e,{excludeKeys:i=>n.includes(i)})}function a(o,e,{isActive:n,isExactActive:i}){if(t.exactQuery==="partial"){if(!s(o.query,e.query))return t.inactiveClass}else if(t.exactQuery===!0&&!P(o.query,e.query))return t.inactiveClass;return t.exactHash&&o.hash!==e.hash?t.inactiveClass:t.exact&&i||!t.exact&&n?t.activeClass:t.inactiveClass}return{resolveLinkClass:a}}}),ye=["href","aria-disabled","role","rel","target","tabindex","onClick"];function xe(t,s,a,o,e,n){const i=Y;return t.to?(v(),w(i,g({key:1},t.$props,{custom:""}),{default:R(({route:r,href:l,target:u,rel:c,navigate:d,isActive:_,isExactActive:m,isExternal:W})=>[X("a",g(t.$attrs,{href:t.disabled?void 0:l,"aria-disabled":t.disabled?"true":void 0,role:t.disabled?"link":void 0,rel:c,target:u,class:t.active!==void 0?t.active?t.activeClass:t.inactiveClass:t.resolveLinkClass(r,t._.provides[G]||t.$route,{isActive:_,isExactActive:m}),tabindex:t.disabled?-1:void 0,onClick:$=>!W&&!t.disabled&&d($)}),[S(t.$slots,"default",A(k({isActive:t.active!==void 0?t.active:t.exact?m:_})))],16,ye)]),_:3},16)):(v(),w(K(t.as),g({key:0,type:t.type,disabled:t.disabled},t.$attrs,{class:t.active?t.activeClass:t.inactiveClass}),{default:R(()=>[S(t.$slots,"default",A(k({isActive:t.active})))]),_:3},16,["type","disabled","class"]))}const Re=z(he,[["render",xe]]),we=f({props:{name:{type:String,required:!0},mode:{type:String,required:!1,default:null},size:{type:[Number,String],required:!1,default:null},customize:{type:Function,required:!1,default:null}}});function ve(t,s,a,o,e,n){const i=J;return v(),w(i,A(k(t.$props)),null,16)}const Se=z(we,[["render",ve]]),je=(t,s,a,o,e=!1)=>{const n=Z(),i=ee(),r=p(()=>{var _;const u=x(s),c=x(a),d=x(o);return te((u==null?void 0:u.strategy)||((_=i.ui)==null?void 0:_.strategy),d?{wrapper:d}:{},u||{},e?oe(i.ui,t,{}):{},c||{})}),l=p(()=>ne(n,["class"]));return{ui:r,attrs:l}};function Me(t){const a=`group-${E().uid}`,o=b({children:[],register(e){this.children.push(e)},unregister(e){const n=this.children.indexOf(e);n>-1&&this.children.splice(n,1)},...t});T(a,o)}function Ie({ui:t,props:s}){const a=E();if(T("ButtonGroupContextConsumer",!0),C("ButtonGroupContextConsumer",!1))return{size:p(()=>s.size),rounded:p(()=>t.value.rounded)};let e=a.parent,n;for(;e&&!n;){if(e.type.name==="ButtonGroup"){n=C(`group-${e.uid}`);break}e=e.parent}const i=p(()=>n==null?void 0:n.value.children.indexOf(a));return ae(()=>{n==null||n.value.register(a)}),se(()=>{n==null||n.value.unregister(a)}),{size:p(()=>n!=null&&n.value?(n==null?void 0:n.value.size)??t.value.default.size:s.size),rounded:p(()=>!n||i.value===-1?t.value.rounded:n.value.children.length===1?n.value.ui.rounded:i.value===0?n.value.rounded.start:i.value===n.value.children.length-1?n.value.rounded.end:"rounded-none")}}const Q={base:"invisible before:visible before:block before:rotate-45 before:z-[-1] before:w-2 before:h-2",ring:"before:ring-1 before:ring-gray-200 dark:before:ring-gray-800",rounded:"before:rounded-sm",background:"before:bg-gray-200 dark:before:bg-gray-800",shadow:"before:shadow",placement:"group-data-[popper-placement*='right']:-left-1 group-data-[popper-placement*='left']:-right-1 group-data-[popper-placement*='top']:-bottom-1 group-data-[popper-placement*='bottom']:-top-1"},H={container:"z-20 group",trigger:"flex items-center w-full",width:"w-full",height:"max-h-60",base:"relative focus:outline-none overflow-y-auto scroll-py-1",background:"bg-white dark:bg-gray-800",shadow:"shadow-lg",rounded:"rounded-md",padding:"p-1",ring:"ring-1 ring-gray-200 dark:ring-gray-700",empty:"text-sm text-gray-400 dark:text-gray-500 px-2 py-1.5",option:{base:"cursor-default select-none relative flex items-center justify-between gap-1",rounded:"rounded-md",padding:"px-1.5 py-1.5",size:"text-sm",color:"text-gray-900 dark:text-white",container:"flex items-center gap-1.5 min-w-0",active:"bg-gray-100 dark:bg-gray-900",inactive:"",selected:"pe-7",disabled:"cursor-not-allowed opacity-50",empty:"text-sm text-gray-400 dark:text-gray-500 px-2 py-1.5",icon:{base:"flex-shrink-0 h-5 w-5",active:"text-gray-900 dark:text-white",inactive:"text-gray-400 dark:text-gray-500"},selectedIcon:{wrapper:"absolute inset-y-0 end-0 flex items-center",padding:"pe-2",base:"h-5 w-5 text-gray-900 dark:text-white flex-shrink-0"},avatar:{base:"flex-shrink-0",size:"2xs"},chip:{base:"flex-shrink-0 w-2 h-2 mx-1 rounded-full"}},transition:{leaveActiveClass:"transition ease-in duration-100",leaveFromClass:"opacity-100",leaveToClass:"opacity-0"},popper:{placement:"bottom-end"},default:{selectedIcon:"i-heroicons-check-20-solid",trailingIcon:"i-heroicons-chevron-down-20-solid",empty:{label:"No options."},optionEmpty:{label:'No results for "{query}".'}},arrow:{...Q,ring:"before:ring-1 before:ring-gray-200 dark:before:ring-gray-700",background:"before:bg-white dark:before:bg-gray-700"}},Pe={...H,select:"inline-flex items-center text-left cursor-default",input:"block w-[calc(100%+0.5rem)] focus:ring-transparent text-sm px-3 py-1.5 text-gray-700 dark:text-gray-200 bg-white dark:bg-gray-800 border-0 border-b border-gray-200 dark:border-gray-700 sticky -top-1 -mt-1 mb-1 -mx-1 z-10 placeholder-gray-400 dark:placeholder-gray-500 focus:outline-none",required:"absolute inset-0 w-px opacity-0 cursor-default",label:"block truncate",option:{...H.option,create:"block truncate"},transition:{leaveActiveClass:"transition ease-in duration-100",leaveFromClass:"opacity-100",leaveToClass:"opacity-0"},popper:{placement:"bottom-end"},default:{selectedIcon:"i-heroicons-check-20-solid",clearSearchOnClose:!1,showCreateOptionWhen:"empty",searchablePlaceholder:{label:"Search..."},empty:{label:"No options."},optionEmpty:{label:'No results for "{query}".'}},arrow:{...Q,ring:"before:ring-1 before:ring-gray-200 dark:before:ring-gray-700",background:"before:bg-white dark:before:bg-gray-700"}};function Ve(t){return new Date(t).toLocaleDateString(void 0,{year:"numeric",month:"short",day:"numeric"})}const Ae=JSON.parse('[{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/sra-manifest-to-concatenated-fastqs.ga","testParameterFiles":["/sra-manifest-to-concatenated-fastqs-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"},{"name":"Pierre Osteil","orcid":"0000-0002-5832-6703"},{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow takes as input a SRA_manifest from SRA Run Selector and will generate one fastq file or fastq pair of file for each experiment (concatenated multiple runs if necessary). Output will be relabelled to match the column specified by the user.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-5832-6703","name":"Pierre Osteil"},{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","release":"0.7","license":"MIT","name":"sra_manifest_to_concatenated_fastqs_parallel","steps":{"0":{"annotation":"Input tabular from SRA Run Selector or home made (First row needs to be a header)","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Input tabular from SRA Run Selector or home made (First row needs to be a header)","name":"SRA_manifest"}],"label":"SRA_manifest","name":"Input dataset","outputs":[],"position":{"left":0,"top":44.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"7125f877-571a-4f10-b9de-48b750cf7964","when":null,"workflow_outputs":[]},"1":{"annotation":"Column number in the SRA_manifest with SRA ID (usually 1)","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Column number in the SRA_manifest with SRA ID (usually 1)","name":"Column number with SRA ID"}],"label":"Column number with SRA ID","name":"Input parameter","outputs":[],"position":{"left":23,"top":145},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"de844377-af66-4f4e-bc67-327e9705d5b5","when":null,"workflow_outputs":[]},"2":{"annotation":"Column number in the SRA_list with final identifier","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Column number in the SRA_list with final identifier","name":"Column number with final identifier"}],"label":"Column number with final identifier","name":"Input parameter","outputs":[],"position":{"left":59,"top":243},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3f0fca4f-b920-4ae6-bda8-8ef61ebbf409","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":3,"input_connections":{"input_param_type|input_param":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Set SRA column to 1 if at 0","name":"Map parameter value","outputs":[{"name":"output_param_integer","type":"expression.json"}],"position":{"left":305,"top":152.5},"post_job_actions":{"HideDatasetActionoutput_param_integer":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_integer"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"integer\\", \\"__current_case__\\": 1, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"0\\", \\"to\\": \\"1\\"}]}, \\"output_param_type\\": \\"integer\\", \\"unmapped\\": {\\"on_unmapped\\": \\"input\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"98c1bbbc-35f3-4a9e-ad2d-1a8420b827b5","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":4,"input_connections":{"components_1|param_type|component_value":{"id":3,"output_name":"output_param_integer"},"components_3|param_type|component_value":{"id":2,"output_name":"output"}},"inputs":[],"label":"Compute column expression","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":527,"top":227},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\",c\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"f93a3f4b-c1f8-45f7-b44f-b42fe0aa1856","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"Cut1","errors":null,"id":5,"input_connections":{"columnList":{"id":4,"output_name":"out1"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"Cut columns of interest","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":816,"top":0},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"dc7f7cb5-cc22-4ec8-8fe9-025f8f265e31","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":6,"input_connections":{"infile":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":"generate table for relabelling","name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1096,"top":189},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"[^\\\\\\\\w\\\\\\\\- .,\\\\\\\\t](?=[^\\\\\\\\t]+$)\\", \\"replace_pattern\\": \\"-\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}, {\\"__index__\\": 1, \\"find_pattern\\": \\"(.+)\\\\\\\\t(.+)\\", \\"replace_pattern\\": \\"$1\\\\\\\\t$1___$2\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"7b4781ed-57b0-4962-a647-21e54375c195","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":"Cut to get only SRA","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1347.75,"top":4.48333740234375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input|__identifier__\\": \\"SRR031709\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"f0405bd8-19f0-4a3d-afe9-3cfd1a0f01a3","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":8,"input_connections":{"split_parms|input":{"id":7,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split file to get one SRA per file + header","name":"Split file","outputs":[{"name":"list_output_tab","type":"input"}],"position":{"left":1645,"top":155.5},"post_job_actions":{"HideDatasetActionlist_output_tab":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"top\\": \\"1\\", \\"split_by\\": {\\"select_split_by\\": \\"col\\", \\"__current_case__\\": 0, \\"id_col\\": \\"1\\", \\"match_regex\\": \\"(.*)\\", \\"sub_regex\\": \\"\\\\\\\\1\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"4a1ad675-4291-4001-a748-d009299f9d40","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0","errors":null,"id":9,"input_connections":{"input|file_list":{"id":8,"output_name":"list_output_tab"}},"inputs":[{"description":"runtime parameter for tool Faster Download and Extract Reads in FASTQ","name":"input"}],"label":"get Fastqs from SRA IDs","name":"Faster Download and Extract Reads in FASTQ","outputs":[{"name":"list_paired","type":"input"},{"name":"output_collection","type":"input"},{"name":"output_collection_other","type":"input"},{"name":"log","type":"txt"}],"position":{"left":1920,"top":435.5},"post_job_actions":{"HideDatasetActionlist_paired":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_paired"},"HideDatasetActionlog":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log"},"HideDatasetActionoutput_collection":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_collection"},"HideDatasetActionoutput_collection_other":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_collection_other"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"516a54ddf218","name":"sra_tools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"seq_defline\\": \\"@$sn/$ri\\", \\"minlen\\": null, \\"split\\": \\"--split-3\\", \\"skip_technical\\": true}, \\"input\\": {\\"input_select\\": \\"file_list\\", \\"__current_case__\\": 2, \\"file_list\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"fa409017-1eea-4426-83f6-22a10b9f762c","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":10,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":9,"output_name":"list_paired"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"relabel pair collec to get SRA+sample","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":2159,"top":329.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"tabular\\", \\"__current_case__\\": 1, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ccdc51b2-444e-4b96-90a0-c7caba6a53e8","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":11,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":9,"output_name":"output_collection"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"relabel single collec to get SRA+sample","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":2138,"top":718.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"tabular\\", \\"__current_case__\\": 1, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"4efa4ed9-6768-4b1e-82f6-055e97af2d2d","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":2390,"top":359.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [3, 1], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}, {\\"columns\\": [2], \\"type\\": \\"paired_identifier\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier2\\", \\"warn\\": null}, {\\"error\\": null, \\"expression\\": \\"(.*?)___(.*)\\", \\"group_count\\": null, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"9d65a5a8-dedb-4d63-9164-47ceefc746b3","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":13,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":2364,"top":738.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [2, 0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}, {\\"error\\": null, \\"expression\\": \\"(.*?)___(.*)\\", \\"group_count\\": null, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"3938066e-faa5-469e-96c0-d4ede1fb5f11","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.3","errors":null,"id":14,"input_connections":{"global_condition|inputs":{"id":12,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Concatenate multiple datasets","name":"global_condition"}],"label":null,"name":"Concatenate multiple datasets","outputs":[{"name":"paired_output","type":"input"}],"position":{"left":2617,"top":347.5},"post_job_actions":{"RenameDatasetActionpaired_output":{"action_arguments":{"newname":"Concatenated paired-end"},"action_type":"RenameDatasetAction","output_name":"paired_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.3","tool_shed_repository":{"changeset_revision":"5b1b635232ed","name":"concatenate_multiple_datasets","owner":"artbio","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"fastqsanger.gz\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"dataset_names\\": false, \\"global_condition\\": {\\"input_type\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paired_cat_type\\": \\"by_strand\\"}, \\"headers\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.3","type":"tool","uuid":"37b6fecb-1c87-4c41-8598-95d43b418ee0","when":null,"workflow_outputs":[{"label":"paired_output","output_name":"paired_output","uuid":"8abbbe2a-9b3e-4472-81ec-a8fbede8a044"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.3","errors":null,"id":15,"input_connections":{"global_condition|inputs":{"id":13,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Concatenate multiple datasets","name":"global_condition"}],"label":null,"name":"Concatenate multiple datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2595,"top":719.5},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Concatenated Single-read"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.3","tool_shed_repository":{"changeset_revision":"5b1b635232ed","name":"concatenate_multiple_datasets","owner":"artbio","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"fastqsanger.gz\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"dataset_names\\": false, \\"global_condition\\": {\\"input_type\\": \\"singles\\", \\"__current_case__\\": 0, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"headers\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.3","type":"tool","uuid":"49a70885-3374-4680-bbd2-ad57b8be543d","when":null,"workflow_outputs":[{"label":"single_output","output_name":"out_file1","uuid":"71cbe11d-24ff-4e86-954f-acb439332d0a"}]}},"tags":[],"uuid":"363898af-e598-4f0e-abd9-e6ded395ce66","version":3},"readme":"# SRA manifest to concatenated fastqs\\n\\nThis workflow takes as input a SRA manifest from SRA Run Selector (or a tabular with a header line), downloads all sequencing run data from the SRA and arranges it into per-sample fastq or pairs of fastq datasets.\\n\\nIt will work out the relationship between runs and samples from the user-indicated run and sample columns in the input and will concatenate sequencing run data as needed to obtain per-sample datasets.\\n\\n## Input dataset\\n\\n- The workflow needs a single tabular input dataset, which is supposed to list SRA run identifiers in one column and sample names in another, and which needs to have a header line.\\n- SRA manifests obtained via the SRA Run Selector and turned into tabular format represent valid input.\\n\\n## Input values\\n\\n- Column number with SRA run ID\\n\\n  For manifests obtained through the SRA Run Selector this is column 1\\n\\n- Column number with sample names\\n\\n  The number of the column that should be used to assign sequencing runs to samples\\n  The names in the column will also serve as the labels of datasets in the output collection.\\n  For manifests obtained through the SRA Run Selector suitable columns might be number 6 (BioSample), 16 (Experiment) or 36 (Sample Name).\\n\\n## Processing\\n\\n- The workflow downloads sequencing run data in fastq format with fasterqdump (one job per SRA run ID).\\n- Run data gets concatenated if it comes from the same sample.\\n\\n## Outputs\\n\\n- There are 2 outputs, one with paired-end datasets, one with single-read datasets.\\n\\n## Limitations\\n\\n- Special characters in sample names (anything that is not an English alphabet character, digit, underscore, dash, space, dot or comma (`[a-zA-Z0-9_\\\\- \\\\.,]`) will be converted to dashes (`-`).\\n","changelog":"# Changelog\\n\\n## [0.7] - 2024-06-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.2` was updated to `toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.3`\\n\\n## [0.6] - 2024-06-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0`\\n\\n## [0.5] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy1`\\n\\n## [0.4] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.10+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy0`\\n\\n## [0.3] - 2024-03-11\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.10+galaxy0`\\n\\n## [0.2.4] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0`\\n\\n## [0.2.3] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0`\\n\\n## [0.2.2] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n\\n## [0.2.1] - 2023-11-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy1`\\n\\n## [0.2] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.1` was updated to `toolshed.g2.bx.psu.edu/repos/artbio/concatenate_multiple_datasets/cat_multi_datasets/1.4.2`\\n\\n## [0.1] - 2023-10-23\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## sra_manifest_to_concatenated_fastqs_parallel\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ SRA_manifest\\"]@{ shape: doc }\\n1[\\"ℹ️ Column number with SRA ID\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Column number with final identifier\\"]@{ shape: lean-l }\\n3[\\"Set SRA column to 1 if at 0\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Compute column expression\\"]@{ shape: process }\\n3 --> 4\\n2 --> 4\\n5[\\"Cut columns of interest\\"]@{ shape: process }\\n4 --> 5\\n0 --> 5\\n6[\\"generate table for relabelling\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Cut to get only SRA\\"]@{ shape: process }\\n5 --> 7\\n8[\\"split file to get one SRA per file + header\\"]@{ shape: process }\\n7 --> 8\\n9[\\"get Fastqs from SRA IDs\\"]@{ shape: process }\\n8 --> 9\\n10[\\"relabel pair collec to get SRA+sample\\"]@{ shape: process }\\n6 --> 10\\n9 --> 10\\n11[\\"relabel single collec to get SRA+sample\\"]@{ shape: process }\\n6 --> 11\\n9 --> 11\\n12[\\"Apply rules\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Apply rules\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Concatenate multiple datasets\\"]@{ shape: process }\\n12 --> 14\\n15[\\"Concatenate multiple datasets\\"]@{ shape: process }\\n13 --> 15\\n```\\n","updated":"2024-06-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sra-manifest-to-concatenated-fastqs/main","dockstore_id":24098,"categories":[],"collections":["Data Fetching"],"doi":"10.5281/zenodo.15078895","tests":[{"doc":"Test for sra-list-to-concatenated-fastqs.ga","job":{"SRA_manifest":{"class":"File","path":"test-data/SRA.txt"},"Column number with SRA ID":1,"Column number with final identifier":22},"outputs":{"paired_output":{"element_tests":{"GSM461177-":{"element_tests":{"forward":{"asserts":{"has_size":{"value":294000000,"delta":30000000}}},"reverse":{"asserts":{"has_size":{"value":307000000,"delta":30000000}}}}},"GSM461178___a":{"element_tests":{"forward":{"asserts":{"has_size":{"value":178000000,"delta":10000000}}},"reverse":{"asserts":{"has_size":{"value":205000000,"delta":20000000}}}}}}},"single_output":{"element_tests":{"GSM461176.-":{"asserts":{"has_size":{"value":139000000,"delta":10000000}}},"GSM461179-ID":{"asserts":{"has_size":{"value":298000000,"delta":30000000}}}}}}}]}],"path":"./workflows/data-fetching/sra-manifest-to-concatenated-fastqs"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/parallel-accession-download.ga","testParameterFiles":["/parallel-accession-download-tests.yml"],"authors":[{"name":"Marius van den Beek","orcid":"0000-0002-9676-7032"},{"name":"IWC","url":"https://github.com/galaxyproject/iwc"}],"definition":{"a_galaxy_workflow":"true","annotation":"Downloads fastq files for sequencing run accessions provided in a text file using fasterq-dump. Creates one job per listed run accession.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9676-7032","name":"Marius van den Beek"},{"class":"Organization","name":"IWC","url":"https://github.com/galaxyproject/iwc"}],"format-version":"0.1","license":"MIT","release":"0.1.14","name":"Parallel Accession Download","steps":{"0":{"annotation":"Text file containing run accessions (starting with SRR, ERR or DRR), one per line.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Text file containing run accessions (starting with SRR, ERR or DRR), one per line.","name":"Run accessions"}],"label":"Run accessions","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"txt\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e9e5605e-29e1-4f90-9693-0e40a2ddfd8f","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":1,"input_connections":{"split_parms|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"Split accessions to collection","name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":279.51666259765625,"top":86.01666259765625},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"f8d776ff-e1ba-4bde-9b4b-392d825fd2b7","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0","errors":null,"id":2,"input_connections":{"input|file_list":{"id":1,"output_name":"list_output_txt"}},"inputs":[{"description":"runtime parameter for tool Faster Download and Extract Reads in FASTQ","name":"input"}],"label":"fasterq-dump","name":"Faster Download and Extract Reads in FASTQ","outputs":[{"name":"list_paired","type":"input"},{"name":"output_collection","type":"input"},{"name":"output_collection_other","type":"input"},{"name":"log","type":"txt"}],"position":{"left":570.9250183105469,"top":62.9749755859375},"post_job_actions":{"HideDatasetActionlist_paired":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_paired"},"HideDatasetActionlog":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log"},"HideDatasetActionoutput_collection":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_collection"},"HideDatasetActionoutput_collection_other":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_collection_other"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"516a54ddf218","name":"sra_tools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"seq_defline\\": \\"@$sn/$ri\\", \\"minlen\\": null, \\"split\\": \\"--split-3\\", \\"skip_technical\\": true}, \\"input\\": {\\"input_select\\": \\"file_list\\", \\"__current_case__\\": 2, \\"file_list\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"4d328e7b-df7d-4d3e-a60f-1ea12925f317","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"list_paired"}},"inputs":[],"label":"flatten paired output","name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":898.0000305175781,"top":71.9749755859375},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"name:PE"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [2], \\"connectable\\": true, \\"is_workflow\\": false, \\"type\\": \\"paired_identifier\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier2\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"924d2152-5640-40cd-bec3-0ced45469d84","when":null,"workflow_outputs":[{"label":"Paired End Reads","output_name":"output","uuid":"b431f0a1-391e-45e1-b739-27dd0fc83087"}]},"4":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"output_collection"}},"inputs":[],"label":"flatten single end output","name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":905.0499572753906,"top":215.0250244140625},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"name:SE"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"946d9610-c363-4915-b93a-f074169ef812","when":null,"workflow_outputs":[{"label":"Single End Reads","output_name":"output","uuid":"e2bf276c-0956-473e-a778-b3b3df7351f5"}]}},"tags":[],"uuid":"15c3229d-8786-457f-a020-7aa64c0cf6e8","version":10},"readme":"# Parallel Accession Download\\n\\nDownloads fastq files for sequencing run accessions provided in a text file\\nusing fasterq-dump. Creates one job per listed run accession, and is therefore\\nmuch faster and more robust to errors when many accessions need to be\\ndownloaded.\\n","changelog":"# Changelog\\n\\n## [0.1.14] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.1+galaxy0`\\n\\n## [0.1.13] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy1`\\n\\n## [0.1.12] - 2024-04-16\\n\\n### Changed\\n\\n- Remove unnecessary param_value_from_file step and directly pass file with identifiers to fasterqdump\\n\\n## [0.1.11] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.10+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.1.0+galaxy0`\\n\\n## [0.1.10] - 2024-03-11\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.10+galaxy0`\\n\\n## [0.1.9] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n\\n## [0.1.8] - 2023-11-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy1`\\n\\n## [0.1.7] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.8+galaxy0`\\n\\n## [0.1.6] - 2023-09-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy3`\\n\\n## [0.1.5] - 2023-09-12\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy1`\\n\\n## [0.1.4] - 2023-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.11.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/3.0.5+galaxy0`\\n\\n## [0.1.3] - 2022-02-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.11.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.11.0+galaxy1`\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.1] - 2021-06-23\\n\\n### Added\\n\\n- Initial version of Parallel Accession Download workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## Parallel Accession Download\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Run accessions\\"]@{ shape: doc }\\n1[\\"Split accessions to collection\\"]@{ shape: process }\\n0 --> 1\\n2[\\"fasterq-dump\\"]@{ shape: process }\\n1 --> 2\\n3[\\"flatten paired output\\"]@{ shape: process }\\n2 --> 3\\n4[\\"flatten single end output\\"]@{ shape: process }\\n2 --> 4\\n```\\n","updated":"2024-05-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/parallel-accession-download/main","dockstore_id":16789,"categories":[],"collections":["Data Fetching"],"doi":"10.5281/zenodo.15078771","tests":[{"doc":"Test download of single end accession","job":{"Run accessions":{"class":"File","path":"test-data/input_accession_single_end.txt"}},"outputs":{"Single End Reads":{"element_tests":{"SRR044777":{"file":"test-data/SRR044777_head.fastq","decompress":true,"compare":"contains"}}}}},{"doc":"Test download of single and paired end accession","job":{"Run accessions":{"class":"File","path":"test-data/input_accession_paired_end.txt"}},"outputs":{"Paired End Reads":{"element_tests":{"SRR11953971":{"elements":{"forward":{"file":"test-data/SRR11953971_1_head.fastq","decompress":true,"compare":"contains"},"reverse":{"file":"test-data/SRR11953971_2_head.fastq","decompress":true,"compare":"contains"}}}}}}}]}],"path":"./workflows/data-fetching/parallel-accession-download"},{"version":1.2,"workflows":[{"name":"Velocyto-on10X-from-bundled","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Velocyto-on10X-from-bundled.ga","testParameterFiles":["/Velocyto-on10X-from-bundled-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Automatically extracts cell barcodes from standard 10X bundled output and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output","steps":{"0":{"annotation":"This can be output of CellRanger or STARsolo","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"This can be output of CellRanger or STARsolo","name":"BAM files with CB and UB"}],"label":"BAM files with CB and UB","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"043d198b-3af0-477d-be11-5d1373280379","when":null,"workflow_outputs":[]},"1":{"annotation":"This can be output of STARsolo or DropletUtils","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This can be output of STARsolo or DropletUtils","name":"filtered matrices in bundle"}],"label":"filtered matrices in bundle","name":"Input dataset collection","outputs":[],"position":{"left":141,"top":139},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"03b0d22c-3b46-4460-8c18-b524bc09d655","when":null,"workflow_outputs":[]},"2":{"annotation":"gtf file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"gtf file","name":"gtf file"}],"label":"gtf file","name":"Input dataset","outputs":[],"position":{"left":231,"top":273},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gtf\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f1d04259-f449-44d7-8d36-49f3ad664361","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":3,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Apply rules","name":"input"}],"label":"extract barcodes from bundle","name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":421,"top":133},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}, {\\"error\\": null, \\"invert\\": false, \\"target_column\\": 1, \\"type\\": \\"add_filter_matches\\", \\"value\\": \\"barcodes\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"8c2d032f-596c-4315-8361-ebf1ec2eb655","when":null,"workflow_outputs":[]},"4":{"annotation":"","id":4,"input_connections":{"BAM files with CB and UB":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"filtered barcodes":{"id":3,"input_subworkflow_step_id":1,"output_name":"output"},"gtf file":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Velocyto_on10X_filtered_barcodes","outputs":[],"position":{"left":710,"top":63.5},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Velocyto_on10X_filtered_barcodes","steps":{"0":{"annotation":"This can be output of CellRanger or STARsolo","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"This can be output of CellRanger or STARsolo","name":"BAM files with CB and UB"}],"label":"BAM files with CB and UB","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"043d198b-3af0-477d-be11-5d1373280379","when":null,"workflow_outputs":[]},"1":{"annotation":"This can be output of STARsolo or DropletUtils","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This can be output of STARsolo or DropletUtils","name":"filtered barcodes"}],"label":"filtered barcodes","name":"Input dataset collection","outputs":[],"position":{"left":55.999999999999986,"top":108.33333333333333},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tsv\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"03b0d22c-3b46-4460-8c18-b524bc09d655","when":null,"workflow_outputs":[]},"2":{"annotation":"gtf file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"gtf file","name":"gtf file"}],"label":"gtf file","name":"Input dataset","outputs":[],"position":{"left":148,"top":204.66666666666666},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gtf\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f1d04259-f449-44d7-8d36-49f3ad664361","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2","errors":null,"id":3,"input_connections":{"main|BAM":{"id":0,"output_name":"output"},"main|barcodes":{"id":1,"output_name":"output"},"main|gtffile":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"}],"label":"velocyto","name":"velocyto CLI","outputs":[{"name":"samples","type":"loom"}],"position":{"left":470.66666666666663,"top":116.32222493489576},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2","tool_shed_repository":{"changeset_revision":"90e95e9ee190","name":"velocyto_cli","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"main\\": {\\"do\\": \\"run10x\\", \\"__current_case__\\": 0, \\"sample_definition\\": {\\"sample_definition_select\\": \\"identifier\\", \\"__current_case__\\": 1}, \\"BAM\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gtffile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"s\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"m\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"M\\": false, \\"t\\": \\"uint16\\"}, \\"verbosity\\": \\"-vv\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.17.17+galaxy2","type":"tool","uuid":"f9aef50f-b31d-44dd-aad6-fe4d520e3b1e","when":null,"workflow_outputs":[{"label":"velocyto loom","output_name":"samples","uuid":"2899a85c-c198-436f-a409-9bfbdf90c95f"}]}},"tags":"","uuid":"66e3b8eb-d312-488e-b407-1ed3fa627ebb"},"tool_id":null,"type":"subworkflow","uuid":"39ea418d-135d-4e7d-b2fe-fbde5c7393b7","when":null,"workflow_outputs":[{"label":"velocyto loom","output_name":"velocyto loom","uuid":"865e6dd6-8c01-4188-a1f6-4cd77e433eed"}]}},"tags":["name:single-cell"],"uuid":"0608c7f4-767c-47b3-ae45-4e98c7c83bec","version":8},"readme":"# RNA Velocity Analysis: Velocyto for 10X Genomics Data\\n\\nThese workflows simply run velocyto. There are 2 workflows because one can be easily run after the Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat-Compatible Format\' workflows (RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output). The other can be easily run from uploaded datasets (RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes).\\n\\n## Input datasets\\n\\n- BAM files with CB and UB: A collection of BAM. It accepts BAM from cellranger or STARsolo with the CB and UB tags (if you use the fastq-to-matrix-10x workflows these tags are automatically included).\\n- filtered barcodes (only for Velocyto_on10X_filtered_barcodes workflow): A collection of filtered barcodes (this is what will be used by velocyto). \'Filtered\' means that these barcodes have been identified as potential cells. It should not be the whole list of 3 million possible barcodes from cellranger.\\n- filtered matrices in bundle (only for Velocyto_on10X_from_bundled workflow): A collection of filtered matrices as bundled (like the one which comes from the fastq-to-matrix-10x workflows): A collection with as many items as samples. For each sample, the item is a list with 3 datasets (barcodes, genes, matrix). The workflow will then extract the items which have the \'barcodes\' identifier.\\n- gtf file: A file with annotations where exons are and how they are grouped into genes.\\n\\n## Processing\\n\\n- If you provided matrices, the first step is to extract barcodes.\\n- For both cases velocyto cli is run to get a loom file per sample with spliced and unspliced counts.\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2`\\n## [0.1] - 2024-01-26\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ BAM files with CB and UB\\"]@{ shape: docs }\\n1[\\"ℹ️ filtered matrices in bundle\\"]@{ shape: docs }\\n2[\\"ℹ️ gtf file\\"]@{ shape: doc }\\n3[\\"extract barcodes from bundle\\"]@{ shape: process }\\n1 --> 3\\n4[\\"🛠️ Velocyto_on10X_filtered_barcodes\\"]@{ shape: subprocess }\\n0 --> 4\\n3 --> 4\\n2 --> 4\\n```\\n\\n## Velocyto_on10X_filtered_barcodes\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ BAM files with CB and UB\\"]@{ shape: docs }\\n1[\\"ℹ️ filtered barcodes\\"]@{ shape: docs }\\n2[\\"ℹ️ gtf file\\"]@{ shape: doc }\\n3[\\"velocyto\\"]@{ shape: process }\\n0 --> 3\\n1 --> 3\\n2 --> 3\\n```\\n","updated":"2024-02-05T00:00:00","trsID":"#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-from-bundled","dockstore_id":25201,"categories":[],"collections":["Single Cell"],"doi":"10.5281/zenodo.15078509","tests":[{"doc":"Test outline for Velocyto-on10X (bundled)","job":{"gtf file":{"class":"File","location":"https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz","decompress":true,"filetype":"gtf"},"BAM files with CB and UB":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"subsample","location":"https://zenodo.org/records/10572348/files/subsample.bam","filetype":"bam"}]},"filtered matrices in bundle":{"class":"Collection","collection_type":"list:list","elements":[{"class":"Collection","type":"list","identifier":"subsample","elements":[{"class":"File","identifier":"barcodes","location":"https://zenodo.org/records/10572348/files/barcodes.tsv","filetype":"tsv"},{"class":"File","identifier":"genes","location":"https://zenodo.org/records/10572348/files/genes.tsv","filetype":"tsv"},{"class":"File","identifier":"matrix","location":"https://zenodo.org/records/10572348/files/matrix.mtx","filetype":"mtx"}]}]}},"outputs":{"velocyto loom":{"element_tests":{"subsample":{"asserts":{"has_size":{"value":4639326,"delta":400000}}}}}}}]},{"name":"Velocyto-on10X-filtered-barcodes","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Velocyto-on10X-filtered-barcodes.ga","testParameterFiles":["/Velocyto-on10X-filtered-barcodes-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Processes 10X Genomics single-cell RNA-seq data using Velocyto to quantify spliced and unspliced transcript counts for RNA velocity analysis. Takes pre-filtered cell barcodes as input and generates a loom file containing separate counts for spliced exons, unspliced introns, and ambiguous regions. Enables downstream trajectory inference and cellular dynamics analysis in tools like scVelo.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes","steps":{"0":{"annotation":"This can be output of CellRanger or STARsolo","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"This can be output of CellRanger or STARsolo","name":"BAM files with CB and UB"}],"label":"BAM files with CB and UB","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"043d198b-3af0-477d-be11-5d1373280379","when":null,"workflow_outputs":[]},"1":{"annotation":"This can be output of STARsolo or DropletUtils (too many barcodes will make memory errors)","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This can be output of STARsolo or DropletUtils (too many barcodes will make memory errors)","name":"filtered barcodes"}],"label":"filtered barcodes","name":"Input dataset collection","outputs":[],"position":{"left":55.999999999999986,"top":108.33333333333333},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tsv\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"03b0d22c-3b46-4460-8c18-b524bc09d655","when":null,"workflow_outputs":[]},"2":{"annotation":"gtf file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"gtf file","name":"gtf file"}],"label":"gtf file","name":"Input dataset","outputs":[],"position":{"left":148,"top":204.66666666666666},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gtf\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f1d04259-f449-44d7-8d36-49f3ad664361","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2","errors":null,"id":3,"input_connections":{"main|BAM":{"id":0,"output_name":"output"},"main|barcodes":{"id":1,"output_name":"output"},"main|gtffile":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"},{"description":"runtime parameter for tool velocyto CLI","name":"main"}],"label":"velocyto","name":"velocyto CLI","outputs":[{"name":"samples","type":"loom"}],"position":{"left":470.66666666666663,"top":116.32222493489576},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2","tool_shed_repository":{"changeset_revision":"90e95e9ee190","name":"velocyto_cli","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"main\\": {\\"do\\": \\"run10x\\", \\"__current_case__\\": 0, \\"sample_definition\\": {\\"sample_definition_select\\": \\"identifier\\", \\"__current_case__\\": 1}, \\"BAM\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gtffile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"s\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"m\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"M\\": false, \\"t\\": \\"uint16\\"}, \\"verbosity\\": \\"-vv\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.17.17+galaxy2","type":"tool","uuid":"f9aef50f-b31d-44dd-aad6-fe4d520e3b1e","when":null,"workflow_outputs":[{"label":"velocyto loom","output_name":"samples","uuid":"2899a85c-c198-436f-a409-9bfbdf90c95f"}]}},"tags":["name:single-cell"],"uuid":"33862923-af05-48ba-aec5-14393981cee2","version":4},"readme":"# RNA Velocity Analysis: Velocyto for 10X Genomics Data\\n\\nThese workflows simply run velocyto. There are 2 workflows because one can be easily run after the Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat-Compatible Format\' workflows (RNA Velocity Analysis: Velocyto for 10X Data from Bundled Output). The other can be easily run from uploaded datasets (RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes).\\n\\n## Input datasets\\n\\n- BAM files with CB and UB: A collection of BAM. It accepts BAM from cellranger or STARsolo with the CB and UB tags (if you use the fastq-to-matrix-10x workflows these tags are automatically included).\\n- filtered barcodes (only for Velocyto_on10X_filtered_barcodes workflow): A collection of filtered barcodes (this is what will be used by velocyto). \'Filtered\' means that these barcodes have been identified as potential cells. It should not be the whole list of 3 million possible barcodes from cellranger.\\n- filtered matrices in bundle (only for Velocyto_on10X_from_bundled workflow): A collection of filtered matrices as bundled (like the one which comes from the fastq-to-matrix-10x workflows): A collection with as many items as samples. For each sample, the item is a list with 3 datasets (barcodes, genes, matrix). The workflow will then extract the items which have the \'barcodes\' identifier.\\n- gtf file: A file with annotations where exons are and how they are grouped into genes.\\n\\n## Processing\\n\\n- If you provided matrices, the first step is to extract barcodes.\\n- For both cases velocyto cli is run to get a loom file per sample with spliced and unspliced counts.\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/velocyto_cli/velocyto_cli/0.17.17+galaxy2`\\n## [0.1] - 2024-01-26\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## RNA Velocity Analysis: Velocyto for 10X Data with Filtered Barcodes\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ BAM files with CB and UB\\"]@{ shape: docs }\\n1[\\"ℹ️ filtered barcodes\\"]@{ shape: docs }\\n2[\\"ℹ️ gtf file\\"]@{ shape: doc }\\n3[\\"velocyto\\"]@{ shape: process }\\n0 --> 3\\n1 --> 3\\n2 --> 3\\n```\\n","updated":"2024-02-05T00:00:00","trsID":"#workflow/github.com/iwc-workflows/velocyto/Velocyto-on10X-filtered-barcodes","dockstore_id":25202,"categories":[],"collections":["Single Cell"],"doi":"10.5281/zenodo.15078887","tests":[{"doc":"Test outline for Velocyto-on10X (barcodes)","job":{"gtf file":{"class":"File","location":"https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz","decompress":true,"filetype":"gtf"},"BAM files with CB and UB":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"subsample","location":"https://zenodo.org/records/10572348/files/subsample.bam","filetype":"bam"}]},"filtered barcodes":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"subsample","location":"https://zenodo.org/records/10572348/files/barcodes.tsv","filetype":"tsv"}]}},"outputs":{"velocyto loom":{"element_tests":{"subsample":{"asserts":{"has_size":{"value":4639326,"delta":400000}}}}}}}]}],"path":"./workflows/scRNAseq/velocyto"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/pseudo-bulk_edgeR.ga","testParameterFiles":["/pseudo-bulk_edgeR-tests.yml"],"authors":[{"name":"Diana Chiang Jurado","orcid":"0000-0002-5857-1477"},{"name":"Pavankumar Videm","orcid":"0000-0002-5192-126X"},{"name":"Pablo Moreno","orcid":"0000-0002-9856-1679"}],"definition":{"a_galaxy_workflow":"true","annotation":"Performs differential gene expression analysis on single-cell data using a pseudobulk approach. Aggregates cell-level counts from an annotated AnnData object by cell type or other metadata using Decoupler, then applies edgeR for robust statistical testing between conditions. Includes data preprocessing steps for compatibility with edgeR and generates interactive volcano plots to visualize significantly differentially expressed genes.","comments":[],"creator":[{"class":"Person","identifier":"0000-0002-5857-1477","name":"Diana Chiang Jurado"},{"class":"Person","identifier":"0000-0002-5192-126X","name":"Pavankumar Videm"},{"class":"Person","identifier":"0000-0002-9856-1679","name":"Pablo Moreno"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.1.1","name":"Single-Cell Pseudobulk Differential Expression Analysis with edgeR","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Ensure your AnnData object contains all necessary layers before using the decoupler tool. The raw counts should be included in AnnData. If they are missing, create a new layer (e.g., \'raw_counts\') and copy the raw counts into it.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Ensure your AnnData object contains all necessary layers before using the decoupler tool. The raw counts should be included in AnnData. If they are missing, create a new layer (e.g., \'raw_counts\') and copy the raw counts into it.","name":"Source AnnData file"}],"label":"Source AnnData file","name":"Input dataset","outputs":[],"position":{"left":0,"top":114.82675544995811},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"h5\\", \\"h5ad\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"96604e3b-861d-4d7c-8408-9d68489c1e41","when":null,"workflow_outputs":[]},"1":{"annotation":"Merge Obs fields before pseudo-bulk analysis to create new categories for grouping, such as \'sample,phase\' or \'sample,louvain\'. Ensure the fields exist in the Obs of the AnnData object. Multiple groups can be merged with a colon (\':\'), e.g., \'sample,phase\\n,phase\' creates \'sample_phase\' and \'louvain_phase\'.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Merge Obs fields before pseudo-bulk analysis to create new categories for grouping, such as \'sample,phase\' or \'sample,louvain\'. Ensure the fields exist in the Obs of the AnnData object. Multiple groups can be merged with a colon (\':\'), e.g., \'sample,phase\\n,phase\' creates \'sample_phase\' and \'louvain_phase\'.","name":"Pseudo-bulk: Fields to merge"}],"label":"Pseudo-bulk: Fields to merge","name":"Input parameter","outputs":[],"position":{"left":60.495251676792236,"top":230.35374378726198},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b765c928-15de-4a9b-8bf2-96f9a9a5cdba","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c70d7068-d535-430f-ba40-199bbfed605c"}]},"2":{"annotation":"Typically, the column in obs that you want to use for comparisons later (the main contrast field) should be specified here. This column will also be used for plotting the pseudo-bulk samples, showing the number of counts and cells.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Typically, the column in obs that you want to use for comparisons later (the main contrast field) should be specified here. This column will also be used for plotting the pseudo-bulk samples, showing the number of counts and cells.","name":"Group by column"}],"label":"Group by column","name":"Input parameter","outputs":[],"position":{"left":123.70374778716867,"top":341.8746744746166},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"fa4b1127-560a-4ce5-8ae7-65b1cbd466f1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"9b6c5134-f9fb-413a-8a65-62ca29401dd9"}]},"3":{"annotation":"The field used to create the pseudo-bulk replicates is typically a combination of multiple Obs fields merged together.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"The field used to create the pseudo-bulk replicates is typically a combination of multiple Obs fields merged together.","name":"Sample key column"}],"label":"Sample key column","name":"Input parameter","outputs":[],"position":{"left":151.20372670956235,"top":465.91578876790373},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"5daa239e-1acc-4730-993d-d7a62ae30575","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"e6d9cb6d-1128-4bd1-bb7e-e8d0d4dd11f6"}]},"4":{"annotation":"Name of the layer containing your raw (non-normalized) counts.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Name of the layer containing your raw (non-normalized) counts.","name":"Name Your Raw Counts Layer"}],"label":"Name Your Raw Counts Layer","name":"Input parameter","outputs":[],"position":{"left":198.40796524172788,"top":596.2242453471495},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e2b06f23-4090-4014-b194-9afcd48e3905","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"ed10dd0c-7c42-4c37-a2ec-788b25900c69"}]},"5":{"annotation":"The fields from Obs to be provided to EdgeR as factors. The first field should represent the main contrast for comparisons, while the subsequent fields will be used as covariates.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"The fields from Obs to be provided to EdgeR as factors. The first field should represent the main contrast for comparisons, while the subsequent fields will be used as covariates.","name":"Factor fields"}],"label":"Factor fields","name":"Input parameter","outputs":[],"position":{"left":223.45860152251703,"top":713.2366686237074},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d7f40c82-930a-45d4-8392-ecd071162040","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"b3885b84-8cf3-47ce-9e57-51890d8f6aa8"}]},"6":{"annotation":"Example 1: ~ 0 + Factor_1\\n(Use this formula when you only want to account for one factor, Factor_1).\\n\\nExample 2 (With covariate adjustment): ~ 0 + Factor_1 + Factor_2\\n(Use this formula if you need to adjust for additional factors, such as Factor_2, which serves as a covariate).\\n\\nNote: Ensure that all factors (e.g., Factor_1, Factor_2) included in the formula are defined in your factor file.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Example 1: ~ 0 + Factor_1\\n(Use this formula when you only want to account for one factor, Factor_1).\\n\\nExample 2 (With covariate adjustment): ~ 0 + Factor_1 + Factor_2\\n(Use this formula if you need to adjust for additional factors, such as Factor_2, which serves as a covariate).\\n\\nNote: Ensure that all factors (e.g., Factor_1, Factor_2) included in the formula are defined in your factor file.","name":"Formula"}],"label":"Formula","name":"Input parameter","outputs":[],"position":{"left":1084.8910833860734,"top":839.4664158173962},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"71d2cf0a-7eaa-41fc-9ffd-f0ffbdbf9657","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"e1911fda-16a7-4f48-8a17-5d9f4bd635ee"}]},"7":{"annotation":"Specify the name of the column containing your gene symbols. For example: gene_symbol, gene_name, x, etc.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Specify the name of the column containing your gene symbols. For example: gene_symbol, gene_name, x, etc.","name":"Gene symbol column"}],"label":"Gene symbol column","name":"Input parameter","outputs":[],"position":{"left":1357.37679290235,"top":445.59148543814814},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e22b8666-3c73-41e2-8d3d-e6a8f4e2ba62","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"7b3c4345-840d-4305-97ec-20ffe71a3a36"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/ebi-gxa/decoupler_pseudobulk/decoupler_pseudobulk/1.4.0+galaxy8","errors":null,"id":8,"input_connections":{"adata_obs_fields_to_merge":{"id":1,"output_name":"output"},"factor_fields":{"id":5,"output_name":"output"},"groupby":{"id":2,"output_name":"output"},"input_file":{"id":0,"output_name":"output"},"layer":{"id":4,"output_name":"output"},"sample_key":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Decoupler pseudo-bulk","outputs":[{"name":"count_matrix","type":"tabular"},{"name":"samples_metadata","type":"tabular"},{"name":"genes_metadata","type":"tabular"},{"name":"plot_output","type":"png"},{"name":"filter_by_expr_plot","type":"png"},{"name":"genes_ignore_per_contrast_field","type":"tabular"}],"position":{"left":516.0467803847828,"top":418.1072761184286},"post_job_actions":{"DeleteIntermediatesActioncount_matrix":{"action_arguments":{},"action_type":"DeleteIntermediatesAction","output_name":"count_matrix"},"HideDatasetActiongenes_metadata":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"genes_metadata"},"HideDatasetActionsamples_metadata":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"samples_metadata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/ebi-gxa/decoupler_pseudobulk/decoupler_pseudobulk/1.4.0+galaxy8","tool_shed_repository":{"changeset_revision":"ef054892d47f","name":"decoupler_pseudobulk","owner":"ebi-gxa","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata_obs_fields_to_merge\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"factor_fields\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"filter_expr\\": true, \\"filter_per_contrast\\": {\\"filter\\": \\"no\\", \\"__current_case__\\": 1}, \\"groupby\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"layer\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_cells\\": \\"10\\", \\"min_counts\\": \\"10\\", \\"min_counts_per_sample\\": \\"20\\", \\"min_total_counts\\": \\"1000\\", \\"mode\\": \\"sum\\", \\"plot_filtering_figsize\\": \\"13 13\\", \\"plot_samples_figsize\\": \\"13 13\\", \\"produce_anndata\\": false, \\"produce_plots\\": true, \\"sample_key\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"use_raw\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.0+galaxy8","type":"tool","uuid":"12b43361-2dbc-4dd5-89c9-6a9842bb1531","when":null,"workflow_outputs":[{"label":"Pseudobulk count matrix","output_name":"count_matrix","uuid":"b28fbf89-5621-459e-9321-6cd0a72261ea"},{"label":"Pseudobulk Plot","output_name":"plot_output","uuid":"6e8b4090-3ab7-4158-805a-2d757fd5e0fb"},{"label":"Filtered by expression","output_name":"filter_by_expr_plot","uuid":"93e42181-49b2-47cf-ab73-98373376caf9"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":8,"output_name":"count_matrix"}},"inputs":[],"label":"Sanitize matrix","name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":817.694628152607,"top":326.31168881503567},"post_job_actions":{"DeleteIntermediatesActionoutfile":{"action_arguments":{},"action_type":"DeleteIntermediatesAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"matrix.tsv"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"[ --+*^]+\\", \\"replace_pattern\\": \\"_\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"15d8b67b-a232-4ea9-9639-4308a23823f1","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"samples_metadata"}},"inputs":[],"label":"Sanitize factors","name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":818.0112059966837,"top":492.3401614099875},"post_job_actions":{"DeleteIntermediatesActionoutfile":{"action_arguments":{},"action_type":"DeleteIntermediatesAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"factors.tsv"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"[ --+*^]+\\", \\"replace_pattern\\": \\"_\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"a001701e-e3ae-491a-b5cb-8ccf5a1fe80d","when":null,"workflow_outputs":[]},"11":{"annotation":"A column that may affect EdgeR and DESeq2.","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0","errors":null,"id":11,"input_connections":{"input_tabular":{"id":8,"output_name":"genes_metadata"}},"inputs":[],"label":"Remove start, end, width","name":"Remove columns","outputs":[{"name":"output_tabular","type":"tabular"}],"position":{"left":834.6675907510573,"top":728.5754761171477},"post_job_actions":{"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"genes_metadata.tsv"},"action_type":"RenameDatasetAction","output_name":"output_tabular"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0","tool_shed_repository":{"changeset_revision":"2040e4c2750a","name":"column_remove_by_header","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"headers\\": [{\\"__index__\\": 0, \\"name\\": \\"start\\"}, {\\"__index__\\": 1, \\"name\\": \\"end\\"}, {\\"__index__\\": 2, \\"name\\": \\"width\\"}], \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_columns\\": false, \\"strip_characters\\": \\"#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0","type":"tool","uuid":"b48e0354-6a07-4ae0-a5fe-1dbb84010a26","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":10,"output_name":"outfile"}},"inputs":[],"label":"Sanitize first factor for leading digits","name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1073.6410833860734,"top":630.5406345673962},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"column\\": \\"2\\", \\"find_pattern\\": \\"^([0-9])(.+)\\", \\"replace_pattern\\": \\"GG_\\\\\\\\\\\\\\\\1\\\\\\\\\\\\\\\\2\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"2505066d-49d0-4ed0-888d-455867e6f07d","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":13,"input_connections":{"infile":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1364.7928022139954,"top":910.5816068561169},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN { print \\\\\\"header\\\\\\" } NR > 1 { if (!seen[$2]++) words[++count]=$2 } END { for (i=1; i<=count; i++) for (j=i+1; j<=count; j++) print words[i]\\\\\\"-\\\\\\"words[j] }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"5701bab4-15b7-401a-9689-c127d17e5bdf","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/edger/edger/3.36.0+galaxy5","errors":null,"id":14,"input_connections":{"anno|geneanno":{"id":11,"output_name":"output_tabular"},"contrasts|cinfo":{"id":13,"output_name":"outfile"},"formula":{"id":6,"output_name":"output"},"input|counts":{"id":9,"output_name":"outfile"},"input|fact|finfo":{"id":12,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool edgeR","name":"anno"},{"description":"runtime parameter for tool edgeR","name":"contrasts"},{"description":"runtime parameter for tool edgeR","name":"input"}],"label":null,"name":"edgeR","outputs":[{"name":"outTables","type":"input"},{"name":"outReport","type":"html"}],"position":{"left":1623.5448063037918,"top":655.3696689453593},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/edger/edger/3.36.0+galaxy5","tool_shed_repository":{"changeset_revision":"ae2aad0a6d50","name":"edger","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"lfc\\": \\"0.0\\", \\"pVal\\": \\"0.05\\", \\"pAdjust\\": \\"BH\\", \\"normalisationOption\\": \\"TMM\\", \\"robOption\\": true, \\"lrtOption\\": false}, \\"anno\\": {\\"annoOpt\\": \\"yes\\", \\"__current_case__\\": 0, \\"geneanno\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"contrasts\\": {\\"contrastOpt\\": \\"file\\", \\"__current_case__\\": 1, \\"cinfo\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"f\\": {\\"filt\\": {\\"filt_select\\": \\"no\\", \\"__current_case__\\": 1}}, \\"formula\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"format\\": \\"matrix\\", \\"__current_case__\\": 1, \\"counts\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fact\\": {\\"ffile\\": \\"yes\\", \\"__current_case__\\": 0, \\"finfo\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"out\\": {\\"normCounts\\": false, \\"rscript\\": false, \\"rdaOption\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.36.0+galaxy5","type":"tool","uuid":"33e1a4fc-7001-4d1a-88c2-d492d8faf7e3","when":null,"workflow_outputs":[{"label":"Tables: DEG","output_name":"outTables","uuid":"851ac5bc-9c57-4f36-b469-33a2e8dde894"},{"label":"Report Results: HTML File","output_name":"outReport","uuid":"5ad48faa-d0a7-4bb1-b307-7861f540ec29"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":15,"input_connections":{"input_collection":{"id":14,"output_name":"outTables"}},"inputs":[],"label":"Get contrast labels","name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1855.0434149796085,"top":281.6915885082918},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"6f390fc9-b4ac-485e-867c-f338c902eacb","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0","errors":null,"id":16,"input_connections":{"headers_0|name":{"id":7,"output_name":"output"},"input_tabular":{"id":14,"output_name":"outTables"}},"inputs":[],"label":"Select gene symbols, logFC, PValue and FDR","name":"Remove columns","outputs":[{"name":"output_tabular","type":"tabular"}],"position":{"left":1940.8759531861979,"top":513.5126602860244},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0","tool_shed_repository":{"changeset_revision":"2040e4c2750a","name":"column_remove_by_header","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"headers\\": [{\\"__index__\\": 0, \\"name\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"name\\": \\"logFC\\"}, {\\"__index__\\": 2, \\"name\\": \\"PValue\\"}, {\\"__index__\\": 3, \\"name\\": \\"FDR\\"}], \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_columns\\": true, \\"strip_characters\\": \\"#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0","type":"tool","uuid":"9935a582-7775-4706-be28-720224f5ba9e","when":null,"workflow_outputs":[{"label":"Tables for volcano plot","output_name":"output_tabular","uuid":"647df402-93cb-4f9e-9bde-7afcbcffad5b"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","errors":null,"id":17,"input_connections":{"infile":{"id":15,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2097.058869383114,"top":160.46839568505237},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"edgeR_\\", \\"replace_pattern\\": \\"\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"6595e60f-c1a5-4667-946a-69f5bf201d0f","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":18,"input_connections":{"split_parms|input":{"id":17,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"Split contrasts","name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":2376.5570395203194,"top":0},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"batch\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"0eb6f773-5ef2-4b50-b4fa-21fdbd7671ff","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":19,"input_connections":{"input1":{"id":18,"output_name":"list_output_txt"}},"inputs":[],"label":"Contrast as parameters","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":2601.5947847657694,"top":142.72866520857775},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81b2bd80-00c7-4ce5-8d26-f48b0f9f7f08","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.6","errors":null,"id":20,"input_connections":{"input":{"id":16,"output_name":"output_tabular"},"plot_options|title":{"id":19,"output_name":"text_param"}},"inputs":[{"description":"runtime parameter for tool Volcano Plot","name":"plot_options"}],"label":null,"name":"Volcano Plot","outputs":[{"name":"plot","type":"pdf"}],"position":{"left":2858.6197555934036,"top":367.5558994050945},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.6","tool_shed_repository":{"changeset_revision":"2f557f6abbfb","name":"volcanoplot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fdr_col\\": \\"4\\", \\"header\\": \\"yes\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"label_col\\": \\"1\\", \\"labels\\": {\\"label_select\\": \\"signif\\", \\"__current_case__\\": 0, \\"top_num\\": \\"40\\"}, \\"lfc_col\\": \\"2\\", \\"lfc_thresh\\": \\"0.58\\", \\"out_options\\": {\\"rscript_out\\": false}, \\"plot_options\\": {\\"boxes\\": false, \\"title\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"xlab\\": \\"\\", \\"ylab\\": \\"\\", \\"xmin\\": null, \\"xmax\\": null, \\"ymax\\": null, \\"legend\\": \\"\\", \\"legend_labs\\": \\"Down,Not Sig,Up\\"}, \\"pval_col\\": \\"3\\", \\"signif_thresh\\": \\"0.05\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.6","type":"tool","uuid":"36e45e14-8bec-4321-9b9f-557db6d6ed97","when":null,"workflow_outputs":[{"label":"Volcano Plot on input dataset(s): PDF","output_name":"plot","uuid":"02cc64ea-d34d-4081-8a2b-2dc59e102c66"}]}},"tags":[],"uuid":"e4e4055c-7e3f-4014-a8e6-343199fd9b01","version":31},"readme":"# Single-Cell Pseudobulk Differential Expression Analysis with edgeR\\n\\nThis workflow uses the decoupler tool in Galaxy to generate pseudobulk counts from an annotated AnnData file obtained from scRNA-seq analysis. Following the pseudobulk step, differential expression genes (DEG) are calculated \\nusing the edgeR tool. The workflow also includes data sanitation steps to ensure smooth operation of edgeR and minimizing potential issues. Additionally, a Volcano plot tool is used to visualize the results after the DEG \\nanalysis.\\n\\nThe workflow deposited here is based on an earlier version of the [Persist-SEQ](https://persist-seq.org/)[^1] Pseudo-bulk scRNA-seq pipeline, of which the latest version is available [here](https://usegalaxy.eu/published/workflow?id=c3a11e1ac1aa8383). In terms of core procedures, the main differences with the IWC workflow are that the Persist-SEQ workflow:\\n\\n[^1]: The PERSIST-SEQ consortium is funded by the Innovative Medicines Initiative (IMI) Joint Undertaking, which receives support from the European Union\'s Horizon 2020 research and innovation program and EFPIA.\\n\\n- Is more opinionanted on the downstream enrichment analysis for the cancer biology use case.\\n- Enables filtering out of seldomly expressed genes (with a configurable threshold) per contrast after DE calling, which reduces poorly supported highly DE genes and improves the signal for downstream enrichment analysis.\\n\\n## Inputs\\n\\n- deCoupler: Source AnnData (`h5ad`).\\n    - Parameter: Pseudobulk: Fields to merge / optional \\n    - Parameter: Group by column / has to be given\\n    - Parameter: Sample key column / has to be given\\n    - Parameter: Name your raw count layer / has to be given\\n    - Parameter: Factor Field / has to be given\\n- edgeR:\\n    - Sanitzed Count Matrix\\n    - Sanitized Factor File\\n    - Cleaned Gene Annotations file\\n    - Parameter: Formula for linear model / has to be given\\n    - Contrast file / has to be given\\n- Volcano Plot:\\n    - Input (`tabular`) file with genesymbol, logFC, Pvalue and FDR columns.\\n\\n## Processing\\n\\nSanitzation steps after decoupler:\\n- Sanitize Matrix and Factors(`tabular`): finds [ --+*^]+ and replace with -\\n- Remove start, end with (`tabular`): A column that may affect EdgeR and DESeq2.\\n- Sanitize First Factor for leading digits (`tabular`): Finds ^([0-9])(.+) and replace it with GG_\\\\\\\\1\\\\\\\\2\\n- Get Contrast labels\\n- Replace text\\n- Split Contrasts\\n- Contrasts as Parameters: Plot title\\n- Select columns for volcano plot using (`Remove columns`) from DEG edgeR (`Table`)output.\\n\\n\\n## Outputs\\n\\n  - Pseudobulk_count_matrix (`tabular`)\\n  - Pseudobulk Plot (`png`)\\n  - Filtered by expression (`png`)\\n  - Table DEG\\n  - Results (`HTML`) File and plots for download within the output as (`png`)\\n  - Volcano plot (`PDF`)\\n","changelog":"# Changelog\\n\\n## [0.1.1] - 2024-11-18\\n### Added\\n- Extended attribution to credit [Persist-SEQ](https://persist-seq.org/) as the original authors of the workflow that this workflow is based on.\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/ebi-gxa/decoupler_pseudobulk/decoupler_pseudobulk/1.4.0+galaxy5` was updated to `toolshed.g2.bx.psu.edu/repos/ebi-gxa/decoupler_pseudobulk/decoupler_pseudobulk/1.4.0+galaxy8`\\n\\n## [0.1] - 2024-10-14\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell Pseudobulk Differential Expression Analysis with edgeR\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Source AnnData file\\"]@{ shape: doc }\\n1[\\"ℹ️ Pseudo-bulk: Fields to merge\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Group by column\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Sample key column\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Name Your Raw Counts Layer\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Factor fields\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Formula\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Gene symbol column\\"]@{ shape: lean-l }\\n8[\\"Decoupler pseudo-bulk\\"]@{ shape: process }\\n1 --> 8\\n5 --> 8\\n2 --> 8\\n0 --> 8\\n4 --> 8\\n3 --> 8\\n9[\\"Sanitize matrix\\"]@{ shape: process }\\n8 --> 9\\n10[\\"Sanitize factors\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Remove start, end, width\\"]@{ shape: process }\\n8 --> 11\\n12[\\"Sanitize first factor for leading digits\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Text reformatting\\"]@{ shape: process }\\n12 --> 13\\n14[\\"edgeR\\"]@{ shape: process }\\n11 --> 14\\n13 --> 14\\n6 --> 14\\n9 --> 14\\n12 --> 14\\n15[\\"Get contrast labels\\"]@{ shape: process }\\n14 --> 15\\n16[\\"Select gene symbols, logFC, PValue and FDR\\"]@{ shape: process }\\n7 --> 16\\n14 --> 16\\n17[\\"Replace Text\\"]@{ shape: process }\\n15 --> 17\\n18[\\"Split contrasts\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Contrast as parameters\\"]@{ shape: process }\\n18 --> 19\\n20[\\"Volcano Plot\\"]@{ shape: process }\\n16 --> 20\\n19 --> 20\\n```\\n","updated":"2024-11-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/pseudobulk-worflow-decoupler-edger/main","dockstore_id":29413,"categories":[],"collections":["Transcriptomics","Single Cell"],"doi":"10.5281/zenodo.15078872","tests":[{"doc":"Test outline for pseudo-bulk_edgeR","job":{"Source AnnData file":{"class":"File","location":"https://zenodo.org/records/13929549/files/Source%20AnnData%20file.h5ad","filetype":"h5ad"},"Pseudo-bulk: Fields to merge":null,"Group by column":"cell_type","Sample key column":"individual","Name Your Raw Counts Layer":"counts","Factor fields":"disease","Gene symbol column":"gene_symbol","Formula":"~ 0 + disease"},"outputs":{"Pseudobulk count matrix":{"has_text_matching":{"expression":"ACER3\\t4.0\\t25.0\\t21.0\\t110.0\\t82.0\\t91.0\\t22.0\\t326.0\\t297.0\\t211.0\\t1004.0\\t574.0\\t370.0\\t108.0\\t0.0\\t0.0\\t2.0\\t2.0\\t188.0\\t113.0\\t135.0\\t322.0\\t324.0\\t159.0\\t7.0\\t7.0\\t32.0\\t5.0\\t33.0\\t89.0\\t2.0\\t2.0\\t8.0\\t48.0"}},"Pseudobulk Plot":{"element_test":{"has_size":40116,"delta":2000}},"Filtered by expression":{"element_test":{"has_size":23490,"delta":2000}},"Report Results: HTML File":{"element_test":{"has_size":531761,"delta":25000}},"Tables: DEG":{"element_tests":{"edgeR_normal-COVID_19":{"has_text_matching":{"expression":"NAPA\\tENSG00000105402\\tTrue\\t0.342[0-9]\\t1.686[0-9]\\t0.846[0-9]\\t4\\tFalse\\t0.180[0-9]\\t0.440[0-9]\\t-1.059[0-9]\\t6.833[0-9]\\t3.291[0-9]\\t0.076[0-9]\\t0.619[0-9]"},"has_n_lines":{"n":1430,"delta":1}}}},"Tables for volcano plot":{"element_tests":{"edgeR_normal-COVID_19":{"has_text_matching":{"expression":"FGFR1OP2\\t-2.367[0-9]\\t0.004[0-9]\\t0.458[0-9]"}}}},"Volcano Plot on input dataset(s): PDF":{"element_tests":{"edgeR_normal-COVID_19":{"has_size":85052,"delta":2000}}}}}]}],"path":"./workflows/scRNAseq/pseudobulk-worflow-decoupler-edger"},{"version":1.2,"workflows":[{"name":"baredSC-1d-logNorm","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/baredSC-1d-logNorm.ga","testParameterFiles":["/baredSC-1d-logNorm-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Applies baredSC algorithm to fit and combine one-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on expression of genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.6","name":"Single-Cell Mixture Analysis: baredSC 1D Log-Normalized Models","steps":{"0":{"annotation":"The dataset must have a first row with row names. Among them should be \'nCount_RNA\' and the gene you want to get PDF of.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"The dataset must have a first row with row names. Among them should be \'nCount_RNA\' and the gene you want to get PDF of.","name":"Tabular with raw expression values"}],"label":"Tabular with raw expression values","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"247797f8-537d-434c-b7f0-b5212d6a6139","when":null,"workflow_outputs":[]},"1":{"annotation":"Must be included in the first row of the tabular","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Must be included in the first row of the tabular","name":"Gene name"}],"label":"Gene name","name":"Input parameter","outputs":[],"position":{"left":51,"top":95},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8486d61d-03ce-44df-a74c-0aa6a5b0e0b7","when":null,"workflow_outputs":[]},"2":{"annotation":"Maximum value to explore in logNorm","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Maximum value to explore in logNorm","name":"Maximum value in logNorm"}],"label":"Maximum value in logNorm","name":"Input parameter","outputs":[],"position":{"left":97,"top":176.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a54a81dc-8b4c-4ed3-8d4e-0748755873e8","when":null,"workflow_outputs":[]},"3":{"annotation":"Usually 4 is enough","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Usually 4 is enough","name":"Maximum number of Gaussians to study"}],"label":"Maximum number of Gaussians to study","name":"Input parameter","outputs":[],"position":{"left":268,"top":291},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b7482c04-93ff-4f34-acbd-9c46d1f067cd","when":null,"workflow_outputs":[]},"4":{"annotation":"","id":4,"input_connections":{"Maximum number in param output":{"id":3,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"generate_param_list_one_to_number","outputs":[],"position":{"left":547,"top":231.5},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"generate_param_list_one_to_number","steps":{"0":{"annotation":"Maximum number in param output","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Maximum number in param output","name":"Maximum number in param output"}],"label":"Maximum number in param output","name":"Input parameter","outputs":[],"position":{"left":0,"top":36},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"655f8763-6f41-401c-a7c2-cbc0926222c0","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"996f5ca8-629e-4887-9bc0-2d719b790868"}]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0","errors":null,"id":1,"input_connections":{"token_set_0|repeat_select|times":{"id":0,"output_name":"output"}},"inputs":[],"label":"create first tabular with good number of rows.","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":280,"top":0},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": \\"whatever\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"c96eeaba-7b46-478f-bc6d-ac3d64f72523","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"addValue","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"outfile"}},"inputs":[],"label":"add column with numbers starting at 1","name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":560,"top":20},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"addValue","tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"20631933-3793-45f0-9fb5-6f3d9463457b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"Cut1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":"retrieve numbers","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":840,"top":20},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"b467476c-8691-4fd4-aea0-d783b14b997a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":4,"input_connections":{"split_parms|input":{"id":3,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split numbers to collection","name":"Split file","outputs":[{"name":"list_output_tab","type":"input"}],"position":{"left":1120,"top":20},"post_job_actions":{"HideDatasetActionlist_output_tab":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"top\\": \\"0\\", \\"split_by\\": {\\"select_split_by\\": \\"row\\", \\"__current_case__\\": 1, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"b621baec-46f7-4dfe-b6fa-f8c44c2822f2","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":4,"output_name":"list_output_tab"}},"inputs":[],"label":"numbers to param","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1400,"top":29},"post_job_actions":{},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"1aff7c8b-0554-4e98-962d-3d6b161fa133","when":null,"workflow_outputs":[{"label":"params_list","output_name":"integer_param","uuid":"e05e466c-4694-4d6e-90dc-ea4e1aa3ff58"}]}},"tags":[],"uuid":"2c6dc95f-cd21-4867-b848-57bfd47cb80f"},"tool_id":null,"type":"subworkflow","uuid":"e5607959-5b5e-4898-92ab-078c7cafb327","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.3+galaxy0","errors":null,"id":5,"input_connections":{"MCMC|nnorm":{"id":4,"output_name":"params_list"},"MCMC|xmax":{"id":2,"output_name":"output"},"geneColName":{"id":1,"output_name":"output"},"input_counts|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool baredSC 1d","name":"MCMC"},{"description":"runtime parameter for tool baredSC 1d","name":"MCMC"},{"description":"runtime parameter for tool baredSC 1d","name":"input_counts"}],"label":"baredSC","name":"baredSC 1d","outputs":[{"name":"qc_plots","type":"input"},{"name":"other_outputs","type":"input"},{"name":"output","type":"npz"},{"name":"neff","type":"txt"},{"name":"pdf","type":"tabular"},{"name":"plot","type":"png"},{"name":"logevidence","type":"txt"}],"position":{"left":867,"top":114.5},"post_job_actions":{"HideDatasetActionlogevidence":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"logevidence"},"HideDatasetActionother_outputs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"other_outputs"},"HideDatasetActionpdf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pdf"},"HideDatasetActionplot":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.3+galaxy0","tool_shed_repository":{"changeset_revision":"769d815e4b0d","name":"baredsc_1d","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"MCMC\\": {\\"xmin\\": \\"0.0\\", \\"xmax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nx\\": \\"100\\", \\"minScalex\\": \\"0.1\\", \\"scale\\": {\\"type\\": \\"Seurat\\", \\"__current_case__\\": 0, \\"targetSum\\": \\"10000.0\\"}, \\"seed\\": \\"1\\", \\"nnorm\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nsampMCMC\\": \\"100000\\", \\"automaticRestart\\": {\\"set_minNeff\\": \\"yes\\", \\"__current_case__\\": 0, \\"minNeff\\": \\"200.0\\"}}, \\"advanced\\": {\\"osampx\\": \\"10\\", \\"osampxpdf\\": \\"5\\", \\"coviscale\\": \\"1.0\\", \\"nis\\": \\"1000\\", \\"burn\\": {\\"custom\\": \\"no\\", \\"__current_case__\\": 0}}, \\"filter\\": {\\"nb\\": \\"0\\", \\"__current_case__\\": 0}, \\"geneColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_counts\\": {\\"filetype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"plots\\": {\\"image_file_format\\": \\"png\\", \\"title\\": \\"\\", \\"removeFirstSamples\\": \\"-1\\", \\"nsampInPlot\\": \\"100000\\", \\"prettyBins\\": \\"-1\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3+galaxy0","type":"tool","uuid":"75379c38-7040-4651-b875-5ad7382de31d","when":null,"workflow_outputs":[{"label":"baredsc_qc_plots","output_name":"qc_plots","uuid":"269f21b9-0499-40bc-88d5-555ee46d166d"},{"label":"baredsc_neff","output_name":"neff","uuid":"b2f23ac4-3634-4c1c-a17c-89a6fec675d8"},{"label":"baredsc_numpy","output_name":"output","uuid":"f5339d63-a667-4cd2-879c-d35c633e3ffd"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.3+galaxy0","errors":null,"id":6,"input_connections":{"MCMC|outputs":{"id":5,"output_name":"output"},"MCMC|xmax":{"id":2,"output_name":"output"},"geneColName":{"id":1,"output_name":"output"},"input_counts|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Combine multiple 1D Models","name":"MCMC"},{"description":"runtime parameter for tool Combine multiple 1D Models","name":"MCMC"},{"description":"runtime parameter for tool Combine multiple 1D Models","name":"input_counts"}],"label":"combine baredSC 1d","name":"Combine multiple 1D Models","outputs":[{"name":"other_outputs","type":"input"},{"name":"pdf","type":"tabular"},{"name":"plot","type":"png"}],"position":{"left":1220,"top":170.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.3+galaxy0","tool_shed_repository":{"changeset_revision":"5b7c81b35b89","name":"baredsc_combine_1d","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"MCMC\\": {\\"outputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"xmin\\": \\"0.0\\", \\"xmax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nx\\": \\"100\\", \\"minScalex\\": \\"0.1\\", \\"scale\\": {\\"type\\": \\"Seurat\\", \\"__current_case__\\": 0, \\"targetSum\\": \\"10000.0\\"}, \\"seed\\": \\"1\\"}, \\"advanced\\": {\\"osampx\\": \\"10\\", \\"osampxpdf\\": \\"5\\", \\"coviscale\\": \\"1.0\\", \\"nis\\": \\"1000\\"}, \\"filter\\": {\\"nb\\": \\"0\\", \\"__current_case__\\": 0}, \\"geneColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_counts\\": {\\"filetype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"plots\\": {\\"image_file_format\\": \\"png\\", \\"title\\": \\"\\", \\"removeFirstSamples\\": \\"-1\\", \\"nsampInPlot\\": \\"100000\\", \\"prettyBins\\": \\"-1\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3+galaxy0","type":"tool","uuid":"752ba293-93e9-4769-85d0-ebc9c2c100b6","when":null,"workflow_outputs":[{"label":"combined_pdf","output_name":"pdf","uuid":"2aa68ed9-2ca6-4f31-be28-bcead2a78e72"},{"label":"combined_plot","output_name":"plot","uuid":"f1cd8147-a286-43a7-bb82-9ee5f881e8fa"},{"label":"combined_other_outputs","output_name":"other_outputs","uuid":"b8da3ab7-2fed-445e-a3a8-97ddcfa68b8e"}]}},"tags":[],"uuid":"1ea0b92d-d863-435a-96ac-1e547fdbc994","version":7},"readme":"# Single-Cell Mixture Analysis: baredSC Log-Normalized Models\\n\\nThese workflows allow to run a baredSC analysis from a table with counts in a single click. It uses models from 1 to N Gaussians and combine them. It uses the logNorm scale, 100 bins for 1 dimension and 25 bins on each axis in 2 dimensions.\\n\\n## Inputs dataset\\n\\n- Both workflows need a tabular dataset where each row is a cell. The tabular needs to have a header line with column names. There must be at least two columns: \'nCount_RNA\' and another one with the counts for the gene(s) of interest. A way to get such table in R from a Seurat object (`seurat.obj`) is:\\n\\n```r\\nmy.genes <- c(\\"Hoxa13\\", \\"Hoxd13\\")\\ndf <- cbind(seurat.obj[[]], # This will give you all metadata including nCount_RNA\\n            FetchData(seurat.obj, slot = \\"counts\\", vars = my.genes))\\n\\nwrite.table(df, \\"input_for_baredSC.txt\\", quote = F, sep = \\"\\\\t\\", row.names = F)\\n```\\n\\n## Inputs values\\n\\nFor the 1D:\\n\\n- Gene name: The name of the column with the counts of your gene of interest.\\n- Maximum value in logNorm: The maximum value to explore in PDF. This value should be large enough so the PDF is at 0 at this value.\\n- Maximum number of Gaussians to study: All models between models with 1 Gaussians to models with this number of Gaussians will be combined.\\n\\nFor the 2D:\\n\\n- Gene name for x axis: The name of the column with the counts of your gene in x axis.\\n- Gene name for y axis: The name of the column with the counts of your gene in y axis.\\n- maximum value in logNorm for x-axis: The maximum value to explore in PDF in the x axis. This value should be large enough so the PDF is at 0 at this value.\\n- maximum value in logNorm for y-axis: The maximum value to explore in PDF in the y axis. This value should be large enough so the PDF is at 0 at this value.\\n- Maximum number of Gaussians to study: All models between models with 1 2D-Gaussians to models with this number of 2D-Gaussians will be combined.\\n- compute p-value: Whether you want to get a p-value. As a consequence, less samples than available will be used for plots as p-value computation requires to have independent samples.\\n\\n## Processing\\n\\n- The workflow will generate paramater values from 1 to the maximum number of Gaussians to study.\\n- baredSC_1d or baredSC_2d is run for each of these number of Gaussians\\n- All models are combined into a single result.\\n","changelog":"# Changelog\\n\\n## [0.6] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0`\\n\\n## [0.5] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n\\n## [0.4] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.3+galaxy0`\\n\\n## [0.3] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0`\\n\\n## [0.2] - 2023-12-01\\n\\n### Add \'compute p-value\' parameter in 2d\\n\\n### Semi-automatic update\\n- all baredsc tools were updated from `1.1.2+galaxy0` to `1.1.2+galaxy1`\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n\\n## [0.1] - 2023-10-03\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell Mixture Analysis: baredSC 1D Log-Normalized Models\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Tabular with raw expression values\\"]@{ shape: doc }\\n1[\\"ℹ️ Gene name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Maximum value in logNorm\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Maximum number of Gaussians to study\\"]@{ shape: lean-l }\\n4[\\"🛠️ generate_param_list_one_to_number\\"]@{ shape: subprocess }\\n3 --> 4\\n5[\\"baredSC\\"]@{ shape: process }\\n4 --> 5\\n2 --> 5\\n1 --> 5\\n0 --> 5\\n6[\\"combine baredSC 1d\\"]@{ shape: process }\\n5 --> 6\\n2 --> 6\\n1 --> 6\\n0 --> 6\\n```\\n\\n## generate_param_list_one_to_number\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Maximum number in param output\\"]@{ shape: lean-l }\\n1[\\"create first tabular with good number of rows.\\"]@{ shape: process }\\n0 --> 1\\n2[\\"add column with numbers starting at 1\\"]@{ shape: process }\\n1 --> 2\\n3[\\"retrieve numbers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"split numbers to collection\\"]@{ shape: process }\\n3 --> 4\\n5[\\"numbers to param\\"]@{ shape: process }\\n4 --> 5\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/baredsc/baredSC-1d-logNorm","dockstore_id":24061,"categories":[],"collections":["Transcriptomics","Single Cell"],"doi":"10.5281/zenodo.15078889","tests":[{"doc":"Test outline for baredSC_1d","job":{"Tabular with raw expression values":{"class":"File","path":"test-data/nih3t3_generated_2d_2.txt","filetype":"tabular"},"Gene name":"0.5_0_0_0.5_x","Maximum value in logNorm":"2.5","Maximum number of Gaussians to study":"4"},"outputs":{"baredsc_numpy":{"element_tests":{"split_file_000000.tabular":{"asserts":{"has_size":{"value":1257919,"delta":100000}}},"split_file_000001.tabular":{"asserts":{"has_size":{"value":1601519,"delta":100000}}},"split_file_000002.tabular":{"asserts":{"has_size":{"value":2180423,"delta":200000}}},"split_file_000003.tabular":{"asserts":{"has_size":{"value":28234812,"delta":2000000}}}}},"baredsc_qc_plots":{"element_tests":{"split_file_000000.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":20304,"delta":1000}}},"corner":{"asserts":{"has_size":{"value":63603,"delta":6000}}},"p":{"asserts":{"has_size":{"value":45302,"delta":4000}}}}},"split_file_000001.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":27704,"delta":2000}}},"corner":{"asserts":{"has_size":{"value":387365,"delta":30000}}},"p":{"asserts":{"has_size":{"value":97623,"delta":90000}}}}},"split_file_000002.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":32463,"delta":3000}}},"corner":{"asserts":{"has_size":{"value":1103840,"delta":100000}}},"p":{"asserts":{"has_size":{"value":159797,"delta":10000}}}}},"split_file_000003.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":31343,"delta":3000}}},"corner":{"asserts":{"has_size":{"value":2139411,"delta":200000}}},"p":{"asserts":{"has_size":{"value":175774,"delta":10000}}}}}}},"baredsc_neff":{"element_tests":{"split_file_000000.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"80[0-9][0-9].[0-9]*"}}},"split_file_000001.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"13[0-9][0-9].[0-9]*"}}},"split_file_000002.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"2[0-9][0-9].[0-9]*"}}},"split_file_000003.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"[7-9][0-9][0-9].[0-9]*"}}}}},"combined_other_outputs":{"element_tests":{"individuals":{"asserts":{"has_size":{"value":108407,"delta":10000}}},"means":{"asserts":{"has_n_lines":{"n":99998,"delta":4000},"has_line_matching":{"expression":"6.[0-9]*e-01"}}},"posterior_andco":{"asserts":{"has_size":{"value":197980,"delta":10000}}},"posterior_individuals":{"asserts":{"has_size":{"value":105262,"delta":10000}}},"posterior_per_cell":{"asserts":{"has_n_lines":{"n":2362},"has_line_matching":{"expression":"mu\\tsd"}}},"with_posterior":{"asserts":{"has_size":{"value":234303,"delta":20000}}}}},"combined_pdf":{"asserts":{"has_line":{"line":"x\\tlow\\tmean\\thigh\\tmedian"},"has_text":{"text":"0.0125\\t"}}},"combined_plot":{"path":"test-data/combined_1d_plot.png","compare":"sim_size","delta_frac":0.1}}}]},{"name":"baredSC-2d-logNorm","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/baredSC-2d-logNorm.ga","testParameterFiles":["/baredSC-2d-logNorm-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Applies baredSC algorithm to fit and combine two-dimensional Gaussian mixture models (from 1 to N components) on log-normalized single-cell gene expression data. Enables identification of subpopulations based on co-expression patterns of two genes of interest and provides statistical assessment of the optimal number of components in heterogeneous cell populations.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.6","name":"Single-Cell Mixture Analysis: baredSC 2D Log-Normalized Models","steps":{"0":{"annotation":"The dataset must have a first row with row names. Among them should be \'nCount_RNA\' and the gene you want to get PDF of.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"The dataset must have a first row with row names. Among them should be \'nCount_RNA\' and the gene you want to get PDF of.","name":"Tabular with raw expression values"}],"label":"Tabular with raw expression values","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"247797f8-537d-434c-b7f0-b5212d6a6139","when":null,"workflow_outputs":[]},"1":{"annotation":"Must be included in the first row of the tabular","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Must be included in the first row of the tabular","name":"Gene name for x axis"}],"label":"Gene name for x axis","name":"Input parameter","outputs":[],"position":{"left":28,"top":91},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8486d61d-03ce-44df-a74c-0aa6a5b0e0b7","when":null,"workflow_outputs":[]},"2":{"annotation":"Maximum value to explore in logNorm in x-axis (usually between 2 and 4)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Maximum value to explore in logNorm in x-axis (usually between 2 and 4)","name":"maximum value in logNorm for x-axis"}],"label":"maximum value in logNorm for x-axis","name":"Input parameter","outputs":[],"position":{"left":78,"top":187.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a54a81dc-8b4c-4ed3-8d4e-0748755873e8","when":null,"workflow_outputs":[]},"3":{"annotation":"Must be included in the first row of the tabular","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Must be included in the first row of the tabular","name":"Gene name for y axis"}],"label":"Gene name for y axis","name":"Input parameter","outputs":[],"position":{"left":152,"top":297.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"73f322d4-ac8e-47f9-930b-e38600ac44f8","when":null,"workflow_outputs":[]},"4":{"annotation":"Maximum value to explore in logNorm in y-axis (usually between 2 and 4)","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Maximum value to explore in logNorm in y-axis (usually between 2 and 4)","name":"maximum value in logNorm for y-axis"}],"label":"maximum value in logNorm for y-axis","name":"Input parameter","outputs":[],"position":{"left":254,"top":390.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"35403c8f-7cb4-40bf-8d6c-a259627daf01","when":null,"workflow_outputs":[]},"5":{"annotation":"Usually 4 is enough","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Usually 4 is enough","name":"Maximum number of Gaussians to study"}],"label":"Maximum number of Gaussians to study","name":"Input parameter","outputs":[],"position":{"left":320,"top":481},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b7482c04-93ff-4f34-acbd-9c46d1f067cd","when":null,"workflow_outputs":[]},"6":{"annotation":"Whether you want to get a p-value. As a consequence, less samples than available will be used for plots.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Whether you want to get a p-value. As a consequence, less samples than available will be used for plots.","name":"compute p-value"}],"label":"compute p-value","name":"Input parameter","outputs":[],"position":{"left":378,"top":578.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"4b7990a3-4fff-4c74-a9ee-0a8716e33866","when":null,"workflow_outputs":[]},"7":{"annotation":"","id":7,"input_connections":{"Maximum number in param output":{"id":5,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"generate_param_list_one_to_number","outputs":[],"position":{"left":677,"top":346.5},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"generate_param_list_one_to_number","steps":{"0":{"annotation":"Maximum number in param output","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Maximum number in param output","name":"Maximum number in param output"}],"label":"Maximum number in param output","name":"Input parameter","outputs":[],"position":{"left":0,"top":36},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"655f8763-6f41-401c-a7c2-cbc0926222c0","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"996f5ca8-629e-4887-9bc0-2d719b790868"}]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0","errors":null,"id":1,"input_connections":{"token_set_0|repeat_select|times":{"id":0,"output_name":"output"}},"inputs":[],"label":"create first tabular with good number of rows.","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":280,"top":0},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": \\"whatever\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"c96eeaba-7b46-478f-bc6d-ac3d64f72523","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"addValue","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"outfile"}},"inputs":[],"label":"add column with numbers starting at 1","name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":560,"top":20},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"addValue","tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"20631933-3793-45f0-9fb5-6f3d9463457b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"Cut1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":"retrieve numbers","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":840,"top":20},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"b467476c-8691-4fd4-aea0-d783b14b997a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":4,"input_connections":{"split_parms|input":{"id":3,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split numbers to collection","name":"Split file","outputs":[{"name":"list_output_tab","type":"input"}],"position":{"left":1120,"top":20},"post_job_actions":{"HideDatasetActionlist_output_tab":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"top\\": \\"0\\", \\"split_by\\": {\\"select_split_by\\": \\"row\\", \\"__current_case__\\": 1, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"b621baec-46f7-4dfe-b6fa-f8c44c2822f2","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":4,"output_name":"list_output_tab"}},"inputs":[],"label":"numbers to param","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1400,"top":29},"post_job_actions":{},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"1aff7c8b-0554-4e98-962d-3d6b161fa133","when":null,"workflow_outputs":[{"label":"params_list","output_name":"integer_param","uuid":"e05e466c-4694-4d6e-90dc-ea4e1aa3ff58"}]}},"tags":[],"uuid":"15e721b4-9f4c-453a-b8c1-2a7f22476db4"},"tool_id":null,"type":"subworkflow","uuid":"e5607959-5b5e-4898-92ab-078c7cafb327","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.3+galaxy0","errors":null,"id":8,"input_connections":{"MCMC|nnorm":{"id":7,"output_name":"params_list"},"MCMC|xmax":{"id":2,"output_name":"output"},"MCMC|ymax":{"id":4,"output_name":"output"},"geneXColName":{"id":1,"output_name":"output"},"geneYColName":{"id":3,"output_name":"output"},"input_counts|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool baredSC 2d","name":"MCMC"},{"description":"runtime parameter for tool baredSC 2d","name":"MCMC"},{"description":"runtime parameter for tool baredSC 2d","name":"MCMC"},{"description":"runtime parameter for tool baredSC 2d","name":"input_counts"}],"label":null,"name":"baredSC 2d","outputs":[{"name":"qc_plots","type":"input"},{"name":"other_outputs","type":"input"},{"name":"output","type":"npz"},{"name":"neff","type":"txt"},{"name":"pdf2d","type":"tabular"},{"name":"pdf2d_flat","type":"tabular"},{"name":"plot","type":"png"},{"name":"logevidence","type":"txt"}],"position":{"left":923,"top":35.5},"post_job_actions":{"HideDatasetActionlogevidence":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"logevidence"},"HideDatasetActionother_outputs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"other_outputs"},"HideDatasetActionpdf2d":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pdf2d"},"HideDatasetActionpdf2d_flat":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pdf2d_flat"},"HideDatasetActionplot":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.3+galaxy0","tool_shed_repository":{"changeset_revision":"a90f8efe08b4","name":"baredsc_2d","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"MCMC\\": {\\"xmin\\": \\"0.0\\", \\"xmax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nx\\": \\"25\\", \\"minScalex\\": \\"0.1\\", \\"ymin\\": \\"0.0\\", \\"ymax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ny\\": \\"25\\", \\"minScaley\\": \\"0.1\\", \\"scale\\": {\\"type\\": \\"Seurat\\", \\"__current_case__\\": 0, \\"targetSum\\": \\"10000.0\\"}, \\"seed\\": \\"1\\", \\"nnorm\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nsampMCMC\\": \\"100000\\", \\"automaticRestart\\": {\\"set_minNeff\\": \\"yes\\", \\"__current_case__\\": 0, \\"minNeff\\": \\"200.0\\"}}, \\"advanced\\": {\\"osampx\\": \\"10\\", \\"osampxpdf\\": \\"4\\", \\"osampy\\": \\"10\\", \\"osampypdf\\": \\"4\\", \\"coviscale\\": \\"1.0\\", \\"nis\\": \\"1000\\", \\"burn\\": {\\"custom\\": \\"no\\", \\"__current_case__\\": 0}, \\"scalePrior\\": \\"0.3\\"}, \\"filter\\": {\\"nb\\": \\"0\\", \\"__current_case__\\": 0}, \\"geneXColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"geneYColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_counts\\": {\\"filetype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"plots\\": {\\"image_file_format\\": \\"png\\", \\"title\\": \\"\\", \\"removeFirstSamples\\": \\"-1\\", \\"nsampInPlot\\": \\"100000\\", \\"prettyBinsx\\": \\"-1\\", \\"prettyBinsy\\": \\"-1\\", \\"log1pColorScale\\": false, \\"splity\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3+galaxy0","type":"tool","uuid":"3fb4de69-9b40-4243-92fb-75553ff8ef09","when":null,"workflow_outputs":[{"label":"baredsc_qc_plots","output_name":"qc_plots","uuid":"005ef683-a231-4b3f-b8ef-125a1ffd2d91"},{"label":"baredsc_neff","output_name":"neff","uuid":"873dbd6d-8345-426b-8c90-809b478c9e24"},{"label":"baredsc_numpy","output_name":"output","uuid":"2f026269-bd89-4c20-a0df-bd1387f549d4"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.3+galaxy0","errors":null,"id":9,"input_connections":{"MCMC|outputs":{"id":8,"output_name":"output"},"MCMC|xmax":{"id":2,"output_name":"output"},"MCMC|ymax":{"id":4,"output_name":"output"},"advanced|getPVal":{"id":6,"output_name":"output"},"geneXColName":{"id":1,"output_name":"output"},"geneYColName":{"id":3,"output_name":"output"},"input_counts|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Combine multiple 2D Models","name":"MCMC"},{"description":"runtime parameter for tool Combine multiple 2D Models","name":"MCMC"},{"description":"runtime parameter for tool Combine multiple 2D Models","name":"MCMC"},{"description":"runtime parameter for tool Combine multiple 2D Models","name":"advanced"},{"description":"runtime parameter for tool Combine multiple 2D Models","name":"input_counts"}],"label":null,"name":"Combine multiple 2D Models","outputs":[{"name":"other_outputs","type":"input"},{"name":"pdf2d","type":"tabular"},{"name":"pdf2d_flat","type":"tabular"},{"name":"plot","type":"png"}],"position":{"left":1240,"top":199.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.3+galaxy0","tool_shed_repository":{"changeset_revision":"02fc39c8610d","name":"baredsc_combine_2d","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"MCMC\\": {\\"outputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"xmin\\": \\"0.0\\", \\"xmax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nx\\": \\"25\\", \\"minScalex\\": \\"0.1\\", \\"ymin\\": \\"0.0\\", \\"ymax\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ny\\": \\"25\\", \\"minScaley\\": \\"0.1\\", \\"scale\\": {\\"type\\": \\"Seurat\\", \\"__current_case__\\": 0, \\"targetSum\\": \\"10000.0\\"}, \\"seed\\": \\"1\\"}, \\"advanced\\": {\\"getPVal\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"osampx\\": \\"10\\", \\"osampxpdf\\": \\"4\\", \\"osampy\\": \\"10\\", \\"osampypdf\\": \\"4\\", \\"coviscale\\": \\"1.0\\", \\"nis\\": \\"1000\\", \\"scalePrior\\": \\"0.3\\"}, \\"filter\\": {\\"nb\\": \\"0\\", \\"__current_case__\\": 0}, \\"geneXColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"geneYColName\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_counts\\": {\\"filetype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"plots\\": {\\"image_file_format\\": \\"png\\", \\"title\\": \\"\\", \\"removeFirstSamples\\": \\"-1\\", \\"nsampInPlot\\": \\"100000\\", \\"prettyBinsx\\": \\"-1\\", \\"prettyBinsy\\": \\"-1\\", \\"log1pColorScale\\": false, \\"splity\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3+galaxy0","type":"tool","uuid":"e4eebae4-0952-4d21-8bb6-d16d1b5397b3","when":null,"workflow_outputs":[{"label":"combined_pdf2d","output_name":"pdf2d","uuid":"0fff7566-a7d9-4dac-b77b-416e6a2f56ef"},{"label":"combined_pdf2d_flat","output_name":"pdf2d_flat","uuid":"79eb786a-7465-4f24-aafa-d2520028527d"},{"label":"combined_plot","output_name":"plot","uuid":"db367996-7c6c-4cd3-8692-c0483fbc671c"},{"label":"combined_other_outputs","output_name":"other_outputs","uuid":"eca7f33c-3a1d-41d7-a56a-925167baf364"}]}},"tags":[],"uuid":"1936816b-958a-4140-b839-58be4cdae1a5","version":1},"readme":"# Single-Cell Mixture Analysis: baredSC Log-Normalized Models\\n\\nThese workflows allow to run a baredSC analysis from a table with counts in a single click. It uses models from 1 to N Gaussians and combine them. It uses the logNorm scale, 100 bins for 1 dimension and 25 bins on each axis in 2 dimensions.\\n\\n## Inputs dataset\\n\\n- Both workflows need a tabular dataset where each row is a cell. The tabular needs to have a header line with column names. There must be at least two columns: \'nCount_RNA\' and another one with the counts for the gene(s) of interest. A way to get such table in R from a Seurat object (`seurat.obj`) is:\\n\\n```r\\nmy.genes <- c(\\"Hoxa13\\", \\"Hoxd13\\")\\ndf <- cbind(seurat.obj[[]], # This will give you all metadata including nCount_RNA\\n            FetchData(seurat.obj, slot = \\"counts\\", vars = my.genes))\\n\\nwrite.table(df, \\"input_for_baredSC.txt\\", quote = F, sep = \\"\\\\t\\", row.names = F)\\n```\\n\\n## Inputs values\\n\\nFor the 1D:\\n\\n- Gene name: The name of the column with the counts of your gene of interest.\\n- Maximum value in logNorm: The maximum value to explore in PDF. This value should be large enough so the PDF is at 0 at this value.\\n- Maximum number of Gaussians to study: All models between models with 1 Gaussians to models with this number of Gaussians will be combined.\\n\\nFor the 2D:\\n\\n- Gene name for x axis: The name of the column with the counts of your gene in x axis.\\n- Gene name for y axis: The name of the column with the counts of your gene in y axis.\\n- maximum value in logNorm for x-axis: The maximum value to explore in PDF in the x axis. This value should be large enough so the PDF is at 0 at this value.\\n- maximum value in logNorm for y-axis: The maximum value to explore in PDF in the y axis. This value should be large enough so the PDF is at 0 at this value.\\n- Maximum number of Gaussians to study: All models between models with 1 2D-Gaussians to models with this number of 2D-Gaussians will be combined.\\n- compute p-value: Whether you want to get a p-value. As a consequence, less samples than available will be used for plots as p-value computation requires to have independent samples.\\n\\n## Processing\\n\\n- The workflow will generate paramater values from 1 to the maximum number of Gaussians to study.\\n- baredSC_1d or baredSC_2d is run for each of these number of Gaussians\\n- All models are combined into a single result.\\n","changelog":"# Changelog\\n\\n## [0.6] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.5+galaxy0`\\n\\n## [0.5] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n\\n## [0.4] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_1d/baredsc_1d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_1d/baredsc_combine_1d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_2d/baredsc_2d/1.1.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/baredsc_combine_2d/baredsc_combine_2d/1.1.3+galaxy0`\\n\\n## [0.3] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy0`\\n\\n## [0.2] - 2023-12-01\\n\\n### Add \'compute p-value\' parameter in 2d\\n\\n### Semi-automatic update\\n- all baredsc tools were updated from `1.1.2+galaxy0` to `1.1.2+galaxy1`\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n\\n## [0.1] - 2023-10-03\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell Mixture Analysis: baredSC 2D Log-Normalized Models\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Tabular with raw expression values\\"]@{ shape: doc }\\n1[\\"ℹ️ Gene name for x axis\\"]@{ shape: lean-l }\\n2[\\"ℹ️ maximum value in logNorm for x-axis\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Gene name for y axis\\"]@{ shape: lean-l }\\n4[\\"ℹ️ maximum value in logNorm for y-axis\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Maximum number of Gaussians to study\\"]@{ shape: lean-l }\\n6[\\"ℹ️ compute p-value\\"]@{ shape: lean-l }\\n7[\\"🛠️ generate_param_list_one_to_number\\"]@{ shape: subprocess }\\n5 --> 7\\n8[\\"baredSC 2d\\"]@{ shape: process }\\n7 --> 8\\n2 --> 8\\n4 --> 8\\n1 --> 8\\n3 --> 8\\n0 --> 8\\n9[\\"Combine multiple 2D Models\\"]@{ shape: process }\\n8 --> 9\\n2 --> 9\\n4 --> 9\\n6 --> 9\\n1 --> 9\\n3 --> 9\\n0 --> 9\\n```\\n\\n## generate_param_list_one_to_number\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Maximum number in param output\\"]@{ shape: lean-l }\\n1[\\"create first tabular with good number of rows.\\"]@{ shape: process }\\n0 --> 1\\n2[\\"add column with numbers starting at 1\\"]@{ shape: process }\\n1 --> 2\\n3[\\"retrieve numbers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"split numbers to collection\\"]@{ shape: process }\\n3 --> 4\\n5[\\"numbers to param\\"]@{ shape: process }\\n4 --> 5\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/baredsc/baredSC-2d-logNorm","dockstore_id":24063,"categories":[],"collections":["Transcriptomics","Single Cell"],"doi":"10.5281/zenodo.15078777","tests":[{"doc":"Test outline for baredSC_2d","job":{"Tabular with raw expression values":{"class":"File","path":"test-data/nih3t3_generated_2d_2.txt","filetype":"tabular"},"Gene name for x axis":"0.5_0_0_0.5_x","Gene name for y axis":"0.5_0_0_0.5_y","maximum value in logNorm for x-axis":"2.5","maximum value in logNorm for y-axis":"2.5","Maximum number of Gaussians to study":"4","compute p-value":"true"},"outputs":{"baredsc_numpy":{"element_tests":{"split_file_000000.tabular":{"asserts":{"has_size":{"value":1654994,"delta":100000}}},"split_file_000001.tabular":{"asserts":{"has_size":{"value":2650966,"delta":200000}}},"split_file_000002.tabular":{"asserts":{"has_size":{"value":39060170,"delta":3000000}}},"split_file_000003.tabular":{"asserts":{"has_size":{"value":49913268,"delta":5000000}}}}},"baredsc_qc_plots":{"element_tests":{"split_file_000000.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":24300,"delta":2000}}},"corner":{"asserts":{"has_size":{"value":300500,"delta":30000}}},"p":{"asserts":{"has_size":{"value":98200,"delta":10000}}}}},"split_file_000001.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":28600,"delta":2000}}},"corner":{"asserts":{"has_size":{"value":2000000,"delta":200000}}},"p":{"asserts":{"has_size":{"value":203100,"delta":20000}}}}},"split_file_000002.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":30200,"delta":3000}}},"corner":{"asserts":{"has_size":{"value":5100000,"delta":500000}}},"p":{"asserts":{"has_size":{"value":269500,"delta":20000}}}}},"split_file_000003.tabular":{"element_tests":{"convergence":{"asserts":{"has_size":{"value":31500,"delta":3000}}},"corner":{"asserts":{"has_size":{"value":9400000,"delta":900000}}},"p":{"asserts":{"has_size":{"value":373300,"delta":30000}}}}}}},"baredsc_neff":{"element_tests":{"split_file_000000.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"[1-2]?[0-9][0-9][0-9].[0-9]*"}}},"split_file_000001.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"[1-2]?[0-9][0-9][0-9].[0-9]*"}}},"split_file_000002.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"[1-2]?[0-9][0-9][0-9].[0-9]*"}}},"split_file_000003.tabular":{"asserts":{"has_n_lines":{"n":1},"has_line_matching":{"expression":"[1-2]?[0-9][0-9][0-9].[0-9]*"}}}}},"combined_other_outputs":{"element_tests":{"corr":{"asserts":{"has_n_lines":{"n":2},"has_text_matching":{"expression":"0.89[0-9]*"},"has_text":{"text":"mean\\tmedian\\tlow\\thigh\\tpval\\terror"}}},"individuals":{"asserts":{"has_size":{"value":54400,"delta":5000}}},"median":{"asserts":{"has_size":{"value":37100,"delta":3000}}}}},"combined_pdf2d":{"asserts":{"has_n_lines":{"n":26},"has_text":{"text":"xy\\t0.05\\t0.15"}}},"combined_pdf2d_flat":{"asserts":{"has_n_lines":{"n":626},"has_line":{"line":"x\\ty\\tlow\\tmean\\thigh\\tmedian"},"has_text":{"text":"0.05\\t0.05"}}},"combined_plot":{"path":"test-data/combined_2d_plot.png","compare":"sim_size","delta_frac":0.1}}}]}],"path":"./workflows/scRNAseq/baredsc"},{"version":1.2,"workflows":[{"name":"scrna-seq-fastq-to-matrix-10x-cellplex","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/scrna-seq-fastq-to-matrix-10x-cellplex.ga","testParameterFiles":["/scrna-seq-fastq-to-matrix-10x-cellplex-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"},{"name":"Mehmet Tekman","orcid":"0000-0002-4181-2676"},{"name":"Hans-Rudolf Hotz","orcid":"0000-0002-2799-424X"},{"name":"Daniel Blankenberg","orcid":"0000-0002-6833-9049"},{"name":"Wendi Bacon","orcid":"0000-0002-8170-8806"}],"definition":{"a_galaxy_workflow":"true","annotation":"Comprehensive preprocessing for 10X Genomics CellPlex multiplexed single-cell RNA-seq data. Processes Cell Multiplexing Oligo (CMO) FASTQ files with CITE-seq-Count including required CellPlex-specific translation steps. Simultaneously processes gene expression FASTQ files with STARsolo and DropletUtils for alignment and cell filtering, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-4181-2676","name":"Mehmet Tekman"},{"class":"Person","identifier":"https://orcid.org/0000-0002-2799-424X","name":"Hans-Rudolf Hotz"},{"class":"Person","identifier":"https://orcid.org/0000-0002-6833-9049","name":"Daniel Blankenberg"},{"class":"Person","identifier":"https://orcid.org/0000-0002-8170-8806","name":"Wendi Bacon"}],"format-version":"0.1","license":"MIT","release":"0.6.2","name":"Single-Cell RNA-seq Preprocessing: 10X Genomics CellPlex Multiplexed Samples","steps":{"0":{"annotation":"reads containing genes","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"reads containing genes","name":"fastq PE collection GEX"}],"label":"fastq PE collection GEX","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"eea58879-4368-4edf-ad0a-2dbd91de2db7","when":null,"workflow_outputs":[]},"1":{"annotation":"Choose your reference genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Choose your reference genome","name":"reference genome"}],"label":"reference genome","name":"Input parameter","outputs":[],"position":{"left":32.183349609375,"top":99.48333740234375},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"32f09d94-82fe-4c79-b86a-3af89fdb3088","when":null,"workflow_outputs":[]},"2":{"annotation":"Gene annotations","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Gene annotations","name":"gtf"}],"label":"gtf","name":"Input dataset","outputs":[],"position":{"left":67.95001220703125,"top":209.95001220703125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"8724cc04-fa9f-48df-ba31-6c47a03c9bde","when":null,"workflow_outputs":[]},"3":{"annotation":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","name":"cellranger_barcodes_3M-february-2018.txt"}],"label":"cellranger_barcodes_3M-february-2018.txt","name":"Input dataset","outputs":[],"position":{"left":111,"top":281.48333740234375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"bc4574f1-2a6f-44db-bed3-6d066d5ce42f","when":null,"workflow_outputs":[]},"4":{"annotation":"Barcode Size is same size of the Read","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Barcode Size is same size of the Read","name":"Barcode Size is same size of the Read"}],"label":"Barcode Size is same size of the Read","name":"Input parameter","outputs":[],"position":{"left":146,"top":394.66668701171875},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"12127fdb-d0d1-486c-a22e-532f714dbaaa","when":null,"workflow_outputs":[]},"5":{"annotation":"reads containing CMO","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"reads containing CMO","name":"fastq PE collection CMO"}],"label":"fastq PE collection CMO","name":"Input dataset collection","outputs":[],"position":{"left":237.68331909179688,"top":493.57719175292976},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"4a52f2cf-ee11-47ac-9f24-ce5e49b8434e","when":null,"workflow_outputs":[]},"6":{"annotation":"csv: first column is the sequence and second column is the name","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"csv: first column is the sequence and second column is the name","name":"sample name and CMO sequence collection"}],"label":"sample name and CMO sequence collection","name":"Input dataset collection","outputs":[],"position":{"left":264.6833190917969,"top":593.5771917529298},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"csv\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"dce97235-a4ba-4bfd-9d1f-60d3942f624b","when":null,"workflow_outputs":[]},"7":{"annotation":"Used by CITE-seq-Count, does not really seem to impact result. You can use 24000.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Used by CITE-seq-Count, does not really seem to impact result. You can use 24000.","name":"Number of expected cells"}],"label":"Number of expected cells","name":"Input parameter","outputs":[],"position":{"left":306.2167053222656,"top":705.6105169482423},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"5ca25d6a-112f-4870-b5a4-163a055360c0","when":null,"workflow_outputs":[]},"8":{"annotation":"","id":8,"input_connections":{"Barcode Size is same size of the Read":{"id":4,"input_subworkflow_step_id":4,"output_name":"output"},"cellranger_barcodes_3M-february-2018.txt":{"id":3,"input_subworkflow_step_id":3,"output_name":"output"},"fastq PE collection":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"gtf":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"},"reference genome":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[{"description":"","name":"process GEX reads"}],"label":"process GEX reads","name":"scRNA-seq_preprocessing_10X_v3_Bundle","outputs":[],"position":{"left":662.4507693654512,"top":75.05904568073197},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-4181-2676","name":"Mehmet Tekman"},{"class":"Person","identifier":"https://orcid.org/0000-0002-2799-424X","name":"Hans-Rudolf Hotz"},{"class":"Person","identifier":"https://orcid.org/0000-0002-6833-9049","name":"Daniel Blankenberg"},{"class":"Person","identifier":"https://orcid.org/0000-0002-8170-8806","name":"Wendi Bacon"}],"format-version":"0.1","license":"MIT","name":"scRNA-seq_preprocessing_10X_v3_Bundle","steps":{"0":{"annotation":"Collection of Pairs with fastqs one item per sample","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection of Pairs with fastqs one item per sample","name":"fastq PE collection"}],"label":"fastq PE collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"eea58879-4368-4edf-ad0a-2dbd91de2db7","when":null,"workflow_outputs":[]},"1":{"annotation":"Choose your reference genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Choose your reference genome","name":"reference genome"}],"label":"reference genome","name":"Input parameter","outputs":[],"position":{"left":58.94316181636054,"top":104},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"adf0f555-5974-44b9-a067-37e4054cd662","when":null,"workflow_outputs":[]},"2":{"annotation":"Gene annotations","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Gene annotations","name":"gtf"}],"label":"gtf","name":"Input dataset","outputs":[],"position":{"left":121.00981220698554,"top":197},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"8724cc04-fa9f-48df-ba31-6c47a03c9bde","when":null,"workflow_outputs":[]},"3":{"annotation":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","name":"cellranger_barcodes_3M-february-2018.txt"}],"label":"cellranger_barcodes_3M-february-2018.txt","name":"Input dataset","outputs":[],"position":{"left":194.97648701167304,"top":293},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"bc4574f1-2a6f-44db-bed3-6d066d5ce42f","when":null,"workflow_outputs":[]},"4":{"annotation":"Barcode Size is same size of the Read","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Barcode Size is same size of the Read","name":"Barcode Size is same size of the Read"}],"label":"Barcode Size is same size of the Read","name":"Input parameter","outputs":[],"position":{"left":260.683349609375,"top":414.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a9d14d27-ca59-444d-9fe1-485b3354b9a7","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1","errors":null,"id":5,"input_connections":{"refGenomeSource|GTFconditional|genomeDir":{"id":1,"output_name":"output"},"refGenomeSource|GTFconditional|sjdbGTFfile":{"id":2,"output_name":"output"},"sc|input_types|input_collection":{"id":0,"output_name":"output"},"sc|soloBarcodeReadLength":{"id":4,"output_name":"output"},"sc|soloCBwhitelist":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool RNA STARSolo","name":"sc"},{"description":"runtime parameter for tool RNA STARSolo","name":"sc"}],"label":null,"name":"RNA STARSolo","outputs":[{"name":"output_log","type":"txt"},{"name":"output_genes","type":"tsv"},{"name":"output_barcodes","type":"tsv"},{"name":"output_matrix","type":"mtx"},{"name":"output_BAM","type":"bam"},{"name":"output_stats","type":"txt"},{"name":"reads_per_gene","type":"tabular"}],"position":{"left":568.7432106444855,"top":374},"post_job_actions":{"HideDatasetActionoutput_BAM":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_BAM"},"HideDatasetActionoutput_barcodes":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_barcodes"},"HideDatasetActionoutput_genes":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_genes"},"HideDatasetActionoutput_log":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_log"},"HideDatasetActionoutput_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_matrix"},"HideDatasetActionoutput_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_stats"},"HideDatasetActionreads_per_gene":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reads_per_gene"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1","tool_shed_repository":{"changeset_revision":"381a32c51141","name":"rna_starsolo","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"outWig\\": {\\"outWigType\\": \\"None\\", \\"__current_case__\\": 0, \\"outWigStrand\\": \\"false\\"}, \\"refGenomeSource\\": {\\"geneSource\\": \\"indexed\\", \\"__current_case__\\": 0, \\"GTFconditional\\": {\\"GTFselect\\": \\"without-gtf-with-gtf\\", \\"__current_case__\\": 1, \\"genomeDir\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfeatureExon\\": \\"exon\\", \\"sjdbOverhang\\": \\"100\\"}}, \\"sc\\": {\\"solo_type\\": \\"CB_UMI_Simple\\", \\"__current_case__\\": 0, \\"input_types\\": {\\"use\\": \\"list_paired\\", \\"__current_case__\\": 1, \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"soloCBwhitelist\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"params\\": {\\"chemistry\\": \\"Cv3\\", \\"__current_case__\\": 1}, \\"soloBarcodeReadLength\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"umidedup\\": {\\"soloUMIdedup\\": \\"1MM_CR\\", \\"__current_case__\\": 4, \\"soloUMIfiltering\\": \\"-\\"}, \\"soloCBmatchWLtype\\": \\"1MM_multi\\"}, \\"solo\\": {\\"soloStrand\\": \\"Forward\\", \\"soloFeatures\\": \\"Gene\\", \\"wasp_conditional\\": {\\"waspOutputMode\\": \\"\\", \\"__current_case__\\": 1}, \\"filter\\": {\\"filter_type\\": \\"no_filter\\", \\"__current_case__\\": 3, \\"output_raw\\": \\"true\\"}, \\"soloOutFormatFeaturesGeneField3\\": \\"Gene Expression\\", \\"outSAMattributes\\": [\\"NH\\", \\"HI\\", \\"AS\\", \\"nM\\", \\"GX\\", \\"GN\\", \\"CB\\", \\"UB\\"], \\"quantModeGene\\": true, \\"outSAMunmapped\\": false, \\"outSAMmapqUnique\\": \\"60\\", \\"junction_limits\\": {\\"limitOutSJoneRead\\": \\"1000\\", \\"limitOutSJcollapsed\\": \\"1000000\\", \\"limitSjdbInsertNsj\\": \\"1000000\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.11a+galaxy1","type":"tool","uuid":"cef740b0-fa95-4542-b08f-dbfca4bfb3e5","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":6,"input_connections":{"results_0|software_cond|output_0|type|input":{"id":5,"output_name":"output_log"},"results_0|software_cond|output_1|type|input":{"id":5,"output_name":"reads_per_gene"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":"multiQC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":848.0098122069855,"top":366.98333740234375},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"a16bf20e-2bc9-45e7-a0a3-b5590212b973","when":null,"workflow_outputs":[{"label":"MultiQC on input dataset(s): Webpage","output_name":"html_report","uuid":"eb55ba68-0473-4b09-87b9-007be3a86098"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dropletutils/dropletutils/1.10.0+galaxy2","errors":null,"id":7,"input_connections":{"tenx_format|input":{"id":5,"output_name":"output_matrix"},"tenx_format|input_barcodes":{"id":5,"output_name":"output_barcodes"},"tenx_format|input_genes":{"id":5,"output_name":"output_genes"}},"inputs":[{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"},{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"},{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"}],"label":"filter cells","name":"DropletUtils","outputs":[{"name":"barcodes_out","type":"tsv"},{"name":"genes_out","type":"tsv"},{"name":"matrix_out","type":"mtx"},{"name":"plot","type":"png"},{"name":"table","type":"tsv"}],"position":{"left":848.0098122069855,"top":640.9833374023438},"post_job_actions":{"HideDatasetActionbarcodes_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"barcodes_out"},"HideDatasetActiongenes_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"genes_out"},"HideDatasetActionmatrix_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"matrix_out"},"HideDatasetActionplot":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plot"},"HideDatasetActiontable":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"table"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dropletutils/dropletutils/1.10.0+galaxy2","tool_shed_repository":{"changeset_revision":"a9caad671439","name":"dropletutils","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation\\": {\\"use\\": \\"filter\\", \\"__current_case__\\": 0, \\"method\\": {\\"use\\": \\"emptydrops\\", \\"__current_case__\\": 1, \\"lower\\": \\"100\\", \\"fdr_thresh\\": \\"0.01\\"}, \\"outformat\\": \\"directory\\"}, \\"seed\\": \\"100\\", \\"tenx_format\\": {\\"use\\": \\"directory\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_genes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.0+galaxy2","type":"tool","uuid":"2ba61d0f-82ae-4044-a7ad-41df1cf8c9b6","when":null,"workflow_outputs":[{"label":"DropletUtils Barcodes","output_name":"barcodes_out","uuid":"0fea83f0-7416-457b-91b0-9fcc38645ceb"},{"label":"DropletUtils Genes","output_name":"genes_out","uuid":"a3046d47-51e9-4b86-9a71-65a2352b0336"},{"label":"DropletUtils Matrix","output_name":"matrix_out","uuid":"75e8d5cc-b2db-45ee-9a41-0197535e7fb1"}]},"8":{"annotation":"","id":8,"input_connections":{"STARsolo Barcodes":{"id":7,"input_subworkflow_step_id":1,"output_name":"barcodes_out"},"STARsolo Genes":{"id":7,"input_subworkflow_step_id":0,"output_name":"genes_out"},"STARsolo Matrix Gene Counts":{"id":7,"input_subworkflow_step_id":2,"output_name":"matrix_out"}},"inputs":[{"description":"","name":"Re-organize STAR-solo output"}],"label":"Re-organize STAR-solo output","name":"Re-organize-STAR-solo-output","outputs":[],"position":{"left":1125.976487011673,"top":774},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Re-organize-STAR-solo-output","steps":{"0":{"annotation":"Collection named Genes raw or Genes filtered from STARsolo output","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection named Genes raw or Genes filtered from STARsolo output","name":"STARsolo Genes"}],"label":"STARsolo Genes","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3e312de3-0895-4b46-ac51-9c1c7e341194","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","name":"STARsolo Barcodes"}],"label":"STARsolo Barcodes","name":"Input dataset collection","outputs":[],"position":{"left":23,"top":126},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4f8d1d99-0adc-40a8-a53b-ce9c1c614aeb","when":null,"workflow_outputs":[]},"2":{"annotation":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","name":"STARsolo Matrix Gene Counts"}],"label":"STARsolo Matrix Gene Counts","name":"Input dataset collection","outputs":[],"position":{"left":84,"top":249},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"266958e3-aacf-4b7f-8f74-6cbcdd5db4c0","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":3,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":337,"top":438},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tsv\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"ff023a14-a401-4ede-8dfb-2187d2d8ad84","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":4,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":579,"top":63},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_genes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"e8eed7c4-a41f-4bf7-bf5d-b4ed5023f1b8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":584,"top":205},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_barcodes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"de152d11-daaa-4c96-ada8-7917c17e9543","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":581,"top":383},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_matrix\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ac2f7a73-a742-4977-b902-aba871059cb6","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":7,"input_connections":{"how|labels":{"id":4,"output_name":"outfile"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":829,"top":58},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"d627ba62-eeb9-49a3-b46f-120dd8c8e3d8","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":8,"input_connections":{"how|labels":{"id":5,"output_name":"outfile"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":827,"top":213},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"e4fad6a7-30e3-4b66-a9bd-5f11f7ea18db","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":9,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":823,"top":364},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"fe80aa41-a8e4-4000-af6b-6c9713de9006","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":10,"input_connections":{"inputs_0|input":{"id":9,"output_name":"output"},"inputs_1|input":{"id":8,"output_name":"output"},"inputs_2|input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1064,"top":128},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"a494a5d5-d908-448e-a406-c7f78d81c10b","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":11,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":1284,"top":128},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"All matrices in cell-ranger format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1, 2], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\1\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"95eb63d6-f1a2-4f0d-b426-fc46a5133471","when":null,"workflow_outputs":[{"label":"STARsolo cell ranger-like output","output_name":"output","uuid":"3a6385df-d937-4c8a-9430-9bafe0cb2f57"}]}},"tags":[],"uuid":"80a6f743-ad90-42e3-87af-0537f22488da"},"tool_id":null,"type":"subworkflow","uuid":"697a68da-9cb9-469b-8102-30c3a3ee589f","when":null,"workflow_outputs":[{"label":"Seurat input for gene expression (filtered)","output_name":"STARsolo cell ranger-like output","uuid":"6cb072df-9e26-456b-ab4f-bc95345b02b4"}]}},"tags":[],"uuid":"1be69a4e-9849-477a-9ca4-8054b67ba6f8"},"tool_id":null,"type":"subworkflow","uuid":"20cd1d79-f289-40d5-a239-ef1f18c0df35","when":null,"workflow_outputs":[{"label":"MultiQC_STARsolo","output_name":"MultiQC on input dataset(s): Webpage","uuid":"84c213a8-9b9a-413e-b944-be94e6ac80ac"},{"label":"DropletUtils Matrix","output_name":"DropletUtils Matrix","uuid":"626c806e-c6d2-4555-8859-8cbb96b361b9"},{"label":"Seurat input for gene expression (filtered)","output_name":"Seurat input for gene expression (filtered)","uuid":"1fc82f04-a2b3-4036-b07f-09e27045d251"},{"label":"DropletUtils Genes","output_name":"DropletUtils Genes","uuid":"9fa61da5-c27a-43aa-a3a2-b36b31a835c5"},{"label":"DropletUtils Barcodes","output_name":"DropletUtils Barcodes","uuid":"844b5fe0-22cb-42b3-bc08-11c2f359178b"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/cite_seq_count/cite_seq_count/1.4.4+galaxy0","errors":null,"id":9,"input_connections":{"expected_cells":{"id":7,"output_name":"output"},"input_types|input_collection":{"id":5,"output_name":"output"},"tags":{"id":6,"output_name":"output"},"whitelist":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool CITE-seq-Count","name":"input_types"}],"label":null,"name":"CITE-seq-Count","outputs":[{"name":"report","type":"yaml"},{"name":"output_features","type":"tsv"},{"name":"output_barcodes","type":"tsv"},{"name":"output_matrix","type":"mtx"},{"name":"output_features_filtered","type":"tsv"},{"name":"output_barcodes_filtered","type":"tsv"},{"name":"output_matrix_filtered","type":"mtx"}],"position":{"left":667.6833801269531,"top":481.593854350586},"post_job_actions":{"HideDatasetActionoutput_barcodes":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_barcodes"},"HideDatasetActionoutput_barcodes_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_barcodes_filtered"},"HideDatasetActionoutput_features":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_features"},"HideDatasetActionoutput_features_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_features_filtered"},"HideDatasetActionoutput_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_matrix"},"HideDatasetActionoutput_matrix_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_matrix_filtered"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/cite_seq_count/cite_seq_count/1.4.4+galaxy0","tool_shed_repository":{"changeset_revision":"3df3d1b51110","name":"cite_seq_count","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bc_collapsing_dist\\": \\"1\\", \\"dense\\": false, \\"expected_cells\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"first_n\\": \\"0\\", \\"input_types\\": {\\"use\\": \\"list_paired\\", \\"__current_case__\\": 1, \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"max_error\\": \\"2\\", \\"no_umi_correction\\": false, \\"params\\": {\\"chemistry\\": \\"v3\\", \\"__current_case__\\": 1}, \\"sliding_window\\": false, \\"start_trim\\": \\"0\\", \\"tags\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"umi_collapsing_dist\\": \\"2\\", \\"unmapped\\": {\\"output\\": \\"false\\", \\"__current_case__\\": 0}, \\"whitelist\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.4+galaxy0","type":"tool","uuid":"0aa67706-b4c9-4d4f-98b0-d019ed33db0b","when":null,"workflow_outputs":[{"label":"CITE-seq-Count report","output_name":"report","uuid":"ee9c2582-bbae-4081-8367-67eb36ad1c49"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":10,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":8,"output_name":"Seurat input for gene expression (filtered)"}},"inputs":[],"label":"Rename STAR-solo output","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":1380.7500305175781,"top":364.6771978564454},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Seurat input for gene expression (filtered)"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"c67e69ba-ebd3-4331-9e8e-a9d8c99c3ae6","when":null,"workflow_outputs":[{"label":"Renamed Seurat input for gene expression (filtered)","output_name":"data_param","uuid":"1d245db6-d570-4c68-a830-b4fc6ce46499"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":11,"input_connections":{"infile":{"id":9,"output_name":"output_barcodes_filtered"}},"inputs":[],"label":"translate 10x barcode v2 to 10x barcode v1","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":928.6666564941406,"top":654.5771917529298},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{split($1,a,\\\\\\"\\\\\\"); a[8]=a[8]==\\\\\\"A\\\\\\"?\\\\\\"T\\\\\\":a[8]==\\\\\\"T\\\\\\"?\\\\\\"A\\\\\\":a[8]==\\\\\\"C\\\\\\"?\\\\\\"G\\\\\\":\\\\\\"C\\\\\\"; a[9]=a[9]==\\\\\\"A\\\\\\"?\\\\\\"T\\\\\\":a[9]==\\\\\\"T\\\\\\"?\\\\\\"A\\\\\\":a[9]==\\\\\\"C\\\\\\"?\\\\\\"G\\\\\\":\\\\\\"C\\\\\\"; print a[1]a[2]a[3]a[4]a[5]a[6]a[7]a[8]a[9]a[10]a[11]a[12]a[13]a[14]a[15]a[16]}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"72572815-061b-47f0-8234-d99d5c28261f","when":null,"workflow_outputs":[]},"12":{"annotation":"","id":12,"input_connections":{"STARsolo Barcodes":{"id":11,"input_subworkflow_step_id":1,"output_name":"outfile"},"STARsolo Genes":{"id":9,"input_subworkflow_step_id":0,"output_name":"output_features_filtered"},"STARsolo Matrix Gene Counts":{"id":9,"input_subworkflow_step_id":2,"output_name":"output_matrix_filtered"}},"inputs":[{"description":"","name":"Re-organize CITE-seq-Count output"}],"label":"Re-organize CITE-seq-Count output","name":"Re-organize-STAR-solo-output","outputs":[],"position":{"left":1173.6666564941406,"top":750.6105169482423},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Re-organize-STAR-solo-output","steps":{"0":{"annotation":"Collection named Genes raw or Genes filtered from STARsolo output","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection named Genes raw or Genes filtered from STARsolo output","name":"STARsolo Genes"}],"label":"STARsolo Genes","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3e312de3-0895-4b46-ac51-9c1c7e341194","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","name":"STARsolo Barcodes"}],"label":"STARsolo Barcodes","name":"Input dataset collection","outputs":[],"position":{"left":23,"top":126},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4f8d1d99-0adc-40a8-a53b-ce9c1c614aeb","when":null,"workflow_outputs":[]},"2":{"annotation":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","name":"STARsolo Matrix Gene Counts"}],"label":"STARsolo Matrix Gene Counts","name":"Input dataset collection","outputs":[],"position":{"left":84,"top":249},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"266958e3-aacf-4b7f-8f74-6cbcdd5db4c0","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":3,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":337,"top":438},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tsv\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"ff023a14-a401-4ede-8dfb-2187d2d8ad84","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":4,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":579,"top":63},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_genes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"e8eed7c4-a41f-4bf7-bf5d-b4ed5023f1b8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":584,"top":205},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_barcodes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"de152d11-daaa-4c96-ada8-7917c17e9543","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":581,"top":383},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_matrix\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ac2f7a73-a742-4977-b902-aba871059cb6","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":7,"input_connections":{"how|labels":{"id":4,"output_name":"outfile"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":829,"top":58},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"d627ba62-eeb9-49a3-b46f-120dd8c8e3d8","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":8,"input_connections":{"how|labels":{"id":5,"output_name":"outfile"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":827,"top":213},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"e4fad6a7-30e3-4b66-a9bd-5f11f7ea18db","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":9,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":823,"top":364},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"fe80aa41-a8e4-4000-af6b-6c9713de9006","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":10,"input_connections":{"inputs_0|input":{"id":9,"output_name":"output"},"inputs_1|input":{"id":8,"output_name":"output"},"inputs_2|input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1064,"top":128},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"a494a5d5-d908-448e-a406-c7f78d81c10b","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":11,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":1284,"top":128},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"All matrices in cell-ranger format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1, 2], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\1\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"95eb63d6-f1a2-4f0d-b426-fc46a5133471","when":null,"workflow_outputs":[{"label":"STARsolo cell ranger-like output","output_name":"output","uuid":"3a6385df-d937-4c8a-9430-9bafe0cb2f57"}]}},"tags":[],"uuid":"6ddcf5bf-a002-4b30-86f2-a136d133a53e"},"tool_id":null,"type":"subworkflow","uuid":"1626e9ae-6466-4ab1-852b-e237a5d3b601","when":null,"workflow_outputs":[{"label":"Seurat input for CMO (UMI)","output_name":"STARsolo cell ranger-like output","uuid":"f307e08b-8a9b-421d-b2dd-378c801b474e"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":13,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":12,"output_name":"STARsolo cell ranger-like output"}},"inputs":[],"label":"rename CITE-seq output","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":1450.6999816894531,"top":841.5938543505861},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Seurat input for CMO (UMI)"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"3e819040-a643-4aa7-949c-42becbf21edd","when":null,"workflow_outputs":[{"label":"Renamed Seurat input for CMO (UMI)","output_name":"data_param","uuid":"a490fb25-f9f4-49f3-8e1c-a36a2edf43fc"}]}},"tags":["#single-cell"],"uuid":"470536e7-ac58-4859-94d6-2085be9c2e17","version":3},"readme":"# Single-Cell RNA-seq Preprocessing: 10X Genomics Data to Analysis-Ready Formats\\n\\nThese workflows are inspired by the [training material](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html). The output is in a \'bundle\' format with three files (one matrix, one with genes, one with barcodes) which is similar to the cellranger output format, and is compatible with any Read10X function (Seurat or Scanpy). There are 2 types of output collections: either one collection with all matrices, one collection with all genes and one collection with all barcodes (compatible with post processing in galaxy) or a single nested collection with one sub-collection per sample with the 3 files (easier for local downstream analysis).\\n\\nBoth workflows are designed for fastqs from 10X libraries v3. One is for regular 10X library (one library per sample), while the other one is for CellPlex 10X library which allows to multiplex samples using CMOs (see [this blog article](https://www.10xgenomics.com/blog/answering-your-questions-about-sample-multiplexing-for-single-cell-gene-expression)).\\n\\n## Input datasets\\n\\n- Specific for each experiment:\\n    - For both workflows: you need a list of pairs of fastqs with gene expression.\\n    - For CellPlex: you need in addition a list of pairs of fastqs with CMO.\\n    - For CellPlex: you need a list of csv which describes samples and CMO used:\\n        - first column is the sequence and second column is the name\\n    /!\\\\ The order of samples need to be exactly the same between the collection of fastqs of CMO and the collection of csv.\\n\\n- Common for all experiments:\\n    - Gene annotations: A gtf file with gene locations\\n    - List of barcodes used by 10X. You can download it at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz\\n\\n## Input values\\n\\n- reference genome: this genome needs to be available for STAR\\n- Barcode Size is same size of the Read: if the length of your R1 of GEX matches the size of cell barcode + UMI set to true. If your R1 contains trailling A, put false.\\n- number of cells: If you make it too large no cell barcode correction will be performed to demultiplex CMOs.\\n\\n## Processing\\n- Gene expression processing:\\n    - Reads are aligned to the genome, asigned to genes, cell barcode and UMI with STAR Solo\\n    - MultiQC report the mapping rate and the number of reads attributed to genes\\n    - The output of STAR Solo is filtered with Droplet Utils to remove cellular barcodes which are probably empty.\\n    - The output of Droplet Utils is reorganized to be:\\n```\\nMain Collection:\\n    - Sample 1:\\n        - matrix.mtx\\n        - barcodes.tsv\\n        - genes.tsv\\n    - Sample 2:\\n        - matrix.mtx\\n        - barcodes.tsv\\n        - genes.tsv\\n...\\n```\\nFor the CellPlex workflow:\\n- CMO processing:\\n    - CITE-Seq Count is used to asign reads and generate a matrix where \'genes\' are the CMO and \'unmapped\'.\\n    - Cellular barcodes are translated to match the cellular barcodes of Gene expression see [this article](https://kb.10xgenomics.com/hc/en-us/articles/360031133451-Why-is-there-a-discrepancy-in-the-3M-february-2018-txt-barcode-whitelist-).\\n    - Reorganize the output with UMI matrices to match the same structure as gene expression matrices.\\n\\n## Test data\\n\\nThe test dataset has been produced to make it as small as possible in order to make the workflow pass on CI.\\n\\n- The CMO reads come from [zenodo](https://zenodo.org/records/10229382) and have been sampled to 0.1 with seqtk.\\n- The GEX reads come from SRR13948489 but have been subsetted to the cells selected in the above zenodo.\\n","changelog":"# Changelog\\n\\n## [0.6.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.6.1] - 2025-03-03\\n\\n### Added\\n\\n- Added individual DropletUtils outputs for processing with scanpy tools\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.5] - 2024-09-25\\n\\n### Manual update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.4] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n\\n## [0.3] - 2024-02-12\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.10b+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n\\n## [0.2] - 2024-02-05\\n\\n### Tool updates\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4`\\n- `pick_value` was replaced by `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0`\\n\\n\\n\\n## [0.1] - 2023-12-21\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell RNA-seq Preprocessing: 10X Genomics CellPlex Multiplexed Samples\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ fastq PE collection GEX\\"]@{ shape: docs }\\n1[\\"ℹ️ reference genome\\"]@{ shape: lean-l }\\n2[\\"ℹ️ gtf\\"]@{ shape: doc }\\n3[\\"ℹ️ cellranger_barcodes_3M-february-2018.txt\\"]@{ shape: doc }\\n4[\\"ℹ️ Barcode Size is same size of the Read\\"]@{ shape: lean-l }\\n5[\\"ℹ️ fastq PE collection CMO\\"]@{ shape: docs }\\n6[\\"ℹ️ sample name and CMO sequence collection\\"]@{ shape: docs }\\n7[\\"ℹ️ Number of expected cells\\"]@{ shape: lean-l }\\n8[\\"🛠️ process GEX reads\\"]@{ shape: subprocess }\\n4 --> 8\\n3 --> 8\\n0 --> 8\\n2 --> 8\\n1 --> 8\\n9[\\"CITE-seq-Count\\"]@{ shape: process }\\n7 --> 9\\n5 --> 9\\n6 --> 9\\n3 --> 9\\n10[\\"Rename STAR-solo output\\"]@{ shape: process }\\n8 --> 10\\n11[\\"translate 10x barcode v2 to 10x barcode v1\\"]@{ shape: process }\\n9 --> 11\\n12[\\"🛠️ Re-organize CITE-seq-Count output\\"]@{ shape: subprocess }\\n11 --> 12\\n9 --> 12\\n9 --> 12\\n13[\\"rename CITE-seq output\\"]@{ shape: process }\\n12 --> 13\\n```\\n\\n## scRNA-seq_preprocessing_10X_v3_Bundle\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ fastq PE collection\\"]@{ shape: docs }\\n1[\\"ℹ️ reference genome\\"]@{ shape: lean-l }\\n2[\\"ℹ️ gtf\\"]@{ shape: doc }\\n3[\\"ℹ️ cellranger_barcodes_3M-february-2018.txt\\"]@{ shape: doc }\\n4[\\"ℹ️ Barcode Size is same size of the Read\\"]@{ shape: lean-l }\\n5[\\"RNA STARSolo\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n0 --> 5\\n4 --> 5\\n3 --> 5\\n6[\\"multiQC\\"]@{ shape: process }\\n5 --> 6\\n5 --> 6\\n7[\\"filter cells\\"]@{ shape: process }\\n5 --> 7\\n5 --> 7\\n5 --> 7\\n8[\\"🛠️ Re-organize STAR-solo output\\"]@{ shape: subprocess }\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n```\\n\\n## Re-organize-STAR-solo-output\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ STARsolo Genes\\"]@{ shape: docs }\\n1[\\"ℹ️ STARsolo Barcodes\\"]@{ shape: docs }\\n2[\\"ℹ️ STARsolo Matrix Gene Counts\\"]@{ shape: docs }\\n3[\\"Extract element identifiers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Replace Text\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Replace Text\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Replace Text\\"]@{ shape: process }\\n3 --> 6\\n7[\\"Relabel identifiers\\"]@{ shape: process }\\n4 --> 7\\n0 --> 7\\n8[\\"Relabel identifiers\\"]@{ shape: process }\\n5 --> 8\\n1 --> 8\\n9[\\"Relabel identifiers\\"]@{ shape: process }\\n6 --> 9\\n2 --> 9\\n10[\\"Merge collections\\"]@{ shape: process }\\n9 --> 10\\n8 --> 10\\n7 --> 10\\n11[\\"Apply rules\\"]@{ shape: process }\\n10 --> 11\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-cellplex","dockstore_id":24781,"categories":[],"collections":["Transcriptomics","Single Cell"],"doi":"10.5281/zenodo.15078821","tests":[{"doc":"Test outline for scrna-seq-fastq-to-matrix-10X-cellplex.ga","job":{"fastq PE collection GEX":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"subsample","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/10412836/files/GEX_R1.fastqsanger.gz","filetype":"fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/10412836/files/GEX_R2.fastqsanger.gz","filetype":"fastqsanger.gz"}]}]},"reference genome":"dm6","Barcode Size is same size of the Read":false,"gtf":{"class":"File","location":"https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz","decompress":true,"filetype":"gtf"},"cellranger_barcodes_3M-february-2018.txt":{"class":"File","path":"test-data/all_cell_barcodes_both_versions.txt","filetype":"tabular"},"fastq PE collection CMO":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"subsample","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/10412836/files/CMO_R1.fastqsanger.gz","filetype":"fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/10412836/files/CMO_R2.fastqsanger.gz","filetype":"fastqsanger.gz"}]}]},"sample name and CMO sequence collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"subsample","location":"https://zenodo.org/records/10229382/files/CMO.csv","filetype":"csv"}]},"Number of expected cells":500},"outputs":{"MultiQC_STARsolo":{"asserts":{"has_text_matching":{"expression":"<span class=\\"val\\">33.[0-9]<span class=[\'\\"]mqc_small_space[\'\\"]>"}}},"Seurat input for gene expression (filtered)":{"attributes":{"collection_type":"list:list"},"element_tests":{"subsample":{"elements":{"matrix":{"asserts":{"has_line":{"line":"23932 3 1171"}}},"barcodes":{"asserts":{"has_line":{"line":"CACATGATCATAAGGA"}}},"genes":{"asserts":{"has_line":{"line":"FBgn0250732\\tgfzf"}}}}}}},"CITE-seq-Count report":{"attributes":{"collection_type":"list"},"element_tests":{"subsample":{"asserts":[{"that":"has_line","line":"CITE-seq-Count Version: 1.4.4"},{"that":"has_line","line":"Reads processed: 116993"},{"that":"has_line","line":"Percentage mapped: 99"},{"that":"has_line","line":"Percentage unmapped: 1"}]}}},"Seurat input for CMO (UMI)":{"attributes":{"collection_type":"list:list"},"element_tests":{"subsample":{"elements":{"matrix":{"asserts":{"has_line":{"line":"8 1218 2289"}}},"barcodes":{"asserts":{"has_line":{"line":"CACATGATCATAAGGA"}}},"genes":{"asserts":{"has_line":{"line":"mESC.rep1-CGCGATATGGTCGGA"}}}}}}}}}]},{"name":"scrna-seq-fastq-to-matrix-10x-v3","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/scrna-seq-fastq-to-matrix-10x-v3.ga","testParameterFiles":["/scrna-seq-fastq-to-matrix-10x-v3-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"},{"name":"Mehmet Tekman","orcid":"0000-0002-4181-2676"},{"name":"Hans-Rudolf Hotz","orcid":"0000-0002-2799-424X"},{"name":"Daniel Blankenberg","orcid":"0000-0002-6833-9049"},{"name":"Wendi Bacon","orcid":"0000-0002-8170-8806"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete preprocessing pipeline for 10X Genomics v3 single-cell RNA-seq data. Aligns raw FASTQ files using STARsolo, performs cell calling and quality filtering with DropletUtils, and formats outputs for seamless import into Seurat/Scanpy (Read10X function).","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-4181-2676","name":"Mehmet Tekman"},{"class":"Person","identifier":"https://orcid.org/0000-0002-2799-424X","name":"Hans-Rudolf Hotz"},{"class":"Person","identifier":"https://orcid.org/0000-0002-6833-9049","name":"Daniel Blankenberg"},{"class":"Person","identifier":"https://orcid.org/0000-0002-8170-8806","name":"Wendi Bacon"}],"format-version":"0.1","license":"MIT","release":"0.6.2","name":"Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat and Scanpy Compatible Format","steps":{"0":{"annotation":"Collection of Pairs with fastqs one item per sample","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection of Pairs with fastqs one item per sample","name":"fastq PE collection"}],"label":"fastq PE collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"eea58879-4368-4edf-ad0a-2dbd91de2db7","when":null,"workflow_outputs":[]},"1":{"annotation":"Choose your reference genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Choose your reference genome","name":"reference genome"}],"label":"reference genome","name":"Input parameter","outputs":[],"position":{"left":58.94316181636054,"top":104},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"adf0f555-5974-44b9-a067-37e4054cd662","when":null,"workflow_outputs":[]},"2":{"annotation":"Gene annotations","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Gene annotations","name":"gtf"}],"label":"gtf","name":"Input dataset","outputs":[],"position":{"left":121.00981220698554,"top":197},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"8724cc04-fa9f-48df-ba31-6c47a03c9bde","when":null,"workflow_outputs":[]},"3":{"annotation":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Available at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz","name":"cellranger_barcodes_3M-february-2018.txt"}],"label":"cellranger_barcodes_3M-february-2018.txt","name":"Input dataset","outputs":[],"position":{"left":194.97648701167304,"top":293},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"bc4574f1-2a6f-44db-bed3-6d066d5ce42f","when":null,"workflow_outputs":[]},"4":{"annotation":"Barcode Size is same size of the Read","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Barcode Size is same size of the Read","name":"Barcode Size is same size of the Read"}],"label":"Barcode Size is same size of the Read","name":"Input parameter","outputs":[],"position":{"left":260.683349609375,"top":414.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a9d14d27-ca59-444d-9fe1-485b3354b9a7","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1","errors":null,"id":5,"input_connections":{"refGenomeSource|GTFconditional|genomeDir":{"id":1,"output_name":"output"},"refGenomeSource|GTFconditional|sjdbGTFfile":{"id":2,"output_name":"output"},"sc|input_types|input_collection":{"id":0,"output_name":"output"},"sc|soloBarcodeReadLength":{"id":4,"output_name":"output"},"sc|soloCBwhitelist":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool RNA STARSolo","name":"sc"},{"description":"runtime parameter for tool RNA STARSolo","name":"sc"}],"label":null,"name":"RNA STARSolo","outputs":[{"name":"output_log","type":"txt"},{"name":"output_genes","type":"tsv"},{"name":"output_barcodes","type":"tsv"},{"name":"output_matrix","type":"mtx"},{"name":"output_BAM","type":"bam"},{"name":"output_stats","type":"txt"},{"name":"reads_per_gene","type":"tabular"}],"position":{"left":568.7432106444855,"top":374},"post_job_actions":{"HideDatasetActionoutput_BAM":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_BAM"},"HideDatasetActionoutput_barcodes":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_barcodes"},"HideDatasetActionoutput_genes":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_genes"},"HideDatasetActionoutput_log":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_log"},"HideDatasetActionoutput_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_matrix"},"HideDatasetActionoutput_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_stats"},"HideDatasetActionreads_per_gene":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reads_per_gene"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1","tool_shed_repository":{"changeset_revision":"381a32c51141","name":"rna_starsolo","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"outWig\\": {\\"outWigType\\": \\"None\\", \\"__current_case__\\": 0, \\"outWigStrand\\": \\"false\\"}, \\"refGenomeSource\\": {\\"geneSource\\": \\"indexed\\", \\"__current_case__\\": 0, \\"GTFconditional\\": {\\"GTFselect\\": \\"without-gtf-with-gtf\\", \\"__current_case__\\": 1, \\"genomeDir\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfeatureExon\\": \\"exon\\", \\"sjdbOverhang\\": \\"100\\"}}, \\"sc\\": {\\"solo_type\\": \\"CB_UMI_Simple\\", \\"__current_case__\\": 0, \\"input_types\\": {\\"use\\": \\"list_paired\\", \\"__current_case__\\": 1, \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"soloCBwhitelist\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"params\\": {\\"chemistry\\": \\"Cv3\\", \\"__current_case__\\": 1}, \\"soloBarcodeReadLength\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"umidedup\\": {\\"soloUMIdedup\\": \\"1MM_CR\\", \\"__current_case__\\": 4, \\"soloUMIfiltering\\": \\"-\\"}, \\"soloCBmatchWLtype\\": \\"1MM_multi\\"}, \\"solo\\": {\\"soloStrand\\": \\"Forward\\", \\"soloFeatures\\": \\"Gene\\", \\"wasp_conditional\\": {\\"waspOutputMode\\": \\"\\", \\"__current_case__\\": 1}, \\"filter\\": {\\"filter_type\\": \\"no_filter\\", \\"__current_case__\\": 3, \\"output_raw\\": \\"true\\"}, \\"soloOutFormatFeaturesGeneField3\\": \\"Gene Expression\\", \\"outSAMattributes\\": [\\"NH\\", \\"HI\\", \\"AS\\", \\"nM\\", \\"GX\\", \\"GN\\", \\"CB\\", \\"UB\\"], \\"quantModeGene\\": true, \\"outSAMunmapped\\": false, \\"outSAMmapqUnique\\": \\"60\\", \\"junction_limits\\": {\\"limitOutSJoneRead\\": \\"1000\\", \\"limitOutSJcollapsed\\": \\"1000000\\", \\"limitSjdbInsertNsj\\": \\"1000000\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.11a+galaxy1","type":"tool","uuid":"cef740b0-fa95-4542-b08f-dbfca4bfb3e5","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":6,"input_connections":{"results_0|software_cond|output_0|type|input":{"id":5,"output_name":"output_log"},"results_0|software_cond|output_1|type|input":{"id":5,"output_name":"reads_per_gene"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":"multiQC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":848.0098122069855,"top":366.98333740234375},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"a16bf20e-2bc9-45e7-a0a3-b5590212b973","when":null,"workflow_outputs":[{"label":"MultiQC on input dataset(s): Webpage","output_name":"html_report","uuid":"eb55ba68-0473-4b09-87b9-007be3a86098"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dropletutils/dropletutils/1.10.0+galaxy2","errors":null,"id":7,"input_connections":{"tenx_format|input":{"id":5,"output_name":"output_matrix"},"tenx_format|input_barcodes":{"id":5,"output_name":"output_barcodes"},"tenx_format|input_genes":{"id":5,"output_name":"output_genes"}},"inputs":[{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"},{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"},{"description":"runtime parameter for tool DropletUtils","name":"tenx_format"}],"label":"filter cells","name":"DropletUtils","outputs":[{"name":"barcodes_out","type":"tsv"},{"name":"genes_out","type":"tsv"},{"name":"matrix_out","type":"mtx"},{"name":"plot","type":"png"},{"name":"table","type":"tsv"}],"position":{"left":848.0098122069855,"top":640.9833374023438},"post_job_actions":{"HideDatasetActionbarcodes_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"barcodes_out"},"HideDatasetActiongenes_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"genes_out"},"HideDatasetActionmatrix_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"matrix_out"},"HideDatasetActionplot":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plot"},"HideDatasetActiontable":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"table"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dropletutils/dropletutils/1.10.0+galaxy2","tool_shed_repository":{"changeset_revision":"a9caad671439","name":"dropletutils","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation\\": {\\"use\\": \\"filter\\", \\"__current_case__\\": 0, \\"method\\": {\\"use\\": \\"emptydrops\\", \\"__current_case__\\": 1, \\"lower\\": \\"100\\", \\"fdr_thresh\\": \\"0.01\\"}, \\"outformat\\": \\"directory\\"}, \\"seed\\": \\"100\\", \\"tenx_format\\": {\\"use\\": \\"directory\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_genes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.0+galaxy2","type":"tool","uuid":"2ba61d0f-82ae-4044-a7ad-41df1cf8c9b6","when":null,"workflow_outputs":[{"label":"DropletUtils Barcodes","output_name":"barcodes_out","uuid":"c3275463-7e0a-4f34-a52b-eeb4dd0877b7"},{"label":"DropletUtils Genes","output_name":"genes_out","uuid":"9aa1c3b1-c4df-42c7-b72d-f5524b06ce33"},{"label":"DropletUtils Matrix","output_name":"matrix_out","uuid":"a7d56334-f655-4459-964f-e6f0738b8591"}]},"8":{"annotation":"","id":8,"input_connections":{"STARsolo Barcodes":{"id":7,"input_subworkflow_step_id":1,"output_name":"barcodes_out"},"STARsolo Genes":{"id":7,"input_subworkflow_step_id":0,"output_name":"genes_out"},"STARsolo Matrix Gene Counts":{"id":7,"input_subworkflow_step_id":2,"output_name":"matrix_out"}},"inputs":[{"description":"","name":"Re-organize STAR-solo output"}],"label":"Re-organize STAR-solo output","name":"Re-organize-STAR-solo-output","outputs":[],"position":{"left":1125.976487011673,"top":774},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Re-organize-STAR-solo-output","steps":{"0":{"annotation":"Collection named Genes raw or Genes filtered from STARsolo output","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection named Genes raw or Genes filtered from STARsolo output","name":"STARsolo Genes"}],"label":"STARsolo Genes","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3e312de3-0895-4b46-ac51-9c1c7e341194","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection named Barcodes raw or Barcodes filtered from STARsolo output","name":"STARsolo Barcodes"}],"label":"STARsolo Barcodes","name":"Input dataset collection","outputs":[],"position":{"left":23,"top":126},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4f8d1d99-0adc-40a8-a53b-ce9c1c614aeb","when":null,"workflow_outputs":[]},"2":{"annotation":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Collection named Matrix Gene Counts raw or Matrix Gene Counts filtered from STARsolo output","name":"STARsolo Matrix Gene Counts"}],"label":"STARsolo Matrix Gene Counts","name":"Input dataset collection","outputs":[],"position":{"left":84,"top":249},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"266958e3-aacf-4b7f-8f74-6cbcdd5db4c0","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":3,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":337,"top":438},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tsv\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"ff023a14-a401-4ede-8dfb-2187d2d8ad84","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":4,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":579,"top":63},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_genes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"e8eed7c4-a41f-4bf7-bf5d-b4ed5023f1b8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":584,"top":205},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_barcodes\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"de152d11-daaa-4c96-ada8-7917c17e9543","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":581,"top":383},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"(.*)\\", \\"replace_pattern\\": \\"\\\\\\\\1_matrix\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ac2f7a73-a742-4977-b902-aba871059cb6","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":7,"input_connections":{"how|labels":{"id":4,"output_name":"outfile"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":829,"top":58},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"d627ba62-eeb9-49a3-b46f-120dd8c8e3d8","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":8,"input_connections":{"how|labels":{"id":5,"output_name":"outfile"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":827,"top":213},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"e4fad6a7-30e3-4b66-a9bd-5f11f7ea18db","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":9,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":823,"top":364},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"fe80aa41-a8e4-4000-af6b-6c9713de9006","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":10,"input_connections":{"inputs_0|input":{"id":9,"output_name":"output"},"inputs_1|input":{"id":8,"output_name":"output"},"inputs_2|input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1064,"top":128},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"a494a5d5-d908-448e-a406-c7f78d81c10b","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":11,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":1284,"top":128},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"All matrices in cell-ranger format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1, 2], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\1\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"(.*)_(.*)\\", \\"group_count\\": null, \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"95eb63d6-f1a2-4f0d-b426-fc46a5133471","when":null,"workflow_outputs":[{"label":"STARsolo cell ranger-like output","output_name":"output","uuid":"3a6385df-d937-4c8a-9430-9bafe0cb2f57"}]}},"tags":[],"uuid":"a5dc2693-ad94-4b65-8767-0abb3a5129a7"},"tool_id":null,"type":"subworkflow","uuid":"697a68da-9cb9-469b-8102-30c3a3ee589f","when":null,"workflow_outputs":[{"label":"Seurat input for gene expression (filtered)","output_name":"STARsolo cell ranger-like output","uuid":"6cb072df-9e26-456b-ab4f-bc95345b02b4"}]}},"tags":["#single-cell"],"uuid":"4a5c18da-1897-41af-b625-29d149ae5f88","version":2},"readme":"# Single-Cell RNA-seq Preprocessing: 10X Genomics Data to Analysis-Ready Formats\\n\\nThese workflows are inspired by the [training material](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html). The output is in a \'bundle\' format with three files (one matrix, one with genes, one with barcodes) which is similar to the cellranger output format, and is compatible with any Read10X function (Seurat or Scanpy). There are 2 types of output collections: either one collection with all matrices, one collection with all genes and one collection with all barcodes (compatible with post processing in galaxy) or a single nested collection with one sub-collection per sample with the 3 files (easier for local downstream analysis).\\n\\nBoth workflows are designed for fastqs from 10X libraries v3. One is for regular 10X library (one library per sample), while the other one is for CellPlex 10X library which allows to multiplex samples using CMOs (see [this blog article](https://www.10xgenomics.com/blog/answering-your-questions-about-sample-multiplexing-for-single-cell-gene-expression)).\\n\\n## Input datasets\\n\\n- Specific for each experiment:\\n    - For both workflows: you need a list of pairs of fastqs with gene expression.\\n    - For CellPlex: you need in addition a list of pairs of fastqs with CMO.\\n    - For CellPlex: you need a list of csv which describes samples and CMO used:\\n        - first column is the sequence and second column is the name\\n    /!\\\\ The order of samples need to be exactly the same between the collection of fastqs of CMO and the collection of csv.\\n\\n- Common for all experiments:\\n    - Gene annotations: A gtf file with gene locations\\n    - List of barcodes used by 10X. You can download it at https://zenodo.org/record/3457880/files/3M-february-2018.txt.gz\\n\\n## Input values\\n\\n- reference genome: this genome needs to be available for STAR\\n- Barcode Size is same size of the Read: if the length of your R1 of GEX matches the size of cell barcode + UMI set to true. If your R1 contains trailling A, put false.\\n- number of cells: If you make it too large no cell barcode correction will be performed to demultiplex CMOs.\\n\\n## Processing\\n- Gene expression processing:\\n    - Reads are aligned to the genome, asigned to genes, cell barcode and UMI with STAR Solo\\n    - MultiQC report the mapping rate and the number of reads attributed to genes\\n    - The output of STAR Solo is filtered with Droplet Utils to remove cellular barcodes which are probably empty.\\n    - The output of Droplet Utils is reorganized to be:\\n```\\nMain Collection:\\n    - Sample 1:\\n        - matrix.mtx\\n        - barcodes.tsv\\n        - genes.tsv\\n    - Sample 2:\\n        - matrix.mtx\\n        - barcodes.tsv\\n        - genes.tsv\\n...\\n```\\nFor the CellPlex workflow:\\n- CMO processing:\\n    - CITE-Seq Count is used to asign reads and generate a matrix where \'genes\' are the CMO and \'unmapped\'.\\n    - Cellular barcodes are translated to match the cellular barcodes of Gene expression see [this article](https://kb.10xgenomics.com/hc/en-us/articles/360031133451-Why-is-there-a-discrepancy-in-the-3M-february-2018-txt-barcode-whitelist-).\\n    - Reorganize the output with UMI matrices to match the same structure as gene expression matrices.\\n\\n## Test data\\n\\nThe test dataset has been produced to make it as small as possible in order to make the workflow pass on CI.\\n\\n- The CMO reads come from [zenodo](https://zenodo.org/records/10229382) and have been sampled to 0.1 with seqtk.\\n- The GEX reads come from SRR13948489 but have been subsetted to the cells selected in the above zenodo.\\n","changelog":"# Changelog\\n\\n## [0.6.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.6.1] - 2025-03-03\\n\\n### Added\\n\\n- Added individual DropletUtils outputs for processing with scanpy tools\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.5] - 2024-09-25\\n\\n### Manual update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.4] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n\\n## [0.3] - 2024-02-12\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.10b+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rna_starsolo/rna_starsolo/2.7.11a+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n\\n## [0.2] - 2024-02-05\\n\\n### Tool updates\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4`\\n- `pick_value` was replaced by `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0`\\n\\n\\n\\n## [0.1] - 2023-12-21\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell RNA-seq Preprocessing: 10X Genomics v3 to Seurat and Scanpy Compatible Format\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ fastq PE collection\\"]@{ shape: docs }\\n1[\\"ℹ️ reference genome\\"]@{ shape: lean-l }\\n2[\\"ℹ️ gtf\\"]@{ shape: doc }\\n3[\\"ℹ️ cellranger_barcodes_3M-february-2018.txt\\"]@{ shape: doc }\\n4[\\"ℹ️ Barcode Size is same size of the Read\\"]@{ shape: lean-l }\\n5[\\"RNA STARSolo\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n0 --> 5\\n4 --> 5\\n3 --> 5\\n6[\\"multiQC\\"]@{ shape: process }\\n5 --> 6\\n5 --> 6\\n7[\\"filter cells\\"]@{ shape: process }\\n5 --> 7\\n5 --> 7\\n5 --> 7\\n8[\\"🛠️ Re-organize STAR-solo output\\"]@{ shape: subprocess }\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n```\\n\\n## Re-organize-STAR-solo-output\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ STARsolo Genes\\"]@{ shape: docs }\\n1[\\"ℹ️ STARsolo Barcodes\\"]@{ shape: docs }\\n2[\\"ℹ️ STARsolo Matrix Gene Counts\\"]@{ shape: docs }\\n3[\\"Extract element identifiers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Replace Text\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Replace Text\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Replace Text\\"]@{ shape: process }\\n3 --> 6\\n7[\\"Relabel identifiers\\"]@{ shape: process }\\n4 --> 7\\n0 --> 7\\n8[\\"Relabel identifiers\\"]@{ shape: process }\\n5 --> 8\\n1 --> 8\\n9[\\"Relabel identifiers\\"]@{ shape: process }\\n6 --> 9\\n2 --> 9\\n10[\\"Merge collections\\"]@{ shape: process }\\n9 --> 10\\n8 --> 10\\n7 --> 10\\n11[\\"Apply rules\\"]@{ shape: process }\\n10 --> 11\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/fastq-to-matrix-10x/scrna-seq-fastq-to-matrix-10x-v3","dockstore_id":24784,"categories":[],"collections":["Transcriptomics","Single Cell"],"doi":"10.5281/zenodo.15078477","tests":[{"doc":"Test outline for scrna-seq-fastq-to-matrix-10x-v3.ga","job":{"fastq PE collection":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"subsample","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/10412836/files/GEX_R1.fastqsanger.gz","filetype":"fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/10412836/files/GEX_R2.fastqsanger.gz","filetype":"fastqsanger.gz"}]}]},"reference genome":"dm6","Barcode Size is same size of the Read":false,"gtf":{"class":"File","location":"https://zenodo.org/record/6457007/files/Drosophila_melanogaster.BDGP6.32.109_UCSC.gtf.gz","decompress":true,"filetype":"gtf"},"cellranger_barcodes_3M-february-2018.txt":{"class":"File","path":"test-data/all_cell_barcodes_both_versions.txt","filetype":"tabular"}},"outputs":{"MultiQC on input dataset(s): Webpage":{"asserts":{"has_text_matching":{"expression":"<span class=\\"val\\">33.[0-9]<span class=[\'\\"]mqc_small_space[\'\\"]>"}}},"Seurat input for gene expression (filtered)":{"attributes":{"collection_type":"list:list"},"element_tests":{"subsample":{"elements":{"matrix":{"asserts":{"has_line":{"line":"23932 3 1171"}}},"barcodes":{"asserts":{"has_line":{"line":"CACATGATCATAAGGA"}}},"genes":{"asserts":{"has_line":{"line":"FBgn0250732\\tgfzf"}}}}}}}}}]}],"path":"./workflows/scRNAseq/fastq-to-matrix-10x"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Preprocessing-and-Clustering-of-single-cell-RNA-seq-data-with-Scanpy.ga","testParameterFiles":["/Preprocessing-and-Clustering-of-single-cell-RNA-seq-data-with-Scanpy-tests.yml"],"authors":[{"name":"Pavankumar Videm","orcid":"0000-0002-5192-126X"},{"name":"Hans-Rudolf Hotz","orcid":"0000-0002-2799-424X"},{"name":"Mehmet Tekman","orcid":"0000-0002-4181-2676"},{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987"}],"definition":{"a_galaxy_workflow":"true","annotation":"End-to-end analysis of single-cell RNA-seq data using the Scanpy/AnnData ecosystem. Imports count matrices, applies quality control filtering of low-quality cells and genes, normalizes and scales data, performs dimension reduction (PCA), and identifies cell clusters using the Louvain algorithm. Generates publication-ready visualizations including UMAP plots colored by clusters and marker genes.","comments":[],"creator":[{"class":"Person","identifier":"0000-0002-5192-126X","name":"Pavankumar Videm"},{"class":"Person","identifier":"0000-0002-2799-424X","name":"Hans-Rudolf Hotz"},{"class":"Person","identifier":"0000-0002-4181-2676","name":"Mehmet Tekman"},{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.1","name":"Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"use a small number. Generally 3-5. If not set, a default of 3 is used.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"use a small number. Generally 3-5. If not set, a default of 3 is used.","name":"Minimum number of cells expressed"}],"label":"Minimum number of cells expressed","name":"Input parameter","outputs":[],"position":{"left":665.7492402230988,"top":1083.0068800199208},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"03449fcf-0a59-49a3-bf5f-530006292f70","when":null,"workflow_outputs":[]},"1":{"annotation":"A cell barcodes file with a single barcode in each line. The barcodes should correspond to the cells in the matrix file","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"A cell barcodes file with a single barcode in each line. The barcodes should correspond to the cells in the matrix file","name":"Barcodes"}],"label":"Barcodes","name":"Input dataset","outputs":[],"position":{"left":300,"top":1410.8299913463652},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"51e797e4-a9e9-496a-92c0-e45f4c9ebd5d","when":null,"workflow_outputs":[]},"2":{"annotation":"A genes/features tabular file with gene ids and gene symbols","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"A genes/features tabular file with gene ids and gene symbols","name":"Genes"}],"label":"Genes","name":"Input dataset","outputs":[],"position":{"left":300,"top":1510.8299913463652},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"614e3951-97a8-4b9a-8052-5e900febcc9f","when":null,"workflow_outputs":[]},"3":{"annotation":"A single-cell count matrix file in Matrix Market Exchange format","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"A single-cell count matrix file in Matrix Market Exchange format","name":"Matrix"}],"label":"Matrix","name":"Input dataset","outputs":[],"position":{"left":300.99998815229196,"top":1611.8299931134695},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"mtx\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e08d9bbd-0761-47dd-b87a-9a0aff2b963b","when":null,"workflow_outputs":[]},"4":{"annotation":"v2 genes.tsv file contains two columns with Gene ID and Gene Name.\\nv3 features.tsv file contains three columns Feature ID, Feature Name,\\nFeature Type","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"v2 genes.tsv file contains two columns with Gene ID and Gene Name.\\nv3 features.tsv file contains three columns Feature ID, Feature Name,\\nFeature Type","name":"Input is from Cell Ranger v2 or earlier versions"}],"label":"Input is from Cell Ranger v2 or earlier versions","name":"Input parameter","outputs":[],"position":{"left":0,"top":1702.8299913463652},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f9e8fb8a-1886-4934-8213-d41b2e5250ce","when":null,"workflow_outputs":[]},"5":{"annotation":"For eg MT- or M-","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"For eg MT- or M-","name":"Mitochondrial genes start with pattern"}],"label":"Mitochondrial genes start with pattern","name":"Input parameter","outputs":[],"position":{"left":1185.4053609341802,"top":1477.0762060510267},"tool_id":null,"tool_state":"{\\"default\\": \\"MT-\\", \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"101b9500-34a5-44f6-b8e5-9760c907833a","when":null,"workflow_outputs":[]},"6":{"annotation":"Minimum number of genes expressed per cell. If not set, a default of 200 is used.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Minimum number of genes expressed per cell. If not set, a default of 200 is used.","name":"Minimum number of genes expressed"}],"label":"Minimum number of genes expressed","name":"Input parameter","outputs":[],"position":{"left":1519.8162409468177,"top":1625.5901525196762},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b3038a05-80e7-4211-9818-2f1a33bd7844","when":null,"workflow_outputs":[]},"7":{"annotation":"Maximum number of genes expressed per cell. If not set, a default of 2500 is used.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Maximum number of genes expressed per cell. If not set, a default of 2500 is used.","name":"Maximum number of genes expressed"}],"label":"Maximum number of genes expressed","name":"Input parameter","outputs":[],"position":{"left":1683.345743008954,"top":2017.656345078871},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2ba40bf4-c7a6-41a6-b877-3145e8f050de","when":null,"workflow_outputs":[]},"8":{"annotation":"Size of the local neighborhood. If not set a default of 15 is used.","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Size of the local neighborhood. If not set a default of 15 is used.","name":"Number of neighbours for computing neighborhood graph"}],"label":"Number of neighbours for computing neighborhood graph","name":"Input parameter","outputs":[],"position":{"left":4510.956333770236,"top":805.636987791653},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"0a1a9943-7f98-465a-a7e2-fc8925e474de","when":null,"workflow_outputs":[]},"9":{"annotation":"This parameter can be estimated from an Elbow plot of PC loadings.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"This parameter can be estimated from an Elbow plot of PC loadings.","name":"Number of PCs to use for computing neighborhood graph"}],"label":"Number of PCs to use for computing neighborhood graph","name":"Input parameter","outputs":[],"position":{"left":4502.422260098537,"top":1010.9182333917911},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2ed9ac9a-b45e-403b-9b6b-ecdbdaf419c1","when":null,"workflow_outputs":[]},"10":{"annotation":"Louvain clustering resolution. Higher resolution means finding more and smaller clusters. If not set, a default value of 1.0 is used.","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Louvain clustering resolution. Higher resolution means finding more and smaller clusters. If not set, a default value of 1.0 is used.","name":"Louvain resolution"}],"label":"Louvain resolution","name":"Input parameter","outputs":[],"position":{"left":5100.6155568749255,"top":451.4393039466668},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"bdea5eae-6fd8-4a83-b2c1-b407f35fec3a","when":null,"workflow_outputs":[]},"11":{"annotation":"You must have run the workflow at least once, to know the number of clusters and the cell types inferred.","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"You must have run the workflow at least once, to know the number of clusters and the cell types inferred.","name":"Manually annotate celltypes?"}],"label":"Manually annotate celltypes?","name":"Input parameter","outputs":[],"position":{"left":5884.356987738044,"top":1484.705188730196},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d63b7d2c-f23b-4c55-86c6-4d7ef72fae37","when":null,"workflow_outputs":[]},"12":{"annotation":"Provide a comma-separated list of cell types to annotate the louvain clusters.","content_id":null,"errors":null,"id":12,"input_connections":{},"inputs":[{"description":"Provide a comma-separated list of cell types to annotate the louvain clusters.","name":"Annotate louvain clusters with these cell types"}],"label":"Annotate louvain clusters with these cell types","name":"Input parameter","outputs":[],"position":{"left":5890.2226608717665,"top":1622.225384627387},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"dc4bbd2f-7594-41c6-bba2-b06e7c8ffaec","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":13,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":931.8168037645809,"top":980.8745203767638},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": \\"3\\"}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"55976e73-1347-4480-871b-867f72b1c34a","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/0.10.9+galaxy0","errors":null,"id":14,"input_connections":{"in|matrix":{"id":3,"output_name":"output"},"in|tenx|barcodes":{"id":1,"output_name":"output"},"in|tenx|genes":{"id":2,"output_name":"output"},"when":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Import Anndata","name":"in"}],"label":"Anndata from 10x v2 or earlier","name":"Import Anndata","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":630,"top":1672.8299913463652},"post_job_actions":{"RenameDatasetActionanndata":{"action_arguments":{"newname":"Input 3k PBMC"},"action_type":"RenameDatasetAction","output_name":"anndata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"d330b3082107","name":"anndata_import","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"in\\": {\\"adata_format\\": \\"mtx\\", \\"__current_case__\\": 3, \\"matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tenx\\": {\\"use\\": \\"legacy_10x\\", \\"__current_case__\\": 1, \\"genes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"var_names\\": \\"gene_symbols\\", \\"make_unique\\": true, \\"gex_only\\": true}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"2623e77b-49bc-4571-b5b5-46680544ab25","when":"$(inputs.when)","workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":15,"input_connections":{"input_param_type|input_param":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Map and switch the input CellRager version","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":310,"top":1852.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"boolean\\", \\"__current_case__\\": 3, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": true, \\"to\\": \\"False\\"}, {\\"__index__\\": 1, \\"from\\": false, \\"to\\": \\"True\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"input\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"8513eb66-b971-41ab-8aa4-03d0d15895ec","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":16,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1774.49879800967,"top":1737.570900245479},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": \\"200\\"}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"86f34174-c9b5-4d0b-a77a-134b269349b0","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":17,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":2002.5144505736228,"top":1996.1826325715392},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": \\"2500\\"}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"1312edfc-99ce-4d09-9bc8-8da090a5d391","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":18,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":4812.018833670425,"top":679.5206660447487},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": null}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"4834bd08-b69c-4388-8774-f5335c748a79","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":19,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":5411.291777668312,"top":416.9706212794444},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": \\"1.0\\"}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"c8b040bb-a2e1-4e3b-8240-2bd7abf581e6","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/0.10.9+galaxy0","errors":null,"id":20,"input_connections":{"in|matrix":{"id":3,"output_name":"output"},"in|tenx|barcodes":{"id":1,"output_name":"output"},"in|tenx|features":{"id":2,"output_name":"output"},"when":{"id":15,"output_name":"output_param_boolean"}},"inputs":[{"description":"runtime parameter for tool Import Anndata","name":"in"}],"label":"Anndata from 10x V3 or later","name":"Import Anndata","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":630,"top":1350.8299913463652},"post_job_actions":{"RenameDatasetActionanndata":{"action_arguments":{"newname":"Input 3k PBMC"},"action_type":"RenameDatasetAction","output_name":"anndata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"d330b3082107","name":"anndata_import","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"in\\": {\\"adata_format\\": \\"mtx\\", \\"__current_case__\\": 3, \\"matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tenx\\": {\\"use\\": \\"v3_10x\\", \\"__current_case__\\": 2, \\"features\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"var_names\\": \\"gene_symbols\\", \\"make_unique\\": true, \\"gex_only\\": true}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"7ef2904a-415f-45f6-98cd-a15af9fdb880","when":"$(inputs.when)","workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":21,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":20,"output_name":"anndata"},"style_cond|type_cond|pick_from_1|value":{"id":14,"output_name":"anndata"}},"inputs":[],"label":"Choose the Anndata","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":910,"top":1780.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"03d6b72c-5f66-48c9-965d-f9422c5c7291","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","errors":null,"id":22,"input_connections":{"adata":{"id":21,"output_name":"data_param"},"method|filter|min_cells":{"id":13,"output_name":"integer_param"}},"inputs":[],"label":"Filter genes by cells expressed","name":"Scanpy filter","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":1202.0003527723472,"top":1161.8299913463652},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"8cb783c81905","name":"scanpy_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"pp.filter_genes\\", \\"__current_case__\\": 1, \\"filter\\": {\\"filter\\": \\"min_cells\\", \\"__current_case__\\": 2, \\"min_cells\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"e46b9c19-fe18-4b68-9cac-ec31027f548a","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","errors":null,"id":23,"input_connections":{"input":{"id":21,"output_name":"data_param"}},"inputs":[],"label":null,"name":"Inspect AnnData","outputs":[{"name":"general","type":"txt"}],"position":{"left":1210,"top":1852.8299913463652},"post_job_actions":{"RenameDatasetActiongeneral":{"action_arguments":{"newname":"Initial Anndata General Info"},"action_type":"RenameDatasetAction","output_name":"general"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"0bf7fddf90c7","name":"anndata_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inspect\\": {\\"info\\": \\"general\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"0433e8be-743b-4186-9ab0-37c3bf238c3f","when":null,"workflow_outputs":[{"label":"Initial Anndata General Info","output_name":"general","uuid":"d0ecbc55-7639-4470-8854-d1a7d24715f2"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","errors":null,"id":24,"input_connections":{"input":{"id":22,"output_name":"anndata_out"},"manipulate|gene_flags_0|startswith":{"id":5,"output_name":"output"}},"inputs":[],"label":"Flag Mitochondrial genes","name":"Manipulate AnnData","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":1482.000448709542,"top":1089.8299913463652},"post_job_actions":{"RenameDatasetActionanndata":{"action_arguments":{"newname":"3k PBMC with mito annotation"},"action_type":"RenameDatasetAction","output_name":"anndata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"7e8c677a7b71","name":"anndata_manipulate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"manipulate\\": {\\"function\\": \\"flag_genes\\", \\"__current_case__\\": 5, \\"gene_flags\\": [{\\"__index__\\": 0, \\"startswith\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"col_name\\": \\"mito\\"}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"d802d24a-1cc5-4e95-82f2-8d84fc2c49e1","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","errors":null,"id":25,"input_connections":{"adata":{"id":24,"output_name":"anndata"}},"inputs":[],"label":"Calculate quality metrics","name":"Scanpy Inspect and manipulate","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":1762.0004970088444,"top":1160.8299913463652},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with mito annotation and qc metrics"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","tool_shed_repository":{"changeset_revision":"d86cb2ce311b","name":"scanpy_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.calculate_qc_metrics\\", \\"__current_case__\\": 0, \\"expr_type\\": \\"counts\\", \\"var_type\\": \\"genes\\", \\"qc_vars\\": \\"mito\\", \\"percent_top\\": \\"\\", \\"layer\\": \\"\\", \\"use_raw\\": false, \\"log1p\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy1","type":"tool","uuid":"6a1b0ee7-070c-4464-b8e0-68f14da607b2","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":26,"input_connections":{"adata":{"id":25,"output_name":"anndata_out"}},"inputs":[],"label":"Scatter plot n_genes vs pct_mito","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":2051.9999683720544,"top":900.8299737231888},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.scatter\\", \\"__current_case__\\": 0, \\"type\\": {\\"type\\": \\"xy\\", \\"__current_case__\\": 0, \\"x\\": \\"n_genes_by_counts\\", \\"y\\": \\"pct_counts_mito\\", \\"log\\": false, \\"color\\": \\"\\", \\"layers\\": {\\"use_layers\\": \\"false\\", \\"__current_case__\\": 1}}, \\"use_raw\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"none\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"size\\": null, \\"title\\": \\"\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"8cd94050-8147-45a7-8c16-274bdc987a35","when":null,"workflow_outputs":[{"label":"Scatter plot of n_genes vs pct_mito","output_name":"out_png","uuid":"b4a00acb-37ed-4e8d-b2fa-aecf3eb0688b"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":27,"input_connections":{"adata":{"id":25,"output_name":"anndata_out"}},"inputs":[],"label":"Scatter plot n_counts vs n_genes","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":2049.9999683720544,"top":1089.8299737231887},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.scatter\\", \\"__current_case__\\": 0, \\"type\\": {\\"type\\": \\"xy\\", \\"__current_case__\\": 0, \\"x\\": \\"total_counts\\", \\"y\\": \\"n_genes_by_counts\\", \\"log\\": false, \\"color\\": \\"\\", \\"layers\\": {\\"use_layers\\": \\"false\\", \\"__current_case__\\": 1}}, \\"use_raw\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"none\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"size\\": null, \\"title\\": \\"\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"748883c6-9abf-464b-8dad-746f9d7ae9af","when":null,"workflow_outputs":[{"label":"Scatter plot of plot n_counts vs n_genes","output_name":"out_png","uuid":"03555fcf-9885-41b4-bba5-c6daa06b7913"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":28,"input_connections":{"adata":{"id":25,"output_name":"anndata_out"}},"inputs":[],"label":"Violin plot n_genes, n_counts, pct_mito","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":2055.9999683720544,"top":1283.8299737231887},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.violin\\", \\"__current_case__\\": 4, \\"key_variables\\": {\\"type\\": \\"custom\\", \\"__current_case__\\": 2, \\"keys\\": \\"n_genes_by_counts, total_counts, pct_counts_mito\\"}, \\"groupby\\": \\"\\", \\"log\\": false, \\"use_raw\\": false, \\"violin_plot\\": {\\"stripplot\\": {\\"stripplot\\": \\"True\\", \\"__current_case__\\": 0, \\"jitter\\": {\\"jitter\\": \\"True\\", \\"__current_case__\\": 0, \\"size\\": \\"1.0\\"}}, \\"multi_panel\\": {\\"multi_panel\\": \\"True\\", \\"__current_case__\\": 1}}, \\"xlabel\\": \\"\\", \\"rotation\\": null, \\"seaborn_violinplot\\": {\\"bw_method\\": \\"scott\\", \\"bw_adjust\\": \\"1.0\\", \\"orient\\": null, \\"linewidth\\": \\"0.0\\", \\"color\\": null, \\"saturation\\": \\"0.75\\"}, \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"dbea9c99-c513-45bb-be11-f1e995ea29cc","when":null,"workflow_outputs":[{"label":"Violin plot n_genes, n_counts, pct_mito","output_name":"out_png","uuid":"d0e14eb2-ab45-4f1c-bf1a-4a70349036cc"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","errors":null,"id":29,"input_connections":{"adata":{"id":25,"output_name":"anndata_out"},"method|filter|min_genes":{"id":16,"output_name":"integer_param"}},"inputs":[],"label":"Filter cells by min genes expressed","name":"Scanpy filter","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":2053.6916490875606,"top":1538.8173034887368},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"8cb783c81905","name":"scanpy_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"pp.filter_cells\\", \\"__current_case__\\": 0, \\"filter\\": {\\"filter\\": \\"min_genes\\", \\"__current_case__\\": 2, \\"min_genes\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"d7762c6b-85db-478f-b6b5-bf582802f06e","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","errors":null,"id":30,"input_connections":{"adata":{"id":29,"output_name":"anndata_out"},"method|filter|max_genes":{"id":17,"output_name":"integer_param"}},"inputs":[],"label":"Filter cells by max genes expressed","name":"Scanpy filter","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":2332.000472347359,"top":1583.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"8cb783c81905","name":"scanpy_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"pp.filter_cells\\", \\"__current_case__\\": 0, \\"filter\\": {\\"filter\\": \\"max_genes\\", \\"__current_case__\\": 3, \\"max_genes\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"a3195b4d-445b-497f-88c8-96bf1edf3a48","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","errors":null,"id":31,"input_connections":{"input":{"id":30,"output_name":"anndata_out"}},"inputs":[],"label":"Filter cells by percentage of mito genes","name":"Manipulate AnnData","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":2612.0003041107857,"top":1608.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"7e8c677a7b71","name":"anndata_manipulate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"manipulate\\": {\\"function\\": \\"filter\\", \\"__current_case__\\": 10, \\"var_obs\\": \\"obs\\", \\"filter\\": {\\"filter\\": \\"key\\", \\"__current_case__\\": 0, \\"key\\": \\"pct_counts_mito\\", \\"filter_key\\": {\\"type\\": \\"number\\", \\"__current_case__\\": 0, \\"filter\\": \\"less\\", \\"value\\": \\"5.0\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"212223c0-de42-4572-8259-f313a131582f","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.10.2+galaxy0","errors":null,"id":32,"input_connections":{"adata":{"id":31,"output_name":"anndata"}},"inputs":[],"label":"Normalize by target sum","name":"Scanpy normalize","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":2892.0001358742124,"top":1619.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"56c0d4acef02","name":"scanpy_normalize","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.normalize_total\\", \\"__current_case__\\": 0, \\"target_sum\\": \\"10000.0\\", \\"exclude_highly_expressed\\": {\\"exclude_highly_expressed\\": \\"False\\", \\"__current_case__\\": 1}, \\"key_added\\": \\"norm\\", \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"aa33e80c-8534-48db-9ef7-486e2eab7bdf","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","errors":null,"id":33,"input_connections":{"adata":{"id":32,"output_name":"anndata_out"}},"inputs":[],"label":"Logarithmize","name":"Scanpy Inspect and manipulate","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":3172.0002924414525,"top":1619.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","tool_shed_repository":{"changeset_revision":"d86cb2ce311b","name":"scanpy_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.log1p\\", \\"__current_case__\\": 6, \\"base\\": null, \\"layer\\": \\"\\", \\"obsm\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy1","type":"tool","uuid":"78f29e9d-025c-430c-9945-f5d0e9ee18a5","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","errors":null,"id":34,"input_connections":{"input":{"id":33,"output_name":"anndata_out"}},"inputs":[],"label":"Freeze raw attribute","name":"Manipulate AnnData","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":3452.0004490086926,"top":1608.8299913463652},"post_job_actions":{"RenameDatasetActionanndata":{"action_arguments":{"newname":"3k PBMC after QC filtering and normalization"},"action_type":"RenameDatasetAction","output_name":"anndata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"7e8c677a7b71","name":"anndata_manipulate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"manipulate\\": {\\"function\\": \\"save_raw\\", \\"__current_case__\\": 11}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"904d4bae-945b-43eb-8151-18b9132f722f","when":null,"workflow_outputs":[{"label":"Anndata with raw attribute","output_name":"anndata","uuid":"bdc0eabb-af2f-4ae5-a72b-97041681db60"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","errors":null,"id":35,"input_connections":{"adata":{"id":34,"output_name":"anndata"}},"inputs":[],"label":"Annotate highly variable genes","name":"Scanpy filter","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":3732.0002807721194,"top":1619.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"8cb783c81905","name":"scanpy_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.highly_variable_genes\\", \\"__current_case__\\": 3, \\"flavor\\": {\\"flavor\\": \\"seurat\\", \\"__current_case__\\": 0, \\"min_mean\\": \\"0.0125\\", \\"max_mean\\": \\"3.0\\", \\"min_disp\\": \\"0.5\\", \\"max_disp\\": null}, \\"n_bins\\": \\"20\\", \\"subset\\": false, \\"layer\\": \\"\\", \\"batch_key\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"f618a000-196f-473c-b8ca-dd8620e1f347","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","errors":null,"id":36,"input_connections":{"input":{"id":35,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Manipulate AnnData","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":4012.0004373393595,"top":1619.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"7e8c677a7b71","name":"anndata_manipulate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"manipulate\\": {\\"function\\": \\"filter\\", \\"__current_case__\\": 10, \\"var_obs\\": \\"var\\", \\"filter\\": {\\"filter\\": \\"key\\", \\"__current_case__\\": 0, \\"key\\": \\"highly_variable\\", \\"filter_key\\": {\\"type\\": \\"boolean\\", \\"__current_case__\\": 2, \\"value\\": true}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"4cce267f-b856-4350-aa6f-227830b3746a","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":37,"input_connections":{"adata":{"id":35,"output_name":"anndata_out"}},"inputs":[],"label":"Plot highly variable","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":4012.0004373393595,"top":1772.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.highly_variable_genes\\", \\"__current_case__\\": 9, \\"log\\": false, \\"highly_variable_genes\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"b74ac6fe-664f-4751-aa40-e584c4846be3","when":null,"workflow_outputs":[{"label":"Plot highly variable","output_name":"out_png","uuid":"62b14303-41ac-46b2-a6f4-70bf14e19d68"}]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_remove_confounders/scanpy_remove_confounders/1.10.2+galaxy0","errors":null,"id":38,"input_connections":{"adata":{"id":36,"output_name":"anndata"}},"inputs":[],"label":"Regress out","name":"Scanpy remove confounders","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":4292.0005939065995,"top":1619.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_remove_confounders/scanpy_remove_confounders/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"9fb627495ec2","name":"scanpy_remove_confounders","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.regress_out\\", \\"__current_case__\\": 0, \\"layer\\": \\"\\", \\"keys\\": \\"total_counts, pct_counts_mito\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"4440363b-aea1-4eed-9f33-8b2a0f354903","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","errors":null,"id":39,"input_connections":{"adata":{"id":38,"output_name":"anndata_out"}},"inputs":[],"label":"Scale data","name":"Scanpy Inspect and manipulate","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":4572.000425670027,"top":1619.8299913463652},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","tool_shed_repository":{"changeset_revision":"d86cb2ce311b","name":"scanpy_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"pp.scale\\", \\"__current_case__\\": 7, \\"zero_center\\": true, \\"max_value\\": \\"10.0\\", \\"layer\\": \\"\\", \\"obsm\\": \\"\\", \\"mask_obs\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy1","type":"tool","uuid":"e4e1ea88-eb3a-47b3-bd3d-797e7a99f956","when":null,"workflow_outputs":[]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","errors":null,"id":40,"input_connections":{"adata":{"id":39,"output_name":"anndata_out"}},"inputs":[],"label":"Compute PCA","name":"Scanpy cluster, embed","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":4852.000582237267,"top":1619.8299913463652},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling and PCA"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"38db3fef7aad","name":"scanpy_cluster_reduce_dimension","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"pp.pca\\", \\"__current_case__\\": 2, \\"n_comps\\": \\"50\\", \\"layer\\": \\"\\", \\"dtype\\": \\"float32\\", \\"pca\\": {\\"chunked\\": \\"False\\", \\"__current_case__\\": 1, \\"zero_center\\": true, \\"svd_solver\\": null, \\"random_state\\": \\"0\\"}, \\"mask_var\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"ae913706-975b-4b63-8432-313b325a4306","when":null,"workflow_outputs":[]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","errors":null,"id":41,"input_connections":{"adata":{"id":40,"output_name":"anndata_out"},"method|n_neighbors":{"id":18,"output_name":"integer_param"},"method|n_pcs":{"id":9,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Scanpy Inspect and manipulate","name":"method"},{"description":"runtime parameter for tool Scanpy Inspect and manipulate","name":"method"}],"label":"Compute Neighborhood graph","name":"Scanpy Inspect and manipulate","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":5132.000738804507,"top":735.8299913463653},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling, PCA and KNN graph"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","tool_shed_repository":{"changeset_revision":"d86cb2ce311b","name":"scanpy_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"pp.neighbors\\", \\"__current_case__\\": 1, \\"n_neighbors\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"n_pcs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"use_rep\\": \\"\\", \\"knn\\": true, \\"pp_neighbors_method\\": \\"umap\\", \\"metric\\": \\"euclidean\\", \\"random_state\\": \\"0\\", \\"key_added\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy1","type":"tool","uuid":"005d6af1-ad00-4ef6-8fcd-e5df4569ba34","when":null,"workflow_outputs":[]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":42,"input_connections":{"adata":{"id":40,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":5135.543225031574,"top":1477.1861932178354},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.pca_loadings\\", \\"__current_case__\\": 12, \\"components\\": \\"1,2,3\\", \\"include_lowest\\": true, \\"n_points\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"e074111c-968e-444a-827d-9928727e9792","when":null,"workflow_outputs":[{"label":"Plot genes according to contributions to PCs","output_name":"out_png","uuid":"c4213c40-ba33-49bd-bb51-b5694b53a629"}]},"43":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":43,"input_connections":{"adata":{"id":40,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":5137.895983849266,"top":1677.4128884822696},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.pca_overview\\", \\"__current_case__\\": 14, \\"color\\": \\"\\", \\"use_raw\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [{\\"__index__\\": 0, \\"axis1\\": \\"1\\", \\"axis2\\": \\"2\\"}, {\\"__index__\\": 1, \\"axis1\\": \\"2\\", \\"axis2\\": \\"3\\"}], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"right margin\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"size\\": null, \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"ncols\\": \\"2\\", \\"wspace\\": \\"0.1\\", \\"hspace\\": \\"0.25\\", \\"title\\": \\"\\"}, \\"add_outline\\": false, \\"outline_color_border\\": null, \\"outline_color_gap\\": null, \\"outline_width_border\\": \\"0.3\\", \\"outline_width_gap\\": \\"0.05\\", \\"matplotlib_pyplot_scatter\\": {\\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"linewidths\\": \\"0.0\\", \\"edgecolors\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"4884a876-9340-4e66-843f-f1849df0f52b","when":null,"workflow_outputs":[{"label":"Plot PCA over view with genes","output_name":"out_png","uuid":"869246fa-71b6-44c6-a9b0-142ce2c83a0b"}]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":44,"input_connections":{"adata":{"id":40,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":5135.543225031574,"top":1852.530082426005},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.pca_variance_ratio\\", \\"__current_case__\\": 13, \\"n_pcs\\": \\"50\\", \\"log\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"bb708b36-d4d4-424e-8262-4a792dd2098f","when":null,"workflow_outputs":[{"label":"Elbow plot of PCs and variance","output_name":"out_png","uuid":"a0152a66-2dc5-4745-804f-b1d5f275348d"}]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","errors":null,"id":45,"input_connections":{"adata":{"id":41,"output_name":"anndata_out"}},"inputs":[],"label":"Embed using UMAP","name":"Scanpy cluster, embed","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":5412.000895371747,"top":807.8299913463653},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"38db3fef7aad","name":"scanpy_cluster_reduce_dimension","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"tl.umap\\", \\"__current_case__\\": 5, \\"min_dist\\": \\"0.5\\", \\"spread\\": \\"1.0\\", \\"n_components\\": \\"2\\", \\"maxiter\\": null, \\"alpha\\": \\"1.0\\", \\"gamma\\": \\"1.0\\", \\"negative_sample_rate\\": \\"5\\", \\"init_pos\\": \\"spectral\\", \\"random_state\\": \\"0\\", \\"neighbors_key\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"881f04b7-3d5f-4012-97e8-d29dce958211","when":null,"workflow_outputs":[]},"46":{"annotation":"Higher resolution means finding more and smaller clusters.","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","errors":null,"id":46,"input_connections":{"adata":{"id":45,"output_name":"anndata_out"},"method|flavor|resolution":{"id":19,"output_name":"float_param"}},"inputs":[],"label":"Louvain clustering","name":"Scanpy cluster, embed","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":5692.001389734953,"top":652.8299913463653},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_cluster_reduce_dimension/scanpy_cluster_reduce_dimension/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"38db3fef7aad","name":"scanpy_cluster_reduce_dimension","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"method\\": {\\"method\\": \\"tl.louvain\\", \\"__current_case__\\": 0, \\"flavor\\": {\\"flavor\\": \\"vtraag\\", \\"__current_case__\\": 0, \\"resolution\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"random_state\\": \\"0\\", \\"key_added\\": \\"louvain\\", \\"directed\\": true, \\"use_weights\\": false, \\"neighbors_key\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"06f497ab-c22e-4f6d-962b-2fc43f99ad90","when":null,"workflow_outputs":[]},"47":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":47,"input_connections":{"adata":{"id":46,"output_name":"anndata_out"}},"inputs":[],"label":"UMAP of louvain","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":5972.00122149838,"top":690.8299913463653},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.umap\\", \\"__current_case__\\": 16, \\"color\\": \\"louvain\\", \\"use_raw\\": false, \\"edges\\": {\\"edges\\": \\"False\\", \\"__current_case__\\": 1}, \\"arrows\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"right margin\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"size\\": null, \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"ncols\\": \\"2\\", \\"wspace\\": \\"0.1\\", \\"hspace\\": \\"0.25\\", \\"title\\": \\"\\"}, \\"add_outline\\": false, \\"outline_color_border\\": null, \\"outline_color_gap\\": null, \\"outline_width_border\\": \\"0.3\\", \\"outline_width_gap\\": \\"0.05\\", \\"matplotlib_pyplot_scatter\\": {\\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"linewidths\\": \\"0.0\\", \\"edgecolors\\": \\"face\\"}, \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"6c2fe5ab-0d77-45cd-bc95-52c40ebea722","when":null,"workflow_outputs":[{"label":"UMAP of louvain","output_name":"out_png","uuid":"7bdb4206-ce50-4252-9161-9c0f60810e6b"}]},"48":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","errors":null,"id":48,"input_connections":{"adata":{"id":46,"output_name":"anndata_out"}},"inputs":[],"label":"Rank genes by Wilcoxon test","name":"Scanpy Inspect and manipulate","outputs":[{"name":"anndata_out","type":"h5ad"}],"position":{"left":5980,"top":842.8299913463653},"post_job_actions":{"RenameDatasetActionanndata_out":{"action_arguments":{"newname":"3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test"},"action_type":"RenameDatasetAction","output_name":"anndata_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.10.2+galaxy1","tool_shed_repository":{"changeset_revision":"d86cb2ce311b","name":"scanpy_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method\\": {\\"method\\": \\"tl.rank_genes_groups\\", \\"__current_case__\\": 4, \\"get_df\\": {\\"get_df_select\\": \\"False\\", \\"__current_case__\\": 0}, \\"groupby\\": \\"louvain\\", \\"use_raw\\": true, \\"groups\\": \\"\\", \\"layer\\": \\"\\", \\"ref\\": {\\"rest\\": \\"rest\\", \\"__current_case__\\": 0}, \\"n_genes\\": \\"100\\", \\"tl_rank_genes_groups_method\\": {\\"method\\": \\"wilcoxon\\", \\"__current_case__\\": 1, \\"corr_method\\": \\"benjamini-hochberg\\", \\"tie_correct\\": false}, \\"key_added\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy1","type":"tool","uuid":"636ce34b-e448-4108-8c36-08bb9552f3d9","when":null,"workflow_outputs":[]},"49":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","errors":null,"id":49,"input_connections":{"input":{"id":48,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Inspect AnnData","outputs":[{"name":"uns_rank_genes_groups_names","type":"tabular"},{"name":"uns_rank_genes_groups_scores","type":"tabular"},{"name":"uns_rank_genes_groups_logfoldchanges","type":"tabular"},{"name":"uns_rank_genes_groups_pvals","type":"tabular"},{"name":"uns_rank_genes_groups_pvals_adj","type":"tabular"}],"position":{"left":6252.001715861586,"top":344.8299913463653},"post_job_actions":{"RenameDatasetActionuns_rank_genes_groups_names":{"action_arguments":{"newname":"Ranked genes with Wilcoxon test"},"action_type":"RenameDatasetAction","output_name":"uns_rank_genes_groups_names"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"0bf7fddf90c7","name":"anndata_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inspect\\": {\\"info\\": \\"uns\\", \\"__current_case__\\": 5, \\"uns_info\\": \\"rank_genes_groups\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"4755a9d9-1cda-4ade-8b1b-4514835fd3d1","when":null,"workflow_outputs":[{"label":"Ranked genes with Wilcoxon test","output_name":"uns_rank_genes_groups_names","uuid":"f1ead4ac-9308-47bb-92e6-47da581c08d3"}]},"50":{"annotation":"Here I inspected the louviain clusters...","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","errors":null,"id":50,"input_connections":{"input":{"id":48,"output_name":"anndata_out"}},"inputs":[],"label":"Inspect clusters","name":"Inspect AnnData","outputs":[{"name":"obs","type":"tabular"}],"position":{"left":6252.001715861586,"top":843.8299913463653},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"0bf7fddf90c7","name":"anndata_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"h5ad\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inspect\\": {\\"info\\": \\"obs\\", \\"__current_case__\\": 3}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"6767734f-9e0b-4650-8c28-9d4b547f2614","when":null,"workflow_outputs":[]},"51":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":51,"input_connections":{"adata":{"id":48,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":6252.001715861586,"top":1149.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.rank_genes_groups\\", \\"__current_case__\\": 24, \\"groups\\": \\"\\", \\"n_genes\\": \\"20\\", \\"gene_symbols\\": \\"\\", \\"fontsize\\": \\"8\\", \\"ncols\\": \\"3\\", \\"sharey\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"a2316a70-98ff-4108-a42a-99700fefbb5d","when":null,"workflow_outputs":[{"label":"Plot Rank gene groups","output_name":"out_png","uuid":"8d0bc61a-7cd1-4516-9064-3483bd868f6c"}]},"52":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","errors":null,"id":52,"input_connections":{"input":{"id":48,"output_name":"anndata_out"},"manipulate|categories":{"id":12,"output_name":"output"},"when":{"id":11,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Manipulate AnnData","name":"manipulate"}],"label":null,"name":"Manipulate AnnData","outputs":[{"name":"anndata","type":"h5ad"}],"position":{"left":6250,"top":1602.8299913463652},"post_job_actions":{"RenameDatasetActionanndata":{"action_arguments":{"newname":"Anndata with Celltype Annotation"},"action_type":"RenameDatasetAction","output_name":"anndata"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"7e8c677a7b71","name":"anndata_manipulate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"manipulate\\": {\\"function\\": \\"rename_categories\\", \\"__current_case__\\": 3, \\"key\\": \\"louvain\\", \\"categories\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"6ac832fd-39f6-4e07-b386-56001eacc343","when":"$(inputs.when)","workflow_outputs":[{"label":"Anndata with Celltype Annotation","output_name":"anndata","uuid":"5d3b148b-6611-4187-a87f-0e8c13312f52"}]},"53":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","errors":null,"id":53,"input_connections":{"input":{"id":48,"output_name":"anndata_out"}},"inputs":[],"label":"General information about the final Anndata object","name":"Inspect AnnData","outputs":[{"name":"general","type":"txt"}],"position":{"left":7277.957184864016,"top":877.9894353996245},"post_job_actions":{"RenameDatasetActiongeneral":{"action_arguments":{"newname":"Final Anndata General Info"},"action_type":"RenameDatasetAction","output_name":"general"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.9+galaxy0","tool_shed_repository":{"changeset_revision":"0bf7fddf90c7","name":"anndata_inspect","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inspect\\": {\\"info\\": \\"general\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.9+galaxy0","type":"tool","uuid":"3086e8de-bf0a-4229-9a38-2889c129c20d","when":null,"workflow_outputs":[{"label":"General information about the final Anndata object","output_name":"general","uuid":"c52fdbb0-794b-430b-8358-731c150d647e"}]},"54":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":54,"input_connections":{"infile":{"id":49,"output_name":"uns_rank_genes_groups_names"}},"inputs":[],"label":"Select top ranked genes","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":6552.001547625013,"top":365.8299913463653},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==2{gsub(/\\\\\\\\t/,\\\\\\",\\\\\\",$0); print}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"da886056-c68e-480f-bab8-586f5209dd9b","when":null,"workflow_outputs":[]},"55":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":55,"input_connections":{"infile":{"id":49,"output_name":"uns_rank_genes_groups_names"}},"inputs":[],"label":"Count number of clusters","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":6561.265841513936,"top":537.2187315680474},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{print NF}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"5f979ec3-a30c-4990-aa58-6a512df2b078","when":null,"workflow_outputs":[]},"56":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":56,"input_connections":{"infile":{"id":49,"output_name":"uns_rank_genes_groups_names"}},"inputs":[],"label":"Select top ranked genes with louvain","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":6552.001547625013,"top":1040.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==2{gsub(/\\\\\\\\t/,\\\\\\",\\\\\\",$0); print \\\\\\"louvain,\\\\\\"$0}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"5d39d890-01fd-43ae-a0a2-eecbbbeeb9a9","when":null,"workflow_outputs":[]},"57":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":57,"input_connections":{"in_file":{"id":50,"output_name":"obs"}},"inputs":[],"label":"Count number of cells per cluster","name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":6498.868219233225,"top":767.2452219288871},"post_job_actions":{"RenameDatasetActionout_file":{"action_arguments":{"newname":"Number of cells per cluster"},"action_type":"RenameDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"15\\", \\"header_in\\": true, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"count\\", \\"op_column\\": \\"1\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"6d4cf180-725d-4a2a-9512-be939a1af8c8","when":null,"workflow_outputs":[{"label":"Number of cells per cluster","output_name":"out_file","uuid":"6b711421-aec9-47ae-b987-2c13aa9c2e8e"}]},"58":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":58,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":52,"output_name":"anndata"},"style_cond|type_cond|pick_from_1|value":{"id":48,"output_name":"anndata_out"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":6570,"top":1202.8299913463652},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"06b51d40-f84e-4ac6-a92b-547be86ded38","when":null,"workflow_outputs":[]},"59":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":59,"input_connections":{"input1":{"id":54,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":6832.00137938844,"top":344.8299913463653},"post_job_actions":{},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"27559b56-2e68-43b5-8533-a4ca05e900de","when":null,"workflow_outputs":[]},"60":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":60,"input_connections":{"input1":{"id":55,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":6916.044358370236,"top":613.875261824484},"post_job_actions":{},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"a5518c7a-2900-43c3-b299-b2b20291d195","when":null,"workflow_outputs":[]},"61":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":61,"input_connections":{"input1":{"id":56,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":6832.00137938844,"top":1019.8299913463653},"post_job_actions":{},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"e175404d-4379-4bfa-aed7-79d001fce5ea","when":null,"workflow_outputs":[]},"62":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":62,"input_connections":{"adata":{"id":58,"output_name":"data_param"},"when":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7284.276436074986,"top":1669.7494225002392},"post_job_actions":{"RenameDatasetActionout_png":{"action_arguments":{"newname":"UMAP with annotated cell types"},"action_type":"RenameDatasetAction","output_name":"out_png"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.umap\\", \\"__current_case__\\": 16, \\"color\\": \\"louvain\\", \\"use_raw\\": true, \\"edges\\": {\\"edges\\": \\"False\\", \\"__current_case__\\": 1}, \\"arrows\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"on data\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"size\\": null, \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"ncols\\": \\"4\\", \\"wspace\\": \\"0.1\\", \\"hspace\\": \\"0.25\\", \\"title\\": \\"\\"}, \\"add_outline\\": false, \\"outline_color_border\\": null, \\"outline_color_gap\\": null, \\"outline_width_border\\": \\"0.3\\", \\"outline_width_gap\\": \\"0.05\\", \\"matplotlib_pyplot_scatter\\": {\\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"linewidths\\": \\"0.0\\", \\"edgecolors\\": \\"face\\"}, \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"90fc9851-6092-46b7-a5ec-c3f5b49345c9","when":"$(inputs.when)","workflow_outputs":[{"label":"UMAP with annotated cell types","output_name":"out_png","uuid":"82e1dab8-bae6-4bc9-b4cb-cc0d8b6c3417"}]},"63":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":63,"input_connections":{"adata":{"id":48,"output_name":"anndata_out"},"method|key_variables|keys":{"id":59,"output_name":"text_param"}},"inputs":[],"label":"Violin plot of top genes on clusters","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7274.993354573022,"top":339.6802685334286},"post_job_actions":{"RenameDatasetActionout_png":{"action_arguments":{"newname":"Violin plot of top genes on clusters"},"action_type":"RenameDatasetAction","output_name":"out_png"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.violin\\", \\"__current_case__\\": 4, \\"key_variables\\": {\\"type\\": \\"custom\\", \\"__current_case__\\": 2, \\"keys\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"groupby\\": \\"louvain\\", \\"log\\": false, \\"use_raw\\": false, \\"violin_plot\\": {\\"stripplot\\": {\\"stripplot\\": \\"True\\", \\"__current_case__\\": 0, \\"jitter\\": {\\"jitter\\": \\"True\\", \\"__current_case__\\": 0, \\"size\\": \\"1.0\\"}}, \\"multi_panel\\": {\\"multi_panel\\": \\"None\\", \\"__current_case__\\": 0}}, \\"xlabel\\": \\"\\", \\"rotation\\": null, \\"seaborn_violinplot\\": {\\"bw_method\\": \\"scott\\", \\"bw_adjust\\": \\"1.0\\", \\"orient\\": null, \\"linewidth\\": \\"0.0\\", \\"color\\": \\"AliceBlue\\", \\"saturation\\": \\"0.75\\"}, \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"8030a8a4-0fb2-41b4-b7b6-a3fee0e1348c","when":null,"workflow_outputs":[{"label":"Violin plot of top genes on clusters","output_name":"out_png","uuid":"281eb7a4-767a-4b0c-b451-ded5693065bb"}]},"64":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":64,"input_connections":{"adata":{"id":48,"output_name":"anndata_out"},"method|figsize|height":{"id":60,"output_name":"integer_param"},"method|figsize|width":{"id":60,"output_name":"integer_param"},"method|num_categories":{"id":60,"output_name":"integer_param"},"method|var_names|var_names":{"id":59,"output_name":"text_param"}},"inputs":[{"description":"runtime parameter for tool Scanpy plot","name":"method"}],"label":"Stacked violin of top genes on clusters","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7278.7873569974545,"top":0},"post_job_actions":{"RenameDatasetActionout_png":{"action_arguments":{"newname":"Stacked violin of top genes on clusters"},"action_type":"RenameDatasetAction","output_name":"out_png"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.stacked_violin\\", \\"__current_case__\\": 5, \\"var_names\\": {\\"type\\": \\"custom\\", \\"__current_case__\\": 1, \\"var_names\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"groupby\\": \\"louvain\\", \\"num_categories\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_symbols\\": \\"\\", \\"log\\": false, \\"use_raw\\": true, \\"dendrogram\\": false, \\"var_group_positions\\": [], \\"var_group_rotation\\": null, \\"figsize\\": {\\"custom\\": \\"yes\\", \\"__current_case__\\": 0, \\"width\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"height\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"layer\\": \\"\\", \\"swap_axes\\": true, \\"violin_plot\\": {\\"stripplot\\": {\\"stripplot\\": \\"False\\", \\"__current_case__\\": 1}, \\"density_norm\\": \\"area\\"}, \\"row_palette\\": null, \\"standard_scale\\": \\"None\\", \\"seaborn_violinplot\\": {\\"bw_method\\": \\"scott\\", \\"bw_adjust\\": \\"1.0\\", \\"orient\\": null, \\"linewidth\\": \\"0.0\\", \\"color\\": null, \\"saturation\\": \\"0.75\\"}, \\"yticklabels\\": false, \\"vmin\\": null, \\"vmax\\": null, \\"vcenter\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"9386a889-5f57-4c24-b804-d1018a2c4d24","when":null,"workflow_outputs":[{"label":"Stacked violin of top genes on clusters","output_name":"out_png","uuid":"803ac216-4803-4e88-a1f1-9b3a8597ada0"}]},"65":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":65,"input_connections":{"adata":{"id":58,"output_name":"data_param"},"method|num_categories":{"id":60,"output_name":"integer_param"},"method|var_names|var_names":{"id":59,"output_name":"text_param"}},"inputs":[{"description":"runtime parameter for tool Scanpy plot","name":"method"}],"label":"Dotplot of top genes on clusters","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7283.8412597064325,"top":547.1373222113621},"post_job_actions":{"RenameDatasetActionout_png":{"action_arguments":{"newname":"Dotplot of top genes on clusters"},"action_type":"RenameDatasetAction","output_name":"out_png"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.dotplot\\", \\"__current_case__\\": 2, \\"var_names\\": {\\"type\\": \\"custom\\", \\"__current_case__\\": 1, \\"var_names\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"groupby\\": \\"louvain\\", \\"num_categories\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_symbols\\": \\"\\", \\"log\\": false, \\"use_raw\\": true, \\"dendrogram\\": true, \\"var_group_positions\\": [], \\"var_group_rotation\\": null, \\"figsize\\": {\\"custom\\": \\"no\\", \\"__current_case__\\": 1}, \\"layer\\": \\"\\", \\"color_map\\": \\"viridis\\", \\"dot_max\\": null, \\"dot_min\\": null, \\"standard_scale\\": \\"None\\", \\"expression_cutoff\\": \\"0.0\\", \\"mean_only_expressed\\": false, \\"matplotlib_pyplot_scatter\\": {\\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"linewidths\\": \\"0.0\\", \\"edgecolors\\": \\"face\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"7234b677-3502-42bb-a1ca-7c62a6118a52","when":null,"workflow_outputs":[{"label":"Dotplot of top genes on clusters","output_name":"out_png","uuid":"5ee15bc7-6913-48bf-8a1e-38b6e009b452"}]},"66":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":66,"input_connections":{"adata":{"id":48,"output_name":"anndata_out"},"method|figsize|height":{"id":60,"output_name":"integer_param"},"method|figsize|width":{"id":60,"output_name":"integer_param"}},"inputs":[],"label":"Heatmap of 20 genes","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7279.9117328730135,"top":1420.1202714194746},"post_job_actions":{"RenameDatasetActionout_png":{"action_arguments":{"newname":"Heatmap of top 20 highly ranked genes"},"action_type":"RenameDatasetAction","output_name":"out_png"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.rank_genes_groups_heatmap\\", \\"__current_case__\\": 27, \\"var_names\\": {\\"type\\": \\"all\\", \\"__current_case__\\": 0}, \\"groups\\": \\"\\", \\"n_genes\\": \\"20\\", \\"key\\": \\"\\", \\"min_logfoldchange\\": null, \\"log\\": false, \\"use_raw\\": true, \\"dendrogram\\": true, \\"var_group_positions\\": [], \\"var_group_rotation\\": null, \\"figsize\\": {\\"custom\\": \\"yes\\", \\"__current_case__\\": 0, \\"width\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"height\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"layer\\": \\"\\", \\"gene_symbols\\": \\"\\", \\"swap_axes\\": false, \\"show_gene_labels\\": false, \\"standard_scale\\": \\"None\\", \\"matplotlib_pyplot_imshow\\": {\\"cmap\\": \\"viridis\\", \\"interpolation\\": \\"None\\", \\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"origin\\": \\"upper\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"92ad3339-6642-4452-a993-aade06db8c56","when":null,"workflow_outputs":[{"label":"Heatmap of top 20 highly ranked genes","output_name":"out_png","uuid":"2e246f95-55da-43b6-9c8e-7d21ed139794"}]},"67":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","errors":null,"id":67,"input_connections":{"adata":{"id":48,"output_name":"anndata_out"},"method|color":{"id":61,"output_name":"text_param"}},"inputs":[{"description":"runtime parameter for tool Scanpy plot","name":"method"}],"label":"UMAP of louvain and top ranked genes","name":"Scanpy plot","outputs":[{"name":"out_png","type":"png"}],"position":{"left":7280.869257089268,"top":1146.0872774690602},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.10.2+galaxy0","tool_shed_repository":{"changeset_revision":"0185bd174ea0","name":"scanpy_plot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_common\\": {\\"show_log\\": false}, \\"format\\": \\"png\\", \\"method\\": {\\"method\\": \\"pl.umap\\", \\"__current_case__\\": 16, \\"color\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"use_raw\\": true, \\"edges\\": {\\"edges\\": \\"False\\", \\"__current_case__\\": 1}, \\"arrows\\": false, \\"sort_order\\": true, \\"groups\\": \\"\\", \\"plot\\": {\\"components\\": [], \\"projection\\": \\"2d\\", \\"legend_loc\\": \\"right margin\\", \\"legend_fontsize\\": null, \\"legend_fontweight\\": \\"normal\\", \\"size\\": null, \\"color_map\\": null, \\"palette\\": null, \\"frameon\\": true, \\"ncols\\": \\"2\\", \\"wspace\\": \\"0.1\\", \\"hspace\\": \\"0.25\\", \\"title\\": \\"\\"}, \\"add_outline\\": false, \\"outline_color_border\\": null, \\"outline_color_gap\\": null, \\"outline_width_border\\": \\"0.3\\", \\"outline_width_gap\\": \\"0.05\\", \\"matplotlib_pyplot_scatter\\": {\\"alpha\\": null, \\"vmin\\": null, \\"vmax\\": null, \\"linewidths\\": \\"0.0\\", \\"edgecolors\\": null}, \\"layer\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10.2+galaxy0","type":"tool","uuid":"f54debde-f7a9-4615-8a81-6a4406ebcc02","when":null,"workflow_outputs":[{"label":"UMAP of louvain and top ranked genes","output_name":"out_png","uuid":"fabdf65a-483a-4295-a0e4-f7e582614091"}]}},"tags":["name:scRNA-seq","name:single-cell"],"uuid":"d6b2d1bb-6aaf-47cc-8956-5ae80b775de5","version":40},"readme":"# Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering\\n\\nThis workflow follows Scanpy legacy workflow [clustering 3k PBMCs](https://scanpy.readthedocs.io/en/stable/tutorials/basics/clustering-2017.html). For more details on concepts and parameters, please refer to the equivalent Galaxy-based [tutorial](https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html).\\n\\n## Inputs\\n\\n### Input datasets\\n- This workflow needs 3 files as input\\n    - A single-cell count matrix file in Matrix Market Exchange format\\n    - A cell barcodes file with a single barcode in each line. The barcodes should correspond to the cells in the matrix file\\n    - A genes/features tabular file with gene ids and gene names.\\n        - Cell Ranger v2 or earlier version call this file as `genes.tsv` and contains two columns:\\n            - Gene ID (Ensembl gene ID or other identifiers)\\n            - Gene Name (common gene name or symbol)\\n        - Cell Ranger v2 or earlier version call this file as `features.tsv` and contains three columns:\\n            - Feature ID (Ensembl gene ID or other identifiers)\\n            - Feature Name (common gene name or symbol)\\n            - Feature Type (e.g., Gene Expression, Antibody Capture, CRISPR Guide Capture, etc.)\\n### Input parameters\\n- The following parameters should be configured according to the data.\\n    - Minimum number of cells expressed per gene. The workflow default is 3.\\n    - Minimum number of genes expressed per cell. The workflow default is 200.\\n    - Maximum number of genes expressed per cell. The workflow default is 2500.\\n    - Size of the local neighborhood. Number of neighbours for computing neighborhood graph. The default is 15.\\n    - Size of the local neighborhood (aka resolution) in louvain algorithm. The default is 1.\\n\\n## Processing\\n\\n- The workflow creates an **Anndata** object from the given input files.\\n- Quality control performed. Cells are filtered by number of genes expressed, cells with high mitochondial content are removed.\\n- Then counts are normlized and scaled\\n- PCA is used for dimensionality reduction and 50 PCs are computed. Various plots are generated to inspect the PCA and PCA loadings that helps in chodeterminingnumber of  PCs to keep for further analysis.\\n- Clustering is performed by computing a neighbourhood graph, and then using **louvain** algorithm. neighborhood graph is embeded into UMAP and plotted.\\n- Marker genes are identified using **Wilcoxon rank sum test**. Marker genes expressions are visualized in various plots.\\n- Optionally, louvain clusters can be annotated with cell types based on the marker genes.\\n\\n## Outputs\\n\\n- Final output is an Anndata object with annotations of louvain clusters.\\n- Some informative plots from QC to the end results\\n","changelog":"## [0.1] - 2024-10-09\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Single-Cell RNA-seq Analysis: Scanpy Preprocessing and Clustering\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Minimum number of cells expressed\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Barcodes\\"]@{ shape: doc }\\n2[\\"ℹ️ Genes\\"]@{ shape: doc }\\n3[\\"ℹ️ Matrix\\"]@{ shape: doc }\\n4[\\"ℹ️ Input is from Cell Ranger v2 or earlier versions\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Mitochondrial genes start with pattern\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Minimum number of genes expressed\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Maximum number of genes expressed\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Number of neighbours for computing neighborhood graph\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Number of PCs to use for computing neighborhood graph\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Louvain resolution\\"]@{ shape: lean-l }\\n11[\\"ℹ️ Manually annotate celltypes?\\"]@{ shape: lean-l }\\n12[\\"ℹ️ Annotate louvain clusters with these cell types\\"]@{ shape: lean-l }\\n13[\\"Pick parameter value\\"]@{ shape: process }\\n0 --> 13\\n14[\\"Anndata from 10x v2 or earlier\\"]@{ shape: process }\\n3 --> 14\\n1 --> 14\\n2 --> 14\\n4 --> 14\\n15[\\"Map and switch the input CellRager version\\"]@{ shape: process }\\n4 --> 15\\n16[\\"Pick parameter value\\"]@{ shape: process }\\n6 --> 16\\n17[\\"Pick parameter value\\"]@{ shape: process }\\n7 --> 17\\n18[\\"Pick parameter value\\"]@{ shape: process }\\n8 --> 18\\n19[\\"Pick parameter value\\"]@{ shape: process }\\n10 --> 19\\n20[\\"Anndata from 10x V3 or later\\"]@{ shape: process }\\n3 --> 20\\n1 --> 20\\n2 --> 20\\n15 --> 20\\n21[\\"Choose the Anndata\\"]@{ shape: process }\\n20 --> 21\\n14 --> 21\\n22[\\"Filter genes by cells expressed\\"]@{ shape: process }\\n21 --> 22\\n13 --> 22\\n23[\\"Inspect AnnData\\"]@{ shape: process }\\n21 --> 23\\n24[\\"Flag Mitochondrial genes\\"]@{ shape: process }\\n22 --> 24\\n5 --> 24\\n25[\\"Calculate quality metrics\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Scatter plot n_genes vs pct_mito\\"]@{ shape: process }\\n25 --> 26\\n27[\\"Scatter plot n_counts vs n_genes\\"]@{ shape: process }\\n25 --> 27\\n28[\\"Violin plot n_genes, n_counts, pct_mito\\"]@{ shape: process }\\n25 --> 28\\n29[\\"Filter cells by min genes expressed\\"]@{ shape: process }\\n25 --> 29\\n16 --> 29\\n30[\\"Filter cells by max genes expressed\\"]@{ shape: process }\\n29 --> 30\\n17 --> 30\\n31[\\"Filter cells by percentage of mito genes\\"]@{ shape: process }\\n30 --> 31\\n32[\\"Normalize by target sum\\"]@{ shape: process }\\n31 --> 32\\n33[\\"Logarithmize\\"]@{ shape: process }\\n32 --> 33\\n34[\\"Freeze raw attribute\\"]@{ shape: process }\\n33 --> 34\\n35[\\"Annotate highly variable genes\\"]@{ shape: process }\\n34 --> 35\\n36[\\"Manipulate AnnData\\"]@{ shape: process }\\n35 --> 36\\n37[\\"Plot highly variable\\"]@{ shape: process }\\n35 --> 37\\n38[\\"Regress out\\"]@{ shape: process }\\n36 --> 38\\n39[\\"Scale data\\"]@{ shape: process }\\n38 --> 39\\n40[\\"Compute PCA\\"]@{ shape: process }\\n39 --> 40\\n41[\\"Compute Neighborhood graph\\"]@{ shape: process }\\n40 --> 41\\n18 --> 41\\n9 --> 41\\n42[\\"Scanpy plot\\"]@{ shape: process }\\n40 --> 42\\n43[\\"Scanpy plot\\"]@{ shape: process }\\n40 --> 43\\n44[\\"Scanpy plot\\"]@{ shape: process }\\n40 --> 44\\n45[\\"Embed using UMAP\\"]@{ shape: process }\\n41 --> 45\\n46[\\"Louvain clustering\\"]@{ shape: process }\\n45 --> 46\\n19 --> 46\\n47[\\"UMAP of louvain\\"]@{ shape: process }\\n46 --> 47\\n48[\\"Rank genes by Wilcoxon test\\"]@{ shape: process }\\n46 --> 48\\n49[\\"Inspect AnnData\\"]@{ shape: process }\\n48 --> 49\\n50[\\"Inspect clusters\\"]@{ shape: process }\\n48 --> 50\\n51[\\"Scanpy plot\\"]@{ shape: process }\\n48 --> 51\\n52[\\"Manipulate AnnData\\"]@{ shape: process }\\n48 --> 52\\n12 --> 52\\n11 --> 52\\n53[\\"General information about the final Anndata object\\"]@{ shape: process }\\n48 --> 53\\n54[\\"Select top ranked genes\\"]@{ shape: process }\\n49 --> 54\\n55[\\"Count number of clusters\\"]@{ shape: process }\\n49 --> 55\\n56[\\"Select top ranked genes with louvain\\"]@{ shape: process }\\n49 --> 56\\n57[\\"Count number of cells per cluster\\"]@{ shape: process }\\n50 --> 57\\n58[\\"Pick parameter value\\"]@{ shape: process }\\n52 --> 58\\n48 --> 58\\n59[\\"Parse parameter value\\"]@{ shape: process }\\n54 --> 59\\n60[\\"Parse parameter value\\"]@{ shape: process }\\n55 --> 60\\n61[\\"Parse parameter value\\"]@{ shape: process }\\n56 --> 61\\n62[\\"Scanpy plot\\"]@{ shape: process }\\n58 --> 62\\n11 --> 62\\n63[\\"Violin plot of top genes on clusters\\"]@{ shape: process }\\n48 --> 63\\n59 --> 63\\n64[\\"Stacked violin of top genes on clusters\\"]@{ shape: process }\\n48 --> 64\\n60 --> 64\\n60 --> 64\\n60 --> 64\\n59 --> 64\\n65[\\"Dotplot of top genes on clusters\\"]@{ shape: process }\\n58 --> 65\\n60 --> 65\\n59 --> 65\\n66[\\"Heatmap of 20 genes\\"]@{ shape: process }\\n48 --> 66\\n60 --> 66\\n60 --> 66\\n67[\\"UMAP of louvain and top ranked genes\\"]@{ shape: process }\\n48 --> 67\\n61 --> 67\\n```\\n","updated":"2024-10-09T00:00:00","trsID":"#workflow/github.com/iwc-workflows/scanpy-clustering/main","dockstore_id":30355,"categories":[],"collections":["Single Cell"],"doi":"10.5281/zenodo.15078824","tests":[{"doc":"Test outline for Preprocessing-and-Clustering-of-single-cell-RNA-seq-data-with-Scanpy","job":{"Barcodes":{"class":"File","location":"https://zenodo.org/record/3581213/files/barcodes.tsv","filetype":"txt"},"Genes":{"class":"File","location":"https://zenodo.org/record/3581213/files/genes.tsv","filetype":"tabular"},"Matrix":{"class":"File","location":"https://zenodo.org/record/3581213/files/matrix.mtx","filetype":"mtx"},"Minimum number of cells expressed":3,"Input is from Cell Ranger v2 or earlier versions":true,"Mitochondrial genes start with pattern":"MT-","Minimum number of genes expressed":200,"Maximum number of genes expressed":2500,"Number of neighbours for computing neighborhood graph":10,"Number of PCs to use for computing neighborhood graph":10,"Louvain resolution":0.45,"Manually annotate celltypes?":true,"Annotate louvain clusters with these cell types":"CD4+ T, CD14+, B, CD8+ T, FCGR3A+, NK, Dendritic, Megakaryocytes"},"outputs":{"Anndata with Celltype Annotation":{"asserts":{"has_h5_keys":{"keys":"uns/rank_genes_groups"}}},"UMAP of louvain and top ranked genes":{"has_size":{"size":696399,"delta":60000},"has_image_width":{"width":1146,"delta":60},"has_image_height":{"height":2099,"delta":105}},"Dotplot of top genes on clusters":{"has_size":{"size":39916,"delta":3900},"has_image_width":{"width":535,"delta":27},"has_image_height":{"height":379,"delta":19}},"Initial Anndata General Info":{"asserts":{"has_text":{"text":"AnnData object with n_obs × n_vars = 2700 × 32738"}}},"Scatter plot of plot n_counts vs n_genes":{"has_size":{"size":27381,"delta":2700},"has_image_width":{"width":630,"delta":40},"has_image_height":{"height":413,"delta":25}},"Plot highly variable":{"has_size":{"size":67135,"delta":6500},"has_image_width":{"width":1061,"delta":50},"has_image_height":{"height":432,"delta":27}},"Plot PCA over view with genes":{"has_size":{"size":61974,"delta":600},"has_image_width":{"width":1158,"delta":60},"has_image_height":{"height":409,"delta":25}},"UMAP of louvain":{"has_size":{"size":68416,"delta":6000},"has_image_width":{"width":601,"delta":30},"has_image_height":{"height":429,"delta":25}},"Ranked genes with Wilcoxon test":{"asserts":{"has_line":{"line":"LDHB\\tLYZ\\tCD74\\tCCL5\\tLST1\\tNKG7\\tHLA-DPA1\\tPF4"}}},"Plot Rank gene groups":{"has_size":{"size":164953,"delta":16000},"has_image_width":{"width":1559,"delta":75},"has_image_height":{"height":1192,"delta":50}},"UMAP with annotated cell types":{"has_size":{"size":70265,"delta":7000},"has_image_width":{"width":535,"delta":27},"has_image_height":{"height":429,"delta":25}},"Stacked violin of top genes on clusters":{"has_size":{"size":22147,"delta":2200},"has_image_width":{"width":299,"delta":15},"has_image_height":{"height":258,"delta":15}},"Heatmap of top 20 highly ranked genes":{"has_size":{"size":162549,"delta":16000},"has_image_width":{"width":788,"delta":40},"has_image_height":{"height":505,"delta":25}},"Scatter plot of n_genes vs pct_mito":{"has_size":{"size":33175},"has_image_width":{"width":617,"delta":30},"has_image_height":{"height":413,"delta":20}},"Violin plot n_genes, n_counts, pct_mito":{"has_size":{"size":71188},"has_image_width":{"width":1508,"delta":75},"has_image_height":{"height":490,"delta":25}},"Anndata with raw attribute":{"asserts":{"has_h5_keys":{"keys":"uns/log1p"}}},"Plot genes according to contributions to PCs":{"has_size":{"size":71020,"delta":7000},"has_image_width":{"width":1601,"delta":80},"has_image_height":{"height":430,"delta":27}},"Elbow plot of PCs and variance":{"has_size":{"size":20174,"delta":2000},"has_image_width":{"width":569,"delta":25},"has_image_height":{"height":453,"delta":25}},"General information about the final Anndata object":{"path":"test-data/General information about the final Anndata object.txt"},"Number of cells per cluster":{"asserts":{"has_line":{"line":"6\\t34"}}},"Violin plot of top genes on clusters":{"has_size":{"size":261433,"delta":26000},"has_image_width":{"width":5532,"delta":280},"has_image_height":{"height":418,"delta":20}}}}]}],"path":"./workflows/scRNAseq/scanpy-clustering"},{"version":1.2,"workflows":[{"name":"COVID-19-SE-WGS-ILLUMINA","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/se-wgs-variation.ga","testParameterFiles":["/se-wgs-variation-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflows performs single end read mapping with bowtie2 followed by sensitive variant calling across a wide range of AFs with lofreq","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: variation analysis on WGS SE data","release":"0.1.6","steps":{"0":{"annotation":"Illumina reads with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Illumina reads with fastqsanger encoding","name":"Single End Collection"}],"label":"Single End Collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":189.96875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fac0f79d-54bd-4c26-b18e-2804067a6522","when":null,"workflow_outputs":[]},"1":{"annotation":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","name":"NC_045512.2 FASTA sequence of SARS-CoV-2"}],"label":"NC_045512.2 FASTA sequence of SARS-CoV-2","name":"Input dataset","outputs":[],"position":{"left":95.984375,"top":601.984375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\", \\"fasta.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c9a062f4-3154-47de-b482-d8168a70f939","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":2,"input_connections":{"single_paired|in1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"out1","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":237.53125,"top":0},"post_job_actions":{"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"},"RenameDatasetActionout1":{"action_arguments":{"newname":"Trimmed and quality-filtered reads (fastp result)"},"action_type":"RenameDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"in1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"226bdff2-28d7-46ec-9a17-887d577056f6","when":null,"workflow_outputs":[{"label":"preprocessed_reads","output_name":"out1","uuid":"21a2c281-bb1a-4632-9cdc-f67b7465041a"},{"label":"fastp_html_report","output_name":"report_html","uuid":"d31c518e-b26c-42e7-9d62-c4adde285a27"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","errors":null,"id":3,"input_connections":{"library|input_1":{"id":2,"output_name":"out1"},"reference_genome|own_file":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bowtie2","name":"library"},{"description":"runtime parameter for tool Bowtie2","name":"reference_genome"}],"label":null,"name":"Bowtie2","outputs":[{"name":"output","type":"bam"},{"name":"mapping_stats","type":"txt"}],"position":{"left":476.203125,"top":210.890625},"post_job_actions":{"HideDatasetActionmapping_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"mapping_stats"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Mapped reads (bowtie2 result)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","tool_shed_repository":{"changeset_revision":"d5ceb9f3c25b","name":"bowtie2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"simple\\", \\"__current_case__\\": 0, \\"presets\\": \\"--very-sensitive\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unaligned_file\\": false, \\"aligned_file\\": false}, \\"reference_genome\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 1, \\"own_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"set_id_auto\\", \\"__current_case__\\": 2}, \\"sam_options\\": {\\"sam_options_selector\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_mapping_stats\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.3+galaxy1","type":"tool","uuid":"befd5b2a-7f19-48bf-9133-55dddc3791f2","when":null,"workflow_outputs":[{"label":"mapped_reads","output_name":"output","uuid":"cea7a009-c5b0-418f-82b0-4a379ece0fe5"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","errors":null,"id":4,"input_connections":{"inputFile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":752.484375,"top":362.140625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"bam\\", \\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"chromInfo\\": \\"/cvmfs/data.galaxyproject.org/managed/len/ucsc/?.len\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFile|__identifier__\\": \\"SRR11247078\\", \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": true, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"298f0df2-22f1-405d-92b3-272f73e3ef5b","when":null,"workflow_outputs":[{"label":"markduplicates_stats","output_name":"metrics_file","uuid":"873c1736-4a15-4805-a1ee-e64808b10a0a"},{"label":"markduplicates_reads","output_name":"outFile","uuid":"10acc683-8d34-4017-892e-21c7bea450eb"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":5,"input_connections":{"results_0|software_cond|input":{"id":2,"output_name":"report_json"},"results_1|software_cond|input":{"id":3,"output_name":"mapping_stats"},"results_2|software_cond|output_0|input":{"id":4,"output_name":"metrics_file"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":831.90625,"top":18.71875},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"},"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"bowtie2\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"cba65a56-8563-4777-8a87-b4c0158d8015","when":null,"workflow_outputs":[{"label":"preprocessing_and_mapping_reports","output_name":"html_report","uuid":"01575724-b29c-465d-b9f0-8577ce82fd00"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","errors":null,"id":6,"input_connections":{"reads":{"id":4,"output_name":"outFile"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Realign reads","name":"reference_source"}],"label":null,"name":"Realign reads","outputs":[{"name":"realigned","type":"bam"}],"position":{"left":980.125,"top":538.46875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"aa35ee7f3ab2","name":"lofreq_viterbi","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"keepflags\\": false, \\"bq2_handling\\": {\\"replace_bq2\\": \\"keep\\", \\"__current_case__\\": 0, \\"defqual\\": \\"2\\"}}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"967ad159-752b-4117-82f8-7fa9296897eb","when":null,"workflow_outputs":[{"label":"realigned_deduplicated_mapped_reads","output_name":"realigned","uuid":"d42cf1f6-fe7e-4559-8d47-d881ff16308e"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1","errors":null,"id":7,"input_connections":{"reads":{"id":6,"output_name":"realigned"},"strategy|reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Insert indel qualities","outputs":[{"name":"output","type":"bam"}],"position":{"left":1217.5,"top":636.09375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Fully processed reads for variant calling (deduplicated, realigned reads with added indelquals)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1","tool_shed_repository":{"changeset_revision":"971e07ca4456","name":"lofreq_indelqual","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strategy\\": {\\"selector\\": \\"dindel\\", \\"__current_case__\\": 1, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy1","type":"tool","uuid":"b2abc152-bdea-418f-af6b-566f83418ffb","when":null,"workflow_outputs":[{"label":"realigned_deduplicated_mapped_reads_with_indel_quals","output_name":"output","uuid":"9fc19a53-7ff5-4e69-9aa2-05400670ff3b"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","errors":null,"id":8,"input_connections":{"reads":{"id":7,"output_name":"output"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Call variants","name":"reference_source"}],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"left":1450.90625,"top":733.734375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","tool_shed_repository":{"changeset_revision":"b11e8e9c23bf","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"5\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": false}, \\"bc_quals\\": {\\"min_bq\\": \\"30\\", \\"min_alt_bq\\": \\"30\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": true}}}, \\"map_quals\\": {\\"min_mq\\": \\"20\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_custom\\", \\"__current_case__\\": 3, \\"sig\\": \\"0.0005\\", \\"bonf\\": \\"0\\", \\"others\\": false}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy3","type":"tool","uuid":"c1515432-2f61-498d-9dcd-b44da6f923c7","when":null,"workflow_outputs":[{"label":"called_variant","output_name":"variants","uuid":"1f968e8a-6574-4002-b2c6-a7faf96111ef"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","errors":null,"id":9,"input_connections":{"invcf":{"id":8,"output_name":"variants"}},"inputs":[],"label":null,"name":"Lofreq filter","outputs":[{"name":"outvcf","type":"vcf"}],"position":{"left":1498.515625,"top":612.421875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"7dfca164d2e3","name":"lofreq_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"af\\": {\\"af_min\\": \\"0.0\\", \\"af_max\\": \\"0.0\\"}, \\"coverage\\": {\\"cov_min\\": \\"0\\", \\"cov_max\\": \\"0\\"}, \\"filter_by_type\\": {\\"keep_only\\": \\"\\", \\"__current_case__\\": 0, \\"qual\\": {\\"snvqual_filter\\": {\\"snvqual\\": \\"no\\", \\"__current_case__\\": 0}, \\"indelqual_filter\\": {\\"indelqual\\": \\"no\\", \\"__current_case__\\": 0}}}, \\"flag_or_drop\\": \\"--print-all\\", \\"invcf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sb\\": {\\"sb_filter\\": {\\"strand_bias\\": \\"mtc\\", \\"__current_case__\\": 2, \\"sb_alpha\\": \\"0.001\\", \\"sb_mtc\\": \\"fdr\\", \\"sb_compound\\": true, \\"sb_indels\\": false}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"f9db3062-6214-4c07-a006-001277dc0be5","when":null,"workflow_outputs":[{"label":"soft_filtered_variants","output_name":"outvcf","uuid":"9f6aebda-a899-4303-b186-7ae5faff79e2"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","errors":null,"id":10,"input_connections":{"input":{"id":9,"output_name":"outvcf"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":"SnpEff eff covid19 version","name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"left":1544.40625,"top":285.5},"post_job_actions":{"RenameDatasetActionsnpeff_output":{"action_arguments":{"newname":"Final (SnpEff-)annotated variants"},"action_type":"RenameDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","tool_shed_repository":{"changeset_revision":"2a3a00c1fa0a","name":"snpeff_sars_cov_2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": false, \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-upstream\\", \\"-no-utr\\"], \\"generate_stats\\": true, \\"genome_version\\": \\"NC_045512.2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": true, \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5covid19","type":"tool","uuid":"6b370dfb-970b-49c2-8060-c017f50508e8","when":null,"workflow_outputs":[{"label":"annotated_variants","output_name":"snpeff_output","uuid":"c9297840-841f-430d-81c9-bfd721b8203b"},{"label":"annotated_variants_stats","output_name":"statsFile","uuid":"adfc092c-6cae-410c-b86c-36fdb0703e9b"}]}},"tags":["COVID-19","covid19.galaxyproject.org"],"uuid":"5171bc59-590c-4e97-8db8-004f73928627","version":6},"readme":"COVID-19: variation analysis on WGS SE data\\n-------------------------------------------\\n\\nThis workflows performs single end read mapping with bowtie2 followed by\\nsensitive variant calling across a wide range of AFs with lofreq and variant\\nannotation with snpEff 4.5covid19.\\n","changelog":"# Changelog\\n\\n## [0.1.6] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3`\\n\\n## [0.1.5] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.3.4.2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy2`\\n\\n## [0.1.4] - 2023-11-20\\n\\n- Fix author in dockstore\\n- Fix input fastq URL for test\\n- Update output test\\n\\n## [0.1.3] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.2] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.1.1] - 2021-06-14\\n\\n### Fixed\\n\\nFix reference to test file in .dockstore.yml\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: variation analysis on WGS SE data workflow\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: variation analysis on WGS SE data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Single End Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ NC_045512.2 FASTA sequence of SARS-CoV-2\\"]@{ shape: doc }\\n2[\\"fastp\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Bowtie2\\"]@{ shape: process }\\n2 --> 3\\n1 --> 3\\n4[\\"MarkDuplicates\\"]@{ shape: process }\\n3 --> 4\\n5[\\"MultiQC\\"]@{ shape: process }\\n2 --> 5\\n3 --> 5\\n4 --> 5\\n6[\\"Realign reads\\"]@{ shape: process }\\n4 --> 6\\n1 --> 6\\n7[\\"Insert indel qualities\\"]@{ shape: process }\\n6 --> 7\\n1 --> 7\\n8[\\"Call variants\\"]@{ shape: process }\\n7 --> 8\\n1 --> 8\\n9[\\"Lofreq filter\\"]@{ shape: process }\\n8 --> 9\\n10[\\"SnpEff eff covid19 version\\"]@{ shape: process }\\n9 --> 10\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-se-illumina-wgs-variant-calling/COVID-19-SE-WGS-ILLUMINA","dockstore_id":16244,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078497","tests":[{"doc":"Test workflow execution using paired end illumina accession","job":{"NC_045512.2 FASTA sequence of SARS-CoV-2":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1"},"Single End Collection":{"class":"Collection","collection_type":"list","elements":[{"identifier":"SRR11605118","class":"File","location":"https://www.be-md.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR11605118"}]}},"outputs":{"annotated_variants":{"attributes":{},"element_tests":{"SRR11605118":{"asserts":{"has_line":{"line":"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tFILTER\\tINFO"},"has_text_matching":{"expression":"NC_045512.2\\t16111\\t.\\tC\\tT\\t[0-9.]*\\tPASS\\tDP=[0-9]*;AF=0.[89][0-9]*;SB=0;DP4=[0-9]*,[0-9]*,[0-9]*,[0-9]*;EFF=SYNONYMOUS_CODING\\\\(LOW|SILENT|Cta/Tta|L5283|7096|ORF1ab|protein_coding|CODING|GU280_gp01|2|T\\\\),SYNONYMOUS_CODING\\\\(LOW|SILENT|Cta/Tta|L891|931|ORF1ab|protein_coding|CODING|YP_009725307.1|2|T|WARNING_TRANSCRIPT_NO_START_CODON\\\\)"}}}}}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-se-illumina-wgs-variant-calling"},{"version":1.2,"workflows":[{"name":"COVID-19-VARIATION-REPORTING","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/variation-reporting.ga","testParameterFiles":["/variation-reporting-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow takes a VCF dataset of variants produced by any of the *-variant-calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: variation analysis reporting","release":"0.3.4","steps":{"0":{"annotation":"Variation data in VCF format. Can be the output of any of the workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Variation data in VCF format. Can be the output of any of the workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling","name":"Variation data to report"}],"label":"Variation data to report","name":"Input dataset collection","outputs":[],"position":{"left":10.28125,"top":34.08333333333334},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"vcf\\", \\"vcf_bgzip\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"577438f0-38b1-4fc1-9145-935566d62347","when":null,"workflow_outputs":[]},"1":{"annotation":"Allele Frequency Filter. This is the minimum allele frequency required for variants to be included in the reports.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Allele Frequency Filter. This is the minimum allele frequency required for variants to be included in the reports.","name":"AF Filter"}],"label":"AF Filter","name":"Input parameter","outputs":[],"position":{"left":5.3125,"top":186.91145833333331},"tool_id":null,"tool_state":"{\\"default\\": 0.05, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2e5a5b38-c204-45a2-98e2-e113bce5a14b","when":null,"workflow_outputs":[{"label":"af_filter_threshold","output_name":"output","uuid":"e55d47c5-7ea0-45c4-8844-47bf29b88542"}]},"2":{"annotation":"Depth Filter. This is the minimum depth of all alignments at a variant site. ","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Depth Filter. This is the minimum depth of all alignments at a variant site. ","name":"DP Filter"}],"label":"DP Filter","name":"Input parameter","outputs":[],"position":{"left":0.78125,"top":284.9270833333333},"tool_id":null,"tool_state":"{\\"default\\": 1, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c7fc64c7-51c7-465b-9184-ca834d4cf6b2","when":null,"workflow_outputs":[{"label":"dp_filter_threshold","output_name":"output","uuid":"220a9662-8b6f-4161-8539-ecb613b4892a"}]},"3":{"annotation":"Depth Filter for variant allele. This is the minimum depth of alignments supporting a variant.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Depth Filter for variant allele. This is the minimum depth of alignments supporting a variant.","name":"DP_ALT Filter"}],"label":"DP_ALT Filter","name":"Input parameter","outputs":[],"position":{"left":0,"top":367.42708333333337},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"145bbc55-8893-413d-9620-f3d85e65527a","when":null,"workflow_outputs":[{"label":"dp_alt_filter_threshold","output_name":"output","uuid":"7f7b09f6-e4e5-43af-881b-f336906bcb91"}]},"4":{"annotation":"A custom tabular file mapping NCBI RefSeq Protein identifiers as used by snpEff version 4.5covid19 to their commonly used names. Can be obtained from https://doi.org/10.5281/zenodo.4555734","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"A custom tabular file mapping NCBI RefSeq Protein identifiers as used by snpEff version 4.5covid19 to their commonly used names. Can be obtained from https://doi.org/10.5281/zenodo.4555734","name":"gene products translations"}],"label":"gene products translations","name":"Input dataset","outputs":[],"position":{"left":690.015625,"top":38.41145833333334},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8b89e674-9012-4f66-a59a-ce05d2fe4223","when":null,"workflow_outputs":[]},"5":{"annotation":"Number of Clusters to use in Variant Frequency Plot.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Number of Clusters to use in Variant Frequency Plot.","name":"Number of Clusters"}],"label":"Number of Clusters","name":"Input parameter","outputs":[],"position":{"left":3413.578125,"top":671.9114583333334},"tool_id":null,"tool_state":"{\\"default\\": 1, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5ca7b1a7-e1ee-4b52-aab4-eaf23949e1da","when":null,"workflow_outputs":[{"label":"plot_number_of_clusters","output_name":"output","uuid":"c02cc297-3f36-4f78-bede-3aa378436672"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":6,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":246.65625,"top":32.08333333333334},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": \\"DP >= 1\\"}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"386eb313-89c1-47df-bd3d-976780eff740","when":null,"workflow_outputs":[{"label":"prefiltered_variants","output_name":"output","uuid":"498b63d3-2449-4633-b9a4-360553d14844"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":7,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"},"components_3|param_type|component_value":{"id":2,"output_name":"output"},"components_5|param_type|component_value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":245,"top":156.36458333333331},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( ( DP4[2] + DP4[3] ) < ( \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) ) | ( DP < \\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" ) | ( ( AF * DP ) < ( \\"}}, {\\"__index__\\": 5, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 6, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" - 0.5 ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"ec1a4872-b9bc-41f0-8675-1f26fdd19f16","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":8,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"},"components_3|param_type|component_value":{"id":2,"output_name":"output"},"components_5|param_type|component_value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":241.71875,"top":463.83333333333337},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"min_af_\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"|min_dp_\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"|min_dp_alt_\\"}}, {\\"__index__\\": 5, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"c9916563-c914-4f5f-90c6-d6c7619e1a98","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":9,"input_connections":{"filter_expression|expr":{"id":7,"output_name":"out1"},"filtering|add_filter":{"id":8,"output_name":"out1"},"input":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"},{"description":"runtime parameter for tool SnpSift Filter","name":"filtering"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":477.421875,"top":322.0677083333333},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"add_filter\\", \\"__current_case__\\": 3, \\"add_filter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"9a6cb40d-476b-4096-adcc-cb601398d0b4","when":null,"workflow_outputs":[{"label":"filtered_variants","output_name":"output","uuid":"d6db8dad-1774-4ec8-a858-26b17a287252"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":10,"input_connections":{"input":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":713.4375,"top":219.05208333333331},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered extracted variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\".\\", \\"extract\\": \\"CHROM POS FILTER REF ALT DP AF DP4 SB EFF[*].IMPACT EFF[*].FUNCLASS EFF[*].EFFECT EFF[*].GENE EFF[*].CODON EFF[*].AA EFF[*].TRID DP4[2] DP4[3]\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": true, \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"085e8f20-3eea-4d97-8e52-6e8a9aa3fbd7","when":null,"workflow_outputs":[{"label":"filtered_extracted_variants","output_name":"output","uuid":"578ee06f-e803-42e2-8afe-b9e88f465cdb"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2","errors":null,"id":11,"input_connections":{"original_file":{"id":10,"output_name":"output"},"replace_information":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace column","outputs":[{"name":"outfile_replace","type":"txt"}],"position":{"left":944.3125,"top":83.03645833333331},"post_job_actions":{"ChangeDatatypeActionoutfile_replace":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"outfile_replace"},"RenameDatasetActionoutfile_replace":{"action_arguments":{"newname":"Extracted variants with filtered and renamed effects"},"action_type":"RenameDatasetAction","output_name":"outfile_replace"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2","tool_shed_repository":{"changeset_revision":"d533e4b75800","name":"replace_column_by_key_value_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column_replace\\": \\"16\\", \\"delimiter\\": \\"tab\\", \\"original_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pass_comments\\": \\"#\\", \\"replace_information\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skip_lines\\": \\"1\\", \\"unknowns_strategy\\": \\"skip\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2","type":"tool","uuid":"0850f6d5-ff1f-47f9-8e49-b7eb5782828a","when":null,"workflow_outputs":[{"label":"filtered_and_renamed_effects","output_name":"outfile_replace","uuid":"25460c36-c35c-4233-97ba-ec0a7aec39e4"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":12,"input_connections":{"input":{"id":11,"output_name":"outfile_replace"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1164.734375,"top":240.34895833333331},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Extracted variants with filtered and renamed effects and AF recalculated"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c7\\", \\"add_column\\": {\\"mode\\": \\"I\\", \\"__current_case__\\": 1, \\"pos\\": \\"8\\"}, \\"new_column_name\\": \\"AFcaller\\"}, {\\"__index__\\": 1, \\"cond\\": \\"round((c18 + c19) / c6, 6)\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"7\\"}, \\"new_column_name\\": \\"AF\\"}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c457f29e-dea5-449a-84c4-4b192da88257","when":null,"workflow_outputs":[{"label":"af_recalculated","output_name":"out_file1","uuid":"86d866eb-b5ff-409c-aaee-d23deac15a47"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":13,"input_connections":{"in_file":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":1331.15625,"top":20.583333333333343},"post_job_actions":{"RenameDatasetActionout_file":{"action_arguments":{"newname":"Collapsed effects (Datamash on all variants)"},"action_type":"RenameDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"1,2,3,4,5,6,7,8,9,10\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"11\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"12\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"13\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"14\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"15\\"}, {\\"__index__\\": 5, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"16\\"}, {\\"__index__\\": 6, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"17\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"ec8aca39-fe1f-404a-9460-7a960cec37e6","when":null,"workflow_outputs":[{"label":"collapsed_effects","output_name":"out_file","uuid":"995044cd-d883-451e-b8a0-6905967f636b"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":14,"input_connections":{"infile":{"id":13,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1681.234375,"top":0},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Extracted variants with highest impact effects"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\s]+\\", \\"replace_pattern\\": \\"\\\\\\\\t$1\\\\\\\\t$2\\\\\\\\t$3\\\\\\\\t$4\\\\\\\\t$5\\\\\\\\t$6\\\\\\\\t$7\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": true, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}, {\\"__index__\\": 1, \\"find_pattern\\": \\"(GroupBy|collapse)\\\\\\\\(([^)]+)\\\\\\\\)\\", \\"replace_pattern\\": \\"$2\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"25f665b6-2b6a-49a3-a395-29706cd772ab","when":null,"workflow_outputs":[{"label":"highest_impact_effects","output_name":"outfile","uuid":"689eb8e9-393c-4d2f-8ab6-98bcfc07892d"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":15,"input_connections":{"input_list":{"id":14,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1982.5,"top":28.067708333333343},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_multiple\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"e5459c74-11e1-45a5-beee-88c2dde61eab","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":16,"input_connections":{"input":{"id":15,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2227.890625,"top":212.38020833333331},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": false, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c5 + \'>\' + c6\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}, \\"new_column_name\\": \\"change\\"}, {\\"__index__\\": 1, \\"cond\\": \\"c3 + \':\' + c19\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}, \\"new_column_name\\": \\"change_with_pos\\"}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"cc8a768f-8078-4bdd-ae98-903f08619070","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":17,"input_connections":{"infile":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2434.0625,"top":80.17708333333331},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"All variants of all samples"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"EFF[*].\\", \\"replace_pattern\\": \\"\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"82276b6c-5c6e-49eb-8ee5-46041c273d30","when":null,"workflow_outputs":[{"label":"all_variants_all_samples","output_name":"outfile","uuid":"708f0b5e-2bc0-4a08-96e2-d590517c7ac9"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":18,"input_connections":{"in_file":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":2685.328125,"top":31.598958333333343},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"20\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"min\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"max\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"8\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"c40528c9-4fed-48c8-9c34-be2cf2c32745","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"Filter1","errors":null,"id":19,"input_connections":{"input":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2678.125,"top":244.73958333333331},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"c4==\'PASS\' or c4==\'.\'\\", \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"c02e3958-04f4-499e-b78e-f304e0fdaeed","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":20,"input_connections":{"in_file":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":2969.03125,"top":441.66145833333337},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"},"RenameDatasetActionout_file":{"action_arguments":{"newname":"Cross-sample variant stats (Datamash)"},"action_type":"RenameDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"3\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"min\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"max\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"19\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"13\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"a9af0104-3e05-48cb-9650-36f58d174cdf","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","errors":null,"id":21,"input_connections":{"infile1":{"id":19,"output_name":"out_file1"},"infile2":{"id":18,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":2926.71875,"top":51.98958333333334},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants of all samples with stats"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"20\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": true, \\"ignore_case\\": false, \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"7e27aac8-1d1e-4d82-88e3-59c4cd61baeb","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":22,"input_connections":{"in_file":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":2935.140625,"top":312.1458333333333},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"3\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"fac0bae2-3415-4aa3-a5f5-7950ea411d98","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":23,"input_connections":{"in_file":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":2900.640625,"top":646.4114583333334},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"20\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"90b934e8-ba30-49aa-9fa4-a371fdf71285","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":24,"input_connections":{"in_file":{"id":21,"output_name":"output"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":3170.828125,"top":15.098958333333343},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"1\\", \\"header_in\\": true, \\"header_out\\": true, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": true, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"unique\\", \\"op_column\\": \\"2\\"}], \\"print_full_line\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"5ba44453-dabf-420f-a49e-2918eae031d3","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile1":{"id":17,"output_name":"outfile"},"infile2":{"id":22,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3172.890625,"top":224.66145833333331},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"3\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": true, \\"ignore_case\\": false, \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3927af37-c003-4fac-a8af-88ba2b656b22","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","errors":null,"id":26,"input_connections":{"infile1":{"id":16,"output_name":"out_file1"},"infile2":{"id":23,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3130.890625,"top":592.9270833333334},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"20\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": true, \\"ignore_case\\": false, \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"472af513-7900-41a5-b049-feb2d630ad2c","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"Cut1","errors":null,"id":27,"input_connections":{"input":{"id":24,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3394.890625,"top":119.89583333333331},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c4,c6,c7,c13,c14,c15,c16,c17,c18,c19,c21,c22,c23,c26,c24,c25,c20\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"30858ca2-a64b-4841-9bb5-5de4d38350cb","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","errors":null,"id":28,"input_connections":{"infile1":{"id":25,"output_name":"output"},"infile2":{"id":20,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3390.921875,"top":317.8177083333333},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants of all samples with per-position stats"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"1\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": true, \\"ignore_case\\": false, \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1b250a36-d3ae-415e-9764-006182096a60","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"Cut1","errors":null,"id":29,"input_connections":{"input":{"id":26,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3356.359375,"top":528.6770833333334},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c3,c4,c5,c6,c7,c8,c9,c11,c12,c13,c14,c15,c16,c17,c18,c19\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"65a8b0dd-4b98-4646-bee8-a89f591816fb","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":30,"input_connections":{"infile":{"id":27,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3604.359375,"top":19.333333333333343},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"All variants of all samples"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"unique\\\\\\\\(Sample\\\\\\\\)\\\\\\\\tcollapse\\\\\\\\(Sample\\\\\\\\)\\\\\\\\tcollapse\\\\\\\\(AF\\\\\\\\)\\", \\"replace_pattern\\": \\"SAMPLES(above-thresholds)\\\\\\\\tSAMPLES(all)\\\\\\\\tAFs(all)\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"6116c1c6-7c7c-43fe-9c91-609c91c6ed85","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"Cut1","errors":null,"id":31,"input_connections":{"input":{"id":28,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3608.171875,"top":344.31770833333337},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c1,c4,c5,c6,c7,c8,c9,c11,c10,c12,c13,c14,c15,c16,c17,c18,c23,c24,c25,c26,c19\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"a925b191-2cbc-4f45-b4cf-bbc25ba823d8","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":32,"input_connections":{"split_parms|input":{"id":29,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":null,"name":"Split file","outputs":[{"name":"list_output_tab","type":"input"}],"position":{"left":3579.625,"top":528.8489583333334},"post_job_actions":{"RenameDatasetActionlist_output_tab":{"action_arguments":{"newname":"Per-sample variants for plotting"},"action_type":"RenameDatasetAction","output_name":"list_output_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"top\\": \\"1\\", \\"split_by\\": {\\"select_split_by\\": \\"col\\", \\"__current_case__\\": 0, \\"id_col\\": \\"1\\", \\"match_regex\\": \\"(.*)\\", \\"sub_regex\\": \\"\\\\\\\\1\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"9f938b6a-e7fb-4509-9c15-271c475092aa","when":null,"workflow_outputs":[{"label":"variants_for_plotting","output_name":"list_output_tab","uuid":"6bed4a2c-611a-4dd7-92bd-df5f07f0b581"}]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1","errors":null,"id":33,"input_connections":{"infile":{"id":30,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3826.890625,"top":120.89583333333331},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Combined Variant Report by Variant"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"1\\", \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sortkeys\\": [{\\"__index__\\": 0, \\"column\\": \\"1\\", \\"order\\": \\"\\", \\"style\\": \\"g\\"}], \\"unique\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"0b2f751b-03c6-4302-9060-16856bb4157c","when":null,"workflow_outputs":[{"label":"by_variant_report","output_name":"outfile","uuid":"bbb19b95-70e3-43fb-baae-525f752b86e9"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1","errors":null,"id":34,"input_connections":{"infile":{"id":31,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3830.6875,"top":330.55208333333337},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Combined Variant Report by Sample"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"1\\", \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sortkeys\\": [{\\"__index__\\": 0, \\"column\\": \\"1\\", \\"order\\": \\"\\", \\"style\\": \\"\\"}, {\\"__index__\\": 1, \\"column\\": \\"2\\", \\"order\\": \\"\\", \\"style\\": \\"n\\"}], \\"unique\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"e55320de-be51-4fa5-908e-76c03de24a77","when":null,"workflow_outputs":[{"label":"combined_variant_report","output_name":"outfile","uuid":"3211bebc-9cdd-417b-b96d-d10cae6fd310"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpfreqplot/snpfreqplot/1.0+galaxy3","errors":null,"id":35,"input_connections":{"clustering|nclust":{"id":5,"output_name":"output"},"sinputs":{"id":32,"output_name":"list_output_tab"}},"inputs":[{"description":"runtime parameter for tool Variant Frequency Plot","name":"clustering"}],"label":null,"name":"Variant Frequency Plot","outputs":[{"name":"outfile","type":"pdf"}],"position":{"left":3821.671875,"top":596.9895833333334},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpfreqplot/snpfreqplot/1.0+galaxy3","tool_shed_repository":{"changeset_revision":"3d0adeee3f2b","name":"snpfreqplot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced\\": {\\"output_type\\": \\"svg\\", \\"ratio\\": \\"0.67\\", \\"color\\": \\"Set3\\"}, \\"clustering\\": {\\"do\\": \\"TRUE\\", \\"__current_case__\\": 0, \\"nclust\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"method\\": \\"ward.D2\\"}, \\"sinputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"varfreq\\": \\"0.0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0+galaxy3","type":"tool","uuid":"583b8d59-2cc0-468e-a0b7-3b1a788c7613","when":null,"workflow_outputs":[{"label":"variant_frequency_plot","output_name":"outfile","uuid":"986f1d1a-02c0-4d91-9f6e-a09a088732ee"}]}},"tags":["COVID-19","covid19.galaxyproject.org"],"uuid":"b08c744d-7c61-4b58-ac5f-4b5886c3c643"},"readme":"COVID-19: variation analysis reporting\\n--------------------------------------\\n\\nThis workflow takes VCF datasets of variants produced by any of the\\n\\"*-variant-calling\\" workflows in\\nhttps://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling\\nand generates tabular reports of variants by samples and by variant, along with\\nan overview plot of variants and their allele-frequencies across all samples.\\n","changelog":"# Changelog\\n\\n## [0.3.4] - 2024-09-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1`\\n\\n## [0.3.2] - 2023-11-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n\\n## [0.3.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.3] - 2022-10-13\\n\\n### Changed\\n\\n- Upgraded to new, more flexible version of column_maker tool.\\n\\n  This change allows a simplification of the workflow, which now uses two\\n  steps less for producing identical results.\\n\\n- Upgraded to latest version of tp_find_and_replace tool, which can handle\\n  multiple substitutions per tool run. Saves another step in the workflow.\\n\\n- Upgraded datamash to its latest tool wrapper version.\\n\\nThese changes should not have any effects on results except this one:\\n\\n- the AFcaller column of the \\"Combined Variant Report by Sample\\" could\\n  previously contain values in scientific notation for very low allele\\n  frequencies (< 0.001). In the new version all values in the column will be\\n  reported consistently in regular floating point format.\\n\\n## [0.2] - 2022-02-11\\n\\n### Changed\\n\\n- Altered the exact meaning of the user-supplied AF and DP_ALT thresholds\\n\\n  The AF threshold is now compared against (DP4[2] + DP4[3]) / DP, i.e. the\\n  unbiased AF based on all bases at the site instead of against the AF reported\\n  by the variant caller (which in the case of lofreq would consider only\\n  variant-supporting bases >= the caller\'s --min-bq in the numerator).\\n\\n  Conversely, the DP_ALT threshold is now compared against DP * AF (as reported\\n  by the variant caller), i.e. in the case of lofreq as the caller will\\n  consider only bases with >= --min-bq base quality.\\n\\n- Reported AF values in the by-sample and the by-variant reports are now\\n  unbiased AF values calculated from (DP4[2] + DP4[3]) / DP as explained above.\\n\\n  The by-sample report reports the original AF value provided by the variant\\n  caller in an additional AFcaller column.\\n\\n- Tile colors in the variant frequency summary plot are now based on unbiased\\n  AF values.\\n\\n- An extra step removing potential called variants with a caller-reported\\n  DP of 0 has been added to avoid division by zero errors in the calculation\\n  of unbiased AF values.\\n\\n### Fixed\\n\\n- Reverted the autoupdated wrapper version change for the snpsift tools\\n\\n  The autoupdate script had problems sorting their non-standard version strings\\n  correctly and downgraded these wrappers instead of upgrading them\\n\\n- Made sure only a single changeset revision of the text_processing tools is\\n  used in the workflow.\\n\\n## [0.1.3] - 2022-02-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3.0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3.0`\\n- `toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2` was updated to `toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.5` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6`\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: variation analysis reporting\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: variation analysis reporting\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Variation data to report\\"]@{ shape: docs }\\n1[\\"ℹ️ AF Filter\\"]@{ shape: lean-l }\\n2[\\"ℹ️ DP Filter\\"]@{ shape: lean-l }\\n3[\\"ℹ️ DP_ALT Filter\\"]@{ shape: lean-l }\\n4[\\"ℹ️ gene products translations\\"]@{ shape: doc }\\n5[\\"ℹ️ Number of Clusters\\"]@{ shape: lean-l }\\n6[\\"SnpSift Filter\\"]@{ shape: process }\\n0 --> 6\\n7[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 7\\n2 --> 7\\n3 --> 7\\n8[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 8\\n2 --> 8\\n3 --> 8\\n9[\\"SnpSift Filter\\"]@{ shape: process }\\n7 --> 9\\n8 --> 9\\n6 --> 9\\n10[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n9 --> 10\\n11[\\"Replace column\\"]@{ shape: process }\\n10 --> 11\\n4 --> 11\\n12[\\"Compute\\"]@{ shape: process }\\n11 --> 12\\n13[\\"Datamash\\"]@{ shape: process }\\n12 --> 13\\n14[\\"Replace\\"]@{ shape: process }\\n13 --> 14\\n15[\\"Collapse Collection\\"]@{ shape: process }\\n14 --> 15\\n16[\\"Compute\\"]@{ shape: process }\\n15 --> 16\\n17[\\"Replace\\"]@{ shape: process }\\n16 --> 17\\n18[\\"Datamash\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Filter\\"]@{ shape: process }\\n17 --> 19\\n20[\\"Datamash\\"]@{ shape: process }\\n17 --> 20\\n21[\\"Join\\"]@{ shape: process }\\n19 --> 21\\n18 --> 21\\n22[\\"Datamash\\"]@{ shape: process }\\n19 --> 22\\n23[\\"Datamash\\"]@{ shape: process }\\n19 --> 23\\n24[\\"Datamash\\"]@{ shape: process }\\n21 --> 24\\n25[\\"Join\\"]@{ shape: process }\\n17 --> 25\\n22 --> 25\\n26[\\"Join\\"]@{ shape: process }\\n16 --> 26\\n23 --> 26\\n27[\\"Cut\\"]@{ shape: process }\\n24 --> 27\\n28[\\"Join\\"]@{ shape: process }\\n25 --> 28\\n20 --> 28\\n29[\\"Cut\\"]@{ shape: process }\\n26 --> 29\\n30[\\"Replace\\"]@{ shape: process }\\n27 --> 30\\n31[\\"Cut\\"]@{ shape: process }\\n28 --> 31\\n32[\\"Split file\\"]@{ shape: process }\\n29 --> 32\\n33[\\"Sort\\"]@{ shape: process }\\n30 --> 33\\n34[\\"Sort\\"]@{ shape: process }\\n31 --> 34\\n35[\\"Variant Frequency Plot\\"]@{ shape: process }\\n5 --> 35\\n32 --> 35\\n```\\n","updated":"2024-09-24T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-variation-reporting/COVID-19-VARIATION-REPORTING","dockstore_id":16304,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078519","tests":[{"doc":"Test generating variant report","job":{"gene products translations":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_feature_mapping.tsv?download=1"},"Variation data to report":{"class":"Collection","collection_type":"list","elements":[{"identifier":"artic","class":"File","path":"test-data/ont-artic.vcf"},{"identifier":"illumina-wgs-se","class":"File","path":"test-data/se-illumina-wgs.vcf"},{"identifier":"illumina-wgs-pe","class":"File","path":"test-data/pe-illumina-wgs.vcf"},{"identifier":"pe-artic","class":"File","path":"test-data/pe-artic.vcf"}]}},"outputs":{"by_variant_report":{"file":"test-data/by_variant_report.tsv"},"combined_variant_report":{"file":"test-data/combined_variant_report.tsv"}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-variation-reporting"},{"version":1.2,"workflows":[{"name":"SARS-COV-2-ILLUMINA-AMPLICON-IVAR-PANGOLIN-NEXTCLADE","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/pe-wgs-ivar-analysis.ga","authors":[{"name":"Peter van Heusden","orcid":"0000-0001-6553-5274"}],"definition":{"a_galaxy_workflow":"true","annotation":"Find and annotate variants in ampliconic SARS-CoV-2 Illumina sequencing data and classify samples with pangolin and nextclade","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0001-6553-5274","name":"Peter van Heusden"}],"release":"0.2.3","format-version":"0.1","license":"MIT","name":"SARS-CoV-2 Illumina Amplicon pipeline - iVar based","steps":{"0":{"annotation":"FASTQ format Illumina Reads (Amplicon Protocol)","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"FASTQ format Illumina Reads (Amplicon Protocol)","name":"Paired read collection for samples"}],"label":"Paired read collection for samples","name":"Input dataset collection","outputs":[],"position":{"left":2,"top":33.99998474121094},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"ce39f6b7-b9f6-4431-8831-7a284fe826a7","when":null,"workflow_outputs":[]},"1":{"annotation":"SARS-CoV-2 reference genome (typically MN908947.3)","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"SARS-CoV-2 reference genome (typically MN908947.3)","name":"Reference FASTA"}],"label":"Reference FASTA","name":"Input dataset","outputs":[],"position":{"left":0,"top":279.99998474121094},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"25167e70-195c-4cda-9219-dbfe2f70c863","when":null,"workflow_outputs":[]},"2":{"annotation":"Primer BED file (from ARTIC project or similar)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Primer BED file (from ARTIC project or similar)","name":"Primer BED"}],"label":"Primer BED","name":"Input dataset","outputs":[],"position":{"left":6,"top":391.49998474121094},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"6c8951a5-f254-4564-8d17-f78e018077ec","when":null,"workflow_outputs":[]},"3":{"annotation":"Specify the proportion of reads that need to agree with each other to call a variant. This is a floating point value between 0 and 1. ","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Specify the proportion of reads that need to agree with each other to call a variant. This is a floating point value between 0 and 1. ","name":"Read fraction to call variant"}],"label":"Read fraction to call variant","name":"Input parameter","outputs":[],"position":{"left":34.35003662109375,"top":635.9999847412109},"tool_id":null,"tool_state":"{\\"default\\": 0.7, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"57a3622f-6908-4c1d-b88d-3bc318ac9d73","when":null,"workflow_outputs":[]},"4":{"annotation":"Minimum base quality score to count a base towards the sequence consensus.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Minimum base quality score to count a base towards the sequence consensus.","name":"Minimum quality score to call base"}],"label":"Minimum quality score to call base","name":"Input parameter","outputs":[],"position":{"left":29.4000244140625,"top":802.0000457763672},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"00fdc318-b4b7-404e-900b-54087a891f55","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","errors":null,"id":5,"input_connections":{"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[],"label":"fastp: Trimmed Illumina Reads","name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":280.00001525878906,"top":0},"post_job_actions":{"HideDatasetActionoutput_paired_coll":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_paired_coll"},"HideDatasetActionreport_html":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_html"},"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","tool_shed_repository":{"changeset_revision":"65b93b623c77","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.2+galaxy0","type":"tool","uuid":"fb6601de-99e2-4271-813d-0e97027e54c7","when":null,"workflow_outputs":[]},"6":{"annotation":"If the reference is named MN908947.3 (Genbank name of SARS-CoV-2 reference genome), rename it to NC_045512.2 (RefSeq name of SARS-CoV-2 reference genome)","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","errors":null,"id":6,"input_connections":{"infile":{"id":1,"output_name":"output"}},"inputs":[],"label":"Rename reference to NC_045512.2","name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":270.00001525878906,"top":274.49998474121094},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","tool_shed_repository":{"changeset_revision":"ddf54b12c295","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"/^>/s/^>MN908947.3/>NC_045512.2/\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"5828628c-ef20-428d-b76e-8d502a8d3097","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","errors":null,"id":7,"input_connections":{"fastq_input|fastq_input1":{"id":5,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":558.0000152587891,"top":95.99998474121094},"post_job_actions":{"HideDatasetActionbam_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bam_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","tool_shed_repository":{"changeset_revision":"e188dc7a68e6","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.17.2","type":"tool","uuid":"11cea040-0e8a-4b53-8d16-7bd55f9e1074","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.4","errors":null,"id":8,"input_connections":{"input":{"id":7,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":836,"top":95.99998474121094},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.4","tool_shed_repository":{"changeset_revision":"3a0efe14891f","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_group\\": null, \\"read_length\\": null, \\"remove_dups\\": false, \\"remove_overlaps\\": false, \\"sparse\\": false, \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.4","type":"tool","uuid":"217c653d-84a7-49dd-ba0f-601bb651ee7d","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0","errors":null,"id":9,"input_connections":{"input":{"id":7,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":839.3500366210938,"top":295.6666717529297},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0","tool_shed_repository":{"changeset_revision":"5826298f6a73","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy0","type":"tool","uuid":"1972d86f-45de-4b7b-aca2-e53ccc240fb8","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","errors":null,"id":10,"input_connections":{"input1":{"id":9,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"QualiMap BamQC","outputs":[{"name":"raw_data","type":"input"},{"name":"output_html","type":"html"}],"position":{"left":1114,"top":172.99998474121094},"post_job_actions":{"HideDatasetActionoutput_html":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_html"},"HideDatasetActionraw_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"raw_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","tool_shed_repository":{"changeset_revision":"19ece8afbaab","name":"qualimap_bamqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"duplicate_skipping\\": [\\"0\\"], \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"per_base_coverage\\": false, \\"plot_specific\\": {\\"n_bins\\": \\"400\\", \\"paint_chromosome_limits\\": true, \\"genome_gc_distr\\": null, \\"homopolymer_size\\": \\"3\\"}, \\"stats_regions\\": {\\"region_select\\": \\"all\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.2d+galaxy3","type":"tool","uuid":"a0720c38-9db1-494e-8976-8baa83f69613","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0","errors":null,"id":11,"input_connections":{"input_bam":{"id":9,"output_name":"outputsam"},"primer|input_bed":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"ivar trim","outputs":[{"name":"output_bam","type":"bam"}],"position":{"left":1114,"top":415.99998474121094},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0","tool_shed_repository":{"changeset_revision":"0893a1dbb807","name":"ivar_trim","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"amplicons\\": {\\"filter_by\\": \\"yes_compute\\", \\"__current_case__\\": 0}, \\"inc_primers\\": true, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_len\\": \\"30\\", \\"min_qual\\": \\"20\\", \\"primer\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 0, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"primer_pos_wiggle\\": \\"0\\", \\"trimmed_length\\": {\\"filter\\": \\"auto\\", \\"__current_case__\\": 1}, \\"window_width\\": \\"4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.2+galaxy0","type":"tool","uuid":"8841f373-8ce6-4aba-aefa-70acd5daff73","when":null,"workflow_outputs":[{"label":"primer_trimmed_bam","output_name":"output_bam","uuid":"ebdd8f89-6559-44a9-9616-c57b78e7a915"}]},"12":{"annotation":"","content_id":"__FLATTEN__","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"raw_data"}},"inputs":[],"label":null,"name":"Flatten collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1392,"top":218.99998474121094},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FLATTEN__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"join_identifier\\": \\"_\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"f2fe26ec-f30b-4bc7-9dd6-62f3c9538ae7","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_variants/ivar_variants/1.4.2+galaxy1","errors":null,"id":13,"input_connections":{"input_bam":{"id":11,"output_name":"output_bam"},"min_freq":{"id":3,"output_name":"output"},"min_qual":{"id":4,"output_name":"output"},"ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ivar variants","name":"output_format"}],"label":null,"name":"ivar variants","outputs":[{"name":"output_variants_tabular","type":"tabular"},{"name":"output_variants_vcf","type":"vcf"}],"position":{"left":1414,"top":448.49998474121094},"post_job_actions":{"HideDatasetActionoutput_variants_vcf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_variants_vcf"},"RenameDatasetActionoutput_variants_tabular":{"action_arguments":{"newname":"ivar variants tabular output on #{input}"},"action_type":"RenameDatasetAction","output_name":"output_variants_tabular"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_variants/ivar_variants/1.4.2+galaxy1","tool_shed_repository":{"changeset_revision":"045d6d00f606","name":"ivar_variants","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_freq\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_qual\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_format\\": {\\"choice\\": \\"tabular_and_vcf\\", \\"__current_case__\\": 1, \\"pass_only\\": false, \\"gtf\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.2+galaxy1","type":"tool","uuid":"0b2aa4bf-69dd-4ccc-b0a2-4a734848e2b8","when":null,"workflow_outputs":[{"label":"ivar_variants_tabular","output_name":"output_variants_tabular","uuid":"319f50e1-71fb-445e-b3e3-8015d2787f5f"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0","errors":null,"id":14,"input_connections":{"input_bam":{"id":11,"output_name":"output_bam"},"min_freq":{"id":3,"output_name":"output"},"min_qual":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"ivar consensus","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":1392,"top":780.9999847412109},"post_job_actions":{"HideDatasetActionconsensus":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0","tool_shed_repository":{"changeset_revision":"17f911830a8c","name":"ivar_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"depth_action\\": \\"-n N\\", \\"filter_depth\\": \\"false\\", \\"gap\\": \\"true\\", \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_depth\\": \\"50\\", \\"min_freq\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_indel_freq\\": \\"0.8\\", \\"min_qual\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.2+galaxy0","type":"tool","uuid":"5fb477e6-e7ac-4e02-a31f-335228becdf3","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","errors":null,"id":15,"input_connections":{"results_0|software_cond|input":{"id":5,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":8,"output_name":"output"},"results_2|software_cond|input":{"id":12,"output_name":"output"}},"inputs":[],"label":"Quality Control Report","name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"left":1670,"top":97.99998474121094},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Quality Control report"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","tool_shed_repository":{"changeset_revision":"abfd8a6544d7","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"qualimap\\", \\"__current_case__\\": 20, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": false, \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy1","type":"tool","uuid":"d03312e1-3b85-4883-b601-6081acf7503d","when":null,"workflow_outputs":[{"label":"bamqc_report_html","output_name":"html_report","uuid":"0489d543-59ec-4f03-be4c-d13b14b03ac7"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","errors":null,"id":16,"input_connections":{"input":{"id":13,"output_name":"output_variants_vcf"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":"Annotated variants","name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"left":2174,"top":447.99998474121094},"post_job_actions":{"HideDatasetActionstatsFile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"statsFile"},"RenameDatasetActionsnpeff_output":{"action_arguments":{"newname":"Annotated variants"},"action_type":"RenameDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","tool_shed_repository":{"changeset_revision":"2a3a00c1fa0a","name":"snpeff_sars_cov_2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": null, \\"chr\\": \\"\\", \\"csvStats\\": false, \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": null, \\"generate_stats\\": true, \\"genome_version\\": \\"NC_045512.2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": true, \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5covid19","type":"tool","uuid":"a8b63b67-2795-4e21-8614-f7d24cec2224","when":null,"workflow_outputs":[{"label":"snpeff_annotated_vcf","output_name":"snpeff_output","uuid":"8f6057ec-4a34-4c49-ad4b-f0d93ba33878"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","errors":null,"id":17,"input_connections":{"infile":{"id":14,"output_name":"consensus"}},"inputs":[],"label":"Consensus genome (masked for depth)","name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":1670,"top":802.0000457763672},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Consensus genome"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","tool_shed_repository":{"changeset_revision":"ddf54b12c295","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"/^>/s/Consensus_(.*)_threshold_.*/\\\\\\\\1/\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"877a609d-d2fa-430d-8b78-2b484af52bda","when":null,"workflow_outputs":[{"label":"ivar_consensus_genome","output_name":"output","uuid":"cabe916a-b8ba-49e1-af9d-30dad1bd44ae"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","errors":null,"id":18,"input_connections":{"inputs":{"id":17,"output_name":"output"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1913,"top":913.5000457763672},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Combined Consensus Genomes"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"87e8ec0a-d8e8-48f3-b943-04c75152fec2","when":null,"workflow_outputs":[{"label":"combined_multifasta","output_name":"out_file1","uuid":"d58378c8-fe46-4cab-9317-5991a3561221"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pangolin/pangolin/4.3+galaxy1","errors":null,"id":19,"input_connections":{"input1":{"id":18,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Pangolin","outputs":[{"name":"output1","type":"tabular"}],"position":{"left":2177,"top":830.5000457763672},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pangolin/pangolin/4.3+galaxy1","tool_shed_repository":{"changeset_revision":"2f1019071464","name":"pangolin","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment\\": false, \\"constellations\\": {\\"source\\": \\"default\\", \\"__current_case__\\": 0}, \\"db\\": {\\"__current_case__\\": 0, \\"source\\": \\"download\\"}, \\"engine\\": {\\"analysis_mode\\": \\"usher\\", \\"__current_case__\\": 0, \\"pangolin_data\\": {\\"source\\": \\"default\\", \\"__current_case__\\": 0}}, \\"expanded_lineage\\": false, \\"include_header\\": true, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_ambig\\": \\"0.5\\", \\"min_length\\": \\"10000\\", \\"usher\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+galaxy1","type":"tool","uuid":"15ff9812-4ff9-4ac2-9f95-c65bab4eb070","when":null,"workflow_outputs":[{"label":"all_samples_pangolin","output_name":"output1","uuid":"a0468d09-08a4-43de-be10-4dc4e6973b99"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/nextclade/nextclade/2.7.0+galaxy0","errors":null,"id":20,"input_connections":{"input_fasta":{"id":18,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Nextclade","name":"input_fasta"}],"label":null,"name":"Nextclade","outputs":[{"name":"report_tsv","type":"tabular"}],"position":{"left":2176,"top":999.0000457763672},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/nextclade/nextclade/2.7.0+galaxy0","tool_shed_repository":{"changeset_revision":"128ba8da994f","name":"nextclade","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"advanced_options\\": \\"no\\", \\"__current_case__\\": 1}, \\"db\\": {\\"source\\": \\"download\\", \\"__current_case__\\": 1}, \\"include_header\\": true, \\"input_fasta\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"organism\\": \\"sars-cov-2\\", \\"outputs\\": [\\"report_tsv\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.0+galaxy0","type":"tool","uuid":"4b821f22-47f8-4251-bc32-5aa167498ba7","when":null,"workflow_outputs":[{"label":"all_samples_nextclade","output_name":"report_tsv","uuid":"4cd1f6bd-be94-43c3-9d4a-773f178b96c4"}]}},"tags":["COVID-19","ARTIC","iwc"],"uuid":"3d86e5fc-cad1-4e21-8ecc-d0e6637899bc","version":1},"readme":"# COVID-19 sequence analysis on Illumina Amplicon PE data\\n\\nThis workflow implements an [iVar](https://github.com/andersen-lab/ivar) based analysis similar to\\nthe one in [ncov2019-artic-nf](https://github.com/connor-lab/ncov2019-artic-nf), [covid-19-signal](https://github.com/jaleezyy/covid-19-signal/) and the Thiagen [Titan workflow](https://github.com/theiagen/public_health_viral_genomics). These workflows (written in  Nextflow, Snakemake and WDL) are widely in use in [COG UK](https://www.cogconsortium.uk/), [CanCOGeN](https://www.genomecanada.ca/en/cancogen) and some US state public health laboratories.\\n\\nThis workflow is also the subject of a Galaxy Training Network tutorial (currently a [Work in Progress](https://github.com/galaxyproject/training-material/pull/2633)).\\nIt differs from [this workflow](https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-variant-calling) in\\nthat it does not use `lofreq` and is aimed at rapid analysis of majority variants and lineage/clade assignment with `pangolin` and `nextclade`.\\n\\nTODO:\\n\\n1. Add support for QC using negative and positive controls\\n2. Integrate with phylogeny tools including IQTree and UShER (and possibly more).\\n","changelog":"# Changelog\\n\\n## [0.3] - 2022-11-22\\n\\n- update all tools:\\n  - fastp from 0.20.1 to 0.23.2\\n  - bwa_mem from 0.7.17.1 to 0.7.17.2\\n  - samtools_stats from 2.0.2 to 2.0.4\\n  - samtools_view from 1.9 to 1.15.1\\n  - ivar_trim from 1.3.1 to 1.4.2\\n  - ivar_variants from 1.3.1 to 1.4.2\\n  - ivar_consensus from 1.3.1 to 1.4.2\\n  - multiqc from 1.9 to 1.11\\n  - pangolin from 3.1.14 to 4.3\\n  - nextclade from 1.4.1 to 2.7.0\\n\\n## [0.2.3] - 2022-05-25\\n\\n### Changed\\n- Changed creator ORCID to absolute URI\\n\\n## [0.2.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.2.1] - 2021-11-04\\n\\n### Added\\n- Added .workflowhub.yml\\n\\n## [0.2] - 2021-10-13\\n\\n### Changed\\n\\n- Upgrade pangolin to 3.1.14 \\n  - this is a bugfix release that addresses problems with commas in sequence IDs\\n- Upgrade nextclade to 1.4.1\\n  - nextclade has a new way of dealing with the nextclade database, so it is now downloaded using nextclade itself\\n  - the latest release supports detecting and reporting frameshifts\\n  - version 1.4.1 fixes a problem with formatting columns that was present in previous versions\\n- A header line is now part of the tabular reports produced by both pangolin and nextclade\\n- The aligned primer-trimmed reads BAM and the multi-fasta combined consensus genomes datasets are no longer hidden as part of the workflow execution\\n\\n## [0.1] - 2021-06-20\\n\\n- Initial version of SARS-CoV-2 Illumina Amplicon pipeline - iVar based for IWC\\n","diagrams":"# Workflow diagrams\\n\\n## SARS-CoV-2 Illumina Amplicon pipeline - iVar based\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired read collection for samples\\"]@{ shape: docs }\\n1[\\"ℹ️ Reference FASTA\\"]@{ shape: doc }\\n2[\\"ℹ️ Primer BED\\"]@{ shape: doc }\\n3[\\"ℹ️ Read fraction to call variant\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Minimum quality score to call base\\"]@{ shape: lean-l }\\n5[\\"fastp: Trimmed Illumina Reads\\"]@{ shape: process }\\n0 --> 5\\n6[\\"Rename reference to NC_045512.2\\"]@{ shape: process }\\n1 --> 6\\n7[\\"Map with BWA-MEM\\"]@{ shape: process }\\n5 --> 7\\n6 --> 7\\n8[\\"Samtools stats\\"]@{ shape: process }\\n7 --> 8\\n9[\\"Samtools view\\"]@{ shape: process }\\n7 --> 9\\n10[\\"QualiMap BamQC\\"]@{ shape: process }\\n9 --> 10\\n11[\\"ivar trim\\"]@{ shape: process }\\n9 --> 11\\n2 --> 11\\n12[\\"Flatten collection\\"]@{ shape: process }\\n10 --> 12\\n13[\\"ivar variants\\"]@{ shape: process }\\n11 --> 13\\n3 --> 13\\n4 --> 13\\n1 --> 13\\n14[\\"ivar consensus\\"]@{ shape: process }\\n11 --> 14\\n3 --> 14\\n4 --> 14\\n15[\\"Quality Control Report\\"]@{ shape: process }\\n5 --> 15\\n8 --> 15\\n12 --> 15\\n16[\\"Annotated variants\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Consensus genome (masked for depth)\\"]@{ shape: process }\\n14 --> 17\\n18[\\"Concatenate datasets\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Pangolin\\"]@{ shape: process }\\n18 --> 19\\n20[\\"Nextclade\\"]@{ shape: process }\\n18 --> 20\\n```\\n","updated":"2022-11-22T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-ivar-analysis/SARS-COV-2-ILLUMINA-AMPLICON-IVAR-PANGOLIN-NEXTCLADE","dockstore_id":16951,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078785"}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-ivar-analysis"},{"version":1.2,"workflows":[{"name":"COVID-19-CONSENSUS-CONSTRUCTION","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/consensus-from-variation.ga","testParameterFiles":["/consensus-from-variation-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"Build a consensus sequence from FILTER PASS variants with intrasample allele-frequency above a configurable consensus threshold.\\nHard-mask regions with low coverage (but not consensus variants within them) and ambiguous sites.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: consensus construction","release":"0.4.3","steps":{"0":{"annotation":"Collection of VCFs produced by upstream workflows for variation analysis","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection of VCFs produced by upstream workflows for variation analysis","name":"Variant calls"}],"label":"Variant calls","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":306.13330078125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fca4e710-87a7-4cb6-9ff1-f9a8dc84ca37","when":null,"workflow_outputs":[]},"1":{"annotation":"Only variant calls with an allele-frequency greater this value will be considered consensus variants.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Only variant calls with an allele-frequency greater this value will be considered consensus variants.","name":"min-AF for consensus variant"}],"label":"min-AF for consensus variant","name":"Input parameter","outputs":[],"position":{"left":0.45001220703125,"top":469},"tool_id":null,"tool_state":"{\\"default\\": 0.75, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"52664bd7-b500-40a1-935b-8ac6df7003e5","when":null,"workflow_outputs":[{"label":"consensus_af_threshold","output_name":"output","uuid":"0d802d31-bb4c-41e3-8e84-c1d99b914ee9"}]},"2":{"annotation":"Variant calls with an allele frequency higher than this value, but lower than the AF threshold for consensus variants will be considered questionable and the respective sites be masked (with Ns) in the consensus sequence.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Variant calls with an allele frequency higher than this value, but lower than the AF threshold for consensus variants will be considered questionable and the respective sites be masked (with Ns) in the consensus sequence.","name":"min-AF for failed variants"}],"label":"min-AF for failed variants","name":"Input parameter","outputs":[],"position":{"left":0.45001220703125,"top":640.6500244140625},"tool_id":null,"tool_state":"{\\"default\\": 0.25, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"a2a15cce-94ec-4a76-a3ff-3ea898e9c678","when":null,"workflow_outputs":[{"label":"non_consensus_af_threshold","output_name":"output","uuid":"16c62493-ad4b-46da-8790-a8094d01a130"}]},"3":{"annotation":"Fully processed BAMs as generated by upstream workflows for variation analysis. Note: for ARTIC data, these BAMs should NOT have undergone processing with ivar removereads.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Fully processed BAMs as generated by upstream workflows for variation analysis. Note: for ARTIC data, these BAMs should NOT have undergone processing with ivar removereads.","name":"aligned reads data for depth calculation"}],"label":"aligned reads data for depth calculation","name":"Input dataset collection","outputs":[],"position":{"left":581.3167114257812,"top":395.16668701171875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"23ae69ea-f9a7-4d5f-85f5-63b9d2bd2c50","when":null,"workflow_outputs":[]},"4":{"annotation":"Sites in the viral genome covered by less than this number of reads are considered questionable and will be masked (with Ns) in the consensus sequence independent of whether a variant has been called at them or not.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Sites in the viral genome covered by less than this number of reads are considered questionable and will be masked (with Ns) in the consensus sequence independent of whether a variant has been called at them or not.","name":"Depth-threshold for masking"}],"label":"Depth-threshold for masking","name":"Input parameter","outputs":[],"position":{"left":585.25,"top":557.8499755859375},"tool_id":null,"tool_state":"{\\"default\\": 5, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"cfdc65b0-8393-4ee3-9e27-16b23135fbec","when":null,"workflow_outputs":[{"label":"depth_threshold","output_name":"output","uuid":"f677edfd-5faf-44af-8e20-c5e6d72ba24a"}]},"5":{"annotation":"The SARS-CoV-2 reference genome","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"The SARS-CoV-2 reference genome","name":"Reference genome"}],"label":"Reference genome","name":"Input dataset","outputs":[],"position":{"left":2283.8167114257812,"top":346.1500244140625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"4a2c6df2-fb1e-42d8-96d5-6765ca85c214","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":282.79998779296875,"top":285.3500061035156},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( ( FILTER = \'PASS\' ) | ( na FILTER ) ) & ( ( DP4[2] + DP4[3] ) >= ( \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"8ec54933-36d6-404c-b0a6-5a14b04eb30e","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":7,"input_connections":{"components_1|param_type|component_value":{"id":2,"output_name":"output"},"components_3|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":233,"top":802.0166625976562},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( ( DP4[2] + DP4[3] ) > ( \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) ) & ( ( DP4[2] + DP4[3] ) < ( \\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) ) & ( ( FILTER = \'PASS\' ) | ( na FILTER ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"51ac3588-c801-4551-8345-b04cc3571fea","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","errors":null,"id":8,"input_connections":{"input_type|input":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bedtools Genome Coverage","name":"input_type"}],"label":null,"name":"bedtools Genome Coverage","outputs":[{"name":"output","type":"bedgraph"}],"position":{"left":880.7999877929688,"top":403.3499755859375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Depth of coverage"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"d\\": false, \\"dz\\": false, \\"five\\": false, \\"input_type\\": {\\"input_type_select\\": \\"bam\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"report\\": {\\"report_select\\": \\"bg\\", \\"__current_case__\\": 0, \\"zero_regions\\": true, \\"scale\\": \\"1.0\\"}, \\"split\\": true, \\"strand\\": \\"\\", \\"three\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"d89684f5-5a49-4b4c-821b-6c8a87f3a46e","when":null,"workflow_outputs":[{"label":"coverage_depth","output_name":"output","uuid":"df923e92-8241-4c8b-9983-4773b753a144"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":9,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":879.3167114257812,"top":559.8499755859375},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c4 < \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"089b2659-d4c8-44c8-9ecc-ffe71676e650","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":10,"input_connections":{"filter_expression|expr":{"id":6,"output_name":"out1"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":485.2833251953125,"top":0},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Consensus variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"3673b687-1ba0-42f2-ba13-fa3130462079","when":null,"workflow_outputs":[{"label":"consensus_variants","output_name":"output","uuid":"1db759fb-634a-4b5f-abd9-a0579a003570"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":11,"input_connections":{"filter_expression|expr":{"id":7,"output_name":"out1"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":456.66668701171875,"top":733.3499755859375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filter-failed variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"fb232cba-e3b2-43f9-bb1f-a9ebc69c9b97","when":null,"workflow_outputs":[{"label":"filter_failed_variants","output_name":"output","uuid":"73373a44-4651-49c2-813d-e4ea0ffd46cf"}]},"12":{"annotation":"","content_id":"Filter1","errors":null,"id":12,"input_connections":{"cond":{"id":9,"output_name":"out1"},"input":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1213.8333129882812,"top":427.3499755859375},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Low-coverage regions"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"8ac27ee3-9e9e-4d6f-b424-2c21f80c87ef","when":null,"workflow_outputs":[{"label":"low_cov_regions","output_name":"out_file1","uuid":"20b0e19e-6916-4709-b8db-8c828ccde324"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":13,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":807.4500122070312,"top":183},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Called variants - CHROM_POS_REF"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\"\\", \\"extract\\": \\"CHROM POS REF\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": false, \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"5aca2952-a87c-4c77-a259-a22daceab557","when":null,"workflow_outputs":[{"label":"chrom_pos_ref_called_variants","output_name":"output","uuid":"555dff08-7ce6-4d48-b00f-2c4143f83489"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":14,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":673.9500122070312,"top":899},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filter-failed variants - CHROM_POS_REF"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\"\\", \\"extract\\": \\"CHROM POS REF\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": false, \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"d99d1a1a-24a7-4379-a7d1-1980303ee08a","when":null,"workflow_outputs":[{"label":"chrom_pos_ref_failed_variants","output_name":"output","uuid":"13574d4b-8c55-4c2f-a5da-76b82ed31deb"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":15,"input_connections":{"input":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1148.1333618164062,"top":249.64999389648438},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Called variant sites"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": false, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--skip-non-computable\\", \\"__current_case__\\": 1}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"int(c2) - (len(c3) == 1)\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"2\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"int(c2) + ((len(c3) - 1) or 1)\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"3\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"fd260e32-723d-4a22-8d64-4795557f8159","when":null,"workflow_outputs":[{"label":"called_variant_sites","output_name":"out_file1","uuid":"0033def6-7531-40b6-8b01-d16a18726fe4"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":16,"input_connections":{"input":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":901.6166381835938,"top":919.1500244140625},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Filter-failed variant sites"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": false, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--skip-non-computable\\", \\"__current_case__\\": 1}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"int(c2) - (len(c3) == 1)\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"2\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"int(c2) + ((len(c3) - 1) or 1)\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"3\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"116e8e99-53ca-4366-a7e1-e3a9fb0bcfc3","when":null,"workflow_outputs":[{"label":"failed_variant_sites","output_name":"out_file1","uuid":"8ba74008-5f23-4b79-993b-ae8ddac44456"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":17,"input_connections":{"input1":{"id":12,"output_name":"out_file1"},"input2":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1468.8833618164062,"top":537.4166870117188},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Cocatenated low-coverage regions and filter-failed sites"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"2b7570fb-9ce5-4a9a-863c-e33efbc88912","when":null,"workflow_outputs":[{"label":"low_cov_regions_plus_filter_failed","output_name":"output","uuid":"aff4b045-799f-4337-a8e6-af87be35bda3"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/merge/gops_merge_1/1.0.0","errors":null,"id":18,"input_connections":{"input1":{"id":17,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge","outputs":[{"name":"output","type":"input"}],"position":{"left":1703.3167114257812,"top":533.6666870117188},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Combined low-coverage regions and filter-failed sites"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/merge/gops_merge_1/1.0.0","tool_shed_repository":{"changeset_revision":"debffd27642d","name":"merge","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"returntype\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3e63167b-6560-4b6d-a409-7d3707039222","when":null,"workflow_outputs":[{"label":"low_cov_regions_plus_filter_failed_combined","output_name":"output","uuid":"8db1e8f6-9f76-4fd0-b6c9-10eeada9f8c4"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/subtract/gops_subtract_1/1.0.0","errors":null,"id":19,"input_connections":{"input1":{"id":18,"output_name":"output"},"input2":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Subtract","outputs":[{"name":"output","type":"input"}],"position":{"left":1959.4666137695312,"top":408},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Masking regions"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/subtract/gops_subtract_1/1.0.0","tool_shed_repository":{"changeset_revision":"0427ca314f3d","name":"subtract","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min\\": \\"1\\", \\"returntype\\": \\"-p\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"51b26960-4ee7-4a3d-96cb-a6b3950c173e","when":null,"workflow_outputs":[{"label":"masking_regions","output_name":"output","uuid":"1bdf0a04-75d5-4b7b-bd52-880a869548a1"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":20,"input_connections":{"input":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2144.8333129882812,"top":585.6666870117188},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"interval"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"1-based masking regions"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"int(c2) + 1\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"2\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"e06fe045-f0f0-4c32-80ad-a8e070946a2f","when":null,"workflow_outputs":[{"label":"1_based_masking_regions","output_name":"out_file1","uuid":"dc85cc18-db50-4c90-97d7-58eb5dcad70b"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4","errors":null,"id":21,"input_connections":{"input_file":{"id":10,"output_name":"output"},"reference_source|fasta_ref":{"id":5,"output_name":"output"},"sec_default|mask":{"id":20,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bcftools consensus","name":"reference_source"},{"description":"runtime parameter for tool bcftools consensus","name":"sec_default"}],"label":null,"name":"bcftools consensus","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":2603.8333129882812,"top":243.66668701171875},"post_job_actions":{"RenameDatasetActionoutput_file":{"action_arguments":{"newname":"Consensus sequence with masking"},"action_type":"RenameDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4","tool_shed_repository":{"changeset_revision":"8aa352093d73","name":"bcftools_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"absent\\": \\"\\", \\"chain\\": false, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"fasta_ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rename\\": true, \\"sec_default\\": {\\"mask\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iupac_codes\\": false, \\"sample\\": \\"\\", \\"select_haplotype\\": null, \\"mark_del\\": \\"\\", \\"mark_ins\\": null, \\"mark_snv\\": null, \\"conditional_mask\\": {\\"selector\\": \\"disabled\\", \\"__current_case__\\": 0}}, \\"sec_restrict\\": {\\"include\\": \\"\\", \\"exclude\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy4","type":"tool","uuid":"698a9a8c-f6fe-430b-a7e0-cf9e1d058d02","when":null,"workflow_outputs":[{"label":"consensus","output_name":"output_file","uuid":"5e1e40be-fab7-4592-82b4-a02536f618ef"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":22,"input_connections":{"input_list":{"id":21,"output_name":"output_file"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2863.5000610351562,"top":267.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Multisample consensus FASTA"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"acae0f3e-448b-483e-a44a-3e09ce9b3e77","when":null,"workflow_outputs":[{"label":"multisample_consensus_fasta","output_name":"output","uuid":"105f13c3-9c94-4924-b950-9526d680562f"}]}},"tags":["COVID-19","covid19.galaxyproject.org"],"uuid":"06dc40a7-99f2-4b3c-ae21-b3dcb239306f"},"readme":"COVID-19: consensus construction\\n--------------------------------\\n\\nThis workflow aims at generating reliable consensus sequences from variant\\ncalls according to transparent criteria that capture at least some of the\\ncomplexity of variant calling.\\n\\nIt takes a collection of VCFs (with DP and DP4 INFO fields) and a collection of\\nthe corresponding aligned reads (for the purpose of calculating genome-wide\\ncoverage) such as produced by any of the variant calling workflows in\\nhttps://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling\\nand generates a collection of viral consensus sequences and a multisample FASTA\\nof all these sequences.\\n\\nEach consensus sequence is guaranteed to capture all called, filter-passing (as\\nper the FILTER column of the VCF input) variants found in the VCF of its sample\\nthat reach a user-defined consensus allele frequency threshold.\\n\\nFilter-failing variants and variants below a second user-defined minimal\\nallele frequency threshold will be ignored.\\n\\nGenomic positions of filter-passing variants with an allele frequency in\\nbetween the two thresholds will be hard-masked (with N) in the consensus\\nsequence of their sample.\\n\\nGenomic positions with a coverage (calculated from the read alignments input)\\nbelow another user-defined threshold will be hard-masked, too, unless they are\\nconsensus variant sites.\\n","changelog":"# Changelog\\n\\n## [0.4.3] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4`\\n\\n## [0.4.2] - 2024-03-06\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.29.2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.30.0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3`\\n\\n## [0.4.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.4] - 2022-10-21\\n\\n### Fixed\\n- Restored original functionality that was dropped accidentally in release 0.2:\\n\\n  when there are zero consensus variants to be integrated into the reference\\n  genome (input VCF with no variants), bcftools consensus 1.10 would silently\\n  skip processing of that sequence *including the incorpartion of masking\\n  regions* into the final consensus genome. As a result, a genome covered by\\n  few or no reads at all would have its consensus sequence reported as\\n  all-reference instead of all Ns.\\n  The initial release of the workflow worked around this problem by\\n  pre-masking the reference before passing it to bcftools consensus, but the\\n  corresponding step was dropped as seemingly redundant in release 0.2.\\n\\n  With v1.14, the edge case behavior with zero variants in the input VCF has\\n  been fixed as a bug in bcftools consensus\\n  (https://github.com/samtools/bcftools/issues/1592)\\n  so the workflow fix in this release consists of a simple update to bcftools\\n  consensus 1.15.1.\\n\\n### Changed\\n- Upgraded to new, more flexible version of column_maker tool.\\n\\n  This change allows a simplification of the workflow, which now uses five\\n  steps less for producing identical results.\\n\\n- Updated bcftools consensus to v1.15.1.\\n\\n## [0.3] - 2022-02-02\\n\\n### Fixed\\n- Apply AF thresholds on unbiased AF values recalculated from DP4 and DP fields\\n  instead of on AF values provided by the variant caller to ensure proper\\n  variant gating (into consensus and ambiguous variants) for lofreq-called\\n  data.\\n\\n  https://github.com/CSB5/lofreq/issues/80 means that lofreq-calculated AF\\n  values are lower bounds of true AFs when bases are excluded from calling\\n  based on base quality. The extent of AF underestimation depends on the\\n  fraction of bases with sub-threshold (30 for this workflow) base qualities.\\n\\n  By recalculating AFs as (DP4[2] + DP4[3]) / DP we avoid this issue for\\n  variant calls generated by lofreq. For variants called with Galaxy\'s medaka\\n  consensus/variant wrappers, original AFs are computed with the exact same\\n  formula so the recalculation by the workflow does not affect variant gating\\n  in that case.\\n\\n### Changed\\n- Increase the default for consensus variant allele frequency threshold to 0.75.\\n  Correct calculation of unbiased AF values increases the typical AF of\\n  consensus variants more than enough to justify the change.\\n- The following tools are updated to their latest wrapper versions or revisions:\\n\\n  - bcftools_consensus\\n  - collapse_collections\\n  - the gops subtract and merge tools\\n  - snpsift\\n\\n  None of these updates are expected to impact the generated consensus\\n  sequences.\\n\\n## [0.2.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.2.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.2] - 2021-04-30\\n\\n### Changed\\n- Lower the default for consensus variant allele frequency threshold to 0.7\\n  (from 0.8).\\n  This was empirically determined to capture a relevant number of variants at\\n  sites that are problematic to call.\\n- Increase the default variant allele frequency threshold for ambiguous sites\\n  to 0.25 (from 0.2) to obtain somewhat cleaner consensus sequences.\\n- Use *SnpSift extract*, instead of *VCFtoTab-delimited* followed by *Cut*, to\\n  generate required tabular views of variants in one step.\\n- Eliminate the use of *bedtools MaskFastaBed*, which was redundant with\\n  specifying masking regions directly at the *bcftools consensus* step.\\n- Reduce the number of processing steps further by changing result dataset\\n  types through post-job actions appropriately to avoid implicit conversions,\\n  and by avoiding a redundant *SnpSift filter* step.\\n\\n### Fixed\\n- Deletions are now reliably incorporated into the consensus sequence.\\n  Before they were, in most cases, N-masked because of low coverage in the\\n  deleted region.\\n- Generally, each consensus sequence is now guaranteed to capture all consensus\\n  variants of its samples as stated in the README, independent of it residing\\n  in low-coverage regions or not.\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: consensus construction\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: consensus construction\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Variant calls\\"]@{ shape: docs }\\n1[\\"ℹ️ min-AF for consensus variant\\"]@{ shape: lean-l }\\n2[\\"ℹ️ min-AF for failed variants\\"]@{ shape: lean-l }\\n3[\\"ℹ️ aligned reads data for depth calculation\\"]@{ shape: docs }\\n4[\\"ℹ️ Depth-threshold for masking\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Reference genome\\"]@{ shape: doc }\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 6\\n7[\\"Compose text parameter value\\"]@{ shape: process }\\n2 --> 7\\n1 --> 7\\n8[\\"bedtools Genome Coverage\\"]@{ shape: process }\\n3 --> 8\\n9[\\"Compose text parameter value\\"]@{ shape: process }\\n4 --> 9\\n10[\\"SnpSift Filter\\"]@{ shape: process }\\n6 --> 10\\n0 --> 10\\n11[\\"SnpSift Filter\\"]@{ shape: process }\\n7 --> 11\\n0 --> 11\\n12[\\"Filter\\"]@{ shape: process }\\n9 --> 12\\n8 --> 12\\n13[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n10 --> 13\\n14[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n11 --> 14\\n15[\\"Compute\\"]@{ shape: process }\\n13 --> 15\\n16[\\"Compute\\"]@{ shape: process }\\n14 --> 16\\n17[\\"Concatenate\\"]@{ shape: process }\\n12 --> 17\\n16 --> 17\\n18[\\"Merge\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Subtract\\"]@{ shape: process }\\n18 --> 19\\n15 --> 19\\n20[\\"Compute\\"]@{ shape: process }\\n19 --> 20\\n21[\\"bcftools consensus\\"]@{ shape: process }\\n10 --> 21\\n5 --> 21\\n20 --> 21\\n22[\\"Collapse Collection\\"]@{ shape: process }\\n21 --> 22\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-consensus-from-variation/COVID-19-CONSENSUS-CONSTRUCTION","dockstore_id":16546,"categories":["COVID-19"],"collections":["SARS-COV-2"],"doi":"10.5281/zenodo.15078787","tests":[{"doc":"Test consensus building from called variants","job":{"Reference genome":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1"},"aligned reads data for depth calculation":{"class":"Collection","collection_type":"list","elements":[{"identifier":"SRR11578257","class":"File","path":"test-data/aligned_reads_for_coverage.bam"}]},"Variant calls":{"class":"Collection","collection_type":"list","elements":[{"identifier":"SRR11578257","class":"File","path":"test-data/final_snpeff_annotated_variants.vcf"}]}},"outputs":{"multisample_consensus_fasta":{"file":"test-data/masked_consensus.fa"}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-consensus-from-variation"},{"version":1.2,"workflows":[{"name":"COVID-19-PE-WGS-ILLUMINA","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/pe-wgs-variation.ga","testParameterFiles":["/pe-wgs-variation-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflows performs paired end read mapping with bwa-mem followed by sensitive variant calling across a wide range of AFs with lofreq","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: variation analysis on WGS PE data","release":"0.2.4","steps":{"0":{"annotation":"Illumina reads with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Illumina reads with fastqsanger encoding","name":"Paired Collection"}],"label":"Paired Collection","name":"Input dataset collection","outputs":[],"position":{"bottom":531.03125,"height":81,"left":-513,"right":-313,"top":450.03125,"width":200,"x":-513,"y":450.03125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"b21522ed-fd9a-44b6-8ed3-a8640727961d","workflow_outputs":[]},"1":{"annotation":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","name":"NC_045512.2 FASTA sequence of SARS-CoV-2"}],"label":"NC_045512.2 FASTA sequence of SARS-CoV-2","name":"Input dataset","outputs":[],"position":{"bottom":729.796875,"height":101,"left":-460.59375,"right":-260.59375,"top":628.796875,"width":200,"x":-460.59375,"y":628.796875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\", \\"fasta.gz\\"]}","tool_version":null,"type":"data_input","uuid":"179e3f76-b1f0-4096-864b-a0f143428cfb","workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","errors":null,"id":2,"input_connections":{"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"bottom":436.953125,"height":192,"left":-246.53125,"right":-46.53125,"top":244.953125,"width":200,"x":-246.53125,"y":244.953125},"post_job_actions":{"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","tool_shed_repository":{"changeset_revision":"65b93b623c77","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": \\"false\\", \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": \\"false\\", \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": \\"false\\", \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": \\"true\\", \\"report_json\\": \\"true\\"}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": \\"false\\", \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": \\"false\\", \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": \\"false\\", \\"cut_by_quality3\\": \\"false\\", \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": \\"false\\"}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": \\"false\\", \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.2+galaxy0","type":"tool","uuid":"e3e15355-92b3-4336-a048-facf7b80a456","workflow_outputs":[{"label":"fastp_pe","output_name":"output_paired_coll","uuid":"e12cf072-ff7d-44dc-86a4-d5db9c7b7bc2"},{"label":"fastp_html_report","output_name":"report_html","uuid":"e5330623-a13b-4e91-a7ed-81318648cc75"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","errors":null,"id":3,"input_connections":{"fastq_input|fastq_input1":{"id":2,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"bottom":698.46875,"height":222,"left":19,"right":219,"top":476.46875,"width":200,"x":19,"y":476.46875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","tool_shed_repository":{"changeset_revision":"64f11cf59c6e","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.17.2","type":"tool","uuid":"ff1eb4ed-6403-4017-8538-c16200b33cdb","workflow_outputs":[{"label":"Map with BWA-MEM on input dataset(s) (mapped reads in BAM format)","output_name":"bam_output","uuid":"37ee7bd6-fddf-46cf-8ce4-079e5920cba2"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy2","errors":null,"id":4,"input_connections":{"input":{"id":3,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"bottom":469.234375,"height":112,"left":252.46875,"right":452.46875,"top":357.234375,"width":200,"x":252.46875,"y":357.234375},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Mapped read pairs (filtered bwa-mem result)"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy2","tool_shed_repository":{"changeset_revision":"0dbf49c414ae","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": \\"false\\", \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": \\"false\\"}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.13+galaxy2","type":"tool","uuid":"e5907460-126f-40a0-9cfd-be33fb79c990","workflow_outputs":[{"label":"filtered_mapped_reads","output_name":"outputsam","uuid":"711e76f8-acf5-49f2-a74d-f3eb87a87ec5"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.3","errors":null,"id":5,"input_connections":{"input":{"id":4,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"bottom":399.015625,"height":112,"left":527.671875,"right":727.671875,"top":287.015625,"width":200,"x":527.671875,"y":287.015625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.3","tool_shed_repository":{"changeset_revision":"1cc79f49b8d5","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_group\\": null, \\"read_length\\": null, \\"remove_dups\\": \\"false\\", \\"remove_overlaps\\": \\"false\\", \\"sparse\\": \\"false\\", \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.3","type":"tool","uuid":"290db975-2f61-42a9-8ab9-53f9c9bf638a","workflow_outputs":[{"label":"mapped_reads_stats","output_name":"output","uuid":"ba8ba259-2114-47d7-8555-16b74d482626"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.3","errors":null,"id":6,"input_connections":{"inputFile":{"id":4,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"bottom":653.421875,"height":182,"left":472.046875,"right":672.046875,"top":471.421875,"width":200,"x":472.046875,"y":471.421875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.3","tool_shed_repository":{"changeset_revision":"881d7645d1bf","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": \\"true\\", \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": \\"true\\", \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.18.2.3","type":"tool","uuid":"dfeec15a-ab76-44e8-8e41-ce4f53289f01","workflow_outputs":[{"label":"markduplicates_stats","output_name":"metrics_file","uuid":"90c70f5f-8fee-4648-884d-ebdbf05779cb"},{"label":"markduplicates_reads","output_name":"outFile","uuid":"eb88f5cb-a4d8-41bd-a9ca-617d2edaf041"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","errors":null,"id":7,"input_connections":{"reads":{"id":6,"output_name":"outFile"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Realign reads","outputs":[{"name":"realigned","type":"bam"}],"position":{"bottom":843.140625,"height":162,"left":609.671875,"right":809.671875,"top":681.140625,"width":200,"x":609.671875,"y":681.140625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"aa35ee7f3ab2","name":"lofreq_viterbi","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"keepflags\\": \\"false\\", \\"bq2_handling\\": {\\"replace_bq2\\": \\"keep\\", \\"__current_case__\\": 0, \\"defqual\\": \\"2\\"}}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"4b611028-980b-436f-8fda-e604d4b1fd71","workflow_outputs":[{"label":"realigned_deduplicated_filtered_mapped_reads","output_name":"realigned","uuid":"ad859e0d-38ec-4e17-aefe-42ca2cbbfc20"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","errors":null,"id":8,"input_connections":{"results_0|software_cond|input":{"id":2,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":5,"output_name":"output"},"results_2|software_cond|output_0|input":{"id":6,"output_name":"metrics_file"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"bottom":431.46875,"height":282,"left":965.171875,"right":1165.171875,"top":149.46875,"width":200,"x":965.171875,"y":149.46875},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","tool_shed_repository":{"changeset_revision":"9a913cdee30e","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": \\"false\\", \\"flat\\": \\"false\\", \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy0","type":"tool","uuid":"f87cae1e-1c65-4e6b-b5ab-6e077ae3cfb2","workflow_outputs":[{"label":"preprocessing_and_mapping_reports","output_name":"html_report","uuid":"4ffd2a85-5ed3-4365-a425-301ff8cb4413"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0","errors":null,"id":9,"input_connections":{"reads":{"id":7,"output_name":"realigned"},"strategy|reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Insert indel qualities","outputs":[{"name":"output","type":"bam"}],"position":{"bottom":1045.1875,"height":182,"left":725.734375,"right":925.734375,"top":863.1875,"width":200,"x":725.734375,"y":863.1875},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Fully processed reads for variant calling (deduplicated, realigned reads with added indelquals)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"426d707dfc47","name":"lofreq_indelqual","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strategy\\": {\\"selector\\": \\"dindel\\", \\"__current_case__\\": 1, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"59600224-396f-49d2-add0-30f66519e98f","workflow_outputs":[{"label":"realigned_deduplicated_filtered_mapped_reads_with_indel_quals","output_name":"output","uuid":"46d8cf28-7c9c-4109-a0c3-14f90db52e47"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1","errors":null,"id":10,"input_connections":{"reads":{"id":9,"output_name":"output"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"bottom":1180.671875,"height":122,"left":854.796875,"right":1054.796875,"top":1058.671875,"width":200,"x":854.796875,"y":1058.671875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1","tool_shed_repository":{"changeset_revision":"e1461b5c52a0","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"5\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": \\"false\\"}, \\"bc_quals\\": {\\"min_bq\\": \\"30\\", \\"min_alt_bq\\": \\"30\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": \\"true\\"}}}, \\"map_quals\\": {\\"min_mq\\": \\"20\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_custom\\", \\"__current_case__\\": 3, \\"sig\\": \\"0.0005\\", \\"bonf\\": \\"0\\", \\"others\\": \\"false\\"}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy1","type":"tool","uuid":"04fddca2-1b1e-4483-a269-24302ef7359c","workflow_outputs":[{"label":"called_variants","output_name":"variants","uuid":"e1da347a-271a-4e22-bfc3-9dd9ac7bf929"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","errors":null,"id":11,"input_connections":{"invcf":{"id":10,"output_name":"variants"}},"inputs":[],"label":null,"name":"Lofreq filter","outputs":[{"name":"outvcf","type":"vcf"}],"position":{"bottom":995.296875,"height":112,"left":1052.3125,"right":1252.3125,"top":883.296875,"width":200,"x":1052.3125,"y":883.296875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"7dfca164d2e3","name":"lofreq_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"af\\": {\\"af_min\\": \\"0.0\\", \\"af_max\\": \\"0.0\\"}, \\"coverage\\": {\\"cov_min\\": \\"0\\", \\"cov_max\\": \\"0\\"}, \\"filter_by_type\\": {\\"keep_only\\": \\"\\", \\"__current_case__\\": 0, \\"qual\\": {\\"snvqual_filter\\": {\\"snvqual\\": \\"no\\", \\"__current_case__\\": 0}, \\"indelqual_filter\\": {\\"indelqual\\": \\"no\\", \\"__current_case__\\": 0}}}, \\"flag_or_drop\\": \\"--print-all\\", \\"invcf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sb\\": {\\"sb_filter\\": {\\"strand_bias\\": \\"mtc\\", \\"__current_case__\\": 2, \\"sb_alpha\\": \\"0.001\\", \\"sb_mtc\\": \\"fdr\\", \\"sb_compound\\": \\"true\\", \\"sb_indels\\": \\"false\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"1b2ce667-2a63-4376-b711-b103ea37ba8b","workflow_outputs":[{"label":"soft_filtered_variants","output_name":"outvcf","uuid":"7a8648d0-c844-43a5-85f8-27220011725a"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","errors":null,"id":12,"input_connections":{"input":{"id":11,"output_name":"outvcf"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":"SnpEff eff covid19 version","name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"bottom":751.359375,"height":302,"left":1235.359375,"right":1435.359375,"top":449.359375,"width":200,"x":1235.359375,"y":449.359375},"post_job_actions":{"RenameDatasetActionsnpeff_output":{"action_arguments":{"newname":"Final (SnpEff-)annotated variants"},"action_type":"RenameDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","tool_shed_repository":{"changeset_revision":"2a3a00c1fa0a","name":"snpeff_sars_cov_2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": \\"false\\", \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-upstream\\", \\"-no-utr\\"], \\"generate_stats\\": \\"true\\", \\"genome_version\\": \\"NC_045512.2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": \\"true\\", \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5covid19","type":"tool","uuid":"6808f1c5-55da-4e39-a453-e215650d22e7","workflow_outputs":[{"label":"annotated_variants","output_name":"snpeff_output","uuid":"218ed74c-5d87-472a-88f4-6b06559e7885"},{"label":"annotated_variants_stats","output_name":"statsFile","uuid":"1986e97d-92a8-4812-99e7-42818115e675"}]}},"tags":["COVID-19","covid19.galaxyproject.org","iwc","emergen_validated"],"uuid":"6695eb16-30a5-42ea-8591-56f2630cf8bf","version":4},"readme":"COVID-19: variation analysis on WGS PE data\\n-------------------------------------------\\n\\nThis workflows performs paired end read mapping with bwa-mem followed by\\nsensitive variant calling across a wide range of AFs with lofreq and variant\\nannotation with snpEff 4.5covid19.\\n","changelog":"# Changelog\\n\\n## [0.2.4] - 2023-11-20\\n\\n- Fix author in dockstore\\n- Use zenodo link instead of googleapis for fastq\\n\\n## [0.2.3] - 2022-02-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.3`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1`\\n\\n## [0.2.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.2.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.2] - 2021-06-17\\n\\n### Changed\\n\\n- Upgrade multiqc to 1.9+galaxy1\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: variation analysis on WGS PE data workflow\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: variation analysis on WGS PE data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ NC_045512.2 FASTA sequence of SARS-CoV-2\\"]@{ shape: doc }\\n2[\\"fastp\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Map with BWA-MEM\\"]@{ shape: process }\\n2 --> 3\\n1 --> 3\\n4[\\"Samtools view\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Samtools stats\\"]@{ shape: process }\\n4 --> 5\\n6[\\"MarkDuplicates\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Realign reads\\"]@{ shape: process }\\n6 --> 7\\n1 --> 7\\n8[\\"MultiQC\\"]@{ shape: process }\\n2 --> 8\\n5 --> 8\\n6 --> 8\\n9[\\"Insert indel qualities\\"]@{ shape: process }\\n7 --> 9\\n1 --> 9\\n10[\\"Call variants\\"]@{ shape: process }\\n9 --> 10\\n1 --> 10\\n11[\\"Lofreq filter\\"]@{ shape: process }\\n10 --> 11\\n12[\\"SnpEff eff covid19 version\\"]@{ shape: process }\\n11 --> 12\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-wgs-variant-calling/COVID-19-PE-WGS-ILLUMINA","dockstore_id":16243,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078781","tests":[{"doc":"Test workflow execution using paired end illumina accession","job":{"NC_045512.2 FASTA sequence of SARS-CoV-2":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1"},"Paired Collection":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"SRR11578257","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/10174466/files/SRR11578257_R1.fastq.gz?download=1"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/10174466/files/SRR11578257_R2.fastq.gz?download=1"}]}]}},"outputs":{"annotated_variants":{"attributes":{},"element_tests":{"SRR11578257":{"path":"test-data/final_snpeff_annotated_variants.vcf","compare":"diff","lines_diff":6}}}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-wgs-variant-calling"},{"version":1.2,"workflows":[{"name":"COVID-19-ARTIC-ONT","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/ont-artic-variation.ga","testParameterFiles":["/ont-artic-variation-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: variation analysis of ARTIC ONT data","release":"0.3.2","steps":{"0":{"annotation":"ONT reads from ARTIC assay with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"ONT reads from ARTIC assay with fastqsanger encoding","name":"ONT-sequenced reads"}],"label":"ONT-sequenced reads","name":"Input dataset collection","outputs":[],"position":{"bottom":230.265625,"height":105,"left":-1042.453125,"right":-842.453125,"top":125.265625,"width":200,"x":-1042.453125,"y":125.265625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"27eb7472-10b2-479a-b57f-d55aee92605a","workflow_outputs":[]},"1":{"annotation":"Discard reads that are shorter than this length","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Discard reads that are shorter than this length","name":"Minimum read length"}],"label":"Minimum read length","name":"Input parameter","outputs":[],"position":{"bottom":362.609375,"height":105,"left":-1042.5,"right":-842.5,"top":257.609375,"width":200,"x":-1042.5,"y":257.609375},"tool_id":null,"tool_state":"{\\"default\\": 400, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d10a48bc-95d0-4f53-b5a5-bff17e020aea","workflow_outputs":[{"label":null,"output_name":"output","uuid":"446daa93-2c4a-41ce-88d8-3213cb61dcaa"}]},"2":{"annotation":"Discard reads that are longer than this length","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Discard reads that are longer than this length","name":"Maximum read length"}],"label":"Maximum read length","name":"Input parameter","outputs":[],"position":{"bottom":499.546875,"height":105,"left":-1035.46875,"right":-835.46875,"top":394.546875,"width":200,"x":-1035.46875,"y":394.546875},"tool_id":null,"tool_state":"{\\"default\\": 700, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"be62c106-273a-4102-a9a1-1376525e4f4e","workflow_outputs":[{"label":null,"output_name":"output","uuid":"bb83c99a-636c-45a7-85fd-438ef84a6dfb"}]},"3":{"annotation":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","name":"NC_045512.2 FASTA sequence of SARS-CoV-2"}],"label":"NC_045512.2 FASTA sequence of SARS-CoV-2","name":"Input dataset","outputs":[],"position":{"bottom":636.578125,"height":105,"left":-1034.6875,"right":-834.6875,"top":531.578125,"width":200,"x":-1034.6875,"y":531.578125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\", \\"fasta.gz\\"]}","tool_version":null,"type":"data_input","uuid":"b67b0bfd-6771-444a-ae7d-5d02f7f9157b","workflow_outputs":[]},"4":{"annotation":"BED file containing (ARTIC) primer positions. For ARTIC primer schemes the file can be obtained from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"BED file containing (ARTIC) primer positions. For ARTIC primer schemes the file can be obtained from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources.","name":"Primer binding sites info in BED format"}],"label":"Primer binding sites info in BED format","name":"Input dataset","outputs":[],"position":{"bottom":785.640625,"height":105,"left":-1031.953125,"right":-831.953125,"top":680.640625,"width":200,"x":-1031.953125,"y":680.640625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bed\\"]}","tool_version":null,"type":"data_input","uuid":"4faea19f-da1d-4683-9317-f7861174a8b6","workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0","errors":null,"id":5,"input_connections":{"filter_options|length_filtering_options|length_limit":{"id":2,"output_name":"output"},"filter_options|length_filtering_options|length_required":{"id":1,"output_name":"output"},"single_paired|in1":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"fastp","outputs":[{"name":"out1","type":"input"},{"name":"report_html","type":"html"}],"position":{"bottom":337,"height":209,"left":-790.8125,"right":-590.8125,"top":128,"width":200,"x":-790.8125,"y":128},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0","tool_shed_repository":{"changeset_revision":"dbf9c561ef29","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": \\"true\\", \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": \\"false\\", \\"length_required\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_limit\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": \\"false\\", \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": \\"true\\", \\"report_json\\": \\"false\\"}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": \\"false\\", \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"-G\\", \\"__current_case__\\": 2}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": \\"false\\", \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": \\"false\\", \\"cut_by_quality3\\": \\"false\\", \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": \\"false\\"}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"in1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": \\"true\\", \\"adapter_sequence1\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.20.1+galaxy0","type":"tool","uuid":"69ab6918-2d3e-4512-81bf-692fb776bb74","workflow_outputs":[{"label":"length_filtered_reads","output_name":"out1","uuid":"dd4c3b98-91d8-4963-b40f-e7e5a5e99e24"},{"label":"fastp_report","output_name":"report_html","uuid":"c89895ce-50a7-4c19-aa3c-badc0cb534d3"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","errors":null,"id":6,"input_connections":{"input":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"bottom":817.796875,"height":95,"left":-563.46875,"right":-363.46875,"top":722.796875,"width":200,"x":-563.46875,"y":722.796875},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","tool_shed_repository":{"changeset_revision":"02026300aa45","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": \\"false\\", \\"cond\\": \\"c3-c2\\", \\"header_lines_conditional\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"round\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6","type":"tool","uuid":"d8c4dcca-6715-4c2d-8598-8f9621a26e6b","workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3","errors":null,"id":7,"input_connections":{"infile":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"bottom":1039.5,"height":116,"left":77.578125,"right":277.578125,"top":923.5,"width":200,"x":77.578125,"y":923.5},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3","tool_shed_repository":{"changeset_revision":"ddf54b12c295","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"column\\": \\"1\\", \\"find_pattern\\": \\"MN908947.3\\", \\"replace_pattern\\": \\"NC_045512.2\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"06a64d18-798b-4b40-b744-f20b62b8e23b","workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2","errors":null,"id":8,"input_connections":{"fastq_input|fastq_input1":{"id":5,"output_name":"out1"},"reference_source|ref_file":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"bottom":478.65625,"height":210,"left":-545.5,"right":-345.5,"top":268.65625,"width":200,"x":-545.5,"y":268.65625},"post_job_actions":{"RenameDatasetActionalignment_output":{"action_arguments":{"newname":"Mapped reads (output of minimap2)"},"action_type":"RenameDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2","tool_shed_repository":{"changeset_revision":"8c6cd2650d1f","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": \\"true\\"}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"map-ont\\"}, \\"indexing_options\\": {\\"H\\": \\"false\\", \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"BAM\\", \\"Q\\": \\"false\\", \\"L\\": \\"false\\", \\"K\\": null, \\"cs\\": null, \\"c\\": \\"false\\", \\"eqx\\": \\"false\\", \\"Y\\": \\"false\\"}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"min_occ_floor\\": null, \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": \\"false\\", \\"p\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.17+galaxy2","type":"tool","uuid":"b07d928c-302e-4fc7-909a-e090813c1875","workflow_outputs":[{"label":"minimap2_mapped_reads","output_name":"alignment_output","uuid":"0877f387-f32c-48ee-9824-055583bd99fd"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":9,"input_connections":{"in_file":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"bottom":858.71875,"height":116,"left":-320.8125,"right":-120.8125,"top":742.71875,"width":200,"x":-320.8125,"y":742.71875},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": \\"false\\", \\"header_out\\": \\"false\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"false\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"max\\", \\"op_column\\": \\"7\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"e2c31a21-c133-4d7d-9e2f-608864b34af6","workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy2","errors":null,"id":10,"input_connections":{"input":{"id":8,"output_name":"alignment_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"bottom":407.734375,"height":137,"left":-311.625,"right":-111.625,"top":270.734375,"width":200,"x":-311.625,"y":270.734375},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Mapped reads (filtered minimap2 result)"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy2","tool_shed_repository":{"changeset_revision":"bf328cec6a42","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"1\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": [\\"4\\", \\"256\\"], \\"exclusive_filter_all\\": null}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": \\"false\\", \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": \\"false\\"}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.9+galaxy2","type":"tool","uuid":"790d9f16-849c-4e47-af55-c9393504ae5c","workflow_outputs":[{"label":"filtered_mapped_reads","output_name":"outputsam","uuid":"10318fc5-8dc2-41e0-b6f5-48b9a419d72e"}]},"11":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":11,"input_connections":{"input1":{"id":9,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"bottom":883.578125,"height":158,"left":-74.328125,"right":125.671875,"top":725.578125,"width":200,"x":-74.328125,"y":725.578125},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"max primer length to be used as padding width for calling in primer binding sites"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"6b010208-e861-4d9a-b0e5-f656e792f3f7","workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"bottom":111,"height":137,"left":-74.09375,"right":125.90625,"top":-26,"width":200,"x":-74.09375,"y":-26},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","tool_shed_repository":{"changeset_revision":"145f6d74ff5e","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_length\\": null, \\"remove_dups\\": \\"false\\", \\"remove_overlaps\\": \\"false\\", \\"sparse\\": \\"false\\", \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.2+galaxy2","type":"tool","uuid":"9bfcce2b-6fb4-44f9-b893-20d3f32be170","workflow_outputs":[{"label":"mapped_reads_stats","output_name":"output","uuid":"dcde6643-5ac7-4bca-92ca-169544c0a6d8"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1","errors":null,"id":13,"input_connections":{"reference_source|input_bam":{"id":10,"output_name":"outputsam"},"reference_source|ref_file":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"BamLeftAlign","outputs":[{"name":"output_bam","type":"bam"}],"position":{"bottom":554.5,"height":189,"left":-75.953125,"right":124.046875,"top":365.5,"width":200,"x":-75.953125,"y":365.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1","tool_shed_repository":{"changeset_revision":"ef2c525bd8cd","name":"freebayes","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"iterations\\": \\"5\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.1","type":"tool","uuid":"78489e65-6f95-47ec-aacd-9023d8c2c3bc","workflow_outputs":[{"label":"realigned_filtered_mapped_reads","output_name":"output_bam","uuid":"6deefe2d-57d7-4e61-b929-a1d6bcc877d2"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy2","errors":null,"id":14,"input_connections":{"input_bam":{"id":13,"output_name":"output_bam"},"primer|input_bed":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"ivar trim","outputs":[{"name":"output_bam","type":"bam"}],"position":{"bottom":425.375,"height":210,"left":158.125,"right":358.125,"top":215.375,"width":200,"x":158.125,"y":215.375},"post_job_actions":{"RenameDatasetActionoutput_bam":{"action_arguments":{"newname":"Fully processed reads for variant calling (primer-trimmed, realigned reads)"},"action_type":"RenameDatasetAction","output_name":"output_bam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy2","tool_shed_repository":{"changeset_revision":"c092052ed673","name":"ivar_trim","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"amplicons\\": {\\"filter_by\\": \\"\\", \\"__current_case__\\": 0}, \\"inc_primers\\": \\"false\\", \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_len\\": \\"1\\", \\"min_qual\\": \\"0\\", \\"primer\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 0, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"primer_pos_wiggle\\": \\"0\\", \\"window_width\\": \\"4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.1+galaxy2","type":"tool","uuid":"cee59eba-91f0-4beb-82d5-3b827e867657","workflow_outputs":[{"label":"primer_trimmed_realigned_filtered_mapped_reads","output_name":"output_bam","uuid":"790186ed-d660-4035-9ea9-5ed6b3bf58c2"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","errors":null,"id":15,"input_connections":{"input1":{"id":14,"output_name":"output_bam"}},"inputs":[],"label":null,"name":"QualiMap BamQC","outputs":[{"name":"raw_data","type":"input"},{"name":"output_html","type":"html"}],"position":{"bottom":297.984375,"height":210,"left":386.53125,"right":586.53125,"top":87.984375,"width":200,"x":386.53125,"y":87.984375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","tool_shed_repository":{"changeset_revision":"19ece8afbaab","name":"qualimap_bamqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"duplicate_skipping\\": [\\"0\\"], \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"per_base_coverage\\": \\"false\\", \\"plot_specific\\": {\\"n_bins\\": \\"400\\", \\"paint_chromosome_limits\\": \\"true\\", \\"genome_gc_distr\\": null, \\"homopolymer_size\\": \\"3\\"}, \\"stats_regions\\": {\\"region_select\\": \\"all\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.2d+galaxy3","type":"tool","uuid":"63d12f6d-9a45-487c-ad82-841aba341ef5","workflow_outputs":[{"label":"bamqc_raw_output","output_name":"raw_data","uuid":"0c2a68ba-319a-4aa8-bbd3-17ca49d5e9ab"},{"label":"bamqc_html_output","output_name":"output_html","uuid":"e81ce71a-2ab8-4a54-bf6d-d65a1599c530"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus/medaka_consensus/1.0.3+galaxy2","errors":null,"id":16,"input_connections":{"bam":{"id":14,"output_name":"output_bam"}},"inputs":[],"label":null,"name":"medaka consensus tool","outputs":[{"name":"out_result","type":"h5"}],"position":{"bottom":664.84375,"height":158,"left":370.78125,"right":570.78125,"top":506.84375,"width":200,"x":370.78125,"y":506.84375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus/medaka_consensus/1.0.3+galaxy2","tool_shed_repository":{"changeset_revision":"db25ec99c3ea","name":"medaka_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"RG\\": \\"\\", \\"bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"batch_size\\": \\"100\\", \\"check_output\\": \\"false\\", \\"chunk_len\\": \\"800\\", \\"chunk_ovlp\\": \\"400\\", \\"disable_cudnn\\": \\"false\\", \\"model\\": \\"r941_min_high_g351\\", \\"out\\": [\\"result\\"], \\"regions_cond\\": {\\"regions_sel\\": \\"none\\", \\"__current_case__\\": 0}, \\"save_features\\": \\"false\\", \\"tag_keep_missing\\": \\"false\\", \\"tag_name\\": \\"\\", \\"tag_value\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3+galaxy2","type":"tool","uuid":"d35f7df5-5b3f-43c3-b558-dd94dade926e","workflow_outputs":[{"label":"medaka_consensus_data","output_name":"out_result","uuid":"eeadcbf5-1300-41d6-8e0f-a251dd1c0299"}]},"17":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":17,"input_connections":{"input":{"id":15,"output_name":"raw_data"}},"inputs":[],"label":null,"name":"Filter failed","outputs":[{"name":"output","type":"input"}],"position":{"bottom":255.453125,"height":116,"left":618.953125,"right":818.953125,"top":139.453125,"width":200,"x":618.953125,"y":139.453125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"717562b8-2694-44ff-94e2-47d1aed4420a","workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_variant/medaka_variant/1.3.2+galaxy1","errors":null,"id":18,"input_connections":{"output_annotated|in_bam":{"id":13,"output_name":"output_bam"},"pool|input":{"id":16,"output_name":"out_result"},"reference_source|ref_file":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"medaka variant tool","outputs":[{"name":"out_variants","type":"vcf"}],"position":{"bottom":564.796875,"height":262,"left":605.109375,"right":805.109375,"top":302.796875,"width":200,"x":605.109375,"y":302.796875},"post_job_actions":{"RenameDatasetActionout_variants":{"action_arguments":{"newname":"Variant calls"},"action_type":"RenameDatasetAction","output_name":"out_variants"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_variant/medaka_variant/1.3.2+galaxy1","tool_shed_repository":{"changeset_revision":"0f5f4a208660","name":"medaka_variant","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"ambig_ref\\": \\"false\\", \\"gvcf\\": \\"false\\", \\"output_annotated\\": {\\"output_annotated_select\\": \\"true\\", \\"__current_case__\\": 0, \\"in_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pad\\": \\"25\\"}, \\"output_log_bool\\": \\"false\\", \\"pool\\": {\\"pool_mode\\": \\"No\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": \\"\\", \\"verbose\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.2+galaxy1","type":"tool","uuid":"dbc1b19c-5ada-4209-94bb-4c6eb628944a","workflow_outputs":[{"label":"medaka_variants_general","output_name":"out_variants","uuid":"23525921-bef8-4f12-8325-233c80204bac"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_variant/medaka_variant/1.3.2+galaxy1","errors":null,"id":19,"input_connections":{"output_annotated|in_bam":{"id":13,"output_name":"output_bam"},"output_annotated|pad":{"id":11,"output_name":"integer_param"},"pool|input":{"id":16,"output_name":"out_result"},"reference_source|ref_file":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"medaka variant tool","outputs":[{"name":"out_variants","type":"vcf"}],"position":{"bottom":930.734375,"height":314,"left":605.21875,"right":805.21875,"top":616.734375,"width":200,"x":605.21875,"y":616.734375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_variant/medaka_variant/1.3.2+galaxy1","tool_shed_repository":{"changeset_revision":"0f5f4a208660","name":"medaka_variant","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"ambig_ref\\": \\"false\\", \\"gvcf\\": \\"false\\", \\"output_annotated\\": {\\"output_annotated_select\\": \\"true\\", \\"__current_case__\\": 0, \\"in_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pad\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"output_log_bool\\": \\"false\\", \\"pool\\": {\\"pool_mode\\": \\"No\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": \\"\\", \\"verbose\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.2+galaxy1","type":"tool","uuid":"8ee2fde4-9631-4e51-9c50-4419cfe39b58","workflow_outputs":[{"label":"medaka_variants_pbs","output_name":"out_variants","uuid":"06999460-a1c2-4dc9-9c55-e0650f8f052d"}]},"20":{"annotation":"","content_id":"__FLATTEN__","errors":null,"id":20,"input_connections":{"input":{"id":17,"output_name":"output"}},"inputs":[],"label":null,"name":"Flatten Collection","outputs":[{"name":"output","type":"input"}],"position":{"bottom":307.421875,"height":158,"left":860.46875,"right":1060.46875,"top":149.421875,"width":200,"x":860.46875,"y":149.421875},"post_job_actions":{},"tool_id":"__FLATTEN__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"join_identifier\\": \\"_\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"de07859e-5ed1-46d0-9cd0-2f3848b4ef63","workflow_outputs":[{"label":"bamqc_raw_output_flattened","output_name":"output","uuid":"68801b40-55d8-4db3-a065-578161bdaeeb"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0","errors":null,"id":21,"input_connections":{"inputA":{"id":19,"output_name":"out_variants"},"reduce_or_iterate|inputB":{"id":7,"output_name":"outfile"}},"inputs":[],"label":null,"name":"bedtools Intersect intervals","outputs":[{"name":"output","type":"input"}],"position":{"bottom":982.46875,"height":147,"left":825.828125,"right":1025.828125,"top":835.46875,"width":200,"x":825.828125,"y":835.46875},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Adjusted variant calls within primer binding sites"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0","tool_shed_repository":{"changeset_revision":"a68aa6c1204a","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed\\": \\"false\\", \\"count\\": \\"false\\", \\"fraction_cond\\": {\\"fraction_select\\": \\"default\\", \\"__current_case__\\": 0}, \\"header\\": \\"true\\", \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"false\\", \\"once\\": \\"true\\", \\"overlap_mode\\": [\\"-wa\\"], \\"reduce_or_iterate\\": {\\"reduce_or_iterate_selector\\": \\"iterate\\", \\"__current_case__\\": 0, \\"inputB\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"sorted\\": \\"false\\", \\"split\\": \\"false\\", \\"strand\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.30.0","type":"tool","uuid":"7e2468f7-9392-4f82-b6d2-11a338ed81a2","workflow_outputs":[{"label":"variants_pbs","output_name":"output","uuid":"209c5be7-ef0d-4009-8ac1-b0b3dd812207"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1","errors":null,"id":22,"input_connections":{"results_0|software_cond|output_0|type|input":{"id":12,"output_name":"output"},"results_1|software_cond|input":{"id":20,"output_name":"output"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"plots","type":"input"},{"name":"html_report","type":"html"}],"position":{"bottom":365.953125,"height":335,"left":1097.328125,"right":1297.328125,"top":30.953125,"width":200,"x":1097.328125,"y":30.953125},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"},"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1","tool_shed_repository":{"changeset_revision":"5e33b465d8d5","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"flat\\": \\"false\\", \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 23, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"qualimap\\", \\"__current_case__\\": 19, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy1","type":"tool","uuid":"815cbc27-8533-42e0-b5b8-c9312aa8cdfe","workflow_outputs":[{"label":"preprocessing_and_mapping_reports","output_name":"html_report","uuid":"6c317485-2591-4984-b163-378faa100f1e"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.10","errors":null,"id":23,"input_connections":{"input_file":{"id":18,"output_name":"out_variants"},"sec_annofile|annofile|annotations":{"id":21,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bcftools annotate","name":"input_file"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_annotate"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_restrict"}],"label":null,"name":"bcftools annotate","outputs":[{"name":"output_file","type":"vcf"}],"position":{"bottom":828,"height":230,"left":1046.5,"right":1246.5,"top":598,"width":200,"x":1046.5,"y":598},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.10","tool_shed_repository":{"changeset_revision":"403553ae19f6","name":"bcftools_annotate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_type\\": \\"v\\", \\"sec_annofile\\": {\\"columns\\": \\"INFO\\", \\"annofile\\": {\\"anno_fmt\\": \\"vcf\\", \\"__current_case__\\": 1, \\"annotations\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"mark_sites\\": \\"\\", \\"set_id\\": \\"\\"}, \\"sec_annotate\\": {\\"remove\\": \\"FORMAT\\", \\"rename_chrs\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"sec_restrict\\": {\\"include\\": \\"\\", \\"exclude\\": \\"\\", \\"collapse\\": null, \\"regions\\": {\\"regions_src\\": \\"__none__\\", \\"__current_case__\\": 0}, \\"samples\\": \\"\\", \\"invert_samples\\": \\"false\\", \\"samples_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"invert_samples_file\\": \\"false\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.10","type":"tool","uuid":"bf733d38-0a4a-4f40-92cf-92e22ff12f5b","workflow_outputs":[{"label":"variants_combined","output_name":"output_file","uuid":"47e2fdc8-dd5a-4f46-8084-ba37d80091db"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","errors":null,"id":24,"input_connections":{"input":{"id":23,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":"SnpEff eff covid19 version","name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"bottom":833.90625,"height":335,"left":1287.59375,"right":1487.59375,"top":498.90625,"width":200,"x":1287.59375,"y":498.90625},"post_job_actions":{"RenameDatasetActionsnpeff_output":{"action_arguments":{"newname":"Final (SnpEff-) annotated variants"},"action_type":"RenameDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","tool_shed_repository":{"changeset_revision":"2a3a00c1fa0a","name":"snpeff_sars_cov_2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": \\"false\\", \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-upstream\\"], \\"generate_stats\\": \\"true\\", \\"genome_version\\": \\"NC_045512.2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": \\"true\\", \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5covid19","type":"tool","uuid":"9eb5a77f-3c5f-43cb-9811-5d4c6cf7931a","workflow_outputs":[{"label":"annotated_variants","output_name":"snpeff_output","uuid":"22a622ff-26df-4d6c-a08c-24fe57703982"},{"label":"annotated_variants_stats","output_name":"statsFile","uuid":"47b851dc-c115-441a-9408-22f579f59a7e"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","errors":null,"id":25,"input_connections":{"invcf":{"id":24,"output_name":"snpeff_output"}},"inputs":[],"label":null,"name":"Lofreq filter","outputs":[{"name":"outvcf","type":"vcf"}],"position":{"bottom":598.1875,"height":95,"left":1512.21875,"right":1712.21875,"top":503.1875,"width":200,"x":1512.21875,"y":503.1875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"7dfca164d2e3","name":"lofreq_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"af\\": {\\"af_min\\": \\"0.0\\", \\"af_max\\": \\"0.0\\"}, \\"coverage\\": {\\"cov_min\\": \\"0\\", \\"cov_max\\": \\"0\\"}, \\"filter_by_type\\": {\\"keep_only\\": \\"\\", \\"__current_case__\\": 0, \\"qual\\": {\\"snvqual_filter\\": {\\"snvqual\\": \\"no\\", \\"__current_case__\\": 0}, \\"indelqual_filter\\": {\\"indelqual\\": \\"no\\", \\"__current_case__\\": 0}}}, \\"flag_or_drop\\": \\"--print-all\\", \\"invcf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sb\\": {\\"sb_filter\\": {\\"strand_bias\\": \\"mtc\\", \\"__current_case__\\": 2, \\"sb_alpha\\": \\"0.001\\", \\"sb_mtc\\": \\"fdr\\", \\"sb_compound\\": \\"true\\", \\"sb_indels\\": \\"false\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"f0e396ff-d72b-4d02-9c4f-885557073ddd","workflow_outputs":[{"label":"lofreq_filtered","output_name":"outvcf","uuid":"7fb4cdec-b28c-40a8-a36a-dd30ab4987d2"}]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":26,"input_connections":{"infile":{"id":25,"output_name":"outvcf"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"bottom":641.015625,"height":137,"left":1729.296875,"right":1929.296875,"top":504.015625,"width":200,"x":1729.296875,"y":504.015625},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Final (SnpEff-) annotated variants with strand-bias soft filter applied"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"ddf54b12c295","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"#CHROM\\\\\\\\tPOS\\\\\\\\tID\\\\\\\\tREF\\\\\\\\tALT\\\\\\\\tQUAL\\\\\\\\tFILTER\\\\\\\\tINFO.*\\", \\"global\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"true\\", \\"replace_pattern\\": \\"#CHROM\\\\\\\\tPOS\\\\\\\\tID\\\\\\\\tREF\\\\\\\\tALT\\\\\\\\tQUAL\\\\\\\\tFILTER\\\\\\\\tINFO\\\\\\\\tFORMAT\\\\\\\\tSAMPLE\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"false\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"0932d832-0ab4-4399-a4eb-a065e7cf1b7a","workflow_outputs":[{"label":"annotated_softfiltered_variants","output_name":"outfile","uuid":"22a290c5-0f48-4140-b759-52ee0b73d1b2"}]}},"tags":["COVID-19","ARTIC","ONT","covid19.galaxyproject.org"],"uuid":"a1ceaf35-f3d7-452d-b5ed-53a4f429d1cf"},"readme":"COVID-19: variation analysis on ARTIC ONT data\\n----------------------------------------------\\n\\nThis workflow for ONT-sequenced ARTIC data is modeled after the alignment/variant-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/en/latest/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with minimap2 and variant calling with medaka. Like the Illumina ARTIC workflow it uses ivar for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefor plagued more by false-positive variant calls, this workflow does make no attempt to handle amplicons affected by potential primer-binding site mutations.\\n","changelog":"# Changelog\\n\\n## [0.3.2] - 2023-11-20\\n\\n- Fix author in dockstore\\n- Use ncbi link instead of googleapis for fastq\\n\\n## [0.3.1] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.3] - 2021-09-22\\n\\n### Changed\\n\\nThis version changes the way variants get called and and how key call\\nstatistics are calculated:\\n\\n- Switch to medaka_variant version 1.3.2+galaxy1 for extracting variants from\\n  medaka consensus data.\\n\\n  This new version of the tool is more robust against input data peculiarities\\n  at the VCF annotation stage:\\n\\n  * it doesn\'t fail on empty BAM input\\n  * it doesn\'t crash on variant calls of unusually high quality that previously\\n    resulted in math domain errors when trying to calculate PHRED scores from\\n    very small error probabilities.\\n\\n  This tool update also means that key INFO fields (DP, DP4, AF) are\\n  now based on calculations carried out by medaka tools annotate instead of by\\n  custom code using samtools mpileup. This has the following consequences:\\n\\n  * the tool can now emit variant calls at complex sites with > 1 lengths of\\n    both the REF and the ALT allele, which were previously dropped\\n  * the workflow became more complex; to account for shortcomings of medaka\\n    tools annotate, the variant call statistics of regular variants and of\\n    primer binding site variants have to be determined in separate runs of the\\n    tool\\n  * All key INFO fields (DP, DP4, AF) will change slightly in this version of\\n    the workflow\\n\\nThis version also adds some of the changes around trimming of primer sequences,\\nwhich have been introduced into version 0.3 of the PE Illumina worflow for\\namplicon data before:\\n\\n- Update ivar trim to version 1.3.1\\n- Run ivar trim as the last mapped reads processing step before variant\\n  calling, i.e., after left-alignment of indels\\n\\nand:\\n\\n- Rename the output of the ivar trim step to \\"Fully processed reads for\\n  variant calling (primer-trimmed, realigned reads)\\" like the corresponding\\n  output of the PE Illumina workflow\\n- Fix a typo in the allowed input formats for the collection of sequenced\\n  reads, which caused fastqsanger.gz data to undergo an implicit and\\n  unnecessary decompression step.\\n\\n### Added\\n\\n- Add a step to filter out failed datasets before flattening the Qualimap BamQC\\n  data for use by MultiQC.\\n\\n  Qualimap BamQC fails on empty BAM input and trying to flatten the resulting\\n  collection containing failed datasets would cause the invocation of the\\n  workflow to fail.\\n\\n## [0.2.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.2]\\n\\n- Apply the strand-bias filter only after variant annotation with snpEff. By\\n  producing fully annotated VCFs with and without filtering, downstream\\n  workflows can easily be switched between filtered/unfiltered input data\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: variation analysis on ARTIC ONT data workflow\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: variation analysis of ARTIC ONT data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ ONT-sequenced reads\\"]@{ shape: docs }\\n1[\\"ℹ️ Minimum read length\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Maximum read length\\"]@{ shape: lean-l }\\n3[\\"ℹ️ NC_045512.2 FASTA sequence of SARS-CoV-2\\"]@{ shape: doc }\\n4[\\"ℹ️ Primer binding sites info in BED format\\"]@{ shape: doc }\\n5[\\"fastp\\"]@{ shape: process }\\n2 --> 5\\n1 --> 5\\n0 --> 5\\n6[\\"Compute\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Replace Text\\"]@{ shape: process }\\n4 --> 7\\n8[\\"Map with minimap2\\"]@{ shape: process }\\n5 --> 8\\n3 --> 8\\n9[\\"Datamash\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Samtools view\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Parse parameter value\\"]@{ shape: process }\\n9 --> 11\\n12[\\"Samtools stats\\"]@{ shape: process }\\n10 --> 12\\n13[\\"BamLeftAlign\\"]@{ shape: process }\\n10 --> 13\\n3 --> 13\\n14[\\"ivar trim\\"]@{ shape: process }\\n13 --> 14\\n4 --> 14\\n15[\\"QualiMap BamQC\\"]@{ shape: process }\\n14 --> 15\\n16[\\"medaka consensus tool\\"]@{ shape: process }\\n14 --> 16\\n17[\\"Filter failed\\"]@{ shape: process }\\n15 --> 17\\n18[\\"medaka variant tool\\"]@{ shape: process }\\n13 --> 18\\n16 --> 18\\n3 --> 18\\n19[\\"medaka variant tool\\"]@{ shape: process }\\n13 --> 19\\n11 --> 19\\n16 --> 19\\n3 --> 19\\n20[\\"Flatten Collection\\"]@{ shape: process }\\n17 --> 20\\n21[\\"bedtools Intersect intervals\\"]@{ shape: process }\\n19 --> 21\\n7 --> 21\\n22[\\"MultiQC\\"]@{ shape: process }\\n12 --> 22\\n20 --> 22\\n23[\\"bcftools annotate\\"]@{ shape: process }\\n18 --> 23\\n21 --> 23\\n24[\\"SnpEff eff covid19 version\\"]@{ shape: process }\\n23 --> 24\\n25[\\"Lofreq filter\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Replace\\"]@{ shape: process }\\n25 --> 26\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-ont-artic-variant-calling/COVID-19-ARTIC-ONT","dockstore_id":16246,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078464","tests":[{"doc":"Test workflow execution using ONT ARTIC accession","job":{"NC_045512.2 FASTA sequence of SARS-CoV-2":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1"},"Primer binding sites info in BED format":{"class":"File","location":"https://zenodo.org/record/4555735/files/ARTIC_nCoV-2019_v3.bed?download=1"},"ONT-sequenced reads":{"class":"Collection","collection_type":"list","elements":[{"identifier":"SRR12447380","class":"File","filetype":"fastqsanger.gz","location":"https://www.be-md.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR12447380"}]}},"outputs":{"annotated_softfiltered_variants":{"attributes":{},"element_tests":{"SRR12447380":{"path":"test-data/final_snpeff_annotated_variants.vcf","compare":"diff","lines_diff":6}}}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-ont-artic-variant-calling"},{"version":1.2,"workflows":[{"name":"COVID-19-PE-ARTIC-ILLUMINA","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/pe-artic-variation.ga","testParameterFiles":["/pe-artic-variation-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"The workflow for Illumina-sequenced ARTIC data builds on the RNASeq workflow for paired-end data using the same steps for mapping and variant calling, but adds extra logic for trimming ARTIC primer sequences off reads with the ivar package. In addition, this workflow uses ivar also to identify amplicons affected by ARTIC primer-binding site mutations and tries to exclude reads derived from such tainted amplicons when calculating allele-frequencies of other variants.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","name":"COVID-19: variation analysis on ARTIC PE data","release":"0.5.4","steps":{"0":{"annotation":"Illumina reads from ARTIC assay with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Illumina reads from ARTIC assay with fastqsanger encoding","name":"Paired Collection"}],"label":"Paired Collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":506.796875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"dfab5b21-4abb-4269-978d-7c7e24479e4a","when":null,"workflow_outputs":[]},"1":{"annotation":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Fasta sequence for Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome","name":"NC_045512.2 FASTA sequence of SARS-CoV-2"}],"label":"NC_045512.2 FASTA sequence of SARS-CoV-2","name":"Input dataset","outputs":[],"position":{"left":269.78125,"top":883.5625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\", \\"fasta.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"03043a6d-ee9e-4af5-abf4-f960beb6b4e9","when":null,"workflow_outputs":[]},"2":{"annotation":"BED file containing ARTIC primer positions. Can be retrieved from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"BED file containing ARTIC primer positions. Can be retrieved from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources","name":"ARTIC primer BED"}],"label":"ARTIC primer BED","name":"Input dataset","outputs":[],"position":{"left":962.171875,"top":326.90625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bed\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"cd6caddf-88ee-47a2-b91d-141fd90b046f","when":null,"workflow_outputs":[]},"3":{"annotation":"Used by ivar trim and ivar removereads for assigning primers to amplicons. Should have one line of tab-separated primer names per amplicon. Can be retrieved from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Used by ivar trim and ivar removereads for assigning primers to amplicons. Should have one line of tab-separated primer names per amplicon. Can be retrieved from https://usegalaxy.eu/u/wolfgang-maier/h/covid-19-resources","name":"ARTIC primers to amplicon assignments"}],"label":"ARTIC primers to amplicon assignments","name":"Input dataset","outputs":[],"position":{"left":961.625,"top":431.5625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"686cdb4d-a3da-468f-af92-8b04e1824ae0","when":null,"workflow_outputs":[]},"4":{"annotation":"Minimum allele-frequency required for a candidate primer binding site mutation to trigger amplicon removal. Variants with AF values below this threshold are treated as possible false-positives, which are not worth the coverage loss associated with amplicon removal.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Minimum allele-frequency required for a candidate primer binding site mutation to trigger amplicon removal. Variants with AF values below this threshold are treated as possible false-positives, which are not worth the coverage loss associated with amplicon removal.","name":"Read removal minimum AF"}],"label":"Read removal minimum AF","name":"Input parameter","outputs":[],"position":{"left":1093.265625,"top":46.640625},"tool_id":null,"tool_state":"{\\"default\\": 0.1, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"29f15ec0-8fa4-4476-b649-cf68ea096744","when":null,"workflow_outputs":[]},"5":{"annotation":"Maximum allele-frequency allowed for a primer binding site mutation to trigger amplicon removal. Variants with AF values above this threshold are treated as fixed variants, which won\'t generate amplicon bias.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Maximum allele-frequency allowed for a primer binding site mutation to trigger amplicon removal. Variants with AF values above this threshold are treated as fixed variants, which won\'t generate amplicon bias.","name":"Read removal maximum AF"}],"label":"Read removal maximum AF","name":"Input parameter","outputs":[],"position":{"left":1094.515625,"top":156.640625},"tool_id":null,"tool_state":"{\\"default\\": 1.0, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b13941c0-31c0-419d-a619-7474c24a582c","when":null,"workflow_outputs":[]},"6":{"annotation":"At any given variant site use the amplicon bias-corrected recall only if the depth of coverage of the site retains at least this value after amplicon removal.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"At any given variant site use the amplicon bias-corrected recall only if the depth of coverage of the site retains at least this value after amplicon removal.","name":"Minimum DP required after amplicon bias correction"}],"label":"Minimum DP required after amplicon bias correction","name":"Input parameter","outputs":[],"position":{"left":2171.5625,"top":0},"tool_id":null,"tool_state":"{\\"default\\": 1, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"e33b575d-e44e-451e-a2ba-5c253fdc9e5a","when":null,"workflow_outputs":[]},"7":{"annotation":"At any given variant site use the amplicon bias-corrected recall only if the depth of variant-supporting reads at the site retains at least this value after amplicon removal.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"At any given variant site use the amplicon bias-corrected recall only if the depth of variant-supporting reads at the site retains at least this value after amplicon removal.","name":"Minimum DP_ALT required after amplicon bias correction"}],"label":"Minimum DP_ALT required after amplicon bias correction","name":"Input parameter","outputs":[],"position":{"left":2172.34375,"top":117.796875},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"07e46388-c546-4425-8702-caf5d19667c1","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":8,"input_connections":{"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":217.765625,"top":618.734375},"post_job_actions":{"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\", \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"849d9cac-14b8-4e45-823d-5747709e8b60","when":null,"workflow_outputs":[{"label":"fastp_html_report","output_name":"report_html","uuid":"341566b1-181a-4bb0-9d0f-37db45003b22"},{"label":"fastp_json_report","output_name":"report_json","uuid":"04955ac5-df51-41ff-bd19-a0a99c7ef761"},{"label":"fastp_reads_output","output_name":"output_paired_coll","uuid":"8685689a-fbae-4a3e-b88b-7b4d212bafde"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":9,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"},"components_3|param_type|component_value":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1331.390625,"top":45.046875},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( ( DP4[2] + DP4[3] ) >= ( \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) ) & ( ( DP4[2] + DP4[3] ) <= ( \\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"d163b9b9-ee13-4b2b-8d64-b6d77758fac9","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":10,"input_connections":{"components_1|param_type|component_value":{"id":6,"output_name":"output"},"components_3|param_type|component_value":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":2400.96875,"top":39.765625},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( DP > \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" ) & ( ( AF * DP ) >= ( \\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" - 0.5 ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"258a1a99-145b-42a0-b858-5220b4199946","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","errors":null,"id":11,"input_connections":{"fastq_input|fastq_input1":{"id":8,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":455.09375,"top":352.5625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","tool_shed_repository":{"changeset_revision":"2477830927ec","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.18","type":"tool","uuid":"3b5d7080-c168-4bf7-946d-9045c0a4bc4c","when":null,"workflow_outputs":[{"label":"mapped_reads","output_name":"bam_output","uuid":"d9641fec-7639-47c3-881f-74165e0b5889"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":12,"input_connections":{"input":{"id":11,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":664.6875,"top":569.921875},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Mapped read pairs (filtered bwa-mem result)"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\"], \\"exclusive_filter\\": [\\"4\\", \\"8\\", \\"256\\"], \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"2f7744df-3811-4b7f-8f1f-cf8bf0ec7a0b","when":null,"workflow_outputs":[{"label":"filtered_mapped_reads","output_name":"outputsam","uuid":"1f75f9a7-8793-4c9e-b64d-5219a1e266db"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","errors":null,"id":13,"input_connections":{"reads":{"id":12,"output_name":"outputsam"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Realign reads","name":"reference_source"}],"label":null,"name":"Realign reads","outputs":[{"name":"realigned","type":"bam"}],"position":{"left":893.5,"top":616.53125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"aa35ee7f3ab2","name":"lofreq_viterbi","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"keepflags\\": false, \\"bq2_handling\\": {\\"replace_bq2\\": \\"keep\\", \\"__current_case__\\": 0, \\"defqual\\": \\"2\\"}}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"67b3fe84-0164-4494-9cfc-0bd0c86d7a2a","when":null,"workflow_outputs":[{"label":"realigned_primer_trimmed_filtered_mapped_reads","output_name":"realigned","uuid":"a9186bec-17b7-4993-89d8-dcaf5a0f89a9"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","errors":null,"id":14,"input_connections":{"input":{"id":12,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":918.8125,"top":922.90625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","tool_shed_repository":{"changeset_revision":"541082d03bef","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_group\\": null, \\"read_length\\": null, \\"remove_dups\\": false, \\"remove_overlaps\\": false, \\"sparse\\": false, \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"271efb08-46c0-4801-9869-b9948dfdebb4","when":null,"workflow_outputs":[{"label":"mapped_reads_stats","output_name":"output","uuid":"7d923dc5-84b5-4f32-90ce-5d97b87ec3e1"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1","errors":null,"id":15,"input_connections":{"reads":{"id":13,"output_name":"realigned"},"strategy|reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Insert indel qualities","outputs":[{"name":"output","type":"bam"}],"position":{"left":1122.875,"top":618.046875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1","tool_shed_repository":{"changeset_revision":"971e07ca4456","name":"lofreq_indelqual","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strategy\\": {\\"selector\\": \\"dindel\\", \\"__current_case__\\": 1, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy1","type":"tool","uuid":"70897728-2744-4a82-9b7d-38190594fec6","when":null,"workflow_outputs":[{"label":"realigned_primer_trimmed_filtered_mapped_reads_with_indel_quals","output_name":"output","uuid":"a3c94ad6-de7b-4b2f-8374-34c5bbff7d55"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0","errors":null,"id":16,"input_connections":{"amplicons|amplicon_info":{"id":3,"output_name":"output"},"input_bam":{"id":15,"output_name":"output"},"primer|input_bed":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ivar trim","name":"amplicons"},{"description":"runtime parameter for tool ivar trim","name":"primer"}],"label":null,"name":"ivar trim","outputs":[{"name":"output_bam","type":"bam"}],"position":{"left":1325.6875,"top":364.9375},"post_job_actions":{"RenameDatasetActionoutput_bam":{"action_arguments":{"newname":"Fully processed reads for variant calling (primer-trimmed, realigned reads with added indelquals)"},"action_type":"RenameDatasetAction","output_name":"output_bam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0","tool_shed_repository":{"changeset_revision":"7bd020da0ce5","name":"ivar_trim","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"amplicons\\": {\\"filter_by\\": \\"yes\\", \\"__current_case__\\": 1, \\"amplicon_info\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"inc_primers\\": true, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_len\\": \\"1\\", \\"min_qual\\": \\"0\\", \\"primer\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 0, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"primer_pos_wiggle\\": \\"0\\", \\"trimmed_length\\": {\\"filter\\": \\"auto\\", \\"__current_case__\\": 1}, \\"window_width\\": \\"4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.4+galaxy0","type":"tool","uuid":"7730e635-853f-4f9f-9451-bffbe6aedd15","when":null,"workflow_outputs":[{"label":"primer_trimmed_filtered_mapped_reads","output_name":"output_bam","uuid":"a03a0ced-394c-4b52-ab17-e596ed15469d"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","errors":null,"id":17,"input_connections":{"reads":{"id":16,"output_name":"output_bam"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Call variants","name":"reference_source"}],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"left":1560.453125,"top":527.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","tool_shed_repository":{"changeset_revision":"b11e8e9c23bf","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"5\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": false}, \\"bc_quals\\": {\\"min_bq\\": \\"30\\", \\"min_alt_bq\\": \\"30\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": true}}}, \\"map_quals\\": {\\"min_mq\\": \\"20\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_custom\\", \\"__current_case__\\": 3, \\"sig\\": \\"0.0005\\", \\"bonf\\": \\"0\\", \\"others\\": false}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy3","type":"tool","uuid":"422c7955-7768-4745-95db-b6882f37cd4b","when":null,"workflow_outputs":[{"label":"preliminary_variants_1","output_name":"variants","uuid":"f4a5e53e-f13e-419d-b729-a5177022406e"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0","errors":null,"id":18,"input_connections":{"input1":{"id":16,"output_name":"output_bam"}},"inputs":[],"label":null,"name":"QualiMap BamQC","outputs":[{"name":"raw_data","type":"input"},{"name":"output_html","type":"html"}],"position":{"left":1497.5625,"top":772},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0","tool_shed_repository":{"changeset_revision":"30a201c9c310","name":"qualimap_bamqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"duplicate_skipping\\": [\\"0\\"], \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"per_base_coverage\\": false, \\"plot_specific\\": {\\"n_bins\\": \\"400\\", \\"paint_chromosome_limits\\": true, \\"genome_gc_distr\\": null, \\"homopolymer_size\\": \\"3\\"}, \\"stats_regions\\": {\\"region_select\\": \\"all\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.3+galaxy0","type":"tool","uuid":"e2c69f7d-e6fd-4c9a-bbfd-05b70e5b2129","when":null,"workflow_outputs":[{"label":"bamqc_raw_output","output_name":"raw_data","uuid":"cd94a67a-edb4-4f54-b71a-ba6c138c8e1a"},{"label":"bamqc_html_output","output_name":"output_html","uuid":"2358cbeb-7bfc-434a-9b5e-decc2021cd8f"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":19,"input_connections":{"filter_expression|expr":{"id":9,"output_name":"out1"},"input":{"id":17,"output_name":"variants"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":1753.1875,"top":326.265625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"a2b784b0-28c1-45e3-93b8-d1d9d598de01","when":null,"workflow_outputs":[{"label":"filtered_preliminary_variants","output_name":"output","uuid":"f0948734-a0aa-4dfe-938f-98b63293898c"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":20,"input_connections":{"filter_expression|expr":{"id":10,"output_name":"out1"},"input":{"id":17,"output_name":"variants"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":2648.703125,"top":531.546875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"b96f89fd-30c1-4e70-ae70-ca37121e3c61","when":null,"workflow_outputs":[{"label":"preliminary_variants_1_filtered","output_name":"output","uuid":"e9c3d008-71df-43b6-919a-c482fc0572ee"}]},"21":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":21,"input_connections":{"input":{"id":18,"output_name":"raw_data"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1730.21875,"top":783.796875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"e4e04a37-f4bf-41d7-9470-5c96125ebc49","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.4.4+galaxy0","errors":null,"id":22,"input_connections":{"amplicons|amplicon_info":{"id":3,"output_name":"output"},"input_bam":{"id":16,"output_name":"output_bam"},"input_bed":{"id":2,"output_name":"output"},"variants_tsv":{"id":19,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ivar removereads","name":"amplicons"}],"label":null,"name":"ivar removereads","outputs":[{"name":"output_bam","type":"bam"}],"position":{"left":1949.546875,"top":48.96875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.4.4+galaxy0","tool_shed_repository":{"changeset_revision":"55a7867809f4","name":"ivar_removereads","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"amplicons\\": {\\"computed\\": \\"no\\", \\"__current_case__\\": 1, \\"amplicon_info\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"variants_tsv\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.4+galaxy0","type":"tool","uuid":"af7d3b75-d16c-43c7-82e4-3f1388edd22c","when":null,"workflow_outputs":[{"label":"amplicon_removal_output","output_name":"output_bam","uuid":"abe39037-7f89-4b93-9431-b0287bcee2d6"}]},"23":{"annotation":"","content_id":"__FLATTEN__","errors":null,"id":23,"input_connections":{"input":{"id":21,"output_name":"output"}},"inputs":[],"label":null,"name":"Flatten collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1955.65625,"top":789.5},"post_job_actions":{},"tool_id":"__FLATTEN__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"join_identifier\\": \\"_\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"396c9b85-8ec7-405a-a742-5b47e40cf0de","when":null,"workflow_outputs":[{"label":"bamqc_raw_output_flattened","output_name":"output","uuid":"b3942cb8-ac12-4b37-88d2-6c25dc58b636"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","errors":null,"id":24,"input_connections":{"reads":{"id":22,"output_name":"output_bam"},"reference_source|ref":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Call variants","name":"reference_source"}],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"left":2204.984375,"top":334.296875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","tool_shed_repository":{"changeset_revision":"b11e8e9c23bf","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"5\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": false}, \\"bc_quals\\": {\\"min_bq\\": \\"30\\", \\"min_alt_bq\\": \\"30\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": true}}}, \\"map_quals\\": {\\"min_mq\\": \\"20\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_custom\\", \\"__current_case__\\": 3, \\"sig\\": \\"0.0005\\", \\"bonf\\": \\"0\\", \\"others\\": false}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy3","type":"tool","uuid":"bef93d5b-c023-459c-888c-ac3382515322","when":null,"workflow_outputs":[{"label":"preliminary_variants_2","output_name":"variants","uuid":"6a59b687-340f-4ef6-b908-31fd69bd081e"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":25,"input_connections":{"results_0|software_cond|input":{"id":8,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":14,"output_name":"output"},"results_2|software_cond|input":{"id":23,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"plots","type":"input"},{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2186.515625,"top":925.765625},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": true, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"qualimap\\", \\"__current_case__\\": 20, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"ee2b164f-213b-4d95-9ab7-febdf5a71085","when":null,"workflow_outputs":[{"label":"preprocessing_and_mapping_reports","output_name":"html_report","uuid":"935dfa12-2c9a-4e44-85fa-a2daa3e04683"},{"label":"preprocessing_and_mapping_plots","output_name":"plots","uuid":"97cbd06a-8be5-4e1f-802b-7928c94de928"}]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy4","errors":null,"id":26,"input_connections":{"input_file":{"id":17,"output_name":"variants"},"sec_annofile|annofile|annotations":{"id":24,"output_name":"variants"}},"inputs":[{"description":"runtime parameter for tool bcftools annotate","name":"sec_annotate"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_annotate"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_restrict"}],"label":null,"name":"bcftools annotate","outputs":[{"name":"output_file","type":"vcf"}],"position":{"left":2409.21875,"top":689.4375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy4","tool_shed_repository":{"changeset_revision":"2c372041ed8c","name":"bcftools_annotate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_type\\": \\"v\\", \\"sec_annofile\\": {\\"columns\\": \\"QUAL,INFO\\", \\"annofile\\": {\\"anno_fmt\\": \\"vcf\\", \\"__current_case__\\": 1, \\"annotations\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"mark_sites\\": \\"-AmpliconBias\\", \\"min_overlap\\": \\"\\", \\"set_id\\": \\"\\"}, \\"sec_annotate\\": {\\"remove\\": \\"\\", \\"rename_chrs\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"rename_annots\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"sec_restrict\\": {\\"include\\": \\"\\", \\"exclude\\": \\"\\", \\"collapse\\": null, \\"regions\\": {\\"regions_src\\": \\"__none__\\", \\"__current_case__\\": 0}, \\"regions_overlap\\": null, \\"samples\\": \\"\\", \\"invert_samples\\": false, \\"samples_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"invert_samples_file\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy4","type":"tool","uuid":"c466b7da-f50c-4093-9eb5-c014ba41c1b8","when":null,"workflow_outputs":[{"label":"variants_fixed_partial","output_name":"output_file","uuid":"7c918bc9-7234-497c-9349-e977bc261510"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":27,"input_connections":{"filter_expression|expr":{"id":10,"output_name":"out1"},"input":{"id":24,"output_name":"variants"}},"inputs":[{"description":"runtime parameter for tool SnpSift Filter","name":"filter_expression"}],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":2655.703125,"top":349.046875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"c3d5ad01-385e-4405-bdc1-51f01dee4071","when":null,"workflow_outputs":[{"label":"preliminary_variants_2_filtered","output_name":"output","uuid":"b7f7c28f-1a7c-4098-bb84-b82fbfea2321"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/vcfvcfintersect/vcfvcfintersect/1.0.0_rc3+galaxy0","errors":null,"id":28,"input_connections":{"reference_source|ref_file":{"id":1,"output_name":"output"},"vcf_input1":{"id":27,"output_name":"output"},"vcf_input2":{"id":20,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool VCF-VCFintersect:","name":"reference_source"}],"label":null,"name":"VCF-VCFintersect:","outputs":[{"name":"out_file1","type":"vcf"}],"position":{"left":2885.859375,"top":501},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/vcfvcfintersect/vcfvcfintersect/1.0.0_rc3+galaxy0","tool_shed_repository":{"changeset_revision":"166c4f50525f","name":"vcfvcfintersect","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"adv_options_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"invert\\": true, \\"isect_union\\": \\"-i\\", \\"loci\\": false, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"vcf_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"vcf_input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"window_size\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0_rc3+galaxy0","type":"tool","uuid":"816eeae7-4ad5-4f2d-b709-285931d2be4b","when":null,"workflow_outputs":[{"label":"lost_filter_passing_variants","output_name":"out_file1","uuid":"2b336081-1278-4a1d-9991-58d0d33adfcf"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy4","errors":null,"id":29,"input_connections":{"input_file":{"id":26,"output_name":"output_file"},"sec_annofile|annofile|annotations":{"id":28,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bcftools annotate","name":"sec_annotate"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_annotate"},{"description":"runtime parameter for tool bcftools annotate","name":"sec_restrict"}],"label":null,"name":"bcftools annotate","outputs":[{"name":"output_file","type":"vcf"}],"position":{"left":3112.953125,"top":656.28125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy4","tool_shed_repository":{"changeset_revision":"2c372041ed8c","name":"bcftools_annotate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_type\\": \\"v\\", \\"sec_annofile\\": {\\"columns\\": \\"QUAL,INFO\\", \\"annofile\\": {\\"anno_fmt\\": \\"vcf\\", \\"__current_case__\\": 1, \\"annotations\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"mark_sites\\": \\"+AmpliconBias\\", \\"min_overlap\\": \\"\\", \\"set_id\\": \\"\\"}, \\"sec_annotate\\": {\\"remove\\": \\"\\", \\"rename_chrs\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"rename_annots\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"sec_restrict\\": {\\"include\\": \\"\\", \\"exclude\\": \\"\\", \\"collapse\\": null, \\"regions\\": {\\"regions_src\\": \\"__none__\\", \\"__current_case__\\": 0}, \\"regions_overlap\\": null, \\"samples\\": \\"\\", \\"invert_samples\\": false, \\"samples_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"invert_samples_file\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy4","type":"tool","uuid":"c466b7da-f50c-4093-9eb5-c014ba41c1b7","when":null,"workflow_outputs":[{"label":"variants_fixed","output_name":"output_file","uuid":"d06d1e97-0016-48f7-9f93-4d68a0d99dde"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":30,"input_connections":{"infile":{"id":29,"output_name":"output_file"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3338.28125,"top":745.546875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^##INFO=<ID=AmpliconBias,.+$\\", \\"replace_pattern\\": \\"##INFO=<ID=AmpliconBias,Number=0,Type=Flag,Description=\\\\\\"Indicates that the AF value of the variant could not be corrected for potential amplicon bias.\\\\\\">\\", \\"sed_options\\": null}, {\\"__index__\\": 1, \\"find_pattern\\": \\"^##INFO=<ID=AF,.+$\\", \\"replace_pattern\\": \\"##INFO=<ID=AF,Number=1,Type=Float,Description=\\\\\\"Lofreq Allele Frequency; Fraction of variant-supporting bases with q > --min-bq among all bases at the site\\\\\\">\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"490dbabb-04ee-426d-a8e5-e2ec9e59de3a","when":null,"workflow_outputs":[{"label":"variants_fixed_header","output_name":"outfile","uuid":"d96147a4-6d8b-426f-9c51-f77efc81a08b"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","errors":null,"id":31,"input_connections":{"input":{"id":30,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":"SnpEff eff covid19 version","name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"left":3566.25,"top":598.96875},"post_job_actions":{"RenameDatasetActionsnpeff_output":{"action_arguments":{"newname":"Final (SnpEff-) annotated variants"},"action_type":"RenameDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19","tool_shed_repository":{"changeset_revision":"2a3a00c1fa0a","name":"snpeff_sars_cov_2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": false, \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-upstream\\"], \\"generate_stats\\": true, \\"genome_version\\": \\"NC_045512.2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": true, \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5covid19","type":"tool","uuid":"a6b0b8e1-a914-4f59-95d4-87fc47ed4171","when":null,"workflow_outputs":[{"label":"annotated_variants","output_name":"snpeff_output","uuid":"ded982f4-601c-4d3c-888d-92716a47bb5c"},{"label":"annotated_variants_stats","output_name":"statsFile","uuid":"890819cb-f5a8-416d-a6f7-39a448fd2cd4"}]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","errors":null,"id":32,"input_connections":{"invcf":{"id":31,"output_name":"snpeff_output"}},"inputs":[],"label":null,"name":"Lofreq filter","outputs":[{"name":"outvcf","type":"vcf"}],"position":{"left":3802.5,"top":614.625},"post_job_actions":{"RenameDatasetActionoutvcf":{"action_arguments":{"newname":"Final (SnpEff-) annotated variants with strand-bias soft filter applied"},"action_type":"RenameDatasetAction","output_name":"outvcf"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"7dfca164d2e3","name":"lofreq_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"af\\": {\\"af_min\\": \\"0.0\\", \\"af_max\\": \\"0.0\\"}, \\"coverage\\": {\\"cov_min\\": \\"0\\", \\"cov_max\\": \\"0\\"}, \\"filter_by_type\\": {\\"keep_only\\": \\"\\", \\"__current_case__\\": 0, \\"qual\\": {\\"snvqual_filter\\": {\\"snvqual\\": \\"no\\", \\"__current_case__\\": 0}, \\"indelqual_filter\\": {\\"indelqual\\": \\"no\\", \\"__current_case__\\": 0}}}, \\"flag_or_drop\\": \\"--print-all\\", \\"invcf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sb\\": {\\"sb_filter\\": {\\"strand_bias\\": \\"mtc\\", \\"__current_case__\\": 2, \\"sb_alpha\\": \\"0.001\\", \\"sb_mtc\\": \\"fdr\\", \\"sb_compound\\": true, \\"sb_indels\\": false}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"4c9e8621-e930-4593-b0be-d84304f86ad4","when":null,"workflow_outputs":[{"label":"annotated_softfiltered_variants","output_name":"outvcf","uuid":"7af82fbc-67e3-46ca-9b17-95de59de8cf5"}]}},"tags":["COVID-19","ARTIC","covid19.galaxyproject.org"],"uuid":"864427b0-d25a-499c-b437-f83f2d6b83f2"},"readme":"COVID-19: variation analysis on ARTIC PE data\\n---------------------------------------------\\n\\nThe workflow for Illumina-sequenced ampliconic data builds on the RNASeq workflow\\nfor paired-end data using the same steps for mapping and variant calling, but\\nadds extra logic for trimming amplicon primer sequences off reads with the ivar\\npackage. In addition, this workflow uses ivar also to identify amplicons\\naffected by primer-binding site mutations and, if possible, excludes reads\\nderived from such \\"tainted\\" amplicons when calculating allele-frequencies\\nof other variants.\\n","changelog":"# Changelog\\n\\n## [0.5.4] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.4.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n\\n## [0.5.3] - 2024-10-01\\n\\n### Tool updates\\n\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0` updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n\\n## [0.5.2] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2c+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.3.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_removereads/ivar_removereads/1.4.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.10` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_annotate/bcftools_annotate/1.15.1+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n\\n## [0.5.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n- Use zenodo link instead of googleapis for fastq in test\\n- Update output test\\n\\n## [0.5] - 2022-02-08\\n\\n### Fixed\\n\\n- Base selection of variants that should trigger amplicon removal on unbiased\\n  AF values recalculated from DP4 and DP fields instead of on AF values\\n  provided by lofreq call.\\n\\n  https://github.com/CSB5/lofreq/issues/80 means that lofreq-calculated AF\\n  values are lower bounds of true AFs when bases are excluded from calling\\n  based on base quality. The extent of AF underestimation depends on the\\n  fraction of bases with sub-threshold (30 for this workflow) base qualities.\\n\\n  By recalculating AFs as (DP4[2] + DP4[3]) / DP we are avoiding this issue.\\n  From this version on we also add a proper description to the header INFO\\n  line description of the AF field to be explicit about the meaning of lofreq\'s\\n  AF value.\\n\\n### Changed\\n\\n- Increase/deactivate the default upper AF threshold for biased amplicon detection\\n\\n  By increasing the default AF threshold to 1.0 amplicon removal now gets\\n  triggered by default for all primer binding site mutations with unbiased\\n  (see AF discussion above) AF > 0.1.\\n\\n  This change is intended to allow removal of amplicons resulting from even\\n  trace amounts of contamination when the intended target of the amplicon\\n  primers drops out because of mutations severely impacting primer binding.\\n  The expectation is that this would either remove contamination-contributed\\n  variants (like contributed by traces of delta virus in omicron preparations\\n  in omicron-triggered amplicon dropout regions), or at least cause them\\n  getting flagged as AmpliconBias calls at the VCF level.\\n  The exact consequences of this change need to be evaluated, but users can\\n  restore the previous behavior by reducing the upper AF threshold back to 0.9,\\n  which will reduce the extent of attempted amplicon removals substantially.\\n\\n- Make the amplicon bias correction more robust and better interoperable with\\n  the [Reporting workflow](https://github.com/iwc-workflows/sars-cov-2-variation-reporting).\\n\\n  First and second (after amplicon removal) round of variant calling are now\\n  carried out with identical lofreq parameter settings.\\n  bcftools annotate is then used to carry over the bias-corrected call stats to\\n  the variant calls obtained in the first round. At the same time, both variant\\n  call lists are filtered with identical DP and DP_ALT filters and stats of\\n  filter-passing variants from the first round that fail to pass those filters\\n  after the second round of calling aretransferred back to the initial\\n  bcftools annotate output.\\n  If the DP and DP_ALT thresholds are chosen as in the Reporting workflow (as\\n  is the case with the default settings), this ensures that no initially called\\n  variant gets lost as a consequence of amplicon bias correction and that no\\n  initially filter-passing variant gets filtered out after correction.\\n  The AmpliconBias INFO flag is used to mark all such variants, for which\\n  amplicon bias correction was skipped to rescue the call.\\n\\n- Upgrade the Galaxy wrapper versions of ivar trim and ivar removereads.\\n\\n  This makes it easy for users to calculate primer amplicon info from suitable\\n  primer scheme bed files instead of passing the info as a separate file.\\n  This workflow sticks to the previous behavior to avoid new requirements on\\n  primer names.\\n- Upgrade fastp to 0.23.2.\\n  This update restores plots contained in the tools html output, should\\n  provide better performance for compressed input data, but also has a moderate\\n  effect on trimming results.\\n- Upgrade other tools to their latest versions or wrapper versions:\\n\\n  - bwa_mem to wrapper version 0.7.17.2\\n  - lofreq_call to wrapper version 2.1.5+galaxy1\\n  - multiqc to version 1.11\\n\\n  None of these are expected to change the variant output produced by the\\n  workflow.\\n\\n### Added\\n\\n- Add a step to filter out failed datasets before flattening the Qualimap BamQC\\n  data for use by MultiQC.\\n\\n  Qualimap BamQC fails on empty BAM input and trying to flatten the resulting\\n  collection containing failed datasets would cause the invocation of the\\n  workflow to fail.\\n\\n## [0.4.2] - 2021-12-13\\n\\n### Added\\n\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.4.1] - 2021-07-23\\n\\n### Added\\n\\nAdded RO-Crate metadata file. No functional changes.\\n\\n## [0.4] - 2021-06-16\\n\\n### Changed\\n\\n- Upgrade multiqc to 1.9+galaxy1\\n\\n## [0.3] - 2021-05-19\\n\\n### Changed\\n\\nThis version brings a number of tweaks to the ivar-dependent steps of the\\nworkflow. Together, these are expected to make variant allele frequency\\ncalculations more precise, in general, and robust in the face of an increasing\\nnumber of variants at primer binding sites:\\n\\n- Upgrade ivar from version 1.2.2 to 1.3.1\\n  This affects ivar trim and ivar removereads\\n- Use the newly introduced -f option of ivar trim to exclude read pairs from\\n  further analysis that extend beyond amplicon boundaries.\\n  This change should be benefitial for accurate AF calculations in general,\\n  but in particular for corrected AF values after removal of biased amplicons,\\n  where aberrant read pairs often represent a larger fraction of the remaining\\n  reads.\\n- Run ivar trim only after realignment and addition of indel qualities by\\n  lofeq. This should make sure that indels close to primer sequences are\\n  seen as read-internal events.\\n- Turn the lower and upper thresholds for variant AF that triggers readremoval\\n  into workflow input parameters and adjust their defaults to trigger read\\n  removal only in more obvious cases of non-fixed variants.\\n- Require a minimum depth of coverage for recalled variants after read removal\\n  of 20 to ensure reliable AF values.\\n  This change also prevents situations where variants are recalled successfully\\n  after read removal, but are later excluded from variant reports generated by\\n  the reporting workflow due to that workflow\'s min_dp_alt >= 10 filter.\\n\\n## [0.2]\\n\\n### Changed\\n\\n- Turn the AmpliconRemoval variant FILTER into an AmpliconBias INFO flag\\n- Apply the strand-bias filter only after variant annotation with snpEff. By\\n  producing fully annotated VCFs with and without filtering, downstream\\n  workflows can easily be switched between filtered/unfiltered input data\\n\\n### Fixed\\n\\n- Make sure the header information about the added flag gets propagated to the\\n  final VCF\\n\\n## [0.1]\\n\\n- Initial version of COVID-19: variation analysis on ARTIC PE data workflow\\n","diagrams":"# Workflow diagrams\\n\\n## COVID-19: variation analysis on ARTIC PE data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ NC_045512.2 FASTA sequence of SARS-CoV-2\\"]@{ shape: doc }\\n2[\\"ℹ️ ARTIC primer BED\\"]@{ shape: doc }\\n3[\\"ℹ️ ARTIC primers to amplicon assignments\\"]@{ shape: doc }\\n4[\\"ℹ️ Read removal minimum AF\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Read removal maximum AF\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Minimum DP required after amplicon bias correction\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Minimum DP_ALT required after amplicon bias correction\\"]@{ shape: lean-l }\\n8[\\"fastp\\"]@{ shape: process }\\n0 --> 8\\n9[\\"Compose text parameter value\\"]@{ shape: process }\\n4 --> 9\\n5 --> 9\\n10[\\"Compose text parameter value\\"]@{ shape: process }\\n6 --> 10\\n7 --> 10\\n11[\\"Map with BWA-MEM\\"]@{ shape: process }\\n8 --> 11\\n1 --> 11\\n12[\\"Samtools view\\"]@{ shape: process }\\n11 --> 12\\n13[\\"Realign reads\\"]@{ shape: process }\\n12 --> 13\\n1 --> 13\\n14[\\"Samtools stats\\"]@{ shape: process }\\n12 --> 14\\n15[\\"Insert indel qualities\\"]@{ shape: process }\\n13 --> 15\\n1 --> 15\\n16[\\"ivar trim\\"]@{ shape: process }\\n3 --> 16\\n15 --> 16\\n2 --> 16\\n17[\\"Call variants\\"]@{ shape: process }\\n16 --> 17\\n1 --> 17\\n18[\\"QualiMap BamQC\\"]@{ shape: process }\\n16 --> 18\\n19[\\"SnpSift Filter\\"]@{ shape: process }\\n9 --> 19\\n17 --> 19\\n20[\\"SnpSift Filter\\"]@{ shape: process }\\n10 --> 20\\n17 --> 20\\n21[\\"Filter failed datasets\\"]@{ shape: process }\\n18 --> 21\\n22[\\"ivar removereads\\"]@{ shape: process }\\n3 --> 22\\n16 --> 22\\n2 --> 22\\n19 --> 22\\n23[\\"Flatten collection\\"]@{ shape: process }\\n21 --> 23\\n24[\\"Call variants\\"]@{ shape: process }\\n22 --> 24\\n1 --> 24\\n25[\\"MultiQC\\"]@{ shape: process }\\n8 --> 25\\n14 --> 25\\n23 --> 25\\n26[\\"bcftools annotate\\"]@{ shape: process }\\n17 --> 26\\n24 --> 26\\n27[\\"SnpSift Filter\\"]@{ shape: process }\\n10 --> 27\\n24 --> 27\\n28[\\"VCF-VCFintersect:\\"]@{ shape: process }\\n1 --> 28\\n27 --> 28\\n20 --> 28\\n29[\\"bcftools annotate\\"]@{ shape: process }\\n26 --> 29\\n28 --> 29\\n30[\\"Replace Text\\"]@{ shape: process }\\n29 --> 30\\n31[\\"SnpEff eff covid19 version\\"]@{ shape: process }\\n30 --> 31\\n32[\\"Lofreq filter\\"]@{ shape: process }\\n31 --> 32\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-variant-calling/COVID-19-PE-ARTIC-ILLUMINA","dockstore_id":16247,"categories":["COVID-19"],"collections":["SARS-COV-2","Variant Calling"],"doi":"10.5281/zenodo.15078877","tests":[{"doc":"Test workflow execution using paired end illumina ARTIC accession","job":{"NC_045512.2 FASTA sequence of SARS-CoV-2":{"class":"File","location":"https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1","filetype":"fasta"},"ARTIC primer BED":{"class":"File","location":"https://zenodo.org/record/4555735/files/ARTIC_nCoV-2019_v3.bed?download=1","filetype":"bed"},"ARTIC primers to amplicon assignments":{"class":"File","location":"https://zenodo.org/record/4555735/files/ARTIC_amplicon_info_v3.tsv?download=1","filetype":"tabular"},"Paired Collection":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"SRR11578257","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/10174466/files/SRR11578257_R1.fastq.gz?download=1"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/10174466/files/SRR11578257_R2.fastq.gz?download=1"}]}]}},"outputs":{"annotated_softfiltered_variants":{"attributes":{},"element_tests":{"SRR11578257":{"asserts":{"has_line":{"line":"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tFILTER\\tINFO"},"has_text_matching":{"expression":"NC_045512.2\\t20209\\t.\\tAGTAGAAATT\\tA\\t[0-9]*\\tPASS\\tDP=[0-9]*;AF=0.9[0-9]*;SB=24;DP4=[0-9]*,[0-9]*,[0-9]*,[0-9]*;INDEL;HRUN=1;EFF=CODON_CHANGE_PLUS_CODON_DELETION\\\\(MODERATE||agtagaaattta/ata|SRNL6649I|7096|ORF1ab|protein_coding|CODING|GU280_gp01|2|A\\\\),CODON_CHANGE_PLUS_CODON_DELETION\\\\(MODERATE||agtagaaattta/ata|SRNL197I|345|ORF1ab|protein_coding|CODING|YP_009725310.1|1|A|WARNING_TRANSCRIPT_NO_START_CODON\\\\)"}}}}}}}]}],"path":"./workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-variant-calling"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/taxonomic-rank-abundance-summary-table.ga","testParameterFiles":["/taxonomic-rank-abundance-summary-table-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053","url":"https://github.com/RZ9082"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow creates taxonomic summary tables for a specified taxonomic rank out of MAPseq\'s OTU tables output collection.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Taxonomic abundance summary tables for a specified taxonomic rank","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":25.297647871305003},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9a5f6c83-e6c6-4b46-a78a-2890ec4e736a","when":null,"workflow_outputs":[]},"1":{"annotation":"Choose a taxonomic rank for a dedicated taxonomic abundance summary table.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Choose a taxonomic rank for a dedicated taxonomic abundance summary table.","name":"Taxonomic rank"}],"label":"Taxonomic rank","name":"Input parameter","outputs":[],"position":{"left":9.007353607619836,"top":235.5333251953125},"tool_id":null,"tool_state":"{\\"default\\": \\"phylum\\", \\"validators\\": [], \\"restrictions\\": [\\"superkingdom\\", \\"kingdom\\", \\"phylum\\", \\"class\\", \\"order\\", \\"family\\", \\"genus\\", \\"species\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"acd7e798-86d0-458b-bce9-ca17ccf7bcd7","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":2,"input_connections":{"input_param_type|input_param":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":null,"name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":312.01935360761985,"top":253.5333251953125},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"superkingdom\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 1;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 1, \\"from\\": \\"kingdom\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 2;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 2, \\"from\\": \\"phylum\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 3;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 3, \\"from\\": \\"class\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 4;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 4, \\"from\\": \\"order\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 5;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 5, \\"from\\": \\"family\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 6;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 6, \\"from\\": \\"genus\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 7;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}, {\\"__index__\\": 7, \\"from\\": \\"species\\", \\"to\\": \\"NR > 2 {          split($3, taxa, \\\\\\";\\\\\\");         merged = \\\\\\"\\\\\\";         levels = 8;         for (i = 1; i <= levels; i++) {             if (i <= length(taxa) && taxa[i] ~ /__/) {                 split(taxa[i], parts, \\\\\\"__\\\\\\");                 if (parts[2] != \\\\\\"\\\\\\") {                     value = parts[2];                 } else {                     value = \\\\\\"unassigned\\\\\\";                 }             } else {                 value = \\\\\\"unassigned\\\\\\";             }             merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;         }         print merged \\\\\\"\\\\\\\\t\\\\\\" $2;     }\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"4489644f-e82c-4b70-9cf0-1f90c3bcbdd0","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":3,"input_connections":{"input_param_type|input_param":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":null,"name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":303.0073536076199,"top":463.5333251953125},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"superkingdom\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 1; i++) { if (i < 1) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 1, \\"from\\": \\"kingdom\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 2; i++) { if (i < 2) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 2, \\"from\\": \\"phylum\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 3; i++) { if (i < 3) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 3, \\"from\\": \\"class\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\\\tclass\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 4; i++) { if (i < 4) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 4, \\"from\\": \\"order\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\\\tclass\\\\\\\\torder\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 5; i++) { if (i < 5) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 5, \\"from\\": \\"family\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\\\tclass\\\\\\\\torder\\\\\\\\tfamily\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 6; i++) { if (i < 6) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 6, \\"from\\": \\"genus\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\\\tclass\\\\\\\\torder\\\\\\\\tfamily\\\\\\\\tgenus\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 7; i++) { if (i < 7) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}, {\\"__index__\\": 7, \\"from\\": \\"species\\", \\"to\\": \\"BEGIN { FS = OFS = \\\\\\"\\\\\\\\t\\\\\\" } NR == 1 { printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\\\tclass\\\\\\\\torder\\\\\\\\tfamily\\\\\\\\tgenus\\\\\\\\tspecies\\\\\\"; for (i = 2; i <= NF; i++) { split($i, name_parts, \\\\\\"_\\\\\\"); printf \\\\\\"\\\\\\\\t%s\\\\\\", name_parts[1] } printf \\\\\\"\\\\\\\\n\\\\\\" } NR > 1 { split($1, taxa_parts, \\\\\\";\\\\\\"); for (i = 1; i <= 8; i++) { if (i < 8) { printf \\\\\\"%s\\\\\\\\t\\\\\\", taxa_parts[i] } else { printf \\\\\\"%s\\\\\\", taxa_parts[i] } } for (i = 2; i <= NF; i++) { if (i < NF) { printf \\\\\\"\\\\\\\\t%s\\\\\\", $i } else { printf \\\\\\"\\\\\\\\t%s\\\\\\\\n\\\\\\", $i } } }\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"9c63fe51-fe0c-4f92-8aee-fc412e272f59","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":4,"input_connections":{"code":{"id":2,"output_name":"output_param_text"},"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":299.74718715848013,"top":1.7389953573637058},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"7cf9ee51-56ef-4270-8d30-f13ef88cfbc8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"Grouping1","errors":null,"id":5,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":520.5962466215664,"top":3.110728645954282},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"80ff4e60-4b7e-40d7-b375-7ec7ee7b16f0","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":6,"input_connections":{"input":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":745.5605059504245,"top":3.110728645954282},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"428d0d7f-339e-4f48-9420-1259bae0de17","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":7,"input_connections":{"infile":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.4095654135107,"top":1.7389953573637058},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b52578ff-8185-42db-b3a5-d04601217243","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":8,"input_connections":{"input_tabular":{"id":7,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1187.2586248765972,"top":0.3672620687731296},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"fc14bb27-e0b9-4a8e-bc17-006deaef59bc","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"code":{"id":3,"output_name":"output_param_text"},"infile":{"id":8,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1414.969304779495,"top":0},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Taxonomic rank summary table"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"6f47b97a-8e69-4c3c-ba7b-c0470ef3c37e","when":null,"workflow_outputs":[{"label":"Taxonomic rank summary table","output_name":"outfile","uuid":"d44762e6-8306-4965-b146-d9ebb35f984c"}]}},"tags":["MGnify_Amplicon","amplicon","metagenomics","name:microgalaxy"],"uuid":"31b19088-dad2-41a0-b086-94037bce4784","version":4},"readme":"# Taxonomic rank abundance summary table\\n\\nThis subworkflow generates taxonomic abundance summary tables for a user-chosen taxonomic rank.\\n\\n## Input datasets\\n\\n- **OTU table collection** containing the taxonomic abundance tables of all datasets.\\n\\n## Outputs\\n\\n- Specified taxonomic level abundance summary","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## Taxonomic abundance summary tables for a specified taxonomic rank\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ OTU tables\\"]@{ shape: docs }\\n1[\\"ℹ️ Taxonomic rank\\"]@{ shape: lean-l }\\n2[\\"Map parameter value\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Map parameter value\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Text reformatting\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Group\\"]@{ shape: process }\\n4 --> 5\\n6[\\"Filter Tabular\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Column join\\"]@{ shape: process }\\n7 --> 8\\n9[\\"Text reformatting\\"]@{ shape: process }\\n3 --> 9\\n8 --> 9\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/taxonomic-rank-abundance-summary-table/main","dockstore_id":30524,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078789","tests":[{"doc":"Test outline for mgnify-amplicon-summary-tables.ga","job":{"OTU tables":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"ERR3046414_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR3046414_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR3046440_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR3046440_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR4319664_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR4319664_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR4319712_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR4319712_MERGED_FASTQ_SSU_OTU.tabular"}]},"Taxonomic rank":"family"},"outputs":{"Taxonomic rank summary table":{"asserts":[{"that":"has_size","size":11263},{"that":"has_text","text":"superkingdom\\tkingdom\\tphylum\\tclass\\torder\\tfamily\\tERR3046414\\tERR3046440\\tERR4319664\\tERR4319712"},{"that":"has_text","text":"Bacteria\\tunassigned\\tActinobacteria\\tAcidimicrobiia\\tunassigned\\tunassigned\\t0\\t0\\t1\\t3"},{"that":"has_text","text":"Bacteria\\tunassigned\\tActinobacteria\\tActinobacteria\\tActinomycetales\\tActinomycetaceae\\t4\\t1\\t0\\t0"},{"that":"has_n_lines","n":129},{"that":"has_n_columns","n":10}]}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/taxonomic-rank-abundance-summary-table"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-pipeline-v5-its.ga","testParameterFiles":["/mgnify-amplicon-pipeline-v5-its-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X"},{"name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"}],"definition":{"a_galaxy_workflow":"true","annotation":"Classification and visualization of ITS regions.","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - ITS","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A BED file containing the LSU and SSU coordinates.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A BED file containing the LSU and SSU coordinates.","name":"LSU and SSU BED"}],"label":"LSU and SSU BED","name":"Input dataset collection","outputs":[],"position":{"left":289.95001220703125,"top":225.9127507839193},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"c17b5c1a-8aac-4e43-be86-11ce0d59779b","when":null,"workflow_outputs":[]},"1":{"annotation":"Processed sequences.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Processed sequences.","name":"Processed sequences"}],"label":"Processed sequences","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":658.1099407040649},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9c72651d-0c23-41bb-8720-f9e603a4726e","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":730.4937643933138,"top":202.9418588363548},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"85b4e860-a20a-4dfa-8831-26a0819975d8","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":3,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":187.50176439331375,"top":493.9418588363548},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"9a909506-5071-466a-8f5b-63cd2bb81254","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":4,"input_connections":{"how|filter_source":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":514.0843031728684,"top":525.9078034884216},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"25bbbc01-9c16-484c-925a-6758ac7e8c08","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_maskfastabed/2.31.1","errors":null,"id":5,"input_connections":{"fasta":{"id":4,"output_name":"output_filtered"},"input":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"bedtools MaskFastaBed","outputs":[{"name":"output","type":"fasta"}],"position":{"left":788.2104391980013,"top":514.7805968487736},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_maskfastabed/2.31.1","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fullheader\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mc\\": \\"N\\", \\"soft\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"7448f5ef-6091-4e73-91fb-dfb97b657aa2","when":null,"workflow_outputs":[{"label":"ITS FASTA files","output_name":"output","uuid":"0527bde7-8e9d-49f1-88d1-b6e4095e2095"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":6,"input_connections":{"sequences":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":1071.4000244140625,"top":429.3085275375267},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format ITSonedb"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid ITSonedb"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_its_itsonedb_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"8ba4b427-6376-44b4-b77a-8cb07dd7daef","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":7,"input_connections":{"sequences":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":1091.1791513437531,"top":944.89646009585},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format ITS UNITE"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid ITS UNITE"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_its_unite_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"5978536f-8c00-49a7-b426-8120f23a3875","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":8,"input_connections":{"infile":{"id":6,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1364.155596001663,"top":122.12686327418507},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"12a9401c-95a3-4da6-8ca6-ced17b6b3b70","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"infile":{"id":6,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1371.014262444616,"top":339.04363878350205},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"94df1cca-c1c3-4643-94ff-dca5cbca8467","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":10,"input_connections":{"infile":{"id":6,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1726.2815522788949,"top":455.65530978203265},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"3d0ba884-d155-4bf6-990c-f589898bae51","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":11,"input_connections":{"input":{"id":6,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1820.6340128732131,"top":615.8965977845029},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"85c966df-4f74-4c8b-81ba-b4f73829c18b","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":12,"input_connections":{"infile":{"id":7,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1314.1580115527963,"top":890.3766404571136},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"165a1fc2-87da-4a8c-8e5c-16fb4d9e84ce","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":13,"input_connections":{"infile":{"id":7,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1280.5052342467761,"top":1234.4196311901392},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"87917797-f8cc-4670-b3b1-28b571fc9e07","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":14,"input_connections":{"infile":{"id":7,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1701.3738942940215,"top":1121.663903849118},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"a2784daf-528d-49a6-a183-acb2c56b3c5e","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":15,"input_connections":{"input":{"id":7,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1943.182478535459,"top":1424.172257451561},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"447e2f51-931c-428d-97fd-0e3e03ee3737","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":16,"input_connections":{"input":{"id":8,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1578.1459890217961,"top":123.20139745137587},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic classifications using ITSoneDB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"974b7a42-0e82-4e0e-8630-3827185e275b","when":null,"workflow_outputs":[{"label":"ITS taxonomic classifications using ITSoneDB","output_name":"output","uuid":"26936de2-fb56-493a-89fe-dbcd1c98df2c"}]},"17":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":17,"input_connections":{"input":{"id":9,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1624.7849208338766,"top":294.6680585252006},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS OTU tables (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d0054748-c9d4-4f29-8640-818060d1a8ff","when":null,"workflow_outputs":[{"label":"ITS OTU tables (ITSoneDB)","output_name":"output","uuid":"030ee411-0b04-4ade-ba0c-715bbd561460"}]},"18":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":18,"input_connections":{"input":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1999.2564767084239,"top":452.431770043427},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"52c2edac-b2b7-434a-9feb-a5e0faa430fe","when":null,"workflow_outputs":[]},"19":{"annotation":"","id":19,"input_connections":{"Input collection":{"id":11,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2141.802455522449,"top":611.9414781218429},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"ac1892f3-2a86-4f96-b522-51132fede268"},"tool_id":null,"type":"subworkflow","uuid":"50932610-dbc1-4fa4-84af-e6c605206eda","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":20,"input_connections":{"input":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1573.7895192434253,"top":872.3032171286626},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic classifications using UNITE DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"c425bf1a-bda0-454c-8d3e-359144e49706","when":null,"workflow_outputs":[{"label":"ITS taxonomic classifications using UNITE DB","output_name":"output","uuid":"febb8ce1-72a0-44c8-a103-490bec12c30c"}]},"21":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":21,"input_connections":{"input":{"id":13,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1515.2157831570244,"top":1272.8548262411377},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS OTU tables (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"15cb2eee-88c1-43dc-a58c-211ed446bb99","when":null,"workflow_outputs":[{"label":"ITS OTU tables (UNITE DB)","output_name":"output","uuid":"8304da79-296c-4cfa-bd38-671cbe727354"}]},"22":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":22,"input_connections":{"input":{"id":14,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1973.1727179528675,"top":1141.672125434937},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"16b97913-9901-4b40-ac60-7b2c5e465b48","when":null,"workflow_outputs":[]},"23":{"annotation":"","id":23,"input_connections":{"Input collection":{"id":15,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2216.80260235576,"top":1483.9821779220883},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"3c05ccd3-5e5a-4ae6-842a-1fb99646e96d"},"tool_id":null,"type":"subworkflow","uuid":"e51d163f-6f26-4964-97a1-47d0aece539c","when":null,"workflow_outputs":[]},"24":{"annotation":"","id":24,"input_connections":{"Input collection":{"id":18,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2222.7347195492944,"top":271.7516225513746},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"84fd93e6-4154-4498-8437-ec9442a0971f"},"tool_id":null,"type":"subworkflow","uuid":"ebe35b8e-b72c-432a-ad92-f79e6b44ff21","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":25,"input_connections":{"type_of_data|input":{"id":11,"output_name":"output"},"when":{"id":19,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona ITSoneDB","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":2497.0678472201726,"top":673.2064707870127},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic abundance pie charts (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"d8f34e02-6822-443b-b4b4-938c06209006","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS taxonomic abundance pie charts (ITSoneDB)","output_name":"output","uuid":"c28aa604-a717-4744-acf9-aa0d4cb52af1"}]},"26":{"annotation":"","id":26,"input_connections":{"Input collection":{"id":22,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2223.6555735015286,"top":1091.0953467133058},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"710712b5-ed3c-4c54-bca2-79dbf11d1ce8"},"tool_id":null,"type":"subworkflow","uuid":"e5486218-e068-4d84-92b4-7dd0d3bf1b8f","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":27,"input_connections":{"type_of_data|input":{"id":15,"output_name":"output"},"when":{"id":23,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona ITS UNITE","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":2518.6399730672456,"top":1456.679375085977},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic abundance pie charts (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"526509f4-006f-4c4c-938b-f40d883fe7d2","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS taxonomic abundance pie charts (UNITE DB)","output_name":"output","uuid":"73a03453-b02c-4626-82a5-7d8268180b42"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":28,"input_connections":{"input|input_fp":{"id":18,"output_name":"output"},"when":{"id":24,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 ITSoneDB","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2489.545576441207,"top":0},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in HDF5 format (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"41a033f9-248d-46b6-8ac0-043b7d95d1df","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in HDF5 format (ITSoneDB)","output_name":"output_fp","uuid":"3f5626b9-9e0b-4efe-9e23-ddc7072dec7b"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":29,"input_connections":{"input|input_fp":{"id":18,"output_name":"output"},"when":{"id":24,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON ITSoneDB","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2491.8688075155605,"top":364.1607245502264},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in JSON format (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"7b86eed5-628f-4a38-ae64-d664e76ce54f","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in JSON format (ITSoneDB)","output_name":"output_fp","uuid":"3ef29ecf-ea57-457e-b244-cb7fea77203a"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":30,"input_connections":{"input|input_fp":{"id":22,"output_name":"output"},"when":{"id":26,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 ITS UNITE","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2511.9725936708223,"top":870.9977416620931},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in HDF5 format (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"5ae011cb-274f-48a9-90c5-b570d2893eb2","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in HDF5 format (UNITE DB)","output_name":"output_fp","uuid":"a51b6afd-2b7e-4e59-a4fa-73103d2366b0"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":31,"input_connections":{"input|input_fp":{"id":22,"output_name":"output"},"when":{"id":26,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON ITS UNITE","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2512.0796733881634,"top":1169.6892588090332},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in JSON format (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"d70deaeb-19f3-4323-8553-688d95be40fd","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in JSON format (UNITE DB)","output_name":"output_fp","uuid":"96a0362f-8eed-4577-820b-a4f341092b95"}]}},"tags":["MGnify_Amplicon","amplicon","metagenomics","name:microgalaxy"],"uuid":"04fa2e92-8fcd-4a20-8af1-063c7f514f12","version":28},"readme":"# MGnify\'s amplicon (v5.0) ITS subworkflow\\n\\nClassification and visualization of rRNA sequences based on ITS.\\n\\n## Input datasets\\n\\n- **Processed sequences** Post quality control\\n- **Clan information file** This file lists which models belong to the same clan `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.claninfo`\\n- **Covariance models** `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.cm`\\n\\n## Processing\\n\\nThis subworkflow executes the following classification and visualization steps:\\n\\n- **bedtools MaskFastaBed**  is used to mask specific regions in a FASTA file based on provided genomic intervals\\n- **MAPseq** Sequence read classification\\n- **biom-convert** Converts OTU tables to HDF5 and JSON formats\\n- **Krona** Generates pie charts out of the OTU tables","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify\'s amplicon pipeline v5.0 - ITS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ LSU and SSU BED\\"]@{ shape: docs }\\n1[\\"ℹ️ Processed sequences\\"]@{ shape: docs }\\n2[\\"Filter empty datasets\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Extract element identifiers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Filter collection\\"]@{ shape: process }\\n3 --> 4\\n1 --> 4\\n5[\\"bedtools MaskFastaBed\\"]@{ shape: process }\\n4 --> 5\\n2 --> 5\\n6[\\"MAPseq\\"]@{ shape: process }\\n5 --> 6\\n7[\\"MAPseq\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Filter empty datasets\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 12\\n13[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 13\\n14[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 14\\n15[\\"Filter empty datasets\\"]@{ shape: process }\\n7 --> 15\\n16[\\"Filter empty datasets\\"]@{ shape: process }\\n8 --> 16\\n17[\\"Filter empty datasets\\"]@{ shape: process }\\n9 --> 17\\n18[\\"Filter empty datasets\\"]@{ shape: process }\\n10 --> 18\\n19[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n11 --> 19\\n20[\\"Filter empty datasets\\"]@{ shape: process }\\n12 --> 20\\n21[\\"Filter empty datasets\\"]@{ shape: process }\\n13 --> 21\\n22[\\"Filter empty datasets\\"]@{ shape: process }\\n14 --> 22\\n23[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n15 --> 23\\n24[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n18 --> 24\\n25[\\"Krona ITSoneDB\\"]@{ shape: process }\\n11 --> 25\\n19 --> 25\\n26[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n22 --> 26\\n27[\\"Krona ITS UNITE\\"]@{ shape: process }\\n15 --> 27\\n23 --> 27\\n28[\\"HDF5 ITSoneDB\\"]@{ shape: process }\\n18 --> 28\\n24 --> 28\\n29[\\"JSON ITSoneDB\\"]@{ shape: process }\\n18 --> 29\\n24 --> 29\\n30[\\"HDF5 ITS UNITE\\"]@{ shape: process }\\n22 --> 30\\n26 --> 30\\n31[\\"JSON ITS UNITE\\"]@{ shape: process }\\n22 --> 31\\n26 --> 31\\n```\\n\\n## Map empty/not empty collection to boolean \\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input collection\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Line/Word/Character count\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Compute\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 4\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-its/main","dockstore_id":30445,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078466","tests":[{"doc":"Test outline for mgnify-amplicon-pipeline-v5-its.ga","job":{"LSU and SSU BED":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"DRR010481_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/DRR010481.bed"},{"class":"File","identifier":"ERR2715528_MERGED_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/ERR2715528.bed"}]},"Processed sequences":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"DRR010481_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/DRR010481_SE.fasta"},{"class":"File","identifier":"ERR2715528_MERGED_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/ERR2715528_PE.fasta"}]}},"outputs":{"ITS FASTA files":{"element_tests":{"DRR010481_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/DRR010481_ITS.fasta","compare":"contains"},"ERR2715528_MERGED_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_ITS.fasta","compare":"contains"}}},"ITS OTU tables (ITSoneDB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"ITS taxonomic classifications using ITSoneDB":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"ITSone"},{"that":"has_n_columns","comment":"#","n":15}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"ITSone"},{"that":"has_n_columns","comment":"#","n":15}]}}},"ITS OTU tables (UNITE DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"ITS taxonomic classifications using UNITE DB":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"UNITE"},{"that":"has_n_columns","comment":"#","n":15}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"UNITE"},{"that":"has_n_columns","comment":"#","n":15}]}}},"ITS OTU tables in HDF5 format (ITSoneDB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_size","value":72000,"delta":10000}]}}},"ITS OTU tables in JSON format (UNITE DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"ITS taxonomic abundance pie charts (ITSoneDB)":{"asserts":[{"that":"has_text","text":"DRR010481_FASTQ_fasta"},{"that":"has_text","text":"ERR2715528_MERGED_FASTQ_fasta"}]},"ITS OTU tables in HDF5 format (UNITE DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]},"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_size","value":72000,"delta":10000}]}}},"ITS taxonomic abundance pie charts (UNITE DB)":{"asserts":[{"that":"has_text","text":"DRR010481_FASTQ_fasta"},{"that":"has_text","text":"ERR2715528_MERGED_FASTQ_fasta"}]}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-pipeline-v5-its"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-pipeline-v5-complete.ga","testParameterFiles":["/mgnify-amplicon-pipeline-v5-complete-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X"},{"name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"}],"definition":{"a_galaxy_workflow":"true","annotation":"MGnify\'s amplicon pipeline v5.0. Including the Quality control for single-end and paired-end reads, rRNA-prediction, and ITS sub-WFs.","comments":[],"creator":[{"class":"Organization","name":" EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"List of SRA accession IDs, one per line.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"List of SRA accession IDs, one per line.","name":"SRA accession list"}],"label":"SRA accession list","name":"Input dataset","outputs":[],"position":{"left":302.9666748046875,"top":162.81400305175785},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"830c700a-be07-46b4-b64e-83e7c75e53ca","when":null,"workflow_outputs":[]},"1":{"annotation":"Clan information file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Clan information file.","name":"Clan information file"}],"label":"Clan information file","name":"Input dataset","outputs":[],"position":{"left":409.48333740234375,"top":367.99733129882816},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"fd16b9ce-b602-4535-9fa0-4218666c2290","when":null,"workflow_outputs":[]},"2":{"annotation":"Number of bases to average across.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Number of bases to average across.","name":"Trimmomatic sliding window number of bases to average across"}],"label":"Trimmomatic sliding window number of bases to average across","name":"Input parameter","outputs":[],"position":{"left":0,"top":819.4217522631932},"tool_id":null,"tool_state":"{\\"default\\": 4, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"39e0bbb8-def5-4cbe-875d-0e75d5e37495","when":null,"workflow_outputs":[]},"3":{"annotation":"Average quality required.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Average quality required.","name":"Trimmomatic sliding window average quality"}],"label":"Trimmomatic sliding window average quality","name":"Input parameter","outputs":[],"position":{"left":0.3513585914411692,"top":978.9369402822651},"tool_id":null,"tool_state":"{\\"default\\": 15, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"21106922-bb34-4647-9bee-446710c7d8d5","when":null,"workflow_outputs":[]},"4":{"annotation":"Enable base correction in overlapped regions. (Paired-end)","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Enable base correction in overlapped regions. (Paired-end)","name":"PE fastp - Enable base correction"}],"label":"PE fastp - Enable base correction","name":"Input parameter","outputs":[],"position":{"left":505.29394856845715,"top":1028.289063251411},"tool_id":null,"tool_state":"{\\"default\\": false, \\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"90776758-eb1e-494a-9c34-d891fb22552b","when":null,"workflow_outputs":[]},"5":{"annotation":"Cut bases off the start of a read, if below a threshold quality.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Cut bases off the start of a read, if below a threshold quality.","name":"Trimmomatic leading"}],"label":"Trimmomatic leading","name":"Input parameter","outputs":[],"position":{"left":8.755817129764921,"top":1146.685722505959},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"869e3248-10ca-44f5-98bb-5e520658b9bd","when":null,"workflow_outputs":[]},"6":{"annotation":"The quality value that a base is qualified. (Paired-end)","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"The quality value that a base is qualified. (Paired-end)","name":"PE fastp - Qualified quality phred"}],"label":"PE fastp - Qualified quality phred","name":"Input parameter","outputs":[],"position":{"left":505.9266149884791,"top":1113.6330085401016},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"9d0bfec9-8ab2-4d46-af38-65e8d9f38445","when":null,"workflow_outputs":[]},"7":{"annotation":"Cut bases off the end of a read, if below a threshold quality.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Cut bases off the end of a read, if below a threshold quality.","name":"Trimmomatic trailing"}],"label":"Trimmomatic trailing","name":"Input parameter","outputs":[],"position":{"left":14.271697331373844,"top":1261.7442025767366},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"a5f602ee-06ac-4c40-a5db-2bac77112f97","when":null,"workflow_outputs":[]},"8":{"annotation":"How many percents of bases are allowed to be unqualified (0~100). (Paired-end)","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"How many percents of bases are allowed to be unqualified (0~100). (Paired-end)","name":"PE fastp - Unqualified percent limit "}],"label":"PE fastp - Unqualified percent limit ","name":"Input parameter","outputs":[],"position":{"left":502.2599890119166,"top":1202.899671137758},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"3ccea377-db95-4eea-83d8-864f942e3347","when":null,"workflow_outputs":[]},"9":{"annotation":"Minimum length of reads to be kept.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Minimum length of reads to be kept.","name":"Trimmomatic min length"}],"label":"Trimmomatic min length","name":"Input parameter","outputs":[],"position":{"left":17.80621052877467,"top":1361.1919895694223},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"183a6499-b9e0-4844-a4cd-f5de8804f1e1","when":null,"workflow_outputs":[]},"10":{"annotation":"Reads shorter than this value will be discarded. (Paired-end)","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Reads shorter than this value will be discarded. (Paired-end)","name":"PE fastp - Length required"}],"label":"PE fastp - Length required","name":"Input parameter","outputs":[],"position":{"left":503.44327758613537,"top":1290.0829902295552},"tool_id":null,"tool_state":"{\\"default\\": 70, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"4c9fd5af-fd9b-45f2-8fe5-31852d05cab1","when":null,"workflow_outputs":[]},"11":{"annotation":"Minimum sequence length.","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"Minimum sequence length.","name":"Length filtering minimum size"}],"label":"Length filtering minimum size","name":"Input parameter","outputs":[],"position":{"left":21.664564064241176,"top":1451.7558900147008},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d5361c5a-2500-4dbd-bbc2-907d0b067fc4","when":null,"workflow_outputs":[]},"12":{"annotation":"Maximal N percentage threshold to conserve sequences.","content_id":null,"errors":null,"id":12,"input_connections":{},"inputs":[{"description":"Maximal N percentage threshold to conserve sequences.","name":"Ambiguity filtering maximal N percentage threshold to conserve sequences"}],"label":"Ambiguity filtering maximal N percentage threshold to conserve sequences","name":"Input parameter","outputs":[],"position":{"left":35.454411655787396,"top":1678.1509474535085},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1503e81e-64d4-473c-bde9-751ba2bb9143","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":null,"errors":null,"id":13,"input_connections":{},"inputs":[{"description":"","name":"SSU taxonomic abundance summary table name"}],"label":"SSU taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2227,"top":312.316},"tool_id":null,"tool_state":"{\\"default\\": \\"SSU taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"44cdc1f5-5482-4843-915c-3eeec13e63ff","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":null,"errors":null,"id":14,"input_connections":{},"inputs":[{"description":"","name":"SSU phylum level taxonomic abundance summary table name"}],"label":"SSU phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2225,"top":434.316},"tool_id":null,"tool_state":"{\\"default\\": \\"SSU phylum level taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"7a186613-22f7-406c-a464-566889fba9b6","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":null,"errors":null,"id":15,"input_connections":{},"inputs":[{"description":"","name":"LSU taxonomic abundance summary table name"}],"label":"LSU taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2244,"top":709.316},"tool_id":null,"tool_state":"{\\"default\\": \\"LSU taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c5976795-446d-465b-a17b-89f88cd36609","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":null,"errors":null,"id":16,"input_connections":{},"inputs":[{"description":"","name":"LSU phylum level taxonomic abundance summary table name"}],"label":"LSU phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2243,"top":849.316},"tool_id":null,"tool_state":"{\\"default\\": \\"LSU phylum level taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"0d031f24-2524-4111-8199-fde1fed46abd","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":null,"errors":null,"id":17,"input_connections":{},"inputs":[{"description":"","name":"ITSoneDB taxonomic abundance summary table name"}],"label":"ITSoneDB taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2479,"top":1091.316},"tool_id":null,"tool_state":"{\\"default\\": \\"ITSoneDB taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"78990ece-405a-4a82-8a7c-87c4efbe396f","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":null,"errors":null,"id":18,"input_connections":{},"inputs":[{"description":"","name":"ITSoneDB phylum level taxonomic abundance summary table name"}],"label":"ITSoneDB phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2483,"top":1216.316},"tool_id":null,"tool_state":"{\\"default\\": \\"ITSoneDB phylum level taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"25b414b3-afe1-48b1-81cd-37059285b496","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":null,"errors":null,"id":19,"input_connections":{},"inputs":[{"description":"","name":"ITS UNITE DB taxonomic abundance summary table name"}],"label":"ITS UNITE DB taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2489,"top":1502.316},"tool_id":null,"tool_state":"{\\"default\\": \\"ITS UNITE DB taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d6da448e-070a-423e-9f7e-182b7d20340c","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":null,"errors":null,"id":20,"input_connections":{},"inputs":[{"description":"","name":"ITS UNITE DB phylum level taxonomic abundance summary table name"}],"label":"ITS UNITE DB phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":2491,"top":1629.316},"tool_id":null,"tool_state":"{\\"default\\": \\"ITS UNITE DB phylum level taxonomic abundance summary table\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2f17ad42-aebe-4069-9907-ca85aec8df82","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastq_dl/fastq_dl/3.0.0+galaxy0","errors":null,"id":21,"input_connections":{"input_type|accessions_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastq-dl","name":"input_type"}],"label":null,"name":"fastq-dl","outputs":[{"name":"metadata","type":"input"},{"name":"single_end_collection","type":"input"},{"name":"paired_end_collection","type":"input"}],"position":{"left":583,"top":163.64733129882816},"post_job_actions":{"HideDatasetActionmetadata":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"metadata"},"HideDatasetActionpaired_end_collection":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"paired_end_collection"},"HideDatasetActionsingle_end_collection":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"single_end_collection"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastq_dl/fastq_dl/3.0.0+galaxy0","tool_shed_repository":{"changeset_revision":"5e7401777990","name":"fastq_dl","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"group_by_experiment\\": false, \\"group_by_sample\\": false, \\"input_type\\": {\\"select_input_type\\": \\"accessions_list\\", \\"__current_case__\\": 1, \\"accessions_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"only_download_metadata\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.0.0+galaxy0","type":"tool","uuid":"f0deb7f2-7d87-4b03-ba36-9aa0c697fb71","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"CONVERTER_gz_to_uncompressed","errors":null,"id":22,"input_connections":{"input1":{"id":21,"output_name":"single_end_collection"}},"inputs":[],"label":null,"name":"Convert compressed file to uncompressed.","outputs":[{"name":"output1","type":"auto"}],"position":{"left":675,"top":0},"post_job_actions":{"ChangeDatatypeActionoutput1":{"action_arguments":{"newtype":"fastq"},"action_type":"ChangeDatatypeAction","output_name":"output1"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"CONVERTER_gz_to_uncompressed","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d08c592b-f96d-466a-955f-79f94d7c6583","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"CONVERTER_gz_to_uncompressed","errors":null,"id":23,"input_connections":{"input1":{"id":21,"output_name":"paired_end_collection"}},"inputs":[],"label":null,"name":"Convert compressed file to uncompressed.","outputs":[{"name":"output1","type":"auto"}],"position":{"left":610.73738778973,"top":574.4055251629275},"post_job_actions":{"ChangeDatatypeActionoutput1":{"action_arguments":{"newtype":"fastq"},"action_type":"ChangeDatatypeAction","output_name":"output1"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"CONVERTER_gz_to_uncompressed","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"846a5e86-bb32-4a51-aead-67e885c47626","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":24,"input_connections":{"infile":{"id":22,"output_name":"output1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":830.6518995672985,"top":159.3217795262707},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR % 4 == 1 { gsub(/[ \\\\\\\\/]/, \\\\\\"-\\\\\\", $0) } { print }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"55dc8c71-6a78-42b3-ac88-1908de9386e0","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":23,"output_name":"output1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":726.4229425503371,"top":915.1239648998104},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR % 4 == 1 { gsub(/[ \\\\\\\\/]/, \\\\\\"-\\\\\\", $0) } { print }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"e69435e0-8589-4010-af09-52d505b5408c","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"CONVERTER_uncompressed_to_gz","errors":null,"id":26,"input_connections":{"input1":{"id":24,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Convert uncompressed file to compressed","outputs":[{"name":"output1","type":"auto"}],"position":{"left":856,"top":319},"post_job_actions":{"ChangeDatatypeActionoutput1":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"output1"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"CONVERTER_uncompressed_to_gz","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.16+galaxy0","type":"tool","uuid":"f88f472c-f9e4-4693-b47d-ac2938f4fd47","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"CONVERTER_uncompressed_to_gz","errors":null,"id":27,"input_connections":{"input1":{"id":25,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Convert uncompressed file to compressed","outputs":[{"name":"output1","type":"auto"}],"position":{"left":752.1779884302049,"top":1038.0952072043892},"post_job_actions":{"ChangeDatatypeActionoutput1":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"output1"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"CONVERTER_uncompressed_to_gz","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.16+galaxy0","type":"tool","uuid":"146869a3-e9f7-4f0b-a6a9-e33532825c61","when":null,"workflow_outputs":[]},"28":{"annotation":"","id":28,"input_connections":{"Ambiguity filtering - Maximal N percentage threshold to conserve sequences":{"id":12,"input_subworkflow_step_id":7,"output_name":"output"},"Length filtering - Minimum size":{"id":11,"input_subworkflow_step_id":6,"output_name":"output"},"Single-end reads":{"id":26,"input_subworkflow_step_id":0,"output_name":"output1"},"Trimmomatic - LEADING":{"id":5,"input_subworkflow_step_id":2,"output_name":"output"},"Trimmomatic - MINLEN":{"id":9,"input_subworkflow_step_id":5,"output_name":"output"},"Trimmomatic - SLIDING WINDOW - Average quality required":{"id":3,"input_subworkflow_step_id":1,"output_name":"output"},"Trimmomatic - SLIDING WINDOW - Number of bases to average across":{"id":2,"input_subworkflow_step_id":3,"output_name":"output"},"Trimmomatic - TRAILING":{"id":7,"input_subworkflow_step_id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"MGnify\'s amplicon pipeline v5.0 - Quality control SE","outputs":[],"position":{"left":1118.958717320336,"top":128.4131041435775},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"MGnify - EMBL","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - Quality control SE","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Single-end reads collection.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Single-end reads collection.","name":"Single-end reads"}],"label":"Single-end reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":219.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"743430cc-e893-4211-9107-663adf3b28b4","when":null,"workflow_outputs":[]},"1":{"annotation":"Average quality required.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Average quality required.","name":"Trimmomatic - SLIDING WINDOW - Average quality required"}],"label":"Trimmomatic - SLIDING WINDOW - Average quality required","name":"Input parameter","outputs":[],"position":{"left":284.8237910552773,"top":0},"tool_id":null,"tool_state":"{\\"default\\": 15, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"7b3ec08b-9dd4-4215-8901-c0eb6f0be3e7","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"dc036460-5da1-4fd8-bc90-39f026ed738f"}]},"2":{"annotation":"Cut bases off the start of a read, if below a threshold quality.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Cut bases off the start of a read, if below a threshold quality.","name":"Trimmomatic - LEADING"}],"label":"Trimmomatic - LEADING","name":"Input parameter","outputs":[],"position":{"left":235.82379105527724,"top":373.00000000000085},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2f3fe5f1-44e3-4117-9417-d68b8a90aa23","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"cbc31256-f1f7-40f6-a21d-1e2dc968bc64"}]},"3":{"annotation":"Number of bases to average across.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Number of bases to average across.","name":"Trimmomatic - SLIDING WINDOW - Number of bases to average across"}],"label":"Trimmomatic - SLIDING WINDOW - Number of bases to average across","name":"Input parameter","outputs":[],"position":{"left":519.8817910552779,"top":90.00000000000021},"tool_id":null,"tool_state":"{\\"default\\": 4, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5b70e0b8-0631-4059-aa36-d1ce896e45b1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"225ef644-f8cf-489a-8de3-95e7c67d471d"}]},"4":{"annotation":"Cut bases off the end of a read, if below a threshold quality.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Cut bases off the end of a read, if below a threshold quality.","name":"Trimmomatic - TRAILING"}],"label":"Trimmomatic - TRAILING","name":"Input parameter","outputs":[],"position":{"left":236.21679574468544,"top":501.2064765902164},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"eed3e996-2baf-4908-ba47-449254adc3cb","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"87a0516a-a12b-44ca-b4a4-bc3e3f9ae30b"}]},"5":{"annotation":"Minimum length of reads to be kept.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Minimum length of reads to be kept.","name":"Trimmomatic - MINLEN"}],"label":"Trimmomatic - MINLEN","name":"Input parameter","outputs":[],"position":{"left":218.92879574468543,"top":624.2064765902168},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"08f7f8d1-ad75-42cf-91e0-bb549c8edd14","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"4cf4eeb3-bc3f-4134-bc0b-dc5f31e061c5"}]},"6":{"annotation":"Minimum sequence length.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Minimum sequence length.","name":"Length filtering - Minimum size"}],"label":"Length filtering - Minimum size","name":"Input parameter","outputs":[],"position":{"left":734.3396679241868,"top":219.62078434990826},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5db2e477-4d58-4380-bc0d-e02a2249045c","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"62b85a60-f703-47d4-9e73-6e11cedd4b7b"}]},"7":{"annotation":"Maximal N percentage threshold to conserve sequences.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Maximal N percentage threshold to conserve sequences.","name":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences"}],"label":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences","name":"Input parameter","outputs":[],"position":{"left":1241.4736763919316,"top":192.6829501818756},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1d6681c7-d75f-49ce-a21b-e5779b0cadca","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d1f7d15b-61fb-425b-81f7-b7345dcebb1d"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":8,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":137.83379001997133,"top":1070.203938085305},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"0ef7f04f-c955-4909-81d0-51665f07604f","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","errors":null,"id":9,"input_connections":{"operations_0|operation|required_quality":{"id":1,"output_name":"output"},"operations_0|operation|window_size":{"id":3,"output_name":"output"},"operations_1|operation|leading":{"id":2,"output_name":"output"},"operations_2|operation|trailing":{"id":4,"output_name":"output"},"operations_3|operation|minlen":{"id":5,"output_name":"output"},"readtype|fastq_in":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Trimmomatic","name":"readtype"}],"label":"Trimming","name":"Trimmomatic","outputs":[{"name":"fastq_out","type":"input"}],"position":{"left":647.7861331470302,"top":337.8731636019347},"post_job_actions":{"ChangeDatatypeActionfastq_out":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"fastq_out"},"HideDatasetActionfastq_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"fastq_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","tool_shed_repository":{"changeset_revision":"b9aaed85cbd1","name":"trimmomatic","owner":"pjbriggs","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"illuminaclip\\": {\\"do_illuminaclip\\": \\"no\\", \\"__current_case__\\": 1}, \\"operations\\": [{\\"__index__\\": 0, \\"operation\\": {\\"name\\": \\"SLIDINGWINDOW\\", \\"__current_case__\\": 0, \\"window_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"required_quality\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"operation\\": {\\"name\\": \\"LEADING\\", \\"__current_case__\\": 2, \\"leading\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"operation\\": {\\"name\\": \\"TRAILING\\", \\"__current_case__\\": 3, \\"trailing\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"operation\\": {\\"name\\": \\"MINLEN\\", \\"__current_case__\\": 1, \\"minlen\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"output_err\\": false, \\"output_logs\\": false, \\"quality_score\\": \\"-phred33\\", \\"readtype\\": {\\"single_or_paired\\": \\"se\\", \\"__current_case__\\": 0, \\"fastq_in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.39+galaxy2","type":"tool","uuid":"6a258b5c-2dbd-4615-88f8-1908bf555374","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":521.8083204698337,"top":1301.3107808231161},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_initial_reads\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"93d70102-188c-4c9e-b0b4-fcb24acd8a28","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","errors":null,"id":11,"input_connections":{"input_file":{"id":9,"output_name":"fastq_out"},"min_size":{"id":6,"output_name":"output"}},"inputs":[],"label":"Length filtering","name":"Filter FASTQ","outputs":[{"name":"output_file","type":"input"}],"position":{"left":891.3091779432864,"top":370.3116497676374},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"144275714590","name":"fastq_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fastq_filters\\": [], \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_num_deviants\\": \\"0\\", \\"max_quality\\": \\"0.0\\", \\"max_size\\": \\"0\\", \\"min_quality\\": \\"0.0\\", \\"min_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paired_end\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"c22e8401-0e9e-404e-8fa4-3cc5e28670b2","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":12,"input_connections":{"input_file":{"id":9,"output_name":"fastq_out"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":644.4179064261575,"top":955.8740730752126},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"18505eb1-1b95-431e-b91f-1b73b50e2e47","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","errors":null,"id":13,"input_connections":{"filter_treatments|base_content_filter_treatments|N_percentage_content_filter_treatments|N_percentage_content_filter_treatment_value":{"id":7,"output_name":"output"},"seq_type|input_singles":{"id":11,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool PRINSEQ","name":"seq_type"}],"label":"Ambiguity filtering","name":"PRINSEQ","outputs":[{"name":"good_sequence_file","type":"input"},{"name":"rejected_sequence_file","type":"input"}],"position":{"left":1165.055726007993,"top":432.1991937304916},"post_job_actions":{"HideDatasetActiongood_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"good_sequence_file"},"HideDatasetActionrejected_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"rejected_sequence_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","tool_shed_repository":{"changeset_revision":"1ee282794de3","name":"prinseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_treatments\\": {\\"apply_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"length_filter_treatments\\": {\\"apply_length_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"quality_filter_treatments\\": {\\"apply_quality_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"base_content_filter_treatments\\": {\\"apply_base_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"GC_perc_content_filter_treatments\\": {\\"apply_GC_perc_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_number_content_filter_treatments\\": {\\"apply_N_number_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_percentage_content_filter_treatments\\": {\\"apply_N_percentage_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"N_percentage_content_filter_treatment_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"apply_other_base_content_filter_treatments\\": true}, \\"complexity_filter_treatments\\": {\\"apply_complexity_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}}, \\"seq_type\\": {\\"seq_type_opt\\": \\"single\\", \\"__current_case__\\": 0, \\"input_singles\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"trimming_treatments\\": {\\"apply_trimming_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.20.4+galaxy2","type":"tool","uuid":"c7c4f665-6256-4693-b13f-3feec07b2bd6","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":14,"input_connections":{"input_file":{"id":11,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1019.7935387142895,"top":800.338235605041},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"6fc925d9-c644-4f8d-9435-a1e94b151eca","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":15,"input_connections":{"infile":{"id":12,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":947.4179064261575,"top":1153.2574043740406},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_trimming\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"687e1130-33fc-4890-8099-e2c6da98f836","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","errors":null,"id":16,"input_connections":{"input_file":{"id":13,"output_name":"good_sequence_file"}},"inputs":[],"label":null,"name":"FASTQ to FASTA","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":1648.8164640063083,"top":634.3621610446716},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"ac3fed111eb6","name":"fastqtofasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"9bc910c0-f85a-474e-9a38-3c5cc2c3434c","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":17,"input_connections":{"input_file":{"id":13,"output_name":"good_sequence_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1612.1550210976457,"top":812.2383769545825},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"5082dc16-2032-46fe-a469-cf79af5da99c","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":18,"input_connections":{"infile":{"id":14,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1331.4716094560185,"top":1063.8186665413482},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_length_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1650ae45-7ee2-4765-8953-48a6548b1c67","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":19,"input_connections":{"input":{"id":16,"output_name":"output_file"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1941.646496697877,"top":647.1088294674363},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Single-end post quality control FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"91a2a702-a2e0-49ec-9864-8d4edb56986e","when":null,"workflow_outputs":[{"label":"Single-end post quality control FASTA files","output_name":"output","uuid":"5f19c1d8-2a5e-48ad-a962-012d20b6cac5"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":20,"input_connections":{"infile":{"id":17,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1877.7115272508015,"top":1026.8671426905246},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_ambiguous_base_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"284c9abf-2ebb-490c-a4d9-1fe997a1f562","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":21,"input_connections":{"results_0|software_cond|output_0|input":{"id":10,"output_name":"outfile"},"results_0|software_cond|output_1|input":{"id":15,"output_name":"outfile"},"results_0|software_cond|output_2|input":{"id":18,"output_name":"outfile"},"results_0|software_cond|output_3|input":{"id":20,"output_name":"outfile"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1387.0957816692048,"top":1445.9486517075318},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Single-end MultiQC report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Single-end MultiQC statistics"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}}, {\\"__index__\\": 1, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}}, {\\"__index__\\": 2, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}}, {\\"__index__\\": 3, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}}]}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"640f4f45-c12a-47e0-846b-9b150ed8beca","when":null,"workflow_outputs":[{"label":"Single-end MultiQC statistics","output_name":"stats","uuid":"125cc43d-33aa-4b6f-a3fa-1c7c4ef3488f"},{"label":"Single-end MultiQC report","output_name":"html_report","uuid":"81732dfa-d5cb-46f5-ba34-13242ab5e3fa"}]}},"tags":[],"uuid":"68bf2a73-d68a-4e25-ba52-5586c057df77"},"tool_id":null,"type":"subworkflow","uuid":"147188ec-4e08-448a-be45-c8eec144d9f2","when":null,"workflow_outputs":[{"label":"Single-end MultiQC report","output_name":"Single-end MultiQC report","uuid":"524978e4-ce01-4152-95e5-4e1ef5d48c3a"},{"label":"Single-end post quality control FASTA files","output_name":"Single-end post quality control FASTA files","uuid":"306311fc-4935-4aae-b9ad-ade3901e134d"},{"label":"Single-end MultiQC statistics","output_name":"Single-end MultiQC statistics","uuid":"90e90edf-c318-42bf-a359-02d7d2bab053"}]},"29":{"annotation":"","id":29,"input_connections":{"Ambiguity filtering - Maximal N percentage threshold to conserve sequences":{"id":12,"input_subworkflow_step_id":11,"output_name":"output"},"Length filtering - Minimum size":{"id":11,"input_subworkflow_step_id":9,"output_name":"output"},"Paired-end reads":{"id":27,"input_subworkflow_step_id":0,"output_name":"output1"},"Trimmomatic - LEADING":{"id":5,"input_subworkflow_step_id":7,"output_name":"output"},"Trimmomatic - MINLEN":{"id":9,"input_subworkflow_step_id":10,"output_name":"output"},"Trimmomatic - SLIDINGWINDOW - Average quality required":{"id":3,"input_subworkflow_step_id":6,"output_name":"output"},"Trimmomatic - SLIDINGWINDOW - Number of bases to average across":{"id":2,"input_subworkflow_step_id":5,"output_name":"output"},"Trimmomatic - TRAILING":{"id":7,"input_subworkflow_step_id":8,"output_name":"output"},"fastp - Enable base correction":{"id":4,"input_subworkflow_step_id":1,"output_name":"output"},"fastp - Length required":{"id":10,"input_subworkflow_step_id":4,"output_name":"output"},"fastp - Qualified quality phred":{"id":6,"input_subworkflow_step_id":2,"output_name":"output"},"fastp - Unqualified percent limit ":{"id":8,"input_subworkflow_step_id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"MGnify\'s amplicon pipeline v5.0 - Quality control PE","outputs":[],"position":{"left":959.2056200080917,"top":1225.7198800364831},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - Quality control PE","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Paired-end reads collection.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Paired-end reads collection.","name":"Paired-end reads"}],"label":"Paired-end reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":134.00843148565127},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"4a419fd7-0b26-4f9d-83ec-2471f862c610","when":null,"workflow_outputs":[]},"1":{"annotation":" Enable base correction in overlapped regions.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":" Enable base correction in overlapped regions.","name":"fastp - Enable base correction"}],"label":"fastp - Enable base correction","name":"Input parameter","outputs":[],"position":{"left":67.94649253008049,"top":317.0543088585308},"tool_id":null,"tool_state":"{\\"default\\": false, \\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"81367449-c0ca-41be-b69d-5bf1568e0361","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"2cc2526b-c51d-42bd-94ac-540d45cce0aa"}]},"2":{"annotation":"The quality value that a base is qualified.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"The quality value that a base is qualified.","name":"fastp - Qualified quality phred"}],"label":"fastp - Qualified quality phred","name":"Input parameter","outputs":[],"position":{"left":309.2157898611554,"top":329.9376208519377},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"cb8a7bac-c007-45c6-814b-9aacae0e37b3","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"8ed6f5eb-5fa9-4d25-8e8b-5f1252c4cf59"}]},"3":{"annotation":"How many percents of bases are allowed to be unqualified (0~100).","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"How many percents of bases are allowed to be unqualified (0~100).","name":"fastp - Unqualified percent limit "}],"label":"fastp - Unqualified percent limit ","name":"Input parameter","outputs":[],"position":{"left":311.55821022359305,"top":420.12080480578607},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"26ed5910-22cf-4a8e-a144-edbb8d6b1d50","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c24e8e77-678b-4fa3-9436-8f0d6592ed7e"}]},"4":{"annotation":"Reads shorter than this value will be discarded.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Reads shorter than this value will be discarded.","name":"fastp - Length required"}],"label":"fastp - Length required","name":"Input parameter","outputs":[],"position":{"left":310.3870000423742,"top":506.790358215978},"tool_id":null,"tool_state":"{\\"default\\": 70, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6521e9f3-6d1b-4e95-8934-3345f858bcf8","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"3221d128-de9e-489e-8497-e357f686112f"}]},"5":{"annotation":"Number of bases to average across.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Number of bases to average across.","name":"Trimmomatic - SLIDINGWINDOW - Number of bases to average across"}],"label":"Trimmomatic - SLIDINGWINDOW - Number of bases to average across","name":"Input parameter","outputs":[],"position":{"left":310.8945435169014,"top":618.8166406435558},"tool_id":null,"tool_state":"{\\"default\\": 4, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1b2652b0-235b-4120-9c82-e4d3210c9d2f","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"77bf0209-c48a-4422-8478-722a924ee1f3"}]},"6":{"annotation":"Average quality required.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Average quality required.","name":"Trimmomatic - SLIDINGWINDOW - Average quality required"}],"label":"Trimmomatic - SLIDINGWINDOW - Average quality required","name":"Input parameter","outputs":[],"position":{"left":312.5205763417023,"top":751.1494950478717},"tool_id":null,"tool_state":"{\\"default\\": 15, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"53e87ca4-6745-4835-b25f-878705737f0d","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d4ea0915-89cf-4a64-8c27-3e0a1443168d"}]},"7":{"annotation":"Minimum quality required to keep a base.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Minimum quality required to keep a base.","name":"Trimmomatic - LEADING"}],"label":"Trimmomatic - LEADING","name":"Input parameter","outputs":[],"position":{"left":315.85885539878177,"top":880.5228796617778},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"4e1cfaad-b067-4bf3-a6b8-33d3fd826285","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"fe7fcea9-5651-491a-b079-db434960e968"}]},"8":{"annotation":"Minimum quality required to keep a base.","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Minimum quality required to keep a base.","name":"Trimmomatic - TRAILING"}],"label":"Trimmomatic - TRAILING","name":"Input parameter","outputs":[],"position":{"left":330.0774689415621,"top":973.9211632212501},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d48c0437-1ab6-4fd4-8dfa-29dd056acaec","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"5f620484-e1bf-4bc5-a227-1b40d783eadd"}]},"9":{"annotation":"Minimum sequence length.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Minimum sequence length.","name":"Length filtering - Minimum size"}],"label":"Length filtering - Minimum size","name":"Input parameter","outputs":[],"position":{"left":993.3486541402174,"top":364.32685651072273},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2957cfb6-b542-4354-b94f-fd171df4c9b1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"4792bf14-48e1-43da-b086-5e137cd9c2a0"}]},"10":{"annotation":"Minimum length of reads to be kept.","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Minimum length of reads to be kept.","name":"Trimmomatic - MINLEN"}],"label":"Trimmomatic - MINLEN","name":"Input parameter","outputs":[],"position":{"left":332.17940873206794,"top":1062.1550248633116},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c898eada-b136-40bf-a78d-240b8d02c660","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"b5501135-286f-410f-8aaa-ce79dfa52b38"}]},"11":{"annotation":"Maximal N percentage threshold to conserve sequences.","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"Maximal N percentage threshold to conserve sequences.","name":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences"}],"label":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences","name":"Input parameter","outputs":[],"position":{"left":1443.0933637282412,"top":237.69561171734486},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d8341e6e-1401-48ef-ac1a-68c98309d0e1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"f55c448d-f67e-4e0f-bede-c7937f2eb298"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":12,"input_connections":{"filter_options|length_filtering_options|length_required":{"id":4,"output_name":"output"},"filter_options|quality_filtering_options|qualified_quality_phred":{"id":2,"output_name":"output"},"filter_options|quality_filtering_options|unqualified_percent_limit":{"id":3,"output_name":"output"},"read_mod_options|base_correction_options|correction":{"id":1,"output_name":"output"},"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":331.5,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unqualified_percent_limit\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": null, \\"umi_len\\": null, \\"umi_prefix\\": null}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"paired_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": null, \\"adapter_sequence2\\": null, \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"9b7f1d2d-d39f-407d-a709-a5095fca62c2","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__UNZIP_COLLECTION__","errors":null,"id":13,"input_connections":{"input":{"id":12,"output_name":"output_paired_coll"}},"inputs":[],"label":null,"name":"Unzip collection","outputs":[{"name":"forward","type":"input"},{"name":"reverse","type":"input"}],"position":{"left":610.2833251953125,"top":342.9666748046875},"post_job_actions":{"ChangeDatatypeActionforward":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"forward"},"ChangeDatatypeActionreverse":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"reverse"},"HideDatasetActionforward":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"forward"},"HideDatasetActionreverse":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reverse"}},"tool_id":"__UNZIP_COLLECTION__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"f151e04c-8502-45b7-8503-38838efe39ee","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mgnify_seqprep/mgnify_seqprep/1.2+galaxy0","errors":null,"id":14,"input_connections":{"input1":{"id":13,"output_name":"forward"},"input2":{"id":13,"output_name":"reverse"}},"inputs":[],"label":null,"name":"Merging paired-end Illumina reads (SeqPrep, modified for use with MGnify piplines)","outputs":[{"name":"output1","type":"fastq.gz"},{"name":"output2","type":"fastq.gz"},{"name":"merged","type":"fastq.gz"}],"position":{"left":788.0499877929688,"top":649.379055631159},"post_job_actions":{"ChangeDatatypeActionmerged":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"merged"},"HideDatasetActionmerged":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"merged"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mgnify_seqprep/mgnify_seqprep/1.2+galaxy0","tool_shed_repository":{"changeset_revision":"76ea9d4604bc","name":"mgnify_seqprep","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"general_options\\": {\\"first_read_discarded\\": null, \\"second_read_discarded\\": null, \\"phred64\\": false, \\"quality_cutoff\\": \\"13\\", \\"min_length\\": \\"30\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": true, \\"merging_options\\": {\\"maximum_quality_score\\": null, \\"print_overhang\\": false, \\"min_base_pair_overlap\\": \\"15\\", \\"max_mismatch_fraction\\": \\"0.02\\", \\"min_match_fraction\\": \\"0.9\\"}, \\"trimming_options\\": {\\"adapter_a\\": \\"AGATCGGAAGAGCGGTTCAG\\", \\"adapter_b\\": \\"AGATCGGAAGAGCGTCGTGT\\", \\"adapter_overlap\\": \\"10\\", \\"max_mismatch_fraction\\": \\"0.02\\", \\"min_match_fraction\\": \\"0.87\\", \\"adapter_bandwidth\\": \\"50\\", \\"gap_open\\": \\"8\\", \\"gap_extend\\": \\"2\\", \\"gap_end\\": \\"2\\", \\"local_alignment_score\\": \\"26\\", \\"read_alignment_bandwidth\\": \\"50\\", \\"read_alignment_gap_open\\": \\"26\\", \\"read_alignment_gap_extend\\": \\"9\\", \\"read_alignment_gap_end\\": \\"5\\", \\"read_alignment_max_gap_fraction\\": \\"0.125\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2+galaxy0","type":"tool","uuid":"670c6abb-2a49-4a77-9cad-eaae9ebfe74e","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","errors":null,"id":15,"input_connections":{"operations_0|operation|required_quality":{"id":6,"output_name":"output"},"operations_0|operation|window_size":{"id":5,"output_name":"output"},"operations_1|operation|leading":{"id":7,"output_name":"output"},"operations_2|operation|trailing":{"id":8,"output_name":"output"},"operations_3|operation|minlen":{"id":10,"output_name":"output"},"readtype|fastq_in":{"id":14,"output_name":"merged"}},"inputs":[{"description":"runtime parameter for tool Trimmomatic","name":"readtype"}],"label":"Trimming","name":"Trimmomatic","outputs":[{"name":"fastq_out","type":"input"}],"position":{"left":1099.8073621070885,"top":478.93954202505694},"post_job_actions":{"HideDatasetActionfastq_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"fastq_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","tool_shed_repository":{"changeset_revision":"b9aaed85cbd1","name":"trimmomatic","owner":"pjbriggs","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"illuminaclip\\": {\\"do_illuminaclip\\": \\"no\\", \\"__current_case__\\": 1}, \\"operations\\": [{\\"__index__\\": 0, \\"operation\\": {\\"name\\": \\"SLIDINGWINDOW\\", \\"__current_case__\\": 0, \\"window_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"required_quality\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"operation\\": {\\"name\\": \\"LEADING\\", \\"__current_case__\\": 2, \\"leading\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"operation\\": {\\"name\\": \\"TRAILING\\", \\"__current_case__\\": 3, \\"trailing\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"operation\\": {\\"name\\": \\"MINLEN\\", \\"__current_case__\\": 1, \\"minlen\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"output_err\\": false, \\"output_logs\\": false, \\"quality_score\\": \\"-phred33\\", \\"readtype\\": {\\"single_or_paired\\": \\"se\\", \\"__current_case__\\": 0, \\"fastq_in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.39+galaxy2","type":"tool","uuid":"c2079512-2019-41aa-beea-d1b597e59b42","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":16,"input_connections":{"input_file":{"id":14,"output_name":"merged"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1250.6521000986795,"top":1200.9140694839034},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"c48ed7ed-5a1d-40e8-8423-434578e91568","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","errors":null,"id":17,"input_connections":{"input_file":{"id":15,"output_name":"fastq_out"},"min_size":{"id":9,"output_name":"output"}},"inputs":[],"label":"Length filtering","name":"Filter FASTQ","outputs":[{"name":"output_file","type":"input"}],"position":{"left":1359,"top":440.365741330622},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"144275714590","name":"fastq_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fastq_filters\\": [], \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_num_deviants\\": \\"0\\", \\"max_quality\\": \\"0.0\\", \\"max_size\\": \\"0\\", \\"min_quality\\": \\"0.0\\", \\"min_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paired_end\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"a12d9770-de9e-45cf-8d18-2be9cc7873ed","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":18,"input_connections":{"input_file":{"id":15,"output_name":"fastq_out"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1546.9682759470388,"top":1199.7428593026848},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"01bb817a-6e2a-4a81-b5a7-ec42f0355b3e","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":19,"input_connections":{"infile":{"id":16,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1593.8207131348126,"top":1563.9834870611885},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_initial_reads\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"680b293b-0b1e-4b95-9dc9-233354651b61","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","errors":null,"id":20,"input_connections":{"filter_treatments|base_content_filter_treatments|N_percentage_content_filter_treatments|N_percentage_content_filter_treatment_value":{"id":11,"output_name":"output"},"seq_type|input_singles":{"id":17,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool PRINSEQ","name":"seq_type"}],"label":"Ambiguity filtering","name":"PRINSEQ","outputs":[{"name":"good_sequence_file","type":"input"},{"name":"rejected_sequence_file","type":"input"}],"position":{"left":1646.903450664335,"top":523.0258360658718},"post_job_actions":{"HideDatasetActiongood_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"good_sequence_file"},"HideDatasetActionrejected_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"rejected_sequence_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","tool_shed_repository":{"changeset_revision":"1ee282794de3","name":"prinseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_treatments\\": {\\"apply_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"length_filter_treatments\\": {\\"apply_length_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"quality_filter_treatments\\": {\\"apply_quality_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"base_content_filter_treatments\\": {\\"apply_base_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"GC_perc_content_filter_treatments\\": {\\"apply_GC_perc_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_number_content_filter_treatments\\": {\\"apply_N_number_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_percentage_content_filter_treatments\\": {\\"apply_N_percentage_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"N_percentage_content_filter_treatment_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"apply_other_base_content_filter_treatments\\": true}, \\"complexity_filter_treatments\\": {\\"apply_complexity_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}}, \\"seq_type\\": {\\"seq_type_opt\\": \\"single\\", \\"__current_case__\\": 0, \\"input_singles\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"trimming_treatments\\": {\\"apply_trimming_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.20.4+galaxy2","type":"tool","uuid":"d65b0e38-5f52-425a-a3c3-9cb126ead9f7","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":21,"input_connections":{"input_file":{"id":17,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1536.6496929362308,"top":818.1691488145751},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"f291100e-19ae-4baf-824b-16244df1e9f3","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":22,"input_connections":{"infile":{"id":18,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1919.413113574621,"top":1164.60655386612},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_trimming\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"2bb5af38-51bd-4828-bc34-7b4c7273c9e2","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","errors":null,"id":23,"input_connections":{"input_file":{"id":20,"output_name":"good_sequence_file"}},"inputs":[],"label":null,"name":"FASTQ to FASTA","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":1952.1162504932327,"top":541.8794796760812},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"ac3fed111eb6","name":"fastqtofasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"7b5c5aeb-0201-4378-984c-ded57d114157","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":24,"input_connections":{"input_file":{"id":20,"output_name":"good_sequence_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":2282.6440047444867,"top":666.8629867716423},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"9afd3a4f-fdae-4d9d-9afc-5990276dc8c5","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":21,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2169.018738976497,"top":996.4944763313531},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_length_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"8c5dc662-6bcc-4772-bc0e-d0b747113ef3","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":26,"input_connections":{"input":{"id":23,"output_name":"output_file"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2306.5295226390967,"top":402.29376277487876},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Paired-end post quality control FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"b899fa36-d2c9-4cbe-880a-9984f2b48462","when":null,"workflow_outputs":[{"label":"Paired-end post quality control FASTA files","output_name":"output","uuid":"ab65a776-f7bb-4495-8eeb-12079046ae1b"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":27,"input_connections":{"infile":{"id":24,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2529.1111984955273,"top":831.6091291601473},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_ambiguous_base_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"aacb9221-34de-46f7-91ec-ff796e98db10","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":28,"input_connections":{"results_0|software_cond|output_0|input":{"id":19,"output_name":"outfile"},"results_0|software_cond|output_1|input":{"id":22,"output_name":"outfile"},"results_0|software_cond|output_2|input":{"id":25,"output_name":"outfile"},"results_0|software_cond|output_3|input":{"id":27,"output_name":"outfile"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2884.8980879090345,"top":597.9223680616317},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Paired-end MultiQC report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Paired-end MultiQC statistics"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 3, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"e04ee136-c52f-4d7e-85c5-3b3ed78b0cce","when":null,"workflow_outputs":[{"label":"Paired-end MultiQC report","output_name":"html_report","uuid":"1d870a5b-942b-4f61-b8fd-a68b121b7f8c"},{"label":"Paired-end MultiQC statistics","output_name":"stats","uuid":"fc6047b2-201d-4e77-ab0d-9c8107897088"}]}},"tags":[],"uuid":"3af45a28-126d-4c9d-9948-cabfe43566f3"},"tool_id":null,"type":"subworkflow","uuid":"a6cb4bbc-057a-4326-b5c8-2c2a17514a62","when":null,"workflow_outputs":[{"label":"Paired-end MultiQC statistics","output_name":"Paired-end MultiQC statistics","uuid":"9f83bbc4-916e-4379-a018-cef375aa6abb"},{"label":"Paired-end post quality control FASTA files","output_name":"Paired-end post quality control FASTA files","uuid":"6b08b8f6-8f53-4a23-92a1-40a064674f17"},{"label":"Paired-end MultiQC report","output_name":"Paired-end MultiQC report","uuid":"ec044dec-9f20-447f-8108-6040204f6054"}]},"30":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":30,"input_connections":{"inputs_0|input":{"id":28,"output_name":"Single-end post quality control FASTA files"},"inputs_1|input":{"id":29,"output_name":"Paired-end post quality control FASTA files"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1390.536415507545,"top":1164.366596783564},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"7ae8f68b-ebaf-4d46-af2b-804c8983dd7d","when":null,"workflow_outputs":[]},"31":{"annotation":"","id":31,"input_connections":{"Clan information file":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"},"Processed sequences":{"id":30,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"MGnify\'s amplicon pipeline v5.0 - rRNA prediction","outputs":[],"position":{"left":1748.9500122070312,"top":386.6499938964844},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - rRNA prediction","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Post quality control.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Post quality control.","name":"Processed sequences"}],"label":"Processed sequences","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":613.5582334322104},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4ddc6490-b2f3-4b9f-9a71-2da0c8857457","when":null,"workflow_outputs":[]},"1":{"annotation":"This file lists which models belong to the same clan.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This file lists which models belong to the same clan.","name":"Clan information file"}],"label":"Clan information file","name":"Input dataset","outputs":[],"position":{"left":345.04235240350044,"top":524.7051496457693},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e981bbe9-5515-4d95-aacf-7047f052f5ef","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/infernal/infernal_cmsearch/1.1.5+galaxy0","errors":null,"id":2,"input_connections":{"seqdb":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"cmsearch","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":353.04235240350044,"top":627.7051496457692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/infernal/infernal_cmsearch/1.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"3711f654044e","name":"infernal","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"A\\": false, \\"Z\\": \\"1000.0\\", \\"acceleration_huristics\\": {\\"acceleration_huristics_selector\\": \\"--hmmonly\\", \\"__current_case__\\": 5}, \\"acyk\\": false, \\"anytrunc\\": false, \\"bottomonly\\": false, \\"cm_opts\\": {\\"cm_opts_selector\\": \\"db\\", \\"__current_case__\\": 0, \\"database\\": \\"mgnify_5_0_rfam\\"}, \\"cyk\\": false, \\"g\\": false, \\"inclusion_thresholds_opts\\": {\\"inclusion_thresholds_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"model_thresholds\\": {\\"cut_ga\\": false, \\"cut_nc\\": false, \\"cut_tc\\": false}, \\"mxsize\\": \\"128.0\\", \\"noali\\": true, \\"nonull3\\": false, \\"notrunc\\": false, \\"reporting_thresholds_opts\\": {\\"reporting_thresholds_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"seqdb\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"smxsize\\": \\"128.0\\", \\"toponly\\": false, \\"verbose\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy0","type":"tool","uuid":"3c4422a0-7d9e-4cc0-97ab-86a6005314c3","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":448.5761351863496,"top":924.4124853144156},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"9fc8ec4b-d8b5-4a98-aa57-32d3178a2362","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/rnateam/cmsearch_deoverlap/cmsearch_deoverlap/0.08+galaxy2","errors":null,"id":4,"input_connections":{"clan_information":{"id":1,"output_name":"output"},"cmsearch_matches":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"CMsearch deoverlap","outputs":[{"name":"deoverlapped_matches","type":"tabular"}],"position":{"left":627.0499877929688,"top":620.7119790448688},"post_job_actions":{"HideDatasetActiondeoverlapped_matches":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"deoverlapped_matches"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/rnateam/cmsearch_deoverlap/cmsearch_deoverlap/0.08+galaxy2","tool_shed_repository":{"changeset_revision":"874ad23526a0","name":"cmsearch_deoverlap","owner":"rnateam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"clan_information\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"cmsearch_matches\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"dirty\\": false, \\"maxkeep\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.08+galaxy2","type":"tool","uuid":"6500ab8a-8bcc-4c40-be3d-ba61769c7ed8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":5,"input_connections":{"tables_0|table":{"id":4,"output_name":"deoverlapped_matches"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"},{"name":"output1","type":"tabular"},{"name":"output2","type":"tabular"},{"name":"output3","type":"tabular"}],"position":{"left":947.7324324034817,"top":678.7051496457692},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionoutput3":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output3"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": [{\\"__index__\\": 0, \\"sqlquery\\": \\"select c1, c9, c8\\\\nfrom t1\\\\nwhere c3 like \'SSU%\' and c8 > c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}, {\\"__index__\\": 1, \\"sqlquery\\": \\"select c1, c8, c9\\\\nfrom t1\\\\nwhere c3 like \'LSU%\' and c8 <= c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}, {\\"__index__\\": 2, \\"sqlquery\\": \\"select c1, c9, c8\\\\nfrom t1\\\\nwhere c3 like \'LSU%\' and c8 > c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}]}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"select c1, c8, c9\\\\nfrom t1\\\\nwhere c3 like \'SSU%\' and c8 <= c9;\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"691d3274-8f28-417f-b4c6-4b52c6581caa","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":6,"input_connections":{"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"SSU BED forward","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1223.5622225911102,"top":634.7040213846066},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"forward\\\\\\", \\\\\\"1\\\\\\", \\\\\\"+\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"49102ad6-550d-46ae-bc61-d02e7a2cd4b1","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":7,"input_connections":{"infile":{"id":5,"output_name":"output1"}},"inputs":[],"label":"SSU BED reverse","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1227.7324324034817,"top":774.7051496457692},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"reverse\\\\\\", \\\\\\"1\\\\\\", \\\\\\"-\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b6ab449c-691d-43f2-bc4b-136a530805ab","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":8,"input_connections":{"infile":{"id":5,"output_name":"output2"}},"inputs":[],"label":"LSU BED forward","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1233.7324324034817,"top":907.706567048101},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"forward\\\\\\", \\\\\\"1\\\\\\", \\\\\\"+\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"52b15892-3864-439a-8526-f520a1f3e048","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":5,"output_name":"output3"}},"inputs":[],"label":"LSU BED reverse","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1224.7324324034817,"top":1033.2232296457578},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"reverse\\\\\\", \\\\\\"1\\\\\\", \\\\\\"-\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"2dc2caf9-8607-42a9-a73d-d570de075554","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":10,"input_connections":{"input1":{"id":6,"output_name":"outfile"},"input2":{"id":7,"output_name":"outfile"}},"inputs":[],"label":"SSU BED","name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1526.7324324034817,"top":664.2232296457572},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU_BED"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"f72fb224-f211-4059-9ee5-9a10efdc0c28","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":11,"input_connections":{"input1":{"id":8,"output_name":"outfile"},"input2":{"id":9,"output_name":"outfile"}},"inputs":[],"label":"LSU BED","name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1505.4794369783956,"top":979.4333768742844},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU_BED"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"802aedfc-dd96-4e90-a31d-b4a5e0c0324c","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1414.4017321950623,"top":282.81198514838457},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"a3fafb37-f848-4c6f-9257-99d9d6cc5607","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":13,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1230.527671267726,"top":1226.1422039003264},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"fc4e1062-e37e-4fda-b064-555d23cc2a67","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":14,"input_connections":{"input1":{"id":10,"output_name":"output"},"input2":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1966.56641902912,"top":1498.3057522310658},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU and SSU BED regions"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"975d4e12-b21c-46ce-b0eb-f9ac7b26f699","when":null,"workflow_outputs":[{"label":"LSU and SSU BED regions","output_name":"output","uuid":"5b559599-e3a4-4d82-86fa-b4561ada0b9a"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":15,"input_connections":{"input_collection":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1639.1793949459288,"top":303.2738932291369},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"80a5e054-f163-4e90-b40f-f0c281add51b","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":16,"input_connections":{"input_collection":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1473.4218235553517,"top":1215.607942233841},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"3082ec60-c7f2-44fd-8289-27c3aaba02f6","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":17,"input_connections":{"how|filter_source":{"id":15,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":1864.5137008919908,"top":391.13461794967463},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"83193908-189b-425b-bf16-a40e35f04fd1","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":18,"input_connections":{"how|filter_source":{"id":16,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":1705.6931247050977,"top":1198.1687448048988},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b07075b5-c465-4bb7-9480-38436e5d42fe","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","errors":null,"id":19,"input_connections":{"fasta_source|fasta":{"id":17,"output_name":"output_filtered"},"input":{"id":12,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1903.3953651059337,"top":674.4302045250132},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"5ba0a701-dd95-4e11-afdd-4cb20e4a8169","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","errors":null,"id":20,"input_connections":{"fasta_source|fasta":{"id":18,"output_name":"output_filtered"},"input":{"id":13,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1907.7391313705002,"top":1010.1390284057046},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"2e79af1d-62d7-49ed-bfe1-e2f3a78faced","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":21,"input_connections":{"input":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2199.186741916243,"top":780.6023526895299},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"cd8fcbf1-866a-4214-bdea-874c1f94f10d","when":null,"workflow_outputs":[{"label":"SSU FASTA files","output_name":"output","uuid":"71c69978-674b-496a-a3a1-e0ff2bf1c4db"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":22,"input_connections":{"input":{"id":20,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2049.6678134598683,"top":1251.106870676105},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"362611c9-757f-424a-ad76-58ee1e3ad479","when":null,"workflow_outputs":[{"label":"LSU FASTA files","output_name":"output","uuid":"0bf0c613-7c89-489e-adbb-e86a57ad13a1"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":23,"input_connections":{"sequences":{"id":21,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":2131.9295017384748,"top":507.60960896405254},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format SSU SILVA"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid SSU SILVA"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_ssu_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"786e29b6-b447-4705-beee-1fbbd6cc64c8","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":24,"input_connections":{"sequences":{"id":22,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":2216.9166870117188,"top":998.5201836686707},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format LSU SILVA"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid LSU SILVA"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_lsu_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"89474568-b342-4376-a52d-20878582d5db","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":23,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2341.918447993722,"top":343.73564945106307},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"6bbc2f91-cdd5-4cda-8923-1c4aafc793d1","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":26,"input_connections":{"infile":{"id":23,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2484.875930049533,"top":671.8771045356158},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"18302666-59ae-4441-b6cd-b46d8a400789","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":27,"input_connections":{"infile":{"id":23,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2682.405523606579,"top":452.39977836112007},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ba332896-373b-475b-98e1-39a4457c8153","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":28,"input_connections":{"input":{"id":23,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3020.6201097810404,"top":729.4160842644015},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"257b8f02-f88a-4b1a-926d-c2f7bd0a2a56","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":29,"input_connections":{"infile":{"id":24,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2459.472750071673,"top":964.1860471272424},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"d4ab7aa0-e6eb-44a9-8e5b-2f275f7dc41c","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":30,"input_connections":{"infile":{"id":24,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2274.421541439101,"top":1375.2808207350452},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"fd27a4ca-fd35-4c6b-aafe-b442d42531f0","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":31,"input_connections":{"infile":{"id":24,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2904.532618934821,"top":1143.3812048537206},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"fe9d4c36-faa8-4bdc-8c3c-3c230dd44ffc","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":32,"input_connections":{"input":{"id":24,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3004.2019554771114,"top":1427.490860606865},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0f82c7b0-6313-4637-8572-0e25e4b61222","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":33,"input_connections":{"input":{"id":25,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2501.3333740234375,"top":170.56666564941406},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU taxonomic classifications using SILVA DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"02c81ee5-d2a0-42da-b7d1-cc432cc58431","when":null,"workflow_outputs":[{"label":"SSU taxonomic classifications using SILVA DB","output_name":"output","uuid":"11632f64-a951-49d9-a46c-85285ba9234d"}]},"34":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":34,"input_connections":{"input":{"id":26,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2755.107387901881,"top":663.6467048040722},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU OTU tables (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"db62999a-75ce-4448-a4bb-ffd0a0c8f070","when":null,"workflow_outputs":[{"label":"SSU OTU tables (SILVA DB)","output_name":"output","uuid":"9fa11853-687d-49c9-94e2-4ac1b6a9de02"}]},"35":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":35,"input_connections":{"input":{"id":27,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2929.3175155528866,"top":456.5149782268919},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ad1c259c-a1e5-496b-aaae-e9841f4086bb","when":null,"workflow_outputs":[]},"36":{"annotation":"","id":36,"input_connections":{"Input collection":{"id":28,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3253.470085366978,"top":701.8494338737765},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"a5aa7d4c-ece2-4cff-82ff-f6320ad251ff","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":37,"input_connections":{"input":{"id":29,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2692.5435761342164,"top":970.0420980333365},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU taxonomic classifications using SILVA DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"fc13fe38-e24b-485d-98d7-da38c4ecea06","when":null,"workflow_outputs":[{"label":"LSU taxonomic classifications using SILVA DB","output_name":"output","uuid":"90446073-dd91-4132-92fc-b3d38644a47c"}]},"38":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":38,"input_connections":{"input":{"id":30,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2513.3484184077383,"top":1365.9111392852947},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU OTU tables (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0e810c9b-8f1a-417e-906e-78d600cd51c8","when":null,"workflow_outputs":[{"label":"LSU OTU tables (SILVA DB)","output_name":"output","uuid":"ecb9dba5-a182-463e-a06a-18edbd37a46e"}]},"39":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":39,"input_connections":{"input":{"id":31,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3149.3155468095524,"top":1143.3812048537206},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"552cb861-a006-4dc7-8ff7-f5af10d5cb93","when":null,"workflow_outputs":[]},"40":{"annotation":"","id":40,"input_connections":{"Input collection":{"id":32,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3314.9631045505002,"top":1517.3226958033472},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"fa552732-31b6-4928-991e-5b966dc4da69","when":null,"workflow_outputs":[]},"41":{"annotation":"","id":41,"input_connections":{"Input collection":{"id":35,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3179.470085366978,"top":408.84943387377643},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"aee25341-bf68-4cd8-b3fe-769a94512bf4","when":null,"workflow_outputs":[]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":42,"input_connections":{"type_of_data|input":{"id":28,"output_name":"output"},"when":{"id":36,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona SSU SILVA","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":3521.7643846129777,"top":626.8193285739657},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU taxonomic abundance pie charts (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"1da968ab-cec0-4db1-b53a-ebe6f035c2fe","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU taxonomic abundance pie charts (SILVA DB)","output_name":"output","uuid":"2571ed1e-04ee-4078-8f43-a406b6e67dbe"}]},"43":{"annotation":"","id":43,"input_connections":{"Input collection":{"id":39,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3408.3308199842286,"top":1137.1888001240086},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"c26c4423-280d-4d7e-bed2-7390b14873ce","when":null,"workflow_outputs":[]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":44,"input_connections":{"type_of_data|input":{"id":32,"output_name":"output"},"when":{"id":40,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona LSU SILVA","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":3670.2216351015004,"top":1492.9828576583009},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU taxonomic abundance pie charts (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"a6de961c-2f40-478a-9ae6-6c33fd9e28bc","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU taxonomic abundance pie charts (SILVA DB)","output_name":"output","uuid":"f4c8977e-7b91-4f4f-b112-4eaa744d7834"}]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":45,"input_connections":{"input|input_fp":{"id":35,"output_name":"output"},"when":{"id":41,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 SSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3498.725031827062,"top":0},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"SSU OTU tables in HDF5 format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"26c1df0e-7c23-4a6f-8859-420a2787dee0","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU OTU tables in HDF5 format (SILVA DB)","output_name":"output_fp","uuid":"e9ed7fe6-a7f3-42a4-bb51-0a4631d38a73"}]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":46,"input_connections":{"input|input_fp":{"id":35,"output_name":"output"},"when":{"id":41,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON SSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3500.6630859470188,"top":321.52388658376185},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"SSU OTU tables in JSON format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"879dd786-fbb9-4054-9bc4-42eb4a9b1193","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU OTU tables in JSON format (SILVA DB)","output_name":"output_fp","uuid":"3937d133-7118-40b9-b88a-ed382b51603d"}]},"47":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":47,"input_connections":{"input|input_fp":{"id":39,"output_name":"output"},"when":{"id":43,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 LSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3674.4440326059353,"top":837.6260676655736},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"LSU OTU tables in HDF5 format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"0429ca95-6661-4467-9585-56e4310d7a05","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU OTU tables in HDF5 format (SILVA DB)","output_name":"output_fp","uuid":"ed10309d-7874-4c3b-952e-4a9e3f85d918"}]},"48":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":48,"input_connections":{"input|input_fp":{"id":39,"output_name":"output"},"when":{"id":43,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON LSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3670.48882921932,"top":1146.8376201123672},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"LSU OTU tables in JSON format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"850a6179-db1c-4865-a689-0e7dd4b471f0","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU OTU tables in JSON format (SILVA DB)","output_name":"output_fp","uuid":"2f2a6bc2-cc77-4266-aeb7-eae31654d930"}]}},"tags":[],"uuid":"53c5bc12-4d7c-4799-b053-ced394665d61"},"tool_id":null,"type":"subworkflow","uuid":"ce11d4ec-e858-4aa8-a6f3-fe7c8d31be7d","when":null,"workflow_outputs":[{"label":"LSU and SSU BED regions","output_name":"LSU and SSU BED regions","uuid":"2c0868a6-e45b-4738-9f2e-8395c49f6af1"},{"label":"LSU OTU tables (SILVA DB)","output_name":"LSU OTU tables (SILVA DB)","uuid":"6b36b9af-d003-4cc2-a179-a2a6c05090ad"},{"label":"SSU taxonomic classifications using SILVA DB","output_name":"SSU taxonomic classifications using SILVA DB","uuid":"3d7739ed-d444-4dbd-b686-b27e1aa1b7d0"},{"label":"SSU OTU tables in HDF5 format (SILVA DB)","output_name":"SSU OTU tables in HDF5 format (SILVA DB)","uuid":"6028cb8e-cb14-4447-a619-f848d39c4f57"},{"label":"SSU OTU tables (SILVA DB)","output_name":"SSU OTU tables (SILVA DB)","uuid":"7f21d964-f8bc-47b6-972d-7b39eb4b7b02"},{"label":"LSU taxonomic classifications using SILVA DB","output_name":"LSU taxonomic classifications using SILVA DB","uuid":"09f8843c-6250-4c8a-a2b3-3ceee8915d23"},{"label":"LSU OTU tables in JSON format (SILVA DB)","output_name":"LSU OTU tables in JSON format (SILVA DB)","uuid":"800565e9-4663-4b11-869a-d68fd09152ea"},{"label":"LSU OTU tables in HDF5 format (SILVA DB)","output_name":"LSU OTU tables in HDF5 format (SILVA DB)","uuid":"42c9e089-6d5b-4af5-9c7e-23165c8569eb"},{"label":"SSU OTU tables in JSON format (SILVA DB)","output_name":"SSU OTU tables in JSON format (SILVA DB)","uuid":"480b5147-b6f0-404b-838c-36f2818100f8"},{"label":"LSU taxonomic abundance pie charts (SILVA DB)","output_name":"LSU taxonomic abundance pie charts (SILVA DB)","uuid":"45a9891a-b535-4f6b-9a66-e86c53ee154b"},{"label":"SSU FASTA files","output_name":"SSU FASTA files","uuid":"ee71d9de-28d9-4177-acb6-9ef170d2ce25"},{"label":"LSU FASTA files","output_name":"LSU FASTA files","uuid":"a4371314-8837-4670-a6b2-1e528264ebd4"},{"label":"SSU taxonomic abundance pie charts (SILVA DB)","output_name":"SSU taxonomic abundance pie charts (SILVA DB)","uuid":"c57d0bc6-d624-4be5-a10b-74c3ad15ab17"}]},"32":{"annotation":"","id":32,"input_connections":{"LSU and SSU BED":{"id":31,"input_subworkflow_step_id":0,"output_name":"LSU and SSU BED regions"},"Processed sequences":{"id":30,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"MGnify\'s amplicon pipeline v5.0 - ITS","outputs":[],"position":{"left":2184.035198956191,"top":1032.541468976701},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - ITS","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A BED file containing the LSU and SSU coordinates.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A BED file containing the LSU and SSU coordinates.","name":"LSU and SSU BED"}],"label":"LSU and SSU BED","name":"Input dataset collection","outputs":[],"position":{"left":289.95001220703125,"top":225.9127507839193},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"c17b5c1a-8aac-4e43-be86-11ce0d59779b","when":null,"workflow_outputs":[]},"1":{"annotation":"Processed sequences.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Processed sequences.","name":"Processed sequences"}],"label":"Processed sequences","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":658.1099407040649},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9c72651d-0c23-41bb-8720-f9e603a4726e","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":730.4937643933138,"top":202.9418588363548},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"85b4e860-a20a-4dfa-8831-26a0819975d8","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":3,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":187.50176439331375,"top":493.9418588363548},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"9a909506-5071-466a-8f5b-63cd2bb81254","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":4,"input_connections":{"how|filter_source":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":514.0843031728684,"top":525.9078034884216},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"25bbbc01-9c16-484c-925a-6758ac7e8c08","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_maskfastabed/2.31.1","errors":null,"id":5,"input_connections":{"fasta":{"id":4,"output_name":"output_filtered"},"input":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"bedtools MaskFastaBed","outputs":[{"name":"output","type":"fasta"}],"position":{"left":788.2104391980013,"top":514.7805968487736},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_maskfastabed/2.31.1","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fullheader\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mc\\": \\"N\\", \\"soft\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"7448f5ef-6091-4e73-91fb-dfb97b657aa2","when":null,"workflow_outputs":[{"label":"ITS FASTA files","output_name":"output","uuid":"0527bde7-8e9d-49f1-88d1-b6e4095e2095"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":6,"input_connections":{"sequences":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":1071.4000244140625,"top":429.3085275375267},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format ITSonedb"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid ITSonedb"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_its_itsonedb_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"8ba4b427-6376-44b4-b77a-8cb07dd7daef","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":7,"input_connections":{"sequences":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":1091.1791513437531,"top":944.89646009585},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format ITS UNITE"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid ITS UNITE"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_its_unite_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"5978536f-8c00-49a7-b426-8120f23a3875","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":8,"input_connections":{"infile":{"id":6,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1364.155596001663,"top":122.12686327418507},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"12a9401c-95a3-4da6-8ca6-ced17b6b3b70","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":6,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1371.014262444616,"top":339.04363878350205},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"94df1cca-c1c3-4643-94ff-dca5cbca8467","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":6,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1726.2815522788949,"top":455.65530978203265},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3d0ba884-d155-4bf6-990c-f589898bae51","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":11,"input_connections":{"input":{"id":6,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1820.6340128732131,"top":615.8965977845029},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"85c966df-4f74-4c8b-81ba-b4f73829c18b","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":7,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1314.1580115527963,"top":890.3766404571136},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"165a1fc2-87da-4a8c-8e5c-16fb4d9e84ce","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":13,"input_connections":{"infile":{"id":7,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1280.5052342467761,"top":1234.4196311901392},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"87917797-f8cc-4670-b3b1-28b571fc9e07","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":14,"input_connections":{"infile":{"id":7,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1701.3738942940215,"top":1121.663903849118},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"a2784daf-528d-49a6-a183-acb2c56b3c5e","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":15,"input_connections":{"input":{"id":7,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1943.182478535459,"top":1424.172257451561},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"447e2f51-931c-428d-97fd-0e3e03ee3737","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":16,"input_connections":{"input":{"id":8,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1578.1459890217961,"top":123.20139745137587},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic classifications using ITSoneDB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"974b7a42-0e82-4e0e-8630-3827185e275b","when":null,"workflow_outputs":[{"label":"ITS taxonomic classifications using ITSoneDB","output_name":"output","uuid":"26936de2-fb56-493a-89fe-dbcd1c98df2c"}]},"17":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":17,"input_connections":{"input":{"id":9,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1624.7849208338766,"top":294.6680585252006},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS OTU tables (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d0054748-c9d4-4f29-8640-818060d1a8ff","when":null,"workflow_outputs":[{"label":"ITS OTU tables (ITSoneDB)","output_name":"output","uuid":"030ee411-0b04-4ade-ba0c-715bbd561460"}]},"18":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":18,"input_connections":{"input":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1999.2564767084239,"top":452.431770043427},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"52c2edac-b2b7-434a-9feb-a5e0faa430fe","when":null,"workflow_outputs":[]},"19":{"annotation":"","id":19,"input_connections":{"Input collection":{"id":11,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2141.802455522449,"top":611.9414781218429},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"50932610-dbc1-4fa4-84af-e6c605206eda","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":20,"input_connections":{"input":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1573.7895192434253,"top":872.3032171286626},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic classifications using UNITE DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"c425bf1a-bda0-454c-8d3e-359144e49706","when":null,"workflow_outputs":[{"label":"ITS taxonomic classifications using UNITE DB","output_name":"output","uuid":"febb8ce1-72a0-44c8-a103-490bec12c30c"}]},"21":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":21,"input_connections":{"input":{"id":13,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1515.2157831570244,"top":1272.8548262411377},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS OTU tables (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"15cb2eee-88c1-43dc-a58c-211ed446bb99","when":null,"workflow_outputs":[{"label":"ITS OTU tables (UNITE DB)","output_name":"output","uuid":"8304da79-296c-4cfa-bd38-671cbe727354"}]},"22":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":22,"input_connections":{"input":{"id":14,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1973.1727179528675,"top":1141.672125434937},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"16b97913-9901-4b40-ac60-7b2c5e465b48","when":null,"workflow_outputs":[]},"23":{"annotation":"","id":23,"input_connections":{"Input collection":{"id":15,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2216.80260235576,"top":1483.9821779220883},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"e51d163f-6f26-4964-97a1-47d0aece539c","when":null,"workflow_outputs":[]},"24":{"annotation":"","id":24,"input_connections":{"Input collection":{"id":18,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2222.7347195492944,"top":271.7516225513746},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"ebe35b8e-b72c-432a-ad92-f79e6b44ff21","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":25,"input_connections":{"type_of_data|input":{"id":11,"output_name":"output"},"when":{"id":19,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona ITSoneDB","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":2497.0678472201726,"top":673.2064707870127},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic abundance pie charts (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"d8f34e02-6822-443b-b4b4-938c06209006","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS taxonomic abundance pie charts (ITSoneDB)","output_name":"output","uuid":"c28aa604-a717-4744-acf9-aa0d4cb52af1"}]},"26":{"annotation":"","id":26,"input_connections":{"Input collection":{"id":22,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2223.6555735015286,"top":1091.0953467133058},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"0b57266d-c6ff-4b87-926a-0226838507d4"},"tool_id":null,"type":"subworkflow","uuid":"e5486218-e068-4d84-92b4-7dd0d3bf1b8f","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":27,"input_connections":{"type_of_data|input":{"id":15,"output_name":"output"},"when":{"id":23,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona ITS UNITE","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":2518.6399730672456,"top":1456.679375085977},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"ITS taxonomic abundance pie charts (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"526509f4-006f-4c4c-938b-f40d883fe7d2","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS taxonomic abundance pie charts (UNITE DB)","output_name":"output","uuid":"73a03453-b02c-4626-82a5-7d8268180b42"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":28,"input_connections":{"input|input_fp":{"id":18,"output_name":"output"},"when":{"id":24,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 ITSoneDB","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2489.545576441207,"top":0},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in HDF5 format (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"41a033f9-248d-46b6-8ac0-043b7d95d1df","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in HDF5 format (ITSoneDB)","output_name":"output_fp","uuid":"3f5626b9-9e0b-4efe-9e23-ddc7072dec7b"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":29,"input_connections":{"input|input_fp":{"id":18,"output_name":"output"},"when":{"id":24,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON ITSoneDB","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2491.8688075155605,"top":364.1607245502264},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in JSON format (ITSoneDB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"7b86eed5-628f-4a38-ae64-d664e76ce54f","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in JSON format (ITSoneDB)","output_name":"output_fp","uuid":"3ef29ecf-ea57-457e-b244-cb7fea77203a"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":30,"input_connections":{"input|input_fp":{"id":22,"output_name":"output"},"when":{"id":26,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 ITS UNITE","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2511.9725936708223,"top":870.9977416620931},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in HDF5 format (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"5ae011cb-274f-48a9-90c5-b570d2893eb2","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in HDF5 format (UNITE DB)","output_name":"output_fp","uuid":"a51b6afd-2b7e-4e59-a4fa-73103d2366b0"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":31,"input_connections":{"input|input_fp":{"id":22,"output_name":"output"},"when":{"id":26,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON ITS UNITE","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":2512.0796733881634,"top":1169.6892588090332},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"ITS OTU tables in JSON format (UNITE DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"d70deaeb-19f3-4323-8553-688d95be40fd","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS OTU tables in JSON format (UNITE DB)","output_name":"output_fp","uuid":"96a0362f-8eed-4577-820b-a4f341092b95"}]}},"tags":[],"uuid":"04fa2e92-8fcd-4a20-8af1-063c7f514f12"},"tool_id":null,"type":"subworkflow","uuid":"169871f2-1966-4600-a8f9-4c5d4d40435d","when":null,"workflow_outputs":[{"label":"ITS OTU tables in HDF5 format (ITSoneDB)","output_name":"ITS OTU tables in HDF5 format (ITSoneDB)","uuid":"d7b793f8-f828-44e3-9af5-798158bd11d6"},{"label":"ITS FASTA files","output_name":"ITS FASTA files","uuid":"b7a20037-e419-4d1d-9e05-b96c6a9efca5"},{"label":"ITS OTU tables (ITSoneDB)","output_name":"ITS OTU tables (ITSoneDB)","uuid":"9d7fe164-5af7-4ef2-a439-a163705abfbb"},{"label":"ITS OTU tables in JSON format (UNITE DB)","output_name":"ITS OTU tables in JSON format (UNITE DB)","uuid":"6ef5a97b-63a2-4308-90a2-08a0f81057ea"},{"label":"ITS OTU tables in HDF5 format (UNITE DB)","output_name":"ITS OTU tables in HDF5 format (UNITE DB)","uuid":"a92dea0e-4bcc-4eb6-8052-cc897908e605"},{"label":"ITS taxonomic classifications using ITSoneDB","output_name":"ITS taxonomic classifications using ITSoneDB","uuid":"2aa27435-9a26-4f50-8d2a-9e07cfda66ba"},{"label":"ITS taxonomic abundance pie charts (UNITE DB)","output_name":"ITS taxonomic abundance pie charts (UNITE DB)","uuid":"f9c37e34-a124-4a1f-b9df-e834efbe0e53"},{"label":"ITS taxonomic classifications using UNITE DB","output_name":"ITS taxonomic classifications using UNITE DB","uuid":"384bb22d-6926-4833-89ed-1c9ec309aa7f"},{"label":"ITS taxonomic abundance pie charts (ITSoneDB)","output_name":"ITS taxonomic abundance pie charts (ITSoneDB)","uuid":"db0a8042-4964-4619-93e1-0d6a4ffdc5f1"},{"label":"ITS OTU tables in JSON format (ITSoneDB)","output_name":"ITS OTU tables in JSON format (ITSoneDB)","uuid":"13e54088-7391-4c8c-9fde-2dd63080fafd"},{"label":"ITS OTU tables (UNITE DB)","output_name":"ITS OTU tables (UNITE DB)","uuid":"582f0478-2a6b-4158-bd78-8144fc3a47bc"}]},"33":{"annotation":"","id":33,"input_connections":{"Input collection":{"id":31,"input_subworkflow_step_id":0,"output_name":"SSU OTU tables (SILVA DB)"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2502.0019974444936,"top":349.1472491921183},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"6c693925-c633-46d6-8a5a-78b4056ebe7e"},"tool_id":null,"type":"subworkflow","uuid":"04b6be05-c23d-45af-9510-af8c948db3b9","when":null,"workflow_outputs":[]},"34":{"annotation":"","id":34,"input_connections":{"Input collection":{"id":31,"input_subworkflow_step_id":0,"output_name":"LSU OTU tables (SILVA DB)"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2483.0019974444936,"top":714.1472491921184},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"6c693925-c633-46d6-8a5a-78b4056ebe7e"},"tool_id":null,"type":"subworkflow","uuid":"cfba285b-8cbc-49b9-81fb-4045df5da94a","when":null,"workflow_outputs":[]},"35":{"annotation":"","id":35,"input_connections":{"Input collection":{"id":32,"input_subworkflow_step_id":0,"output_name":"ITS OTU tables (ITSoneDB)"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2486,"top":1345.1493129882813},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"6c693925-c633-46d6-8a5a-78b4056ebe7e"},"tool_id":null,"type":"subworkflow","uuid":"2d700305-0a7f-42fd-8abe-70af6993966b","when":null,"workflow_outputs":[]},"36":{"annotation":"","id":36,"input_connections":{"Input collection":{"id":32,"input_subworkflow_step_id":0,"output_name":"ITS OTU tables (UNITE DB)"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":2503,"top":1768.5160122070313},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"6c693925-c633-46d6-8a5a-78b4056ebe7e"},"tool_id":null,"type":"subworkflow","uuid":"f63abbc2-d1c8-48fe-8e9f-99a70e9b541b","when":null,"workflow_outputs":[]},"37":{"annotation":"","id":37,"input_connections":{"OTU tables":{"id":31,"input_subworkflow_step_id":0,"output_name":"SSU OTU tables (SILVA DB)"},"Phylum level taxonomic abundance summary table name":{"id":14,"input_subworkflow_step_id":2,"output_name":"output"},"Taxonomic abundance summary table name":{"id":13,"input_subworkflow_step_id":1,"output_name":"output"},"when":{"id":33,"output_name":"boolean_param"}},"inputs":[],"label":null,"name":"MGnify amplicon summary tables","outputs":[],"position":{"left":2736.61669921875,"top":280.6660061035156},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[{"color":"turquoise","data":{"text":"phylum summary table"},"id":1,"position":[270.9406915224165,380.7370220536662],"size":[1353,243],"type":"markdown"},{"color":"green","data":{"text":"taxonomic summary table"},"id":0,"position":[272.3406915224165,120.13702205366614],"size":[909,241],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"MGnify amplicon summary tables","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":337.4772484997172},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ea3accdc-9c03-462f-b58e-8fedc647c253","when":null,"workflow_outputs":[]},"1":{"annotation":"specifies the name of taxonomic summary file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file.","name":"Taxonomic abundance summary table name"}],"label":"Taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":968.63330078125,"top":0},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"98862eac-2fa1-4e7d-95f9-ef4832b40b45","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"1b0794db-a154-4fd8-a701-763bd96c309e"}]},"2":{"annotation":"specifies the name of taxonomic summary file, for phylum level.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file, for phylum level.","name":"Phylum level taxonomic abundance summary table name"}],"label":"Phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":1415.5626469798628,"top":643.2277124832184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e948e6d0-a7f3-4d47-a50f-003ee055865a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c890d873-bb58-49ae-a1e7-a3e6d19368e0"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":3,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":298.42962390165496,"top":166.45843163713073},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c4,\\\\n    c3\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"b12026e3-5bd8-479d-a942-4858beae4a6a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":4,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.7278335508603,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 { \\\\n        split($3, taxa, \\\\\\";\\\\\\");\\\\n        merged = \\\\\\"\\\\\\";\\\\n        levels = 3; # We need sk__, k__, and p__ levels\\\\n        for (i = 1; i <= levels; i++) {\\\\n            if (i <= length(taxa) && taxa[i] ~ /__/) {\\\\n                split(taxa[i], parts, \\\\\\"__\\\\\\");\\\\n                if (parts[2] != \\\\\\"\\\\\\") {\\\\n                    value = parts[2];\\\\n                } else {\\\\n                    value = \\\\\\"unassigned\\\\\\";\\\\n                }\\\\n            } else {\\\\n                value = \\\\\\"unassigned\\\\\\";\\\\n            }\\\\n            merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;\\\\n        }\\\\n        print merged \\\\\\"\\\\\\\\t\\\\\\" $2;\\\\n    }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1df0be4a-13b7-438a-8940-c5c01f6b1707","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":523.9070991902965,"top":167.47348284368692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==1 {\\\\n    OFS=\\\\\\"\\\\\\\\t\\\\\\";\\\\n    print \\\\\\"#SampleID\\\\\\", $1;\\\\n}\\\\nNR>0 {\\\\n    print $2, $3;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"41005051-0bae-41c8-a28c-abacd01ebebc","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Grouping1","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":513.5768930139466,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"4c5830bb-82f1-4407-b37f-9e424cb8e797","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":7,"input_connections":{"input_tabular":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":743.2357210275393,"top":166.08754960449474},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"7239102d-4e4a-4aba-971b-bbb14c560d54","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":738.5411523428047,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"76a42aee-056b-4b82-b0b3-59a474e3dd07","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.0499877929688,"top":166.12169075483803},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the first header as it is\\\\n  printf \\\\\\"%s\\\\\\", $1\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Print the rest of the lines as they are\\\\n  for (i = 1; i <= NF; i++) {\\\\n    printf \\\\\\"%s%s\\\\\\", $i, (i < NF ? OFS : ORS)\\\\n  }\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3b0a6563-2da1-407f-ba77-2978a61ac2bc","when":null,"workflow_outputs":[{"label":"Taxonomic abundance summary table","output_name":"outfile","uuid":"ce7e544e-3f02-4ec9-a12d-0af3886b6bc5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":959.3902118058909,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b02b2b8c-1968-4e9c-89f0-0e1a26692663","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":11,"input_connections":{"input_tabular":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1180.2392712689773,"top":424.9709589271268},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"b6121ed1-f88b-4c5f-94f4-ed90b31cc089","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1407.949951171875,"top":423.5999755859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Phylum level taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the renamed headers for the new columns\\\\n  printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Split the first column by \';\' into three parts\\\\n  split($1, taxa_parts, \\\\\\";\\\\\\")\\\\n  superkingdom = (taxa_parts[1] != \\\\\\"\\\\\\" ? taxa_parts[1] : \\\\\\"unassigned\\\\\\")\\\\n  kingdom = (length(taxa_parts[2]) ? taxa_parts[2] : \\\\\\"unassigned\\\\\\")\\\\n  phylum = (length(taxa_parts[3]) ? taxa_parts[3] : \\\\\\"unassigned\\\\\\")\\\\n  printf \\\\\\"%s\\\\\\\\t%s\\\\\\\\t%s\\\\\\", superkingdom, kingdom, phylum\\\\n  # Process the remaining columns\\\\n  for (i = 2; i <= NF; i++) {\\\\n    if ($i != \\\\\\"\\\\\\") {\\\\n      split($i, part, \\\\\\"_\\\\\\")\\\\n      printf \\\\\\"\\\\\\\\t%s\\\\\\", part[1]\\\\n    } else {\\\\n      printf \\\\\\"\\\\\\\\tunassigned\\\\\\"\\\\n    }\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ae416cc9-c888-4eef-bc43-b15a1ac9899f","when":null,"workflow_outputs":[{"label":"phylum level taxonomic abundance summary table","output_name":"outfile","uuid":"58153320-f0c2-4d63-aa04-43f04c791da2"}]}},"tags":[],"uuid":"dda70895-1e39-42a3-8b4f-b6493cf15e8f"},"tool_id":null,"type":"subworkflow","uuid":"4ddff127-ed1a-4eab-bcfe-8715f2284618","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU taxonomic abundance summary table","output_name":"Taxonomic abundance summary table","uuid":"4efd5e53-32f9-4e1b-b850-ed9c6e96cbb3"},{"label":"SSU phylum level taxonomic abundance summary table","output_name":"phylum level taxonomic abundance summary table","uuid":"b8be6baa-41d2-4652-87dc-f1fb9abc07ee"}]},"38":{"annotation":"","id":38,"input_connections":{"OTU tables":{"id":31,"input_subworkflow_step_id":0,"output_name":"LSU OTU tables (SILVA DB)"},"Phylum level taxonomic abundance summary table name":{"id":16,"input_subworkflow_step_id":2,"output_name":"output"},"Taxonomic abundance summary table name":{"id":15,"input_subworkflow_step_id":1,"output_name":"output"},"when":{"id":34,"output_name":"boolean_param"}},"inputs":[],"label":null,"name":"MGnify amplicon summary tables","outputs":[],"position":{"left":2740.7999877929688,"top":697.5499877929688},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[{"color":"turquoise","data":{"text":"phylum summary table"},"id":1,"position":[270.9406915224165,380.7370220536662],"size":[1353,243],"type":"markdown"},{"color":"green","data":{"text":"taxonomic summary table"},"id":0,"position":[272.3406915224165,120.13702205366614],"size":[909,241],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"MGnify amplicon summary tables","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":337.4772484997172},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ea3accdc-9c03-462f-b58e-8fedc647c253","when":null,"workflow_outputs":[]},"1":{"annotation":"specifies the name of taxonomic summary file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file.","name":"Taxonomic abundance summary table name"}],"label":"Taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":968.63330078125,"top":0},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"98862eac-2fa1-4e7d-95f9-ef4832b40b45","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"1b0794db-a154-4fd8-a701-763bd96c309e"}]},"2":{"annotation":"specifies the name of taxonomic summary file, for phylum level.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file, for phylum level.","name":"Phylum level taxonomic abundance summary table name"}],"label":"Phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":1415.5626469798628,"top":643.2277124832184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e948e6d0-a7f3-4d47-a50f-003ee055865a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c890d873-bb58-49ae-a1e7-a3e6d19368e0"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":3,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":298.42962390165496,"top":166.45843163713073},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c4,\\\\n    c3\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"b12026e3-5bd8-479d-a942-4858beae4a6a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":4,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.7278335508603,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 { \\\\n        split($3, taxa, \\\\\\";\\\\\\");\\\\n        merged = \\\\\\"\\\\\\";\\\\n        levels = 3; # We need sk__, k__, and p__ levels\\\\n        for (i = 1; i <= levels; i++) {\\\\n            if (i <= length(taxa) && taxa[i] ~ /__/) {\\\\n                split(taxa[i], parts, \\\\\\"__\\\\\\");\\\\n                if (parts[2] != \\\\\\"\\\\\\") {\\\\n                    value = parts[2];\\\\n                } else {\\\\n                    value = \\\\\\"unassigned\\\\\\";\\\\n                }\\\\n            } else {\\\\n                value = \\\\\\"unassigned\\\\\\";\\\\n            }\\\\n            merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;\\\\n        }\\\\n        print merged \\\\\\"\\\\\\\\t\\\\\\" $2;\\\\n    }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1df0be4a-13b7-438a-8940-c5c01f6b1707","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":523.9070991902965,"top":167.47348284368692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==1 {\\\\n    OFS=\\\\\\"\\\\\\\\t\\\\\\";\\\\n    print \\\\\\"#SampleID\\\\\\", $1;\\\\n}\\\\nNR>0 {\\\\n    print $2, $3;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"41005051-0bae-41c8-a28c-abacd01ebebc","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Grouping1","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":513.5768930139466,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"4c5830bb-82f1-4407-b37f-9e424cb8e797","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":7,"input_connections":{"input_tabular":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":743.2357210275393,"top":166.08754960449474},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"7239102d-4e4a-4aba-971b-bbb14c560d54","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":738.5411523428047,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"76a42aee-056b-4b82-b0b3-59a474e3dd07","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.0499877929688,"top":166.12169075483803},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the first header as it is\\\\n  printf \\\\\\"%s\\\\\\", $1\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Print the rest of the lines as they are\\\\n  for (i = 1; i <= NF; i++) {\\\\n    printf \\\\\\"%s%s\\\\\\", $i, (i < NF ? OFS : ORS)\\\\n  }\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3b0a6563-2da1-407f-ba77-2978a61ac2bc","when":null,"workflow_outputs":[{"label":"Taxonomic abundance summary table","output_name":"outfile","uuid":"ce7e544e-3f02-4ec9-a12d-0af3886b6bc5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":959.3902118058909,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b02b2b8c-1968-4e9c-89f0-0e1a26692663","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":11,"input_connections":{"input_tabular":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1180.2392712689773,"top":424.9709589271268},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"b6121ed1-f88b-4c5f-94f4-ed90b31cc089","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1407.949951171875,"top":423.5999755859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Phylum level taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the renamed headers for the new columns\\\\n  printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Split the first column by \';\' into three parts\\\\n  split($1, taxa_parts, \\\\\\";\\\\\\")\\\\n  superkingdom = (taxa_parts[1] != \\\\\\"\\\\\\" ? taxa_parts[1] : \\\\\\"unassigned\\\\\\")\\\\n  kingdom = (length(taxa_parts[2]) ? taxa_parts[2] : \\\\\\"unassigned\\\\\\")\\\\n  phylum = (length(taxa_parts[3]) ? taxa_parts[3] : \\\\\\"unassigned\\\\\\")\\\\n  printf \\\\\\"%s\\\\\\\\t%s\\\\\\\\t%s\\\\\\", superkingdom, kingdom, phylum\\\\n  # Process the remaining columns\\\\n  for (i = 2; i <= NF; i++) {\\\\n    if ($i != \\\\\\"\\\\\\") {\\\\n      split($i, part, \\\\\\"_\\\\\\")\\\\n      printf \\\\\\"\\\\\\\\t%s\\\\\\", part[1]\\\\n    } else {\\\\n      printf \\\\\\"\\\\\\\\tunassigned\\\\\\"\\\\n    }\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ae416cc9-c888-4eef-bc43-b15a1ac9899f","when":null,"workflow_outputs":[{"label":"phylum level taxonomic abundance summary table","output_name":"outfile","uuid":"58153320-f0c2-4d63-aa04-43f04c791da2"}]}},"tags":[],"uuid":"dda70895-1e39-42a3-8b4f-b6493cf15e8f"},"tool_id":null,"type":"subworkflow","uuid":"048da8ae-fbf7-4afe-ad77-85428df7b486","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU phylum level taxonomic abundance summary table","output_name":"phylum level taxonomic abundance summary table","uuid":"e7425613-0d16-47f5-86c9-b881908ca21e"},{"label":"LSU taxonomic abundance summary table","output_name":"Taxonomic abundance summary table","uuid":"203f7e9c-346a-44b3-82fb-c00beab8049f"}]},"39":{"annotation":"","id":39,"input_connections":{"OTU tables":{"id":32,"input_subworkflow_step_id":0,"output_name":"ITS OTU tables (ITSoneDB)"},"Phylum level taxonomic abundance summary table name":{"id":18,"input_subworkflow_step_id":2,"output_name":"output"},"Taxonomic abundance summary table name":{"id":17,"input_subworkflow_step_id":1,"output_name":"output"},"when":{"id":35,"output_name":"boolean_param"}},"inputs":[{"description":"","name":"ITSoneDB summary tables"}],"label":"ITSoneDB summary tables","name":"MGnify amplicon summary tables","outputs":[],"position":{"left":2798.1953869283334,"top":1113.7202008303634},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[{"color":"turquoise","data":{"text":"phylum summary table"},"id":1,"position":[270.9406915224165,380.7370220536662],"size":[1353,243],"type":"markdown"},{"color":"green","data":{"text":"taxonomic summary table"},"id":0,"position":[272.3406915224165,120.13702205366614],"size":[909,241],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"MGnify amplicon summary tables","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":337.4772484997172},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ea3accdc-9c03-462f-b58e-8fedc647c253","when":null,"workflow_outputs":[]},"1":{"annotation":"specifies the name of taxonomic summary file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file.","name":"Taxonomic abundance summary table name"}],"label":"Taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":968.63330078125,"top":0},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"98862eac-2fa1-4e7d-95f9-ef4832b40b45","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"1b0794db-a154-4fd8-a701-763bd96c309e"}]},"2":{"annotation":"specifies the name of taxonomic summary file, for phylum level.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file, for phylum level.","name":"Phylum level taxonomic abundance summary table name"}],"label":"Phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":1415.5626469798628,"top":643.2277124832184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e948e6d0-a7f3-4d47-a50f-003ee055865a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c890d873-bb58-49ae-a1e7-a3e6d19368e0"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":3,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":298.42962390165496,"top":166.45843163713073},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c4,\\\\n    c3\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"b12026e3-5bd8-479d-a942-4858beae4a6a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":4,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.7278335508603,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 { \\\\n        split($3, taxa, \\\\\\";\\\\\\");\\\\n        merged = \\\\\\"\\\\\\";\\\\n        levels = 3; # We need sk__, k__, and p__ levels\\\\n        for (i = 1; i <= levels; i++) {\\\\n            if (i <= length(taxa) && taxa[i] ~ /__/) {\\\\n                split(taxa[i], parts, \\\\\\"__\\\\\\");\\\\n                if (parts[2] != \\\\\\"\\\\\\") {\\\\n                    value = parts[2];\\\\n                } else {\\\\n                    value = \\\\\\"unassigned\\\\\\";\\\\n                }\\\\n            } else {\\\\n                value = \\\\\\"unassigned\\\\\\";\\\\n            }\\\\n            merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;\\\\n        }\\\\n        print merged \\\\\\"\\\\\\\\t\\\\\\" $2;\\\\n    }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1df0be4a-13b7-438a-8940-c5c01f6b1707","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":523.9070991902965,"top":167.47348284368692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==1 {\\\\n    OFS=\\\\\\"\\\\\\\\t\\\\\\";\\\\n    print \\\\\\"#SampleID\\\\\\", $1;\\\\n}\\\\nNR>0 {\\\\n    print $2, $3;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"41005051-0bae-41c8-a28c-abacd01ebebc","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Grouping1","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":513.5768930139466,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"4c5830bb-82f1-4407-b37f-9e424cb8e797","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":7,"input_connections":{"input_tabular":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":743.2357210275393,"top":166.08754960449474},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"7239102d-4e4a-4aba-971b-bbb14c560d54","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":738.5411523428047,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"76a42aee-056b-4b82-b0b3-59a474e3dd07","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.0499877929688,"top":166.12169075483803},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the first header as it is\\\\n  printf \\\\\\"%s\\\\\\", $1\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Print the rest of the lines as they are\\\\n  for (i = 1; i <= NF; i++) {\\\\n    printf \\\\\\"%s%s\\\\\\", $i, (i < NF ? OFS : ORS)\\\\n  }\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3b0a6563-2da1-407f-ba77-2978a61ac2bc","when":null,"workflow_outputs":[{"label":"Taxonomic abundance summary table","output_name":"outfile","uuid":"ce7e544e-3f02-4ec9-a12d-0af3886b6bc5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":959.3902118058909,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b02b2b8c-1968-4e9c-89f0-0e1a26692663","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":11,"input_connections":{"input_tabular":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1180.2392712689773,"top":424.9709589271268},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"b6121ed1-f88b-4c5f-94f4-ed90b31cc089","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1407.949951171875,"top":423.5999755859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Phylum level taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the renamed headers for the new columns\\\\n  printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Split the first column by \';\' into three parts\\\\n  split($1, taxa_parts, \\\\\\";\\\\\\")\\\\n  superkingdom = (taxa_parts[1] != \\\\\\"\\\\\\" ? taxa_parts[1] : \\\\\\"unassigned\\\\\\")\\\\n  kingdom = (length(taxa_parts[2]) ? taxa_parts[2] : \\\\\\"unassigned\\\\\\")\\\\n  phylum = (length(taxa_parts[3]) ? taxa_parts[3] : \\\\\\"unassigned\\\\\\")\\\\n  printf \\\\\\"%s\\\\\\\\t%s\\\\\\\\t%s\\\\\\", superkingdom, kingdom, phylum\\\\n  # Process the remaining columns\\\\n  for (i = 2; i <= NF; i++) {\\\\n    if ($i != \\\\\\"\\\\\\") {\\\\n      split($i, part, \\\\\\"_\\\\\\")\\\\n      printf \\\\\\"\\\\\\\\t%s\\\\\\", part[1]\\\\n    } else {\\\\n      printf \\\\\\"\\\\\\\\tunassigned\\\\\\"\\\\n    }\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ae416cc9-c888-4eef-bc43-b15a1ac9899f","when":null,"workflow_outputs":[{"label":"phylum level taxonomic abundance summary table","output_name":"outfile","uuid":"58153320-f0c2-4d63-aa04-43f04c791da2"}]}},"tags":[],"uuid":"dda70895-1e39-42a3-8b4f-b6493cf15e8f"},"tool_id":null,"type":"subworkflow","uuid":"7c750ebc-941e-4173-9684-23f98f9ec758","when":"$(inputs.when)","workflow_outputs":[{"label":"ITSoneDB taxonomic abundance summary table","output_name":"Taxonomic abundance summary table","uuid":"05613f35-22d4-4a32-8ddc-69d6a7281470"},{"label":"ITSoneDB phylum level taxonomic abundance summary table","output_name":"phylum level taxonomic abundance summary table","uuid":"82dce358-6820-47c4-a1a1-6e02506e66be"}]},"40":{"annotation":"","id":40,"input_connections":{"OTU tables":{"id":32,"input_subworkflow_step_id":0,"output_name":"ITS OTU tables (UNITE DB)"},"Phylum level taxonomic abundance summary table name":{"id":20,"input_subworkflow_step_id":2,"output_name":"output"},"Taxonomic abundance summary table name":{"id":19,"input_subworkflow_step_id":1,"output_name":"output"},"when":{"id":36,"output_name":"boolean_param"}},"inputs":[{"description":"","name":"UNITE summary tables"}],"label":"UNITE summary tables","name":"MGnify amplicon summary tables","outputs":[],"position":{"left":2798.5579081844076,"top":1500.7454629519505},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[{"color":"turquoise","data":{"text":"phylum summary table"},"id":1,"position":[270.9406915224165,380.7370220536662],"size":[1353,243],"type":"markdown"},{"color":"green","data":{"text":"taxonomic summary table"},"id":0,"position":[272.3406915224165,120.13702205366614],"size":[909,241],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"MGnify amplicon summary tables","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":337.4772484997172},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ea3accdc-9c03-462f-b58e-8fedc647c253","when":null,"workflow_outputs":[]},"1":{"annotation":"specifies the name of taxonomic summary file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file.","name":"Taxonomic abundance summary table name"}],"label":"Taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":968.63330078125,"top":0},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"98862eac-2fa1-4e7d-95f9-ef4832b40b45","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"1b0794db-a154-4fd8-a701-763bd96c309e"}]},"2":{"annotation":"specifies the name of taxonomic summary file, for phylum level.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file, for phylum level.","name":"Phylum level taxonomic abundance summary table name"}],"label":"Phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":1415.5626469798628,"top":643.2277124832184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e948e6d0-a7f3-4d47-a50f-003ee055865a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c890d873-bb58-49ae-a1e7-a3e6d19368e0"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":3,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":298.42962390165496,"top":166.45843163713073},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c4,\\\\n    c3\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"b12026e3-5bd8-479d-a942-4858beae4a6a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":4,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.7278335508603,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 { \\\\n        split($3, taxa, \\\\\\";\\\\\\");\\\\n        merged = \\\\\\"\\\\\\";\\\\n        levels = 3; # We need sk__, k__, and p__ levels\\\\n        for (i = 1; i <= levels; i++) {\\\\n            if (i <= length(taxa) && taxa[i] ~ /__/) {\\\\n                split(taxa[i], parts, \\\\\\"__\\\\\\");\\\\n                if (parts[2] != \\\\\\"\\\\\\") {\\\\n                    value = parts[2];\\\\n                } else {\\\\n                    value = \\\\\\"unassigned\\\\\\";\\\\n                }\\\\n            } else {\\\\n                value = \\\\\\"unassigned\\\\\\";\\\\n            }\\\\n            merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;\\\\n        }\\\\n        print merged \\\\\\"\\\\\\\\t\\\\\\" $2;\\\\n    }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"1df0be4a-13b7-438a-8940-c5c01f6b1707","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":523.9070991902965,"top":167.47348284368692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==1 {\\\\n    OFS=\\\\\\"\\\\\\\\t\\\\\\";\\\\n    print \\\\\\"#SampleID\\\\\\", $1;\\\\n}\\\\nNR>0 {\\\\n    print $2, $3;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"41005051-0bae-41c8-a28c-abacd01ebebc","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Grouping1","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":513.5768930139466,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"4c5830bb-82f1-4407-b37f-9e424cb8e797","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":7,"input_connections":{"input_tabular":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":743.2357210275393,"top":166.08754960449474},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"7239102d-4e4a-4aba-971b-bbb14c560d54","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":738.5411523428047,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"76a42aee-056b-4b82-b0b3-59a474e3dd07","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.0499877929688,"top":166.12169075483803},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the first header as it is\\\\n  printf \\\\\\"%s\\\\\\", $1\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Print the rest of the lines as they are\\\\n  for (i = 1; i <= NF; i++) {\\\\n    printf \\\\\\"%s%s\\\\\\", $i, (i < NF ? OFS : ORS)\\\\n  }\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3b0a6563-2da1-407f-ba77-2978a61ac2bc","when":null,"workflow_outputs":[{"label":"Taxonomic abundance summary table","output_name":"outfile","uuid":"ce7e544e-3f02-4ec9-a12d-0af3886b6bc5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":959.3902118058909,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b02b2b8c-1968-4e9c-89f0-0e1a26692663","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":11,"input_connections":{"input_tabular":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1180.2392712689773,"top":424.9709589271268},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"b6121ed1-f88b-4c5f-94f4-ed90b31cc089","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1407.949951171875,"top":423.5999755859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Phylum level taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the renamed headers for the new columns\\\\n  printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Split the first column by \';\' into three parts\\\\n  split($1, taxa_parts, \\\\\\";\\\\\\")\\\\n  superkingdom = (taxa_parts[1] != \\\\\\"\\\\\\" ? taxa_parts[1] : \\\\\\"unassigned\\\\\\")\\\\n  kingdom = (length(taxa_parts[2]) ? taxa_parts[2] : \\\\\\"unassigned\\\\\\")\\\\n  phylum = (length(taxa_parts[3]) ? taxa_parts[3] : \\\\\\"unassigned\\\\\\")\\\\n  printf \\\\\\"%s\\\\\\\\t%s\\\\\\\\t%s\\\\\\", superkingdom, kingdom, phylum\\\\n  # Process the remaining columns\\\\n  for (i = 2; i <= NF; i++) {\\\\n    if ($i != \\\\\\"\\\\\\") {\\\\n      split($i, part, \\\\\\"_\\\\\\")\\\\n      printf \\\\\\"\\\\\\\\t%s\\\\\\", part[1]\\\\n    } else {\\\\n      printf \\\\\\"\\\\\\\\tunassigned\\\\\\"\\\\n    }\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ae416cc9-c888-4eef-bc43-b15a1ac9899f","when":null,"workflow_outputs":[{"label":"phylum level taxonomic abundance summary table","output_name":"outfile","uuid":"58153320-f0c2-4d63-aa04-43f04c791da2"}]}},"tags":[],"uuid":"dda70895-1e39-42a3-8b4f-b6493cf15e8f"},"tool_id":null,"type":"subworkflow","uuid":"a1fbce13-d09f-4106-be36-b2034451af81","when":"$(inputs.when)","workflow_outputs":[{"label":"ITS UNITE DB taxonomic abundance summary table","output_name":"Taxonomic abundance summary table","uuid":"a17b7e83-6e97-4923-a8ea-542838729f5d"},{"label":"ITS UNITE DB phylum level taxonomic abundance summary table","output_name":"phylum level taxonomic abundance summary table","uuid":"1813a260-c36c-4b3d-862a-fb244ce8f4a9"}]}},"tags":["MGnify_Amplicon","metagenomics","amplicon","name:microgalaxy"],"uuid":"5d06fcca-ca23-4a0b-9361-89812a67f9e9","version":155},"readme":"# MGnify\'s amplicon pipeline (v5.0)\\n\\n## Input datasets\\n\\n- **SRA accession list** a File listing the SRA accession IDs\\n- **Clan information file** This file lists which models belong to the same clan `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.claninfo`\\n- **Covariance models** `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.cm`\\n\\n## Processing\\n\\nThis pipeline executes the following subworkflows:\\n\\n- **Quality control - single-end**\\n- **Quality control - paired-end**\\n- **rRNA-prediction** Classification and visualization of rRNA sequences based on SSUs and LSUs\\n- **ITS** Classification and visualization of rRNA sequences based on ITS\\n- **Summary tables** Generates taxonomic abundance summary tables\\n","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify\'s amplicon pipeline v5.0\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ SRA accession list\\"]@{ shape: doc }\\n1[\\"ℹ️ Clan information file\\"]@{ shape: doc }\\n2[\\"ℹ️ Trimmomatic sliding window number of bases to average across\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Trimmomatic sliding window average quality\\"]@{ shape: lean-l }\\n4[\\"ℹ️ PE fastp - Enable base correction\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimmomatic leading\\"]@{ shape: lean-l }\\n6[\\"ℹ️ PE fastp - Qualified quality phred\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Trimmomatic trailing\\"]@{ shape: lean-l }\\n8[\\"ℹ️ PE fastp - Unqualified percent limit \\"]@{ shape: lean-l }\\n9[\\"ℹ️ Trimmomatic min length\\"]@{ shape: lean-l }\\n10[\\"ℹ️ PE fastp - Length required\\"]@{ shape: lean-l }\\n11[\\"ℹ️ Length filtering minimum size\\"]@{ shape: lean-l }\\n12[\\"ℹ️ Ambiguity filtering maximal N percentage threshold to conserve sequences\\"]@{ shape: lean-l }\\n13[\\"ℹ️ SSU taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n14[\\"ℹ️ SSU phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n15[\\"ℹ️ LSU taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n16[\\"ℹ️ LSU phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n17[\\"ℹ️ ITSoneDB taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n18[\\"ℹ️ ITSoneDB phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n19[\\"ℹ️ ITS UNITE DB taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n20[\\"ℹ️ ITS UNITE DB phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n21[\\"fastq-dl\\"]@{ shape: process }\\n0 --> 21\\n22[\\"Convert compressed file to uncompressed.\\"]@{ shape: process }\\n21 --> 22\\n23[\\"Convert compressed file to uncompressed.\\"]@{ shape: process }\\n21 --> 23\\n24[\\"Text reformatting\\"]@{ shape: process }\\n22 --> 24\\n25[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 25\\n26[\\"Convert uncompressed file to compressed\\"]@{ shape: process }\\n24 --> 26\\n27[\\"Convert uncompressed file to compressed\\"]@{ shape: process }\\n25 --> 27\\n28[\\"🛠️ MGnify\'s amplicon pipeline v5.0 - Quality control SE\\"]@{ shape: subprocess }\\n12 --> 28\\n11 --> 28\\n26 --> 28\\n5 --> 28\\n9 --> 28\\n3 --> 28\\n2 --> 28\\n7 --> 28\\n29[\\"🛠️ MGnify\'s amplicon pipeline v5.0 - Quality control PE\\"]@{ shape: subprocess }\\n12 --> 29\\n11 --> 29\\n27 --> 29\\n5 --> 29\\n9 --> 29\\n3 --> 29\\n2 --> 29\\n7 --> 29\\n4 --> 29\\n10 --> 29\\n6 --> 29\\n8 --> 29\\n30[\\"Merge collections\\"]@{ shape: process }\\n28 --> 30\\n29 --> 30\\n31[\\"🛠️ MGnify\'s amplicon pipeline v5.0 - rRNA prediction\\"]@{ shape: subprocess }\\n1 --> 31\\n30 --> 31\\n32[\\"🛠️ MGnify\'s amplicon pipeline v5.0 - ITS\\"]@{ shape: subprocess }\\n31 --> 32\\n30 --> 32\\n33[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n31 --> 33\\n34[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n31 --> 34\\n35[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n32 --> 35\\n36[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n32 --> 36\\n37[\\"🛠️ MGnify amplicon summary tables\\"]@{ shape: subprocess }\\n31 --> 37\\n14 --> 37\\n13 --> 37\\n33 --> 37\\n38[\\"🛠️ MGnify amplicon summary tables\\"]@{ shape: subprocess }\\n31 --> 38\\n16 --> 38\\n15 --> 38\\n34 --> 38\\n39[\\"🛠️ ITSoneDB summary tables\\"]@{ shape: subprocess }\\n32 --> 39\\n18 --> 39\\n17 --> 39\\n35 --> 39\\n40[\\"🛠️ UNITE summary tables\\"]@{ shape: subprocess }\\n32 --> 40\\n20 --> 40\\n19 --> 40\\n36 --> 40\\n```\\n\\n## MGnify amplicon summary tables\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ OTU tables\\"]@{ shape: docs }\\n1[\\"ℹ️ Taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n3[\\"Query Tabular\\"]@{ shape: process }\\n0 --> 3\\n4[\\"Text reformatting\\"]@{ shape: process }\\n0 --> 4\\n5[\\"Text reformatting\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Group\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Column join\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Filter Tabular\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Text reformatting\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Column join\\"]@{ shape: process }\\n10 --> 11\\n12[\\"Text reformatting\\"]@{ shape: process }\\n11 --> 12\\n```\\n\\n## MGnify\'s amplicon pipeline v5.0 - ITS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ LSU and SSU BED\\"]@{ shape: docs }\\n1[\\"ℹ️ Processed sequences\\"]@{ shape: docs }\\n2[\\"Filter empty datasets\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Extract element identifiers\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Filter collection\\"]@{ shape: process }\\n3 --> 4\\n1 --> 4\\n5[\\"bedtools MaskFastaBed\\"]@{ shape: process }\\n4 --> 5\\n2 --> 5\\n6[\\"MAPseq\\"]@{ shape: process }\\n5 --> 6\\n7[\\"MAPseq\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Filter empty datasets\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 12\\n13[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 13\\n14[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 14\\n15[\\"Filter empty datasets\\"]@{ shape: process }\\n7 --> 15\\n16[\\"Filter empty datasets\\"]@{ shape: process }\\n8 --> 16\\n17[\\"Filter empty datasets\\"]@{ shape: process }\\n9 --> 17\\n18[\\"Filter empty datasets\\"]@{ shape: process }\\n10 --> 18\\n19[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n11 --> 19\\n20[\\"Filter empty datasets\\"]@{ shape: process }\\n12 --> 20\\n21[\\"Filter empty datasets\\"]@{ shape: process }\\n13 --> 21\\n22[\\"Filter empty datasets\\"]@{ shape: process }\\n14 --> 22\\n23[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n15 --> 23\\n24[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n18 --> 24\\n25[\\"Krona ITSoneDB\\"]@{ shape: process }\\n11 --> 25\\n19 --> 25\\n26[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n22 --> 26\\n27[\\"Krona ITS UNITE\\"]@{ shape: process }\\n15 --> 27\\n23 --> 27\\n28[\\"HDF5 ITSoneDB\\"]@{ shape: process }\\n18 --> 28\\n24 --> 28\\n29[\\"JSON ITSoneDB\\"]@{ shape: process }\\n18 --> 29\\n24 --> 29\\n30[\\"HDF5 ITS UNITE\\"]@{ shape: process }\\n22 --> 30\\n26 --> 30\\n31[\\"JSON ITS UNITE\\"]@{ shape: process }\\n22 --> 31\\n26 --> 31\\n```\\n\\n## MGnify\'s amplicon pipeline v5.0 - rRNA prediction\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Processed sequences\\"]@{ shape: docs }\\n1[\\"ℹ️ Clan information file\\"]@{ shape: doc }\\n2[\\"cmsearch\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Filter empty datasets\\"]@{ shape: process }\\n2 --> 3\\n4[\\"CMsearch deoverlap\\"]@{ shape: process }\\n1 --> 4\\n3 --> 4\\n5[\\"Query Tabular\\"]@{ shape: process }\\n4 --> 5\\n6[\\"SSU BED forward\\"]@{ shape: process }\\n5 --> 6\\n7[\\"SSU BED reverse\\"]@{ shape: process }\\n5 --> 7\\n8[\\"LSU BED forward\\"]@{ shape: process }\\n5 --> 8\\n9[\\"LSU BED reverse\\"]@{ shape: process }\\n5 --> 9\\n10[\\"SSU BED\\"]@{ shape: process }\\n6 --> 10\\n7 --> 10\\n11[\\"LSU BED\\"]@{ shape: process }\\n8 --> 11\\n9 --> 11\\n12[\\"Filter empty datasets\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Filter empty datasets\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Concatenate\\"]@{ shape: process }\\n10 --> 14\\n11 --> 14\\n15[\\"Extract element identifiers\\"]@{ shape: process }\\n12 --> 15\\n16[\\"Extract element identifiers\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Filter collection\\"]@{ shape: process }\\n15 --> 17\\n0 --> 17\\n18[\\"Filter collection\\"]@{ shape: process }\\n16 --> 18\\n0 --> 18\\n19[\\"bedtools getfasta\\"]@{ shape: process }\\n17 --> 19\\n12 --> 19\\n20[\\"bedtools getfasta\\"]@{ shape: process }\\n18 --> 20\\n13 --> 20\\n21[\\"FASTA Width\\"]@{ shape: process }\\n19 --> 21\\n22[\\"FASTA Width\\"]@{ shape: process }\\n20 --> 22\\n23[\\"MAPseq\\"]@{ shape: process }\\n21 --> 23\\n24[\\"MAPseq\\"]@{ shape: process }\\n22 --> 24\\n25[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 25\\n26[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 26\\n27[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 27\\n28[\\"Filter empty datasets\\"]@{ shape: process }\\n23 --> 28\\n29[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 29\\n30[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 30\\n31[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 31\\n32[\\"Filter empty datasets\\"]@{ shape: process }\\n24 --> 32\\n33[\\"Filter empty datasets\\"]@{ shape: process }\\n25 --> 33\\n34[\\"Filter empty datasets\\"]@{ shape: process }\\n26 --> 34\\n35[\\"Filter empty datasets\\"]@{ shape: process }\\n27 --> 35\\n36[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n28 --> 36\\n37[\\"Filter empty datasets\\"]@{ shape: process }\\n29 --> 37\\n38[\\"Filter empty datasets\\"]@{ shape: process }\\n30 --> 38\\n39[\\"Filter empty datasets\\"]@{ shape: process }\\n31 --> 39\\n40[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n32 --> 40\\n41[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n35 --> 41\\n42[\\"Krona SSU SILVA\\"]@{ shape: process }\\n28 --> 42\\n36 --> 42\\n43[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n39 --> 43\\n44[\\"Krona LSU SILVA\\"]@{ shape: process }\\n32 --> 44\\n40 --> 44\\n45[\\"HDF5 SSU SILVA\\"]@{ shape: process }\\n35 --> 45\\n41 --> 45\\n46[\\"JSON SSU SILVA\\"]@{ shape: process }\\n35 --> 46\\n41 --> 46\\n47[\\"HDF5 LSU SILVA\\"]@{ shape: process }\\n39 --> 47\\n43 --> 47\\n48[\\"JSON LSU SILVA\\"]@{ shape: process }\\n39 --> 48\\n43 --> 48\\n```\\n\\n## Map empty/not empty collection to boolean \\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input collection\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Line/Word/Character count\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Compute\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 4\\n```\\n\\n## MGnify\'s amplicon pipeline v5.0 - Quality control PE\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired-end reads\\"]@{ shape: docs }\\n1[\\"ℹ️ fastp - Enable base correction\\"]@{ shape: lean-l }\\n2[\\"ℹ️ fastp - Qualified quality phred\\"]@{ shape: lean-l }\\n3[\\"ℹ️ fastp - Unqualified percent limit \\"]@{ shape: lean-l }\\n4[\\"ℹ️ fastp - Length required\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimmomatic - SLIDINGWINDOW - Number of bases to average across\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Trimmomatic - SLIDINGWINDOW - Average quality required\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Trimmomatic - LEADING\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Trimmomatic - TRAILING\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Length filtering - Minimum size\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Trimmomatic - MINLEN\\"]@{ shape: lean-l }\\n11[\\"ℹ️ Ambiguity filtering - Maximal N percentage threshold to conserve sequences\\"]@{ shape: lean-l }\\n12[\\"fastp\\"]@{ shape: process }\\n4 --> 12\\n2 --> 12\\n3 --> 12\\n1 --> 12\\n0 --> 12\\n13[\\"Unzip collection\\"]@{ shape: process }\\n12 --> 13\\n14[\\"Merging paired-end Illumina reads (SeqPrep, modified for use with MGnify piplines)\\"]@{ shape: process }\\n13 --> 14\\n13 --> 14\\n15[\\"Trimming\\"]@{ shape: process }\\n6 --> 15\\n5 --> 15\\n7 --> 15\\n8 --> 15\\n10 --> 15\\n14 --> 15\\n16[\\"FastQC\\"]@{ shape: process }\\n14 --> 16\\n17[\\"Length filtering\\"]@{ shape: process }\\n15 --> 17\\n9 --> 17\\n18[\\"FastQC\\"]@{ shape: process }\\n15 --> 18\\n19[\\"Replace\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Ambiguity filtering\\"]@{ shape: process }\\n11 --> 20\\n17 --> 20\\n21[\\"FastQC\\"]@{ shape: process }\\n17 --> 21\\n22[\\"Replace\\"]@{ shape: process }\\n18 --> 22\\n23[\\"FASTQ to FASTA\\"]@{ shape: process }\\n20 --> 23\\n24[\\"FastQC\\"]@{ shape: process }\\n20 --> 24\\n25[\\"Replace\\"]@{ shape: process }\\n21 --> 25\\n26[\\"FASTA Width\\"]@{ shape: process }\\n23 --> 26\\n27[\\"Replace\\"]@{ shape: process }\\n24 --> 27\\n28[\\"MultiQC\\"]@{ shape: process }\\n19 --> 28\\n22 --> 28\\n25 --> 28\\n27 --> 28\\n```\\n\\n## MGnify\'s amplicon pipeline v5.0 - Quality control SE\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Single-end reads\\"]@{ shape: docs }\\n1[\\"ℹ️ Trimmomatic - SLIDING WINDOW - Average quality required\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Trimmomatic - LEADING\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Trimmomatic - SLIDING WINDOW - Number of bases to average across\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Trimmomatic - TRAILING\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimmomatic - MINLEN\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Length filtering - Minimum size\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Ambiguity filtering - Maximal N percentage threshold to conserve sequences\\"]@{ shape: lean-l }\\n8[\\"FastQC\\"]@{ shape: process }\\n0 --> 8\\n9[\\"Trimming\\"]@{ shape: process }\\n1 --> 9\\n3 --> 9\\n2 --> 9\\n4 --> 9\\n5 --> 9\\n0 --> 9\\n10[\\"Replace\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Length filtering\\"]@{ shape: process }\\n9 --> 11\\n6 --> 11\\n12[\\"FastQC\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Ambiguity filtering\\"]@{ shape: process }\\n7 --> 13\\n11 --> 13\\n14[\\"FastQC\\"]@{ shape: process }\\n11 --> 14\\n15[\\"Replace\\"]@{ shape: process }\\n12 --> 15\\n16[\\"FASTQ to FASTA\\"]@{ shape: process }\\n13 --> 16\\n17[\\"FastQC\\"]@{ shape: process }\\n13 --> 17\\n18[\\"Replace\\"]@{ shape: process }\\n14 --> 18\\n19[\\"FASTA Width\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Replace\\"]@{ shape: process }\\n17 --> 20\\n21[\\"MultiQC\\"]@{ shape: process }\\n10 --> 21\\n15 --> 21\\n18 --> 21\\n20 --> 21\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-complete/main","dockstore_id":30443,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078460","tests":[{"doc":"Test outline for mgnify-amplicon-pipeline-v5-complete.ga","job":{"SRA accession list":{"class":"File","location":"https://zenodo.org/records/13710235/files/accessions.csv"},"Clan information file":{"class":"File","location":"ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.claninfo"}},"outputs":{"Single-end MultiQC report":{"asserts":[{"that":"has_text","text":"DRR010481_ambiguous_base_filtering"},{"that":"has_text","text":"84.0"},{"that":"has_text","text":"DRR010481_initial_reads"},{"that":"has_text","text":"84.8"},{"that":"has_text","text":"DRR010481_length_filtering"},{"that":"has_text","text":"DRR010481_trimming"}]},"Single-end post quality control FASTA files":{"element_tests":{"DRR010481":{"location":"https://zenodo.org/records/13710235/files/DRR010481_SE.fasta"}}},"Single-end MultiQC statistics":{"asserts":[{"that":"has_text","text":"DRR010481_ambiguous_base_filtering\\t84.0\\t47.0\\t242.02\\t244\\t40.0\\t2.4999999999999998e-05"},{"that":"has_text","text":"DRR010481_initial_reads\\t84.84848484848484\\t47.0\\t545.0454545454545\\t550\\t40.0\\t3.2999999999999996e-05"},{"that":"has_text","text":"DRR010481_length_filtering\\t84.0\\t47.0\\t242.02\\t244\\t40.0\\t2.4999999999999998e-05"},{"that":"has_text","text":"DRR010481_trimming\\t84.0\\t47.0\\t242.02\\t244\\t40.0\\t2.4999999999999998e-05"}]},"Paired-end MultiQC report":{"asserts":[{"that":"has_text","text":"ERR2715528_ambiguous_base_filtering"},{"that":"has_text","text":"ERR2715528_initial_reads"},{"that":"has_text","text":"ERR2715528_length_filtering"},{"that":"has_text","text":"ERR2715528_trimming"},{"that":"has_text","text":"92.2"}]},"Paired-end MultiQC statistics":{"location":"https://zenodo.org/records/14746033/files/general_stats_PE.tabular"},"Paired-end post quality control FASTA files":{"element_tests":{"ERR2715528":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_PE.fasta"}}},"LSU FASTA files":{"element_tests":{"DRR010481":{"location":"https://zenodo.org/records/13710235/files/DRR010481_LSU.fasta","compare":"contains"}}},"SSU taxonomic classifications using SILVA DB":{"element_tests":{"ERR2715528":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"SILVA"},{"that":"has_n_columns","comment":"#","n":15}]}}},"SSU OTU tables (SILVA DB)":{"element_tests":{"ERR2715528":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"LSU OTU tables (SILVA DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"SSU OTU tables in HDF5 format (SILVA DB)":{"element_tests":{"ERR2715528":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]}}},"SSU OTU tables in JSON format (SILVA DB)":{"element_tests":{"ERR2715528":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"SSU taxonomic abundance pie charts (SILVA DB)":{"asserts":[{"that":"has_text","text":"ERR2715528"}]},"LSU OTU tables in HDF5 format (SILVA DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]}}},"LSU OTU tables in JSON format (SILVA DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"LSU taxonomic abundance pie charts (SILVA DB)":{"asserts":[{"that":"has_text","text":"DRR010481"}]},"LSU taxonomic classifications using SILVA DB":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"SILVA"},{"that":"has_n_columns","comment":"#","n":15}]}}},"SSU FASTA files":{"element_tests":{"ERR2715528":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_SSU.fasta","compare":"contains"}}},"LSU and SSU BED regions":{"element_tests":{"DRR010481":{"location":"https://zenodo.org/records/13710235/files/DRR010481.bed"},"ERR2715528":{"location":"https://zenodo.org/records/13710235/files/ERR2715528.bed"}}},"ITS taxonomic classifications using ITSoneDB":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"ITSone"},{"that":"has_n_columns","comment":"#","n":15}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"ITSone"},{"that":"has_n_columns","comment":"#","n":15}]}}},"ITS FASTA files":{"element_tests":{"DRR010481":{"location":"https://zenodo.org/records/13710235/files/DRR010481_ITS.fasta","compare":"contains"},"ERR2715528":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_ITS.fasta","compare":"contains"}}},"ITS OTU tables in JSON format (UNITE DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"ITS taxonomic abundance pie charts (UNITE DB)":{"asserts":[{"that":"has_text","text":"DRR010481"},{"that":"has_text","text":"ERR2715528"}]},"ITS OTU tables in HDF5 format (UNITE DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]},"ERR2715528":{"asserts":[{"that":"has_size","value":72000,"delta":10000}]}}},"ITS taxonomic abundance pie charts (ITSoneDB)":{"asserts":[{"that":"has_text","text":"DRR010481"},{"that":"has_text","text":"ERR2715528"}]},"ITS OTU tables in JSON format (ITSoneDB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"ITS OTU tables in HDF5 format (ITSoneDB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]},"ERR2715528":{"asserts":[{"that":"has_size","value":72000,"delta":10000}]}}},"ITS taxonomic classifications using UNITE DB":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"UNITE"},{"that":"has_n_columns","comment":"#","n":15}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"UNITE"},{"that":"has_n_columns","comment":"#","n":15}]}}},"ITS OTU tables (UNITE DB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"ITS OTU tables (ITSoneDB)":{"element_tests":{"DRR010481":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]},"ERR2715528":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"ITSoneDB phylum level taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"superkingdom\\tkingdom\\tphylum\\tDRR010481\\tERR2715528"},{"that":"has_n_columns","n":5},{"that":"has_n_lines","n":10,"delta":4}]},"ITSoneDB taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"#SampleID\\tDRR010481\\tERR2715528"},{"that":"has_n_columns","n":3},{"that":"has_n_lines","n":117,"delta":17}]},"LSU phylum level taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"superkingdom\\tkingdom\\tphylum\\tDRR010481"},{"that":"has_n_columns","n":4},{"that":"has_n_lines","n":6,"delta":4}]},"LSU taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"#SampleID\\tDRR010481"},{"that":"has_n_columns","n":2},{"that":"has_n_lines","n":6,"delta":4}]},"SSU phylum level taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"superkingdom\\tkingdom\\tphylum\\tERR2715528"},{"that":"has_n_columns","n":4},{"that":"has_n_lines","n":12,"delta":4}]},"SSU taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"#SampleID\\tERR2715528"},{"that":"has_n_columns","n":2},{"that":"has_n_lines","n":13,"delta":4}]},"ITS UNITE DB phylum level taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"superkingdom\\tkingdom\\tphylum\\tDRR010481\\tERR2715528"},{"that":"has_n_columns","n":5},{"that":"has_n_lines","n":13,"delta":4}]},"ITS UNITE DB taxonomic abundance summary table":{"asserts":[{"that":"has_text","text":"#SampleID\\tDRR010481\\tERR2715528"},{"that":"has_n_columns","n":3},{"that":"has_n_lines","n":130,"delta":19}]}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-pipeline-v5-complete"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-pipeline-v5-rrna-prediction.ga","testParameterFiles":["/mgnify-amplicon-pipeline-v5-rrna-prediction-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X"},{"name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"}],"definition":{"a_galaxy_workflow":"true","annotation":"Classification and visualization of SSU, LSU sequences.","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - rRNA prediction","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Post quality control.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Post quality control.","name":"Processed sequences"}],"label":"Processed sequences","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":613.5582334322104},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4ddc6490-b2f3-4b9f-9a71-2da0c8857457","when":null,"workflow_outputs":[]},"1":{"annotation":"This file lists which models belong to the same clan.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This file lists which models belong to the same clan.","name":"Clan information file"}],"label":"Clan information file","name":"Input dataset","outputs":[],"position":{"left":345.04235240350044,"top":524.7051496457693},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e981bbe9-5515-4d95-aacf-7047f052f5ef","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/infernal/infernal_cmsearch/1.1.5+galaxy0","errors":null,"id":2,"input_connections":{"seqdb":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"cmsearch","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":353.04235240350044,"top":627.7051496457692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/infernal/infernal_cmsearch/1.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"3711f654044e","name":"infernal","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"A\\": false, \\"Z\\": \\"1000.0\\", \\"acceleration_huristics\\": {\\"acceleration_huristics_selector\\": \\"--hmmonly\\", \\"__current_case__\\": 5}, \\"acyk\\": false, \\"anytrunc\\": false, \\"bottomonly\\": false, \\"cm_opts\\": {\\"cm_opts_selector\\": \\"db\\", \\"__current_case__\\": 0, \\"database\\": \\"mgnify_5_0_rfam\\"}, \\"cyk\\": false, \\"g\\": false, \\"inclusion_thresholds_opts\\": {\\"inclusion_thresholds_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"model_thresholds\\": {\\"cut_ga\\": false, \\"cut_nc\\": false, \\"cut_tc\\": false}, \\"mxsize\\": \\"128.0\\", \\"noali\\": true, \\"nonull3\\": false, \\"notrunc\\": false, \\"reporting_thresholds_opts\\": {\\"reporting_thresholds_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"seqdb\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"smxsize\\": \\"128.0\\", \\"toponly\\": false, \\"verbose\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy0","type":"tool","uuid":"3c4422a0-7d9e-4cc0-97ab-86a6005314c3","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":448.5761351863496,"top":924.4124853144156},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"9fc8ec4b-d8b5-4a98-aa57-32d3178a2362","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/rnateam/cmsearch_deoverlap/cmsearch_deoverlap/0.08+galaxy2","errors":null,"id":4,"input_connections":{"clan_information":{"id":1,"output_name":"output"},"cmsearch_matches":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"CMsearch deoverlap","outputs":[{"name":"deoverlapped_matches","type":"tabular"}],"position":{"left":627.0499877929688,"top":620.7119790448688},"post_job_actions":{"HideDatasetActiondeoverlapped_matches":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"deoverlapped_matches"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/rnateam/cmsearch_deoverlap/cmsearch_deoverlap/0.08+galaxy2","tool_shed_repository":{"changeset_revision":"874ad23526a0","name":"cmsearch_deoverlap","owner":"rnateam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"clan_information\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"cmsearch_matches\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"dirty\\": false, \\"maxkeep\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.08+galaxy2","type":"tool","uuid":"6500ab8a-8bcc-4c40-be3d-ba61769c7ed8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":5,"input_connections":{"tables_0|table":{"id":4,"output_name":"deoverlapped_matches"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"},{"name":"output1","type":"tabular"},{"name":"output2","type":"tabular"},{"name":"output3","type":"tabular"}],"position":{"left":947.7324324034817,"top":678.7051496457692},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionoutput3":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output3"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": [{\\"__index__\\": 0, \\"sqlquery\\": \\"select c1, c9, c8\\\\nfrom t1\\\\nwhere c3 like \'SSU%\' and c8 > c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}, {\\"__index__\\": 1, \\"sqlquery\\": \\"select c1, c8, c9\\\\nfrom t1\\\\nwhere c3 like \'LSU%\' and c8 <= c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}, {\\"__index__\\": 2, \\"sqlquery\\": \\"select c1, c9, c8\\\\nfrom t1\\\\nwhere c3 like \'LSU%\' and c8 > c9;\\", \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}}]}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"select c1, c8, c9\\\\nfrom t1\\\\nwhere c3 like \'SSU%\' and c8 <= c9;\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"691d3274-8f28-417f-b4c6-4b52c6581caa","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"SSU BED forward","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1223.5622225911102,"top":634.7040213846066},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"forward\\\\\\", \\\\\\"1\\\\\\", \\\\\\"+\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"49102ad6-550d-46ae-bc61-d02e7a2cd4b1","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":7,"input_connections":{"infile":{"id":5,"output_name":"output1"}},"inputs":[],"label":"SSU BED reverse","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1227.7324324034817,"top":774.7051496457692},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"reverse\\\\\\", \\\\\\"1\\\\\\", \\\\\\"-\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b6ab449c-691d-43f2-bc4b-136a530805ab","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":8,"input_connections":{"infile":{"id":5,"output_name":"output2"}},"inputs":[],"label":"LSU BED forward","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1233.7324324034817,"top":907.706567048101},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"forward\\\\\\", \\\\\\"1\\\\\\", \\\\\\"+\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"52b15892-3864-439a-8526-f520a1f3e048","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"infile":{"id":5,"output_name":"output3"}},"inputs":[],"label":"LSU BED reverse","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1224.7324324034817,"top":1033.2232296457578},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"\'BEGIN {OFS=\\\\\\"\\\\\\\\t\\\\\\"} {print $1, $2 - 1, $3, \\\\\\"reverse\\\\\\", \\\\\\"1\\\\\\", \\\\\\"-\\\\\\"}\'\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"2dc2caf9-8607-42a9-a73d-d570de075554","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":10,"input_connections":{"input1":{"id":6,"output_name":"outfile"},"input2":{"id":7,"output_name":"outfile"}},"inputs":[],"label":"SSU BED","name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1526.7324324034817,"top":664.2232296457572},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU_BED"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"f72fb224-f211-4059-9ee5-9a10efdc0c28","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":11,"input_connections":{"input1":{"id":8,"output_name":"outfile"},"input2":{"id":9,"output_name":"outfile"}},"inputs":[],"label":"LSU BED","name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1505.4794369783956,"top":979.4333768742844},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU_BED"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"802aedfc-dd96-4e90-a31d-b4a5e0c0324c","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1414.4017321950623,"top":282.81198514838457},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"a3fafb37-f848-4c6f-9257-99d9d6cc5607","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":13,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1230.527671267726,"top":1226.1422039003264},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"fc4e1062-e37e-4fda-b064-555d23cc2a67","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","errors":null,"id":14,"input_connections":{"input1":{"id":10,"output_name":"output"},"input2":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Concatenate","outputs":[{"name":"output","type":"input"}],"position":{"left":1966.56641902912,"top":1498.3057522310658},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU and SSU BED regions"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1","tool_shed_repository":{"changeset_revision":"73ca13a7ec5f","name":"concat","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sameformat\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"975d4e12-b21c-46ce-b0eb-f9ac7b26f699","when":null,"workflow_outputs":[{"label":"LSU and SSU BED regions","output_name":"output","uuid":"5b559599-e3a4-4d82-86fa-b4561ada0b9a"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":15,"input_connections":{"input_collection":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1639.1793949459288,"top":303.2738932291369},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"80a5e054-f163-4e90-b40f-f0c281add51b","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":16,"input_connections":{"input_collection":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1473.4218235553517,"top":1215.607942233841},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"3082ec60-c7f2-44fd-8289-27c3aaba02f6","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":17,"input_connections":{"how|filter_source":{"id":15,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":1864.5137008919908,"top":391.13461794967463},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"83193908-189b-425b-bf16-a40e35f04fd1","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"__FILTER_FROM_FILE__","errors":null,"id":18,"input_connections":{"how|filter_source":{"id":16,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter collection","name":"how"}],"label":null,"name":"Filter collection","outputs":[{"name":"output_filtered","type":"input"},{"name":"output_discarded","type":"input"}],"position":{"left":1705.6931247050977,"top":1198.1687448048988},"post_job_actions":{"HideDatasetActionoutput_discarded":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_discarded"},"HideDatasetActionoutput_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_filtered"}},"tool_id":"__FILTER_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_filter\\": \\"remove_if_absent\\", \\"__current_case__\\": 0, \\"filter_source\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b07075b5-c465-4bb7-9480-38436e5d42fe","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","errors":null,"id":19,"input_connections":{"fasta_source|fasta":{"id":17,"output_name":"output_filtered"},"input":{"id":12,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1903.3953651059337,"top":674.4302045250132},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"5ba0a701-dd95-4e11-afdd-4cb20e4a8169","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","errors":null,"id":20,"input_connections":{"fasta_source|fasta":{"id":18,"output_name":"output_filtered"},"input":{"id":13,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1907.7391313705002,"top":1010.1390284057046},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"2e79af1d-62d7-49ed-bfe1-e2f3a78faced","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":21,"input_connections":{"input":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2199.186741916243,"top":780.6023526895299},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"cd8fcbf1-866a-4214-bdea-874c1f94f10d","when":null,"workflow_outputs":[{"label":"SSU FASTA files","output_name":"output","uuid":"71c69978-674b-496a-a3a1-e0ff2bf1c4db"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":22,"input_connections":{"input":{"id":20,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2049.6678134598683,"top":1251.106870676105},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"362611c9-757f-424a-ad76-58ee1e3ad479","when":null,"workflow_outputs":[{"label":"LSU FASTA files","output_name":"output","uuid":"0bf0c613-7c89-489e-adbb-e86a57ad13a1"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":23,"input_connections":{"sequences":{"id":21,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":2131.9295017384748,"top":507.60960896405254},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format SSU SILVA"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid SSU SILVA"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_ssu_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"786e29b6-b447-4705-beee-1fbbd6cc64c8","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","errors":null,"id":24,"input_connections":{"sequences":{"id":22,"output_name":"output"}},"inputs":[],"label":null,"name":"MAPseq","outputs":[{"name":"classifications","type":"tabular"},{"name":"otu_tsv","type":"tabular"},{"name":"otu_tsv_notaxid","type":"tabular"},{"name":"krona_format","type":"tabular"}],"position":{"left":2216.9166870117188,"top":998.5201836686707},"post_job_actions":{"ChangeDatatypeActionclassifications":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"classifications"},"ChangeDatatypeActionotu_tsv":{"action_arguments":{"newtype":"tsv"},"action_type":"ChangeDatatypeAction","output_name":"otu_tsv"},"HideDatasetActionclassifications":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"classifications"},"HideDatasetActionkrona_format":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"krona_format"},"HideDatasetActionotu_tsv":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv"},"HideDatasetActionotu_tsv_notaxid":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"otu_tsv_notaxid"},"RenameDatasetActionkrona_format":{"action_arguments":{"newname":"krona_format LSU SILVA"},"action_type":"RenameDatasetAction","output_name":"krona_format"},"RenameDatasetActionotu_tsv_notaxid":{"action_arguments":{"newname":"otu_tsv_notaxid LSU SILVA"},"action_type":"RenameDatasetAction","output_name":"otu_tsv_notaxid"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/mapseq/mapseq/2.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"f1d1dafc836d","name":"mapseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"minid1\\": \\"1\\", \\"minid2\\": \\"1\\", \\"minscore\\": \\"30\\", \\"otulim\\": \\"50\\", \\"outfmt\\": \\"simple\\", \\"ref_db\\": {\\"db_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"db_cached\\": \\"mgnify_lsu_from_2023-09-26\\", \\"mapseq2biom\\": {\\"mapseq2biom\\": \\"yes\\", \\"__current_case__\\": 0, \\"krona_input\\": true}}, \\"sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tophits\\": \\"80\\", \\"topotus\\": \\"40\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy0","type":"tool","uuid":"89474568-b342-4376-a52d-20878582d5db","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":25,"input_connections":{"infile":{"id":23,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2341.918447993722,"top":343.73564945106307},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"6bbc2f91-cdd5-4cda-8923-1c4aafc793d1","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":26,"input_connections":{"infile":{"id":23,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2484.875930049533,"top":671.8771045356158},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"18302666-59ae-4441-b6cd-b46d8a400789","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":27,"input_connections":{"infile":{"id":23,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2682.405523606579,"top":452.39977836112007},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ba332896-373b-475b-98e1-39a4457c8153","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":28,"input_connections":{"input":{"id":23,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3020.6201097810404,"top":729.4160842644015},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"257b8f02-f88a-4b1a-926d-c2f7bd0a2a56","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":29,"input_connections":{"infile":{"id":24,"output_name":"classifications"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2459.472750071673,"top":964.1860471272424},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"d4ab7aa0-e6eb-44a9-8e5b-2f275f7dc41c","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":30,"input_connections":{"infile":{"id":24,"output_name":"otu_tsv"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2274.421541439101,"top":1375.2808207350452},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"fd27a4ca-fd35-4c6b-aafe-b442d42531f0","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":31,"input_connections":{"infile":{"id":24,"output_name":"otu_tsv_notaxid"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2904.532618934821,"top":1143.3812048537206},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR == 1 {first = $0}\\\\nNR == 2 {second = $0}\\\\nNR > 2 {print first; print second; print; while (getline) print; exit}\\\\nEND {if (NR == 2) exit}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"fe9d4c36-faa8-4bdc-8c3c-3c230dd44ffc","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":32,"input_connections":{"input":{"id":24,"output_name":"krona_format"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3004.2019554771114,"top":1427.490860606865},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0f82c7b0-6313-4637-8572-0e25e4b61222","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":33,"input_connections":{"input":{"id":25,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2501.3333740234375,"top":170.56666564941406},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU taxonomic classifications using SILVA DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"02c81ee5-d2a0-42da-b7d1-cc432cc58431","when":null,"workflow_outputs":[{"label":"SSU taxonomic classifications using SILVA DB","output_name":"output","uuid":"11632f64-a951-49d9-a46c-85285ba9234d"}]},"34":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":34,"input_connections":{"input":{"id":26,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2755.107387901881,"top":663.6467048040722},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU OTU tables (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"db62999a-75ce-4448-a4bb-ffd0a0c8f070","when":null,"workflow_outputs":[{"label":"SSU OTU tables (SILVA DB)","output_name":"output","uuid":"9fa11853-687d-49c9-94e2-4ac1b6a9de02"}]},"35":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":35,"input_connections":{"input":{"id":27,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2929.3175155528866,"top":456.5149782268919},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ad1c259c-a1e5-496b-aaae-e9841f4086bb","when":null,"workflow_outputs":[]},"36":{"annotation":"","id":36,"input_connections":{"Input collection":{"id":28,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3253.470085366978,"top":701.8494338737765},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"61653fbd-a405-4a8b-8dc1-bf7e9ed388d9"},"tool_id":null,"type":"subworkflow","uuid":"a5aa7d4c-ece2-4cff-82ff-f6320ad251ff","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":37,"input_connections":{"input":{"id":29,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2692.5435761342164,"top":970.0420980333365},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU taxonomic classifications using SILVA DB"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"fc13fe38-e24b-485d-98d7-da38c4ecea06","when":null,"workflow_outputs":[{"label":"LSU taxonomic classifications using SILVA DB","output_name":"output","uuid":"90446073-dd91-4132-92fc-b3d38644a47c"}]},"38":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":38,"input_connections":{"input":{"id":30,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2513.3484184077383,"top":1365.9111392852947},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU OTU tables (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0e810c9b-8f1a-417e-906e-78d600cd51c8","when":null,"workflow_outputs":[{"label":"LSU OTU tables (SILVA DB)","output_name":"output","uuid":"ecb9dba5-a182-463e-a06a-18edbd37a46e"}]},"39":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":39,"input_connections":{"input":{"id":31,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3149.3155468095524,"top":1143.3812048537206},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"552cb861-a006-4dc7-8ff7-f5af10d5cb93","when":null,"workflow_outputs":[]},"40":{"annotation":"","id":40,"input_connections":{"Input collection":{"id":32,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3314.9631045505002,"top":1517.3226958033472},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"247de06c-d328-4e91-a4fa-f6cdfabecc48"},"tool_id":null,"type":"subworkflow","uuid":"fa552732-31b6-4928-991e-5b966dc4da69","when":null,"workflow_outputs":[]},"41":{"annotation":"","id":41,"input_connections":{"Input collection":{"id":35,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3179.470085366978,"top":408.84943387377643},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"59e3ed59-1683-4dab-8664-73d5b381b3c3"},"tool_id":null,"type":"subworkflow","uuid":"aee25341-bf68-4cd8-b3fe-769a94512bf4","when":null,"workflow_outputs":[]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":42,"input_connections":{"type_of_data|input":{"id":28,"output_name":"output"},"when":{"id":36,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona SSU SILVA","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":3521.7643846129777,"top":626.8193285739657},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"SSU taxonomic abundance pie charts (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"1da968ab-cec0-4db1-b53a-ebe6f035c2fe","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU taxonomic abundance pie charts (SILVA DB)","output_name":"output","uuid":"2571ed1e-04ee-4078-8f43-a406b6e67dbe"}]},"43":{"annotation":"","id":43,"input_connections":{"Input collection":{"id":39,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Map empty/not empty collection to boolean ","outputs":[],"position":{"left":3408.3308199842286,"top":1137.1888001240086},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"Map empty/not empty collection to boolean ","steps":{"0":{"annotation":"Collection to map to boolean parameter.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Collection to map to boolean parameter.","name":"Input collection"}],"label":"Input collection","name":"Input dataset collection","outputs":[],"position":{"left":0.15615897466262751,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e6d389a5-f18e-4d28-8fe1-429cd7ef36e9","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":0,"top":85.40073046637212},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"d9ec30dc-178b-484b-938b-c1fed725e697","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"wc_gnu","errors":null,"id":2,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1.385951110441301,"top":204.8836843547113},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5d2d8e7a-faab-4a80-b572-94e0437b0bfc","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":3,"input_connections":{"input":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4.196812654368596,"top":322.88311011250636},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c1 != 0\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5dbc764c-01f3-411f-8987-9c3fd84f22e7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":3,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":4.977714755176354,"top":419.5469987423484},"post_job_actions":{"HideDatasetActionboolean_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"boolean_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"boolean\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"64f1f502-d047-4d1f-8999-b7ec04489def","when":null,"workflow_outputs":[{"label":"boolean_param","output_name":"boolean_param","uuid":"890f9593-92c6-40c3-a348-f124fd2e84b1"}]}},"tags":[],"uuid":"78de0767-d351-4672-ac6b-d8b3b9bd3ac7"},"tool_id":null,"type":"subworkflow","uuid":"c26c4423-280d-4d7e-bed2-7390b14873ce","when":null,"workflow_outputs":[]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":44,"input_connections":{"type_of_data|input":{"id":32,"output_name":"output"},"when":{"id":40,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":"Krona LSU SILVA","name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":3670.2216351015004,"top":1492.9828576583009},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"LSU taxonomic abundance pie charts (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"a6de961c-2f40-478a-9ae6-6c33fd9e28bc","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU taxonomic abundance pie charts (SILVA DB)","output_name":"output","uuid":"f4c8977e-7b91-4f4f-b112-4eaa744d7834"}]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":45,"input_connections":{"input|input_fp":{"id":35,"output_name":"output"},"when":{"id":41,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 SSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3498.725031827062,"top":0},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"SSU OTU tables in HDF5 format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"26c1df0e-7c23-4a6f-8859-420a2787dee0","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU OTU tables in HDF5 format (SILVA DB)","output_name":"output_fp","uuid":"e9ed7fe6-a7f3-42a4-bb51-0a4631d38a73"}]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":46,"input_connections":{"input|input_fp":{"id":35,"output_name":"output"},"when":{"id":41,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON SSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3500.6630859470188,"top":321.52388658376185},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"SSU OTU tables in JSON format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"879dd786-fbb9-4054-9bc4-42eb4a9b1193","when":"$(inputs.when)","workflow_outputs":[{"label":"SSU OTU tables in JSON format (SILVA DB)","output_name":"output_fp","uuid":"3937d133-7118-40b9-b88a-ed382b51603d"}]},"47":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":47,"input_connections":{"input|input_fp":{"id":39,"output_name":"output"},"when":{"id":43,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"HDF5 LSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3674.4440326059353,"top":837.6260676655736},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"LSU OTU tables in HDF5 format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"hdf5\\", \\"__current_case__\\": 1, \\"collapsed_observations\\": false, \\"collapsed_samples\\": false}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"0429ca95-6661-4467-9585-56e4310d7a05","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU OTU tables in HDF5 format (SILVA DB)","output_name":"output_fp","uuid":"ed10309d-7874-4c3b-952e-4a9e3f85d918"}]},"48":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","errors":null,"id":48,"input_connections":{"input|input_fp":{"id":39,"output_name":"output"},"when":{"id":43,"output_name":"boolean_param"}},"inputs":[{"description":"runtime parameter for tool Convert","name":"input"},{"description":"runtime parameter for tool Convert","name":"output"},{"description":"runtime parameter for tool Convert","name":"output"}],"label":"JSON LSU SILVA","name":"Convert","outputs":[{"name":"output_fp","type":"biom1"}],"position":{"left":3670.48882921932,"top":1146.8376201123672},"post_job_actions":{"RenameDatasetActionoutput_fp":{"action_arguments":{"newname":"LSU OTU tables in JSON format (SILVA DB)"},"action_type":"RenameDatasetAction","output_name":"output_fp"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/biom_convert/biom_convert/2.1.15+galaxy1","tool_shed_repository":{"changeset_revision":"d4d4e9febf5c","name":"biom_convert","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"type_selector\\": \\"tsv\\", \\"__current_case__\\": 0, \\"input_fp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"process_obs_metadata\\": \\"taxonomy\\"}, \\"output\\": {\\"type_selector\\": \\"biom\\", \\"__current_case__\\": 1, \\"table_type\\": \\"OTU table\\", \\"biom_type\\": {\\"selector\\": \\"json\\", \\"__current_case__\\": 0}, \\"sample_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"observation_metadata_fp\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.15+galaxy1","type":"tool","uuid":"850a6179-db1c-4865-a689-0e7dd4b471f0","when":"$(inputs.when)","workflow_outputs":[{"label":"LSU OTU tables in JSON format (SILVA DB)","output_name":"output_fp","uuid":"2f2a6bc2-cc77-4266-aeb7-eae31654d930"}]}},"tags":["MGnify_Amplicon","metagenomics","amplicon","name:microgalaxy"],"uuid":"53c5bc12-4d7c-4799-b053-ced394665d61","version":38},"readme":"# MGnify\'s amplicon (v5.0) rRNA-prediction subworkflow\\n\\nClassification and visualization of rRNA sequences based on SSUs and LSUs.\\n\\n## Input datasets\\n\\n- **Processed sequences** Post quality control\\n- **Clan information file** This file lists which models belong to the same clan `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.claninfo`\\n- **Covariance models** `wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.cm`\\n\\n## Processing\\n\\nThis subworkflow executes the following classification and visualization steps:\\n\\n- **cmsearch** Searches covariance models against a sequence database\\n- **CMsearch-deoverlap** Removes lower scoring overlaps from cmsearch output files\\n- **bedtools getfasta**  Extracts sequences from a FASTA file\\n- **MAPseq** Sequence read classification\\n- **biom-convert** Converts OTU tables to HDF5 and JSON formats\\n- **Krona** Generates pie charts out of the OTU tables","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify\'s amplicon pipeline v5.0 - rRNA prediction\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Processed sequences\\"]@{ shape: docs }\\n1[\\"ℹ️ Clan information file\\"]@{ shape: doc }\\n2[\\"cmsearch\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Filter empty datasets\\"]@{ shape: process }\\n2 --> 3\\n4[\\"CMsearch deoverlap\\"]@{ shape: process }\\n1 --> 4\\n3 --> 4\\n5[\\"Query Tabular\\"]@{ shape: process }\\n4 --> 5\\n6[\\"SSU BED forward\\"]@{ shape: process }\\n5 --> 6\\n7[\\"SSU BED reverse\\"]@{ shape: process }\\n5 --> 7\\n8[\\"LSU BED forward\\"]@{ shape: process }\\n5 --> 8\\n9[\\"LSU BED reverse\\"]@{ shape: process }\\n5 --> 9\\n10[\\"SSU BED\\"]@{ shape: process }\\n6 --> 10\\n7 --> 10\\n11[\\"LSU BED\\"]@{ shape: process }\\n8 --> 11\\n9 --> 11\\n12[\\"Filter empty datasets\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Filter empty datasets\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Concatenate\\"]@{ shape: process }\\n10 --> 14\\n11 --> 14\\n15[\\"Extract element identifiers\\"]@{ shape: process }\\n12 --> 15\\n16[\\"Extract element identifiers\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Filter collection\\"]@{ shape: process }\\n15 --> 17\\n0 --> 17\\n18[\\"Filter collection\\"]@{ shape: process }\\n16 --> 18\\n0 --> 18\\n19[\\"bedtools getfasta\\"]@{ shape: process }\\n17 --> 19\\n12 --> 19\\n20[\\"bedtools getfasta\\"]@{ shape: process }\\n18 --> 20\\n13 --> 20\\n21[\\"FASTA Width\\"]@{ shape: process }\\n19 --> 21\\n22[\\"FASTA Width\\"]@{ shape: process }\\n20 --> 22\\n23[\\"MAPseq\\"]@{ shape: process }\\n21 --> 23\\n24[\\"MAPseq\\"]@{ shape: process }\\n22 --> 24\\n25[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 25\\n26[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 26\\n27[\\"Text reformatting\\"]@{ shape: process }\\n23 --> 27\\n28[\\"Filter empty datasets\\"]@{ shape: process }\\n23 --> 28\\n29[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 29\\n30[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 30\\n31[\\"Text reformatting\\"]@{ shape: process }\\n24 --> 31\\n32[\\"Filter empty datasets\\"]@{ shape: process }\\n24 --> 32\\n33[\\"Filter empty datasets\\"]@{ shape: process }\\n25 --> 33\\n34[\\"Filter empty datasets\\"]@{ shape: process }\\n26 --> 34\\n35[\\"Filter empty datasets\\"]@{ shape: process }\\n27 --> 35\\n36[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n28 --> 36\\n37[\\"Filter empty datasets\\"]@{ shape: process }\\n29 --> 37\\n38[\\"Filter empty datasets\\"]@{ shape: process }\\n30 --> 38\\n39[\\"Filter empty datasets\\"]@{ shape: process }\\n31 --> 39\\n40[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n32 --> 40\\n41[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n35 --> 41\\n42[\\"Krona SSU SILVA\\"]@{ shape: process }\\n28 --> 42\\n36 --> 42\\n43[\\"🛠️ Map empty/not empty collection to boolean \\"]@{ shape: subprocess }\\n39 --> 43\\n44[\\"Krona LSU SILVA\\"]@{ shape: process }\\n32 --> 44\\n40 --> 44\\n45[\\"HDF5 SSU SILVA\\"]@{ shape: process }\\n35 --> 45\\n41 --> 45\\n46[\\"JSON SSU SILVA\\"]@{ shape: process }\\n35 --> 46\\n41 --> 46\\n47[\\"HDF5 LSU SILVA\\"]@{ shape: process }\\n39 --> 47\\n43 --> 47\\n48[\\"JSON LSU SILVA\\"]@{ shape: process }\\n39 --> 48\\n43 --> 48\\n```\\n\\n## Map empty/not empty collection to boolean \\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input collection\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Line/Word/Character count\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Compute\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 4\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-rrna-prediction/main","dockstore_id":30451,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078499","tests":[{"doc":"Test outline for mgnify-amplicon-pipeline-v5-rrna-prediction.ga","job":{"Processed sequences":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"DRR010481_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/DRR010481_SE.fasta"},{"class":"File","identifier":"ERR2715528_MERGED_FASTQ.fasta","location":"https://zenodo.org/records/13710235/files/ERR2715528_PE.fasta"}]},"Clan information file":{"class":"File","location":"ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/pipeline-5.0/ref-dbs/rfam_models/ribosomal_models/ribo.claninfo"}},"outputs":{"LSU and SSU BED regions":{"element_tests":{"DRR010481_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/DRR010481.bed"},"ERR2715528_MERGED_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/ERR2715528.bed"}}},"SSU FASTA files":{"element_tests":{"ERR2715528_MERGED_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_SSU.fasta","compare":"contains"}}},"LSU FASTA files":{"element_tests":{"DRR010481_FASTQ.fasta":{"location":"https://zenodo.org/records/13710235/files/DRR010481_LSU.fasta","compare":"contains"}}},"SSU taxonomic classifications using SILVA DB":{"element_tests":{"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"SILVA"},{"that":"has_n_columns","comment":"#","n":15}]}}},"SSU OTU tables (SILVA DB)":{"element_tests":{"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"LSU OTU tables (SILVA DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# Constructed from biom file"},{"that":"has_text","text":"# OTU ID"},{"that":"has_text","text":"Unspecified"},{"that":"has_text","text":"taxonomy"},{"that":"has_text","text":"taxid"},{"that":"has_n_columns","comment":"#","n":4}]}}},"LSU taxonomic classifications using SILVA DB":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"# mapseq v1.2.6 (Jan 20 2023)"},{"that":"has_text","text":"SILVA"},{"that":"has_n_columns","comment":"#","n":15}]}}},"SSU OTU tables in HDF5 format (SILVA DB)":{"element_tests":{"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]}}},"SSU OTU tables in JSON format (SILVA DB)":{"element_tests":{"ERR2715528_MERGED_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"SSU taxonomic abundance pie charts (SILVA DB)":{"asserts":[{"that":"has_text","text":"ERR2715528_MERGED_FASTQ_fasta"}]},"LSU OTU tables in HDF5 format (SILVA DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_size","value":37000,"delta":10000}]}}},"LSU OTU tables in JSON format (SILVA DB)":{"element_tests":{"DRR010481_FASTQ.fasta":{"asserts":[{"that":"has_text","text":"\\"type\\": \\"OTU table\\""}]}}},"LSU taxonomic abundance pie charts (SILVA DB)":{"asserts":[{"that":"has_text","text":"DRR010481_FASTQ_fasta"}]}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-pipeline-v5-rrna-prediction"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mapseq-to-ampvis2.ga","testParameterFiles":["/mapseq-to-ampvis2-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Mara Besemer","url":"https://github.com/MaraBesemer"}],"definition":{"a_galaxy_workflow":"true","annotation":"The MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.","comments":[{"color":"green","data":{"text":"tax_table"},"id":1,"position":[282,429.53420657186683],"size":[978,217],"type":"markdown"},{"color":"orange","data":{"text":"OTU_table"},"id":0,"position":[284,115.53420657186683],"size":[1031,220],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","name":"Mara Besemer","url":"https://github.com/MaraBesemer"}],"format-version":"0.1","license":"MIT","name":"MAPseq to ampvis2","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A dataset collection containing the tab-separated OTU outputs of MAPseq.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A dataset collection containing the tab-separated OTU outputs of MAPseq.","name":"MAPseq OTU tables"}],"label":"MAPseq OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":282.53420657186683},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"d273ae7f-de2e-4f1f-a6ea-582804a31dc5","when":null,"workflow_outputs":[]},"1":{"annotation":"Contextual information associated with the data from the OTU tables.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Contextual information associated with the data from the OTU tables.","name":"OTU table metadata"}],"label":"OTU table metadata","name":"Input dataset","outputs":[],"position":{"left":1088.3581803500276,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"1e3d42f9-942e-461b-ae51-e9049bd51d7e","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":2,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":358.7833251953125,"top":148.36751956014808},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c2,\\\\n    c3,\\\\n    c3 * 100 / SUM(c3) OVER() AS relative_abundance\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"ee2e3ed8-e95e-4c6f-8898-f417916dbeb4","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":3,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":302,"top":464.0333251953125},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 {\\\\n    # Extract OTU ID and add \\\\\\"OTU_\\\\\\" prefix\\\\n    otu_id = \\\\\\"OTU_\\\\\\" $1\\\\n\\\\n    # Initialize variables for taxonomic levels\\\\n    superkingdom = kingdom = phylum = class = order = family = genus = species = \\\\\\"\\\\\\"\\\\n\\\\n    # Split the taxonomy column by \\\\\\";\\\\\\"\\\\n    split($3, taxonomy, \\\\\\";\\\\\\")\\\\n\\\\n    # Extract taxonomic levels\\\\n    for (i in taxonomy) {\\\\n        if (taxonomy[i] ~ /^sk__/) {\\\\n            superkingdom = taxonomy[i]\\\\n        }else if (taxonomy[i] ~ /^k__/) {\\\\n            kingdom = taxonomy[i]\\\\n        }else if (taxonomy[i] ~ /^p__/) {\\\\n            phylum = taxonomy[i]\\\\n        } else if (taxonomy[i] ~ /^c__/) {\\\\n            class = taxonomy[i]\\\\n        } else if (taxonomy[i] ~ /^o__/) {\\\\n            order = taxonomy[i]\\\\n        } else if (taxonomy[i] ~ /^f__/) {\\\\n            family = taxonomy[i]\\\\n        } else if (taxonomy[i] ~ /^g__/) {\\\\n            genus = taxonomy[i]\\\\n        } else if (taxonomy[i] ~ /^s__/) {\\\\n            species = taxonomy[i]\\\\n        }\\\\n    }\\\\n\\\\n    # Print the row with extracted values\\\\n    print otu_id \\\\\\"\\\\\\\\t\\\\\\" superkingdom \\\\\\"\\\\\\\\t\\\\\\" kingdom \\\\\\"\\\\\\\\t\\\\\\" phylum \\\\\\"\\\\\\\\t\\\\\\" class \\\\\\"\\\\\\\\t\\\\\\" order \\\\\\"\\\\\\\\t\\\\\\" family \\\\\\"\\\\\\\\t\\\\\\" genus \\\\\\"\\\\\\\\t\\\\\\" species\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"542e2486-608c-48c3-949f-bf693f35e503","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":4,"input_connections":{"infile":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":587.316650390625,"top":148.00088137655433},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n    FS = \\\\\\"\\\\\\\\t\\\\\\";\\\\n    OFS = \\\\\\"\\\\\\\\t\\\\\\";\\\\n}\\\\n\\\\nNR == 1 {\\\\n    # Extract the prefix before the first underscore from the filename in the first row, first column\\\\n    split($1, filename_parts, \\\\\\"_\\\\\\");\\\\n    filename_prefix = filename_parts[1];\\\\n    print \\\\\\"OTU\\\\\\", filename_prefix;\\\\n}\\\\n\\\\n{\\\\n    # Print the OTU ID with \'OTU_\' prefix and the relative_abundance\\\\n    print \\\\\\"OTU_\\\\\\" $2, $4;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b3a78171-289c-40dc-95c7-f1ea65b12167","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":5,"input_connections":{"input_list":{"id":3,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":556,"top":464.03420657186683},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"cbdaeb1f-a7ad-4a7a-afd4-20a5567732c1","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":6,"input_connections":{"input_tabular":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":814.316650390625,"top":145.16753787069496},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"0\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"653811aa-3f19-4f7c-84d4-be3d091a4550","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":7,"input_connections":{"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":777,"top":468.03420657186683},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n    # Print the header\\\\n    print \\\\\\"OTU\\\\\\\\tSuperkingdom\\\\\\\\tKingdom\\\\\\\\tPhylum\\\\\\\\tClass\\\\\\\\tOrder\\\\\\\\tFamily\\\\\\\\tGenus\\\\\\\\tSpecies\\\\\\"\\\\n}\\\\n\\\\n# Print each line as it is\\\\n{\\\\n    print\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"25d25ac8-20a2-491e-8847-d49874922821","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":8,"input_connections":{"infile":{"id":6,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1041.816650390625,"top":145.01754397421058},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"OTU table"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==2 {for(i=2; i<=NF; i++) if (index($i, \\\\\\".\\\\\\") > 0) $i = substr($i, 1, index($i, \\\\\\".\\\\\\")-1)} NR>1 {print}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"8ada9fc3-1ec1-4ec8-995e-b2f6411c777f","when":null,"workflow_outputs":[{"label":"OTU table","output_name":"outfile","uuid":"80d60c58-cde2-4101-9fdd-d53a7be1ace2"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":9,"input_connections":{"tables_0|table":{"id":7,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1032.75,"top":446.25},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Taxonomy table"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT c1,\\\\n       GROUP_CONCAT(DISTINCT c2) AS c2,\\\\n       GROUP_CONCAT(DISTINCT c3) AS c3,\\\\n       GROUP_CONCAT(DISTINCT c4) AS c4,\\\\n       GROUP_CONCAT(DISTINCT c5) AS c5,\\\\n       GROUP_CONCAT(DISTINCT c6) AS c6,\\\\n       GROUP_CONCAT(DISTINCT c7) AS c7,\\\\n       GROUP_CONCAT(DISTINCT c8) AS c8,\\\\n       GROUP_CONCAT(DISTINCT c9) AS c9\\\\nFROM t1\\\\nGROUP BY c1;\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"fb4a4528-4617-4416-94fa-c3e1a2ddea0e","when":null,"workflow_outputs":[{"label":"Taxonomy table","output_name":"output","uuid":"b4a1b7c0-a4df-4a2e-a381-87f5159b3b7c"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ampvis2_load/ampvis2_load/2.8.9+galaxy0","errors":null,"id":10,"input_connections":{"metadata":{"id":1,"output_name":"output"},"otutable":{"id":8,"output_name":"outfile"},"taxonomy":{"id":9,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ampvis2 load","name":"fasta"},{"description":"runtime parameter for tool ampvis2 load","name":"tree"}],"label":null,"name":"ampvis2 load","outputs":[{"name":"ampvis","type":"ampvis2"},{"name":"metadata_list_out","type":"tabular"},{"name":"taxonomy_list_out","type":"tabular"}],"position":{"left":1514.0344635260071,"top":201.03822045770914},"post_job_actions":{"RenameDatasetActionampvis":{"action_arguments":{"newname":"Ampvis2 object"},"action_type":"RenameDatasetAction","output_name":"ampvis"},"RenameDatasetActionmetadata_list_out":{"action_arguments":{"newname":"Ampvis2 metadata table"},"action_type":"RenameDatasetAction","output_name":"metadata_list_out"},"RenameDatasetActiontaxonomy_list_out":{"action_arguments":{"newname":"Ampvis2 taxonomy table"},"action_type":"RenameDatasetAction","output_name":"taxonomy_list_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ampvis2_load/ampvis2_load/2.8.9+galaxy0","tool_shed_repository":{"changeset_revision":"738474880da8","name":"ampvis2_load","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"asv_otu_col_empty\\": false, \\"asv_sequences\\": false, \\"fasta\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"guess_column_types\\": true, \\"metadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"otutable\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pruneSingletons\\": false, \\"taxonomy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tree\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"write_lists\\": [\\"tax\\", \\"metadata\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.8.9+galaxy0","type":"tool","uuid":"283c89ea-5c69-4ad0-9905-146d721621b8","when":null,"workflow_outputs":[{"label":"Ampvis2 object","output_name":"ampvis","uuid":"c0f259f9-28d8-451d-99c4-e276fdbc8a42"},{"label":"Ampvis2 metadata table","output_name":"metadata_list_out","uuid":"67b437eb-9293-4024-a216-1fd21059810e"},{"label":"Ampvis2 taxonomy table","output_name":"taxonomy_list_out","uuid":"6a54207f-293e-4810-a1b0-29a05f284b24"}]}},"tags":[],"uuid":"e24060a0-1cfe-40f1-9251-48e96f7e1121","version":40},"readme":"# MAPseq to ampvis workflow\\n\\nThe MAPseq to Ampvis workflow processes MAPseq OTU tables and associated metadata for analysis in Ampvis2. This workflow involves reformatting MAPseq output datasets to produce structured output files suitable for Ampvis2.\\n\\n## Input datasets\\n\\n- **MAPseq OTU tables** These tables, generated by MAPseq, include sequence counts and taxonomic classifications.\\n- **Metadata** Contextual information associated with the data from the OTU tables.\\n\\n## Outputs\\n\\n- **otu_table** A newly formatted OTU table structured for Ampvis2.\\n- **tax_table** A taxonomy table containing the taxonomic classifications from the OTU tables.\\n- **ampvis2** An R object compatible with the Ampvis2 tool for further analysis.\\n- **metadata_list_out**  A structured list of metadata used in the analysis.\\n- **taxonomy_list_out** A processed taxonomy list ready for integration with Ampvis2.","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## MAPseq to ampvis2\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ MAPseq OTU tables\\"]@{ shape: docs }\\n1[\\"ℹ️ OTU table metadata\\"]@{ shape: doc }\\n2[\\"Query Tabular\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Text reformatting\\"]@{ shape: process }\\n0 --> 3\\n4[\\"Text reformatting\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Collapse Collection\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Column join\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Text reformatting\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Text reformatting\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Query Tabular\\"]@{ shape: process }\\n7 --> 9\\n10[\\"ampvis2 load\\"]@{ shape: process }\\n1 --> 10\\n8 --> 10\\n9 --> 10\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mapseq-to-ampvis2/main","dockstore_id":30442,"categories":[],"collections":[],"doi":"10.5281/zenodo.15078899","tests":[{"doc":"Test outline for mapseq-to-ampvis2.ga","job":{"MAPseq OTU tables":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"SRR11032273","location":"https://zenodo.org/records/13347829/files/SRR11032273.tsv"},{"class":"File","identifier":"SRR11032274","location":"https://zenodo.org/records/13347829/files/SRR11032274.tsv"},{"class":"File","identifier":"SRR11032275","location":"https://zenodo.org/records/13347829/files/SRR11032275.tsv"},{"class":"File","identifier":"SRR11032276","location":"https://zenodo.org/records/13347829/files/SRR11032276.tsv"},{"class":"File","identifier":"SRR11032277","location":"https://zenodo.org/records/13347829/files/SRR11032277.tsv"},{"class":"File","identifier":"SRR11038211","location":"https://zenodo.org/records/13347829/files/SRR11038211.tsv"},{"class":"File","identifier":"SRR11038212","location":"https://zenodo.org/records/13347829/files/SRR11038212.tsv"},{"class":"File","identifier":"SRR11038213","location":"https://zenodo.org/records/13347829/files/SRR11038213.tsv"},{"class":"File","identifier":"SRR11038214","location":"https://zenodo.org/records/13347829/files/SRR11038214.tsv"},{"class":"File","identifier":"SRR11038215","location":"https://zenodo.org/records/13347829/files/SRR11038215.tsv"}]},"OTU table metadata":{"class":"File","location":"https://zenodo.org/records/13347829/files/test_metadata_formatted.tabular"}},"outputs":{"OTU table":{"location":"https://zenodo.org/records/13347829/files/OTU_table.tabular"},"Taxonomy table":{"location":"https://zenodo.org/records/14745803/files/tax_table.tabular"},"Ampvis2 object":{"asserts":[{"that":"has_size","value":15000,"delta":1000}]},"Ampvis2 metadata table":{"location":"https://zenodo.org/records/13347829/files/metadata_list.tabular"},"Ampvis2 taxonomy table":{"location":"https://zenodo.org/records/14745803/files/taxonomy_list.tabular"}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mapseq-to-ampvis2"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-summary-tables.ga","testParameterFiles":["/mgnify-amplicon-summary-tables-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow creates taxonomic summary tables out of the amplicon pipeline results. ","comments":[{"color":"green","data":{"text":"taxonomic summary table"},"id":0,"position":[272.3406915224165,120.13702205366614],"size":[909,241],"type":"markdown"},{"color":"turquoise","data":{"text":"phylum summary table"},"id":1,"position":[270.9406915224165,380.7370220536662],"size":[1353,243],"type":"markdown"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"}],"format-version":"0.1","license":"MIT","name":"MGnify amplicon summary tables","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"OTU tables.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"OTU tables.","name":"OTU tables"}],"label":"OTU tables","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":337.4772484997172},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ea3accdc-9c03-462f-b58e-8fedc647c253","when":null,"workflow_outputs":[]},"1":{"annotation":"specifies the name of taxonomic summary file.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file.","name":"Taxonomic abundance summary table name"}],"label":"Taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":968.63330078125,"top":0},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"98862eac-2fa1-4e7d-95f9-ef4832b40b45","when":null,"workflow_outputs":[]},"2":{"annotation":"specifies the name of taxonomic summary file, for phylum level.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"specifies the name of taxonomic summary file, for phylum level.","name":"Phylum level taxonomic abundance summary table name"}],"label":"Phylum level taxonomic abundance summary table name","name":"Input parameter","outputs":[],"position":{"left":1415.5626469798628,"top":643.2277124832184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e948e6d0-a7f3-4d47-a50f-003ee055865a","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":3,"input_connections":{"tables_0|table":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":null,"name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":298.42962390165496,"top":166.45843163713073},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT \\\\n    c1,\\\\n    c4,\\\\n    c3\\\\nFROM \\\\n    t1;\\\\n\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"comment\\", \\"__current_case__\\": 1, \\"comment_char\\": \\"35\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"b12026e3-5bd8-479d-a942-4858beae4a6a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":4,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.7278335508603,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR > 2 { \\\\n        split($3, taxa, \\\\\\";\\\\\\");\\\\n        merged = \\\\\\"\\\\\\";\\\\n        levels = 3; # We need sk__, k__, and p__ levels\\\\n        for (i = 1; i <= levels; i++) {\\\\n            if (i <= length(taxa) && taxa[i] ~ /__/) {\\\\n                split(taxa[i], parts, \\\\\\"__\\\\\\");\\\\n                if (parts[2] != \\\\\\"\\\\\\") {\\\\n                    value = parts[2];\\\\n                } else {\\\\n                    value = \\\\\\"unassigned\\\\\\";\\\\n                }\\\\n            } else {\\\\n                value = \\\\\\"unassigned\\\\\\";\\\\n            }\\\\n            merged = merged (merged ? \\\\\\";\\\\\\" : \\\\\\"\\\\\\") value;\\\\n        }\\\\n        print merged \\\\\\"\\\\\\\\t\\\\\\" $2;\\\\n    }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"1df0be4a-13b7-438a-8940-c5c01f6b1707","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":5,"input_connections":{"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":523.9070991902965,"top":167.47348284368692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"NR==1 {\\\\n    OFS=\\\\\\"\\\\\\\\t\\\\\\";\\\\n    print \\\\\\"#SampleID\\\\\\", $1;\\\\n}\\\\nNR>0 {\\\\n    print $2, $3;\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"41005051-0bae-41c8-a28c-abacd01ebebc","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Grouping1","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":513.5768930139466,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"sum\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"4c5830bb-82f1-4407-b37f-9e424cb8e797","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":7,"input_connections":{"input_tabular":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":743.2357210275393,"top":166.08754960449474},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"7239102d-4e4a-4aba-971b-bbb14c560d54","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":738.5411523428047,"top":427.71442550430794},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.1","tool_shed_repository":{"changeset_revision":"90f657745fea","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"prepend_dataset_name\\", \\"__current_case__\\": 4}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.1","type":"tool","uuid":"76a42aee-056b-4b82-b0b3-59a474e3dd07","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":966.0499877929688,"top":166.12169075483803},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the first header as it is\\\\n  printf \\\\\\"%s\\\\\\", $1\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Print the rest of the lines as they are\\\\n  for (i = 1; i <= NF; i++) {\\\\n    printf \\\\\\"%s%s\\\\\\", $i, (i < NF ? OFS : ORS)\\\\n  }\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"3b0a6563-2da1-407f-ba77-2978a61ac2bc","when":null,"workflow_outputs":[{"label":"Taxonomic abundance summary table","output_name":"outfile","uuid":"ce7e544e-3f02-4ec9-a12d-0af3886b6bc5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":959.3902118058909,"top":426.34269221571736},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS = OFS = \\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  print \\\\\\"taxa\\\\\\", $1\\\\n}\\\\nNR > 0 {\\\\n  print $2, $3\\\\n}\\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b02b2b8c-1968-4e9c-89f0-0e1a26692663","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":11,"input_connections":{"input_tabular":{"id":10,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1180.2392712689773,"top":424.9709589271268},"post_job_actions":{"HideDatasetActiontabular_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tabular_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"1\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"b6121ed1-f88b-4c5f-94f4-ed90b31cc089","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1407.949951171875,"top":423.5999755859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"${Phylum level taxonomic abundance summary table name}"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN {\\\\n  FS=OFS=\\\\\\"\\\\\\\\t\\\\\\"\\\\n}\\\\nNR == 1 {\\\\n  # Print the renamed headers for the new columns\\\\n  printf \\\\\\"superkingdom\\\\\\\\tkingdom\\\\\\\\tphylum\\\\\\"\\\\n  # Adjust the remaining original headers to only include the part before the first underscore\\\\n  for (i = 2; i <= NF; i++) {\\\\n    split($i, header_parts, \\\\\\"_\\\\\\")\\\\n    printf \\\\\\"\\\\\\\\t%s\\\\\\", header_parts[1]\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\\\nNR > 1 {\\\\n  # Split the first column by \';\' into three parts\\\\n  split($1, taxa_parts, \\\\\\";\\\\\\")\\\\n  superkingdom = (taxa_parts[1] != \\\\\\"\\\\\\" ? taxa_parts[1] : \\\\\\"unassigned\\\\\\")\\\\n  kingdom = (length(taxa_parts[2]) ? taxa_parts[2] : \\\\\\"unassigned\\\\\\")\\\\n  phylum = (length(taxa_parts[3]) ? taxa_parts[3] : \\\\\\"unassigned\\\\\\")\\\\n  printf \\\\\\"%s\\\\\\\\t%s\\\\\\\\t%s\\\\\\", superkingdom, kingdom, phylum\\\\n  # Process the remaining columns\\\\n  for (i = 2; i <= NF; i++) {\\\\n    if ($i != \\\\\\"\\\\\\") {\\\\n      split($i, part, \\\\\\"_\\\\\\")\\\\n      printf \\\\\\"\\\\\\\\t%s\\\\\\", part[1]\\\\n    } else {\\\\n      printf \\\\\\"\\\\\\\\tunassigned\\\\\\"\\\\n    }\\\\n  }\\\\n  printf \\\\\\"\\\\\\\\n\\\\\\"\\\\n}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ae416cc9-c888-4eef-bc43-b15a1ac9899f","when":null,"workflow_outputs":[{"label":"phylum level taxonomic abundance summary table","output_name":"outfile","uuid":"58153320-f0c2-4d63-aa04-43f04c791da2"}]}},"tags":["MGnify_Amplicon","amplicon","metagenomics","name:microgalaxy"],"uuid":"dda70895-1e39-42a3-8b4f-b6493cf15e8f","version":75},"readme":"# MGnify\'s summary tables (v5.0)\\n\\nThis subworkflow generates taxonomic abundance summary tables.\\n\\n## Input datasets\\n\\n- **OTU table collection** containing the OTU tables of all datasets.\\n\\n## Outputs\\n\\n- Phylum level abundance summary table\\n- Taxonomic abundance summary table consisting of all levels.","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify amplicon summary tables\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ OTU tables\\"]@{ shape: docs }\\n1[\\"ℹ️ Taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Phylum level taxonomic abundance summary table name\\"]@{ shape: lean-l }\\n3[\\"Query Tabular\\"]@{ shape: process }\\n0 --> 3\\n4[\\"Text reformatting\\"]@{ shape: process }\\n0 --> 4\\n5[\\"Text reformatting\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Group\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Column join\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Filter Tabular\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Text reformatting\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Text reformatting\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Column join\\"]@{ shape: process }\\n10 --> 11\\n12[\\"Text reformatting\\"]@{ shape: process }\\n11 --> 12\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-taxonomic-summary-tables/main","dockstore_id":30453,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078773","tests":[{"doc":"Test outline for mgnify-amplicon-summary-tables.ga","job":{"OTU tables":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"ERR3046414_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR3046414_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR3046440_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR3046440_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR4319664_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR4319664_MERGED_FASTQ_SSU_OTU.tabular"},{"class":"File","identifier":"ERR4319712_MERGED_FASTQ_SSU_OTU.tabular","location":"https://zenodo.org/records/13710235/files/ERR4319712_MERGED_FASTQ_SSU_OTU.tabular"}]},"Taxonomic abundance summary table name":"taxonomic_summary_table","Phylum level taxonomic abundance summary table name":"phylum_taxonomic_summary_table"},"outputs":{"Taxonomic abundance summary table":{"location":"https://zenodo.org/records/13710235/files/taxonomic_summary_table.tabular"},"phylum level taxonomic abundance summary table":{"location":"https://zenodo.org/records/13710235/files/phylum_taxonomic_summary_table.tabular"}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-taxonomic-summary-tables"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-pipeline-v5-quality-control-single-end.ga","testParameterFiles":["/mgnify-amplicon-pipeline-v5-quality-control-single-end-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X"},{"name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"}],"definition":{"a_galaxy_workflow":"true","annotation":"Quality control subworkflow for single-end reads.","comments":[],"creator":[{"class":"Organization","name":"MGnify - EMBL","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - Quality control SE","release":"0.3","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Single-end reads collection.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Single-end reads collection.","name":"Single-end reads"}],"label":"Single-end reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":219.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"743430cc-e893-4211-9107-663adf3b28b4","when":null,"workflow_outputs":[]},"1":{"annotation":"Average quality required.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Average quality required.","name":"Trimmomatic - SLIDING WINDOW - Average quality required"}],"label":"Trimmomatic - SLIDING WINDOW - Average quality required","name":"Input parameter","outputs":[],"position":{"left":284.8237910552773,"top":0},"tool_id":null,"tool_state":"{\\"default\\": 15, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"7b3ec08b-9dd4-4215-8901-c0eb6f0be3e7","when":null,"workflow_outputs":[]},"2":{"annotation":"Cut bases off the start of a read, if below a threshold quality.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Cut bases off the start of a read, if below a threshold quality.","name":"Trimmomatic - LEADING"}],"label":"Trimmomatic - LEADING","name":"Input parameter","outputs":[],"position":{"left":235.82379105527724,"top":373.00000000000085},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2f3fe5f1-44e3-4117-9417-d68b8a90aa23","when":null,"workflow_outputs":[]},"3":{"annotation":"Number of bases to average across.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Number of bases to average across.","name":"Trimmomatic - SLIDING WINDOW - Number of bases to average across"}],"label":"Trimmomatic - SLIDING WINDOW - Number of bases to average across","name":"Input parameter","outputs":[],"position":{"left":519.8817910552779,"top":90.00000000000021},"tool_id":null,"tool_state":"{\\"default\\": 4, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5b70e0b8-0631-4059-aa36-d1ce896e45b1","when":null,"workflow_outputs":[]},"4":{"annotation":"Cut bases off the end of a read, if below a threshold quality.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Cut bases off the end of a read, if below a threshold quality.","name":"Trimmomatic - TRAILING"}],"label":"Trimmomatic - TRAILING","name":"Input parameter","outputs":[],"position":{"left":236.21679574468544,"top":501.2064765902164},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"eed3e996-2baf-4908-ba47-449254adc3cb","when":null,"workflow_outputs":[]},"5":{"annotation":"Minimum length of reads to be kept.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Minimum length of reads to be kept.","name":"Trimmomatic - MINLEN"}],"label":"Trimmomatic - MINLEN","name":"Input parameter","outputs":[],"position":{"left":218.92879574468543,"top":624.2064765902168},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"08f7f8d1-ad75-42cf-91e0-bb549c8edd14","when":null,"workflow_outputs":[]},"6":{"annotation":"Minimum sequence length.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Minimum sequence length.","name":"Length filtering - Minimum size"}],"label":"Length filtering - Minimum size","name":"Input parameter","outputs":[],"position":{"left":734.3396679241868,"top":219.62078434990826},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5db2e477-4d58-4380-bc0d-e02a2249045c","when":null,"workflow_outputs":[]},"7":{"annotation":"Maximal N percentage threshold to conserve sequences.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Maximal N percentage threshold to conserve sequences.","name":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences"}],"label":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences","name":"Input parameter","outputs":[],"position":{"left":1241.4736763919316,"top":192.6829501818756},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1d6681c7-d75f-49ce-a21b-e5779b0cadca","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":8,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":137.83379001997133,"top":1070.203938085305},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"0ef7f04f-c955-4909-81d0-51665f07604f","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","errors":null,"id":9,"input_connections":{"operations_0|operation|required_quality":{"id":1,"output_name":"output"},"operations_0|operation|window_size":{"id":3,"output_name":"output"},"operations_1|operation|leading":{"id":2,"output_name":"output"},"operations_2|operation|trailing":{"id":4,"output_name":"output"},"operations_3|operation|minlen":{"id":5,"output_name":"output"},"readtype|fastq_in":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Trimmomatic","name":"readtype"}],"label":"Trimming","name":"Trimmomatic","outputs":[{"name":"fastq_out","type":"input"}],"position":{"left":647.7861331470302,"top":337.8731636019347},"post_job_actions":{"ChangeDatatypeActionfastq_out":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"fastq_out"},"HideDatasetActionfastq_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"fastq_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","tool_shed_repository":{"changeset_revision":"b9aaed85cbd1","name":"trimmomatic","owner":"pjbriggs","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"illuminaclip\\": {\\"do_illuminaclip\\": \\"no\\", \\"__current_case__\\": 1}, \\"operations\\": [{\\"__index__\\": 0, \\"operation\\": {\\"name\\": \\"SLIDINGWINDOW\\", \\"__current_case__\\": 0, \\"window_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"required_quality\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"operation\\": {\\"name\\": \\"LEADING\\", \\"__current_case__\\": 2, \\"leading\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"operation\\": {\\"name\\": \\"TRAILING\\", \\"__current_case__\\": 3, \\"trailing\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"operation\\": {\\"name\\": \\"MINLEN\\", \\"__current_case__\\": 1, \\"minlen\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"output_err\\": false, \\"output_logs\\": false, \\"quality_score\\": \\"-phred33\\", \\"readtype\\": {\\"single_or_paired\\": \\"se\\", \\"__current_case__\\": 0, \\"fastq_in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.39+galaxy2","type":"tool","uuid":"6a258b5c-2dbd-4615-88f8-1908bf555374","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":10,"input_connections":{"infile":{"id":8,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":521.8083204698337,"top":1301.3107808231161},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_initial_reads\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"93d70102-188c-4c9e-b0b4-fcb24acd8a28","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","errors":null,"id":11,"input_connections":{"input_file":{"id":9,"output_name":"fastq_out"},"min_size":{"id":6,"output_name":"output"}},"inputs":[],"label":"Length filtering","name":"Filter FASTQ","outputs":[{"name":"output_file","type":"input"}],"position":{"left":891.3091779432864,"top":370.3116497676374},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"144275714590","name":"fastq_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fastq_filters\\": [], \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_num_deviants\\": \\"0\\", \\"max_quality\\": \\"0.0\\", \\"max_size\\": \\"0\\", \\"min_quality\\": \\"0.0\\", \\"min_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paired_end\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"c22e8401-0e9e-404e-8fa4-3cc5e28670b2","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":12,"input_connections":{"input_file":{"id":9,"output_name":"fastq_out"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":644.4179064261575,"top":955.8740730752126},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"18505eb1-1b95-431e-b91f-1b73b50e2e47","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","errors":null,"id":13,"input_connections":{"filter_treatments|base_content_filter_treatments|N_percentage_content_filter_treatments|N_percentage_content_filter_treatment_value":{"id":7,"output_name":"output"},"seq_type|input_singles":{"id":11,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool PRINSEQ","name":"seq_type"}],"label":"Ambiguity filtering","name":"PRINSEQ","outputs":[{"name":"good_sequence_file","type":"input"},{"name":"rejected_sequence_file","type":"input"}],"position":{"left":1165.055726007993,"top":432.1991937304916},"post_job_actions":{"HideDatasetActiongood_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"good_sequence_file"},"HideDatasetActionrejected_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"rejected_sequence_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","tool_shed_repository":{"changeset_revision":"1ee282794de3","name":"prinseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_treatments\\": {\\"apply_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"length_filter_treatments\\": {\\"apply_length_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"quality_filter_treatments\\": {\\"apply_quality_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"base_content_filter_treatments\\": {\\"apply_base_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"GC_perc_content_filter_treatments\\": {\\"apply_GC_perc_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_number_content_filter_treatments\\": {\\"apply_N_number_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_percentage_content_filter_treatments\\": {\\"apply_N_percentage_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"N_percentage_content_filter_treatment_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"apply_other_base_content_filter_treatments\\": true}, \\"complexity_filter_treatments\\": {\\"apply_complexity_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}}, \\"seq_type\\": {\\"seq_type_opt\\": \\"single\\", \\"__current_case__\\": 0, \\"input_singles\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"trimming_treatments\\": {\\"apply_trimming_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.20.4+galaxy2","type":"tool","uuid":"c7c4f665-6256-4693-b13f-3feec07b2bd6","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":14,"input_connections":{"input_file":{"id":11,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1019.7935387142895,"top":800.338235605041},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"6fc925d9-c644-4f8d-9435-a1e94b151eca","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":15,"input_connections":{"infile":{"id":12,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":947.4179064261575,"top":1153.2574043740406},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_trimming\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"687e1130-33fc-4890-8099-e2c6da98f836","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","errors":null,"id":16,"input_connections":{"input_file":{"id":13,"output_name":"good_sequence_file"}},"inputs":[],"label":null,"name":"FASTQ to FASTA","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":1648.8164640063083,"top":634.3621610446716},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"ac3fed111eb6","name":"fastqtofasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"9bc910c0-f85a-474e-9a38-3c5cc2c3434c","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":17,"input_connections":{"input_file":{"id":13,"output_name":"good_sequence_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1612.1550210976457,"top":812.2383769545825},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"5082dc16-2032-46fe-a469-cf79af5da99c","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":18,"input_connections":{"infile":{"id":14,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1331.4716094560185,"top":1063.8186665413482},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_length_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"1650ae45-7ee2-4765-8953-48a6548b1c67","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":19,"input_connections":{"input":{"id":16,"output_name":"output_file"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1941.646496697877,"top":647.1088294674363},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Single-end post quality control FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"91a2a702-a2e0-49ec-9864-8d4edb56986e","when":null,"workflow_outputs":[{"label":"Single-end post quality control FASTA files","output_name":"output","uuid":"5f19c1d8-2a5e-48ad-a962-012d20b6cac5"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":20,"input_connections":{"infile":{"id":17,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1877.7115272508015,"top":1026.8671426905246},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_ambiguous_base_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"284c9abf-2ebb-490c-a4d9-1fe997a1f562","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":21,"input_connections":{"results_0|software_cond|output_0|input":{"id":10,"output_name":"outfile"},"results_0|software_cond|output_1|input":{"id":15,"output_name":"outfile"},"results_0|software_cond|output_2|input":{"id":18,"output_name":"outfile"},"results_0|software_cond|output_3|input":{"id":20,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1387.0957816692048,"top":1445.9486517075318},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Single-end MultiQC report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Single-end MultiQC statistics"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 3, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"640f4f45-c12a-47e0-846b-9b150ed8beca","when":null,"workflow_outputs":[{"label":"Single-end MultiQC statistics","output_name":"stats","uuid":"125cc43d-33aa-4b6f-a3fa-1c7c4ef3488f"},{"label":"Single-end MultiQC report","output_name":"html_report","uuid":"81732dfa-d5cb-46f5-ba34-13242ab5e3fa"}]}},"tags":["MGnify_Amplicon","amplicon","metagenomics","name:microgalaxy"],"uuid":"68bf2a73-d68a-4e25-ba52-5586c057df77","version":20},"readme":"# MGnify\'s amplicon (v5.0) quality control subworkflow for single end data\\n\\n## Input datasets\\n\\n- **Single-end reada** A collection containing single-end reads.\\n\\n## Input values\\n\\n- **Trimmomatic - SLIDING WINDOW - Average quality required** (Default = 15)\\n- **Trimmomatic - LEADING** Minimum quality required to keep a base, (Default = 3)\\n- **Trimmomatic - TRAILING** Minimum quality required to keep a base, (Default = 3)\\n- **Trimmomatic - SLIDING WINDOW - Number of bases to average across** (Default = 4)\\n- **Trimmomatic - MINLEN** Minimum length of reads to be kept, (Default = 100)\\n- **Trimmomatic - Quality score encoding** The phred+64 encoding works the same as the phred+33 encoding, except you add 64 to the phred score to determine the ascii code of the quality character, (Default = Phred33)\\n- **Length filtering - Minimum size** Minimum sequence length, (Default = 100)\\n- **Ambiguity filtering - Maximal N percentage threshold to conserve sequences** Maximal N percentage threshold to conserve sequences, (Default = 10)\\n\\n## Processing\\n\\nThis subworkflow executes the following quality control steps:\\n\\n- **Trimmomatic** Filtering and trimming reads using trimmomatic\\n- **Filter FASTQ** Filter reads shorter than 100 bp\\n- **PRINSEQ** Ambiguity filtering","changelog":"# Changelog\\n\\n## [0.3] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.2] - 2025-02-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.1] - 10-05-2024\\n\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify\'s amplicon pipeline v5.0 - Quality control SE\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Single-end reads\\"]@{ shape: docs }\\n1[\\"ℹ️ Trimmomatic - SLIDING WINDOW - Average quality required\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Trimmomatic - LEADING\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Trimmomatic - SLIDING WINDOW - Number of bases to average across\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Trimmomatic - TRAILING\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimmomatic - MINLEN\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Length filtering - Minimum size\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Ambiguity filtering - Maximal N percentage threshold to conserve sequences\\"]@{ shape: lean-l }\\n8[\\"FastQC\\"]@{ shape: process }\\n0 --> 8\\n9[\\"Trimming\\"]@{ shape: process }\\n1 --> 9\\n3 --> 9\\n2 --> 9\\n4 --> 9\\n5 --> 9\\n0 --> 9\\n10[\\"Replace\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Length filtering\\"]@{ shape: process }\\n9 --> 11\\n6 --> 11\\n12[\\"FastQC\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Ambiguity filtering\\"]@{ shape: process }\\n7 --> 13\\n11 --> 13\\n14[\\"FastQC\\"]@{ shape: process }\\n11 --> 14\\n15[\\"Replace\\"]@{ shape: process }\\n12 --> 15\\n16[\\"FASTQ to FASTA\\"]@{ shape: process }\\n13 --> 16\\n17[\\"FastQC\\"]@{ shape: process }\\n13 --> 17\\n18[\\"Replace\\"]@{ shape: process }\\n14 --> 18\\n19[\\"FASTA Width\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Replace\\"]@{ shape: process }\\n17 --> 20\\n21[\\"MultiQC\\"]@{ shape: process }\\n10 --> 21\\n15 --> 21\\n18 --> 21\\n20 --> 21\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-single-end/main","dockstore_id":30449,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078839","tests":[{"doc":"Test outline for mgnify-amplicon-pipeline-v5-quality-control-single-end.ga","job":{"Single-end reads":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"DRR010481","location":"https://zenodo.org/records/13710235/files/DRR010481.fastqsanger.gz"}]}},"outputs":{"Single-end MultiQC statistics":{"location":"https://zenodo.org/records/14746033/files/general_stats_SE.tabular"},"Single-end MultiQC report":{"asserts":[{"that":"has_text","text":"DRR010481_ambiguous_base_filtering"},{"that":"has_text","text":"84.0"},{"that":"has_text","text":"DRR010481_initial_reads"},{"that":"has_text","text":"47"},{"that":"has_text","text":"DRR010481_length_filtering"},{"that":"has_text","text":"DRR010481_trimming"},{"that":"has_text","text":"DRR010481_ambiguous_base_filtering"},{"that":"has_text","text":"48"},{"that":"has_text","text":"553"},{"that":"has_text","text":"242"},{"that":"has_text","text":"DRR010481_initial_reads"},{"that":"has_text","text":"DRR010481_length_filtering"},{"that":"has_text","text":"DRR010481_trimming"}]},"Single-end post quality control FASTA files":{"element_tests":{"DRR010481":{"location":"https://zenodo.org/records/13710235/files/DRR010481_SE.fasta"}}}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-pipeline-v5-quality-control-single-end"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/mgnify-amplicon-pipeline-v5-quality-control-paired-end.ga","testParameterFiles":["/mgnify-amplicon-pipeline-v5-quality-control-paired-end-tests.yml"],"authors":[{"name":"Rand Zoabi","orcid":"0009-0000-2501-8053"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X"},{"name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"}],"definition":{"a_galaxy_workflow":"true","annotation":"Quality control subworkflow for paired-end reads. ","comments":[],"creator":[{"class":"Organization","name":"EMBL\'s European Bioinformatics Institute","url":"https://www.ebi.ac.uk/metagenomics"},{"class":"Person","identifier":"https://orcid.org/0009-0000-2501-8053","name":"Rand Zoabi","url":"https://github.com/RZ9082"},{"class":"Person","identifier":"https://orcid.org/0000-0003-2982-388X","name":"Paul Zierep","url":"https://github.com/paulzierep"}],"format-version":"0.1","license":"Apache-2.0","name":"MGnify\'s amplicon pipeline v5.0 - Quality control PE","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Paired-end reads collection.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Paired-end reads collection.","name":"Paired-end reads"}],"label":"Paired-end reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":134.00843148565127},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"4a419fd7-0b26-4f9d-83ec-2471f862c610","when":null,"workflow_outputs":[]},"1":{"annotation":" Enable base correction in overlapped regions.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":" Enable base correction in overlapped regions.","name":"fastp - Enable base correction"}],"label":"fastp - Enable base correction","name":"Input parameter","outputs":[],"position":{"left":67.94649253008049,"top":317.0543088585308},"tool_id":null,"tool_state":"{\\"default\\": false, \\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"81367449-c0ca-41be-b69d-5bf1568e0361","when":null,"workflow_outputs":[]},"2":{"annotation":"The quality value that a base is qualified.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"The quality value that a base is qualified.","name":"fastp - Qualified quality phred"}],"label":"fastp - Qualified quality phred","name":"Input parameter","outputs":[],"position":{"left":309.2157898611554,"top":329.9376208519377},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"cb8a7bac-c007-45c6-814b-9aacae0e37b3","when":null,"workflow_outputs":[]},"3":{"annotation":"How many percents of bases are allowed to be unqualified (0~100).","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"How many percents of bases are allowed to be unqualified (0~100).","name":"fastp - Unqualified percent limit "}],"label":"fastp - Unqualified percent limit ","name":"Input parameter","outputs":[],"position":{"left":311.55821022359305,"top":420.12080480578607},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"26ed5910-22cf-4a8e-a144-edbb8d6b1d50","when":null,"workflow_outputs":[]},"4":{"annotation":"Reads shorter than this value will be discarded.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Reads shorter than this value will be discarded.","name":"fastp - Length required"}],"label":"fastp - Length required","name":"Input parameter","outputs":[],"position":{"left":310.3870000423742,"top":506.790358215978},"tool_id":null,"tool_state":"{\\"default\\": 70, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6521e9f3-6d1b-4e95-8934-3345f858bcf8","when":null,"workflow_outputs":[]},"5":{"annotation":"Number of bases to average across.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Number of bases to average across.","name":"Trimmomatic - SLIDINGWINDOW - Number of bases to average across"}],"label":"Trimmomatic - SLIDINGWINDOW - Number of bases to average across","name":"Input parameter","outputs":[],"position":{"left":310.8945435169014,"top":618.8166406435558},"tool_id":null,"tool_state":"{\\"default\\": 4, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1b2652b0-235b-4120-9c82-e4d3210c9d2f","when":null,"workflow_outputs":[]},"6":{"annotation":"Average quality required.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Average quality required.","name":"Trimmomatic - SLIDINGWINDOW - Average quality required"}],"label":"Trimmomatic - SLIDINGWINDOW - Average quality required","name":"Input parameter","outputs":[],"position":{"left":312.5205763417023,"top":751.1494950478717},"tool_id":null,"tool_state":"{\\"default\\": 15, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"53e87ca4-6745-4835-b25f-878705737f0d","when":null,"workflow_outputs":[]},"7":{"annotation":"Minimum quality required to keep a base.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Minimum quality required to keep a base.","name":"Trimmomatic - LEADING"}],"label":"Trimmomatic - LEADING","name":"Input parameter","outputs":[],"position":{"left":315.85885539878177,"top":880.5228796617778},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"4e1cfaad-b067-4bf3-a6b8-33d3fd826285","when":null,"workflow_outputs":[]},"8":{"annotation":"Minimum quality required to keep a base.","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Minimum quality required to keep a base.","name":"Trimmomatic - TRAILING"}],"label":"Trimmomatic - TRAILING","name":"Input parameter","outputs":[],"position":{"left":330.0774689415621,"top":973.9211632212501},"tool_id":null,"tool_state":"{\\"default\\": 3, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d48c0437-1ab6-4fd4-8dfa-29dd056acaec","when":null,"workflow_outputs":[]},"9":{"annotation":"Minimum sequence length.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Minimum sequence length.","name":"Length filtering - Minimum size"}],"label":"Length filtering - Minimum size","name":"Input parameter","outputs":[],"position":{"left":993.3486541402174,"top":364.32685651072273},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2957cfb6-b542-4354-b94f-fd171df4c9b1","when":null,"workflow_outputs":[]},"10":{"annotation":"Minimum length of reads to be kept.","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Minimum length of reads to be kept.","name":"Trimmomatic - MINLEN"}],"label":"Trimmomatic - MINLEN","name":"Input parameter","outputs":[],"position":{"left":332.17940873206794,"top":1062.1550248633116},"tool_id":null,"tool_state":"{\\"default\\": 100, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c898eada-b136-40bf-a78d-240b8d02c660","when":null,"workflow_outputs":[]},"11":{"annotation":"Maximal N percentage threshold to conserve sequences.","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"Maximal N percentage threshold to conserve sequences.","name":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences"}],"label":"Ambiguity filtering - Maximal N percentage threshold to conserve sequences","name":"Input parameter","outputs":[],"position":{"left":1443.0933637282412,"top":237.69561171734486},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d8341e6e-1401-48ef-ac1a-68c98309d0e1","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":12,"input_connections":{"filter_options|length_filtering_options|length_required":{"id":4,"output_name":"output"},"filter_options|quality_filtering_options|qualified_quality_phred":{"id":2,"output_name":"output"},"filter_options|quality_filtering_options|unqualified_percent_limit":{"id":3,"output_name":"output"},"read_mod_options|base_correction_options|correction":{"id":1,"output_name":"output"},"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":331.5,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unqualified_percent_limit\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": null, \\"umi_len\\": null, \\"umi_prefix\\": null}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"paired_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": null, \\"adapter_sequence2\\": null, \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"9b7f1d2d-d39f-407d-a709-a5095fca62c2","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__UNZIP_COLLECTION__","errors":null,"id":13,"input_connections":{"input":{"id":12,"output_name":"output_paired_coll"}},"inputs":[],"label":null,"name":"Unzip collection","outputs":[{"name":"forward","type":"input"},{"name":"reverse","type":"input"}],"position":{"left":610.2833251953125,"top":342.9666748046875},"post_job_actions":{"ChangeDatatypeActionforward":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"forward"},"ChangeDatatypeActionreverse":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"reverse"},"HideDatasetActionforward":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"forward"},"HideDatasetActionreverse":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reverse"}},"tool_id":"__UNZIP_COLLECTION__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"f151e04c-8502-45b7-8503-38838efe39ee","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mgnify_seqprep/mgnify_seqprep/1.2+galaxy0","errors":null,"id":14,"input_connections":{"input1":{"id":13,"output_name":"forward"},"input2":{"id":13,"output_name":"reverse"}},"inputs":[],"label":null,"name":"Merging paired-end Illumina reads (SeqPrep, modified for use with MGnify piplines)","outputs":[{"name":"output1","type":"fastq.gz"},{"name":"output2","type":"fastq.gz"},{"name":"merged","type":"fastq.gz"}],"position":{"left":788.0499877929688,"top":649.379055631159},"post_job_actions":{"ChangeDatatypeActionmerged":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"merged"},"HideDatasetActionmerged":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"merged"},"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mgnify_seqprep/mgnify_seqprep/1.2+galaxy0","tool_shed_repository":{"changeset_revision":"76ea9d4604bc","name":"mgnify_seqprep","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"general_options\\": {\\"first_read_discarded\\": null, \\"second_read_discarded\\": null, \\"phred64\\": false, \\"quality_cutoff\\": \\"13\\", \\"min_length\\": \\"30\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": true, \\"merging_options\\": {\\"maximum_quality_score\\": null, \\"print_overhang\\": false, \\"min_base_pair_overlap\\": \\"15\\", \\"max_mismatch_fraction\\": \\"0.02\\", \\"min_match_fraction\\": \\"0.9\\"}, \\"trimming_options\\": {\\"adapter_a\\": \\"AGATCGGAAGAGCGGTTCAG\\", \\"adapter_b\\": \\"AGATCGGAAGAGCGTCGTGT\\", \\"adapter_overlap\\": \\"10\\", \\"max_mismatch_fraction\\": \\"0.02\\", \\"min_match_fraction\\": \\"0.87\\", \\"adapter_bandwidth\\": \\"50\\", \\"gap_open\\": \\"8\\", \\"gap_extend\\": \\"2\\", \\"gap_end\\": \\"2\\", \\"local_alignment_score\\": \\"26\\", \\"read_alignment_bandwidth\\": \\"50\\", \\"read_alignment_gap_open\\": \\"26\\", \\"read_alignment_gap_extend\\": \\"9\\", \\"read_alignment_gap_end\\": \\"5\\", \\"read_alignment_max_gap_fraction\\": \\"0.125\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2+galaxy0","type":"tool","uuid":"670c6abb-2a49-4a77-9cad-eaae9ebfe74e","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","errors":null,"id":15,"input_connections":{"operations_0|operation|required_quality":{"id":6,"output_name":"output"},"operations_0|operation|window_size":{"id":5,"output_name":"output"},"operations_1|operation|leading":{"id":7,"output_name":"output"},"operations_2|operation|trailing":{"id":8,"output_name":"output"},"operations_3|operation|minlen":{"id":10,"output_name":"output"},"readtype|fastq_in":{"id":14,"output_name":"merged"}},"inputs":[{"description":"runtime parameter for tool Trimmomatic","name":"readtype"}],"label":"Trimming","name":"Trimmomatic","outputs":[{"name":"fastq_out","type":"input"}],"position":{"left":1099.8073621070885,"top":478.93954202505694},"post_job_actions":{"HideDatasetActionfastq_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"fastq_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.39+galaxy2","tool_shed_repository":{"changeset_revision":"b9aaed85cbd1","name":"trimmomatic","owner":"pjbriggs","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"illuminaclip\\": {\\"do_illuminaclip\\": \\"no\\", \\"__current_case__\\": 1}, \\"operations\\": [{\\"__index__\\": 0, \\"operation\\": {\\"name\\": \\"SLIDINGWINDOW\\", \\"__current_case__\\": 0, \\"window_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"required_quality\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"operation\\": {\\"name\\": \\"LEADING\\", \\"__current_case__\\": 2, \\"leading\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"operation\\": {\\"name\\": \\"TRAILING\\", \\"__current_case__\\": 3, \\"trailing\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"operation\\": {\\"name\\": \\"MINLEN\\", \\"__current_case__\\": 1, \\"minlen\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"output_err\\": false, \\"output_logs\\": false, \\"quality_score\\": \\"-phred33\\", \\"readtype\\": {\\"single_or_paired\\": \\"se\\", \\"__current_case__\\": 0, \\"fastq_in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.39+galaxy2","type":"tool","uuid":"c2079512-2019-41aa-beea-d1b597e59b42","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":16,"input_connections":{"input_file":{"id":14,"output_name":"merged"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1250.6521000986795,"top":1200.9140694839034},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"c48ed7ed-5a1d-40e8-8423-434578e91568","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","errors":null,"id":17,"input_connections":{"input_file":{"id":15,"output_name":"fastq_out"},"min_size":{"id":9,"output_name":"output"}},"inputs":[],"label":"Length filtering","name":"Filter FASTQ","outputs":[{"name":"output_file","type":"input"}],"position":{"left":1359,"top":440.365741330622},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastq_filter/fastq_filter/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"144275714590","name":"fastq_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fastq_filters\\": [], \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_num_deviants\\": \\"0\\", \\"max_quality\\": \\"0.0\\", \\"max_size\\": \\"0\\", \\"min_quality\\": \\"0.0\\", \\"min_size\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paired_end\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"a12d9770-de9e-45cf-8d18-2be9cc7873ed","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":18,"input_connections":{"input_file":{"id":15,"output_name":"fastq_out"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1546.9682759470388,"top":1199.7428593026848},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"01bb817a-6e2a-4a81-b5a7-ec42f0355b3e","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":19,"input_connections":{"infile":{"id":16,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1593.8207131348126,"top":1563.9834870611885},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_initial_reads\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"680b293b-0b1e-4b95-9dc9-233354651b61","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","errors":null,"id":20,"input_connections":{"filter_treatments|base_content_filter_treatments|N_percentage_content_filter_treatments|N_percentage_content_filter_treatment_value":{"id":11,"output_name":"output"},"seq_type|input_singles":{"id":17,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool PRINSEQ","name":"seq_type"}],"label":"Ambiguity filtering","name":"PRINSEQ","outputs":[{"name":"good_sequence_file","type":"input"},{"name":"rejected_sequence_file","type":"input"}],"position":{"left":1646.903450664335,"top":523.0258360658718},"post_job_actions":{"HideDatasetActiongood_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"good_sequence_file"},"HideDatasetActionrejected_sequence_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"rejected_sequence_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/prinseq/prinseq/0.20.4+galaxy2","tool_shed_repository":{"changeset_revision":"1ee282794de3","name":"prinseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_treatments\\": {\\"apply_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"length_filter_treatments\\": {\\"apply_length_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"quality_filter_treatments\\": {\\"apply_quality_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"base_content_filter_treatments\\": {\\"apply_base_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"GC_perc_content_filter_treatments\\": {\\"apply_GC_perc_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_number_content_filter_treatments\\": {\\"apply_N_number_content_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"N_percentage_content_filter_treatments\\": {\\"apply_N_percentage_content_filter_treatments\\": \\"true\\", \\"__current_case__\\": 0, \\"N_percentage_content_filter_treatment_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"apply_other_base_content_filter_treatments\\": true}, \\"complexity_filter_treatments\\": {\\"apply_complexity_filter_treatments\\": \\"false\\", \\"__current_case__\\": 1}}, \\"seq_type\\": {\\"seq_type_opt\\": \\"single\\", \\"__current_case__\\": 0, \\"input_singles\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"trimming_treatments\\": {\\"apply_trimming_treatments\\": \\"false\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.20.4+galaxy2","type":"tool","uuid":"d65b0e38-5f52-425a-a3c3-9cb126ead9f7","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":21,"input_connections":{"input_file":{"id":17,"output_name":"output_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1536.6496929362308,"top":818.1691488145751},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"f291100e-19ae-4baf-824b-16244df1e9f3","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":22,"input_connections":{"infile":{"id":18,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1919.413113574621,"top":1164.60655386612},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_trimming\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"2bb5af38-51bd-4828-bc34-7b4c7273c9e2","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","errors":null,"id":23,"input_connections":{"input_file":{"id":20,"output_name":"good_sequence_file"}},"inputs":[],"label":null,"name":"FASTQ to FASTA","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":1952.1162504932327,"top":541.8794796760812},"post_job_actions":{"HideDatasetActionoutput_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5+galaxy2","tool_shed_repository":{"changeset_revision":"ac3fed111eb6","name":"fastqtofasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.5+galaxy2","type":"tool","uuid":"7b5c5aeb-0201-4378-984c-ded57d114157","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":24,"input_connections":{"input_file":{"id":20,"output_name":"good_sequence_file"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":2282.6440047444867,"top":666.8629867716423},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"FastQC after Ambiguity filtering"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"9afd3a4f-fdae-4d9d-9afc-5990276dc8c5","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":21,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2169.018738976497,"top":996.4944763313531},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_length_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"8c5dc662-6bcc-4772-bc0e-d0b747113ef3","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","errors":null,"id":26,"input_connections":{"input":{"id":23,"output_name":"output_file"}},"inputs":[],"label":null,"name":"FASTA Width","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2306.5295226390967,"top":402.29376277487876},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Paired-end post quality control FASTA files"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_formatter/cshl_fasta_formatter/1.0.1+galaxy2","tool_shed_repository":{"changeset_revision":"516b97a23d7e","name":"fasta_formatter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"width\\": \\"60\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1+galaxy2","type":"tool","uuid":"b899fa36-d2c9-4cbe-880a-9984f2b48462","when":null,"workflow_outputs":[{"label":"Paired-end post quality control FASTA files","output_name":"output","uuid":"ab65a776-f7bb-4495-8eeb-12079046ae1b"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":27,"input_connections":{"infile":{"id":24,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2529.1111984955273,"top":831.6091291601473},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"\\\\\\\\t(.*?)\\\\\\\\.gz\\", \\"replace_pattern\\": \\"\\\\\\\\t$1_ambiguous_base_filtering\\", \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"aacb9221-34de-46f7-91ec-ff796e98db10","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":28,"input_connections":{"results_0|software_cond|output_0|input":{"id":19,"output_name":"outfile"},"results_0|software_cond|output_1|input":{"id":22,"output_name":"outfile"},"results_0|software_cond|output_2|input":{"id":25,"output_name":"outfile"},"results_0|software_cond|output_3|input":{"id":27,"output_name":"outfile"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2884.8980879090345,"top":597.9223680616317},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Paired-end MultiQC report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Paired-end MultiQC statistics"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 3, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"e04ee136-c52f-4d7e-85c5-3b3ed78b0cce","when":null,"workflow_outputs":[{"label":"Paired-end MultiQC statistics","output_name":"stats","uuid":"fc6047b2-201d-4e77-ab0d-9c8107897088"},{"label":"Paired-end MultiQC report","output_name":"html_report","uuid":"1d870a5b-942b-4f61-b8fd-a68b121b7f8c"}]}},"tags":["MGnify_Amplicon","amplicon","metagenomics","name:microgalaxy"],"uuid":"3af45a28-126d-4c9d-9948-cabfe43566f3","version":21},"readme":"# MGnify\'s amplicon (v5.0) quality control subworkflow for paired end data\\n\\n## Input datasets\\n\\n- **Paired-end reads** a paired list collection containing the forward and reverse reads.\\n\\n## Input values\\n\\n- **fastp - Enable base correction** Enable base correction in overlapped regions, (Default = No)\\n- **fastp - Qualified quality phred** The quality value that a base is qualified, (Default = 20)\\n- **fastp - Unqualified percent limit** How many percents of bases are allowed to be unqualified, (Default = 20)\\n- **fastp - Length required** Reads shorter than this value will be discarded, (Default = 70)\\n- **Trimmomatic - SLIDINGWINDOW - Number of bases to average across** (Default = 4)\\n- **Trimmomatic - SLIDINGWINDOW - Average quality required** (Default = 15)\\n- **Trimmomatic - LEADING** Minimum quality required to keep a base, (Default = 3)\\n- **Trimmomatic - TRAILING** Minimum quality required to keep a base, (Default = 3)\\n- **Trimmomatic - MINLEN** Minimum length of reads to be kept, (Default = 100)\\n- **Trimmomatic - Quality score encoding** The phred+64 encoding works the same as the phred+33 encoding, except you add 64 to the phred score to determine the ascii code of the quality character, (Default = Phred33)\\n- **Length filtering - Minimum size** Minimum sequence length, (Default = 100)\\n- **Ambiguity filtering - Maximal N percentage threshold to conserve sequences** Maximal N percentage threshold to conserve sequences, (Default = 10)\\n\\n## Processing\\n\\nThis subworkflow executes the following quality control steps:\\n\\n- **fastp** Quality and length filtering\\n- **SeqPrep** Merges overlapping reads into a single longer reads using SeqPrep\\n- **Trimmomatic** Filtering and trimming reads using trimmomatic\\n- **Filter FASTQ** Filter reads shorter than 100 bp\\n- **PRINSEQ** Ambiguity filtering\\n","changelog":"# Changelog\\n\\n## [0.2] - 2025-03-10\\n\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4`\\n\\n## [0.1] - 2025-01-27\\n\\n- First release\\n\\n","diagrams":"# Workflow diagrams\\n\\n## MGnify\'s amplicon pipeline v5.0 - Quality control PE\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired-end reads\\"]@{ shape: docs }\\n1[\\"ℹ️ fastp - Enable base correction\\"]@{ shape: lean-l }\\n2[\\"ℹ️ fastp - Qualified quality phred\\"]@{ shape: lean-l }\\n3[\\"ℹ️ fastp - Unqualified percent limit \\"]@{ shape: lean-l }\\n4[\\"ℹ️ fastp - Length required\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimmomatic - SLIDINGWINDOW - Number of bases to average across\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Trimmomatic - SLIDINGWINDOW - Average quality required\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Trimmomatic - LEADING\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Trimmomatic - TRAILING\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Length filtering - Minimum size\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Trimmomatic - MINLEN\\"]@{ shape: lean-l }\\n11[\\"ℹ️ Ambiguity filtering - Maximal N percentage threshold to conserve sequences\\"]@{ shape: lean-l }\\n12[\\"fastp\\"]@{ shape: process }\\n4 --> 12\\n2 --> 12\\n3 --> 12\\n1 --> 12\\n0 --> 12\\n13[\\"Unzip collection\\"]@{ shape: process }\\n12 --> 13\\n14[\\"Merging paired-end Illumina reads (SeqPrep, modified for use with MGnify piplines)\\"]@{ shape: process }\\n13 --> 14\\n13 --> 14\\n15[\\"Trimming\\"]@{ shape: process }\\n6 --> 15\\n5 --> 15\\n7 --> 15\\n8 --> 15\\n10 --> 15\\n14 --> 15\\n16[\\"FastQC\\"]@{ shape: process }\\n14 --> 16\\n17[\\"Length filtering\\"]@{ shape: process }\\n15 --> 17\\n9 --> 17\\n18[\\"FastQC\\"]@{ shape: process }\\n15 --> 18\\n19[\\"Replace\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Ambiguity filtering\\"]@{ shape: process }\\n11 --> 20\\n17 --> 20\\n21[\\"FastQC\\"]@{ shape: process }\\n17 --> 21\\n22[\\"Replace\\"]@{ shape: process }\\n18 --> 22\\n23[\\"FASTQ to FASTA\\"]@{ shape: process }\\n20 --> 23\\n24[\\"FastQC\\"]@{ shape: process }\\n20 --> 24\\n25[\\"Replace\\"]@{ shape: process }\\n21 --> 25\\n26[\\"FASTA Width\\"]@{ shape: process }\\n23 --> 26\\n27[\\"Replace\\"]@{ shape: process }\\n24 --> 27\\n28[\\"MultiQC\\"]@{ shape: process }\\n19 --> 28\\n22 --> 28\\n25 --> 28\\n27 --> 28\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/mgnify-amplicon-pipeline-v5-quality-control-paired-end/main","dockstore_id":30448,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.15078875","tests":[{"doc":"Test outline for mgnify-amplicon-pipeline-v5-quality-control-paired-end.ga","job":{"Paired-end reads":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"ERR2715528","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/records/13710235/files/ERR2715528_forward.fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/records/13710235/files/ERR2715528_reverse.fastqsanger.gz"}]}]}},"outputs":{"Paired-end post quality control FASTA files":{"element_tests":{"ERR2715528":{"location":"https://zenodo.org/records/13710235/files/ERR2715528_PE.fasta"}}},"Paired-end MultiQC statistics":{"location":"https://zenodo.org/records/14746033/files/general_stats_PE.tabular"},"Paired-end MultiQC report":{"asserts":[{"that":"has_text","text":"ERR2715528_ambiguous_base_filtering"},{"that":"has_text","text":"ERR2715528_initial_reads"},{"that":"has_text","text":"ERR2715528_length_filtering"},{"that":"has_text","text":"ERR2715528_trimming"},{"that":"has_text","text":"92.2"},{"that":"has_text","text":"51"}]}}}]}],"path":"./workflows/amplicon/amplicon-mgnify/mgnify-amplicon-pipeline-v5-quality-control-paired-end"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/dada2_paired.ga","testParameterFiles":["/dada2_paired-tests.yml"],"authors":[{"name":"Matthias Bernt","orcid":"0000-0003-3763-0797"},{"name":"UFZ Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"dada2 amplicon analysis for paired end data\\n\\nThe workflow has three main outputs: \\n- the sequence table (output of makeSequenceTable)\\n- the taxonomy (output of assignTaxonomy)\\n- the counts which allow to track the number of sequences in the samples through the steps (output of sequence counts)","creator":[{"class":"Person","identifier":"0000-0003-3763-0797","name":"Matthias Bernt"},{"class":"Organization","name":"UFZ Leipzig"}],"format-version":"0.1","license":"MIT","name":"dada2 amplicon analysis pipeline - for paired end data","release":"0.3","steps":{"0":{"annotation":"Paired input data","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Paired input data","name":"Paired input data"}],"label":"Paired input data","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":6},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"9bccdfaa-3c84-4c13-acfe-6c41aebb0d10","when":null,"workflow_outputs":[]},"1":{"annotation":"Forward reads will be truncated to this length. Empty means no truncation.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Forward reads will be truncated to this length. Empty means no truncation.","name":"Read length forward read"}],"label":"Read length forward read","name":"Input parameter","outputs":[],"position":{"left":12,"top":242.3237599999942},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"42e4d027-9232-4280-8972-86df76743ffd","when":null,"workflow_outputs":[]},"2":{"annotation":"Reverse reads will be truncated to this length. Empty means no truncation.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Reverse reads will be truncated to this length. Empty means no truncation.","name":"Read length reverse read"}],"label":"Read length reverse read","name":"Input parameter","outputs":[],"position":{"left":11,"top":367.3237599999942},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"0e8416f9-fa56-4c60-a824-e2d103210858","when":null,"workflow_outputs":[]},"3":{"annotation":"Pooling may increase sensitivity","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Pooling may increase sensitivity","name":"Pool samples"}],"label":"Pool samples","name":"Input parameter","outputs":[],"position":{"left":10.577839999617993,"top":522.3237599999942},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"69ae9a01-0659-4a34-9c68-727038d75bb6","when":null,"workflow_outputs":[]},"4":{"annotation":"for taxonomic assignment. If a cached reference is used no reference from history is used. ","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"for taxonomic assignment. If a cached reference is used no reference from history is used. ","name":"Cached reference database"}],"label":"Cached reference database","name":"Input parameter","outputs":[],"position":{"left":38.24183470363951,"top":889.0398222178009},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8f563388-f136-4cfe-8f86-a4f6e15da12c","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":5,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"Sort samples","name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":280,"top":0},"post_job_actions":{},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [0], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1], \\"connectable\\": true, \\"is_workflow\\": false, \\"type\\": \\"paired_identifier\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"numeric\\": false, \\"target_column\\": 0, \\"type\\": \\"sort\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"a14e5e93-4814-440c-8009-e93e23a6ce73","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.34.0+galaxy0","errors":null,"id":6,"input_connections":{"batch_cond|paired_cond|reads":{"id":5,"output_name":"output"}},"inputs":[],"label":"QualityProfile before filterAndTrim","name":"dada2: plotQualityProfile","outputs":[{"name":"output_fwd","type":"pdf"},{"name":"output_rev","type":"pdf"}],"position":{"left":560,"top":5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"4e469b758915","name":"dada2_plotqualityprofile","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_cond\\": {\\"batch_select\\": \\"joint\\", \\"__current_case__\\": 0, \\"paired_cond\\": {\\"paired_select\\": \\"paired\\", \\"__current_case__\\": 0, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"aggregate\\": true}, \\"n\\": \\"10000000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"4186778f-e4dd-4253-afca-28da14e20bd8","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.34.0+galaxy0","errors":null,"id":7,"input_connections":{"paired_cond|reads":{"id":5,"output_name":"output"},"seprev_cond|trim|truncLen":{"id":2,"output_name":"output"},"trim|truncLen":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool dada2: filterAndTrim","name":"paired_cond"},{"description":"runtime parameter for tool dada2: filterAndTrim","name":"trim"}],"label":null,"name":"dada2: filterAndTrim","outputs":[{"name":"paired_output","type":"input"},{"name":"outtab","type":"tabular"}],"position":{"left":560,"top":233},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"00e0347751b1","name":"dada2_filterandtrim","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter\\": {\\"maxLen\\": null, \\"minLen\\": \\"20\\", \\"maxN\\": \\"0\\", \\"minQ\\": \\"0\\", \\"maxEE\\": \\"2\\"}, \\"orientFwd\\": \\"\\", \\"output_statistics\\": true, \\"paired_cond\\": {\\"paired_select\\": \\"paired\\", \\"__current_case__\\": 0, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rmPhiX\\": true, \\"rmlowcomplex\\": \\"0\\", \\"seprev_cond\\": {\\"seprev_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"trim\\": {\\"truncQ\\": \\"2\\", \\"trimLeft\\": \\"0\\", \\"trimRight\\": \\"0\\", \\"truncLen\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filter\\": {\\"maxLen\\": null, \\"minLen\\": \\"20\\", \\"maxN\\": \\"0\\", \\"minQ\\": \\"0\\", \\"maxEE\\": \\"2\\"}}, \\"trim\\": {\\"truncQ\\": \\"2\\", \\"trimLeft\\": \\"0\\", \\"trimRight\\": \\"0\\", \\"truncLen\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"ff8fd36b-517c-4ddb-8f18-22dcc068bb76","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.34.0+galaxy0","errors":null,"id":8,"input_connections":{"batch_cond|paired_cond|reads":{"id":7,"output_name":"paired_output"}},"inputs":[],"label":"QualityProfile after filterAndTrim","name":"dada2: plotQualityProfile","outputs":[{"name":"output_fwd","type":"pdf"},{"name":"output_rev","type":"pdf"}],"position":{"left":840,"top":263},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"4e469b758915","name":"dada2_plotqualityprofile","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_cond\\": {\\"batch_select\\": \\"joint\\", \\"__current_case__\\": 0, \\"paired_cond\\": {\\"paired_select\\": \\"paired\\", \\"__current_case__\\": 0, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"aggregate\\": true}, \\"n\\": \\"1000000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"9e4e40e4-7ea5-4ccf-99c6-cd377fe35dbd","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__UNZIP_COLLECTION__","errors":null,"id":9,"input_connections":{"input":{"id":7,"output_name":"paired_output"}},"inputs":[],"label":null,"name":"Unzip collection","outputs":[{"name":"forward","type":"input"},{"name":"reverse","type":"input"}],"position":{"left":840,"top":491},"post_job_actions":{"TagDatasetActionforward":{"action_arguments":{"tags":"#forward"},"action_type":"TagDatasetAction","output_name":"forward"},"TagDatasetActionreverse":{"action_arguments":{"tags":"#reverse"},"action_type":"TagDatasetAction","output_name":"reverse"}},"tool_id":"__UNZIP_COLLECTION__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"c0f0013f-b065-46dc-9eb7-15a89b27ab14","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.34.0+galaxy0","errors":null,"id":10,"input_connections":{"fls":{"id":9,"output_name":"forward"}},"inputs":[],"label":null,"name":"dada2: learnErrors","outputs":[{"name":"errors","type":"dada2_errorrates"},{"name":"plot","type":"pdf"}],"position":{"left":1120,"top":362},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"a14a18a47c36","name":"dada2_learnerrors","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced\\": {\\"errfoo\\": \\"loessErrfun\\", \\"randomize\\": false, \\"maxconsist\\": \\"10\\", \\"omegac\\": \\"0.0\\"}, \\"chromInfo\\": \\"/gpfs1/data/galaxy_server/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"fls\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nbases\\": \\"8\\", \\"plotopt\\": {\\"obs\\": true, \\"err_out\\": true, \\"err_in\\": false, \\"nominalQ\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"725e57a5-35c9-4b20-860a-40c91ce4a6b8","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.34.0+galaxy0","errors":null,"id":11,"input_connections":{"fls":{"id":9,"output_name":"reverse"}},"inputs":[],"label":null,"name":"dada2: learnErrors","outputs":[{"name":"errors","type":"dada2_errorrates"},{"name":"plot","type":"pdf"}],"position":{"left":1113,"top":716.3237599999942},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"a14a18a47c36","name":"dada2_learnerrors","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced\\": {\\"errfoo\\": \\"loessErrfun\\", \\"randomize\\": false, \\"maxconsist\\": \\"10\\", \\"omegac\\": \\"0.0\\"}, \\"chromInfo\\": \\"/gpfs1/data/galaxy_server/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"fls\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nbases\\": \\"8\\", \\"plotopt\\": {\\"obs\\": true, \\"err_out\\": true, \\"err_in\\": false, \\"nominalQ\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"53cb573f-65bd-4226-ad69-bc03a8f71961","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.34.0+galaxy0","errors":null,"id":12,"input_connections":{"batch_cond|derep":{"id":9,"output_name":"forward"},"batch_cond|pool":{"id":3,"output_name":"output"},"err":{"id":10,"output_name":"errors"}},"inputs":[{"description":"runtime parameter for tool dada2: dada","name":"batch_cond"},{"description":"runtime parameter for tool dada2: dada","name":"batch_cond"}],"label":null,"name":"dada2: dada","outputs":[{"name":"data_collection","type":"input"}],"position":{"left":1400,"top":406},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"b35a5e8ff0fe","name":"dada2_dada","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_cond\\": {\\"batch_select\\": \\"no\\", \\"__current_case__\\": 1, \\"derep\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pool\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"err\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"d4bf96a4-79bf-45dc-87d2-ccbd1eba28bd","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.34.0+galaxy0","errors":null,"id":13,"input_connections":{"batch_cond|derep":{"id":9,"output_name":"reverse"},"batch_cond|pool":{"id":3,"output_name":"output"},"err":{"id":11,"output_name":"errors"}},"inputs":[{"description":"runtime parameter for tool dada2: dada","name":"batch_cond"},{"description":"runtime parameter for tool dada2: dada","name":"batch_cond"}],"label":null,"name":"dada2: dada","outputs":[{"name":"data_collection","type":"input"}],"position":{"left":1390,"top":715.0071096093692},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"b35a5e8ff0fe","name":"dada2_dada","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_cond\\": {\\"batch_select\\": \\"no\\", \\"__current_case__\\": 1, \\"derep\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pool\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"err\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"0d8aff43-250d-4a9b-9c35-34589d02b3ec","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.34.0+galaxy0","errors":null,"id":14,"input_connections":{"dadaF":{"id":12,"output_name":"data_collection"},"dadaR":{"id":13,"output_name":"data_collection"},"derepF":{"id":9,"output_name":"forward"},"derepR":{"id":9,"output_name":"reverse"}},"inputs":[],"label":null,"name":"dada2: mergePairs","outputs":[{"name":"merged","type":"dada2_mergepairs"}],"position":{"left":1683,"top":468.0071096093692},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"646c58820267","name":"dada2_mergepairs","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chromInfo\\": \\"/gpfs1/data/galaxy_server/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"dadaF\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"dadaF|__identifier__\\": \\"pool.feces.fastq\\", \\"dadaR\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"dadaR|__identifier__\\": \\"pool.feces.fastq\\", \\"derepF\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"derepF|__identifier__\\": \\"pool.feces.fastq\\", \\"derepR\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"derepR|__identifier__\\": \\"pool.feces.fastq\\", \\"justConcatenate\\": false, \\"maxMismatch\\": \\"0\\", \\"minOverlap\\": \\"12\\", \\"output_details\\": false, \\"trimOverhang\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"574508a6-3b02-40e0-8d54-9423e6ddd852","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.34.0+galaxy0","errors":null,"id":15,"input_connections":{"samples":{"id":14,"output_name":"merged"}},"inputs":[],"label":null,"name":"dada2: makeSequenceTable","outputs":[{"name":"stable","type":"dada2_sequencetable"},{"name":"plot_output","type":"pdf"}],"position":{"left":1960.0001199993603,"top":521.0070943505801},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"ebd632e95aae","name":"dada2_makesequencetable","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_cond\\": {\\"filter_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"orderBy\\": \\"abundance\\", \\"plot\\": true, \\"samples\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"6b6b4d9b-573f-4eee-a406-51930e761f28","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.34.0+galaxy0","errors":null,"id":16,"input_connections":{"unqs":{"id":15,"output_name":"stable"}},"inputs":[],"label":null,"name":"dada2: removeBimeraDenovo","outputs":[{"name":"stable_uniques","type":"dada2_uniques"},{"name":"stable_mergepairs","type":"dada2_mergepairs"},{"name":"stable_sequencetable","type":"dada2_sequencetable"}],"position":{"left":2240,"top":554},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"efe26de231ec","name":"dada2_removebimeradenovo","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"method\\": \\"consensus\\", \\"unqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"76a3ebb9-4d87-470d-aea6-abb90184c7ec","when":null,"workflow_outputs":[{"label":"Sequence Table","output_name":"stable_sequencetable","uuid":"d8690ce6-0468-4b59-a824-bcb34218ae87"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.34.0+galaxy0","errors":null,"id":17,"input_connections":{"inrep_0|input":{"id":7,"output_name":"outtab"},"inrep_1|input":{"id":12,"output_name":"data_collection"},"inrep_2|input":{"id":13,"output_name":"data_collection"},"inrep_3|input":{"id":14,"output_name":"merged"},"inrep_4|input":{"id":15,"output_name":"stable"},"inrep_5|input":{"id":16,"output_name":"stable_sequencetable"}},"inputs":[],"label":null,"name":"dada2: sequence counts","outputs":[{"name":"counts","type":"tabular"}],"position":{"left":2523.672846294445,"top":53.76299803013818},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"390bc2645138","name":"dada2_seqcounts","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inrep\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"FiltTrim\\"}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"Dada Fwd\\"}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"Dada Rev\\"}, {\\"__index__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"MakePairs\\"}, {\\"__index__\\": 4, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"SeqTab\\"}, {\\"__index__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"Bimera\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"bd73b51d-5c84-4de0-94c7-453879379bd8","when":null,"workflow_outputs":[{"label":"Counts","output_name":"counts","uuid":"f5ba8eba-84da-4a58-9bba-5d016c7dcc40"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.34.0+galaxy0","errors":null,"id":18,"input_connections":{"reference_cond|refFasta":{"id":4,"output_name":"output"},"seqs":{"id":16,"output_name":"stable_sequencetable"}},"inputs":[{"description":"runtime parameter for tool dada2: assignTaxonomy and addSpecies","name":"reference_cond"}],"label":"dada2: assignTaxonomy","name":"dada2: assignTaxonomy and addSpecies","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2647.6612392582533,"top":838.3432775089392},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.34.0+galaxy0","tool_shed_repository":{"changeset_revision":"705d26fbea02","name":"dada2_assigntaxonomyaddspecies","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addSpecies_cond\\": {\\"addSpecies_select\\": \\"FALSE\\", \\"__current_case__\\": 0}, \\"minBoot\\": \\"50\\", \\"outputBootstraps\\": false, \\"reference_cond\\": {\\"reference_select\\": \\"builtin\\", \\"__current_case__\\": 0, \\"refFasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"seed\\": null, \\"seqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tryRC\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.34.0+galaxy0","type":"tool","uuid":"7f71f23c-bcb2-4c6e-a716-33964fd602c4","when":null,"workflow_outputs":[{"label":"Taxonomy","output_name":"output","uuid":"f0bc83d0-6d2f-4e78-98f7-c4ce06365c23"}]}},"tags":["name:amplicon"],"uuid":"11d717f8-92ef-4dd7-bda7-c193175fece2","version":1},"readme":"# Dada2: amplicon analysis for paired end data\\n\\n## Inputs dataset\\n\\n- `Paired input data` paired input collection in FASTQ format\\n\\n## Inputs values\\n\\n- `Read length forward/reverse reads` length of the forward/reverse reads to which they should be truncated in the filter and trim step\\n- `Pool samples` pooling may increase sensitivity\\n- `Reference database` that should be used for taxonomic assignment\\n\\n## Processing\\n\\nThe workflow follows the steps described in the [dada2 tutorial](https://benjjneb.github.io/dada2/tutorial.html).\\n\\nAs a first step the input collection is sorted. This is important because the dada2 step outputs\\na collection in sorted order. If the input collection would not be sorted then the mergePairs step\\nsamples would be mixed up.\\n\\n- `FilterAndTrim` Quality control by filtering and trimming reads\\n- `QualityProfile` is called before and after the FilterAndTrim step\\n- `Unzip Collection` separates forward and reverse reads (the next steps are evaluated separately on forward and reverse reads)\\n- `learnErrors` learn error rates\\n- `dada` filter noisy reads\\n- `mergePairs` merge forward and reverse reads\\n- `makeSequenceTable` create the sequence table\\n- `removeBimeraDenovo` remove chimeric sequencs\\n- `assignTaxonomy` assign taxonomic information from a reference data base\\n\\n## TODO\\n\\nSome possibilities to extend/improve the workflow\\n\\n- output BIOM\\n- use ASV1, ... in sequence table and taxonomy output, and output additional fasta\\n- allow to use custom taxonomy / make it optional\\n","changelog":"# Changelog\\n\\n## [0.3] - 2025-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.34.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.30.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.34.0+galaxy0`\\n\\n## [0.2] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_plotqualityprofile/dada2_plotQualityProfile/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_filterandtrim/dada2_filterAndTrim/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_learnerrors/dada2_learnErrors/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_dada/dada2_dada/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_mergepairs/dada2_mergePairs/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_makesequencetable/dada2_makeSequenceTable/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_removebimeradenovo/dada2_removeBimeraDenovo/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_seqcounts/dada2_seqCounts/1.30.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/dada2_assigntaxonomyaddspecies/dada2_assignTaxonomyAddspecies/1.30.0+galaxy0`\\n\\n## [0.1] - 2024-01-09\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## dada2 amplicon analysis pipeline - for paired end data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired input data\\"]@{ shape: docs }\\n1[\\"ℹ️ Read length forward read\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Read length reverse read\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Pool samples\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Cached reference database\\"]@{ shape: lean-l }\\n5[\\"Sort samples\\"]@{ shape: process }\\n0 --> 5\\n6[\\"QualityProfile before filterAndTrim\\"]@{ shape: process }\\n5 --> 6\\n7[\\"dada2: filterAndTrim\\"]@{ shape: process }\\n5 --> 7\\n2 --> 7\\n1 --> 7\\n8[\\"QualityProfile after filterAndTrim\\"]@{ shape: process }\\n7 --> 8\\n9[\\"Unzip collection\\"]@{ shape: process }\\n7 --> 9\\n10[\\"dada2: learnErrors\\"]@{ shape: process }\\n9 --> 10\\n11[\\"dada2: learnErrors\\"]@{ shape: process }\\n9 --> 11\\n12[\\"dada2: dada\\"]@{ shape: process }\\n9 --> 12\\n3 --> 12\\n10 --> 12\\n13[\\"dada2: dada\\"]@{ shape: process }\\n9 --> 13\\n3 --> 13\\n11 --> 13\\n14[\\"dada2: mergePairs\\"]@{ shape: process }\\n12 --> 14\\n13 --> 14\\n9 --> 14\\n9 --> 14\\n15[\\"dada2: makeSequenceTable\\"]@{ shape: process }\\n14 --> 15\\n16[\\"dada2: removeBimeraDenovo\\"]@{ shape: process }\\n15 --> 16\\n17[\\"dada2: sequence counts\\"]@{ shape: process }\\n7 --> 17\\n12 --> 17\\n13 --> 17\\n14 --> 17\\n15 --> 17\\n16 --> 17\\n18[\\"dada2: assignTaxonomy\\"]@{ shape: process }\\n4 --> 18\\n16 --> 18\\n```\\n","updated":"2025-02-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/dada2/main","dockstore_id":25853,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.15078505","tests":[{"doc":"Test outline for dada---paired","job":{"Paired input data":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"F3D0","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/800651/files/F3D0_R1.fastq"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/800651/files/F3D0_R2.fastq"}]},{"class":"Collection","type":"paired","identifier":"F3D5","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/800651/files/F3D5_R1.fastq"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/800651/files/F3D5_R2.fastq"}]},{"class":"Collection","type":"paired","identifier":"F3D145","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/800651/files/F3D145_R1.fastq"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/800651/files/F3D145_R2.fastq"}]},{"class":"Collection","type":"paired","identifier":"F3D150","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/800651/files/F3D150_R1.fastq"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/800651/files/F3D150_R2.fastq"}]},{"class":"Collection","type":"paired","identifier":"Mock","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/800651/files/Mock_R1.fastq"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/800651/files/Mock_R2.fastq"}]}]},"Read length forward read":240,"Read length reverse read":160,"Pool samples":"FALSE","Cached reference database":"silva_132"},"outputs":{"Sequence Table":{"path":"test-data/Sequence Table.dada2_sequencetable","asserts":[{"has_n_columns":{"n":6}},{"has_n_lines":{"n":171}}]},"Counts":{"path":"test-data/Counts.tabular","asserts":[{"has_n_columns":{"n":8}},{"has_n_lines":{"n":6}}]},"Taxonomy":{"ftype":"tabular","sorted":true,"asserts":[{"has_text":{"text":"Firmicutes","n":131}},{"has_n_columns":{"n":7}},{"has_n_lines":{"n":171}}]}}}]}],"path":"./workflows/amplicon/dada2"},{"version":1.2,"workflows":[{"name":"IIa-denoising-se","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-IIa-denoising-and-feature-table-creation-single-end.ga","testParameterFiles":["/QIIME2-IIa-denoising-and-feature-table-creation-single-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","release":"0.3","license":"MIT","name":"QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)","steps":{"0":{"annotation":"Tab separated metadata file","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":0,"top":140.20380466285837},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8dde0c69-40f1-415a-a866-5d1115e80cdf","when":null,"workflow_outputs":[]},"1":{"annotation":"Demultiplexed sequences in qza format","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Demultiplexed sequences in qza format","name":"Demultiplexed sequences"}],"label":"Demultiplexed sequences","name":"Input dataset","outputs":[],"position":{"left":0.7889777582004348,"top":256.1421426050462},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e250ddc1-f567-40ca-9f2b-e07bbf84cdec","when":null,"workflow_outputs":[]},"2":{"annotation":"Length to which the sequence sould be truncated. Will remove bases from the 3\' end.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Length to which the sequence sould be truncated. Will remove bases from the 3\' end.","name":"Truncation length"}],"label":"Truncation length","name":"Input parameter","outputs":[],"position":{"left":1.1220203717959043,"top":387.6692612809613},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"02f3a9bf-0e92-4682-b6b3-ab6b5a332937","when":null,"workflow_outputs":[]},"3":{"annotation":"Number of bases at the 5\' end that should be removed. Default: 0.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Number of bases at the 5\' end that should be removed. Default: 0.","name":"Trimming length"}],"label":"Trimming length","name":"Input parameter","outputs":[],"position":{"left":1.464599609375,"top":519.0800476074219},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"8d947f65-cb01-4477-b833-4e446f2a92d5","when":null,"workflow_outputs":[]},"4":{"annotation":"qiime2 dada2 denoise-paired\\nDenoise and dereplicate paired-end sequences\\nSmall insight: if the denoising percentage is lower or poorer, reduce \'trunc_q: Int\' from 2 to 1.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":4,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|trim_left":{"id":3,"output_name":"output"},"demultiplexed_seqs":{"id":1,"output_name":"output"},"trunc_len":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool qiime2 dada2 denoise-single","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Denoising the datasets","name":"qiime2 dada2 denoise-single","outputs":[{"name":"table","type":"qza"},{"name":"representative_sequences","type":"qza"},{"name":"denoising_stats","type":"qza"}],"position":{"left":336.80960206844844,"top":133.89925462837766},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"354f63459d37","name":"qiime2__dada2__denoise_single","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"trim_left\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_ee\\": \\"2.0\\", \\"trunc_q\\": \\"2\\", \\"pooling_method\\": \\"independent\\", \\"chimera_method\\": \\"consensus\\", \\"min_fold_parent_over_abundance\\": \\"1.0\\", \\"allow_one_off\\": false, \\"n_reads_learn\\": \\"1000000\\", \\"hashed_feature_ids\\": true, \\"retain_all_samples\\": true}, \\"demultiplexed_seqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trunc_len\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"a6369693-372e-4258-9e81-b7085746e441","when":null,"workflow_outputs":[{"label":"Denoising output table","output_name":"table","uuid":"5be21ff5-eb12-442b-85c3-932ed240cb2c"},{"label":"Representative denoised sequences","output_name":"representative_sequences","uuid":"2c8b68a1-ffb9-4a6c-aa28-336f770acd5e"},{"label":"Denoising statistics","output_name":"denoising_stats","uuid":"2bba5561-d09b-4843-a988-465b54a07022"}]},"5":{"annotation":"qiime2 feature-table tabulate-seqs\\nView sequence associated with each feature","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":5,"input_connections":{"data":{"id":4,"output_name":"representative_sequences"}},"inputs":[{"description":"runtime parameter for tool qiime2 feature-table tabulate-seqs","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Tabulate DADA2 denoised representative sequences","name":"qiime2 feature-table tabulate-seqs","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":690.0833740234375,"top":0},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_rep_seq_visualization"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"c4570ced1d88","name":"qiime2__feature_table__tabulate_seqs","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"taxonomy\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"metadata\\": [], \\"merge_method\\": \\"strict\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"221fe314-49e9-48bd-979e-1b6d0c2f4192","when":null,"workflow_outputs":[{"label":"DADA2 representative sequences list visualisation","output_name":"visualization","uuid":"6e7fb017-526c-4467-8082-501ffc3cb6c3"}]},"6":{"annotation":"qiime2 metadata tabulate\\nInteractively explore Metadata in a table","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":6,"input_connections":{"input_0|__q2galaxy__GUI__conditional__input__|source":{"id":4,"output_name":"denoising_stats"}},"inputs":[],"label":"Tabulate DADA2 statistical metadata ","name":"qiime2 metadata tabulate","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":690.0390625,"top":249.63809204101562},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_stats_visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"9857be21dcaf","name":"qiime2__metadata__tabulate","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"page_size\\": \\"100\\"}, \\"input\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__input__\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"082c2df8-422c-4661-b39d-181ef7ff3072","when":null,"workflow_outputs":[{"label":"DADA2 statistics visualisation","output_name":"visualization","uuid":"1118ba34-2f79-4385-adfb-adbe6821479e"}]},"7":{"annotation":"qiime2 feature-table summarize\\nSummarize table\\n","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":7,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|sample_metadata_0|__q2galaxy__GUI__conditional__sample_metadata__|source":{"id":0,"output_name":"output"},"table":{"id":4,"output_name":"table"}},"inputs":[],"label":"Summing up the dada2 output table","name":"qiime2 feature-table summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":692.4243820904538,"top":479.7321793212501},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_table_visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"d0f2249afe30","name":"qiime2__feature_table__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"sample_metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__sample_metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"f6376403-1be3-4bc0-af85-d5e2e04984b2","when":null,"workflow_outputs":[{"label":"DADA2 output table","output_name":"visualization","uuid":"13960a03-ee65-4385-8936-b7ff5e5a98f3"}]}},"tags":[],"uuid":"718df5b9-9cbb-4792-be50-604e18ef0548","version":2},"readme":"# QIIME2 workflows\\n\\n## Available workflows\\n\\nDenoising (using `qiime2`\'s `dada2` integration for paired / single end data.\\n\\n## Inputs\\n\\n- Demultiplexed sequences as a qiime2 aertifact file (`qza`) containing the sequence information.\\n- Metadata table (`tabular`)\\n- Truncation length\\n- Trimming length (optional)\\n\\nFor the paired end workflow the truncation and trimming length for the reverse reads can / has to be given.\\n\\n\\n## Processing\\n\\n- Denoising with `qiime2 dada2 denoise-single`/`paired`\\n- For each of the three outputs (see below) another tool is started to prepare a corresponding qzv file\\n  - representative sequences `qiime2 feature-table tabulate-seqs `\\n  - denoising statistics `qiime2 metadata tabulate`\\n  - summary of the feature table\\n\\n## Outputs\\n\\n  - representative sequences \\n  - denoising statistics \\n  - summary of the feature table (how many sequences are lost in the corresponding steps)\\n","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 IIa: Denoising (sequence quality control) and feature table creation (single-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n1[\\"ℹ️ Demultiplexed sequences\\"]@{ shape: doc }\\n2[\\"ℹ️ Truncation length\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Trimming length\\"]@{ shape: lean-l }\\n4[\\"Denoising the datasets\\"]@{ shape: process }\\n3 --> 4\\n1 --> 4\\n2 --> 4\\n5[\\"Tabulate DADA2 denoised representative sequences\\"]@{ shape: process }\\n4 --> 5\\n6[\\"Tabulate DADA2 statistical metadata \\"]@{ shape: process }\\n4 --> 6\\n7[\\"Summing up the dada2 output table\\"]@{ shape: process }\\n0 --> 7\\n4 --> 7\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIa-denoising-se","dockstore_id":26898,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244073","tests":[{"doc":"Test outline for QIIME2-IIa-denoising--and-feature-table-creation-single-end","job":{"Metadata":{"class":"File","location":"https://docs.qiime2.org/2021.11/data/tutorials/moving-pictures-usage/sample-metadata.tsv","filetype":"tabular"},"Demultiplexed sequences":{"class":"File","location":"https://docs.qiime2.org/2021.11/data/tutorials/moving-pictures-usage/demux.qza","filetype":"qza"},"Truncation length":120,"Trimming length":0},"outputs":{"Denoising output table":{"asserts":[{"has_size":{"min":"20k"}},{"has_archive_member":{"path":"^[^/]*/data/feature-table.biom","n":1}}]},"Representative denoised sequences":{"asserts":[{"has_size":{"min":"20k"}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1}},{"has_archive_member":{"path":"^[^/]*/data/dna-sequences.fasta","n":1,"asserts":[{"has_text_matching":{"expression":">.*","n":770}}]}}]},"Denoising statistics":{"asserts":[{"has_size":{"min":0}},{"has_size":{"min":"10k"}},{"has_archive_member":{"path":"^[^/]*/data/stats.tsv","n":1,"asserts":[{"has_n_columns":{"n":7}},{"has_n_lines":{"n":36}}]}}]}}}]},{"name":"IIb-denoising-pe","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-IIb-denoising-and-feature-table-creation-paired-end.ga","testParameterFiles":["/QIIME2-IIb-denoising-and-feature-table-creation-paired-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Use DADA2 for sequence quality control. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the q2-dada2 plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","release":"0.3","license":"MIT","name":"QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)","steps":{"0":{"annotation":"Tab separated metadata file","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":1.6647546421401653,"top":139.9560324460615},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8dde0c69-40f1-415a-a866-5d1115e80cdf","when":null,"workflow_outputs":[]},"1":{"annotation":"Demultiplexed sequences in qza format","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Demultiplexed sequences in qza format","name":"Demultiplexed sequences"}],"label":"Demultiplexed sequences","name":"Input dataset","outputs":[],"position":{"left":2.4506271621231845,"top":238.07957958589887},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e250ddc1-f567-40ca-9f2b-e07bbf84cdec","when":null,"workflow_outputs":[]},"2":{"annotation":"Length to which the forward read sequence should be truncated. Will remove bases from the 3\' end.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Length to which the forward read sequence should be truncated. Will remove bases from the 3\' end.","name":"Truncation length (forward)"}],"label":"Truncation length (forward)","name":"Input parameter","outputs":[],"position":{"left":1.5784534434758744,"top":353.55220274941445},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"02f3a9bf-0e92-4682-b6b3-ab6b5a332937","when":null,"workflow_outputs":[]},"3":{"annotation":"Length to which the reverse read sequence should be truncated. Will remove bases from the 3\' end.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Length to which the reverse read sequence should be truncated. Will remove bases from the 3\' end.","name":"Truncation length (reverse)"}],"label":"Truncation length (reverse)","name":"Input parameter","outputs":[],"position":{"left":0.12398288926847933,"top":478.1294436628793},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"2af02538-19c9-4ffc-924e-5108987064a1","when":null,"workflow_outputs":[]},"4":{"annotation":"Number of bases at the 5\' end of the forward read that should be removed. Default: 0.\\n\\n\\nMention the length up to which you want to trim your sequences from the left.\\nIf not needed, put 0.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Number of bases at the 5\' end of the forward read that should be removed. Default: 0.\\n\\n\\nMention the length up to which you want to trim your sequences from the left.\\nIf not needed, put 0.","name":"Trimming length (forward)"}],"label":"Trimming length (forward)","name":"Input parameter","outputs":[],"position":{"left":0,"top":592.1359047583886},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"cf76e446-d3c1-4682-820c-f6a2f76343d1","when":null,"workflow_outputs":[]},"5":{"annotation":"Number of bases at the 5\' end of the reverse read that should be removed. Default: 0.\\n","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Number of bases at the 5\' end of the reverse read that should be removed. Default: 0.\\n","name":"Trimming length (reverse)"}],"label":"Trimming length (reverse)","name":"Input parameter","outputs":[],"position":{"left":0.14257701997123728,"top":708.0834213595151},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"ccc4f7a3-5e34-4933-ab8b-167e8c944aa7","when":null,"workflow_outputs":[]},"6":{"annotation":"qiime2 dada2 denoise-paired\\nDenoise and dereplicate paired-end sequences\\nSmall insight: if the denoising percentage is lower or poorer, reduce \'trunc_q: Int\' from 2 to 1.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":6,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|trim_left_f":{"id":4,"output_name":"output"},"__q2galaxy__GUI__section__extra_opts__|trim_left_r":{"id":5,"output_name":"output"},"demultiplexed_seqs":{"id":1,"output_name":"output"},"trunc_len_f":{"id":2,"output_name":"output"},"trunc_len_r":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool qiime2 dada2 denoise-paired","name":"__q2galaxy__GUI__section__extra_opts__"},{"description":"runtime parameter for tool qiime2 dada2 denoise-paired","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Denoising the datasets","name":"qiime2 dada2 denoise-paired","outputs":[{"name":"table","type":"qza"},{"name":"representative_sequences","type":"qza"},{"name":"denoising_stats","type":"qza"}],"position":{"left":353.695690215759,"top":186.09945678710938},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"933b92011b91","name":"qiime2__dada2__denoise_paired","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"trim_left_f\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trim_left_r\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_ee_f\\": \\"2.0\\", \\"max_ee_r\\": \\"2.0\\", \\"trunc_q\\": \\"2\\", \\"min_overlap\\": \\"12\\", \\"pooling_method\\": \\"independent\\", \\"chimera_method\\": \\"consensus\\", \\"min_fold_parent_over_abundance\\": \\"1.0\\", \\"allow_one_off\\": false, \\"n_reads_learn\\": \\"1000000\\", \\"hashed_feature_ids\\": true, \\"retain_all_samples\\": true}, \\"demultiplexed_seqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trunc_len_f\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trunc_len_r\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"93e5c884-d782-4b97-96ba-d9abdf4ce4ed","when":null,"workflow_outputs":[{"label":"Denoising statistics","output_name":"denoising_stats","uuid":"971dcd45-80de-44e3-a0f0-37a24d87c89b"},{"label":"Denoising output table","output_name":"table","uuid":"cbcb0a29-a64f-46ff-8018-dea44208d8c8"},{"label":"Representative denoised sequences","output_name":"representative_sequences","uuid":"299bee3a-7747-4a0e-957d-2fb16bbbc277"}]},"7":{"annotation":"qiime2 feature-table tabulate-seqs\\nView sequence associated with each feature","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":7,"input_connections":{"data":{"id":6,"output_name":"representative_sequences"}},"inputs":[{"description":"runtime parameter for tool qiime2 feature-table tabulate-seqs","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Tabulate DADA2 denoised representative sequences","name":"qiime2 feature-table tabulate-seqs","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":691.7523737176135,"top":0},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_rep_seq_visualization"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"c4570ced1d88","name":"qiime2__feature_table__tabulate_seqs","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"taxonomy\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"metadata\\": [], \\"merge_method\\": \\"strict\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"221fe314-49e9-48bd-979e-1b6d0c2f4192","when":null,"workflow_outputs":[{"label":"DADA2 representative sequences list visualisation","output_name":"visualization","uuid":"90ca4cf9-cd24-4a88-a1aa-bd3698387f92"}]},"8":{"annotation":"qiime2 metadata tabulate\\nInteractively explore Metadata in an HTML table","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":8,"input_connections":{"input_0|__q2galaxy__GUI__conditional__input__|source":{"id":6,"output_name":"denoising_stats"}},"inputs":[],"label":"Tabulate DADA2 statistical metadata ","name":"qiime2 metadata tabulate","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":691.5803874570937,"top":249.50780313619012},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_stats_visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"9857be21dcaf","name":"qiime2__metadata__tabulate","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"page_size\\": \\"100\\"}, \\"input\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__input__\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"082c2df8-422c-4661-b39d-181ef7ff3072","when":null,"workflow_outputs":[{"label":"DADA2 statistics visualisation","output_name":"visualization","uuid":"4501c38f-a3ad-4873-82a7-036b3314a0b4"}]},"9":{"annotation":"qiime2 feature-table summarize\\nSummarize table\\n","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":9,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|sample_metadata_0|__q2galaxy__GUI__conditional__sample_metadata__|source":{"id":0,"output_name":"output"},"table":{"id":6,"output_name":"table"}},"inputs":[],"label":"Summing up the dada2 output table","name":"qiime2 feature-table summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":693.96337890625,"top":479.484375},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"dada2_table_visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"d0f2249afe30","name":"qiime2__feature_table__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"sample_metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__sample_metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"f6376403-1be3-4bc0-af85-d5e2e04984b2","when":null,"workflow_outputs":[{"label":"DADA2 output table visualisation","output_name":"visualization","uuid":"bf6baadb-1272-48b1-95b6-9aeeb8ee9e1a"}]}},"tags":[],"uuid":"8b8c4f4c-7cc0-422d-9a6f-47357faa0082","version":1},"readme":"# QIIME2 workflows\\n\\n## Available workflows\\n\\nDenoising (using `qiime2`\'s `dada2` integration for paired / single end data.\\n\\n## Inputs\\n\\n- Demultiplexed sequences as a qiime2 aertifact file (`qza`) containing the sequence information.\\n- Metadata table (`tabular`)\\n- Truncation length\\n- Trimming length (optional)\\n\\nFor the paired end workflow the truncation and trimming length for the reverse reads can / has to be given.\\n\\n\\n## Processing\\n\\n- Denoising with `qiime2 dada2 denoise-single`/`paired`\\n- For each of the three outputs (see below) another tool is started to prepare a corresponding qzv file\\n  - representative sequences `qiime2 feature-table tabulate-seqs `\\n  - denoising statistics `qiime2 metadata tabulate`\\n  - summary of the feature table\\n\\n## Outputs\\n\\n  - representative sequences \\n  - denoising statistics \\n  - summary of the feature table (how many sequences are lost in the corresponding steps)\\n","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_single/qiime2__dada2__denoise_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__tabulate_seqs/qiime2__feature_table__tabulate_seqs/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_table__summarize/qiime2__feature_table__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__dada2__denoise_paired/qiime2__dada2__denoise_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 IIb: Denoising (sequence quality control) and feature table creation (paired-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n1[\\"ℹ️ Demultiplexed sequences\\"]@{ shape: doc }\\n2[\\"ℹ️ Truncation length (forward)\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Truncation length (reverse)\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Trimming length (forward)\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Trimming length (reverse)\\"]@{ shape: lean-l }\\n6[\\"Denoising the datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n1 --> 6\\n2 --> 6\\n3 --> 6\\n7[\\"Tabulate DADA2 denoised representative sequences\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Tabulate DADA2 statistical metadata \\"]@{ shape: process }\\n6 --> 8\\n9[\\"Summing up the dada2 output table\\"]@{ shape: process }\\n0 --> 9\\n6 --> 9\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-II-denoising/IIb-denoising-pe","dockstore_id":26899,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244079","tests":[{"doc":"Test outline for QIIME2-IIb-and-feature-table-creation-paired-end","job":{"Metadata":{"class":"File","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/020-metadata/sample-metadata.tsv","filetype":"tabular"},"Demultiplexed sequences":{"class":"File","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/demultiplexed-sequences.qza","filetype":"qza"},"Truncation length (forward)":204,"Truncation length (reverse)":205,"Trimming length (forward)":1,"Trimming length (reverse)":1},"outputs":{"Denoising output table":{"asserts":[{"has_size":{"min":"20k"}},{"has_archive_member":{"path":"^[^/]*/data/feature-table.biom","n":1}}]},"Representative denoised sequences":{"asserts":[{"has_size":{"min":"20k"}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1}},{"has_archive_member":{"path":"^[^/]*/data/dna-sequences.fasta","n":1,"asserts":[{"has_text_matching":{"expression":">.*","n":661}}]}}]},"Denoising statistics":{"asserts":[{"has_size":{"min":"10k"}},{"has_archive_member":{"path":"^[^/]*/data/stats.tsv","n":1,"asserts":[{"has_n_columns":{"n":9}},{"has_n_lines":{"n":43}}]}}]}}}]}],"path":"./workflows/amplicon/qiime2/qiime2-II-denoising"},{"version":1.2,"workflows":[{"name":"Ia-import-multiplexed-se","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-Ia-multiplexed-data-single-end.ga","testParameterFiles":["/QIIME2-Ia-multiplexed-data-single-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Importing single-end multiplexed data (not demultiplexed yet)","creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"QIIME2 Ia: multiplexed data (single-end)","steps":{"0":{"annotation":"Single-end sequences","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Single-end sequences","name":"Sequences"}],"label":"Sequences","name":"Input dataset","outputs":[],"position":{"left":2.349740917339593,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"2789d643-0609-420d-848b-cc63959f796c","when":null,"workflow_outputs":[]},"1":{"annotation":"Barcode sequences","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Barcode sequences","name":"Barcodes"}],"label":"Barcodes","name":"Input dataset","outputs":[],"position":{"left":0.5855346656953526,"top":97.72533948535487},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"a4cddcc4-c1c9-4ae2-a726-762b01b46cbf","when":null,"workflow_outputs":[]},"2":{"annotation":"A tab separated metadata file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"A tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":0.5183062535642966,"top":204.0675768088665},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c8103f12-e7ab-4850-bd54-a2e7fffefdb1","when":null,"workflow_outputs":[]},"3":{"annotation":"Column name of the metadata to be used for demultiplexing","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Column name of the metadata to be used for demultiplexing","name":"Metadata parameter"}],"label":"Metadata parameter","name":"Input parameter","outputs":[],"position":{"left":0,"top":310.7679254201888},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"336f8551-d936-4c2c-ad92-323b2ffd2c68","when":null,"workflow_outputs":[]},"4":{"annotation":"If reverse complement of barcodes is necessary for the analysis, choose Yes.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"If reverse complement of barcodes is necessary for the analysis, choose Yes.","name":"Reverse complement barcodes"}],"label":"Reverse complement barcodes","name":"Input parameter","outputs":[],"position":{"left":2.073822021484375,"top":430.05810546875},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"75545228-5faf-4a46-b5a2-9ca52a85e86b","when":null,"workflow_outputs":[]},"5":{"annotation":"qiime2 tools input\\nThis step allows the input of the files to be analysed and then generate a qza output, necessary for the QIIME2 pipeline.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":5,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import_barcodes|data":{"id":1,"output_name":"output"},"import_root|__q2galaxy__GUI__cond__format__|import_sequences|data":{"id":0,"output_name":"output"}},"inputs":[],"label":"Input files","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":296.96873815702213,"top":157.00082696025927},"post_job_actions":{"HideDatasetActionimported_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"imported_data"},"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-data-single-end"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"EMPSingleEndSequences\\", \\"__current_case__\\": 17, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"EMPSingleEndDirFmt\\", \\"__current_case__\\": 1, \\"import_sequences\\": {\\"name\\": \\"sequences.fastq.gz\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"import_barcodes\\": {\\"name\\": \\"barcodes.fastq.gz\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"b39f05da-9f23-47ff-9695-b21bb61f18a2","when":null,"workflow_outputs":[]},"6":{"annotation":"Import the metadata as artifact","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":6,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import|data":{"id":2,"output_name":"output"}},"inputs":[],"label":"Metadata as artifact","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":297.4600499643163,"top":3.3186015520126793},"post_job_actions":{"HideDatasetActionimported_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"imported_data"},"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-metadata"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"ImmutableMetadata\\", \\"__current_case__\\": 58, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"ImmutableMetadataFormat\\", \\"__current_case__\\": 0, \\"import\\": {\\"name\\": \\"__q2galaxy__::literal::None\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"9caaabce-3a1a-4bf9-9fa0-d6fbe24a4bb4","when":null,"workflow_outputs":[]},"7":{"annotation":"qiime2 demux emp-single\\nDemultiplex sequence data generated with the EMP protocol.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":7,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|rev_comp_mapping_barcodes":{"id":4,"output_name":"output"},"barcodes|column":{"id":3,"output_name":"output"},"barcodes|source":{"id":6,"output_name":"imported_data"},"seqs":{"id":5,"output_name":"imported_data"}},"inputs":[{"description":"runtime parameter for tool qiime2 demux emp-single","name":"__q2galaxy__GUI__section__extra_opts__"},{"description":"runtime parameter for tool qiime2 demux emp-single","name":"barcodes"},{"description":"runtime parameter for tool qiime2 demux emp-single","name":"barcodes"}],"label":"Demultiplex single-end data","name":"qiime2 demux emp-single","outputs":[{"name":"per_sample_sequences","type":"qza"},{"name":"error_correction_details","type":"qza"}],"position":{"left":588.0871883326284,"top":1.7404776473888466},"post_job_actions":{"HideDatasetActionerror_correction_details":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"error_correction_details"},"RenameDatasetActionerror_correction_details":{"action_arguments":{"newname":"error-correction-demux"},"action_type":"RenameDatasetAction","output_name":"error_correction_details"},"RenameDatasetActionper_sample_sequences":{"action_arguments":{"newname":"demux-single-end"},"action_type":"RenameDatasetAction","output_name":"per_sample_sequences"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"8e01d149bcdb","name":"qiime2__demux__emp_single","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"golay_error_correction\\": true, \\"rev_comp_barcodes\\": false, \\"rev_comp_mapping_barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_description_mismatch\\": false}, \\"barcodes\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"seqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"23dd9199-b8aa-4daf-893f-1ca504ee0292","when":null,"workflow_outputs":[{"label":"Demultiplexed single-end data","output_name":"per_sample_sequences","uuid":"9cdc6089-1547-403c-a790-d2070d111e08"}]},"8":{"annotation":"qiime2 demux summarize\\nSummarize counts per sample.\\n(Galaxy Version 2023.5.0+q2galaxy.2023.5.0.2)","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":8,"input_connections":{"data":{"id":7,"output_name":"per_sample_sequences"}},"inputs":[],"label":"Summarize demultiplexed output","name":"qiime2 demux summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":869.9317626953125,"top":4.6328125},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"demux-single-end-visualization"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"1a13e87462a0","name":"qiime2__demux__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"n\\": \\"10000\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"3b70c32e-3cd7-4444-b5e8-3ecf0916a14f","when":null,"workflow_outputs":[{"label":"Summary of the demultiplexing","output_name":"visualization","uuid":"954e49a7-25e4-4484-9dab-b725348a57c6"}]}},"tags":[],"uuid":"951c418e-f57e-491d-af38-f2c44c6089ce","version":45},"readme":"# QIIME2 import workflows\\n\\n\\n## Available workflows\\n\\nImport of fastqsanger.gz data into QIIME artifact files.\\n\\nAvailable for:\\n\\n- paired / single end data\\n- demultiplexed / multiplexed data (the former according to the EMP protocol)\\n\\nFor data that is multiplexed with another protocol the Galaxy cutadapt tool can be use.\\n\\n## Inputs\\n\\n- Single end or paired end reads in fastq format.\\n- For demultiplexed data all datasets must be in a single (flat) collection\\n  (also paired data).\\n\\n### Demultiplexed data\\n\\n- Demultiplexed data must follow the naming scheme `.+_.+_R[12]_001\\\\.fastq\\\\.gz`.\\n  Any lane information (in the form of `L[0-9][0-9][0-9]_`) in the dataset names\\n  is automatically removed.\\n\\n### Mulmultiplexed data\\n\\n- Multiplexed data in a single or two fastq.gz dataset(s)\\n- Barcodes as fastq.gz file\\n- Metadata (a table describing the samples) and a metadata parameter (the name of the column that contains the barcode sequences)\\n- A boolean determining if there reverse complement of the barcode sequences shoul dbe used\\n\\n## Processing\\n\\nFor demultiplexed data\\n\\n1. Lane information is removed from the collection identifiers (using `Extract element identifiers`, `Regex Find And Replace` and `Relabel identifiers`)\\n2. Import of sequence data using `qiime2 tools import` with `Casava One Eight Laneless Per Sample Directory Format`\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\nFor multiplexed data\\n\\n1. Import sequences and metadata with `qiime2 tools import` as `EMP Paired End Directory Format` and `Immutable Metadata Format`, resp.\\n2. Demultiplex the sequences with `qiime2 demux emp-paired`/`paired` (using sequences and metadata information)\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\n\\n## Outputs\\n\\n- Sequence data in `qza` format\\n- A corresponding qiime visualization file in `qzv` format\\n\\n## TODOs\\n\\n- The import workflows for multiplexed data currently first convert the metadata into qza and require the user to enter a column as free text. If Galaxy allows for data-column workflow parameters this step can be removed.","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml and test.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 Ia: multiplexed data (single-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Sequences\\"]@{ shape: doc }\\n1[\\"ℹ️ Barcodes\\"]@{ shape: doc }\\n2[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n3[\\"ℹ️ Metadata parameter\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Reverse complement barcodes\\"]@{ shape: lean-l }\\n5[\\"Input files\\"]@{ shape: process }\\n1 --> 5\\n0 --> 5\\n6[\\"Metadata as artifact\\"]@{ shape: process }\\n2 --> 6\\n7[\\"Demultiplex single-end data\\"]@{ shape: process }\\n4 --> 7\\n3 --> 7\\n6 --> 7\\n5 --> 7\\n8[\\"Summarize demultiplexed output\\"]@{ shape: process }\\n7 --> 8\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-I-import/Ia-import-multiplexed-se","dockstore_id":26894,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244045","tests":[{"doc":"Test outline for QIIME2-Ia:-Demultiplexing-(single-end)","job":{"Sequences":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/moving-pictures/emp-single-end-sequences/sequences.fastq.gz","filetype":"fastqsanger.gz"},"Barcodes":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/moving-pictures/emp-single-end-sequences/barcodes.fastq.gz","filetype":"fastqsanger.gz"},"Metadata":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/moving-pictures/sample_metadata.tsv","filetype":"tabular"},"Metadata parameter":"barcode-sequence","Reverse complement barcodes":false},"outputs":{"Demultiplexed single-end data":{"asserts":[{"has_size":{"value":"20M","delta":"1M"}},{"has_archive_member":{"path":".*data/L[0-9]S[0-9]{1,3}_[0-9]{1,2}_L001_R1_001\\\\.fastq\\\\.gz","n":34}},{"has_archive_member":{"path":".*/data/metadata.yml","asserts":[{"has_text":{"text":"phred-offset: 33"}}]}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_text":{"text":"uuid:"}},{"has_text":{"text":"type: SampleData[SequencesWithQuality]"}},{"has_text":{"text":"format: SingleLanePerSampleSingleEndFastqDirFmt"}}]}}]}}}]},{"name":"Ib-import-multiplexed-pe","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-Ib-multiplexed-data-paired-end.ga","testParameterFiles":["/QIIME2-Ib-multiplexed-data-paired-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Importing paired-end multiplexed data (not demultiplexed yet)","creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"QIIME2 Ib: multiplexed data (paired-end)","steps":{"0":{"annotation":"All the forward-directed FASTQ sequences","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"All the forward-directed FASTQ sequences","name":"Forward sequences"}],"label":"Forward sequences","name":"Input dataset","outputs":[],"position":{"left":1.551926327108045,"top":5.8683936952556905},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"5d5aa3bf-a1d1-4df4-ac95-a33d09990a08","when":null,"workflow_outputs":[]},"1":{"annotation":"All the reverse-directed FASTQ sequences","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"All the reverse-directed FASTQ sequences","name":"Reverse sequences"}],"label":"Reverse sequences","name":"Input dataset","outputs":[],"position":{"left":0.384033203125,"top":131.51597595214844},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"229c1d1d-f1be-4565-b2ab-c99cdaf32d13","when":null,"workflow_outputs":[]},"2":{"annotation":"Barcode sequences","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Barcode sequences","name":"Barcodes"}],"label":"Barcodes","name":"Input dataset","outputs":[],"position":{"left":0,"top":259.2206573486328},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"0f49dd68-9962-40b5-bedc-ef31d3c0388e","when":null,"workflow_outputs":[]},"3":{"annotation":"The tab separated file for the metadata","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"The tab separated file for the metadata","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":0.727783203125,"top":389.2863006591797},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"1c2abb24-f3b0-4401-8801-5b22551f8ad0","when":null,"workflow_outputs":[]},"4":{"annotation":"Type in the name of the parameter which would be used as a demultiplexing reference","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Type in the name of the parameter which would be used as a demultiplexing reference","name":"Metadata parameter"}],"label":"Metadata parameter","name":"Input parameter","outputs":[],"position":{"left":299.46905517578125,"top":368.38479653625535},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3e62bbaf-435d-4d2a-85d7-3194ef82d6c8","when":null,"workflow_outputs":[]},"5":{"annotation":"If reverse complement of barcodes is necessary for the analysis, choose Yes.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"If reverse complement of barcodes is necessary for the analysis, choose Yes.","name":"Reverse complement of barcodes needed?"}],"label":"Reverse complement of barcodes needed?","name":"Input parameter","outputs":[],"position":{"left":1.358734130859375,"top":487.51402282714844},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c111c91e-170b-4097-b804-c58b963b7d29","when":null,"workflow_outputs":[]},"6":{"annotation":"qiime2 tools import\\nImport data into a QIIME 2 artifact","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":6,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import_barcodes|data":{"id":2,"output_name":"output"},"import_root|__q2galaxy__GUI__cond__format__|import_forward|data":{"id":0,"output_name":"output"},"import_root|__q2galaxy__GUI__cond__format__|import_reverse|data":{"id":1,"output_name":"output"}},"inputs":[],"label":"Import data into the pipeline","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":299.16656494140625,"top":151.00979653625532},"post_job_actions":{"HideDatasetActionimported_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"imported_data"},"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-data-paired-end"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"EMPPairedEndSequences\\", \\"__current_case__\\": 16, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"EMPPairedEndDirFmt\\", \\"__current_case__\\": 1, \\"import_forward\\": {\\"name\\": \\"forward.fastq.gz\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"import_reverse\\": {\\"name\\": \\"reverse.fastq.gz\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"import_barcodes\\": {\\"name\\": \\"barcodes.fastq.gz\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"6a264e2d-ac11-498f-bdcf-c115221bbc91","when":null,"workflow_outputs":[]},"7":{"annotation":"Import metadata into a QIIME 2 artifact","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":7,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import|data":{"id":3,"output_name":"output"}},"inputs":[],"label":"Metadata as artifact","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":301.67230224609375,"top":3.8492282989506634},"post_job_actions":{"HideDatasetActionimported_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"imported_data"},"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-metadata"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"ImmutableMetadata\\", \\"__current_case__\\": 58, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"ImmutableMetadataFormat\\", \\"__current_case__\\": 0, \\"import\\": {\\"name\\": \\"__q2galaxy__::literal::None\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"b2974890-9277-4369-81af-4fa0ff80c5fe","when":null,"workflow_outputs":[]},"8":{"annotation":"qiime2 demux emp-paired\\nDemultiplex paired-end sequence data generated with the EMP protocol.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":8,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|rev_comp_mapping_barcodes":{"id":5,"output_name":"output"},"barcodes|column":{"id":4,"output_name":"output"},"barcodes|source":{"id":7,"output_name":"imported_data"},"seqs":{"id":6,"output_name":"imported_data"}},"inputs":[{"description":"runtime parameter for tool qiime2 demux emp-paired","name":"__q2galaxy__GUI__section__extra_opts__"},{"description":"runtime parameter for tool qiime2 demux emp-paired","name":"barcodes"},{"description":"runtime parameter for tool qiime2 demux emp-paired","name":"barcodes"}],"label":"Demultiplex paired-end sequences","name":"qiime2 demux emp-paired","outputs":[{"name":"per_sample_sequences","type":"qza"},{"name":"error_correction_details","type":"qza"}],"position":{"left":592.7421875,"top":0},"post_job_actions":{"HideDatasetActionerror_correction_details":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"error_correction_details"},"RenameDatasetActionerror_correction_details":{"action_arguments":{"newname":"error-correction-demux-paired-end"},"action_type":"RenameDatasetAction","output_name":"error_correction_details"},"RenameDatasetActionper_sample_sequences":{"action_arguments":{"newname":"demux-paired-end"},"action_type":"RenameDatasetAction","output_name":"per_sample_sequences"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"8edb4dc16d6e","name":"qiime2__demux__emp_paired","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"golay_error_correction\\": true, \\"rev_comp_barcodes\\": false, \\"rev_comp_mapping_barcodes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_description_mismatch\\": false}, \\"barcodes\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"seqs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"e3395c76-e4fc-44f4-a944-2fb25f5d99fb","when":null,"workflow_outputs":[{"label":"Demultiplexed paired-end output","output_name":"per_sample_sequences","uuid":"d582d835-3dfd-43d5-b31f-79ddcc04ff70"}]},"9":{"annotation":"qiime2 demux summarize\\nSummarize counts per sample.\\n(Galaxy Version 2023.5.0+q2galaxy.2023.5.0.2)","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":9,"input_connections":{"data":{"id":8,"output_name":"per_sample_sequences"}},"inputs":[],"label":"Summarising the demultiplexed output","name":"qiime2 demux summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":876.4692949602977,"top":1.584142709419993},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"demux-paired-end-visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"1a13e87462a0","name":"qiime2__demux__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"n\\": \\"10000\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"cf48f270-6c6c-4fad-8c1d-2b02bb73ae49","when":null,"workflow_outputs":[{"label":"Summary of the demultiplexing","output_name":"visualization","uuid":"1aa2fe61-6a59-44c9-8079-a1d4a7ed8345"}]}},"tags":[],"uuid":"b09b5420-bb7c-4f27-9c90-7e2031889379","version":49},"readme":"# QIIME2 import workflows\\n\\n\\n## Available workflows\\n\\nImport of fastqsanger.gz data into QIIME artifact files.\\n\\nAvailable for:\\n\\n- paired / single end data\\n- demultiplexed / multiplexed data (the former according to the EMP protocol)\\n\\nFor data that is multiplexed with another protocol the Galaxy cutadapt tool can be use.\\n\\n## Inputs\\n\\n- Single end or paired end reads in fastq format.\\n- For demultiplexed data all datasets must be in a single (flat) collection\\n  (also paired data).\\n\\n### Demultiplexed data\\n\\n- Demultiplexed data must follow the naming scheme `.+_.+_R[12]_001\\\\.fastq\\\\.gz`.\\n  Any lane information (in the form of `L[0-9][0-9][0-9]_`) in the dataset names\\n  is automatically removed.\\n\\n### Mulmultiplexed data\\n\\n- Multiplexed data in a single or two fastq.gz dataset(s)\\n- Barcodes as fastq.gz file\\n- Metadata (a table describing the samples) and a metadata parameter (the name of the column that contains the barcode sequences)\\n- A boolean determining if there reverse complement of the barcode sequences shoul dbe used\\n\\n## Processing\\n\\nFor demultiplexed data\\n\\n1. Lane information is removed from the collection identifiers (using `Extract element identifiers`, `Regex Find And Replace` and `Relabel identifiers`)\\n2. Import of sequence data using `qiime2 tools import` with `Casava One Eight Laneless Per Sample Directory Format`\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\nFor multiplexed data\\n\\n1. Import sequences and metadata with `qiime2 tools import` as `EMP Paired End Directory Format` and `Immutable Metadata Format`, resp.\\n2. Demultiplex the sequences with `qiime2 demux emp-paired`/`paired` (using sequences and metadata information)\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\n\\n## Outputs\\n\\n- Sequence data in `qza` format\\n- A corresponding qiime visualization file in `qzv` format\\n\\n## TODOs\\n\\n- The import workflows for multiplexed data currently first convert the metadata into qza and require the user to enter a column as free text. If Galaxy allows for data-column workflow parameters this step can be removed.","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml and test.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 Ib: multiplexed data (paired-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Forward sequences\\"]@{ shape: doc }\\n1[\\"ℹ️ Reverse sequences\\"]@{ shape: doc }\\n2[\\"ℹ️ Barcodes\\"]@{ shape: doc }\\n3[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n4[\\"ℹ️ Metadata parameter\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Reverse complement of barcodes needed?\\"]@{ shape: lean-l }\\n6[\\"Import data into the pipeline\\"]@{ shape: process }\\n2 --> 6\\n0 --> 6\\n1 --> 6\\n7[\\"Metadata as artifact\\"]@{ shape: process }\\n3 --> 7\\n8[\\"Demultiplex paired-end sequences\\"]@{ shape: process }\\n5 --> 8\\n4 --> 8\\n7 --> 8\\n6 --> 8\\n9[\\"Summarising the demultiplexed output\\"]@{ shape: process }\\n8 --> 9\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-I-import/Ib-import-multiplexed-pe","dockstore_id":26895,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244055","tests":[{"doc":"Test outline for QIIME2-Ib:-Demultiplexing-(paired-end)","job":{"Forward sequences":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/atacama-soils/10p/forward.fastq.gz","filetype":"fastqsanger.gz"},"Reverse sequences":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/atacama-soils/10p/reverse.fastq.gz","filetype":"fastqsanger.gz"},"Barcodes":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/atacama-soils/10p/barcodes.fastq.gz","filetype":"fastqsanger.gz"},"Metadata":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/atacama-soils/sample_metadata.tsv","filetype":"tabular"},"Metadata parameter":"barcode-sequence","Reverse complement of barcodes needed?":true},"outputs":{"Demultiplexed paired-end output":{"asserts":[{"has_size":{"value":"95M","delta":"4M"}},{"has_archive_member":{"path":".*data/[A-Z]{3}[0-9]+(\\\\.[0-9])?\\\\.[0-9]_[0-9]{1,2}_L001_R[12]_001\\\\.fastq\\\\.gz","n":150}},{"has_archive_member":{"path":".*/data/metadata.yml","asserts":[{"has_text":{"text":"phred-offset: 33"}}]}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_text":{"text":"uuid:"}},{"has_text":{"text":"SampleData[PairedEndSequencesWithQuality]"}},{"has_text":{"text":"format: SingleLanePerSamplePairedEndFastqDirFmt"}}]}}]}}}]},{"name":"Ic-import-demultiplexed-se","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-Ic-demultiplexed-data-single-end.ga","testParameterFiles":["/QIIME2-Ic-demultiplexed-data-single-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Importing demultiplexed data (single-end)","creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"QIIME2 Ic: Demultiplexed data (single-end)","steps":{"0":{"annotation":"A collection of single-end demultiplexed sequences","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A collection of single-end demultiplexed sequences","name":"Sequence collection"}],"label":"Sequence collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":9.339782603705014},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"35c1a1b1-daeb-4cd1-bd91-e5f0016ec79a","when":null,"workflow_outputs":[]},"1":{"annotation":"Extract a list of the names of sequence files","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":"Extract element identifiers","name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":262.551090424436,"top":137.8447786862222},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"08cc79e2-3fad-483f-aebd-4f444c978d84","when":null,"workflow_outputs":[]},"2":{"annotation":"This is mainly to clean out lane parameters to continue import in laneless format","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":"Screening laneless and single-lane","name":"Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":505.17041015625,"top":135.84155315653058},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"_L[0-9][0-9][0-9]_\\", \\"replacement\\": \\"_\\"}], \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"3e2bbefc-5452-4320-b258-140f892d69af","when":null,"workflow_outputs":[]},"3":{"annotation":"Renaming of sequence files to fit into laneless format","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":3,"input_connections":{"how|labels":{"id":2,"output_name":"out_file1"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabel sequence files","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":804.3797337280766,"top":0},"post_job_actions":{},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"c7dcacd6-1f62-4764-a984-ae47e88c94ec","when":null,"workflow_outputs":[]},"4":{"annotation":"qiime2 tools import\\nImport data into a QIIME 2 artifact","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":4,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import_sequences|__q2galaxy__GUI__cond__sequences__|elements":{"id":3,"output_name":"output"}},"inputs":[],"label":"Import data into the pipeline","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":1079.0168624775688,"top":1.637809220122261},"post_job_actions":{"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-data-demultiplexed"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"SampleData__ob__SequencesWithQuality__cb__\\", \\"__current_case__\\": 103, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"CasavaOneEightLanelessPerSampleDirFmt\\", \\"__current_case__\\": 0, \\"import_sequences\\": {\\"__q2galaxy__GUI__cond__sequences__\\": {\\"__q2galaxy__GUI__select__picker__\\": \\"collection\\", \\"__current_case__\\": 0, \\"elements\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__q2galaxy__GUI__cond__add_ext__\\": {\\"__q2galaxy__GUI__select__ext_pick__\\": \\"no\\", \\"__current_case__\\": 1}}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"6a264e2d-ac11-498f-bdcf-c115221bbc91","when":null,"workflow_outputs":[{"label":"Demultiplexed data","output_name":"imported_data","uuid":"0015293e-bcd7-433b-8b98-e05fda148602"}]},"5":{"annotation":"qiime2 demux summarize\\nSummarize counts per sample.\\n(Galaxy Version 2023.5.0+q2galaxy.2023.5.0.2)","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":5,"input_connections":{"data":{"id":4,"output_name":"imported_data"}},"inputs":[],"label":"Summarising the demultiplexed output","name":"qiime2 demux summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":1339.6939711359755,"top":5.9268220678864205},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"demux-visualization"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"1a13e87462a0","name":"qiime2__demux__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"n\\": \\"10000\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"cf48f270-6c6c-4fad-8c1d-2b02bb73ae49","when":null,"workflow_outputs":[{"label":"Visualization","output_name":"visualization","uuid":"5c67b911-b178-4f6f-8ef8-0071d72d3a25"}]}},"tags":[],"uuid":"5a57f873-52f8-4c51-9e8d-e747d32d183a","version":1},"readme":"# QIIME2 import workflows\\n\\n\\n## Available workflows\\n\\nImport of fastqsanger.gz data into QIIME artifact files.\\n\\nAvailable for:\\n\\n- paired / single end data\\n- demultiplexed / multiplexed data (the former according to the EMP protocol)\\n\\nFor data that is multiplexed with another protocol the Galaxy cutadapt tool can be use.\\n\\n## Inputs\\n\\n- Single end or paired end reads in fastq format.\\n- For demultiplexed data all datasets must be in a single (flat) collection\\n  (also paired data).\\n\\n### Demultiplexed data\\n\\n- Demultiplexed data must follow the naming scheme `.+_.+_R[12]_001\\\\.fastq\\\\.gz`.\\n  Any lane information (in the form of `L[0-9][0-9][0-9]_`) in the dataset names\\n  is automatically removed.\\n\\n### Mulmultiplexed data\\n\\n- Multiplexed data in a single or two fastq.gz dataset(s)\\n- Barcodes as fastq.gz file\\n- Metadata (a table describing the samples) and a metadata parameter (the name of the column that contains the barcode sequences)\\n- A boolean determining if there reverse complement of the barcode sequences shoul dbe used\\n\\n## Processing\\n\\nFor demultiplexed data\\n\\n1. Lane information is removed from the collection identifiers (using `Extract element identifiers`, `Regex Find And Replace` and `Relabel identifiers`)\\n2. Import of sequence data using `qiime2 tools import` with `Casava One Eight Laneless Per Sample Directory Format`\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\nFor multiplexed data\\n\\n1. Import sequences and metadata with `qiime2 tools import` as `EMP Paired End Directory Format` and `Immutable Metadata Format`, resp.\\n2. Demultiplex the sequences with `qiime2 demux emp-paired`/`paired` (using sequences and metadata information)\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\n\\n## Outputs\\n\\n- Sequence data in `qza` format\\n- A corresponding qiime visualization file in `qzv` format\\n\\n## TODOs\\n\\n- The import workflows for multiplexed data currently first convert the metadata into qza and require the user to enter a column as free text. If Galaxy allows for data-column workflow parameters this step can be removed.","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml and test.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 Ic: Demultiplexed data (single-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Sequence collection\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Screening laneless and single-lane\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Relabel sequence files\\"]@{ shape: process }\\n2 --> 3\\n0 --> 3\\n4[\\"Import data into the pipeline\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Summarising the demultiplexed output\\"]@{ shape: process }\\n4 --> 5\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-I-import/Ic-import-demultiplexed-se","dockstore_id":26896,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244065","tests":[{"doc":"Test outline for QIIME2-Ic-demultiplexed-data-single-end","job":{"Sequence collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"FMT.0093C_5_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093C_5_L001_R1_001.fastq.gz"},{"class":"File","identifier":"FMT.0093D_2_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093D_2_L001_R1_001.fastq.gz"},{"class":"File","identifier":"FMT.0093E_25_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093E_25_L001_R1_001.fastq.gz"}]}},"outputs":{"Demultiplexed data":{"asserts":[{"has_size":{"value":"12M","delta":"1M"}},{"has_archive_member":{"path":".*fastq.gz","n":3}},{"has_archive_member":{"path":".*/data/metadata.yml","asserts":[{"has_text":{"text":"phred-offset: 33"}}]}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_text":{"text":"uuid:"}},{"has_text":{"text":"type: SampleData[SequencesWithQuality]"}},{"has_text":{"text":"format: SingleLanePerSampleSingleEndFastqDirFmt"}}]}}]}}}]},{"name":"Id-import-demultiplexed-pe","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-Id-demultiplexed-data-paired-end.ga","testParameterFiles":["/QIIME2-Id-demultiplexed-data-paired-end-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"Importing demultiplexed data (paired-end)","creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","release":"0.3","license":"MIT","name":"QIIME2 Id: Demultiplexed data (paired-end)","steps":{"0":{"annotation":"Import of the paired-end demultiplexed sequences","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Import of the paired-end demultiplexed sequences","name":"Sequence collection"}],"label":"Sequence collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":59.91667175292969},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger.gz\\", \\"fastqillumina.gz\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"35c1a1b1-daeb-4cd1-bd91-e5f0016ec79a","when":null,"workflow_outputs":[]},"1":{"annotation":"Extract a list of the names of sequence files","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":"Extract element identifiers","name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":304.79388003144334,"top":263.4010896400516},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"5570c46c-5a48-43aa-9e5c-5f7c05c057d8","when":null,"workflow_outputs":[]},"2":{"annotation":"This is mainly to clean out lane parameters to continue import in laneless format ","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":"Screening laneless and single-lane","name":"Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":571.1741505024197,"top":264.71925121294674},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"_L[0-9][0-9][0-9]_\\", \\"replacement\\": \\"_\\"}], \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"bde7f68d-6c55-4431-92bf-bcf61539ff8f","when":null,"workflow_outputs":[]},"3":{"annotation":"Renaming of sequence files to fit into laneless format","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":3,"input_connections":{"how|labels":{"id":2,"output_name":"out_file1"},"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabel sequence files","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":803.9640223409692,"top":32.109518819964116},"post_job_actions":{},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"7347626e-4600-48c1-9ab1-dc989ccebc65","when":null,"workflow_outputs":[]},"4":{"annotation":"qiime2 tools import\\nImport data into a QIIME 2 artifact","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":4,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import_sequences|__q2galaxy__GUI__cond__sequences__|elements":{"id":3,"output_name":"output"}},"inputs":[],"label":"Import data into the pipeline","name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":1078.219667111558,"top":27.908915490176845},"post_job_actions":{"RenameDatasetActionimported_data":{"action_arguments":{"newname":"input-data-demultiplexed"},"action_type":"RenameDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"SampleData__ob__PairedEndSequencesWithQuality__cb__\\", \\"__current_case__\\": 100, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"CasavaOneEightLanelessPerSampleDirFmt\\", \\"__current_case__\\": 0, \\"import_sequences\\": {\\"__q2galaxy__GUI__cond__sequences__\\": {\\"__q2galaxy__GUI__select__picker__\\": \\"collection\\", \\"__current_case__\\": 0, \\"elements\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__q2galaxy__GUI__cond__add_ext__\\": {\\"__q2galaxy__GUI__select__ext_pick__\\": \\"no\\", \\"__current_case__\\": 1}}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"6a264e2d-ac11-498f-bdcf-c115221bbc91","when":null,"workflow_outputs":[{"label":"Demultiplexed data","output_name":"imported_data","uuid":"7cb96ead-ed0f-4874-9fa4-0c766f78b123"}]},"5":{"annotation":"qiime2 demux summarize\\nSummarize counts per sample.\\n(Galaxy Version 2023.5.0+q2galaxy.2023.5.0.2)","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":5,"input_connections":{"data":{"id":4,"output_name":"imported_data"}},"inputs":[],"label":"Summarising the demultiplexed output","name":"qiime2 demux summarize","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":1358.9000244140625,"top":0},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"demux-visualization"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"1a13e87462a0","name":"qiime2__demux__summarize","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"n\\": \\"10000\\"}, \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"cf48f270-6c6c-4fad-8c1d-2b02bb73ae49","when":null,"workflow_outputs":[{"label":"visualization","output_name":"visualization","uuid":"e7d259fd-ffa7-4d18-af74-343cedca4d8f"}]}},"tags":[],"uuid":"f4887024-6f68-4d53-a889-5635366dd18d","version":2},"readme":"# QIIME2 import workflows\\n\\n\\n## Available workflows\\n\\nImport of fastqsanger.gz data into QIIME artifact files.\\n\\nAvailable for:\\n\\n- paired / single end data\\n- demultiplexed / multiplexed data (the former according to the EMP protocol)\\n\\nFor data that is multiplexed with another protocol the Galaxy cutadapt tool can be use.\\n\\n## Inputs\\n\\n- Single end or paired end reads in fastq format.\\n- For demultiplexed data all datasets must be in a single (flat) collection\\n  (also paired data).\\n\\n### Demultiplexed data\\n\\n- Demultiplexed data must follow the naming scheme `.+_.+_R[12]_001\\\\.fastq\\\\.gz`.\\n  Any lane information (in the form of `L[0-9][0-9][0-9]_`) in the dataset names\\n  is automatically removed.\\n\\n### Mulmultiplexed data\\n\\n- Multiplexed data in a single or two fastq.gz dataset(s)\\n- Barcodes as fastq.gz file\\n- Metadata (a table describing the samples) and a metadata parameter (the name of the column that contains the barcode sequences)\\n- A boolean determining if there reverse complement of the barcode sequences shoul dbe used\\n\\n## Processing\\n\\nFor demultiplexed data\\n\\n1. Lane information is removed from the collection identifiers (using `Extract element identifiers`, `Regex Find And Replace` and `Relabel identifiers`)\\n2. Import of sequence data using `qiime2 tools import` with `Casava One Eight Laneless Per Sample Directory Format`\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\nFor multiplexed data\\n\\n1. Import sequences and metadata with `qiime2 tools import` as `EMP Paired End Directory Format` and `Immutable Metadata Format`, resp.\\n2. Demultiplex the sequences with `qiime2 demux emp-paired`/`paired` (using sequences and metadata information)\\n3. Prepare visualisation dataset with `qiime2 demux summarize`\\n\\n\\n## Outputs\\n\\n- Sequence data in `qza` format\\n- A corresponding qiime visualization file in `qzv` format\\n\\n## TODOs\\n\\n- The import workflows for multiplexed data currently first convert the metadata into qza and require the user to enter a column as free text. If Galaxy allows for data-column workflow parameters this step can be removed.","changelog":"# Changelog\\n\\n## [0.3] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_single/qiime2__demux__emp_single/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__summarize/qiime2__demux__summarize/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__demux__emp_paired/qiime2__demux__emp_paired/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.2] - 2024-05-23\\nFix dockstore.yml and test.\\n\\n## [0.1] - 2024-04-08\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 Id: Demultiplexed data (paired-end)\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Sequence collection\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Screening laneless and single-lane\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Relabel sequence files\\"]@{ shape: process }\\n2 --> 3\\n0 --> 3\\n4[\\"Import data into the pipeline\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Summarising the demultiplexed output\\"]@{ shape: process }\\n4 --> 5\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-I-import/Id-import-demultiplexed-pe","dockstore_id":26897,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244067","tests":[{"doc":"Test outline for QIIME2-Id:-Demultiplexed-data-(paired-end)","job":{"Sequence collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"FMT.0093C_46_L001_R2_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093C_46_L001_R2_001.fastq.gz"},{"class":"File","identifier":"FMT.0093C_5_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093C_5_L001_R1_001.fastq.gz"},{"class":"File","identifier":"FMT.0093D_2_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093D_2_L001_R1_001.fastq.gz"},{"class":"File","identifier":"FMT.0093D_43_L001_R2_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093D_43_L001_R2_001.fastq.gz"},{"class":"File","identifier":"FMT.0093E_25_L001_R1_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093E_25_L001_R1_001.fastq.gz"},{"class":"File","identifier":"FMT.0093E_66_L001_R2_001.fastq.gz","location":"https://docs.qiime2.org/jupyterbooks/cancer-microbiome-intervention-tutorial/data/020-tutorial-upstream/030-importing/data_to_import/FMT.0093E_66_L001_R2_001.fastq.gz"}]}},"outputs":{"Demultiplexed data":{"asserts":[{"has_size":{"value":"24M","delta":"1M"}},{"has_archive_member":{"path":".*fastq.gz","n":6}},{"has_archive_member":{"path":".*/data/metadata.yml","asserts":[{"has_text":{"text":"phred-offset: 33"}}]}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_text":{"text":"uuid:"}},{"has_text":{"text":"type: SampleData[PairedEndSequencesWithQuality]"}},{"has_text":{"text":"format: SingleLanePerSamplePairedEndFastqDirFmt"}}]}}]}}}]}],"path":"./workflows/amplicon/qiime2/qiime2-I-import"},{"version":1.2,"workflows":[{"name":"QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis.ga","testParameterFiles":["/QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow \\n- Reconstruct phylogeny (insert fragments in a reference)\\n- Alpha rarefaction analysis\\n- Taxonomic analysis","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","name":"QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis","release":"0.2","steps":{"0":{"annotation":"Representative sequences, e.g. from DADA2","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Representative sequences, e.g. from DADA2","name":"Representative sequences"}],"label":"Representative sequences","name":"Input dataset","outputs":[],"position":{"left":1.1907255852393628,"top":32.84463293555338},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ffb9d0c7-787e-4b70-9ed4-73624641b185","when":null,"workflow_outputs":[]},"1":{"annotation":"Feature table. e.g. from DADA2","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Feature table. e.g. from DADA2","name":"Feature table"}],"label":"Feature table","name":"Input dataset","outputs":[],"position":{"left":0,"top":141.5833282470703},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"388c0b1a-4e14-44e3-83d5-97a5052f5d26","when":null,"workflow_outputs":[]},"2":{"annotation":"Tab separated metadata file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":3.298961099430351,"top":256.6628878437186},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tsv\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"64e30001-38ae-4d19-a93c-6d0572d10252","when":null,"workflow_outputs":[]},"3":{"annotation":"Minimum depth used for the rarefaction analysis (minimum value : 1)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Minimum depth used for the rarefaction analysis (minimum value : 1)","name":"Minimum depth"}],"label":"Minimum depth","name":"Input parameter","outputs":[],"position":{"left":5.446155936666033,"top":345.49918723766496},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"93c5a1ed-7222-4e61-bfde-7c2ee74bb1fe","when":null,"workflow_outputs":[]},"4":{"annotation":"Maximum depth used for the rarefaction analysis (minimum value : 1)","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Maximum depth used for the rarefaction analysis (minimum value : 1)","name":"Maximum depth"}],"label":"Maximum depth","name":"Input parameter","outputs":[],"position":{"left":6.523874261050689,"top":444.34720870793564},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"37b64f9f-e47a-4f70-aec2-ef2f3e005c4c","when":null,"workflow_outputs":[]},"5":{"annotation":"Reference data for the SEPP fragmentation insertion phylogenetic tree generation","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Reference data for the SEPP fragmentation insertion phylogenetic tree generation","name":"SEPP fragment insertion reference"}],"label":"SEPP fragment insertion reference","name":"Input dataset","outputs":[],"position":{"left":14.95001220703125,"top":634.2999877929688},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"3cb0a84c-dc8c-4325-929b-2d3915e7738a","when":null,"workflow_outputs":[]},"6":{"annotation":"Taxonomic classifier (e.g. Greengenes or SILVA)","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Taxonomic classifier (e.g. Greengenes or SILVA)","name":"Taxonomic classifier"}],"label":"Taxonomic classifier","name":"Input dataset","outputs":[],"position":{"left":13.29724891100554,"top":751.0166625976562},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8c477a8c-f360-4a8a-95ef-c4c9b8168ae8","when":null,"workflow_outputs":[]},"7":{"annotation":"Phylogenetic tree for diversity analysis","id":7,"input_connections":{"DADA2 representative sequences":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"SEPP fragment insertion source file":{"id":5,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[{"description":"Phylogenetic tree for diversity analysis","name":"Phylogenetic tree for diversity analysis"}],"label":"Phylogenetic tree for diversity analysis","name":"QIIME2 III: Phylogenetic tree for diversity analysis","outputs":[],"position":{"left":457.7998825174995,"top":0},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","name":"QIIME2 III: Phylogenetic tree for diversity analysis","steps":{"0":{"annotation":"DADA2 representative sequences","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"DADA2 representative sequences","name":"DADA2 representative sequences"}],"label":"DADA2 representative sequences","name":"Input dataset","outputs":[],"position":{"left":0,"top":19.497570176200952},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"71e6440a-8aa6-4899-bd11-901dffae36a5","when":null,"workflow_outputs":[]},"1":{"annotation":"Reference data for the SEPP fragmentation insertion phylogenetic tree generation","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Reference data for the SEPP fragmentation insertion phylogenetic tree generation","name":"SEPP fragment insertion source file"}],"label":"SEPP fragment insertion source file","name":"Input dataset","outputs":[],"position":{"left":0.34857177734375,"top":144.6569787455369},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"2bf4a3e7-51d9-4323-ab9a-f6a85c3b0cd7","when":null,"workflow_outputs":[]},"2":{"annotation":"qiime2 fragment-insertion sepp\\nInsert fragment sequences using SEPP into reference phylogenies.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":2,"input_connections":{"reference_database":{"id":1,"output_name":"output"},"representative_sequences":{"id":0,"output_name":"output"}},"inputs":[],"label":"Phylogenetic tree generation","name":"qiime2 fragment-insertion sepp","outputs":[{"name":"tree","type":"qza"},{"name":"placements","type":"qza"}],"position":{"left":310.42930324654003,"top":0},"post_job_actions":{"HideDatasetActionplacements":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"placements"},"RenameDatasetActionplacements":{"action_arguments":{"newname":"rooted_tree_placements"},"action_type":"RenameDatasetAction","output_name":"placements"},"RenameDatasetActiontree":{"action_arguments":{"newname":"rooted_tree"},"action_type":"RenameDatasetAction","output_name":"tree"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"a74f06287b8c","name":"qiime2__fragment_insertion__sepp","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"alignment_subset_size\\": \\"1000\\", \\"placement_subset_size\\": \\"5000\\", \\"debug\\": false}, \\"reference_database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"representative_sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"f6aa969f-1bdf-4d74-8b0a-af93a0640040","when":null,"workflow_outputs":[{"label":"Rooted tree","output_name":"tree","uuid":"e572bf3b-64aa-4bd5-a629-f53ba6a383be"}]}},"tags":[],"uuid":"ea7b99e7-ee7b-4fd5-bda8-cd00ddfe6ade"},"tool_id":null,"type":"subworkflow","uuid":"ccd1436d-4d55-401f-b794-8396eef8288f","when":null,"workflow_outputs":[{"label":"Rooted tree","output_name":"Rooted tree","uuid":"86b75240-ab72-47df-a1c1-a70911d19d66"}]},"8":{"annotation":"Taxonomic analysis","id":8,"input_connections":{"DADA2 feature table":{"id":1,"input_subworkflow_step_id":3,"output_name":"output"},"DADA2 representative sequences":{"id":0,"input_subworkflow_step_id":1,"output_name":"output"},"Metadata":{"id":2,"input_subworkflow_step_id":0,"output_name":"output"},"Taxonomic classifier":{"id":6,"input_subworkflow_step_id":2,"output_name":"output"}},"inputs":[{"description":"Taxonomic analysis","name":"Taxonomic analysis"}],"label":"Taxonomic analysis","name":"QIIME2 V: Taxonomic analysis","outputs":[],"position":{"left":484.1197712977931,"top":565.4134141655171},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","name":"QIIME2 V: Taxonomic analysis","steps":{"0":{"annotation":"Tab separated metadata file","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":0.791015625,"top":2.6916668368741092},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c2deedc3-713c-4030-9fb8-5734c016ad5c","when":null,"workflow_outputs":[]},"1":{"annotation":"Demultiplexed sequences in qza format","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Demultiplexed sequences in qza format","name":"DADA2 representative sequences"}],"label":"DADA2 representative sequences","name":"Input dataset","outputs":[],"position":{"left":1.8453224170332305,"top":111.47987345989156},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e250ddc1-f567-40ca-9f2b-e07bbf84cdec","when":null,"workflow_outputs":[]},"2":{"annotation":"Taxonomic classifier (e.g. Greengenes or SILVA)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Taxonomic classifier (e.g. Greengenes or SILVA)","name":"Taxonomic classifier"}],"label":"Taxonomic classifier","name":"Input dataset","outputs":[],"position":{"left":0.451446533203125,"top":241.0041668368741},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ac0cdf1d-2a29-481c-aaa3-21f1b8f9b97d","when":null,"workflow_outputs":[]},"3":{"annotation":"DADA2 feature table","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"DADA2 feature table","name":"DADA2 feature table"}],"label":"DADA2 feature table","name":"Input dataset","outputs":[],"position":{"left":0,"top":365.0612957431241},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"81495001-5b55-4676-8d7f-cb0659d6acc3","when":null,"workflow_outputs":[]},"4":{"annotation":"qiime2 feature-classifier classify-sklearn\\nPre-fitted sklearn-based taxonomy classifier","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":4,"input_connections":{"classifier":{"id":2,"output_name":"output"},"reads":{"id":1,"output_name":"output"}},"inputs":[],"label":"Taxonomy classification","name":"qiime2 feature-classifier classify-sklearn","outputs":[{"name":"classification","type":"qza"}],"position":{"left":334.0095438830636,"top":1.9720584701849528},"post_job_actions":{"RenameDatasetActionclassification":{"action_arguments":{"newname":"taxonomy_classification"},"action_type":"RenameDatasetAction","output_name":"classification"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"771d8be25d7a","name":"qiime2__feature_classifier__classify_sklearn","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"__q2galaxy__GUI__conditional__reads_per_batch__\\": {\\"__q2galaxy__GUI__select__\\": \\"auto\\", \\"__current_case__\\": 0, \\"reads_per_batch\\": \\"auto\\"}, \\"pre_dispatch\\": \\"2*n_jobs\\", \\"__q2galaxy__GUI__conditional__confidence__\\": {\\"__q2galaxy__GUI__select__\\": \\"__q2galaxy__::control::Float X Range(0__comma__ 1__comma__ inclusive_end=True)\\", \\"__current_case__\\": 1, \\"confidence\\": \\"0.7\\"}, \\"read_orientation\\": \\"auto\\"}, \\"classifier\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"cf9e003e-bcd4-4819-88cf-98d3bc8bf531","when":null,"workflow_outputs":[{"label":"Taxonomy classification","output_name":"classification","uuid":"c2f6d710-2fec-45de-a31d-bfa001afb26c"}]},"5":{"annotation":"qiime2 taxa barplot\\nVisualize taxonomy with an interactive bar plot","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":5,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":0,"output_name":"output"},"__q2galaxy__GUI__section__extra_opts__|taxonomy":{"id":4,"output_name":"classification"},"table":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool qiime2 taxa barplot","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Taxonomy barplot","name":"qiime2 taxa barplot","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":645.4988052205106,"top":0},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"taxa_barplot"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"3da2f87501ee","name":"qiime2__taxa__barplot","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"taxonomy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__q2galaxy__GUI__conditional__level_delimiter__\\": {\\"__q2galaxy__GUI__select__\\": \\"__q2galaxy__::control::default\\", \\"__current_case__\\": 0, \\"level_delimiter\\": \\"__q2galaxy__::literal::None\\"}}, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"a2146130-d5a3-4f0d-814f-61677f57709b","when":null,"workflow_outputs":[{"label":"Taxa barplot","output_name":"visualization","uuid":"cef83374-d99f-4a8c-98ec-90afd3a37da3"}]},"6":{"annotation":"qiime2 metadata tabulate\\nInteractively explore Metadata in an HTML table","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":6,"input_connections":{"input_0|__q2galaxy__GUI__conditional__input__|source":{"id":4,"output_name":"classification"}},"inputs":[],"label":"Tabulate taxonomy classification","name":"qiime2 metadata tabulate","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":646.2365246851235,"top":304.22497686132425},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"taxonomy_classification_visualisation"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"9857be21dcaf","name":"qiime2__metadata__tabulate","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"page_size\\": \\"100\\"}, \\"input\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__input__\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"380548c3-b9b6-4ebd-ae68-e3fbb891f7e5","when":null,"workflow_outputs":[{"label":"Taxonomy classification table","output_name":"visualization","uuid":"9b1b4f66-d7a7-42cd-ac81-00c417300f47"}]}},"tags":[],"uuid":"d0309a9d-66d8-46e2-b103-674377fa0de8"},"tool_id":null,"type":"subworkflow","uuid":"b145de22-67f9-4577-99d9-cffc76d210ce","when":null,"workflow_outputs":[{"label":"Taxonomy classification","output_name":"Taxonomy classification","uuid":"622c6287-2cea-4b85-82cc-26a2d3abfb3d"},{"label":"Taxa barplot","output_name":"Taxa barplot","uuid":"1857691f-3d38-45be-a6e5-4dca9a0d836a"},{"label":"Taxonomy classification table","output_name":"Taxonomy classification table","uuid":"8a273aca-d2bc-4038-a13c-63cea1666839"}]},"9":{"annotation":"Rarefaction","id":9,"input_connections":{"DADA2 feature table":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"},"Maximum depth":{"id":4,"input_subworkflow_step_id":4,"output_name":"output"},"Metadata":{"id":2,"input_subworkflow_step_id":0,"output_name":"output"},"Minimum depth":{"id":3,"input_subworkflow_step_id":3,"output_name":"output"},"Rooted tree":{"id":7,"input_subworkflow_step_id":2,"output_name":"Rooted tree"}},"inputs":[{"description":"Rarefaction","name":"Rarefaction"}],"label":"Rarefaction","name":"QIIME2 IV: Rarefaction","outputs":[],"position":{"left":766.4972916356149,"top":298.3500061035156},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","name":"QIIME2 IV: Rarefaction","steps":{"0":{"annotation":"Tab separated metadata file","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Tab separated metadata file","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8dde0c69-40f1-415a-a866-5d1115e80cdf","when":null,"workflow_outputs":[]},"1":{"annotation":"DADA2 feature table","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"DADA2 feature table","name":"DADA2 feature table"}],"label":"DADA2 feature table","name":"Input dataset","outputs":[],"position":{"left":2.3167686199511115,"top":140.50124094027063},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"92e667de-c04f-420c-9747-ffedf77041fa","when":null,"workflow_outputs":[]},"2":{"annotation":"Import rooted_tree output from phylogenetic tree generation step","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Import rooted_tree output from phylogenetic tree generation step","name":"Rooted tree"}],"label":"Rooted tree","name":"Input dataset","outputs":[],"position":{"left":2.0428123211229865,"top":260.4983035867503},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"86b5c5cc-362f-4dc3-990d-17843406df1a","when":null,"workflow_outputs":[]},"3":{"annotation":"The depth value is based on the feature sampling depth from DADA2 table output (minimum value : 1)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"The depth value is based on the feature sampling depth from DADA2 table output (minimum value : 1)","name":"Minimum depth"}],"label":"Minimum depth","name":"Input parameter","outputs":[],"position":{"left":1.766632080078125,"top":353.64630126953125},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3c3e503d-c3ff-4257-8ce8-7acb2cb4ab3c","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"5f3015ba-bfed-48bc-a22e-03eb8f2576e2"}]},"4":{"annotation":"The depth value is based on the feature sampling depth from DADA2 table output","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"The depth value is based on the feature sampling depth from DADA2 table output","name":"Maximum depth"}],"label":"Maximum depth","name":"Input parameter","outputs":[],"position":{"left":0.92767333984375,"top":472.1087646484375},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"2af02538-19c9-4ffc-924e-5108987064a1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"b4075e2a-6d99-4102-a396-05ef3f4b3cca"}]},"5":{"annotation":"qiime2 diversity alpha-rarefaction\\nAlpha rarefaction curves\\n(Galaxy Version 2023.5.1+q2galaxy.2023.5.0.2)","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":5,"input_connections":{"__q2galaxy__GUI__section__extra_opts__|metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":0,"output_name":"output"},"__q2galaxy__GUI__section__extra_opts__|min_depth":{"id":3,"output_name":"output"},"__q2galaxy__GUI__section__extra_opts__|phylogeny":{"id":2,"output_name":"output"},"max_depth":{"id":4,"output_name":"output"},"table":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool qiime2 diversity alpha-rarefaction","name":"__q2galaxy__GUI__section__extra_opts__"},{"description":"runtime parameter for tool qiime2 diversity alpha-rarefaction","name":"__q2galaxy__GUI__section__extra_opts__"}],"label":"Alpha rarefaction","name":"qiime2 diversity alpha-rarefaction","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":348.86456298828125,"top":75.385009765625},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"alpha_rarefaction"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"22bfda174157","name":"qiime2__diversity__alpha_rarefaction","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"phylogeny\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metrics\\": [], \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"min_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"steps\\": \\"10\\", \\"iterations\\": \\"10\\"}, \\"max_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"489f6ea1-d13a-48ae-b71d-40c413f3b45e","when":null,"workflow_outputs":[{"label":"Rarefaction curve","output_name":"visualization","uuid":"5675c59e-7e5e-4b0f-a7e7-263a0fa42f1d"}]}},"tags":[],"uuid":"28382e50-9000-48f1-ae97-d468934f4962"},"tool_id":null,"type":"subworkflow","uuid":"ba53440b-8543-44d1-9f96-16845342c339","when":null,"workflow_outputs":[{"label":"Rarefaction curve","output_name":"Rarefaction curve","uuid":"12394a5b-15fc-44f2-80cd-2b38de92517a"}]}},"tags":[],"uuid":"3163175d-7673-42d9-8aae-505afbd74946","version":30},"readme":"# QIIME2 workflows\\n\\n## Available workflows\\n\\n- III-V Downstream analyses: III) reconstruct a taxonomy for diversity analysis, IV) rarefaction analysis, V) taxonomic analysis.\\n- VI: Computation of diversity metrics and estimations\\n\\nAnalogous to the procedures described in the Parkinson’s Mouse Tutorial: https://docs.qiime2.org/2024.5/tutorials/pd-mice/\\n\\n## Inputs\\n\\nThe two workflows have two inputs in common \\n\\n- Feature table: Count data\\n- Metadata: Metadata table\\n\\nand the following extra inputs\\n\\nIII-V\\n\\n- Representative sequences: Representative (ASV) sequences\\n- Minimum depth: Lower limit of the sampling depth for the alpha rarefaction analysis\\n- Maximum depth: Upper limit of the sampling depth for the alpha rarefaction analysis\\n- SEPP fragment insertion reference: used for the reconstruction of the phylogenetic tree\\n- Taxonomic classifier: The classifier to assign taxonomic information to the ASVs\\n\\nVI:\\n\\nSampling depth: For the metric calculation (should be based on the rarefaction analysis done in IV)\\nTarget metadata parameter: that should be used for beta diversity calculations\\nRooted Tree: for instance the tree computed in III\\n\\n## Processing\\n\\nIII-V\\n\\n- Phylogenetic tree generation using `qiime2 fragment-insertion sepp`\\n- Alpha rarefaction analysis using `qiime2 diversity alpha-rarefaction`\\n- Taxonomic classification using `qiime2 feature-classifier classify-sklearn` and compute barplot and tabular output\\n\\nVI: \\n\\n- compute alpha and beta diversity metrics using `qiime2 diversity core-metrics-phylogenetic`\\n- organize these metrics in 4 collections:\\n  1. Distance matrix collection (weighted and unweighted unifrac, jaccard and bray curtis)\\n  2. PCoA collection (same as the distance matrices)\\n  3. Emperor plot collection (same as the distance matrices)\\n  4. Richness and evenness collection (rarefied table, faith pd vector observed features vector, shannon vector, evenness vector)\\n- get visualization for alpha diversity:\\n  - Pielou\'s eveness\\n  - Observed features\\n  - Shannons diversity index\\n- get visualization for beta diversity\\n  - Jaccard distance matrix\\n  - Bray curtis distance matrix\\n  - Unifrac distance metrix \\n\\n## Outputs\\n\\nIII-V:\\n\\n- Phylogenetic tree\\n- Rarefaction curve\\n- Taxonomic classification (as qza, barplot and table)\\n\\nVI:\\n\\n- Four collections containing: distance matrix, PCoA, Emperor plots, Richness and evenness\\n- Visualization for alpha diversity: Pielou\'s eveness, Observed features, Shannons diversity index\\n- Visualization for beta diversity: Jaccard, Bray curtis, Unifrac","changelog":"# Changelog\\n\\n## [0.2] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.1] - 2024-06-22\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Representative sequences\\"]@{ shape: doc }\\n1[\\"ℹ️ Feature table\\"]@{ shape: doc }\\n2[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n3[\\"ℹ️ Minimum depth\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Maximum depth\\"]@{ shape: lean-l }\\n5[\\"ℹ️ SEPP fragment insertion reference\\"]@{ shape: doc }\\n6[\\"ℹ️ Taxonomic classifier\\"]@{ shape: doc }\\n7[\\"🛠️ Phylogenetic tree for diversity analysis\\"]@{ shape: subprocess }\\n0 --> 7\\n5 --> 7\\n8[\\"🛠️ Taxonomic analysis\\"]@{ shape: subprocess }\\n1 --> 8\\n0 --> 8\\n2 --> 8\\n6 --> 8\\n9[\\"🛠️ Rarefaction\\"]@{ shape: subprocess }\\n1 --> 9\\n4 --> 9\\n2 --> 9\\n3 --> 9\\n7 --> 9\\n```\\n\\n## QIIME2 IV: Rarefaction\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n1[\\"ℹ️ DADA2 feature table\\"]@{ shape: doc }\\n2[\\"ℹ️ Rooted tree\\"]@{ shape: doc }\\n3[\\"ℹ️ Minimum depth\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Maximum depth\\"]@{ shape: lean-l }\\n5[\\"Alpha rarefaction\\"]@{ shape: process }\\n0 --> 5\\n3 --> 5\\n2 --> 5\\n4 --> 5\\n1 --> 5\\n```\\n\\n## QIIME2 V: Taxonomic analysis\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n1[\\"ℹ️ DADA2 representative sequences\\"]@{ shape: doc }\\n2[\\"ℹ️ Taxonomic classifier\\"]@{ shape: doc }\\n3[\\"ℹ️ DADA2 feature table\\"]@{ shape: doc }\\n4[\\"Taxonomy classification\\"]@{ shape: process }\\n2 --> 4\\n1 --> 4\\n5[\\"Taxonomy barplot\\"]@{ shape: process }\\n0 --> 5\\n4 --> 5\\n3 --> 5\\n6[\\"Tabulate taxonomy classification\\"]@{ shape: process }\\n4 --> 6\\n```\\n\\n## QIIME2 III: Phylogenetic tree for diversity analysis\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ DADA2 representative sequences\\"]@{ shape: doc }\\n1[\\"ℹ️ SEPP fragment insertion source file\\"]@{ shape: doc }\\n2[\\"Phylogenetic tree generation\\"]@{ shape: process }\\n1 --> 2\\n0 --> 2\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-III-V-Phylogeny-Rarefaction-Taxonomic-Analysis","dockstore_id":27882,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244085","tests":[{"doc":"Test outline for QIIME2-III:-Phylogenetic-tree-for-diversity-analysis","job":{"Metadata":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/pd-mice/sample_metadata.tsv","filetype":"tabular"},"Representative sequences":{"class":"File","location":"https://docs.qiime2.org/2024.2/data/tutorials/pd-mice/dada2_rep_set.qza","filetype":"qza"},"Feature table":{"class":"File","location":"https://docs.qiime2.org/2024.2/data/tutorials/pd-mice/dada2_table.qza","filetype":"qza"},"Minimum depth":1,"Maximum depth":2019,"Taxonomic classifier":{"class":"File","location":"https://docs.qiime2.org/2024.10/data/tutorials/pd-mice/gg-13-8-99-515-806-nb-classifier.qza","filetype":"qza"},"SEPP fragment insertion reference":{"class":"File","location":"https://data.qiime2.org/2024.2/common/sepp-refs-gg-13-8.qza","filetype":"qza"}},"outputs":{"Rooted tree":{"ftype":"qza","asserts":[{"has_size":{"min":"2M","max":"3M"}},{"has_archive_member":{"path":"^[^/]*/data/tree.nwk","n":1,"asserts":[{"has_text_matching":{"expression":"k__Bacteria"}}]}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_line":{"line":"type: Phylogeny[Rooted]"}},{"has_line":{"line":"format: NewickDirectoryFormat"}}]}}]},"Rarefaction curve":{"ftype":"qzv","asserts":[{"has_size":{"min":"400k","max":"500k"}},{"has_archive_member":{"path":"^[^/]*/data/index.html","n":1}},{"has_archive_member":{"path":"^[^/]*/data/.*\\\\.csv","n":3}},{"has_archive_member":{"path":"^[^/]*/data/.*\\\\.jsonp","n":21}}]},"Taxonomy classification":{"ftype":"qza","asserts":[{"has_size":{"min":"40k","max":"80k"}},{"has_archive_member":{"path":"^[^/]*/metadata.yaml","n":1,"asserts":[{"has_line":{"line":"type: FeatureData[Taxonomy]"}},{"has_line":{"line":"format: TSVTaxonomyDirectoryFormat"}}]}},{"has_archive_member":{"path":"^[^/]*/data/taxonomy.tsv","n":1,"asserts":[{"has_n_lines":{"n":288}}]}}]},"Taxa barplot":{"ftype":"qzv","asserts":[{"has_size":{"min":"400k","max":"500k"}},{"has_archive_member":{"path":"^[^/]*/data/.*\\\\.csv","n":7}},{"has_archive_member":{"path":"^[^/]*/data/.*\\\\.jsonp","n":7}}]},"Taxonomy classification table":{"ftype":"qza","asserts":[{"has_size":{"min":"1M","max":"2M"}},{"has_archive_member":{"path":"^[^/]*/data/metadata.tsv","n":1,"asserts":[{"has_n_lines":{"n":289}}]}}]}}}]},{"name":"QIIME2-VI-diversity-metrics-and-estimations","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/QIIME2-VI-diversity-metrics-and-estimations.ga","testParameterFiles":["/QIIME2-VI-diversity-metrics-and-estimations-tests.yml"],"authors":[{"name":"Debjyoti Ghosh","orcid":"0009-0008-1496-1677"},{"name":"Helmholtz-Zentrum für Umweltforschung - UFZ","address":"Permoserstraße 15, 04318 Leipzig"}],"definition":{"a_galaxy_workflow":"true","annotation":"The first step in hypothesis testing in microbial ecology is typically to look at within- (alpha) and between-sample (beta) diversity. We can calculate diversity metrics, apply appropriate statistical tests, and visualize the data using the q2-diversity plugin.","comments":[{"color":"none","data":{"size":1,"text":"From DADA2 table output or rarefaction"},"id":1,"position":[0,221.8],"size":[152,76],"type":"text"},{"color":"none","data":{"text":"Input files"},"id":0,"position":[136.6,3.8000000000000114],"size":[101,40],"type":"markdown"},{"color":"none","data":{"text":"Alpha diversity"},"id":3,"position":[844.9,0.7000000000000028],"size":[134,40],"type":"markdown"},{"color":"none","data":{"text":"Diversity metrics"},"id":2,"position":[513,1.8000000000000114],"size":[139,39],"type":"markdown"},{"color":"none","data":{"text":"Beta diversity"},"id":4,"position":[1184.8999999999999,0],"size":[120,39],"type":"markdown"},{"child_steps":[12,14,15,16],"color":"none","data":{"title":"Collections"},"id":5,"position":[794.4,1062.3],"size":[1263,352],"type":"frame"}],"creator":[{"class":"Person","identifier":"0009-0008-1496-1677","name":"Debjyoti Ghosh"},{"address":"Permoserstraße 15, 04318 Leipzig","class":"Organization","name":"Helmholtz-Zentrum für Umweltforschung - UFZ"}],"format-version":"0.1","license":"MIT","name":"QIIME2 VI: Diversity metrics and estimations","release":"0.2","steps":{"0":{"annotation":"This value is based on the DADA2 table features.\\nNote: the choice of the value is prioritising either the inclusion of all the samples or a higher feature retention.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"This value is based on the DADA2 table features.\\nNote: the choice of the value is prioritising either the inclusion of all the samples or a higher feature retention.","name":"Sampling depth"}],"label":"Sampling depth","name":"Input parameter","outputs":[],"position":{"left":134.6,"top":224.1206788736988},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a926d4ba-b238-4499-a12b-a95840970cef","when":null,"workflow_outputs":[]},"1":{"annotation":"The tab separated file for the metadata","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"The tab separated file for the metadata","name":"Metadata"}],"label":"Metadata","name":"Input dataset","outputs":[],"position":{"left":125.04715373257632,"top":312.2417148460719},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qiime2.tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8546caa2-9723-4c36-b693-988d43717ccf","when":null,"workflow_outputs":[]},"2":{"annotation":"For instance the dada2_table from DADA2","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"For instance the dada2_table from DADA2","name":"Feature table"}],"label":"Feature table","name":"Input dataset","outputs":[],"position":{"left":131.65948794697385,"top":519.9723458836958},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"f8b970d0-8396-4793-8131-36cc24fcc630","when":null,"workflow_outputs":[]},"3":{"annotation":"Rooted phylogenetic tree","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Rooted phylogenetic tree","name":"Rooted tree"}],"label":"Rooted tree","name":"Input dataset","outputs":[],"position":{"left":131.88431251326813,"top":637.5877672262973},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"qza\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"0bbab7fb-adb4-45b1-926e-3183f6708658","when":null,"workflow_outputs":[]},"4":{"annotation":"Type in the name of the parameter which would be used for beta diversity estimation","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Type in the name of the parameter which would be used for beta diversity estimation","name":"Target metadata parameter (for beta diversity)"}],"label":"Target metadata parameter (for beta diversity)","name":"Input parameter","outputs":[],"position":{"left":132.27305627796053,"top":773.5167861361558},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"63ee0d89-71cd-4488-8c14-776f44470aba","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","errors":null,"id":5,"input_connections":{"import_root|__q2galaxy__GUI__cond__format__|import|data":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"qiime2 tools import","outputs":[{"name":"imported_data","type":"qza"}],"position":{"left":183.95541284754188,"top":384.72561308872656},"post_job_actions":{"HideDatasetActionimported_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"imported_data"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27","tool_shed_repository":{"changeset_revision":"8a0db69428c7","name":"qiime2_core__tools__import","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"import_root\\": {\\"type\\": \\"ImmutableMetadata\\", \\"__current_case__\\": 58, \\"__q2galaxy__GUI__cond__format__\\": {\\"format\\": \\"ImmutableMetadataFormat\\", \\"__current_case__\\": 0, \\"import\\": {\\"name\\": \\"__q2galaxy__::literal::None\\", \\"data\\": {\\"__class__\\": \\"ConnectedValue\\"}}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+dist.h3d8a7e27","type":"tool","uuid":"be1fae33-1b8d-4756-854f-a0587934da09","when":null,"workflow_outputs":[]},"6":{"annotation":"QIIME 2’s diversity analyses are available through the q2-diversity plugin, which supports computing alpha and beta diversity metrics, applying related statistical tests, and generating interactive visualizations. We’ll first apply the core-metrics-phylogenetic method, which rarefies a FeatureTable[Frequency] to a user-specified depth, computes several alpha and beta diversity metrics, and generates principle coordinates analysis (PCoA) plots using Emperor for each of the beta diversity metrics.","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":6,"input_connections":{"metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":1,"output_name":"output"},"phylogeny":{"id":3,"output_name":"output"},"sampling_depth":{"id":0,"output_name":"output"},"table":{"id":2,"output_name":"output"}},"inputs":[],"label":"Diversity metrics","name":"qiime2 diversity core-metrics-phylogenetic","outputs":[{"name":"rarefied_table","type":"qza"},{"name":"faith_pd_vector","type":"qza"},{"name":"observed_features_vector","type":"qza"},{"name":"shannon_vector","type":"qza"},{"name":"evenness_vector","type":"qza"},{"name":"unweighted_unifrac_distance_matrix","type":"qza"},{"name":"weighted_unifrac_distance_matrix","type":"qza"},{"name":"jaccard_distance_matrix","type":"qza"},{"name":"bray_curtis_distance_matrix","type":"qza"},{"name":"unweighted_unifrac_pcoa_results","type":"qza"},{"name":"weighted_unifrac_pcoa_results","type":"qza"},{"name":"jaccard_pcoa_results","type":"qza"},{"name":"bray_curtis_pcoa_results","type":"qza"},{"name":"unweighted_unifrac_emperor","type":"qzv"},{"name":"weighted_unifrac_emperor","type":"qzv"},{"name":"jaccard_emperor","type":"qzv"},{"name":"bray_curtis_emperor","type":"qzv"}],"position":{"left":512.5765871918118,"top":81.4},"post_job_actions":{"HideDatasetActionbray_curtis_distance_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bray_curtis_distance_matrix"},"HideDatasetActionbray_curtis_emperor":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bray_curtis_emperor"},"HideDatasetActionbray_curtis_pcoa_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bray_curtis_pcoa_results"},"HideDatasetActionevenness_vector":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"evenness_vector"},"HideDatasetActionfaith_pd_vector":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"faith_pd_vector"},"HideDatasetActionjaccard_distance_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"jaccard_distance_matrix"},"HideDatasetActionjaccard_emperor":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"jaccard_emperor"},"HideDatasetActionjaccard_pcoa_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"jaccard_pcoa_results"},"HideDatasetActionobserved_features_vector":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"observed_features_vector"},"HideDatasetActionrarefied_table":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"rarefied_table"},"HideDatasetActionshannon_vector":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"shannon_vector"},"HideDatasetActionunweighted_unifrac_distance_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"unweighted_unifrac_distance_matrix"},"HideDatasetActionunweighted_unifrac_emperor":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"unweighted_unifrac_emperor"},"HideDatasetActionunweighted_unifrac_pcoa_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"unweighted_unifrac_pcoa_results"},"HideDatasetActionweighted_unifrac_distance_matrix":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"weighted_unifrac_distance_matrix"},"HideDatasetActionweighted_unifrac_emperor":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"weighted_unifrac_emperor"},"HideDatasetActionweighted_unifrac_pcoa_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"weighted_unifrac_pcoa_results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"7fc930fc43cf","name":"qiime2__diversity__core_metrics_phylogenetic","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"with_replacement\\": false, \\"ignore_missing_samples\\": false}, \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"phylogeny\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampling_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"cd47d601-3186-40c0-bd1a-1564728a776d","when":null,"workflow_outputs":[]},"7":{"annotation":"qiime2 diversity alpha-group-significance\\nAlpha diversity comparisons","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":7,"input_connections":{"alpha_diversity":{"id":6,"output_name":"evenness_vector"},"metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":1,"output_name":"output"}},"inputs":[],"label":"Alpha diversity metrics - Pielou\'s evenness","name":"qiime2 diversity alpha-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":846.7461791992188,"top":80.1315673828125},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"alpha_diversity_pielou"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"96988c25ca01","name":"qiime2__diversity__alpha_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alpha_diversity\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"b1e9c678-9093-4e47-a860-89a0ce54e658","when":null,"workflow_outputs":[{"label":"Alpha diversity - Pielou\'s evenness","output_name":"visualization","uuid":"ec1ab736-8fd7-41b6-b554-38195c2ec954"}]},"8":{"annotation":"qiime2 diversity alpha-group-significance\\nAlpha diversity comparisons","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":8,"input_connections":{"alpha_diversity":{"id":6,"output_name":"observed_features_vector"},"metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":1,"output_name":"output"}},"inputs":[],"label":"Alpha diversity metrics - Observed features","name":"qiime2 diversity alpha-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":846.8121938591562,"top":402.04402416735945},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"alpha_diversity_observed"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"96988c25ca01","name":"qiime2__diversity__alpha_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alpha_diversity\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"867c60b4-2a64-4b17-82ce-524dcd8b3362","when":null,"workflow_outputs":[{"label":"Alpha diversity - Observed features","output_name":"visualization","uuid":"62b5bb9a-75c4-4962-815e-fe03e3b4ec59"}]},"9":{"annotation":"qiime2 diversity alpha-group-significance\\nAlpha diversity comparisons","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":9,"input_connections":{"alpha_diversity":{"id":6,"output_name":"shannon_vector"},"metadata_0|__q2galaxy__GUI__conditional__metadata__|source":{"id":1,"output_name":"output"}},"inputs":[],"label":"Alpha diversity metrics - Shannon\'s diversity index","name":"qiime2 diversity alpha-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":842.9852181246822,"top":734.0335085712542},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"alpha_diversity_shannon"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"96988c25ca01","name":"qiime2__diversity__alpha_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alpha_diversity\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": [{\\"__index__\\": 0, \\"__q2galaxy__GUI__conditional__metadata__\\": {\\"type\\": \\"tsv\\", \\"__current_case__\\": 0, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"6b2cd5b5-982b-42a2-b8c3-67d768219937","when":null,"workflow_outputs":[{"label":"Alpha diversity - Shannon\'s diversity index","output_name":"visualization","uuid":"077d6bde-1788-4d4f-b3f4-164c04d52c5b"}]},"10":{"annotation":"qiime2 diversity beta-group-significance\\nBeta diversity group","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":10,"input_connections":{"distance_matrix":{"id":6,"output_name":"jaccard_distance_matrix"},"metadata|column":{"id":4,"output_name":"output"},"metadata|source":{"id":5,"output_name":"imported_data"}},"inputs":[{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"},{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"}],"label":"Beta diversity - Jaccard distance matrix","name":"qiime2 diversity beta-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":1183.3696675834228,"top":79.0184079082341},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"beta_diversity_jaccard"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"a0d20b8b681e","name":"qiime2__diversity__beta_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"method\\": \\"permanova\\", \\"pairwise\\": false, \\"permutations\\": \\"999\\"}, \\"distance_matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"df0800f4-d40f-4245-b15f-e7c35e434f41","when":null,"workflow_outputs":[{"label":"Beta diversity - Jaccard distance metrics","output_name":"visualization","uuid":"d95ad0c9-e953-4328-94ee-db4cd2457931"}]},"11":{"annotation":"qiime2 diversity beta-group-significance\\nBeta diversity group","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":11,"input_connections":{"distance_matrix":{"id":6,"output_name":"bray_curtis_distance_matrix"},"metadata|column":{"id":4,"output_name":"output"},"metadata|source":{"id":5,"output_name":"imported_data"}},"inputs":[{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"},{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"}],"label":"Beta diversity - Bray-Curtis distance matrix","name":"qiime2 diversity beta-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":1181.5534417354834,"top":399.32788547201847},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"beta_diversity_bray_curtis"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"a0d20b8b681e","name":"qiime2__diversity__beta_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"method\\": \\"permanova\\", \\"pairwise\\": false, \\"permutations\\": \\"999\\"}, \\"distance_matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"7ca4a86e-7a49-4d04-b52e-10c375d856dc","when":null,"workflow_outputs":[{"label":"Beta diversity - Bray-Curtis distance metrics","output_name":"visualization","uuid":"32ed3c8e-eb1f-4382-a75f-de9a56c2ddbb"}]},"12":{"annotation":"Creating a collection for the emperor plots created from core-metrics","content_id":"__BUILD_LIST__","errors":null,"id":12,"input_connections":{"datasets_0|input":{"id":6,"output_name":"unweighted_unifrac_emperor"},"datasets_1|input":{"id":6,"output_name":"weighted_unifrac_emperor"},"datasets_2|input":{"id":6,"output_name":"jaccard_emperor"},"datasets_3|input":{"id":6,"output_name":"bray_curtis_emperor"}},"inputs":[],"label":"Emperor plot collection","name":"Build list","outputs":[{"name":"output","type":"input"}],"position":{"left":816.9327117801281,"top":1111.237105588374},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"collection_emperor"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__BUILD_LIST__","tool_state":"{\\"datasets\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"unweighted_unifrac\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"weighted_unifrac\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"jaccard\\"}}, {\\"__index__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"bray_curtis\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"5fa94d54-8e21-4f2d-89f5-5953b32d3fed","when":null,"workflow_outputs":[{"label":"Emperor plot collection","output_name":"output","uuid":"c9c67cb8-7b03-47c6-9672-8e08788b6b4f"}]},"13":{"annotation":"qiime2 diversity beta-group-significance\\nBeta diversity group","content_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","errors":null,"id":13,"input_connections":{"distance_matrix":{"id":6,"output_name":"weighted_unifrac_distance_matrix"},"metadata|column":{"id":4,"output_name":"output"},"metadata|source":{"id":5,"output_name":"imported_data"}},"inputs":[{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"},{"description":"runtime parameter for tool qiime2 diversity beta-group-significance","name":"metadata"}],"label":"Beta diversity - weighted UniFrac distance matrix","name":"qiime2 diversity beta-group-significance","outputs":[{"name":"visualization","type":"qzv"}],"position":{"left":1189.8275957319759,"top":730.5232235877588},"post_job_actions":{"RenameDatasetActionvisualization":{"action_arguments":{"newname":"beta_diversity_weighted_unifrac"},"action_type":"RenameDatasetAction","output_name":"visualization"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0","tool_shed_repository":{"changeset_revision":"a0d20b8b681e","name":"qiime2__diversity__beta_group_significance","owner":"q2d2","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__q2galaxy__GUI__section__extra_opts__\\": {\\"method\\": \\"permanova\\", \\"pairwise\\": false, \\"permutations\\": \\"999\\"}, \\"distance_matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"metadata\\": {\\"type\\": \\"qza\\", \\"__current_case__\\": 1, \\"source\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2024.10.0+q2galaxy.2024.10.0","type":"tool","uuid":"e16130db-b499-4d24-8950-b3779969258d","when":null,"workflow_outputs":[{"label":"Beta diversity - weighted UniFrac distance metrics","output_name":"visualization","uuid":"2bd1213c-b75b-45c4-b8e2-69a2b0d7a000"}]},"14":{"annotation":"Creating a collection for the pcoa indices created from core-metrics","content_id":"__BUILD_LIST__","errors":null,"id":14,"input_connections":{"datasets_0|input":{"id":6,"output_name":"bray_curtis_pcoa_results"},"datasets_1|input":{"id":6,"output_name":"weighted_unifrac_pcoa_results"},"datasets_2|input":{"id":6,"output_name":"unweighted_unifrac_pcoa_results"},"datasets_3|input":{"id":6,"output_name":"jaccard_pcoa_results"}},"inputs":[],"label":"PCoA collection","name":"Build list","outputs":[{"name":"output","type":"input"}],"position":{"left":1154.1830644819759,"top":1143.3454485226546},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"collection_pcoa"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__BUILD_LIST__","tool_state":"{\\"datasets\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"unweighted_unifrac\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"weighted_unifrac\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"jaccard\\"}}, {\\"__index__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"bray_curtis\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"ecfc4438-5355-436b-8508-8bd4bfcb03ba","when":null,"workflow_outputs":[{"label":"PCoA collection","output_name":"output","uuid":"96716878-d1c1-4dd1-9acd-d4cdc736f884"}]},"15":{"annotation":"Creating a collection for distance matrix created from core-metrics","content_id":"__BUILD_LIST__","errors":null,"id":15,"input_connections":{"datasets_0|input":{"id":6,"output_name":"bray_curtis_distance_matrix"},"datasets_1|input":{"id":6,"output_name":"weighted_unifrac_distance_matrix"},"datasets_2|input":{"id":6,"output_name":"unweighted_unifrac_distance_matrix"},"datasets_3|input":{"id":6,"output_name":"jaccard_distance_matrix"}},"inputs":[],"label":"Distance matrix collection","name":"Build list","outputs":[{"name":"output","type":"input"}],"position":{"left":1467.0719804976009,"top":1126.3676653195296},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"collection_distance_matrix"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__BUILD_LIST__","tool_state":"{\\"datasets\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"unweighted_unifrac\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"weighted_unifrac\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"jaccard\\"}}, {\\"__index__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"bray_curtis\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"9a17d575-227f-4f98-a949-d5d90e733a41","when":null,"workflow_outputs":[{"label":"Distance matrix collection","output_name":"output","uuid":"81caf090-13e7-4304-8348-4cf87352fb30"}]},"16":{"annotation":"Creating a collection for indices of community richness and evenness created from core-metrics","content_id":"__BUILD_LIST__","errors":null,"id":16,"input_connections":{"datasets_0|input":{"id":6,"output_name":"rarefied_table"},"datasets_1|input":{"id":6,"output_name":"faith_pd_vector"},"datasets_2|input":{"id":6,"output_name":"observed_features_vector"},"datasets_3|input":{"id":6,"output_name":"evenness_vector"},"datasets_4|input":{"id":6,"output_name":"shannon_vector"}},"inputs":[],"label":"Richness and evenness collection","name":"Build list","outputs":[{"name":"output","type":"input"}],"position":{"left":1788.2942298465593,"top":1104.9898983924463},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"collection_richness_evenness"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__BUILD_LIST__","tool_state":"{\\"datasets\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"rarefied_table\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"faith_pd_vector\\"}}, {\\"__index__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"observed_features_vector\\"}}, {\\"__index__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"shannon_vector\\"}}, {\\"__index__\\": 4, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"manual\\", \\"__current_case__\\": 2, \\"identifier\\": \\"eveness_vector\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"032f5d65-2ed6-47b2-8837-25aa360fd355","when":null,"workflow_outputs":[{"label":"Richness and evenness collection","output_name":"output","uuid":"b17fd336-7d7c-4b1f-a0ef-0337323ba9d5"}]}},"tags":[],"uuid":"300b5a7d-f763-486a-a8ea-87ea84571f19","version":2},"readme":"# QIIME2 workflows\\n\\n## Available workflows\\n\\n- III-V Downstream analyses: III) reconstruct a taxonomy for diversity analysis, IV) rarefaction analysis, V) taxonomic analysis.\\n- VI: Computation of diversity metrics and estimations\\n\\nAnalogous to the procedures described in the Parkinson’s Mouse Tutorial: https://docs.qiime2.org/2024.5/tutorials/pd-mice/\\n\\n## Inputs\\n\\nThe two workflows have two inputs in common \\n\\n- Feature table: Count data\\n- Metadata: Metadata table\\n\\nand the following extra inputs\\n\\nIII-V\\n\\n- Representative sequences: Representative (ASV) sequences\\n- Minimum depth: Lower limit of the sampling depth for the alpha rarefaction analysis\\n- Maximum depth: Upper limit of the sampling depth for the alpha rarefaction analysis\\n- SEPP fragment insertion reference: used for the reconstruction of the phylogenetic tree\\n- Taxonomic classifier: The classifier to assign taxonomic information to the ASVs\\n\\nVI:\\n\\nSampling depth: For the metric calculation (should be based on the rarefaction analysis done in IV)\\nTarget metadata parameter: that should be used for beta diversity calculations\\nRooted Tree: for instance the tree computed in III\\n\\n## Processing\\n\\nIII-V\\n\\n- Phylogenetic tree generation using `qiime2 fragment-insertion sepp`\\n- Alpha rarefaction analysis using `qiime2 diversity alpha-rarefaction`\\n- Taxonomic classification using `qiime2 feature-classifier classify-sklearn` and compute barplot and tabular output\\n\\nVI: \\n\\n- compute alpha and beta diversity metrics using `qiime2 diversity core-metrics-phylogenetic`\\n- organize these metrics in 4 collections:\\n  1. Distance matrix collection (weighted and unweighted unifrac, jaccard and bray curtis)\\n  2. PCoA collection (same as the distance matrices)\\n  3. Emperor plot collection (same as the distance matrices)\\n  4. Richness and evenness collection (rarefied table, faith pd vector observed features vector, shannon vector, evenness vector)\\n- get visualization for alpha diversity:\\n  - Pielou\'s eveness\\n  - Observed features\\n  - Shannons diversity index\\n- get visualization for beta diversity\\n  - Jaccard distance matrix\\n  - Bray curtis distance matrix\\n  - Unifrac distance metrix \\n\\n## Outputs\\n\\nIII-V:\\n\\n- Phylogenetic tree\\n- Rarefaction curve\\n- Taxonomic classification (as qza, barplot and table)\\n\\nVI:\\n\\n- Four collections containing: distance matrix, PCoA, Emperor plots, Richness and evenness\\n- Visualization for alpha diversity: Pielou\'s eveness, Observed features, Shannons diversity index\\n- Visualization for beta diversity: Jaccard, Bray curtis, Unifrac","changelog":"# Changelog\\n\\n## [0.2] - 2024-11-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__fragment_insertion__sepp/qiime2__fragment_insertion__sepp/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__feature_classifier__classify_sklearn/qiime2__feature_classifier__classify_sklearn/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__taxa__barplot/qiime2__taxa__barplot/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2023.5.0+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__metadata__tabulate/qiime2__metadata__tabulate/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_rarefaction/qiime2__diversity__alpha_rarefaction/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2023.5.0+dist.h193f7cc9.3` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2_core__tools__import/qiime2_core__tools__import/2024.10.0+dist.h3d8a7e27`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__core_metrics_phylogenetic/qiime2__diversity__core_metrics_phylogenetic/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__alpha_group_significance/qiime2__diversity__alpha_group_significance/2024.10.0+q2galaxy.2024.10.0`\\n- `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2023.5.1+q2galaxy.2023.5.0.2` was updated to `toolshed.g2.bx.psu.edu/repos/q2d2/qiime2__diversity__beta_group_significance/qiime2__diversity__beta_group_significance/2024.10.0+q2galaxy.2024.10.0`\\n\\n## [0.1] - 2024-06-22\\n","diagrams":"# Workflow diagrams\\n\\n## QIIME2 VI: Diversity metrics and estimations\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Sampling depth\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Metadata\\"]@{ shape: doc }\\n2[\\"ℹ️ Feature table\\"]@{ shape: doc }\\n3[\\"ℹ️ Rooted tree\\"]@{ shape: doc }\\n4[\\"ℹ️ Target metadata parameter (for beta diversity)\\"]@{ shape: lean-l }\\n5[\\"qiime2 tools import\\"]@{ shape: process }\\n1 --> 5\\n6[\\"Diversity metrics\\"]@{ shape: process }\\n1 --> 6\\n3 --> 6\\n0 --> 6\\n2 --> 6\\n7[\\"Alpha diversity metrics - Pielou\'s evenness\\"]@{ shape: process }\\n6 --> 7\\n1 --> 7\\n8[\\"Alpha diversity metrics - Observed features\\"]@{ shape: process }\\n6 --> 8\\n1 --> 8\\n9[\\"Alpha diversity metrics - Shannon\'s diversity index\\"]@{ shape: process }\\n6 --> 9\\n1 --> 9\\n10[\\"Beta diversity - Jaccard distance matrix\\"]@{ shape: process }\\n6 --> 10\\n4 --> 10\\n5 --> 10\\n11[\\"Beta diversity - Bray-Curtis distance matrix\\"]@{ shape: process }\\n6 --> 11\\n4 --> 11\\n5 --> 11\\n12[\\"Emperor plot collection\\"]@{ shape: process }\\n6 --> 12\\n6 --> 12\\n6 --> 12\\n6 --> 12\\n13[\\"Beta diversity - weighted UniFrac distance matrix\\"]@{ shape: process }\\n6 --> 13\\n4 --> 13\\n5 --> 13\\n14[\\"PCoA collection\\"]@{ shape: process }\\n6 --> 14\\n6 --> 14\\n6 --> 14\\n6 --> 14\\n15[\\"Distance matrix collection\\"]@{ shape: process }\\n6 --> 15\\n6 --> 15\\n6 --> 15\\n6 --> 15\\n16[\\"Richness and evenness collection\\"]@{ shape: process }\\n6 --> 16\\n6 --> 16\\n6 --> 16\\n6 --> 16\\n6 --> 16\\n```\\n","updated":"2024-11-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/qiime2-III-VI-downsteam/QIIME2-VI-diversity-metrics-and-estimations","dockstore_id":27884,"categories":[],"collections":["Metabarcoding"],"doi":"10.5281/zenodo.14244089","tests":[{"doc":"Test outline for QIIME2-VI:-Diversity-metrics-and-estimations","job":{"Metadata":{"class":"File","location":"https://data.qiime2.org/2024.2/tutorials/pd-mice/sample_metadata.tsv","filetype":"qiime2.tabular"},"Feature table":{"class":"File","location":"https://docs.qiime2.org/2024.2/data/tutorials/pd-mice/dada2_table.qza","filetype":"qza"},"Rooted tree":{"class":"File","location":"https://docs.qiime2.org/2024.2/data/tutorials/pd-mice/tree.qza","filetype":"qza"},"Sampling depth":"2000","Target metadata parameter (for beta diversity)":"cage_id"},"outputs":{"Alpha diversity - Pielou\'s evenness":{"ftype":"qzv","asserts":[{"has_size":{"min":"300k","max":"400k"}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.csv","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.jsonp","n":6}}]},"Alpha diversity - Shannon\'s diversity index":{"ftype":"qzv","asserts":[{"has_size":{"min":"300k","max":"400k"}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.csv","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.jsonp","n":6}}]},"Alpha diversity - Observed features":{"ftype":"qzv","asserts":[{"has_size":{"min":"300k","max":"400k"}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.csv","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.jsonp","n":6}}]},"Beta diversity - Bray-Curtis distance metrics":{"ftype":"qzv","asserts":[{"has_size":{"min":"400k","max":"500k"}},{"has_archive_member":{"path":"^[^/]*/data/raw_data\\\\.tsv","n":1}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.png","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.pdf","n":6}}]},"Beta diversity - Jaccard distance metrics":{"ftype":"qzv","asserts":[{"has_size":{"min":"400k","max":"500k"}},{"has_archive_member":{"path":"^[^/]*/data/raw_data\\\\.tsv","n":1}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.png","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.pdf","n":6}}]},"Beta diversity - weighted UniFrac distance metrics":{"ftype":"qzv","asserts":[{"has_size":{"min":"400k","max":"500k"}},{"has_archive_member":{"path":"^[^/]*/data/raw_data\\\\.tsv","n":1}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.png","n":6}},{"has_archive_member":{"path":"^[^/]*/data/[^/]*\\\\.pdf","n":6}}]},"Emperor plot collection":{"element_tests":{"unweighted_unifrac":{"ftype":"qzv"},"weighted_unifrac":{"ftype":"qzv"},"jaccard":{"ftype":"qzv"},"bray_curtis":{"ftype":"qzv"}}},"Distance matrix collection":{"element_tests":{"unweighted_unifrac":{"ftype":"qzv"},"weighted_unifrac":{"ftype":"qzv"},"jaccard":{"ftype":"qzv"},"bray_curtis":{"ftype":"qzv"}}},"PCoA collection":{"element_tests":{"unweighted_unifrac":{"ftype":"qzv"},"weighted_unifrac":{"ftype":"qzv"},"jaccard":{"ftype":"qzv"},"bray_curtis":{"ftype":"qzv"}}},"Richness and evenness collection":{"element_tests":{"rarefied_table":{"ftype":"qzv"},"faith_pd_vector":{"ftype":"qzv"},"observed_features_vector":{"ftype":"qzv"},"shannon_vector":{"ftype":"qzv"},"eveness_vector":{"ftype":"qzv"}}}}}]}],"path":"./workflows/amplicon/qiime2/qiime2-III-VI-downsteam"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Assembly-Hifi-only-VGP3.ga","testParameterFiles":["/Assembly-Hifi-only-VGP3-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.2.8","name":"Assembly-Hifi-only-VGP3","report":{"markdown":"\\n# Workflow Execution Report\\n\\nTime workflow was invoked:\\n\\n```galaxy\\ninvocation_time()\\n```\\nGalaxy version :\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Raw unitig graph\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"raw unitig graph image\\")\\n```\\n\\n## Merqury results\\n\\n### Merqury QV\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_qv\\")\\n```\\n\\n### Merqury completeness\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_stats\\")\\n```\\n\\n### Merqury plots\\n\\n\\nspectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-cn.fl\\")\\n```\\n\\nspectra-asm:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-asm.fl\\")\\n```\\n\\nhap1 spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_01.spectra-cn.fl\\")\\n```\\n\\n\\n## BUSCO results (Vertebrata database)\\n\\n\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Pacbio Reads Collection"}],"label":"Pacbio Reads Collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":549.7803459425681},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"188272eb-651c-4cad-9cdc-57dc6425947d","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Meryl Database"}],"label":"Meryl Database","name":"Input dataset","outputs":[],"position":{"left":95.49981346886014,"top":713.083417070356},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a3b31a30-b60f-4e4a-96d1-f4c0513ef476","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Genomescope Summary"}],"label":"Genomescope Summary","name":"Input dataset","outputs":[],"position":{"left":178.7993156668467,"top":854.7276031666358},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"705c22f0-68e3-4f5e-93da-e1dda2a80f98","when":null,"workflow_outputs":[]},"3":{"annotation":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","name":"Bits for Hifiasm bloom filter"}],"label":"Bits for Hifiasm bloom filter","name":"Input parameter","outputs":[],"position":{"left":272.75742759831104,"top":981.8662360058211},"tool_id":null,"tool_state":"{\\"default\\": 37, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"3ce6ecc2-3262-4695-b4ef-1a58f52a48bd","when":null,"workflow_outputs":[]},"4":{"annotation":"If empty, read coverage will be estimated from the Genomescope parameters. ","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"If empty, read coverage will be estimated from the Genomescope parameters. ","name":"Homozygous Read Coverage"}],"label":"Homozygous Read Coverage","name":"Input parameter","outputs":[],"position":{"left":330.6677396409291,"top":1112.6998884025538},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1e16198c-a515-4b6d-9a72-d1de93974d2b","when":null,"workflow_outputs":[]},"5":{"annotation":"GenomeScope model parameters generated by K-mer profiling workflow","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"GenomeScope model parameters generated by K-mer profiling workflow","name":"Genomescope Model Parameters"}],"label":"Genomescope Model Parameters","name":"Input dataset","outputs":[],"position":{"left":376.4742856427239,"top":1238.8834014045185},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"eb7272f6-9895-42b9-9a89-79ad3f342588","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":422.96681360823476,"top":1352.5435082963184},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"af7ad6ad-2718-4a16-ad1f-6858b1b4fb50","when":null,"workflow_outputs":[]},"7":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":500.2702810213637,"top":1472.6772609095954},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"df030d72-fca9-4b7e-b58b-dcd27c453543","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"","name":"SAK input file (Optional)"}],"label":"SAK input file (Optional)","name":"Input dataset","outputs":[],"position":{"left":2169.708999517185,"top":982.8125},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"b7958e14-2495-4d38-8da1-ea583f5ae8e2","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":3650.0963286187475,"top":1145.9790267944336},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"163c033f-85ce-46e5-b00a-3b3904dea278","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":3688.0509184624975,"top":1293.919334411621},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"88e40054-8605-4746-b91b-38ff85c70fd6","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":11,"input_connections":{"library|input_1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":null,"name":"Cutadapt","outputs":[{"name":"out1","type":"fastqsanger"},{"name":"report","type":"txt"},{"name":"json_stats","type":"json"}],"position":{"left":620.9277495171852,"top":992.890625},"post_job_actions":{"HideDatasetActionjson_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"json_stats"},"TagDatasetActionout1":{"action_arguments":{"tags":"trimmed_"},"action_type":"TagDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"35\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": true}, \\"filter_options\\": {\\"discard_trimmed\\": true, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"1\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [], \\"front_adapters\\": [], \\"anywhere_adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT\\"}, \\"single_noindels\\": false}, {\\"__index__\\": 1, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCTTTTCCTCCTCCTCCGTTGTTGTTGTTGAGAGAGAT\\"}, \\"single_noindels\\": false}]}}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"0\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\", \\"json_stats\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"f679b5eb-cc26-4005-9499-3ad2eb4698d4","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":12,"input_connections":{"infile":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1323.7328032281227,"top":1528.905174255371},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-G\\", \\"url_paste\\": \\"Haploid\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"620435fe-f48b-4867-9bc2-60655b153b2f","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":13,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1541.8138037091583,"top":1164.7394625964785},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": \\"37\\"}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"ef31ebb9-5de0-4c65-a3c7-30624dbddc05","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":14,"input_connections":{"input":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":589.7410652330907,"top":1966.7977210262934},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c3*2\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"0203b86a-9457-48fc-87af-e251b24bad2b","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":15,"input_connections":{"results_0|software_cond|input":{"id":11,"output_name":"report"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1078.5644682671852,"top":1052.86328125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"24ef7e9a-d7eb-4cda-8675-26e8ea167b54","when":null,"workflow_outputs":[{"label":"multiqc html report","output_name":"html_report","uuid":"86a436db-4089-46e3-ad77-c6247302f4ce"},{"label":"cutadapt multiqc stats","output_name":"stats","uuid":"c274f7aa-095e-4245-b6aa-cfadcf0d1736"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":16,"input_connections":{"infile":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1466.7015532281227,"top":1363.5642318725586},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"bp\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 1, \\"find_pattern\\": \\",\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 2, \\"find_pattern\\": \\"([a-z])\\\\\\\\s+([A-Z])\\", \\"replace_pattern\\": \\"\\\\\\\\1_\\\\\\\\2\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"2d652b72-7e96-48d0-9d65-efc08a749eb2","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"Cut1","errors":null,"id":17,"input_connections":{"input":{"id":14,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":749.523911043862,"top":1812.7203552573856},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fb6d8a81-7921-4b8b-be3c-e6f37772caa9","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"Convert characters1","errors":null,"id":18,"input_connections":{"input":{"id":16,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Convert","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1619.6987017419167,"top":1546.9875717163086},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Convert characters1","tool_state":"{\\"condense\\": true, \\"convert_from\\": \\"s\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strip\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ebf57da9-a17b-423c-a16e-3c05a0de5a3b","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":19,"input_connections":{"input1":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":969.434668724552,"top":1963.4408548125773},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated homozygous read coverage"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"65b07e97-c8ea-4fbb-b37e-aab4c93c61fa","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"Cut1","errors":null,"id":20,"input_connections":{"input":{"id":18,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1774.698699834568,"top":1332.044334411621},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"7486cd7c-ff1e-4d8f-a793-7d107e053c2b","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":21,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":4,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":19,"output_name":"integer_param"}},"inputs":[],"label":"Homozygous read coverage for Hifiasm","name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1113.916320765605,"top":1651.3174976390737},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Homozygous read coverage for Hifiasm"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"ddb202e3-53f2-406b-96eb-f8a8c442f8da","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":22,"input_connections":{"input1":{"id":20,"output_name":"out_file1"}},"inputs":[],"label":"Estimated genome size","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":2059.1305083062475,"top":1149.6296005249023},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated Genome size"},"action_type":"RenameDatasetAction","output_name":"integer_param"},"TagDatasetActioninteger_param":{"action_arguments":{"tags":"estimated_genome_size"},"action_type":"TagDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"2915bf7f-75c0-49cc-8f6f-af13b4510f4e","when":null,"workflow_outputs":[{"label":"Estimated Genome size","output_name":"integer_param","uuid":"28396392-6855-400b-a28c-6297d43b7e23"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","errors":null,"id":23,"input_connections":{"assembly_options|hom_cov":{"id":21,"output_name":"integer_param"},"filter_bits":{"id":13,"output_name":"integer_param"},"mode|reads":{"id":11,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool Hifiasm","name":"assembly_options"},{"description":"runtime parameter for tool Hifiasm","name":"mode"}],"label":null,"name":"Hifiasm","outputs":[{"name":"noseq_files","type":"input"},{"name":"raw_unitigs","type":"gfa1"},{"name":"processed_unitigs","type":"gfa1"},{"name":"primary_contig_graph","type":"gfa1"},{"name":"alternate_contig_graph","type":"gfa1"},{"name":"log_file","type":"txt"}],"position":{"left":1990.2952727105446,"top":516.4193801879883},"post_job_actions":{"TagDatasetActionalternate_contig_graph":{"action_arguments":{"tags":"hifiasm_alternate_gfa"},"action_type":"TagDatasetAction","output_name":"alternate_contig_graph"},"TagDatasetActionnoseq files":{"action_arguments":{"tags":"noseq_hifiasm"},"action_type":"TagDatasetAction","output_name":"noseq files"},"TagDatasetActionprimary_contig_graph":{"action_arguments":{"tags":"hifiasm_primary_gfa"},"action_type":"TagDatasetAction","output_name":"primary_contig_graph"},"TagDatasetActionprocessed_unitigs":{"action_arguments":{"tags":"hifiasm_processed_gfa"},"action_type":"TagDatasetAction","output_name":"processed_unitigs"},"TagDatasetActionraw_unitigs":{"action_arguments":{"tags":"hifiasm_raw_gfa"},"action_type":"TagDatasetAction","output_name":"raw_unitigs"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","tool_shed_repository":{"changeset_revision":"170061c052ad","name":"hifiasm","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"advanced_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"hifiasm_kmer_length\\": \\"51\\", \\"window_size\\": \\"51\\", \\"drop_kmers\\": \\"5.0\\", \\"max_overlaps\\": \\"100\\", \\"correction_rounds\\": \\"3\\", \\"min_hist_cnt\\": null, \\"max_kocc\\": \\"20000\\", \\"hg_size\\": \\"\\"}, \\"assembly_options\\": {\\"assembly_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"cleaning_rounds\\": \\"4\\", \\"adapter_length\\": \\"0\\", \\"pop_contigs\\": \\"10000000\\", \\"pop_unitigs\\": \\"100000\\", \\"remove_tips\\": \\"3\\", \\"max_overlap\\": \\"0.8\\", \\"min_overlap\\": \\"0.2\\", \\"disable_post_join\\": false, \\"ignore_error_corrected\\": false, \\"hom_cov\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"bins_out\\": false, \\"filter_bits\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hic_partition\\": {\\"hic_partition_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"log_out\\": true, \\"mode\\": {\\"mode_selector\\": \\"standard\\", \\"__current_case__\\": 0, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"ont_integration\\": {\\"ont_integration_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"purge_options\\": {\\"purge_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"purge_level\\": \\"0\\", \\"similarity_threshold\\": \\"0.75\\", \\"minimum_overlap\\": \\"1\\", \\"purge_max\\": null, \\"n_hap\\": null}, \\"scaffolding_options\\": {\\"scaffold_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy0","type":"tool","uuid":"cecbcd76-97f0-4e7a-9c90-061d3e5be432","when":null,"workflow_outputs":[{"label":"Hifiasm Alternate gfa","output_name":"alternate_contig_graph","uuid":"7a14aa59-72a2-4bb5-9d74-2d501577db87"},{"label":"Hifiasm Primary gfa","output_name":"primary_contig_graph","uuid":"15537a4a-9d13-48c6-9a99-42edb5a5cad0"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":24,"input_connections":{"input_file":{"id":23,"output_name":"raw_unitigs"}},"inputs":[],"label":"Raw Unitig Image","name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":2210.498706500735,"top":14.534266196161356},"post_job_actions":{"TagDatasetActionoutfile":{"action_arguments":{"tags":"utg_png"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"2000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": \\"25.0\\", \\"output_format\\": \\"jpg\\", \\"width\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"c072feab-30cb-4311-8a45-bc126bf82e51","when":null,"workflow_outputs":[{"label":"raw unitig graph image","output_name":"outfile","uuid":"9a8feefa-ea89-402b-b2ab-336b417d5fa4"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":25,"input_connections":{"input_file":{"id":23,"output_name":"alternate_contig_graph"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2604.404312017185,"top":222.83203125},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Usable GFA Alternate"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"solo_alternate_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"18417334-ada9-4ac4-bc1f-9707f3200dfa","when":null,"workflow_outputs":[{"label":"Usable GFA Alternate","output_name":"output","uuid":"23d72fbe-6df8-431d-a711-01f766f1b08c"}]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":26,"input_connections":{"input_file":{"id":23,"output_name":"primary_contig_graph"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2613.958999517185,"top":0},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Usable GFA Primary"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"solo_primary_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"2dcb95b0-743f-4855-8a3c-d17e9d0656e7","when":null,"workflow_outputs":[{"label":"Usable GFA Primary","output_name":"output","uuid":"e1a82af3-c947-41f7-bb72-6c8c3842a9b8"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":27,"input_connections":{"input_file":{"id":23,"output_name":"primary_contig_graph"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2610.130874517185,"top":432.58203125},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm Primary assembly"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_primary"},"action_type":"TagDatasetAction","output_name":"output"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_primary, #c1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"ebb8800f-c58b-4118-93a3-a417631919db","when":null,"workflow_outputs":[{"label":"Hifiasm Primary assembly","output_name":"output","uuid":"f30af626-f47c-4967-a92b-f697b8e977d3"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":28,"input_connections":{"input_file":{"id":23,"output_name":"alternate_contig_graph"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2660.6503447320288,"top":702.3852767944336},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm Alternate assembly"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_alternate"},"action_type":"TagDatasetAction","output_name":"output"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_primary, #c1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"c6bd9a9c-c058-47dc-85cb-d5ba5e38490a","when":null,"workflow_outputs":[{"label":"Hifiasm Alternate assembly","output_name":"output","uuid":"76bc77a2-fd89-494d-a646-091560aeb1a1"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":29,"input_connections":{"input_file":{"id":23,"output_name":"primary_contig_graph"},"mode_condition|statistics_condition|expected_genomesize":{"id":22,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3293.2914594656368,"top":451.8153245133906},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_primary, #c1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"29d8eb08-28d9-4b2a-b61c-fcfbe634d4e9","when":null,"workflow_outputs":[{"label":"Assembly Stats on Primary assembly","output_name":"stats","uuid":"2acbea03-21b2-4bf6-b5dc-212dff771dc8"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":30,"input_connections":{"input_file":{"id":23,"output_name":"alternate_contig_graph"},"mode_condition|statistics_condition|expected_genomesize":{"id":22,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3295.19031529559,"top":753.6082826147187},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_alternate, #c2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"6b62822f-2c9f-41d6-acdd-45e13c52b1fb","when":null,"workflow_outputs":[{"label":"Assembly Stats on Alternate Assembly","output_name":"stats","uuid":"6362e546-0c71-47c9-8c78-939308613459"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":31,"input_connections":{"input_file":{"id":23,"output_name":"primary_contig_graph"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3281.6167045876036,"top":1012.1853499711735},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_primary"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"0e65f363-c145-4b0e-987e-9a13e822dbdf","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":32,"input_connections":{"input_file":{"id":23,"output_name":"alternate_contig_graph"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3296.71562955564,"top":1167.9942812334957},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_alternate"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"70b12ad8-64c9-401c-a50d-69178b9e7321","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":33,"input_connections":{"input":{"id":27,"output_name":"output"},"lineage_conditional|cached_db":{"id":6,"output_name":"output"},"lineage|lineage_dataset":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":3757.7159709943294,"top":1426.3228945055312},"post_job_actions":{"RenameDatasetActionbusco_sum":{"action_arguments":{"newname":"Busco Summary"},"action_type":"RenameDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"#c1,busco_c1_gff"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_c1_missing, #c1"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_c1_summ, #c1"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_c1_full, #c1"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_c1_img, #c1"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"1cd23635-275c-411f-b91d-f53f2bb229cb","when":null,"workflow_outputs":[{"label":"Busco Summary","output_name":"busco_sum","uuid":"3d6d5b4e-de88-4a54-ab6f-1a34aae9ecd8"},{"label":"Busco summary image","output_name":"summary_image","uuid":"0fb12967-65f6-477f-9f68-960ca6f56f37"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","errors":null,"id":34,"input_connections":{"mode|assembly_options|assembly_01":{"id":27,"output_name":"output"},"mode|assembly_options|assembly_02":{"id":28,"output_name":"output"},"mode|meryldb_F1":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Merqury","name":"mode"}],"label":null,"name":"Merqury","outputs":[{"name":"qv_files","type":"input"},{"name":"png_files","type":"input"},{"name":"stats_files","type":"input"},{"name":"hist_files","type":"input"},{"name":"log_file","type":"txt"}],"position":{"left":2984.96852100156,"top":1277.655174255371},"post_job_actions":{"HideDatasetActionlog_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log_file"},"TagDatasetActionbed_files":{"action_arguments":{"tags":"merqury_c_bed, #c"},"action_type":"TagDatasetAction","output_name":"bed_files"},"TagDatasetActionhist_files":{"action_arguments":{"tags":"#c"},"action_type":"TagDatasetAction","output_name":"hist_files"},"TagDatasetActionpng_files":{"action_arguments":{"tags":"merqury_c_png, #c"},"action_type":"TagDatasetAction","output_name":"png_files"},"TagDatasetActionqv_files":{"action_arguments":{"tags":"merqury_c_qv, #c"},"action_type":"TagDatasetAction","output_name":"qv_files"},"TagDatasetActionsizes_files":{"action_arguments":{"tags":"merqury_c_sizes, #c"},"action_type":"TagDatasetAction","output_name":"sizes_files"},"TagDatasetActionstats_files":{"action_arguments":{"tags":"merqury_c_stats, #c"},"action_type":"TagDatasetAction","output_name":"stats_files"},"TagDatasetActionwig_files":{"action_arguments":{"tags":"merqury_c_wig, #c"},"action_type":"TagDatasetAction","output_name":"wig_files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","tool_shed_repository":{"changeset_revision":"eb7b00ec347d","name":"merqury","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"label\\": \\"output_merqury\\", \\"mode\\": {\\"options\\": \\"default\\", \\"__current_case__\\": 0, \\"meryldb_F1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_options\\": {\\"number_assemblies\\": \\"two\\", \\"__current_case__\\": 1, \\"assembly_01\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_02\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_add_headers\\": true, \\"output_selector\\": [\\"qv\\", \\"plots\\", \\"stats\\", \\"log\\", \\"hist\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy4","type":"tool","uuid":"da32529d-f84c-4af6-b443-eddfc1705d78","when":null,"workflow_outputs":[{"label":"Merqury png","output_name":"png_files","uuid":"1040586b-b0aa-41c3-98e4-00ead4e43a5c"},{"label":"Merqury Histograms","output_name":"hist_files","uuid":"9bb427f1-2afd-4e63-b7a1-2c9b54c6bee0"},{"label":"Merqury completeness stats","output_name":"stats_files","uuid":"640107ca-7175-4624-9851-33911c197b18"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":35,"input_connections":{"infile":{"id":29,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3689.64186084531,"top":357.68359375},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\tPrimary\\\\\\"}; {print}; \\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b59dc0fc-7ac7-4e06-a282-89638b228be5","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":36,"input_connections":{"infile":{"id":30,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3690.7071684624975,"top":556.3767623901367},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\tAlternate\\\\\\"}; {print}; \\\\n\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"535e61f0-806f-4ee4-9bed-562968291171","when":null,"workflow_outputs":[]},"37":{"annotation":"","id":37,"input_connections":{"gfa_stats":{"id":31,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data Prep Primary"}],"label":"Data Prep Primary","name":"gfastats_data_prep","outputs":[],"position":{"left":3965.646499517185,"top":761.265625},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7f250de3-98cc-448b-b573-6b8a85c71352","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"68af2c3b-8f48-4bde-9dd1-a78151d6de1a"}]}},"tags":[],"uuid":"facbee5d-65b6-4a6d-9f02-7e8bfba15312"},"tool_id":null,"type":"subworkflow","uuid":"f7f6d8e0-74b0-479f-bb34-2f23450ffc4f","when":null,"workflow_outputs":[]},"38":{"annotation":"","id":38,"input_connections":{"gfa_stats":{"id":32,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data Prep Alternate"}],"label":"Data Prep Alternate","name":"gfastats_data_prep","outputs":[],"position":{"left":3961.68727100156,"top":946.4477767944336},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7f250de3-98cc-448b-b573-6b8a85c71352","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"68af2c3b-8f48-4bde-9dd1-a78151d6de1a"}]}},"tags":[],"uuid":"1723ac19-500a-4252-9b34-dc5318839dab"},"tool_id":null,"type":"subworkflow","uuid":"be05bdc0-5016-44a9-b438-ddca2492d2d6","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":39,"input_connections":{"input":{"id":34,"output_name":"qv_files"}},"inputs":[],"label":"merqury_qv","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":3368.915208782228,"top":1378.4880425062945},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"22d65bf8-2507-44ab-862a-f9a319fb45a4","when":null,"workflow_outputs":[{"label":"merqury_qv","output_name":"output","uuid":"392ff784-c7dc-4564-8f33-fd73b6cd4635"}]},"40":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":40,"input_connections":{"input":{"id":34,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":3383.7818577068465,"top":1495.6596354006524},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"a773ee61-4660-4e8a-8308-e33094c75299","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-cn.fl","output_name":"output","uuid":"c37f517e-c25d-44df-b2b6-926c5b1246cd"}]},"41":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":41,"input_connections":{"input":{"id":34,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-asm.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":3387.3075902982096,"top":1660.8402706956817},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-asm.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"45d20c6f-e68d-4f56-b82d-dad963673bd4","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-asm.fl","output_name":"output","uuid":"8505620c-0d68-4f62-b2e6-5d019e834ba7"}]},"42":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":42,"input_connections":{"input":{"id":34,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_01.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":3390.1728026881706,"top":1834.5536606231856},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_01.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"976df436-bf6e-4586-b979-39d154a037e2","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_01.spectra-cn.fl","output_name":"output","uuid":"cc37b1d2-e371-47b4-9d66-837733c4c04b"}]},"43":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":43,"input_connections":{"input":{"id":34,"output_name":"stats_files"}},"inputs":[],"label":"merqury_stats","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":3684.797014458241,"top":1970.3126850008684},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.completeness\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"a5704471-f618-4dcc-b34f-81e6598e650d","when":null,"workflow_outputs":[{"label":"merqury_stats","output_name":"output","uuid":"0dbd43e2-8b7a-47fb-aeb7-cb5abfccd98f"}]},"44":{"annotation":"","content_id":"join1","errors":null,"id":44,"input_connections":{"input1":{"id":35,"output_name":"outfile"},"input2":{"id":36,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4054.1211995685508,"top":281.72424025026436},"post_job_actions":{},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"fill_empty\\", \\"__current_case__\\": 1, \\"fill_columns_by\\": \\"fill_unjoined_only\\", \\"do_fill_empty_columns\\": {\\"column_fill_type\\": \\"single_fill_value\\", \\"__current_case__\\": 0, \\"fill_value\\": \\".\\"}}, \\"header\\": \\"\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"3049cad6-465c-4726-8a99-e0ac18e119c7","when":null,"workflow_outputs":[]},"45":{"annotation":"","id":45,"input_connections":{"Alternate data":{"id":38,"input_subworkflow_step_id":1,"output_name":"gfastats data for plotting"},"Name of alternate assembly":{"id":10,"input_subworkflow_step_id":3,"output_name":"output"},"Name of primary assembly":{"id":9,"input_subworkflow_step_id":2,"output_name":"output"},"Primary data":{"id":37,"input_subworkflow_step_id":0,"output_name":"gfastats data for plotting"}},"inputs":[{"description":"","name":"Plotting Nx and Sizes"}],"label":"Plotting Nx and Sizes","name":"gfastats_plot","outputs":[],"position":{"left":4513.6757963921855,"top":917.4278793334961},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","name":"gfastats_plot","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Primary data"}],"label":"Primary data","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a9a972e8-76f3-416f-9148-77203f6df3b8","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Alternate data"}],"label":"Alternate data","name":"Input dataset","outputs":[],"position":{"left":6.960205078125,"top":143.53692626953125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a5ce76b6-ae69-4c5d-b42f-dc211c9b64cc","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4.55963134765625,"top":296.3210144042969},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6118c82a-61da-4856-9e25-1f0d93297527","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"894be62c-079d-4209-a098-1858de70948a"}]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":7.428955078125,"top":440.4403076171875},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"3204b6e0-bd83-4a8d-8ddb-3b0dd89dbd04","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"3d7657f7-0651-43c1-80bb-57c4fa6d3502"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"exp":{"id":2,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1139.7584686279297,"top":107.42897033691406},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"8bc6adbb-1346-4521-bd69-ffa2738bad07","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"exp":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1151.7044982910156,"top":392.65625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"63c7b63c-5179-4ab3-9682-7d150029446e","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":6,"input_connections":{"inputs":{"id":4,"output_name":"out_file1"},"queries_0|inputs2":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1441.193115234375,"top":216.59091186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"edbf8363-cf44-4536-b45c-60d54088e0fb","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1780.3265991210938,"top":182.99716186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ffa38570-79c9-41e6-9a55-3245d5427fa1","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1781.349365234375,"top":387.96875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6567b833-44b5-41e6-885b-d1d17b5d9376","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2068.1390991210938,"top":31.988632202148438},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"18c6ac70-f4d3-4d68-93f8-c08afafbc6c5","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"d1dd55bd-0441-4cfd-baf2-767ef768d01e"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2083.3805541992188,"top":319.8011169433594},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"cce1fe43-55c6-4c1a-a957-ac7af9271c46","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"c979a10d-de93-4c02-b286-0ec0fca016b9"}]}},"tags":[],"uuid":"d9b0af24-02ec-456a-9de7-65410da2dbab"},"tool_id":null,"type":"subworkflow","uuid":"08a613ce-a4c1-46df-930c-153b491114d2","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"Size Plot","uuid":"e46888e6-9ae3-4310-9639-de1cdfff1871"},{"label":"Nx Plot","output_name":"Nx Plot","uuid":"e34782d2-17d1-4598-92d1-93ca3d7ec7dc"}]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":46,"input_connections":{"input":{"id":44,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":4479.7665298216425,"top":303.2318241078274},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Assembly statistics"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"gfastats_asm_pri_alt"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": [\\"3\\"]}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"3470d58d-da54-43ef-842c-acfee1b35a56","when":null,"workflow_outputs":[{"label":"Assembly statistics","output_name":"output","uuid":"323d0ecc-1fec-4126-94de-1f0deb6b6880"}]},"47":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":47,"input_connections":{"infile":{"id":46,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":4748.6890863468225,"top":495.54555212695516},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ec90b8f9-bcc2-4b8a-88ee-03b3e19979ba","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"365867c1-fec9-4c10-8b31-3f615992910a"}]}},"tags":["VGP","Reviewed"],"uuid":"380b07b6-6444-4aaf-b2f8-788eec425986","version":2},"readme":"## Contiging Solo:\\n\\nGenerate assembly based on PacBio Hifi Reads.\\n\\n\\n### Inputs\\n\\n\\n1. Hifi long reads [fastq]\\n2. K-mer database [meryldb]\\n3. Genome profile summary generated by Genomescope [txt]\\n4. Homozygous Read Coverage. Optional, use if you think the estimation from Genomescope is inacurate. \\n5. Genomescope Model Parameters generated by Genomescope [tabular]\\n6. Database for busco lineage (recommended: latest)\\n7. Busco lineage (recommended: vertebrata)\\n8. Name of first assembly\\n9. Name of second assembly\\n\\n\\n### Outputs\\n\\n1. Primary assembly\\n2. Alternate assembly\\n3. QC: Bandage image for the raw unitigs\\n4. QC: BUSCO report for both assemblies\\n5. QC: Merqury report for both assemblies\\n6. QC: Assembly statistics for both assemblies\\n7. QC: Nx plot for both assemblies\\n8. QC: Size plot for both assemblie\\n","changelog":"# Changelog\\n\\n## [0.2.8] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.2.7] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.2.6] - 2025-02-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.2.5] - 2025-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy1`\\n\\n## [0.2.4] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n## [0.2.3] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n\\n## [0.2.2] - 2024-10-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n\\n## [0.2.1] - 2024-08-13\\n\\n- Expose Busco lineage database parameter\\n\\n## [0.2] - 2024-08-01\\n\\n### Added\\n\\n- Add bandage image\\n- Add workflow report\\n\\n### Changed\\n\\n- Fix bug in merging assembly statistics\\n\\n## [0.1.8] - 2024-07-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n\\n## [0.1.7] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0`\\n\\n## [0.1.6] - 2024-05-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0`\\n\\n\\n## [0.1.5] - 2024-04-23\\n\\n### Changed\\n\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n\\n## [0.1.4] - 2024-04-19\\n\\n### Manual Updates\\n\\n- Add Homozygous Read coverage parameter and calculation\\n- Expose Lineage parameter\\n\\n### Automatic update\\n\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4`\\n\\n## [0.1.3] - 2024-03-25\\n\\n### Manual Updates\\n- Add Gff outputs to Busco\\n- Add histogram outputs to Merqury\\n- Add labels to subworkflows\\n\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1`\\n\\n\\n## [0.1.2] - 2023-11-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.16.1+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n\\n## [0.1.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1] - 2023-09-26\\n\\nAddition of the workflow to the iwc repository.\\n","diagrams":"# Workflow diagrams\\n\\n## Assembly-Hifi-only-VGP3\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Pacbio Reads Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ Meryl Database\\"]@{ shape: doc }\\n2[\\"ℹ️ Genomescope Summary\\"]@{ shape: doc }\\n3[\\"ℹ️ Bits for Hifiasm bloom filter\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Homozygous Read Coverage\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Genomescope Model Parameters\\"]@{ shape: doc }\\n6[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n8[\\"ℹ️ SAK input file (Optional)\\"]@{ shape: doc }\\n9[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n11[\\"Cutadapt\\"]@{ shape: process }\\n0 --> 11\\n12[\\"Search in textfiles\\"]@{ shape: process }\\n2 --> 12\\n13[\\"Pick parameter value\\"]@{ shape: process }\\n3 --> 13\\n14[\\"Compute\\"]@{ shape: process }\\n5 --> 14\\n15[\\"MultiQC\\"]@{ shape: process }\\n11 --> 15\\n16[\\"Replace Text\\"]@{ shape: process }\\n12 --> 16\\n17[\\"Cut\\"]@{ shape: process }\\n14 --> 17\\n18[\\"Convert\\"]@{ shape: process }\\n16 --> 18\\n19[\\"Parse parameter value\\"]@{ shape: process }\\n17 --> 19\\n20[\\"Cut\\"]@{ shape: process }\\n18 --> 20\\n21[\\"Homozygous read coverage for Hifiasm\\"]@{ shape: process }\\n4 --> 21\\n19 --> 21\\n22[\\"Estimated genome size\\"]@{ shape: process }\\n20 --> 22\\n23[\\"Hifiasm\\"]@{ shape: process }\\n21 --> 23\\n13 --> 23\\n11 --> 23\\n24[\\"Raw Unitig Image\\"]@{ shape: process }\\n23 --> 24\\n25[\\"gfastats\\"]@{ shape: process }\\n23 --> 25\\n8 --> 25\\n26[\\"gfastats\\"]@{ shape: process }\\n23 --> 26\\n8 --> 26\\n27[\\"gfastats\\"]@{ shape: process }\\n23 --> 27\\n8 --> 27\\n28[\\"gfastats\\"]@{ shape: process }\\n23 --> 28\\n8 --> 28\\n29[\\"gfastats\\"]@{ shape: process }\\n23 --> 29\\n22 --> 29\\n30[\\"gfastats\\"]@{ shape: process }\\n23 --> 30\\n22 --> 30\\n31[\\"gfastats\\"]@{ shape: process }\\n23 --> 31\\n32[\\"gfastats\\"]@{ shape: process }\\n23 --> 32\\n33[\\"Busco\\"]@{ shape: process }\\n27 --> 33\\n6 --> 33\\n7 --> 33\\n34[\\"Merqury\\"]@{ shape: process }\\n27 --> 34\\n28 --> 34\\n1 --> 34\\n35[\\"Text reformatting\\"]@{ shape: process }\\n29 --> 35\\n36[\\"Text reformatting\\"]@{ shape: process }\\n30 --> 36\\n37[\\"🛠️ Data Prep Primary\\"]@{ shape: subprocess }\\n31 --> 37\\n38[\\"🛠️ Data Prep Alternate\\"]@{ shape: subprocess }\\n32 --> 38\\n39[\\"merqury_qv\\"]@{ shape: process }\\n34 --> 39\\n40[\\"output_merqury.spectra-cn.fl\\"]@{ shape: process }\\n34 --> 40\\n41[\\"output_merqury.spectra-asm.fl\\"]@{ shape: process }\\n34 --> 41\\n42[\\"output_merqury.assembly_01.spectra-cn.fl\\"]@{ shape: process }\\n34 --> 42\\n43[\\"merqury_stats\\"]@{ shape: process }\\n34 --> 43\\n44[\\"Join two Datasets\\"]@{ shape: process }\\n35 --> 44\\n36 --> 44\\n45[\\"🛠️ Plotting Nx and Sizes\\"]@{ shape: subprocess }\\n38 --> 45\\n10 --> 45\\n9 --> 45\\n37 --> 45\\n46[\\"Advanced Cut\\"]@{ shape: process }\\n44 --> 46\\n47[\\"Replace\\"]@{ shape: process }\\n46 --> 47\\n```\\n\\n## gfastats_plot\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Primary data\\"]@{ shape: doc }\\n1[\\"ℹ️ Alternate data\\"]@{ shape: doc }\\n2[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"Concatenate datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Cut\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Nx Plot\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Size Plot\\"]@{ shape: process }\\n8 --> 10\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Assembly-Hifi-only-VGP3/main","dockstore_id":24056,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078483","tests":[{"doc":"Test for Assembly-Hifi-only-VGP3","job":{"Meryl Database":{"class":"File","location":"https://zenodo.org/record/8371053/files/Meryl-db.meryldb?download=1","filetype":"meryldb"},"Genomescope Summary":{"class":"File","location":"https://zenodo.org/record/8371053/files/GenomeScope_Summary.txt?download=1","filetype":"txt"},"Genomescope Model Parameters":{"class":"File","path":"test-data/GenomeScope_Model_parameters.tabular","filetype":"tabular"},"Pacbio Reads Collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"yeast_reads_sub1.fastq.gz","location":"https://zenodo.org/record/8371053/files/yeast_reads_sub1.fastq.gz?download=1","filetype":"fastqsanger.gz"}]},"Bits for Hifiasm bloom filter":32,"SAK input file (Optional)":null,"Homozygous Read Coverage":null,"Database for Busco Lineage":"v5","Lineage":"vertebrata_odb10","Name of primary assembly":"Primary","Name of alternate assembly":"Alternate"},"outputs":{"Hifiasm Primary assembly":{"asserts":{"has_n_lines":{"n":168}}},"Hifiasm Alternate assembly":{"asserts":{"has_n_lines":{"n":6}}},"Assembly Stats on Alternate Assembly":{"asserts":{"has_line":{"line":"# scaffolds\\t3"}}},"Estimated Genome size":2288021,"Assembly Stats on Primary assembly":{"asserts":{"has_line":{"line":"# scaffolds\\t84"}}},"Assembly statistics":{"asserts":{"has_line":{"line":"# scaffolds\\t84\\t3"}}},"Busco Summary":{"asserts":{"has_text":{"text":"C:1.2%[S:1.2%,D:0.0%],F:0.4%,M:98.4%,n:3354"}}},"Nx Plot":{"asserts":{"has_size":{"value":60000,"delta":10000}}}}}]}],"path":"./workflows/VGP-assembly-v2/Assembly-Hifi-only-VGP3"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Assembly-decontamination-VGP9.ga","testParameterFiles":["/Assembly-decontamination-VGP9-tests.yml"],"authors":[{"name":"Nadolina Brajuka","url":"https://github.com/Nadolina"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Nadolina Brajuka","url":"https://github.com/Nadolina"}],"format-version":"0.1","license":"BSD-3-Clause","release":"0.4","name":"Assembly-decontamination-VGP9","report":{"markdown":"\\n# Workflow Execution Report\\n\\n### Worflow ran on: \\n\\n```galaxy\\ninvocation_time()\\n```\\n\\n# List of contaminants\\n\\n\\r\\n```galaxy\\nhistory_dataset_as_table(output=\\"contaminants_table\\")\\n```\\r\\n\\n\\n# List of Mitochondrial scaffolds\\n\\r\\n```galaxy\\nhistory_dataset_as_table(output=\\"mito_scaff_names\\")\\n```\\r\\n\\n\\n\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Scaffolded assembly (fasta)"}],"label":"Scaffolded assembly (fasta)","name":"Input dataset","outputs":[],"position":{"left":0,"top":573.2100224386197},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"f3335a9f-1793-4ba2-98d8-253eb5380a3f","when":null,"workflow_outputs":[]},"1":{"annotation":"Select the database to query to identify non-target contaminants.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Select the database to query to identify non-target contaminants.","name":"Database for Kraken2"}],"label":"Database for Kraken2","name":"Input parameter","outputs":[],"position":{"left":0.5475811709443406,"top":692.7734375},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"4bd1b150-bdb0-4458-9ddc-62bc29356aa2","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"34f0edb5-742e-4de2-923f-1fb93e511ca9"}]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy2","errors":null,"id":2,"input_connections":{"db_opts|subject":{"id":0,"output_name":"output"}},"inputs":[],"label":"soft-masking ","name":"NCBI BLAST+ dustmasker","outputs":[{"name":"outfile","type":"maskinfo-asn1"}],"position":{"left":463.79758117094434,"top":292.69140625},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy2","tool_shed_repository":{"changeset_revision":"cbf3f518b668","name":"ncbi_blast_plus","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"db_opts\\": {\\"db_opts_selector\\": \\"file\\", \\"__current_case__\\": 2, \\"database\\": \\"\\", \\"histdb\\": \\"\\", \\"subject\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"level\\": \\"40\\", \\"linker\\": \\"1\\", \\"outformat\\": \\"fasta\\", \\"window\\": \\"64\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.14.1+galaxy2","type":"tool","uuid":"903a13af-2be0-48a9-8355-1e2486d857cf","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","errors":null,"id":3,"input_connections":{"infile":{"id":2,"output_name":"outfile"}},"inputs":[],"label":"hard-masking","name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":827.4147686709443,"top":312.24609375},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"fasta"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"/^>/!y/atcgn/NNNNN/\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"5089350b-c1a9-498a-9786-9864d484f77a","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1","errors":null,"id":4,"input_connections":{"kraken2_database":{"id":1,"output_name":"output"},"single_paired|input_sequences":{"id":3,"output_name":"output"}},"inputs":[],"label":"ID non-target contaminants","name":"Kraken2","outputs":[{"name":"classified_out_s","type":"input"},{"name":"unclassified_out_s","type":"input"},{"name":"output","type":"tabular"}],"position":{"left":1300.1257061709443,"top":147.58203125},"post_job_actions":{"HideDatasetActionunclassified_out_s":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"unclassified_out_s"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1","tool_shed_repository":{"changeset_revision":"cdee7158adf3","name":"kraken2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"confidence\\": \\"0.3\\", \\"kraken2_database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_base_quality\\": \\"0\\", \\"minimum_hit_groups\\": \\"2\\", \\"quick\\": false, \\"report\\": {\\"create_report\\": false, \\"use_mpa_style\\": false, \\"report_zero_counts\\": false, \\"report_minimizer_data\\": false}, \\"single_paired\\": {\\"single_paired_selector\\": \\"no\\", \\"__current_case__\\": 2, \\"input_sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split_reads\\": true, \\"use_names\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3+galaxy1","type":"tool","uuid":"17141f21-ae3c-44ac-b0a5-d3bb9b8726a4","when":null,"workflow_outputs":[{"label":"Kraken2 classification scores","output_name":"output","uuid":"c380857b-5180-4e82-a14f-8b575f10abaa"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy2","errors":null,"id":5,"input_connections":{"query":{"id":3,"output_name":"output"}},"inputs":[],"label":"blast mitochondria DB","name":"NCBI BLAST+ blastn","outputs":[{"name":"output1","type":"tabular"}],"position":{"left":1293.4616436709443,"top":539.6171875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy2","tool_shed_repository":{"changeset_revision":"cbf3f518b668","name":"ncbi_blast_plus","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"blast_type\\": \\"blastn\\", \\"db_opts\\": {\\"db_opts_selector\\": \\"db\\", \\"__current_case__\\": 0, \\"database\\": [\\"refseq_mitochondrion\\"], \\"histdb\\": \\"\\", \\"subject\\": \\"\\"}, \\"evalue_cutoff\\": \\"0.001\\", \\"output\\": {\\"out_format\\": \\"cols\\", \\"__current_case__\\": 2, \\"std_cols\\": [\\"qseqid\\", \\"sseqid\\", \\"length\\", \\"qstart\\", \\"qend\\", \\"evalue\\"], \\"ext_cols\\": [\\"qlen\\"], \\"ids_cols\\": null, \\"misc_cols\\": [\\"qcovs\\", \\"qcovhsp\\"], \\"tax_cols\\": null}, \\"query\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.14.1+galaxy2","type":"tool","uuid":"1737bad0-48d9-489e-9a45-d6f3d3fcb900","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"Cut1","errors":null,"id":6,"input_connections":{"input":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1654.7624249209443,"top":53.078125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1,c2,c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"bb39eb8b-cad8-44f1-aa56-64a11d985e3c","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/parse_mito_blast/parse_mito_blast/1.0.2+galaxy0","errors":null,"id":7,"input_connections":{"blastout":{"id":5,"output_name":"output1"}},"inputs":[],"label":"parsing blast output","name":"Parse mitochondrial blast","outputs":[{"name":"cov_report","type":"tabular"},{"name":"mito_scaff_names","type":"txt"}],"position":{"left":1821.3783868183484,"top":482.19337215633186},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/parse_mito_blast/parse_mito_blast/1.0.2+galaxy0","tool_shed_repository":{"changeset_revision":"3c9ad4adf8d2","name":"parse_mito_blast","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"blastout\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2+galaxy0","type":"tool","uuid":"4350e47a-2394-4381-a274-e238985195df","when":null,"workflow_outputs":[{"label":"mito_scaff_names","output_name":"mito_scaff_names","uuid":"1478af1e-52f4-48fe-b966-0a61e2014a58"}]},"8":{"annotation":"","content_id":"Filter1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2001.9694561709443,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"List of contaminants"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"c1!=\'U\'\\", \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"f5d80335-9b98-40ca-b706-4bb54d9801e6","when":null,"workflow_outputs":[{"label":"contaminants_table","output_name":"out_file1","uuid":"64156e28-aaf0-4578-9a7e-d6d21774c2c8"}]},"9":{"annotation":"","content_id":"Cut1","errors":null,"id":9,"input_connections":{"input":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2237.4640689838916,"top":241.19478945940944},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"8cc3e58f-2940-454a-9cf6-b32c1ab19016","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":10,"input_connections":{"inputs":[{"id":7,"output_name":"mito_scaff_names"},{"id":9,"output_name":"out_file1"}]},"inputs":[],"label":"concatenate scaffold lists","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2538.6178936709443,"top":423.9453125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"5b7e33a0-0ec4-4539-bb91-0c862597df69","when":null,"workflow_outputs":[{"label":"full contaminant + mito scaffold list","output_name":"out_file1","uuid":"77084ec5-a145-40c1-9cfc-476d091105ae"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":11,"input_connections":{"input_file":{"id":0,"output_name":"output"},"target_condition|exclude_bed":{"id":10,"output_name":"out_file1"}},"inputs":[],"label":"removing scaffolds ","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2977.7819561709443,"top":798.078125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta.gz\\", \\"__current_case__\\": 1, \\"line_length\\": null}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"true\\", \\"__current_case__\\": 1, \\"target_sequence\\": \\"\\", \\"include_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"exclude_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"5113c945-c811-4cbc-81ac-20d1d80e97ed","when":null,"workflow_outputs":[{"label":"final decontaminated assembly","output_name":"output","uuid":"88a8e9ed-d92c-4340-aaad-c6d450a336aa"}]}},"tags":["VGP_curated"],"uuid":"aa4960d9-198d-4990-a393-a98efcd3070a","version":3},"readme":"## Decontamination Workflow\\n\\nDecontamination (foreign contaminants and mitochondrial sequences) of genome assembly after scaffolding step. Part of the VGP Suite. \\n\\n### Inputs\\n\\n- Genome Assembly [fasta]\\n- Database for Kraken2. Database containing the possible contaminants.\\n\\n### Ouput\\n\\n- List of contaminant scaffolds\\n- List of mitochondrial scaffolds\\n- Decontaminated assembly\\n","changelog":"# Changelog\\n\\n## [0.4] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.3] - 2024-08-26\\n\\n- Changed the regex to only change the sequences. The workflow is now more robust to different (non VGP) scaffold names and VGP names with prefixes. \\n- Reduce the number of steps in the workflow. \\n\\n## [0.2] - 2024-08-15\\n\\n- Bug Fix: a missing regex caused the list of contaminants to empty out and the contaminants were not removed. \\n\\n## [0.1.6] - 2024-06-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1`\\n\\n## [0.1.5] - 2024-04-01\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n\\n\\n## [0.1.4] - 2024-03-11\\n\\n- Fix regex that was causing the headers to change. \\n\\n\\n## [0.1.3] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.2] - 2023-11-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy1`\\n\\n## [0.1.1] - 2023-11-09\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.10.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/2.14.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.10.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.14.1+galaxy0`\\n\\n## [0.1] - 2023-11-07\\n\\n- Creation of workflow for Assembly decontamination\\n","diagrams":"# Workflow diagrams\\n\\n## Assembly-decontamination-VGP9\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Scaffolded assembly (fasta)\\"]@{ shape: doc }\\n1[\\"ℹ️ Database for Kraken2\\"]@{ shape: lean-l }\\n2[\\"soft-masking \\"]@{ shape: process }\\n0 --> 2\\n3[\\"hard-masking\\"]@{ shape: process }\\n2 --> 3\\n4[\\"ID non-target contaminants\\"]@{ shape: process }\\n1 --> 4\\n3 --> 4\\n5[\\"blast mitochondria DB\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Cut\\"]@{ shape: process }\\n4 --> 6\\n7[\\"parsing blast output\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Filter\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Cut\\"]@{ shape: process }\\n8 --> 9\\n10[\\"concatenate scaffold lists\\"]@{ shape: process }\\n7 --> 10\\n9 --> 10\\n11[\\"removing scaffolds \\"]@{ shape: process }\\n0 --> 11\\n10 --> 11\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Assembly-decontamination-VGP9/main","dockstore_id":24303,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078881","tests":[{"doc":"Test outline for Assembly-decontamination-VGP9","job":{"Scaffolded assembly (fasta)":{"class":"File","location":"https://zenodo.org/records/13327557/files/Scaffolded_assembly.fasta?download=1","filetype":"fasta"},"Database for Kraken2":"viral2019-03"},"outputs":{"mito_scaff_names":{"asserts":{"has_text":{"text":"scaffold_82"}}},"final decontaminated assembly":{"decompress":true,"asserts":{"not_has_text":{"text":"scaffold_200"}}},"contaminants_table":{"asserts":{"has_text":{"text":"scaffold_200"}}}}}]}],"path":"./workflows/VGP-assembly-v2/Assembly-decontamination-VGP9"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Scaffolding-HiC-VGP8.ga","testParameterFiles":["/Scaffolding-HiC-VGP8-tests.yml"],"authors":[{"name":"VGP","url":"https://vertebrategenomeproject.org"},{"name":"Galaxy"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow perform the scaffolding of a genome assemble using HiC data with YAHS. Part of the VGP set of workflows.","comments":[],"creator":[{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"},{"class":"Organization","name":"Galaxy"}],"format-version":"0.1","license":"CC-BY-4.0","release":"1.2","name":"Scaffolding with Hi-C data VGP8","report":{"markdown":"# Workflow Execution Report\\n\\nTime workflow was invoked\\n\\n```galaxy\\ninvocation_time()\\n```\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n\\n## BUSCO results (Vertebrata database)\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco Summary image\\")\\n```\\n\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n## PretextMap\\n\\n### Before Scaffolding\\n\\n\\r\\n```galaxy\\nhistory_dataset_as_image(output=\\"Pretext Map Before HiC scaffolding\\")\\n```\\r\\n\\n\\n### After Scaffolding\\r\\n```galaxy\\nhistory_dataset_as_image(output=\\"Pretext Map After HiC scaffolding\\")\\n```\\r\\n\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"The input GFA must conform to gfa1.2 standards, i.e. should have \'P\' lines defined, and contain sequences. Output GFAs from assemblers can be run through a GFA-GFA conversion using gfastats to ensure this. \\n","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"The input GFA must conform to gfa1.2 standards, i.e. should have \'P\' lines defined, and contain sequences. Output GFAs from assemblers can be run through a GFA-GFA conversion using gfastats to ensure this. \\n","name":"Input GFA"}],"label":"Input GFA","name":"Input dataset","outputs":[],"position":{"left":0,"top":528.1154743404687},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"84c4861b-1fa5-4c1e-86f4-2a9211d56b41","when":null,"workflow_outputs":[]},"1":{"annotation":"Select the haplotype being scaffolded.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Select the haplotype being scaffolded.","name":"Haplotype"}],"label":"Haplotype","name":"Input parameter","outputs":[],"position":{"left":56.014594183762775,"top":621.6149969681851},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictions\\": [\\"hap1\\", \\"hap2\\", \\"mat\\", \\"pat\\", \\"pri\\", \\"alt\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3495a63c-bcd5-45fa-b327-ed7ec0234e78","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Hi-C reads"}],"label":"Hi-C reads","name":"Input dataset collection","outputs":[],"position":{"left":113.93357467974579,"top":715.4531348279907},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"644d86d9-1ab3-49aa-8dda-cf01b985d374","when":null,"workflow_outputs":[]},"3":{"annotation":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\".  (Select No if you are using the trimmed data generated by a previous workflow)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\".  (Select No if you are using the trimmed data generated by a previous workflow)","name":"Trim Hi-C Data?"}],"label":"Trim Hi-C Data?","name":"Input parameter","outputs":[],"position":{"left":184.66040815892518,"top":805.5160931691415},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"68c2f24e-7f3f-40f9-b0dc-bbdd9ca46874","when":null,"workflow_outputs":[]},"4":{"annotation":"Select the database to use for Busco lineage. ","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Select the database to use for Busco lineage. ","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":236.5698456297109,"top":918.1387332663845},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1f4997fc-1880-4416-8857-2a63d3501a6c","when":null,"workflow_outputs":[]},"5":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis.\\n","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis.\\n","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":271.45421676866704,"top":1019.3608529326734},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b45e6bda-be0e-45e7-a9b1-17325a08dd76","when":null,"workflow_outputs":[]},"6":{"annotation":"Restriction enzymes used in preparation of Hi-C libraries.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Restriction enzymes used in preparation of Hi-C libraries.","name":"Restriction enzymes"}],"label":"Restriction enzymes","name":"Input parameter","outputs":[],"position":{"left":312.93646897918677,"top":1102.9756599942102},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b899a210-60c2-4e9e-8a34-e1bc82ef373d","when":null,"workflow_outputs":[]},"7":{"annotation":"Estimated genome size from contiging workflow.","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Estimated genome size from contiging workflow.","name":"Estimated genome size - Parameter File"}],"label":"Estimated genome size - Parameter File","name":"Input dataset","outputs":[],"position":{"left":349.30065073286784,"top":1212.5625422446765},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"427eea61-8233-4c44-9868-805f70ab76ce","when":null,"workflow_outputs":[]},"8":{"annotation":"Do not use. ","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Do not use. ","name":"SAK input file"}],"label":"SAK input file","name":"Input dataset","outputs":[],"position":{"left":388.73117716462275,"top":1321.8543101858209},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"bc993517-ae23-417d-8eb1-891a127f544c","when":null,"workflow_outputs":[]},"9":{"annotation":"Provide AGP for Scaffolding.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Provide AGP for Scaffolding.","name":"Sequence graph"}],"label":"Sequence graph","name":"Input dataset","outputs":[],"position":{"left":442.9594740249754,"top":1402.4600651685134},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"13cd4c1c-968b-4077-9845-e06738a2efb8","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":10,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":768.555619765986,"top":468.1468249917974},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"$&.\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"0951bd8c-a2c2-4bd4-b4aa-4b090bf19857","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":11,"input_connections":{"input1":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1864.297833780668,"top":1578.552969536336},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"c224c32e-36a4-451f-a74a-fa07a9bf9c7c","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":12,"input_connections":{"input_file":{"id":0,"output_name":"output"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":767.5961403849224,"top":260.08813758262755},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"47c9d134-9895-4055-a3cf-98a77dc6a0b7","when":null,"workflow_outputs":[]},"13":{"annotation":"","id":13,"input_connections":{"Do you want to trim the Hi-C data?":{"id":3,"input_subworkflow_step_id":2,"output_name":"output"},"Hi-C reads":{"id":2,"input_subworkflow_step_id":1,"output_name":"output"},"Reference":{"id":12,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Trim and Align Hi-C paired collection","outputs":[],"position":{"left":1194.609375,"top":859.7421875},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0001-6228-2785","name":"Patrik Smeds"},{"class":"Person","identifier":"https://orcid.org/0000-0001-6421-3484","name":"Delphine Lariviere"}],"format-version":"0.1","license":"MIT","name":"Trim and Align Hi-C paired collection","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Assembly for Haplotype 1.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Assembly for Haplotype 1.","name":"Reference"}],"label":"Reference","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"fa895dbf-8f06-4728-ac6c-d526fcff2578","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Hi-C reads"}],"label":"Hi-C reads","name":"Input dataset collection","outputs":[],"position":{"left":124.6511831732837,"top":159.35032776516732},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"a381377e-3cd7-4ed0-b7ed-8f5ddcedcb6e","when":null,"workflow_outputs":[]},"2":{"annotation":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\".","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\".","name":"Do you want to trim the Hi-C data?"}],"label":"Do you want to trim the Hi-C data?","name":"Input parameter","outputs":[],"position":{"left":234.54989137895723,"top":295.2235674741496},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"6d1391db-7fef-4b5c-a236-26016d6d0aab","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":3,"input_connections":{"library|input_1":{"id":1,"output_name":"output"},"when":{"id":2,"output_name":"output"}},"inputs":[],"label":"Trim Hi-C reads 2","name":"Cutadapt","outputs":[{"name":"out_pairs","type":"input"},{"name":"report","type":"txt"}],"position":{"left":782.453125,"top":137.4375},"post_job_actions":{"HideDatasetActionout_pairs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_pairs"},"RenameDatasetActionout1":{"action_arguments":{"newname":"Trimmed Hi-C Forward Reads"},"action_type":"RenameDatasetAction","output_name":"out1"},"RenameDatasetActionout2":{"action_arguments":{"newname":"Trimmed Hi-C Reverse Reads"},"action_type":"RenameDatasetAction","output_name":"out2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"1\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}, \\"r2\\": {\\"adapters2\\": [], \\"front_adapters2\\": [], \\"anywhere_adapters2\\": []}, \\"pair_adapters\\": false}, \\"other_trimming_options\\": {\\"cut\\": \\"5\\", \\"cut2\\": \\"5\\", \\"quality_cutoff\\": \\"0\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": null, \\"length_tag\\": null, \\"rename\\": null, \\"zero_cap\\": false}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"15c5cc40-ce0a-4e29-a891-87f55bf2bd6b","when":"$(inputs.when)","workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":4,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":3,"output_name":"out_pairs"},"style_cond|type_cond|pick_from_1|value":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":1070.82421875,"top":405.32421875},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Trimmed Hi-C data"},"action_type":"RenameDatasetAction","output_name":"data_param"},"TagDatasetActiondata_param":{"action_arguments":{"tags":"hic_trimmed"},"action_type":"TagDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"325884ac-c2c1-4a5c-b74f-c1c422662b78","when":null,"workflow_outputs":[{"label":"Trimmed Hi-C data","output_name":"data_param","uuid":"73027165-8c18-45a9-9df1-76e3cc2421fb"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy1","errors":null,"id":5,"input_connections":{"fastq_input|fastq_input1":{"id":4,"output_name":"data_param"},"reference_source|ref_file":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"BWA-MEM2","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":1319.1117419971706,"top":62.41230832146982},"post_job_actions":{"HideDatasetActionbam_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bam_output"},"RenameDatasetActionbam_output":{"action_arguments":{"newname":"Merged Hi-C Alignments"},"action_type":"RenameDatasetAction","output_name":"bam_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy1","tool_shed_repository":{"changeset_revision":"bfaa0d22c2e4","name":"bwa_mem2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"full\\", \\"__current_case__\\": 4, \\"algorithmic_options\\": {\\"algorithmic_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"k\\": \\"19\\", \\"w\\": \\"100\\", \\"d\\": \\"100\\", \\"r\\": \\"1.5\\", \\"y\\": \\"20\\", \\"c\\": \\"500\\", \\"D\\": \\"0.5\\", \\"W\\": \\"0\\", \\"m\\": \\"50\\", \\"S\\": true, \\"P\\": true, \\"e\\": false}, \\"scoring_options\\": {\\"scoring_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"io_options\\": {\\"io_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"five\\": false, \\"q\\": false, \\"T\\": \\"20\\", \\"h\\": \\"5\\", \\"a\\": true, \\"C\\": false, \\"V\\": false, \\"Y\\": false, \\"M\\": false, \\"K\\": null}}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"iset_stats\\": null}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1+galaxy1","type":"tool","uuid":"45d2eaea-40f9-461e-bfb0-cadd9b9738b8","when":null,"workflow_outputs":[{"label":"Hi-C Alignments","output_name":"bam_output","uuid":"16019374-1e25-4b5b-a89e-a9a4fe7dcb95"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/3.1.1.0","errors":null,"id":6,"input_connections":{"inputFile":{"id":5,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"MergeSamFiles","outputs":[{"name":"outFile","type":"bam"}],"position":{"left":1555.6341685495568,"top":91.70871028133104},"post_job_actions":{"TagDatasetActionoutFile":{"action_arguments":{"tags":"merged_hic_mapping"},"action_type":"TagDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"comments\\": [], \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_sequence_dictionaries\\": false, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"b6f0b737-f2db-414f-bb64-ec99b1b4225b","when":null,"workflow_outputs":[{"label":"Merged Hi-C Alignments","output_name":"outFile","uuid":"41c24dd9-be91-4b0c-8a8b-f1346eecd250"}]}},"tags":[],"uuid":"be80dcee-1fcf-45aa-b249-a7dd2d75b4cd"},"tool_id":null,"type":"subworkflow","uuid":"d037482e-d0ed-4382-92be-5e9654e0d15c","when":null,"workflow_outputs":[{"label":"s1 Trimmed Hi-C data","output_name":"Trimmed Hi-C data","uuid":"bfee12c8-1715-48ad-a35b-d938e9038c87"},{"label":"s1 Hi-C Alignments","output_name":"Hi-C Alignments","uuid":"936f703d-5288-4bc6-85a1-81bd91ce77ef"},{"label":"s1 Merged Hi-C Alignments","output_name":"Merged Hi-C Alignments","uuid":"0d093dde-847a-4e1c-9fe9-d64c2162eae4"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy2","errors":null,"id":14,"input_connections":{"function|agp":{"id":9,"output_name":"output"},"function|bfile":{"id":13,"output_name":"Merged Hi-C Alignments"},"function|enzyme_conditional|preconfigured_enzymes":{"id":6,"output_name":"output"},"function|fasta":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"YAHS","outputs":[{"name":"initial_agp_break","type":"input"},{"name":"agp_break","type":"input"},{"name":"agp_out","type":"input"},{"name":"final_agp_out","type":"agp"},{"name":"final_fasta_out","type":"fasta"},{"name":"log_file","type":"txt"}],"position":{"left":1635.9148202090612,"top":289.1388196257464},"post_job_actions":{"TagDatasetActionfinal_agp_out":{"action_arguments":{"tags":"s2_agp, #s2"},"action_type":"TagDatasetAction","output_name":"final_agp_out"},"TagDatasetActionfinal_fasta_out":{"action_arguments":{"tags":"s2_fasta, #s2"},"action_type":"TagDatasetAction","output_name":"final_fasta_out"},"TagDatasetActionlog_file":{"action_arguments":{"tags":"s2_log, #s2"},"action_type":"TagDatasetAction","output_name":"log_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy2","tool_shed_repository":{"changeset_revision":"ff4031bfaa22","name":"yahs","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function\\": {\\"function_select\\": \\"yahs\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bfile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"agp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"res\\": \\"\\", \\"enzyme_conditional\\": {\\"enzyme_options\\": \\"preconfigured\\", \\"__current_case__\\": 1, \\"preconfigured_enzymes\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"length\\": null, \\"quality\\": null, \\"no_contig_ec\\": true, \\"no_scaffold_ec\\": false}, \\"log_out\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2a.2+galaxy2","type":"tool","uuid":"b3b3bfbc-39b4-4302-967a-220420e3fd36","when":null,"workflow_outputs":[{"label":"YAHS on input dataset(s): Final scaffolds agp output","output_name":"final_agp_out","uuid":"bf7b1438-ec6d-4da7-9699-feb6a70a3234"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy1","errors":null,"id":15,"input_connections":{"input":{"id":13,"output_name":"Merged Hi-C Alignments"}},"inputs":[],"label":null,"name":"PretextMap","outputs":[{"name":"pretext_map_out","type":"pretext"}],"position":{"left":1695.9718300925392,"top":1239.278936866479},"post_job_actions":{"TagDatasetActionpretext_map_out":{"action_arguments":{"tags":"pretextmap_s1, #s1"},"action_type":"TagDatasetAction","output_name":"pretext_map_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy1","tool_shed_repository":{"changeset_revision":"c6cec5ab35c1","name":"pretext_map","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter\\": {\\"filter_type\\": \\"\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"map_qual\\": \\"20\\", \\"sorting\\": {\\"sortby\\": \\"nosort\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.9+galaxy1","type":"tool","uuid":"357b7d30-1c06-40c5-813d-ea009d3c33e7","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":16,"input_connections":{"find_and_replace_0|replace_pattern":{"id":10,"output_name":"out1"},"infile":{"id":14,"output_name":"final_agp_out"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2009.0757979719058,"top":658.9971487352432},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"scaffold_[0-9]*\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": true, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"4d5d4a06-e906-4c73-8159-050d49158dd3","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.4+galaxy0","errors":null,"id":17,"input_connections":{"input":{"id":15,"output_name":"pretext_map_out"}},"inputs":[],"label":null,"name":"Pretext Snapshot","outputs":[{"name":"pretext_snap_out","type":"input"}],"position":{"left":2029.0667900499222,"top":1119.045538825013},"post_job_actions":{"HideDatasetActionpretext_snap_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pretext_snap_out"},"TagDatasetActionpretext_snap_out":{"action_arguments":{"tags":"pretext_s1"},"action_type":"TagDatasetAction","output_name":"pretext_snap_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.4+galaxy0","tool_shed_repository":{"changeset_revision":"26f2fe7e53a1","name":"pretext_snapshot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"colormap\\": \\"5\\", \\"formats\\": {\\"outformat\\": \\"png\\", \\"__current_case__\\": 0}, \\"grid\\": {\\"showGrid\\": \\"yes\\", \\"__current_case__\\": 0, \\"gridsize\\": \\"1\\", \\"gridcolor\\": \\"black\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mintexels\\": \\"64\\", \\"resolution\\": \\"1000\\", \\"sequencenames\\": false, \\"sequences\\": \\"=full\\", \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.4+galaxy0","type":"tool","uuid":"0d79214f-0f8b-4d4f-b836-9935dd334da8","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":18,"input_connections":{"input_file":{"id":12,"output_name":"output"},"mode_condition|agp_to_path":{"id":16,"output_name":"outfile"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2366.497237130829,"top":169.60721977132718},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Reconciliated Scaffolds: gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"s2_gfa_noseq"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"scaffolding\\", \\"__current_case__\\": 2, \\"agp_to_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"discover_paths\\": false, \\"no_sequence\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"f22b9382-90b7-4a21-9b6a-8ea860e0b61f","when":null,"workflow_outputs":[{"label":"Reconciliated Scaffolds: gfa no sequence","output_name":"output","uuid":"ae66bdd9-9a05-427c-b622-5fecf43f722c"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":19,"input_connections":{"input_file":{"id":12,"output_name":"output"},"mode_condition|agp_to_path":{"id":16,"output_name":"outfile"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2375.1851623715397,"top":524.5980397194448},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Reconciliated Scaffolds: gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"s2_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"scaffolding\\", \\"__current_case__\\": 2, \\"agp_to_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"discover_paths\\": false, \\"no_sequence\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"c9168832-ba68-4d06-bbba-86e6b1b82d5f","when":null,"workflow_outputs":[{"label":"Reconciliated Scaffolds: gfa","output_name":"output","uuid":"61df9811-7dff-4c08-a06f-635d9f857a95"}]},"20":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":20,"input_connections":{"input":{"id":17,"output_name":"pretext_snap_out"}},"inputs":[],"label":null,"name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2287.019437743355,"top":1336.1150045876932},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"pretext_s1"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"first\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"d99db0c7-f705-4861-92a8-2f49b3e597e4","when":null,"workflow_outputs":[{"label":"Pretext Map Before HiC scaffolding","output_name":"output","uuid":"40c397f4-282b-46b3-8e73-56475a411cf5"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":21,"input_connections":{"input_file":{"id":19,"output_name":"output"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2610.6208530840718,"top":788.2900418666428},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"s2_fasta"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"d59cf1b1-67dd-4b31-856a-28061e27e0b7","when":null,"workflow_outputs":[{"label":"Reconciliated Scaffolds: fasta","output_name":"output","uuid":"85707e8b-f3ab-41ba-a761-a1b5effed8d5"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":22,"input_connections":{"input_file":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3220.0028253083537,"top":478.6222575397917},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_scaffolds_s2, #s2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"s\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"6b973613-a8f7-497a-a7e1-203affa3690b","when":null,"workflow_outputs":[{"label":"Scaffold sizes for s2","output_name":"stats","uuid":"b651f4bd-5dc8-4a29-a984-b6a2c10ff0b4"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":23,"input_connections":{"input_file":{"id":19,"output_name":"output"},"mode_condition|statistics_condition|expected_genomesize":{"id":11,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3192.3016722361835,"top":1433.9523540523207},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_s2, #s2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"80bbd119-3f2b-4bdf-86ca-c1c1871031c8","when":null,"workflow_outputs":[{"label":"Assembly Statistics for s2","output_name":"stats","uuid":"373bb382-8584-4ebd-a4a1-9bb87bd50f58"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy1","errors":null,"id":24,"input_connections":{"fastq_input|fastq_input1":{"id":13,"output_name":"Trimmed Hi-C data"},"reference_source|ref_file":{"id":21,"output_name":"output"}},"inputs":[],"label":null,"name":"BWA-MEM2","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":3024.3896307623727,"top":990.659429713819},"post_job_actions":{"HideDatasetActionbam_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bam_output"},"RenameDatasetActionbam_output":{"action_arguments":{"newname":"Merged Hi-C Alignments"},"action_type":"RenameDatasetAction","output_name":"bam_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy1","tool_shed_repository":{"changeset_revision":"bfaa0d22c2e4","name":"bwa_mem2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"full\\", \\"__current_case__\\": 4, \\"algorithmic_options\\": {\\"algorithmic_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"k\\": \\"19\\", \\"w\\": \\"100\\", \\"d\\": \\"100\\", \\"r\\": \\"1.5\\", \\"y\\": \\"20\\", \\"c\\": \\"500\\", \\"D\\": \\"0.5\\", \\"W\\": \\"0\\", \\"m\\": \\"50\\", \\"S\\": true, \\"P\\": true, \\"e\\": false}, \\"scoring_options\\": {\\"scoring_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"io_options\\": {\\"io_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"five\\": false, \\"q\\": false, \\"T\\": \\"20\\", \\"h\\": \\"5\\", \\"a\\": true, \\"C\\": false, \\"V\\": false, \\"Y\\": false, \\"M\\": false, \\"K\\": null}}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"iset_stats\\": null}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1+galaxy1","type":"tool","uuid":"6b35b1f3-b654-4ef2-82cc-ea9d5c55f0ac","when":null,"workflow_outputs":[{"label":"Hi-C Alignments","output_name":"bam_output","uuid":"16019374-1e25-4b5b-a89e-a9a4fe7dcb95"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":25,"input_connections":{"input":{"id":21,"output_name":"output"},"lineage_conditional|cached_db":{"id":4,"output_name":"output"},"lineage|lineage_dataset":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"}],"position":{"left":3655.4017691809954,"top":502.80723157808995},"post_job_actions":{"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_s2_missing"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_s2_summ"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":" busco_s2_full"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_s2_img"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"c30b907f-05c9-4709-a939-b66f52daf6e4","when":null,"workflow_outputs":[{"label":"Busco Summary image","output_name":"summary_image","uuid":"cf7a5b93-9021-4029-8107-015787fa2ca5"},{"label":"Busco Summary","output_name":"busco_sum","uuid":"49589149-e16f-4a88-9d84-c1b06288021f"}]},"26":{"annotation":"","id":26,"input_connections":{"gfa_stats":{"id":22,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":3864.7200568670196,"top":246.43472522442866},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7032c6a5-c663-4966-bd5a-ccc245d0d438","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"e5ef68c9-e4c5-4154-ae94-d3e69d4e1ad1","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"187c2b5d-e55c-42c1-b492-0b2a7322615c","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"dbb9b471-00f2-418e-81cc-eb1b6bd289a5","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"2cc6118c-3db3-48d8-8e7a-14a69ebf8702","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"afa7d423-3a43-4391-9ab6-b64beed063ae","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"108303be-4124-4990-bbd8-9973fef9e101","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"ee776542-b181-4dc3-8a15-eae6c1958292","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"409ff147-cd33-4832-b28f-8f1ecc64974c"}]}},"tags":[],"uuid":"11876808-b502-4dd0-9981-e29771f3e53f"},"tool_id":null,"type":"subworkflow","uuid":"d0ccdae0-c2ae-49f7-a517-87a51f835252","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":27,"input_connections":{"infile":{"id":23,"output_name":"stats"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3543.108297125495,"top":1713.1581326877538},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"4b0a88b6-47cc-4381-949c-fc18a99799ac","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"6013195c-2adc-4f6f-8193-199f97b711c9"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/3.1.1.0","errors":null,"id":28,"input_connections":{"inputFile":{"id":24,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"MergeSamFiles","outputs":[{"name":"outFile","type":"bam"}],"position":{"left":3388.8964328961006,"top":902.6748672256023},"post_job_actions":{"TagDatasetActionoutFile":{"action_arguments":{"tags":"merged_hic_mapping_s2"},"action_type":"TagDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"comments\\": [], \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_sequence_dictionaries\\": false, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"816584b3-16a8-47af-99f8-d83cfe07f793","when":null,"workflow_outputs":[{"label":"s2 Merged Hi-C Alignments","output_name":"outFile","uuid":"ef9671eb-6d8f-438b-9255-0c6b9880ad35"}]},"29":{"annotation":"","content_id":"Cut1","errors":null,"id":29,"input_connections":{"input":{"id":26,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4303.912125964182,"top":0},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"a2a87cfa-4202-4755-bb85-0c7eef887084","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"Cut1","errors":null,"id":30,"input_connections":{"input":{"id":26,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4316.01460236021,"top":231.4900897412639},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"f722cb99-55a1-4de3-8631-1093acad3a38","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","errors":null,"id":31,"input_connections":{"input":{"id":28,"output_name":"outFile"}},"inputs":[],"label":null,"name":"bedtools BAM to BED","outputs":[{"name":"output","type":"bed"}],"position":{"left":3782.111881160324,"top":1187.3026120969967},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"2892111d91f8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"ed_score\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"option\\": \\"-bed12\\", \\"split\\": false, \\"tag\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"a9748992-845b-4ae4-9e6d-cd7b6b63ceb2","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy1","errors":null,"id":32,"input_connections":{"input":{"id":28,"output_name":"outFile"}},"inputs":[],"label":null,"name":"PretextMap","outputs":[{"name":"pretext_map_out","type":"pretext"}],"position":{"left":3792.5747975916825,"top":1430.0825550937245},"post_job_actions":{"TagDatasetActionpretext_map_out":{"action_arguments":{"tags":"pretextmap_s2, #s2"},"action_type":"TagDatasetAction","output_name":"pretext_map_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy1","tool_shed_repository":{"changeset_revision":"c6cec5ab35c1","name":"pretext_map","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter\\": {\\"filter_type\\": \\"\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"map_qual\\": \\"20\\", \\"sorting\\": {\\"sortby\\": \\"nosort\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.9+galaxy1","type":"tool","uuid":"57abb4c5-797e-4604-9121-704867ad3e1e","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":33,"input_connections":{"input1":{"id":29,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":4656.06841167563,"top":1.6453506289526558},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Default\\", \\"__current_case__\\": 0}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"1\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"d2c8aaf8-7924-4536-b2a4-027113575635","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"e6c248d6-fff2-408f-a31e-2077e36b5708"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":34,"input_connections":{"input1":{"id":30,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":4643.364968494731,"top":285.5996888343209},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Default\\", \\"__current_case__\\": 0}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"1\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"e02afb1a-3e33-46a7-be51-1bb30c86b089","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"6756b2b5-d6f6-4dac-8c08-7d1c00f1f276"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.5+galaxy0","errors":null,"id":35,"input_connections":{"infile":{"id":31,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"outfile","type":"input"}],"position":{"left":4209.461255249279,"top":933.2325126274696},"post_job_actions":{"TagDatasetActionoutfile":{"action_arguments":{"tags":"sorted_merged_alignments_s2"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"0\\", \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sortkeys\\": [{\\"__index__\\": 0, \\"column\\": \\"4\\", \\"order\\": \\"\\", \\"style\\": \\"\\"}], \\"unique\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"88684eea-7e72-40e5-8410-d9d31a90792f","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.4+galaxy0","errors":null,"id":36,"input_connections":{"input":{"id":32,"output_name":"pretext_map_out"}},"inputs":[],"label":null,"name":"Pretext Snapshot","outputs":[{"name":"pretext_snap_out","type":"input"}],"position":{"left":4065.2630146526653,"top":1530.8383632305972},"post_job_actions":{"HideDatasetActionpretext_snap_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pretext_snap_out"},"TagDatasetActionpretext_snap_out":{"action_arguments":{"tags":"pretext_s2"},"action_type":"TagDatasetAction","output_name":"pretext_snap_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.4+galaxy0","tool_shed_repository":{"changeset_revision":"26f2fe7e53a1","name":"pretext_snapshot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"colormap\\": \\"5\\", \\"formats\\": {\\"outformat\\": \\"png\\", \\"__current_case__\\": 0}, \\"grid\\": {\\"showGrid\\": \\"yes\\", \\"__current_case__\\": 0, \\"gridsize\\": \\"1\\", \\"gridcolor\\": \\"black\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mintexels\\": \\"64\\", \\"resolution\\": \\"1000\\", \\"sequencenames\\": false, \\"sequences\\": \\"=full\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.4+galaxy0","type":"tool","uuid":"04d0356b-428e-4801-875c-6b64cac4d530","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":37,"input_connections":{"input":{"id":36,"output_name":"pretext_snap_out"}},"inputs":[],"label":null,"name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4341.157886975588,"top":1351.6133474498854},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"pretext_s2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"first\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"a13de545-44b8-444c-b457-c026b6051203","when":null,"workflow_outputs":[{"label":"Pretext Map After HiC scaffolding","output_name":"output","uuid":"d7fed6f0-7fb1-4a9b-a112-bce5bc04e873"}]}},"tags":["VGP_curated"],"uuid":"25ccf329-b18b-4c85-8d24-7f2f7ba5843f","version":3},"readme":"# Scaffolding with HiC data\\n\\nThis workflow perfoms scaffolding using HiC data with YAHS. It is designed to be run as part of one the VGP analysis trajectories. \\nExample of trajectory : \\n- VGP1 : Kmer profiling \\n- VGP4 : Genome assembly with HiC phasing\\n- VGP6 : Purge duplicated haplotigs\\n- VGP8 : Scaffolding with HiC\\n\\n## Inputs\\n\\n1. Scaffolded assembly [gfa]\\n2. Name of the haplotype\\n3. HiC reads paired collection [fastq]\\n5. Trim Hi-C data? If `yes` Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\". \\n6. Database for busco lineage (recommended: latest)\\n7. Busco lineage (recommended: vertebrata)\\n8. Restriction enzyme sequence (recommended for VGP data: Arima Hi-C 2.0)\\n9. Estimated genome size [txt]\\n\\n\\n### Outputs\\n\\n1. Scaffolds in [fasta] and [gfa] format\\n2. If you selected `yes` for Hi-C trimming, the trimmed collections of Hi-C reads\\n3. QC: Assembly statistics\\n4. QC: Nx plot\\n5. QC: Size plot\\n6. QC: BUSCO report\\n7. QC: Pretext Maps before and after scaffolding\\n","changelog":"# Changelog\\n\\n## [1.2] - 2025-03-17\\n\\n### Changes\\n\\n- Remove the unnecessary sorting of alignments by names that was slowing down the workflow.\\n\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [1.1] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.5+galaxy0`\\n\\n## [1.0] - 2025-02-13\\n\\n### Changes\\n\\n - Remove the tool `filter and merge` and replace it by the use of the -SP option in bwa-mem to skip mate rescue and pairing. \\n - Hi-C data are now input as a paired collection\\n - Generate lighter `gfa` file without sequences\\n \\n### Updates \\n\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n\\n## [0.3] - 2025-01-30\\n\\n### Changes\\n\\n- Hi-C data are now input as collections\\n- Add optional trimming of Hi-C data\\n- Set Metaeuk as Busco gene predictor\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.4+galaxy0`\\n\\n\\n## [0.2.8] - 2024-09-12\\n\\n- Add suffix to the scaffold names corresponding to the haplotype being assembled\\n\\n## [0.2.7] - 2024-08-13\\n\\n- Expose Busco lineage database parameter\\n\\n## [0.2.6] - 2024-08-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy2`\\n\\n## [0.2.5] - 2024-07-31\\n\\n- Addition of Workflow report\\n\\n## [0.2.4] - 2024-05-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n\\n## [0.2.3] - 2024-04-01\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy1`\\n\\n## [0.2.2] - 2024-02-05\\n\\n### Manual update\\n\\n- remove tag filtering\\n- set \\"Show grid\\" of Pretext Snapshot to \\"yes\\"\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/9.3+galaxy0`\\n\\n\\n## [0.2.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.2] - 2023-11-15\\n\\n### Added\\n\\n- more descriptive labels for inputs\\n\\n## [0.1.2] - 2023-11-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/yahs/yahs/1.2a.2+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.1.1] - 2023-10-26\\n\\nAdded tags for better visibility of outputs in the history, and exposure of the BUSCO lineage parameter\\n\\n## [0.1] - 2023-09-27\\n\\nCreation of the workflow and tests\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Scaffolding with Hi-C data VGP8\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input GFA\\"]@{ shape: doc }\\n1[\\"ℹ️ Haplotype\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Hi-C reads\\"]@{ shape: docs }\\n3[\\"ℹ️ Trim Hi-C Data?\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Restriction enzymes\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Estimated genome size - Parameter File\\"]@{ shape: doc }\\n8[\\"ℹ️ SAK input file\\"]@{ shape: doc }\\n9[\\"ℹ️ Sequence graph\\"]@{ shape: doc }\\n10[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 10\\n11[\\"Parse parameter value\\"]@{ shape: process }\\n7 --> 11\\n12[\\"gfastats\\"]@{ shape: process }\\n0 --> 12\\n8 --> 12\\n13[\\"🛠️ Trim and Align Hi-C paired collection\\"]@{ shape: subprocess }\\n3 --> 13\\n2 --> 13\\n12 --> 13\\n14[\\"YAHS\\"]@{ shape: process }\\n9 --> 14\\n13 --> 14\\n6 --> 14\\n12 --> 14\\n15[\\"PretextMap\\"]@{ shape: process }\\n13 --> 15\\n16[\\"Replace\\"]@{ shape: process }\\n10 --> 16\\n14 --> 16\\n17[\\"Pretext Snapshot\\"]@{ shape: process }\\n15 --> 17\\n18[\\"gfastats\\"]@{ shape: process }\\n12 --> 18\\n16 --> 18\\n19[\\"gfastats\\"]@{ shape: process }\\n12 --> 19\\n16 --> 19\\n20[\\"Extract dataset\\"]@{ shape: process }\\n17 --> 20\\n21[\\"gfastats\\"]@{ shape: process }\\n19 --> 21\\n8 --> 21\\n22[\\"gfastats\\"]@{ shape: process }\\n19 --> 22\\n23[\\"gfastats\\"]@{ shape: process }\\n19 --> 23\\n11 --> 23\\n24[\\"BWA-MEM2\\"]@{ shape: process }\\n13 --> 24\\n21 --> 24\\n25[\\"Busco\\"]@{ shape: process }\\n21 --> 25\\n4 --> 25\\n5 --> 25\\n26[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n22 --> 26\\n27[\\"Replace\\"]@{ shape: process }\\n23 --> 27\\n28[\\"MergeSamFiles\\"]@{ shape: process }\\n24 --> 28\\n29[\\"Cut\\"]@{ shape: process }\\n26 --> 29\\n30[\\"Cut\\"]@{ shape: process }\\n26 --> 30\\n31[\\"bedtools BAM to BED\\"]@{ shape: process }\\n28 --> 31\\n32[\\"PretextMap\\"]@{ shape: process }\\n28 --> 32\\n33[\\"Nx Plot\\"]@{ shape: process }\\n29 --> 33\\n34[\\"Size Plot\\"]@{ shape: process }\\n30 --> 34\\n35[\\"Sort\\"]@{ shape: process }\\n31 --> 35\\n36[\\"Pretext Snapshot\\"]@{ shape: process }\\n32 --> 36\\n37[\\"Extract dataset\\"]@{ shape: process }\\n36 --> 37\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n\\n## Trim and Align Hi-C paired collection\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Reference\\"]@{ shape: doc }\\n1[\\"ℹ️ Hi-C reads\\"]@{ shape: docs }\\n2[\\"ℹ️ Do you want to trim the Hi-C data?\\"]@{ shape: lean-l }\\n3[\\"Trim Hi-C reads 2\\"]@{ shape: process }\\n1 --> 3\\n2 --> 3\\n4[\\"Pick parameter value\\"]@{ shape: process }\\n3 --> 4\\n1 --> 4\\n5[\\"BWA-MEM2\\"]@{ shape: process }\\n4 --> 5\\n0 --> 5\\n6[\\"MergeSamFiles\\"]@{ shape: process }\\n5 --> 6\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Scaffolding-HiC-VGP8/main","dockstore_id":24128,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078489","tests":[{"doc":"Test outline for Scaffolding-HiC-VGP8","job":{"Input GFA":{"class":"File","location":"https://zenodo.org/records/10037496/files/input%20GFA.gfa1","filetype":"gfa1"},"Hi-C reads":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"Hi-C reads","elements":[{"identifier":"forward","class":"File","path":"https://zenodo.org/records/10037496/files/HiC%20Forward%20reads.fastqsanger.gz"},{"identifier":"reverse","class":"File","path":"https://zenodo.org/records/10037496/files/HiC%20reverse%20reads.fastqsanger.gz"}]}]},"Estimated genome size - Parameter File":{"class":"File","location":"https://zenodo.org/records/10037496/files/Estimated%20genome%20size%20-%20Parameter%20File.txt","filetype":"txt"},"Database for Busco Lineage":"v5","Lineage":"vertebrata_odb10","Restriction enzymes":"arima2","Haplotype":"hap1","Trim Hi-C Data?":false},"outputs":{"Reconciliated Scaffolds: gfa":{"asserts":{"has_n_lines":{"n":168}}},"Reconciliated Scaffolds: fasta":{"asserts":{"has_n_lines":{"n":166},"has_text":{"text":"scaffold_53.hap1"}}},"Busco Summary":{"asserts":{"has_text":{"text":"C:1.1%[S:1.0%,D:0.0%],F:0.4%,M:98.5%,n:3354"}}},"Assembly Statistics for s2":{"asserts":{"has_line":{"line":"# scaffolds\\t83"}}},"Nx Plot":{"asserts":{"has_size":{"value":47000,"delta":5000}}},"Pretext Map After HiC scaffolding":{"asserts":{"has_size":{"value":1200000,"delta":200000}}}}}]}],"path":"./workflows/VGP-assembly-v2/Scaffolding-HiC-VGP8"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Assembly-Hifi-Trio-phasing-VGP5.ga","testParameterFiles":["/Assembly-Hifi-Trio-phasing-VGP5-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.9.2","name":"Genome Assembly with Pacbio Hifi reads and Trio data for phasing - VGP5","report":{"markdown":"\\n# Workflow Execution Report\\n\\nTime workflow was invoked:\\n\\n```galaxy\\ninvocation_time()\\n```\\nGalaxy version :\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Raw unitig graph\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"raw unitig graph image\\")\\n```\\n\\n## Merqury results\\n\\n### Merqury QV\\n\\n```galaxy\\nhistory_dataset_as_table(output=merqury_stats)\\n```\\n\\n### Merqury completeness\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_stats\\")\\n```\\n\\n### Merqury plots\\n\\n\\nspectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-cn.fl\\")\\n```\\n\\nspectra-asm:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-asm.fl\\")\\n```\\n\\n\\nhap1 spectra-cn:\\n\\n\\r\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_01.spectra-cn.fl\\")\\n```\\r\\n\\n\\nhap2 spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_02.spectra-cn.fl\\")\\n```\\n\\n## BUSCO results (Vertebrata database)\\n\\nHap1\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco Summary Image Hap1\\")\\n```\\n\\nHap2\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco Summary Image Hap2\\")\\n```\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Pacbio Reads Collection : child"}],"label":"Pacbio Reads Collection : child","name":"Input dataset collection","outputs":[],"position":{"left":5.828125,"top":188.5546875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"54f368c3-47cb-4fcf-8271-628becc1051e","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Paternal Illumina reads (hap1)"}],"label":"Paternal Illumina reads (hap1)","name":"Input dataset collection","outputs":[],"position":{"left":9.9375,"top":297.7890625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"4e8847f2-4257-4ccc-a375-edfcbd7451f6","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Maternal Illumina reads (hap2)"}],"label":"Maternal Illumina reads (hap2)","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":428.5859375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"76917fe2-487e-4780-84e7-917615a2e444","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Meryl Database : Child"}],"label":"Meryl Database : Child","name":"Input dataset","outputs":[],"position":{"left":5.828125,"top":558.4765625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a67ad2b7-a50f-4e8f-8bb1-a433a7662c45","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"","name":"Hapmer Database : Paternal"}],"label":"Hapmer Database : Paternal","name":"Input dataset","outputs":[],"position":{"left":0.625,"top":665.9921875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"1575eb2b-6dd8-4d11-af33-9e1f8f023c6a","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"","name":"Hapmer Database : Maternal"}],"label":"Hapmer Database : Maternal","name":"Input dataset","outputs":[],"position":{"left":1.921875,"top":774.3828125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"3fa39765-f349-4465-a2d1-947fdbad21ba","when":null,"workflow_outputs":[]},"6":{"annotation":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","name":"Bits for bloom filter"}],"label":"Bits for bloom filter","name":"Input parameter","outputs":[],"position":{"left":33.17695496536256,"top":890.2068997739406},"tool_id":null,"tool_state":"{\\"default\\": 37, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"a5987ed3-6fdb-49c7-92d0-5b2f78a9bbb9","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":28.142955132748718,"top":1078.9295496884674},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ca553be3-e5c9-4770-bd61-d50ed05c69ae","when":null,"workflow_outputs":[]},"8":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":61.152110333438976,"top":1189.4200518565176},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"9762d451-eb6f-4f0c-8203-7094f46ea771","when":null,"workflow_outputs":[]},"9":{"annotation":"If empty, read coverage will be estimated from the Genomescope parameters. ","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"If empty, read coverage will be estimated from the Genomescope parameters. ","name":"Homozygous Read Coverage"}],"label":"Homozygous Read Coverage","name":"Input parameter","outputs":[],"position":{"left":98.92882500063175,"top":1275.4014318043698},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"ffcf874f-594d-4d58-99dc-73ef566bcc88","when":null,"workflow_outputs":[]},"10":{"annotation":"GenomeScope model parameters generated by K-mer profiling workflow","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"GenomeScope model parameters generated by K-mer profiling workflow","name":"Genomescope Model Parameters"}],"label":"Genomescope Model Parameters","name":"Input dataset","outputs":[],"position":{"left":146.82808789081685,"top":1368.9007539001223},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ae6dbf08-a610-4d51-9995-8211c8ca5a81","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"","name":"Genomescope Summary"}],"label":"Genomescope Summary","name":"Input dataset","outputs":[],"position":{"left":217.51082966945373,"top":1473.9095926545822},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"81d79a21-1ba4-415c-a6ce-8a8e04ec6f59","when":null,"workflow_outputs":[]},"12":{"annotation":"--trio-dual option","content_id":null,"errors":null,"id":12,"input_connections":{},"inputs":[{"description":"--trio-dual option","name":"Utilize homology information to correct trio-phasing errors"}],"label":"Utilize homology information to correct trio-phasing errors","name":"Input parameter","outputs":[],"position":{"left":306.5808483755939,"top":1565.8147646057243},"tool_id":null,"tool_state":"{\\"default\\": true, \\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"eb8c0bcc-14c8-4b3d-b73f-442eae454b82","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":null,"errors":null,"id":13,"input_connections":{},"inputs":[{"description":"","name":"Name for Haplotype 1"}],"label":"Name for Haplotype 1","name":"Input parameter","outputs":[],"position":{"left":4518.28125,"top":939.7734375},"tool_id":null,"tool_state":"{\\"default\\": \\"Hap1\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c5651e7e-31d4-41b7-a590-e2fadc7f59e9","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":null,"errors":null,"id":14,"input_connections":{},"inputs":[{"description":"","name":"Name for Haplotype 2"}],"label":"Name for Haplotype 2","name":"Input parameter","outputs":[],"position":{"left":4516.25,"top":1056.8359375},"tool_id":null,"tool_state":"{\\"default\\": \\"Hap2\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1ae103f4-38e9-4414-8e46-b8a763159937","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":15,"input_connections":{"library|input_1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":null,"name":"Cutadapt","outputs":[{"name":"out1","type":"fastqsanger"},{"name":"report","type":"txt"},{"name":"json_stats","type":"json"}],"position":{"left":1148.515625,"top":267.7578125},"post_job_actions":{"HideDatasetActionjson_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"json_stats"},"RenameDatasetActionout1":{"action_arguments":{"newname":"Cutadapt on #{library.input_1 }"},"action_type":"RenameDatasetAction","output_name":"out1"},"TagDatasetActionout1":{"action_arguments":{"tags":"trimmed_hifi"},"action_type":"TagDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"35\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": true}, \\"filter_options\\": {\\"discard_trimmed\\": true, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"1\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [], \\"front_adapters\\": [], \\"anywhere_adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT\\"}, \\"single_noindels\\": false}, {\\"__index__\\": 1, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCTTTTCCTCCTCCTCCGTTGTTGTTGTTGAGAGAGAT\\"}, \\"single_noindels\\": false}]}}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"0\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"json_stats\\", \\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"b8770ddb-2ec4-4ef5-8362-6531e559554d","when":null,"workflow_outputs":[{"label":"json_stats","output_name":"json_stats","uuid":"96389258-09f1-4a1a-8e17-0bbae8e999f5"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":16,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":523.0869347835544,"top":911.0977745488228},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"37\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"707e7317-530b-4c47-8c3f-5037373b0e80","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":17,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":427.36900336229525,"top":1298.6882377122847},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c3*2\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"e6e81aeb-6a44-40f8-9d11-7a3dcffd2323","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":18,"input_connections":{"infile":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1909.84375,"top":1620.4453125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-G\\", \\"url_paste\\": \\"Haploid\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"d199b295-3372-4be9-8550-415ca90d5abd","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":19,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"boolean_param","type":"expression.json"}],"position":{"left":727.9970202699099,"top":1601.2400817249174},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"boolean\\", \\"__current_case__\\": 3, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": true}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"e39a1d1b-1641-4606-be9b-87ac9c118da6","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":20,"input_connections":{"results_0|software_cond|input":{"id":15,"output_name":"report"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1523.27734375,"top":79.2265625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"b64b5b58-cf21-4fb6-af57-2e8a56e85142","when":null,"workflow_outputs":[{"label":"multiqc html report","output_name":"html_report","uuid":"9bd9c4ff-97c3-4557-a550-a360207e08ab"},{"label":"cutadapt multiqc stats","output_name":"stats","uuid":"110b5141-3025-4ee6-8d4a-d4987fa6dfe5"}]},"21":{"annotation":"","content_id":"Cut1","errors":null,"id":21,"input_connections":{"input":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":697.2536393070069,"top":1146.3570121128503},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"e87a329b-9ff4-4e11-b496-fb6cc01aa334","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":22,"input_connections":{"infile":{"id":18,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2199.15625,"top":1460.4296875},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"bp\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 1, \\"find_pattern\\": \\",\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 2, \\"find_pattern\\": \\"([a-z])\\\\\\\\s+([A-Z])\\", \\"replace_pattern\\": \\"\\\\\\\\1_\\\\\\\\2\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"fd5e8b7e-2a79-4335-aa09-1a4a4ef9dbf2","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":23,"input_connections":{"input1":{"id":21,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":931.1985205684734,"top":1287.8800646673546},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"1d774f9d-e850-4ebb-bf5a-952030b23978","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"Convert characters1","errors":null,"id":24,"input_connections":{"input":{"id":22,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Convert","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2462.265625,"top":1292.3203125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Convert characters1","tool_state":"{\\"condense\\": true, \\"convert_from\\": \\"s\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strip\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ad520fed-bb69-4fd0-a72c-53086c9aa58b","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":25,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":9,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":23,"output_name":"integer_param"}},"inputs":[],"label":"Homozygous read coverage for Hifiasm","name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1251.8098747874983,"top":959.7102546174094},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Homozygous read coverage for Hifiasm"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"c45d9a2c-344d-4de1-91d1-72903248cc7e","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"Cut1","errors":null,"id":26,"input_connections":{"input":{"id":24,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2674.140625,"top":1194.9609375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"d3b276c2-8062-4552-9cb9-7f54cf8bb287","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","errors":null,"id":27,"input_connections":{"assembly_options|hom_cov":{"id":25,"output_name":"integer_param"},"filter_bits":{"id":16,"output_name":"integer_param"},"mode|reads":{"id":15,"output_name":"out1"},"mode|trio_dual":{"id":19,"output_name":"boolean_param"},"mode|trioinput|hap1_reads":{"id":1,"output_name":"output"},"mode|trioinput|hap2_reads":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Hifiasm","name":"assembly_options"},{"description":"runtime parameter for tool Hifiasm","name":"mode"},{"description":"runtime parameter for tool Hifiasm","name":"mode"}],"label":null,"name":"Hifiasm","outputs":[{"name":"noseq_files","type":"input"},{"name":"bin_files","type":"input"},{"name":"hap1_contigs","type":"gfa1"},{"name":"hap2_contigs","type":"gfa1"},{"name":"raw_unitigs_trio","type":"gfa1"},{"name":"processed_unitigs_trio","type":"gfa1"},{"name":"log_file","type":"txt"}],"position":{"left":1843.692405931764,"top":411.04687749931213},"post_job_actions":{"TagDatasetActionhap1_contigs":{"action_arguments":{"tags":"hifiasm_trio_hap1"},"action_type":"TagDatasetAction","output_name":"hap1_contigs"},"TagDatasetActionhap2_contigs":{"action_arguments":{"tags":"hifiasm_trio_hap2"},"action_type":"TagDatasetAction","output_name":"hap2_contigs"},"TagDatasetActionlog_file":{"action_arguments":{"tags":"hifiasm_trio_log"},"action_type":"TagDatasetAction","output_name":"log_file"},"TagDatasetActionnoseq files":{"action_arguments":{"tags":"noseq_hifiasm"},"action_type":"TagDatasetAction","output_name":"noseq files"},"TagDatasetActionprocessed_unitigs_trio":{"action_arguments":{"tags":"hifiasm_trio_processed"},"action_type":"TagDatasetAction","output_name":"processed_unitigs_trio"},"TagDatasetActionraw_unitigs_trio":{"action_arguments":{"tags":"hifiasm_trio_raw_unitig"},"action_type":"TagDatasetAction","output_name":"raw_unitigs_trio"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","tool_shed_repository":{"changeset_revision":"170061c052ad","name":"hifiasm","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"advanced_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"assembly_options\\": {\\"assembly_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"cleaning_rounds\\": \\"4\\", \\"adapter_length\\": \\"0\\", \\"pop_contigs\\": \\"10000000\\", \\"pop_unitigs\\": \\"100000\\", \\"remove_tips\\": \\"3\\", \\"max_overlap\\": \\"0.8\\", \\"min_overlap\\": \\"0.2\\", \\"disable_post_join\\": false, \\"ignore_error_corrected\\": false, \\"hom_cov\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"bins_out\\": true, \\"filter_bits\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hic_partition\\": {\\"hic_partition_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"log_out\\": true, \\"mode\\": {\\"mode_selector\\": \\"trio\\", \\"__current_case__\\": 1, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trioinput\\": {\\"trio_input_selector\\": \\"reads\\", \\"__current_case__\\": 0, \\"hap1_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hap2_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"max_kmers\\": \\"2\\", \\"min_kmers\\": \\"5\\", \\"yak_kmer_length\\": \\"31\\", \\"trio_dual\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"ont_integration\\": {\\"ont_integration_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"purge_options\\": {\\"purge_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"scaffolding_options\\": {\\"scaffold_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy0","type":"tool","uuid":"50aee4e7-5172-49a0-a448-4020942301af","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":28,"input_connections":{"input1":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":"Estimated genome size","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":3071.875,"top":1025.8359375},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated Genome size"},"action_type":"RenameDatasetAction","output_name":"integer_param"},"TagDatasetActioninteger_param":{"action_arguments":{"tags":"estimated_genome_size"},"action_type":"TagDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"f4f9a27e-f8fc-4cc2-8bea-94a2bf225b66","when":null,"workflow_outputs":[{"label":"Estimated Genome size","output_name":"integer_param","uuid":"5ae5a387-d898-4272-bc29-608e5fc33655"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":29,"input_connections":{"input_file":{"id":27,"output_name":"hap1_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2292.796875,"top":0},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"usable hap1 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"trio_hap1_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"ae31efe9-7f56-4a2d-b811-d77d4f8b0abf","when":null,"workflow_outputs":[{"label":"usable hap1 gfa","output_name":"output","uuid":"4fc29101-1e84-407c-8e97-4899fda4300c"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":30,"input_connections":{"input_file":{"id":27,"output_name":"hap2_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"No Sequence hap2 gfa","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2289.15625,"top":289.3125},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"No Sequence hap2 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"trio_hap2_gfa_noseq"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": true}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"f2a02b12-43b7-4969-a999-dbc18b820c2f","when":null,"workflow_outputs":[{"label":"No Sequence hap2 gfa","output_name":"output","uuid":"10b452af-b1fa-4dbe-b2ed-eeeab8dfb650"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":31,"input_connections":{"input_file":{"id":27,"output_name":"raw_unitigs_trio"}},"inputs":[],"label":"Raw Unitig Image","name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":2274.525103645694,"top":773.0738945855107},"post_job_actions":{"TagDatasetActionoutfile":{"action_arguments":{"tags":"utg_png"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"2000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": \\"25.0\\", \\"output_format\\": \\"png\\", \\"width\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"f9af6aa8-4ee4-4d4d-a54f-653881706285","when":null,"workflow_outputs":[{"label":"raw unitig graph image","output_name":"outfile","uuid":"c5d8c5b8-3d33-473d-96bc-cedd33d14786"}]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":32,"input_connections":{"input_file":{"id":27,"output_name":"hap1_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"No Sequence hap1 gfa","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2589.3287059192635,"top":3.551765520199534},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"No Sequence hap1 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"trio_hap1_gfa_noseq"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": true}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"6f1276d9-1e55-49cb-9a12-2b2a680443bd","when":null,"workflow_outputs":[{"label":"No Sequence hap1 gfa","output_name":"output","uuid":"8abadd24-eb18-4a90-b628-6607b47c503f"}]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":33,"input_connections":{"input_file":{"id":27,"output_name":"hap2_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2601.996610470895,"top":270.42086615372415},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"usable hap2 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"trio_hap2_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"3f087297-a3c5-4f93-9d85-95e954252369","when":null,"workflow_outputs":[{"label":"usable hap2 gfa","output_name":"output","uuid":"ca9f8434-d7e9-4cf7-84b8-ce9cad5548f0"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":34,"input_connections":{"input_file":{"id":27,"output_name":"hap1_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":3131.875,"top":321.6796875},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm Trio hap1"},"action_type":"RenameDatasetAction","output_name":"output"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Hifiasm Trio hap1"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_trio_hap1"},"action_type":"TagDatasetAction","output_name":"output"},"TagDatasetActionstats":{"action_arguments":{"tags":"hifiasm_trio_hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"653200d3-c548-4dd2-87cb-e8b40fbc8235","when":null,"workflow_outputs":[{"label":"Hifiasm Trio hap1","output_name":"output","uuid":"711071f5-4b23-4a9f-af19-abcf4b731da9"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":35,"input_connections":{"input_file":{"id":27,"output_name":"hap2_contigs"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":3117.859375,"top":644.6484375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm Trio hap2"},"action_type":"RenameDatasetAction","output_name":"output"},"RenameDatasetActionstats":{"action_arguments":{"newname":"Hifiasm Trio hap1"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_trio_hap2"},"action_type":"TagDatasetAction","output_name":"output"},"TagDatasetActionstats":{"action_arguments":{"tags":"hifiasm_trio_hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"77c35546-0668-45d5-aa10-2c5048fd4256","when":null,"workflow_outputs":[{"label":"Hifiasm Trio hap2","output_name":"output","uuid":"f4438ca6-d970-4263-ad68-f6a488b56bbb"}]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":36,"input_connections":{"input_file":{"id":27,"output_name":"hap1_contigs"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3892.703125,"top":516.5390625},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_hap1, #hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"fb185c69-e1a1-47a4-b983-d25d94f25522","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":37,"input_connections":{"input_file":{"id":27,"output_name":"hap2_contigs"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3886.140625,"top":907.5078125},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_hap2, #hap2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"a3de4200-b7d2-4cd5-bcca-fa6f548ed6a0","when":null,"workflow_outputs":[]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":38,"input_connections":{"input_file":{"id":27,"output_name":"hap2_contigs"},"mode_condition|statistics_condition|expected_genomesize":{"id":28,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3843.6875,"top":676.8359375},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_hap2, #hap2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"60492710-1411-45ff-abe6-1c94644bcca4","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":39,"input_connections":{"input_file":{"id":27,"output_name":"hap1_contigs"},"mode_condition|statistics_condition|expected_genomesize":{"id":28,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":3919.75,"top":321.0703125},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_hap1, #hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"904feec3-140a-4aff-a8af-29592733175e","when":null,"workflow_outputs":[]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":40,"input_connections":{"input":{"id":34,"output_name":"output"},"lineage_conditional|cached_db":{"id":7,"output_name":"output"},"lineage|lineage_dataset":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":3656.2421875,"top":1226.00390625},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"#hap1,busco_hap1_gff"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_hap1_missing, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_hap1_summ, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_hap1_full, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_hap1_img, #hap1"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"bc06da8e-e5cb-493e-98d1-c974a3e46cff","when":null,"workflow_outputs":[{"label":"Busco Summary Image Hap1","output_name":"summary_image","uuid":"783cb816-dead-4035-911f-d4dd7590efbf"},{"label":"Busco Summary Hap1","output_name":"busco_sum","uuid":"bb3f3b2b-d455-4c53-a4b4-044bd4a90c7a"},{"label":"Busco Gff Hap1","output_name":"busco_gff","uuid":"0946a7b1-b95e-425a-b039-eddb66d7f203"}]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":41,"input_connections":{"input":{"id":35,"output_name":"output"},"lineage_conditional|cached_db":{"id":7,"output_name":"output"},"lineage|lineage_dataset":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":3652.9087639414856,"top":1832.782414405416},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"#hap2,busco_hap2_gff"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_hap2_missing, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_hap2_summ, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_hap2_full, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_hap2_img, #hap2"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"4931c626-b636-43c3-b19b-3c778199b9a8","when":null,"workflow_outputs":[{"label":"Busco Summary Hap2","output_name":"busco_sum","uuid":"b1bae72f-2729-47d4-925c-ddbba82e2098"},{"label":"Busco Gff Hap2","output_name":"busco_gff","uuid":"948be37c-7cde-4c13-805c-2d9806380939"},{"label":"Busco Summary Image Hap2","output_name":"summary_image","uuid":"caf3c70a-11cf-468f-aab8-144a38880d07"}]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","errors":null,"id":42,"input_connections":{"mode|assembly_options|assembly_01":{"id":34,"output_name":"output"},"mode|assembly_options|assembly_02":{"id":35,"output_name":"output"},"mode|meryldb_F1":{"id":3,"output_name":"output"},"mode|meryldb_MAT":{"id":5,"output_name":"output"},"mode|meryldb_PAT":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Merqury","name":"mode"},{"description":"runtime parameter for tool Merqury","name":"mode"},{"description":"runtime parameter for tool Merqury","name":"mode"}],"label":null,"name":"Merqury","outputs":[{"name":"bed_files","type":"input"},{"name":"wig_files","type":"input"},{"name":"qv_files","type":"input"},{"name":"png_files","type":"input"},{"name":"sizes_files","type":"input"},{"name":"stats_files","type":"input"},{"name":"hist_files","type":"input"},{"name":"log_file","type":"txt"}],"position":{"left":3990.2355401989075,"top":1314.0148504041067},"post_job_actions":{"HideDatasetActionlog_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log_file"},"RenameDatasetActionwig_files":{"action_arguments":{"newname":"merqury_trio_wig, #trio"},"action_type":"RenameDatasetAction","output_name":"wig_files"},"TagDatasetActionbed_files":{"action_arguments":{"tags":"merqury_trio_bed"},"action_type":"TagDatasetAction","output_name":"bed_files"},"TagDatasetActionhist_files":{"action_arguments":{"tags":"#trio"},"action_type":"TagDatasetAction","output_name":"hist_files"},"TagDatasetActionpng_files":{"action_arguments":{"tags":"merqury_trio_png, #trio"},"action_type":"TagDatasetAction","output_name":"png_files"},"TagDatasetActionqv_files":{"action_arguments":{"tags":"merqury_trio_qv, #trio"},"action_type":"TagDatasetAction","output_name":"qv_files"},"TagDatasetActionsizes_files":{"action_arguments":{"tags":"merqury_trio_sizes, #trio"},"action_type":"TagDatasetAction","output_name":"sizes_files"},"TagDatasetActionstats_files":{"action_arguments":{"tags":"merqury_trio_stats, #trio"},"action_type":"TagDatasetAction","output_name":"stats_files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","tool_shed_repository":{"changeset_revision":"eb7b00ec347d","name":"merqury","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"label\\": \\"output_merqury\\", \\"mode\\": {\\"options\\": \\"trio\\", \\"__current_case__\\": 1, \\"meryldb_F1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"meryldb_PAT\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"meryldb_MAT\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_options\\": {\\"number_assemblies\\": \\"two\\", \\"__current_case__\\": 1, \\"assembly_01\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_02\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_add_headers\\": true, \\"output_selector\\": [\\"qv\\", \\"plots\\", \\"sizes\\", \\"stats\\", \\"bed\\", \\"wig\\", \\"log\\", \\"hist\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy4","type":"tool","uuid":"25a1c5c3-56b1-490a-b73e-ad8cd94c1c26","when":null,"workflow_outputs":[{"label":"Merqury Images","output_name":"png_files","uuid":"703f1b5c-9fd8-4f49-8dae-b5c3d0770738"},{"label":"Merqury QV","output_name":"qv_files","uuid":"947a9ca1-437f-4003-9871-ad38e76a19f6"},{"label":"Merqury Trio Histogram","output_name":"hist_files","uuid":"82446904-1649-4dbf-ba0d-48576419a32c"}]},"43":{"annotation":"","id":43,"input_connections":{"gfa_stats":{"id":36,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data prep Hap1"}],"label":"Data prep Hap1","name":"gfastats_data_prep","outputs":[],"position":{"left":4881.21875,"top":481.3671875},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"09cefa4f-7cb2-47de-9ec3-d68701a12c71","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"40d1fc30-02a7-458b-b1aa-72c38d5d55ea","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"680af783-3108-4617-a793-276977f7c2ac","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"5c3447fa-bf5d-42b9-8aab-6b80c8da3dda","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"4b662ba7-bc49-4259-a7fa-dfdcf1616dea","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"ed8dca6d-0806-4119-9bb3-e09fdc5947f0","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"b289c325-e9e7-492d-98e8-98580acd14cd","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"bc2c5737-d84f-4347-85ec-a1291e21eee7","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"23d98a71-9790-4cd5-a936-ed6ee6060103"}]}},"tags":[],"uuid":"ea3754aa-6322-42a9-8ab1-7d009ccd46de"},"tool_id":null,"type":"subworkflow","uuid":"813bcf16-aed8-4f57-a64e-37dc7d111cac","when":null,"workflow_outputs":[]},"44":{"annotation":"","id":44,"input_connections":{"gfa_stats":{"id":37,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data Prep Hap2"}],"label":"Data Prep Hap2","name":"gfastats_data_prep","outputs":[],"position":{"left":4884.75,"top":635.0078125},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"28582012-66c6-41cb-8c8d-893aea2f11e5","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"7e8e2879-09cb-429a-a14d-2f4e5492ff65","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"d7aba069-39bd-4f32-bf18-eac978a1a0cf","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"c29a59e2-11d8-415f-989f-756ddfe1bb94","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"c455907e-6018-4b6c-8873-76f6ac866ebf","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"39812bd4-43d7-4e29-baae-43b694a8a2df","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"5a6e3382-a9a0-4937-811b-eef1807a4176","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"28afe0ea-6c13-49a8-99cc-ded6f1feeaca","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"97d0a87d-453f-40db-89b3-e48dce13e49b"}]}},"tags":[],"uuid":"9b855cb3-b937-4223-890c-217388329b0d"},"tool_id":null,"type":"subworkflow","uuid":"6b4dfe1c-e8c9-49d7-a52a-44959e30f3e7","when":null,"workflow_outputs":[]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":45,"input_connections":{"infile":{"id":38,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":4294.579496270747,"top":669.0668107299717},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\thap2-Maternal\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"124adb19-8437-4d65-bbed-41de62ca1788","when":null,"workflow_outputs":[]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":46,"input_connections":{"infile":{"id":39,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":4292.78125,"top":294.7734375},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\thap1-Paternal\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"5786321e-4c58-4ac2-baec-8407aeefac72","when":null,"workflow_outputs":[]},"47":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":47,"input_connections":{"input":{"id":42,"output_name":"qv_files"}},"inputs":[],"label":"merqury_qv","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4601.312365069848,"top":1296.4801576959867},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"111081e7-8909-446e-a3b0-1f6fab8de92e","when":null,"workflow_outputs":[{"label":"merqury_qv","output_name":"output","uuid":"cdbf5aec-670b-4da2-b6f2-f6608ae252f0"}]},"48":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":48,"input_connections":{"input":{"id":42,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4595.0227870283525,"top":1419.242140314577},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"be91dc83-4b56-42cf-95e0-3e44f68cac33","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-cn.fl","output_name":"output","uuid":"dbfe025a-b595-4f5e-80b6-1d94fb363f0c"}]},"49":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":49,"input_connections":{"input":{"id":42,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-asm.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4605.757916018367,"top":1590.818754817862},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-asm.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"dd0f7e82-6bf2-43c6-b53b-ccf00f34ca49","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-asm.fl","output_name":"output","uuid":"c34a15dc-3af9-4bb2-b08e-6c0eae63178f"}]},"50":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":50,"input_connections":{"input":{"id":42,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_01.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4595.029655522313,"top":1757.5278966007343},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_01.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"b0a8b7e9-b28c-4bb7-9f0f-53aa2b52ba4a","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_01.spectra-cn.fl","output_name":"output","uuid":"6241bc27-84fc-4694-aa64-7676ea130542"}]},"51":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":51,"input_connections":{"input":{"id":42,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_02.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4601.294376157093,"top":1945.7697669515674},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_02.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"6752c9bb-3633-4cc7-b505-45cf7264d6f1","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_02.spectra-cn.fl","output_name":"output","uuid":"761c07e7-fc3b-415b-a91a-ee6841355d54"}]},"52":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":52,"input_connections":{"input":{"id":42,"output_name":"stats_files"}},"inputs":[],"label":"merqury_stats","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":4605.861270498924,"top":2140.8315611990333},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.completeness\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"14c501db-2128-4e14-bca5-b8134897004e","when":null,"workflow_outputs":[{"label":"merqury_stats","output_name":"output","uuid":"9b4d468d-235c-4d4b-b687-ee59343549df"}]},"53":{"annotation":"","id":53,"input_connections":{"Alternate data":{"id":44,"input_subworkflow_step_id":1,"output_name":"gfastats data for plotting"},"Name of alternate assembly":{"id":14,"input_subworkflow_step_id":3,"output_name":"output"},"Name of primary assembly":{"id":13,"input_subworkflow_step_id":2,"output_name":"output"},"Primary data":{"id":43,"input_subworkflow_step_id":0,"output_name":"gfastats data for plotting"}},"inputs":[{"description":"","name":"Plots"}],"label":"Plots","name":"gfastats_plot","outputs":[],"position":{"left":5196.359375,"top":343.4765625},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","name":"gfastats_plot","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Primary data"}],"label":"Primary data","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"eb649656-22ce-4f5a-8689-907504927bed","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Alternate data"}],"label":"Alternate data","name":"Input dataset","outputs":[],"position":{"left":6.960205078125,"top":143.53692626953125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"9f75a894-8ba9-4d05-a0b6-b43f39db9f65","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4.55963134765625,"top":296.3210144042969},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6e74523f-750b-4fc8-b550-baf2a4524110","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"068a9d5c-f82a-467a-8900-0c29b0d735c4"}]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":7.428955078125,"top":440.4403076171875},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"5a976fab-1491-4217-8f91-c92352bab10a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"b0705c7b-2eb6-48e5-9c38-fff300d2b154"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"exp":{"id":2,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1139.7584686279297,"top":107.42897033691406},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"a3bec00c-fdd6-4e7e-9335-9a24656fb7c3","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"exp":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1151.7044982910156,"top":392.65625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"0c2f2aaa-f1dd-4272-96b5-99d1d213b079","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":6,"input_connections":{"inputs":{"id":4,"output_name":"out_file1"},"queries_0|inputs2":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1441.193115234375,"top":216.59091186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"c02c7b07-b7b3-4f03-8b05-a23a4210e109","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1780.3265991210938,"top":182.99716186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"a671894f-e5e4-46ca-ae09-0637961e8b46","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1781.349365234375,"top":387.96875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"57b84cc1-e745-46b1-bf0f-acf263eb9c1b","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2068.1390991210938,"top":31.988632202148438},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"79e4b8a2-c9bf-4cf0-b943-7de18ec50ed6","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"8039e581-f8ae-4239-bd7c-0bc7a3c9b345"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2083.3805541992188,"top":319.8011169433594},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"350f3535-47b5-4870-9f09-1be8f702bc23","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"edb70288-7a24-4499-9f0f-e56aaf796522"}]}},"tags":[],"uuid":"81b31fdd-9c5b-4b53-b27f-68e8fdc65cf4"},"tool_id":null,"type":"subworkflow","uuid":"bc14963b-acf4-4064-ad92-fbf7cad3730f","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"Nx Plot","uuid":"4235c91f-2071-483e-b905-bed50bc53b05"},{"label":"Size Plot","output_name":"Size Plot","uuid":"044f639b-f8f9-44b9-85b0-bbea417e4af7"}]},"54":{"annotation":"","content_id":"join1","errors":null,"id":54,"input_connections":{"input1":{"id":46,"output_name":"outfile"},"input2":{"id":45,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4618.30993263997,"top":35.10939711899408},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"fill_empty\\", \\"__current_case__\\": 1, \\"fill_columns_by\\": \\"fill_unjoined_only\\", \\"do_fill_empty_columns\\": {\\"column_fill_type\\": \\"single_fill_value\\", \\"__current_case__\\": 0, \\"fill_value\\": \\".\\"}}, \\"header\\": \\"\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"5c346836-f071-457a-844f-1a4c9739ccc7","when":null,"workflow_outputs":[]},"55":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":55,"input_connections":{"input":{"id":54,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":5013.618579933229,"top":35.30375914453537},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Assembly statistics for Hap1 and Hap2"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"gfastats_asm_hap1_hap2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": [\\"3\\"]}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b6796224-1789-45fd-8392-3237462c9071","when":null,"workflow_outputs":[{"label":"Assembly statistics for Hap1 and Hap2","output_name":"output","uuid":"473ae324-f8d1-43ee-b57c-9b1a8a9705ec"}]},"56":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":56,"input_connections":{"infile":{"id":55,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":5346.826883826449,"top":86.40715310618705},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"68545a0f-7aa4-43e3-bf1a-5dcb75108c51","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"fe375d8e-3e97-4373-b3ac-0d2e409d0145"}]}},"tags":["VGP","Reviewed"],"uuid":"bbcefc9f-8a6f-4eee-b4d6-9738bbd18eb9","version":1},"readme":"# Genome Assembly with Hifi reads and Trio Data\\n\\nGenerate phased assembly based on PacBio Hifi Reads using parental Illumina data for phasing. Part of the VGP workflow suite, it needs to be run after the Trio k-mer Profiling workflow VGP2.\\n\\n## Inputs\\n\\n1. Hifi long reads [fastq]\\n2. Concatenated Illumina reads : Paternal [fastq]\\n3. Concatenated Illumina reads : Maternal [fastq]\\n4. K-mer database [meryldb] generated by VGP2 workflow.\\n5. Paternal hapmer database [meryldb] generated by VGP2 workflow.\\n6. Maternal hapmer database [meryldb] generated by VGP2 workflow.\\n7. Bits for Bloom Filter. Change for large genomes to save memory.\\n8. Database to use for Busco lineages. Recommended : latest version.\\n8. Lineage. Select the taxonomic lineage of the assembled species.  \\n9.  Homozygous read coverage (Estimated from the Genomescope model if not provided)\\n10. Genome model parameters generated by Genomescope [tabular] generated by VGP2 workflow.\\n11. Genome profile summary generated by Genomescope [txt] generated by VGP2 workflow.\\n12. Name of first haplotype\\n13. Name of second haplotype\\n\\n## Outputs\\n\\n1. Haplotype 1 assembly [fasta] and [gfa]\\n2. Haplotype 2 assembly [fasta] and [gfa]\\n9. QC: Size plot for both assemblies\\n5. QC: MultiQC report for HiFi reads trimming\\n6. QC: BUSCO report for both assemblies\\n7. QC: Merqury report for both assemblies\\n8. QC: Assembly statistics for both assemblies\\n9. QC: Nx plot for both assemblies\\n10. QC: Size plot for both assemblies\\n\\n","changelog":"# Changelog\\n\\n## [0.9.2] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.9.1] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.9.0] - 2025-02-26\\n\\n### Changes\\n\\n- Output bin files from Hifiasm. Helpful for debugging\\n- Output lighter gfas without sequences along with the regular gfas \\n\\n## [0.8] - 2025-02-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.7] - 2025-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy1`\\n\\n## [0.6] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n## [0.5] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.4] - 2024-12-16\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1`\\n\\n## [0.3] - 2024-11-11\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n\\n## [0.2] - 2024-08-12\\n\\n### Added\\n\\n- Workflow report\\n\\n### Changed \\n\\n- Expose the Busco lineage database parameter\\n- Fix a bug happening when merging assembly statistics : The presence of the \'#\' character caused the tables to join incorrectly, and the numbers did not match the metrics\\n\\n## [0.1.8] - 2024-07-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n\\n## [0.1.7] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0`\\n\\n## [0.1.6] - 2024-05-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n\\n## [0.1.5] - 2024-05-06\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n\\n## [0.1.4] - 2024-04-22\\n\\n### Added\\n\\n- Add Homozygous Read coverage parameter and calculation\\n- Expose Lineage parameter\\n\\n### Automatic update\\n\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n\\n## [0.1.3] - 2024-03-25\\n\\n### Manual update\\n\\n- Add gff output to Busco\\n- Add histogram output to Merqury\\n- Label subworkflows\\n\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1`\\n\\n## [0.1.2] - 2023-11-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.16.1+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.1.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n\\n## [0.1] - 2023-09-27\\n\\nCreation of workflow for Trio assembly. \\n\\n","diagrams":"# Workflow diagrams\\n\\n## Genome Assembly with Pacbio Hifi reads and Trio data for phasing - VGP5\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Pacbio Reads Collection : child\\"]@{ shape: docs }\\n1[\\"ℹ️ Paternal Illumina reads (hap1)\\"]@{ shape: docs }\\n2[\\"ℹ️ Maternal Illumina reads (hap2)\\"]@{ shape: docs }\\n3[\\"ℹ️ Meryl Database : Child\\"]@{ shape: doc }\\n4[\\"ℹ️ Hapmer Database : Paternal\\"]@{ shape: doc }\\n5[\\"ℹ️ Hapmer Database : Maternal\\"]@{ shape: doc }\\n6[\\"ℹ️ Bits for bloom filter\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Homozygous Read Coverage\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Genomescope Model Parameters\\"]@{ shape: doc }\\n11[\\"ℹ️ Genomescope Summary\\"]@{ shape: doc }\\n12[\\"ℹ️ Utilize homology information to correct trio-phasing errors\\"]@{ shape: lean-l }\\n13[\\"ℹ️ Name for Haplotype 1\\"]@{ shape: lean-l }\\n14[\\"ℹ️ Name for Haplotype 2\\"]@{ shape: lean-l }\\n15[\\"Cutadapt\\"]@{ shape: process }\\n0 --> 15\\n16[\\"Pick parameter value\\"]@{ shape: process }\\n6 --> 16\\n17[\\"Compute\\"]@{ shape: process }\\n10 --> 17\\n18[\\"Search in textfiles\\"]@{ shape: process }\\n11 --> 18\\n19[\\"Pick parameter value\\"]@{ shape: process }\\n12 --> 19\\n20[\\"MultiQC\\"]@{ shape: process }\\n15 --> 20\\n21[\\"Cut\\"]@{ shape: process }\\n17 --> 21\\n22[\\"Replace Text\\"]@{ shape: process }\\n18 --> 22\\n23[\\"Parse parameter value\\"]@{ shape: process }\\n21 --> 23\\n24[\\"Convert\\"]@{ shape: process }\\n22 --> 24\\n25[\\"Homozygous read coverage for Hifiasm\\"]@{ shape: process }\\n9 --> 25\\n23 --> 25\\n26[\\"Cut\\"]@{ shape: process }\\n24 --> 26\\n27[\\"Hifiasm\\"]@{ shape: process }\\n25 --> 27\\n16 --> 27\\n15 --> 27\\n19 --> 27\\n1 --> 27\\n2 --> 27\\n28[\\"Estimated genome size\\"]@{ shape: process }\\n26 --> 28\\n29[\\"gfastats\\"]@{ shape: process }\\n27 --> 29\\n30[\\"No Sequence hap2 gfa\\"]@{ shape: process }\\n27 --> 30\\n31[\\"Raw Unitig Image\\"]@{ shape: process }\\n27 --> 31\\n32[\\"No Sequence hap1 gfa\\"]@{ shape: process }\\n27 --> 32\\n33[\\"gfastats\\"]@{ shape: process }\\n27 --> 33\\n34[\\"gfastats\\"]@{ shape: process }\\n27 --> 34\\n35[\\"gfastats\\"]@{ shape: process }\\n27 --> 35\\n36[\\"gfastats\\"]@{ shape: process }\\n27 --> 36\\n37[\\"gfastats\\"]@{ shape: process }\\n27 --> 37\\n38[\\"gfastats\\"]@{ shape: process }\\n27 --> 38\\n28 --> 38\\n39[\\"gfastats\\"]@{ shape: process }\\n27 --> 39\\n28 --> 39\\n40[\\"Busco\\"]@{ shape: process }\\n34 --> 40\\n7 --> 40\\n8 --> 40\\n41[\\"Busco\\"]@{ shape: process }\\n35 --> 41\\n7 --> 41\\n8 --> 41\\n42[\\"Merqury\\"]@{ shape: process }\\n34 --> 42\\n35 --> 42\\n3 --> 42\\n5 --> 42\\n4 --> 42\\n43[\\"🛠️ Data prep Hap1\\"]@{ shape: subprocess }\\n36 --> 43\\n44[\\"🛠️ Data Prep Hap2\\"]@{ shape: subprocess }\\n37 --> 44\\n45[\\"Text reformatting\\"]@{ shape: process }\\n38 --> 45\\n46[\\"Text reformatting\\"]@{ shape: process }\\n39 --> 46\\n47[\\"merqury_qv\\"]@{ shape: process }\\n42 --> 47\\n48[\\"output_merqury.spectra-cn.fl\\"]@{ shape: process }\\n42 --> 48\\n49[\\"output_merqury.spectra-asm.fl\\"]@{ shape: process }\\n42 --> 49\\n50[\\"output_merqury.assembly_01.spectra-cn.fl\\"]@{ shape: process }\\n42 --> 50\\n51[\\"output_merqury.assembly_02.spectra-cn.fl\\"]@{ shape: process }\\n42 --> 51\\n52[\\"merqury_stats\\"]@{ shape: process }\\n42 --> 52\\n53[\\"🛠️ Plots\\"]@{ shape: subprocess }\\n44 --> 53\\n14 --> 53\\n13 --> 53\\n43 --> 53\\n54[\\"Join two Datasets\\"]@{ shape: process }\\n46 --> 54\\n45 --> 54\\n55[\\"Advanced Cut\\"]@{ shape: process }\\n54 --> 55\\n56[\\"Replace\\"]@{ shape: process }\\n55 --> 56\\n```\\n\\n## gfastats_plot\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Primary data\\"]@{ shape: doc }\\n1[\\"ℹ️ Alternate data\\"]@{ shape: doc }\\n2[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"Concatenate datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Cut\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Nx Plot\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Size Plot\\"]@{ shape: process }\\n8 --> 10\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Assembly-Hifi-Trio-phasing-VGP5/main","dockstore_id":24241,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078832","tests":[{"doc":"Test outline for Assembly-Hifi-Trio-phasing-VGP5","job":{"Meryl Database : Child":{"class":"File","location":"https://zenodo.org/records/10056319/files/Meryl%20Database%20-%20Child.meryldb?download=1","filetype":"meryldb"},"Hapmer Database : Paternal":{"class":"File","location":"https://zenodo.org/records/10056319/files/Hapmer%20Database%20-%20Paternal.meryldb?download=1","filetype":"meryldb"},"Hapmer Database : Maternal":{"class":"File","location":"https://zenodo.org/records/10056319/files/Hapmer%20Database%20-%20Maternal.meryldb?download=1","filetype":"meryldb"},"Genomescope Summary":{"class":"File","location":"https://zenodo.org/records/10056319/files/Genomescope%20Summary.txt?download=1","filetype":"txt"},"Genomescope Model Parameters":{"class":"File","path":"test-data/GenomeScope_Model_parameters.tabular","filetype":"tabular"},"Pacbio Reads Collection : child":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"yeast_reads_sub1.fastq.gz","location":"https://zenodo.org/records/10056319/files/Pacbio%20Reads%20Collection.fastq.gz?download=1"}]},"Paternal Illumina reads (hap1)":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"hap1_2.fq","location":"https://zenodo.org/records/10056319/files/Sub_hap1_2.fastqsanger?download=1"},{"class":"File","identifier":"hap1_1.fq","location":"https://zenodo.org/records/10056319/files/Sub_hap1_1.fastqsanger?download=1"}]},"Maternal Illumina reads (hap2)":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"hap2_2.fq","location":"https://zenodo.org/records/10056319/files/Sub_hap2_2.fastqsanger?download=1"},{"class":"File","identifier":"hap2_1.fq","location":"https://zenodo.org/records/10056319/files/Sub_hap2_1.fastqsanger?download=1"}]},"SAK input file (Optional)":null,"Homozygous Read Coverage":null,"Bits for bloom filter":32,"Name for Haplotype 1":"Hap1","Name for Haplotype 2":"Hap2","Database for Busco Lineage":"v5","Lineage":"vertebrata_odb10"},"outputs":{"Estimated Genome size":2288021,"Assembly statistics for Hap1 and Hap2":{"asserts":{"has_line":{"line":"# contigs\\t81\\t27"}}},"usable hap1 gfa":{"asserts":{"has_n_lines":{"n":167}}},"Hifiasm Trio hap1":{"asserts":{"has_n_lines":{"n":162}}},"Busco Summary Hap1":{"asserts":{"has_text":{"text":"C:1.2%[S:1.1%,D:0.0%],F:0.4%,M:98.4%"}}},"Nx Plot":{"asserts":{"has_size":{"value":65000,"delta":5000}}},"No Sequence hap1 gfa":{"asserts":{"not_has_text":{"text":"GTCAAGGCGA"}}}}}]}],"path":"./workflows/VGP-assembly-v2/Assembly-Hifi-Trio-phasing-VGP5"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Purge-duplicate-contigs-VGP6.ga","testParameterFiles":["/Purge-duplicate-contigs-VGP6-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"Purge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication). This workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5)","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.8","name":"Purge duplicate contigs from a diploid assembly VGP6","report":{"markdown":"\\n# Workflow Execution Report\\n\\nTime workflow was invoked\\n\\n```galaxy\\ninvocation_time()\\n```\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Merqury results\\n\\nMerqury QV:\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_QV\\")\\n```\\n\\nMerqury completeness:\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_stats\\")\\n```\\n\\nMerqury plots:\\n\\nspectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-cn.fl\\")\\n```\\n\\n\\nspectra-asm:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-asm.fl\\")\\n```\\n\\n\\nhap1/Primary spectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_01.spectra-cn.fl\\")\\n```\\n\\n\\nhap2/alternate spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_02.spectra-cn.fl\\")\\n```\\n\\n\\n\\n\\n## BUSCO results (Vertebrata database)\\n\\nPurged hap1/primary Assembly\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco on Purged Primary assembly: summary image\\")\\n```\\n\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A simple list containing PacBio data in either fasta or fastq formats with trimmed adapters. These are usually generated by contiging workflows and have tag \\"trimmed_hifi\\" associated with them.\\n","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A simple list containing PacBio data in either fasta or fastq formats with trimmed adapters. These are usually generated by contiging workflows and have tag \\"trimmed_hifi\\" associated with them.\\n","name":"Pacbio Reads Collection - Trimmed"}],"label":"Pacbio Reads Collection - Trimmed","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"0df002ab-91cf-4767-a348-9ce31f439a01","when":null,"workflow_outputs":[]},"1":{"annotation":"Primary assembly generated by contiging workflow in fasta format.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Primary assembly generated by contiging workflow in fasta format.","name":"Hifiasm Primary assembly"}],"label":"Hifiasm Primary assembly","name":"Input dataset","outputs":[],"position":{"left":21.7890625,"top":101.30078125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"158c873a-b3bc-4135-95d4-d435a4849786","when":null,"workflow_outputs":[]},"2":{"annotation":"Alternate assembly generated by contiging workflow in fasta format.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Alternate assembly generated by contiging workflow in fasta format.","name":"Hifiasm Alternate assembly"}],"label":"Hifiasm Alternate assembly","name":"Input dataset","outputs":[],"position":{"left":52.028900146484375,"top":200.42266845703125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"681334cb-47f1-49cb-9f86-612e17655419","when":null,"workflow_outputs":[]},"3":{"annotation":"Meryl database generated by K-mer profiling workflow.\\n","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Meryl database generated by K-mer profiling workflow.\\n","name":"Meryl Database"}],"label":"Meryl Database","name":"Input dataset","outputs":[],"position":{"left":101.96484375,"top":297.9609375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0c3b5f68-3534-4d76-b291-e844707606ec","when":null,"workflow_outputs":[]},"4":{"annotation":"GenomeScope model parameters generated by K-mer profiling workflow.\\n","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"GenomeScope model parameters generated by K-mer profiling workflow.\\n","name":"Genomescope model parameters"}],"label":"Genomescope model parameters","name":"Input dataset","outputs":[],"position":{"left":146.16796875,"top":390.59765625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"5c8af4f2-5764-49a6-89fa-dd8730b11828","when":null,"workflow_outputs":[]},"5":{"annotation":"Estimated genome size generated by contiging workflows. ","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Estimated genome size generated by contiging workflows. ","name":"Estimated genome size - Parameter File"}],"label":"Estimated genome size - Parameter File","name":"Input dataset","outputs":[],"position":{"left":188.4163818359375,"top":507.23516845703125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"9cc9d8e4-a917-4258-a9fc-74de593c42de","when":null,"workflow_outputs":[]},"6":{"annotation":"Database to use for Busco Lineages.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Database to use for Busco Lineages.","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":265.7339336519477,"top":639.7798327397703},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b9c24106-80b7-46ba-9f50-1dcfc7d434d7","when":null,"workflow_outputs":[]},"7":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis.\\n","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis.\\n","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":331.03774471723193,"top":766.9125081362662},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a489e27a-9a41-41e2-ab57-db2e855b1018","when":null,"workflow_outputs":[]},"8":{"annotation":"Do not use. ","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Do not use. ","name":"SAK input file"}],"label":"SAK input file","name":"Input dataset","outputs":[],"position":{"left":2647.0749499476224,"top":1846.1729865822142},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"f53ab5b0-6b4d-43ea-8876-a1e2476fc7c6","when":null,"workflow_outputs":[]},"9":{"annotation":"Name of the first assembly.","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Name of the first assembly.","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4388.98868377012,"top":201.3119349646822},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"multiple\\": false, \\"validators\\": [{\\"message\\": \\"Please do not use blank spaces\\", \\"expression\\": \\"^\\\\\\\\S+$\\", \\"negate\\": false, \\"type\\": \\"regex\\"}], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2952f3e6-cc98-4cbe-ac46-32adb482ff3d","when":null,"workflow_outputs":[]},"10":{"annotation":"Name of the second assembly.","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Name of the second assembly.","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":4420.62890625,"top":321.19140625},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"multiple\\": false, \\"validators\\": [{\\"message\\": \\"Please do not use blank spaces\\", \\"expression\\": \\"^\\\\\\\\S+$\\", \\"negate\\": false, \\"type\\": \\"regex\\"}], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"fd024c72-fe6c-4cd9-ac28-c8e86eaafc4f","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":11,"input_connections":{"fastq_input|fastq_input1":{"id":0,"output_name":"output"},"reference_source|ref_file":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":1118.17578125,"top":45.06640625},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"},"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"asm5\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"37c8f9b6-ec2e-43f7-9055-5f4f92436903","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":12,"input_connections":{"function_select|input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"split_fasta","type":"fasta"}],"position":{"left":1414.81640625,"top":1257.89453125},"post_job_actions":{"HideDatasetActionsplit_fasta":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"split_fasta"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"split_fa\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"bc4c132e-b355-4996-aa7e-11423e4253c5","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":13,"input_connections":{"input":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":905.01953125,"top":946.48828125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"1.5*c3\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"3*c7\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"104983c1-2416-47ce-a805-9932cc0300a3","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":14,"input_connections":{"input1":{"id":5,"output_name":"output"}},"inputs":[],"label":"Estimated genome size","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":4162.67578125,"top":1607.08203125},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated Genome size"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"66e97598-4765-4430-afd3-df587f986c81","when":null,"workflow_outputs":[{"label":"Estimated Genome Size","output_name":"integer_param","uuid":"a6d1a01f-ee44-4faa-a8a1-0f20fde412e8"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":15,"input_connections":{"components_1|param_type|component_value":{"id":9,"output_name":"output"}},"inputs":[],"label":"Awk command for primary assembly","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":4651.365068171187,"top":984.2645791215833},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\t\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"\\\\\\"}; {print}; \\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"6cc103fa-7ff9-4594-a380-2a740d05fd85","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":16,"input_connections":{"components_1|param_type|component_value":{"id":10,"output_name":"output"}},"inputs":[],"label":"Awk command for alternate assembly 2","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":4644.5776730115,"top":504.2187556460902},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\t\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"\\\\\\"}; {print}; \\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"4b54a0ae-8c81-4f58-9902-470660558866","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":17,"input_connections":{"fastq_input|fastq_input1":{"id":12,"output_name":"split_fasta"},"reference_source|ref_file":{"id":12,"output_name":"split_fasta"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":1709.6757657717624,"top":1173.1835801091086},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": \\"1\\", \\"B\\": \\"19\\", \\"O\\": \\"39\\", \\"O2\\": \\"81\\", \\"E\\": \\"3\\", \\"E2\\": \\"1\\", \\"z\\": \\"200\\", \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"self-homology\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": \\"100\\", \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": \\"40\\", \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"9bceadad-bf25-46f7-8d3f-d9ab18dde578","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"Cut1","errors":null,"id":18,"input_connections":{"input":{"id":13,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1385.28515625,"top":793.75390625},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6fc4a474-db85-4503-8879-4698250564ed","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"Cut1","errors":null,"id":19,"input_connections":{"input":{"id":13,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1380.22265625,"top":1043.83203125},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"294539f6-ccc6-4bc3-baec-15a668a80899","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":20,"input_connections":{"input1":{"id":18,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1729.67578125,"top":560.94140625},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"TagDatasetActioncalcuts_cutoff":{"action_arguments":{"tags":"p1_calcuts_cutoffs"},"action_type":"TagDatasetAction","output_name":"calcuts_cutoff"},"TagDatasetActionhist":{"action_arguments":{"tags":"p1_calcuts_hist"},"action_type":"TagDatasetAction","output_name":"hist"},"TagDatasetActionpbcstat_cov":{"action_arguments":{"tags":"p1_calcuts_basecov"},"action_type":"TagDatasetAction","output_name":"pbcstat_cov"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"f88e694e-edcf-47b2-905d-b9d5be3886eb","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":21,"input_connections":{"input1":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1664.98828125,"top":879.00390625},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"56cf2739-93df-4d25-baf5-1d1d8ce4aa0f","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":22,"input_connections":{"function_select|input":{"id":11,"output_name":"alignment_output"},"function_select|section_calcuts|transition":{"id":21,"output_name":"integer_param"},"function_select|section_calcuts|upper_depth":{"id":20,"output_name":"integer_param"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"stat_file","type":"tabular"},{"name":"pbcstat_cov","type":"tabular"},{"name":"pbcstat_wig","type":"wig"},{"name":"hist","type":"png"},{"name":"calcuts_log","type":"txt"},{"name":"calcuts_cutoff","type":"tabular"}],"position":{"left":2284.2421875,"top":327.12109375},"post_job_actions":{"TagDatasetActioncalcuts_cutoff":{"action_arguments":{"tags":"p1_calcuts_cutoffs, #p1"},"action_type":"TagDatasetAction","output_name":"calcuts_cutoff"},"TagDatasetActioncalcuts_log":{"action_arguments":{"tags":"p1_calcuts_log, #p1"},"action_type":"TagDatasetAction","output_name":"calcuts_log"},"TagDatasetActionhist":{"action_arguments":{"tags":"p1_calcuts_hist, #p1"},"action_type":"TagDatasetAction","output_name":"hist"},"TagDatasetActionpbcstat_cov":{"action_arguments":{"tags":"p1_calcuts_basecov, #p1"},"action_type":"TagDatasetAction","output_name":"pbcstat_cov"},"TagDatasetActionpbcstat_wig":{"action_arguments":{"tags":"p1_calcuts_basecov_wig, #p1"},"action_type":"TagDatasetAction","output_name":"pbcstat_wig"},"TagDatasetActionstat_file":{"action_arguments":{"tags":"p1_calcuts_depths, #p1"},"action_type":"TagDatasetAction","output_name":"stat_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"pbcstat\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pbcstat_options\\": {\\"max_cov\\": \\"500\\", \\"min_map_ratio\\": \\"0.0\\", \\"min_map_qual\\": null, \\"flank\\": \\"0\\", \\"primary_alignments\\": true}, \\"section_calcuts\\": {\\"min_depth\\": \\"0.1\\", \\"low_depth\\": \\"1\\", \\"transition\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"upper_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ploidy\\": \\"-d 0\\"}, \\"section_hist\\": {\\"ymin\\": null, \\"ymax\\": null, \\"xmin\\": null, \\"xmax\\": null, \\"title\\": \\"Read depth histogram plot\\"}, \\"output_options\\": [\\"pbcstat_coverage\\", \\"pbcstat_wig\\", \\"depth_stats\\", \\"histogram\\", \\"calcuts_cutoff\\", \\"calcuts_log\\"]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"97ce37f6-db2d-483c-abab-813827929e9b","when":null,"workflow_outputs":[{"label":"Cutoffs for primary assembly","output_name":"calcuts_cutoff","uuid":"d2c6d9bd-e28f-4ff1-ac7b-f9068e49a582"},{"label":"Read Coverage and cutoffs calculation on primary assembly : Histogram plot","output_name":"hist","uuid":"2e045d9d-99e4-457f-9cee-691616ed6590"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":23,"input_connections":{"function_select|coverage":{"id":22,"output_name":"pbcstat_cov"},"function_select|cutoffs":{"id":22,"output_name":"calcuts_cutoff"},"function_select|input":{"id":17,"output_name":"alignment_output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"purge_dups_log","type":"txt"},{"name":"purge_dups_bed","type":"bed"}],"position":{"left":2455.08203125,"top":1344.78515625},"post_job_actions":{"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"p1_purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"},"TagDatasetActionpurge_dups_log":{"action_arguments":{"tags":"p1_purgeoverlap_log"},"action_type":"TagDatasetAction","output_name":"purge_dups_log"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"purge_dups\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"coverage\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"cutoffs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_bad\\": \\"0.8\\", \\"min_align\\": \\"70\\", \\"min_match\\": \\"200\\", \\"min_chain\\": \\"500\\", \\"max_gap\\": \\"20000\\", \\"double_chain\\": {\\"chaining_rounds\\": \\"one\\", \\"__current_case__\\": 1}, \\"min_chain_score\\": \\"10000\\", \\"max_extend\\": \\"15000\\", \\"log_file\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"e724b339-6e6b-498c-a62f-33c1bf8d0a9b","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":24,"input_connections":{"function_select|bed_input":{"id":23,"output_name":"purge_dups_bed"},"function_select|fasta_input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"get_seqs_hap","type":"fasta"},{"name":"get_seqs_purged","type":"fasta"}],"position":{"left":2737.47265625,"top":1214.14453125},"post_job_actions":{"RenameDatasetActionget_seqs_purged":{"action_arguments":{"newname":"Purged Primary Assembly"},"action_type":"RenameDatasetAction","output_name":"get_seqs_purged"},"TagDatasetActionget_seqs_hap":{"action_arguments":{"tags":"seq_purged_p1"},"action_type":"TagDatasetAction","output_name":"get_seqs_hap"},"TagDatasetActionget_seqs_purged":{"action_arguments":{"tags":"p1"},"action_type":"TagDatasetAction","output_name":"get_seqs_purged"},"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"p1_purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"get_seqs\\", \\"__current_case__\\": 5, \\"fasta_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bed_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_options\\": {\\"coverage\\": false, \\"haplotigs\\": false, \\"length\\": \\"10000\\", \\"min_ratio\\": \\"0.05\\", \\"end_trim\\": true, \\"split\\": false, \\"min_gap\\": \\"10000\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"a25a74ec-9030-4852-88c4-60dc462cd364","when":null,"workflow_outputs":[{"label":"Purged Primary Assembly","output_name":"get_seqs_purged","uuid":"cba2e095-7c28-488c-ba6a-8fb2b1612bc1"}]},"25":{"annotation":"","content_id":"cat1","errors":null,"id":25,"input_connections":{"input1":{"id":24,"output_name":"get_seqs_hap"},"queries_0|input2":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2955.67578125,"top":1020.36328125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"RemoveTagDatasetActionout_file1":{"action_arguments":{"tags":"#p1"},"action_type":"RemoveTagDatasetAction","output_name":"out_file1"}},"tool_id":"cat1","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b81b1694-8694-4b23-8cfe-ec2c92b36808","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":26,"input_connections":{"input":{"id":24,"output_name":"get_seqs_purged"},"lineage_conditional|cached_db":{"id":6,"output_name":"output"},"lineage|lineage_dataset":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":3285.7071094555,"top":1520.5883525119295},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"busco_p1_gff,name:p1"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_p1_missing, #p1"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_p1_summ, #p1"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_p1_full, #p1"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_p1_img, #p1"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"88073363-2f5c-4486-a23a-1cfdc4f5e6a6","when":null,"workflow_outputs":[{"label":"Busco on Purged Primary assembly: summary image","output_name":"summary_image","uuid":"d1ecf9fe-3f76-465d-bcac-cf5b4986a8e8"},{"label":"Busco on Purged Primary assembly: short summary","output_name":"busco_sum","uuid":"87515aeb-4e2e-494f-a271-44013d0944e4"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":27,"input_connections":{"input_file":{"id":24,"output_name":"get_seqs_purged"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":3638.4140625,"top":1864.09765625},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"p1_gfa,PurgedPrimaryAssembly"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"terminal_overlaps_length\\": \\"500\\"}, \\"no_sequence\\": false}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"55a89d15-73d3-4e8b-b9da-06e650bd00cf","when":null,"workflow_outputs":[{"label":"Purged Primary Assembly (gfa)","output_name":"output","uuid":"e04e7950-3bcf-46e8-918c-781f71ef64c9"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":28,"input_connections":{"input_file":{"id":24,"output_name":"get_seqs_purged"},"mode_condition|statistics_condition|expected_genomesize":{"id":14,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":4895.80078125,"top":888.28515625},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_p1, #p1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"68dd6694-1839-4ec3-b258-c30a35901ebf","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":29,"input_connections":{"input_file":{"id":24,"output_name":"get_seqs_purged"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":4895.86328125,"top":1122.89453125},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_p1,  #p1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"2a8b5983-7a56-420d-9825-5ba18ac1e560","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":30,"input_connections":{"fastq_input|fastq_input1":{"id":0,"output_name":"output"},"reference_source|ref_file":{"id":25,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":3254.25390625,"top":655.11328125},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"},"TagDatasetActionalignment_output":{"action_arguments":{"tags":"readCoverage"},"action_type":"TagDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"asm5\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"895ca00e-3eaf-4c8a-9147-be1d62459cf9","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":31,"input_connections":{"function_select|input":{"id":25,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"split_fasta","type":"fasta"}],"position":{"left":3275.16015625,"top":919.81640625},"post_job_actions":{"HideDatasetActionsplit_fasta":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"split_fasta"},"RemoveTagDatasetActionsplit_fasta":{"action_arguments":{"tags":"#p1"},"action_type":"RemoveTagDatasetAction","output_name":"split_fasta"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"split_fa\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"f14c9eb4-89f8-4697-9267-ea58035a9126","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":32,"input_connections":{"code":{"id":15,"output_name":"out1"},"infile":{"id":28,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":5238.44140625,"top":1132.58203125},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"8ad0b9e5-e002-402d-9d3a-ec20c331f864","when":null,"workflow_outputs":[]},"33":{"annotation":"","id":33,"input_connections":{"gfa_stats":{"id":29,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":5764.59765625,"top":329.87890625},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"6f1d6192-d1ef-4117-b3af-436b08bffb72","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"3000707c-422b-4e57-b965-b06a7b9d824a","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"eb19b23a-04eb-4cf6-8389-ca7c21458194","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"4beb22a7-74f7-4dff-abf9-3d1bfd263ecd","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"fdf47dde-b43d-431e-a8a4-7f476ada62ae","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"f7687678-170e-4287-973d-ee285e5a4d62","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"9e083ec1-ee5c-4aa0-83e8-2d4f01db18ca","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"9cab2e16-083c-48ed-ada3-566c9639e819","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"5156457e-0da2-4478-a298-7b4612d7e311"}]}},"tags":[],"uuid":"36fe2942-d819-4adc-8c23-a596b55d224c"},"tool_id":null,"type":"subworkflow","uuid":"509bf106-157c-4565-b2fc-bf28b3818a10","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":34,"input_connections":{"function_select|input":{"id":30,"output_name":"alignment_output"},"function_select|section_calcuts|transition":{"id":21,"output_name":"integer_param"},"function_select|section_calcuts|upper_depth":{"id":20,"output_name":"integer_param"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"stat_file","type":"tabular"},{"name":"pbcstat_cov","type":"tabular"},{"name":"pbcstat_wig","type":"wig"},{"name":"hist","type":"png"},{"name":"calcuts_log","type":"txt"},{"name":"calcuts_cutoff","type":"tabular"}],"position":{"left":3614.9135005370044,"top":361.1132969887206},"post_job_actions":{"TagDatasetActioncalcuts_cutoff":{"action_arguments":{"tags":"p2_calcuts_cutoffs"},"action_type":"TagDatasetAction","output_name":"calcuts_cutoff"},"TagDatasetActioncalcuts_log":{"action_arguments":{"tags":" p2_calcuts_log"},"action_type":"TagDatasetAction","output_name":"calcuts_log"},"TagDatasetActionhist":{"action_arguments":{"tags":"p2_calcuts_hist"},"action_type":"TagDatasetAction","output_name":"hist"},"TagDatasetActionpbcstat_cov":{"action_arguments":{"tags":"p2_calcuts_basecov"},"action_type":"TagDatasetAction","output_name":"pbcstat_cov"},"TagDatasetActionpbcstat_wig":{"action_arguments":{"tags":" p2_calcuts_basecov_wig"},"action_type":"TagDatasetAction","output_name":"pbcstat_wig"},"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"p2_purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"},"TagDatasetActionstat_file":{"action_arguments":{"tags":"p2_calcuts_depths"},"action_type":"TagDatasetAction","output_name":"stat_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"pbcstat\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pbcstat_options\\": {\\"max_cov\\": \\"500\\", \\"min_map_ratio\\": \\"0.0\\", \\"min_map_qual\\": null, \\"flank\\": \\"0\\", \\"primary_alignments\\": true}, \\"section_calcuts\\": {\\"min_depth\\": \\"0.1\\", \\"low_depth\\": \\"1\\", \\"transition\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"upper_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ploidy\\": \\"-d 0\\"}, \\"section_hist\\": {\\"ymin\\": null, \\"ymax\\": null, \\"xmin\\": null, \\"xmax\\": null, \\"title\\": \\"Read depth histogram plot\\"}, \\"output_options\\": [\\"pbcstat_coverage\\", \\"pbcstat_wig\\", \\"depth_stats\\", \\"histogram\\", \\"calcuts_cutoff\\", \\"calcuts_log\\"]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"0496de4b-ec62-48d5-8c36-e2f946ca728d","when":null,"workflow_outputs":[{"label":"Cutoffs for alternate assembly","output_name":"calcuts_cutoff","uuid":"49bf0c95-0290-4f8b-a9d7-2f17f53e9d26"},{"label":"Read Coverage and cutoffs calculation on alternate assembly : Histogram Plot","output_name":"hist","uuid":"f12c1cc7-a12a-4314-ae16-2738b80afe70"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":35,"input_connections":{"fastq_input|fastq_input1":{"id":31,"output_name":"split_fasta"},"reference_source|ref_file":{"id":31,"output_name":"split_fasta"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":3648.0662896482727,"top":1119.8737079721939},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"},"TagDatasetActionalignment_output":{"action_arguments":{"tags":"SelfvsSelf"},"action_type":"TagDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": \\"1\\", \\"B\\": \\"19\\", \\"O\\": \\"39\\", \\"O2\\": \\"81\\", \\"E\\": \\"3\\", \\"E2\\": \\"1\\", \\"z\\": \\"200\\", \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"self-homology\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": \\"100\\", \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": \\"40\\", \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"735ef0da-2e48-404e-b57d-837dc2f161fd","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":36,"input_connections":{"function_select|coverage":{"id":34,"output_name":"pbcstat_cov"},"function_select|cutoffs":{"id":34,"output_name":"calcuts_cutoff"},"function_select|input":{"id":35,"output_name":"alignment_output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"purge_dups_log","type":"txt"},{"name":"purge_dups_bed","type":"bed"}],"position":{"left":3926.34765625,"top":743.31640625},"post_job_actions":{"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"p2_purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"},"TagDatasetActionpurge_dups_log":{"action_arguments":{"tags":"p2_purgeoverlap_log"},"action_type":"TagDatasetAction","output_name":"purge_dups_log"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"purge_dups\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"coverage\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"cutoffs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_bad\\": \\"0.8\\", \\"min_align\\": \\"70\\", \\"min_match\\": \\"200\\", \\"min_chain\\": \\"500\\", \\"max_gap\\": \\"20000\\", \\"double_chain\\": {\\"chaining_rounds\\": \\"one\\", \\"__current_case__\\": 1}, \\"min_chain_score\\": \\"10000\\", \\"max_extend\\": \\"15000\\", \\"log_file\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"065dcba7-a905-4fc2-8cec-0da64f2cae80","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":37,"input_connections":{"function_select|bed_input":{"id":36,"output_name":"purge_dups_bed"},"function_select|fasta_input":{"id":25,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"get_seqs_hap","type":"fasta"},{"name":"get_seqs_purged","type":"fasta"}],"position":{"left":4191.601976906268,"top":960.190063897382},"post_job_actions":{"RenameDatasetActionget_seqs_purged":{"action_arguments":{"newname":"Purged Alternate assembly"},"action_type":"RenameDatasetAction","output_name":"get_seqs_purged"},"TagDatasetActionget_seqs_hap":{"action_arguments":{"tags":"seq_purged_p2"},"action_type":"TagDatasetAction","output_name":"get_seqs_hap"},"TagDatasetActionget_seqs_purged":{"action_arguments":{"tags":"p2"},"action_type":"TagDatasetAction","output_name":"get_seqs_purged"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"get_seqs\\", \\"__current_case__\\": 5, \\"fasta_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bed_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_options\\": {\\"coverage\\": false, \\"haplotigs\\": false, \\"length\\": \\"10000\\", \\"min_ratio\\": \\"0.05\\", \\"end_trim\\": true, \\"split\\": false, \\"min_gap\\": \\"10000\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"d38860b9-4bad-4498-b76f-7c5da1d91de0","when":null,"workflow_outputs":[{"label":"Purged Alternate assembly","output_name":"get_seqs_purged","uuid":"fc24cb81-cb83-4d9e-8e22-d2e967409f1e"}]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":38,"input_connections":{"input_file":{"id":37,"output_name":"get_seqs_purged"},"mode_condition|statistics_condition|expected_genomesize":{"id":14,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":4886.50390625,"top":450.41015625},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_p2, #p2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"6ddef707-fe4d-4020-bebb-2ab9b262df62","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":39,"input_connections":{"input_file":{"id":37,"output_name":"get_seqs_purged"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":4894.92578125,"top":679.81640625},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_p2, #p2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"d6ed7c62-c214-4856-a926-2985b003dbd9","when":null,"workflow_outputs":[]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","errors":null,"id":40,"input_connections":{"mode|assembly_options|assembly_01":{"id":24,"output_name":"get_seqs_purged"},"mode|assembly_options|assembly_02":{"id":37,"output_name":"get_seqs_purged"},"mode|meryldb_F1":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Merqury","name":"mode"}],"label":null,"name":"Merqury","outputs":[{"name":"qv_files","type":"input"},{"name":"png_files","type":"input"},{"name":"stats_files","type":"input"},{"name":"hist_files","type":"input"},{"name":"log_file","type":"txt"}],"position":{"left":4745.81640625,"top":1420.23828125},"post_job_actions":{"HideDatasetActionlog_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log_file"},"TagDatasetActionbed_files":{"action_arguments":{"tags":"merqury_p_bed, #p"},"action_type":"TagDatasetAction","output_name":"bed_files"},"TagDatasetActionhist_files":{"action_arguments":{"tags":"merqury_p_hist,name:p"},"action_type":"TagDatasetAction","output_name":"hist_files"},"TagDatasetActionpng_files":{"action_arguments":{"tags":"merqury_p_png, #p"},"action_type":"TagDatasetAction","output_name":"png_files"},"TagDatasetActionqv_files":{"action_arguments":{"tags":"merqury_p_qv, #p"},"action_type":"TagDatasetAction","output_name":"qv_files"},"TagDatasetActionsizes_files":{"action_arguments":{"tags":"merqury_p_sizes, #p"},"action_type":"TagDatasetAction","output_name":"sizes_files"},"TagDatasetActionstats_files":{"action_arguments":{"tags":"merqury_p_stats, #p"},"action_type":"TagDatasetAction","output_name":"stats_files"},"TagDatasetActionwig_files":{"action_arguments":{"tags":"merqury_p_wig, #p"},"action_type":"TagDatasetAction","output_name":"wig_files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","tool_shed_repository":{"changeset_revision":"eb7b00ec347d","name":"merqury","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"label\\": \\"output_merqury\\", \\"mode\\": {\\"options\\": \\"default\\", \\"__current_case__\\": 0, \\"meryldb_F1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_options\\": {\\"number_assemblies\\": \\"two\\", \\"__current_case__\\": 1, \\"assembly_01\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_02\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_add_headers\\": false, \\"output_selector\\": [\\"qv\\", \\"plots\\", \\"stats\\", \\"log\\", \\"hist\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy4","type":"tool","uuid":"133c60bc-4171-44b4-bf22-c70c623265b2","when":null,"workflow_outputs":[{"label":"Merqury on Phased assemblies: Images","output_name":"png_files","uuid":"ff9cfc96-449e-41d8-9c10-7f586ab6f479"},{"label":"Merqury on Phased assemblies: stats","output_name":"stats_files","uuid":"70f44477-6a3b-4ac4-b01a-f812ea880bae"}]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":41,"input_connections":{"input_file":{"id":37,"output_name":"get_seqs_purged"},"mode_condition|swiss_army_knife":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":4990.376211055945,"top":1286.3843684734534},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"p2_gfa,PurgedAlternateAssembly"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"terminal_overlaps_length\\": \\"500\\"}, \\"no_sequence\\": false}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"f15eb7b7-c881-490b-8734-cc4d934eaa3a","when":null,"workflow_outputs":[{"label":"Purged Alternate assembly (gfa)","output_name":"output","uuid":"5323cc49-9368-44f9-b1db-e027bcc77abd"}]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":42,"input_connections":{"code":{"id":16,"output_name":"out1"},"infile":{"id":38,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":5253.42578125,"top":708.59765625},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ed133b72-e945-4fe5-91d6-c71efa30678c","when":null,"workflow_outputs":[]},"43":{"annotation":"","id":43,"input_connections":{"gfa_stats":{"id":39,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":5760.26953125,"top":526.87890625},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"56311775-399e-4e9e-aa48-8fa9fff12ad3","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"7a31b528-ee29-4b2a-9270-6ae34ce198ef","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"1a29dcd8-0ecd-4e4b-b7c4-ba688290eef3","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"053d0561-98e5-46b2-b554-685802fdb0ed","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"3b744e29-af55-4a10-9e56-bb637456ef54","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"b66fba8c-0e58-44ae-97eb-38fa3b408904","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"0477114d-32f5-4560-a3c5-3995c4cd8469","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"602d39d3-ab8c-4f94-9119-f04ca5d2e750","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"38ba0514-f27a-45fd-889b-5778f32e5ec6"}]}},"tags":[],"uuid":"7eb9d6c8-f018-464e-8232-7ad3e6c76857"},"tool_id":null,"type":"subworkflow","uuid":"71a56572-9780-41fd-a92a-9f306d93ac29","when":null,"workflow_outputs":[]},"44":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":44,"input_connections":{"input":{"id":40,"output_name":"qv_files"}},"inputs":[],"label":"merqury_QV","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5380.134532792207,"top":1426.0925046670234},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"7d4f1ffc-5e09-4db5-aebd-1d1c9383e6b8","when":null,"workflow_outputs":[{"label":"merqury_QV","output_name":"output","uuid":"2f152e08-eaf2-4539-802c-cec3b72e6afe"}]},"45":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":45,"input_connections":{"input":{"id":40,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5383.847035629067,"top":1539.9431415667343},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"706cf485-92c5-43c3-8a04-23af75cf56c8","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-cn.fl","output_name":"output","uuid":"f34e855e-09ed-4da3-970c-ec7ad337c651"}]},"46":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":46,"input_connections":{"input":{"id":40,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-asm.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5392.201156836791,"top":1705.0261196501608},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-asm.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"4edba3bb-2444-4d91-87cc-50acc24923e9","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-asm.fl","output_name":"output","uuid":"71ffd225-48bd-4e96-8025-ebaa2ea02bad"}]},"47":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":47,"input_connections":{"input":{"id":40,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_01.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5353.028840865742,"top":1867.990707717229},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_01.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"41fd4355-7ef4-43be-b71c-0c4a897ddc5e","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_01.spectra-cn.fl","output_name":"output","uuid":"05b7387f-ab38-403e-ab71-b1d65e1cb025"}]},"48":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":48,"input_connections":{"input":{"id":40,"output_name":"stats_files"}},"inputs":[],"label":"merqury_stats","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5343.486929912844,"top":2088.537692922351},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"first\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"5a62f7c5-2e97-44d8-8002-ab66dd0cd17c","when":null,"workflow_outputs":[{"label":"merqury_stats","output_name":"output","uuid":"bd69aaf0-461e-4c12-a8ad-48d97de3b00c"}]},"49":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":49,"input_connections":{"input":{"id":40,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_02.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":5608.286176870008,"top":1876.5434537656731},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_02.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"035085e5-6b04-4a50-bb0a-6b3477528921","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_02.spectra-cn.fl","output_name":"output","uuid":"8bf731d1-0eee-4877-b93c-6989cec7d77f"}]},"50":{"annotation":"","content_id":"join1","errors":null,"id":50,"input_connections":{"input1":{"id":32,"output_name":"outfile"},"input2":{"id":42,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":5636.118290732048,"top":1129.4974470583106},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"fill_empty\\", \\"__current_case__\\": 1, \\"fill_columns_by\\": \\"fill_unjoined_only\\", \\"do_fill_empty_columns\\": {\\"column_fill_type\\": \\"single_fill_value\\", \\"__current_case__\\": 0, \\"fill_value\\": \\".\\"}}, \\"header\\": \\"\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"ac5c2ad5-434a-4c52-802b-eb14db33b07d","when":null,"workflow_outputs":[]},"51":{"annotation":"","id":51,"input_connections":{"Alternate data":{"id":43,"input_subworkflow_step_id":1,"output_name":"gfastats data for plotting"},"Name of alternate assembly":{"id":10,"input_subworkflow_step_id":3,"output_name":"output"},"Name of primary assembly":{"id":9,"input_subworkflow_step_id":2,"output_name":"output"},"Primary data":{"id":33,"input_subworkflow_step_id":0,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"gfastats_plot","outputs":[],"position":{"left":6121.59765625,"top":374.23828125},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","name":"gfastats_plot","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Primary data"}],"label":"Primary data","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"bb14a035-f95c-4e8e-864a-6b69caad5a89","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Alternate data"}],"label":"Alternate data","name":"Input dataset","outputs":[],"position":{"left":6.960205078125,"top":143.53692626953125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"262f039f-40c5-41c2-a36d-adcb840aea19","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4.55963134765625,"top":296.3210144042969},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1e24b121-b6e3-4478-b8c4-badfb50cf30e","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"74a3fc5f-212c-43d5-adc2-103f490fa766"}]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":7.428955078125,"top":440.4403076171875},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c2411d50-9786-4ff3-b316-5d60334dc340","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"b886bb65-a905-4e2c-b6de-ae3dcc3749d7"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"exp":{"id":2,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1139.7584686279297,"top":107.42897033691406},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"0fde4e58-dadf-4c75-8231-d5c838c05710","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"exp":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1151.7044982910156,"top":392.65625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"f8602bb7-6924-4b52-932b-b862f5698ae0","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":6,"input_connections":{"inputs":{"id":4,"output_name":"out_file1"},"queries_0|inputs2":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1441.193115234375,"top":216.59091186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"a2dc3c75-6398-4a22-bbd1-254fa1112764","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1780.3265991210938,"top":182.99716186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fc421b7d-5c43-4bef-8904-b08a79f517e6","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1781.349365234375,"top":387.96875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"91e26e50-21d0-4380-bf84-a5bacef80541","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2068.1390991210938,"top":31.988632202148438},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"80c6d25c-25e8-41dd-9f73-10aeda771d1d","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"10ea479b-6055-4295-ab4d-b4a3448ba241"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2083.3805541992188,"top":319.8011169433594},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"d8440c79-e959-4f80-95ad-4b903c987979","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"59e0d92b-8e0f-4e97-b6be-8656f1721abf"}]}},"tags":[],"uuid":"2f7bc429-ee69-4548-aa90-b6e7f1db026f"},"tool_id":null,"type":"subworkflow","uuid":"78a19ed6-ca17-4b55-9822-bd7bb42a4b22","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"Size Plot","uuid":"b5516866-8b4f-41b8-a19a-12e1362d344c"},{"label":"Nx Plot","output_name":"Nx Plot","uuid":"a15bf8fc-7eae-497b-9f75-ddaf474ade51"}]},"52":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":52,"input_connections":{"input":{"id":50,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":5984.510589781477,"top":1063.1774682487087},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Assembly statistics for purged assemblies"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"gfastats_p1_p2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": [\\"3\\"]}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"1963d91c-104e-4845-8045-6ae47c991241","when":null,"workflow_outputs":[{"label":"Assembly statistics for purged assemblies","output_name":"output","uuid":"5c1dae47-1920-41b6-b86f-8627ea5e5721"}]},"53":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":53,"input_connections":{"infile":{"id":52,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":6339.231196702902,"top":1019.7029697491412},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"94fb5ab8-0ebf-4302-992f-b8ca42b01780","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"a022ab8b-306a-4035-8604-810d39f1bde7"}]}},"tags":["VGP_curated"],"uuid":"4893d49b-6e4e-41f3-abf6-6dac5a62a4f1","version":4},"readme":"# Purge Duplicate Contigs\\n\\nPurge contigs marked as duplicates by purge_dups (could be haplotypic duplication or overlap duplication)\\nThis workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5)\\n\\n## Inputs\\n\\n1. Hifi long reads - trimmed [fastq] (Generated by Cutadapt in the contigging workflow)\\n2. Primary Assembly (hap1) [fasta] (Generated by the contigging workflow)\\n3. Alternate Assembly (hap2) [fasta] (Generated by the contigging workflow)\\n4. K-mer database [meryldb]  (Generated by the k-mer profiling workflow)\\n5. Genomescope model parameters [txt] (Generated by the k-mer profiling workflow)\\n6. Estimated Genome Size [txt]\\n7. Database for busco lineage (recommended: latest) \\n8. Lineage of your species for Busco Orthologs (recommended: vertebrata)\\n9. Name of first haplotype\\n10. Name of second haplotype\\n\\n\\n## Outputs\\n\\n1. Haplotype 1 purged assembly (Fasta and gfa)\\n2. Haplotype 2 purged assembly (Fasta and gfa)\\n3. QC: BUSCO report for both assemblies\\n4. QC: Merqury report for both assemblies\\n5. QC: Assembly statistics for both assemblies\\n6. QC: Nx plot for both assemblies\\n7. QC: Size plot for both assemblies\\n","changelog":"# Changelog\\n\\n## [0.8] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.7] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.6] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n## [0.5] - 2025-01-30\\n\\n## Added\\n\\n- Validator preventing the use of blank spaces in assembly names\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n\\n## [0.4] - 2024-08-13\\n\\n### Added\\n\\n- Workflow report\\n\\n### Changed\\n\\n- Expose Busco lineage database parameter\\n- Fix bug that was causing issues when merging the assembly statistics: The presence of the \'#\' character caused the tables to join incorrectly, and the numbers did not match the metrics\\n\\n## [0.3.8] - 2024-04-23\\n\\n### Added\\n\\n- Add Busco Gff output\\n\\n## [0.3.7] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4`\\n\\n## [0.3.6] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.27+galaxy0`\\n\\n## [0.3.5] - 2024-03-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0`\\n\\n## [0.3.4] - 2024-02-15\\n\\n- Remove tag filtering\\n- Add merqury histogram outputs\\n\\n## [0.3.3] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.3.2] - 2023-11-15\\n\\n### Added\\n\\n- more descriptive lables for inputs\\n\\n## [0.3.1] - 2023-11-09\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.24+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.3] - 2023-11-08\\n\\n### Added\\n\\n- Made BUSCO lineage selectable from workflow launch interface\\n\\n### Changed\\n\\n- Change BUSCO lineage input parameter name from `Provide lineage for BUSCO (e.g., Vertebrata)` to `Lineage`\\n\\n\\n## [0.2.0] - 2023-11-07\\n\\n### Added\\n\\n- Tag filtering for inputs\\n- Changed parameters of gfastats for postprocessing of contigs generated with purge_dups. Specfically:\\n  - \\"Terminal overlap selection\\" is set to \\"Yes\\"\\n  - \\"Generated the initial set of paths\\" is set to \\"No\\"\\n\\n\\n## [0.1.1] - 2023-11-01\\n\\n### Added\\n\\n- More explicit tags for readability\\n\\n\\n## [0.1] - 2023-10-26\\n\\n### Added\\n\\n- Workflow for purging duplication in contigs\\n","diagrams":"# Workflow diagrams\\n\\n## Purge duplicate contigs from a diploid assembly VGP6\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Pacbio Reads Collection - Trimmed\\"]@{ shape: docs }\\n1[\\"ℹ️ Hifiasm Primary assembly\\"]@{ shape: doc }\\n2[\\"ℹ️ Hifiasm Alternate assembly\\"]@{ shape: doc }\\n3[\\"ℹ️ Meryl Database\\"]@{ shape: doc }\\n4[\\"ℹ️ Genomescope model parameters\\"]@{ shape: doc }\\n5[\\"ℹ️ Estimated genome size - Parameter File\\"]@{ shape: doc }\\n6[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n8[\\"ℹ️ SAK input file\\"]@{ shape: doc }\\n9[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n11[\\"Map with minimap2\\"]@{ shape: process }\\n0 --> 11\\n1 --> 11\\n12[\\"Purge overlaps\\"]@{ shape: process }\\n1 --> 12\\n13[\\"Compute\\"]@{ shape: process }\\n4 --> 13\\n14[\\"Estimated genome size\\"]@{ shape: process }\\n5 --> 14\\n15[\\"Awk command for primary assembly\\"]@{ shape: process }\\n9 --> 15\\n16[\\"Awk command for alternate assembly 2\\"]@{ shape: process }\\n10 --> 16\\n17[\\"Map with minimap2\\"]@{ shape: process }\\n12 --> 17\\n12 --> 17\\n18[\\"Cut\\"]@{ shape: process }\\n13 --> 18\\n19[\\"Cut\\"]@{ shape: process }\\n13 --> 19\\n20[\\"Parse parameter value\\"]@{ shape: process }\\n18 --> 20\\n21[\\"Parse parameter value\\"]@{ shape: process }\\n19 --> 21\\n22[\\"Purge overlaps\\"]@{ shape: process }\\n11 --> 22\\n21 --> 22\\n20 --> 22\\n23[\\"Purge overlaps\\"]@{ shape: process }\\n22 --> 23\\n22 --> 23\\n17 --> 23\\n24[\\"Purge overlaps\\"]@{ shape: process }\\n23 --> 24\\n1 --> 24\\n25[\\"Concatenate datasets\\"]@{ shape: process }\\n24 --> 25\\n2 --> 25\\n26[\\"Busco\\"]@{ shape: process }\\n24 --> 26\\n6 --> 26\\n7 --> 26\\n27[\\"gfastats\\"]@{ shape: process }\\n24 --> 27\\n8 --> 27\\n28[\\"gfastats\\"]@{ shape: process }\\n24 --> 28\\n14 --> 28\\n29[\\"gfastats\\"]@{ shape: process }\\n24 --> 29\\n30[\\"Map with minimap2\\"]@{ shape: process }\\n0 --> 30\\n25 --> 30\\n31[\\"Purge overlaps\\"]@{ shape: process }\\n25 --> 31\\n32[\\"Text reformatting\\"]@{ shape: process }\\n15 --> 32\\n28 --> 32\\n33[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n29 --> 33\\n34[\\"Purge overlaps\\"]@{ shape: process }\\n30 --> 34\\n21 --> 34\\n20 --> 34\\n35[\\"Map with minimap2\\"]@{ shape: process }\\n31 --> 35\\n31 --> 35\\n36[\\"Purge overlaps\\"]@{ shape: process }\\n34 --> 36\\n34 --> 36\\n35 --> 36\\n37[\\"Purge overlaps\\"]@{ shape: process }\\n36 --> 37\\n25 --> 37\\n38[\\"gfastats\\"]@{ shape: process }\\n37 --> 38\\n14 --> 38\\n39[\\"gfastats\\"]@{ shape: process }\\n37 --> 39\\n40[\\"Merqury\\"]@{ shape: process }\\n24 --> 40\\n37 --> 40\\n3 --> 40\\n41[\\"gfastats\\"]@{ shape: process }\\n37 --> 41\\n8 --> 41\\n42[\\"Text reformatting\\"]@{ shape: process }\\n16 --> 42\\n38 --> 42\\n43[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n39 --> 43\\n44[\\"merqury_QV\\"]@{ shape: process }\\n40 --> 44\\n45[\\"output_merqury.spectra-cn.fl\\"]@{ shape: process }\\n40 --> 45\\n46[\\"output_merqury.spectra-asm.fl\\"]@{ shape: process }\\n40 --> 46\\n47[\\"output_merqury.assembly_01.spectra-cn.fl\\"]@{ shape: process }\\n40 --> 47\\n48[\\"merqury_stats\\"]@{ shape: process }\\n40 --> 48\\n49[\\"output_merqury.assembly_02.spectra-cn.fl\\"]@{ shape: process }\\n40 --> 49\\n50[\\"Join two Datasets\\"]@{ shape: process }\\n32 --> 50\\n42 --> 50\\n51[\\"🛠️ gfastats_plot\\"]@{ shape: subprocess }\\n43 --> 51\\n10 --> 51\\n9 --> 51\\n33 --> 51\\n52[\\"Advanced Cut\\"]@{ shape: process }\\n50 --> 52\\n53[\\"Replace\\"]@{ shape: process }\\n52 --> 53\\n```\\n\\n## gfastats_plot\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Primary data\\"]@{ shape: doc }\\n1[\\"ℹ️ Alternate data\\"]@{ shape: doc }\\n2[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"Concatenate datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Cut\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Nx Plot\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Size Plot\\"]@{ shape: process }\\n8 --> 10\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Purge-duplicate-contigs-VGP6/main","dockstore_id":24163,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078812","tests":[{"doc":"Test outline for Purge duplicate contigs from a diploid assembly VGP6","job":{"Hifiasm Primary assembly":{"class":"File","location":"https://zenodo.org/records/10047837/files/Hifiasm%20Primary%20assembly.fasta?download=1","filetype":"fasta"},"Hifiasm Alternate assembly":{"class":"File","location":"https://zenodo.org/records/10047837/files/Hifiasm%20Alternate%20assembly.fasta?download=1","filetype":"fasta"},"Meryl Database":{"class":"File","location":"https://zenodo.org/records/10047837/files/Meryl%20Database.meryldb?download=1","filetype":"meryldb"},"Genomescope model parameters":{"class":"File","location":"https://zenodo.org/records/10047837/files/Genomescope%20model%20parameters.tabular?download=1","filetype":"tabular"},"Estimated genome size - Parameter File":{"class":"File","location":"https://zenodo.org/records/10047837/files/Estimated%20genome%20size%20-%20Parameter%20File.expression.json?download=1","filetype":"expression.json"},"Pacbio Reads Collection - Trimmed":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"yeast_reads_sub1.fastq.gz","location":"https://zenodo.org/records/10047837/files/Pacbio%20Reads%20Collection%20-%20Trimmed_yeast_reads_sub1.fastq.gz.fastq.gz?download=1"}]},"Database for Busco Lineage":"v5","Lineage":"vertebrata_odb10","Name of primary assembly":"Primary","Name of alternate assembly":"Alternate"},"outputs":{"Cutoffs for primary assembly":{"asserts":{"has_text":{"text":"1\\t15\\t15\\t16\\t16\\t48"}}},"Purged Primary Assembly":{"asserts":{"has_n_lines":{"n":156}}},"Purged Primary Assembly (gfa)":{"asserts":{"has_n_lines":{"n":157}}},"Busco on Purged Primary assembly: short summary":{"asserts":{"has_text":{"text":"C:1.1%[S:1.1%,D:0.0%],F:0.5%,M:98.4%,n:3354"}}},"Cutoffs for alternate assembly":{"asserts":{"has_text":{"text":"1\\t15\\t15\\t16\\t16\\t48"}}},"Purged Alternate assembly":{"asserts":{"has_n_lines":{"n":144}}},"Purged Alternate assembly (gfa)":{"asserts":{"has_n_lines":{"n":145}}},"Assembly statistics for purged assemblies":{"asserts":{"has_text":{"text":"# contigs\\t78\\t72"}}},"Nx Plot":{"asserts":{"has_size":{"value":61000,"delta":5000}}},"Merqury on Phased assemblies: stats":{"element_tests":{"output_merqury.completeness":{"asserts":{"has_text":{"text":"both\\tall\\t1212740\\t1300032\\t93.2854"}}}}}}}]}],"path":"./workflows/VGP-assembly-v2/Purge-duplicate-contigs-VGP6"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/kmer-profiling-hifi-VGP1.ga","testParameterFiles":["/kmer-profiling-hifi-VGP1-tests.yml"],"authors":[{"name":"VGP","url":"https://vertebrategenomeproject.org"},{"name":"Galaxy"}],"definition":{"a_galaxy_workflow":"true","annotation":"Performs k-mer profiling on PacBio data and generates GenomeScope plots and summary for genome characteristics assessment.","comments":[],"creator":[{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"},{"class":"Organization","name":"Galaxy"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.1.9","name":"kmer-profiling-hifi-VGP1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n\\n```galaxy\\ninvocation_time()\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"GenomeScope linear plot\\")\\n```\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"GenomeScope log plot\\")\\n```\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"GenomeScope transformed linear plot\\")\\n```\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"GenomeScope transformed log plot\\")\\n```\\n\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A simple list collection containing PacBio data in either fastq or fasta format","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A simple list collection containing PacBio data in either fastq or fasta format","name":"Collection of Pacbio Data"}],"label":"Collection of Pacbio Data","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":237.75781276855386},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"861c3a49-1055-4030-9a91-e53cbf1ac436","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"K-mer length "}],"label":"K-mer length ","name":"Input parameter","outputs":[],"position":{"left":68.13671875,"top":408.3515627685539},"tool_id":null,"tool_state":"{\\"default\\": 21, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"946bcadd-8ab0-4595-9985-abb574539844","when":null,"workflow_outputs":[]},"2":{"annotation":"Ploidy for the organism being assembled","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Ploidy for the organism being assembled","name":"Ploidy"}],"label":"Ploidy","name":"Input parameter","outputs":[],"position":{"left":177.15234375,"top":569.6054690185539},"tool_id":null,"tool_state":"{\\"default\\": 2, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"e3848560-55a2-42a5-ac1a-487ccf084d92","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":3,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":329.0234375,"top":381.1640627685539},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"21\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"9a2f7fb6-aaf7-4e81-9d9a-a198db3cd397","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7","errors":null,"id":4,"input_connections":{"input_reads":{"id":0,"output_name":"output"},"options_kmer_size|input_kmer_size":{"id":3,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":419.9140625,"top":102.30468776855386},"post_job_actions":{"HideDatasetActionread_db":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"bb6774c6aeaf","name":"meryl_count_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"count_operation\\": \\"count\\", \\"input_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options_kmer_size\\": {\\"kmer_size\\": \\"provide\\", \\"__current_case__\\": 0, \\"input_kmer_size\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"fc4e77b0-4f5a-4679-9177-87b2a3cd71ed","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","errors":null,"id":5,"input_connections":{"input_meryldb_02":{"id":4,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":702.75390625,"top":0},"post_job_actions":{"TagDatasetActionread_db":{"action_arguments":{"tags":"meryl_db,Meryl Database"},"action_type":"TagDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"c4758f3bf36e","name":"meryl_groups_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"groups_operations\\": \\"union-sum\\", \\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"2ea15dc0-a71e-48a6-8d41-d568120525b4","when":null,"workflow_outputs":[{"label":"Merged Meryl Database","output_name":"read_db","uuid":"72fcc42e-2c61-4e86-a2dc-6fe974f2a35d"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","errors":null,"id":6,"input_connections":{"input_meryldb_02":{"id":5,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db_hist","type":"tabular"}],"position":{"left":992.953125,"top":83.94920349121094},"post_job_actions":{"HideDatasetActionread_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db_hist"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"9a0227fa7521","name":"meryl_histogram_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"12353c6c-2b6c-400f-887d-41a5c3f3f277","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"read_db_hist"},"kmer_length":{"id":3,"output_name":"integer_param"},"ploidy":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"GenomeScope","outputs":[{"name":"linear_plot","type":"png"},{"name":"log_plot","type":"png"},{"name":"transformed_linear_plot","type":"png"},{"name":"transformed_log_plot","type":"png"},{"name":"model","type":"txt"},{"name":"summary","type":"txt"},{"name":"model_params","type":"tabular"}],"position":{"left":1274.636407247791,"top":125.53515651855386},"post_job_actions":{"TagDatasetActionlinear_plot":{"action_arguments":{"tags":"genomescope_linear,Linear Plot"},"action_type":"TagDatasetAction","output_name":"linear_plot"},"TagDatasetActionlog_plot":{"action_arguments":{"tags":"genomescope_log,Log Plot"},"action_type":"TagDatasetAction","output_name":"log_plot"},"TagDatasetActionmodel":{"action_arguments":{"tags":"genomescope_model"},"action_type":"TagDatasetAction","output_name":"model"},"TagDatasetActionmodel_params":{"action_arguments":{"tags":"genomescope_params,GenomeScope Parameters"},"action_type":"TagDatasetAction","output_name":"model_params"},"TagDatasetActionsummary":{"action_arguments":{"tags":"genomescope_summ,GenomeScope Summary"},"action_type":"TagDatasetAction","output_name":"summary"},"TagDatasetActiontransformed_linear_plot":{"action_arguments":{"tags":"genomescope_tr_linear,Transformed Linear Plot"},"action_type":"TagDatasetAction","output_name":"transformed_linear_plot"},"TagDatasetActiontransformed_log_plot":{"action_arguments":{"tags":"genomescope_tr_log,Transformed Log Plot"},"action_type":"TagDatasetAction","output_name":"transformed_log_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","tool_shed_repository":{"changeset_revision":"460fad600dce","name":"genomescope","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"topology\\": null, \\"initial_repetitiveness\\": null, \\"initial_heterozygosities\\": \\"\\", \\"transform_exp\\": null, \\"testing\\": true, \\"true_params\\": \\"\\", \\"trace_flag\\": false, \\"num_rounds\\": null}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmer_length\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lambda\\": null, \\"max_kmercov\\": null, \\"output_options\\": {\\"output_files\\": [\\"model_output\\", \\"summary_output\\"], \\"no_unique_sequence\\": false}, \\"ploidy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.1+galaxy0","type":"tool","uuid":"955de351-ef19-42dc-89d7-ca213994515a","when":null,"workflow_outputs":[{"label":"GenomeScope transformed linear plot","output_name":"transformed_linear_plot","uuid":"48e2c4c5-8658-44e6-8981-673a2f1f719e"},{"label":"GenomeScope transformed log plot","output_name":"transformed_log_plot","uuid":"ea87bd08-a6d9-4c9e-9497-8bfb1725da04"},{"label":"GenomeScope summary","output_name":"summary","uuid":"b74004e1-4f4b-4f43-a9f0-2a7a0a70ca1d"},{"label":"GenomeScope Model Parameters","output_name":"model_params","uuid":"cb88b945-413c-4e2d-81a9-dd838df902ff"},{"label":"GenomeScope log plot","output_name":"log_plot","uuid":"988e2189-d5f2-4e9e-8fc2-4434e68e8501"},{"label":"GenomeScope linear plot","output_name":"linear_plot","uuid":"57230be4-36d5-4307-b4b0-b8a73f6800ce"}]}},"tags":["Reviewed","VGP"],"uuid":"58adb6d9-0345-44ca-82fd-e394eda5a18b","version":3},"readme":"# VGP Workflow #1\\n\\nThis workflow produces a Meryl database and Genomescope outputs that will be used to determine parameters for following workflows, and assess the quality of genome assemblies. Specifically, it provides information about the genomic complexity, such as the genome size and levels of heterozygosity and repeat content, as well about the data quality.\\n\\n### Inputs\\n\\n-   A collection of Hifi long reads in FASTQ format\\n-   *k*-mer length\\n-   Ploidy\\n\\n### Outputs\\n\\n-   Meryl Database of kmer counts\\n-   GenomeScope\\n    -   Linear plot\\n    -   Log plot\\n    -   Transformed linear plot\\n    -   Transformed log plot\\n    -   Summary\\n    -   Model\\n    -   Model parameteres\\n\\n ![image](https://github.com/galaxyproject/iwc/assets/4291636/565238fc-f8a9-46ac-8b31-6276410fa436)\\n","changelog":"# Changelog\\n\\n## [0.1.9] - 2024-12-17\\n### Added\\n- Annotation for workflow describing its function\\n\\n## [0.1.8] - 2024-12-03\\n\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7`\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7`\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7`\\n- Fix invalid connections using optional inputs to set default values for non-optional tool inputs. Now using the \\"pick parameter value\\" tool to set default value \\n\\n\\n## [0.1.7] - 2024-08-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0`\\n\\n## [0.1.6] - 2024-07-12\\n\\n- Add workflow report\\n\\n## [0.1.5] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.4] - 2023-11-15\\n\\n### Added\\n\\n- more descriptive labels for inputs\\n\\n## [0.1.3] - 2023-11-09\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy2`\\n\\n## [0.1.2] - 2023-11-07\\n### Changed\\n- Added tag filtering for inputs\\n\\n## [0.1.1] - 2023-10-30\\n### Changed\\n- Names of the workflow and viles for consistency with other VGP workflows\\n- Labels and tags for user readability\\n  \\n## [0.1] - 2021-08-26\\n### Added\\n- First version of the workflow. \\n","diagrams":"# Workflow diagrams\\n\\n## kmer-profiling-hifi-VGP1\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Collection of Pacbio Data\\"]@{ shape: docs }\\n1[\\"ℹ️ K-mer length \\"]@{ shape: lean-l }\\n2[\\"ℹ️ Ploidy\\"]@{ shape: lean-l }\\n3[\\"Pick parameter value\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Meryl\\"]@{ shape: process }\\n0 --> 4\\n3 --> 4\\n5[\\"Meryl\\"]@{ shape: process }\\n4 --> 5\\n6[\\"Meryl\\"]@{ shape: process }\\n5 --> 6\\n7[\\"GenomeScope\\"]@{ shape: process }\\n6 --> 7\\n3 --> 7\\n2 --> 7\\n```\\n","updated":"2024-12-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/kmer-profiling-hifi-VGP1/main","dockstore_id":24173,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078908","tests":[{"doc":"Test outline for kmer-profiling-hifi-VGP1.ga","job":{"Collection of Pacbio Data":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"pacbio_reads","location":"https://zenodo.org/record/6603774/files/child.fastq?download=1","filetype":"fastqsanger"}]},"K-mer length ":8,"Ploidy":1},"outputs":{"GenomeScope linear plot":{"file":"test-data/GenomeScope_Linear_plot.png","compare":"sim_size","delta":10000},"GenomeScope log plot":{"file":"test-data/GenomeScope_Log_plot.png","compare":"sim_size","delta":10000},"GenomeScope transformed linear plot":{"file":"test-data/GenomeScope_Transformed_linear_plot.png","compare":"sim_size","delta":10000},"GenomeScope transformed log plot":{"file":"test-data/GenomeScope_Transformed_log_plot.png","compare":"sim_size","delta":10000},"GenomeScope summary":{"asserts":[{"has_text_matching":{"expression":"27,84. bp"}},{"has_text_matching":{"expression":"35,91. bp"}}]},"GenomeScope Model Parameters":{"asserts":[{"has_text_matching":{"expression":"0.09184.*"}},{"has_text_matching":{"expression":"27.44.*"}}]},"Merged Meryl Database":{"asserts":{"has_size":{"value":105703,"delta":10000}}}}}]}],"path":"./workflows/VGP-assembly-v2/kmer-profiling-hifi-VGP1"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Scaffolding-BioNano-VGP7.ga","testParameterFiles":["/Scaffolding-BioNano-VGP7-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.1.4","name":"Scaffolding-BioNano-VGP7","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Invocation time\\n```galaxy\\ninvocation_time()\\n```\\n\\n## Assembly statistics\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n## Size plot\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n## Nx plot\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Bionano Data"}],"label":"Bionano Data","name":"Input dataset","outputs":[],"position":{"left":407.7208647554372,"top":102.69171294873553},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"cmap\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"d620279b-9ba3-4758-9701-6d04b3f24137","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Estimated genome size - Parameter File"}],"label":"Estimated genome size - Parameter File","name":"Input dataset","outputs":[],"position":{"left":408.8927397554372,"top":260.0354629487355},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"08b9c6d8-f93a-4f57-8f1c-f26842a6e57c","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Input GFA"}],"label":"Input GFA","name":"Input dataset","outputs":[],"position":{"left":0,"top":504.88089415583244},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gfa1\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"4f728394-0652-4f67-919e-cb7ec3950534","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Conflict resolution files"}],"label":"Conflict resolution files","name":"Input dataset","outputs":[],"position":{"left":434.0958647554372,"top":591.0667129487355},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"404dfe4f-8590-486e-b632-a0564010e213","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1563.423989755437,"top":492.7385879487355},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"2bd22c8a-3266-43ee-93a3-14aaa0ece0b0","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","errors":null,"id":5,"input_connections":{"input_file":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":406.98840818165013,"top":397.6732446781746},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","tool_shed_repository":{"changeset_revision":"3ef480892a9f","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.6+galaxy0","type":"tool","uuid":"1bd73efc-df51-456f-a7b4-6738209d0492","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.7.0+galaxy3","errors":null,"id":6,"input_connections":{"bionano_cmap":{"id":0,"output_name":"output"},"conflict_resolution":{"id":3,"output_name":"output"},"ngs_fasta":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Bionano Hybrid Scaffold","outputs":[{"name":"ngs_contigs_scaffold_trimmed","type":"fasta"},{"name":"ngs_contigs_not_scaffolded_trimmed","type":"fasta"},{"name":"report","type":"txt"},{"name":"conflicts","type":"txt"},{"name":"results","type":"zip"},{"name":"ngs_contigs_scaffold_agp","type":"agp"}],"position":{"left":1172.5426030242018,"top":146.96312689571127},"post_job_actions":{"RenameDatasetActionngs_contigs_scaffold_trimmed":{"action_arguments":{"newname":"Bionano Hybrid Scaffold s1"},"action_type":"RenameDatasetAction","output_name":"ngs_contigs_scaffold_trimmed"},"TagDatasetActionconflicts":{"action_arguments":{"tags":"s1_conflicts"},"action_type":"TagDatasetAction","output_name":"conflicts"},"TagDatasetActionngs_contigs_not_scaffolded_trimmed":{"action_arguments":{"tags":"non_hybridscaffolds_s1"},"action_type":"TagDatasetAction","output_name":"ngs_contigs_not_scaffolded_trimmed"},"TagDatasetActionngs_contigs_scaffold_agp":{"action_arguments":{"tags":"s1_agp"},"action_type":"TagDatasetAction","output_name":"ngs_contigs_scaffold_agp"},"TagDatasetActionngs_contigs_scaffold_trimmed":{"action_arguments":{"tags":"hybridscaffolds_s1"},"action_type":"TagDatasetAction","output_name":"ngs_contigs_scaffold_trimmed"},"TagDatasetActionreport":{"action_arguments":{"tags":"s1_report"},"action_type":"TagDatasetAction","output_name":"report"},"TagDatasetActionresults":{"action_arguments":{"tags":"s1_zip"},"action_type":"TagDatasetAction","output_name":"results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.7.0+galaxy3","tool_shed_repository":{"changeset_revision":"3371c5bdc17a","name":"bionano_scaffold","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"all_files\\": false, \\"bionano_cmap\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"configuration_options\\": {\\"configuration\\": \\"vgp\\", \\"__current_case__\\": 0, \\"enzyme\\": \\"CTTAAG\\"}, \\"conflict_filter_genome\\": \\"3\\", \\"conflict_filter_sequence\\": \\"3\\", \\"conflict_resolution\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ngs_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trim_cut_sites\\": true, \\"zip_file\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.7.0+galaxy3","type":"tool","uuid":"acbf9c02-c273-42a9-a809-d515f724b180","when":null,"workflow_outputs":[{"label":"Scaffolds: agp","output_name":"ngs_contigs_scaffold_agp","uuid":"fc8458cc-c385-406e-94d7-7420901d45be"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","errors":null,"id":7,"input_connections":{"input_file":{"id":2,"output_name":"output"},"mode_condition|agp_to_path":{"id":6,"output_name":"ngs_contigs_scaffold_agp"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":1552.8601758103882,"top":157.7106893820858},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Bionano scaffolds reconciliated (GFA)"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"s1_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","tool_shed_repository":{"changeset_revision":"3ef480892a9f","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"scaffolding\\", \\"__current_case__\\": 2, \\"agp_to_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.6+galaxy0","type":"tool","uuid":"d29608e2-9748-4e5a-b0b5-8690dac6fb2e","when":null,"workflow_outputs":[{"label":"Bionano scaffolds reconciliated: gfa","output_name":"output","uuid":"5748c26a-1859-4f2b-9c27-42e5822a4f64"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","errors":null,"id":8,"input_connections":{"input_file":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2087.756900390276,"top":0},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"s1_fasta"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","tool_shed_repository":{"changeset_revision":"3ef480892a9f","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": \\"60\\"}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.6+galaxy0","type":"tool","uuid":"bfd8e6b3-14cd-4dd7-89ad-1a7b48bdcdfd","when":null,"workflow_outputs":[{"label":"Bionano scaffolds reconciliated: fasta","output_name":"output","uuid":"8a50c9c5-cb5f-4562-b11d-38af4581ff0c"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","errors":null,"id":9,"input_connections":{"input_file":{"id":7,"output_name":"output"},"mode_condition|statistics_condition|expected_genomesize":{"id":4,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2092.095864755437,"top":238.23858794873553},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_s1, #s1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","tool_shed_repository":{"changeset_revision":"3ef480892a9f","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.6+galaxy0","type":"tool","uuid":"15b0e368-9cb2-4be5-9a7e-ac445454356e","when":null,"workflow_outputs":[{"label":"Assembly Statistics for s1","output_name":"stats","uuid":"7cae5064-75a5-4536-97cb-e3d5fd5db33f"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","errors":null,"id":10,"input_connections":{"input_file":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2094.002114755437,"top":524.1292129487355},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0","tool_shed_repository":{"changeset_revision":"3ef480892a9f","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"s\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.6+galaxy0","type":"tool","uuid":"4f4625b9-e967-48e1-baf7-382de5cc949a","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":11,"input_connections":{"infile":{"id":9,"output_name":"stats"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2430.427837638421,"top":236.31263207430692},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"292e7cba-fa66-4f7e-935d-504d64a304cc","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"4a0bdf73-d0f9-4c54-9c82-9a63e19e4695"}]},"12":{"annotation":"","id":12,"input_connections":{"gfa_stats":{"id":10,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":2504.048989755437,"top":598.5979629487355},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.890625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7f250de3-98cc-448b-b573-6b8a85c71352","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.609375,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.125,"top":235.15625},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.09375,"top":116.015625},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"addValue","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.046875,"top":456.171875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"addValue","tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.453125,"top":299.421875},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.09375,"top":493.359375},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.09375,"top":735.515625},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0","tool_shed_repository":{"changeset_revision":"6595517c2dd8","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"6fd3a21d-d317-4bdf-8876-1a1bbf335ce4"}]}},"tags":[],"uuid":"0d43f853-69ee-4f89-9a1d-50cfa45ec0fd"},"tool_id":null,"type":"subworkflow","uuid":"63e0616a-dc9b-4b79-9c12-7a77536f7abc","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"Cut1","errors":null,"id":13,"input_connections":{"input":{"id":12,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2887.048989755437,"top":398.5979629487355},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"1cb7e0ff-ee2a-4c92-be4b-5ae68c37384a","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"Cut1","errors":null,"id":14,"input_connections":{"input":{"id":12,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2877.048989755437,"top":652.5979629487355},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"c447f820-0170-481c-a965-49b6f9903be1","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":15,"input_connections":{"input1":{"id":13,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":3308.048989755437,"top":302.5979629487355},"post_job_actions":{"TagDatasetActionoutput1":{"action_arguments":{"tags":"Nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"3b12bf9b4b87","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Default\\", \\"__current_case__\\": 0}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"7.0\\", \\"height_output_dim\\": \\"7.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"1\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"1be952f7-cfe8-4d39-b0e8-97eef0d0ee08","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"6cdf8b50-64ee-4a4c-add5-90c2ce73d32c"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":16,"input_connections":{"input1":{"id":14,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":3314.048989755437,"top":662.5979629487355},"post_job_actions":{"TagDatasetActionoutput1":{"action_arguments":{"tags":"Size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"3b12bf9b4b87","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Default\\", \\"__current_case__\\": 0}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"7.0\\", \\"height_output_dim\\": \\"7.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"1\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"56f214f7-0d63-4c5c-8464-ff54843ddb4e","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"23a8777f-9aa0-4210-a820-49060278e7c6"}]}},"tags":["VGP_curated"],"uuid":"9fd3c0de-a497-47dc-96e3-0f3e1998173d","version":6},"readme":"# Scaffolding with Bionano\\n\\nScaffolding using Bionano optical map data\\n\\n## Inputs\\n\\n1. Bionano data [cmap]\\n2. Estimated genome size [txt]\\n3. Phased assembly generated by Hifiasm [gfa1]\\n\\n## Outputs\\n\\n1. Scaffolds\\n2. Non-scaffolded contigs\\n3. QC: Assembly statistics\\n4. QC: Nx plot\\n5. QC: Size plot","changelog":"# Changelog\\n\\n## [0.1.4] - 2024-08-13\\n\\n- Addition of workflow report\\n\\n## [0.1.3] - 2024-04-01\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n\\n## [0.1.2] - 2024-03-04\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n\\n## [0.1.1] - 2023-11-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.7.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.7.0+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.1] - 2023-11-01\\n\\nAddition of the workflow to the iwc repository.\\n","diagrams":"# Workflow diagrams\\n\\n## Scaffolding-BioNano-VGP7\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Bionano Data\\"]@{ shape: doc }\\n1[\\"ℹ️ Estimated genome size - Parameter File\\"]@{ shape: doc }\\n2[\\"ℹ️ Input GFA\\"]@{ shape: doc }\\n3[\\"ℹ️ Conflict resolution files\\"]@{ shape: doc }\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n1 --> 4\\n5[\\"gfastats\\"]@{ shape: process }\\n2 --> 5\\n6[\\"Bionano Hybrid Scaffold\\"]@{ shape: process }\\n0 --> 6\\n3 --> 6\\n5 --> 6\\n7[\\"gfastats\\"]@{ shape: process }\\n2 --> 7\\n6 --> 7\\n8[\\"gfastats\\"]@{ shape: process }\\n7 --> 8\\n9[\\"gfastats\\"]@{ shape: process }\\n7 --> 9\\n4 --> 9\\n10[\\"gfastats\\"]@{ shape: process }\\n7 --> 10\\n11[\\"Replace\\"]@{ shape: process }\\n9 --> 11\\n12[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n10 --> 12\\n13[\\"Cut\\"]@{ shape: process }\\n12 --> 13\\n14[\\"Cut\\"]@{ shape: process }\\n12 --> 14\\n15[\\"Scatterplot with ggplot2\\"]@{ shape: process }\\n13 --> 15\\n16[\\"Scatterplot with ggplot2\\"]@{ shape: process }\\n14 --> 16\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2024-08-13T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Scaffolding-Bionano-VGP7/main","dockstore_id":24239,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078893","tests":[{"doc":"Test outline for Scaffolding-BioNano-VGP7","job":{"Bionano Data":{"class":"File","location":"https://zenodo.org/records/10063794/files/Bionano%20Data.cmap?download=1","filetype":"cmap"},"Estimated genome size - Parameter File":{"class":"File","location":"https://zenodo.org/records/10063794/files/Estimated%20genome%20size%20-%20Parameter%20File.expression.json?download=1","filetype":"expression.json"},"Input GFA":{"class":"File","location":"https://zenodo.org/records/10063794/files/Input%20GFA.gfa1?download=1","filetype":"gfa1"}},"outputs":{"Bionano scaffolds reconciliated: gfa":{"asserts":{"has_n_lines":{"n":67}}},"Assembly Statistics for s1":{"asserts":{"has_line":{"line":"# scaffolds\\t24"}}},"Scaffolds: agp":{"asserts":{"has_n_lines":{"n":50}}},"Bionano scaffolds reconciliated: fasta":{"asserts":{"has_n_lines":{"n":390825}}},"Nx Plot":{"asserts":{"has_size":{"value":76000,"delta":5000}}}}}]}],"path":"./workflows/VGP-assembly-v2/Scaffolding-Bionano-VGP7"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/kmer-profiling-hifi-trio-VGP2.ga","testParameterFiles":["/kmer-profiling-hifi-trio-VGP2-tests.yml"],"authors":[{"name":"VGP","url":"https://vertebrategenomeproject.org"},{"name":"Galaxy"}],"definition":{"a_galaxy_workflow":"true","annotation":"Create Meryl Database used for the estimation of assembly parameters and quality control with Merqury. Part of the VGP pipeline.","comments":[],"creator":[{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"},{"class":"Organization","name":"Galaxy"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.1.5","name":"kmer-profiling-hifi-trio-VGP2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Pacbio Hifi reads"}],"label":"Pacbio Hifi reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":251.26255108421145},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"144a390d-3d1d-49a8-a1cd-f8a784eb287c","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection of Illumina reads for the father of the Individual sequenced with Pacbio Hifi.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection of Illumina reads for the father of the Individual sequenced with Pacbio Hifi.","name":"Paternal reads"}],"label":"Paternal reads","name":"Input dataset collection","outputs":[],"position":{"left":16.83203125,"top":399.546875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"670ba121-d332-4943-93e6-b5a2950b4718","when":null,"workflow_outputs":[]},"2":{"annotation":"Collection of Illumina reads for the mother of the Individual sequenced with Pacbio Hifi.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Collection of Illumina reads for the mother of the Individual sequenced with Pacbio Hifi.","name":"Maternal reads"}],"label":"Maternal reads","name":"Input dataset collection","outputs":[],"position":{"left":18.07421875,"top":541.590674999974},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3a16a1cf-cc12-42f4-8d5f-78f307dcf87d","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"K-mer length"}],"label":"K-mer length","name":"Input parameter","outputs":[],"position":{"left":8.30859375,"top":660.512549999974},"tool_id":null,"tool_state":"{\\"default\\": 21, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"834ddad6-9ec5-4f3c-a0c6-73093dd5b7f0","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"","name":"Ploidy"}],"label":"Ploidy","name":"Input parameter","outputs":[],"position":{"left":4.04296875,"top":831.043799999974},"tool_id":null,"tool_state":"{\\"default\\": 2, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"131b3494-ca33-457c-9f53-9c8c608d319a","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":5,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":271.5784037973972,"top":632.6981223972093},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"21\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"89acf50f-a411-4e93-adfd-f90f14306d52","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7","errors":null,"id":6,"input_connections":{"input_reads":{"id":1,"output_name":"output"},"options_kmer_size|input_kmer_size":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":939.3740103274155,"top":110.13449847651007},"post_job_actions":{"HideDatasetActionread_db":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"bb6774c6aeaf","name":"meryl_count_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"count_operation\\": \\"count\\", \\"input_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options_kmer_size\\": {\\"kmer_size\\": \\"provide\\", \\"__current_case__\\": 0, \\"input_kmer_size\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"813fe35d-87e5-4948-8bf0-24f087d872f4","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6","errors":null,"id":7,"input_connections":{"operation_type|input_reads":{"id":2,"output_name":"output"},"operation_type|options_kmer_size|input_kmer_size":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":908.7621267409064,"top":1144.9401063508444},"post_job_actions":{"HideDatasetActionread_db":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6","tool_shed_repository":{"changeset_revision":"29dabd8db6f2","name":"meryl","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation_type\\": {\\"command_type\\": \\"count-kmers\\", \\"__current_case__\\": 0, \\"count_operations\\": \\"count\\", \\"input_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options_kmer_size\\": {\\"kmer_size\\": \\"provide\\", \\"__current_case__\\": 0, \\"input_kmer_size\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy6","type":"tool","uuid":"4d9b4be1-8f1a-4457-8955-0649971b6f63","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6","errors":null,"id":8,"input_connections":{"operation_type|child_reads":{"id":0,"output_name":"output"},"operation_type|maternal_reads":{"id":2,"output_name":"output"},"operation_type|options_kmer_size|input_kmer_size":{"id":5,"output_name":"integer_param"},"operation_type|paternal_reads":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"},{"name":"read_db_hist","type":"tabular"},{"name":"pat_db","type":"meryldb"},{"name":"pat_db_hist","type":"tabular"},{"name":"mat_db","type":"meryldb"},{"name":"mat_db_hist","type":"tabular"}],"position":{"left":1448.4867361305037,"top":422.4463592843802},"post_job_actions":{"HideDatasetActionmat_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"mat_db_hist"},"HideDatasetActionpat_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"pat_db_hist"},"HideDatasetActionread_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db_hist"},"TagDatasetActionmat_db":{"action_arguments":{"tags":"meryl_db_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"mat_db"},"TagDatasetActionmat_db_hist":{"action_arguments":{"tags":"meryl_hist_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"mat_db_hist"},"TagDatasetActionpat_db":{"action_arguments":{"tags":"meryl_db_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"pat_db"},"TagDatasetActionpat_db_hist":{"action_arguments":{"tags":"meryl_hist_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"pat_db_hist"},"TagDatasetActionread_db":{"action_arguments":{"tags":"meryl_db"},"action_type":"TagDatasetAction","output_name":"read_db"},"TagDatasetActionread_db_hist":{"action_arguments":{"tags":"meryl_hist"},"action_type":"TagDatasetAction","output_name":"read_db_hist"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6","tool_shed_repository":{"changeset_revision":"29dabd8db6f2","name":"meryl","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation_type\\": {\\"command_type\\": \\"trio-mode\\", \\"__current_case__\\": 6, \\"child_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"paternal_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maternal_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options_kmer_size\\": {\\"kmer_size\\": \\"provide\\", \\"__current_case__\\": 0, \\"input_kmer_size\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy6","type":"tool","uuid":"384dfbfc-98dc-4fba-b421-77cf81573f85","when":null,"workflow_outputs":[{"label":"Meryl database : maternal","output_name":"mat_db","uuid":"66a5ba61-a290-4350-ac33-3b9a878cca67"},{"label":"Meryl database : Child","output_name":"read_db","uuid":"69021571-e6a3-4014-8449-94b5ff8af000"},{"label":"Meryl database : paternal","output_name":"pat_db","uuid":"84990c33-282b-44dc-8a5e-6db7889ca79c"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","errors":null,"id":9,"input_connections":{"input_meryldb_02":{"id":6,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":1301.3547331422744,"top":25.748345240269117},"post_job_actions":{"HideDatasetActionread_db":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"c4758f3bf36e","name":"meryl_groups_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"groups_operations\\": \\"union-sum\\", \\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"d10ac58d-439d-4555-b911-f1d0fad69603","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","errors":null,"id":10,"input_connections":{"input_meryldb_02":{"id":7,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db","type":"meryldb"}],"position":{"left":1297.8309419338123,"top":1231.0599821130284},"post_job_actions":{"HideDatasetActionread_db":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"c4758f3bf36e","name":"meryl_groups_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"groups_operations\\": \\"union-sum\\", \\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"645ecd54-ea5c-4382-a0a3-6b46edb979bc","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","errors":null,"id":11,"input_connections":{"input":{"id":8,"output_name":"read_db_hist"},"kmer_length":{"id":5,"output_name":"integer_param"},"ploidy":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"GenomeScope","outputs":[{"name":"linear_plot","type":"png"},{"name":"log_plot","type":"png"},{"name":"transformed_linear_plot","type":"png"},{"name":"transformed_log_plot","type":"png"},{"name":"model","type":"txt"},{"name":"summary","type":"txt"},{"name":"model_params","type":"tabular"}],"position":{"left":2134.73828125,"top":0},"post_job_actions":{"TagDatasetActionlinear_plot":{"action_arguments":{"tags":"genomescope_linear"},"action_type":"TagDatasetAction","output_name":"linear_plot"},"TagDatasetActionlog_plot":{"action_arguments":{"tags":"genomescope_log"},"action_type":"TagDatasetAction","output_name":"log_plot"},"TagDatasetActionmodel":{"action_arguments":{"tags":"genomescope_model"},"action_type":"TagDatasetAction","output_name":"model"},"TagDatasetActionmodel_params":{"action_arguments":{"tags":"genomescope_params"},"action_type":"TagDatasetAction","output_name":"model_params"},"TagDatasetActionsummary":{"action_arguments":{"tags":"genomescope_summ"},"action_type":"TagDatasetAction","output_name":"summary"},"TagDatasetActiontransformed_linear_plot":{"action_arguments":{"tags":"genomescope_tr_linear"},"action_type":"TagDatasetAction","output_name":"transformed_linear_plot"},"TagDatasetActiontransformed_log_plot":{"action_arguments":{"tags":"genomescope_tr_log"},"action_type":"TagDatasetAction","output_name":"transformed_log_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","tool_shed_repository":{"changeset_revision":"460fad600dce","name":"genomescope","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"topology\\": null, \\"initial_repetitiveness\\": null, \\"initial_heterozygosities\\": \\"\\", \\"transform_exp\\": null, \\"testing\\": true, \\"true_params\\": \\"\\", \\"trace_flag\\": false, \\"num_rounds\\": null}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmer_length\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lambda\\": null, \\"max_kmercov\\": null, \\"output_options\\": {\\"output_files\\": [\\"model_output\\", \\"summary_output\\"], \\"no_unique_sequence\\": false}, \\"ploidy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.1+galaxy0","type":"tool","uuid":"12e75ec0-5722-4ad7-b333-636c3c9ad9c8","when":null,"workflow_outputs":[{"label":"GenomeScope transformed linear plot (child)","output_name":"transformed_linear_plot","uuid":"acebbb13-a1bd-4866-9e6f-b15708e7c78d"},{"label":"GenomeScope transformed log plot (child)","output_name":"transformed_log_plot","uuid":"bc1789b4-8cc8-4f74-ac92-f6d20c1fbd7f"},{"label":"GenomeScope summary (child)","output_name":"summary","uuid":"12268f46-c221-4417-884b-27cdad90ec05"},{"label":"GenomeScope linear plot (child)","output_name":"linear_plot","uuid":"2fbf41bc-b939-4a05-b788-2be6008aa36a"},{"label":"GenomeScope log plot (child)","output_name":"log_plot","uuid":"4fc5e47d-bae4-44f3-957b-4dce5068610a"},{"label":"GenomeScope Model Parameters (child)","output_name":"model_params","uuid":"3bc4131f-ce9b-4358-a20b-885020d34f09"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","errors":null,"id":12,"input_connections":{"input_meryldb_02":{"id":9,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db_hist","type":"tabular"}],"position":{"left":1578.841276483822,"top":7.625007303688233},"post_job_actions":{"HideDatasetActionread_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db_hist"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"9a0227fa7521","name":"meryl_histogram_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"eb911656-5d33-4381-a2fe-ba00465e9146","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","errors":null,"id":13,"input_connections":{"input_meryldb_02":{"id":10,"output_name":"read_db"}},"inputs":[],"label":null,"name":"Meryl","outputs":[{"name":"read_db_hist","type":"tabular"}],"position":{"left":1560.0367016837042,"top":1140.0239514182024},"post_job_actions":{"HideDatasetActionread_db_hist":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"read_db_hist"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7","tool_shed_repository":{"changeset_revision":"9a0227fa7521","name":"meryl_histogram_kmers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_meryldb_02\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy7","type":"tool","uuid":"2d366d57-1781-4053-b1f5-a4abdeb7b508","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","errors":null,"id":14,"input_connections":{"input":{"id":12,"output_name":"read_db_hist"},"kmer_length":{"id":5,"output_name":"integer_param"},"ploidy":{"id":4,"output_name":"output"}},"inputs":[],"label":"Genomescope on paternal haplotype","name":"GenomeScope","outputs":[{"name":"linear_plot","type":"png"},{"name":"log_plot","type":"png"},{"name":"transformed_linear_plot","type":"png"},{"name":"transformed_log_plot","type":"png"},{"name":"model","type":"txt"},{"name":"summary","type":"txt"}],"position":{"left":1851.80078125,"top":235.52643658225833},"post_job_actions":{"HideDatasetActionmodel":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"model"},"HideDatasetActionsummary":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"summary"},"TagDatasetActionlinear_plot":{"action_arguments":{"tags":"genomescope_linear_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"linear_plot"},"TagDatasetActionlog_plot":{"action_arguments":{"tags":"genomescope_log_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"log_plot"},"TagDatasetActionmodel":{"action_arguments":{"tags":"genomescope_model_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"model"},"TagDatasetActionsummary":{"action_arguments":{"tags":"genomescope_summ_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"summary"},"TagDatasetActiontransformed_linear_plot":{"action_arguments":{"tags":"genomescope_tr_linear_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"transformed_linear_plot"},"TagDatasetActiontransformed_log_plot":{"action_arguments":{"tags":"genomescope_tr_log_pat, #paternal"},"action_type":"TagDatasetAction","output_name":"transformed_log_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","tool_shed_repository":{"changeset_revision":"460fad600dce","name":"genomescope","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"topology\\": null, \\"initial_repetitiveness\\": null, \\"initial_heterozygosities\\": \\"\\", \\"transform_exp\\": null, \\"testing\\": false, \\"true_params\\": \\"\\", \\"trace_flag\\": false, \\"num_rounds\\": null}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmer_length\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lambda\\": null, \\"max_kmercov\\": null, \\"output_options\\": {\\"output_files\\": [\\"model_output\\", \\"summary_output\\"], \\"no_unique_sequence\\": false}, \\"ploidy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.1+galaxy0","type":"tool","uuid":"c65bd0f5-d8d5-4d1d-81bf-6d05458bc160","when":null,"workflow_outputs":[{"label":"GenomeScope log plot (paternal)","output_name":"log_plot","uuid":"cc24ee83-d9ee-4759-9353-1be2e7dbb253"},{"label":"GenomeScope transformed linear plot (paternal)","output_name":"transformed_linear_plot","uuid":"2b73811b-e91e-49c6-a122-5065977f9f0e"},{"label":"GenomeScope linear plot (paternal)","output_name":"linear_plot","uuid":"bba131ee-0c54-404a-87f8-067faebea46b"},{"label":"GenomeScope transformed log plot (paternal)","output_name":"transformed_log_plot","uuid":"eddc8909-df19-48c2-8406-d6bc784ceb8b"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","errors":null,"id":15,"input_connections":{"input":{"id":13,"output_name":"read_db_hist"},"kmer_length":{"id":5,"output_name":"integer_param"},"ploidy":{"id":4,"output_name":"output"}},"inputs":[],"label":"Genomescope on maternal haplotype","name":"GenomeScope","outputs":[{"name":"linear_plot","type":"png"},{"name":"log_plot","type":"png"},{"name":"transformed_linear_plot","type":"png"},{"name":"transformed_log_plot","type":"png"},{"name":"model","type":"txt"},{"name":"summary","type":"txt"}],"position":{"left":1841.77734375,"top":1187.106299999974},"post_job_actions":{"HideDatasetActionmodel":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"model"},"HideDatasetActionsummary":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"summary"},"TagDatasetActionlinear_plot":{"action_arguments":{"tags":"genomescope_linear_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"linear_plot"},"TagDatasetActionlog_plot":{"action_arguments":{"tags":"genomescope_log_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"log_plot"},"TagDatasetActionmodel":{"action_arguments":{"tags":"genomescope_model_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"model"},"TagDatasetActionsummary":{"action_arguments":{"tags":"genomescope_summ_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"summary"},"TagDatasetActiontransformed_linear_plot":{"action_arguments":{"tags":"genomescope_tr_linear_mat"},"action_type":"TagDatasetAction","output_name":"transformed_linear_plot"},"TagDatasetActiontransformed_log_plot":{"action_arguments":{"tags":"genomescope_tr_log_mat, #maternal"},"action_type":"TagDatasetAction","output_name":"transformed_log_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0","tool_shed_repository":{"changeset_revision":"460fad600dce","name":"genomescope","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"topology\\": null, \\"initial_repetitiveness\\": null, \\"initial_heterozygosities\\": \\"\\", \\"transform_exp\\": null, \\"testing\\": false, \\"true_params\\": \\"\\", \\"trace_flag\\": false, \\"num_rounds\\": null}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmer_length\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lambda\\": null, \\"max_kmercov\\": null, \\"output_options\\": {\\"output_files\\": [\\"model_output\\", \\"summary_output\\"], \\"no_unique_sequence\\": false}, \\"ploidy\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.1+galaxy0","type":"tool","uuid":"87497ba1-5c23-4852-9e2f-0bd78dc86753","when":null,"workflow_outputs":[{"label":"GenomeScope linear plot (maternal)","output_name":"linear_plot","uuid":"ed208558-8a6f-4405-bd9c-7e6343e9f71d"},{"label":"GenomeScope log plot (maternal)","output_name":"log_plot","uuid":"e2ef48ab-bbb8-4a89-8d10-4de7998db481"},{"label":"GenomeScope transformed log plot (maternal)","output_name":"transformed_log_plot","uuid":"ebe91290-a02b-498d-adb6-75fab0739fbf"},{"label":"GenomeScope transformed linear plot (maternal)","output_name":"transformed_linear_plot","uuid":"7ecb789d-57e3-40f2-81ff-f853b444609b"}]}},"tags":["Reviewed","VGP"],"uuid":"d146054e-ba07-4d0f-b6af-fef24101d9bc","version":5},"readme":"# VGP Workflow #1\\n\\nThis workflow collects the metrics on the properties of the genome under consideration by analyzing the *k*-mer frequencies. It provides information about the genomic complexity, such as the genome size and levels of heterozygosity and repeat content, as well about the data quality. It uses reads from two parental genomes to partition long reads from the offspring into haplotype-specific *k*-mer databases.\\n\\n### Inputs\\n\\n-   Collection of Hifi long reads [fastq] (Collection)\\n-   Paternal short-read Illumina sequencing reads [fastq] (Collection)\\n-   Maternal short-read Illumina sequencing reads [fastq] (Collection)\\n-   *k*-mer length\\n-   Ploidy\\n\\n### Outputs\\n\\n-   Meryl databases of k-mer counts\\n    - Child\\n    - Paternal haplotype\\n    - Maternal haplotype\\n-   GenomeScope metrics for child and the two parental genomes (three GenomeScope profiles in total)\\n    -   Linear plot\\n    -   Log plot\\n    -   Transformed linear plot\\n    -   Transformed log plot\\n    -   Summary\\n    -   Model\\n    -   Model parameteres\\n \\n    ![image](https://github.com/galaxyproject/iwc/assets/4291636/35282f8e-d021-44f6-8e03-7b58b32d6d00)\\n","changelog":"# Changelog\\n\\n## [0.1.5] - 2024-12-03\\n\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_count_kmers/meryl_count_kmers/1.3+galaxy7`\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_groups_kmers/meryl_groups_kmers/1.3+galaxy7`\\n- `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6` was updated to `https://usegalaxy.org/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/meryl_histogram_kmers/meryl_histogram_kmers/1.3+galaxy7`\\n- Fix invalid connections using optional inputs to set default values for non-optional tool inputs. Now using the \\"pick parameter value\\" tool to set default value\\n- \\n## [0.1.4] - 2024-08-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0.1+galaxy0`\\n\\n## [0.1.3] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.2] - 2023-11-09\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy2`\\n\\n## [0.1] - 2023-10-30\\n\\n### Added\\n- Clearer labels\\n\\n### Removed\\n- Interlacing step that was unneccessary\\n- Requirement for paired list inputs for the Illumina parental reads. A dataset collection list is sufficient now.\\n- Unnecessary workflow intputs\\n\\n## [0.1] - 2021-08-26\\n### Added\\n- First version of the workflow. \\n","diagrams":"# Workflow diagrams\\n\\n## kmer-profiling-hifi-trio-VGP2\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Pacbio Hifi reads\\"]@{ shape: docs }\\n1[\\"ℹ️ Paternal reads\\"]@{ shape: docs }\\n2[\\"ℹ️ Maternal reads\\"]@{ shape: docs }\\n3[\\"ℹ️ K-mer length\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Ploidy\\"]@{ shape: lean-l }\\n5[\\"Pick parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Meryl\\"]@{ shape: process }\\n1 --> 6\\n5 --> 6\\n7[\\"Meryl\\"]@{ shape: process }\\n2 --> 7\\n5 --> 7\\n8[\\"Meryl\\"]@{ shape: process }\\n0 --> 8\\n2 --> 8\\n5 --> 8\\n1 --> 8\\n9[\\"Meryl\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Meryl\\"]@{ shape: process }\\n7 --> 10\\n11[\\"GenomeScope\\"]@{ shape: process }\\n8 --> 11\\n5 --> 11\\n4 --> 11\\n12[\\"Meryl\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Meryl\\"]@{ shape: process }\\n10 --> 13\\n14[\\"Genomescope on paternal haplotype\\"]@{ shape: process }\\n12 --> 14\\n5 --> 14\\n4 --> 14\\n15[\\"Genomescope on maternal haplotype\\"]@{ shape: process }\\n13 --> 15\\n5 --> 15\\n4 --> 15\\n```\\n","updated":"2024-12-03T00:00:00","trsID":"#workflow/github.com/iwc-workflows/kmer-profiling-hifi-trio-VGP2/main","dockstore_id":24175,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078485","tests":[{"doc":"Test outline for kmer-profiling-hifi-trio-VGP2.ga","job":{"Maternal reads":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/6603774/files/paternal.fastq?download=1","filetype":"fastqsanger"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/6603774/files/paternal.fastq?download=1","filetype":"fastqsanger"}]},"Paternal reads":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/record/6603774/files/maternal.fastq?download=1","filetype":"fastqsanger"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/record/6603774/files/maternal.fastq?download=1","filetype":"fastqsanger"}]},"Pacbio Hifi reads":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"pacbio_dataset","location":"https://zenodo.org/record/6603774/files/child.fastq?download=1","filetype":"fastqsanger"}]},"K-mer length":9,"Ploidy":1},"outputs":{"GenomeScope linear plot (maternal)":{"file":"test-data/GenomeScope_Linear_plot_maternal.png","compare":"sim_size","delta":15000},"GenomeScope linear plot (paternal)":{"file":"test-data/GenomeScope_Linear_plot_paternal.png","compare":"sim_size","delta":15000},"GenomeScope linear plot (child)":{"file":"test-data/GenomeScope_Linear_plot_child.png","compare":"sim_size","delta":15000},"GenomeScope summary (child)":{"asserts":{"has_text":{"text":"43,763 bp"}}},"Meryl database : Child":{"asserts":{"has_size":{"value":205051,"delta":1000}}},"Meryl database : paternal":{"asserts":{"has_size":{"value":40338,"delta":1000}}},"Meryl database : maternal":{"asserts":{"has_size":{"value":49534,"delta":1000}}}}}]}],"path":"./workflows/VGP-assembly-v2/kmer-profiling-hifi-trio-VGP2"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Assembly-Hifi-HiC-phasing-VGP4.ga","testParameterFiles":["/Assembly-Hifi-HiC-phasing-VGP4-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"},{"name":"Delphine Lariviere","orcid":"0000-0001-6421-3484"}],"definition":{"a_galaxy_workflow":"true","annotation":"Assemble Genome using PacBio HiFi and HiC data for phasing. Prerequisite: Run k-mer profiling workflow (VGP1). ","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"},{"class":"Person","identifier":"https://orcid.org/0000-0001-6421-3484","name":"Delphine Lariviere"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.3.2","name":"Genome Assembly from Hifi reads with HiC phasing - VGP4","report":{"markdown":"\\n# Workflow Execution Report\\n\\nTime workflow was invoked:\\n\\n```galaxy\\ninvocation_time()\\n```\\nGalaxy version :\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Raw unitig graph\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"raw unitig graph image\\")\\n```\\n\\n## Merqury results\\n\\n### Merqury QV\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_qv\\")\\n```\\n\\n### Merqury completeness\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_stats\\")\\n```\\n\\n### Merqury plots\\n\\n\\nspectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-cn.fl\\")\\n```\\n\\nspectra-asm:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-asm.fl\\")\\n```\\n\\nhap1 spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_01.spectra-cn.fl\\")\\n```\\n\\nhap2 spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_02.spectra-cn.fl\\")\\n```\\n\\n## BUSCO results (Vertebrata database)\\n\\nHap1\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco Summary Image Hap1\\")\\n```\\n\\nHap2\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco Summary Image Hap2\\")\\n```\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A simple list containing PacBio data in either fasta or fastq formats.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A simple list containing PacBio data in either fasta or fastq formats.","name":"Pacbio Reads"}],"label":"Pacbio Reads","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":57.703125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"f7ef62fa-eb54-4489-b444-b6be262a45c0","when":null,"workflow_outputs":[]},"1":{"annotation":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\". ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Trim 5 bases at the beginning of each read. Use with Arima Hi-C data if the Hi-C map looks \\"noisy\\". ","name":"Trim Hi-C reads?"}],"label":"Trim Hi-C reads?","name":"Input parameter","outputs":[],"position":{"left":10.605981367449079,"top":169.1401750148436},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"46d0bd3a-f4fd-48ab-ba51-31838a288248","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Hi-C reads - Forward"}],"label":"Hi-C reads - Forward","name":"Input dataset collection","outputs":[],"position":{"left":23.493046219174857,"top":286.1353582168184},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"1e04875f-8c71-4e12-8b0b-27907cfd353b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Hi-C reads - Reverse"}],"label":"Hi-C reads - Reverse","name":"Input dataset collection","outputs":[],"position":{"left":47.33150610678279,"top":383.67001993611086},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"59acef18-cdb1-4bbd-ad88-97c1e6c1e1da","when":null,"workflow_outputs":[]},"4":{"annotation":"GenomeScope summary generated by K-mer profiling workflow","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"GenomeScope summary generated by K-mer profiling workflow","name":"Genomescope Summary"}],"label":"Genomescope Summary","name":"Input dataset","outputs":[],"position":{"left":60.90386993408604,"top":486.6096965446283},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"1844cbb3-ba5f-4997-a3cd-ce70f4570a99","when":null,"workflow_outputs":[]},"5":{"annotation":"Meryl database generated by K-mer profiling workflow","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Meryl database generated by K-mer profiling workflow","name":"Meryl Database"}],"label":"Meryl Database","name":"Input dataset","outputs":[],"position":{"left":104.55748448943986,"top":579.0493971252434},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"fcb0f86e-455a-4a6e-9c86-d8994caf0493","when":null,"workflow_outputs":[]},"6":{"annotation":"Select the database to use for Busco lineages. ","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Select the database to use for Busco lineages. ","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":147.59214794655475,"top":666.080035105452},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"245a5b51-c40d-4425-81ca-2baf3c157022","when":null,"workflow_outputs":[]},"7":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":178.35453362075538,"top":770.3740188995779},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a9df063c-37ef-440d-93e6-212ac3a598e8","when":null,"workflow_outputs":[]},"8":{"annotation":"Name of the first haplotype without blank space. ","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Name of the first haplotype without blank space. ","name":"Name for Haplotype 1"}],"label":"Name for Haplotype 1","name":"Input parameter","outputs":[],"position":{"left":188.0521006912058,"top":852.6092753452594},"tool_id":null,"tool_state":"{\\"default\\": \\"Hap1\\", \\"multiple\\": false, \\"validators\\": [{\\"message\\": \\"Please do not use blank spaces\\", \\"expression\\": \\"^\\\\\\\\S+$\\", \\"negate\\": false, \\"type\\": \\"regex\\"}], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"41d1bb31-75f9-4034-abad-96aed4916b45","when":null,"workflow_outputs":[]},"9":{"annotation":"Name of the second haplotype without blank space. ","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Name of the second haplotype without blank space. ","name":"Name for Haplotype 2"}],"label":"Name for Haplotype 2","name":"Input parameter","outputs":[],"position":{"left":196.84876717523315,"top":939.8716854194543},"tool_id":null,"tool_state":"{\\"default\\": \\"Hap2\\", \\"multiple\\": false, \\"validators\\": [{\\"message\\": \\"Please do not use blank spaces\\", \\"expression\\": \\"^\\\\\\\\S+$\\", \\"negate\\": false, \\"type\\": \\"regex\\"}], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6d5afa06-c3e8-4eb7-8819-8cfe8300c462","when":null,"workflow_outputs":[]},"10":{"annotation":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Defaults to 37 if not specified. For genomes much larger than human, applying -f38 or even -f39 is preferred to save memory on k-mer counting.","name":"Bits for bloom filter"}],"label":"Bits for bloom filter","name":"Input parameter","outputs":[],"position":{"left":203.3067924052237,"top":1033.0113085913417},"tool_id":null,"tool_state":"{\\"default\\": 37, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b949921b-95a3-4d43-81b2-4006a4909e4b","when":null,"workflow_outputs":[]},"11":{"annotation":"Do not use. ","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"Do not use. ","name":"SAK input file"}],"label":"SAK input file","name":"Input dataset","outputs":[],"position":{"left":220.84188432960252,"top":1142.6599342462896},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"557a2e2b-008c-4566-836a-6509950aa85b","when":null,"workflow_outputs":[]},"12":{"annotation":"If empty, read coverage will be estimated from the Genomescope parameters. ","content_id":null,"errors":null,"id":12,"input_connections":{},"inputs":[{"description":"If empty, read coverage will be estimated from the Genomescope parameters. ","name":"Homozygous Read Coverage"}],"label":"Homozygous Read Coverage","name":"Input parameter","outputs":[],"position":{"left":225.87340829197328,"top":1228.4076056463502},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b251eddc-49e6-4899-a893-a904c29b9e75","when":null,"workflow_outputs":[]},"13":{"annotation":"GenomeScope model parameters generated by K-mer profiling workflow","content_id":null,"errors":null,"id":13,"input_connections":{},"inputs":[{"description":"GenomeScope model parameters generated by K-mer profiling workflow","name":"Genomescope Model Parameters"}],"label":"Genomescope Model Parameters","name":"Input dataset","outputs":[],"position":{"left":294.59996273219525,"top":1368.5129541390972},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0735f7ce-74e8-4ae0-8142-e6d2fd89c89e","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":14,"input_connections":{"library|input_1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":null,"name":"Cutadapt","outputs":[{"name":"out1","type":"fastqsanger"},{"name":"report","type":"txt"},{"name":"json_stats","type":"json"}],"position":{"left":1195.84375,"top":332.72265625},"post_job_actions":{"HideDatasetActionjson_stats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"json_stats"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"},"RenameDatasetActionout1":{"action_arguments":{"newname":"Cutadapt on #{library.input_1 }"},"action_type":"RenameDatasetAction","output_name":"out1"},"TagDatasetActionout1":{"action_arguments":{"tags":"trimmed_hifi"},"action_type":"TagDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"35\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": true}, \\"filter_options\\": {\\"discard_trimmed\\": true, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"1\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [], \\"front_adapters\\": [], \\"anywhere_adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT\\"}, \\"single_noindels\\": false}, {\\"__index__\\": 1, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"\\", \\"adapter\\": \\"ATCTCTCTCTTTTCCTCCTCCTCCGTTGTTGTTGTTGAGAGAGAT\\"}, \\"single_noindels\\": false}]}}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"0\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\", \\"json_stats\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"64ce3a66-9497-4937-a1b6-7f72814f1fe1","when":null,"workflow_outputs":[{"label":"Cutadapt on input dataset(s): Read 1 Output","output_name":"out1","uuid":"24af0265-1dce-43a5-9fd2-51b414f0c1ea"}]},"15":{"annotation":"","content_id":"__SORTLIST__","errors":null,"id":15,"input_connections":{"input":{"id":2,"output_name":"output"}},"inputs":[],"label":"Sort Forward","name":"Sort collection","outputs":[{"name":"output","type":"input"}],"position":{"left":310.6527425605535,"top":248.36123371697312},"post_job_actions":{},"tool_id":"__SORTLIST__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sort_type\\": {\\"sort_type\\": \\"alpha\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"e980165d-9904-48c2-a296-0281da95a30a","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"__SORTLIST__","errors":null,"id":16,"input_connections":{"input":{"id":3,"output_name":"output"}},"inputs":[],"label":"Sort Reverse","name":"Sort collection","outputs":[{"name":"output","type":"input"}],"position":{"left":342.6490550998522,"top":442.3649381616605},"post_job_actions":{},"tool_id":"__SORTLIST__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sort_type\\": {\\"sort_type\\": \\"alpha\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"07c89224-358a-4423-b200-fd01d98e2454","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":17,"input_connections":{"infile":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":433.16701749931593,"top":57.458678282421076},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-G\\", \\"url_paste\\": \\"Haploid\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"72b7138a-e0d4-4922-9895-c6cbacddbb90","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":18,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":645.6652024686778,"top":1018.014255963943},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"37\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"981127b6-37ac-4163-8036-da7aaacf83e7","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":19,"input_connections":{"input":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":608.9761001191155,"top":1536.5226390907337},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"c3*2\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"bfa738cb-d9c1-4b3c-a89b-71135d65a3b6","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":20,"input_connections":{"results_0|software_cond|input":{"id":14,"output_name":"report"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"comment"},{"description":"runtime parameter for tool MultiQC","name":"image_content_input"},{"description":"runtime parameter for tool MultiQC","name":"title"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1485.8575094249647,"top":388},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"title\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"c60d4342-a7fb-4dd1-8f5f-206fa7e423b8","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":21,"input_connections":{"library|input_1":{"id":15,"output_name":"output"},"library|input_2":{"id":16,"output_name":"output"},"when":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"},{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":null,"name":"Cutadapt","outputs":[{"name":"out1","type":"fastqsanger"},{"name":"out2","type":"fastqsanger"}],"position":{"left":621.4406238766769,"top":371.3248476325246},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"1\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"paired\\", \\"__current_case__\\": 1, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}, \\"r2\\": {\\"adapters2\\": [], \\"front_adapters2\\": [], \\"anywhere_adapters2\\": []}, \\"pair_adapters\\": false}, \\"other_trimming_options\\": {\\"cut\\": \\"5\\", \\"cut2\\": \\"5\\", \\"quality_cutoff\\": \\"0\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": null, \\"read_mod_options\\": {\\"strip_suffix\\": null, \\"length_tag\\": null, \\"rename\\": null, \\"zero_cap\\": false}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"b4079de2-e671-4896-93c5-aa57d264d8b0","when":"$(inputs.when)","workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","errors":null,"id":22,"input_connections":{"infile":{"id":17,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":635.8575094249649,"top":202},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"bp\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 1, \\"find_pattern\\": \\",\\", \\"replace_pattern\\": \\"\\", \\"sed_options\\": null}, {\\"__index__\\": 2, \\"find_pattern\\": \\"([a-z])\\\\\\\\s+([A-Z])\\", \\"replace_pattern\\": \\"\\\\\\\\1_\\\\\\\\2\\", \\"sed_options\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"565beb6a-79f6-4ceb-80f3-46939db4d9c9","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"Cut1","errors":null,"id":23,"input_connections":{"input":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":699.3466983298952,"top":1363.8044347342013},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ca6bbaa5-0c32-4504-94ba-6f8644e6f655","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":24,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":21,"output_name":"out1"},"style_cond|type_cond|pick_from_1|value":{"id":15,"output_name":"output"}},"inputs":[],"label":"Pick forward","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":907.8456367727276,"top":416.106078365762},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Clean Hi-C forward reads"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"fbf4bd28-cf48-4bba-863a-1d16b51bb2c8","when":null,"workflow_outputs":[{"label":"Clean Hi-C forward reads","output_name":"data_param","uuid":"99ea6e30-2d9f-45ce-addb-d50a454d13e4"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":25,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":21,"output_name":"out2"},"style_cond|type_cond|pick_from_1|value":{"id":16,"output_name":"output"}},"inputs":[],"label":"Pick reverse","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":907.6455364524222,"top":634.9981928194156},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Clean Hi-C reverse reads"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"38203c0e-036b-4df3-85b7-22a1165043f4","when":null,"workflow_outputs":[{"label":"Clean Hi-C reverse reads","output_name":"data_param","uuid":"6f4438f5-2019-4b00-8991-a46fe52534d1"}]},"26":{"annotation":"","content_id":"Convert characters1","errors":null,"id":26,"input_connections":{"input":{"id":22,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Convert","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":915.8575094249649,"top":202},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Convert characters1","tool_state":"{\\"condense\\": true, \\"convert_from\\": \\"s\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strip\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"e901d941-c0fe-4915-b9e7-1a7e69596bd2","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":27,"input_connections":{"input1":{"id":23,"output_name":"out_file1"}},"inputs":[],"label":"Estimated homozygous read coverage","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":941.0871317158083,"top":1488.40625},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated homozygous read coverage"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"e1360db3-bf77-402c-a450-36d74adbddc6","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":28,"input_connections":{"input_list":{"id":24,"output_name":"data_param"}},"inputs":[],"label":"Collapse forward reads","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1181.8657286235991,"top":617.5639606408357},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"009222f9-3b14-4fb2-817f-01a21354d5c5","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":29,"input_connections":{"input_list":{"id":25,"output_name":"data_param"}},"inputs":[],"label":"Collapse reverse reads","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1187.9269369021918,"top":787.2365514262615},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"daf1b0aa-221a-4abd-b883-1f12a40afdd6","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"Cut1","errors":null,"id":30,"input_connections":{"input":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1195.8575094249647,"top":202},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"34ffee0e-99e4-4c04-ad64-1d3a50b39102","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":31,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":12,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":27,"output_name":"integer_param"}},"inputs":[],"label":"Homozygous read coverage for Hifiasm","name":"Pick parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1054.1457254658083,"top":1063.5234375},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Homozygous read coverage for Hifiasm"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"6f942d67-c6db-46b8-8207-20f00402e86b","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":32,"input_connections":{"input1":{"id":30,"output_name":"out_file1"}},"inputs":[],"label":"Estimated genome size","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1485.8575167538206,"top":151},"post_job_actions":{"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated Genome size"},"action_type":"RenameDatasetAction","output_name":"integer_param"},"TagDatasetActioninteger_param":{"action_arguments":{"tags":"estimated_genome_size"},"action_type":"TagDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"e0531bfd-5007-4e75-8f71-2f06c123831b","when":null,"workflow_outputs":[{"label":"Estimated Genome size","output_name":"integer_param","uuid":"a6854349-56a1-4ebd-845b-237f3600a0a6"}]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","errors":null,"id":33,"input_connections":{"assembly_options|hom_cov":{"id":31,"output_name":"integer_param"},"filter_bits":{"id":18,"output_name":"integer_param"},"hic_partition|h1":{"id":28,"output_name":"output"},"hic_partition|h2":{"id":29,"output_name":"output"},"mode|reads":{"id":14,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool Hifiasm","name":"assembly_options"},{"description":"runtime parameter for tool Hifiasm","name":"hic_partition"},{"description":"runtime parameter for tool Hifiasm","name":"hic_partition"},{"description":"runtime parameter for tool Hifiasm","name":"mode"}],"label":"Hifiasm","name":"Hifiasm","outputs":[{"name":"noseq_files","type":"input"},{"name":"bin_files","type":"input"},{"name":"hic_pcontig_graph","type":"gfa1"},{"name":"hic_acontig_graph","type":"gfa1"},{"name":"hic_balanced_contig_hap1_graph","type":"gfa1"},{"name":"hic_balanced_contig_hap2_graph","type":"gfa1"},{"name":"hic_raw_initig","type":"gfa1"},{"name":"log_file","type":"txt"}],"position":{"left":1485.8515369004044,"top":593.2567305909766},"post_job_actions":{"HideDatasetActionbin_files":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bin_files"},"HideDatasetActionhic_acontig_graph":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"hic_acontig_graph"},"HideDatasetActionhic_pcontig_graph":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"hic_pcontig_graph"},"TagDatasetActionhic_balanced_contig_hap1_graph":{"action_arguments":{"tags":"hifiasm_hic_hap1_gfa, #hap1"},"action_type":"TagDatasetAction","output_name":"hic_balanced_contig_hap1_graph"},"TagDatasetActionhic_balanced_contig_hap2_graph":{"action_arguments":{"tags":"hifiasm_hic_hap2_gfa, #hap2"},"action_type":"TagDatasetAction","output_name":"hic_balanced_contig_hap2_graph"},"TagDatasetActionhic_raw_initig":{"action_arguments":{"tags":"hifiasm_hic_unitig_gfa"},"action_type":"TagDatasetAction","output_name":"hic_raw_initig"},"TagDatasetActionlog_file":{"action_arguments":{"tags":"hifiasm_hic_log"},"action_type":"TagDatasetAction","output_name":"log_file"},"TagDatasetActionnoseq files":{"action_arguments":{"tags":"noseq_hifiasm"},"action_type":"TagDatasetAction","output_name":"noseq files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0","tool_shed_repository":{"changeset_revision":"170061c052ad","name":"hifiasm","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"advanced_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"assembly_options\\": {\\"assembly_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"cleaning_rounds\\": \\"4\\", \\"adapter_length\\": \\"0\\", \\"pop_contigs\\": \\"10000000\\", \\"pop_unitigs\\": \\"100000\\", \\"remove_tips\\": \\"3\\", \\"max_overlap\\": \\"0.8\\", \\"min_overlap\\": \\"0.2\\", \\"disable_post_join\\": false, \\"ignore_error_corrected\\": false, \\"hom_cov\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"bins_out\\": true, \\"filter_bits\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hic_partition\\": {\\"hic_partition_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"h1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"h2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seed\\": null, \\"n_weight\\": null, \\"n_perturb\\": null, \\"f_perturb\\": null, \\"l_msjoin\\": \\"500000\\"}, \\"log_out\\": true, \\"mode\\": {\\"mode_selector\\": \\"standard\\", \\"__current_case__\\": 0, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"ont_integration\\": {\\"ont_integration_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"purge_options\\": {\\"purge_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"scaffolding_options\\": {\\"scaffold_selector\\": \\"blank\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy0","type":"tool","uuid":"0178b5a1-d244-43bc-85e4-e83d1360a94e","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":34,"input_connections":{"input_file":{"id":33,"output_name":"hic_raw_initig"}},"inputs":[],"label":"Raw Unitig Image","name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":1795.8575240826765,"top":0},"post_job_actions":{"TagDatasetActionoutfile":{"action_arguments":{"tags":"utg_png"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"2000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": \\"25.0\\", \\"output_format\\": \\"png\\", \\"width\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"c023af81-3c79-43f9-96df-8d982b19e4ed","when":null,"workflow_outputs":[{"label":"raw unitig graph image","output_name":"outfile","uuid":"973f1ee3-d67f-4126-b9c0-301c9b0be6e7"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":35,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap2_graph"},"mode_condition|statistics_condition|expected_genomesize":{"id":32,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":1792.92938400605,"top":210.64959935584233},"post_job_actions":{"RenameDatasetActionstats":{"action_arguments":{"newname":"Assembly stats for hap2"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_hap2, #hap2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"001f8c0e-234a-4406-9d8b-a26b8e878c56","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":36,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap1_graph"},"mode_condition|statistics_condition|expected_genomesize":{"id":32,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":1795.8575240826765,"top":404},"post_job_actions":{"RenameDatasetActionstats":{"action_arguments":{"newname":"Assembly stats for hap1"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_asm_hap1, #hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"fe43bf41-0e43-482e-99bb-7a855d4bcb2f","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":37,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap2_graph"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":1795.8575240826765,"top":612},"post_job_actions":{"RenameDatasetActionstats":{"action_arguments":{"newname":"Contig sizes for hap2"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_hap2, #hap2"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"723d5c3e-c0ec-4085-bccd-29d5d97ce3f8","when":null,"workflow_outputs":[]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":38,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap2_graph"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"gfastats gfa hap2 no sequences","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":1805.2477750750836,"top":760.7954647500668},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"No Sequences hap2 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hic_hap2_gfa_noseq"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": true}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"4d4a30a0-d2e9-4814-afc2-74e4194ebdf5","when":null,"workflow_outputs":[{"label":"No Sequences hap2 gfa","output_name":"output","uuid":"0bba8e16-b835-40db-a15c-f4d045ec704b"}]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":39,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap1_graph"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":1795.8575240826765,"top":1017},"post_job_actions":{"RenameDatasetActionstats":{"action_arguments":{"newname":"Contig sizes for hap1"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_hap1, #hap1"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": true}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"ed95e95b-68f3-47f4-ac03-eb4fb01d3020","when":null,"workflow_outputs":[]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":40,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap1_graph"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"gfastats gfa hap1","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":1795.8575240826765,"top":1195},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"usable hap1 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hic_hap1_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"2bf6f1be-461a-45e1-aa5f-0af394563a23","when":null,"workflow_outputs":[{"label":"usable hap1 gfa","output_name":"output","uuid":"853d982b-69e8-436f-a791-4f7875034109"}]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":41,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap2_graph"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"gfastats gfa hap2","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2068.1821792575515,"top":911.7949109092888},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"usable hap2 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hic_hap2_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": false}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"9a8c167d-d645-4da3-bb22-105c179b2a6d","when":null,"workflow_outputs":[{"label":"usable hap2 gfa","output_name":"output","uuid":"03228d5f-a300-4321-b715-ffc466c21246"}]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":42,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap2_graph"},"mode_condition|swiss_army_knife":{"id":11,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":1795.8575240826765,"top":1422},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"596ea56e-3c73-402a-b1ae-e2be1bb91227","when":null,"workflow_outputs":[]},"43":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":43,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap1_graph"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"gfastats gfa hap1 no sequences","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2061.8604376614403,"top":1140.328071721209},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"No sequences hap1 gfa"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hic_hap1_gfa_noseq"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"no_sequence\\": true}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"36a45c3e-cb65-4d64-9dee-15310637751e","when":null,"workflow_outputs":[{"label":"No sequences hap1 gfa","output_name":"output","uuid":"b13512c3-caf2-4c4b-abf1-b57749b6b00c"}]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":44,"input_connections":{"input_file":{"id":33,"output_name":"hic_balanced_contig_hap1_graph"},"mode_condition|swiss_army_knife":{"id":11,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":null,"name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":1795.8575240826765,"top":1627},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"line_length\\": null}, \\"discover_paths\\": true, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"cd65bf95-931d-4514-8851-6652198c7ab7","when":null,"workflow_outputs":[]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":45,"input_connections":{"infile":{"id":35,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2075.8575240826767,"top":223},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\thap2\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"3ae908c6-47d7-40b0-b73e-95f1ed49dd24","when":null,"workflow_outputs":[]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":46,"input_connections":{"infile":{"id":36,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2082.567600465808,"top":405.10546875},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\thap1\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"e0c8072a-6132-4669-be68-a44981f659da","when":null,"workflow_outputs":[]},"47":{"annotation":"","id":47,"input_connections":{"gfa_stats":{"id":37,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data Prep Hap2"}],"label":"Data Prep Hap2","name":"gfastats_data_prep","outputs":[],"position":{"left":2075.8575240826767,"top":624},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7f250de3-98cc-448b-b573-6b8a85c71352","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"68af2c3b-8f48-4bde-9dd1-a78151d6de1a"}]}},"tags":[],"uuid":"d4ac965c-03ac-43a4-880a-36d4e713a98b"},"tool_id":null,"type":"subworkflow","uuid":"9d2e48a6-d050-49ee-af0a-31ca952c4535","when":null,"workflow_outputs":[]},"48":{"annotation":"","id":48,"input_connections":{"gfa_stats":{"id":39,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[{"description":"","name":"Data Prep Hap1"}],"label":"Data Prep Hap1","name":"gfastats_data_prep","outputs":[],"position":{"left":2077.376194215808,"top":784.40234375},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7f250de3-98cc-448b-b573-6b8a85c71352","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"68af2c3b-8f48-4bde-9dd1-a78151d6de1a"}]}},"tags":[],"uuid":"f2d914a1-3576-45be-8b91-3f5ebb7261fe"},"tool_id":null,"type":"subworkflow","uuid":"7df6c0b4-4fdc-43fc-bb3e-cc1d43c5ad0b","when":null,"workflow_outputs":[]},"49":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","errors":null,"id":49,"input_connections":{"infile":{"id":42,"output_name":"output"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":2075.8575240826767,"top":1372},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm HiC hap2"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_hic_hap2,hifiasm_Assembly_Haplotype_2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/_path//g\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"d8a10e0c-5a9c-4c4f-910a-8a94cdcba7f0","when":null,"workflow_outputs":[{"label":"Hifiasm Hi-C hap2","output_name":"output","uuid":"dd375833-aa34-4529-af99-c8be93fb013a"}]},"50":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","errors":null,"id":50,"input_connections":{"infile":{"id":44,"output_name":"output"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":2075.8575240826767,"top":1610},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Hifiasm HiC hap1"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"hifiasm_hic_hap1,hifiasm_Assembly_Haplotype_1"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/_path//g\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"82af5c81-e733-4e57-b931-8234ab6c8056","when":null,"workflow_outputs":[{"label":"Hifiasm Hi-C hap1","output_name":"output","uuid":"86addfe0-5939-45d0-bb10-8e9bff0bd220"}]},"51":{"annotation":"","content_id":"join1","errors":null,"id":51,"input_connections":{"input1":{"id":46,"output_name":"outfile"},"input2":{"id":45,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2435.4465067158085,"top":424.50390625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"fill_empty\\", \\"__current_case__\\": 1, \\"fill_columns_by\\": \\"fill_unjoined_only\\", \\"do_fill_empty_columns\\": {\\"column_fill_type\\": \\"single_fill_value\\", \\"__current_case__\\": 0, \\"fill_value\\": \\".\\"}}, \\"header\\": \\"\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"00e95547-9451-4c37-bc19-b29de3864c6a","when":null,"workflow_outputs":[]},"52":{"annotation":"","id":52,"input_connections":{"Alternate data":{"id":47,"input_subworkflow_step_id":1,"output_name":"gfastats data for plotting"},"Name of alternate assembly":{"id":9,"input_subworkflow_step_id":3,"output_name":"output"},"Name of primary assembly":{"id":8,"input_subworkflow_step_id":2,"output_name":"output"},"Primary data":{"id":48,"input_subworkflow_step_id":0,"output_name":"gfastats data for plotting"}},"inputs":[{"description":"","name":"Plot Data"}],"label":"Plot Data","name":"gfastats_plot","outputs":[],"position":{"left":2375.8575240826767,"top":624},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","name":"gfastats_plot","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Primary data"}],"label":"Primary data","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a9a972e8-76f3-416f-9148-77203f6df3b8","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Alternate data"}],"label":"Alternate data","name":"Input dataset","outputs":[],"position":{"left":6.960205078125,"top":143.53692626953125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"a5ce76b6-ae69-4c5d-b42f-dc211c9b64cc","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4.55963134765625,"top":296.3210144042969},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6118c82a-61da-4856-9e25-1f0d93297527","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"894be62c-079d-4209-a098-1858de70948a"}]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":7.428955078125,"top":440.4403076171875},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"3204b6e0-bd83-4a8d-8ddb-3b0dd89dbd04","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"3d7657f7-0651-43c1-80bb-57c4fa6d3502"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"exp":{"id":2,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1139.7584686279297,"top":107.42897033691406},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"8bc6adbb-1346-4521-bd69-ffa2738bad07","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"exp":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1151.7044982910156,"top":392.65625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"63c7b63c-5179-4ab3-9682-7d150029446e","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":6,"input_connections":{"inputs":{"id":4,"output_name":"out_file1"},"queries_0|inputs2":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1441.193115234375,"top":216.59091186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"edbf8363-cf44-4536-b45c-60d54088e0fb","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1780.3265991210938,"top":182.99716186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ffa38570-79c9-41e6-9a55-3245d5427fa1","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1781.349365234375,"top":387.96875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6567b833-44b5-41e6-885b-d1d17b5d9376","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2068.1390991210938,"top":31.988632202148438},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"18c6ac70-f4d3-4d68-93f8-c08afafbc6c5","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"d1dd55bd-0441-4cfd-baf2-767ef768d01e"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2083.3805541992188,"top":319.8011169433594},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"cce1fe43-55c6-4c1a-a957-ac7af9271c46","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"c979a10d-de93-4c02-b286-0ec0fca016b9"}]}},"tags":[],"uuid":"c442d6b4-7893-48ef-a7d6-af90db707cd2"},"tool_id":null,"type":"subworkflow","uuid":"1b8e2d4e-3380-4414-bdb0-b356383c0ee2","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"Size Plot","uuid":"03a0d2a8-c2d5-4ca7-80fe-7d04d1c4894a"},{"label":"Nx Plot","output_name":"Nx Plot","uuid":"781f8511-c3fe-4954-afa8-d21c94c996a4"}]},"53":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":53,"input_connections":{"input":{"id":49,"output_name":"output"},"lineage_conditional|cached_db":{"id":6,"output_name":"output"},"lineage|lineage_dataset":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":"Busco Hap2","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":2375.633304847493,"top":867.7265312174471},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"#hap2,busco_hap2_gff"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_hap2_missing, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_hap2_summ, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_hap2_full, #hap2"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_hap2_img, #hap2"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"08066bc3-0992-4606-844d-16be3c56691a","when":null,"workflow_outputs":[{"label":"Busco Summary Hap2","output_name":"busco_sum","uuid":"bf7e959c-5252-467e-bb58-2b8cabbd0f3b"},{"label":"Busco Gff File - Hap2","output_name":"busco_gff","uuid":"3b8e8b67-ad7d-4d30-a253-e601d3e201d2"},{"label":"Busco Summary Image Hap2","output_name":"summary_image","uuid":"84725fc4-3480-44a2-abc0-abc8f54addb1"}]},"54":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","errors":null,"id":54,"input_connections":{"mode|assembly_options|assembly_01":{"id":50,"output_name":"output"},"mode|assembly_options|assembly_02":{"id":49,"output_name":"output"},"mode|meryldb_F1":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Merqury","name":"mode"}],"label":"Merqury","name":"Merqury","outputs":[{"name":"qv_files","type":"input"},{"name":"png_files","type":"input"},{"name":"stats_files","type":"input"},{"name":"hist_files","type":"input"},{"name":"log_file","type":"txt"}],"position":{"left":2373.6806693400385,"top":1497.1314889797957},"post_job_actions":{"HideDatasetActionlog_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log_file"},"TagDatasetActionbed_files":{"action_arguments":{"tags":"merqury_hifi_hic_bed, #hic_phased"},"action_type":"TagDatasetAction","output_name":"bed_files"},"TagDatasetActionhist_files":{"action_arguments":{"tags":"#hic_phased,merqury_hifi_hic_hist"},"action_type":"TagDatasetAction","output_name":"hist_files"},"TagDatasetActionpng_files":{"action_arguments":{"tags":"merqury_hifi_hic_png, #hic_phased"},"action_type":"TagDatasetAction","output_name":"png_files"},"TagDatasetActionqv_files":{"action_arguments":{"tags":"merqury_hifi_hic_qv, #hic_phased"},"action_type":"TagDatasetAction","output_name":"qv_files"},"TagDatasetActionsizes_files":{"action_arguments":{"tags":"merqury_hifi_hic_sizes, #hic_phased"},"action_type":"TagDatasetAction","output_name":"sizes_files"},"TagDatasetActionstats_files":{"action_arguments":{"tags":"merqury_hifi_hic_stats, #hic_phased"},"action_type":"TagDatasetAction","output_name":"stats_files"},"TagDatasetActionwig_files":{"action_arguments":{"tags":"merqury_hifi_hic_wig, #hic_phased"},"action_type":"TagDatasetAction","output_name":"wig_files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","tool_shed_repository":{"changeset_revision":"eb7b00ec347d","name":"merqury","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"label\\": \\"output_merqury\\", \\"mode\\": {\\"options\\": \\"default\\", \\"__current_case__\\": 0, \\"meryldb_F1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_options\\": {\\"number_assemblies\\": \\"two\\", \\"__current_case__\\": 1, \\"assembly_01\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_02\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_add_headers\\": true, \\"output_selector\\": [\\"qv\\", \\"plots\\", \\"stats\\", \\"log\\", \\"hist\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy4","type":"tool","uuid":"9985e575-3113-4c2c-bc2c-78faab3a1291","when":null,"workflow_outputs":[{"label":"Merqury Histograms","output_name":"hist_files","uuid":"67d9d45d-989d-43c0-9ebd-0da9cc35763a"},{"label":"Merqury QV","output_name":"qv_files","uuid":"5ba07c29-6ca5-42bd-a1e8-f4f6a35c3814"},{"label":"Merqury Stats","output_name":"stats_files","uuid":"22a82fae-74cf-4024-9815-ba6a9892aac3"},{"label":"Merqury images","output_name":"png_files","uuid":"4983a97f-9dd1-44d2-a4be-d13e9a651f41"}]},"55":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":55,"input_connections":{"input":{"id":50,"output_name":"output"},"lineage_conditional|cached_db":{"id":6,"output_name":"output"},"lineage|lineage_dataset":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":"Busco Hap1","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":2665.087072336259,"top":1053.8484191350424},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"#hap1,busco_hap1_gff"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_hap1_missing, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_hap1_summ, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_hap1_full, #hap1"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_hap1_img, #hap1"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"a4957dee-45cc-4f34-ae2e-e07f93b9fcc4","when":null,"workflow_outputs":[{"label":"Busco Gff File - Hap1","output_name":"busco_gff","uuid":"ba26cb24-25d4-4669-982c-abe4816753a6"},{"label":"Busco Summary Hap1","output_name":"busco_sum","uuid":"bd6443ea-5266-4dfc-8f1e-aded27317da8"},{"label":"Busco Summary Image Hap1","output_name":"summary_image","uuid":"cfac3083-4c1e-4bb4-91a5-eba5d19f2757"}]},"56":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":56,"input_connections":{"input":{"id":51,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2737.1183817158085,"top":417.2265625},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Assembly statistics for Hap1 and Hap2"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"gfastats_asm_hap1_hap2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": [\\"3\\"]}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"769fbac6-1e77-4d5f-8dba-b97d89fa8005","when":null,"workflow_outputs":[{"label":"Assembly statistics for Hap1 and Hap2","output_name":"output","uuid":"0de77a25-8f99-4e56-aab0-3cab170cc4b2"}]},"57":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":57,"input_connections":{"input":{"id":54,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2676.118294606436,"top":1728.078125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9b56f0a5-6a0c-4280-9338-c08831e89c3c","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-cn.fl","output_name":"output","uuid":"4fd32211-7056-480e-a847-475eb2c864ac"}]},"58":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":58,"input_connections":{"input":{"id":54,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-asm.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2673.708138356436,"top":1890.59375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-asm.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"7fc6530d-4701-40c7-8b83-ba94ac3f11b5","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-asm.fl","output_name":"output","uuid":"6fad1c89-02e9-4381-b728-33fb5e6b60b1"}]},"59":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":59,"input_connections":{"input":{"id":54,"output_name":"qv_files"}},"inputs":[],"label":"merqury_qv","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2961.453125,"top":1615.171875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9e10d9ad-b16c-4904-a61f-6e1ec40fcf23","when":null,"workflow_outputs":[{"label":"merqury_qv","output_name":"output","uuid":"6d753d76-7437-40aa-9f8d-568b9f19fed4"}]},"60":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":60,"input_connections":{"input":{"id":54,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_01.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2673.821419606436,"top":2059.98046875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_01.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"3d96d433-af93-45f1-aaab-907addea2461","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_01.spectra-cn.fl","output_name":"output","uuid":"cef7fd12-4b0c-4a4f-8f53-e570ae604232"}]},"61":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":61,"input_connections":{"input":{"id":54,"output_name":"stats_files"}},"inputs":[],"label":"merqury_stats","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2960.891732106436,"top":1734.2421875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.completeness\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"c7ad672f-341b-4f2a-a5d2-b5ba9771cfef","when":null,"workflow_outputs":[{"label":"merqury_stats","output_name":"output","uuid":"b560b501-ef22-48f4-abbb-acbf793dfda1"}]},"62":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":62,"input_connections":{"input":{"id":54,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_02.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2670.383919606436,"top":2249.78125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_02.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"75f49e14-af85-4a07-81f1-3b75f591f965","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_02.spectra-cn.fl","output_name":"output","uuid":"c10c5102-6b26-4cdc-bc35-50af09e36d9d"}]},"63":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":63,"input_connections":{"infile":{"id":56,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3039.313898238167,"top":446.09145245110693},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"7528118f-3c8c-47a8-86e8-6085145c38d1","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"0d69a8ab-29c1-4fef-97d9-af243ca8eb83"}]}},"tags":["VGP","Reviewed"],"uuid":"c0a8fe5c-cd56-4070-9431-c646c2aae2c6","version":1},"readme":"# Genome Assembly from Hifi reads with HiC phasing - VGP4\\n\\nGenerate phased assembly based on PacBio Hifi Reads using HiC data from the same individual for assembly phasing. Part of the VGP workflow suite, it needs to be run after the k-mer profiling workflow VGP1.\\n\\n## Inputs\\n\\n1. Hifi long reads [fastq].\\n2. Trim Hi-C reads ? If yes, trim 5 bases at the beginning of each reads. Use with some Arima Hi-C data if the contact map looks \\"noisy\\". \\n3. Collection of Hi-C forward reads [fastq].\\n4. Collection of Hi-C reverse reads [fastq].\\n5. Genome profile summary generated by Genomescope [txt] generated by VGP1 workflow.\\n6. K-mer database [meryldb] generated by VGP1 workflow.\\n7. Database to use for Busco lineages. Recommended : latest version.\\n8. Lineage. Select the taxonomic lineage of the assembled species.  \\n9. Name of first assembly.\\n10. Name of second assembly.\\n11. Bits for bloom filter. Change for large genomes to save memory.\\n12. Homozygous Read Coverage. Optional: specify if the coverage detected by Genomescope in VGP1 in not satisfactory.\\n13. Genomescope model parameters [tabular] generated by VGP1 workflow.\\n\\n## Outputs\\n\\n1. Haplotype 1 assembly ([fasta] and [gfa])\\n2. Haplotype 2 assembly ([fasta] and [gfa])\\n3. Trimmed Hi-C forward reads collection\\n4. Trimmed Hi-C reverse reads collection\\n5. QC: MultiQC report for HiFi reads trimming\\n6. QC: BUSCO report for both assemblies\\n7. QC: Merqury report for both assemblies\\n8. QC: Assembly statistics for both assemblies\\n9. QC: Nx plot for both assemblies\\n10. QC: Size plot for both assemblies\\n","changelog":"# Changelog\\n\\n## [0.3.2] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.24.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.3.1] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.3.0] - 2025-02-26\\n\\n### Changes\\n\\n- Output bin files from Hifiasm. Helpful for debugging\\n- Output lighter gfas without sequences along with the regular gfas \\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.2.4] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n## [0.2.3] - 2025-01-13\\n\\n### Changes\\n\\n- Hi-C inputs are now entered as collections instead of single datasets\\n- Add optional trimming of Hi-C reads\\n- Add inputs annotations\\n- Add validator for haplotype names: No blank spaces \\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.20.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0`\\n\\n## [0.2.2] - 2024-08-07\\n\\n- Expose parameter: Busco lineage database\\n\\n## [0.2.1] - 2024-07-29\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.9+galaxy0`\\n\\n## [0.2] - 2024-07-12\\n\\n### Added \\n\\n- Workflow report\\n\\n### Changed\\n\\n- Fix bug where the columns of the assembly statistics didn\'t join properly. \\n  \\n\\n## [0.1.11] - 2024-05-20\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n\\n\\n\\n## [0.1.10] - 2024-04-23\\n\\n### Changed\\n\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n\\n## [0.1.9] - 2024-04-19\\n\\n### Manual Updates\\n\\n- Add the option to provide Homozygous Read coverage parameter\\n\\n### Automatic update\\n\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4`\\n\\n\\n## [0.1.8] - 2024-03-25\\n\\n### Manual update\\n\\n- Remove tag filtering\\n- Add merqury histogram output\\n- Add output for cutadapt\\n- Output Gff files in Busco \\n\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy1`\\n\\n## [0.1.7] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.5] - 2023-11-15\\n\\n### Added\\n\\n- more descriptive labels for inputs\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0`\\n\\n## [0.1.4] - 2023-11-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.16.1+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.19.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.1.3] - 2023-11-08\\n\\n- Added tags `hifiasm_Assembly_Haplotype_1` and `hifiasm_Assembly_Haplotype_2` to fasta outputs of hifiasm, so they can be easily identified for feeding to purge_dups workflow\\n- Made BUSCO lineages selectable from workflow launch interface\\n\\n## [0.1.2] - 2023-11-07\\n\\n- Added input filtering based on tags. Rearranged input ordering.\\n\\n## [0.1] - 2023-09-27\\n\\n- Creation of workflow for HiFi assembly with Hi-C phasing.\\n","diagrams":"# Workflow diagrams\\n\\n## Genome Assembly from Hifi reads with HiC phasing - VGP4\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Pacbio Reads\\"]@{ shape: docs }\\n1[\\"ℹ️ Trim Hi-C reads?\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Hi-C reads - Forward\\"]@{ shape: docs }\\n3[\\"ℹ️ Hi-C reads - Reverse\\"]@{ shape: docs }\\n4[\\"ℹ️ Genomescope Summary\\"]@{ shape: doc }\\n5[\\"ℹ️ Meryl Database\\"]@{ shape: doc }\\n6[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Name for Haplotype 1\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Name for Haplotype 2\\"]@{ shape: lean-l }\\n10[\\"ℹ️ Bits for bloom filter\\"]@{ shape: lean-l }\\n11[\\"ℹ️ SAK input file\\"]@{ shape: doc }\\n12[\\"ℹ️ Homozygous Read Coverage\\"]@{ shape: lean-l }\\n13[\\"ℹ️ Genomescope Model Parameters\\"]@{ shape: doc }\\n14[\\"Cutadapt\\"]@{ shape: process }\\n0 --> 14\\n15[\\"Sort Forward\\"]@{ shape: process }\\n2 --> 15\\n16[\\"Sort Reverse\\"]@{ shape: process }\\n3 --> 16\\n17[\\"Search in textfiles\\"]@{ shape: process }\\n4 --> 17\\n18[\\"Pick parameter value\\"]@{ shape: process }\\n10 --> 18\\n19[\\"Compute\\"]@{ shape: process }\\n13 --> 19\\n20[\\"MultiQC\\"]@{ shape: process }\\n14 --> 20\\n21[\\"Cutadapt\\"]@{ shape: process }\\n15 --> 21\\n16 --> 21\\n1 --> 21\\n22[\\"Replace Text\\"]@{ shape: process }\\n17 --> 22\\n23[\\"Cut\\"]@{ shape: process }\\n19 --> 23\\n24[\\"Pick forward\\"]@{ shape: process }\\n21 --> 24\\n15 --> 24\\n25[\\"Pick reverse\\"]@{ shape: process }\\n21 --> 25\\n16 --> 25\\n26[\\"Convert\\"]@{ shape: process }\\n22 --> 26\\n27[\\"Estimated homozygous read coverage\\"]@{ shape: process }\\n23 --> 27\\n28[\\"Collapse forward reads\\"]@{ shape: process }\\n24 --> 28\\n29[\\"Collapse reverse reads\\"]@{ shape: process }\\n25 --> 29\\n30[\\"Cut\\"]@{ shape: process }\\n26 --> 30\\n31[\\"Homozygous read coverage for Hifiasm\\"]@{ shape: process }\\n12 --> 31\\n27 --> 31\\n32[\\"Estimated genome size\\"]@{ shape: process }\\n30 --> 32\\n33[\\"Hifiasm\\"]@{ shape: process }\\n31 --> 33\\n18 --> 33\\n28 --> 33\\n29 --> 33\\n14 --> 33\\n34[\\"Raw Unitig Image\\"]@{ shape: process }\\n33 --> 34\\n35[\\"gfastats\\"]@{ shape: process }\\n33 --> 35\\n32 --> 35\\n36[\\"gfastats\\"]@{ shape: process }\\n33 --> 36\\n32 --> 36\\n37[\\"gfastats\\"]@{ shape: process }\\n33 --> 37\\n38[\\"gfastats gfa hap2 no sequences\\"]@{ shape: process }\\n33 --> 38\\n39[\\"gfastats\\"]@{ shape: process }\\n33 --> 39\\n40[\\"gfastats gfa hap1\\"]@{ shape: process }\\n33 --> 40\\n41[\\"gfastats gfa hap2\\"]@{ shape: process }\\n33 --> 41\\n42[\\"gfastats\\"]@{ shape: process }\\n33 --> 42\\n11 --> 42\\n43[\\"gfastats gfa hap1 no sequences\\"]@{ shape: process }\\n33 --> 43\\n44[\\"gfastats\\"]@{ shape: process }\\n33 --> 44\\n11 --> 44\\n45[\\"Text reformatting\\"]@{ shape: process }\\n35 --> 45\\n46[\\"Text reformatting\\"]@{ shape: process }\\n36 --> 46\\n47[\\"🛠️ Data Prep Hap2\\"]@{ shape: subprocess }\\n37 --> 47\\n48[\\"🛠️ Data Prep Hap1\\"]@{ shape: subprocess }\\n39 --> 48\\n49[\\"Text transformation\\"]@{ shape: process }\\n42 --> 49\\n50[\\"Text transformation\\"]@{ shape: process }\\n44 --> 50\\n51[\\"Join two Datasets\\"]@{ shape: process }\\n46 --> 51\\n45 --> 51\\n52[\\"🛠️ Plot Data\\"]@{ shape: subprocess }\\n47 --> 52\\n9 --> 52\\n8 --> 52\\n48 --> 52\\n53[\\"Busco Hap2\\"]@{ shape: process }\\n49 --> 53\\n6 --> 53\\n7 --> 53\\n54[\\"Merqury\\"]@{ shape: process }\\n50 --> 54\\n49 --> 54\\n5 --> 54\\n55[\\"Busco Hap1\\"]@{ shape: process }\\n50 --> 55\\n6 --> 55\\n7 --> 55\\n56[\\"Advanced Cut\\"]@{ shape: process }\\n51 --> 56\\n57[\\"output_merqury.spectra-cn.fl\\"]@{ shape: process }\\n54 --> 57\\n58[\\"output_merqury.spectra-asm.fl\\"]@{ shape: process }\\n54 --> 58\\n59[\\"merqury_qv\\"]@{ shape: process }\\n54 --> 59\\n60[\\"output_merqury.assembly_01.spectra-cn.fl\\"]@{ shape: process }\\n54 --> 60\\n61[\\"merqury_stats\\"]@{ shape: process }\\n54 --> 61\\n62[\\"output_merqury.assembly_02.spectra-cn.fl\\"]@{ shape: process }\\n54 --> 62\\n63[\\"Replace\\"]@{ shape: process }\\n56 --> 63\\n```\\n\\n## gfastats_plot\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Primary data\\"]@{ shape: doc }\\n1[\\"ℹ️ Alternate data\\"]@{ shape: doc }\\n2[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"Concatenate datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Cut\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Nx Plot\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Size Plot\\"]@{ shape: process }\\n8 --> 10\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Assembly-Hifi-HiC-phasing-VGP4/main","dockstore_id":24244,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078891","tests":[{"doc":"Test outline for Assembly-Hifi-HiC-phasing-VGP4","job":{"Hi-C reads - Forward":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"HiC-Forward-reads","location":"https://zenodo.org/records/10068595/files/HiC%20forward%20reads.fastqsanger.gz?download=1","filetype":"fastqsanger.gz"}]},"Hi-C reads - Reverse":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"HiC-Reverse-reads","location":"https://zenodo.org/records/10068595/files/HiC%20reverse%20reads.fastqsanger.gz?download=1","filetype":"fastqsanger.gz"}]},"Genomescope Model Parameters":{"class":"File","path":"test-data/Genomescope Model Parameters.tabular","filetype":"tabular"},"Genomescope Summary":{"class":"File","location":"https://zenodo.org/records/10068595/files/Genomescope%20Summary.txt?download=1","filetype":"txt"},"Meryl Database":{"class":"File","location":"https://zenodo.org/records/10068595/files/Meryl%20Database.meryldb?download=1","filetype":"meryldb"},"Pacbio Reads":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"yeast_reads_sub1.fastq.gz","location":"https://zenodo.org/records/10068595/files/Pacbio%20Reads%20Collection_yeast_reads_sub1.fastq.gz.fastq.gz?download=1"}]},"Lineage":"vertebrata_odb10","Bits for bloom filter":32,"Name for Haplotype 1":"Hap1","Name for Haplotype 2":"Hap2","Homozygous Read Coverage":null,"Database for Busco Lineage":"v5","Trim Hi-C reads?":false},"outputs":{"Hifiasm Hi-C hap1":{"asserts":{"has_n_lines":{"n":114}}},"Estimated Genome size":2288021,"Busco Summary Hap1":{"asserts":{"has_text":{"text":"C:0.8%[S:0.8%,D:0.0%],F:0.3%,M:98.9%,n:3354"}}},"Nx Plot":{"asserts":{"has_size":{"value":65000,"delta":10000}}},"usable hap1 gfa":{"asserts":{"has_n_lines":{"n":119}}},"No Sequences hap2 gfa":{"asserts":{"has_text":{"text":"S\\th2tg000001l\\t*\\tLN:i:43860\\tLN:i:43860\\trd:i:45"}}},"Assembly statistics for Hap1 and Hap2":{"asserts":{"has_text":{"text":"# scaffolds\\t57\\t51"}}}}}]}],"path":"./workflows/VGP-assembly-v2/Assembly-Hifi-HiC-phasing-VGP4"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Generate-Nx-and-Size-plots-for-multiple-assemblies.ga","testParameterFiles":["/Generate-Nx-and-Size-plots-for-multiple-assemblies-tests.yml"],"authors":[{"name":"VGP"},{"name":"Delphine Lariviere","orcid":"0000-0001-6421-3484"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"VGP"},{"class":"Person","identifier":"0000-0001-6421-3484","name":"Delphine Lariviere"}],"format-version":"0.1","release":"0.1.4","name":"Generate Nx and Size plots for multiple assemblies","steps":{"0":{"annotation":"Name the items inside the collection with the names you want to appear on the plot ","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Name the items inside the collection with the names you want to appear on the plot ","name":"Collection of genomes to plot"}],"label":"Collection of genomes to plot","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"307dd5c2-4198-4fbe-81ae-67f94be1ca99","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":1,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":275.89453125,"top":207.73046875},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"1c6d8c57-0554-4e66-aa53-5f03a5022fde","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"sort1","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"stats"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":560.66015625,"top":38.16015625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"55d86a03-5706-4070-b468-5e8407d3c871","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":3,"input_connections":{"infile":{"id":2,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":847.6875,"top":192.03515625},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"69c7b2e7-5c82-49ec-8301-737e5ec1739b","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":4,"input_connections":{"in_file":{"id":3,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":1137.859375,"top":87.41015625},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"1cdafa6b-ccdb-4b90-8dc6-ea1f92425715","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"input":{"id":3,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1154.4453125,"top":525.86328125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a266291-d20c-42ff-b55f-bb334bd520d8","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1401.83984375,"top":304.703125},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"81bf512c-89b3-4b18-b7e9-36ae448a828f","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":7,"input_connections":{"components_1|param_type|component_value":{"id":6,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1650,"top":451.7265625},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bda1a1ff-ab78-4338-a511-d24f851ec102","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":8,"input_connections":{"input":{"id":5,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":7,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1933.33984375,"top":652.31640625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"c93568ca-103f-4d8e-af88-d6de721e30cd","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"68af2c3b-8f48-4bde-9dd1-a78151d6de1a"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/mvdbeek/add_input_name_as_column/addName/0.2.0","errors":null,"id":9,"input_connections":{"input":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Add input name as column","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2259.09765625,"top":708.26171875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/mvdbeek/add_input_name_as_column/addName/0.2.0","tool_shed_repository":{"changeset_revision":"3284b72eef56","name":"add_input_name_as_column","owner":"mvdbeek","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": {\\"contains_header\\": \\"no\\", \\"__current_case__\\": 1}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prepend\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"44431f31-7fb3-469b-87be-154b5a8f14dd","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":10,"input_connections":{"input_list":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2554.69140625,"top":684.2109375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"48dcba84-9af3-4dbf-968b-5a7ab0353f42","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"Cut1","errors":null,"id":11,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2856.78515625,"top":479.484375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"c1d69d96-61e4-4e5d-94b0-a63ea918b659","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"Cut1","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2907.5703125,"top":823.07421875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"0c98ca2a-d4c6-4e6e-898a-aefd7a2d6870","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":13,"input_connections":{"input1":{"id":11,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":3336,"top":541.96484375},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"7.0\\", \\"height_output_dim\\": \\"7.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"d2f3f6a2-61b7-4268-84f9-b3515ef3508b","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"ea8d76de-4b96-42a5-bd99-b4846fc4f869"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":14,"input_connections":{"input1":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":3341.6171875,"top":855.3828125},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"7.0\\", \\"height_output_dim\\": \\"7.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"3c8f4a21-e941-4e58-8d11-0f66c4a7d2c0","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"7051d948-3249-4f85-b772-6bae20c99f4d"}]}},"tags":[],"uuid":"40c0a173-1805-46c6-9005-5ead67bc9f48","version":11},"readme":"## Generate Nx and Size plot for multiple assemblies\\n\\n\\n### Inputs\\n\\nCollection of fasta files. The name of each item in the collection will be used as label for the Nx and Size plots.\\n\\n### Outputs\\n\\n\\n1. Nx plot \\n2. Size plot ","changelog":"# Changelog\\n\\n## [0.1.4] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.1.3] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n\\n## [0.1.2] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n\\n## [0.1.1] - 2024-06-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n\\n## [0.1] - 2024-05-11\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Generate Nx and Size plots for multiple assemblies\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Collection of genomes to plot\\"]@{ shape: docs }\\n1[\\"gfastats\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Sort\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Text reformatting\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Datamash\\"]@{ shape: process }\\n3 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Parse parameter value\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Compose text parameter value\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Compute\\"]@{ shape: process }\\n5 --> 8\\n7 --> 8\\n9[\\"Add input name as column\\"]@{ shape: process }\\n8 --> 9\\n10[\\"Collapse Collection\\"]@{ shape: process }\\n9 --> 10\\n11[\\"Cut\\"]@{ shape: process }\\n10 --> 11\\n12[\\"Cut\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Nx Plot\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Size Plot\\"]@{ shape: process }\\n12 --> 14\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Plot-Nx-Size/main","dockstore_id":27320,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078910","tests":[{"doc":"Test outline for Generate-Nx-and-Size-plots-for-multiple-assemblies","job":{"Collection of genomes to plot":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"Primary","location":"https://zenodo.org/records/10047837/files/Hifiasm%20Primary%20assembly.fasta?download=1"},{"class":"File","identifier":"Alternate","location":"https://zenodo.org/records/10047837/files/Hifiasm%20Alternate%20assembly.fasta?download=1"}]}},"outputs":{"Size Plot":{"asserts":{"has_size":{"value":100000,"delta":2000}}},"Nx Plot":{"asserts":{"has_size":{"value":87000,"delta":2000}}},"gfastats data for plotting":{"element_tests":{"Primary":{"path":"test-data/primary_head.tabular","compare":"contains"},"Alternate":{"path":"test-data/alternate_head.tabular","compare":"contains"}}}}}]}],"path":"./workflows/VGP-assembly-v2/Plot-Nx-Size"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Purging-duplicates-one-haplotype-VGP6b.ga","testParameterFiles":["/Purging-duplicates-one-haplotype-VGP6b-tests.yml"],"authors":[{"name":"Galaxy"},{"name":"VGP","url":"https://vertebrategenomeproject.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"Purge duplicates from one haplotype. Prerequisites: run after a k-mer profiling workflow (VGP 1 or 2) and a contiging workflow (VGP 3,4 or 5).","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","release":"0.7.5","name":"Purging duplicates in one haplotype VGP6b","report":{"markdown":"\\n# Workflow Execution Report\\n\\nTime workflow was invoked\\n\\n```galaxy\\ninvocation_time()\\n```\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Merqury results\\n\\nMerqury QV:\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_QV\\")\\n```\\n\\nMerqury completeness:\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"merqury_stats\\")\\n```\\n\\nMerqury plots:\\n\\nspectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-cn.fl\\")\\n```\\n\\n\\nspectra-asm:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.spectra-asm.fl\\")\\n```\\n\\n\\nhap1 spectra-cn:\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_01.spectra-cn.fl\\")\\n```\\n\\n\\nhap2 spectra-cn:\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"output_merqury.assembly_02.spectra-cn.fl\\")\\n```\\n\\n\\n\\n\\n## BUSCO results (Vertebrata database)\\n\\nPurged Assembly\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Busco on Purged Primary assembly: summary image\\")\\n```\\n\\n\\n## Assembly statistics\\n\\n\\n```galaxy\\nhistory_dataset_as_table(output=\\"clean_stats\\")\\n```\\n\\n\\n## Nx and Size plots\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Nx Plot\\")\\n```\\n\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"Size Plot\\")\\n```\\n\\n\\n\\n## Current Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Model parameters obtained in the k-mer profiling workflow. ","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Model parameters obtained in the k-mer profiling workflow. ","name":"Genomescope model parameters"}],"label":"Genomescope model parameters","name":"Input dataset","outputs":[],"position":{"left":0,"top":473.0537740808015},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0c31ea5d-7cb6-4dff-89d3-7a1643ed74e6","when":null,"workflow_outputs":[]},"1":{"annotation":"Trimmed PacBio HiFi reads—outputs of cutadapt in the contiging workflow. ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Trimmed PacBio HiFi reads—outputs of cutadapt in the contiging workflow. ","name":"Pacbio Reads Collection - Trimmed"}],"label":"Pacbio Reads Collection - Trimmed","name":"Input dataset collection","outputs":[],"position":{"left":353.67187789473076,"top":855.1562350370721},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"5e2ecae6-c42f-4add-b863-e31e0536b8a6","when":null,"workflow_outputs":[]},"2":{"annotation":"Assembly containing duplications to be purged. ","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Assembly containing duplications to be purged. ","name":"Assembly to purge"}],"label":"Assembly to purge","name":"Input dataset","outputs":[],"position":{"left":457.72815230879326,"top":1233.737533865197},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"7460bcbb-c83a-4931-94b2-40e37c7baed0","when":null,"workflow_outputs":[]},"3":{"annotation":"Meryl database obtained in the k-mer profiling workflow. ","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Meryl database obtained in the k-mer profiling workflow. ","name":"Meryl Database"}],"label":"Meryl Database","name":"Input dataset","outputs":[],"position":{"left":1894.8311396080076,"top":708.3121334471537},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0f22d885-78ed-41a2-9dab-2e8576b33c38","when":null,"workflow_outputs":[]},"4":{"annotation":"Assembly that does not need purging. ","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Assembly that does not need purging. ","name":"Assembly to leave alone (For Merqury comparison)"}],"label":"Assembly to leave alone (For Merqury comparison)","name":"Input dataset","outputs":[],"position":{"left":1972.4566892558003,"top":828.0277755820543},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"86fed2c5-2b3d-4917-a376-f30d182a3967","when":null,"workflow_outputs":[]},"5":{"annotation":"Estimated genome file obtained in the contiging workflow. ","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Estimated genome file obtained in the contiging workflow. ","name":"Estimated genome size - Parameter File"}],"label":"Estimated genome size - Parameter File","name":"Input dataset","outputs":[],"position":{"left":1665.2040517264386,"top":1510.890391185622},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f19aebb2-3ee9-44dc-ad1e-af4aada02e31","when":null,"workflow_outputs":[]},"6":{"annotation":"Database to use for Busco lineages. ","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Database to use for Busco lineages. ","name":"Database for Busco Lineage"}],"label":"Database for Busco Lineage","name":"Input parameter","outputs":[],"position":{"left":1705.6584960887728,"top":1674.0155168754275},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1562a9b8-e3b5-473f-9f77-e09693d1a98f","when":null,"workflow_outputs":[]},"7":{"annotation":"Taxonomic lineage for the organism being assembled for Busco analysis","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Taxonomic lineage for the organism being assembled for Busco analysis","name":"Lineage"}],"label":"Lineage","name":"Input parameter","outputs":[],"position":{"left":1746.454413936029,"top":1810.0027445448727},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f95d611e-99bd-43a3-8455-b87df2071d01","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"","name":"Name of purged assembly"}],"label":"Name of purged assembly","name":"Input parameter","outputs":[],"position":{"left":2531.47238801942,"top":831.9098633921083},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"2bb6f78a-3a49-4cd4-87d1-29b7900b7950","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"","name":"Name of un-altered assembly"}],"label":"Name of un-altered assembly","name":"Input parameter","outputs":[],"position":{"left":2540.3923712516594,"top":949.2459649024566},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"c04ef45a-c035-4470-9fe1-87ded0e8bafb","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":10,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":279.14929817609004,"top":468.0364202234945},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"1.5*c3\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"3*c7\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"7d73837b-7c7d-4abf-a9ea-84ff6a1b637a","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":11,"input_connections":{"fastq_input|fastq_input1":{"id":1,"output_name":"output"},"reference_source|ref_file":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":828.456118617387,"top":821.0374911405877},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"},"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"asm5\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"b6259cc9-6c33-48ff-bc0f-c7fcd0643809","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":12,"input_connections":{"function_select|input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"split_fasta","type":"fasta"}],"position":{"left":835.7031135648554,"top":1143.001779839763},"post_job_actions":{"HideDatasetActionsplit_fasta":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"split_fasta"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"split_fa\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"bdf1cdab-99ae-4bae-a1e5-a88460469f20","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":13,"input_connections":{"input_file":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2261.4845426800953,"top":1078.9565991289805},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"2bde750d-f225-498f-bd3c-b954e71b5bdd","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":14,"input_connections":{"input1":{"id":5,"output_name":"output"}},"inputs":[],"label":"Estimated genome size","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1970.8727749962593,"top":1623.4892732441454},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"RenameDatasetActioninteger_param":{"action_arguments":{"newname":"Estimated Genome size"},"action_type":"RenameDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"3a1aa4eb-3f07-46fc-af93-ce8f08f5a8c6","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"Cut1","errors":null,"id":15,"input_connections":{"input":{"id":10,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":558.0468791193098,"top":468.0364202234945},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"36041c70-3530-400c-968a-f5237915dbc7","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"Cut1","errors":null,"id":16,"input_connections":{"input":{"id":10,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":558.0468791193098,"top":660.0676759469875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"95aeabf1-7bc7-421d-a50c-53b027534fe1","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":17,"input_connections":{"fastq_input|fastq_input1":{"id":12,"output_name":"split_fasta"},"reference_source|ref_file":{"id":12,"output_name":"split_fasta"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":1123.36328125,"top":1130.6857156904466},"post_job_actions":{"ChangeDatatypeActionalignment_output":{"action_arguments":{"newtype":"paf"},"action_type":"ChangeDatatypeAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": \\"1\\", \\"B\\": \\"19\\", \\"O\\": \\"39\\", \\"O2\\": \\"81\\", \\"E\\": \\"3\\", \\"E2\\": \\"1\\", \\"z\\": \\"200\\", \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": \\"self-homology\\"}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": \\"100\\", \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": \\"40\\", \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"6f009c42-6298-4db3-999e-b408c4ebedd6","when":null,"workflow_outputs":[]},"18":{"annotation":"","id":18,"input_connections":{"gfa_stats":{"id":13,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":2540.373300171866,"top":1078.9565991289805},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"64a6925e-020e-462a-8bc5-a473f276e2a1","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"007342c9-5000-43e7-a689-221a6a5c4228","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"b00c3f6d-621b-4831-9f78-cc1746d57876","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"22525cda-b084-4b03-84e1-12b720567921","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9a6ea45f-cd94-4597-ab9a-477a2af58fe3","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"fd56b131-b5b3-461b-9a4e-ae65bfa1add7","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"897d75f9-f1ab-4706-a46d-3e62935732a3","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"fb3f5607-de2c-4905-aaa6-998cec97dd50","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"21cc546e-1231-488d-a94e-626f86461888"}]}},"tags":[],"uuid":"d62effdb-fa4c-4047-946d-6a46555cefbf"},"tool_id":null,"type":"subworkflow","uuid":"7934a577-c91d-46f5-bddc-1c0d5dad173f","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":19,"input_connections":{"input_file":{"id":4,"output_name":"output"},"mode_condition|statistics_condition|expected_genomesize":{"id":14,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2263.41796875,"top":1423.7677469404466},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"a7237fd4-e2ee-4b33-bdb7-975e49ccc16c","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":20,"input_connections":{"input1":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":836.9531477685715,"top":437.0294656904467},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"TagDatasetActioncalcuts_cutoff":{"action_arguments":{"tags":"p1_calcuts_cutoffs"},"action_type":"TagDatasetAction","output_name":"calcuts_cutoff"},"TagDatasetActionhist":{"action_arguments":{"tags":"p1_calcuts_hist"},"action_type":"TagDatasetAction","output_name":"hist"},"TagDatasetActionpbcstat_cov":{"action_arguments":{"tags":"p1_calcuts_basecov"},"action_type":"TagDatasetAction","output_name":"pbcstat_cov"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"d4a4cf6e-361f-4b2a-b0e3-bdb1d7203b21","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":21,"input_connections":{"input1":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":836.9531477685715,"top":629.0086517382745},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"27915433-a961-4876-9bef-8fd4695f9e06","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":22,"input_connections":{"infile":{"id":19,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2540.373300171866,"top":1447.0555638451494},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\tAlternate\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"bb591597-add9-4957-85d7-a6f6983eaa8f","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":23,"input_connections":{"function_select|input":{"id":11,"output_name":"alignment_output"},"function_select|section_calcuts|transition":{"id":21,"output_name":"integer_param"},"function_select|section_calcuts|upper_depth":{"id":20,"output_name":"integer_param"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"stat_file","type":"tabular"},{"name":"pbcstat_cov","type":"tabular"},{"name":"pbcstat_wig","type":"wig"},{"name":"hist","type":"png"},{"name":"calcuts_log","type":"txt"},{"name":"calcuts_cutoff","type":"tabular"}],"position":{"left":1125.9375626626565,"top":443.99128200472524},"post_job_actions":{"TagDatasetActioncalcuts_cutoff":{"action_arguments":{"tags":"calcuts_cutoffs, #purged"},"action_type":"TagDatasetAction","output_name":"calcuts_cutoff"},"TagDatasetActioncalcuts_log":{"action_arguments":{"tags":"calcuts_log, #purged"},"action_type":"TagDatasetAction","output_name":"calcuts_log"},"TagDatasetActionhist":{"action_arguments":{"tags":"calcuts_hist, #purged"},"action_type":"TagDatasetAction","output_name":"hist"},"TagDatasetActionpbcstat_cov":{"action_arguments":{"tags":"calcuts_basecov, #purged"},"action_type":"TagDatasetAction","output_name":"pbcstat_cov"},"TagDatasetActionpbcstat_wig":{"action_arguments":{"tags":"calcuts_basecov_wig, #purged"},"action_type":"TagDatasetAction","output_name":"pbcstat_wig"},"TagDatasetActionstat_file":{"action_arguments":{"tags":"calcuts_depths, #purged"},"action_type":"TagDatasetAction","output_name":"stat_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"pbcstat\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pbcstat_options\\": {\\"max_cov\\": \\"500\\", \\"min_map_ratio\\": \\"0.0\\", \\"min_map_qual\\": null, \\"flank\\": \\"0\\", \\"primary_alignments\\": true}, \\"section_calcuts\\": {\\"min_depth\\": \\"0.1\\", \\"low_depth\\": \\"1\\", \\"transition\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"upper_depth\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ploidy\\": \\"-d 0\\"}, \\"section_hist\\": {\\"ymin\\": null, \\"ymax\\": null, \\"xmin\\": null, \\"xmax\\": null, \\"title\\": \\"Read depth histogram plot\\"}, \\"output_options\\": [\\"pbcstat_coverage\\", \\"pbcstat_wig\\", \\"depth_stats\\", \\"histogram\\", \\"calcuts_cutoff\\", \\"calcuts_log\\"]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"15d14426-9cad-4735-a093-dd8401d1f36a","when":null,"workflow_outputs":[{"label":"Cutoffs","output_name":"calcuts_cutoff","uuid":"0c375bcb-c80a-489e-b4fc-98a7c99813d7"},{"label":"Read Coverage and cutoffs calculation: Histogram plot","output_name":"hist","uuid":"dff0569f-d3c4-457c-842e-e62511ea754d"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":24,"input_connections":{"function_select|coverage":{"id":23,"output_name":"pbcstat_cov"},"function_select|cutoffs":{"id":23,"output_name":"calcuts_cutoff"},"function_select|input":{"id":17,"output_name":"alignment_output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"purge_dups_log","type":"txt"},{"name":"purge_dups_bed","type":"bed"}],"position":{"left":1414.6961876965902,"top":1011.031271707235},"post_job_actions":{"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"},"TagDatasetActionpurge_dups_log":{"action_arguments":{"tags":"purgeoverlap_log"},"action_type":"TagDatasetAction","output_name":"purge_dups_log"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"purge_dups\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"coverage\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"cutoffs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_bad\\": \\"0.8\\", \\"min_align\\": \\"70\\", \\"min_match\\": \\"200\\", \\"min_chain\\": \\"500\\", \\"max_gap\\": \\"20000\\", \\"double_chain\\": {\\"chaining_rounds\\": \\"one\\", \\"__current_case__\\": 1}, \\"min_chain_score\\": \\"10000\\", \\"max_extend\\": \\"15000\\", \\"log_file\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"053d0a4c-b7eb-426c-8d3f-e6dccb680fcc","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":25,"input_connections":{"infile":{"id":24,"output_name":"purge_dups_bed"}},"inputs":[],"label":"Remove REPEATs from BED","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1693.6113702942387,"top":1098.0017249661748},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"purge_overlaps_norepeats_bed"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"REPEAT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"6bc68c65-e806-40f1-88e6-dc2f9ae3db87","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","errors":null,"id":26,"input_connections":{"function_select|bed_input":{"id":25,"output_name":"output"},"function_select|fasta_input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Purge overlaps","name":"function_select"},{"description":"runtime parameter for tool Purge overlaps","name":"function_select"}],"label":null,"name":"Purge overlaps","outputs":[{"name":"get_seqs_hap","type":"fasta"},{"name":"get_seqs_purged","type":"fasta"}],"position":{"left":1972.4566892558003,"top":1198.9132315142244},"post_job_actions":{"RenameDatasetActionget_seqs_hap":{"action_arguments":{"newname":"Removed haplotigs"},"action_type":"RenameDatasetAction","output_name":"get_seqs_hap"},"RenameDatasetActionget_seqs_purged":{"action_arguments":{"newname":"Purged assembly"},"action_type":"RenameDatasetAction","output_name":"get_seqs_purged"},"TagDatasetActionget_seqs_hap":{"action_arguments":{"tags":"removed_haplotigs"},"action_type":"TagDatasetAction","output_name":"get_seqs_hap"},"TagDatasetActionget_seqs_purged":{"action_arguments":{"tags":"purged_assembly_fasta"},"action_type":"TagDatasetAction","output_name":"get_seqs_purged"},"TagDatasetActionpurge_dups_bed":{"action_arguments":{"tags":"p1_purgeoverlap_bed"},"action_type":"TagDatasetAction","output_name":"purge_dups_bed"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.6+galaxy0","tool_shed_repository":{"changeset_revision":"e9bd16ba5ebd","name":"purge_dups","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"function_select\\": {\\"functions\\": \\"get_seqs\\", \\"__current_case__\\": 5, \\"fasta_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bed_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"advanced_options\\": {\\"coverage\\": false, \\"haplotigs\\": false, \\"length\\": \\"10000\\", \\"min_ratio\\": \\"0.05\\", \\"end_trim\\": true, \\"split\\": false, \\"min_gap\\": \\"10000\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.6+galaxy0","type":"tool","uuid":"266003a1-1786-43c7-b3a8-6af326c8c541","when":null,"workflow_outputs":[{"label":"Removed haplotigs","output_name":"get_seqs_hap","uuid":"0593d60a-a6ec-48c0-a1dc-c2a4bc5bd537"},{"label":"Purged assembly","output_name":"get_seqs_purged","uuid":"0fb19650-ae22-41da-8ccf-fc12842b8d6b"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","errors":null,"id":27,"input_connections":{"mode|assembly_options|assembly_01":{"id":26,"output_name":"get_seqs_purged"},"mode|assembly_options|assembly_02":{"id":4,"output_name":"output"},"mode|meryldb_F1":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Merqury","name":"mode"}],"label":null,"name":"Merqury","outputs":[{"name":"qv_files","type":"input"},{"name":"png_files","type":"input"},{"name":"stats_files","type":"input"},{"name":"hist_files","type":"input"},{"name":"log_file","type":"txt"}],"position":{"left":2270.374846462181,"top":399.10600822544444},"post_job_actions":{"HideDatasetActionlog_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"log_file"},"TagDatasetActionbed_files":{"action_arguments":{"tags":"merqury_p_bed, #p"},"action_type":"TagDatasetAction","output_name":"bed_files"},"TagDatasetActionhist_files":{"action_arguments":{"tags":"merqury_purged_hist,name:purged"},"action_type":"TagDatasetAction","output_name":"hist_files"},"TagDatasetActionpng_files":{"action_arguments":{"tags":"merqury_purged_png, #purged"},"action_type":"TagDatasetAction","output_name":"png_files"},"TagDatasetActionqv_files":{"action_arguments":{"tags":"merqury_purged_qv, #purged"},"action_type":"TagDatasetAction","output_name":"qv_files"},"TagDatasetActionsizes_files":{"action_arguments":{"tags":"merqury_p_sizes, #p"},"action_type":"TagDatasetAction","output_name":"sizes_files"},"TagDatasetActionstats_files":{"action_arguments":{"tags":"merqury_purged_stats, #purged"},"action_type":"TagDatasetAction","output_name":"stats_files"},"TagDatasetActionwig_files":{"action_arguments":{"tags":"merqury_p_wig, #p"},"action_type":"TagDatasetAction","output_name":"wig_files"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4","tool_shed_repository":{"changeset_revision":"eb7b00ec347d","name":"merqury","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"label\\": \\"output_merqury\\", \\"mode\\": {\\"options\\": \\"default\\", \\"__current_case__\\": 0, \\"meryldb_F1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_options\\": {\\"number_assemblies\\": \\"two\\", \\"__current_case__\\": 1, \\"assembly_01\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"assembly_02\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_add_headers\\": false, \\"output_selector\\": [\\"qv\\", \\"plots\\", \\"stats\\", \\"log\\", \\"hist\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3+galaxy4","type":"tool","uuid":"4c0ee7aa-643d-4583-b507-0def031e329b","when":null,"workflow_outputs":[{"label":"Merqury on Phased assemblies: Images","output_name":"png_files","uuid":"4b7c4a1c-42e6-4d63-999d-bdce891470cd"},{"label":"Merqury on Phased assemblies: stats","output_name":"stats_files","uuid":"c07cd28e-86b5-4e4f-87fa-bc2b3b65da33"},{"label":"qv_files","output_name":"qv_files","uuid":"9fb50959-0da7-4838-9f0e-9c945f04c1f6"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":28,"input_connections":{"input_file":{"id":26,"output_name":"get_seqs_purged"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2267.44140625,"top":1252.4864969404466},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_contigs_purged,  #purged"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"size\\", \\"__current_case__\\": 0, \\"out_size\\": \\"c\\"}, \\"locale\\": false, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"ce91b60e-4bcf-45ff-8334-329f885cc145","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":29,"input_connections":{"input":{"id":26,"output_name":"get_seqs_purged"},"lineage_conditional|cached_db":{"id":6,"output_name":"output"},"lineage|lineage_dataset":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"},{"description":"runtime parameter for tool Busco","name":"lineage_conditional"}],"label":null,"name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":2265.9340461481925,"top":1559.2451038580757},"post_job_actions":{"TagDatasetActionbusco_gff":{"action_arguments":{"tags":"busco_purged_gff,name:purged"},"action_type":"TagDatasetAction","output_name":"busco_gff"},"TagDatasetActionbusco_missing":{"action_arguments":{"tags":"busco_purged_missing, #purged"},"action_type":"TagDatasetAction","output_name":"busco_missing"},"TagDatasetActionbusco_sum":{"action_arguments":{"tags":"busco_purged_summ, #purged"},"action_type":"TagDatasetAction","output_name":"busco_sum"},"TagDatasetActionbusco_table":{"action_arguments":{"tags":"busco_purged_full, #purged"},"action_type":"TagDatasetAction","output_name":"busco_table"},"TagDatasetActionsummary_image":{"action_arguments":{"tags":"busco_purged_img, #purged"},"action_type":"TagDatasetAction","output_name":"summary_image"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"metaeuk\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"outputs\\": [\\"short_summary\\", \\"missing\\", \\"image\\", \\"gff\\"], \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"73b62ef3-0002-418e-9f10-22076c1e48f3","when":null,"workflow_outputs":[{"label":"Busco on Purged Primary assembly: summary image","output_name":"summary_image","uuid":"4545fb20-1b7c-4329-9a94-6087cd1beeea"},{"label":"Busco on Purged Primary assembly: short summary","output_name":"busco_sum","uuid":"b366e7f7-02f9-47e0-b2b3-7bfc3265951d"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":30,"input_connections":{"input_file":{"id":26,"output_name":"get_seqs_purged"}},"inputs":[{"description":"runtime parameter for tool gfastats","name":"mode_condition"}],"label":"Convert purged fasta to gfa","name":"gfastats","outputs":[{"name":"output","type":"fastq"}],"position":{"left":2539.288422879884,"top":1674.902781383633},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Purged assembly (GFA)"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"purged_assembly_gfa"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"manipulation\\", \\"__current_case__\\": 0, \\"swiss_army_knife\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"output_condition\\": {\\"out_format\\": \\"gfa\\", \\"__current_case__\\": 4, \\"terminal_overlaps_condition\\": {\\"terminal_overlaps_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"terminal_overlaps_length\\": \\"1000\\"}, \\"no_sequence\\": false}, \\"discover_paths\\": false, \\"sort\\": \\"\\", \\"remove_terminal_gaps\\": false, \\"homopolymer_compress\\": null}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"90cfbc84-a100-4795-a376-f93ffe2dbd0b","when":null,"workflow_outputs":[{"label":"Purged assembly (GFA)","output_name":"output","uuid":"25213685-ed44-44a7-84fd-8d92f9d1d734"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","errors":null,"id":31,"input_connections":{"input_file":{"id":26,"output_name":"get_seqs_purged"},"mode_condition|statistics_condition|expected_genomesize":{"id":14,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"gfastats","outputs":[{"name":"stats","type":"tabular"}],"position":{"left":2267.472206693373,"top":2157.321113933695},"post_job_actions":{"TagDatasetActionstats":{"action_arguments":{"tags":"gfastats_purged"},"action_type":"TagDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0","tool_shed_repository":{"changeset_revision":"50b6ec4074f4","name":"gfastats","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode_condition\\": {\\"selector\\": \\"statistics\\", \\"__current_case__\\": 1, \\"statistics_condition\\": {\\"selector\\": \\"assembly\\", \\"__current_case__\\": 2, \\"expected_genomesize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"locale\\": true, \\"tabular\\": true, \\"discover_paths\\": false}, \\"target_condition\\": {\\"target_option\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.3.10+galaxy0","type":"tool","uuid":"ff2b02d1-f6dc-4cd6-bd4a-a70e4d97492c","when":null,"workflow_outputs":[{"label":"Purged assembly statistics","output_name":"stats","uuid":"c98fd03e-3a3d-472b-8b13-f13479fc4dad"}]},"32":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":32,"input_connections":{"input":{"id":27,"output_name":"qv_files"}},"inputs":[],"label":"merqury_QV","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2550.2959755790407,"top":0},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"2056c7f8-a087-4008-a166-e55bbfcb4035","when":null,"workflow_outputs":[{"label":"merqury_QV","output_name":"output","uuid":"958c426c-368c-4b7a-bd99-424bca35507a"}]},"33":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":33,"input_connections":{"input":{"id":27,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2634.5836490420215,"top":129.38725910990257},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"png"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"03bc5d8c-8465-447a-bd9b-ea5163549a9a","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-cn.fl","output_name":"output","uuid":"fe3e5228-3805-4941-9063-25f3828742e8"}]},"34":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":34,"input_connections":{"input":{"id":27,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.spectra-asm.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2665.2365619096245,"top":304.8565690227396},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"png"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.spectra-asm.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"2eda2fbc-3e64-4a3d-ab3a-82b5f2f62519","when":null,"workflow_outputs":[{"label":"output_merqury.spectra-asm.fl","output_name":"output","uuid":"003e44c6-ae7c-40cd-94a0-1a467fab9183"}]},"35":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":35,"input_connections":{"input":{"id":27,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_01.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2670.829560068264,"top":464.1789679160868},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"png"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_01.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"3d7d4de8-b260-4139-a6f4-058094cb37d4","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_01.spectra-cn.fl","output_name":"output","uuid":"e8f58510-9603-4995-b37a-89d8f3392a15"}]},"36":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":36,"input_connections":{"input":{"id":27,"output_name":"stats_files"}},"inputs":[],"label":"merqury_stats","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2790.1622429860763,"top":667.0034491512803},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"first\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"7b4b00d4-2c85-4e05-8063-744b17147303","when":null,"workflow_outputs":[{"label":"merqury_stats","output_name":"output","uuid":"f05c41e7-96cf-4933-bb13-71ec5108e8a0"}]},"37":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":37,"input_connections":{"input":{"id":27,"output_name":"png_files"}},"inputs":[],"label":"output_merqury.assembly_02.spectra-cn.fl","name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":2920.3828341709313,"top":460.05753014885033},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"png"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"by_identifier\\", \\"__current_case__\\": 1, \\"identifier\\": \\"output_merqury.assembly_02.spectra-cn.fl\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"47d7ca2f-a60f-4867-96c1-bcf9220da8bb","when":null,"workflow_outputs":[{"label":"output_merqury.assembly_02.spectra-cn.fl","output_name":"output","uuid":"5557187a-4905-449b-b3e7-d1a9202c9b25"}]},"38":{"annotation":"","id":38,"input_connections":{"gfa_stats":{"id":28,"input_subworkflow_step_id":0,"output_name":"stats"}},"inputs":[],"label":null,"name":"gfastats_data_prep","outputs":[],"position":{"left":2540.373300171866,"top":1250.0416485279309},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"format-version":"0.1","name":"gfastats_data_prep","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"gfa_stats"}],"label":"gfa_stats","name":"Input dataset","outputs":[],"position":{"left":0,"top":189.90056800842285},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"39a418ab-2c85-4de4-b800-27e5fe9fd734","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"sort1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.6136245727539,"top":0},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"sort1","tool_state":"{\\"column\\": \\"2\\", \\"column_set\\": [], \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"order\\": \\"DESC\\", \\"style\\": \\"num\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"65c11379-6c6b-4197-8c80-b20f07f93235","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":2,"input_connections":{"infile":{"id":1,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":292.1306838989258,"top":235.1562442779541},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{total += $2; $3 = total}1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"9469cad4-1680-476a-95ff-0567e424364c","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":3,"input_connections":{"in_file":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":595.0994338989258,"top":116.0227222442627},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"absmax\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"bb39d613-0273-4eec-8903-3037d635df94","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"input":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":479.0482864379883,"top":456.17896461486816},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"yes\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"57a71d85-0956-4144-bee4-31fdb01278dd","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":693.4658889770508,"top":299.4318027496338},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"65492414-7486-402f-8dc5-222e7e39a74b","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":5,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":885.0994338989258,"top":493.36646461486816},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c3/\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"15d3a356-7e65-4d17-a0ce-1561408176d5","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":7,"input_connections":{"input":{"id":4,"output_name":"out_file1"},"ops|expressions_0|cond":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1115.0993728637695,"top":735.5255222320557},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"gfastats data for plotting"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 1, \\"cond\\": \\"c2/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}, {\\"__index__\\": 2, \\"cond\\": \\"c3/1000000\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"8d82104b-65ff-4b66-8df7-09405f70e645","when":null,"workflow_outputs":[{"label":"gfastats data for plotting","output_name":"out_file1","uuid":"4a66487b-c3bd-4b9f-a72a-961f0aa0e55b"}]}},"tags":[],"uuid":"732b249a-84e9-4add-9fca-37f7c68f8115"},"tool_id":null,"type":"subworkflow","uuid":"76b01d81-8ba2-41d7-9701-ec77a2b21eb1","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","errors":null,"id":39,"input_connections":{"infile":{"id":31,"output_name":"stats"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2540.373300171866,"top":2034.0086581408366},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"BEGIN{print \\\\\\"Metric\\\\\\\\tPrimary\\\\\\"}; {print}; \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"7c1dd273-10e0-402f-978f-d9df50a5b236","when":null,"workflow_outputs":[]},"40":{"annotation":"","id":40,"input_connections":{"Alternate data":{"id":18,"input_subworkflow_step_id":1,"output_name":"gfastats data for plotting"},"Name of alternate assembly":{"id":9,"input_subworkflow_step_id":3,"output_name":"output"},"Name of primary assembly":{"id":8,"input_subworkflow_step_id":2,"output_name":"output"},"Primary data":{"id":38,"input_subworkflow_step_id":0,"output_name":"gfastats data for plotting"}},"inputs":[],"label":null,"name":"gfastats_plot","outputs":[],"position":{"left":2839.1842345748423,"top":1158.019113211751},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"Galaxy"},{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"}],"format-version":"0.1","license":"CC-BY-4.0","name":"gfastats_plot","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Primary data"}],"label":"Primary data","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"921f1cc2-a25b-4473-bf57-7fa4c6847727","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Alternate data"}],"label":"Alternate data","name":"Input dataset","outputs":[],"position":{"left":6.960205078125,"top":143.53692626953125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"cc90ff07-9a98-4662-80e3-c5cfe35c9498","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Name of primary assembly"}],"label":"Name of primary assembly","name":"Input parameter","outputs":[],"position":{"left":4.55963134765625,"top":296.3210144042969},"tool_id":null,"tool_state":"{\\"default\\": \\"Primary\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"76f3bd32-bf3e-42be-9e79-cee9e401a28b","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"53a7e6a5-2fac-4b10-a78b-023a98d5d761"}]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"Name of alternate assembly"}],"label":"Name of alternate assembly","name":"Input parameter","outputs":[],"position":{"left":7.428955078125,"top":440.4403076171875},"tool_id":null,"tool_state":"{\\"default\\": \\"Alternate\\", \\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"cd232b7f-e618-4548-a1da-55e5be6b3f32","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"76c4def3-9a34-4a6a-8d7d-a374ed695769"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":4,"input_connections":{"exp":{"id":2,"output_name":"output"},"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1139.7584686279297,"top":107.42897033691406},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"cb69df8a-6a6e-4e46-ad3d-83789fd7694e","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","errors":null,"id":5,"input_connections":{"exp":{"id":3,"output_name":"output"},"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Add column","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1151.7044982910156,"top":392.65625},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1","tool_shed_repository":{"changeset_revision":"023f0a3760b3","name":"add_value","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exp\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterate\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"baab034b-2407-4b3a-8c2a-e64022b5bfcd","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":6,"input_connections":{"inputs":{"id":4,"output_name":"out_file1"},"queries_0|inputs2":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1441.193115234375,"top":216.59091186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"8018cb28-c340-4b60-ab26-3e51d24ecc3f","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1780.3265991210938,"top":182.99716186523438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c5,c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"dd1ba6ee-c28a-49f5-ba28-4e25d5edb2dd","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1781.349365234375,"top":387.96875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c8,c4,c7\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"7303b6e1-ea08-40d4-910c-ed3392821c11","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Nx Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2068.1390991210938,"top":31.988632202148438},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Nx Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#nx_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"x\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Nx (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"b0f9006c-44b6-4da5-9a08-a569cf14876c","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"output1","uuid":"d5ed977a-0c77-4e09-87ea-73b3115a5ebc"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"Size Plot","name":"Scatterplot with ggplot2","outputs":[{"name":"output1","type":"png"}],"position":{"left":2083.3805541992188,"top":319.8011169433594},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Size Plot"},"action_type":"RenameDatasetAction","output_name":"output1"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"#size_plot"},"action_type":"TagDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1","tool_shed_repository":{"changeset_revision":"5298e06f14f9","name":"ggplot2_point","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"type_conditional\\": {\\"type_options\\": \\"lines\\", \\"__current_case__\\": 2}, \\"factor\\": {\\"factoring\\": \\"Single\\", \\"__current_case__\\": 1, \\"factorcol\\": \\"1\\", \\"colors\\": \\"Set1\\", \\"colororder\\": \\"1\\"}, \\"axis_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"axis_text_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"plot_title_customization\\": {\\"axis_customization\\": \\"default\\", \\"__current_case__\\": 0}, \\"gridlinecust\\": \\"default\\", \\"transform\\": \\"none\\", \\"scaling\\": {\\"plot_scaling\\": \\"Automatic\\", \\"__current_case__\\": 0}, \\"theme\\": \\"bw\\", \\"legend\\": \\"yes\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out\\": {\\"unit_output_dim\\": \\"in\\", \\"width_output_dim\\": \\"6.0\\", \\"height_output_dim\\": \\"4.0\\", \\"dpi_output_dim\\": \\"300.0\\", \\"additional_output_format\\": \\"none\\"}, \\"title\\": \\"\\", \\"xlab\\": \\"Scaffold number\\", \\"xplot\\": \\"2\\", \\"ylab\\": \\"Cumulative Size (Mb)\\", \\"yplot\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy1","type":"tool","uuid":"feb2b91a-edcb-4b1d-b3ec-eb2e7939a1dd","when":null,"workflow_outputs":[{"label":"Size Plot","output_name":"output1","uuid":"936fed44-f7a8-4973-8baf-9ba9ea7d23da"}]}},"tags":[],"uuid":"f5ca6830-3fb9-47b3-8203-0f16d9d406bd"},"tool_id":null,"type":"subworkflow","uuid":"1a299ea2-ec8f-4730-993c-028c1c47ecc9","when":null,"workflow_outputs":[{"label":"Nx Plot","output_name":"Nx Plot","uuid":"f0d394c3-27d8-4286-a6ea-6c3f1b6fed81"},{"label":"Size Plot","output_name":"Size Plot","uuid":"17b84f94-3fd6-4722-ac27-17998f5da324"}]},"41":{"annotation":"","content_id":"join1","errors":null,"id":41,"input_connections":{"input1":{"id":39,"output_name":"outfile"},"input2":{"id":22,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2877.4741567062574,"top":1682.1529233499416},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"fill_empty\\", \\"__current_case__\\": 1, \\"fill_columns_by\\": \\"fill_unjoined_only\\", \\"do_fill_empty_columns\\": {\\"column_fill_type\\": \\"single_fill_value\\", \\"__current_case__\\": 0, \\"fill_value\\": \\".\\"}}, \\"header\\": \\"\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"a36b6561-88d8-4a2f-8e48-56fcbc0a676d","when":null,"workflow_outputs":[]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":42,"input_connections":{"input":{"id":41,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":3173.52289710744,"top":1656.5699296579871},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Assembly statistics for both assemblies"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"gfastats_p1_p2"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": [\\"3\\"]}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"f004c161-83ca-4bce-8589-36da5fb1eb9a","when":null,"workflow_outputs":[{"label":"Assembly statistics for both assemblies","output_name":"output","uuid":"d8c0c665-710f-44ce-a54f-17d585335d4b"}]},"43":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","errors":null,"id":43,"input_connections":{"infile":{"id":42,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3561.341720003155,"top":1664.190126043544},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#\\", \\"replace_pattern\\": \\"Number of\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"ccba9fad-60fc-47fa-b56b-f24cb1b4ebde","when":null,"workflow_outputs":[{"label":"clean_stats","output_name":"outfile","uuid":"95622471-0d6f-4d6d-91a6-f010a4b6d0c0"}]}},"tags":["VGP_curated"],"uuid":"b4825ad9-fd20-4e21-856b-0b49a67a75f5","version":4},"readme":"# Purge Duplicate Contigs\\n\\nPurge contigs marked as duplicates by purge_dups in a single haplotype(could be haplotypic duplication or overlap duplication)\\nThis workflow is the 6th workflow of the VGP pipeline. It is meant to be run after one of the contigging steps (Workflow 3, 4, or 5)\\n\\n## Inputs\\n\\n1. Genomescope model parameters [txt] (Generated by the k-mer profiling workflow)\\n2. Hifi long reads - trimmed [fastq] (Generated by Cutadapt in the contigging workflow)\\n3. Assembly to purge (e.g. hap1) [fasta] (Generated by the contigging workflow)\\n4. K-mer database [meryldb]  (Generated by the k-mer profiling workflow)\\n5. Assembly to leave alone (used for merqury statistics) (e.g. hap2) [fasta] (Generated by the contigging workflow)\\n6. Estimated Genome Size [txt]\\n7. Database for busco lineage (recommended: latest)\\n8. Busco lineage (recommended: vertebrata)\\n9. Name of un-altered assembly\\n10. Name of purged assembly\\n\\n\\n## Outputs\\n\\n1. Haplotype 1 purged assembly (Fasta and gfa)\\n2. Haplotype 2 purged assembly (Fasta and gfa)\\n3. QC: BUSCO report for both assemblies\\n4. QC: Merqury report for both assemblies\\n5. QC: Assembly statistics for both assemblies\\n6. QC: Nx plot for both assemblies\\n7. QC: Size plot for both assemblies","changelog":"# Changelog\\n\\n## [0.7.5] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.10+galaxy0`\\n\\n## [0.7.4] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.5+galaxy0`\\n\\n## [0.7.3] - 2025-02-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy2`\\n\\n## [0.7.2] - 2025-01-30\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n\\n## [0.7.1] - 2024-08-13\\n\\n- Expose Busco parameters : \\n  - Lineage database \\n  - Lineage\\n\\n## [0.7] - 2024-07-24\\n\\n### Added\\n\\n- Workflow report\\n\\n### Changed\\n\\n- Fix bug where the columns of the assembly statistics didn\'t join properly.\\n- Rename some outputs for clarity\\n\\n## [0.6] - 2024-04-23\\n\\n### Changed\\n\\n- Add Merqury histogram output\\n- Add Busco gff output\\n- Unhide Merqury QV output\\n\\n## [0.5] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0`\\n\\n## [0.4] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.27+galaxy0`\\n\\n## [0.2] - 2024-02-19\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.24+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/add_value/addValue/1.0.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/3.4.0+galaxy1`\\n\\n## [0.1] - 2024-02-07\\n\\n### Added\\n\\n- Workflow for purging duplication in contigs in single haplotype\\n","diagrams":"# Workflow diagrams\\n\\n## Purging duplicates in one haplotype VGP6b\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Genomescope model parameters\\"]@{ shape: doc }\\n1[\\"ℹ️ Pacbio Reads Collection - Trimmed\\"]@{ shape: docs }\\n2[\\"ℹ️ Assembly to purge\\"]@{ shape: doc }\\n3[\\"ℹ️ Meryl Database\\"]@{ shape: doc }\\n4[\\"ℹ️ Assembly to leave alone (For Merqury comparison)\\"]@{ shape: doc }\\n5[\\"ℹ️ Estimated genome size - Parameter File\\"]@{ shape: doc }\\n6[\\"ℹ️ Database for Busco Lineage\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Lineage\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Name of purged assembly\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Name of un-altered assembly\\"]@{ shape: lean-l }\\n10[\\"Compute\\"]@{ shape: process }\\n0 --> 10\\n11[\\"Map with minimap2\\"]@{ shape: process }\\n1 --> 11\\n2 --> 11\\n12[\\"Purge overlaps\\"]@{ shape: process }\\n2 --> 12\\n13[\\"gfastats\\"]@{ shape: process }\\n4 --> 13\\n14[\\"Estimated genome size\\"]@{ shape: process }\\n5 --> 14\\n15[\\"Cut\\"]@{ shape: process }\\n10 --> 15\\n16[\\"Cut\\"]@{ shape: process }\\n10 --> 16\\n17[\\"Map with minimap2\\"]@{ shape: process }\\n12 --> 17\\n12 --> 17\\n18[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n13 --> 18\\n19[\\"gfastats\\"]@{ shape: process }\\n4 --> 19\\n14 --> 19\\n20[\\"Parse parameter value\\"]@{ shape: process }\\n15 --> 20\\n21[\\"Parse parameter value\\"]@{ shape: process }\\n16 --> 21\\n22[\\"Text reformatting\\"]@{ shape: process }\\n19 --> 22\\n23[\\"Purge overlaps\\"]@{ shape: process }\\n11 --> 23\\n21 --> 23\\n20 --> 23\\n24[\\"Purge overlaps\\"]@{ shape: process }\\n23 --> 24\\n23 --> 24\\n17 --> 24\\n25[\\"Remove REPEATs from BED\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Purge overlaps\\"]@{ shape: process }\\n25 --> 26\\n2 --> 26\\n27[\\"Merqury\\"]@{ shape: process }\\n26 --> 27\\n4 --> 27\\n3 --> 27\\n28[\\"gfastats\\"]@{ shape: process }\\n26 --> 28\\n29[\\"Busco\\"]@{ shape: process }\\n26 --> 29\\n6 --> 29\\n7 --> 29\\n30[\\"Convert purged fasta to gfa\\"]@{ shape: process }\\n26 --> 30\\n31[\\"gfastats\\"]@{ shape: process }\\n26 --> 31\\n14 --> 31\\n32[\\"merqury_QV\\"]@{ shape: process }\\n27 --> 32\\n33[\\"output_merqury.spectra-cn.fl\\"]@{ shape: process }\\n27 --> 33\\n34[\\"output_merqury.spectra-asm.fl\\"]@{ shape: process }\\n27 --> 34\\n35[\\"output_merqury.assembly_01.spectra-cn.fl\\"]@{ shape: process }\\n27 --> 35\\n36[\\"merqury_stats\\"]@{ shape: process }\\n27 --> 36\\n37[\\"output_merqury.assembly_02.spectra-cn.fl\\"]@{ shape: process }\\n27 --> 37\\n38[\\"🛠️ gfastats_data_prep\\"]@{ shape: subprocess }\\n28 --> 38\\n39[\\"Text reformatting\\"]@{ shape: process }\\n31 --> 39\\n40[\\"🛠️ gfastats_plot\\"]@{ shape: subprocess }\\n18 --> 40\\n9 --> 40\\n8 --> 40\\n38 --> 40\\n41[\\"Join two Datasets\\"]@{ shape: process }\\n39 --> 41\\n22 --> 41\\n42[\\"Advanced Cut\\"]@{ shape: process }\\n41 --> 42\\n43[\\"Replace\\"]@{ shape: process }\\n42 --> 43\\n```\\n\\n## gfastats_plot\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Primary data\\"]@{ shape: doc }\\n1[\\"ℹ️ Alternate data\\"]@{ shape: doc }\\n2[\\"ℹ️ Name of primary assembly\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Name of alternate assembly\\"]@{ shape: lean-l }\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n0 --> 4\\n5[\\"Add column\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"Concatenate datasets\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Cut\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Nx Plot\\"]@{ shape: process }\\n7 --> 9\\n10[\\"Size Plot\\"]@{ shape: process }\\n8 --> 10\\n```\\n\\n## gfastats_data_prep\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ gfa_stats\\"]@{ shape: doc }\\n1[\\"Sort\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Text reformatting\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Datamash\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Add column\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Compute\\"]@{ shape: process }\\n4 --> 7\\n6 --> 7\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Purge-duplicates-one-haplotype-VGP6b/main","dockstore_id":25505,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078866","tests":[{"doc":"Test outline for Purging-duplicates-one-haplotype-VGP6b","job":{"Genomescope model parameters":{"class":"File","location":"https://zenodo.org/records/10632760/files/Genomescope%20model%20parameters.tabular?download=1","filetype":"tabular"},"Assembly to purge":{"class":"File","location":"https://zenodo.org/records/10632760/files/hap1.fasta?download=1","filetype":"fasta"},"Meryl Database":{"class":"File","location":"https://zenodo.org/records/10632760/files/Meryl%20Database.meryldb?download=1","filetype":"meryldb"},"Estimated genome size - Parameter File":{"class":"File","location":"https://zenodo.org/records/10632760/files/Estimated%20genome%20size%20-%20Parameter%20File.expression.json?download=1","filetype":"expression.json"},"Assembly to leave alone (For Merqury comparison)":{"class":"File","location":"https://zenodo.org/records/10632760/files/hap2.fasta?download=1","filetype":"fasta"},"Pacbio Reads Collection - Trimmed":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"yeast_reads_sub1.fastq.gz","location":"https://zenodo.org/records/10632760/files/Trimmed_yeast_reads_sub1.fastq.gz?download=1"}]},"Name of un-altered assembly":"Hap2","Name of purged assembly":"Hap1","Database for Busco Lineage":"v5","Lineage":"vertebrata_odb10"},"outputs":{"Removed haplotigs":{"asserts":{"has_n_lines":{"n":14}}},"Purged assembly":{"asserts":{"has_n_lines":{"n":158}}},"Purged assembly (GFA)":{"asserts":{"has_n_lines":{"n":160}}},"Assembly statistics for both assemblies":{"asserts":{"has_text":{"text":"# scaffolds\\t79\\t79"}}},"Cutoffs":{"asserts":{"has_text":{"text":"48"}}},"Busco on Purged Primary assembly: short summary":{"asserts":{"has_text":{"text":"C:1.1%[S:1.1%,D:0.0%],F:0.5%,M:98.4%,n:3354"}}},"Purged assembly statistics":{"asserts":{"has_text":{"text":"# scaffolds\\t79"}}},"Nx Plot":{"asserts":{"has_size":{"value":57000,"delta":5000}}},"Size Plot":{"asserts":{"has_size":{"value":84000,"delta":5000}}},"Merqury on Phased assemblies: stats":{"element_tests":{"output_merqury.completeness":{"asserts":{"has_text":{"text":"95.7636"}}}}}}}]}],"path":"./workflows/VGP-assembly-v2/Purge-duplicates-one-haplotype-VGP6b"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Mitogenome-Assembly-VGP0.ga","testParameterFiles":["/Mitogenome-Assembly-VGP0-tests.yml"],"authors":[{"name":"VGP","url":"https://vertebrategenomeproject.org"},{"name":"Galaxy"}],"definition":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Organization","name":"VGP","url":"https://vertebrategenomeproject.org"},{"class":"Organization","name":"Galaxy"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Mitogenome-Assembly-VGP0","release":"0.1","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Collection of Pacbio Data"}],"label":"Collection of Pacbio Data","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"861c3a49-1055-4030-9a91-e53cbf1ac436","when":null,"workflow_outputs":[]},"1":{"annotation":"Latin name of species","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Latin name of species","name":"Species name (latin name)"}],"label":"Species name (latin name)","name":"Input parameter","outputs":[],"position":{"left":131.640625,"top":108.3515625},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"27257feb-04b0-48f9-8bd3-a3a74536eab2","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"65196658-7776-4165-944b-c7876483edf4"}]},"2":{"annotation":"Required for NCBI database query","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Required for NCBI database query","name":"Email adress"}],"label":"Email adress","name":"Input parameter","outputs":[],"position":{"left":200.78125,"top":290.66015625},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"4ddb385f-ecd7-4d48-bc7b-e9c7971a1e62","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"28bbbdb1-62b7-4d3a-9076-09cc835464cc"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mitohifi/mitohifi/3+galaxy0","errors":null,"id":3,"input_connections":{"operation_mode|email":{"id":2,"output_name":"output"},"operation_mode|species":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MitoHiFi","name":"operation_mode"},{"description":"runtime parameter for tool MitoHiFi","name":"operation_mode"}],"label":null,"name":"MitoHiFi","outputs":[{"name":"fasta_reference","type":"fasta"},{"name":"genbank_reference","type":"genbank"}],"position":{"left":463.26953125,"top":242.234375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mitohifi/mitohifi/3+galaxy0","tool_shed_repository":{"changeset_revision":"99ddbf037d98","name":"mitohifi","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation_mode\\": {\\"command\\": \\"find_reference\\", \\"__current_case__\\": 0, \\"species\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"email\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_length\\": \\"15000\\", \\"exact_species\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3+galaxy0","type":"tool","uuid":"e59aea2d-9bc3-4e2e-9af2-711f0cdea373","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mitohifi/mitohifi/3+galaxy0","errors":null,"id":4,"input_connections":{"operation_mode|input_option|input_reads":{"id":0,"output_name":"output"},"operation_mode|reference_fasta":{"id":3,"output_name":"fasta_reference"},"operation_mode|reference_genbank":{"id":3,"output_name":"genbank_reference"}},"inputs":[{"description":"runtime parameter for tool MitoHiFi","name":"operation_mode"},{"description":"runtime parameter for tool MitoHiFi","name":"operation_mode"}],"label":null,"name":"MitoHiFi","outputs":[{"name":"mitogenome_fasta","type":"fasta"},{"name":"mitogenome_genbank","type":"genbank"},{"name":"mitogenome_annotation","type":"png"},{"name":"mitogenome_coverage","type":"png"},{"name":"contigs_stats","type":"tsv"},{"name":"hifiasm","type":"fasta"},{"name":"hifiasm_filtered","type":"fasta"},{"name":"hifiasm_contigs","type":"fasta"}],"position":{"left":767.6689707083681,"top":134.09873055610626},"post_job_actions":{"HideDatasetActionhifiasm":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"hifiasm"},"HideDatasetActionhifiasm_contigs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"hifiasm_contigs"},"HideDatasetActionhifiasm_filtered":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"hifiasm_filtered"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/mitohifi/mitohifi/3+galaxy0","tool_shed_repository":{"changeset_revision":"99ddbf037d98","name":"mitohifi","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"operation_mode\\": {\\"command\\": \\"mitohifi\\", \\"__current_case__\\": 1, \\"input_option\\": {\\"input\\": \\"pacbio\\", \\"__current_case__\\": 0, \\"input_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bloom_filter\\": \\"0\\"}, \\"reference_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_genbank\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"genetic_code\\": \\"2\\", \\"advanced_options\\": {\\"query_blast\\": \\"70\\", \\"circular_size\\": null, \\"circular_offset\\": null, \\"outputs\\": null}, \\"output_zip\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3+galaxy0","type":"tool","uuid":"ccd83410-608a-4b9d-a43a-11cecd6c2d42","when":null,"workflow_outputs":[{"label":"mitogenome_annotation","output_name":"mitogenome_annotation","uuid":"85061673-df5e-480d-920a-a11c72bf8f56"},{"label":"mitogenome_genbank","output_name":"mitogenome_genbank","uuid":"fe4d1d3d-9caa-45f6-944e-c4b101105dd1"},{"label":"contigs_stats","output_name":"contigs_stats","uuid":"31807e4b-e616-4742-92e0-934e6c3c09ae"},{"label":"mitogenome_coverage","output_name":"mitogenome_coverage","uuid":"8393c665-a328-4698-93e5-198c6fea52dd"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compress_file/compress_file/0.1.0","errors":null,"id":5,"input_connections":{"input":{"id":4,"output_name":"mitogenome_fasta"}},"inputs":[{"description":"runtime parameter for tool Compress file(s)","name":"input"}],"label":null,"name":"Compress file(s)","outputs":[{"name":"output_file","type":"gz"}],"position":{"left":1065.5131364872464,"top":265.5261773629027},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compress_file/compress_file/0.1.0","tool_shed_repository":{"changeset_revision":"09ea79f9f260","name":"compress_file","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"5207adc9-f36a-42e9-8631-e01da77ac446","when":null,"workflow_outputs":[{"label":"output_file","output_name":"output_file","uuid":"6f918a08-1ca1-4d6e-a4d8-7e65e13a65ed"}]}},"tags":["Reviewed","VGP"],"uuid":"57920bf9-1741-4170-badf-39fdfcdaa2e9","version":4},"readme":"# Assembly of Mitochondrial DNA from PacBio HiFi reads\\n\\nGenerate mitochondrial assembly based on PacBio Hifi Reads.\\n\\n## Inputs\\n\\n1. Hifi long reads [fastq]\\n2. Latin name of a related Species\\n3. Email adress required for NCBI database query \\n\\n## Outputs\\n\\n1. Contigs Statistics\\n2. Images : \\n   1. Mitogenome Coverage\\n   2. Mitogenome Annotation\\n3. Genbank file of the assembled mitogenome\\n4. Fasta file of the assembled mitogenome","changelog":"# Changelog\\n\\n\\n## [0.1] - 2024-01-03\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Mitogenome-Assembly-VGP0\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Collection of Pacbio Data\\"]@{ shape: docs }\\n1[\\"ℹ️ Species name (latin name)\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Email adress\\"]@{ shape: lean-l }\\n3[\\"MitoHiFi\\"]@{ shape: process }\\n2 --> 3\\n1 --> 3\\n4[\\"MitoHiFi\\"]@{ shape: process }\\n0 --> 4\\n3 --> 4\\n3 --> 4\\n5[\\"Compress file(s)\\"]@{ shape: process }\\n4 --> 5\\n```\\n","updated":"2024-01-03T00:00:00","trsID":"#workflow/github.com/iwc-workflows/Mitogenome-assembly-VGP0/main","dockstore_id":25318,"categories":[],"collections":["Vertebrates Genome Project"],"doi":"10.5281/zenodo.15078848","tests":[{"doc":"Test outline for Mitogenome-Assembly-VGP0","job":{"Collection of Pacbio Data":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"pacbio_01.fasta.gz","location":"https://zenodo.org/records/10454765/files/pacbio_01.fasta.gz?download=1"}]},"Species name (latin name)":"Theretra latreillii lucasii","Email adress":"iwc@galaxyproject.org"},"outputs":{"contigs_stats":{"asserts":{"has_text":{"text":"15316\\t36\\tTrue"}}},"mitogenome_coverage":{"asserts":{"has_size":{"value":19000,"delta":2000}}},"mitogenome_annotation":{"asserts":{"has_size":{"value":68000,"delta":5000}}},"mitogenome_genbank":{"asserts":{"has_n_lines":{"n":480}}}}}]}],"path":"./workflows/VGP-assembly-v2/Mitogenome-assembly-VGP0"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/segmentation-and-counting.ga","testParameterFiles":["/segmentation-and-counting-tests.yml"],"authors":[{"name":"Leonid Kostrykin","orcid":"0000-0003-1323-3762","url":"https://github.com/kostrykin/"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0003-1323-3762","name":"Leonid Kostrykin","url":"https://github.com/kostrykin/"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"Segmentation and counting of cell nuclei in fluorescence microscopy images","steps":{"0":{"annotation":"The fluorescence microscopy images to be segmented. Must be the single image channel, which contains the cell nuclei.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"The fluorescence microscopy images to be segmented. Must be the single image channel, which contains the cell nuclei.","name":"input_image"}],"label":"input_image","name":"Input dataset","outputs":[],"position":{"left":0,"top":123.6936084329719},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"facec436-5d6e-43de-a21e-cebc8272a35c","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_simple_filter/ip_filter_standard/1.12.0+galaxy1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter 2-D image","outputs":[{"name":"output","type":"tiff"}],"position":{"left":320.6808544907898,"top":244.80388558482915},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_simple_filter/ip_filter_standard/1.12.0+galaxy1","tool_shed_repository":{"changeset_revision":"d51310ab328a","name":"2d_simple_filter","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tiff\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"filter\\": {\\"filter_type\\": \\"gaussian\\", \\"__current_case__\\": 0, \\"size\\": \\"3.0\\"}, \\"filter_type\\": \\"gaussian\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"radius\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.12.0+galaxy1","type":"tool","uuid":"0c652b2d-7376-4444-b4c4-70babd1a0f14","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_histogram_equalization/ip_histogram_equalization/0.18.1+galaxy0","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Perform histogram equalization","outputs":[{"name":"output","type":"tiff"}],"position":{"left":438.81238880324617,"top":9.67052833486391},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_histogram_equalization/ip_histogram_equalization/0.18.1+galaxy0","tool_shed_repository":{"changeset_revision":"b1c2c210813c","name":"2d_histogram_equalization","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tiff\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"h_type\\": \\"clahe\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.18.1+galaxy0","type":"tool","uuid":"c4b63332-c0ed-4a40-9670-9d5ba2a5e88f","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_auto_threshold/ip_threshold/0.18.1+galaxy3","errors":null,"id":3,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Threshold image","outputs":[{"name":"output","type":"tiff"}],"position":{"left":565.4531507696091,"top":243.376959571321},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/2d_auto_threshold/ip_threshold/0.18.1+galaxy3","tool_shed_repository":{"changeset_revision":"699a5e9146b3","name":"2d_auto_threshold","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"block_size\\": \\"5\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"dark_bg\\": \\"true\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert_output\\": false, \\"th_method\\": {\\"method_id\\": \\"otsu\\", \\"__current_case__\\": 1, \\"threshold\\": \\"0\\", \\"block_size\\": \\"0\\", \\"offset\\": \\"0.0\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.18.1+galaxy3","type":"tool","uuid":"d75a2027-d855-4f05-9261-25f2f712efce","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/bfconvert/ip_convertimage/6.7.0+galaxy3","errors":null,"id":4,"input_connections":{"input_file":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Convert image format","outputs":[{"name":"output","type":"tiff"}],"position":{"left":689.7987516288889,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/bfconvert/ip_convertimage/6.7.0+galaxy3","tool_shed_repository":{"changeset_revision":"fcadded98e61","name":"bfconvert","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tiff\\", \\"bigtiff\\": false, \\"channel_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"compression\\": \\"False\\", \\"crop_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"noflat\\": true, \\"out_format\\": \\"png\\", \\"pyramid_options\\": {\\"generate\\": \\"False\\", \\"__current_case__\\": 1}, \\"range_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"series_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"tile_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"timepoint_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"z_options\\": {\\"extract\\": \\"False\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"6.7.0+galaxy3","type":"tool","uuid":"a6a689c6-c1d0-4929-87a2-ac5206cf8026","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/binary2labelimage/ip_binary_to_labelimage/0.5+galaxy0","errors":null,"id":5,"input_connections":{"input":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Convert binary image to label map","outputs":[{"name":"output","type":"tiff"}],"position":{"left":812.10278473575,"top":245.5611328413049},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/binary2labelimage/ip_binary_to_labelimage/0.5+galaxy0","tool_shed_repository":{"changeset_revision":"984358e43242","name":"binary2labelimage","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tiff\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"mode_selector\\": \\"cca\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5+galaxy0","type":"tool","uuid":"cd899592-8b30-4a79-9aba-f75a10c75dd7","when":null,"workflow_outputs":[{"label":"label_image","output_name":"output","uuid":"daaee143-81e0-4067-aac0-81f02732c343"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/overlay_images/ip_overlay_images/0.0.4+galaxy4","errors":null,"id":6,"input_connections":{"method_option|im1":{"id":4,"output_name":"output"},"method_option|im2":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Overlay images","name":"method_option"},{"description":"runtime parameter for tool Overlay images","name":"method_option"}],"label":null,"name":"Overlay images","outputs":[{"name":"out_png","type":"png"}],"position":{"left":1085.3470167429655,"top":104.58436042741572},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/overlay_images/ip_overlay_images/0.0.4+galaxy4","tool_shed_repository":{"changeset_revision":"ca362a9bfa20","name":"overlay_images","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"method_option\\": {\\"method\\": \\"seg_contour\\", \\"__current_case__\\": 2, \\"im1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"im2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"thickness\\": \\"2\\", \\"color\\": \\"#ff0000\\", \\"show_label\\": true, \\"label_color\\": \\"#ffff00\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.4+galaxy4","type":"tool","uuid":"066a4e0d-2ae3-46f7-a0c8-77a1a83d119a","when":null,"workflow_outputs":[{"label":"overlay_image","output_name":"out_png","uuid":"5b94aaa3-9034-465e-995f-3d8f1f057f4b"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/imgteam/count_objects/ip_count_objects/0.0.5-2","errors":null,"id":7,"input_connections":{"input":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Count objects in label map","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1086.8749834881503,"top":370.25000974083935},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/imgteam/count_objects/ip_count_objects/0.0.5-2","tool_shed_repository":{"changeset_revision":"b58447a2eed2","name":"count_objects","owner":"imgteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tiff\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.5-2","type":"tool","uuid":"75a4d4d8-0435-441f-a309-d04071cf125f","when":null,"workflow_outputs":[{"label":"objects_count","output_name":"output","uuid":"8b3d51a7-2c1f-4f84-ac80-ef0ef6d25a3f"}]}},"tags":[],"uuid":"8c2d765b-2764-4b66-8542-adc2109c24f2","version":10},"readme":"# Segmentation and counting of cell nuclei in fluorescence microscopy images\\n\\nThis workflow performs segmentation and counting of cell nuclei using fluorescence microscopy images. The segmentation step is performed using Otsu thresholding (Otsu, 1979). The workflow is based on the tutorial: https://training.galaxyproject.org/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html\\n\\n![](test-data/overlay_image.png)\\n\\n## Inputs\\n\\n**`input_image`:** The fluorescence microscopy images to be segmented. Must be the single image channel, which contains the cell nuclei.\\n\\n## Outputs\\n\\n**`overlay_image`:** An overlay of the original image and the outlines of the segmentated objects, each also annotated with a unique number.\\n\\n**`objects_count`:** Table with a single column `objects` and a single row (the actual number of objects).\\n\\n**`label_image`:** The segmentation result (label map, which contains a unique label for each segmented object).\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-11-07\\n\\n- Upgrade the tests of the workflow\\n- Upgrade the segmentation-and-counting.ga workflow to recent tool versions\\n\\n## [0.1] - 2024-02-29\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Segmentation and counting of cell nuclei in fluorescence microscopy images\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ input_image\\"]@{ shape: doc }\\n1[\\"Filter 2-D image\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Perform histogram equalization\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Threshold image\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Convert image format\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Convert binary image to label map\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Overlay images\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n7[\\"Count objects in label map\\"]@{ shape: process }\\n5 --> 7\\n```\\n","updated":"2024-11-07T00:00:00","trsID":"#workflow/github.com/iwc-workflows/fluorescence-nuclei-segmentation-and-counting/main","dockstore_id":25723,"categories":[],"collections":["Imaging"],"doi":"10.5281/zenodo.15078515","tests":[{"doc":"Test outline for segmentation-and-counting.ga","job":{"input_image":{"class":"File","path":"test-data/input_image.tiff","filetype":"tiff"}},"outputs":{"overlay_image":{"path":"test-data/overlay_image.png","compare":"image_diff"},"objects_count":{"path":"test-data/objects_count.tabular","compare":"diff","lines_diff":0},"label_image":{"path":"test-data/label_image.tiff","compare":"image_diff","metric":"iou"}}}]}],"path":"./workflows/imaging/fluorescence-nuclei-segmentation-and-counting"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/atacseq.ga","testParameterFiles":["/atacseq-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete ATAC-seq analysis pipeline for paired-end reads. Processes raw FASTQ data through adapter and bad quality removal (cutadapt), alignment (Bowtie2 end-to-end), and filtering (removes MT reads, discordant pairs, low mapping quality <30, PCR duplicates). Generates 5\' cut site pileups (±100bp), performs peak calling, and quantifies reads in 1kb summit-centered regions. Produces two normalized coverage tracks (per million mapped reads and per million reads in peaks) and fragment length distribution plots for quality assessment.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"1.0","name":"ATAC-seq Analysis: Chromatin Accessibility Profiling","steps":{"0":{"annotation":"Should be a paired collection with ATAC-seq fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with ATAC-seq fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":579.4833374023438},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"3b1d869d-e8a9-49d9-894c-8aa500cd445d","when":null,"workflow_outputs":[]},"1":{"annotation":"reference_genome","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"reference_genome","name":"reference_genome"}],"label":"reference_genome","name":"Input parameter","outputs":[],"position":{"left":30.550048828125,"top":684.38330078125},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c085055c-9d18-49d7-9f87-fae61683d775","when":null,"workflow_outputs":[]},"2":{"annotation":"Used by macs2:\\\\nH. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Used by macs2:\\\\nH. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":98,"top":787.9833374023438},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"43f5c231-ed56-4cda-bd50-0495dfd77d70","when":null,"workflow_outputs":[]},"3":{"annotation":"Bin size for normalized bigwig (usually 50bp is sufficient)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Bin size for normalized bigwig (usually 50bp is sufficient)","name":"bin_size"}],"label":"bin_size","name":"Input parameter","outputs":[],"position":{"left":155.54247082439377,"top":898.8496207394044},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"942c77e6-f462-49ac-87ed-6e4e14589b0b","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1","errors":null,"id":4,"input_connections":{"library|input_1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":"Cutadapt (remove adapter + bad quality bases)","name":"Cutadapt","outputs":[{"name":"out_pairs","type":"input"},{"name":"report","type":"txt"}],"position":{"left":278.16668701171875,"top":492.36669921875},"post_job_actions":{"HideDatasetActionout_pairs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_pairs"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"},"RenameDatasetActionreport":{"action_arguments":{"newname":"cutadapt report"},"action_type":"RenameDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1","tool_shed_repository":{"changeset_revision":"5eb7e84243f2","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"15\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Nextera R1\\", \\"adapter\\": \\"CTGTCTCTTATACACATCTCCGAGCCCACGAGAC\\"}, \\"single_noindels\\": false}], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}, \\"r2\\": {\\"adapters2\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Nextera R2\\", \\"adapter\\": \\"CTGTCTCTTATACACATCTGACGCTGCCGACGA\\"}, \\"single_noindels\\": false}], \\"front_adapters2\\": [], \\"anywhere_adapters2\\": []}, \\"pair_adapters\\": false}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"30\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.9+galaxy1","type":"tool","uuid":"33fa2759-9f3f-431b-b35c-b5c777d5d5b7","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","errors":null,"id":5,"input_connections":{"library|input_1":{"id":4,"output_name":"out_pairs"},"reference_genome|index":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bowtie2","name":"library"},{"description":"runtime parameter for tool Bowtie2","name":"reference_genome"}],"label":"Bowtie2 map on reference","name":"Bowtie2","outputs":[{"name":"output","type":"bam"},{"name":"mapping_stats","type":"txt"}],"position":{"left":555.13330078125,"top":395.6499938964844},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionmapping_stats":{"action_arguments":{"newname":"mapping stats"},"action_type":"RenameDatasetAction","output_name":"mapping_stats"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"bowtie2 output (BAM)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","tool_shed_repository":{"changeset_revision":"d5ceb9f3c25b","name":"bowtie2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"simple\\", \\"__current_case__\\": 0, \\"presets\\": \\"--very-sensitive\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"unaligned_file\\": false, \\"aligned_file\\": false, \\"paired_options\\": {\\"paired_options_selector\\": \\"yes\\", \\"__current_case__\\": 0, \\"I\\": \\"0\\", \\"X\\": \\"1000\\", \\"fr_rf_ff\\": \\"--fr\\", \\"no_mixed\\": false, \\"no_discordant\\": false, \\"dovetail\\": true, \\"no_contain\\": false, \\"no_overlap\\": false}}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"sam_options\\": {\\"sam_options_selector\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_mapping_stats\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.3+galaxy1","type":"tool","uuid":"c32a3847-f673-487f-af98-8d50999f2d21","when":null,"workflow_outputs":[{"label":"mapping stats","output_name":"mapping_stats","uuid":"0a898754-6a14-4282-9269-02f5d516e49f"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","errors":null,"id":6,"input_connections":{"input_bam":{"id":5,"output_name":"output"}},"inputs":[],"label":"filter MAPQ30 concordant pairs and not mitochondrial pairs","name":"Filter BAM","outputs":[{"name":"out_file2","type":"txt"},{"name":"out_file1","type":"bam"}],"position":{"left":826.183349609375,"top":318.58331298828125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"HideDatasetActionout_file2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file2"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"filtered BAM"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","tool_shed_repository":{"changeset_revision":"993b19f20c76","name":"bamtools_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"conditions\\": [{\\"__index__\\": 0, \\"filters\\": [{\\"__index__\\": 0, \\"bam_property\\": {\\"bam_property_selector\\": \\"mapQuality\\", \\"__current_case__\\": 14, \\"bam_property_value\\": \\">=30\\"}}, {\\"__index__\\": 1, \\"bam_property\\": {\\"bam_property_selector\\": \\"isProperPair\\", \\"__current_case__\\": 11, \\"bam_property_value\\": true}}, {\\"__index__\\": 2, \\"bam_property\\": {\\"bam_property_selector\\": \\"reference\\", \\"__current_case__\\": 20, \\"bam_property_value\\": \\"!chrM\\"}}, {\\"__index__\\": 3, \\"bam_property\\": {\\"bam_property_selector\\": \\"mateReference\\", \\"__current_case__\\": 16, \\"bam_property_value\\": \\"!MT\\"}}]}], \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rule_configuration\\": {\\"rules_selector\\": \\"true\\", \\"__current_case__\\": 1, \\"rules\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.2+galaxy2","type":"tool","uuid":"eba4c413-30c9-4bab-a088-dc12d5956c91","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","errors":null,"id":7,"input_connections":{"input":{"id":5,"output_name":"output"}},"inputs":[],"label":"Get number of reads per chromosome","name":"Samtools idxstats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":945.2166748046875,"top":539.63330078125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","tool_shed_repository":{"changeset_revision":"fa5d3e61e429","name":"samtools_idxstats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"024f7338-2712-4fb1-a913-dfd5bc2d4f1e","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","errors":null,"id":8,"input_connections":{"inputFile":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":"remove PCR duplicates","name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":1163.1000061035156,"top":211.4666748046875},"post_job_actions":{"RenameDatasetActionmetrics_file":{"action_arguments":{"newname":"MarkDuplicates metrics"},"action_type":"RenameDatasetAction","output_name":"metrics_file"},"RenameDatasetActionoutFile":{"action_arguments":{"newname":"BAM filtered rmDup"},"action_type":"RenameDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": true, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"d0a9998c-18a9-4579-8d33-f0897e1ceaeb","when":null,"workflow_outputs":[{"label":"MarkDuplicates metrics","output_name":"metrics_file","uuid":"144745aa-8898-45e6-a328-53e2518202b9"},{"label":"BAM filtered rmDup","output_name":"outFile","uuid":"5f4d5cc5-c2f2-40f3-b941-df5dd7347d25"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":9,"input_connections":{"infile":{"id":7,"output_name":"output"}},"inputs":[],"label":"reads in chrM/MT for multiQC","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1190.1499938964844,"top":618.5833129882812},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\" ($1==\\\\\\"chrM\\\\\\" || $1 == \\\\\\"MT\\\\\\"){print $1, $3}($1!=\\\\\\"chrM\\\\\\" && $1!=\\\\\\"MT\\\\\\"){SUM+=$3}END{print \\\\\\"others\\\\\\", SUM} \\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"9ae38aa6-511d-4ab4-b055-7a1dae65649f","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","errors":null,"id":10,"input_connections":{"input":{"id":8,"output_name":"outFile"}},"inputs":[],"label":"convert BAM to BED to improve peak calling","name":"bedtools BAM to BED","outputs":[{"name":"output","type":"bed"}],"position":{"left":1458.7166595458984,"top":193.58331298828125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"BED filtered rmDup"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"ed_score\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"option\\": \\"\\", \\"split\\": false, \\"tag\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"12ca18f2-222b-4e9c-8ae0-b5772cd51bd7","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1","errors":null,"id":11,"input_connections":{"input_bam":{"id":8,"output_name":"outFile"}},"inputs":[],"label":"Compute fragment length histogram","name":"Paired-end histogram","outputs":[{"name":"output1","type":"png"},{"name":"output2","type":"tabular"}],"position":{"left":1478.1666717529297,"top":559.88330078125},"post_job_actions":{"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"RenameDatasetActionoutput1":{"action_arguments":{"newname":"histogram of fragment length"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1","tool_shed_repository":{"changeset_revision":"bd1416eea1f0","name":"pe_histogram","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lower_limit\\": null, \\"upper_limit\\": \\"500\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"cc486ad2-0f48-4636-98b5-ea5fad9b75b3","when":null,"workflow_outputs":[{"label":"histogram of fragment length","output_name":"output1","uuid":"06437eef-c6fa-49f1-81fb-c48a21b87f57"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":12,"input_connections":{"input":{"id":8,"output_name":"outFile"}},"inputs":[],"label":"number of reads","name":"Samtools view","outputs":[{"name":"outputcnt","type":"tabular"}],"position":{"left":1960.1157004761121,"top":545.8157628477595},"post_job_actions":{"HideDatasetActionoutputcnt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcnt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"all_reads\\", \\"__current_case__\\": 0, \\"output_options\\": {\\"reads_report_type\\": \\"count\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"2999f836-c74f-47ed-b1ad-81f43349fe11","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":13,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":10,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"Call Peak with MACS2","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_summits","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"},{"name":"output_extra_files","type":"html"}],"position":{"left":1741.2333374023438,"top":0},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_extra_files":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_extra_files"},"HideDatasetActionoutput_summits":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_summits"},"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_control_lambda":{"action_arguments":{"newname":"MACS2 control coverage"},"action_type":"RenameDatasetAction","output_name":"output_control_lambda"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"MACS2 narrowPeak"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_summits":{"action_arguments":{"newname":"MACS2 summits"},"action_type":"RenameDatasetAction","output_name":"output_summits"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"MACS2 peaks xls"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"MACS2 treatment coverage"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": false, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"all\\", \\"__current_case__\\": 2}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BED\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"-100\\"}, \\"outputs\\": [\\"peaks_tabular\\", \\"summits\\", \\"bdg\\", \\"html\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"57386f5d-96bb-4684-97dc-3b2228189c01","when":null,"workflow_outputs":[{"label":"MACS2 narrowPeak","output_name":"output_narrowpeaks","uuid":"02d92607-381e-41b0-b0ed-b0f89bc02967"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","errors":null,"id":14,"input_connections":{"infile":{"id":11,"output_name":"output2"}},"inputs":[],"label":"remove comments lines","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1780,"top":701.4833374023438},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"^#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"c06960f8-9d4e-483e-8f07-12976f8b802a","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":15,"input_connections":{"input":{"id":12,"output_name":"outputcnt"}},"inputs":[],"label":"compute 1/million reads","name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1681.741700059967,"top":1407.1751610966817},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": true, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"1000000/c1\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"ec62cddf-fb04-4185-b9f4-709de2ce0202","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":16,"input_connections":{"input1":{"id":13,"output_name":"output_treat_pileup"}},"inputs":[],"label":"Bigwig from MACS2 (no norm)","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1444.2833251953125,"top":1510.0499877929688},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Coverage from MACS2 (bigwig)"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"4eea632c-ec31-43e7-88d2-79895b764a06","when":null,"workflow_outputs":[{"label":"Coverage from MACS2 (bigwig)","output_name":"out_file1","uuid":"5ffc5c02-8669-49b2-82a1-682ec3122d39"}]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_slopbed/2.31.1+galaxy0","errors":null,"id":17,"input_connections":{"genome_file_opts|genome":{"id":1,"output_name":"output"},"inputA":{"id":13,"output_name":"output_summits"}},"inputs":[{"description":"runtime parameter for tool bedtools SlopBed","name":"genome_file_opts"}],"label":"get summits +/-500kb","name":"bedtools SlopBed","outputs":[{"name":"output","type":"bed"}],"position":{"left":2186.5,"top":206.29998779296875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"1kb around each summit"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_slopbed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addition\\": {\\"addition_select\\": \\"b\\", \\"__current_case__\\": 0, \\"b\\": \\"500\\"}, \\"genome_file_opts\\": {\\"genome_file_opts_selector\\": \\"loc\\", \\"__current_case__\\": 0, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"header\\": false, \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pct\\": false, \\"strand\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"d1c672ca-ee55-4e48-9d97-75552050534a","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","errors":null,"id":18,"input_connections":{"infile":{"id":13,"output_name":"output_tabular"}},"inputs":[],"label":"summary of MACS2","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":2198.433349609375,"top":79.64999389648438},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"MACS2 report"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"^#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"a12ee458-2fe0-4d1e-8df9-c8fc76d7b277","when":null,"workflow_outputs":[{"label":"MACS2 report","output_name":"output","uuid":"2c50d266-785f-4a9d-ba31-f9fe66b703c7"}]},"19":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":19,"input_connections":{"input1":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":"Convert 1/million reads to parameter","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":1927.9064759856594,"top":1388.3825581400483},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"7afe5b18-2555-4f24-a1e2-348341631008","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":20,"input_connections":{"input":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":"Isolate each bigwig do normalize not average","name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":2134.4445922218906,"top":1595.328936224611},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [0, 1], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"05ca058f-88e4-4e7c-9e93-66e25043883d","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.31.1","errors":null,"id":21,"input_connections":{"input":{"id":17,"output_name":"output"}},"inputs":[],"label":"Merge summits +/-500kb","name":"bedtools MergeBED","outputs":[{"name":"output","type":"bed"}],"position":{"left":2501.5999755859375,"top":273.75},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Summits +-500bp (merged)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"c_and_o_argument_repeat\\": [], \\"distance\\": \\"0\\", \\"header\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strand\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"a74b6905-51b8-4c39-bc8c-3267b1f0a7a5","when":null,"workflow_outputs":[{"label":"1kb around summits","output_name":"output","uuid":"94021253-18e5-4969-9dec-ec901bcc9a3b"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":22,"input_connections":{"advancedOpt|binSize":{"id":3,"output_name":"output"},"advancedOpt|scaleFactors":{"id":19,"output_name":"text_param"},"bigwigs":{"id":20,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"normalize by million reads","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":2394.696221186954,"top":1367.5056983565069},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"bigwig normalized per million reads"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"fa2633cf-af38-43eb-8c9f-b8b656d6a631","when":null,"workflow_outputs":[{"label":"bigwig_norm","output_name":"outFileName","uuid":"42fb26ba-7428-43cf-8668-e865b67c12bc"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.31.1+galaxy0","errors":null,"id":23,"input_connections":{"inputA":{"id":21,"output_name":"output"},"reduce_or_iterate|inputB":{"id":8,"output_name":"outFile"}},"inputs":[{"description":"runtime parameter for tool bedtools Compute both the depth and breadth of coverage","name":"reduce_or_iterate"}],"label":"Compute coverage on summits +/-500kb","name":"bedtools Compute both the depth and breadth of coverage","outputs":[{"name":"output","type":"bed"}],"position":{"left":2734.11669921875,"top":534.88330078125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"coverage on MACS2 peaks"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"a_or_b\\": false, \\"d\\": false, \\"hist\\": false, \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mean\\": false, \\"overlap_a\\": null, \\"overlap_b\\": null, \\"reciprocal_overlap\\": false, \\"reduce_or_iterate\\": {\\"reduce_or_iterate_selector\\": \\"iterate\\", \\"__current_case__\\": 0, \\"inputB\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"sorted\\": false, \\"split\\": false, \\"strandedness\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"f3681ac7-e1ba-4442-971f-8353389c5265","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":24,"input_connections":{"infile":{"id":23,"output_name":"output"}},"inputs":[],"label":"number of reads in peaks","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3097.2166748046875,"top":460.11669921875},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Nb of reads in summits +-500bp"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"{S=S+$4}END{print S}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"f65b3936-595c-4ddb-88bc-2cc41e514e38","when":null,"workflow_outputs":[{"label":"Nb of reads in summits +-500bp","output_name":"outfile","uuid":"ba96198b-0b36-4b3c-a7bf-664d765eff3a"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","errors":null,"id":25,"input_connections":{"input":{"id":24,"output_name":"outfile"}},"inputs":[],"label":"compute 1/million reads in peaks","name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1660.7476485833522,"top":1675.7739064754614},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1","tool_shed_repository":{"changeset_revision":"aff5135563c6","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": true, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"no\\", \\"__current_case__\\": 0, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"1000000/c1\\", \\"add_column\\": {\\"mode\\": \\"R\\", \\"__current_case__\\": 2, \\"pos\\": \\"1\\"}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1","type":"tool","uuid":"5f55e258-a4fe-4b51-ab9b-2a6f0f22d910","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"cat1","errors":null,"id":26,"input_connections":{"input1":{"id":24,"output_name":"outfile"},"queries_0|input2":{"id":12,"output_name":"outputcnt"}},"inputs":[],"label":"Combine number of reads in peaks with total number of reads","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":3026.25,"top":727.7666625976562},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"cat1","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"ae286273-3247-4a8f-b296-33a679fbaf7d","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":27,"input_connections":{"input1":{"id":25,"output_name":"out_file1"}},"inputs":[],"label":"Convert 1/million reads in peaks to parameter","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":1922.9030306495047,"top":1685.3491762772487},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"f0eb1323-041b-4e29-9f07-de15427c267d","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":28,"input_connections":{"infile":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":"reads in peaks multiQC","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2870.2166748046875,"top":944.800048828125},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\" NR==1{print \\\\\\"in peaks\\\\\\",$1;inp=$1}NR==2{print \\\\\\"outside peaks\\\\\\",$1 - inp}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"7fe2bb35-31b5-4da0-83cf-38bb4af2848e","when":null,"workflow_outputs":[]},"29":{"annotation":"Isolate each bigwig do normalize not average","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":29,"input_connections":{"advancedOpt|binSize":{"id":3,"output_name":"output"},"advancedOpt|scaleFactors":{"id":27,"output_name":"text_param"},"bigwigs":{"id":20,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"normalize by million reads in peaks","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":2402.456534423419,"top":1734.6798994525739},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"bigwig normalized per million reads in peaks"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"8eab7f49-605b-462a-9806-da0c795117f3","when":null,"workflow_outputs":[{"label":"bigwig_norm2","output_name":"outFileName","uuid":"60031333-7419-4121-a0c7-e235b70a5cfc"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0","errors":null,"id":30,"input_connections":{"results_0|software_cond|input":{"id":4,"output_name":"report"},"results_1|software_cond|input":{"id":5,"output_name":"mapping_stats"},"results_2|software_cond|input":{"id":9,"output_name":"outfile"},"results_3|software_cond|output_0|input":{"id":8,"output_name":"metrics_file"},"results_4|software_cond|input":{"id":14,"output_name":"output"},"results_5|software_cond|input":{"id":13,"output_name":"output_tabular"},"results_6|software_cond|input":{"id":28,"output_name":"outfile"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"plots","type":"input"},{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2612.75,"top":1012.6666259765625},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0","tool_shed_repository":{"changeset_revision":"f7e2f1eb3a16","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": true, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"bowtie2\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"custom_content\\", \\"__current_case__\\": 32, \\"plot_type\\": \\"bargraph\\", \\"section_name\\": \\"chrM\\", \\"title\\": \\"reads mapping on chrM\\", \\"description\\": \\"\\", \\"xlab\\": \\"\\", \\"ylab\\": \\"\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, {\\"__index__\\": 4, \\"software_cond\\": {\\"software\\": \\"custom_content\\", \\"__current_case__\\": 32, \\"plot_type\\": \\"linegraph\\", \\"section_name\\": \\"Fragment size\\", \\"title\\": \\"Fragment size distribution\\", \\"description\\": \\"\\", \\"xlab\\": \\"\\", \\"ylab\\": \\"\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 5, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 6, \\"software_cond\\": {\\"software\\": \\"custom_content\\", \\"__current_case__\\": 32, \\"plot_type\\": \\"bargraph\\", \\"section_name\\": \\"Reads in peaks\\", \\"title\\": \\"Number of reads in peaks\\", \\"description\\": \\"Number of reads falling 500bp from a summit\\", \\"xlab\\": \\"\\", \\"ylab\\": \\"\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.24.1+galaxy0","type":"tool","uuid":"112d720f-c747-4c92-985f-ebdb52086cc9","when":null,"workflow_outputs":[{"label":"MultiQC webpage","output_name":"html_report","uuid":"c22aafb2-d9f2-43c3-a6c4-cfe4a3166c07"},{"label":"MultiQC on input dataset(s): Stats","output_name":"stats","uuid":"6071150d-48db-4cf8-bfed-2cbdb2635856"}]}},"tags":["ATACseq"],"uuid":"747f2432-7493-44b8-a5ee-82d104407cad","version":1},"readme":"# ATAC-seq Analysis: Chromatin Accessibility Profiling\\n\\nThis workflow is highly concordant with the corresponding training material.\\nYou can have more information about ATAC-seq analysis in the [slides](https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/slides.html) and the [tutorial](https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html).\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of dataset pairs of fastqsanger.\\n\\n## Inputs values\\n\\n- reference_genome: this field will be adapted to the genomes available for bowtie2 and the genomes available for bedtools slopbed (dbkeys table)\\n- effective_genome_size: this is used by macs2 and may be entered manually (indications are provided for heavily used genomes)\\n- bin_size: this is used when normalization of coverage is performed. Large values will allow to have smaller output files but with less resolution while small values will increase computation time and size of output files to produce more resolutive bigwigs.\\n\\n## Processing\\n\\n- The workflow will remove nextera adapters and low quality bases and filter out any read smaller than 15bp.\\n- The filtered reads are mapped with bowtie2 allowing dovetail and fragment length up to 1kb.\\n- The BAM is filtered to keep only MAPQ30, concordant pairs and pairs outside of the mitochondria.\\n- The PCR duplicates are removed with Picard (only from version 0.8).\\n- The BAM is converted to BED to enable macs2 to take both pairs into account.\\n- The peaks are called with macs2 which at the same time generates a coverage file.\\n- The coverage file is converted to bigwig\\n- The amount of reads 500bp from summits and the total number of reads are computed.\\n- Two normalizations are computed:\\n  - By million reads\\n  - By million reads in peaks (500bp from summits)\\n- Other QC are performed:\\n  - A histogram with fragment length is computed.\\n  - The evaluation of percentage of reads to chrM or MT is computed.\\n- A multiQC is run to have an overview of the QC.\\n\\n### Warning\\n\\n- The `reference_genome` parameter value is used to select references in bowtie2 and bedtools slopbed. Only references that are present in bowtie2 **and** bedtools slopbed are selectable. If your favorite reference genome is not available ask your administrator to make sure that each bowtie2 reference has a corresponding len file for use in bedtools slopbed.\\n","changelog":"# Changelog\\n\\n## [1.0] - 2024-11-28\\n\\n### Manual update\\n\\nUse the options like written in the README: in all previous versions, bowtie2 was running with `--very-sensitive` option but without allowing dovetail and without fragment length up to 1kb. Bowtie2 has been adjusted and is now running with these options: allow dovetail and fragment length up to 1kb.\\n\\n## [0.17] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n### Manual update\\n- Add a step to remove comments lines from histogram to be compatible with new multiQC version\\n\\n## [0.16] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.15] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_slopbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_slopbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.31.1+galaxy0`\\n\\n## [0.14] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n## [0.13] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n\\n## [0.12] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n\\n## [0.11] - 2024-03-18\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2`\\n\\n## [0.10] - 2024-03-14\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.4` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.4` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n\\n## [0.9] - 2023-10-23\\n\\nFix the normalization factor. It was coverage per reads and per reads in peaks instead of per million reads and per million reads in peaks.\\n\\n## [0.8] - 2023-10-19\\n\\nFix the remove duplicate step!\\nIn all previous versions, due to an error, PCR duplicates were not removed.\\n\\n## [0.7] - 2023-10-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n\\n## [0.6] - 2023-09-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n\\n## [0.5.1] - 2023-09-22\\n\\nFix bug in normalize profiles when used with multiple samples (in 0.5.0 it is averaging samples instead of normalizing each sample).\\n\\n## [0.5] - 2023-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.3` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.4`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.30.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n\\n### Manual update\\n- add normalization steps for coverage\\n\\n## [0.4] - 2023-01-16\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1`\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0`\\n\\n## [0.1] - 2022-10-12\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## ATAC-seq Analysis: Chromatin Accessibility Profiling\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ reference_genome\\"]@{ shape: lean-l }\\n2[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n3[\\"ℹ️ bin_size\\"]@{ shape: lean-l }\\n4[\\"Cutadapt (remove adapter + bad quality bases)\\"]@{ shape: process }\\n0 --> 4\\n5[\\"Bowtie2 map on reference\\"]@{ shape: process }\\n4 --> 5\\n1 --> 5\\n6[\\"filter MAPQ30 concordant pairs and not mitochondrial pairs\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Get number of reads per chromosome\\"]@{ shape: process }\\n5 --> 7\\n8[\\"remove PCR duplicates\\"]@{ shape: process }\\n6 --> 8\\n9[\\"reads in chrM/MT for multiQC\\"]@{ shape: process }\\n7 --> 9\\n10[\\"convert BAM to BED to improve peak calling\\"]@{ shape: process }\\n8 --> 10\\n11[\\"Compute fragment length histogram\\"]@{ shape: process }\\n8 --> 11\\n12[\\"number of reads\\"]@{ shape: process }\\n8 --> 12\\n13[\\"Call Peak with MACS2\\"]@{ shape: process }\\n2 --> 13\\n10 --> 13\\n14[\\"remove comments lines\\"]@{ shape: process }\\n11 --> 14\\n15[\\"compute 1/million reads\\"]@{ shape: process }\\n12 --> 15\\n16[\\"Bigwig from MACS2 (no norm)\\"]@{ shape: process }\\n13 --> 16\\n17[\\"get summits +/-500kb\\"]@{ shape: process }\\n1 --> 17\\n13 --> 17\\n18[\\"summary of MACS2\\"]@{ shape: process }\\n13 --> 18\\n19[\\"Convert 1/million reads to parameter\\"]@{ shape: process }\\n15 --> 19\\n20[\\"Isolate each bigwig do normalize not average\\"]@{ shape: process }\\n16 --> 20\\n21[\\"Merge summits +/-500kb\\"]@{ shape: process }\\n17 --> 21\\n22[\\"normalize by million reads\\"]@{ shape: process }\\n3 --> 22\\n19 --> 22\\n20 --> 22\\n23[\\"Compute coverage on summits +/-500kb\\"]@{ shape: process }\\n21 --> 23\\n8 --> 23\\n24[\\"number of reads in peaks\\"]@{ shape: process }\\n23 --> 24\\n25[\\"compute 1/million reads in peaks\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Combine number of reads in peaks with total number of reads\\"]@{ shape: process }\\n24 --> 26\\n12 --> 26\\n27[\\"Convert 1/million reads in peaks to parameter\\"]@{ shape: process }\\n25 --> 27\\n28[\\"reads in peaks multiQC\\"]@{ shape: process }\\n26 --> 28\\n29[\\"normalize by million reads in peaks\\"]@{ shape: process }\\n3 --> 29\\n27 --> 29\\n20 --> 29\\n30[\\"MultiQC\\"]@{ shape: process }\\n4 --> 30\\n5 --> 30\\n9 --> 30\\n8 --> 30\\n14 --> 30\\n13 --> 30\\n28 --> 30\\n```\\n","updated":"2024-11-28T00:00:00","trsID":"#workflow/github.com/iwc-workflows/atacseq/main","dockstore_id":19836,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078779","tests":[{"doc":"Test outline for ATACseq","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"SRR891268_chr22_enriched","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R1.fastq.gz","filetype":"fastqsanger"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R2.fastq.gz","filetype":"fastqsanger"}]}]},"reference_genome":"hg19","effective_genome_size":2700000000,"bin_size":1000},"outputs":{"mapping stats":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":[{"that":"has_text","text":"7282 (2.59%) aligned concordantly 0 times"},{"that":"has_text","text":"121059 (43.09%) aligned concordantly exactly 1 time"},{"that":"has_text","text":"152623 (54.32%) aligned concordantly >1 times"}]}}},"MarkDuplicates metrics":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_text":{"text":"0.02"}}}}},"BAM filtered rmDup":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_size":{"value":15810403,"delta":1000000}}}}},"histogram of fragment length":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_size":{"value":47718,"delta":4000}}}}},"MACS2 narrowPeak":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_n_lines":{"n":236}}}}},"MACS2 report":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":[{"that":"has_text","text":"# tag size is determined as 47 bps"},{"that":"has_text","text":"# total tags in treatment: 262080"}]}}},"Coverage from MACS2 (bigwig)":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_size":{"value":2892925,"delta":200000}}}}},"1kb around summits":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_n_lines":{"n":217}}}}},"Nb of reads in summits +-500bp":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_line":{"line":"9548"}}}}},"bigwig_norm":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_size":{"value":1253177,"delta":100000}}}}},"bigwig_norm2":{"element_tests":{"SRR891268_chr22_enriched":{"asserts":{"has_size":{"value":1248419,"delta":100000}}}}},"MultiQC on input dataset(s): Stats":{"asserts":{"has_line":{"line":"Sample\\tMACS2_mqc_generalstats_macs2_d\\tMACS2_mqc_generalstats_macs2_peak_count\\tPicard: Mark Duplicates_mqc_generalstats_picard_mark_duplicates_PERCENT_DUPLICATION\\tBowtie 2 / HiSAT2_mqc_generalstats_bowtie_2_hisat2_overall_alignment_rate\\tCutadapt_mqc_generalstats_cutadapt_percent_trimmed"},"has_text_matching":{"expression":"SRR891268_chr22_enriched\\t200.0\\t236\\t2.7[0-9]*\\t98.[0-9]*\\t4.7[0-9]*"}}},"MultiQC webpage":{"asserts":[{"that":"has_text","text":"<a href=\\"#cutadapt_filtered_reads\\" class=\\"nav-l2\\">Filtered Reads</a>"},{"that":"has_text","text":"<td>% Aligned</td>"},{"that":"has_text","text":"<td>% BP Trimmed</td>"}]}}}]}],"path":"./workflows/epigenetics/atacseq"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/chipseq-sr.ga","testParameterFiles":["/chipseq-sr-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete ChIP-seq analysis for single-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and quality filtering (MAPQ >= 30). Peak calling with MACS2 uses either a fixed extension parameter or built-in model to identify protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.14","name":"ChIP-seq Analysis: Single-End Read Processing","steps":{"0":{"annotation":"Should be a collection with ChIPseq fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a collection with ChIPseq fastqs","name":"SR fastq input"}],"label":"SR fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"e09c0852-1db3-4a68-b88c-1b94c205cb6c","when":null,"workflow_outputs":[]},"1":{"annotation":"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC ","name":"adapter_forward"}],"label":"adapter_forward","name":"Input parameter","outputs":[],"position":{"left":39.0333251953125,"top":77.44999694824219},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"30c1d867-5e73-4348-8969-848f58d94015","when":null,"workflow_outputs":[]},"2":{"annotation":"reference_genome","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"reference_genome","name":"reference_genome"}],"label":"reference_genome","name":"Input parameter","outputs":[],"position":{"left":62.7833251953125,"top":169.9666748046875},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"0119d669-7ce9-46fd-ba6f-3efd92dfb7f2","when":null,"workflow_outputs":[]},"3":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":106.23333740234375,"top":262.3999938964844},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"5bb3b3df-60ab-4ec7-88e4-476be547ffbf","when":null,"workflow_outputs":[]},"4":{"annotation":"Whether you want to have a profile normalized as Signal to Million Reads","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Whether you want to have a profile normalized as Signal to Million Reads","name":"normalize_profile"}],"label":"normalize_profile","name":"Input parameter","outputs":[],"position":{"left":163.95545543674132,"top":359.9555974960296},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b89b796d-df4f-416f-8bc4-4f1951efa449","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":5,"input_connections":{"library|input_1":{"id":0,"output_name":"output"},"library|r1|adapters_0|adapter_source|adapter":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":"Cutadapt (remove adapter + bad quality bases)","name":"Cutadapt","outputs":[{"name":"out1","type":"fastqsanger"},{"name":"report","type":"txt"}],"position":{"left":423.31666564941406,"top":80.16667175292969},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"},"RenameDatasetActionreport":{"action_arguments":{"newname":"cutadapt report"},"action_type":"RenameDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"15\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC\\", \\"adapter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"single_noindels\\": false}], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"30\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"c7846b4c-54fb-458e-982e-c0d8358a9f5d","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","errors":null,"id":6,"input_connections":{"library|input_1":{"id":5,"output_name":"out1"},"reference_genome|index":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bowtie2","name":"library"},{"description":"runtime parameter for tool Bowtie2","name":"reference_genome"}],"label":"Bowtie2 map on reference","name":"Bowtie2","outputs":[{"name":"output","type":"bam"},{"name":"mapping_stats","type":"txt"}],"position":{"left":718.6333312988281,"top":459.8500061035156},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionmapping_stats":{"action_arguments":{"newname":"mapping stats"},"action_type":"RenameDatasetAction","output_name":"mapping_stats"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"bowtie2 output (BAM)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","tool_shed_repository":{"changeset_revision":"d5ceb9f3c25b","name":"bowtie2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"analysis_type\\": {\\"analysis_type_selector\\": \\"simple\\", \\"__current_case__\\": 0, \\"presets\\": \\"no_presets\\"}, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unaligned_file\\": false, \\"aligned_file\\": false}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"sam_options\\": {\\"sam_options_selector\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_mapping_stats\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.3+galaxy1","type":"tool","uuid":"6fd8444b-f305-4daa-a33a-5cb44e063f39","when":null,"workflow_outputs":[{"label":"mapping stats","output_name":"mapping_stats","uuid":"75a21a1c-1a84-4e7e-8b40-2bd11310f590"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","errors":null,"id":7,"input_connections":{"input1":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter SAM or BAM, output SAM or BAM","name":"bed_file"}],"label":"filter MAPQ30","name":"Filter SAM or BAM, output SAM or BAM","outputs":[{"name":"output1","type":"sam"}],"position":{"left":1114.0499877929688,"top":604.2999877929688},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"filtered BAM"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","tool_shed_repository":{"changeset_revision":"649a225999a5","name":"samtool_filter2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"flag\\": {\\"filter\\": \\"no\\", \\"__current_case__\\": 0}, \\"header\\": \\"-h\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": \\"\\", \\"mapq\\": \\"30\\", \\"outputtype\\": \\"bam\\", \\"possibly_select_inverse\\": false, \\"read_group\\": \\"\\", \\"regions\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy1","type":"tool","uuid":"bb6e3ac5-cdb0-493c-b534-264ba530a711","when":null,"workflow_outputs":[{"label":"filtered BAM","output_name":"output1","uuid":"053ba410-9509-42fd-9b88-0cb749e1738d"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":8,"input_connections":{"advanced_options|spmr":{"id":4,"output_name":"output"},"effective_genome_size_options|gsize":{"id":3,"output_name":"output"},"treatment|input_treatment_file":{"id":7,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"advanced_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"Call Peaks with MACS2","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_summits","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"},{"name":"output_extra_files","type":"html"}],"position":{"left":1555.199951171875,"top":337.98333740234375},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_extra_files":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_extra_files"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_control_lambda":{"action_arguments":{"newname":"MACS2 control coverage"},"action_type":"RenameDatasetAction","output_name":"output_control_lambda"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"MACS2 narrowPeak"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_summits":{"action_arguments":{"newname":"MACS2 summits"},"action_type":"RenameDatasetAction","output_name":"output_summits"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"MACS2 peaks xls"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"MACS2 treatment coverage"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAM\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"0\\"}, \\"outputs\\": [\\"peaks_tabular\\", \\"summits\\", \\"bdg\\", \\"html\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"06ca906e-c512-4376-9ffc-8eb2b5774af1","when":null,"workflow_outputs":[{"label":"MACS2 peaks","output_name":"output_tabular","uuid":"1be026e2-7e00-4d96-869c-9763f87477d8"},{"label":"MACS2 narrowPeak","output_name":"output_narrowpeaks","uuid":"5e4fb64b-40f1-4b07-b0c0-6a9fc6eb08b2"},{"label":"MACS2 summits","output_name":"output_summits","uuid":"146bf7b5-1ecb-4099-a52f-e48bc9f45d27"}]},"9":{"annotation":"summary of MACS2","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"infile":{"id":8,"output_name":"output_tabular"}},"inputs":[],"label":"summary of MACS2","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1965.11669921875,"top":208.98333740234375},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"MACS2 report"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"^#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"95832fa1-e96e-4867-8162-d1e39cb1dc46","when":null,"workflow_outputs":[{"label":"MACS2 report","output_name":"output","uuid":"acb6f9ef-7210-4477-a8aa-ed1898e32e96"}]},"10":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"output_treat_pileup"}},"inputs":[],"label":"Bigwig from MACS2","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":2026.9666748046875,"top":487.2833251953125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"coverage from MACS2 (bigwig)"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"0fe57c5d-00a0-4cb6-9bac-97e6d03c6b76","when":null,"workflow_outputs":[{"label":"coverage from MACS2","output_name":"out_file1","uuid":"1ab01ce0-8f89-4695-9e99-4cf969ec7d1e"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":11,"input_connections":{"results_0|software_cond|input":{"id":5,"output_name":"report"},"results_1|software_cond|input":{"id":6,"output_name":"mapping_stats"},"results_2|software_cond|input":{"id":8,"output_name":"output_tabular"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":"MultiQC","name":"MultiQC","outputs":[{"name":"plots","type":"input"},{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2074.183349609375,"top":866.2833251953125},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": true, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"bowtie2\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"fb066848-43df-412a-9767-9613bac7d961","when":null,"workflow_outputs":[{"label":"MultiQC webpage","output_name":"html_report","uuid":"2167289f-6c78-479a-8d3b-95caf11d0c4e"},{"label":"MultiQC on input dataset(s): Stats","output_name":"stats","uuid":"d4c3e0a7-d5b7-4307-8669-0254a3723ce0"}]}},"tags":["ChIP"],"uuid":"fe6bda9f-1fc2-4a86-bf8d-e779c79466fc","version":1},"readme":"# ChIP-seq Analysis: Single-End Read Processing\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of fastqsanger files.\\n\\n## Inputs values\\n\\n- adapters sequence_forward: this depends on the library preparation. If you don\'t know, use FastQC to determine if it is Truseq or Nextera.\\n- reference_genome: this field will be adapted to the genomes available for bowtie2.\\n- effective_genome_size: this is used by MACS2 and may be entered manually (indications are provided for heavily used genomes).\\n- normalize_profile: Whether you want to have a profile normalized as Signal to Million Reads.\\n\\n## Processing\\n\\n- The workflow will remove illumina adapters and low quality bases and filter out any read smaller than 15bp.\\n- The filtered reads are mapped with bowtie2 with default parameters.\\n- The BAM is filtered to keep only MAPQ30.\\n- The peaks are called with MACS2 with a fixed extension of 200bp which at the same time generates a coverage file (normalized or not).\\n- The coverage is converted to bigwig.\\n- A MultiQC is run to have an overview of the QC.\\n\\n### Warning\\n\\n- The filtered bam still has PCR duplicates which are removed by MACS2.\\n","changelog":"# Changelog\\n\\n## [0.14] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.13] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.12] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.11] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.10] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1`\\n\\n## [0.9] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n## [0.8] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n\\n## [0.7] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n\\n## [0.6] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n\\n## [0.5] - 2023-10-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n\\n## [0.4] - 2023-06-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n\\n### Manual update\\n- New parameter to get normalized profile\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0`\\n\\n## [0.1] - 2022-10-18\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## ChIP-seq Analysis: Single-End Read Processing\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ SR fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ adapter_forward\\"]@{ shape: lean-l }\\n2[\\"ℹ️ reference_genome\\"]@{ shape: lean-l }\\n3[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n4[\\"ℹ️ normalize_profile\\"]@{ shape: lean-l }\\n5[\\"Cutadapt (remove adapter + bad quality bases)\\"]@{ shape: process }\\n0 --> 5\\n1 --> 5\\n6[\\"Bowtie2 map on reference\\"]@{ shape: process }\\n5 --> 6\\n2 --> 6\\n7[\\"filter MAPQ30\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Call Peaks with MACS2\\"]@{ shape: process }\\n4 --> 8\\n3 --> 8\\n7 --> 8\\n9[\\"summary of MACS2\\"]@{ shape: process }\\n8 --> 9\\n10[\\"Bigwig from MACS2\\"]@{ shape: process }\\n8 --> 10\\n11[\\"MultiQC\\"]@{ shape: process }\\n5 --> 11\\n6 --> 11\\n8 --> 11\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/chipseq-sr/main","dockstore_id":19837,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078473","tests":[{"doc":"Test outline for ChIPseq_SR","job":{"SR fastq input":{"class":"Collection","collection_type":"list","elements":[{"identifier":"wt_H3K4me3","class":"File","location":"https://zenodo.org/record/1324070/files/wt_H3K4me3_read1.fastq.gz","filetype":"fastqsanger"}]},"adapter_forward":"GATCGGAAGAGCACACGTCTGAACTCCAGTCAC","reference_genome":"mm10","effective_genome_size":1870000000,"normalize_profile":true},"outputs":{"MultiQC webpage":{"asserts":[{"that":"has_text","text":"wt_H3K4me3"},{"that":"has_text","text":"<a href=\\"#cutadapt_filtered_reads\\" class=\\"nav-l2\\">Filtered Reads</a>"},{"that":"has_text","text":"<a href=\\"#bowtie2\\" class=\\"nav-l1\\">Bowtie 2 / HiSAT2</a>"}]},"MultiQC on input dataset(s): Stats":{"asserts":{"has_line":{"line":"Sample\\tmacs2-d\\tmacs2-treatment_redundant_rate\\tmacs2-peak_count\\tbowtie_2_hisat2-overall_alignment_rate\\tcutadapt-percent_trimmed"},"has_text_matching":{"expression":"wt_H3K4me3\\t200.0\\t0.0\\t4\\t98.[0-9]*\\t4.6[0-9]*"}}},"filtered BAM":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_size":{"value":2547967,"delta":200000}}}}},"MACS2 summits":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_n_lines":{"n":4}}}}},"MACS2 peaks":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"# effective genome size = 1.87e+09"},{"that":"has_text","text":"# d = 200"},{"that":"has_text","text":"# tags after filtering in treatment: 44038"}]}}},"MACS2 narrowPeak":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_n_lines":{"n":4}}}}},"MACS2 report":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"# effective genome size = 1.87e+09"},{"that":"has_text","text":"# d = 200"},{"that":"has_text","text":"# tags after filtering in treatment: 44038"}]}}},"coverage from MACS2":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_size":{"value":559925,"delta":10000}}}}},"mapping stats":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"49251 reads; of these:"},{"that":"has_text","text":"43525 (88.37%) aligned exactly 1 time"},{"that":"has_text","text":"98.37% overall alignment rate"}]}}}}}]}],"path":"./workflows/epigenetics/chipseq-sr"},{"version":1.2,"workflows":[{"name":"consensus-peaks-chip-sr","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/consensus-peaks-chip-sr.ga","testParameterFiles":["/consensus-peaks-chip-sr-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Identifies high-confidence consensus peaks from ChIP-seq single-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.","comments":[{"child_steps":[5,8,12],"color":"blue","data":{"title":"Get a BED with intersection of at least x rep"},"id":0,"position":[767.4,0],"size":[525.4,382.7],"type":"frame"},{"child_steps":[4,7,11,10,15,14,16,17,18,19],"color":"none","data":{"title":"Downsample input BAM to get the same number of reads"},"id":2,"position":[310.4,898],"size":[2000,403],"type":"frame"},{"child_steps":[9,13],"color":"green","data":{"title":"Get the average coverage"},"id":1,"position":[721.4,539.8],"size":[548.7,332],"type":"frame"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"1.3","name":"Consensus Peak Calling for ChIP-seq Single-End Replicates","steps":{"0":{"annotation":"A collection with n replicates. BAM should not have duplicates","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A collection with n replicates. BAM should not have duplicates","name":"n rmDup BAMSR"}],"label":"n rmDup BAMSR","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":168.94998779296876},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fc7f3d9f-c1a6-4e00-a65a-ff31ab9e6bca","when":null,"workflow_outputs":[]},"1":{"annotation":"Minimum number of replicates into which the summit should be present.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Minimum number of replicates into which the summit should be present.","name":"Minimum number of overlap"}],"label":"Minimum number of overlap","name":"Input parameter","outputs":[],"position":{"left":25,"top":247.4},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"025fcb5d-6842-4ed0-9bac-b393942b9d29","when":null,"workflow_outputs":[]},"2":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":39,"top":341.94791480468575},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"dac93661-a2aa-4163-a3ec-f64c4075c881","when":null,"workflow_outputs":[]},"3":{"annotation":"Bin size for average of normalized coverage","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Bin size for average of normalized coverage","name":"bin_size"}],"label":"bin_size","name":"Input parameter","outputs":[],"position":{"left":57,"top":438.98123999999825},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ef17ea2d-7f73-44a8-aecf-15279c559fd1","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":4,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"count number of reads","name":"Samtools view","outputs":[{"name":"outputcnt","type":"tabular"}],"position":{"left":330.46457068361184,"top":1016.5181008539595},"post_job_actions":{"HideDatasetActionoutputcnt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcnt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"all_reads\\", \\"__current_case__\\": 0, \\"output_options\\": {\\"reads_report_type\\": \\"count\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"a864ff43-af08-436a-aa78-54491ff0cce6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":5,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":"generate filter rule","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":795.7157982750858,"top":206.5517325344974},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c4 >= \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"e6855df9-0e94-4310-835c-15514bd37974","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":6,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks individually","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"}],"position":{"left":454.449951171875,"top":153.21457740234203},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"individual macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"individual macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"individual macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAM\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"0\\"}, \\"outputs\\": [\\"bdg\\", \\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"8f2a3aa4-704e-4d90-a672-94b97996dd56","when":null,"workflow_outputs":[{"label":"individual_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"4552df5d-4397-4118-af11-f97eb8c748a5"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":7,"input_connections":{"input_list":{"id":4,"output_name":"outputcnt"}},"inputs":[],"label":"put all nb of reads into single dataset","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":585.9498956836106,"top":1016.0836991103162},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"bd8b6b03-19ef-4921-96c3-abe5e6992941","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","errors":null,"id":8,"input_connections":{"tag|inputs":{"id":6,"output_name":"output_narrowpeaks"}},"inputs":[{"description":"runtime parameter for tool bedtools Multiple Intersect","name":"tag"}],"label":"compute multi intersect","name":"bedtools Multiple Intersect","outputs":[{"name":"output","type":"bed"}],"position":{"left":790,"top":49.4},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"cluster\\": false, \\"empty\\": {\\"empty_selector\\": \\"\\", \\"__current_case__\\": 1}, \\"filler\\": \\"0\\", \\"header\\": false, \\"tag\\": {\\"tag_select\\": \\"tag\\", \\"__current_case__\\": 0, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"9edd1aab-eb4c-4fd4-b9f6-a6dbcae46dd2","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":9,"input_connections":{"input1":{"id":6,"output_name":"output_treat_pileup"}},"inputs":[],"label":"individual normalized bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":777.2817744590351,"top":635.7085502853505},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"d0d6f92d-3d7e-44c3-b2f6-c10e429d2369","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","errors":null,"id":10,"input_connections":{"singtabop|input":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Table Compute","name":"singtabop"}],"label":"get min value","name":"Table Compute","outputs":[{"name":"table","type":"tabular"}],"position":{"left":839.4690473102415,"top":1013.9803858976712},"post_job_actions":{"HideDatasetActiontable":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"table"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","tool_shed_repository":{"changeset_revision":"3bf5661c0280","name":"table_compute","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"out_opts\\": null, \\"precision\\": \\"6\\", \\"singtabop\\": {\\"use_type\\": \\"single\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"col_row_names\\": null, \\"adv\\": {\\"header\\": null, \\"nrows\\": null, \\"skipfooter\\": null, \\"skip_blank_lines\\": true}, \\"user\\": {\\"mode\\": \\"matrixapply\\", \\"__current_case__\\": 3, \\"matrixapply_func\\": {\\"vector_op\\": \\"min\\", \\"__current_case__\\": 1}, \\"dimension\\": \\"0\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.4+galaxy0","type":"tool","uuid":"0d63ffd8-0646-4dbf-9ede-14bffcd5de2d","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"wc_gnu","errors":null,"id":11,"input_connections":{"input1":{"id":7,"output_name":"output"}},"inputs":[],"label":"get nb of replicates","name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":852.2032935094636,"top":1131.5836587544356},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"1e4420d0-edfb-4411-b7e0-602b510ff898","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"Filter1","errors":null,"id":12,"input_connections":{"cond":{"id":5,"output_name":"out1"},"input":{"id":8,"output_name":"output"}},"inputs":[],"label":"filter multi intersect","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1058,"top":57.4},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"4acd5289-87cc-4cfb-9863-35248989ab27","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":13,"input_connections":{"advancedOpt|binSize":{"id":3,"output_name":"output"},"bigwigs":{"id":9,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"average coverage from replicates","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":1035.1219194343084,"top":575.1464921044934},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"Average Normalized Coverage of all replicates"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": \\"1\\", \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"36059130-5f2a-4fe0-8b36-58073595fdd5","when":null,"workflow_outputs":[{"label":"average_bigwig","output_name":"outFileName","uuid":"35dfdb8a-121d-41d1-9ad1-1e0d95a8ab8c"}]},"14":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":14,"input_connections":{"input1":{"id":10,"output_name":"table"}},"inputs":[],"label":"convert min value to text","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":1084.3691526034222,"top":938.054981863764},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"cdbee56b-9fff-4205-9cb6-741be770985d","when":null,"workflow_outputs":[]},"15":{"annotation":"get nb of rep as param","content_id":"param_value_from_file","errors":null,"id":15,"input_connections":{"input1":{"id":11,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1092.4166992159298,"top":1123.4218289425305},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"2ea49dca-270b-454a-9d86-6947799ab996","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":16,"input_connections":{"token_set_0|line":{"id":14,"output_name":"text_param"},"token_set_0|repeat_select|times":{"id":15,"output_name":"integer_param"}},"inputs":[],"label":"create a dataset with the min value as many times as there are replicates","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":1355.26353673623,"top":996.1249965061141},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"29ef7038-e10a-47cd-835f-945dccaf9c1c","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":17,"input_connections":{"split_parms|input":{"id":16,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split min value","name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":1583.8149291973814,"top":1143.0376400865628},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"7db836a5-793a-403e-8db1-b97f619f4c69","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":18,"input_connections":{"input1":{"id":17,"output_name":"list_output_txt"}},"inputs":[],"label":"convert min nb of reads to parameter","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1830.6125542203363,"top":1109.3615627344768},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"71b1c454-41db-4a8b-8595-8d3fa86a86a0","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":19,"input_connections":{"input":{"id":0,"output_name":"output"},"mode|subsample_config|subsampling_mode|target":{"id":18,"output_name":"integer_param"}},"inputs":[],"label":"downsample BAM","name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":2090.7845788070495,"top":1085.996383548678},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": null, \\"library\\": null, \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"target\\", \\"__current_case__\\": 1, \\"target\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seed\\": \\"1\\"}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"ae46e826-d985-4437-abae-152b6877eea7","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":20,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":19,"output_name":"outputsam"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks on merge","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"}],"position":{"left":1546.199951171875,"top":579.4979025976545},"post_job_actions":{"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"merged macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"merged macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"merged macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAM\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"0\\"}, \\"outputs\\": [\\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"Yes\\", \\"__current_case__\\": 1, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"e183ec20-d4ab-427a-9a0c-3e9bbced8793","when":null,"workflow_outputs":[{"label":"merged_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"6d97d620-9486-4d85-9a80-e3595c0bb968"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","errors":null,"id":21,"input_connections":{"inputA":{"id":20,"output_name":"output_narrowpeaks"},"reduce_or_iterate|inputB":{"id":12,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools Intersect intervals","name":"reduce_or_iterate"}],"label":"get merged peaks overlapping at least x replicates","name":"bedtools Intersect intervals","outputs":[{"name":"output","type":"input"}],"position":{"left":1643.5166015625,"top":314.897927011717},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed\\": false, \\"count\\": false, \\"fraction_cond\\": {\\"fraction_select\\": \\"default\\", \\"__current_case__\\": 0}, \\"genome_file_opts\\": {\\"genome_file_opts_selector\\": \\"loc\\", \\"__current_case__\\": 0, \\"genome\\": null}, \\"header\\": false, \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": false, \\"once\\": false, \\"overlap_mode\\": [\\"-wa\\", \\"-wb\\"], \\"reduce_or_iterate\\": {\\"reduce_or_iterate_selector\\": \\"iterate\\", \\"__current_case__\\": 0, \\"inputB\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"sorted\\": false, \\"split\\": false, \\"strand\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"d6183a41-f78c-49fc-b826-f8aa95ff84cb","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":22,"input_connections":{"results_0|software_cond|input":{"id":6,"output_name":"output_tabular"},"results_1|software_cond|input":{"id":20,"output_name":"output_tabular"}},"inputs":[],"label":"multiQC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2315.5733416696335,"top":16.13231218303406},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"f7d74248-16e3-4020-8156-caae0e7682ef","when":null,"workflow_outputs":[{"label":"multiqc_output","output_name":"html_report","uuid":"a342a984-235e-4d40-9e09-d835aab35282"}]},"23":{"annotation":"","content_id":"Filter1","errors":null,"id":23,"input_connections":{"input":{"id":21,"output_name":"output"}},"inputs":[],"label":"only keep peaks with summits overlapping intersection of at least x replicates","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1874.401877544738,"top":319.9312522070295},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"(c2+c10)>=c12 and (c2+c10)<c13\\", \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"dd65edce-c587-4e3f-9c08-408fe81c557e","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"Cut1","errors":null,"id":24,"input_connections":{"input":{"id":23,"output_name":"out_file1"}},"inputs":[],"label":"keep only columns of narrowPeak","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2104.095825078475,"top":414.0305050651918},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1-c10\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6a58206f-a7b2-4f25-b837-b17185c86c52","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":24,"output_name":"out_file1"}},"inputs":[],"label":"discard duplicated lines","name":"Unique","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2336.462697917632,"top":435.17882451605976},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"narrowPeak shared by at least x replicates"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_numeric\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"62a36e17-5aea-419d-835d-0fa33dd05e97","when":null,"workflow_outputs":[{"label":"shared_narrowPeak","output_name":"outfile","uuid":"47b85acb-10eb-454c-aa95-1253f223641d"}]}},"tags":["ATAC-seq"],"uuid":"f71eee25-8402-41da-8284-5298656ab379","version":2},"readme":"# Consensus Peak Calling for ChIP-seq, ATAC-seq and CUT&RUN Replicates\\n\\nThe goal of this workflow is to get a list of confident peaks with summits from n replicates.\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of datasets with n BAM where PCR duplicates have been removed (the workflow also works for nested list if you have multiple conditions each with multiple replicates).\\n\\n## Inputs values\\n\\n- Minimum number of overlap: Minimum number of replicates into which the final summit should be present.\\n- effective_genome_size: this is used by MACS2 and may be entered manually (indications are provided for heavily used genomes).\\n- bin_size: this is the bin sized used to compute the average of normalized profiles. Large values will allow to have a smaller output file but with less resolution while small values will increase computation time and size of the output file to produce a more resolutive bigwig.\\n\\n## Strategy summary\\n\\nHere is a generated example to highlight the strategy:\\n![strategy](https://raw.githubusercontent.com/galaxyproject/iwc/main/workflows/epigenetics/consensus-peaks/strategy.png)\\n\\n## Processing\\n\\n- The workflow will:\\n  - first part:\\n    - call peaks and compute normalized coverage on each BAM individually\\n    - average normalized profiles\\n    - compute the intersection between all peaks and filter when at least x replicate overlaps\\n  - second part:\\n    - subset all BAM to get the same number of reads\\n    - call peaks on all subsetted BAM combined\\n  - finally, keep only peaks from the second part that have summits overlapping the filtered intersection of the first part.\\n","changelog":"# Changelog\\n\\n## [1.3] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [1.2] - 2024-09-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [1.1] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n\\n## [1.0] - 2024-05-01\\n\\n### Changed\\nThe workflow changed as it accepts now more than 2 replicates and has a new parameter to define what is the minimum number of replicates where the summit should be found.\\nThe workflow for ATAC and CUT&RUN changed the input datatype as now a BAM is required instead of a BED.\\n\\n## [0.7] - 2024-04-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.6] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.5] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0`\\n\\n## [0.4] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.3] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.2] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n\\n## [0.1] - 2023-08-31\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Consensus Peak Calling for ChIP-seq Single-End Replicates\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ n rmDup BAMSR\\"]@{ shape: docs }\\n1[\\"ℹ️ Minimum number of overlap\\"]@{ shape: lean-l }\\n2[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n3[\\"ℹ️ bin_size\\"]@{ shape: lean-l }\\n4[\\"count number of reads\\"]@{ shape: process }\\n0 --> 4\\n5[\\"generate filter rule\\"]@{ shape: process }\\n1 --> 5\\n6[\\"call peaks individually\\"]@{ shape: process }\\n2 --> 6\\n0 --> 6\\n7[\\"put all nb of reads into single dataset\\"]@{ shape: process }\\n4 --> 7\\n8[\\"compute multi intersect\\"]@{ shape: process }\\n6 --> 8\\n9[\\"individual normalized bigwig\\"]@{ shape: process }\\n6 --> 9\\n10[\\"get min value\\"]@{ shape: process }\\n7 --> 10\\n11[\\"get nb of replicates\\"]@{ shape: process }\\n7 --> 11\\n12[\\"filter multi intersect\\"]@{ shape: process }\\n5 --> 12\\n8 --> 12\\n13[\\"average coverage from replicates\\"]@{ shape: process }\\n3 --> 13\\n9 --> 13\\n14[\\"convert min value to text\\"]@{ shape: process }\\n10 --> 14\\n15[\\"Parse parameter value\\"]@{ shape: process }\\n11 --> 15\\n16[\\"create a dataset with the min value as many times as there are replicates\\"]@{ shape: process }\\n14 --> 16\\n15 --> 16\\n17[\\"split min value\\"]@{ shape: process }\\n16 --> 17\\n18[\\"convert min nb of reads to parameter\\"]@{ shape: process }\\n17 --> 18\\n19[\\"downsample BAM\\"]@{ shape: process }\\n0 --> 19\\n18 --> 19\\n20[\\"call peaks on merge\\"]@{ shape: process }\\n2 --> 20\\n19 --> 20\\n21[\\"get merged peaks overlapping at least x replicates\\"]@{ shape: process }\\n20 --> 21\\n12 --> 21\\n22[\\"multiQC\\"]@{ shape: process }\\n6 --> 22\\n20 --> 22\\n23[\\"only keep peaks with summits overlapping intersection of at least x replicates\\"]@{ shape: process }\\n21 --> 23\\n24[\\"keep only columns of narrowPeak\\"]@{ shape: process }\\n23 --> 24\\n25[\\"discard duplicated lines\\"]@{ shape: process }\\n24 --> 25\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-sr","dockstore_id":23496,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078523","tests":[{"doc":"Test for consensus-peaks-chip-sr.ga","job":{"n rmDup BAMSR":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"rep1","path":"test-data/ChIP_SR_rep1.bam","dbkey":"mm10"},{"class":"File","identifier":"rep2","path":"test-data/ChIP_SR_rep2.bam","dbkey":"mm10"},{"class":"File","identifier":"rep3","path":"test-data/ChIP_SR_rep3.bam","dbkey":"mm10"}]},"Minimum number of overlap":2,"effective_genome_size":1870000000,"bin_size":50},"outputs":{"individual_macs2_narrowPeaks":{"element_tests":{"rep1":{"asserts":{"has_n_lines":{"n":3}}},"rep2":{"asserts":{"has_n_lines":{"n":3}}},"rep3":{"asserts":{"has_n_lines":{"n":3}}}}},"average_bigwig":{"asserts":{"has_size":{"value":1290,"delta":100}}},"merged_macs2_narrowPeaks":{"asserts":{"has_n_lines":{"n":4}}},"multiqc_output":{"asserts":{"has_text":{"text":"MACS2: Fragment Length"}}},"shared_narrowPeak":{"asserts":{"has_n_lines":{"n":2}}}}}]},{"name":"consensus-peaks-chip-pe","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/consensus-peaks-chip-pe.ga","testParameterFiles":["/consensus-peaks-chip-pe-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Identifies high-confidence consensus peaks from ChIP-seq paired-end replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.","comments":[{"child_steps":[9,13],"color":"green","data":{"title":"Get the average coverage"},"id":2,"position":[687.5,395.8],"size":[498.9,345.5],"type":"frame"},{"child_steps":[4,7,11,10,15,14,16,17,18,19],"color":"none","data":{"title":"Downsample input BAM to get the same number of reads"},"id":1,"position":[180.2,744.1],"size":[1889,379],"type":"frame"},{"child_steps":[5,8,12],"color":"blue","data":{"title":"Get a BED with intersection of at least x rep"},"id":0,"position":[764.8,0],"size":[531.7,370.1],"type":"frame"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"1.3","name":"Consensus Peak Calling for ChIP-seq Paired-End Replicates","steps":{"0":{"annotation":"A collection with n replicates. BAM should not have duplicates","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A collection with n replicates. BAM should not have duplicates","name":"n rmDup BAMPE"}],"label":"n rmDup BAMPE","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":155.14998779296874},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fc7f3d9f-c1a6-4e00-a65a-ff31ab9e6bca","when":null,"workflow_outputs":[]},"1":{"annotation":"Minimum number of replicates into which the summit should be present.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Minimum number of replicates into which the summit should be present.","name":"Minimum number of overlap"}],"label":"Minimum number of overlap","name":"Input parameter","outputs":[],"position":{"left":25,"top":233.6},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"025fcb5d-6842-4ed0-9bac-b393942b9d29","when":null,"workflow_outputs":[]},"2":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":39,"top":328.1479148046858},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"dac93661-a2aa-4163-a3ec-f64c4075c881","when":null,"workflow_outputs":[]},"3":{"annotation":"Bin size for average of normalized coverage","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Bin size for average of normalized coverage","name":"bin_size"}],"label":"bin_size","name":"Input parameter","outputs":[],"position":{"left":57,"top":425.1812399999983},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ef17ea2d-7f73-44a8-aecf-15279c559fd1","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":4,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"count number of reads","name":"Samtools view","outputs":[{"name":"outputcnt","type":"tabular"}],"position":{"left":200.2051223265678,"top":859.5510386258217},"post_job_actions":{"HideDatasetActionoutputcnt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcnt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"all_reads\\", \\"__current_case__\\": 0, \\"output_options\\": {\\"reads_report_type\\": \\"count\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"a864ff43-af08-436a-aa78-54491ff0cce6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":5,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":"generate filter rule","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":793.0213699877876,"top":178.96164159176644},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c4 >= \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"e6855df9-0e94-4310-835c-15514bd37974","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":6,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks individually","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"}],"position":{"left":454.449951171875,"top":139.41457740234202},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"individual macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"individual macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"individual macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAMPE\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"create_model\\", \\"__current_case__\\": 0, \\"mfold_lower\\": \\"5\\", \\"mfold_upper\\": \\"50\\", \\"band_width\\": \\"300\\"}, \\"outputs\\": [\\"bdg\\", \\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"8f2a3aa4-704e-4d90-a672-94b97996dd56","when":null,"workflow_outputs":[{"label":"individual_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"4552df5d-4397-4118-af11-f97eb8c748a5"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":7,"input_connections":{"input_list":{"id":4,"output_name":"outputcnt"}},"inputs":[],"label":"put all nb of reads into single dataset","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":437.3050979125053,"top":856.8156282351949},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"bd8b6b03-19ef-4921-96c3-abe5e6992941","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","errors":null,"id":8,"input_connections":{"tag|inputs":{"id":6,"output_name":"output_narrowpeaks"}},"inputs":[{"description":"runtime parameter for tool bedtools Multiple Intersect","name":"tag"}],"label":"compute multi intersect","name":"bedtools Multiple Intersect","outputs":[{"name":"output","type":"bed"}],"position":{"left":790,"top":35.6},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"cluster\\": false, \\"empty\\": {\\"empty_selector\\": \\"\\", \\"__current_case__\\": 1}, \\"filler\\": \\"0\\", \\"header\\": false, \\"tag\\": {\\"tag_select\\": \\"tag\\", \\"__current_case__\\": 0, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"9edd1aab-eb4c-4fd4-b9f6-a6dbcae46dd2","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":9,"input_connections":{"input1":{"id":6,"output_name":"output_treat_pileup"}},"inputs":[],"label":"individual normalized bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":701.9011607009672,"top":524.1953936525294},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"d0d6f92d-3d7e-44c3-b2f6-c10e429d2369","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","errors":null,"id":10,"input_connections":{"singtabop|input":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Table Compute","name":"singtabop"}],"label":"get min value","name":"Table Compute","outputs":[{"name":"table","type":"tabular"}],"position":{"left":661.2051223265678,"top":836.1343516141029},"post_job_actions":{"HideDatasetActiontable":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"table"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","tool_shed_repository":{"changeset_revision":"3bf5661c0280","name":"table_compute","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"out_opts\\": null, \\"precision\\": \\"6\\", \\"singtabop\\": {\\"use_type\\": \\"single\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"col_row_names\\": null, \\"adv\\": {\\"header\\": null, \\"nrows\\": null, \\"skipfooter\\": null, \\"skip_blank_lines\\": true}, \\"user\\": {\\"mode\\": \\"matrixapply\\", \\"__current_case__\\": 3, \\"matrixapply_func\\": {\\"vector_op\\": \\"min\\", \\"__current_case__\\": 1}, \\"dimension\\": \\"0\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.4+galaxy0","type":"tool","uuid":"0d63ffd8-0646-4dbf-9ede-14bffcd5de2d","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"wc_gnu","errors":null,"id":11,"input_connections":{"input1":{"id":7,"output_name":"output"}},"inputs":[],"label":"get nb of replicates","name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":664.2051223265678,"top":946.0510386258217},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"1e4420d0-edfb-4411-b7e0-602b510ff898","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"Filter1","errors":null,"id":12,"input_connections":{"cond":{"id":5,"output_name":"out1"},"input":{"id":8,"output_name":"output"}},"inputs":[],"label":"filter multi intersect","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1058,"top":43.6},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"4acd5289-87cc-4cfb-9863-35248989ab27","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":13,"input_connections":{"advancedOpt|binSize":{"id":3,"output_name":"output"},"bigwigs":{"id":9,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"average coverage from replicates","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":944.6735015666067,"top":451.87316496226794},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"Average Normalized Coverage of all replicates"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": \\"1\\", \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"36059130-5f2a-4fe0-8b36-58073595fdd5","when":null,"workflow_outputs":[{"label":"average_bigwig","output_name":"outFileName","uuid":"35dfdb8a-121d-41d1-9ad1-1e0d95a8ab8c"}]},"14":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":14,"input_connections":{"input1":{"id":10,"output_name":"table"}},"inputs":[],"label":"convert min value to text","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":903.2051223265678,"top":784.0510386258217},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"cdbee56b-9fff-4205-9cb6-741be770985d","when":null,"workflow_outputs":[]},"15":{"annotation":"get nb of rep as param","content_id":"param_value_from_file","errors":null,"id":15,"input_connections":{"input1":{"id":11,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":907.2051223265678,"top":949.0510386258217},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"2ea49dca-270b-454a-9d86-6947799ab996","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":16,"input_connections":{"token_set_0|line":{"id":14,"output_name":"text_param"},"token_set_0|repeat_select|times":{"id":15,"output_name":"integer_param"}},"inputs":[],"label":"create a dataset with the min value as many times as there are replicates","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":1136.2051223265678,"top":843.0510386258217},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"29ef7038-e10a-47cd-835f-945dccaf9c1c","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":17,"input_connections":{"split_parms|input":{"id":16,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split min value","name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":1371.2051223265678,"top":921.0510386258217},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"7db836a5-793a-403e-8db1-b97f619f4c69","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":18,"input_connections":{"input1":{"id":17,"output_name":"list_output_txt"}},"inputs":[],"label":"convert min nb of reads to parameter","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1605.2384475218803,"top":912.9677256375404},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"71b1c454-41db-4a8b-8595-8d3fa86a86a0","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":19,"input_connections":{"input":{"id":0,"output_name":"output"},"mode|subsample_config|subsampling_mode|target":{"id":18,"output_name":"integer_param"}},"inputs":[],"label":"downsample BAM","name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":1848.7423727171654,"top":894.4010142117592},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": null, \\"library\\": null, \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"target\\", \\"__current_case__\\": 1, \\"target\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seed\\": \\"1\\"}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"ae46e826-d985-4437-abae-152b6877eea7","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":20,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":19,"output_name":"outputsam"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks on merge","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"}],"position":{"left":1456.336227527062,"top":482.24653231537656},"post_job_actions":{"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"merged macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"merged macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"merged macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAMPE\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"create_model\\", \\"__current_case__\\": 0, \\"mfold_lower\\": \\"5\\", \\"mfold_upper\\": \\"50\\", \\"band_width\\": \\"300\\"}, \\"outputs\\": [\\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"Yes\\", \\"__current_case__\\": 1, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"e183ec20-d4ab-427a-9a0c-3e9bbced8793","when":null,"workflow_outputs":[{"label":"merged_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"6d97d620-9486-4d85-9a80-e3595c0bb968"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","errors":null,"id":21,"input_connections":{"inputA":{"id":20,"output_name":"output_narrowpeaks"},"reduce_or_iterate|inputB":{"id":12,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools Intersect intervals","name":"reduce_or_iterate"}],"label":"get merged peaks overlapping at least x replicates","name":"bedtools Intersect intervals","outputs":[{"name":"output","type":"input"}],"position":{"left":1643.5166015625,"top":301.09792701171705},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed\\": false, \\"count\\": false, \\"fraction_cond\\": {\\"fraction_select\\": \\"default\\", \\"__current_case__\\": 0}, \\"genome_file_opts\\": {\\"genome_file_opts_selector\\": \\"loc\\", \\"__current_case__\\": 0, \\"genome\\": null}, \\"header\\": false, \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": false, \\"once\\": false, \\"overlap_mode\\": [\\"-wa\\", \\"-wb\\"], \\"reduce_or_iterate\\": {\\"reduce_or_iterate_selector\\": \\"iterate\\", \\"__current_case__\\": 0, \\"inputB\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"sorted\\": false, \\"split\\": false, \\"strand\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"d6183a41-f78c-49fc-b826-f8aa95ff84cb","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":22,"input_connections":{"results_0|software_cond|input":{"id":6,"output_name":"output_tabular"},"results_1|software_cond|input":{"id":20,"output_name":"output_tabular"}},"inputs":[],"label":"multiQC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2091.738158489414,"top":26.470097442576304},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"f7d74248-16e3-4020-8156-caae0e7682ef","when":null,"workflow_outputs":[{"label":"multiqc_output","output_name":"html_report","uuid":"a342a984-235e-4d40-9e09-d835aab35282"}]},"23":{"annotation":"","content_id":"Filter1","errors":null,"id":23,"input_connections":{"input":{"id":21,"output_name":"output"}},"inputs":[],"label":"only keep peaks with summits overlapping intersection of at least x replicates","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1874.401877544738,"top":306.13125220702955},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"(c2+c10)>=c12 and (c2+c10)<c13\\", \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"dd65edce-c587-4e3f-9c08-408fe81c557e","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"Cut1","errors":null,"id":24,"input_connections":{"input":{"id":23,"output_name":"out_file1"}},"inputs":[],"label":"keep only columns of narrowPeak","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2104.095825078475,"top":400.23050506519184},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1-c10\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6a58206f-a7b2-4f25-b837-b17185c86c52","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","errors":null,"id":25,"input_connections":{"infile":{"id":24,"output_name":"out_file1"}},"inputs":[],"label":"discard duplicated lines","name":"Unique","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2336.462697917632,"top":421.3788245160598},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"narrowPeak shared by at least x replicates"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_numeric\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"62a36e17-5aea-419d-835d-0fa33dd05e97","when":null,"workflow_outputs":[{"label":"shared_narrowPeak","output_name":"outfile","uuid":"47b85acb-10eb-454c-aa95-1253f223641d"}]}},"tags":["ATAC-seq"],"uuid":"a2f66906-e9e3-42f9-8629-390017ed2db0","version":6},"readme":"# Consensus Peak Calling for ChIP-seq, ATAC-seq and CUT&RUN Replicates\\n\\nThe goal of this workflow is to get a list of confident peaks with summits from n replicates.\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of datasets with n BAM where PCR duplicates have been removed (the workflow also works for nested list if you have multiple conditions each with multiple replicates).\\n\\n## Inputs values\\n\\n- Minimum number of overlap: Minimum number of replicates into which the final summit should be present.\\n- effective_genome_size: this is used by MACS2 and may be entered manually (indications are provided for heavily used genomes).\\n- bin_size: this is the bin sized used to compute the average of normalized profiles. Large values will allow to have a smaller output file but with less resolution while small values will increase computation time and size of the output file to produce a more resolutive bigwig.\\n\\n## Strategy summary\\n\\nHere is a generated example to highlight the strategy:\\n![strategy](https://raw.githubusercontent.com/galaxyproject/iwc/main/workflows/epigenetics/consensus-peaks/strategy.png)\\n\\n## Processing\\n\\n- The workflow will:\\n  - first part:\\n    - call peaks and compute normalized coverage on each BAM individually\\n    - average normalized profiles\\n    - compute the intersection between all peaks and filter when at least x replicate overlaps\\n  - second part:\\n    - subset all BAM to get the same number of reads\\n    - call peaks on all subsetted BAM combined\\n  - finally, keep only peaks from the second part that have summits overlapping the filtered intersection of the first part.\\n","changelog":"# Changelog\\n\\n## [1.3] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [1.2] - 2024-09-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [1.1] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n\\n## [1.0] - 2024-05-01\\n\\n### Changed\\nThe workflow changed as it accepts now more than 2 replicates and has a new parameter to define what is the minimum number of replicates where the summit should be found.\\nThe workflow for ATAC and CUT&RUN changed the input datatype as now a BAM is required instead of a BED.\\n\\n## [0.7] - 2024-04-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.6] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.5] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0`\\n\\n## [0.4] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.3] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.2] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n\\n## [0.1] - 2023-08-31\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Consensus Peak Calling for ChIP-seq Paired-End Replicates\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ n rmDup BAMPE\\"]@{ shape: docs }\\n1[\\"ℹ️ Minimum number of overlap\\"]@{ shape: lean-l }\\n2[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n3[\\"ℹ️ bin_size\\"]@{ shape: lean-l }\\n4[\\"count number of reads\\"]@{ shape: process }\\n0 --> 4\\n5[\\"generate filter rule\\"]@{ shape: process }\\n1 --> 5\\n6[\\"call peaks individually\\"]@{ shape: process }\\n2 --> 6\\n0 --> 6\\n7[\\"put all nb of reads into single dataset\\"]@{ shape: process }\\n4 --> 7\\n8[\\"compute multi intersect\\"]@{ shape: process }\\n6 --> 8\\n9[\\"individual normalized bigwig\\"]@{ shape: process }\\n6 --> 9\\n10[\\"get min value\\"]@{ shape: process }\\n7 --> 10\\n11[\\"get nb of replicates\\"]@{ shape: process }\\n7 --> 11\\n12[\\"filter multi intersect\\"]@{ shape: process }\\n5 --> 12\\n8 --> 12\\n13[\\"average coverage from replicates\\"]@{ shape: process }\\n3 --> 13\\n9 --> 13\\n14[\\"convert min value to text\\"]@{ shape: process }\\n10 --> 14\\n15[\\"Parse parameter value\\"]@{ shape: process }\\n11 --> 15\\n16[\\"create a dataset with the min value as many times as there are replicates\\"]@{ shape: process }\\n14 --> 16\\n15 --> 16\\n17[\\"split min value\\"]@{ shape: process }\\n16 --> 17\\n18[\\"convert min nb of reads to parameter\\"]@{ shape: process }\\n17 --> 18\\n19[\\"downsample BAM\\"]@{ shape: process }\\n0 --> 19\\n18 --> 19\\n20[\\"call peaks on merge\\"]@{ shape: process }\\n2 --> 20\\n19 --> 20\\n21[\\"get merged peaks overlapping at least x replicates\\"]@{ shape: process }\\n20 --> 21\\n12 --> 21\\n22[\\"multiQC\\"]@{ shape: process }\\n6 --> 22\\n20 --> 22\\n23[\\"only keep peaks with summits overlapping intersection of at least x replicates\\"]@{ shape: process }\\n21 --> 23\\n24[\\"keep only columns of narrowPeak\\"]@{ shape: process }\\n23 --> 24\\n25[\\"discard duplicated lines\\"]@{ shape: process }\\n24 --> 25\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-chip-pe","dockstore_id":23497,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078521","tests":[{"doc":"Test for consensus-peaks-chip-pe.ga","job":{"n rmDup BAMPE":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"rep1","path":"test-data/rep1.bam","dbkey":"mm10"},{"class":"File","identifier":"rep2","path":"test-data/rep2.bam","dbkey":"mm10"},{"class":"File","identifier":"rep3","path":"test-data/rep3.bam","dbkey":"mm10"}]},"Minimum number of overlap":2,"effective_genome_size":1870000000,"bin_size":50},"outputs":{"individual_macs2_narrowPeaks":{"element_tests":{"rep1":{"asserts":{"has_n_lines":{"n":3}}},"rep2":{"asserts":{"has_n_lines":{"n":3}}},"rep3":{"asserts":{"has_n_lines":{"n":2}}}}},"average_bigwig":{"asserts":{"has_size":{"value":1290,"delta":100}}},"merged_macs2_narrowPeaks":{"asserts":{"has_n_lines":{"n":4}}},"multiqc_output":{"asserts":{"has_text":{"text":"MACS2: Fragment Length"}}},"shared_narrowPeak":{"asserts":{"has_n_lines":{"n":2}}}}}]},{"name":"consensus-peaks-atac-cutandrun","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/consensus-peaks-atac-cutandrun.ga","testParameterFiles":["/consensus-peaks-atac-cutandrun-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Identifies high-confidence consensus peaks from ATAC-seq or CUT&RUN replicate experiments. The workflow calls peaks on individual replicates and identifies their intersection. To control for sequencing depth differences, it subsamples all replicates to the smallest library size, performs peak calling on the combined normalized data, and retains only peaks whose summits overlap with intersections from a user-defined minimum number of replicates.","comments":[{"child_steps":[5,8,11,12,15,16,17,18,19,20],"color":"blue","data":{"title":"Downsample input BAM to get the same number of reads"},"id":2,"position":[228,878.7],"size":[1917.8,397.4],"type":"frame"},{"child_steps":[6,9,13],"color":"blue","data":{"title":"Get a BED with intersection of at least x rep"},"id":0,"position":[776.6,0],"size":[508.7,363],"type":"frame"},{"child_steps":[10,14],"color":"green","data":{"title":"Get the average coverage"},"id":1,"position":[647.1,523.9],"size":[504.5,324],"type":"frame"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"1.3","name":"Consensus Peak Calling for ATAC-seq and CUT&RUN Replicates","steps":{"0":{"annotation":"A collection with n replicates. BAM should not have duplicates.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A collection with n replicates. BAM should not have duplicates.","name":"n rmDup BAM"}],"label":"n rmDup BAM","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":194.74998779296874},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fc7f3d9f-c1a6-4e00-a65a-ff31ab9e6bca","when":null,"workflow_outputs":[]},"1":{"annotation":"Minimum number of replicates into which the summit should be present.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Minimum number of replicates into which the summit should be present.","name":"Minimum number of overlap"}],"label":"Minimum number of overlap","name":"Input parameter","outputs":[],"position":{"left":25,"top":273.2},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"025fcb5d-6842-4ed0-9bac-b393942b9d29","when":null,"workflow_outputs":[]},"2":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":39,"top":367.74791480468576},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"dac93661-a2aa-4163-a3ec-f64c4075c881","when":null,"workflow_outputs":[]},"3":{"annotation":"Bin size for average of normalized coverage","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Bin size for average of normalized coverage","name":"bin_size"}],"label":"bin_size","name":"Input parameter","outputs":[],"position":{"left":57,"top":464.78123999999826},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ef17ea2d-7f73-44a8-aecf-15279c559fd1","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","errors":null,"id":4,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"convert BAM to BED","name":"bedtools BAM to BED","outputs":[{"name":"output","type":"bed"}],"position":{"left":217.78016821945388,"top":74.30843071525936},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"ed_score\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"option\\": \\"\\", \\"split\\": false, \\"tag\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"b62c338c-eb40-4ea5-b5e6-819f39790c07","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":5,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"count number of reads","name":"Samtools view","outputs":[{"name":"outputcnt","type":"tabular"}],"position":{"left":248.01056678577842,"top":1011.4507862713202},"post_job_actions":{"HideDatasetActionoutputcnt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcnt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"all_reads\\", \\"__current_case__\\": 0, \\"output_options\\": {\\"reads_report_type\\": \\"count\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"9cff58b2-123e-42df-b562-3b2a4a553934","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"}},"inputs":[],"label":"generate filter rule","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":796.6100753157618,"top":188.622596980592},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c4 >= \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"e6855df9-0e94-4310-835c-15514bd37974","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":7,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks individually","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"}],"position":{"left":454.449951171875,"top":179.014577402342},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"individual macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"individual macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"individual macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"all\\", \\"__current_case__\\": 2}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BED\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"-100\\"}, \\"outputs\\": [\\"bdg\\", \\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"8f2a3aa4-704e-4d90-a672-94b97996dd56","when":null,"workflow_outputs":[{"label":"individual_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"4552df5d-4397-4118-af11-f97eb8c748a5"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":8,"input_connections":{"input_list":{"id":5,"output_name":"outputcnt"}},"inputs":[],"label":"put all nb of reads into single dataset","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":487.0999755859375,"top":1006.4666625976563},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"bd8b6b03-19ef-4921-96c3-abe5e6992941","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","errors":null,"id":9,"input_connections":{"tag|inputs":{"id":7,"output_name":"output_narrowpeaks"}},"inputs":[{"description":"runtime parameter for tool bedtools Multiple Intersect","name":"tag"}],"label":"compute multi intersect","name":"bedtools Multiple Intersect","outputs":[{"name":"output","type":"bed"}],"position":{"left":797.3086613144798,"top":40.02067466199719},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"cluster\\": false, \\"empty\\": {\\"empty_selector\\": \\"\\", \\"__current_case__\\": 1}, \\"filler\\": \\"0\\", \\"header\\": false, \\"tag\\": {\\"tag_select\\": \\"tag\\", \\"__current_case__\\": 0, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"9edd1aab-eb4c-4fd4-b9f6-a6dbcae46dd2","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":10,"input_connections":{"input1":{"id":7,"output_name":"output_treat_pileup"}},"inputs":[],"label":"individual normalized bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":680.6833851669262,"top":635.8069731504326},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"d0d6f92d-3d7e-44c3-b2f6-c10e429d2369","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","errors":null,"id":11,"input_connections":{"singtabop|input":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Table Compute","name":"singtabop"}],"label":"get min value","name":"Table Compute","outputs":[{"name":"table","type":"tabular"}],"position":{"left":719.5,"top":974.7833435058594},"post_job_actions":{"HideDatasetActiontable":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"table"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0","tool_shed_repository":{"changeset_revision":"3bf5661c0280","name":"table_compute","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"out_opts\\": null, \\"precision\\": \\"6\\", \\"singtabop\\": {\\"use_type\\": \\"single\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"col_row_names\\": null, \\"adv\\": {\\"header\\": null, \\"nrows\\": null, \\"skipfooter\\": null, \\"skip_blank_lines\\": true}, \\"user\\": {\\"mode\\": \\"matrixapply\\", \\"__current_case__\\": 3, \\"matrixapply_func\\": {\\"vector_op\\": \\"min\\", \\"__current_case__\\": 1}, \\"dimension\\": \\"0\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.4+galaxy0","type":"tool","uuid":"0d63ffd8-0646-4dbf-9ede-14bffcd5de2d","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"wc_gnu","errors":null,"id":12,"input_connections":{"input1":{"id":8,"output_name":"output"}},"inputs":[],"label":"get nb of replicates","name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":723,"top":1081.7},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"1e4420d0-edfb-4411-b7e0-602b510ff898","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"Filter1","errors":null,"id":13,"input_connections":{"cond":{"id":6,"output_name":"out1"},"input":{"id":9,"output_name":"output"}},"inputs":[],"label":"filter multi intersect","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1065.3086613144797,"top":48.02067466199719},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"4acd5289-87cc-4cfb-9863-35248989ab27","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":14,"input_connections":{"advancedOpt|binSize":{"id":3,"output_name":"output"},"bigwigs":{"id":10,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"},{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"average coverage from replicates","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":938.7350974129533,"top":555.8480470664147},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"Average Normalized Coverage of all replicates"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": \\"1\\", \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"36059130-5f2a-4fe0-8b36-58073595fdd5","when":null,"workflow_outputs":[{"label":"average_bigwig","output_name":"outFileName","uuid":"35dfdb8a-121d-41d1-9ad1-1e0d95a8ab8c"}]},"15":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":15,"input_connections":{"input1":{"id":11,"output_name":"table"}},"inputs":[],"label":"convert min value to text","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":972,"top":918.7},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"},"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"cdbee56b-9fff-4205-9cb6-741be770985d","when":null,"workflow_outputs":[]},"16":{"annotation":"get nb of rep as param","content_id":"param_value_from_file","errors":null,"id":16,"input_connections":{"input1":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":980.362829492841,"top":1090.2503372993485},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"2ea49dca-270b-454a-9d86-6947799ab996","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":17,"input_connections":{"token_set_0|line":{"id":15,"output_name":"text_param"},"token_set_0|repeat_select|times":{"id":16,"output_name":"integer_param"}},"inputs":[],"label":"create a dataset with the min value as many times as there are replicates","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":1210.95940032202,"top":978.1979565079766},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"29ef7038-e10a-47cd-835f-945dccaf9c1c","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":18,"input_connections":{"split_parms|input":{"id":17,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":"split min value","name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":1454.946894305971,"top":1074.616955921246},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"7db836a5-793a-403e-8db1-b97f619f4c69","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":19,"input_connections":{"input1":{"id":18,"output_name":"list_output_txt"}},"inputs":[],"label":"convert min nb of reads to parameter","name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":1679.9454808872808,"top":1066.2435805949858},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"71b1c454-41db-4a8b-8595-8d3fa86a86a0","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":20,"input_connections":{"input":{"id":0,"output_name":"output"},"mode|subsample_config|subsampling_mode|target":{"id":19,"output_name":"integer_param"}},"inputs":[],"label":"select random reads","name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":1925.7976872167908,"top":1061.3309439057557},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": null, \\"library\\": null, \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": null, \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"target\\", \\"__current_case__\\": 1, \\"target\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seed\\": \\"1\\"}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"36713e3f-6110-41f7-8df1-9a9966400fab","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","errors":null,"id":21,"input_connections":{"input":{"id":20,"output_name":"outputsam"}},"inputs":[],"label":"convert subsampled bam to bed","name":"bedtools BAM to BED","outputs":[{"name":"output","type":"bed"}],"position":{"left":1300.0284368938724,"top":585.202166486144},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"ed_score\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"option\\": \\"\\", \\"split\\": false, \\"tag\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"7521eb4d-6cb9-463b-8c03-91ac64396ac3","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":22,"input_connections":{"effective_genome_size_options|gsize":{"id":2,"output_name":"output"},"treatment|input_treatment_file":{"id":21,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"call peaks on merge","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"}],"position":{"left":1546.199951171875,"top":605.2979025976546},"post_job_actions":{"HideDatasetActionoutput_tabular":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"merged macs2 narrowPeaks"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"merged macs2 tabular"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"merged macs2 coverage normalized"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": true, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"all\\", \\"__current_case__\\": 2}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BED\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"-100\\"}, \\"outputs\\": [\\"peaks_tabular\\"], \\"treatment\\": {\\"t_multi_select\\": \\"Yes\\", \\"__current_case__\\": 1, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"e183ec20-d4ab-427a-9a0c-3e9bbced8793","when":null,"workflow_outputs":[{"label":"merged_macs2_narrowPeaks","output_name":"output_narrowpeaks","uuid":"6d97d620-9486-4d85-9a80-e3595c0bb968"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","errors":null,"id":23,"input_connections":{"inputA":{"id":22,"output_name":"output_narrowpeaks"},"reduce_or_iterate|inputB":{"id":13,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools Intersect intervals","name":"reduce_or_iterate"}],"label":"get merged peaks overlapping at least x replicates","name":"bedtools Intersect intervals","outputs":[{"name":"output","type":"input"}],"position":{"left":1643.5166015625,"top":340.697927011717},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed\\": false, \\"count\\": false, \\"fraction_cond\\": {\\"fraction_select\\": \\"default\\", \\"__current_case__\\": 0}, \\"genome_file_opts\\": {\\"genome_file_opts_selector\\": \\"loc\\", \\"__current_case__\\": 0, \\"genome\\": null}, \\"header\\": false, \\"inputA\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": false, \\"once\\": false, \\"overlap_mode\\": [\\"-wa\\", \\"-wb\\"], \\"reduce_or_iterate\\": {\\"reduce_or_iterate_selector\\": \\"iterate\\", \\"__current_case__\\": 0, \\"inputB\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"sorted\\": false, \\"split\\": false, \\"strand\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"d6183a41-f78c-49fc-b826-f8aa95ff84cb","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":24,"input_connections":{"results_0|software_cond|input":{"id":7,"output_name":"output_tabular"},"results_1|software_cond|input":{"id":22,"output_name":"output_tabular"}},"inputs":[],"label":"multiQC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2329.32147475926,"top":153.52074582627785},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"f7d74248-16e3-4020-8156-caae0e7682ef","when":null,"workflow_outputs":[{"label":"multiqc_output","output_name":"html_report","uuid":"a342a984-235e-4d40-9e09-d835aab35282"}]},"25":{"annotation":"","content_id":"Filter1","errors":null,"id":25,"input_connections":{"input":{"id":23,"output_name":"output"}},"inputs":[],"label":"only keep peaks with summits overlapping intersection of at least x replicates","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1874.401877544738,"top":345.7312522070295},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"(c2+c10)>=c12 and (c2+c10)<c13\\", \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"dd65edce-c587-4e3f-9c08-408fe81c557e","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"Cut1","errors":null,"id":26,"input_connections":{"input":{"id":25,"output_name":"out_file1"}},"inputs":[],"label":"keep only columns of narrowPeak","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2104.095825078475,"top":439.8305050651918},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1-c10\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6a58206f-a7b2-4f25-b837-b17185c86c52","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","errors":null,"id":27,"input_connections":{"infile":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":"discard duplicated lines","name":"Unique","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2336.462697917632,"top":460.9788245160598},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"narrowPeak shared by at least x replicates"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_numeric\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"62a36e17-5aea-419d-835d-0fa33dd05e97","when":null,"workflow_outputs":[{"label":"shared_narrowPeak","output_name":"outfile","uuid":"47b85acb-10eb-454c-aa95-1253f223641d"}]}},"tags":["ATAC-seq"],"uuid":"f58371cd-2e2d-41da-b1ca-ef10f7330047","version":13},"readme":"# Consensus Peak Calling for ChIP-seq, ATAC-seq and CUT&RUN Replicates\\n\\nThe goal of this workflow is to get a list of confident peaks with summits from n replicates.\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of datasets with n BAM where PCR duplicates have been removed (the workflow also works for nested list if you have multiple conditions each with multiple replicates).\\n\\n## Inputs values\\n\\n- Minimum number of overlap: Minimum number of replicates into which the final summit should be present.\\n- effective_genome_size: this is used by MACS2 and may be entered manually (indications are provided for heavily used genomes).\\n- bin_size: this is the bin sized used to compute the average of normalized profiles. Large values will allow to have a smaller output file but with less resolution while small values will increase computation time and size of the output file to produce a more resolutive bigwig.\\n\\n## Strategy summary\\n\\nHere is a generated example to highlight the strategy:\\n![strategy](https://raw.githubusercontent.com/galaxyproject/iwc/main/workflows/epigenetics/consensus-peaks/strategy.png)\\n\\n## Processing\\n\\n- The workflow will:\\n  - first part:\\n    - call peaks and compute normalized coverage on each BAM individually\\n    - average normalized profiles\\n    - compute the intersection between all peaks and filter when at least x replicate overlaps\\n  - second part:\\n    - subset all BAM to get the same number of reads\\n    - call peaks on all subsetted BAM combined\\n  - finally, keep only peaks from the second part that have summits overlapping the filtered intersection of the first part.\\n","changelog":"# Changelog\\n\\n## [1.3] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [1.2] - 2024-09-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [1.1] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.31.1+galaxy0`\\n\\n## [1.0] - 2024-05-01\\n\\n### Changed\\nThe workflow changed as it accepts now more than 2 replicates and has a new parameter to define what is the minimum number of replicates where the summit should be found.\\nThe workflow for ATAC and CUT&RUN changed the input datatype as now a BAM is required instead of a BED.\\n\\n## [0.7] - 2024-04-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.6] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1`\\n\\n## [0.5] - 2024-03-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy0`\\n\\n## [0.4] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.3] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n\\n## [0.2] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n\\n## [0.1] - 2023-08-31\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Consensus Peak Calling for ATAC-seq and CUT&RUN Replicates\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ n rmDup BAM\\"]@{ shape: docs }\\n1[\\"ℹ️ Minimum number of overlap\\"]@{ shape: lean-l }\\n2[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n3[\\"ℹ️ bin_size\\"]@{ shape: lean-l }\\n4[\\"convert BAM to BED\\"]@{ shape: process }\\n0 --> 4\\n5[\\"count number of reads\\"]@{ shape: process }\\n0 --> 5\\n6[\\"generate filter rule\\"]@{ shape: process }\\n1 --> 6\\n7[\\"call peaks individually\\"]@{ shape: process }\\n2 --> 7\\n4 --> 7\\n8[\\"put all nb of reads into single dataset\\"]@{ shape: process }\\n5 --> 8\\n9[\\"compute multi intersect\\"]@{ shape: process }\\n7 --> 9\\n10[\\"individual normalized bigwig\\"]@{ shape: process }\\n7 --> 10\\n11[\\"get min value\\"]@{ shape: process }\\n8 --> 11\\n12[\\"get nb of replicates\\"]@{ shape: process }\\n8 --> 12\\n13[\\"filter multi intersect\\"]@{ shape: process }\\n6 --> 13\\n9 --> 13\\n14[\\"average coverage from replicates\\"]@{ shape: process }\\n3 --> 14\\n10 --> 14\\n15[\\"convert min value to text\\"]@{ shape: process }\\n11 --> 15\\n16[\\"Parse parameter value\\"]@{ shape: process }\\n12 --> 16\\n17[\\"create a dataset with the min value as many times as there are replicates\\"]@{ shape: process }\\n15 --> 17\\n16 --> 17\\n18[\\"split min value\\"]@{ shape: process }\\n17 --> 18\\n19[\\"convert min nb of reads to parameter\\"]@{ shape: process }\\n18 --> 19\\n20[\\"select random reads\\"]@{ shape: process }\\n0 --> 20\\n19 --> 20\\n21[\\"convert subsampled bam to bed\\"]@{ shape: process }\\n20 --> 21\\n22[\\"call peaks on merge\\"]@{ shape: process }\\n2 --> 22\\n21 --> 22\\n23[\\"get merged peaks overlapping at least x replicates\\"]@{ shape: process }\\n22 --> 23\\n13 --> 23\\n24[\\"multiQC\\"]@{ shape: process }\\n7 --> 24\\n22 --> 24\\n25[\\"only keep peaks with summits overlapping intersection of at least x replicates\\"]@{ shape: process }\\n23 --> 25\\n26[\\"keep only columns of narrowPeak\\"]@{ shape: process }\\n25 --> 26\\n27[\\"discard duplicated lines\\"]@{ shape: process }\\n26 --> 27\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/consensus-peaks/consensus-peaks-atac-cutandrun","dockstore_id":23499,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078828","tests":[{"doc":"Test for consensus-peaks-atac-cutandrun.ga","job":{"n rmDup BAM":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"rep1","path":"test-data/rep1.bam","dbkey":"mm10"},{"class":"File","identifier":"rep2","path":"test-data/rep2.bam","dbkey":"mm10"},{"class":"File","identifier":"rep3","path":"test-data/rep3.bam","dbkey":"mm10"}]},"Minimum number of overlap":2,"effective_genome_size":1870000000,"bin_size":50},"outputs":{"individual_macs2_narrowPeaks":{"element_tests":{"rep1":{"asserts":{"has_n_lines":{"n":4}}},"rep2":{"asserts":{"has_n_lines":{"n":6}}},"rep3":{"asserts":{"has_n_lines":{"n":4}}}}},"average_bigwig":{"asserts":{"has_size":{"value":1388,"delta":100}}},"merged_macs2_narrowPeaks":{"asserts":{"has_n_lines":{"n":8}}},"multiqc_output":{"asserts":{"has_text":{"text":"MACS2: Fragment Length"}}},"shared_narrowPeak":{"asserts":{"has_n_lines":{"n":4}}}}}]}],"path":"./workflows/epigenetics/consensus-peaks"},{"version":1.2,"workflows":[{"name":"hic-fastq-to-cool-hicup-cooler","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/hic-fastq-to-cool-hicup-cooler.ga","testParameterFiles":["/hic-fastq-to-cool-hicup-cooler-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Comprehensive Hi-C data processing workflow that converts paired-end FASTQ files to balanced contact matrices. Uses HiCUP for pre-processing (adapter trimming, mapping, filtering) with fragment midpoint coordinate assignment, filters reads by mapping quality, and generates sorted tabix files. The final output is a balanced cool file at user-specified resolution created with cooler, ready for multi-resolution contact matrix analysis.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"Hi-C Processing: FASTQ to Balanced Cool Files","steps":{"0":{"annotation":"Should be a paired collection with Hi-C fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with Hi-C fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":101.53334045410156},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"30fe3d0f-541a-478a-b57d-a43c0c16ccad","when":null,"workflow_outputs":[]},"1":{"annotation":"only use genome ids which have bowtie2 indexes","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"only use genome ids which have bowtie2 indexes","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":39.01666259765625,"top":171.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"45194ed6-a1e9-4248-8d3a-f51febc36d61","when":null,"workflow_outputs":[]},"2":{"annotation":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","name":"Restriction enzyme"}],"label":"Restriction enzyme","name":"Input parameter","outputs":[],"position":{"left":71.98333740234375,"top":246.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec011ceb-c601-464f-8661-ca9c0b335bb5","when":null,"workflow_outputs":[]},"3":{"annotation":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","name":"No fill-in"}],"label":"No fill-in","name":"Input parameter","outputs":[],"position":{"left":98.433349609375,"top":332},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"11e64d82-c022-48b2-b616-076bfcdb14ad","when":null,"workflow_outputs":[]},"4":{"annotation":"can be set to 0 for no filtering","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"can be set to 0 for no filtering","name":"minimum MAPQ"}],"label":"minimum MAPQ","name":"Input parameter","outputs":[],"position":{"left":143.9666748046875,"top":401.5333251953125},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"748c8800-5bef-41b7-b2cf-6d6a57c0eb70","when":null,"workflow_outputs":[]},"5":{"annotation":"For example 10000 for 10kb","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"For example 10000 for 10kb","name":"Bin size in bp"}],"label":"Bin size in bp","name":"Input parameter","outputs":[],"position":{"left":186,"top":491.0333251953125},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1e019ddf-171c-4247-ab11-0d2cdbe10f6a","when":null,"workflow_outputs":[]},"6":{"annotation":"Nothing means genome-wide, \'--cis-only\' means only Cis interactions, \'--trans-only\' means only Trans interactions","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Nothing means genome-wide, \'--cis-only\' means only Cis interactions, \'--trans-only\' means only Trans interactions","name":"Interactions to consider to calculate weights in normalization step"}],"label":"Interactions to consider to calculate weights in normalization step","name":"Input parameter","outputs":[],"position":{"left":216.9666748046875,"top":558.25},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"\\", \\"--cis-only\\", \\"--trans-only\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"51990231-540d-4fbc-bf77-bbbb4d1fc461","when":null,"workflow_outputs":[]},"7":{"annotation":"You can use your region of interest or a region known for high level of insulation for example HoxD locus:\\nhg19: chr2:175,689,620-178,604,461\\nhg38: chr2:174,692,032-177,585,317\\nmm10: chr2:73,779,626-75,669,724\\nmm39: ","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"You can use your region of interest or a region known for high level of insulation for example HoxD locus:\\nhg19: chr2:175,689,620-178,604,461\\nhg38: chr2:174,692,032-177,585,317\\nmm10: chr2:73,779,626-75,669,724\\nmm39: ","name":"region for matrix plotting"}],"label":"region for matrix plotting","name":"Input parameter","outputs":[],"position":{"left":282,"top":710},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ea792308-f30e-49c4-9fec-8de03f45f64c","when":null,"workflow_outputs":[]},"8":{"annotation":"","id":8,"input_connections":{"No fill-in":{"id":3,"input_subworkflow_step_id":3,"output_name":"output"},"PE fastq input":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"Restriction enzyme":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"},"genome name":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"},"minimum MAPQ":{"id":4,"input_subworkflow_step_id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Hi-C_fastqToPairs_hicup","outputs":[],"position":{"left":489.01666259765625,"top":0},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Hi-C_fastqToPairs_hicup","steps":{"0":{"annotation":"Should be a paired collection with Hi-C fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with Hi-C fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":12.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"30fe3d0f-541a-478a-b57d-a43c0c16ccad","when":null,"workflow_outputs":[]},"1":{"annotation":"use the bowtie2 indexes to display choices","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"use the bowtie2 indexes to display choices","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":39,"top":82.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"45194ed6-a1e9-4248-8d3a-f51febc36d61","when":null,"workflow_outputs":[]},"2":{"annotation":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","name":"Restriction enzyme"}],"label":"Restriction enzyme","name":"Input parameter","outputs":[],"position":{"left":58,"top":155.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec011ceb-c601-464f-8661-ca9c0b335bb5","when":null,"workflow_outputs":[]},"3":{"annotation":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","name":"No fill-in"}],"label":"No fill-in","name":"Input parameter","outputs":[],"position":{"left":97.5,"top":248.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c807fdbf-1c1d-45b4-851e-47fb919b07d6","when":null,"workflow_outputs":[]},"4":{"annotation":"can be set to 0 for no filtering","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"can be set to 0 for no filtering","name":"minimum MAPQ"}],"label":"minimum MAPQ","name":"Input parameter","outputs":[],"position":{"left":131,"top":320.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"748c8800-5bef-41b7-b2cf-6d6a57c0eb70","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","errors":null,"id":5,"input_connections":{"advanced_options|nofill":{"id":3,"output_name":"output"},"genome":{"id":1,"output_name":"output"},"library|input_1":{"id":0,"output_name":"output"},"re1":{"id":2,"output_name":"output"},"reference_genome|index":{"id":1,"output_name":"output"}},"inputs":[],"label":"HiCUP","name":"Hicup Pipeline","outputs":[{"name":"intermediate_results","type":"input"},{"name":"hicup_results","type":"html"},{"name":"hicup_report","type":"tabular"},{"name":"dataset_hicup","type":"qname_sorted.bam"},{"name":"digester_file","type":"tabular"}],"position":{"left":356.5,"top":0},"post_job_actions":{"HideDatasetActiondataset_hicup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"dataset_hicup"},"HideDatasetActiondigester_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"digester_file"},"HideDatasetActionintermediate_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"intermediate_results"},"RenameDatasetActionhicup_report":{"action_arguments":{"newname":"HiCUP report (tabular)"},"action_type":"RenameDatasetAction","output_name":"hicup_report"},"RenameDatasetActionhicup_results":{"action_arguments":{"newname":"HiCUP report (HTML)"},"action_type":"RenameDatasetAction","output_name":"hicup_results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"68fb19094f84","name":"hicup_hicup","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"re2\\": \\"\\", \\"longest\\": \\"\\", \\"shortest\\": \\"\\", \\"nofill\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"re1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"8aba6fbb-d940-4826-911b-8c148e82085c","when":null,"workflow_outputs":[{"label":"HiCUP report (HTML)","output_name":"hicup_results","uuid":"ba37bb7a-ad3a-487b-b7ba-0023f1c496fd"},{"label":"HiCUP report (tabular)","output_name":"hicup_report","uuid":"fd8f1b17-c68e-4652-b139-acccf51baa30"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"},"components_3|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":"build filtering rule for MAPQ","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":746.5,"top":392.5},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"(c10>=\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\") and (c11>=\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\")\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"ee70cd21-6a36-4e21-97d2-ff0749236523","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","errors":null,"id":7,"input_connections":{"digester|digester_file":{"id":5,"output_name":"digester_file"},"input_file":{"id":5,"output_name":"dataset_hicup"}},"inputs":[],"label":"valid pairs in juicebox format","name":"Hicup to juicer converter","outputs":[{"name":"output","type":"tabular"}],"position":{"left":637.5,"top":197.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"valid pairs in juicebox format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"b4e7244246e2","name":"hicup2juicer","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"digester\\": {\\"provide_digester\\": \\"yes\\", \\"__current_case__\\": 0, \\"digester_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"usemid\\": true}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"659811cd-a212-4949-9693-074f6161ed11","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format","output_name":"output","uuid":"dfc67c68-2569-4727-9ca6-a70f3b13edd9"}]},"8":{"annotation":"","content_id":"Filter1","errors":null,"id":8,"input_connections":{"cond":{"id":6,"output_name":"out1"},"input":{"id":7,"output_name":"output"}},"inputs":[],"label":"valid pairs in juicebox format MAPQ filtered","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1041,"top":304.5},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"valid pairs in juicebox format MAPQ filtered"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"fe9a084d-06ad-4671-a4d6-85dff2f96138","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format MAPQ filtered","output_name":"out_file1","uuid":"b08373c7-59b2-4ae3-90b7-7c9a05ca81b7"}]}},"tags":"","uuid":"f6bb48d4-0392-4c89-825e-01b8523b0943"},"tool_id":null,"type":"subworkflow","uuid":"33846b88-4c1c-4c90-9ff9-63cdb29e9587","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format","output_name":"valid pairs in juicebox format","uuid":"4dffbfdd-cd23-435c-8fac-0cf7d202d8a1"},{"label":"HiCUP report (txt)","output_name":"HiCUP report (tabular)","uuid":"38b61150-fc80-43e3-86c2-027562e128da"},{"label":"valid pairs in juicebox format MAPQ filtered","output_name":"valid pairs in juicebox format MAPQ filtered","uuid":"c82b20be-667c-4085-9619-dba3b551adf3"},{"label":"HiCUP report (html)","output_name":"HiCUP report (HTML)","uuid":"ea4b99e0-c0f4-491a-8992-62d177b60c23"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_csort_tabix/cooler_csort_tabix/0.9.3+galaxy1","errors":null,"id":9,"input_connections":{"input":{"id":8,"output_name":"valid pairs in juicebox format"},"size_source|fasta_cached":{"id":1,"output_name":"output"}},"inputs":[],"label":"Sort pairs and index","name":"cooler csort with tabix","outputs":[{"name":"output","type":"interval_tabix.gz"}],"position":{"left":742.0333251953125,"top":367.5},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"juicer_medium_tabix.gz"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"valid pairs filtered and sorted"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_csort_tabix/cooler_csort_tabix/0.9.3+galaxy1","tool_shed_repository":{"changeset_revision":"8c3cba559c99","name":"cooler_csort_tabix","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"format_sel\\": {\\"format\\": \\"juicer_medium\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"size_source\\": {\\"size_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"fasta_cached\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy1","type":"tool","uuid":"bbb479cd-c67c-4347-9c1d-ac8095d90aa9","when":null,"workflow_outputs":[{"label":"valid pairs filtered and sorted","output_name":"output","uuid":"8f590490-b10c-441a-90f5-77110c7bab77"}]},"10":{"annotation":"","id":10,"input_connections":{"Bin size in bp":{"id":5,"input_subworkflow_step_id":0,"output_name":"output"},"Interactions to consider to calculate weights in normalization step":{"id":6,"input_subworkflow_step_id":3,"output_name":"output"},"Juicer Medium Tabix with validPairs":{"id":9,"input_subworkflow_step_id":2,"output_name":"output"},"genome name":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Hi-C_juicermediumtabixToCool_cooler","outputs":[],"position":{"left":1002.0166015625,"top":441.5},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Hi-C_juicermediumtabixToCool_cooler","steps":{"0":{"annotation":"For example 10000 for 10kb","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"For example 10000 for 10kb","name":"Bin size in bp"}],"label":"Bin size in bp","name":"Input parameter","outputs":[],"position":{"left":0,"top":33.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d0a82863-82f8-426c-addb-06066e9d0a73","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"fd697a0a-0844-470c-a6b3-c96e3a0247ba"}]},"1":{"annotation":"This is used to get the chromosome sizes","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This is used to get the chromosome sizes","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":46,"top":123.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"48506f80-b1b5-4d1d-98a6-c6882de026f1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"ad0f1061-11da-491d-a2d8-bf4cb7174fde"}]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Juicer Medium Tabix with validPairs"}],"label":"Juicer Medium Tabix with validPairs","name":"Input dataset collection","outputs":[],"position":{"left":129.25,"top":208.25},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"588962a4-8ce4-4929-81fd-a9dfdd032598","when":null,"workflow_outputs":[]},"3":{"annotation":"Recommended value: cis-only","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Recommended value: cis-only","name":"Interactions to consider to calculate weights in normalization step"}],"label":"Interactions to consider to calculate weights in normalization step","name":"Input parameter","outputs":[],"position":{"left":221,"top":309.75},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d6375500-a591-47a9-b126-950347e3e7a8","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"6c998f49-38e8-4863-9908-a51e0915965a"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","errors":null,"id":4,"input_connections":{"binsize":{"id":0,"output_name":"output"},"size_source|fasta_cached":{"id":1,"output_name":"output"}},"inputs":[],"label":"make bed with bins","name":"cooler_makebins","outputs":[{"name":"output","type":"bed"}],"position":{"left":489.25,"top":0},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"ebd2a52eb245","name":"cooler_makebins","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"binsize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"size_source\\": {\\"size_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"fasta_cached\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"34e0a213-93e8-466e-b07b-ede1443ee7d6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","errors":null,"id":5,"input_connections":{"assembly":{"id":1,"output_name":"output"},"format_sel|input_pairs":{"id":2,"output_name":"output"},"input_bed":{"id":4,"output_name":"output"}},"inputs":[],"label":"Load pairs in matrix","name":"cooler_cload_tabix","outputs":[{"name":"output","type":"cool"}],"position":{"left":762.25,"top":58},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with raw values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","tool_shed_repository":{"changeset_revision":"0641e1c892c6","name":"cooler_cload_tabix","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assembly\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"format_sel\\": {\\"format\\": \\"juicer_medium\\", \\"__current_case__\\": 0, \\"input_pairs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy1","type":"tool","uuid":"d411eea7-7996-44f2-af2a-feecb854a067","when":null,"workflow_outputs":[{"label":"matrix with raw values","output_name":"output","uuid":"d398b493-9b5d-412f-a06d-d1cfdc904feb"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","errors":null,"id":6,"input_connections":{"cistrans":{"id":3,"output_name":"output"},"input":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool cooler_balance","name":"blacklist"}],"label":"ICE normalization","name":"cooler_balance","outputs":[{"name":"output","type":"cool"}],"position":{"left":991.25,"top":254.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with iced values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"8f9fe0667b89","name":"cooler_balance","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"blacklist\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"cistrans\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"convergencepolicy\\": \\"error\\", \\"ignorediags\\": \\"2\\", \\"ignoredist\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"madmax\\": \\"5\\", \\"maxiters\\": \\"200\\", \\"mincount\\": \\"0\\", \\"minnnz\\": \\"10\\", \\"name\\": \\"weight\\", \\"tol\\": \\"1e-05\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"44270887-1c96-4ede-8bb9-3db2a07eeb6e","when":null,"workflow_outputs":[{"label":"matrix with iced values","output_name":"output","uuid":"1d4fe955-922e-40bf-b879-a71290af3ea7"}]}},"tags":"","uuid":"d7fb120d-41f4-4a67-a6fe-2885dcd33ebf"},"tool_id":null,"type":"subworkflow","uuid":"c4d87b28-5da7-48cb-aa49-cab59a0cb19b","when":null,"workflow_outputs":[{"label":"matrix with raw values","output_name":"matrix with raw values","uuid":"140f460e-a1b0-4868-ac7e-5605c78e18a9"},{"label":"matrix with iced values","output_name":"matrix with iced values","uuid":"b7485ffe-ffcd-4e03-b8ad-7e0d02b05c17"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy2","errors":null,"id":11,"input_connections":{"region":{"id":7,"output_name":"output"},"tracks_0|track_file_style_conditional|matrix_h5_cooler_multiple":{"id":10,"output_name":"matrix with iced values"}},"inputs":[],"label":"final plot","name":"pyGenomeTracks","outputs":[{"name":"outFileName","type":"png"}],"position":{"left":1295,"top":693},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"plot with pyGenomeTracks"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy2","tool_shed_repository":{"changeset_revision":"59fd173ac850","name":"pygenometracks","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"global_args\\": {\\"title\\": \\"\\", \\"fontsize\\": \\"12\\", \\"dpi\\": \\"72\\", \\"width\\": \\"40.0\\", \\"plotWidth\\": null, \\"height\\": null, \\"trackLabelFraction\\": \\"0.05\\", \\"trackLabelHAlign\\": \\"left\\", \\"decreasingXAxis\\": false}, \\"image_file_format\\": \\"png\\", \\"region\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tracks\\": [{\\"__index__\\": 0, \\"track_file_style_conditional\\": {\\"track_file_style_selector\\": \\"hic_matrix_option\\", \\"__current_case__\\": 0, \\"title\\": \\"\\", \\"matrix_h5_cooler_multiple\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"colormap\\": \\"RdYlBu_r\\", \\"min_value\\": null, \\"max_value\\": null, \\"depth\\": \\"8000000\\", \\"transform\\": \\"no\\", \\"height_matrix\\": null, \\"show_masked_bins\\": false, \\"boundaries_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"scale_factor\\": \\"1.0\\", \\"rasterize\\": true, \\"invert_orientation\\": false, \\"spacer_height\\": null}}, {\\"__index__\\": 1, \\"track_file_style_conditional\\": {\\"track_file_style_selector\\": \\"xaxis_option\\", \\"__current_case__\\": 13, \\"title\\": \\"\\", \\"fontsize\\": null, \\"xaxis_where\\": \\"bottom\\", \\"spacer_height\\": null}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.8+galaxy2","type":"tool","uuid":"d8b80397-70dc-4522-8a12-948b1107ac8d","when":null,"workflow_outputs":[{"label":"plot with pyGenomeTracks","output_name":"outFileName","uuid":"27b9c4ea-2a6a-4477-bd5e-7f69a89d69fa"}]}},"tags":["Hi-C"],"uuid":"2196fbff-317e-437d-a022-696a1f95a850","version":4},"readme":"# (Capture) Hi-C Processing: FASTQ to Balanced Cool Files\\n\\nThis can also be used for Hi-ChIP experiments, in that case the output with `matrix with iced values` is ignored and the matrix to use is `matrix with raw values`.\\n\\n## Input datasets\\n\\n- The workflow needs a list of dataset pairs of fastqsanger.\\n\\n## Input values\\n\\n- genome name: suggested from the bowtie2 indices, it is used to map and build the list of bins.\\n- restriction enzyme: Restriction enzyme used e.g. A^GATCT,BglII. The \'^\' is used to express where the enzyme cuts.\\n- No fill-in: If you used a biotin fill-in protocol, put this to false, else, put it to true.\\n- minimum MAPQ: Filtering to apply to pairs you want to keep in your matrix, set it to 0 to not apply filtering (HiCUP already filter for uniquely mapped or MAPQ30).\\n- Bin size in bp: Used to generate your first matrix but you will be able to rerun the subworkflow `hic_tabix_to_cool_cooler` to get other resolutions.\\n- Interactions to consider to calculate weights in normalization step: this is a parameter for the last correction step (ICE).\\n\\nFor the region capture workflow:\\n\\n- chromosome, start and end positions of the capture region\\n\\nFor the Hi-C workflow:\\n\\n- region to use in pyGenomeTracks to check the matrices.\\n\\n## Processing\\n\\n- Reads are processed with HiCUP which comprises these steps:\\n  - Truncation of reads for the religation motif\\n  - Mapping of mates independently with bowtie2\\n  - Pairing the mates when both mates are uniquely mapped or MAPQ30\\n  - Filtering the pairs for undigested, self-ligated...\\n  - Removing duplicates\\n- The output BAM file is converted to medium juicer format: `<readname> <str1> <chr1> <pos1> <frag1> <str2> <chr2> <pos2> <frag2> <mapq1> <mapq2>` where str = strand (0 for forward, anything else for reverse) and pos is the middle of the fragment.\\n- The pairs are filtered for MAPQ if specified.\\n- For the region capture Hi-C workflow the pairs are filtered for both mates in the captured region.\\n- The filtered pairs are sorted and indexed with cooler_csort.\\n- The pairs are loaded into a matrix of the given resolution and balanced with cooler.\\n- A final plot is made with pyGenomeTracks using the balanced matrices on the region provided or the capture region.\\n\\n## Subworkflows\\n\\nThere are 2 subworkflows: `hic_tabix_to_cool_cooler` and `hic_fastq_to_pairs_hicup.ga`.\\n\\n### hic_tabix_to_cool_cooler\\n\\nThis first subworkflow can be used to generate matrices to different resolutions using one of the output of the full workflow (`valid pairs filtered and sorted`).\\n\\nIf the dataset are still in galaxy (format: juicer_medium_tabix.gz), the workflow can be run directly.\\n\\nIf the dataset is not anymore in galaxy, you need to upload and specify the datatype as: juicer_medium_tabix.gz\\n\\n### hic_fastq_to_pairs_hicup\\n\\nThe second subworkflow has no real reason to be launched by itself except for QC tests.\\n\\nIf you want to run the first subworkflow from these results:\\n\\n- You first need to filter the pairs (`valid pairs in juicebox format MAPQ filtered`) for the capture region if relevent using the tool Filter1 (**Filter** data on any column using simple expressions) with the condition `(c3==\'chr2\' and c4<180000000 and c4>170000000) and (c7==\\"chr2\\" and c8<180000000 and c8>170000000)` if your capture region is chr2:170000000-180000000.\\n- Then you need to run cooler_csort (**cooler csort with tabix** Sort and index a contact list.) with as input the `valid pairs in juicebox format MAPQ filtered` or the output of the previous step and for \\"Format of your input file\\" use \\"Juicer Medium Format\\".\\n\\nThe output of `cooler_csort` can be used as input of the first subworkflow.\\n","changelog":"# Changelog\\n\\n## [0.3] - 2023-09-08\\n\\n### Update tools\\n- all cooler tools were updated from 0.8.11 to 0.9.3\\n- pyGenomeTracks was updated from 3.7 to 3.8\\n\\n## [0.2.1] - 2023-09-01 Bug fix\\n\\nFix dockstore (workflows are the same)\\n\\n## [0.2] - 2023-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0`\\n\\n### Manual update\\n- Get fragment id in valid pairs file.\\n- Use the middle of the fragment instead of 5\' of the read in the valid pairs file.\\n\\n## [0.1] - 2023-01-16\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Hi-C Processing: FASTQ to Balanced Cool Files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Restriction enzyme\\"]@{ shape: lean-l }\\n3[\\"ℹ️ No fill-in\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum MAPQ\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Bin size in bp\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Interactions to consider to calculate weights in normalization step\\"]@{ shape: lean-l }\\n7[\\"ℹ️ region for matrix plotting\\"]@{ shape: lean-l }\\n8[\\"🛠️ Hi-C_fastqToPairs_hicup\\"]@{ shape: subprocess }\\n3 --> 8\\n0 --> 8\\n2 --> 8\\n1 --> 8\\n4 --> 8\\n9[\\"Sort pairs and index\\"]@{ shape: process }\\n8 --> 9\\n1 --> 9\\n10[\\"🛠️ Hi-C_juicermediumtabixToCool_cooler\\"]@{ shape: subprocess }\\n5 --> 10\\n6 --> 10\\n9 --> 10\\n1 --> 10\\n11[\\"final plot\\"]@{ shape: process }\\n7 --> 11\\n10 --> 11\\n```\\n\\n## Hi-C_juicermediumtabixToCool_cooler\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Bin size in bp\\"]@{ shape: lean-l }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Juicer Medium Tabix with validPairs\\"]@{ shape: docs }\\n3[\\"ℹ️ Interactions to consider to calculate weights in normalization step\\"]@{ shape: lean-l }\\n4[\\"make bed with bins\\"]@{ shape: process }\\n0 --> 4\\n1 --> 4\\n5[\\"Load pairs in matrix\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n4 --> 5\\n6[\\"ICE normalization\\"]@{ shape: process }\\n3 --> 6\\n5 --> 6\\n```\\n\\n## Hi-C_fastqToPairs_hicup\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Restriction enzyme\\"]@{ shape: lean-l }\\n3[\\"ℹ️ No fill-in\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum MAPQ\\"]@{ shape: lean-l }\\n5[\\"HiCUP\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n0 --> 5\\n2 --> 5\\n1 --> 5\\n6[\\"build filtering rule for MAPQ\\"]@{ shape: process }\\n4 --> 6\\n4 --> 6\\n7[\\"valid pairs in juicebox format\\"]@{ shape: process }\\n5 --> 7\\n5 --> 7\\n8[\\"valid pairs in juicebox format MAPQ filtered\\"]@{ shape: process }\\n6 --> 8\\n7 --> 8\\n```\\n","updated":"2023-09-08T00:00:00","trsID":"#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-cool-hicup-cooler","dockstore_id":20777,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078491","tests":[{"doc":"Test outline for Hi-C_fastqToCool_hicup_cooler","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"test_dataset","elements":[{"class":"File","identifier":"forward","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset1.fastq","filetype":"fastqsanger"},{"class":"File","identifier":"reverse","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset2.fastq","filetype":"fastqsanger"}]}]},"genome name":"hg19","Restriction enzyme":"A^AGCTT,HindIII","No fill-in":"false","minimum MAPQ":30,"Bin size in bp":1000000,"Interactions to consider to calculate weights in normalization step":"","region for matrix plotting":"chr2:175,689,620-178,604,461"},"outputs":{"HiCUP report (html)":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":4602173,"delta":4000000}}}}},"HiCUP report (txt)":{"element_tests":{"test_dataset":{"asserts":{"has_text":{"text":"\\t99742\\t99742\\t92512\\t92628\\t7230\\t7114\\t22.45\\t22.63\\t2658\\t2476\\t73431\\t72142\\t17767\\t18475\\t5886\\t6649\\t57671\\t57671\\t39966\\t1652\\t17997\\t20317\\t17705\\t481\\t2432\\t13452\\t1340\\t0\\t0\\t39962\\t1652\\t17996\\t20314\\t57.82\\t69.30\\t99.99\\t50.83\\t40.07"}}}}},"valid pairs in juicebox format":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"has_line","line":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}},"valid pairs in juicebox format MAPQ filtered":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"not_has_text","text":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}},"valid pairs filtered and sorted":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":735227,"delta":70000}}}}},"matrix with raw values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":117588,"delta":10000}}}}},"matrix with iced values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":149349,"delta":10000}}}}},"plot with pyGenomeTracks":{"file":"test-data/plot_hic.png","compare":"sim_size","delta":1500}}}]},{"name":"chic-fastq-to-cool-hicup-cooler","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/chic-fastq-to-cool-hicup-cooler.ga","testParameterFiles":["/chic-fastq-to-cool-hicup-cooler-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"End-to-end processing of Capture Hi-C paired-end sequencing data. Transforms raw FASTQ files into balanced contact matrices using HiCUP for processing Hi-C reads (adapter trimming, mapping, filtering for valid pairs) with additional filtering for MAPQ and captured regions. The workflow generates sorted tabix files via cooler and produces balanced cool files at user-specified resolution for downstream analysis and visualization.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"Capture Hi-C Processing: FASTQ to Balanced Cool Files","steps":{"0":{"annotation":"Should be a paired collection with Hi-C fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with Hi-C fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":39.12167999999406},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"30fe3d0f-541a-478a-b57d-a43c0c16ccad","when":null,"workflow_outputs":[]},"1":{"annotation":"only use genome ids which have bowtie2 indexes","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"only use genome ids which have bowtie2 indexes","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":39,"top":109.12167999999406},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"45194ed6-a1e9-4248-8d3a-f51febc36d61","when":null,"workflow_outputs":[]},"2":{"annotation":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","name":"Restriction enzyme"}],"label":"Restriction enzyme","name":"Input parameter","outputs":[],"position":{"left":72,"top":184.12167999999406},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec011ceb-c601-464f-8661-ca9c0b335bb5","when":null,"workflow_outputs":[]},"3":{"annotation":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","name":"No fill-in"}],"label":"No fill-in","name":"Input parameter","outputs":[],"position":{"left":98.5,"top":269.62167999999406},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"11e64d82-c022-48b2-b616-076bfcdb14ad","when":null,"workflow_outputs":[]},"4":{"annotation":"can be set to 0 for no filtering","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"can be set to 0 for no filtering","name":"minimum MAPQ"}],"label":"minimum MAPQ","name":"Input parameter","outputs":[],"position":{"left":144,"top":339.12167999999406},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"748c8800-5bef-41b7-b2cf-6d6a57c0eb70","when":null,"workflow_outputs":[]},"5":{"annotation":"For example 10000 for 10kb","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"For example 10000 for 10kb","name":"Bin size in bp"}],"label":"Bin size in bp","name":"Input parameter","outputs":[],"position":{"left":173,"top":409.62167999999406},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1e019ddf-171c-4247-ab11-0d2cdbe10f6a","when":null,"workflow_outputs":[]},"6":{"annotation":"Nothing means genome-wide, \'--cis-only\' means only Cis interactions, \'--trans-only\' means only Trans interactions","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Nothing means genome-wide, \'--cis-only\' means only Cis interactions, \'--trans-only\' means only Trans interactions","name":"Interactions to consider to calculate weights in normalization step"}],"label":"Interactions to consider to calculate weights in normalization step","name":"Input parameter","outputs":[],"position":{"left":236,"top":473.87167999999406},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"\\", \\"--cis-only\\", \\"--trans-only\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"2cf7902e-aa51-45a7-ae75-9107bbb41911","when":null,"workflow_outputs":[]},"7":{"annotation":"For example: chr2","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"For example: chr2","name":"capture region (chromosome)"}],"label":"capture region (chromosome)","name":"Input parameter","outputs":[],"position":{"left":279.5,"top":622.6216799999941},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"386a8e30-aab6-4168-877e-581b883928e5","when":null,"workflow_outputs":[]},"8":{"annotation":"For example: 72402000","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"For example: 72402000","name":"capture region (start)"}],"label":"capture region (start)","name":"Input parameter","outputs":[],"position":{"left":333.5,"top":723.6216799999941},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"47de292d-7499-407e-bf9a-b67a4dc47348","when":null,"workflow_outputs":[]},"9":{"annotation":"For example: 77000000","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"For example: 77000000","name":"capture region (end)"}],"label":"capture region (end)","name":"Input parameter","outputs":[],"position":{"left":397.5,"top":835.6216799999941},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"25480973-63be-4e64-9ba1-26088b4a65c4","when":null,"workflow_outputs":[]},"10":{"annotation":"","id":10,"input_connections":{"No fill-in":{"id":3,"input_subworkflow_step_id":3,"output_name":"output"},"PE fastq input":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"Restriction enzyme":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"},"genome name":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"},"minimum MAPQ":{"id":4,"input_subworkflow_step_id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Hi-C_fastqToPairs_hicup","outputs":[],"position":{"left":520,"top":0},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Hi-C_fastqToPairs_hicup","steps":{"0":{"annotation":"Should be a paired collection with Hi-C fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with Hi-C fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":12.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"30fe3d0f-541a-478a-b57d-a43c0c16ccad","when":null,"workflow_outputs":[]},"1":{"annotation":"use the bowtie2 indexes to display choices","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"use the bowtie2 indexes to display choices","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":39,"top":82.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"45194ed6-a1e9-4248-8d3a-f51febc36d61","when":null,"workflow_outputs":[]},"2":{"annotation":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","name":"Restriction enzyme"}],"label":"Restriction enzyme","name":"Input parameter","outputs":[],"position":{"left":58,"top":155.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec011ceb-c601-464f-8661-ca9c0b335bb5","when":null,"workflow_outputs":[]},"3":{"annotation":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","name":"No fill-in"}],"label":"No fill-in","name":"Input parameter","outputs":[],"position":{"left":97.5,"top":248.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c807fdbf-1c1d-45b4-851e-47fb919b07d6","when":null,"workflow_outputs":[]},"4":{"annotation":"can be set to 0 for no filtering","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"can be set to 0 for no filtering","name":"minimum MAPQ"}],"label":"minimum MAPQ","name":"Input parameter","outputs":[],"position":{"left":131,"top":320.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"748c8800-5bef-41b7-b2cf-6d6a57c0eb70","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","errors":null,"id":5,"input_connections":{"advanced_options|nofill":{"id":3,"output_name":"output"},"genome":{"id":1,"output_name":"output"},"library|input_1":{"id":0,"output_name":"output"},"re1":{"id":2,"output_name":"output"},"reference_genome|index":{"id":1,"output_name":"output"}},"inputs":[],"label":"HiCUP","name":"Hicup Pipeline","outputs":[{"name":"intermediate_results","type":"input"},{"name":"hicup_results","type":"html"},{"name":"hicup_report","type":"tabular"},{"name":"dataset_hicup","type":"qname_sorted.bam"},{"name":"digester_file","type":"tabular"}],"position":{"left":356.5,"top":0},"post_job_actions":{"HideDatasetActiondataset_hicup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"dataset_hicup"},"HideDatasetActiondigester_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"digester_file"},"HideDatasetActionintermediate_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"intermediate_results"},"RenameDatasetActionhicup_report":{"action_arguments":{"newname":"HiCUP report (tabular)"},"action_type":"RenameDatasetAction","output_name":"hicup_report"},"RenameDatasetActionhicup_results":{"action_arguments":{"newname":"HiCUP report (HTML)"},"action_type":"RenameDatasetAction","output_name":"hicup_results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"68fb19094f84","name":"hicup_hicup","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"re2\\": \\"\\", \\"longest\\": \\"\\", \\"shortest\\": \\"\\", \\"nofill\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"re1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"8aba6fbb-d940-4826-911b-8c148e82085c","when":null,"workflow_outputs":[{"label":"HiCUP report (HTML)","output_name":"hicup_results","uuid":"ba37bb7a-ad3a-487b-b7ba-0023f1c496fd"},{"label":"HiCUP report (tabular)","output_name":"hicup_report","uuid":"fd8f1b17-c68e-4652-b139-acccf51baa30"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"},"components_3|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":"build filtering rule for MAPQ","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":746.5,"top":392.5},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"(c10>=\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\") and (c11>=\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\")\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"ee70cd21-6a36-4e21-97d2-ff0749236523","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","errors":null,"id":7,"input_connections":{"digester|digester_file":{"id":5,"output_name":"digester_file"},"input_file":{"id":5,"output_name":"dataset_hicup"}},"inputs":[],"label":"valid pairs in juicebox format","name":"Hicup to juicer converter","outputs":[{"name":"output","type":"tabular"}],"position":{"left":637.5,"top":197.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"valid pairs in juicebox format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"b4e7244246e2","name":"hicup2juicer","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"digester\\": {\\"provide_digester\\": \\"yes\\", \\"__current_case__\\": 0, \\"digester_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"usemid\\": true}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"659811cd-a212-4949-9693-074f6161ed11","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format","output_name":"output","uuid":"dfc67c68-2569-4727-9ca6-a70f3b13edd9"}]},"8":{"annotation":"","content_id":"Filter1","errors":null,"id":8,"input_connections":{"cond":{"id":6,"output_name":"out1"},"input":{"id":7,"output_name":"output"}},"inputs":[],"label":"valid pairs in juicebox format MAPQ filtered","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1041,"top":304.5},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"valid pairs in juicebox format MAPQ filtered"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"fe9a084d-06ad-4671-a4d6-85dff2f96138","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format MAPQ filtered","output_name":"out_file1","uuid":"b08373c7-59b2-4ae3-90b7-7c9a05ca81b7"}]}},"tags":"","uuid":"9df7c037-6fe4-47e9-8311-e40bc5101b58"},"tool_id":null,"type":"subworkflow","uuid":"3584d879-0b7a-4a89-a5f6-8439db6f003e","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format","output_name":"valid pairs in juicebox format","uuid":"bc19676e-9619-43b9-9c21-bc3c31fad3b2"},{"label":"valid pairs in juicebox format MAPQ filtered","output_name":"valid pairs in juicebox format MAPQ filtered","uuid":"ec040a8e-883a-4cdb-b5bf-ea5a611a16a7"},{"label":"HiCUP report (html)","output_name":"HiCUP report (HTML)","uuid":"902f20ef-d51d-4fad-86a0-206cda2ef361"},{"label":"HiCUP report (txt)","output_name":"HiCUP report (tabular)","uuid":"b479e588-6d59-4b60-9dd7-036e58358eae"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":11,"input_connections":{"components_11|param_type|component_value":{"id":8,"output_name":"output"},"components_1|param_type|component_value":{"id":7,"output_name":"output"},"components_3|param_type|component_value":{"id":9,"output_name":"output"},"components_5|param_type|component_value":{"id":8,"output_name":"output"},"components_7|param_type|component_value":{"id":7,"output_name":"output"},"components_9|param_type|component_value":{"id":9,"output_name":"output"}},"inputs":[],"label":"write filtering for capture region","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":698.5,"top":846.6216799999941},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"(c3==\'\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"\' and c4<\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" and c4>\\"}}, {\\"__index__\\": 5, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 6, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\") and (c7==\\\\\\"\\"}}, {\\"__index__\\": 7, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 8, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"\\\\\\" and c8<\\"}}, {\\"__index__\\": 9, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 10, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" and c8>\\"}}, {\\"__index__\\": 11, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 12, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\")\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"5736ad36-505d-45cd-aebc-6763055e9509","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":12,"input_connections":{"components_0|param_type|component_value":{"id":7,"output_name":"output"},"components_2|param_type|component_value":{"id":8,"output_name":"output"},"components_4|param_type|component_value":{"id":9,"output_name":"output"}},"inputs":[],"label":"write region for pyGenomeTracks","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1398,"top":689.6216799999941},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\":\\"}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"-\\"}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"eb2d5028-6f3f-4f6f-9338-ec979be3b4ab","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"Filter1","errors":null,"id":13,"input_connections":{"cond":{"id":11,"output_name":"out1"},"input":{"id":10,"output_name":"valid pairs in juicebox format MAPQ filtered"}},"inputs":[],"label":"Filter for capture region","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":891.5,"top":555.6216799999941},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"f39490d2-503d-4c69-a682-c8472abc8e94","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_csort_tabix/cooler_csort_tabix/0.9.3+galaxy1","errors":null,"id":14,"input_connections":{"input":{"id":13,"output_name":"out_file1"},"size_source|fasta_cached":{"id":1,"output_name":"output"}},"inputs":[],"label":"Sort filtered pairs and index","name":"cooler csort with tabix","outputs":[{"name":"output","type":"interval_tabix.gz"}],"position":{"left":988,"top":228.12167999999406},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"juicer_medium_tabix.gz"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"valid pairs filtered and sorted"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_csort_tabix/cooler_csort_tabix/0.9.3+galaxy1","tool_shed_repository":{"changeset_revision":"8c3cba559c99","name":"cooler_csort_tabix","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"format_sel\\": {\\"format\\": \\"juicer_medium\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"size_source\\": {\\"size_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"fasta_cached\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy1","type":"tool","uuid":"bbb479cd-c67c-4347-9c1d-ac8095d90aa9","when":null,"workflow_outputs":[{"label":"valid pairs filtered and sorted","output_name":"output","uuid":"1a49e9e8-502b-4750-9349-a026a26df5a0"}]},"15":{"annotation":"","id":15,"input_connections":{"Bin size in bp":{"id":5,"input_subworkflow_step_id":0,"output_name":"output"},"Interactions to consider to calculate weights in normalization step":{"id":6,"input_subworkflow_step_id":3,"output_name":"output"},"Juicer Medium Tabix with validPairs":{"id":14,"input_subworkflow_step_id":2,"output_name":"output"},"genome name":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Hi-C_juicermediumtabixToCool_cooler","outputs":[],"position":{"left":1290,"top":309.1166687011719},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Hi-C_juicermediumtabixToCool_cooler","steps":{"0":{"annotation":"For example 10000 for 10kb","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"For example 10000 for 10kb","name":"Bin size in bp"}],"label":"Bin size in bp","name":"Input parameter","outputs":[],"position":{"left":0,"top":33.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d0a82863-82f8-426c-addb-06066e9d0a73","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"9b1ea2f8-0754-426a-b7a4-23385bc7ff6a"}]},"1":{"annotation":"This is used to get the chromosome sizes","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This is used to get the chromosome sizes","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":46,"top":123.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"48506f80-b1b5-4d1d-98a6-c6882de026f1","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"3074ec27-efc0-4aa1-b979-4e9a444420bd"}]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Juicer Medium Tabix with validPairs"}],"label":"Juicer Medium Tabix with validPairs","name":"Input dataset collection","outputs":[],"position":{"left":129.25,"top":208.25},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"588962a4-8ce4-4929-81fd-a9dfdd032598","when":null,"workflow_outputs":[]},"3":{"annotation":"Recommended value: cis-only","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Recommended value: cis-only","name":"Interactions to consider to calculate weights in normalization step"}],"label":"Interactions to consider to calculate weights in normalization step","name":"Input parameter","outputs":[],"position":{"left":221,"top":309.75},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d6375500-a591-47a9-b126-950347e3e7a8","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"fc026c8b-591d-480c-acfa-5f480c2089dd"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","errors":null,"id":4,"input_connections":{"binsize":{"id":0,"output_name":"output"},"size_source|fasta_cached":{"id":1,"output_name":"output"}},"inputs":[],"label":"make bed with bins","name":"cooler_makebins","outputs":[{"name":"output","type":"bed"}],"position":{"left":489.25,"top":0},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"ebd2a52eb245","name":"cooler_makebins","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"binsize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"size_source\\": {\\"size_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"fasta_cached\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"34e0a213-93e8-466e-b07b-ede1443ee7d6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","errors":null,"id":5,"input_connections":{"assembly":{"id":1,"output_name":"output"},"format_sel|input_pairs":{"id":2,"output_name":"output"},"input_bed":{"id":4,"output_name":"output"}},"inputs":[],"label":"Load pairs in matrix","name":"cooler_cload_tabix","outputs":[{"name":"output","type":"cool"}],"position":{"left":762.25,"top":58},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with raw values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","tool_shed_repository":{"changeset_revision":"0641e1c892c6","name":"cooler_cload_tabix","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assembly\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"format_sel\\": {\\"format\\": \\"juicer_medium\\", \\"__current_case__\\": 0, \\"input_pairs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy1","type":"tool","uuid":"d411eea7-7996-44f2-af2a-feecb854a067","when":null,"workflow_outputs":[{"label":"matrix with raw values","output_name":"output","uuid":"d398b493-9b5d-412f-a06d-d1cfdc904feb"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","errors":null,"id":6,"input_connections":{"cistrans":{"id":3,"output_name":"output"},"input":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool cooler_balance","name":"blacklist"}],"label":"ICE normalization","name":"cooler_balance","outputs":[{"name":"output","type":"cool"}],"position":{"left":991.25,"top":254.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with iced values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"8f9fe0667b89","name":"cooler_balance","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"blacklist\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"cistrans\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"convergencepolicy\\": \\"error\\", \\"ignorediags\\": \\"2\\", \\"ignoredist\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"madmax\\": \\"5\\", \\"maxiters\\": \\"200\\", \\"mincount\\": \\"0\\", \\"minnnz\\": \\"10\\", \\"name\\": \\"weight\\", \\"tol\\": \\"1e-05\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"44270887-1c96-4ede-8bb9-3db2a07eeb6e","when":null,"workflow_outputs":[{"label":"matrix with iced values","output_name":"output","uuid":"1d4fe955-922e-40bf-b879-a71290af3ea7"}]}},"tags":"","uuid":"1a6f4422-8f82-4444-a30c-bde38d181e41"},"tool_id":null,"type":"subworkflow","uuid":"63c60960-bd33-436a-ac47-fdaf69c4bdfa","when":null,"workflow_outputs":[{"label":"matrix with iced values","output_name":"matrix with iced values","uuid":"178c2801-e3b3-4398-8448-795d67ae5aa3"},{"label":"matrix with raw values","output_name":"matrix with raw values","uuid":"606c3260-41a2-400e-a580-0db59b3bc11f"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy2","errors":null,"id":16,"input_connections":{"region":{"id":12,"output_name":"out1"},"tracks_0|track_file_style_conditional|matrix_h5_cooler_multiple":{"id":15,"output_name":"matrix with iced values"}},"inputs":[],"label":"final_plot","name":"pyGenomeTracks","outputs":[{"name":"outFileName","type":"png"}],"position":{"left":1673,"top":806.6216799999941},"post_job_actions":{"RenameDatasetActionoutFileName":{"action_arguments":{"newname":"plot with pyGenomeTracks"},"action_type":"RenameDatasetAction","output_name":"outFileName"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy2","tool_shed_repository":{"changeset_revision":"59fd173ac850","name":"pygenometracks","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"global_args\\": {\\"title\\": \\"\\", \\"fontsize\\": \\"12\\", \\"dpi\\": \\"72\\", \\"width\\": \\"40.0\\", \\"plotWidth\\": null, \\"height\\": null, \\"trackLabelFraction\\": \\"0.05\\", \\"trackLabelHAlign\\": \\"left\\", \\"decreasingXAxis\\": false}, \\"image_file_format\\": \\"png\\", \\"region\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tracks\\": [{\\"__index__\\": 0, \\"track_file_style_conditional\\": {\\"track_file_style_selector\\": \\"hic_matrix_option\\", \\"__current_case__\\": 0, \\"title\\": \\"\\", \\"matrix_h5_cooler_multiple\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"colormap\\": \\"RdYlBu_r\\", \\"min_value\\": null, \\"max_value\\": null, \\"depth\\": \\"8000000\\", \\"transform\\": \\"no\\", \\"height_matrix\\": null, \\"show_masked_bins\\": false, \\"boundaries_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"scale_factor\\": \\"1.0\\", \\"rasterize\\": true, \\"invert_orientation\\": false, \\"spacer_height\\": null}}, {\\"__index__\\": 1, \\"track_file_style_conditional\\": {\\"track_file_style_selector\\": \\"xaxis_option\\", \\"__current_case__\\": 13, \\"title\\": \\"\\", \\"fontsize\\": null, \\"xaxis_where\\": \\"bottom\\", \\"spacer_height\\": null}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.8+galaxy2","type":"tool","uuid":"dfbe45de-b2a7-479e-b717-29ec5b31379e","when":null,"workflow_outputs":[{"label":"plot with pyGenomeTracks","output_name":"outFileName","uuid":"32a70740-19aa-463d-a280-6394156bd412"}]}},"tags":["Hi-C"],"uuid":"2a89cbe4-a69f-4bff-9465-2e04b0217e35","version":4},"readme":"# (Capture) Hi-C Processing: FASTQ to Balanced Cool Files\\n\\nThis can also be used for Hi-ChIP experiments, in that case the output with `matrix with iced values` is ignored and the matrix to use is `matrix with raw values`.\\n\\n## Input datasets\\n\\n- The workflow needs a list of dataset pairs of fastqsanger.\\n\\n## Input values\\n\\n- genome name: suggested from the bowtie2 indices, it is used to map and build the list of bins.\\n- restriction enzyme: Restriction enzyme used e.g. A^GATCT,BglII. The \'^\' is used to express where the enzyme cuts.\\n- No fill-in: If you used a biotin fill-in protocol, put this to false, else, put it to true.\\n- minimum MAPQ: Filtering to apply to pairs you want to keep in your matrix, set it to 0 to not apply filtering (HiCUP already filter for uniquely mapped or MAPQ30).\\n- Bin size in bp: Used to generate your first matrix but you will be able to rerun the subworkflow `hic_tabix_to_cool_cooler` to get other resolutions.\\n- Interactions to consider to calculate weights in normalization step: this is a parameter for the last correction step (ICE).\\n\\nFor the region capture workflow:\\n\\n- chromosome, start and end positions of the capture region\\n\\nFor the Hi-C workflow:\\n\\n- region to use in pyGenomeTracks to check the matrices.\\n\\n## Processing\\n\\n- Reads are processed with HiCUP which comprises these steps:\\n  - Truncation of reads for the religation motif\\n  - Mapping of mates independently with bowtie2\\n  - Pairing the mates when both mates are uniquely mapped or MAPQ30\\n  - Filtering the pairs for undigested, self-ligated...\\n  - Removing duplicates\\n- The output BAM file is converted to medium juicer format: `<readname> <str1> <chr1> <pos1> <frag1> <str2> <chr2> <pos2> <frag2> <mapq1> <mapq2>` where str = strand (0 for forward, anything else for reverse) and pos is the middle of the fragment.\\n- The pairs are filtered for MAPQ if specified.\\n- For the region capture Hi-C workflow the pairs are filtered for both mates in the captured region.\\n- The filtered pairs are sorted and indexed with cooler_csort.\\n- The pairs are loaded into a matrix of the given resolution and balanced with cooler.\\n- A final plot is made with pyGenomeTracks using the balanced matrices on the region provided or the capture region.\\n\\n## Subworkflows\\n\\nThere are 2 subworkflows: `hic_tabix_to_cool_cooler` and `hic_fastq_to_pairs_hicup.ga`.\\n\\n### hic_tabix_to_cool_cooler\\n\\nThis first subworkflow can be used to generate matrices to different resolutions using one of the output of the full workflow (`valid pairs filtered and sorted`).\\n\\nIf the dataset are still in galaxy (format: juicer_medium_tabix.gz), the workflow can be run directly.\\n\\nIf the dataset is not anymore in galaxy, you need to upload and specify the datatype as: juicer_medium_tabix.gz\\n\\n### hic_fastq_to_pairs_hicup\\n\\nThe second subworkflow has no real reason to be launched by itself except for QC tests.\\n\\nIf you want to run the first subworkflow from these results:\\n\\n- You first need to filter the pairs (`valid pairs in juicebox format MAPQ filtered`) for the capture region if relevent using the tool Filter1 (**Filter** data on any column using simple expressions) with the condition `(c3==\'chr2\' and c4<180000000 and c4>170000000) and (c7==\\"chr2\\" and c8<180000000 and c8>170000000)` if your capture region is chr2:170000000-180000000.\\n- Then you need to run cooler_csort (**cooler csort with tabix** Sort and index a contact list.) with as input the `valid pairs in juicebox format MAPQ filtered` or the output of the previous step and for \\"Format of your input file\\" use \\"Juicer Medium Format\\".\\n\\nThe output of `cooler_csort` can be used as input of the first subworkflow.\\n","changelog":"# Changelog\\n\\n## [0.3] - 2023-09-08\\n\\n### Update tools\\n- all cooler tools were updated from 0.8.11 to 0.9.3\\n- pyGenomeTracks was updated from 3.7 to 3.8\\n\\n## [0.2.1] - 2023-09-01 Bug fix\\n\\nFix dockstore (workflows are the same)\\n\\n## [0.2] - 2023-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0`\\n\\n### Manual update\\n- Get fragment id in valid pairs file.\\n- Use the middle of the fragment instead of 5\' of the read in the valid pairs file.\\n\\n## [0.1] - 2023-01-16\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Capture Hi-C Processing: FASTQ to Balanced Cool Files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Restriction enzyme\\"]@{ shape: lean-l }\\n3[\\"ℹ️ No fill-in\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum MAPQ\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Bin size in bp\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Interactions to consider to calculate weights in normalization step\\"]@{ shape: lean-l }\\n7[\\"ℹ️ capture region (chromosome)\\"]@{ shape: lean-l }\\n8[\\"ℹ️ capture region (start)\\"]@{ shape: lean-l }\\n9[\\"ℹ️ capture region (end)\\"]@{ shape: lean-l }\\n10[\\"🛠️ Hi-C_fastqToPairs_hicup\\"]@{ shape: subprocess }\\n3 --> 10\\n0 --> 10\\n2 --> 10\\n1 --> 10\\n4 --> 10\\n11[\\"write filtering for capture region\\"]@{ shape: process }\\n8 --> 11\\n7 --> 11\\n9 --> 11\\n8 --> 11\\n7 --> 11\\n9 --> 11\\n12[\\"write region for pyGenomeTracks\\"]@{ shape: process }\\n7 --> 12\\n8 --> 12\\n9 --> 12\\n13[\\"Filter for capture region\\"]@{ shape: process }\\n11 --> 13\\n10 --> 13\\n14[\\"Sort filtered pairs and index\\"]@{ shape: process }\\n13 --> 14\\n1 --> 14\\n15[\\"🛠️ Hi-C_juicermediumtabixToCool_cooler\\"]@{ shape: subprocess }\\n5 --> 15\\n6 --> 15\\n14 --> 15\\n1 --> 15\\n16[\\"final_plot\\"]@{ shape: process }\\n12 --> 16\\n15 --> 16\\n```\\n\\n## Hi-C_juicermediumtabixToCool_cooler\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Bin size in bp\\"]@{ shape: lean-l }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Juicer Medium Tabix with validPairs\\"]@{ shape: docs }\\n3[\\"ℹ️ Interactions to consider to calculate weights in normalization step\\"]@{ shape: lean-l }\\n4[\\"make bed with bins\\"]@{ shape: process }\\n0 --> 4\\n1 --> 4\\n5[\\"Load pairs in matrix\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n4 --> 5\\n6[\\"ICE normalization\\"]@{ shape: process }\\n3 --> 6\\n5 --> 6\\n```\\n\\n## Hi-C_fastqToPairs_hicup\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Restriction enzyme\\"]@{ shape: lean-l }\\n3[\\"ℹ️ No fill-in\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum MAPQ\\"]@{ shape: lean-l }\\n5[\\"HiCUP\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n0 --> 5\\n2 --> 5\\n1 --> 5\\n6[\\"build filtering rule for MAPQ\\"]@{ shape: process }\\n4 --> 6\\n4 --> 6\\n7[\\"valid pairs in juicebox format\\"]@{ shape: process }\\n5 --> 7\\n5 --> 7\\n8[\\"valid pairs in juicebox format MAPQ filtered\\"]@{ shape: process }\\n6 --> 8\\n7 --> 8\\n```\\n","updated":"2023-09-08T00:00:00","trsID":"#workflow/github.com/iwc-workflows/hic-hicup-cooler/chic-fastq-to-cool-hicup-cooler","dockstore_id":20778,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078793","tests":[{"doc":"Test outline for Hi-C_fastqToCool_hicup_cooler","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"test_dataset","elements":[{"class":"File","identifier":"forward","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset1.fastq","filetype":"fastqsanger"},{"class":"File","identifier":"reverse","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset2.fastq","filetype":"fastqsanger"}]}]},"genome name":"hg19","Restriction enzyme":"A^AGCTT,HindIII","No fill-in":"false","minimum MAPQ":30,"Bin size in bp":1000000,"Interactions to consider to calculate weights in normalization step":"--cis-only","capture region (chromosome)":"chr2","capture region (start)":170000000,"capture region (end)":180000000},"outputs":{"HiCUP report (html)":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":4602173,"delta":4000000}}}}},"HiCUP report (txt)":{"element_tests":{"test_dataset":{"asserts":{"has_text":{"text":"\\t99742\\t99742\\t92512\\t92628\\t7230\\t7114\\t22.45\\t22.63\\t2658\\t2476\\t73431\\t72142\\t17767\\t18475\\t5886\\t6649\\t57671\\t57671\\t39966\\t1652\\t17997\\t20317\\t17705\\t481\\t2432\\t13452\\t1340\\t0\\t0\\t39962\\t1652\\t17996\\t20314\\t57.82\\t69.30\\t99.99\\t50.83\\t40.07"}}}}},"valid pairs in juicebox format":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"has_line","line":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}},"valid pairs in juicebox format MAPQ filtered":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"not_has_text","text":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}},"valid pairs filtered and sorted":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":807,"delta":80}}}}},"matrix with raw values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":42118,"delta":4000}}}}},"matrix with iced values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":47830,"delta":4000}}}}},"plot with pyGenomeTracks":{"file":"test-data/plot_chic.png","compare":"sim_size","delta":1500}}}]},{"name":"hic-juicermediumtabix-to-cool-cooler","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/hic-juicermediumtabix-to-cool-cooler.ga","testParameterFiles":["/hic-juicermediumtabix-to-cool-cooler-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Converts Hi-C interaction data from Juicer medium format (tabix files) to balanced cooler format. This streamlined workflow takes a collection of Juicer medium tabix files and a reference genome name as input and generates balanced cool files at user-specified resolution. Enables seamless transition between different Hi-C analysis ecosystems while maintaining data integrity.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"Hi-C Format Conversion: Juicer Medium to Cooler Files","steps":{"0":{"annotation":"For example 10000 for 10kb","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"For example 10000 for 10kb","name":"Bin size in bp"}],"label":"Bin size in bp","name":"Input parameter","outputs":[],"position":{"left":0,"top":33.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d0a82863-82f8-426c-addb-06066e9d0a73","when":null,"workflow_outputs":[]},"1":{"annotation":"This is used to get the chromosome sizes","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This is used to get the chromosome sizes","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":46,"top":123.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"48506f80-b1b5-4d1d-98a6-c6882de026f1","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"Juicer Medium Tabix with validPairs"}],"label":"Juicer Medium Tabix with validPairs","name":"Input dataset collection","outputs":[],"position":{"left":129.25,"top":208.25},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"588962a4-8ce4-4929-81fd-a9dfdd032598","when":null,"workflow_outputs":[]},"3":{"annotation":"Recommended value: cis-only","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Recommended value: cis-only","name":"Interactions to consider to calculate weights in normalization step"}],"label":"Interactions to consider to calculate weights in normalization step","name":"Input parameter","outputs":[],"position":{"left":221,"top":309.75},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d6375500-a591-47a9-b126-950347e3e7a8","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","errors":null,"id":4,"input_connections":{"binsize":{"id":0,"output_name":"output"},"size_source|fasta_cached":{"id":1,"output_name":"output"}},"inputs":[],"label":"make bed with bins","name":"cooler_makebins","outputs":[{"name":"output","type":"bed"}],"position":{"left":489.25,"top":0},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_makebins/cooler_makebins/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"ebd2a52eb245","name":"cooler_makebins","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"binsize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"size_source\\": {\\"size_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"fasta_cached\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"34e0a213-93e8-466e-b07b-ede1443ee7d6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","errors":null,"id":5,"input_connections":{"assembly":{"id":1,"output_name":"output"},"format_sel|input_pairs":{"id":2,"output_name":"output"},"input_bed":{"id":4,"output_name":"output"}},"inputs":[],"label":"Load pairs in matrix","name":"cooler_cload_tabix","outputs":[{"name":"output","type":"cool"}],"position":{"left":762.25,"top":58},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with raw values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_cload_tabix/cooler_cload_tabix/0.9.3+galaxy1","tool_shed_repository":{"changeset_revision":"0641e1c892c6","name":"cooler_cload_tabix","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assembly\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"format_sel\\": {\\"format\\": \\"juicer_medium\\", \\"__current_case__\\": 0, \\"input_pairs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy1","type":"tool","uuid":"d411eea7-7996-44f2-af2a-feecb854a067","when":null,"workflow_outputs":[{"label":"matrix with raw values","output_name":"output","uuid":"d398b493-9b5d-412f-a06d-d1cfdc904feb"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","errors":null,"id":6,"input_connections":{"cistrans":{"id":3,"output_name":"output"},"input":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool cooler_balance","name":"blacklist"}],"label":"ICE normalization","name":"cooler_balance","outputs":[{"name":"output","type":"cool"}],"position":{"left":991.25,"top":254.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"matrix with iced values"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lldelisle/cooler_balance/cooler_balance/0.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"8f9fe0667b89","name":"cooler_balance","owner":"lldelisle","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"blacklist\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"cistrans\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"convergencepolicy\\": \\"error\\", \\"ignorediags\\": \\"2\\", \\"ignoredist\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"madmax\\": \\"5\\", \\"maxiters\\": \\"200\\", \\"mincount\\": \\"0\\", \\"minnnz\\": \\"10\\", \\"name\\": \\"weight\\", \\"tol\\": \\"1e-05\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.3+galaxy0","type":"tool","uuid":"44270887-1c96-4ede-8bb9-3db2a07eeb6e","when":null,"workflow_outputs":[{"label":"matrix with iced values","output_name":"output","uuid":"1d4fe955-922e-40bf-b879-a71290af3ea7"}]}},"tags":["Hi-C"],"uuid":"48733b55-4cde-45d4-a69e-78faff0db3f5","version":1},"readme":"# (Capture) Hi-C Processing: FASTQ to Balanced Cool Files\\n\\nThis can also be used for Hi-ChIP experiments, in that case the output with `matrix with iced values` is ignored and the matrix to use is `matrix with raw values`.\\n\\n## Input datasets\\n\\n- The workflow needs a list of dataset pairs of fastqsanger.\\n\\n## Input values\\n\\n- genome name: suggested from the bowtie2 indices, it is used to map and build the list of bins.\\n- restriction enzyme: Restriction enzyme used e.g. A^GATCT,BglII. The \'^\' is used to express where the enzyme cuts.\\n- No fill-in: If you used a biotin fill-in protocol, put this to false, else, put it to true.\\n- minimum MAPQ: Filtering to apply to pairs you want to keep in your matrix, set it to 0 to not apply filtering (HiCUP already filter for uniquely mapped or MAPQ30).\\n- Bin size in bp: Used to generate your first matrix but you will be able to rerun the subworkflow `hic_tabix_to_cool_cooler` to get other resolutions.\\n- Interactions to consider to calculate weights in normalization step: this is a parameter for the last correction step (ICE).\\n\\nFor the region capture workflow:\\n\\n- chromosome, start and end positions of the capture region\\n\\nFor the Hi-C workflow:\\n\\n- region to use in pyGenomeTracks to check the matrices.\\n\\n## Processing\\n\\n- Reads are processed with HiCUP which comprises these steps:\\n  - Truncation of reads for the religation motif\\n  - Mapping of mates independently with bowtie2\\n  - Pairing the mates when both mates are uniquely mapped or MAPQ30\\n  - Filtering the pairs for undigested, self-ligated...\\n  - Removing duplicates\\n- The output BAM file is converted to medium juicer format: `<readname> <str1> <chr1> <pos1> <frag1> <str2> <chr2> <pos2> <frag2> <mapq1> <mapq2>` where str = strand (0 for forward, anything else for reverse) and pos is the middle of the fragment.\\n- The pairs are filtered for MAPQ if specified.\\n- For the region capture Hi-C workflow the pairs are filtered for both mates in the captured region.\\n- The filtered pairs are sorted and indexed with cooler_csort.\\n- The pairs are loaded into a matrix of the given resolution and balanced with cooler.\\n- A final plot is made with pyGenomeTracks using the balanced matrices on the region provided or the capture region.\\n\\n## Subworkflows\\n\\nThere are 2 subworkflows: `hic_tabix_to_cool_cooler` and `hic_fastq_to_pairs_hicup.ga`.\\n\\n### hic_tabix_to_cool_cooler\\n\\nThis first subworkflow can be used to generate matrices to different resolutions using one of the output of the full workflow (`valid pairs filtered and sorted`).\\n\\nIf the dataset are still in galaxy (format: juicer_medium_tabix.gz), the workflow can be run directly.\\n\\nIf the dataset is not anymore in galaxy, you need to upload and specify the datatype as: juicer_medium_tabix.gz\\n\\n### hic_fastq_to_pairs_hicup\\n\\nThe second subworkflow has no real reason to be launched by itself except for QC tests.\\n\\nIf you want to run the first subworkflow from these results:\\n\\n- You first need to filter the pairs (`valid pairs in juicebox format MAPQ filtered`) for the capture region if relevent using the tool Filter1 (**Filter** data on any column using simple expressions) with the condition `(c3==\'chr2\' and c4<180000000 and c4>170000000) and (c7==\\"chr2\\" and c8<180000000 and c8>170000000)` if your capture region is chr2:170000000-180000000.\\n- Then you need to run cooler_csort (**cooler csort with tabix** Sort and index a contact list.) with as input the `valid pairs in juicebox format MAPQ filtered` or the output of the previous step and for \\"Format of your input file\\" use \\"Juicer Medium Format\\".\\n\\nThe output of `cooler_csort` can be used as input of the first subworkflow.\\n","changelog":"# Changelog\\n\\n## [0.3] - 2023-09-08\\n\\n### Update tools\\n- all cooler tools were updated from 0.8.11 to 0.9.3\\n- pyGenomeTracks was updated from 3.7 to 3.8\\n\\n## [0.2.1] - 2023-09-01 Bug fix\\n\\nFix dockstore (workflows are the same)\\n\\n## [0.2] - 2023-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0`\\n\\n### Manual update\\n- Get fragment id in valid pairs file.\\n- Use the middle of the fragment instead of 5\' of the read in the valid pairs file.\\n\\n## [0.1] - 2023-01-16\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Hi-C Format Conversion: Juicer Medium to Cooler Files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Bin size in bp\\"]@{ shape: lean-l }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Juicer Medium Tabix with validPairs\\"]@{ shape: docs }\\n3[\\"ℹ️ Interactions to consider to calculate weights in normalization step\\"]@{ shape: lean-l }\\n4[\\"make bed with bins\\"]@{ shape: process }\\n0 --> 4\\n1 --> 4\\n5[\\"Load pairs in matrix\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n4 --> 5\\n6[\\"ICE normalization\\"]@{ shape: process }\\n3 --> 6\\n5 --> 6\\n```\\n","updated":"2023-09-08T00:00:00","trsID":"#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-juicermediumtabix-to-cool-cooler","dockstore_id":20779,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078469","tests":[{"doc":"Test outline for Hi-C_fastqToCool_hicup_cooler","job":{"Juicer Medium Tabix with validPairs":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"test_dataset","path":"test-data/valid_pairs_filtered_and_sorted.juicer_medium_tabix.gz","filetype":"juicer_medium_tabix.gz"}]},"genome name":"hg19","Bin size in bp":1000000,"Interactions to consider to calculate weights in normalization step":"--cis-only"},"outputs":{"matrix with raw values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":42118,"delta":4000}}}}},"matrix with iced values":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":47830,"delta":4000}}}}}}}]},{"name":"hic-fastq-to-pairs-hicup","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/hic-fastq-to-pairs-hicup.ga","testParameterFiles":["/hic-fastq-to-pairs-hicup-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Processes Hi-C paired-end FASTQ files to generate validated interaction pairs using HiCUP. The workflow truncates reads at ligation junctions, maps to reference genome, assigns to restriction fragments, and filters out experimental artifacts (self-ligated, dangling ends, internal fragments, and size outliers). Removes duplicates and converts outputs to formats compatible with Juicebox/cooler using fragment midpoints as coordinates. Final filtering by mapping quality ensures high-confidence interaction data.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"Hi-C Data Processing: FASTQ to Valid Interaction Pairs","steps":{"0":{"annotation":"Should be a paired collection with Hi-C fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with Hi-C fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":12.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"30fe3d0f-541a-478a-b57d-a43c0c16ccad","when":null,"workflow_outputs":[]},"1":{"annotation":"use the bowtie2 indexes to display choices","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"use the bowtie2 indexes to display choices","name":"genome name"}],"label":"genome name","name":"Input parameter","outputs":[],"position":{"left":39,"top":82.5},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"45194ed6-a1e9-4248-8d3a-f51febc36d61","when":null,"workflow_outputs":[]},"2":{"annotation":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Restriction enzyme used e.g. A^GATCT,BglII. Some Hi-C protocols may use several enzymes. To specify several enzymes, use the \':\' to separate them e.g. A^GATCT,BglII:A^AGCTT,HindIII:^GATC,DpnII. HiCUP accomodates N in restriction enzyme: e.g. :A^ANCTT","name":"Restriction enzyme"}],"label":"Restriction enzyme","name":"Input parameter","outputs":[],"position":{"left":58,"top":155.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec011ceb-c601-464f-8661-ca9c0b335bb5","when":null,"workflow_outputs":[]},"3":{"annotation":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Hi-C protocol did NOT include a fill-in of sticky ends prior to re-ligation and therefore reads shall be truncated at the restriction site sequence","name":"No fill-in"}],"label":"No fill-in","name":"Input parameter","outputs":[],"position":{"left":97.5,"top":248.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c807fdbf-1c1d-45b4-851e-47fb919b07d6","when":null,"workflow_outputs":[]},"4":{"annotation":"can be set to 0 for no filtering","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"can be set to 0 for no filtering","name":"minimum MAPQ"}],"label":"minimum MAPQ","name":"Input parameter","outputs":[],"position":{"left":131,"top":320.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"748c8800-5bef-41b7-b2cf-6d6a57c0eb70","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","errors":null,"id":5,"input_connections":{"advanced_options|nofill":{"id":3,"output_name":"output"},"genome":{"id":1,"output_name":"output"},"library|input_1":{"id":0,"output_name":"output"},"re1":{"id":2,"output_name":"output"},"reference_genome|index":{"id":1,"output_name":"output"}},"inputs":[],"label":"HiCUP","name":"Hicup Pipeline","outputs":[{"name":"intermediate_results","type":"input"},{"name":"hicup_results","type":"html"},{"name":"hicup_report","type":"tabular"},{"name":"dataset_hicup","type":"qname_sorted.bam"},{"name":"digester_file","type":"tabular"}],"position":{"left":356.5,"top":0},"post_job_actions":{"HideDatasetActiondataset_hicup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"dataset_hicup"},"HideDatasetActiondigester_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"digester_file"},"HideDatasetActionintermediate_results":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"intermediate_results"},"RenameDatasetActionhicup_report":{"action_arguments":{"newname":"HiCUP report (tabular)"},"action_type":"RenameDatasetAction","output_name":"hicup_report"},"RenameDatasetActionhicup_results":{"action_arguments":{"newname":"HiCUP report (HTML)"},"action_type":"RenameDatasetAction","output_name":"hicup_results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"68fb19094f84","name":"hicup_hicup","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"re2\\": \\"\\", \\"longest\\": \\"\\", \\"shortest\\": \\"\\", \\"nofill\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"re1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"8aba6fbb-d940-4826-911b-8c148e82085c","when":null,"workflow_outputs":[{"label":"HiCUP report (tabular)","output_name":"hicup_report","uuid":"fd8f1b17-c68e-4652-b139-acccf51baa30"},{"label":"HiCUP report (HTML)","output_name":"hicup_results","uuid":"ba37bb7a-ad3a-487b-b7ba-0023f1c496fd"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"},"components_3|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":"build filtering rule for MAPQ","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":746.5,"top":392.5},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"(c10>=\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\") and (c11>=\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\")\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"ee70cd21-6a36-4e21-97d2-ff0749236523","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","errors":null,"id":7,"input_connections":{"digester|digester_file":{"id":5,"output_name":"digester_file"},"input_file":{"id":5,"output_name":"dataset_hicup"}},"inputs":[{"description":"runtime parameter for tool Hicup to juicer converter","name":"digester"},{"description":"runtime parameter for tool Hicup to juicer converter","name":"input_file"}],"label":"valid pairs in juicebox format","name":"Hicup to juicer converter","outputs":[{"name":"output","type":"tabular"}],"position":{"left":637.5,"top":197.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"valid pairs in juicebox format"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0","tool_shed_repository":{"changeset_revision":"b4e7244246e2","name":"hicup2juicer","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"digester\\": {\\"provide_digester\\": \\"yes\\", \\"__current_case__\\": 0, \\"digester_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"usemid\\": true}, \\"input_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.2+galaxy0","type":"tool","uuid":"659811cd-a212-4949-9693-074f6161ed11","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format","output_name":"output","uuid":"dfc67c68-2569-4727-9ca6-a70f3b13edd9"}]},"8":{"annotation":"","content_id":"Filter1","errors":null,"id":8,"input_connections":{"cond":{"id":6,"output_name":"out1"},"input":{"id":7,"output_name":"output"}},"inputs":[],"label":"valid pairs in juicebox format MAPQ filtered","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1041,"top":304.5},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"valid pairs in juicebox format MAPQ filtered"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"0\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"fe9a084d-06ad-4671-a4d6-85dff2f96138","when":null,"workflow_outputs":[{"label":"valid pairs in juicebox format MAPQ filtered","output_name":"out_file1","uuid":"b08373c7-59b2-4ae3-90b7-7c9a05ca81b7"}]}},"tags":["Hi-C"],"uuid":"48f579ea-eaac-49ee-b38f-93c82f8b7d76","version":2},"readme":"# (Capture) Hi-C Processing: FASTQ to Balanced Cool Files\\n\\nThis can also be used for Hi-ChIP experiments, in that case the output with `matrix with iced values` is ignored and the matrix to use is `matrix with raw values`.\\n\\n## Input datasets\\n\\n- The workflow needs a list of dataset pairs of fastqsanger.\\n\\n## Input values\\n\\n- genome name: suggested from the bowtie2 indices, it is used to map and build the list of bins.\\n- restriction enzyme: Restriction enzyme used e.g. A^GATCT,BglII. The \'^\' is used to express where the enzyme cuts.\\n- No fill-in: If you used a biotin fill-in protocol, put this to false, else, put it to true.\\n- minimum MAPQ: Filtering to apply to pairs you want to keep in your matrix, set it to 0 to not apply filtering (HiCUP already filter for uniquely mapped or MAPQ30).\\n- Bin size in bp: Used to generate your first matrix but you will be able to rerun the subworkflow `hic_tabix_to_cool_cooler` to get other resolutions.\\n- Interactions to consider to calculate weights in normalization step: this is a parameter for the last correction step (ICE).\\n\\nFor the region capture workflow:\\n\\n- chromosome, start and end positions of the capture region\\n\\nFor the Hi-C workflow:\\n\\n- region to use in pyGenomeTracks to check the matrices.\\n\\n## Processing\\n\\n- Reads are processed with HiCUP which comprises these steps:\\n  - Truncation of reads for the religation motif\\n  - Mapping of mates independently with bowtie2\\n  - Pairing the mates when both mates are uniquely mapped or MAPQ30\\n  - Filtering the pairs for undigested, self-ligated...\\n  - Removing duplicates\\n- The output BAM file is converted to medium juicer format: `<readname> <str1> <chr1> <pos1> <frag1> <str2> <chr2> <pos2> <frag2> <mapq1> <mapq2>` where str = strand (0 for forward, anything else for reverse) and pos is the middle of the fragment.\\n- The pairs are filtered for MAPQ if specified.\\n- For the region capture Hi-C workflow the pairs are filtered for both mates in the captured region.\\n- The filtered pairs are sorted and indexed with cooler_csort.\\n- The pairs are loaded into a matrix of the given resolution and balanced with cooler.\\n- A final plot is made with pyGenomeTracks using the balanced matrices on the region provided or the capture region.\\n\\n## Subworkflows\\n\\nThere are 2 subworkflows: `hic_tabix_to_cool_cooler` and `hic_fastq_to_pairs_hicup.ga`.\\n\\n### hic_tabix_to_cool_cooler\\n\\nThis first subworkflow can be used to generate matrices to different resolutions using one of the output of the full workflow (`valid pairs filtered and sorted`).\\n\\nIf the dataset are still in galaxy (format: juicer_medium_tabix.gz), the workflow can be run directly.\\n\\nIf the dataset is not anymore in galaxy, you need to upload and specify the datatype as: juicer_medium_tabix.gz\\n\\n### hic_fastq_to_pairs_hicup\\n\\nThe second subworkflow has no real reason to be launched by itself except for QC tests.\\n\\nIf you want to run the first subworkflow from these results:\\n\\n- You first need to filter the pairs (`valid pairs in juicebox format MAPQ filtered`) for the capture region if relevent using the tool Filter1 (**Filter** data on any column using simple expressions) with the condition `(c3==\'chr2\' and c4<180000000 and c4>170000000) and (c7==\\"chr2\\" and c8<180000000 and c8>170000000)` if your capture region is chr2:170000000-180000000.\\n- Then you need to run cooler_csort (**cooler csort with tabix** Sort and index a contact list.) with as input the `valid pairs in juicebox format MAPQ filtered` or the output of the previous step and for \\"Format of your input file\\" use \\"Juicer Medium Format\\".\\n\\nThe output of `cooler_csort` can be used as input of the first subworkflow.\\n","changelog":"# Changelog\\n\\n## [0.3] - 2023-09-08\\n\\n### Update tools\\n- all cooler tools were updated from 0.8.11 to 0.9.3\\n- pyGenomeTracks was updated from 3.7 to 3.8\\n\\n## [0.2.1] - 2023-09-01 Bug fix\\n\\nFix dockstore (workflows are the same)\\n\\n## [0.2] - 2023-02-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup_hicup/hicup_hicup/0.9.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.8.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/hicup2juicer/hicup2juicer/0.9.2+galaxy0`\\n\\n### Manual update\\n- Get fragment id in valid pairs file.\\n- Use the middle of the fragment instead of 5\' of the read in the valid pairs file.\\n\\n## [0.1] - 2023-01-16\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Hi-C Data Processing: FASTQ to Valid Interaction Pairs\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ genome name\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Restriction enzyme\\"]@{ shape: lean-l }\\n3[\\"ℹ️ No fill-in\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum MAPQ\\"]@{ shape: lean-l }\\n5[\\"HiCUP\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n0 --> 5\\n2 --> 5\\n1 --> 5\\n6[\\"build filtering rule for MAPQ\\"]@{ shape: process }\\n4 --> 6\\n4 --> 6\\n7[\\"valid pairs in juicebox format\\"]@{ shape: process }\\n5 --> 7\\n5 --> 7\\n8[\\"valid pairs in juicebox format MAPQ filtered\\"]@{ shape: process }\\n6 --> 8\\n7 --> 8\\n```\\n","updated":"2023-09-08T00:00:00","trsID":"#workflow/github.com/iwc-workflows/hic-hicup-cooler/hic-fastq-to-pairs-hicup","dockstore_id":20780,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078511","tests":[{"doc":"Test outline for Hi-C_fastqToPairs_hicup","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"test_dataset","elements":[{"class":"File","identifier":"forward","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset1.fastq","filetype":"fastqsanger"},{"class":"File","identifier":"reverse","location":"https://github.com/bgruening/galaxytools/raw/master/tools/hicup/test-data/dataset2.fastq","filetype":"fastqsanger"}]}]},"genome name":"hg19","Restriction enzyme":"A^AGCTT,HindIII","No fill-in":"false","minimum MAPQ":30},"outputs":{"HiCUP report (HTML)":{"element_tests":{"test_dataset":{"asserts":{"has_size":{"value":4602173,"delta":4000000}}}}},"HiCUP report (tabular)":{"element_tests":{"test_dataset":{"asserts":{"has_text":{"text":"\\t99742\\t99742\\t92512\\t92628\\t7230\\t7114\\t22.45\\t22.63\\t2658\\t2476\\t73431\\t72142\\t17767\\t18475\\t5886\\t6649\\t57671\\t57671\\t39966\\t1652\\t17997\\t20317\\t17705\\t481\\t2432\\t13452\\t1340\\t0\\t0\\t39962\\t1652\\t17996\\t20314\\t57.82\\t69.30\\t99.99\\t50.83\\t40.07"}}}}},"valid pairs in juicebox format":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"has_line","line":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}},"valid pairs in juicebox format MAPQ filtered":{"element_tests":{"test_dataset":{"asserts":[{"that":"has_line","line":"1\\t1\\tchr10\\t100023987\\t28055\\t1\\tchr10\\t101500419\\t28474\\t42\\t42"},{"that":"has_line","line":"2\\t1\\tchr10\\t100091500\\t28079\\t1\\tchr10\\t122245984\\t34516\\t38\\t42"},{"that":"not_has_text","text":"3\\t0\\tchr10\\t100127492\\t28094\\t1\\tchr10\\t50864290\\t13489\\t0\\t42"}]}}}}}]}],"path":"./workflows/epigenetics/hic-hicup-cooler"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/chipseq-pe.ga","testParameterFiles":["/chipseq-pe-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete ChIP-seq analysis for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt), alignment to reference genome (Bowtie2), and stringent quality filtering (MAPQ >= 30, concordant pairs only). Peak calling with MACS2 optimized for paired-end reads identifies protein-DNA binding sites. Generates alignment files, peak calls, and quality metrics for downstream analysis.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.14","name":"ChIP-seq Analysis: Paired-End Read Processing","steps":{"0":{"annotation":"Should be a paired collection with ChIPseq fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with ChIPseq fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"e09c0852-1db3-4a68-b88c-1b94c205cb6c","when":null,"workflow_outputs":[]},"1":{"annotation":"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC ","name":"adapter_forward"}],"label":"adapter_forward","name":"Input parameter","outputs":[],"position":{"left":11.01666259765625,"top":93.5},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"30c1d867-5e73-4348-8969-848f58d94015","when":null,"workflow_outputs":[]},"2":{"annotation":"Please use: For R2: - For Nextera: CTGTCTCTTATACACATCTGACGCTGCCGACGA - For TruSeq: GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Please use: For R2: - For Nextera: CTGTCTCTTATACACATCTGACGCTGCCGACGA - For TruSeq: GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","name":"adapter_reverse"}],"label":"adapter_reverse","name":"Input parameter","outputs":[],"position":{"left":50.066650390625,"top":189.45001220703125},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec244ede-8fa3-4d05-85a0-06839f4cf97d","when":null,"workflow_outputs":[]},"3":{"annotation":"reference_genome","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"reference_genome","name":"reference_genome"}],"label":"reference_genome","name":"Input parameter","outputs":[],"position":{"left":102.9000244140625,"top":333.23333740234375},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"0119d669-7ce9-46fd-ba6f-3efd92dfb7f2","when":null,"workflow_outputs":[]},"4":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":144.8499755859375,"top":437.61669921875},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"5bb3b3df-60ab-4ec7-88e4-476be547ffbf","when":null,"workflow_outputs":[]},"5":{"annotation":"Whether you want to have a profile normalized as Signal to Million Fragments","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Whether you want to have a profile normalized as Signal to Million Fragments","name":"normalize_profile"}],"label":"normalize_profile","name":"Input parameter","outputs":[],"position":{"left":185,"top":542},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8180a47b-2fa4-404e-b644-1f3ec3bea980","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","errors":null,"id":6,"input_connections":{"library|input_1":{"id":0,"output_name":"output"},"library|r1|adapters_0|adapter_source|adapter":{"id":1,"output_name":"output"},"library|r2|adapters2_0|adapter_source|adapter":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":"Cutadapt (remove adapter + bad quality bases)","name":"Cutadapt","outputs":[{"name":"out_pairs","type":"input"},{"name":"report","type":"txt"}],"position":{"left":414.29998779296875,"top":57.25},"post_job_actions":{"HideDatasetActionout_pairs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_pairs"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"},"RenameDatasetActionreport":{"action_arguments":{"newname":"cutadapt report"},"action_type":"RenameDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0","tool_shed_repository":{"changeset_revision":"c0b2c2e39c9c","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"15\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Please use: For R1: - For Nextera: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC - For TrueSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC \\", \\"adapter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"single_noindels\\": false}], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}, \\"r2\\": {\\"adapters2\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Please use: For R2: - For Nextera: CTGTCTCTTATACACATCTGACGCTGCCGACGA - For TruSeq: GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\\", \\"adapter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"single_noindels\\": false}], \\"front_adapters2\\": [], \\"anywhere_adapters2\\": []}, \\"pair_adapters\\": false}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"30\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0+galaxy0","type":"tool","uuid":"c7846b4c-54fb-458e-982e-c0d8358a9f5d","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","errors":null,"id":7,"input_connections":{"library|input_1":{"id":6,"output_name":"out_pairs"},"reference_genome|index":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bowtie2","name":"library"},{"description":"runtime parameter for tool Bowtie2","name":"reference_genome"}],"label":"Bowtie2 map on reference","name":"Bowtie2","outputs":[{"name":"output","type":"bam"},{"name":"mapping_stats","type":"txt"}],"position":{"left":709.6000061035156,"top":436.93333435058594},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionmapping_stats":{"action_arguments":{"newname":"mapping stats"},"action_type":"RenameDatasetAction","output_name":"mapping_stats"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"bowtie2 output (BAM)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","tool_shed_repository":{"changeset_revision":"d5ceb9f3c25b","name":"bowtie2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"analysis_type\\": {\\"analysis_type_selector\\": \\"simple\\", \\"__current_case__\\": 0, \\"presets\\": \\"no_presets\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unaligned_file\\": false, \\"aligned_file\\": false, \\"paired_options\\": {\\"paired_options_selector\\": \\"no\\", \\"__current_case__\\": 1}}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"sam_options\\": {\\"sam_options_selector\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_mapping_stats\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.3+galaxy1","type":"tool","uuid":"6fd8444b-f305-4daa-a33a-5cb44e063f39","when":null,"workflow_outputs":[{"label":"mapping stats","output_name":"mapping_stats","uuid":"c7c0e8ff-234e-4c79-a2e3-a48404fcd712"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","errors":null,"id":8,"input_connections":{"input1":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter SAM or BAM, output SAM or BAM","name":"bed_file"}],"label":"filter MAPQ30 concordent pairs","name":"Filter SAM or BAM, output SAM or BAM","outputs":[{"name":"output1","type":"sam"}],"position":{"left":1105,"top":581.3833312988281},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"filtered BAM"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","tool_shed_repository":{"changeset_revision":"649a225999a5","name":"samtool_filter2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"flag\\": {\\"filter\\": \\"yes\\", \\"__current_case__\\": 1, \\"reqBits\\": [\\"0x0002\\"], \\"skipBits\\": null}, \\"header\\": \\"-h\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": \\"\\", \\"mapq\\": \\"30\\", \\"outputtype\\": \\"bam\\", \\"possibly_select_inverse\\": false, \\"read_group\\": \\"\\", \\"regions\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy1","type":"tool","uuid":"bb6e3ac5-cdb0-493c-b534-264ba530a711","when":null,"workflow_outputs":[{"label":"filtered BAM","output_name":"output1","uuid":"c57ebeed-1344-49e2-a7f1-7ad36de8730a"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":9,"input_connections":{"advanced_options|spmr":{"id":5,"output_name":"output"},"effective_genome_size_options|gsize":{"id":4,"output_name":"output"},"treatment|input_treatment_file":{"id":8,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"advanced_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"Call Peaks with MACS2","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_summits","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"},{"name":"output_extra_files","type":"html"}],"position":{"left":1546.183349609375,"top":314.96665954589844},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_extra_files":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_extra_files"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_control_lambda":{"action_arguments":{"newname":"MACS2 control coverage"},"action_type":"RenameDatasetAction","output_name":"output_control_lambda"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"MACS2 narrowPeak"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_summits":{"action_arguments":{"newname":"MACS2 summits"},"action_type":"RenameDatasetAction","output_name":"output_summits"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"MACS2 peaks xls"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"MACS2 treatment coverage"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"1\\", \\"__current_case__\\": 1}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BAMPE\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"create_model\\", \\"__current_case__\\": 0, \\"mfold_lower\\": \\"5\\", \\"mfold_upper\\": \\"50\\", \\"band_width\\": \\"300\\"}, \\"outputs\\": [\\"peaks_tabular\\", \\"summits\\", \\"bdg\\", \\"html\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"06ca906e-c512-4376-9ffc-8eb2b5774af1","when":null,"workflow_outputs":[{"label":"MACS2 summits","output_name":"output_summits","uuid":"33cc76f5-8763-477e-93e5-391dd88bf529"},{"label":"MACS2 peaks","output_name":"output_tabular","uuid":"10114173-d06e-4ad1-aa62-0de5ca17b364"},{"label":"MACS2 narrowPeak","output_name":"output_narrowpeaks","uuid":"7c28605b-1dad-401d-b057-7fad39ee032c"}]},"10":{"annotation":"summary of MACS2","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","errors":null,"id":10,"input_connections":{"infile":{"id":9,"output_name":"output_tabular"}},"inputs":[],"label":"summary of MACS2","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1956.11669921875,"top":185.96665954589844},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"MACS2 report"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"^#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"95832fa1-e96e-4867-8162-d1e39cb1dc46","when":null,"workflow_outputs":[{"label":"MACS2 report","output_name":"output","uuid":"4bae077a-d85a-4e02-b4cc-d125bd10da47"}]},"11":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":11,"input_connections":{"input1":{"id":9,"output_name":"output_treat_pileup"}},"inputs":[],"label":"Bigwig from MACS2","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":2017.933349609375,"top":464.3333282470703},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"coverage from MACS2 (bigwig)"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"0fe57c5d-00a0-4cb6-9bac-97e6d03c6b76","when":null,"workflow_outputs":[{"label":"coverage from MACS2","output_name":"out_file1","uuid":"ccd1cade-8897-41a6-9b14-8e875f60d4e7"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":12,"input_connections":{"results_0|software_cond|input":{"id":6,"output_name":"report"},"results_1|software_cond|input":{"id":7,"output_name":"mapping_stats"},"results_2|software_cond|input":{"id":9,"output_name":"output_tabular"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":"MultiQC","name":"MultiQC","outputs":[{"name":"plots","type":"input"},{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2065.1666870117188,"top":843.3333129882812},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": true, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"bowtie2\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"fb066848-43df-412a-9767-9613bac7d961","when":null,"workflow_outputs":[{"label":"MultiQC webpage","output_name":"html_report","uuid":"175c983c-cb66-430e-aaeb-252b9f520f3b"},{"label":"MultiQC on input dataset(s): Stats","output_name":"stats","uuid":"409a12c8-c81d-4607-a7d8-3f8fe0ab8d02"}]}},"tags":["ChIP"],"uuid":"1834d09c-c683-4424-bc8d-57df156c2fc8","version":1},"readme":"# ChIP-seq Analysis: Paired-End Read Processing\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of dataset pairs of fastqsanger.\\n\\n## Inputs values\\n\\n- adapters sequences: this depends on the library preparation. If you don\'t know, use FastQC to determine if it is Truseq or Nextera.\\n- reference_genome: this field will be adapted to the genomes available for bowtie2.\\n- effective_genome_size: this is used by MACS2 and may be entered manually (indications are provided for heavily used genomes).\\n- normalize_profile: Whether you want to have a profile normalized as Signal to Million Fragments.\\n\\n## Processing\\n\\n- The workflow will remove illumina adapters and low quality bases and filter out any pair with mate smaller than 15bp.\\n- The filtered reads are mapped with bowtie2 with default parameters.\\n- The BAM is filtered to keep only MAPQ30 and concordant pairs.\\n- The peaks are called with MACS2 which at the same time generates a coverage file (normalized or not).\\n- The coverage is converted to bigwig.\\n- A MultiQC is run to have an overview of the QC.\\n\\n### Warning\\n\\n- The filtered bam still has PCR duplicates which are removed by MACS2.\\n\\n## Contribution\\n\\n@lldelisle wrote the workflow.\\n\\n@nagoue updated the tools, made it work in usegalaxy.org, fixed the best practices and wrote the tests.\\n","changelog":"# Changelog\\n\\n## [0.14] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/5.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n\\n## [0.13] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.12] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.11] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.10] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1`\\n\\n## [0.9] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n## [0.8] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n\\n## [0.7] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n\\n## [0.6.1] - 2024-03-14\\n\\nFix bug introduced in 0.6: the second adapter sequence was not used.\\n\\n## [0.6] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n\\n## [0.5] - 2023-10-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n\\n## [0.4] - 2023-06-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n\\n### Manual update\\n- New parameter to get normalized profile\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0`\\n\\n## 2022-10-20\\nChIPseq_PE has been renamed chipseq-pe (still version 0.1)\\n\\n## [0.1] - 2022-10-06\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## ChIP-seq Analysis: Paired-End Read Processing\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ adapter_forward\\"]@{ shape: lean-l }\\n2[\\"ℹ️ adapter_reverse\\"]@{ shape: lean-l }\\n3[\\"ℹ️ reference_genome\\"]@{ shape: lean-l }\\n4[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n5[\\"ℹ️ normalize_profile\\"]@{ shape: lean-l }\\n6[\\"Cutadapt (remove adapter + bad quality bases)\\"]@{ shape: process }\\n0 --> 6\\n1 --> 6\\n2 --> 6\\n7[\\"Bowtie2 map on reference\\"]@{ shape: process }\\n6 --> 7\\n3 --> 7\\n8[\\"filter MAPQ30 concordent pairs\\"]@{ shape: process }\\n7 --> 8\\n9[\\"Call Peaks with MACS2\\"]@{ shape: process }\\n5 --> 9\\n4 --> 9\\n8 --> 9\\n10[\\"summary of MACS2\\"]@{ shape: process }\\n9 --> 10\\n11[\\"Bigwig from MACS2\\"]@{ shape: process }\\n9 --> 11\\n12[\\"MultiQC\\"]@{ shape: process }\\n6 --> 12\\n7 --> 12\\n9 --> 12\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/chipseq-pe/main","dockstore_id":19834,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15078883","tests":[{"doc":"Test outline for ChIPseq_PE","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"wt_H3K4me3","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/record/1324070/files/wt_H3K4me3_read1.fastq.gz","filetype":"fastqsanger"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/record/1324070/files/wt_H3K4me3_read2.fastq.gz","filetype":"fastqsanger"}]}]},"adapter_forward":"GATCGGAAGAGCACACGTCTGAACTCCAGTCAC","adapter_reverse":"GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","reference_genome":"mm10","effective_genome_size":1870000000,"normalize_profile":true},"outputs":{"MultiQC webpage":{"asserts":[{"that":"has_text","text":"wt_H3K4me3"},{"that":"has_text","text":"<a href=\\"#cutadapt_filtered_reads\\" class=\\"nav-l2\\">Filtered Reads</a>"},{"that":"has_text","text":"<a href=\\"#bowtie2\\" class=\\"nav-l1\\">Bowtie 2 / HiSAT2</a>"}]},"MultiQC on input dataset(s): Stats":{"asserts":{"has_line":{"line":"Sample\\tmacs2-d\\tmacs2-treatment_redundant_rate\\tmacs2-peak_count\\tbowtie_2_hisat2-overall_alignment_rate\\tcutadapt-percent_trimmed"},"has_text_matching":{"expression":"wt_H3K4me3\\t201.0\\t0.0\\t11\\t98.[0-9]*\\t10.[0-9]*"}}},"filtered BAM":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_size":{"value":5008461,"delta":200000}}}}},"MACS2 summits":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_n_lines":{"n":11}}}}},"MACS2 peaks":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"# effective genome size = 1.87e+09"},{"that":"has_text","text":"# fragment size is determined as 201 bps"},{"that":"has_text","text":"# fragments after filtering in treatment: 41724"}]}}},"MACS2 narrowPeak":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_n_lines":{"n":11}}}}},"MACS2 report":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"# effective genome size = 1.87e+09"},{"that":"has_text","text":"# fragment size is determined as 201 bps"},{"that":"has_text","text":"# fragments after filtering in treatment: 41724"}]}}},"coverage from MACS2":{"element_tests":{"wt_H3K4me3":{"asserts":{"has_size":{"value":556477,"delta":10000}}}}},"mapping stats":{"element_tests":{"wt_H3K4me3":{"asserts":[{"that":"has_text","text":"1292 (2.79%) aligned concordantly 0 times"},{"that":"has_text","text":"39329 (84.93%) aligned concordantly exactly 1 time"},{"that":"has_text","text":"5686 (12.28%) aligned concordantly >1 times"}]}}}}}]}],"path":"./workflows/epigenetics/chipseq-pe"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/cutandrun.ga","testParameterFiles":["/cutandrun-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete CUT&RUN/CUT&TAG analysis workflow for paired-end sequencing data. Processes raw FASTQ files through adapter removal (cutadapt) and alignment (Bowtie2 with dovetail option enabled). Applies quality filtering (MAPQ ≥ 30, concordant pairs only), converts BAM to BED format, and performs peak calling using MACS2 with parameters optimized for the punctate signal profile characteristic of CUT&RUN/CUT&TAG experiments.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.14","name":"CUT&RUN/CUT&TAG Analysis: Protein-DNA Interaction Mapping","steps":{"0":{"annotation":"Should be a paired collection with CUT and RUN fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a paired collection with CUT and RUN fastqs","name":"PE fastq input"}],"label":"PE fastq input","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":268},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"9fbb875d-05b1-4557-bd2e-710f80dd1a21","when":null,"workflow_outputs":[]},"1":{"annotation":"Please use: For R1: - For TrueSeq (CUT and RUN): GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTCCGAGCCCACGAGAC ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Please use: For R1: - For TrueSeq (CUT and RUN): GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTCCGAGCCCACGAGAC ","name":"adapter_forward"}],"label":"adapter_forward","name":"Input parameter","outputs":[],"position":{"left":22,"top":375},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1eac70a4-a4b7-42c7-82cd-913f23b4f941","when":null,"workflow_outputs":[]},"2":{"annotation":"Please use: For R2: - For TruSeq (CUT and RUN): GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTGACGCTGCCGACGA","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Please use: For R2: - For TruSeq (CUT and RUN): GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTGACGCTGCCGACGA","name":"adapter_reverse"}],"label":"adapter_reverse","name":"Input parameter","outputs":[],"position":{"left":62,"top":483},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"68b198e5-7de8-445f-807f-7f432090e2c7","when":null,"workflow_outputs":[]},"3":{"annotation":"reference_genome","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"reference_genome","name":"reference_genome"}],"label":"reference_genome","name":"Input parameter","outputs":[],"position":{"left":128,"top":589},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"7f16a988-8ead-4a4a-9be1-8f5fbb744dec","when":null,"workflow_outputs":[]},"4":{"annotation":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Used by MACS2: H. sapiens: 2700000000, M. musculus: 1870000000, D. melanogaster: 120000000, C. elegans: 90000000","name":"effective_genome_size"}],"label":"effective_genome_size","name":"Input parameter","outputs":[],"position":{"left":176,"top":689},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"aae90433-355d-4f27-bdfb-77f8bc4fcff5","when":null,"workflow_outputs":[]},"5":{"annotation":"Whether you want to have a profile normalized as Signal to Million Reads","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Whether you want to have a profile normalized as Signal to Million Reads","name":"normalize_profile"}],"label":"normalize_profile","name":"Input parameter","outputs":[],"position":{"left":228,"top":780.4016399999857},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a2c8d3e2-7542-4feb-8523-6100983ddb3b","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1","errors":null,"id":6,"input_connections":{"library|input_1":{"id":0,"output_name":"output"},"library|r1|adapters_0|adapter_source|adapter":{"id":1,"output_name":"output"},"library|r2|adapters2_0|adapter_source|adapter":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cutadapt","name":"library"}],"label":"Cutadapt (remove adapter + bad quality bases)","name":"Cutadapt","outputs":[{"name":"out_pairs","type":"input"},{"name":"report","type":"txt"}],"position":{"left":288,"top":180},"post_job_actions":{"HideDatasetActionout_pairs":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_pairs"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"},"RenameDatasetActionreport":{"action_arguments":{"newname":"cutadapt report"},"action_type":"RenameDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1","tool_shed_repository":{"changeset_revision":"5eb7e84243f2","name":"cutadapt","owner":"lparsons","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_options\\": {\\"action\\": \\"trim\\", \\"error_rate\\": \\"0.1\\", \\"no_indels\\": false, \\"times\\": \\"1\\", \\"overlap\\": \\"3\\", \\"match_read_wildcards\\": false, \\"no_match_adapter_wildcards\\": true, \\"revcomp\\": false}, \\"filter_options\\": {\\"discard_trimmed\\": false, \\"discard_untrimmed\\": false, \\"minimum_length\\": \\"15\\", \\"minimum_length2\\": null, \\"maximum_length\\": null, \\"maximum_length2\\": null, \\"max_n\\": null, \\"max_expected_errors\\": null, \\"max_average_error_rate\\": null, \\"discard_casava\\": false, \\"pair_filter\\": \\"any\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"r1\\": {\\"adapters\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Please use: For R1: - For TrueSeq (CUT and RUN): GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTCCGAGCCCACGAGAC \\", \\"adapter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"single_noindels\\": false}], \\"front_adapters\\": [], \\"anywhere_adapters\\": []}, \\"r2\\": {\\"adapters2\\": [{\\"__index__\\": 0, \\"adapter_source\\": {\\"adapter_source_list\\": \\"user\\", \\"__current_case__\\": 0, \\"adapter_name\\": \\"Please use: For R2: - For TruSeq (CUT and RUN): GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT - For Nextera (CUT and TAG): CTGTCTCTTATACACATCTGACGCTGCCGACGA\\", \\"adapter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"single_noindels\\": false}], \\"front_adapters2\\": [], \\"anywhere_adapters2\\": []}, \\"pair_adapters\\": false}, \\"other_trimming_options\\": {\\"cut\\": \\"0\\", \\"cut2\\": \\"0\\", \\"quality_cutoff\\": \\"30\\", \\"quality_cutoff2\\": \\"\\", \\"nextseq_trim\\": \\"0\\", \\"trim_n\\": false, \\"poly_a\\": false, \\"shorten_options\\": {\\"shorten_values\\": \\"False\\", \\"__current_case__\\": 1}, \\"shorten_options_r2\\": {\\"shorten_values_r2\\": \\"False\\", \\"__current_case__\\": 1}}, \\"output_selector\\": [\\"report\\"], \\"read_mod_options\\": {\\"strip_suffix\\": \\"\\", \\"length_tag\\": \\"\\", \\"rename\\": \\"\\", \\"zero_cap\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.9+galaxy1","type":"tool","uuid":"774b0604-628f-46a1-9088-a59d082e5317","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","errors":null,"id":7,"input_connections":{"library|input_1":{"id":6,"output_name":"out_pairs"},"reference_genome|index":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bowtie2","name":"library"},{"description":"runtime parameter for tool Bowtie2","name":"reference_genome"}],"label":"Bowtie2 map on reference","name":"Bowtie2","outputs":[{"name":"output","type":"bam"},{"name":"mapping_stats","type":"txt"}],"position":{"left":559,"top":102},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionmapping_stats":{"action_arguments":{"newname":"mapping stats"},"action_type":"RenameDatasetAction","output_name":"mapping_stats"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"bowtie2 output (BAM)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1","tool_shed_repository":{"changeset_revision":"d5ceb9f3c25b","name":"bowtie2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"analysis_type\\": {\\"analysis_type_selector\\": \\"simple\\", \\"__current_case__\\": 0, \\"presets\\": \\"--very-sensitive\\"}, \\"library\\": {\\"type\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"unaligned_file\\": false, \\"aligned_file\\": false, \\"paired_options\\": {\\"paired_options_selector\\": \\"yes\\", \\"__current_case__\\": 0, \\"I\\": \\"0\\", \\"X\\": \\"1000\\", \\"fr_rf_ff\\": \\"--fr\\", \\"no_mixed\\": false, \\"no_discordant\\": false, \\"dovetail\\": true, \\"no_contain\\": false, \\"no_overlap\\": false}}, \\"reference_genome\\": {\\"source\\": \\"indexed\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"sam_options\\": {\\"sam_options_selector\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_mapping_stats\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.3+galaxy1","type":"tool","uuid":"407d46cc-d908-44eb-a4dd-125537498b17","when":null,"workflow_outputs":[{"label":"Mapping stats","output_name":"mapping_stats","uuid":"87e15eab-1e45-4838-88b7-d2d8594f0de1"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","errors":null,"id":8,"input_connections":{"input1":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Filter SAM or BAM, output SAM or BAM","name":"bed_file"}],"label":"filter MAPQ30 concordant pairs","name":"Filter SAM or BAM, output SAM or BAM","outputs":[{"name":"output1","type":"sam"}],"position":{"left":834,"top":23},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"RenameDatasetActionoutput1":{"action_arguments":{"newname":"filtered BAM"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtool_filter2/samtool_filter2/1.8+galaxy1","tool_shed_repository":{"changeset_revision":"649a225999a5","name":"samtool_filter2","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bed_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"flag\\": {\\"filter\\": \\"yes\\", \\"__current_case__\\": 1, \\"reqBits\\": [\\"0x0002\\"], \\"skipBits\\": null}, \\"header\\": \\"-h\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"library\\": \\"\\", \\"mapq\\": \\"30\\", \\"outputtype\\": \\"bam\\", \\"possibly_select_inverse\\": false, \\"read_group\\": \\"\\", \\"regions\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy1","type":"tool","uuid":"b04312a6-1680-4d8e-9c70-f30f2d2be3cc","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","errors":null,"id":9,"input_connections":{"inputFile":{"id":8,"output_name":"output1"}},"inputs":[],"label":"remove PCR duplicates","name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":1112,"top":64},"post_job_actions":{"RenameDatasetActionmetrics_file":{"action_arguments":{"newname":"MarkDuplicates metrics"},"action_type":"RenameDatasetAction","output_name":"metrics_file"},"RenameDatasetActionoutFile":{"action_arguments":{"newname":"BAM filtered rmDup"},"action_type":"RenameDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": true, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"7f78ad5a-5250-4ced-aa7f-cf0802051c82","when":null,"workflow_outputs":[{"label":"BAM filtered rmDup","output_name":"outFile","uuid":"4cb5b980-39bd-40ab-ad0a-8c90beffd11c"},{"label":"MarkDuplicates metrics","output_name":"metrics_file","uuid":"3cec58d5-3851-4ccb-bab5-ae3f550df1e5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","errors":null,"id":10,"input_connections":{"input":{"id":9,"output_name":"outFile"}},"inputs":[],"label":"convert BAM to BED to improve peak calling","name":"bedtools BAM to BED","outputs":[{"name":"output","type":"bed"}],"position":{"left":1387,"top":37},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"BED filtered rmDup"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"ed_score\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"option\\": \\"\\", \\"split\\": false, \\"tag\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1+galaxy0","type":"tool","uuid":"f447290b-bd3d-403d-bc67-0765124b7a97","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","errors":null,"id":11,"input_connections":{"advanced_options|spmr":{"id":5,"output_name":"output"},"effective_genome_size_options|gsize":{"id":4,"output_name":"output"},"treatment|input_treatment_file":{"id":10,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MACS2 callpeak","name":"advanced_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"effective_genome_size_options"},{"description":"runtime parameter for tool MACS2 callpeak","name":"treatment"}],"label":"Call Peaks with MACS2","name":"MACS2 callpeak","outputs":[{"name":"output_tabular","type":"tabular"},{"name":"output_narrowpeaks","type":"bed"},{"name":"output_summits","type":"bed"},{"name":"output_treat_pileup","type":"bedgraph"},{"name":"output_control_lambda","type":"bedgraph"},{"name":"output_extra_files","type":"html"}],"position":{"left":1656,"top":3},"post_job_actions":{"HideDatasetActionoutput_control_lambda":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_control_lambda"},"HideDatasetActionoutput_extra_files":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_extra_files"},"HideDatasetActionoutput_treat_pileup":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_treat_pileup"},"RenameDatasetActionoutput_control_lambda":{"action_arguments":{"newname":"MACS2 control coverage"},"action_type":"RenameDatasetAction","output_name":"output_control_lambda"},"RenameDatasetActionoutput_narrowpeaks":{"action_arguments":{"newname":"MACS2 narrowPeak"},"action_type":"RenameDatasetAction","output_name":"output_narrowpeaks"},"RenameDatasetActionoutput_summits":{"action_arguments":{"newname":"MACS2 summits"},"action_type":"RenameDatasetAction","output_name":"output_summits"},"RenameDatasetActionoutput_tabular":{"action_arguments":{"newname":"MACS2 peaks xls"},"action_type":"RenameDatasetAction","output_name":"output_tabular"},"RenameDatasetActionoutput_treat_pileup":{"action_arguments":{"newname":"MACS2 treatment coverage"},"action_type":"RenameDatasetAction","output_name":"output_treat_pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"86e2413cf3f8","name":"macs2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"to_large\\": false, \\"nolambda\\": false, \\"spmr\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ratio\\": null, \\"slocal\\": null, \\"llocal\\": null, \\"broad_options\\": {\\"broad_options_selector\\": \\"nobroad\\", \\"__current_case__\\": 1, \\"call_summits\\": true}, \\"keep_dup_options\\": {\\"keep_dup_options_selector\\": \\"all\\", \\"__current_case__\\": 2}, \\"d_min\\": \\"20\\", \\"buffer_size\\": \\"100000\\"}, \\"control\\": {\\"c_select\\": \\"No\\", \\"__current_case__\\": 1}, \\"cutoff_options\\": {\\"cutoff_options_selector\\": \\"qvalue\\", \\"__current_case__\\": 1, \\"qvalue\\": \\"0.05\\"}, \\"effective_genome_size_options\\": {\\"effective_genome_size_options_selector\\": \\"user_defined\\", \\"__current_case__\\": 4, \\"gsize\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"format\\": \\"BED\\", \\"nomodel_type\\": {\\"nomodel_type_selector\\": \\"nomodel\\", \\"__current_case__\\": 1, \\"extsize\\": \\"200\\", \\"shift\\": \\"-100\\"}, \\"outputs\\": [\\"peaks_tabular\\", \\"summits\\", \\"bdg\\", \\"html\\"], \\"treatment\\": {\\"t_multi_select\\": \\"No\\", \\"__current_case__\\": 0, \\"input_treatment_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.9.1+galaxy0","type":"tool","uuid":"1a6316d7-3ee1-482a-9264-77bc98090d73","when":null,"workflow_outputs":[{"label":"MACS2 summits","output_name":"output_summits","uuid":"7bed750a-c0a6-4fda-8f10-4605db362045"},{"label":"MACS2 narrowPeak","output_name":"output_narrowpeaks","uuid":"be4e6c87-fb7b-40ed-ac6b-971aeda3631f"},{"label":"MACS2 peaks xls","output_name":"output_tabular","uuid":"45eb5a1f-6036-40bf-8c10-5672cb1258ab"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"output_tabular"}},"inputs":[],"label":"summary of MACS2","name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":1946,"top":0},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"MACS2 report"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"^#\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"c3ff2b9d-d53a-40d9-8f4c-a7a2461dd7f5","when":null,"workflow_outputs":[{"label":"MACS2 report","output_name":"output","uuid":"8bf8fad7-3493-4424-a5bf-278457288166"}]},"13":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":13,"input_connections":{"input1":{"id":11,"output_name":"output_treat_pileup"}},"inputs":[],"label":"Bigwig from MACS2","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1945,"top":139},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Coverage from MACS2 (bigwig)"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"4bac5ae5-8119-4ced-92cc-b2281d6199ba","when":null,"workflow_outputs":[{"label":"Coverage from MACS2 (bigwig)","output_name":"out_file1","uuid":"4a8e2e9d-4896-4ffd-99b2-a3492e71053c"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":14,"input_connections":{"results_0|software_cond|input":{"id":6,"output_name":"report"},"results_1|software_cond|input":{"id":7,"output_name":"mapping_stats"},"results_2|software_cond|output_0|input":{"id":9,"output_name":"metrics_file"},"results_3|software_cond|input":{"id":11,"output_name":"output_tabular"}},"inputs":[],"label":"MultiQC","name":"MultiQC","outputs":[{"name":"plots","type":"input"},{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1936,"top":415},"post_job_actions":{"HideDatasetActionplots":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"plots"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": true, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"cutadapt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"bowtie2\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, {\\"__index__\\": 3, \\"software_cond\\": {\\"software\\": \\"macs2\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"0edf5e04-cc38-414c-9b57-117e919c435b","when":null,"workflow_outputs":[{"label":"MultiQC on input dataset(s): Stats","output_name":"stats","uuid":"59dfba75-8658-4213-8fd9-b3ce5a916f14"},{"label":"MultiQC webpage","output_name":"html_report","uuid":"6d87436a-550e-4d76-a5da-d4463dfd4473"}]}},"tags":["CUTnRUN"],"uuid":"98e9e34b-b203-43db-9a23-5fa72963fa6d","version":1},"readme":"# CUT&RUN/CUT&TAG Analysis: Protein-DNA Interaction Mapping\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of dataset pairs of fastqsanger.\\n\\n## Inputs values\\n\\n- adapter sequences: this depends on the library preparation. Usually CUT&RUN is Truseq and CUT&TAG is Nextera. If you don\'t know, use FastQC to determine if it is Truseq or Nextera\\n- reference_genome: this field will be adapted to the genomes available for bowtie2\\n- effective_genome_size: this is used by macs2 and may be entered manually (indications are provided for heavily used genomes)\\n- normalize_profile: Whether you want to have a profile normalized as Signal to Million Reads.\\n\\n## Processing\\n\\n- The workflow will remove illumina adapters and low quality bases and filter out any read smaller than 15bp\\n- The filtered reads are mapped with bowtie2 allowing dovetail and fragment length up to 1kb\\n- The BAM is filtered to keep only MAPQ30 and concordant pairs\\n- The PCR duplicates are removed with Picard (only from version 0.6)\\n- The BAM is converted to BED to enable macs2 to take both pairs into account\\n- The peaks are called with macs2 which at the same time generates a coverage file (normalized or not).\\n- The coverage file is converted to bigwig\\n- A multiQC is run to have an overview of the QC\\n","changelog":"# Changelog\\n\\n## [0.14] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [0.13] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.12] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.11] - 2024-05-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.31.1+galaxy0`\\n\\n## [0.10] - 2024-04-22\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy0`\\n\\n## [0.9] - 2024-04-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy1`\\n\\n## [0.8] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.7+galaxy0`\\n\\n## [0.7] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.4` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/9.3+galaxy0`\\n\\n## [0.6.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.6] - 2023-10-19\\n\\nFix the remove duplicate step!\\nIn all previous versions, due to an error, PCR duplicates were not removed.\\n\\n## [0.5] - 2023-10-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.9.1+galaxy0`\\n\\n## [0.4] - 2023-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.3` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.4`\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0`\\n\\n### Manual update\\n- New parameter to get normalized profile\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.0+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.5+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0`\\n\\n## [0.1] - 2022-10-06\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## CUT&RUN/CUT&TAG Analysis: Protein-DNA Interaction Mapping\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ PE fastq input\\"]@{ shape: docs }\\n1[\\"ℹ️ adapter_forward\\"]@{ shape: lean-l }\\n2[\\"ℹ️ adapter_reverse\\"]@{ shape: lean-l }\\n3[\\"ℹ️ reference_genome\\"]@{ shape: lean-l }\\n4[\\"ℹ️ effective_genome_size\\"]@{ shape: lean-l }\\n5[\\"ℹ️ normalize_profile\\"]@{ shape: lean-l }\\n6[\\"Cutadapt (remove adapter + bad quality bases)\\"]@{ shape: process }\\n0 --> 6\\n1 --> 6\\n2 --> 6\\n7[\\"Bowtie2 map on reference\\"]@{ shape: process }\\n6 --> 7\\n3 --> 7\\n8[\\"filter MAPQ30 concordant pairs\\"]@{ shape: process }\\n7 --> 8\\n9[\\"remove PCR duplicates\\"]@{ shape: process }\\n8 --> 9\\n10[\\"convert BAM to BED to improve peak calling\\"]@{ shape: process }\\n9 --> 10\\n11[\\"Call Peaks with MACS2\\"]@{ shape: process }\\n5 --> 11\\n4 --> 11\\n10 --> 11\\n12[\\"summary of MACS2\\"]@{ shape: process }\\n11 --> 12\\n13[\\"Bigwig from MACS2\\"]@{ shape: process }\\n11 --> 13\\n14[\\"MultiQC\\"]@{ shape: process }\\n6 --> 14\\n7 --> 14\\n9 --> 14\\n11 --> 14\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/cutandrun/main","dockstore_id":19825,"categories":[],"collections":["Epigenetics"],"doi":"10.5281/zenodo.15077995","tests":[{"doc":"Test outline for CUT&RUN (hg38 bigwig conversion fails for memory issues)","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"Rep1","elements":[{"identifier":"forward","class":"File","location":"https://zenodo.org/record/6823059/files/Rep1_R1.fastq","filetype":"fastqsanger"},{"identifier":"reverse","class":"File","location":"https://zenodo.org/record/6823059/files/Rep1_R2.fastq","filetype":"fastqsanger"}]}]},"adapter_forward":"GATCGGAAGAGCACACGTCTGAACTCCAGTCAC","adapter_reverse":"GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","reference_genome":"hg38canon","effective_genome_size":2700000000,"normalize_profile":false},"outputs":{"Mapping stats":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"289103 reads; of these:"},"has_text_matching":{"expression":"99.39% overall alignment rate"}}}}},"BAM filtered rmDup":{"element_tests":{"Rep1":{"asserts":{"has_size":{"value":8661584,"delta":800000}}}}},"MarkDuplicates metrics":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"0.33"}}}}},"MACS2 summits":{"element_tests":{"Rep1":{"asserts":{"has_n_lines":{"n":5870}}}}},"MACS2 narrowPeak":{"element_tests":{"Rep1":{"asserts":{"has_n_lines":{"n":5870}}}}},"MACS2 peaks xls":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"# tag size is determined as 40 bps"},"has_text_matching":{"expression":"# total tags in treatment: 238930"}}}}}}},{"doc":"Test outline for CUT&RUN (dm6)","job":{"PE fastq input":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"Rep1","elements":[{"identifier":"forward","class":"File","path":"test-data/SRR15904259_subset_forward.fastqsanger.gz","filetype":"fastqsanger"},{"identifier":"reverse","class":"File","path":"test-data/SRR15904259_subset_reverse.fastqsanger.gz","filetype":"fastqsanger"}]}]},"adapter_forward":"GATCGGAAGAGCACACGTCTGAACTCCAGTCAC","adapter_reverse":"GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","reference_genome":"dm6","effective_genome_size":120000000,"normalize_profile":true},"outputs":{"Mapping stats":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"343237 reads; of these:"},"has_text_matching":{"expression":"98.17% overall alignment rate"}}}}},"BAM filtered rmDup":{"element_tests":{"Rep1":{"asserts":{"has_size":{"value":18246620,"delta":1000000}}}}},"MarkDuplicates metrics":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"0.001269"}}}}},"MACS2 summits":{"element_tests":{"Rep1":{"asserts":{"has_n_lines":{"n":3231}}}}},"MACS2 narrowPeak":{"element_tests":{"Rep1":{"asserts":{"has_n_lines":{"n":3231}}}}},"MACS2 peaks xls":{"element_tests":{"Rep1":{"asserts":{"has_text":{"text":"# tag size is determined as 36 bps"},"has_text_matching":{"expression":"# total tags in treatment: 489716"}}}}},"Coverage from MACS2 (bigwig)":{"element_tests":{"Rep1":{"asserts":{"has_size":{"value":5087932,"delta":400000}}}}}}}]}],"path":"./workflows/epigenetics/cutandrun"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/average-bigwig-between-replicates.ga","testParameterFiles":["/average-bigwig-between-replicates-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Calculates average signal values across replicate BigWig files to generate a single consolidated track. Requires input file identifiers formatted as \\"sample_name_replicateID\\". The tool automatically groups files by sample name, computes the mean signal at each position, and outputs consolidated BigWig files named after the common sample prefix (without the replicate ID suffix).","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"BigWig Replicates Averaging Workflow","steps":{"0":{"annotation":"We assume the identifiers are like:\\nsample_name_replicateID","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"We assume the identifiers are like:\\nsample_name_replicateID","name":"Bigwig to average"}],"label":"Bigwig to average","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"45d1d287-ea71-4817-afd2-8ad89aedac46","when":null,"workflow_outputs":[]},"1":{"annotation":"Bin size for normalized bigwig (usually 5bp is sufficient for RNA-seq and 50bp for ChIP/CUTandRUN/ATAC)","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Bin size for normalized bigwig (usually 5bp is sufficient for RNA-seq and 50bp for ChIP/CUTandRUN/ATAC)","name":"bin_size"}],"label":"bin_size","name":"Input parameter","outputs":[],"position":{"left":16,"top":125},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ae41572a-bafb-40d4-8042-87f07552980f","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":254,"top":11.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [1, 2], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"^(.*)_([^_]*)$\\", \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\1\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"expression\\": \\"^(.*)_([^_]*)$\\", \\"is_workflow\\": false, \\"replacement\\": \\"\\\\\\\\2\\", \\"target_column\\": 0, \\"type\\": \\"add_column_regex\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"5532681e-f2b3-415c-87e2-71d5eb982d62","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","errors":null,"id":3,"input_connections":{"advancedOpt|binSize":{"id":1,"output_name":"output"},"bigwigs":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool bigwigAverage","name":"advancedOpt"}],"label":"average bigwigs from different replicates","name":"bigwigAverage","outputs":[{"name":"outFileName","type":"bigwig"}],"position":{"left":474,"top":11.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0","tool_shed_repository":{"changeset_revision":"4a53856a5b85","name":"deeptools_bigwig_average","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advancedOpt\\": {\\"showAdvancedOpt\\": \\"yes\\", \\"__current_case__\\": 1, \\"binSize\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"skipNAs\\": false, \\"scaleFactors\\": \\"1\\", \\"blackListFileName\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"bigwigs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"outFileFormat\\": \\"bigwig\\", \\"region\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.5.4+galaxy0","type":"tool","uuid":"f63daf4e-ce9f-4301-be74-e4c263b6e88b","when":null,"workflow_outputs":[{"label":"average_bigwigs","output_name":"outFileName","uuid":"19023604-eee1-4099-b1c2-abe3de93b3f3"}]}},"tags":[],"uuid":"c6ed4b0e-d719-4997-9860-ba99488ec332","version":6},"readme":"# BigWig Replicates Averaging Workflow\\n\\nThis workflow is very useful when you processed multiple samples in collections and you want to generate an average coverage per condition.\\n\\n## Inputs dataset\\n\\n- The workflow needs a single input which is a list of bigwigs (normalized). The identifiers of your bigwigs must be like:\\n  - whatever_sample1_identificationOfReplicate1\\n  - whatever_sample1_identificationOfReplicate2\\n  - ...\\n  - whatever_sample2_identificationOfReplicate1\\n  - whatever_sample2_identificationOfReplicate2\\n  - ...\\n\\n## Inputs values\\n\\n- bin_size: this is used when average of coverage is performed. Large values will allow to have smaller output files but with less resolution while small values will increase computation time and size of output files to produce more resolutive bigwigs. I suggest 5bp for RNA-seq and 50bp for other applications.\\n\\n## Processing\\n\\n- The workflow will split identifiers between everything which is before the last underscore which will be the *sample* and everything which is after the last underscore which will be the *replicate identifier*. And restructure the collection as list:list:\\n  - whatever_sample1:\\n    - identificationOfReplicate1\\n    - identificationOfReplicate2\\n    - ...\\n  - whatever_sample2:\\n    - identificationOfReplicate1\\n    - identificationOfReplicate2\\n    - ---\\n  - ...\\n- Then it will average bigwigs into each inner list\\n\\n## Outputs\\n\\n- The output is a collection of bigwig datasets like:\\n  - whatever_sample1\\n  - whatever_sample2\\n  - ...\\n","changelog":"# Changelog\\n\\n## [0.2] - 2023-09-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bigwig_average/deeptools_bigwig_average/3.5.4+galaxy0`\\n\\n## [0.1] - 2023-09-22\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## BigWig Replicates Averaging Workflow\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Bigwig to average\\"]@{ shape: docs }\\n1[\\"ℹ️ bin_size\\"]@{ shape: lean-l }\\n2[\\"Apply rules\\"]@{ shape: process }\\n0 --> 2\\n3[\\"average bigwigs from different replicates\\"]@{ shape: process }\\n1 --> 3\\n2 --> 3\\n```\\n","updated":"2023-09-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/average-bigwig-between-replicates/main","dockstore_id":23760,"categories":[],"collections":["Epigenetics","Transcriptomics"],"doi":"10.5281/zenodo.15078852","tests":[{"doc":"Test outline for average-bigwig-between-replicates.ga","job":{"Bigwig to average":{"class":"Collection","collection_type":"list","elements":[{"identifier":"ATAC_HH19_PT_rep1","class":"File","location":"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM6152756&format=file&file=GSM6152756%5FATAC%5FHH19%5FPT%5Frep1%2Ebigwig","filetype":"bigwig"},{"identifier":"ATAC_HH19_PT_rep2","class":"File","location":"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM6152757&format=file&file=GSM6152757%5FATAC%5FHH19%5FPT%5Frep2%2Ebigwig","filetype":"bigwig"},{"identifier":"ATAC_HH35_DS_rep1","class":"File","location":"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM6152758&format=file&file=GSM6152758%5FATAC%5FHH35%5FDS%5Frep1%2Ebigwig","filetype":"bigwig"},{"identifier":"ATAC_HH35_DS_rep2","class":"File","location":"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM6152759&format=file&file=GSM6152759%5FATAC%5FHH35%5FDS%5Frep2%2Ebigwig","filetype":"bigwig"}]},"bin_size":50},"outputs":{"average_bigwigs":{"attributes":{"collection_type":"list"},"element_tests":{"ATAC_HH19_PT":{"asserts":{"has_size":{"value":140203524,"delta":10000000}}},"ATAC_HH35_DS":{"asserts":{"has_size":{"value":124599299,"delta":10000000}}}}}}}]}],"path":"./workflows/epigenetics/average-bigwig-between-replicates"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/fragment-based-docking-scoring.ga","testParameterFiles":["/fragment-based-docking-scoring-tests.yml"],"authors":[{"name":"Simon Bray","orcid":"0000-0002-0621-6705"},{"name":"Tim Dudgeon","orcid":"0000-0001-6879-5194"}],"definition":{"a_galaxy_workflow":"true","annotation":"Virtual screening of the SARS-CoV-2 main protease with rDock and pose scoring","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"},{"class":"Person","identifier":"https://orcid.org/0000-0001-6879-5194","name":"Tim Dudgeon"}],"format-version":"0.1","license":"MIT","name":"Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring","release":"0.1.5","steps":{"0":{"annotation":"Number of docking poses to generate per input compound","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Number of docking poses to generate per input compound","name":"Number of poses"}],"label":"Number of poses","name":"Input parameter","outputs":[],"position":{"bottom":391.3000030517578,"height":61.80000305175781,"left":144,"right":344,"top":329.5,"width":200,"x":144,"y":329.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"52331e17-340d-4c08-8faf-b83ac383f1ac","workflow_outputs":[]},"1":{"annotation":"PDB file for the receptor protein","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"PDB file for the receptor protein","name":"Receptor (PDB)"}],"label":"Receptor (PDB)","name":"Input dataset","outputs":[],"position":{"bottom":441.3000030517578,"height":61.80000305175781,"left":-412,"right":-212,"top":379.5,"width":200,"x":-412,"y":379.5},"tool_id":null,"tool_state":"{\\"optional\\": false}","tool_version":null,"type":"data_input","uuid":"156f4aeb-d160-456a-974e-7884b79cb83a","workflow_outputs":[]},"2":{"annotation":"Fragments in SDF format","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Fragments in SDF format","name":"All fragments (SDF)"}],"label":"All fragments (SDF)","name":"Input dataset","outputs":[],"position":{"bottom":633.6999969482422,"height":82.19999694824219,"left":-412,"right":-212,"top":551.5,"width":200,"x":-412,"y":551.5},"tool_id":null,"tool_state":"{\\"optional\\": false}","tool_version":null,"type":"data_input","uuid":"dd2d3bc0-8138-4e66-b2d9-e5e90e27b659","workflow_outputs":[]},"3":{"annotation":"Parameter for parallelization of docking","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Parameter for parallelization of docking","name":"Collection size for docking"}],"label":"Collection size for docking","name":"Input parameter","outputs":[],"position":{"bottom":790.1000061035156,"height":102.60000610351562,"left":-134,"right":66,"top":687.5,"width":200,"x":-134,"y":687.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e731045a-b2e3-4766-97b0-173be3f8f098","workflow_outputs":[]},"4":{"annotation":"Value between 0 and 1. All poses with a SuCOS score lower than this value will be filtered out.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Value between 0 and 1. All poses with a SuCOS score lower than this value will be filtered out.","name":"SuCOS threshold"}],"label":"SuCOS threshold","name":"Input parameter","outputs":[],"position":{"bottom":205.3000030517578,"height":61.80000305175781,"left":710,"right":910,"top":143.5,"width":200,"x":710,"y":143.5},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f85faf19-ac80-4e33-980e-38a24ccb877d","workflow_outputs":[{"label":null,"output_name":"output","uuid":"b8f1c765-0ea2-46af-93ea-49b325f0f6c9"}]},"5":{"annotation":"SDF or MOL file with the single fragment for SuCOS scoring","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"SDF or MOL file with the single fragment for SuCOS scoring","name":"Fragment for SuCOS scoring (SDF/MOL)"}],"label":"Fragment for SuCOS scoring (SDF/MOL)","name":"Input dataset","outputs":[],"position":{"bottom":325.6999969482422,"height":82.19999694824219,"left":710,"right":910,"top":243.5,"width":200,"x":710,"y":243.5},"tool_id":null,"tool_state":"{\\"optional\\": false}","tool_version":null,"type":"data_input","uuid":"43dd0cc4-fa60-4087-a4c7-7d248f6b2b2e","workflow_outputs":[]},"6":{"annotation":"List of compounds for docking in SMILES format","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"List of compounds for docking in SMILES format","name":"Candidate compounds (SMILES)"}],"label":"Candidate compounds (SMILES)","name":"Input dataset","outputs":[],"position":{"bottom":957.6999969482422,"height":82.19999694824219,"left":-690,"right":-490,"top":875.5,"width":200,"x":-690,"y":875.5},"tool_id":null,"tool_state":"{\\"optional\\": false}","tool_version":null,"type":"data_input","uuid":"47bf8c53-ae1e-4af6-bece-6adfd74b627c","workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","errors":null,"id":7,"input_connections":{"infile":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compound conversion","outputs":[{"name":"outfile","type":"text"}],"position":{"bottom":497.3000030517578,"height":174.8000030517578,"left":-134,"right":66,"top":322.5,"width":200,"x":-134,"y":322.5},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"e2c36f62e22f","name":"openbabel_compound_convert","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"appendtotitle\\": \\"\\", \\"dative_bonds\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"oformat\\": {\\"oformat_opts_selector\\": \\"mol2\\", \\"__current_case__\\": 41, \\"mol2_ignore_res\\": \\"false\\", \\"gen2d\\": \\"false\\", \\"gen3d\\": \\"false\\"}, \\"ph\\": \\"7.4\\", \\"remove_h\\": \\"false\\", \\"split\\": \\"false\\", \\"unique\\": {\\"unique_opts_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"b689d8e6-c474-4763-afb0-3e30610aac57","workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_frankenstein_ligand/ctb_frankenstein_ligand/2013.1-0+galaxy0","errors":null,"id":8,"input_connections":{"infile":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Create Frankenstein ligand","outputs":[{"name":"outfile","type":"sdf"}],"position":{"bottom":649.1000061035156,"height":113.60000610351562,"left":-134,"right":66,"top":535.5,"width":200,"x":-134,"y":535.5},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_frankenstein_ligand/ctb_frankenstein_ligand/2013.1-0+galaxy0","tool_shed_repository":{"changeset_revision":"7255688c77f3","name":"ctb_frankenstein_ligand","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2013.1-0+galaxy0","type":"tool","uuid":"081abe35-347a-4244-99ae-ec35329e3de0","workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":9,"input_connections":{"components_1|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"bottom":251.89999389648438,"height":154.39999389648438,"left":988,"right":1188,"top":97.5,"width":200,"x":988,"y":97.5},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"$SuCOS_Score >= \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"8beea755-90e6-4151-8c26-f8e9e8f24bfa","workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/enumerate_charges/enumerate_charges/2020.03.4+galaxy0","errors":null,"id":10,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Enumerate changes","outputs":[{"name":"output","type":"smi"}],"position":{"bottom":973.1000061035156,"height":113.60000610351562,"left":-412,"right":-212,"top":859.5,"width":200,"x":-412,"y":859.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/enumerate_charges/enumerate_charges/2020.03.4+galaxy0","tool_shed_repository":{"changeset_revision":"bbbf5fb356dd","name":"enumerate_charges","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"max_ph\\": \\"10.4\\", \\"min_ph\\": \\"4.4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2020.03.4+galaxy0","type":"tool","uuid":"73172813-0c22-4d96-846e-c5dbe6702fce","workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rxdock_rbcavity/rxdock_rbcavity/2013.1.1_148c5bd1+galaxy0","errors":null,"id":11,"input_connections":{"ligand":{"id":8,"output_name":"outfile"},"receptor":{"id":7,"output_name":"outfile"}},"inputs":[],"label":null,"name":"rDock cavity definition","outputs":[{"name":"activesite","type":"rdock_as"}],"position":{"bottom":594.8999938964844,"height":164.39999389648438,"left":144,"right":344,"top":430.5,"width":200,"x":144,"y":430.5},"post_job_actions":{"HideDatasetActionactivesite":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"activesite"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rxdock_rbcavity/rxdock_rbcavity/2013.1.1_148c5bd1+galaxy0","tool_shed_repository":{"changeset_revision":"0848d3b1a46f","name":"rxdock_rbcavity","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"gridstep\\": \\"0.5\\", \\"ligand\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_volume\\": \\"100\\", \\"radius\\": \\"3.0\\", \\"receptor\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sphere\\": \\"1.0\\", \\"vol_incr\\": \\"0.0\\", \\"weight\\": \\"1.0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2013.1.1_148c5bd1+galaxy0","type":"tool","uuid":"3ea250c8-be7d-4a19-9d91-044cecaea535","workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","errors":null,"id":12,"input_connections":{"infile":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Compound conversion","outputs":[{"name":"outfile","type":"text"}],"position":{"bottom":1003.3000030517578,"height":174.8000030517578,"left":-134,"right":66,"top":828.5,"width":200,"x":-134,"y":828.5},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"e2c36f62e22f","name":"openbabel_compound_convert","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"appendtotitle\\": \\"\\", \\"dative_bonds\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"oformat\\": {\\"oformat_opts_selector\\": \\"sdf\\", \\"__current_case__\\": 58, \\"sdf_exp_h\\": \\"false\\", \\"sdf_no_prop\\": \\"false\\", \\"sdf_wedge_bonds\\": \\"false\\", \\"sdf_alias_out\\": \\"false\\", \\"gen2d\\": \\"false\\", \\"gen3d\\": \\"true\\"}, \\"ph\\": \\"7.4\\", \\"remove_h\\": \\"false\\", \\"split\\": \\"false\\", \\"unique\\": {\\"unique_opts_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"7f069b7b-a0c0-474a-9e33-052ab9318658","workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","errors":null,"id":13,"input_connections":{"split_parms|input":{"id":12,"output_name":"outfile"},"split_parms|select_mode|numnew":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Split file","outputs":[{"name":"list_output_sdf","type":"input"}],"position":{"bottom":961.5,"height":144,"left":144,"right":344,"top":817.5,"width":200,"x":144,"y":817.5},"post_job_actions":{"HideDatasetActionlist_output_sdf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_sdf"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","tool_shed_repository":{"changeset_revision":"6cbe2f30c2d7","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"sdf\\", \\"__current_case__\\": 4, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"numnew\\", \\"__current_case__\\": 1, \\"numnew\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.0","type":"tool","uuid":"428d69f6-67b5-4377-becc-c4d9cfa6319d","workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rxdock_rbdock/rxdock_rbdock/2013.1.1_148c5bd1+galaxy0","errors":null,"id":14,"input_connections":{"active_site":{"id":11,"output_name":"activesite"},"ligands":{"id":13,"output_name":"list_output_sdf"},"num":{"id":0,"output_name":"output"},"receptor":{"id":7,"output_name":"outfile"}},"inputs":[],"label":null,"name":"rDock docking","outputs":[{"name":"output","type":"sdf"}],"position":{"bottom":522.8999938964844,"height":184.39999389648438,"left":432,"right":632,"top":338.5,"width":200,"x":432,"y":338.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rxdock_rbdock/rxdock_rbdock/2013.1.1_148c5bd1+galaxy0","tool_shed_repository":{"changeset_revision":"b34d068c2782","name":"rxdock_rbdock","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"active_site\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"custom_rec\\": {\\"custom_rec_select\\": \\"false\\", \\"__current_case__\\": 1, \\"flex\\": \\"3.0\\"}, \\"filter\\": {\\"filter_select\\": \\"no_filter\\", \\"__current_case__\\": 1}, \\"ligands\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": \\"false\\", \\"num\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"receptor\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seed\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2013.1.1_148c5bd1+galaxy0","type":"tool","uuid":"4fd94b5e-d09f-4c1a-92ea-12788d9f525e","workflow_outputs":[{"label":"Docking poses","output_name":"output","uuid":"d346dbc4-932a-4c33-90f7-816faaa0db0f"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2","errors":null,"id":15,"input_connections":{"input_list":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"bottom":497.5,"height":134,"left":710,"right":910,"top":363.5,"width":200,"x":710,"y":363.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2","tool_shed_repository":{"changeset_revision":"830961c48e42","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"filename\\": {\\"add_name\\": \\"false\\", \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.2","type":"tool","uuid":"bd5a919a-4c2c-437a-bdd7-e023347ed0f3","workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/sucos_docking_scoring/sucos_docking_scoring/2020.03.4+galaxy1","errors":null,"id":16,"input_connections":{"input":{"id":15,"output_name":"output"},"refmol":{"id":5,"output_name":"output"}},"inputs":[],"label":null,"name":"Score docked poses using SuCOS","outputs":[{"name":"output","type":"sdf"}],"position":{"bottom":453.8999938964844,"height":164.39999389648438,"left":988,"right":1188,"top":289.5,"width":200,"x":988,"y":289.5},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/sucos_docking_scoring/sucos_docking_scoring/2020.03.4+galaxy1","tool_shed_repository":{"changeset_revision":"4f1896782f7c","name":"sucos_docking_scoring","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"refmol\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tanimoto\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2020.03.4+galaxy1","type":"tool","uuid":"75ba98b0-883f-4a22-972e-e951a9c3f3be","workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rdock_sort_filter/rdock_sort_filter/2013.1-0+galaxy0","errors":null,"id":17,"input_connections":{"filter":{"id":9,"output_name":"out1"},"input":{"id":16,"output_name":"output"}},"inputs":[],"label":null,"name":"rDock docking","outputs":[{"name":"output","type":"sdf"}],"position":{"bottom":286.5,"height":144,"left":1266,"right":1466,"top":142.5,"width":200,"x":1266,"y":142.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/rdock_sort_filter/rdock_sort_filter/2013.1-0+galaxy0","tool_shed_repository":{"changeset_revision":"d1ca4b45f615","name":"rdock_sort_filter","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"descending\\": \\"true\\", \\"filter\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"global_sort\\": \\"true\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name_field\\": \\"\\", \\"sort_field\\": \\"SuCOS_Score\\", \\"top\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2013.1.1_148c5bd1+galaxy0","type":"tool","uuid":"7c3dcdd3-023a-442c-8b88-3531a94e3ffc","workflow_outputs":[{"label":"Scored and filtered poses","output_name":"output","uuid":"93062a03-32e2-4852-84d1-5385fa9b908a"}]}},"tags":[],"uuid":"bb767af6-3641-4a5e-843d-78508fd7577a","version":1},"readme":"# Fragment-based virtual screening with docking and pose scoring\\n\\nDock a compound library against a target protein with rDock and validate the\\nposes generated against a reference fragment using SuCOS to compare the feature\\noverlap. Poses are filtered by a user-specified SuCOS threshold.\\n\\nA list of fragments should be specified which will be used to define the cavity\\nfor docking, using the \'Frankenstein ligand\' technique. For more details, please\\nsee https://www.informaticsmatters.com/blog/2018/11/23/cavities-and-frankenstein-molecules.html\\n\\nCompounds are split into collections and then recombined to allow the workflow\\nto be run in a highly parallelized fashion. To specify the level of\\nparallelization, use the \'Collection size\' parameter.\\n","changelog":"# Changelog\\n\\n## [0.1.5] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.4] - 2023-02-14\\n\\n### Fixed\\n- Fix the changeset revisions for 2 tool shed repositories\\n- Sync version of ``openbabel_compund_convert`` step to newest version\\n\\n## [0.1.3] - 2022-05-25\\n\\n### Changed\\n- Changed creator ORCID to absolute URI\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-12-06\\n\\n### Fixed\\n- `.workflowhub.yml`: use `repo-name/dockstore-workflow-name` scheme for workflows.\\n- Moved test data inside repo.\\n\\n## [0.1]\\n\\n- Initial version of fragment-based docking and scoring workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## Fragment-based virtual screening using rDock for docking and SuCOS for pose scoring\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Number of poses\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Receptor (PDB)\\"]@{ shape: doc }\\n2[\\"ℹ️ All fragments (SDF)\\"]@{ shape: doc }\\n3[\\"ℹ️ Collection size for docking\\"]@{ shape: lean-l }\\n4[\\"ℹ️ SuCOS threshold\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Fragment for SuCOS scoring (SDF/MOL)\\"]@{ shape: doc }\\n6[\\"ℹ️ Candidate compounds (SMILES)\\"]@{ shape: doc }\\n7[\\"Compound conversion\\"]@{ shape: process }\\n1 --> 7\\n8[\\"Create Frankenstein ligand\\"]@{ shape: process }\\n2 --> 8\\n9[\\"Compose text parameter value\\"]@{ shape: process }\\n4 --> 9\\n10[\\"Enumerate changes\\"]@{ shape: process }\\n6 --> 10\\n11[\\"rDock cavity definition\\"]@{ shape: process }\\n8 --> 11\\n7 --> 11\\n12[\\"Compound conversion\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Split file\\"]@{ shape: process }\\n12 --> 13\\n3 --> 13\\n14[\\"rDock docking\\"]@{ shape: process }\\n11 --> 14\\n13 --> 14\\n0 --> 14\\n7 --> 14\\n15[\\"Collapse Collection\\"]@{ shape: process }\\n14 --> 15\\n16[\\"Score docked poses using SuCOS\\"]@{ shape: process }\\n15 --> 16\\n5 --> 16\\n17[\\"rDock docking\\"]@{ shape: process }\\n9 --> 17\\n16 --> 17\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/fragment-based-docking-scoring/main","dockstore_id":17269,"categories":["Structural Biology"],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078507","tests":[{"doc":"Test outline for fragment-based-docking-scoring.ga","job":{"All fragments (SDF)":{"class":"File","path":"test-data/all-fragments.sdf","filetype":"sdf"},"Candidate compounds (SMILES)":{"class":"File","path":"test-data/candidates.smi","filetype":"smi"},"Collection size for docking":2,"Fragment for SuCOS scoring (SDF/MOL)":{"class":"File","path":"test-data/single-fragment.sdf","filetype":"sdf"},"Receptor (PDB)":{"class":"File","path":"test-data/receptor.pdb","filetype":"pdb"},"Number of poses":1,"SuCOS threshold":0},"outputs":{"Scored and filtered poses":{"asserts":{"has_text":{"text":"SuCOS_Score","n":4}}}}}]}],"path":"./workflows/computational-chemistry/fragment-based-docking-scoring"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/gromacs-mmgbsa.ga","testParameterFiles":["/gromacs-mmgbsa-tests.yml"],"authors":[{"name":"Simon Bray","orcid":"0000-0002-0621-6705"}],"definition":{"a_galaxy_workflow":"true","annotation":"MMGBSA simulation and calculation","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"}],"format-version":"0.1","license":"MIT","name":"MMGBSA calculations with GROMACS","release":"0.1.5","steps":{"0":{"annotation":"NaCl concentration","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"NaCl concentration","name":"Salt concentration"}],"label":"Salt concentration","name":"Input parameter","outputs":[],"position":{"left":0,"top":184},"tool_id":null,"tool_state":"{\\"default\\": 0.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"be1f6375-076e-4d61-a780-f867af394809","when":null,"workflow_outputs":[]},"1":{"annotation":"Size of the MMGBSA ensemble","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Size of the MMGBSA ensemble","name":"Number of simulations"}],"label":"Number of simulations","name":"Input parameter","outputs":[],"position":{"left":278,"top":7},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"0b26a4eb-6715-4e56-b8c4-1796486f91c5","when":null,"workflow_outputs":[]},"2":{"annotation":"PDB input","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"PDB input","name":"Apoprotein PDB"}],"label":"Apoprotein PDB","name":"Input dataset","outputs":[],"position":{"left":556,"top":302},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"808dce0b-2da4-4cad-865e-5f264164b9a5","when":null,"workflow_outputs":[]},"3":{"annotation":"Water model","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Water model","name":"Water model"}],"label":"Water model","name":"Input parameter","outputs":[],"position":{"left":556,"top":422},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"tip3p\\", \\"tip4p\\", \\"tips3p\\", \\"tip5p\\", \\"spc\\", \\"spce\\", \\"none\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"c556adbf-d074-4467-95dc-1c1966c5cff2","when":null,"workflow_outputs":[]},"4":{"annotation":"pH for protonation (-1.0 to skip)","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"pH for protonation (-1.0 to skip)","name":"pH"}],"label":"pH","name":"Input parameter","outputs":[],"position":{"left":556,"top":522},"tool_id":null,"tool_state":"{\\"default\\": -1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"a7af9e2f-f236-4ed4-b4e6-472d8542d8b5","when":null,"workflow_outputs":[]},"5":{"annotation":"Force field","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Force field","name":"Force field"}],"label":"Force field","name":"Input parameter","outputs":[],"position":{"left":556,"top":622},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"oplsaa\\", \\"gromos43a1\\", \\"amber96\\", \\"gromos53a6\\", \\"amber99sb\\", \\"gromos53a5\\", \\"gromos43a2\\", \\"amberGS\\", \\"charmm27\\", \\"amber03\\", \\"gromos54a7\\", \\"gromos45a3\\", \\"amber99\\", \\"amber94\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8c2f5860-bd30-4351-8bfb-1f9c42a92c4b","when":null,"workflow_outputs":[]},"6":{"annotation":"SDF input","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"SDF input","name":"Ligand SDF"}],"label":"Ligand SDF","name":"Input dataset","outputs":[],"position":{"left":556,"top":722},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"6c2e18cb-d616-43b3-b655-ff9d6dfc3ac9","when":null,"workflow_outputs":[]},"7":{"annotation":"Number of steps for NVT equilibration","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Number of steps for NVT equilibration","name":"NVT equilibration steps"}],"label":"NVT equilibration steps","name":"Input parameter","outputs":[],"position":{"left":1728,"top":570},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"d752a5f0-f16e-4b63-9d2b-139392376e77","when":null,"workflow_outputs":[]},"8":{"annotation":"Number of steps for NPT equilibration","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Number of steps for NPT equilibration","name":"NPT equilibration steps"}],"label":"NPT equilibration steps","name":"Input parameter","outputs":[],"position":{"left":2036,"top":712},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"466e4cd0-4bf7-4238-ac5d-dc0d803557a1","when":null,"workflow_outputs":[]},"9":{"annotation":"Number of steps for production simulation","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Number of steps for production simulation","name":"Production steps"}],"label":"Production steps","name":"Input parameter","outputs":[],"position":{"left":2344,"top":711},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3661db6a-9afa-42d7-8dcd-c43e9cf320ea","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":10,"input_connections":{"components_0|param_type|component_value":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":278,"top":148},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"5ad64e44-e40e-4848-a8b3-cddd18fb0fc1","when":null,"workflow_outputs":[]},"11":{"annotation":"","id":11,"input_connections":{"Apoprotein PDB":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"},"Force field":{"id":5,"input_subworkflow_step_id":4,"output_name":"output"},"Ligand SDF":{"id":6,"input_subworkflow_step_id":0,"output_name":"output"},"Water model":{"id":3,"input_subworkflow_step_id":3,"output_name":"output"},"pH":{"id":4,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Create GRO and TOP complex files","outputs":[],"position":{"left":874,"top":293},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"}],"format-version":"0.1","license":"MIT","name":"Create GRO and TOP complex files","steps":{"0":{"annotation":"SD-file for the input ligand","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"SD-file for the input ligand","name":"Ligand SDF"}],"label":"Ligand SDF","name":"Input dataset","outputs":[],"position":{"left":0,"top":461},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"sdf\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"277627e0-eba8-4de5-855b-63711c0db691","when":null,"workflow_outputs":[]},"1":{"annotation":"pH for protonating the ligand. Set to -1.0 to skip.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"pH for protonating the ligand. Set to -1.0 to skip.","name":"pH"}],"label":"pH","name":"Input parameter","outputs":[],"position":{"left":0,"top":718},"tool_id":null,"tool_state":"{\\"default\\": -1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6893b667-f375-481b-96eb-50d5aceb5030","when":null,"workflow_outputs":[]},"2":{"annotation":"PDB file for the protein (without ligand, cofactor, waters)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"PDB file for the protein (without ligand, cofactor, waters)","name":"Apoprotein PDB"}],"label":"Apoprotein PDB","name":"Input dataset","outputs":[],"position":{"left":1112,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"68f60b7d-13f3-4b21-aedf-986f0f7e48d7","when":null,"workflow_outputs":[]},"3":{"annotation":"Model for water molecules (relevant for subsequent solvation)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Model for water molecules (relevant for subsequent solvation)","name":"Water model"}],"label":"Water model","name":"Input parameter","outputs":[],"position":{"left":1112,"top":120},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"tip3p\\", \\"tip4p\\", \\"tips3p\\", \\"tip5p\\", \\"spc\\", \\"spce\\", \\"none\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8d29ae78-60de-45ba-b403-cdd2646af030","when":null,"workflow_outputs":[]},"4":{"annotation":"Force field for protein modelling. GAFF is used for the ligand.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Force field for protein modelling. GAFF is used for the ligand.","name":"Force field"}],"label":"Force field","name":"Input parameter","outputs":[],"position":{"left":1112,"top":220},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"oplsaa\\", \\"gromos43a1\\", \\"amber96\\", \\"gromos53a6\\", \\"amber99sb\\", \\"amber99sb\\", \\"gromos53a5\\", \\"gromos43a2\\", \\"amberGS\\", \\"charmm27\\", \\"amber03\\", \\"gromos54a7\\", \\"gromos45a3\\", \\"amber99\\", \\"amber94\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"16c399b1-e67c-47a9-acd3-61ddc39473d9","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Descriptors","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":278,"top":445},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","tool_shed_repository":{"changeset_revision":"0993ac4f4a23","name":"ctb_rdkit_descriptors","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_multiple\\": [\\"FormalCharge\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2020.03.4+galaxy1","type":"tool","uuid":"351397ff-a72d-4c7d-b39a-bc97b8f2c357","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":0,"output_name":"output"},"ph":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compound conversion","outputs":[{"name":"outfile","type":"text"}],"position":{"left":278,"top":595},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"e2c36f62e22f","name":"openbabel_compound_convert","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"appendtotitle\\": \\"\\", \\"dative_bonds\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"oformat\\": {\\"oformat_opts_selector\\": \\"pdb\\", \\"__current_case__\\": 51}, \\"ph\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"remove_h\\": false, \\"split\\": false, \\"unique\\": {\\"unique_opts_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"c1f8c291-3b11-4afa-b62d-ee2d2d22b11f","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0","errors":null,"id":7,"input_connections":{"ff":{"id":4,"output_name":"output"},"pdb_input":{"id":2,"output_name":"output"},"water":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"GROMACS initial setup","outputs":[{"name":"output1","type":"top"},{"name":"output2","type":"gro"},{"name":"output3","type":"itp"},{"name":"report","type":"txt"}],"position":{"left":1386,"top":44},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0","tool_shed_repository":{"changeset_revision":"070e3ecc3fda","name":"gmx_setup","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": true, \\"ff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_h\\": false, \\"pdb_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"water\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"b922c050-5df2-42b8-a819-1cbf892e06fa","when":null,"workflow_outputs":[{"label":"Position restraints file","output_name":"output3","uuid":"a8969ed8-83f9-4ff3-9634-56f85e7dd4dd"}]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":556,"top":445},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fd7483c3-52b0-489f-a422-2d399e2cd4e5","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","errors":null,"id":9,"input_connections":{"infile":{"id":6,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":556,"top":651},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"HETATM\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"be1e9539-c3ba-4c3c-a06f-a82c5b4d8437","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":834,"top":415},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"50d4ab40-a78d-4d97-8740-7e00edacae6b","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","errors":null,"id":11,"input_connections":{"allparams|nc":{"id":10,"output_name":"integer_param"},"input1":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"AnteChamber","outputs":[{"name":"output1","type":"mol2"}],"position":{"left":1112,"top":540},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"8a839e6a1e3e","name":"ambertools_antechamber","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"allparams\\": {\\"nc\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"m\\": \\"1\\", \\"resname\\": \\"UNL\\", \\"c\\": \\"bcc\\", \\"at\\": \\"gaff\\", \\"j\\": \\"4\\"}, \\"extraparams\\": {\\"pf\\": true, \\"usenc\\": true}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"selected_output_format\\": \\"mol2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"32a50a09-a9c7-4d08-9d7f-7d5c9efaa3bf","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","errors":null,"id":12,"input_connections":{"charge":{"id":10,"output_name":"integer_param"},"input1":{"id":11,"output_name":"output1"}},"inputs":[],"label":null,"name":"Generate MD topologies for small molecules","outputs":[{"name":"output","type":"itp"},{"name":"gro_output","type":"gro"}],"position":{"left":1390,"top":423},"post_job_actions":{"HideDatasetActiongro_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"gro_output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"095ad4d096d1","name":"ambertools_acpype","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"atomtype\\": \\"gaff\\", \\"charge\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"charge_method\\": \\"user\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"multiplicity\\": \\"1\\", \\"save_gro\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"3ee8d6ce-e7bc-45c5-ae34-ac0348ac959f","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","errors":null,"id":13,"input_connections":{"lig_gro":{"id":12,"output_name":"gro_output"},"lig_top":{"id":12,"output_name":"output"},"prot_gro":{"id":7,"output_name":"output2"},"prot_top":{"id":7,"output_name":"output1"}},"inputs":[],"label":null,"name":"Merge GROMACS topologies","outputs":[{"name":"complex_top","type":"top"},{"name":"complex_gro","type":"gro"}],"position":{"left":1678,"top":203},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","tool_shed_repository":{"changeset_revision":"bc31d02be922","name":"gmx_merge_topology_files","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"lig_gro\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lig_top\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prot_gro\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prot_top\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.3+galaxy0","type":"tool","uuid":"2a22a7fd-3e14-4f27-b1a8-a6d66cef292a","when":null,"workflow_outputs":[{"label":"Complex topology","output_name":"complex_top","uuid":"df11b684-24b0-481f-956d-c7cdc1e334ba"},{"label":"Complex GRO","output_name":"complex_gro","uuid":"193f8774-0334-49d3-b8ae-bdc3484cb1be"}]}},"tags":"","uuid":"1a4e83d6-e4b2-4f01-bcd1-de564a96bf7e"},"tool_id":null,"type":"subworkflow","uuid":"fe2677ce-d0d4-4937-9d34-f9a35db814d8","when":null,"workflow_outputs":[{"label":"Complex topology","output_name":"Complex topology","uuid":"e092ce5a-3fca-4183-b244-daf8a0582e72"},{"label":"Complex GRO","output_name":"Complex GRO","uuid":"ab63a333-8a78-4476-946a-f1bac8fde57c"},{"label":"GROMACS setup (ITP) on input dataset(s)","output_name":"Position restraints file","uuid":"8f1c5eee-7510-4430-b3e0-e2301926285d"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0","errors":null,"id":12,"input_connections":{"token_set_0|line":{"id":10,"output_name":"out1"},"token_set_0|repeat_select|times":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":556,"top":0},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"712bd247-7a81-4e77-ab1c-ba0e3f012d20","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_editconf/gmx_editconf/2022+galaxy0","errors":null,"id":13,"input_connections":{"input_file":{"id":11,"output_name":"Complex GRO"}},"inputs":[],"label":null,"name":"GROMACS structure configuration","outputs":[{"name":"output","type":"gro"}],"position":{"left":1152,"top":405},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_editconf/gmx_editconf/2022+galaxy0","tool_shed_repository":{"changeset_revision":"88076940fa94","name":"gmx_editconf","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"box\\": {\\"config\\": \\"true\\", \\"__current_case__\\": 0, \\"dim\\": \\"1.0\\", \\"type\\": \\"triclinic\\"}, \\"capture_log\\": false, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_format\\": \\"gro\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"08354f00-bea9-4f56-94cd-35b50dcf57c1","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1","errors":null,"id":14,"input_connections":{"split_parms|input":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":874,"top":36},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1","tool_shed_repository":{"changeset_revision":"baabc30154cd","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.1","type":"tool","uuid":"06f666f0-c2a8-4f18-9ad6-42b0e601c731","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":15,"input_connections":{"input1":{"id":14,"output_name":"list_output_txt"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":1152,"top":16},"post_job_actions":{"HideDatasetActionfloat_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"float_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"float\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"ad421d25-27f7-445c-aa4f-6b131bdfad6b","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0","errors":null,"id":16,"input_connections":{"conc":{"id":15,"output_name":"float_param"},"gro_input":{"id":13,"output_name":"output"},"top_input":{"id":11,"output_name":"Complex topology"}},"inputs":[],"label":null,"name":"GROMACS solvation and adding ions","outputs":[{"name":"output1","type":"gro"},{"name":"output2","type":"top"}],"position":{"left":1440,"top":16},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c4fbab8e03c5","name":"gmx_solvate","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": false, \\"conc\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mxw\\": \\"0\\", \\"neutralise\\": \\"-neutral\\", \\"seed\\": \\"1\\", \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"water_model\\": \\"spc216\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"565a9285-ba79-40c5-86ec-1dbdd9a9de17","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0","errors":null,"id":17,"input_connections":{"gro_input":{"id":16,"output_name":"output1"},"top_input":{"id":16,"output_name":"output2"}},"inputs":[],"label":null,"name":"GROMACS energy minimization","outputs":[{"name":"output1","type":"gro"},{"name":"output2","type":"edr"},{"name":"report","type":"txt"}],"position":{"left":1728,"top":153},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0","tool_shed_repository":{"changeset_revision":"b46d4b4d995c","name":"gmx_em","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"steep\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": \\"50000\\", \\"emtol\\": \\"1000.0\\", \\"emstep\\": \\"0.01\\", \\"seed\\": \\"1\\"}, \\"mxw\\": \\"0\\", \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"7c542061-2478-43a3-a3a3-795b20e96b31","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/parmconv/parmconv/21.10+galaxy0","errors":null,"id":18,"input_connections":{"param_inputs|str_in":{"id":16,"output_name":"output1"},"param_inputs|top_in":{"id":16,"output_name":"output2"}},"inputs":[],"label":null,"name":"Convert Parameters","outputs":[{"name":"prmtop_ligand","type":"txt"},{"name":"prmtop_receptor","type":"txt"},{"name":"prmtop_complex","type":"txt"},{"name":"prmtop_solvatedcomplex","type":"txt"}],"position":{"left":2920,"top":40},"post_job_actions":{"HideDatasetActionprmtop_complex":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"prmtop_complex"},"HideDatasetActionprmtop_ligand":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"prmtop_ligand"},"HideDatasetActionprmtop_receptor":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"prmtop_receptor"},"HideDatasetActionprmtop_solvatedcomplex":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"prmtop_solvatedcomplex"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/parmconv/parmconv/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"6d0677a148fe","name":"parmconv","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"param_inputs\\": {\\"fmt\\": \\"GROMACS\\", \\"__current_case__\\": 1, \\"top_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"str_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"modbehaviour\\": {\\"removedihe\\": true, \\"removebox\\": true, \\"radii\\": \\"mbondi\\"}}, \\"stripmask_complex\\": \\":NA,CL,SOL\\", \\"stripmask_ligand\\": \\"!:UNL\\", \\"stripmask_receptor\\": \\":NA,CL,SOL,UNL\\", \\"stripmask_solvatedcomplex\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"9266a7e8-9fad-4d2e-aaf9-1be038313349","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","errors":null,"id":19,"input_connections":{"gro_input":{"id":17,"output_name":"output1"},"inps|itp_in":{"id":11,"output_name":"Position restraints file"},"sets|mdp|md_steps":{"id":7,"output_name":"output"},"top_input":{"id":16,"output_name":"output2"}},"inputs":[{"description":"runtime parameter for tool GROMACS simulation","name":"inps"},{"description":"runtime parameter for tool GROMACS simulation","name":"inps"}],"label":null,"name":"GROMACS simulation","outputs":[{"name":"output1","type":"gro"},{"name":"output4","type":"xtc"},{"name":"output5","type":"cpt"},{"name":"report","type":"txt"}],"position":{"left":2036,"top":153},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput4":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output4"},"HideDatasetActionoutput5":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output5"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c0c9a5024177","name":"gmx_sim","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 1, \\"__job_resource__select\\": \\"yes\\", \\"gpu\\": \\"1\\"}, \\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inps\\": {\\"cpt_in\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"itp_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ndx_in\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"mxw\\": \\"0\\", \\"outps\\": {\\"traj\\": \\"xtc\\", \\"str\\": \\"gro\\", \\"cpt_out\\": \\"true\\", \\"edr_out\\": \\"false\\", \\"xvg_out\\": \\"false\\", \\"tpr_out\\": \\"false\\"}, \\"sets\\": {\\"ensembleselect\\": {\\"ensemble\\": \\"nvt\\", \\"__current_case__\\": 0, \\"startvel\\": \\"false\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"md\\", \\"constraints\\": \\"h-bonds\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"temperature\\": \\"300\\", \\"systemTcouple\\": \\"false\\", \\"step_length\\": \\"0.001\\", \\"write_freq\\": \\"1000\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"a926430a-b1e4-4e09-861f-ba7005f89e6c","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","errors":null,"id":20,"input_connections":{"gro_input":{"id":19,"output_name":"output1"},"inps|cpt_in":{"id":19,"output_name":"output5"},"inps|itp_in":{"id":11,"output_name":"Position restraints file"},"sets|mdp|md_steps":{"id":8,"output_name":"output"},"top_input":{"id":16,"output_name":"output2"}},"inputs":[{"description":"runtime parameter for tool GROMACS simulation","name":"inps"}],"label":null,"name":"GROMACS simulation","outputs":[{"name":"output1","type":"gro"},{"name":"output4","type":"xtc"},{"name":"output5","type":"cpt"},{"name":"report","type":"txt"}],"position":{"left":2344,"top":153},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput4":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output4"},"HideDatasetActionoutput5":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output5"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c0c9a5024177","name":"gmx_sim","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 1, \\"__job_resource__select\\": \\"yes\\", \\"gpu\\": \\"1\\"}, \\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inps\\": {\\"cpt_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"itp_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ndx_in\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"mxw\\": \\"0\\", \\"outps\\": {\\"traj\\": \\"xtc\\", \\"str\\": \\"gro\\", \\"cpt_out\\": \\"true\\", \\"edr_out\\": \\"false\\", \\"xvg_out\\": \\"false\\", \\"tpr_out\\": \\"false\\"}, \\"sets\\": {\\"ensembleselect\\": {\\"ensemble\\": \\"nvt\\", \\"__current_case__\\": 0, \\"startvel\\": \\"false\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"md\\", \\"constraints\\": \\"h-bonds\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"temperature\\": \\"300\\", \\"systemTcouple\\": \\"false\\", \\"step_length\\": \\"0.001\\", \\"write_freq\\": \\"1000\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"fe91fab0-9a55-4480-9618-43a3e0195500","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","errors":null,"id":21,"input_connections":{"gro_input":{"id":20,"output_name":"output1"},"inps|cpt_in":{"id":20,"output_name":"output5"},"sets|mdp|md_steps":{"id":9,"output_name":"output"},"top_input":{"id":16,"output_name":"output2"}},"inputs":[{"description":"runtime parameter for tool GROMACS simulation","name":"inps"}],"label":null,"name":"GROMACS simulation","outputs":[{"name":"output1","type":"gro"},{"name":"output4","type":"xtc"},{"name":"output8","type":"tpr"},{"name":"report","type":"txt"}],"position":{"left":2642,"top":202},"post_job_actions":{"HideDatasetActionoutput8":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output8"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c0c9a5024177","name":"gmx_sim","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 1, \\"__job_resource__select\\": \\"yes\\", \\"gpu\\": \\"1\\"}, \\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inps\\": {\\"cpt_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"itp_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ndx_in\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"mxw\\": \\"0\\", \\"outps\\": {\\"traj\\": \\"xtc\\", \\"str\\": \\"gro\\", \\"cpt_out\\": \\"false\\", \\"edr_out\\": \\"false\\", \\"xvg_out\\": \\"false\\", \\"tpr_out\\": \\"true\\"}, \\"sets\\": {\\"ensembleselect\\": {\\"ensemble\\": \\"nvt\\", \\"__current_case__\\": 0, \\"startvel\\": \\"false\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"md\\", \\"constraints\\": \\"none\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"temperature\\": \\"300\\", \\"systemTcouple\\": \\"false\\", \\"step_length\\": \\"0.001\\", \\"write_freq\\": \\"1000\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"7750144f-b8a9-4543-a13c-68bca8772d8f","when":null,"workflow_outputs":[{"label":"GRO files","output_name":"output1","uuid":"9f4cd61a-8639-4d0e-a848-407cbdaff68c"},{"label":"XTC files","output_name":"output4","uuid":"ae282dc2-9136-4c2a-a812-08bc204ae5a6"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/md_converter/md_converter/1.9.6+galaxy0","errors":null,"id":22,"input_connections":{"input_file":{"id":21,"output_name":"output4"}},"inputs":[],"label":null,"name":"MDTraj file converter","outputs":[{"name":"output","type":"data"}],"position":{"left":2920,"top":334},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/md_converter/md_converter/1.9.6+galaxy0","tool_shed_repository":{"changeset_revision":"ba83f0923369","name":"md_converter","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_format\\": \\"netcdf\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.9.6+galaxy0","type":"tool","uuid":"d98fcd50-1530-4899-820e-ff08b8dde0ae","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/mmpbsa_mmgbsa/mmpbsa_mmgbsa/21.10+galaxy0","errors":null,"id":23,"input_connections":{"input|simulation|complex":{"id":18,"output_name":"prmtop_complex"},"input|simulation|ligand":{"id":18,"output_name":"prmtop_ligand"},"input|simulation|receptor":{"id":18,"output_name":"prmtop_receptor"},"input|simulation|solvatedcomplex":{"id":18,"output_name":"prmtop_solvatedcomplex"},"input|simulation|trajcomplex":{"id":22,"output_name":"output"}},"inputs":[],"label":null,"name":"MMPBSA/MMGBSA","outputs":[{"name":"tempfiles","type":"input"},{"name":"resultoutfile","type":"txt"},{"name":"parameteroutfile","type":"txt"}],"position":{"left":3198,"top":36},"post_job_actions":{"HideDatasetActionparameteroutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"parameteroutfile"},"HideDatasetActiontempfiles":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"tempfiles"},"RenameDatasetActionresultoutfile":{"action_arguments":{"newname":"MMGBSA statistics"},"action_type":"RenameDatasetAction","output_name":"resultoutfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/mmpbsa_mmgbsa/mmpbsa_mmgbsa/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"608098b3668d","name":"mmpbsa_mmgbsa","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"allparams\\": {\\"startframe\\": \\"1\\", \\"endframe\\": \\"9999999\\", \\"interval\\": \\"10\\", \\"entropy\\": false, \\"use_sander\\": false, \\"keep_files\\": true, \\"strip_mask\\": \\":NA,CL,SOL\\"}, \\"calcdetails\\": {\\"gb_pb_calc\\": {\\"calctype\\": \\"gb\\", \\"__current_case__\\": 0, \\"igb\\": \\"5\\", \\"saltcon\\": \\"0.1\\", \\"surfoff\\": \\"0.0\\", \\"molsurf\\": false, \\"probe\\": \\"1.4\\", \\"msoffset\\": \\"0.0\\"}, \\"decomposition\\": {\\"decomposition\\": \\"no\\", \\"__current_case__\\": 1}}, \\"input\\": {\\"simulation\\": {\\"simtype\\": \\"single\\", \\"__current_case__\\": 0, \\"ligand\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"receptor\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"complex\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"solvatedcomplex\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trajcomplex\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"0351786b-5169-450f-8d5f-194dce5e4b77","when":null,"workflow_outputs":[{"label":"MMGBSA statistics","output_name":"resultoutfile","uuid":"cace1632-56e0-4222-9682-a3e079fead2d"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","errors":null,"id":24,"input_connections":{"infile":{"id":23,"output_name":"resultoutfile"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":3476,"top":199},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"DELTA TOTAL\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"3aba0217-8162-43c3-8d19-1b04dcb3984d","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":25,"input_connections":{"input_list":{"id":24,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":3754,"top":189},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"dbf60354-8c38-4735-9dd0-44d776906e02","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"Cut1","errors":null,"id":26,"input_connections":{"input":{"id":25,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4032,"top":209},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c3\\", \\"delimiter\\": \\"Sp\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"073ece68-67cb-497e-bf81-df302004f043","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"Summary_Statistics1","errors":null,"id":27,"input_connections":{"input":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Summary Statistics","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":4310,"top":214},"post_job_actions":{},"tool_id":"Summary_Statistics1","tool_state":"{\\"cond\\": \\"c1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.2","type":"tool","uuid":"f5812d99-5d61-40c0-af3a-5f1eb307c755","when":null,"workflow_outputs":[{"label":"MMGBSA free energy","output_name":"out_file1","uuid":"5b387d49-544a-4462-912a-4d64d775c62a"}]}},"tags":[],"uuid":"16455667-cdef-4979-9603-f338e09aa2a1","version":1},"readme":"# GROMACS MMGBSA free energy calculation\\n\\nPerform an ensemble of MD simulations of a user-specified size using GROMACS,\\nand calculate MMGBSA free energies using AmberTools. An ensemble average is\\ncalculated and returned to the user as the final input.\\n\\nThe input protein (PDB) and ligand (SDF) files provided are parameterized by\\nthe \'Protein-ligand complex parameterization\' subworkflow.\\n","changelog":"# Changelog\\n\\n## [0.1.5] - 2023-11-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2021.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_editconf/gmx_editconf/2021.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_editconf/gmx_editconf/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.1`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2021.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2021.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2021.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/md_converter/md_converter/1.9.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/md_converter/md_converter/1.9.7+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2` was updated to `toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0`\\n\\n## [0.1.4] - 2023-11-20\\n\\n- Fix author in dockstore\\n- Fix changeset_revision of gmx_solvate\\n\\n## [0.1.3] - 2022-05-25\\n\\n### Changed\\n- Changed creator ORCID to absolute URI\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-12-06\\n\\n### Fixed\\n- `.workflowhub.yml`: use `repo-name/dockstore-workflow-name` scheme for workflows.\\n- Moved test data inside repo.\\n\\n## [0.1]\\n\\n- Initial version of GROMACS MMGBSA workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## MMGBSA calculations with GROMACS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Salt concentration\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Number of simulations\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Apoprotein PDB\\"]@{ shape: doc }\\n3[\\"ℹ️ Water model\\"]@{ shape: lean-l }\\n4[\\"ℹ️ pH\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Force field\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Ligand SDF\\"]@{ shape: doc }\\n7[\\"ℹ️ NVT equilibration steps\\"]@{ shape: lean-l }\\n8[\\"ℹ️ NPT equilibration steps\\"]@{ shape: lean-l }\\n9[\\"ℹ️ Production steps\\"]@{ shape: lean-l }\\n10[\\"Compose text parameter value\\"]@{ shape: process }\\n0 --> 10\\n11[\\"🛠️ Create GRO and TOP complex files\\"]@{ shape: subprocess }\\n2 --> 11\\n5 --> 11\\n6 --> 11\\n3 --> 11\\n4 --> 11\\n12[\\"Create text file\\"]@{ shape: process }\\n10 --> 12\\n1 --> 12\\n13[\\"GROMACS structure configuration\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Split file\\"]@{ shape: process }\\n12 --> 14\\n15[\\"Parse parameter value\\"]@{ shape: process }\\n14 --> 15\\n16[\\"GROMACS solvation and adding ions\\"]@{ shape: process }\\n15 --> 16\\n13 --> 16\\n11 --> 16\\n17[\\"GROMACS energy minimization\\"]@{ shape: process }\\n16 --> 17\\n16 --> 17\\n18[\\"Convert Parameters\\"]@{ shape: process }\\n16 --> 18\\n16 --> 18\\n19[\\"GROMACS simulation\\"]@{ shape: process }\\n17 --> 19\\n11 --> 19\\n7 --> 19\\n16 --> 19\\n20[\\"GROMACS simulation\\"]@{ shape: process }\\n19 --> 20\\n19 --> 20\\n11 --> 20\\n8 --> 20\\n16 --> 20\\n21[\\"GROMACS simulation\\"]@{ shape: process }\\n20 --> 21\\n20 --> 21\\n9 --> 21\\n16 --> 21\\n22[\\"MDTraj file converter\\"]@{ shape: process }\\n21 --> 22\\n23[\\"MMPBSA/MMGBSA\\"]@{ shape: process }\\n18 --> 23\\n18 --> 23\\n18 --> 23\\n18 --> 23\\n22 --> 23\\n24[\\"Search in textfiles\\"]@{ shape: process }\\n23 --> 24\\n25[\\"Collapse Collection\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Cut\\"]@{ shape: process }\\n25 --> 26\\n27[\\"Summary Statistics\\"]@{ shape: process }\\n26 --> 27\\n```\\n\\n## Create GRO and TOP complex files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Ligand SDF\\"]@{ shape: doc }\\n1[\\"ℹ️ pH\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Apoprotein PDB\\"]@{ shape: doc }\\n3[\\"ℹ️ Water model\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Force field\\"]@{ shape: lean-l }\\n5[\\"Descriptors\\"]@{ shape: process }\\n0 --> 5\\n6[\\"Compound conversion\\"]@{ shape: process }\\n0 --> 6\\n1 --> 6\\n7[\\"GROMACS initial setup\\"]@{ shape: process }\\n4 --> 7\\n2 --> 7\\n3 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Search in textfiles\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Parse parameter value\\"]@{ shape: process }\\n8 --> 10\\n11[\\"AnteChamber\\"]@{ shape: process }\\n10 --> 11\\n9 --> 11\\n12[\\"Generate MD topologies for small molecules\\"]@{ shape: process }\\n10 --> 12\\n11 --> 12\\n13[\\"Merge GROMACS topologies\\"]@{ shape: process }\\n12 --> 13\\n12 --> 13\\n7 --> 13\\n7 --> 13\\n```\\n","updated":"2023-11-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/gromacs-mmgbsa/main","dockstore_id":17278,"categories":["Structural Biology"],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078816","tests":[{"doc":"Test outline for gromacs-mmgbsa.ga","job":{"Apoprotein PDB":{"class":"File","path":"test-data/mpro.pdb"},"Force field":"amber99sb","Ligand SDF":{"class":"File","path":"test-data/lig.sdf"},"NPT equilibration steps":50,"NVT equilibration steps":50,"Number of simulations":2,"Production steps":100,"Water model":"tip3p","Salt concentration":0.1},"outputs":{"MMGBSA free energy":{"class":"File","assert":{"has_n_lines":2}},"MMGBSA statistics":{"element_tests":{"split_file_000000.txt":{"assert":{"has_text":"Generalized Born ESURF calculated using LCPO"}}}},"XTC files":{"element_tests":{"split_file_000000.txt":{"assert":{"has_size":138900,"delta":500}}}}}}]}],"path":"./workflows/computational-chemistry/gromacs-mmgbsa"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/protein-ligand-complex-parameterization.ga","testParameterFiles":["/protein-ligand-complex-parameterization-tests.yml"],"authors":[{"name":"Simon Bray","orcid":"0000-0002-0621-6705"}],"definition":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"}],"format-version":"0.1","license":"MIT","name":"Create GRO and TOP complex files","release":"0.1.4","steps":{"0":{"annotation":"pH for protonating the ligand. Set to -1.0 to skip.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"pH for protonating the ligand. Set to -1.0 to skip.","name":"pH"}],"label":"pH","name":"Input parameter","outputs":[],"position":{"bottom":384.8000030517578,"height":61.80000305175781,"left":-238,"right":-38,"top":323,"width":200,"x":-238,"y":323},"tool_id":null,"tool_state":"{\\"default\\": -1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6893b667-f375-481b-96eb-50d5aceb5030","workflow_outputs":[]},"1":{"annotation":"SD-file for the input ligand","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"SD-file for the input ligand","name":"Ligand SDF"}],"label":"Ligand SDF","name":"Input dataset","outputs":[],"position":{"bottom":623.8000030517578,"height":61.80000305175781,"left":-238,"right":-38,"top":562,"width":200,"x":-238,"y":562},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"sdf\\"]}","tool_version":null,"type":"data_input","uuid":"277627e0-eba8-4de5-855b-63711c0db691","workflow_outputs":[]},"2":{"annotation":"PDB file for the protein (without ligand, cofactor, waters)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"PDB file for the protein (without ligand, cofactor, waters)","name":"Apoprotein PDB"}],"label":"Apoprotein PDB","name":"Input dataset","outputs":[],"position":{"bottom":105.80000305175781,"height":61.80000305175781,"left":874,"right":1074,"top":44,"width":200,"x":874,"y":44},"tool_id":null,"tool_state":"{\\"optional\\": false}","tool_version":null,"type":"data_input","uuid":"68f60b7d-13f3-4b21-aedf-986f0f7e48d7","workflow_outputs":[]},"3":{"annotation":"Model for water molecules (relevant for subsequent solvation)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Model for water molecules (relevant for subsequent solvation)","name":"Water model"}],"label":"Water model","name":"Input parameter","outputs":[],"position":{"bottom":205.8000030517578,"height":61.80000305175781,"left":874,"right":1074,"top":144,"width":200,"x":874,"y":144},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"tip3p\\", \\"tip4p\\", \\"tips3p\\", \\"tip5p\\", \\"spc\\", \\"spce\\", \\"none\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8d29ae78-60de-45ba-b403-cdd2646af030","workflow_outputs":[]},"4":{"annotation":"Force field for protein modelling. GAFF is used for the ligand.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Force field for protein modelling. GAFF is used for the ligand.","name":"Force field"}],"label":"Force field","name":"Input parameter","outputs":[],"position":{"bottom":305.8000030517578,"height":61.80000305175781,"left":874,"right":1074,"top":244,"width":200,"x":874,"y":244},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"oplsaa\\", \\"gromos43a1\\", \\"amber96\\", \\"gromos53a6\\", \\"amber99sb\\", \\"amber99sb\\", \\"gromos53a5\\", \\"gromos43a2\\", \\"amberGS\\", \\"charmm27\\", \\"amber03\\", \\"gromos54a7\\", \\"gromos45a3\\", \\"amber99\\", \\"amber94\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"16c399b1-e67c-47a9-acd3-61ddc39473d9","workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","errors":null,"id":5,"input_connections":{"infile":{"id":1,"output_name":"output"},"ph":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Compound conversion","outputs":[{"name":"outfile","type":"text"}],"position":{"bottom":507.6000061035156,"height":225.60000610351562,"left":40,"right":240,"top":282,"width":200,"x":40,"y":282},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"e2c36f62e22f","name":"openbabel_compound_convert","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"appendtotitle\\": \\"\\", \\"dative_bonds\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"oformat\\": {\\"oformat_opts_selector\\": \\"pdb\\", \\"__current_case__\\": 51}, \\"ph\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"remove_h\\": \\"false\\", \\"split\\": \\"false\\", \\"unique\\": {\\"unique_opts_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"c1f8c291-3b11-4afa-b62d-ee2d2d22b11f","workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","errors":null,"id":6,"input_connections":{"infile":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Descriptors","outputs":[{"name":"outfile","type":"tabular"}],"position":{"bottom":639.1999969482422,"height":93.19999694824219,"left":40,"right":240,"top":546,"width":200,"x":40,"y":546},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","tool_shed_repository":{"changeset_revision":"a1c53f0533b0","name":"ctb_rdkit_descriptors","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_multiple\\": [\\"FormalCharge\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2020.03.4+galaxy1","type":"tool","uuid":"351397ff-a72d-4c7d-b39a-bc97b8f2c357","workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2021.3+galaxy0","errors":null,"id":7,"input_connections":{"ff":{"id":4,"output_name":"output"},"pdb_input":{"id":2,"output_name":"output"},"water":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"GROMACS initial setup","outputs":[{"name":"output1","type":"top"},{"name":"output2","type":"gro"},{"name":"output3","type":"itp"},{"name":"report","type":"txt"}],"position":{"bottom":409.20001220703125,"height":347.20001220703125,"left":1152,"right":1352,"top":62,"width":200,"x":1152,"y":62},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2021.3+galaxy0","tool_shed_repository":{"changeset_revision":"1d4dd4f908d4","name":"gmx_setup","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": \\"true\\", \\"ff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_h\\": \\"false\\", \\"pdb_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"water\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2021.3+galaxy0","type":"tool","uuid":"b922c050-5df2-42b8-a819-1cbf892e06fa","workflow_outputs":[{"label":"Position restraints file","output_name":"output3","uuid":"b23aa4b2-8ded-4ebe-9499-eb5e8081de22"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","errors":null,"id":8,"input_connections":{"infile":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"bottom":451.6000061035156,"height":113.60000610351562,"left":596,"right":796,"top":338,"width":200,"x":596,"y":338},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","tool_shed_repository":{"changeset_revision":"ddf54b12c295","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"HETATM\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"be1e9539-c3ba-4c3c-a06f-a82c5b4d8437","workflow_outputs":[]},"9":{"annotation":"","content_id":"Cut1","errors":null,"id":9,"input_connections":{"input":{"id":6,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"bottom":639.1999969482422,"height":93.19999694824219,"left":318,"right":518,"top":546,"width":200,"x":318,"y":546},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fd7483c3-52b0-489f-a422-2d399e2cd4e5","workflow_outputs":[]},"10":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":10,"input_connections":{"input1":{"id":9,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"bottom":670.3999938964844,"height":154.39999389648438,"left":596,"right":796,"top":516,"width":200,"x":596,"y":516},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"50d4ab40-a78d-4d97-8740-7e00edacae6b","workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","errors":null,"id":11,"input_connections":{"allparams|nc":{"id":10,"output_name":"integer_param"},"input1":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"AnteChamber","outputs":[{"name":"output1","type":"mol2"}],"position":{"bottom":574.3999938964844,"height":164.39999389648438,"left":874,"right":1074,"top":410,"width":200,"x":874,"y":410},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"4fff93efc0f9","name":"ambertools_antechamber","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"allparams\\": {\\"nc\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"m\\": \\"1\\", \\"resname\\": \\"UNL\\", \\"c\\": \\"bcc\\", \\"at\\": \\"gaff\\", \\"j\\": \\"4\\"}, \\"extraparams\\": {\\"pf\\": \\"true\\", \\"usenc\\": \\"true\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"selected_output_format\\": \\"mol2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"32a50a09-a9c7-4d08-9d7f-7d5c9efaa3bf","workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","errors":null,"id":12,"input_connections":{"charge":{"id":10,"output_name":"integer_param"},"input1":{"id":11,"output_name":"output1"}},"inputs":[],"label":null,"name":"Generate MD topologies for small molecules","outputs":[{"name":"output","type":"itp"},{"name":"gro_output","type":"gro"}],"position":{"bottom":662.1999969482422,"height":215.1999969482422,"left":1152,"right":1352,"top":447,"width":200,"x":1152,"y":447},"post_job_actions":{"HideDatasetActiongro_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"gro_output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"a0c154146234","name":"ambertools_acpype","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"atomtype\\": \\"gaff\\", \\"charge\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"charge_method\\": \\"user\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"multiplicity\\": \\"1\\", \\"save_gro\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"3ee8d6ce-e7bc-45c5-ae34-ac0348ac959f","workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","errors":null,"id":13,"input_connections":{"lig_gro":{"id":12,"output_name":"gro_output"},"lig_top":{"id":12,"output_name":"output"},"prot_gro":{"id":7,"output_name":"output2"},"prot_top":{"id":7,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool Merge GROMACS topologies","name":"lig_gro"},{"description":"runtime parameter for tool Merge GROMACS topologies","name":"lig_top"},{"description":"runtime parameter for tool Merge GROMACS topologies","name":"prot_gro"},{"description":"runtime parameter for tool Merge GROMACS topologies","name":"prot_top"}],"label":null,"name":"Merge GROMACS topologies","outputs":[{"name":"complex_top","type":"top"},{"name":"complex_gro","type":"gro"}],"position":{"bottom":482.6000061035156,"height":255.60000610351562,"left":1440,"right":1640,"top":227,"width":200,"x":1440,"y":227},"post_job_actions":{"RenameDatasetActioncomplex_gro":{"action_arguments":{"newname":"Complex GRO"},"action_type":"RenameDatasetAction","output_name":"complex_gro"},"RenameDatasetActioncomplex_top":{"action_arguments":{"newname":"Complex topology"},"action_type":"RenameDatasetAction","output_name":"complex_top"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","tool_shed_repository":{"changeset_revision":"9389cd867cf2","name":"gmx_merge_topology_files","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"lig_gro\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"lig_top\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"prot_gro\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"prot_top\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.3+galaxy0","type":"tool","uuid":"2a22a7fd-3e14-4f27-b1a8-a6d66cef292a","workflow_outputs":[{"label":"Complex topology","output_name":"complex_top","uuid":"039fdc15-5b7f-4f41-9dcf-545673cbd406"},{"label":"Complex GRO","output_name":"complex_gro","uuid":"1c6090a6-5a1f-44b3-bc95-ae554a9bf67d"}]}},"tags":[],"uuid":"f1aea424-b536-496f-a0e6-fe46df8e6e18","version":1},"readme":"# Protein-ligand complex parameterization\\n\\nParameterizes an input protein (PDB) and ligand (SDF) file prior to molecular\\ndynamics simulation with GROMACS.\\n\\nThis is a simple workflow intended for use as a subworkflow in more complex\\nMD workflows. It is used as a subworkflow by the GROMACS MMGBSA and dcTMD\\nworkflows. \\n","changelog":"# Changelog\\n\\n## [0.1.4] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.3] - 2022-05-25\\n\\n### Changed\\n- Changed creator ORCID to absolute URI\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-12-06\\n\\n### Fixed\\n- `.workflowhub.yml`: use `repo-name/dockstore-workflow-name` scheme for workflows.\\n- Moved test data inside repo.\\n\\n## [0.1]\\n\\n- Initial version of protein-ligand parameterization workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## Create GRO and TOP complex files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ pH\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Ligand SDF\\"]@{ shape: doc }\\n2[\\"ℹ️ Apoprotein PDB\\"]@{ shape: doc }\\n3[\\"ℹ️ Water model\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Force field\\"]@{ shape: lean-l }\\n5[\\"Compound conversion\\"]@{ shape: process }\\n1 --> 5\\n0 --> 5\\n6[\\"Descriptors\\"]@{ shape: process }\\n1 --> 6\\n7[\\"GROMACS initial setup\\"]@{ shape: process }\\n4 --> 7\\n2 --> 7\\n3 --> 7\\n8[\\"Search in textfiles\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Cut\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Parse parameter value\\"]@{ shape: process }\\n9 --> 10\\n11[\\"AnteChamber\\"]@{ shape: process }\\n10 --> 11\\n8 --> 11\\n12[\\"Generate MD topologies for small molecules\\"]@{ shape: process }\\n10 --> 12\\n11 --> 12\\n13[\\"Merge GROMACS topologies\\"]@{ shape: process }\\n12 --> 13\\n12 --> 13\\n7 --> 13\\n7 --> 13\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/protein-ligand-complex-parameterization/main","dockstore_id":17279,"categories":["Structural Biology"],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078454","tests":[{"doc":"Test outline for protein-ligand-complex-parameterization.ga","job":{"Apoprotein PDB":{"class":"File","path":"test-data/mpro.pdb"},"Force field":"amber99sb","Ligand SDF":{"class":"File","path":"test-data/lig.sdf"},"Water model":"tip3p"},"outputs":{"Complex GRO":{"path":"test-data/complex.gro"},"Complex topology":{"path":"test-data/complex.top","lines_diff":16}}}]}],"path":"./workflows/computational-chemistry/protein-ligand-complex-parameterization"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/gromacs-dctmd.ga","testParameterFiles":["/gromacs-dctmd-tests.yml"],"authors":[{"name":"Simon Bray","orcid":"0000-0002-0621-6705"}],"definition":{"a_galaxy_workflow":"true","annotation":"Perform dcTMD free energy simulations and calculations","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"}],"format-version":"0.1","license":"MIT","name":"dcTMD calculations with GROMACS","release":"0.1.5","steps":{"0":{"annotation":"Ligand SDF","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Ligand SDF","name":"Ligand SDF"}],"label":"Ligand SDF","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"59ec9b66-5db1-413a-89f9-dea12549e958","when":null,"workflow_outputs":[]},"1":{"annotation":"Set to -1.0 to skip protonation","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Set to -1.0 to skip protonation","name":"pH to protonate ligand"}],"label":"pH to protonate ligand","name":"Input parameter","outputs":[],"position":{"left":0,"top":100},"tool_id":null,"tool_state":"{\\"default\\": -1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"84fdad78-cd39-4c19-b228-7244c6fe8676","when":null,"workflow_outputs":[]},"2":{"annotation":"Protein PDB","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Protein PDB","name":"Protein PDB"}],"label":"Protein PDB","name":"Input dataset","outputs":[],"position":{"left":0,"top":241},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"16aacb27-77b0-43bd-b5b1-a94d41fa4294","when":null,"workflow_outputs":[]},"3":{"annotation":"NaCl concentration (mol/dm^3)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"NaCl concentration (mol/dm^3)","name":"Salt concentration"}],"label":"Salt concentration","name":"Input parameter","outputs":[],"position":{"left":288,"top":11},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"66a35907-fba1-4e82-9ed8-b46b73ed6b91","when":null,"workflow_outputs":[]},"4":{"annotation":"Water model","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Water model","name":"Water model"}],"label":"Water model","name":"Input parameter","outputs":[],"position":{"left":0,"top":341},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"tip3p\\", \\"tip4p\\", \\"tips3p\\", \\"tip5p\\", \\"spc\\", \\"spce\\", \\"none\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8400f998-7ca3-43ee-ae56-7332b8021bda","when":null,"workflow_outputs":[]},"5":{"annotation":"Force field","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Force field","name":"Force field"}],"label":"Force field","name":"Input parameter","outputs":[],"position":{"left":0,"top":441},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"oplsaa\\", \\"gromos43a1\\", \\"amber96\\", \\"gromos53a6\\", \\"amber99sb\\", \\"gromos53a5\\", \\"gromos43a2\\", \\"amberGS\\", \\"charmm27\\", \\"amber03\\", \\"gromos54a7\\", \\"gromos45a3\\", \\"amber99\\", \\"amber94\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3d20c7aa-e3ca-42d3-9aa8-702abf14d35a","when":null,"workflow_outputs":[]},"6":{"annotation":"Number of simulations in the TMD ensemble","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Number of simulations in the TMD ensemble","name":"Number of simulations"}],"label":"Number of simulations","name":"Input parameter","outputs":[],"position":{"left":0,"top":842},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ca9f5490-71d5-4691-882e-bdb5fba4835e","when":null,"workflow_outputs":[]},"7":{"annotation":"Temperature","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Temperature","name":"Temperature"}],"label":"Temperature","name":"Input parameter","outputs":[],"position":{"left":0,"top":983},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"9a848a36-8add-434a-b179-fe801ccf7022","when":null,"workflow_outputs":[]},"8":{"annotation":"Number of equilibration steps","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Number of equilibration steps","name":"Number of equilibration steps"}],"label":"Number of equilibration steps","name":"Input parameter","outputs":[],"position":{"left":864,"top":897},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a9d402ec-eafb-4fab-9e5a-192765011bba","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/get_online_data/ctb_online_data_fetch/0.4","errors":null,"id":9,"input_connections":{},"inputs":[],"label":null,"name":"Online data","outputs":[{"name":"output","type":"txt"}],"position":{"left":586,"top":1169},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/get_online_data/ctb_online_data_fetch/0.4","tool_shed_repository":{"changeset_revision":"2538366eb8fb","name":"get_online_data","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"url_paste\\": \\"https://raw.githubusercontent.com/galaxyproject/iwc/main/workflows/computational-chemistry/gromacs-dctmd/tmd.mdp\\", \\"whitelist\\": \\"mdp\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.4","type":"tool","uuid":"a83e0693-3471-4736-a01e-188df629e8bb","when":null,"workflow_outputs":[]},"10":{"annotation":"Rate to pull groups apart (nm/ps)","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Rate to pull groups apart (nm/ps)","name":"Pulling rate"}],"label":"Pulling rate","name":"Input parameter","outputs":[],"position":{"left":288,"top":1315},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e371d527-a200-4d15-80ea-f297863090a6","when":null,"workflow_outputs":[]},"11":{"annotation":"MD step length","content_id":null,"errors":null,"id":11,"input_connections":{},"inputs":[{"description":"MD step length","name":"Step length (ps)"}],"label":"Step length (ps)","name":"Input parameter","outputs":[],"position":{"left":864,"top":1160},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"bbc28f7f-ac02-42af-ad8b-e5d4137d348b","when":null,"workflow_outputs":[]},"12":{"annotation":"List of atom indices (separated with spaces) which make up the protein pull group","content_id":null,"errors":null,"id":12,"input_connections":{},"inputs":[{"description":"List of atom indices (separated with spaces) which make up the protein pull group","name":"Protein pull group"}],"label":"Protein pull group","name":"Input parameter","outputs":[],"position":{"left":1440,"top":550},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"5df8964e-e7ad-4398-b073-482069ec918f","when":null,"workflow_outputs":[]},"13":{"annotation":"Number of steps for the production simulation","content_id":null,"errors":null,"id":13,"input_connections":{},"inputs":[{"description":"Number of steps for the production simulation","name":"Number of steps"}],"label":"Number of steps","name":"Input parameter","outputs":[],"position":{"left":586,"top":1468},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"4ec27bd6-fc77-484e-87a6-78f6631f06e0","when":null,"workflow_outputs":[]},"14":{"annotation":"pull-group1-pbcatom - the reference atom index for the treatment of periodic boundary conditions inside the group ","content_id":null,"errors":null,"id":14,"input_connections":{},"inputs":[{"description":"pull-group1-pbcatom - the reference atom index for the treatment of periodic boundary conditions inside the group ","name":"Pull group pbcatom"}],"label":"Pull group pbcatom","name":"Input parameter","outputs":[],"position":{"left":1162,"top":1101},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"integer\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a098f94d-3e47-4076-9eb2-4e08612313f2","when":null,"workflow_outputs":[]},"15":{"annotation":"","id":15,"input_connections":{"Apoprotein PDB":{"id":2,"input_subworkflow_step_id":2,"output_name":"output"},"Force field":{"id":5,"input_subworkflow_step_id":4,"output_name":"output"},"Ligand SDF":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"Water model":{"id":4,"input_subworkflow_step_id":3,"output_name":"output"},"pH":{"id":1,"input_subworkflow_step_id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Create GRO and TOP complex files","outputs":[],"position":{"left":285.5,"top":167},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-0621-6705","name":"Simon Bray"}],"format-version":"0.1","license":"MIT","name":"Create GRO and TOP complex files","steps":{"0":{"annotation":"SD-file for the input ligand","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"SD-file for the input ligand","name":"Ligand SDF"}],"label":"Ligand SDF","name":"Input dataset","outputs":[],"position":{"left":0,"top":461},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"sdf\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"277627e0-eba8-4de5-855b-63711c0db691","when":null,"workflow_outputs":[]},"1":{"annotation":"pH for protonating the ligand. Set to -1.0 to skip.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"pH for protonating the ligand. Set to -1.0 to skip.","name":"pH"}],"label":"pH","name":"Input parameter","outputs":[],"position":{"left":0,"top":718},"tool_id":null,"tool_state":"{\\"default\\": -1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6893b667-f375-481b-96eb-50d5aceb5030","when":null,"workflow_outputs":[]},"2":{"annotation":"PDB file for the protein (without ligand, cofactor, waters)","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"PDB file for the protein (without ligand, cofactor, waters)","name":"Apoprotein PDB"}],"label":"Apoprotein PDB","name":"Input dataset","outputs":[],"position":{"left":1112,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"68f60b7d-13f3-4b21-aedf-986f0f7e48d7","when":null,"workflow_outputs":[]},"3":{"annotation":"Model for water molecules (relevant for subsequent solvation)","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Model for water molecules (relevant for subsequent solvation)","name":"Water model"}],"label":"Water model","name":"Input parameter","outputs":[],"position":{"left":1112,"top":120},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"tip3p\\", \\"tip4p\\", \\"tips3p\\", \\"tip5p\\", \\"spc\\", \\"spce\\", \\"none\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"8d29ae78-60de-45ba-b403-cdd2646af030","when":null,"workflow_outputs":[]},"4":{"annotation":"Force field for protein modelling. GAFF is used for the ligand.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Force field for protein modelling. GAFF is used for the ligand.","name":"Force field"}],"label":"Force field","name":"Input parameter","outputs":[],"position":{"left":1112,"top":220},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"oplsaa\\", \\"gromos43a1\\", \\"amber96\\", \\"gromos53a6\\", \\"amber99sb\\", \\"amber99sb\\", \\"gromos53a5\\", \\"gromos43a2\\", \\"amberGS\\", \\"charmm27\\", \\"amber03\\", \\"gromos54a7\\", \\"gromos45a3\\", \\"amber99\\", \\"amber94\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"16c399b1-e67c-47a9-acd3-61ddc39473d9","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Descriptors","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":278,"top":445},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/ctb_rdkit_descriptors/ctb_rdkit_descriptors/2020.03.4+galaxy1","tool_shed_repository":{"changeset_revision":"0993ac4f4a23","name":"ctb_rdkit_descriptors","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_multiple\\": [\\"FormalCharge\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2020.03.4+galaxy1","type":"tool","uuid":"351397ff-a72d-4c7d-b39a-bc97b8f2c357","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","errors":null,"id":6,"input_connections":{"infile":{"id":0,"output_name":"output"},"ph":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Compound conversion","outputs":[{"name":"outfile","type":"text"}],"position":{"left":278,"top":595},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0","tool_shed_repository":{"changeset_revision":"e2c36f62e22f","name":"openbabel_compound_convert","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"appendtotitle\\": \\"\\", \\"dative_bonds\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"oformat\\": {\\"oformat_opts_selector\\": \\"pdb\\", \\"__current_case__\\": 51}, \\"ph\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"remove_h\\": false, \\"split\\": false, \\"unique\\": {\\"unique_opts_selector\\": \\"\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1+galaxy0","type":"tool","uuid":"c1f8c291-3b11-4afa-b62d-ee2d2d22b11f","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0","errors":null,"id":7,"input_connections":{"ff":{"id":4,"output_name":"output"},"pdb_input":{"id":2,"output_name":"output"},"water":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"GROMACS initial setup","outputs":[{"name":"output1","type":"top"},{"name":"output2","type":"gro"},{"name":"output3","type":"itp"},{"name":"report","type":"txt"}],"position":{"left":1386,"top":44},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0","tool_shed_repository":{"changeset_revision":"070e3ecc3fda","name":"gmx_setup","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": true, \\"ff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_h\\": false, \\"pdb_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"water\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"b922c050-5df2-42b8-a819-1cbf892e06fa","when":null,"workflow_outputs":[{"label":"Position restraints file","output_name":"output3","uuid":"a8969ed8-83f9-4ff3-9634-56f85e7dd4dd"}]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":556,"top":445},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fd7483c3-52b0-489f-a422-2d399e2cd4e5","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","errors":null,"id":9,"input_connections":{"infile":{"id":6,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Search in textfiles","outputs":[{"name":"output","type":"input"}],"position":{"left":556,"top":651},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"case_sensitive\\": \\"-i\\", \\"color\\": \\"NOCOLOR\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"lines_after\\": \\"0\\", \\"lines_before\\": \\"0\\", \\"regex_type\\": \\"-P\\", \\"url_paste\\": \\"HETATM\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"be1e9539-c3ba-4c3c-a06f-a82c5b4d8437","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":10,"input_connections":{"input1":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":834,"top":415},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"50d4ab40-a78d-4d97-8740-7e00edacae6b","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","errors":null,"id":11,"input_connections":{"allparams|nc":{"id":10,"output_name":"integer_param"},"input1":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"AnteChamber","outputs":[{"name":"output1","type":"mol2"}],"position":{"left":1112,"top":540},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_antechamber/ambertools_antechamber/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"8a839e6a1e3e","name":"ambertools_antechamber","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"allparams\\": {\\"nc\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"m\\": \\"1\\", \\"resname\\": \\"UNL\\", \\"c\\": \\"bcc\\", \\"at\\": \\"gaff\\", \\"j\\": \\"4\\"}, \\"extraparams\\": {\\"pf\\": true, \\"usenc\\": true}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"selected_output_format\\": \\"mol2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"32a50a09-a9c7-4d08-9d7f-7d5c9efaa3bf","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","errors":null,"id":12,"input_connections":{"charge":{"id":10,"output_name":"integer_param"},"input1":{"id":11,"output_name":"output1"}},"inputs":[],"label":null,"name":"Generate MD topologies for small molecules","outputs":[{"name":"output","type":"itp"},{"name":"gro_output","type":"gro"}],"position":{"left":1390,"top":423},"post_job_actions":{"HideDatasetActiongro_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"gro_output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/ambertools_acpype/ambertools_acpype/21.10+galaxy0","tool_shed_repository":{"changeset_revision":"095ad4d096d1","name":"ambertools_acpype","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"atomtype\\": \\"gaff\\", \\"charge\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"charge_method\\": \\"user\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"multiplicity\\": \\"1\\", \\"save_gro\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"21.10+galaxy0","type":"tool","uuid":"3ee8d6ce-e7bc-45c5-ae34-ac0348ac959f","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","errors":null,"id":13,"input_connections":{"lig_gro":{"id":12,"output_name":"gro_output"},"lig_top":{"id":12,"output_name":"output"},"prot_gro":{"id":7,"output_name":"output2"},"prot_top":{"id":7,"output_name":"output1"}},"inputs":[],"label":null,"name":"Merge GROMACS topologies","outputs":[{"name":"complex_top","type":"top"},{"name":"complex_gro","type":"gro"}],"position":{"left":1678,"top":203},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_merge_topology_files/gmx_merge_topology_files/3.4.3+galaxy0","tool_shed_repository":{"changeset_revision":"bc31d02be922","name":"gmx_merge_topology_files","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"lig_gro\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lig_top\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prot_gro\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prot_top\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.3+galaxy0","type":"tool","uuid":"2a22a7fd-3e14-4f27-b1a8-a6d66cef292a","when":null,"workflow_outputs":[{"label":"Complex topology","output_name":"complex_top","uuid":"df11b684-24b0-481f-956d-c7cdc1e334ba"},{"label":"Complex GRO","output_name":"complex_gro","uuid":"193f8774-0334-49d3-b8ae-bdc3484cb1be"}]}},"tags":"","uuid":"6cd49957-0863-42cd-81de-171e57516221"},"tool_id":null,"type":"subworkflow","uuid":"9e64bd77-92d9-4c39-9b37-17636deeb1dd","when":null,"workflow_outputs":[{"label":"Topology","output_name":"Complex topology","uuid":"e092ce5a-3fca-4183-b244-daf8a0582e72"},{"label":"Structure file (GRO format, optional)","output_name":"Complex GRO","uuid":"ab63a333-8a78-4476-946a-f1bac8fde57c"},{"label":"GROMACS setup (ITP) on input dataset(s)","output_name":"Position restraints file","uuid":"8f1c5eee-7510-4430-b3e0-e2301926285d"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0","errors":null,"id":16,"input_connections":{"token_set_0|line":{"id":7,"output_name":"output"},"token_set_0|repeat_select|times":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":288,"top":902},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"0213cfb8-a156-4ddc-bdc6-262706119f10","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":17,"input_connections":{"components_1|param_type|component_value":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":586,"top":1269},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"pull_coord1_rate = \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"74c216e8-326c-4d5d-afc8-d905fb55068b","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":18,"input_connections":{"components_1|param_type|component_value":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1162,"top":1221},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"dt = \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"366aaf2a-5aa9-45ea-93fd-b032c567e8b9","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":19,"input_connections":{"components_1|param_type|component_value":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":864,"top":1422},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"nsteps = \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"44832635-6428-4a4e-a8cb-d0e088542703","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":20,"input_connections":{"components_1|param_type|component_value":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1440,"top":1065},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" pull-group1-pbcatom = \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"c6a76b14-657f-4ae9-8cc4-7d25f89ba8e4","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0","errors":null,"id":21,"input_connections":{"conc":{"id":3,"output_name":"output"},"gro_input":{"id":15,"output_name":"Complex GRO"},"top_input":{"id":15,"output_name":"Complex topology"}},"inputs":[],"label":null,"name":"GROMACS solvation and adding ions","outputs":[{"name":"output1","type":"gro"},{"name":"output2","type":"top"}],"position":{"left":586,"top":374},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c4fbab8e03c5","name":"gmx_solvate","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": false, \\"conc\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mxw\\": \\"0\\", \\"neutralise\\": \\"-neutral\\", \\"seed\\": \\"1\\", \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"water_model\\": \\"spc216\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"5adf2a94-fac0-41aa-8cdb-cf475bf0bf21","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","errors":null,"id":22,"input_connections":{"split_parms|input":{"id":16,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":586,"top":725},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","tool_shed_repository":{"changeset_revision":"6cbe2f30c2d7","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.0","type":"tool","uuid":"40134c51-d93d-4e92-81b5-86cbff22a479","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","errors":null,"id":23,"input_connections":{"infile":{"id":9,"output_name":"output"},"text_input":{"id":17,"output_name":"out1"}},"inputs":[],"label":null,"name":"Add line to file","outputs":[{"name":"outfile","type":"input"}],"position":{"left":864,"top":1260},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","tool_shed_repository":{"changeset_revision":"8e251055b1a9","name":"add_line_to_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": \\"footer\\", \\"text_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"36e82014-346e-4739-b7cb-91c0480e696a","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0","errors":null,"id":24,"input_connections":{"gro_input":{"id":21,"output_name":"output1"},"top_input":{"id":21,"output_name":"output2"}},"inputs":[],"label":null,"name":"GROMACS energy minimization","outputs":[{"name":"output1","type":"gro"},{"name":"output2","type":"edr"}],"position":{"left":864,"top":473},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0","tool_shed_repository":{"changeset_revision":"b46d4b4d995c","name":"gmx_em","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": false, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"steep\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": \\"50000\\", \\"emtol\\": \\"1000.0\\", \\"emstep\\": \\"0.01\\", \\"seed\\": \\"1\\"}, \\"mxw\\": \\"0\\", \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"effbec33-36ce-4232-8081-0ebef2ccee91","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":25,"input_connections":{"input1":{"id":22,"output_name":"list_output_txt"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":864,"top":705},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"34f23531-275e-4dae-957b-c01859485e21","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","errors":null,"id":26,"input_connections":{"infile":{"id":23,"output_name":"outfile"},"text_input":{"id":19,"output_name":"out1"}},"inputs":[],"label":null,"name":"Add line to file","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1162,"top":1413},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","tool_shed_repository":{"changeset_revision":"8e251055b1a9","name":"add_line_to_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": \\"footer\\", \\"text_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"92e7bb53-d6ca-4290-8a8c-af3f0e93b533","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_makendx/gmx_makendx/2022+galaxy0","errors":null,"id":27,"input_connections":{"input_file":{"id":24,"output_name":"output1"}},"inputs":[],"label":null,"name":"Create GROMACS index files","outputs":[{"name":"ndx","type":"ndx"}],"position":{"left":1162,"top":473},"post_job_actions":{"HideDatasetActionndx":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"ndx"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_makendx/gmx_makendx/2022+galaxy0","tool_shed_repository":{"changeset_revision":"e3e536aa2633","name":"gmx_makendx","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"capture_log\\": false, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sel\\": \\"13 & ! a H*\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"3380956f-6363-46fa-87b1-f065ab4d4b56","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","errors":null,"id":28,"input_connections":{"gro_input":{"id":24,"output_name":"output1"},"inps|itp_in":{"id":15,"output_name":"Position restraints file"},"sets|mdp|md_steps":{"id":8,"output_name":"output"},"sets|mdp|step_length":{"id":11,"output_name":"output"},"sets|mdp|temperature":{"id":25,"output_name":"integer_param"},"top_input":{"id":21,"output_name":"output2"}},"inputs":[{"description":"runtime parameter for tool GROMACS simulation","name":"inps"},{"description":"runtime parameter for tool GROMACS simulation","name":"inps"}],"label":null,"name":"GROMACS simulation","outputs":[{"name":"output1","type":"gro"},{"name":"output4","type":"xtc"},{"name":"output5","type":"cpt"},{"name":"report","type":"txt"}],"position":{"left":1162,"top":625},"post_job_actions":{"HideDatasetActionoutput1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output1"},"HideDatasetActionoutput4":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output4"},"HideDatasetActionoutput5":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output5"},"HideDatasetActionreport":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c0c9a5024177","name":"gmx_sim","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inps\\": {\\"cpt_in\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"itp_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ndx_in\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"mxw\\": \\"0\\", \\"outps\\": {\\"traj\\": \\"xtc\\", \\"str\\": \\"gro\\", \\"cpt_out\\": \\"true\\", \\"edr_out\\": \\"false\\", \\"xvg_out\\": \\"false\\", \\"tpr_out\\": \\"false\\"}, \\"sets\\": {\\"ensembleselect\\": {\\"ensemble\\": \\"nvt\\", \\"__current_case__\\": 0, \\"startvel\\": \\"false\\"}, \\"mdp\\": {\\"mdpfile\\": \\"default\\", \\"__current_case__\\": 1, \\"integrator\\": \\"md\\", \\"constraints\\": \\"h-bonds\\", \\"cutoffscheme\\": \\"Verlet\\", \\"coulombtype\\": \\"PME\\", \\"temperature\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"systemTcouple\\": \\"false\\", \\"step_length\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"write_freq\\": \\"1000\\", \\"rcoulomb\\": \\"1.0\\", \\"rlist\\": \\"1.0\\", \\"rvdw\\": \\"1.0\\", \\"md_steps\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"2c06b8d7-b07c-4dc7-9237-80c19b731bde","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","errors":null,"id":29,"input_connections":{"infile":{"id":26,"output_name":"outfile"},"text_input":{"id":18,"output_name":"out1"}},"inputs":[],"label":null,"name":"Add line to file","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1440,"top":1260},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","tool_shed_repository":{"changeset_revision":"8e251055b1a9","name":"add_line_to_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": \\"footer\\", \\"text_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"612d572b-541f-4e76-9a9a-d477abd10a07","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","errors":null,"id":30,"input_connections":{"infile":{"id":27,"output_name":"ndx"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":1440,"top":670},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/System/pullgrp/;t;d\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"95cac327-99f3-455e-8765-a7db33343894","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","errors":null,"id":31,"input_connections":{"infile":{"id":29,"output_name":"outfile"},"text_input":{"id":20,"output_name":"out1"}},"inputs":[],"label":null,"name":"Add line to file","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1728,"top":863},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"mdp"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","tool_shed_repository":{"changeset_revision":"8e251055b1a9","name":"add_line_to_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": \\"footer\\", \\"text_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"506adfe5-8c0c-41d1-8a13-42f8821677de","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","errors":null,"id":32,"input_connections":{"infile":{"id":30,"output_name":"output"},"text_input":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Add line to file","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1728,"top":553},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0","tool_shed_repository":{"changeset_revision":"8e251055b1a9","name":"add_line_to_file","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": \\"footer\\", \\"text_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"45466bad-4528-4558-89e2-c80cd3d73029","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","errors":null,"id":33,"input_connections":{"inputs":{"id":27,"output_name":"ndx"},"queries_0|inputs2":{"id":32,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2006,"top":485},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"ndx"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"1a933020-6bde-4378-bae2-a401389e2d63","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","errors":null,"id":34,"input_connections":{"gro_input":{"id":28,"output_name":"output1"},"inps|cpt_in":{"id":28,"output_name":"output5"},"inps|ndx_in":{"id":33,"output_name":"out_file1"},"sets|mdp|mdp_input":{"id":31,"output_name":"outfile"},"top_input":{"id":21,"output_name":"output2"}},"inputs":[{"description":"runtime parameter for tool GROMACS simulation","name":"inps"}],"label":null,"name":"GROMACS simulation","outputs":[{"name":"output1","type":"gro"},{"name":"output4","type":"xtc"},{"name":"output7","type":"xvg"},{"name":"output7_1","type":"xvg"},{"name":"output8","type":"tpr"},{"name":"report","type":"txt"}],"position":{"left":2294,"top":690},"post_job_actions":{"ChangeDatatypeActionoutput8":{"action_arguments":{"newtype":"tpr"},"action_type":"ChangeDatatypeAction","output_name":"output8"},"HideDatasetActionoutput7_1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output7_1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0","tool_shed_repository":{"changeset_revision":"c0c9a5024177","name":"gmx_sim","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 1, \\"__job_resource__select\\": \\"yes\\", \\"gpu\\": \\"1\\"}, \\"capture_log\\": true, \\"gro_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inps\\": {\\"cpt_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"itp_in\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"ndx_in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"mxw\\": \\"0\\", \\"outps\\": {\\"traj\\": \\"xtc\\", \\"str\\": \\"gro\\", \\"cpt_out\\": \\"false\\", \\"edr_out\\": \\"false\\", \\"xvg_out\\": \\"true\\", \\"tpr_out\\": \\"true\\"}, \\"sets\\": {\\"ensembleselect\\": {\\"ensemble\\": \\"nvt\\", \\"__current_case__\\": 0, \\"startvel\\": \\"false\\"}, \\"mdp\\": {\\"mdpfile\\": \\"custom\\", \\"__current_case__\\": 0, \\"mdp_input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"top_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022+galaxy0","type":"tool","uuid":"c724b44d-5a73-4e92-8406-bf45ebbdb716","when":null,"workflow_outputs":[{"label":"XVG files","output_name":"output7","uuid":"0b3cd371-b269-46ac-992d-fad21c9fea35"},{"label":"XTC files","output_name":"output4","uuid":"30596358-e89f-48c8-a81f-8c58c28d82b1"},{"label":"TPR files","output_name":"output8","uuid":"addf9510-b7d7-418d-baa5-8e10aade66fe"},{"label":"GRO files","output_name":"output1","uuid":"77e25856-9b12-42dc-9a49-634bb0955851"},{"label":"Log","output_name":"report","uuid":"7edd7681-6e75-406d-ad24-465a496a3c3e"}]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/chemteam/biomd_neqgamma/biomd_neqgamma/0.1.5.2+galaxy1","errors":null,"id":35,"input_connections":{"xvgs":{"id":34,"output_name":"output7"}},"inputs":[{"description":"runtime parameter for tool dcTMD friction correction","name":"json"}],"label":null,"name":"dcTMD friction correction","outputs":[{"name":"outp_col","type":"input"},{"name":"outp_frict_col","type":"input"}],"position":{"left":2572,"top":875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/chemteam/biomd_neqgamma/biomd_neqgamma/0.1.5.2+galaxy1","tool_shed_repository":{"changeset_revision":"afcb925def69","name":"biomd_neqgamma","owner":"chemteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"T\\": \\"300.0\\", \\"av\\": \\"1\\", \\"json\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"sigma\\": \\"1\\", \\"vel\\": \\"0.001\\", \\"xvgs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.5.2+galaxy1","type":"tool","uuid":"7446ff96-7c10-4363-9067-045c05dfddf6","when":null,"workflow_outputs":[{"label":"Friction data","output_name":"outp_frict","uuid":"59a92ec0-f2cc-4052-bb0c-d0445dcc03b5"},{"label":"Free energy data","output_name":"outp","uuid":"13ec40b1-9c75-40b5-ac04-932801eaa671"}]}},"tags":[],"uuid":"7950b716-2bc9-4cb6-8472-6506c702adea","version":1},"readme":"# GROMACS dcTMD free energy calculation\\n\\nPerform an ensemble of targeted MD simulations of a user-specified size using\\nthe GROMACS PULL code and calculate dcTMD free energy and friction profiles\\nfor the resulting dissocation pathway. Note that pathway separation is not\\nperformed by the workflow; the user is responsible for checking the ensemble themselves.\\n\\nThe input protein (PDB) and ligand (SDF) files provided are parameterized by\\nthe \'Protein-ligand complex parameterization\' subworkflow.\\n\\nNote that the workflow uses a MDP file for configuring the TMD simulations; this\\nis packaged alongside the workflow as `tmd.mdp`.\\n\\n## Citations\\n* Steffen Wolf and Gerhard Stock (2018), Targeted Molecular Dynamics Calculations of Free Energy Profiles Using a Nonequilibrium Friction Correction, J. Chem. Theory Comput. doi:10.1021/acs.jctc.8b00835\\n* Steffen Wolf, Benjamin Lickert, Simon Bray and Gerhard Stock (2020), Multisecond ligand dissociation dynamics from atomistic simulations, Nat. Commun. doi:10.1038/s41467-020-16655-1\\n","changelog":"# Changelog\\n\\n## [0.1.5] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1.4] - 2023-11-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2021.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_setup/gmx_setup/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2021.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_solvate/gmx_solvate/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2021.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_em/gmx_em/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_makendx/gmx_makendx/2021.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_makendx/gmx_makendx/2022+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2021.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/chemteam/gmx_sim/gmx_sim/2022+galaxy0`\\n\\n## [0.1.3] - 2022-05-25\\n\\n### Changed\\n- Changed creator ORCID to absolute URI\\n\\n## [0.1.2] - 2021-12-13\\n\\n### Added\\n- Added GitHub Actions workflow. No functional changes.\\n\\n## [0.1.1] - 2021-12-06\\n\\n### Fixed\\n- `.workflowhub.yml`: use `repo-name/dockstore-workflow-name` scheme for workflows.\\n- Moved test data inside repo.\\n\\n## [0.1]\\n\\n- Initial version of GROMACS dcTMD workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## dcTMD calculations with GROMACS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Ligand SDF\\"]@{ shape: doc }\\n1[\\"ℹ️ pH to protonate ligand\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Protein PDB\\"]@{ shape: doc }\\n3[\\"ℹ️ Salt concentration\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Water model\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Force field\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Number of simulations\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Temperature\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Number of equilibration steps\\"]@{ shape: lean-l }\\n9[\\"Online data\\"]@{ shape: process }\\n10[\\"ℹ️ Pulling rate\\"]@{ shape: lean-l }\\n11[\\"ℹ️ Step length (ps)\\"]@{ shape: lean-l }\\n12[\\"ℹ️ Protein pull group\\"]@{ shape: lean-l }\\n13[\\"ℹ️ Number of steps\\"]@{ shape: lean-l }\\n14[\\"ℹ️ Pull group pbcatom\\"]@{ shape: lean-l }\\n15[\\"🛠️ Create GRO and TOP complex files\\"]@{ shape: subprocess }\\n2 --> 15\\n5 --> 15\\n0 --> 15\\n4 --> 15\\n1 --> 15\\n16[\\"Create text file\\"]@{ shape: process }\\n7 --> 16\\n6 --> 16\\n17[\\"Compose text parameter value\\"]@{ shape: process }\\n10 --> 17\\n18[\\"Compose text parameter value\\"]@{ shape: process }\\n11 --> 18\\n19[\\"Compose text parameter value\\"]@{ shape: process }\\n13 --> 19\\n20[\\"Compose text parameter value\\"]@{ shape: process }\\n14 --> 20\\n21[\\"GROMACS solvation and adding ions\\"]@{ shape: process }\\n3 --> 21\\n15 --> 21\\n15 --> 21\\n22[\\"Split file\\"]@{ shape: process }\\n16 --> 22\\n23[\\"Add line to file\\"]@{ shape: process }\\n9 --> 23\\n17 --> 23\\n24[\\"GROMACS energy minimization\\"]@{ shape: process }\\n21 --> 24\\n21 --> 24\\n25[\\"Parse parameter value\\"]@{ shape: process }\\n22 --> 25\\n26[\\"Add line to file\\"]@{ shape: process }\\n23 --> 26\\n19 --> 26\\n27[\\"Create GROMACS index files\\"]@{ shape: process }\\n24 --> 27\\n28[\\"GROMACS simulation\\"]@{ shape: process }\\n24 --> 28\\n15 --> 28\\n8 --> 28\\n11 --> 28\\n25 --> 28\\n21 --> 28\\n29[\\"Add line to file\\"]@{ shape: process }\\n26 --> 29\\n18 --> 29\\n30[\\"Text transformation\\"]@{ shape: process }\\n27 --> 30\\n31[\\"Add line to file\\"]@{ shape: process }\\n29 --> 31\\n20 --> 31\\n32[\\"Add line to file\\"]@{ shape: process }\\n30 --> 32\\n12 --> 32\\n33[\\"Concatenate datasets\\"]@{ shape: process }\\n27 --> 33\\n32 --> 33\\n34[\\"GROMACS simulation\\"]@{ shape: process }\\n28 --> 34\\n28 --> 34\\n33 --> 34\\n31 --> 34\\n21 --> 34\\n35[\\"dcTMD friction correction\\"]@{ shape: process }\\n34 --> 35\\n```\\n\\n## Create GRO and TOP complex files\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Ligand SDF\\"]@{ shape: doc }\\n1[\\"ℹ️ pH\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Apoprotein PDB\\"]@{ shape: doc }\\n3[\\"ℹ️ Water model\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Force field\\"]@{ shape: lean-l }\\n5[\\"Descriptors\\"]@{ shape: process }\\n0 --> 5\\n6[\\"Compound conversion\\"]@{ shape: process }\\n0 --> 6\\n1 --> 6\\n7[\\"GROMACS initial setup\\"]@{ shape: process }\\n4 --> 7\\n2 --> 7\\n3 --> 7\\n8[\\"Cut\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Search in textfiles\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Parse parameter value\\"]@{ shape: process }\\n8 --> 10\\n11[\\"AnteChamber\\"]@{ shape: process }\\n10 --> 11\\n9 --> 11\\n12[\\"Generate MD topologies for small molecules\\"]@{ shape: process }\\n10 --> 12\\n11 --> 12\\n13[\\"Merge GROMACS topologies\\"]@{ shape: process }\\n12 --> 13\\n12 --> 13\\n7 --> 13\\n7 --> 13\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/gromacs-dctmd/main","dockstore_id":17276,"categories":["Structural Biology"],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078513","tests":[{"doc":"Test outline for gromacs-dctmd.ga","job":{"Force field":"amber99sb","Ligand SDF":{"class":"File","path":"test-data/lig.sdf"},"Number of simulations":"2","Number of steps":"100","Protein PDB":{"class":"File","path":"test-data/mpro.pdb"},"Protein pull group":"350 364 378 606 623 658 669 683 721 2164 2184 2203 2217 2224 2235 2543 2558 2585 2841 2853 2877 2894","Pull group pbcatom":"606","Pulling rate":"0.001","Step length (ps)":"0.001","Water model":"tip3p","Temperature":"300","Number of equilibration steps":"50","Salt concentration":0.1},"outputs":{"XTC files":{"element_tests":{"split_file_000000.txt":{"assert":{"has_size":138900,"delta":500}}}},"Free energy data":{"class":"File","assert":{"has_n_lines":101,"has_n_columns":5}},"Friction data":{"class":"File","assert":{"has_n_lines":101,"has_n_columns":5}}}}]}],"path":"./workflows/computational-chemistry/gromacs-dctmd"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/pox-virus-half-genome.ga","testParameterFiles":["/pox-virus-half-genome-tests.yml"],"authors":[{"name":"Viktoria Isabel Schwarz","orcid":"0000-0001-6897-1215"},{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"A workflow for the analysis of pox virus genomes sequenced as half-genomes (for ITR resolution) in a tiled-amplicon approach","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0001-6897-1215","name":"Viktoria Isabel Schwarz"},{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","license":"MIT","release":"0.3","name":"Pox Virus Illumina Amplicon Workflow from half-genomes","steps":{"0":{"annotation":"The viral reference sequence to map sequenced reads against","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"The viral reference sequence to map sequenced reads against","name":"Reference FASTA"}],"label":"Reference FASTA","name":"Input dataset","outputs":[],"position":{"left":244.316650390625,"top":642.8833312988281},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f17c7022-fc94-42b5-ae90-7b3ec3d70ffb","when":null,"workflow_outputs":[]},"1":{"annotation":"The workflow expects a primer scheme split into two separate sequencing pools. These pools must be denoted as pool1/pool2 in the BED score column. The pool ids may, optionally, be followed by indicators of the subpool for tiled PCR amplification (e.g., pool1a/pool1b/pool2a/pool2b)","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"The workflow expects a primer scheme split into two separate sequencing pools. These pools must be denoted as pool1/pool2 in the BED score column. The pool ids may, optionally, be followed by indicators of the subpool for tiled PCR amplification (e.g., pool1a/pool1b/pool2a/pool2b)","name":"Primer Scheme"}],"label":"Primer Scheme","name":"Input dataset","outputs":[],"position":{"left":0,"top":722},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"1829d4e0-ee74-4a94-8dc5-98a11d96b961","when":null,"workflow_outputs":[]},"2":{"annotation":"A collection of the sequencing data obtained from the **pool1** run of all samples","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"A collection of the sequencing data obtained from the **pool1** run of all samples","name":"PE Reads Pool1"}],"label":"PE Reads Pool1","name":"Input dataset collection","outputs":[],"position":{"left":1261.3000030517578,"top":554.6000061035156},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"c154fa4b-feab-49f5-b1d7-b0a070d78a11","when":null,"workflow_outputs":[]},"3":{"annotation":"A collection of the sequencing data obtained from the **pool2** run of all samples","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"A collection of the sequencing data obtained from the **pool2** run of all samples","name":"PE Reads Pool2"}],"label":"PE Reads Pool2","name":"Input dataset collection","outputs":[],"position":{"left":1264.6999969482422,"top":664.9666748046875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"9c159726-6213-4bf3-a8ef-7e901ae2e1f3","when":null,"workflow_outputs":[]},"4":{"annotation":"Only sequenced bases with at least this  base calling quality will be considered for consensus calling.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Only sequenced bases with at least this  base calling quality will be considered for consensus calling.","name":"Minimum quality score to call base"}],"label":"Minimum quality score to call base","name":"Input parameter","outputs":[],"position":{"left":3694.38330078125,"top":984.5999755859375},"tool_id":null,"tool_state":"{\\"default\\": 20, \\"validators\\": [], \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"43b722a0-385f-4ce7-9255-5041224ae149","when":null,"workflow_outputs":[]},"5":{"annotation":"A consensus **SNV** call requires that the alternate base is seen in at least this fraction of reads covering it.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"A consensus **SNV** call requires that the alternate base is seen in at least this fraction of reads covering it.","name":"Allele frequency to call SNV"}],"label":"Allele frequency to call SNV","name":"Input parameter","outputs":[],"position":{"left":3696.166748046875,"top":1097.88330078125},"tool_id":null,"tool_state":"{\\"default\\": 0.7, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"378c2c1d-9250-4871-add1-1744ebd83eea","when":null,"workflow_outputs":[]},"6":{"annotation":"A consensus **indel** call requires that the indel is seen in at least this fraction of reads covering it.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"A consensus **indel** call requires that the indel is seen in at least this fraction of reads covering it.","name":"Allele frequency to call indel"}],"label":"Allele frequency to call indel","name":"Input parameter","outputs":[],"position":{"left":3699.13330078125,"top":1211.86669921875},"tool_id":null,"tool_state":"{\\"default\\": 0.8, \\"validators\\": [], \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"8ccccd63-0323-4791-8549-99b322d27ac3","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.4","errors":null,"id":7,"input_connections":{"ref|input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Compute sequence length","name":"ref"}],"label":null,"name":"Compute sequence length","outputs":[{"name":"output","type":"tabular"}],"position":{"left":465.9666748046875,"top":438.8666687011719},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.4","tool_shed_repository":{"changeset_revision":"5cbde03c1103","name":"fasta_compute_length","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"ref\\": {\\"ref_source\\": \\"history\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_first\\": \\"0\\", \\"keep_first_word\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"f562eb06-797e-4efb-a464-6d368f573dbe","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Grep1","errors":null,"id":8,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":"Select pool1 primers","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":238.066650390625,"top":318.5500030517578},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Pool1 primers"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"[p|P]ool1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"c46843eb-0908-493a-958b-fbc236247626","when":null,"workflow_outputs":[{"label":"pool1_primers","output_name":"out_file1","uuid":"0159e87f-d2cb-4eed-af76-dbf37e9f79f4"}]},"9":{"annotation":"","content_id":"Grep1","errors":null,"id":9,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":"Select pool2 primers","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":206.0333251953125,"top":917.13330078125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Pool2 primers"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"[p|P]ool2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"edcdbb82-3a77-433d-92c7-4e954af22c0c","when":null,"workflow_outputs":[{"label":"pool2_primers","output_name":"out_file1","uuid":"6d14606b-2ddf-4a5f-9535-97f5ff2bca20"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":10,"input_connections":{"input_collection":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":1548.6333312988281,"top":644.2999877929688},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"b524acb2-97ff-41b0-ae89-9f8d5b71918d","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":11,"input_connections":{"single_paired|paired_input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":"fastp: Trimmed Illumina Reads Pool1","name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":2014.4000244140625,"top":335.21665954589844},"post_job_actions":{"RenameDatasetActionoutput_paired_coll":{"action_arguments":{"newname":"fastp on pool1 - processed reads"},"action_type":"RenameDatasetAction","output_name":"output_paired_coll"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"fastp on pool1 - report"},"action_type":"RenameDatasetAction","output_name":"report_html"},"TagDatasetActionoutput_paired_coll":{"action_arguments":{"tags":"#pool1"},"action_type":"TagDatasetAction","output_name":"output_paired_coll"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"#pool1"},"action_type":"TagDatasetAction","output_name":"report_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\", \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"bf2dc639-b721-4ca5-a1a3-0a55d1185143","when":null,"workflow_outputs":[{"label":"fastp_pool1_html","output_name":"report_html","uuid":"bc9529db-6ffb-4cd0-84b0-460d8c58de2a"},{"label":"fastp_pool1_reads","output_name":"output_paired_coll","uuid":"35f23a63-de11-4b78-a325-683636a1c151"},{"label":"fastp_pool1_json","output_name":"report_json","uuid":"1736b78c-f7d9-4d21-a3b0-b20a55f0e470"}]},"12":{"annotation":"","content_id":"Cut1","errors":null,"id":12,"input_connections":{"input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":732.38330078125,"top":408.96665954589844},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"889acb03-b55a-40fa-a1d9-804e1eb1d958","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":13,"input_connections":{"in_file":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":478.13330078125,"top":184.21665954589844},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"max\\", \\"op_column\\": \\"3\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"6999e1c0-819e-4f9c-8ec0-75e985802a54","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","errors":null,"id":14,"input_connections":{"in_file":{"id":9,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"input"}],"position":{"left":452.13330078125,"top":933.2666625976562},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0","tool_shed_repository":{"changeset_revision":"4c07ddedc198","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"\\", \\"header_in\\": false, \\"header_out\\": false, \\"ignore_case\\": false, \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"narm\\": false, \\"need_sort\\": false, \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"min\\", \\"op_column\\": \\"2\\"}], \\"print_full_line\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.8+galaxy0","type":"tool","uuid":"2130b998-eb44-42dc-81ee-ad098bb5d721","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","errors":null,"id":15,"input_connections":{"split_parms|input":{"id":10,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":null,"name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":1801.1333312988281,"top":648.816650390625},"post_job_actions":{"HideDatasetActionlist_output_txt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2","tool_shed_repository":{"changeset_revision":"2dae863c8f42","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"pool1_ids\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.2","type":"tool","uuid":"b912e921-c821-4ede-86d6-d1752eb18ff7","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"__SORTLIST__","errors":null,"id":16,"input_connections":{"input":{"id":3,"output_name":"output"},"sort_type|sort_file":{"id":10,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Sort collection","name":"sort_type"}],"label":null,"name":"Sort collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1763.3999938964844,"top":1043.5999755859375},"post_job_actions":{},"tool_id":"__SORTLIST__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sort_type\\": {\\"sort_type\\": \\"file\\", \\"__current_case__\\": 2, \\"sort_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"83a64b0a-a56a-454b-9c8b-10ff0916fa4b","when":null,"workflow_outputs":[{"label":"input dataset(s) (sorted)","output_name":"output","uuid":"84d9f9ca-c0ef-4c3a-9ae6-dd58a3ad54e0"}]},"17":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":17,"input_connections":{"input1":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":"Get end position of sequence","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":971,"top":252.71665954589844},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"ec25cd53-697b-4de1-91fd-4cf6d08af23d","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":18,"input_connections":{"input1":{"id":13,"output_name":"out_file"}},"inputs":[],"label":"Get end position of Pool1","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":725.433349609375,"top":108.66667175292969},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"27c913a7-f1cc-47ec-8e19-a942cd5ce8a3","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":19,"input_connections":{"input1":{"id":14,"output_name":"out_file"}},"inputs":[],"label":"Get start position of Pool2","name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":707.1500244140625,"top":933.75},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"33b91954-218c-46bf-9cf8-8a0d9535c1be","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":20,"input_connections":{"input1":{"id":15,"output_name":"list_output_txt"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":2063.7833251953125,"top":621.2166748046875},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"3b0c55ad-66dc-4ce2-9525-f0982ab5f8a7","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","errors":null,"id":21,"input_connections":{"single_paired|paired_input":{"id":16,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":"fastp: Trimmed Illumina Reads Pool2","name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":2007.3499755859375,"top":1011.933349609375},"post_job_actions":{"RenameDatasetActionoutput_paired_coll":{"action_arguments":{"newname":"fastp on pool2 - processed reads"},"action_type":"RenameDatasetAction","output_name":"output_paired_coll"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"fastp on pool2 - report"},"action_type":"RenameDatasetAction","output_name":"report_html"},"TagDatasetActionoutput_paired_coll":{"action_arguments":{"tags":"#pool2"},"action_type":"TagDatasetAction","output_name":"output_paired_coll"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"#pool2"},"action_type":"TagDatasetAction","output_name":"report_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4","tool_shed_repository":{"changeset_revision":"10678d49d39e","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 1, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\", \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy4","type":"tool","uuid":"9c9dcad3-f146-42fb-9fbc-650703ecc901","when":null,"workflow_outputs":[{"label":"fastp_pool2_json","output_name":"report_json","uuid":"238d7dcd-ffc8-48ad-922a-8577d71bb998"},{"label":"fastp_pool2_reads","output_name":"output_paired_coll","uuid":"0d6bd58c-00bf-45e5-bec6-edffd77e962f"},{"label":"fastp_pool2_html","output_name":"report_html","uuid":"794a733f-a08b-4577-9ca0-7eda43e1aaf8"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":22,"input_connections":{"components_0|param_type|component_value":{"id":18,"output_name":"text_param"},"components_2|param_type|component_value":{"id":17,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1201.9166717529297,"top":102.05000305175781},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"-\\"}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"cfe61da6-24c0-4f7c-9ef0-428edd13b964","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":23,"input_connections":{"components_1|param_type|component_value":{"id":19,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":966.2333374023438,"top":920.86669921875},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"1-\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"576bae3e-38d9-48cb-8afc-80d10e1cd43f","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":24,"input_connections":{"components_0|param_type|component_value":{"id":20,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":2326.9833374023438,"top":586.8500061035156},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"_pool1\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"874cd418-17df-4ccf-b66e-869601aec73b","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":25,"input_connections":{"components_0|param_type|component_value":{"id":20,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":2314.183349609375,"top":1257.86669921875},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"_pool2\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"f121cfa5-f48e-4eb4-b40b-f078b3f52987","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskseq51/5.0.0","errors":null,"id":26,"input_connections":{"input1":{"id":0,"output_name":"output"},"regions":{"id":22,"output_name":"out1"}},"inputs":[],"label":"Mask Reference for Pool1","name":"maskseq","outputs":[{"name":"out_file1","type":"fasta"}],"position":{"left":1452.6333312988281,"top":278.1666717529297},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Masked reference for mapping of pool1"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskseq51/5.0.0","tool_shed_repository":{"changeset_revision":"63dd26468588","name":"emboss_5","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maskchar\\": \\"N\\", \\"out_format1\\": \\"fasta\\", \\"regions\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tolower\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.0","type":"tool","uuid":"3c8b86c1-7e53-4b72-b836-88a921a368ba","when":null,"workflow_outputs":[{"label":"masked_ref_pool1","output_name":"out_file1","uuid":"0c3b5b20-c426-428c-a08b-c835a82477c1"}]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskseq51/5.0.0","errors":null,"id":27,"input_connections":{"input1":{"id":0,"output_name":"output"},"regions":{"id":23,"output_name":"out1"}},"inputs":[],"label":"Mask Reference for Pool2","name":"maskseq","outputs":[{"name":"out_file1","type":"fasta"}],"position":{"left":1350.1000061035156,"top":851},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Masked reference for mapping of pool2"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskseq51/5.0.0","tool_shed_repository":{"changeset_revision":"63dd26468588","name":"emboss_5","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maskchar\\": \\"N\\", \\"out_format1\\": \\"fasta\\", \\"regions\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tolower\\": \\"no\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.0","type":"tool","uuid":"140608b3-48b3-46f2-92e3-2b9c3fb9e3b2","when":null,"workflow_outputs":[{"label":"masked_ref_pool2","output_name":"out_file1","uuid":"af755a5e-2b20-49bc-9d52-34781f937f48"}]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","errors":null,"id":28,"input_connections":{"fastq_input|fastq_input1":{"id":11,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":26,"output_name":"out_file1"},"rg|read_group_id_conditional|ID":{"id":24,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":2542.316650390625,"top":98.73333740234375},"post_job_actions":{"RenameDatasetActionbam_output":{"action_arguments":{"newname":"Mapping of Pool1"},"action_type":"RenameDatasetAction","output_name":"bam_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","tool_shed_repository":{"changeset_revision":"2477830927ec","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"read_group_id_conditional\\": {\\"do_auto_name\\": false, \\"__current_case__\\": 1, \\"ID\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"read_group_sm_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"PL\\": \\"ILLUMINA\\", \\"read_group_lb_conditional\\": {\\"do_auto_name\\": false, \\"__current_case__\\": 1, \\"LB\\": \\"\\"}, \\"CN\\": \\"\\", \\"DS\\": \\"\\", \\"DT\\": \\"\\", \\"FO\\": \\"\\", \\"KS\\": \\"\\", \\"PG\\": \\"\\", \\"PI\\": null, \\"PU\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.18","type":"tool","uuid":"c0a73a1f-0a76-40b6-bdd7-259e59ba2d94","when":null,"workflow_outputs":[{"label":"Mapping of Pool1","output_name":"bam_output","uuid":"a9caa907-b6bf-4af9-94f4-51422194b076"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","errors":null,"id":29,"input_connections":{"fastq_input|fastq_input1":{"id":21,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":27,"output_name":"out_file1"},"rg|read_group_id_conditional|ID":{"id":25,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":2538.550048828125,"top":838.8333129882812},"post_job_actions":{"RenameDatasetActionbam_output":{"action_arguments":{"newname":"Mapping of Pool2"},"action_type":"RenameDatasetAction","output_name":"bam_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","tool_shed_repository":{"changeset_revision":"2477830927ec","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"read_group_id_conditional\\": {\\"do_auto_name\\": false, \\"__current_case__\\": 1, \\"ID\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"read_group_sm_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"PL\\": \\"ILLUMINA\\", \\"read_group_lb_conditional\\": {\\"do_auto_name\\": false, \\"__current_case__\\": 1, \\"LB\\": \\"\\"}, \\"CN\\": \\"\\", \\"DS\\": \\"\\", \\"DT\\": \\"\\", \\"FO\\": \\"\\", \\"KS\\": \\"\\", \\"PG\\": \\"\\", \\"PI\\": null, \\"PU\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.18","type":"tool","uuid":"4dfc264e-7453-4c5f-a073-9e2bff28dc9c","when":null,"workflow_outputs":[{"label":"mapping_pool2","output_name":"bam_output","uuid":"6f18769d-0650-42f1-9a5d-b57dbe27e50f"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":30,"input_connections":{"input":{"id":28,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":2788.683349609375,"top":298.3500061035156},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Filtered mapping of pool1"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"16002ad4-640a-4e3c-bf60-dbcb05de92ae","when":null,"workflow_outputs":[{"label":"filtered_mapping_pool1","output_name":"outputsam","uuid":"deed668a-97fc-44cb-9671-0a9d1805a5ff"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","errors":null,"id":31,"input_connections":{"input":{"id":28,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2906.0166015625,"top":157},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Mapping stats for pool1"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","tool_shed_repository":{"changeset_revision":"541082d03bef","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_group\\": null, \\"read_length\\": null, \\"remove_dups\\": false, \\"remove_overlaps\\": false, \\"sparse\\": false, \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"cc2b1c1d-5eac-4b0f-8864-44a230b47510","when":null,"workflow_outputs":[{"label":"mapping_stats_pool1","output_name":"output","uuid":"325f7653-63c3-4081-aa4c-d351184e6990"}]},"32":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":32,"input_connections":{"input":{"id":29,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":2794.7833251953125,"top":827.0833129882812},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Filtered mapping of pool2"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"229c546c-d0bb-4152-b01a-719085f0e830","when":null,"workflow_outputs":[{"label":"filtered_mapping_pool2","output_name":"outputsam","uuid":"d8e8f874-6bb1-42ae-8025-c7cfd5345385"}]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","errors":null,"id":33,"input_connections":{"input":{"id":29,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2873.7833251953125,"top":1029.5333251953125},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Mapping stats for pool2"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5","tool_shed_repository":{"changeset_revision":"541082d03bef","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_group\\": null, \\"read_length\\": null, \\"remove_dups\\": false, \\"remove_overlaps\\": false, \\"sparse\\": false, \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"1f931160-fc0a-4768-bf44-730dd4f42cda","when":null,"workflow_outputs":[{"label":"mapping_stats_pool2","output_name":"output","uuid":"b27308f7-9c7b-444d-a13a-96f358a4d2ba"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":34,"input_connections":{"results_0|software_cond|input":{"id":11,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":31,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":3134.6500244140625,"top":0},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Quality Report (sequencing pool)"},"action_type":"RenameDatasetAction","output_name":"html_report"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"e69dc815-e73b-4003-b556-37a82db778fd","when":null,"workflow_outputs":[{"label":"multiqc_pool1_stats","output_name":"stats","uuid":"d7c05c17-8a9d-4f6d-8f66-d4bafeb78152"},{"label":"pool1_quality_report","output_name":"html_report","uuid":"3bb518fd-cd6b-4f99-8dd6-f86eb5d63005"}]},"35":{"annotation":"","content_id":"__ZIP_COLLECTION__","errors":null,"id":35,"input_connections":{"input_forward":{"id":30,"output_name":"outputsam"},"input_reverse":{"id":32,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Zip collections","outputs":[{"name":"output","type":"input"}],"position":{"left":3010.183349609375,"top":585.2833251953125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__ZIP_COLLECTION__","tool_state":"{\\"input_forward\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_reverse\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b6493f6e-f845-43b9-8f8e-a26a270b9547","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":36,"input_connections":{"results_0|software_cond|input":{"id":21,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":33,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":3106.3333740234375,"top":1146.316650390625},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Quality Report (sequenced pools)"},"action_type":"RenameDatasetAction","output_name":"html_report"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"ee53d363-2fc7-4eb7-82fd-dfd3bea3d938","when":null,"workflow_outputs":[{"label":"pool2_quality_report","output_name":"html_report","uuid":"600931e4-238e-4c6f-b016-1f1347bab8d3"},{"label":"multiqc_pool2_stats","output_name":"stats","uuid":"0c6a719b-3b91-4de4-bfca-f4e1a432fa21"}]},"37":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":37,"input_connections":{"input":{"id":35,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":3239.2833251953125,"top":599.3500061035156},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"columns\\": [0, 1], \\"connectable\\": true, \\"editing\\": false, \\"is_workflow\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"collapsible_value\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"connectable\\": true, \\"error\\": null, \\"is_workflow\\": false, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"6de011ce-321e-40e7-a37a-a55a63683e49","when":null,"workflow_outputs":[]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.20+galaxy2","errors":null,"id":38,"input_connections":{"bamfiles":{"id":37,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Samtools merge","name":"bed_file"},{"description":"runtime parameter for tool Samtools merge","name":"headerbam"}],"label":null,"name":"Samtools merge","outputs":[{"name":"output","type":"bam"}],"position":{"left":3470,"top":508.5},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Merged Mappings of both pools"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.20+galaxy2","tool_shed_repository":{"changeset_revision":"cae2794d2deb","name":"samtools_merge","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bamfiles\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bed_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"headerbam\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"idpg\\": true, \\"idrg\\": false, \\"region\\": \\"\\", \\"seed\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy2","type":"tool","uuid":"8ae30344-c06e-4951-b0e9-45cb102253aa","when":null,"workflow_outputs":[{"label":"mapping_merged","output_name":"output","uuid":"88e8b60f-bb4c-4ef0-aa13-e84c7157718c"}]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0","errors":null,"id":39,"input_connections":{"input1":{"id":38,"output_name":"output"}},"inputs":[],"label":null,"name":"QualiMap BamQC","outputs":[{"name":"raw_data","type":"input"},{"name":"output_html","type":"html"}],"position":{"left":3658.35009765625,"top":175.5500030517578},"post_job_actions":{"RenameDatasetActionoutput_html":{"action_arguments":{"newname":"QualiMap BamQC report for merged mappings"},"action_type":"RenameDatasetAction","output_name":"output_html"},"TagDatasetActionoutput_html":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"output_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0","tool_shed_repository":{"changeset_revision":"30a201c9c310","name":"qualimap_bamqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"duplicate_skipping\\": [\\"0\\"], \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"per_base_coverage\\": false, \\"plot_specific\\": {\\"n_bins\\": \\"400\\", \\"paint_chromosome_limits\\": true, \\"genome_gc_distr\\": null, \\"homopolymer_size\\": \\"3\\"}, \\"stats_regions\\": {\\"region_select\\": \\"all\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.3+galaxy0","type":"tool","uuid":"e02ad651-ed91-41e0-ade2-633f63464ae9","when":null,"workflow_outputs":[{"label":"qualimap_pool1_raw","output_name":"raw_data","uuid":"f27a8689-62fb-4b4e-b179-ed1490df3767"},{"label":"qualimap_merged_html","output_name":"output_html","uuid":"51471b47-e71a-4b5f-828c-0166cc42a8a0"}]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0","errors":null,"id":40,"input_connections":{"input_bam":{"id":38,"output_name":"output"},"primer|input_bed":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ivar trim","name":"primer"}],"label":null,"name":"ivar trim","outputs":[{"name":"output_bam","type":"bam"}],"position":{"left":3704.14990234375,"top":694.8333129882812},"post_job_actions":{"RenameDatasetActionoutput_bam":{"action_arguments":{"newname":"Primer trimmed merged mapped reads"},"action_type":"RenameDatasetAction","output_name":"output_bam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0","tool_shed_repository":{"changeset_revision":"7bd020da0ce5","name":"ivar_trim","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"amplicons\\": {\\"filter_by\\": \\"yes_compute\\", \\"__current_case__\\": 0}, \\"inc_primers\\": true, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_len\\": \\"30\\", \\"min_qual\\": \\"20\\", \\"primer\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 0, \\"input_bed\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"primer_pos_wiggle\\": \\"0\\", \\"trimmed_length\\": {\\"filter\\": \\"auto\\", \\"__current_case__\\": 1}, \\"window_width\\": \\"4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.4+galaxy0","type":"tool","uuid":"735ed414-14ed-475c-a01b-c13cc5e708d9","when":null,"workflow_outputs":[{"label":"trimmed_merged_mapping","output_name":"output_bam","uuid":"9dffaa75-2a82-40da-8e0d-dd6320f5e3cd"}]},"41":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":41,"input_connections":{"input":{"id":39,"output_name":"raw_data"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":3874.11669921875,"top":5.6666717529296875},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"48597f73-b230-4bc3-8c52-fbfcd1d11392","when":null,"workflow_outputs":[]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.4+galaxy0","errors":null,"id":42,"input_connections":{"input_bam":{"id":40,"output_name":"output_bam"},"min_freq":{"id":5,"output_name":"output"},"min_indel_freq":{"id":6,"output_name":"output"},"min_qual":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"ivar consensus","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":3963.7666015625,"top":737.2333374023438},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.4+galaxy0","tool_shed_repository":{"changeset_revision":"253935875d96","name":"ivar_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"depth_action\\": \\"-n N\\", \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_depth\\": \\"50\\", \\"min_freq\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_indel_freq\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_qual\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.4+galaxy0","type":"tool","uuid":"e2ae5c77-f8f6-42a8-b44e-be7afac35cd9","when":null,"workflow_outputs":[{"label":"ivar_consensus_out","output_name":"consensus","uuid":"eb5ded94-5ee3-4b8b-a756-11b13b4d24d2"}]},"43":{"annotation":"","content_id":"__FLATTEN__","errors":null,"id":43,"input_connections":{"input":{"id":41,"output_name":"output"}},"inputs":[],"label":null,"name":"Flatten collection","outputs":[{"name":"output","type":"input"}],"position":{"left":3990.166748046875,"top":203.68333435058594},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FLATTEN__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"join_identifier\\": \\"_\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b60960bb-6a5c-4088-bc9b-ce5ae83df5bf","when":null,"workflow_outputs":[]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","errors":null,"id":44,"input_connections":{"infile":{"id":42,"output_name":"consensus"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":4200.7666015625,"top":960.9833374023438},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Per-sample consensus sequences"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"/^>/s/Consensus_(.*)_threshold_.*/\\\\\\\\1/\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"54fc1053-c37e-44ff-a198-d2a893d6b415","when":null,"workflow_outputs":[{"label":"per_sample_consensus","output_name":"output","uuid":"f26548a3-4473-4dac-999c-51b624899a58"}]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","errors":null,"id":45,"input_connections":{"results_0|software_cond|input":{"id":43,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MultiQC","name":"image_content_input"}],"label":null,"name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":4218.13330078125,"top":200.68333435058594},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Quality Report (samples)"},"action_type":"RenameDatasetAction","output_name":"html_report"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3","tool_shed_repository":{"changeset_revision":"31c42a2c02d3","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"image_content_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"qualimap\\", \\"__current_case__\\": 20, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy3","type":"tool","uuid":"798f3fc5-4165-4594-ba13-0d11e0acf95c","when":null,"workflow_outputs":[{"label":"quality_by_sample_report","output_name":"html_report","uuid":"b00e6c4d-99c5-41b8-90f7-8a51320ec02f"},{"label":"multiqc_sample_stats","output_name":"stats","uuid":"90ba0204-4d91-47f6-9039-15d50065f56c"}]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","errors":null,"id":46,"input_connections":{"inputs":{"id":44,"output_name":"output"}},"inputs":[],"label":null,"name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":4448.88330078125,"top":963.13330078125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Combined Consensus Genomes"},"action_type":"RenameDatasetAction","output_name":"out_file1"},"TagDatasetActionout_file1":{"action_arguments":{"tags":"🏁"},"action_type":"TagDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"6a39c34c-ef4c-4a94-8ac1-f244bdc87f98","when":null,"workflow_outputs":[{"label":"combined_consensus_multifasta","output_name":"out_file1","uuid":"a088a17e-ff50-4545-a64c-f150da167bd4"}]}},"tags":["pox","virology"],"uuid":"f8552617-4af2-4fa6-883f-0d605225939e","version":1},"readme":"# Pox Virus Illumina Amplicon Workflow for half-genomes sequencing data\\n\\nThis workflow generates consensus sequences from Illumina PE-sequenced ARTIC data of pox virus samples.\\n\\nIt requires that all samples have been sequenced in two halves in two separate sequencing runs, and utilizes this property to resolve the inverted terminal repeat (ITR) sequences of pox virus genomes.\\n\\nThe workflow uses BWA-MEM for mapping the reads from each half-genome sequencing run to a correspondingly masked version of the reference genome, merges the resulting two read mappings, and uses iVar for primer trimming and consensus sequence generation.\\n\\nConceptually, this workflow builds on https://github.com/iwc-workflows/sars-cov-2-pe-illumina-artic-ivar-analysis and adds the logic for the split genome mapping and merging of the results.","changelog":"# Changelog\\n\\n## [0.3] - 2025-03-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.3` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.4`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.15.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.20+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.5+galaxy0`\\n\\n## [0.2] - 2024-06-19\\n\\n### Tool version updates\\n\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0+galaxy2` was updated tp `toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.4` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.5`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.15.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/samtools_merge/samtools_merge/1.15.1+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy6` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.4.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1`\\n\\n## [0.1]\\n\\nInitial version of Pox Virus Illumina Amplicon workflow (iVar based) for half-genomes sequencing data\\n","diagrams":"# Workflow diagrams\\n\\n## Pox Virus Illumina Amplicon Workflow from half-genomes\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Reference FASTA\\"]@{ shape: doc }\\n1[\\"ℹ️ Primer Scheme\\"]@{ shape: doc }\\n2[\\"ℹ️ PE Reads Pool1\\"]@{ shape: docs }\\n3[\\"ℹ️ PE Reads Pool2\\"]@{ shape: docs }\\n4[\\"ℹ️ Minimum quality score to call base\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Allele frequency to call SNV\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Allele frequency to call indel\\"]@{ shape: lean-l }\\n7[\\"Compute sequence length\\"]@{ shape: process }\\n0 --> 7\\n8[\\"Select pool1 primers\\"]@{ shape: process }\\n1 --> 8\\n9[\\"Select pool2 primers\\"]@{ shape: process }\\n1 --> 9\\n10[\\"Extract element identifiers\\"]@{ shape: process }\\n2 --> 10\\n11[\\"fastp: Trimmed Illumina Reads Pool1\\"]@{ shape: process }\\n2 --> 11\\n12[\\"Cut\\"]@{ shape: process }\\n7 --> 12\\n13[\\"Datamash\\"]@{ shape: process }\\n8 --> 13\\n14[\\"Datamash\\"]@{ shape: process }\\n9 --> 14\\n15[\\"Split file\\"]@{ shape: process }\\n10 --> 15\\n16[\\"Sort collection\\"]@{ shape: process }\\n3 --> 16\\n10 --> 16\\n17[\\"Get end position of sequence\\"]@{ shape: process }\\n12 --> 17\\n18[\\"Get end position of Pool1\\"]@{ shape: process }\\n13 --> 18\\n19[\\"Get start position of Pool2\\"]@{ shape: process }\\n14 --> 19\\n20[\\"Parse parameter value\\"]@{ shape: process }\\n15 --> 20\\n21[\\"fastp: Trimmed Illumina Reads Pool2\\"]@{ shape: process }\\n16 --> 21\\n22[\\"Compose text parameter value\\"]@{ shape: process }\\n18 --> 22\\n17 --> 22\\n23[\\"Compose text parameter value\\"]@{ shape: process }\\n19 --> 23\\n24[\\"Compose text parameter value\\"]@{ shape: process }\\n20 --> 24\\n25[\\"Compose text parameter value\\"]@{ shape: process }\\n20 --> 25\\n26[\\"Mask Reference for Pool1\\"]@{ shape: process }\\n0 --> 26\\n22 --> 26\\n27[\\"Mask Reference for Pool2\\"]@{ shape: process }\\n0 --> 27\\n23 --> 27\\n28[\\"Map with BWA-MEM\\"]@{ shape: process }\\n11 --> 28\\n26 --> 28\\n24 --> 28\\n29[\\"Map with BWA-MEM\\"]@{ shape: process }\\n21 --> 29\\n27 --> 29\\n25 --> 29\\n30[\\"Samtools view\\"]@{ shape: process }\\n28 --> 30\\n31[\\"Samtools stats\\"]@{ shape: process }\\n28 --> 31\\n32[\\"Samtools view\\"]@{ shape: process }\\n29 --> 32\\n33[\\"Samtools stats\\"]@{ shape: process }\\n29 --> 33\\n34[\\"MultiQC\\"]@{ shape: process }\\n11 --> 34\\n31 --> 34\\n35[\\"Zip collections\\"]@{ shape: process }\\n30 --> 35\\n32 --> 35\\n36[\\"MultiQC\\"]@{ shape: process }\\n21 --> 36\\n33 --> 36\\n37[\\"Apply rules\\"]@{ shape: process }\\n35 --> 37\\n38[\\"Samtools merge\\"]@{ shape: process }\\n37 --> 38\\n39[\\"QualiMap BamQC\\"]@{ shape: process }\\n38 --> 39\\n40[\\"ivar trim\\"]@{ shape: process }\\n38 --> 40\\n1 --> 40\\n41[\\"Filter failed datasets\\"]@{ shape: process }\\n39 --> 41\\n42[\\"ivar consensus\\"]@{ shape: process }\\n40 --> 42\\n5 --> 42\\n6 --> 42\\n4 --> 42\\n43[\\"Flatten collection\\"]@{ shape: process }\\n41 --> 43\\n44[\\"Text transformation\\"]@{ shape: process }\\n42 --> 44\\n45[\\"MultiQC\\"]@{ shape: process }\\n43 --> 45\\n46[\\"Concatenate datasets\\"]@{ shape: process }\\n44 --> 46\\n```\\n","updated":"2025-03-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/pox-virus-amplicon/main","dockstore_id":21142,"categories":[],"collections":["Virology"],"doi":"10.5281/zenodo.15078918","tests":[{"doc":"Testing Pox Virus Illumina Amplicon Workflow with Lumpy Skin Disease Virus half genome sequenced data","job":{"Reference FASTA":{"class":"File","location":"https://www.ebi.ac.uk/ena/browser/api/fasta/AF325528.1?download=true","filetype":"fasta"},"Primer Scheme":{"class":"File","path":"test-data/CaPV-V1_primer_scheme.bed6","filetype":"bed"},"PE Reads Pool1":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"20L70","elements":[{"identifier":"forward","class":"File","location":"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR151/076/SRR15145276/SRR15145276_1.fastq.gz","filetype":"fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR151/076/SRR15145276/SRR15145276_2.fastq.gz","filetype":"fastqsanger.gz"}]}]},"PE Reads Pool2":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"20L70","elements":[{"identifier":"forward","class":"File","location":"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR151/075/SRR15145275/SRR15145275_1.fastq.gz","filetype":"fastqsanger.gz"},{"identifier":"reverse","class":"File","location":"ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR151/075/SRR15145275/SRR15145275_2.fastq.gz","filetype":"fastqsanger.gz"}]}]}},"outputs":{"combined_consensus_multifasta":{"file":"test-data/combined_consensus_multifasta.fasta"}}}]}],"path":"./workflows/virology/pox-virus-amplicon"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/influenza-consensus-and-subtyping.ga","testParameterFiles":["/influenza-consensus-and-subtyping-test.yml"],"authors":[{"name":"Viktoria Isabel Schwarz","orcid":"0000-0001-6897-1215"},{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow performs subtyping and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates.","comments":[{"child_steps":[0,1,2,4,5,6,7,8,9,12,13],"color":"lime","data":{"title":"Prepare m x n VAPOR jobs mapping over sequenced samples and genome segments"},"id":0,"position":[0,0],"size":[1077,849],"type":"frame"},{"child_steps":[24,28,30,32,33,34,35,36,37,38,31],"color":"none","data":{"title":"Build per-sample x per-segment consensus sequences"},"id":4,"position":[2553.1,469.3],"size":[1645.5,711.4],"type":"frame"},{"child_steps":[42,41],"color":"turquoise","data":{"title":"Compare samples segment-wise"},"id":5,"position":[4592.5,72.1],"size":[346,1319],"type":"frame"},{"child_steps":[21,22,18,20,25,27,29],"color":"orange","data":{"title":"Generate Subtyping report"},"id":2,"position":[1422.2,958.9],"size":[1010.4,824.1],"type":"frame"},{"child_steps":[14,15,16],"color":"blue","data":{"title":"Run VAPOR and extract IDs of best hits"},"id":1,"position":[1129.1000000000001,469.6],"size":[557.1,420.7],"type":"frame"},{"child_steps":[19,23,17,26],"color":"green","data":{"title":"Compile ref genome from per-segment VAPOR hits and map against it"},"id":3,"position":[1722.4,411.7],"size":[797.6,524.2],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0002-9464-6640","name":"Wolfgang Maier"},{"class":"Person","identifier":"0000-0001-6897-1215","name":"Viktoria Isabel Schwarz"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Influenza A isolate subtyping and consensus sequence generation","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"A collection of influenza A reference sequences organized by segment","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"A collection of influenza A reference sequences organized by segment","name":"References per segment collection"}],"label":"References per segment collection","name":"Input dataset collection","outputs":[],"position":{"left":68.9,"top":78},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"09e6896a-8058-4bee-8969-c04a6695dc23","when":null,"workflow_outputs":[]},"1":{"annotation":"Paired collection of forward and reverse reads of the sequenced isolate","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Paired collection of forward and reverse reads of the sequenced isolate","name":"Sequenced paired-end data"}],"label":"Sequenced paired-end data","name":"Input dataset collection","outputs":[],"position":{"left":72.76249485877617,"top":712.1781858684114},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"628ec5eb-ecc6-4ea2-856c-80bb78324fed","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":2,"input_connections":{"input_collection":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":136.76676208116686,"top":453.4062124875933},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"6f573ad0-03c4-4f16-bbca-77ac8d5cbd09","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0","errors":null,"id":3,"input_connections":{"single_paired|paired_input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"}],"position":{"left":481.4647580560187,"top":878.4588036869068},"post_job_actions":{"HideDatasetActionoutput_paired_coll":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_paired_coll"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"fastp reports"},"action_type":"RenameDatasetAction","output_name":"report_html"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"report_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0","tool_shed_repository":{"changeset_revision":"c59d48774d03","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": \\"30\\", \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": false}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": true, \\"cut_by_quality3\\": true, \\"cut_window_size\\": null, \\"cut_mean_quality\\": \\"30\\"}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.4+galaxy0","type":"tool","uuid":"9240a429-6fd4-488a-a397-52c40e68b9f3","when":null,"workflow_outputs":[{"label":"fastp reports","output_name":"report_html","uuid":"3ba6ac5d-db7a-43a3-adf1-46f79bfcc4da"}]},"4":{"annotation":"","content_id":"wc_gnu","errors":null,"id":4,"input_connections":{"input1":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Line/Word/Character count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":227.27182504593,"top":256.60694781390674},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"wc_gnu","tool_state":"{\\"include_header\\": false, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options\\": [\\"lines\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d16e7c5b-a924-4169-ac73-c3e903f28bda","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"__UNZIP_COLLECTION__","errors":null,"id":5,"input_connections":{"input":{"id":3,"output_name":"output_paired_coll"}},"inputs":[],"label":null,"name":"Unzip collection","outputs":[{"name":"forward","type":"input"},{"name":"reverse","type":"input"}],"position":{"left":395.184459729144,"top":579.1904362376869},"post_job_actions":{"HideDatasetActionforward":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"forward"},"HideDatasetActionreverse":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reverse"}},"tool_id":"__UNZIP_COLLECTION__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d9b4f336-85fa-43e0-b8ef-fbaa647379c1","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":6,"input_connections":{"input1":{"id":4,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"integer_param","type":"expression.json"}],"position":{"left":445.339540573655,"top":308.6365185896452},"post_job_actions":{"HideDatasetActioninteger_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"integer_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"integer\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"8562f219-b541-4910-a673-5baaaba10112","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"__DUPLICATE_FILE_TO_COLLECTION__","errors":null,"id":7,"input_connections":{"input":{"id":5,"output_name":"forward"}},"inputs":[],"label":null,"name":"Duplicate file to collection","outputs":[{"name":"output","type":"input"}],"position":{"left":701.4020466610151,"top":497.96176181070183},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__DUPLICATE_FILE_TO_COLLECTION__","tool_state":"{\\"element_identifier\\": \\"segment\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"number\\": \\"8\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0c27055c-2890-4c15-917c-6cd7b1e6c327","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__DUPLICATE_FILE_TO_COLLECTION__","errors":null,"id":8,"input_connections":{"input":{"id":5,"output_name":"reverse"}},"inputs":[],"label":null,"name":"Duplicate file to collection","outputs":[{"name":"output","type":"input"}],"position":{"left":699.9601177895548,"top":630.6432442656766},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__DUPLICATE_FILE_TO_COLLECTION__","tool_state":"{\\"element_identifier\\": \\"segment\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"number\\": \\"8\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"54dd807e-0c64-4a49-a417-7ce56efeb9b1","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__DUPLICATE_FILE_TO_COLLECTION__","errors":null,"id":9,"input_connections":{"input":{"id":0,"output_name":"output"},"number":{"id":6,"output_name":"integer_param"}},"inputs":[],"label":null,"name":"Duplicate file to collection","outputs":[{"name":"output","type":"input"}],"position":{"left":610.8131065769353,"top":99.39531876213022},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__DUPLICATE_FILE_TO_COLLECTION__","tool_state":"{\\"element_identifier\\": \\"seq\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"number\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"51029cc7-325e-4953-a951-ce1faa3a0efc","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":10,"input_connections":{"input_param_type|input_param":{"id":6,"output_name":"integer_param"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Run IQ-Tree (at least 3 samples)","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":2914.2167433441446,"top":88.24261883436165},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"RenameDatasetActionoutput_param_boolean":{"action_arguments":{"newname":"IQ-Tree flag"},"action_type":"RenameDatasetAction","output_name":"output_param_boolean"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"integer\\", \\"__current_case__\\": 1, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"1\\", \\"to\\": \\"false\\"}, {\\"__index__\\": 1, \\"from\\": \\"2\\", \\"to\\": \\"false\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"default\\", \\"__current_case__\\": 2, \\"default_value\\": \\"true\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"084f559e-e02d-4f2e-8b84-afd66b571ddb","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":11,"input_connections":{"input_param_type|input_param":{"id":6,"output_name":"integer_param"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Multiple sequences flag","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":3676.209740366546,"top":237.44118277330173},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"integer\\", \\"__current_case__\\": 1, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"1\\", \\"to\\": \\"false\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"default\\", \\"__current_case__\\": 2, \\"default_value\\": \\"true\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"af835b41-07e1-4123-9788-51484647f80f","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":12,"input_connections":{"input":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":846.8171456303577,"top":183.27355273774876},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [1, 0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"ef68a12d-d216-4029-a853-08c3b76cb42d","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":13,"input_connections":{"how|labels":{"id":2,"output_name":"output"},"input":{"id":12,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":785.4245477462141,"top":318.04767383103456},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": true}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3788b42c-b86a-4f05-af5d-c51b88407a07","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3","errors":null,"id":14,"input_connections":{"fasta_file":{"id":13,"output_name":"output"},"fastq_input|fastq1":{"id":7,"output_name":"output"},"fastq_input|fastq2":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool VAPOR","name":"fastq_input"},{"description":"runtime parameter for tool VAPOR","name":"fastq_input"}],"label":null,"name":"VAPOR","outputs":[{"name":"output_scores","type":"tabular"}],"position":{"left":1156.7986831355713,"top":519.1100241959895},"post_job_actions":{"RenameDatasetActionoutput_scores":{"action_arguments":{"newname":"Vapor - closest references"},"action_type":"RenameDatasetAction","output_name":"output_scores"},"TagDatasetActionoutput_scores":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"output_scores"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3","tool_shed_repository":{"changeset_revision":"244812f5bd1f","name":"vapor","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired\\", \\"__current_case__\\": 1, \\"fastq1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fastq2\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"opt\\": {\\"kmer_length\\": \\"21\\", \\"threshold\\": \\"0.1\\", \\"min_kmer_cov\\": \\"5\\", \\"min_kmer_prop\\": \\"0.1\\", \\"top_seed_frac\\": \\"1.0\\"}, \\"output_type\\": \\"scores\\", \\"return_best_n\\": \\"500\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2+galaxy3","type":"tool","uuid":"4483138c-4bef-4a76-81a4-2afe4fabbcd8","when":null,"workflow_outputs":[{"label":"Vapor - closest references","output_name":"output_scores","uuid":"827db122-2ddd-4416-8aaf-35752a84825b"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":15,"input_connections":{"infile":{"id":14,"output_name":"output_scores"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1396.7646298414118,"top":546.6159249079953},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^.+\\\\\\\\t>(.+)$\\", \\"replace_pattern\\": \\"$1\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"6ccab17e-c55c-457b-947f-82e5cf4729ef","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"Show beginning1","errors":null,"id":16,"input_connections":{"input":{"id":15,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Select first","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1422.8063658972999,"top":703.3761105306064},"post_job_actions":{},"tool_id":"Show beginning1","tool_state":"{\\"header\\": false, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineNum\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"d4cdb8cd-5892-4322-88b7-978c7f8e30fc","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.4+galaxy0","errors":null,"id":17,"input_connections":{"in_file":{"id":13,"output_name":"output"},"source|name_list":{"id":16,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool seqtk_subseq","name":"source"}],"label":null,"name":"seqtk_subseq","outputs":[{"name":"default","type":"input"}],"position":{"left":1745.3865144373408,"top":443.62606496341493},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.4+galaxy0","tool_shed_repository":{"changeset_revision":"a019807f4e67","name":"seqtk","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"l\\": \\"0\\", \\"source\\": {\\"type\\": \\"name\\", \\"__current_case__\\": 1, \\"name_list\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"t\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4+galaxy0","type":"tool","uuid":"df2f3828-807e-48cd-a30e-b274fe120292","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":18,"input_connections":{"input_list":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1532.0815683391677,"top":987.9238806639823},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"00bd9284-86b1-49e2-87ea-69e0c66dde62","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":19,"input_connections":{"input_list":{"id":17,"output_name":"default"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1980.4301950287581,"top":447.74854938603744},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Top scoring segments combined"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"4abd34a3-1e91-4101-84b2-bd49171cd43d","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"__EXTRACT_DATASET__","errors":null,"id":20,"input_connections":{"input":{"id":18,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract dataset","outputs":[{"name":"output","type":"data"}],"position":{"left":1848.8191275691797,"top":993.1032099697546},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__EXTRACT_DATASET__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"which\\": {\\"which_dataset\\": \\"first\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"9922ea40-a153-41c1-861b-ca079cf59891","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"Grep1","errors":null,"id":21,"input_connections":{"input":{"id":18,"output_name":"output"}},"inputs":[],"label":null,"name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1854.0668838127747,"top":1143.4495210124612},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"^HA.+\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"1a3a6027-721a-4ea8-bbed-525ced3fa5a6","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"Grep1","errors":null,"id":22,"input_connections":{"input":{"id":18,"output_name":"output"}},"inputs":[],"label":null,"name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1851.8655420320315,"top":1267.6601041651043},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"^NA.+\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"c2f159ef-78da-41ee-8f93-c8d7e8f6b610","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":23,"input_connections":{"infile":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2209.3806148420026,"top":457.56505821777},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Hybrid reference genomes used for mapping"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\">([^|]+).+$\\", \\"replace_pattern\\": \\">$1\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"f01ffacc-006f-4edd-ac50-dc0da78c4f02","when":null,"workflow_outputs":[{"label":"Hybrid reference genomes used for mapping","output_name":"outfile","uuid":"2cf026c7-3cca-4af6-b4c9-e46ef78de9ae"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":24,"input_connections":{"infile":{"id":20,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2798.4469946063095,"top":1056.5422223493317},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^([^|]+).+$\\", \\"replace_pattern\\": \\"$1\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b7e8dbf0-ee6d-46fc-bd65-8c6aca5883be","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"Paste1","errors":null,"id":25,"input_connections":{"input1":{"id":21,"output_name":"out_file1"},"input2":{"id":22,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Paste","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2153.0739616556525,"top":1228.4842665319438},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Paste1","tool_state":"{\\"delimiter\\": \\"T\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b09f413d-21d2-479f-b1a3-590d8a1e4e96","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","errors":null,"id":26,"input_connections":{"fastq_input|fastq_input1":{"id":3,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":23,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":2289.6694675065078,"top":686.8321592357391},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18","tool_shed_repository":{"changeset_revision":"2477830927ec","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"illumina\\", \\"__current_case__\\": 0}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"read_group_id_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"read_group_sm_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"PL\\": \\"ILLUMINA\\", \\"read_group_lb_conditional\\": {\\"do_auto_name\\": false, \\"__current_case__\\": 1, \\"LB\\": null}, \\"CN\\": null, \\"DS\\": null, \\"DT\\": null, \\"FO\\": null, \\"KS\\": null, \\"PG\\": null, \\"PI\\": null, \\"PU\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.18","type":"tool","uuid":"b7ee4713-86ab-40d3-a1df-aff33e1b27e7","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":27,"input_connections":{"input_list":{"id":25,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2154.6046014682556,"top":1404.299968663298},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_multiple\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"66a1dfe9-f0c0-445f-b2b2-30b9950393db","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2","errors":null,"id":28,"input_connections":{"input":{"id":26,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":2557.1234301153786,"top":762.7716901183999},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Final read mapping results"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy2","tool_shed_repository":{"changeset_revision":"6be888be75f9","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"20\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy2","type":"tool","uuid":"daf75af5-ae66-41c5-8aa7-5644110740ea","when":null,"workflow_outputs":[{"label":"Final read mapping results","output_name":"outputsam","uuid":"e022c539-044c-4a36-a6ff-9ff43bdaa53c"}]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":29,"input_connections":{"infile":{"id":27,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2161.453666034377,"top":1574.982352705691},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Subtyping results"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"🌟"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)\\\\\\\\tHA.+\\\\\\\\|.*(H\\\\\\\\d{1,2})N\\\\\\\\d{1,2}\\\\\\\\|[^|]+\\\\\\\\tNA.+\\\\\\\\|.*H\\\\\\\\d{1,2}(N\\\\\\\\d{1,2})\\\\\\\\|.+$\\", \\"replace_pattern\\": \\"$1\\\\\\\\t$2$3\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"18b38230-16aa-4017-bb84-4d83cdaa662e","when":null,"workflow_outputs":[{"label":"Subtyping results","output_name":"outfile","uuid":"e589d9e3-2c25-48e2-9710-2f04ca9104eb"}]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","errors":null,"id":30,"input_connections":{"input1":{"id":28,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"QualiMap BamQC","outputs":[{"name":"raw_data","type":"input"},{"name":"output_html","type":"html"}],"position":{"left":2806.5530105890093,"top":649.2678440521755},"post_job_actions":{"HideDatasetActionraw_data":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"raw_data"},"RenameDatasetActionoutput_html":{"action_arguments":{"newname":"QC reports for mapping results"},"action_type":"RenameDatasetAction","output_name":"output_html"},"TagDatasetActionoutput_html":{"action_arguments":{"tags":"🔍"},"action_type":"TagDatasetAction","output_name":"output_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3","tool_shed_repository":{"changeset_revision":"19ece8afbaab","name":"qualimap_bamqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"duplicate_skipping\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"per_base_coverage\\": false, \\"plot_specific\\": {\\"n_bins\\": \\"40\\", \\"paint_chromosome_limits\\": true, \\"genome_gc_distr\\": null, \\"homopolymer_size\\": \\"3\\"}, \\"stats_regions\\": {\\"region_select\\": \\"all\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.2d+galaxy3","type":"tool","uuid":"885b9257-1d9f-40b0-bf2b-2d1327b16776","when":null,"workflow_outputs":[{"label":"QC reports for mapping results","output_name":"output_html","uuid":"59b76d32-7890-48f3-9885-c7c7c11d942f"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.2+galaxy2","errors":null,"id":31,"input_connections":{"input_bam":{"id":28,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Split BAM by reference","outputs":[{"name":"output_bams","type":"input"}],"position":{"left":2804.149854475478,"top":883.1487246046947},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.2+galaxy2","tool_shed_repository":{"changeset_revision":"9b8feb118f9e","name":"bamtools_split_ref","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"refs\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.2+galaxy2","type":"tool","uuid":"2a534e6a-0542-4470-9775-80591966b922","when":null,"workflow_outputs":[]},"32":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":32,"input_connections":{"input":{"id":31,"output_name":"output_bams"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":3043.9512700139053,"top":873.3196571359551},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [1, 0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"3c5be312-d3a6-46de-a9f0-7d31741aff3a","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":33,"input_connections":{"how|labels":{"id":24,"output_name":"outfile"},"input":{"id":32,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":null,"name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":3265.919754268056,"top":983.9937307995837},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": true}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"0bcb9db0-4160-4856-b769-d40feb29955c","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":34,"input_connections":{"input":{"id":33,"output_name":"output"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":3378.69433149217,"top":727.5370045095681},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [1, 0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"666137d6-7a9d-4b2f-92da-476da9b2da8f","when":null,"workflow_outputs":[]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.3+galaxy0","errors":null,"id":35,"input_connections":{"input_bam":{"id":34,"output_name":"output"}},"inputs":[],"label":null,"name":"ivar consensus","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":3373.00829259856,"top":498.47642565852516},"post_job_actions":{"RenameDatasetActionconsensus":{"action_arguments":{"newname":"Per-sample consensus sequences"},"action_type":"RenameDatasetAction","output_name":"consensus"},"TagDatasetActionconsensus":{"action_arguments":{"tags":"🌟"},"action_type":"TagDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.3+galaxy0","tool_shed_repository":{"changeset_revision":"3a72fede57ce","name":"ivar_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"depth_action\\": \\"-n N\\", \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_depth\\": \\"10\\", \\"min_freq\\": \\"0.7\\", \\"min_indel_freq\\": \\"0.8\\", \\"min_qual\\": \\"20\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.4.3+galaxy0","type":"tool","uuid":"b1295c1b-530d-4ec0-86ab-83c2ed370b83","when":null,"workflow_outputs":[{"label":"Per-sample consensus sequences","output_name":"consensus","uuid":"772f61e0-4af2-4a6f-b0c6-9359470d7b5c"}]},"36":{"annotation":"","content_id":"__APPLY_RULES__","errors":null,"id":36,"input_connections":{"input":{"id":35,"output_name":"consensus"}},"inputs":[],"label":null,"name":"Apply rules","outputs":[{"name":"output","type":"input"}],"position":{"left":3658.1266044610875,"top":740.8707182575123},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Per-segment consensus sequences"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__APPLY_RULES__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rules\\": {\\"mapping\\": [{\\"columns\\": [1, 0], \\"editing\\": false, \\"type\\": \\"list_identifiers\\"}], \\"rules\\": [{\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier0\\", \\"warn\\": null}, {\\"error\\": null, \\"type\\": \\"add_column_metadata\\", \\"value\\": \\"identifier1\\", \\"warn\\": null}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"773f9899-f5c9-4804-ac43-8b98a14669c5","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":37,"input_connections":{"input_list":{"id":36,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":3941.7086336866,"top":565.5606948208823},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"txt"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"d16db5b4-85ce-4e3c-8a94-dea5790f8d2f","when":null,"workflow_outputs":[]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":38,"input_connections":{"infile":{"id":37,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3949.894759441091,"top":767.0987708391559},"post_job_actions":{"ChangeDatatypeActionoutfile":{"action_arguments":{"newtype":"fasta"},"action_type":"ChangeDatatypeAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Per-segment consensus sequences with samples combined"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"🌟"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)\\\\\\\\t>Consensus_(.+)_threshold.+\\", \\"replace_pattern\\": \\">$1|$2\\", \\"is_regex\\": true, \\"global\\": false, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"cd5cfbd9-309a-42e5-957f-83258ef2a022","when":null,"workflow_outputs":[{"label":"Per-segment consensus sequences with samples combined","output_name":"outfile","uuid":"b7a61ef2-ddfa-402f-be33-a6afb461361f"}]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.520+galaxy0","errors":null,"id":39,"input_connections":{"input|batches_0|inputs":{"id":38,"output_name":"outfile"},"when":{"id":11,"output_name":"output_param_boolean"}},"inputs":[],"label":null,"name":"MAFFT","outputs":[{"name":"outputAlignment","type":"fasta"}],"position":{"left":4146.950961736724,"top":263.3305934704922},"post_job_actions":{"RenameDatasetActionoutputAlignment":{"action_arguments":{"newname":"Multiple sequence alignments per segment"},"action_type":"RenameDatasetAction","output_name":"outputAlignment"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.520+galaxy0","tool_shed_repository":{"changeset_revision":"bf28a8cff401","name":"mafft","owner":"rnateam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"datatype_selection\\": {\\"datatype\\": \\"--nuc\\", \\"__current_case__\\": 1, \\"scoring_matrix\\": {\\"type\\": \\"--kimura\\", \\"__current_case__\\": 0, \\"coefficient\\": \\"200\\"}, \\"fmodel\\": false, \\"gap_costs\\": {\\"use_defaults\\": \\"yes\\", \\"__current_case__\\": 0}}, \\"flavour\\": {\\"type\\": \\"mafft-fftns\\", \\"__current_case__\\": 2}, \\"input\\": {\\"mapping\\": \\"implicit\\", \\"__current_case__\\": 0, \\"batches\\": [{\\"__index__\\": 0, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"outputFormat\\": \\"\\", \\"reorder\\": false, \\"treeout\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"7.520+galaxy0","type":"tool","uuid":"86f16717-083c-4cee-ac15-3620bf04c5c8","when":"$(inputs.when)","workflow_outputs":[{"label":"Multiple sequence alignments per segment","output_name":"outputAlignment","uuid":"8e954c40-3cc3-4dca-ae9d-b6f71b96d524"}]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":40,"input_connections":{"style_cond|type_cond|default_value":{"id":38,"output_name":"outfile"},"style_cond|type_cond|pick_from_0|value":{"id":39,"output_name":"outputAlignment"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":4355.300937322661,"top":453.4472621716641},"post_job_actions":{"HideDatasetActiondata_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"e29dfb1e-10d1-4d6c-b57a-1a1354aed811","when":null,"workflow_outputs":[]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.3.3+galaxy0","errors":null,"id":41,"input_connections":{"general_options|s":{"id":40,"output_name":"data_param"},"when":{"id":10,"output_name":"output_param_boolean"}},"inputs":[{"description":"runtime parameter for tool IQ-TREE","name":"general_options"},{"description":"runtime parameter for tool IQ-TREE","name":"general_options"},{"description":"runtime parameter for tool IQ-TREE","name":"likelihood_mapping"}],"label":null,"name":"IQ-TREE","outputs":[{"name":"bionj","type":"nhx"},{"name":"treefile","type":"nhx"},{"name":"mldist","type":"mldist"},{"name":"splits","type":"nex"},{"name":"iqtree","type":"iqtree"}],"position":{"left":4658.398071254695,"top":154.6799234732899},"post_job_actions":{"HideDatasetActionbionj":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bionj"},"HideDatasetActionsplits":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"splits"},"RenameDatasetActioniqtree":{"action_arguments":{"newname":"IQ-Tree per-segment report"},"action_type":"RenameDatasetAction","output_name":"iqtree"},"RenameDatasetActionmldist":{"action_arguments":{"newname":"IQ-Tree per-segment ML distance matrix"},"action_type":"RenameDatasetAction","output_name":"mldist"},"RenameDatasetActiontreefile":{"action_arguments":{"newname":"IQ-Tree per-segment ML tree"},"action_type":"RenameDatasetAction","output_name":"treefile"},"TagDatasetActioniqtree":{"action_arguments":{"tags":"🌟"},"action_type":"TagDatasetAction","output_name":"iqtree"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.3.3+galaxy0","tool_shed_repository":{"changeset_revision":"70acec670c30","name":"iqtree","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"bootstrap_parameters\\": {\\"ultrafast_bootstrap\\": {\\"ufboot\\": null, \\"ufjack\\": null, \\"jack_prop\\": null, \\"boot_trees\\": false, \\"wbtl\\": false, \\"nmax\\": \\"1000\\", \\"bcor\\": \\"0.99\\", \\"nstep\\": \\"100\\", \\"beps\\": \\"0.5\\", \\"sampling\\": null, \\"bnni\\": false}, \\"nonparametric_bootstrap\\": {\\"boot\\": null, \\"jack\\": null, \\"jack_prop\\": null, \\"bcon\\": null, \\"bonly\\": null, \\"tbe\\": false}}, \\"general_options\\": {\\"s\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"o\\": null, \\"short_alignments\\": false, \\"seqtype\\": \\"DNA\\", \\"t\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"seed\\": null, \\"keep_ident\\": false, \\"epsilon\\": null, \\"safe\\": false}, \\"likelihood_mapping\\": {\\"lmap\\": null, \\"lmclust\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"quartetlh\\": false}, \\"miscellaneous_options\\": {\\"fconst\\": null}, \\"modelling_parameters\\": {\\"automatic_model\\": {\\"cond_model\\": {\\"opt_custommodel\\": \\"true\\", \\"__current_case__\\": 0, \\"m\\": null}, \\"merge_strategy\\": {\\"merge\\": \\"none\\", \\"__current_case__\\": 0}, \\"mset\\": null, \\"msub\\": \\"nuclear\\", \\"mfreq\\": null, \\"mrate\\": null, \\"cmin\\": \\"2\\", \\"cmax\\": \\"10\\", \\"merit\\": \\"AIC\\", \\"mtree\\": false, \\"madd\\": null, \\"mdef\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"modelomatic\\": false}, \\"specifying_substitution\\": {\\"mix_opt\\": false}, \\"rate_heterogeneity\\": {\\"alpha_min\\": null, \\"gamma_median\\": false, \\"i\\": null, \\"opt_gamma_inv\\": false, \\"rate\\": false, \\"mlrate\\": false}, \\"partition_model\\": {\\"model_selection\\": {\\"model\\": \\"none\\", \\"__current_case__\\": 0}}, \\"site_specific_frequency\\": {\\"tree_freq\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"site_freq\\": null, \\"freq_max\\": false}}, \\"time_tree\\": {\\"date_source\\": {\\"select_source\\": \\"none\\", \\"__current_case__\\": 0}, \\"date_tip\\": null, \\"date_root\\": null, \\"date_ci\\": null, \\"clock_sd\\": null, \\"date_no_outgroup\\": false, \\"date_outlier\\": null, \\"date_options\\": null}, \\"tree_parameters\\": {\\"tree_search\\": {\\"n\\": null, \\"ninit\\": \\"100\\", \\"ntop\\": \\"20\\", \\"nbest\\": \\"5\\", \\"nstop\\": \\"100\\", \\"radius\\": \\"6\\", \\"perturb\\": \\"0.5\\", \\"allnni\\": false, \\"tree_fix\\": false, \\"g\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"fast\\": false, \\"polytomy\\": false, \\"treels\\": false, \\"show_lh\\": false, \\"terrace\\": false}, \\"single_branch\\": {\\"alrt\\": null, \\"abayes\\": false, \\"lbp\\": null}, \\"tree_topology\\": {\\"trees\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"test\\": null, \\"test_weight\\": false, \\"test_au\\": false}, \\"constructing_consensus\\": {\\"contree\\": false, \\"connet\\": false, \\"sup_min\\": \\"0.0\\", \\"burnin\\": null, \\"support\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"suptag\\": null}, \\"computing_robinson_foulds\\": {\\"tree_dist_all\\": false, \\"tree_dist\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"tree_dist2\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"generating_random\\": {\\"r\\": null, \\"rand\\": null, \\"branch_min\\": null, \\"branch_mean\\": null, \\"branch_max\\": null}, \\"ancestral_state\\": {\\"ancestral\\": false, \\"asr_min\\": null}, \\"symmetry_test\\": {\\"symtest\\": false, \\"symtest_remove_bad\\": false, \\"symtest_remove_good\\": false, \\"symtest_type\\": null, \\"symtest_pval\\": null, \\"symtest_keep_zero\\": false}, \\"concordance_factor\\": {\\"gcf\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"df_tree\\": false, \\"scf\\": null, \\"cf_verbose\\": false, \\"cf_quartet\\": false}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.3.3+galaxy0","type":"tool","uuid":"a4745843-80b5-4017-94f3-4ed895e3d08c","when":"$(inputs.when)","workflow_outputs":[{"label":"IQ-Tree per-segment ML tree","output_name":"treefile","uuid":"08ab78b7-2612-4315-ab3a-fb2588c82172"},{"label":"IQ-Tree per-segment ML distance matrix","output_name":"mldist","uuid":"c5d752a6-d45f-46ad-a7fd-600b28ad5454"},{"label":"IQ-Tree per-segment report","output_name":"iqtree","uuid":"b8fc5bb9-c8dc-4db1-be8c-69de315bff49"}]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snipit/snipit/1.2+galaxy0","errors":null,"id":42,"input_connections":{"alignment":{"id":40,"output_name":"data_param"},"when":{"id":11,"output_name":"output_param_boolean"}},"inputs":[],"label":null,"name":"snipit","outputs":[{"name":"snp_plot","type":"png"}],"position":{"left":4656.336680310707,"top":1184.260547419398},"post_job_actions":{"RenameDatasetActionsnp_plot":{"action_arguments":{"newname":"Snipit plots per segment"},"action_type":"RenameDatasetAction","output_name":"snp_plot"},"TagDatasetActionsnp_plot":{"action_arguments":{"tags":"🌟"},"action_type":"TagDatasetAction","output_name":"snp_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snipit/snipit/1.2+galaxy0","tool_shed_repository":{"changeset_revision":"92b9fd5f1f9f","name":"snipit","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"dims\\": {\\"width\\": \\"0.0\\", \\"height\\": \\"0.0\\", \\"size_option\\": \\"\\"}, \\"plot\\": {\\"format\\": \\"png\\", \\"__current_case__\\": 2, \\"transparent_background\\": false}, \\"pos_restrict\\": {\\"include_positions\\": [], \\"exclude_positions\\": [], \\"exclude_ambig_pos\\": false}, \\"ref\\": {\\"select\\": \\"first\\", \\"__current_case__\\": 0}, \\"show_indels\\": false, \\"style\\": {\\"labels\\": {\\"choose\\": \\"\\", \\"__current_case__\\": 0}, \\"colouring\\": {\\"palette\\": \\"classic\\", \\"__current_case__\\": 1}, \\"sort\\": {\\"order\\": \\"--sort-by-mutation-number\\", \\"__current_case__\\": 2, \\"high_to_low\\": false}, \\"flip_vertical\\": false}, \\"write_snps\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2+galaxy0","type":"tool","uuid":"bb67a577-8d17-48bc-a966-d678b0d82bc2","when":"$(inputs.when)","workflow_outputs":[{"label":"Snipit plots per segment","output_name":"snp_plot","uuid":"ff202345-aa4f-426f-aaf1-46208d738fcd"}]}},"tags":["AIV","virology"],"uuid":"26114133-2c7d-4f90-8f62-0ed5837b5948","version":1},"readme":"# Subtyping and consensus sequence generation for batches of Influenza A isolates\\n\\nThis workflow performs subtyping (with respect to the HA and NA genes) and consensus sequence generation for batches of Illumina PE sequenced Influenza A isolates. It overcomes Influenza genome variability by compiling the best possible reference for read mapping from a collection of reference sequences for each viral genome segment.\\n\\nIt uses:\\n- **VAPOR** for identifying, from a collection of reference sequences of each of the eight viral genome segments, the individual segment sequences that match the sequencing data for each sample most closely\\n- **fastp** for sequenced reads pre-processing,\\n- **bwa mem** for mapping the preprocessed reads to a reference compiled from the best-matching segment sequences identified by *VAPOR*\\n- **ivar consensus** for generating consensus sequences from the mapped reads data\\n- **MAFFT** for generating multiple sequence alignments from the consensus sequences of the samples in a batch\\n- **Snipit** and **IQ-Tree** for visualizing differences and analyzing relationships between sequences from a batch\\n\\nThe workflow provides a subtyping report, the consensus sequences arranged by gene segment or by sample, quality control at the level of sequenced reads and of mapping results, and batch-level visualization and phylogenetic insight.\\n\\n## Input datasets\\n\\n- **Sequenced paired-end data**: a list of pairs of sequencing datasets, one fw-/rv-reads pair per sequenced isolate\\n- **References per segment collection**: this must be provided as a list of 8 FASTA datasets, one for each Influenza genome segment. Each of these datasets should contain all reference sequences of the corresponding segment that you wish to use in the analysis.\\n\\n  **NOTE 1**: For subtyping to work correctly *all* FASTA sequence title lines in the input need to follow the scheme (note the use of both `|` and `/`):\\n\\n  `>[segment]|[influenza type]/[host]/[region]/[internal reference number]/[collection year]|subtype|[accession number]`\\n\\n  where `[host]` can be omitted for samples obtained from human patients.\\n\\n  Examples:\\n\\n  - `>NS|A/California/07/2009|H1N1|NC_026432.1` (for the NS segment sequence of a H1N1 sample obtained from a human patient in California in 2009)\\n  - `>NA|A/chicken/Zimbabwe/AI4935/2017|H5N8|MF973227.1` (for the NA segment sequence of a H5N8 sample obtained from chicken in Zimbabwe in 2017)\\n\\n  **NOTE 2**: Sequences containing the ambiguity symbol N will be ignored by the workflow entirely.\\n\\n  **NOTE 3**: A well-formatted collection of reference sequences suitable for most analysis needs is linked from the \\"Data resources\\" section of the page: https://virology.usegalaxy.eu/published/page?id=a04ab8d6ecb698fa.\\n\\n## Outputs:\\n\\n- **Vapor - closest references**: a nested collection listing, for each sample and segment, the up to 500 best matching reference sequences according to VAPOR; inspect this collection if you are curious how many good matches to your data there were in the reference collection; useful for debugging, for example, if generated consensus sequences contain unresolved bases (Ns)\\n- **Subtyping results**: a table listing one sample and its detected subtype (with respect to the HA and NA segments) per row\\n- **Hybrid reference genomes used for mapping**: collection of compiled reference genomes (in FASTA format, one per sample) that served as input for bwa-mem; each reference genome consists of the genome segments from the reference collections that best matched the corresponding sample\'s sequencing data\\n- **fastp reports**: QC and read trimming and filtering results from fastp\\n- **Final read mapping results**: bwa-mem mapping results in BAM format post-processed with samtools view\\n- **QC reports for mapping results**: a collection of reports of QC metrics for the \\"Final read mapping results\\" generated with Qualimap\\n- **Per-sample consensus sequences**: a nested collection of consensus sequences organized first by sample, then by segment\\n- **Per-segment consensus sequences with samples combined**: a collection of the same consensus sequences organized by segment; each collection element is a multi-sequence FASTA dataset with the segment-specific sequences of all samples\\n- **Multiple sequence alignments per segment**: a collection of multiple sequence alignments generated with MAFFT from each of the \\"Per-segment consensus sequences with samples combined\\" above; generated only if you provided input sequencing data of at least two samples\\n- **Snipit plots per segment**: a collection of SNP plots across samples generated with Snipit; one plot per segment; generated  only if you provided input sequencing data of at least two samples; the first input sample will be used as the reference in the plot\\n- **IQ-Tree per-segment ML tree**, **IQ-Tree per-segment ML distance matrix** and **IQ-Tree per-segment report**: collections of IQ-Tree ML trees, distance matrices and reports; each collection has one element per segment; generated  only if you provided input sequencing data of at least three samples\\n\\n## Related training material\\n\\nhttps://gxy.io/GTN:T00308 guides you through a simplified, manual analysis that still includes the key steps of this workflow.\\n\\n","changelog":"# Changelog\\n\\n## [0.1] - 2025-01-09\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Influenza A isolate subtyping and consensus sequence generation\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ References per segment collection\\"]@{ shape: docs }\\n1[\\"ℹ️ Sequenced paired-end data\\"]@{ shape: docs }\\n2[\\"Extract element identifiers\\"]@{ shape: process }\\n1 --> 2\\n3[\\"fastp\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Line/Word/Character count\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Unzip collection\\"]@{ shape: process }\\n3 --> 5\\n6[\\"Parse parameter value\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Duplicate file to collection\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Duplicate file to collection\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Duplicate file to collection\\"]@{ shape: process }\\n0 --> 9\\n6 --> 9\\n10[\\"Run IQ-Tree (at least 3 samples)\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Multiple sequences flag\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Apply rules\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Relabel identifiers\\"]@{ shape: process }\\n2 --> 13\\n12 --> 13\\n14[\\"VAPOR\\"]@{ shape: process }\\n13 --> 14\\n7 --> 14\\n8 --> 14\\n15[\\"Replace\\"]@{ shape: process }\\n14 --> 15\\n16[\\"Select first\\"]@{ shape: process }\\n15 --> 16\\n17[\\"seqtk_subseq\\"]@{ shape: process }\\n13 --> 17\\n16 --> 17\\n18[\\"Collapse Collection\\"]@{ shape: process }\\n16 --> 18\\n19[\\"Collapse Collection\\"]@{ shape: process }\\n17 --> 19\\n20[\\"Extract dataset\\"]@{ shape: process }\\n18 --> 20\\n21[\\"Select\\"]@{ shape: process }\\n18 --> 21\\n22[\\"Select\\"]@{ shape: process }\\n18 --> 22\\n23[\\"Replace\\"]@{ shape: process }\\n19 --> 23\\n24[\\"Replace\\"]@{ shape: process }\\n20 --> 24\\n25[\\"Paste\\"]@{ shape: process }\\n21 --> 25\\n22 --> 25\\n26[\\"Map with BWA-MEM\\"]@{ shape: process }\\n3 --> 26\\n23 --> 26\\n27[\\"Collapse Collection\\"]@{ shape: process }\\n25 --> 27\\n28[\\"Samtools view\\"]@{ shape: process }\\n26 --> 28\\n29[\\"Replace\\"]@{ shape: process }\\n27 --> 29\\n30[\\"QualiMap BamQC\\"]@{ shape: process }\\n28 --> 30\\n31[\\"Split BAM by reference\\"]@{ shape: process }\\n28 --> 31\\n32[\\"Apply rules\\"]@{ shape: process }\\n31 --> 32\\n33[\\"Relabel identifiers\\"]@{ shape: process }\\n24 --> 33\\n32 --> 33\\n34[\\"Apply rules\\"]@{ shape: process }\\n33 --> 34\\n35[\\"ivar consensus\\"]@{ shape: process }\\n34 --> 35\\n36[\\"Apply rules\\"]@{ shape: process }\\n35 --> 36\\n37[\\"Collapse Collection\\"]@{ shape: process }\\n36 --> 37\\n38[\\"Replace\\"]@{ shape: process }\\n37 --> 38\\n39[\\"MAFFT\\"]@{ shape: process }\\n38 --> 39\\n11 --> 39\\n40[\\"Pick parameter value\\"]@{ shape: process }\\n38 --> 40\\n39 --> 40\\n41[\\"IQ-TREE\\"]@{ shape: process }\\n40 --> 41\\n10 --> 41\\n42[\\"snipit\\"]@{ shape: process }\\n40 --> 42\\n11 --> 42\\n```\\n","updated":"2025-01-09T00:00:00","trsID":"#workflow/github.com/iwc-workflows/influenza-isolates-consensus-and-subtyping/main","dockstore_id":30392,"categories":[],"collections":["Virology"],"doi":"10.5281/zenodo.15078914"}],"path":"./workflows/virology/influenza-isolates-consensus-and-subtyping"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Generic-variation-analysis-on-WGS-PE-data.ga","testParameterFiles":["/Generic-variation-analysis-on-WGS-PE-data-tests.yml"],"authors":[{"name":"Anton Nekrutenko","orcid":"0000-0002-5987-8032"}],"definition":{"a_galaxy_workflow":"true","annotation":"Workflow for variant analysis against  a reference genome in GenBank format","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-5987-8032","name":"Anton Nekrutenko"}],"format-version":"0.1","release":"0.1.1","license":"MIT","name":"Generic variation analysis on WGS PE data","steps":{"0":{"annotation":"Illumina reads with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Illumina reads with fastqsanger encoding","name":"Paired Collection"}],"label":"Paired Collection","name":"Input dataset collection","outputs":[],"position":{"left":58.531223149503546,"top":300.6406189710691},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"2352cd98-29f2-456b-a807-ede1227f6089","workflow_outputs":[]},"1":{"annotation":"GenBank with annotations for the genome of interest","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"GenBank with annotations for the genome of interest","name":"GenBank genome"}],"label":"GenBank genome","name":"Input dataset","outputs":[],"position":{"left":0,"top":631.656273804954},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"genbank\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"5e6e8cac-8a75-42e8-9932-6c63bcebb861","workflow_outputs":[]},"2":{"annotation":"Should describe your reference genome, e.g. mpxv for Monkeypox virus.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Should describe your reference genome, e.g. mpxv for Monkeypox virus.","name":"Name for genome database"}],"label":"Name for genome database","name":"Input parameter","outputs":[],"position":{"left":20.98434665780934,"top":745.593744965532},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"7e5e0a5f-4386-44cb-8941-f04ec7d8b725","workflow_outputs":[{"label":null,"output_name":"output","uuid":"a543d03c-6cc6-4d50-8dee-ec7a839b16b9"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","errors":null,"id":3,"input_connections":{"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":325.0155944202679,"top":95.51562991016024},"post_job_actions":{"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","tool_shed_repository":{"changeset_revision":"65b93b623c77","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": \\"false\\", \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": \\"false\\", \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": \\"false\\", \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": \\"true\\", \\"report_json\\": \\"true\\"}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": \\"false\\", \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": \\"false\\", \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": \\"false\\", \\"cut_by_quality3\\": \\"false\\", \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": \\"false\\"}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": \\"false\\", \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.2+galaxy0","type":"tool","uuid":"6410c94d-827d-4d1b-b15d-be13df2db0ac","workflow_outputs":[{"label":"fastp_pe","output_name":"output_paired_coll","uuid":"c49d5f48-ab0e-4f59-8b2f-27b3a8f65609"},{"label":"fastp_html_report","output_name":"report_html","uuid":"c77d6708-0d13-4f9f-aed8-b3e30de5bafa"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_build_gb/4.3+T.galaxy4","errors":null,"id":4,"input_connections":{"genome_version":{"id":2,"output_name":"output"},"input_type|input_gbk":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool SnpEff build:","name":"input_type"}],"label":null,"name":"SnpEff build:","outputs":[{"name":"snpeff_output","type":"snpeffdb"},{"name":"output_fasta","type":"fasta"}],"position":{"left":289.04683721041243,"top":692.6562509323089},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_build_gb/4.3+T.galaxy4","tool_shed_repository":{"changeset_revision":"74aebe30fb52","name":"snpeff","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"codon_table\\": \\"Standard\\", \\"genome_version\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_type\\": {\\"input_type_selector\\": \\"gb\\", \\"__current_case__\\": 0, \\"input_gbk\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"fasta\\": \\"yes\\", \\"remove_version\\": \\"true\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+T.galaxy4","type":"tool","uuid":"1406c6b2-51c9-47f7-9f9d-190c0e9ad515","workflow_outputs":[{"label":"SnpEff4.3 database","output_name":"snpeff_output","uuid":"c43258f1-ae40-45d9-a559-a7e20f2ac107"},{"label":"Fasta sequences for genbank file","output_name":"output_fasta","uuid":"6a81041a-b077-4211-8d48-aafbe78c0713"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","errors":null,"id":5,"input_connections":{"fastq_input|fastq_input1":{"id":3,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":4,"output_name":"output_fasta"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":590.5546666162805,"top":327.0468702141476},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","tool_shed_repository":{"changeset_revision":"64f11cf59c6e","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"full\\", \\"__current_case__\\": 4, \\"algorithmic_options\\": {\\"algorithmic_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"scoring_options\\": {\\"scoring_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"io_options\\": {\\"io_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"five\\": \\"false\\", \\"q\\": \\"true\\", \\"T\\": \\"30\\", \\"h\\": \\"5\\", \\"a\\": \\"false\\", \\"C\\": \\"false\\", \\"V\\": \\"false\\", \\"Y\\": \\"true\\", \\"M\\": \\"false\\"}}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"do_not_set\\", \\"__current_case__\\": 3}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.17.2","type":"tool","uuid":"a6bd6dbf-d3e7-439f-b826-16ec60bfb33f","workflow_outputs":[{"label":"bwa_mem_alignments","output_name":"bam_output","uuid":"4fa2a1d1-3df1-43a0-9df8-687f55df83a4"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy1","errors":null,"id":6,"input_connections":{"input":{"id":5,"output_name":"bam_output"}},"inputs":[{"description":"runtime parameter for tool Samtools view","name":"input"}],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":823.9531041162805,"top":207.86718147106907},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Mapped read pairs (filtered bwa-mem result)"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy1","tool_shed_repository":{"changeset_revision":"c370440f901e","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"0\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": \\"false\\", \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": \\"false\\"}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.13+galaxy1","type":"tool","uuid":"6186b9e0-5841-4572-bec3-3f5c11a1c1cf","workflow_outputs":[{"label":"filtered_alignment","output_name":"outputsam","uuid":"68a53be7-7332-4c7f-9615-32770064ffb0"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2","errors":null,"id":7,"input_connections":{"inputFile":{"id":6,"output_name":"outputsam"}},"inputs":[{"description":"runtime parameter for tool MarkDuplicates","name":"inputFile"}],"label":null,"name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":1043.617201919711,"top":322.0156341366272},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2","tool_shed_repository":{"changeset_revision":"a1f0b3f4b781","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": \\"true\\", \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": \\"true\\", \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.18.2.2","type":"tool","uuid":"cd152e82-b33b-4cca-9126-a4bd63138593","workflow_outputs":[{"label":"markduplicates_stats","output_name":"metrics_file","uuid":"2a24ab1a-6f0f-4896-b436-d6eeeb7ca22c"},{"label":"markduplicates_reads","output_name":"outFile","uuid":"9fa98a79-4a12-41df-82c7-6fc4c5a8a1fa"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","errors":null,"id":8,"input_connections":{"input":{"id":6,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1099.2499388405358,"top":137.63280895722616},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","tool_shed_repository":{"changeset_revision":"145f6d74ff5e","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_length\\": null, \\"remove_dups\\": \\"false\\", \\"remove_overlaps\\": \\"false\\", \\"sparse\\": \\"false\\", \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.2+galaxy2","type":"tool","uuid":"0ba8c049-f7f1-487f-aa23-3bc6cb2d23aa","workflow_outputs":[{"label":"mapped_reads_stats","output_name":"output","uuid":"acc7e52b-636c-4f28-a49a-0052f23c8737"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","errors":null,"id":9,"input_connections":{"reads":{"id":7,"output_name":"outFile"},"reference_source|ref":{"id":4,"output_name":"output_fasta"}},"inputs":[{"description":"runtime parameter for tool Realign reads","name":"reads"},{"description":"runtime parameter for tool Realign reads","name":"reference_source"}],"label":null,"name":"Realign reads","outputs":[{"name":"realigned","type":"bam"}],"position":{"left":1181.2499875692147,"top":531.7812529212346},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"af7e416d8176","name":"lofreq_viterbi","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"keepflags\\": \\"false\\", \\"bq2_handling\\": {\\"replace_bq2\\": \\"keep\\", \\"__current_case__\\": 0, \\"defqual\\": \\"2\\"}}, \\"reads\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"f22634d5-0472-4b37-9f1f-55693be5109d","workflow_outputs":[{"label":"realigned_deduplicated_filtered_mapped_reads","output_name":"realigned","uuid":"436348c9-ad6f-46aa-9cd4-7744563ca643"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","errors":null,"id":10,"input_connections":{"results_0|software_cond|input":{"id":3,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":8,"output_name":"output"},"results_2|software_cond|output_0|input":{"id":7,"output_name":"metrics_file"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"left":1536.7109061744209,"top":0},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","tool_shed_repository":{"changeset_revision":"9a913cdee30e","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": \\"false\\", \\"flat\\": \\"false\\", \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"saveLog\\": \\"false\\", \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy0","type":"tool","uuid":"1eab5364-f845-4668-9bb0-c820b2c61ab1","workflow_outputs":[{"label":"preprocessing_and_mapping_reports","output_name":"html_report","uuid":"710f4e3f-c7a5-465a-96ec-7eb558a44b35"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0","errors":null,"id":11,"input_connections":{"reads":{"id":9,"output_name":"realigned"},"strategy|reference_source|ref":{"id":4,"output_name":"output_fasta"}},"inputs":[],"label":null,"name":"Insert indel qualities","outputs":[{"name":"output","type":"bam"}],"position":{"left":1296.3124448073129,"top":712.8437220928869},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Fully processed reads for variant calling (deduplicated, realigned reads with added indelquals)"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"354b534eeab7","name":"lofreq_indelqual","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strategy\\": {\\"selector\\": \\"dindel\\", \\"__current_case__\\": 1, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"8568254e-a9a6-4d67-a4a9-97947d7bc10e","workflow_outputs":[{"label":"realigned_deduplicated_filtered_mapped_reads_with_indel_quals","output_name":"output","uuid":"71283f3d-16ce-43a6-af6a-89e0d6d0aa4f"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1","errors":null,"id":12,"input_connections":{"reads":{"id":11,"output_name":"output"},"reference_source|ref":{"id":4,"output_name":"output_fasta"}},"inputs":[],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"left":1423.3749607187183,"top":908.2577816094984},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1","tool_shed_repository":{"changeset_revision":"e1461b5c52a0","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"5\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": \\"false\\"}, \\"bc_quals\\": {\\"min_bq\\": \\"30\\", \\"min_alt_bq\\": \\"30\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": \\"true\\"}}}, \\"map_quals\\": {\\"min_mq\\": \\"0\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_custom\\", \\"__current_case__\\": 3, \\"sig\\": \\"0.0005\\", \\"bonf\\": \\"0\\", \\"others\\": \\"false\\"}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy1","type":"tool","uuid":"b8c2b602-396e-4727-8774-cd87edbe3a07","workflow_outputs":[{"label":"called_variants","output_name":"variants","uuid":"f0daf1f1-9200-45de-8c6e-776879096b3e"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","errors":null,"id":13,"input_connections":{"invcf":{"id":12,"output_name":"variants"}},"inputs":[],"label":null,"name":"Lofreq filter","outputs":[{"name":"outvcf","type":"vcf"}],"position":{"left":1623.8827831633464,"top":733.9296904212346},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_filter/lofreq_filter/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"950d1d49d678","name":"lofreq_filter","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"af\\": {\\"af_min\\": \\"0.0\\", \\"af_max\\": \\"0.0\\"}, \\"coverage\\": {\\"cov_min\\": \\"0\\", \\"cov_max\\": \\"0\\"}, \\"filter_by_type\\": {\\"keep_only\\": \\"\\", \\"__current_case__\\": 0, \\"qual\\": {\\"snvqual_filter\\": {\\"snvqual\\": \\"no\\", \\"__current_case__\\": 0}, \\"indelqual_filter\\": {\\"indelqual\\": \\"no\\", \\"__current_case__\\": 0}}}, \\"flag_or_drop\\": \\"--print-all\\", \\"invcf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sb\\": {\\"sb_filter\\": {\\"strand_bias\\": \\"mtc\\", \\"__current_case__\\": 2, \\"sb_alpha\\": \\"0.001\\", \\"sb_mtc\\": \\"fdr\\", \\"sb_compound\\": \\"true\\", \\"sb_indels\\": \\"false\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"72fc0a88-0fff-4434-9c7b-cff1694f8fbf","workflow_outputs":[{"label":"soft_filtered_variants","output_name":"outvcf","uuid":"58a73993-57b0-4590-b462-65f0a0f7978a"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy1","errors":null,"id":14,"input_connections":{"input":{"id":13,"output_name":"outvcf"},"snpDb|snpeff_db":{"id":4,"output_name":"snpeff_output"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"input"},{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"snpDb"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":null,"name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"left":1973.0859200969005,"top":761.6406458215656},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy1","tool_shed_repository":{"changeset_revision":"74aebe30fb52","name":"snpeff","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": \\"false\\", \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-intron\\", \\"-no-upstream\\", \\"-no-utr\\"], \\"generate_stats\\": \\"true\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": \\"true\\", \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"snpDb\\": {\\"genomeSrc\\": \\"custom\\", \\"__current_case__\\": 3, \\"snpeff_db\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"codon_table\\": \\"Standard\\"}, \\"spliceRegion\\": {\\"setSpliceRegions\\": \\"no\\", \\"__current_case__\\": 0}, \\"spliceSiteSize\\": null, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+T.galaxy1","type":"tool","uuid":"2fa5e8fd-9282-4156-ba29-93d6ad2b65e1","workflow_outputs":[{"label":"SnpEff eff: stats","output_name":"statsFile","uuid":"5587a236-9d63-4d3b-b368-0643fa07529e"},{"label":"SnpEff variants","output_name":"snpeff_output","uuid":"4a062f86-0ab4-4ba1-b8fc-70e35d1f1df7"}]}},"tags":["mpxv","generic"],"uuid":"be0d055d-25ac-422a-ae9c-b0efdfb5bf86","version":5},"readme":"Generic variation analysis on WGS PE data\\n-------------------------------------------\\n\\nThis workflows performs paired end read mapping with bwa-mem followed by\\nsensitive variant calling across a wide range of AFs with lofreq and variant\\nannotation with snpEff. The reference genome can be provided as a GenBank file.\\n","changelog":"# Changelog\\n\\n\\n## [0.1.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1]\\n\\n- Initial version of Generic variation analysis on WGS PE data workflow\\n","diagrams":"# Workflow diagrams\\n\\n## Generic variation analysis on WGS PE data\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ GenBank genome\\"]@{ shape: doc }\\n2[\\"ℹ️ Name for genome database\\"]@{ shape: lean-l }\\n3[\\"fastp\\"]@{ shape: process }\\n0 --> 3\\n4[\\"SnpEff build:\\"]@{ shape: process }\\n2 --> 4\\n1 --> 4\\n5[\\"Map with BWA-MEM\\"]@{ shape: process }\\n3 --> 5\\n4 --> 5\\n6[\\"Samtools view\\"]@{ shape: process }\\n5 --> 6\\n7[\\"MarkDuplicates\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Samtools stats\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Realign reads\\"]@{ shape: process }\\n7 --> 9\\n4 --> 9\\n10[\\"MultiQC\\"]@{ shape: process }\\n3 --> 10\\n8 --> 10\\n7 --> 10\\n11[\\"Insert indel qualities\\"]@{ shape: process }\\n9 --> 11\\n4 --> 11\\n12[\\"Call variants\\"]@{ shape: process }\\n11 --> 12\\n4 --> 12\\n13[\\"Lofreq filter\\"]@{ shape: process }\\n12 --> 13\\n14[\\"SnpEff eff:\\"]@{ shape: process }\\n13 --> 14\\n4 --> 14\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/generic-variant-calling-wgs-pe/main","dockstore_id":18261,"categories":["Variant Calling"],"collections":["Variant Calling"],"doi":"10.5281/zenodo.15078897","tests":[{"doc":"Test outline for Generic-variation-analysis-on-WGS-PE-data","job":{"GenBank genome":{"class":"File","filetype":"genbank","path":"test-data/GenBank genome.genbank"},"Name for genome database":"mpxv","Paired Collection":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"ERR3485802","elements":[{"identifier":"forward","class":"File","path":"test-data/ERR3485802.forward.fastq.gz"},{"identifier":"reverse","class":"File","path":"test-data/ERR3485802.reverse.fastq.gz"}]}]}},"outputs":{"Fasta sequences for genbank file":{"path":"test-data/Fasta sequences for genbank file.fasta","compare":"contains"},"SnpEff variants":{"element_tests":{"ERR3485802":{"path":"test-data/SnpEff variants.vcf","compare":"diff","lines_diff":6}}},"called_variants":{"element_tests":{"ERR3485802":{"path":"test-data/called_variants.vcf","compare":"diff","lines_diff":6}}},"mapped_reads_stats":{"element_tests":{"ERR3485802":{"path":"test-data/mapped_reads_stats.tabular"}}},"soft_filtered_variants":{"element_tests":{"ERR3485802":{"path":"test-data/soft_filtered_variants.vcf","compare":"diff","lines_diff":6}}}}}]}],"path":"./workflows/variant-calling/generic-variant-calling-wgs-pe"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/WGS-PE-variant-calling-in-haploid-system.ga","testParameterFiles":["/WGS-PE-variant-calling-in-haploid-system-tests.yml"],"authors":[{"name":"Anton Nekrutenko","orcid":"0000-0002-5987-8032"}],"definition":{"a_galaxy_workflow":"true","annotation":"Workflow for variant analysis against a reference genome in GenBank format","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-5987-8032","name":"Anton Nekrutenko"}],"format-version":"0.1","license":"MIT","name":"Paired end variant calling in haploid system","release":"0.1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Illumina reads with fastqsanger encoding","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Illumina reads with fastqsanger encoding","name":"Paired Collection"}],"label":"Paired Collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"87e2a757-c63f-42c4-ac41-2ff9802fc230","when":null,"workflow_outputs":[]},"1":{"annotation":"GTF with gene annotation, will be used for annotating variants","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"GTF with gene annotation, will be used for annotating variants","name":"Annotation GTF"}],"label":"Annotation GTF","name":"Input dataset","outputs":[],"position":{"left":15.4000244140625,"top":149.5},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"cc6b08f6-0cac-43b9-a6ac-5cb9e809aef0","when":null,"workflow_outputs":[]},"2":{"annotation":"Fasta file to use as reference for variant calling","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Fasta file to use as reference for variant calling","name":"Genome fasta"}],"label":"Genome fasta","name":"Input dataset","outputs":[],"position":{"left":68.79998779296875,"top":361.1807963463069},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"56f0bac5-1144-424e-a2ee-e7ce408ae855","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","errors":null,"id":3,"input_connections":{"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":472.81558221323667,"top":210.69642625646713},"post_job_actions":{"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0","tool_shed_repository":{"changeset_revision":"65b93b623c77","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\", \\"adapter_sequence2\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.2+galaxy0","type":"tool","uuid":"aab66ae3-1955-46f0-b0b8-9d6374b81d8a","when":null,"workflow_outputs":[{"label":"fastp html report","output_name":"report_html","uuid":"f780b8d7-8dbb-4a04-9522-8a2c99b317d9"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_build_gb/4.3+T.galaxy6","errors":null,"id":4,"input_connections":{"input_type|input":{"id":1,"output_name":"output"},"input_type|reference_source|input_fasta":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool SnpEff build:","name":"input_type"}],"label":null,"name":"SnpEff build:","outputs":[{"name":"snpeff_output","type":"snpeffdb"}],"position":{"left":437.79998779296875,"top":805.1807963463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_build_gb/4.3+T.galaxy6","tool_shed_repository":{"changeset_revision":"6322be79bd8e","name":"snpeff","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"codon_table\\": \\"Standard\\", \\"genome_version\\": \\"snpeff_db\\", \\"input_type\\": {\\"input_type_selector\\": \\"gtf\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+T.galaxy6","type":"tool","uuid":"691404df-5586-476d-a11e-28ccba3ddd55","when":null,"workflow_outputs":[{"label":"SnpEff4.3 database","output_name":"snpeff_output","uuid":"8d680888-a2b7-465f-890d-81e78aaa0396"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","errors":null,"id":5,"input_connections":{"fastq_input|fastq_input1":{"id":3,"output_name":"output_paired_coll"},"reference_source|ref_file":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map with BWA-MEM","name":"fastq_input"},{"description":"runtime parameter for tool Map with BWA-MEM","name":"reference_source"}],"label":null,"name":"Map with BWA-MEM","outputs":[{"name":"bam_output","type":"bam"}],"position":{"left":738.3546544092493,"top":442.2276665604545},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2","tool_shed_repository":{"changeset_revision":"64f11cf59c6e","name":"bwa","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"analysis_type\\": {\\"analysis_type_selector\\": \\"full\\", \\"__current_case__\\": 4, \\"algorithmic_options\\": {\\"algorithmic_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"scoring_options\\": {\\"scoring_options_selector\\": \\"do_not_set\\", \\"__current_case__\\": 1}, \\"io_options\\": {\\"io_options_selector\\": \\"set\\", \\"__current_case__\\": 0, \\"five\\": false, \\"q\\": true, \\"T\\": \\"30\\", \\"h\\": \\"5\\", \\"a\\": false, \\"C\\": false, \\"V\\": false, \\"Y\\": true, \\"M\\": false}}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iset_stats\\": \\"\\"}, \\"output_sort\\": \\"coordinate\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"index_a\\": \\"auto\\"}, \\"rg\\": {\\"rg_selector\\": \\"set\\", \\"__current_case__\\": 1, \\"read_group_id_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"read_group_sm_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"PL\\": \\"ILLUMINA\\", \\"read_group_lb_conditional\\": {\\"do_auto_name\\": true, \\"__current_case__\\": 0}, \\"CN\\": null, \\"DS\\": null, \\"DT\\": null, \\"FO\\": null, \\"KS\\": null, \\"PG\\": null, \\"PI\\": null, \\"PU\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.7.17.2","type":"tool","uuid":"b227a5b7-bd10-45f2-8b3c-9797ec9fe6c9","when":null,"workflow_outputs":[]},"6":{"annotation":"Retains alignments if read is paired and mapped in proper orientation ","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy1","errors":null,"id":6,"input_connections":{"input":{"id":5,"output_name":"bam_output"}},"inputs":[],"label":null,"name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":996.7531127929688,"top":307.0401713463069},"post_job_actions":{"RenameDatasetActionoutputsam":{"action_arguments":{"newname":"Mapped read pairs (filtered bwa-mem result)"},"action_type":"RenameDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.13+galaxy1","tool_shed_repository":{"changeset_revision":"c370440f901e","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": \\"0\\", \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": [\\"1\\", \\"2\\"], \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"fraction\\", \\"__current_case__\\": 0, \\"factor\\": \\"1.0\\", \\"seed\\": null}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.13+galaxy1","type":"tool","uuid":"9cd7e5e8-fdf0-4c0b-9921-a6fd0d8cc735","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","errors":null,"id":7,"input_connections":{"input":{"id":6,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"Samtools stats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1274.0343627929688,"top":236.8057963463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_stats/samtools_stats/2.0.2+galaxy2","tool_shed_repository":{"changeset_revision":"145f6d74ff5e","name":"samtools_stats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cov_threshold\\": null, \\"coverage_cond\\": {\\"coverage_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"filter_by_flags\\": {\\"filter_flags\\": \\"nofilter\\", \\"__current_case__\\": 1}, \\"gc_depth\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"insert_size\\": null, \\"most_inserts\\": null, \\"read_length\\": null, \\"remove_dups\\": false, \\"remove_overlaps\\": false, \\"sparse\\": false, \\"split_output_cond\\": {\\"split_output_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"trim_quality\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.2+galaxy2","type":"tool","uuid":"1c709200-f5e5-4699-a881-428fe57ec1fd","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2","errors":null,"id":8,"input_connections":{"inputFile":{"id":6,"output_name":"outputsam"}},"inputs":[],"label":null,"name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":1306.4093627929688,"top":431.1964213463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2","tool_shed_repository":{"changeset_revision":"a1f0b3f4b781","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": true, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.18.2.2","type":"tool","uuid":"32e3534c-283c-45a3-845e-edadfe29343a","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","errors":null,"id":9,"input_connections":{"results_0|software_cond|input":{"id":3,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":7,"output_name":"output"},"results_2|software_cond|output_0|input":{"id":8,"output_name":"metrics_file"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"left":1684.5108939673896,"top":115.18079634630689},"post_job_actions":{"HideDatasetActionstats":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"stats"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"Preprocessing and mapping reports"},"action_type":"RenameDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0","tool_shed_repository":{"changeset_revision":"9a913cdee30e","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"stats\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"saveLog\\": false, \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy0","type":"tool","uuid":"827f0bc5-b7b3-472f-95fc-5bf34fffb9fd","when":null,"workflow_outputs":[{"label":"Preprocessing and mapping reports","output_name":"html_report","uuid":"73117bba-89ef-49b8-a29f-df28a3bba689"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","errors":null,"id":10,"input_connections":{"reads":{"id":8,"output_name":"outFile"},"reference_source|ref":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Realign reads","name":"reference_source"}],"label":null,"name":"Realign reads","outputs":[{"name":"realigned","type":"bam"}],"position":{"left":1652.2999877929688,"top":581.6807963463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"aa35ee7f3ab2","name":"lofreq_viterbi","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_options\\": {\\"keepflags\\": false, \\"bq2_handling\\": {\\"replace_bq2\\": \\"keep\\", \\"__current_case__\\": 0, \\"defqual\\": \\"2\\"}}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy0","type":"tool","uuid":"299c5dae-cb7b-4de5-bdeb-6f5f2470488a","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","errors":null,"id":11,"input_connections":{"reads":{"id":10,"output_name":"realigned"},"reference_source|ref":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Call variants","name":"reference_source"}],"label":null,"name":"Call variants","outputs":[{"name":"variants","type":"vcf"}],"position":{"left":1911.7999877929688,"top":689.1807963463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy3","tool_shed_repository":{"changeset_revision":"b11e8e9c23bf","name":"lofreq_call","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"call_control\\": {\\"set_call_options\\": \\"yes\\", \\"__current_case__\\": 1, \\"coverage\\": {\\"min_cov\\": \\"10\\", \\"max_depth\\": \\"1000000\\"}, \\"pe\\": {\\"use_orphan\\": false}, \\"bc_quals\\": {\\"min_bq\\": \\"20\\", \\"min_alt_bq\\": \\"20\\", \\"alt_bq\\": {\\"modify\\": \\"\\", \\"__current_case__\\": 0}}, \\"align_quals\\": {\\"alnqual\\": {\\"use_alnqual\\": \\"\\", \\"__current_case__\\": 0, \\"alnqual_choice\\": {\\"alnquals_to_use\\": \\"\\", \\"__current_case__\\": 1, \\"extended_baq\\": true}}}, \\"map_quals\\": {\\"min_mq\\": \\"0\\", \\"use_mq\\": {\\"no_mq\\": \\"\\", \\"__current_case__\\": 0, \\"max_mq\\": \\"255\\"}}, \\"source_qual\\": {\\"use_src_qual\\": {\\"src_qual\\": \\"\\", \\"__current_case__\\": 0}}, \\"joint_qual\\": {\\"min_jq\\": \\"0\\", \\"min_alt_jq\\": \\"0\\", \\"def_alt_jq\\": \\"0\\"}}, \\"filter_control\\": {\\"filter_type\\": \\"set_lofreq_standard\\", \\"__current_case__\\": 2, \\"sig\\": \\"0.01\\", \\"bonf\\": \\"dynamic\\", \\"others\\": \\"\\"}, \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"ref_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"regions\\": {\\"restrict_to_region\\": \\"genome\\", \\"__current_case__\\": 0}, \\"variant_types\\": \\"--call-indels\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.5+galaxy3","type":"tool","uuid":"6df71467-a7b6-45f5-ab1d-1a9672d4f81c","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"variants"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2075.2999877929688,"top":471.4999963463248},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"/^#/ { print; next } ($4 ~ /^[ACGT]$/ && $5 ~ /^[ACGT]$/) { print }\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"60e608c1-4b33-4d09-b3e5-be4c9334a233","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy2","errors":null,"id":13,"input_connections":{"input":{"id":12,"output_name":"outfile"},"snpDb|snpeff_db":{"id":4,"output_name":"snpeff_output"}},"inputs":[{"description":"runtime parameter for tool SnpEff eff:","name":"intervals"},{"description":"runtime parameter for tool SnpEff eff:","name":"snpDb"},{"description":"runtime parameter for tool SnpEff eff:","name":"transcripts"}],"label":null,"name":"SnpEff eff:","outputs":[{"name":"snpeff_output","type":"vcf"},{"name":"statsFile","type":"html"}],"position":{"left":2109.8781127929688,"top":850.8214213463069},"post_job_actions":{"TagDatasetActionsnpeff_output":{"action_arguments":{"tags":"VariantsAsVCF"},"action_type":"TagDatasetAction","output_name":"snpeff_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy2","tool_shed_repository":{"changeset_revision":"6322be79bd8e","name":"snpeff","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotations\\": [\\"-formatEff\\", \\"-classic\\"], \\"chr\\": \\"\\", \\"csvStats\\": false, \\"filter\\": {\\"specificEffects\\": \\"no\\", \\"__current_case__\\": 0}, \\"filterOut\\": [\\"-no-downstream\\", \\"-no-intergenic\\", \\"-no-intron\\", \\"-no-upstream\\", \\"-no-utr\\"], \\"generate_stats\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputFormat\\": \\"vcf\\", \\"intervals\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"noLog\\": true, \\"offset\\": \\"default\\", \\"outputConditional\\": {\\"outputFormat\\": \\"vcf\\", \\"__current_case__\\": 0}, \\"snpDb\\": {\\"genomeSrc\\": \\"custom\\", \\"__current_case__\\": 3, \\"snpeff_db\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"codon_table\\": \\"Standard\\"}, \\"spliceRegion\\": {\\"setSpliceRegions\\": \\"no\\", \\"__current_case__\\": 0}, \\"spliceSiteSize\\": null, \\"transcripts\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"udLength\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+T.galaxy2","type":"tool","uuid":"e3d2c332-c029-463c-bb8b-771716db7d67","when":null,"workflow_outputs":[{"label":"SnpEff variants","output_name":"snpeff_output","uuid":"8f3cd914-554d-4fab-97ca-245245a63cb3"},{"label":"SnpEff eff reports","output_name":"statsFile","uuid":"37d8fb6f-b2e6-4f91-9f8e-58a30354b35d"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":14,"input_connections":{"input":{"id":13,"output_name":"snpeff_output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2416.2999877929688,"top":510.1807963463069},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\".\\", \\"extract\\": \\"CHROM POS FILTER REF ALT DP AF DP4 SB EFF[*].IMPACT EFF[*].FUNCLASS EFF[*].EFFECT EFF[*].GENE EFF[*].CODON EFF[*].AA EFF[*].TRID\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": true, \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"767ff87e-6a49-4bd4-ba05-27b4d349e695","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":15,"input_connections":{"input_list":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2588.2999877929688,"top":751.1807963463069},"post_job_actions":{"TagDatasetActionoutput":{"action_arguments":{"tags":"VariantsAsTSV"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_multiple\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"5feceb7a-ba26-4974-9215-dfdbbcc5b296","when":null,"workflow_outputs":[{"label":"Annotated Variants","output_name":"output","uuid":"301f944e-0f65-415f-a56f-9899dca85e88"}]}},"tags":["generic","VeuPath","Haploid"],"uuid":"d5b4eace-d876-4999-bad3-14c1cb02b0c9","version":5},"readme":"# Haploid variant calling for whole genome sequencing paired end data\\n\\nThis workflow uses Illumina or Element read data to discover variants (short nucleotide polymorphisms, SNPs, and small indels) in haploid genomes with multiple genomic sequences (contigs, scaffolds, or chromosomes).\\n\\n## Inputs dataset\\n\\n- The workflow needs a list of paired end fastq files\\n- A GTF containtaing the Gene annotation for the selected haploid genome\\n- A fasta file for the haploid genome to call variants against\\n\\n## Outputs\\n\\n- Tab-delimited summary of annotated variants\\n- Report summarizing the quality of input data and mapping results\\n\\n## Processing\\n\\n- The workflow will remove adapters using fastp\\n- The filtered reads are aligned with bwa-mem.\\n- Only properly aligned mate pairs are retained, PCR duplicates are removed.\\n- Alignments are re-aligned using lofreq viterbi and variants are called with lofreq call.\\n- Variants are annotated with snpeff eff\\n","changelog":"# Changelog\\n\\n\\n## [0.1] - 2024-10-29\\n\\n- Initial version of Paired end variant calling in haploid system workflow\\n","diagrams":"# Workflow diagrams\\n\\n## Paired end variant calling in haploid system\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Paired Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ Annotation GTF\\"]@{ shape: doc }\\n2[\\"ℹ️ Genome fasta\\"]@{ shape: doc }\\n3[\\"fastp\\"]@{ shape: process }\\n0 --> 3\\n4[\\"SnpEff build:\\"]@{ shape: process }\\n1 --> 4\\n2 --> 4\\n5[\\"Map with BWA-MEM\\"]@{ shape: process }\\n3 --> 5\\n2 --> 5\\n6[\\"Samtools view\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Samtools stats\\"]@{ shape: process }\\n6 --> 7\\n8[\\"MarkDuplicates\\"]@{ shape: process }\\n6 --> 8\\n9[\\"MultiQC\\"]@{ shape: process }\\n3 --> 9\\n7 --> 9\\n8 --> 9\\n10[\\"Realign reads\\"]@{ shape: process }\\n8 --> 10\\n2 --> 10\\n11[\\"Call variants\\"]@{ shape: process }\\n10 --> 11\\n2 --> 11\\n12[\\"Text reformatting\\"]@{ shape: process }\\n11 --> 12\\n13[\\"SnpEff eff:\\"]@{ shape: process }\\n12 --> 13\\n4 --> 13\\n14[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n13 --> 14\\n15[\\"Collapse Collection\\"]@{ shape: process }\\n14 --> 15\\n```\\n","updated":"2024-10-29T00:00:00","trsID":"#workflow/github.com/iwc-workflows/haploid-variant-calling-wgs-pe/main","dockstore_id":29174,"categories":[],"collections":["Variant Calling"],"doi":"10.5281/zenodo.15078462","tests":[{"doc":"Test outline for WGS-PE-variant-calling-in-haploid-system","job":{"Annotation GTF":{"class":"File","location":"https://zenodo.org/records/14009320/files/Annotation%20GTF.gtf?download=1","filetype":"gtf"},"Genome fasta":{"class":"File","location":"https://zenodo.org/records/14009320/files/Genome%20fasta.fasta.gz?download=1","filetype":"fasta.gz"},"Paired Collection":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"ERR018930","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/records/14009320/files/ERR018930_forward.fastqsanger.gz?download=1"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/records/14009320/files/ERR018930_reverse.fastqsanger.gz?download=1"}]},{"class":"Collection","type":"paired","identifier":"ERR1035492","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/records/14009320/files/ERR1035492_forward.fastqsanger.gz?download=1"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/records/14009320/files/ERR1035492_reverse.fastqsanger.gz?download=1"}]}]}},"outputs":{"Annotated Variants":{"path":"test-data/Annotated Variants.tabular"},"SnpEff variants":{"element_tests":{"ERR018930":{"asserts":[{"has_line":{"line":"NC_009906.1\\t3204\\t.\\tA\\tG\\t120.0\\tPASS\\tDP=22;AF=0.727273;SB=2;DP4=2,3,3,14;EFF=INTRAGENIC(MODIFIER|||||PVX_087665||NON_CODING|||G)"}},{"has_line":{"line":"NC_009906.1\\t3261\\t.\\tC\\tA\\t52.0\\tPASS\\tDP=15;AF=0.333333;SB=0;DP4=3,7,2,3;EFF=INTRAGENIC(MODIFIER|||||PVX_087665||NON_CODING|||A)"}}]},"ERR1035492":{"asserts":{"has_line":{"line":"NC_009906.1\\t2975\\t.\\tA\\tG\\t75.0\\tPASS\\tDP=26;AF=0.692308;SB=0;DP4=5,3,12,6;EFF=INTRAGENIC(MODIFIER|||||PVX_087665||NON_CODING|||G)"}}}}}}}]}],"path":"./workflows/variant-calling/haploid-variant-calling-wgs-pe"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Generic-variation-analysis-reporting.ga","testParameterFiles":["/Generic-variation-analysis-reporting-tests.yml"],"authors":[{"name":"Wolfgang Maier","orcid":"0000-0002-9464-6640"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow takes a VCF dataset of variants produced by any of the variant calling workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling and generates tabular lists of variants by Samples and by Variant, and an overview plot of variants and their allele-frequencies.","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9464-6640","name":"Wolfgang Maier"}],"format-version":"0.1","release":"0.1.1","license":"MIT","name":"Generic variation analysis reporting","steps":{"0":{"annotation":"Variation data in VCF format. Can be the output of any of the workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Variation data in VCF format. Can be the output of any of the workflows in https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling","name":"Variation data to report"}],"label":"Variation data to report","name":"Input dataset collection","outputs":[],"position":{"left":29.25,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"vcf\\", \\"vcf_bgzip\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"900f58cd-3ac6-4371-9295-6596726b1ee9","workflow_outputs":[]},"1":{"annotation":"Allele Frequency Filter. This is the minimum allele frequency required for variants to be included in the reports.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Allele Frequency Filter. This is the minimum allele frequency required for variants to be included in the reports.","name":"AF Filter"}],"label":"AF Filter","name":"Input parameter","outputs":[],"position":{"left":28.3125,"top":208.8203125},"tool_id":null,"tool_state":"{\\"default\\": 0.05, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"664a4fe1-a981-4e22-b6c4-628c87baf28a","workflow_outputs":[{"label":null,"output_name":"output","uuid":"15fcff98-3e33-4deb-a083-a6c3c94b1b8d"}]},"2":{"annotation":"Depth Filter. This is the minimum depth of all alignments at a variant site. ","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Depth Filter. This is the minimum depth of all alignments at a variant site. ","name":"DP Filter"}],"label":"DP Filter","name":"Input parameter","outputs":[],"position":{"left":26.75,"top":358.8359375},"tool_id":null,"tool_state":"{\\"default\\": 1, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"8d6df2cd-f16c-476e-bc00-e83a5e568cdd","workflow_outputs":[{"label":null,"output_name":"output","uuid":"a01b2b83-81d7-4cd5-a307-797046910821"}]},"3":{"annotation":"Depth Filter for variant allele. This is the minimum depth of alignments supporting a variant.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Depth Filter for variant allele. This is the minimum depth of alignments supporting a variant.","name":"DP_ALT Filter"}],"label":"DP_ALT Filter","name":"Input parameter","outputs":[],"position":{"left":0,"top":572.3359375},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"842ca5f3-acfd-4243-88cd-625d1834ad89","workflow_outputs":[{"label":null,"output_name":"output","uuid":"89133c97-8b6f-4c7b-abbd-799fd385ac6a"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":4,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":348.65625,"top":159},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"2e497a770bca","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": \\"DP >= 1\\"}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"8adcdaba-fea6-4cb1-9ec3-70972604145e","workflow_outputs":[{"label":"prefiltered_variants","output_name":"output","uuid":"7e953ef0-9e3a-485b-8201-290512552269"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":5,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"},"components_3|param_type|component_value":{"id":2,"output_name":"output"},"components_5|param_type|component_value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":347,"top":283.2734375},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"( ( DP4[2] + DP4[3] ) < ( \\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" * DP ) ) | ( DP < \\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" ) | ( ( AF * DP ) < ( \\"}}, {\\"__index__\\": 5, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 6, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\" - 0.5 ) )\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"7b42a044-ea48-4aae-82d0-7e134e47e6ee","workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":6,"input_connections":{"components_1|param_type|component_value":{"id":1,"output_name":"output"},"components_3|param_type|component_value":{"id":2,"output_name":"output"},"components_5|param_type|component_value":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":343.71875,"top":590.75},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"min_af_\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"|min_dp_\\"}}, {\\"__index__\\": 3, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"|min_dp_alt_\\"}}, {\\"__index__\\": 5, \\"param_type\\": {\\"select_param_type\\": \\"integer\\", \\"__current_case__\\": 1, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"dd9b21a6-e2e1-4637-aca7-dd485b52ddf9","workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":7,"input_connections":{"filter_expression|expr":{"id":5,"output_name":"out1"},"filtering|add_filter":{"id":6,"output_name":"out1"},"input":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":579.40625,"top":448.984375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"2e497a770bca","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"filtering\\": {\\"mode\\": \\"add_filter\\", \\"__current_case__\\": 3, \\"add_filter\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"8c1888cd-fceb-45fc-add4-2e7edc582ca1","workflow_outputs":[{"label":"filtered_variants","output_name":"output","uuid":"a923399c-9faa-40a7-be8d-caa2ca72c5be"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":8,"input_connections":{"input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":815.4375,"top":345.9609375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered extracted variants"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"2e497a770bca","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\".\\", \\"extract\\": \\"CHROM POS FILTER REF ALT DP AF DP4 SB EFF[*].IMPACT EFF[*].FUNCLASS EFF[*].EFFECT EFF[*].GENE EFF[*].CODON EFF[*].AA EFF[*].TRID DP4[2] DP4[3]\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": \\"true\\", \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"dce57343-cb31-469a-b886-d62e7ce53bad","workflow_outputs":[{"label":"filtered_extracted_variants","output_name":"output","uuid":"fb09ef77-9522-452a-a0e9-73a4a580b8aa"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","errors":null,"id":9,"input_connections":{"input":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1266.7265625,"top":367.2578125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Extracted variants with filtered and renamed effects and AF recalculated"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","tool_shed_repository":{"changeset_revision":"427903d47026","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": \\"false\\", \\"cond\\": \\"round((c17 + c18) / c6, 6)\\", \\"header_lines_conditional\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"header_new_column_name\\": \\"AFrecalc\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"round\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6","type":"tool","uuid":"726b7b08-31a8-4ec7-b5b9-027b1f104eeb","workflow_outputs":[{"label":"af_recalculated","output_name":"out_file1","uuid":"a47c05f5-0ea1-4649-a7c1-7b88cca0e6e5"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":10,"input_connections":{"in_file":{"id":9,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":1433.15625,"top":147.5},"post_job_actions":{"RenameDatasetActionout_file":{"action_arguments":{"newname":"Collapsed effects (Datamash on all variants)"},"action_type":"RenameDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"1,2,3,4,5,6,19,7,8,9\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"false\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"10\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"11\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"12\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"13\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"14\\"}, {\\"__index__\\": 5, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"15\\"}, {\\"__index__\\": 6, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"16\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"a5100b84-498a-47fe-b408-fd159a8d729a","workflow_outputs":[{"label":"collapsed_effects","output_name":"out_file","uuid":"407d9cee-4b7d-4855-b62c-a166ecf8f50e"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":11,"input_connections":{"infile":{"id":10,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1556.34375,"top":323.5625},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Extracted variants with highest impact effects"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\t]+\\\\\\\\t([^\\\\\\\\t,]+),[^\\\\\\\\s]+\\", \\"global\\": \\"true\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"true\\", \\"replace_pattern\\": \\"\\\\\\\\t$1\\\\\\\\t$2\\\\\\\\t$3\\\\\\\\t$4\\\\\\\\t$5\\\\\\\\t$6\\\\\\\\t$7\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"true\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"6e156bbc-a207-4941-bd6a-d654473fde37","workflow_outputs":[{"label":"highest_impact_effects","output_name":"outfile","uuid":"aacf2235-fd9c-43e9-a886-dde7ccc7c691"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":12,"input_connections":{"infile":{"id":11,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1707.1953125,"top":165.234375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Variants with cleaned up header"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"(GroupBy|collapse)\\\\\\\\(([^)]+)\\\\\\\\)\\", \\"global\\": \\"true\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"true\\", \\"replace_pattern\\": \\"$2\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"false\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"4d002a96-055c-4892-8ad4-d6266de51e4e","workflow_outputs":[{"label":"cleaned_header","output_name":"outfile","uuid":"0bade229-08f2-491c-8361-77ec91ceed98"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":13,"input_connections":{"infile":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1825.5,"top":329.8828125},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Fully processed variants collection for reporting"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"AFrecalc\\\\\\\\tAF\\", \\"global\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"true\\", \\"replace_pattern\\": \\"AF\\\\\\\\tAFcaller\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"false\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"9ee0a745-0f40-4579-91ed-39aaa9895302","workflow_outputs":[{"label":"processed_variants_collection","output_name":"outfile","uuid":"2707f4c6-dd54-44bb-adad-a3703d165823"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":14,"input_connections":{"input_list":{"id":13,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1984.5078125,"top":154.984375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": \\"true\\", \\"__current_case__\\": 0, \\"place_name\\": \\"same_multiple\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"c8ad21ea-c72b-4063-9974-380869b87d58","workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","errors":null,"id":15,"input_connections":{"input":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2106.3828125,"top":338.484375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","tool_shed_repository":{"changeset_revision":"427903d47026","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": \\"false\\", \\"cond\\": \\"str(c5) + \'>\' + str(c6)\\", \\"header_lines_conditional\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"header_new_column_name\\": \\"change\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"round\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6","type":"tool","uuid":"61cc455a-4330-4518-bf8a-b099780bdb30","workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","errors":null,"id":16,"input_connections":{"input":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2329.8984375,"top":339.296875},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6","tool_shed_repository":{"changeset_revision":"427903d47026","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": \\"false\\", \\"cond\\": \\"str(int(c3)) + \':\' + str(c19)\\", \\"header_lines_conditional\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"header_new_column_name\\": \\"change_with_pos\\"}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"round\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6","type":"tool","uuid":"5a0297af-22c4-4550-8673-cbe044496410","workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":17,"input_connections":{"infile":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2536.0703125,"top":207.0859375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"All variants of all samples"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"EFF[*].\\", \\"global\\": \\"true\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"false\\", \\"replace_pattern\\": \\"\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"false\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"23c73601-2fdb-4a1e-be60-ff0f1ec96be8","workflow_outputs":[{"label":"all_variants_all_samples","output_name":"outfile","uuid":"c3bf5f59-3363-4305-ab99-7c38fceab34c"}]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":18,"input_connections":{"in_file":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":2787.3359375,"top":158.5234375},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"20\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"true\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"min\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"max\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"collapse\\", \\"op_column\\": \\"8\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"61adea76-ce05-489d-af2b-1ad74bd141fa","workflow_outputs":[]},"19":{"annotation":"","content_id":"Filter1","errors":null,"id":19,"input_connections":{"input":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2780.1328125,"top":371.6484375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"c4==\'PASS\' or c4==\'.\'\\", \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"c5568630-a6f4-468c-b320-5f28547f7c14","workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":20,"input_connections":{"in_file":{"id":17,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":3071.03125,"top":568.5703125},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"},"RenameDatasetActionout_file":{"action_arguments":{"newname":"Cross-sample variant stats (Datamash)"},"action_type":"RenameDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"3\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"true\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}, {\\"__index__\\": 1, \\"op_name\\": \\"min\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 2, \\"op_name\\": \\"max\\", \\"op_column\\": \\"8\\"}, {\\"__index__\\": 3, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"19\\"}, {\\"__index__\\": 4, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"13\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"517d8ba6-1df6-417c-96e9-f297b3b8f51f","workflow_outputs":[]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":21,"input_connections":{"in_file":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":2895.6484375,"top":789.3203125},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"20\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"true\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"6f97bd54-aeea-4615-b6f0-b9cc8e99b519","workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","errors":null,"id":22,"input_connections":{"infile1":{"id":19,"output_name":"out_file1"},"infile2":{"id":18,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3028.71875,"top":178.8984375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants of all samples with stats"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"20\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.2","type":"tool","uuid":"ce58eb8c-118f-467e-80e4-b0bf35c2919f","workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":23,"input_connections":{"in_file":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":3037.1484375,"top":439.0625},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"3\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"true\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"countunique\\", \\"op_column\\": \\"1\\"}], \\"print_full_line\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"aee9b1d2-a707-4101-8b22-23ebaa3a32a1","workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","errors":null,"id":24,"input_connections":{"infile1":{"id":16,"output_name":"out_file1"},"infile2":{"id":21,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3232.8984375,"top":719.8359375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"20\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.2","type":"tool","uuid":"9aeb39d3-d040-43ee-a3fa-2a65eb1a7c34","workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","errors":null,"id":25,"input_connections":{"in_file":{"id":22,"output_name":"output"}},"inputs":[],"label":null,"name":"Datamash","outputs":[{"name":"out_file","type":"tabular"}],"position":{"left":3272.8359375,"top":142.0234375},"post_job_actions":{"HideDatasetActionout_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0","tool_shed_repository":{"changeset_revision":"562f3c677828","name":"datamash_ops","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"grouping\\": \\"1\\", \\"header_in\\": \\"true\\", \\"header_out\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"in_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"need_sort\\": \\"true\\", \\"operations\\": [{\\"__index__\\": 0, \\"op_name\\": \\"unique\\", \\"op_column\\": \\"2\\"}], \\"print_full_line\\": \\"true\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"74886968-16db-41a0-9d2a-98475d1d163e","workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","errors":null,"id":26,"input_connections":{"infile1":{"id":17,"output_name":"outfile"},"infile2":{"id":23,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3274.8984375,"top":351.5703125},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"3\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.2","type":"tool","uuid":"9c3e2d08-a103-497b-b22c-667f1ca23152","workflow_outputs":[]},"27":{"annotation":"","content_id":"Cut1","errors":null,"id":27,"input_connections":{"input":{"id":24,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3458.375,"top":655.5859375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c3,c4,c5,c6,c7,c8,c9,c11,c12,c13,c14,c15,c16,c17,c18,c19\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"c74fc520-a76c-4b7b-b846-8cd8310ead70","workflow_outputs":[]},"28":{"annotation":"","content_id":"Cut1","errors":null,"id":28,"input_connections":{"input":{"id":25,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3496.8984375,"top":246.8125},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c4,c6,c7,c13,c14,c15,c16,c17,c18,c19,c21,c22,c23,c26,c24,c25,c20\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"102b3ed6-1c6c-4545-9985-dfd3ac735acf","workflow_outputs":[]},"29":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","errors":null,"id":29,"input_connections":{"infile1":{"id":26,"output_name":"output"},"infile2":{"id":20,"output_name":"out_file"}},"inputs":[],"label":null,"name":"Join","outputs":[{"name":"output","type":"input"}],"position":{"left":3492.9375,"top":444.734375},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtered variants of all samples with per-position stats"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column1\\": \\"1\\", \\"column2\\": \\"1\\", \\"empty_string_filler\\": \\"0\\", \\"header\\": \\"true\\", \\"ignore_case\\": \\"false\\", \\"infile1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"jointype\\": \\" \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.2","type":"tool","uuid":"95dcccd0-0323-4448-b8ca-ad1fce8d2a45","workflow_outputs":[]},"30":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","errors":null,"id":30,"input_connections":{"split_parms|input":{"id":27,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Split file","outputs":[{"name":"list_output_tab","type":"input"}],"position":{"left":3681.6328125,"top":655.7578125},"post_job_actions":{"RenameDatasetActionlist_output_tab":{"action_arguments":{"newname":"Per-sample variants for plotting"},"action_type":"RenameDatasetAction","output_name":"list_output_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","tool_shed_repository":{"changeset_revision":"6cbe2f30c2d7","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"tabular\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"top\\": \\"1\\", \\"split_by\\": {\\"select_split_by\\": \\"col\\", \\"__current_case__\\": 0, \\"id_col\\": \\"1\\", \\"match_regex\\": \\"(.*)\\", \\"sub_regex\\": \\"\\\\\\\\1\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.0","type":"tool","uuid":"c0accd6f-76af-46f7-a5f0-647000d54444","workflow_outputs":[{"label":"variants_for_plotting","output_name":"list_output_tab","uuid":"e62f958f-e749-43b4-9c02-17fced624689"}]},"31":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","errors":null,"id":31,"input_connections":{"infile":{"id":28,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3706.375,"top":146.25},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"All variants of all samples"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"caseinsensitive\\": \\"false\\", \\"find_pattern\\": \\"unique\\\\\\\\(Sample\\\\\\\\)\\\\\\\\tcollapse\\\\\\\\(Sample\\\\\\\\)\\\\\\\\tcollapse\\\\\\\\(AF\\\\\\\\)\\", \\"global\\": \\"true\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": \\"true\\", \\"replace_pattern\\": \\"SAMPLES(above-thresholds)\\\\\\\\tSAMPLES(all)\\\\\\\\tAFs(all)\\", \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}, \\"skip_first_line\\": \\"false\\", \\"wholewords\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.3","type":"tool","uuid":"cf84a2f6-689b-4bed-bc0d-324d40f46dd7","workflow_outputs":[]},"32":{"annotation":"","content_id":"Cut1","errors":null,"id":32,"input_connections":{"input":{"id":29,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3710.1875,"top":471.234375},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c1,c4,c5,c6,c7,c8,c9,c11,c10,c12,c13,c14,c15,c16,c17,c18,c23,c24,c25,c26,c19\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"3d0237fc-33b6-4305-b6e8-2502175f624f","workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1","errors":null,"id":33,"input_connections":{"infile":{"id":31,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3928.8984375,"top":247.8125},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Combined Variant Report by Variant"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"1\\", \\"ignore_case\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sortkeys\\": [{\\"__index__\\": 0, \\"column\\": \\"1\\", \\"order\\": \\"\\", \\"style\\": \\"g\\"}], \\"unique\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"6edca54f-9af8-4305-bd11-4c74251af900","workflow_outputs":[{"label":"by_variant_report","output_name":"outfile","uuid":"30808300-2789-48b0-9073-98ec1bddcb3f"}]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1","errors":null,"id":34,"input_connections":{"infile":{"id":32,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Sort","outputs":[{"name":"outfile","type":"input"}],"position":{"left":3932.6953125,"top":457.4609375},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"Combined Variant Report by Sample"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1","tool_shed_repository":{"changeset_revision":"f46f0e4f75c4","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"header\\": \\"1\\", \\"ignore_case\\": \\"false\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sortkeys\\": [{\\"__index__\\": 0, \\"column\\": \\"1\\", \\"order\\": \\"\\", \\"style\\": \\"\\"}, {\\"__index__\\": 1, \\"column\\": \\"2\\", \\"order\\": \\"\\", \\"style\\": \\"n\\"}], \\"unique\\": \\"false\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"eb754361-9ec5-46c8-80b5-6bb040345719","workflow_outputs":[{"label":"combined_variant_report","output_name":"outfile","uuid":"2f0680a4-d809-4f85-8df9-4fcd6c4f3d80"}]}},"tags":["mpvx","generic"],"uuid":"9e94da53-d8ab-4713-8aac-7c6674fa1d39","version":2},"readme":"Generic variation analysis reporting\\n--------------------------------------\\n\\nThis workflow takes table of variants produced by any of the variant calling workflows in\\nhttps://github.com/galaxyproject/iwc/tree/main/workflows/variant-calling\\nand generates a list of variants by Samples and by Variant.\\n","changelog":"# Changelog\\n\\n## [0.1.1] - 2023-11-20\\n\\n- Fix author in dockstore\\n\\n## [0.1]\\n\\n- Initial version generic variation analysis reporting workflow\\n","diagrams":"# Workflow diagrams\\n\\n## Generic variation analysis reporting\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Variation data to report\\"]@{ shape: docs }\\n1[\\"ℹ️ AF Filter\\"]@{ shape: lean-l }\\n2[\\"ℹ️ DP Filter\\"]@{ shape: lean-l }\\n3[\\"ℹ️ DP_ALT Filter\\"]@{ shape: lean-l }\\n4[\\"SnpSift Filter\\"]@{ shape: process }\\n0 --> 4\\n5[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 5\\n2 --> 5\\n3 --> 5\\n6[\\"Compose text parameter value\\"]@{ shape: process }\\n1 --> 6\\n2 --> 6\\n3 --> 6\\n7[\\"SnpSift Filter\\"]@{ shape: process }\\n5 --> 7\\n6 --> 7\\n4 --> 7\\n8[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n7 --> 8\\n9[\\"Compute\\"]@{ shape: process }\\n8 --> 9\\n10[\\"Datamash\\"]@{ shape: process }\\n9 --> 10\\n11[\\"Replace\\"]@{ shape: process }\\n10 --> 11\\n12[\\"Replace\\"]@{ shape: process }\\n11 --> 12\\n13[\\"Replace\\"]@{ shape: process }\\n12 --> 13\\n14[\\"Collapse Collection\\"]@{ shape: process }\\n13 --> 14\\n15[\\"Compute\\"]@{ shape: process }\\n14 --> 15\\n16[\\"Compute\\"]@{ shape: process }\\n15 --> 16\\n17[\\"Replace\\"]@{ shape: process }\\n16 --> 17\\n18[\\"Datamash\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Filter\\"]@{ shape: process }\\n17 --> 19\\n20[\\"Datamash\\"]@{ shape: process }\\n17 --> 20\\n21[\\"Datamash\\"]@{ shape: process }\\n19 --> 21\\n22[\\"Join\\"]@{ shape: process }\\n19 --> 22\\n18 --> 22\\n23[\\"Datamash\\"]@{ shape: process }\\n19 --> 23\\n24[\\"Join\\"]@{ shape: process }\\n16 --> 24\\n21 --> 24\\n25[\\"Datamash\\"]@{ shape: process }\\n22 --> 25\\n26[\\"Join\\"]@{ shape: process }\\n17 --> 26\\n23 --> 26\\n27[\\"Cut\\"]@{ shape: process }\\n24 --> 27\\n28[\\"Cut\\"]@{ shape: process }\\n25 --> 28\\n29[\\"Join\\"]@{ shape: process }\\n26 --> 29\\n20 --> 29\\n30[\\"Split file\\"]@{ shape: process }\\n27 --> 30\\n31[\\"Replace\\"]@{ shape: process }\\n28 --> 31\\n32[\\"Cut\\"]@{ shape: process }\\n29 --> 32\\n33[\\"Sort\\"]@{ shape: process }\\n31 --> 33\\n34[\\"Sort\\"]@{ shape: process }\\n32 --> 34\\n```\\n","updated":"2023-11-20T00:00:00","trsID":"#workflow/github.com/iwc-workflows/variation-reporting/main","dockstore_id":24515,"categories":[],"collections":["Variant Calling"],"doi":"10.5281/zenodo.15078801","tests":[{"doc":"Test outline for Generic-variation-analysis-reporting","job":{"AF Filter":0.05,"DP Filter":1,"DP_ALT Filter":10,"Variation data to report":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"filtered variants ERR3485802","path":"test-data/Variation data to report_filtered variants ERR3485802.vcf"}]}},"outputs":{"af_recalculated":{"element_tests":{"filtered variants ERR3485802":{"path":"test-data/af_recalculated.tabular","compare":"diff","lines_diff":6}}},"all_variants_all_samples":{"path":"test-data/all_variants_all_samples.tabular","compare":"diff","lines_diff":6},"by_variant_report":{"path":"test-data/by_variant_report.tabular","compare":"diff","lines_diff":6},"combined_variant_report":{"path":"test-data/combined_variant_report.tabular","compare":"diff","lines_diff":6}}}]}],"path":"./workflows/variant-calling/variation-reporting"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/rnaseq-pe.ga","testParameterFiles":["/rnaseq-pe-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"},{"name":"Pavankumar Videm","orcid":"0000-0002-5192-126X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete RNA-Seq analysis for paired-end data: Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.","comments":[{"child_steps":[15,16,17],"color":"lime","data":{"title":"Map Strandedness parameter"},"id":0,"position":[855.5086321940169,1252.5],"size":[251,558],"type":"frame"},{"child_steps":[19,20],"color":"red","data":{"title":"Coverage Files"},"id":1,"position":[1603.68749985966,0],"size":[375,538],"type":"frame"},{"child_steps":[14,22,23],"color":"yellow","data":{"title":"Additional quantification"},"id":2,"position":[1391.2086321940171,1073.6999999999998],"size":[860,716],"type":"frame"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","release":"1.2","name":"RNA-Seq Analysis: Paired-End Read Processing and Quantification","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Should be a list of paired-end RNA-seq fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a list of paired-end RNA-seq fastqs","name":"Collection paired FASTQ files"}],"label":"Collection paired FASTQ files","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":462.87500890145213},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list:paired\\"}","tool_version":null,"type":"data_collection_input","uuid":"020f7580-f647-471f-b210-dbf9dfcf80bf","when":null,"workflow_outputs":[]},"1":{"annotation":"This is optional. Fastp will use overlapping. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC, for TruSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This is optional. Fastp will use overlapping. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC, for TruSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC","name":"Forward adapter"}],"label":"Forward adapter","name":"Input parameter","outputs":[],"position":{"left":35.17186494140515,"top":567.328103383874},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6d1de206-97c4-41ba-8702-a980f748a689","when":null,"workflow_outputs":[]},"2":{"annotation":"This is optional. Fastp will use overlapping. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTGACGCTGCCGACGA, for TruSeq: GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"This is optional. Fastp will use overlapping. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTGACGCTGCCGACGA, for TruSeq: GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT or AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT","name":"Reverse adapter"}],"label":"Reverse adapter","name":"Input parameter","outputs":[],"position":{"left":77.546875,"top":667.4062589014521},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"7aae38b8-39a8-4fba-bc6b-bda8c1ea9162","when":null,"workflow_outputs":[]},"3":{"annotation":"Whether to compute additional QC like fastQC, gene body coverage etc...","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Whether to compute additional QC like fastQC, gene body coverage etc...","name":"Generate additional QC reports"}],"label":"Generate additional QC reports","name":"Input parameter","outputs":[],"position":{"left":114.671875,"top":764.5781339014521},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"bea331ce-d6fa-4312-8dce-a8b13bc4e77b","when":null,"workflow_outputs":[]},"4":{"annotation":"Select the reference genome","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Select the reference genome","name":"Reference genome"}],"label":"Reference genome","name":"Input parameter","outputs":[],"position":{"left":157.109375,"top":870.218728383874},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a72b29ae-65a7-4e4d-a996-2943ab5b02aa","when":null,"workflow_outputs":[]},"5":{"annotation":"GTF compatible with the reference genome. Mind the UCSC/Ensembl differences in chromosome naming.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"GTF compatible with the reference genome. Mind the UCSC/Ensembl differences in chromosome naming.","name":"GTF file of annotation"}],"label":"GTF file of annotation","name":"Input dataset","outputs":[],"position":{"left":199.671875,"top":977.3906033838739},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"4987fd01-b72c-4bc0-9d49-a840a0d04b8e","when":null,"workflow_outputs":[]},"6":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"Strandedness"}],"label":"Strandedness","name":"Input parameter","outputs":[],"position":{"left":251.890625,"top":1073.781228383874},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"2ccf2242-dddd-45b6-8cea-746d9b5d49bc","when":null,"workflow_outputs":[]},"7":{"annotation":"Use featureCounts tool instead of RNA STAR?","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Use featureCounts tool instead of RNA STAR?","name":"Use featureCounts for generating count tables"}],"label":"Use featureCounts for generating count tables","name":"Input parameter","outputs":[],"position":{"left":291.34375,"top":1173.124978383874},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f43d45e6-5692-41cb-8cfc-205dbeab1bd0","when":null,"workflow_outputs":[]},"8":{"annotation":"Whether FPKM values should be computed with Cufflinks","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"Whether FPKM values should be computed with Cufflinks","name":"Compute Cufflinks FPKM"}],"label":"Compute Cufflinks FPKM","name":"Input parameter","outputs":[],"position":{"left":346.83340121591186,"top":1309.3879390873915},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"78868cdb-baff-4388-a41a-a700259da1ed","when":null,"workflow_outputs":[]},"9":{"annotation":"It could be a GTF with for example one entry for the chrM forward and one entry for the chrM reverse","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"It could be a GTF with for example one entry for the chrM forward and one entry for the chrM reverse","name":"GTF with regions to exclude from FPKM normalization with Cufflinks"}],"label":"GTF with regions to exclude from FPKM normalization with Cufflinks","name":"Input dataset","outputs":[],"position":{"left":380.53912206977975,"top":1425.307227138327},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0d4e1623-adc9-4ff3-8156-7e1322b1e278","when":null,"workflow_outputs":[]},"10":{"annotation":"Whether FPKM values should be computed with StringTie","content_id":null,"errors":null,"id":10,"input_connections":{},"inputs":[{"description":"Whether FPKM values should be computed with StringTie","name":"Compute StringTie FPKM"}],"label":"Compute StringTie FPKM","name":"Input parameter","outputs":[],"position":{"left":406.9259176995931,"top":1559.4856968335116},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"056f11f4-bdd9-4361-b037-2c0822320cc2","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__FLATTEN__","errors":null,"id":11,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Flatten collection","outputs":[{"name":"output","type":"input"}],"position":{"left":430.77112544776327,"top":623.8873433527641},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__FLATTEN__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"join_identifier\\": \\"_\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"dddbd652-fe4f-42c8-a2ff-bdc259d4928c","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","errors":null,"id":12,"input_connections":{"single_paired|adapter_trimming_options|adapter_sequence1":{"id":1,"output_name":"output"},"single_paired|adapter_trimming_options|adapter_sequence2":{"id":2,"output_name":"output"},"single_paired|paired_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":"remove adapters + bad quality bases","name":"fastp","outputs":[{"name":"output_paired_coll","type":"input"},{"name":"report_json","type":"json"}],"position":{"left":392.6147686047791,"top":201.9284685268642},"post_job_actions":{"HideDatasetActionoutput_paired_coll":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_paired_coll"},"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","tool_shed_repository":{"changeset_revision":"a626e8c0e1ba","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": \\"30\\", \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": \\"15\\", \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": false, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": null, \\"umi_len\\": null, \\"umi_prefix\\": null}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"paired_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_sequence2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy3","type":"tool","uuid":"80d855a6-7a77-46e9-9124-9dd3d9d72cc1","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":13,"input_connections":{"input_param_type|input_param":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"no additional QC","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":2159.2104562234385,"top":23.726446900364408},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"boolean\\", \\"__current_case__\\": 3, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": false, \\"to\\": \\"true\\"}, {\\"__index__\\": 1, \\"from\\": true, \\"to\\": \\"false\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"input\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"fd5cb0a4-a2a4-4677-8a1d-e81e1043533b","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":14,"input_connections":{"components_0|param_type|component_value":{"id":4,"output_name":"output"}},"inputs":[],"label":"get reference_genome as text parameter","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1497.7951400347756,"top":1173.104961480357},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bcceb886-0592-4b3f-8b39-26fe79caeb62","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":15,"input_connections":{"input_param_type|input_param":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get featureCounts strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":880.1168847150709,"top":1303.1637555172579},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"1\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"2\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"0\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"72d7d4d1-d1be-481c-b2a1-1e85192b2661","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":16,"input_connections":{"input_param_type|input_param":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get cufflinks strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":880.334534933914,"top":1483.3515528952898},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"fr-secondstrand\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"fr-firststrand\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"fr-unstranded\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"f027f3d5-3b91-4061-b5cd-78553509ce5e","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":17,"input_connections":{"input_param_type|input_param":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get Stringtie strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":875.5945006269037,"top":1653.5777444451405},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"--fr\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"--rf\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"63e6501b-e1de-4184-af42-720cc9a6dfb0","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1","errors":null,"id":18,"input_connections":{"refGenomeSource|GTFconditional|genomeDir":{"id":4,"output_name":"output"},"refGenomeSource|GTFconditional|sjdbGTFfile":{"id":5,"output_name":"output"},"singlePaired|input":{"id":12,"output_name":"output_paired_coll"}},"inputs":[{"description":"runtime parameter for tool RNA STAR","name":"singlePaired"}],"label":"STAR: map and count and coverage splitted","name":"RNA STAR","outputs":[{"name":"output_log","type":"txt"},{"name":"splice_junctions","type":"interval"},{"name":"mapped_reads","type":"bam"},{"name":"reads_per_gene","type":"tabular"},{"name":"signal_unique_str1","type":"bedgraph"},{"name":"signal_uniquemultiple_str1","type":"bedgraph"},{"name":"signal_unique_str2","type":"bedgraph"},{"name":"signal_uniquemultiple_str2","type":"bedgraph"}],"position":{"left":732.1609477893844,"top":246.8799136582777},"post_job_actions":{"HideDatasetActionoutput_log":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_log"},"HideDatasetActionreads_per_gene":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reads_per_gene"},"HideDatasetActionsignal_unique_str1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_unique_str1"},"HideDatasetActionsignal_unique_str2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_unique_str2"},"HideDatasetActionsignal_uniquemultiple_str1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_uniquemultiple_str1"},"HideDatasetActionsignal_uniquemultiple_str2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_uniquemultiple_str2"},"HideDatasetActionsplice_junctions":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"splice_junctions"},"RenameDatasetActionmapped_reads":{"action_arguments":{"newname":"Mapped Reads"},"action_type":"RenameDatasetAction","output_name":"mapped_reads"},"RenameDatasetActionreads_per_gene":{"action_arguments":{"newname":"Reads per gene from STAR"},"action_type":"RenameDatasetAction","output_name":"reads_per_gene"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1","tool_shed_repository":{"changeset_revision":"53255f6eecfc","name":"rgrnastar","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"algo\\": {\\"params\\": {\\"settingsType\\": \\"full\\", \\"__current_case__\\": 3, \\"seed\\": {\\"seedSearchStartLmax\\": \\"50\\", \\"seedSearchStartLmaxOverLread\\": \\"1.0\\", \\"seedSearchLmax\\": \\"0\\", \\"seedMultimapNmax\\": \\"10000\\", \\"seedPerReadNmax\\": \\"1000\\", \\"seedPerWindowNmax\\": \\"50\\", \\"seedNoneLociPerWindow\\": \\"10\\"}, \\"align\\": {\\"alignIntronMin\\": \\"20\\", \\"alignIntronMax\\": \\"1000000\\", \\"alignMatesGapMax\\": \\"1000000\\", \\"alignSJoverhangMin\\": \\"8\\", \\"alignSJstitchMismatchNmax\\": {\\"alignSJstitchMismatchNmax1\\": \\"0\\", \\"alignSJstitchMismatchNmax2\\": \\"-1\\", \\"alignSJstitchMismatchNmax3\\": \\"0\\", \\"alignSJstitchMismatchNmax4\\": \\"0\\"}, \\"alignSJDBoverhangMin\\": \\"1\\", \\"alignSplicedMateMapLmin\\": \\"0\\", \\"alignSplicedMateMapLminOverLmate\\": \\"0.66\\", \\"alignWindowsPerReadNmax\\": \\"10000\\", \\"alignTranscriptsPerWindowNmax\\": \\"100\\", \\"alignTranscriptsPerReadNmax\\": \\"10000\\", \\"alignEndsType\\": \\"Local\\", \\"peOverlapNbasesMin\\": \\"0\\", \\"peOverlapMMp\\": \\"0.01\\"}, \\"chim_settings\\": {\\"chimSegmentMin\\": \\"12\\", \\"chimScoreMin\\": \\"0\\", \\"chimScoreDropMax\\": \\"20\\", \\"chimScoreSeparation\\": \\"10\\", \\"chimScoreJunctionNonGTAG\\": \\"-1\\", \\"chimJunctionOverhangMin\\": \\"20\\", \\"chimSegmentReadGapMax\\": \\"0\\", \\"chimFilter\\": true, \\"chimMainSegmentMultNmax\\": \\"10\\", \\"chimMultimapNmax\\": \\"1\\", \\"chimMultimapScoreRange\\": \\"1\\"}, \\"junction_limits\\": {\\"limitOutSJoneRead\\": \\"1000\\", \\"limitOutSJcollapsed\\": \\"1000000\\", \\"limitSjdbInsertNsj\\": \\"1000000\\"}}}, \\"chimOutType\\": \\"\\", \\"filter\\": {\\"basic_filters\\": [\\"exclude_unmapped\\"], \\"output_params2\\": {\\"output_select2\\": \\"yes\\", \\"__current_case__\\": 0, \\"outFilterType\\": true, \\"outFilterMultimapScoreRange\\": \\"1\\", \\"outFilterMultimapNmax\\": \\"20\\", \\"outFilterMismatchNmax\\": \\"999\\", \\"outFilterMismatchNoverLmax\\": \\"0.3\\", \\"outFilterMismatchNoverReadLmax\\": \\"0.04\\", \\"outFilterScoreMin\\": \\"0\\", \\"outFilterScoreMinOverLread\\": \\"0.66\\", \\"outFilterMatchNmin\\": \\"0\\", \\"outFilterMatchNminOverLread\\": \\"0.66\\", \\"outSAMmultNmax\\": \\"-1\\", \\"outSAMtlen\\": \\"1\\"}}, \\"oformat\\": {\\"outSAMattributes\\": [\\"NH\\", \\"HI\\", \\"AS\\", \\"nM\\"], \\"HI_offset\\": \\"1\\", \\"outSAMprimaryFlag\\": \\"OneBestScore\\", \\"outSAMmapqUnique\\": \\"255\\", \\"wasp_conditional\\": {\\"waspOutputMode\\": \\"\\", \\"__current_case__\\": 1}}, \\"outWig\\": {\\"outWigType\\": \\"bedGraph\\", \\"__current_case__\\": 1, \\"outWigTypeSecondWord\\": \\"\\", \\"outWigStrand\\": true, \\"outWigReferencesPrefix\\": \\"-\\", \\"outWigNorm\\": true}, \\"perf\\": {\\"outBAMsortingBinsN\\": \\"50\\", \\"winAnchorMultimapNmax\\": \\"50\\"}, \\"refGenomeSource\\": {\\"geneSource\\": \\"indexed\\", \\"__current_case__\\": 0, \\"GTFconditional\\": {\\"GTFselect\\": \\"without-gtf-with-gtf\\", \\"__current_case__\\": 1, \\"genomeDir\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfeatureExon\\": \\"exon\\", \\"sjdbOverhang\\": \\"100\\", \\"quantmode_output\\": {\\"quantMode\\": \\"GeneCounts\\", \\"__current_case__\\": 1}}}, \\"singlePaired\\": {\\"sPaired\\": \\"paired_collection\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"twopass\\": {\\"twopassMode\\": \\"None\\", \\"__current_case__\\": 0, \\"twopass_read_subset\\": \\"\\", \\"sj_precalculated\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.11a+galaxy1","type":"tool","uuid":"be1a277d-6d0d-4aaf-acea-9735426932b0","when":null,"workflow_outputs":[{"label":"Mapped Reads","output_name":"mapped_reads","uuid":"f5722954-fd5c-4e66-aed0-4cd099dd9b1c"}]},"19":{"annotation":"","id":19,"input_connections":{"STAR BAM":{"id":18,"input_subworkflow_step_id":1,"output_name":"mapped_reads"},"STAR log":{"id":18,"input_subworkflow_step_id":0,"output_name":"output_log"}},"inputs":[{"description":"","name":"Generate Unstranded Coverage"}],"label":"Generate Unstranded Coverage","name":"Get Uniquely mapped unstranded coverage","outputs":[],"position":{"left":1683.2031849964808,"top":86.70584740647254},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Get Uniquely mapped unstranded coverage","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"RNA STAR log files","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"RNA STAR log files","name":"STAR log"}],"label":"STAR log","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":46},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3e3f7463-a635-44a2-a1f4-7e12aae3d25b","when":null,"workflow_outputs":[]},"1":{"annotation":"Mapped BAM files generate from RNA STAR","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Mapped BAM files generate from RNA STAR","name":"STAR BAM"}],"label":"STAR BAM","name":"Input dataset collection","outputs":[],"position":{"left":280,"top":209},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"10dddc85-d3fc-4cba-a637-0c40153d284f","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":2,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":"get scaling factor","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":280,"top":20},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"$0~\\\\\\"Uniquely mapped reads number\\\\\\"{print 1000000/$NF}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"77cf2304-36bf-4337-bc2d-bc4d8b831668","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","errors":null,"id":3,"input_connections":{"input_bam":{"id":1,"output_name":"output"}},"inputs":[],"label":"keep uniquely mapped reads","name":"Filter BAM","outputs":[{"name":"out_file2","type":"txt"},{"name":"out_file1","type":"bam"}],"position":{"left":560,"top":195},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"HideDatasetActionout_file2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","tool_shed_repository":{"changeset_revision":"993b19f20c76","name":"bamtools_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"conditions\\": [{\\"__index__\\": 0, \\"filters\\": [{\\"__index__\\": 0, \\"bam_property\\": {\\"bam_property_selector\\": \\"tag\\", \\"__current_case__\\": 21, \\"bam_property_value\\": \\"NH=1\\"}}]}], \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rule_configuration\\": {\\"rules_selector\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.2+galaxy2","type":"tool","uuid":"59db7088-85bf-42ba-9dcd-76836f9d6310","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":560,"top":0},"post_job_actions":{"HideDatasetActionfloat_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"float_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"float\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"83e6d856-5809-419a-b4bc-639662a95751","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","errors":null,"id":5,"input_connections":{"input_type|input":{"id":3,"output_name":"out_file1"},"report|scale":{"id":4,"output_name":"float_param"}},"inputs":[{"description":"runtime parameter for tool bedtools Genome Coverage","name":"input_type"},{"description":"runtime parameter for tool bedtools Genome Coverage","name":"report"}],"label":"Scaled Coverage both strands combined","name":"bedtools Genome Coverage","outputs":[{"name":"output","type":"bedgraph"}],"position":{"left":840,"top":12},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"d\\": false, \\"dz\\": false, \\"five\\": false, \\"input_type\\": {\\"input_type_select\\": \\"bam\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"report\\": {\\"report_select\\": \\"bg\\", \\"__current_case__\\": 0, \\"zero_regions\\": false, \\"scale\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split\\": true, \\"strand\\": \\"\\", \\"three\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"df53e789-be2d-4b09-be8d-a07a35c14913","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":6,"input_connections":{"input1":{"id":5,"output_name":"output"}},"inputs":[],"label":"convert both strands coverage to bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1120,"top":67},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Both Strands Coverage"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"62d76728-f614-48c0-9560-3abd05b8366d","when":null,"workflow_outputs":[{"label":"Both Strands Coverage","output_name":"out_file1","uuid":"73763ad9-c132-4764-ada6-eece0c30e74f"}]}},"tags":[],"uuid":"2ebe04f9-4b13-4811-9f32-aed6d18682fb"},"tool_id":null,"type":"subworkflow","uuid":"6ce6314b-d339-40bf-9f0e-53ca525204be","when":null,"workflow_outputs":[{"label":"Unstranded Coverage","output_name":"Both Strands Coverage","uuid":"f90cc885-438b-49ee-b1d1-8ed45f7dea5c"}]},"20":{"annotation":"","id":20,"input_connections":{"Bedgraph strand 1":{"id":18,"input_subworkflow_step_id":1,"output_name":"signal_unique_str1"},"Bedgraph strand 2":{"id":18,"input_subworkflow_step_id":2,"output_name":"signal_unique_str2"},"strandedness":{"id":6,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[{"description":"","name":"Generate Stranded Coverage"}],"label":"Generate Stranded Coverage","name":"Re-arrange Stranded RNA-seq coverage","outputs":[],"position":{"left":1692.210489743303,"top":282.0471550654421},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Re-arrange Stranded RNA-seq coverage","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"strandedness"}],"label":"strandedness","name":"Input parameter","outputs":[],"position":{"left":0,"top":370},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"56955f25-7e9e-4201-a056-b61f47359c84","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d363c181-7f37-45ef-9732-86aaf879478a"}]},"1":{"annotation":"From STAR strand 1","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"From STAR strand 1","name":"Bedgraph strand 1"}],"label":"Bedgraph strand 1","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":472},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bedgraph\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"7919f2ed-554a-4e08-93eb-6896a2195293","when":null,"workflow_outputs":[]},"2":{"annotation":"From STAR strand 2","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"From STAR strand 2","name":"Bedgraph strand 2"}],"label":"Bedgraph strand 2","name":"Input dataset collection","outputs":[],"position":{"left":569.999952609169,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bedgraph\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9318223c-4443-4341-8448-11df606ccfdf","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":3,"input_connections":{"input_param_type|input_param":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get replacement for strand2","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":289.9999763045845,"top":51},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"$1_reverse\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"$1_forward\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"$1_reverse\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"2ec50fce-ef76-4f56-9fc6-5c6b34c6593e","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":4,"input_connections":{"input_param_type|input_param":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get replacement for strand1","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":289.9999763045845,"top":359},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"$1_forward\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"$1_reverse\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"$1_forward\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"b7e2cedf-e846-4553-9e48-4aa9a93d40b2","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":5,"input_connections":{"input_collection":{"id":1,"output_name":"output"}},"inputs":[],"label":"get identifiers","name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":289.9999763045845,"top":225},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"7d866aad-dd6c-4d17-bf6d-8f0026652265","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":6,"input_connections":{"find_and_replace_0|replace_pattern":{"id":3,"output_name":"output_param_text"},"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"New labels strand 2","name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":569.999952609169,"top":123},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ec10a98b-adb9-4037-9137-1717068d7157","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":7,"input_connections":{"find_and_replace_0|replace_pattern":{"id":4,"output_name":"output_param_text"},"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"New labels strand 1","name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":569.999952609169,"top":328},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"4eb0d67c-f53c-4f84-90fc-3db41e4f0aa7","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":8,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabelled strand 2","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":849.9998437853558,"top":106},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"147ccbb0-b951-4851-9bf5-2c6b439ac1b6","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":9,"input_connections":{"how|labels":{"id":7,"output_name":"outfile"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabelled strand 1","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":849.9999005376209,"top":365},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"e9dcd2df-2834-4e9a-bef2-7666de012b8a","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":10,"input_connections":{"inputs_0|input":{"id":9,"output_name":"output"},"inputs_1|input":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1129.9999903467356,"top":120},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"f47d7381-13b2-4615-b947-17a0ac2ad500","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":11,"input_connections":{"input1":{"id":10,"output_name":"output"}},"inputs":[],"label":"convert to bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1410.000136908115,"top":171},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"uniquely mapped stranded coverage"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"852f2d23-b48e-44d0-aaf2-aed33532d98a","when":null,"workflow_outputs":[{"label":"Uniquely Mapped Stranded Coverage","output_name":"out_file1","uuid":"0a31ff80-b3d1-4959-869f-1a6a3c627dde"}]}},"tags":[],"uuid":"1972b03b-02a8-4828-9206-a6b502030280"},"tool_id":null,"type":"subworkflow","uuid":"bf990b24-1ec8-4e67-acec-fecf26a3d49f","when":null,"workflow_outputs":[{"label":"Stranded Coverage","output_name":"Uniquely Mapped Stranded Coverage","uuid":"9161f772-6904-4479-aff4-566295f5b3b2"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0","errors":null,"id":21,"input_connections":{"alignment":{"id":18,"output_name":"mapped_reads"},"anno|reference_gene_sets":{"id":5,"output_name":"output"},"strand_specificity":{"id":15,"output_name":"output_param_text"},"when":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool featureCounts","name":"anno"}],"label":null,"name":"featureCounts","outputs":[{"name":"output_short","type":"tabular"},{"name":"output_summary","type":"tabular"}],"position":{"left":1713.913195680125,"top":672.9994811085061},"post_job_actions":{"HideDatasetActionoutput_short":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_short"},"HideDatasetActionoutput_summary":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_summary"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0","tool_shed_repository":{"changeset_revision":"5dbff3a5845a","name":"featurecounts","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"anno\\": {\\"anno_select\\": \\"history\\", \\"__current_case__\\": 2, \\"reference_gene_sets\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gff_feature_type\\": \\"exon\\", \\"gff_feature_attribute\\": \\"gene_id\\", \\"summarization_level\\": false}, \\"extended_parameters\\": {\\"multifeatures\\": {\\"multifeat\\": \\"\\", \\"__current_case__\\": 0}, \\"exon_exon_junction_read_counting_enabled\\": {\\"count_exon_exon_junction_reads\\": \\"\\", \\"__current_case__\\": 1}, \\"long_reads\\": false, \\"by_read_group\\": false, \\"largest_overlap\\": false, \\"min_overlap\\": \\"1\\", \\"frac_overlap\\": \\"0\\", \\"frac_overlap_feature\\": \\"0\\", \\"read_extension_5p\\": \\"0\\", \\"read_extension_3p\\": \\"0\\", \\"read_reduction\\": \\"\\", \\"R\\": false}, \\"format\\": \\"tabdel_short\\", \\"include_feature_length_file\\": false, \\"pe_parameters\\": {\\"paired_end_status\\": \\"PE_fragments\\", \\"__current_case__\\": 2, \\"check_distance_enabled\\": {\\"checkFragLength\\": \\"false\\", \\"__current_case__\\": 1}, \\"only_both_ends\\": false, \\"exclude_chimerics\\": true}, \\"read_filtering_parameters\\": {\\"mapping_quality\\": \\"0\\", \\"splitonly\\": \\"\\", \\"primary\\": false, \\"ignore_dup\\": false}, \\"strand_specificity\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.8+galaxy0","type":"tool","uuid":"5a03b546-eb3f-4e99-b845-24c8f54ed183","when":"$(inputs.when)","workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","errors":null,"id":22,"input_connections":{"guide|guide_source|ref_hist":{"id":5,"output_name":"output"},"input_options|input_bam":{"id":18,"output_name":"mapped_reads"},"rna_strandness":{"id":17,"output_name":"output_param_text"},"when":{"id":10,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"input_options"}],"label":"Compute FPKM with StringTie","name":"StringTie","outputs":[{"name":"output_gtf","type":"gtf"},{"name":"gene_abundance_estimation","type":"tabular"}],"position":{"left":1731.7359886400766,"top":1424.2142641508385},"post_job_actions":{"HideDatasetActionoutput_gtf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_gtf"},"RenameDatasetActiongene_abundance_estimation":{"action_arguments":{"newname":"Gene abundance estimates from StringTie"},"action_type":"RenameDatasetAction","output_name":"gene_abundance_estimation"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","tool_shed_repository":{"changeset_revision":"cbf488da3b2c","name":"stringtie","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"abundance_estimation\\": true, \\"omit_sequences\\": \\"\\", \\"name_prefix\\": null, \\"fraction\\": \\"0.01\\", \\"min_tlen\\": \\"200\\", \\"min_anchor_len\\": \\"10\\", \\"min_anchor_cov\\": \\"1\\", \\"min_bundle_cov\\": \\"1\\", \\"bdist\\": \\"50\\", \\"bundle_fraction\\": \\"1.0\\", \\"disable_trimming\\": false, \\"multi_mapping\\": false, \\"point_features\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"guide\\": {\\"use_guide\\": \\"yes\\", \\"__current_case__\\": 1, \\"guide_source\\": {\\"guide_gff_select\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_hist\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_estimation\\": true, \\"special_outputs\\": {\\"special_outputs_select\\": \\"no\\", \\"__current_case__\\": 2}, \\"coverage_file\\": false}, \\"input_options\\": {\\"input_mode\\": \\"short_reads\\", \\"__current_case__\\": 0, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rna_strandness\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.3+galaxy0","type":"tool","uuid":"5a796064-9c52-4b5c-b5da-02c43f6c742f","when":"$(inputs.when)","workflow_outputs":[{"label":"Gene Abundance Estimates from StringTie","output_name":"gene_abundance_estimation","uuid":"9bcb277a-bb4d-4629-8c71-4cac7cee4c63"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4","errors":null,"id":23,"input_connections":{"advanced_settings|library_type":{"id":16,"output_name":"output_param_text"},"advanced_settings|mask_file":{"id":9,"output_name":"output"},"bias_correction|seq_source|index":{"id":14,"output_name":"out1"},"input":{"id":18,"output_name":"mapped_reads"},"reference_annotation|reference_annotation_file":{"id":5,"output_name":"output"},"when":{"id":8,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cufflinks","name":"advanced_settings"},{"description":"runtime parameter for tool Cufflinks","name":"advanced_settings"},{"description":"runtime parameter for tool Cufflinks","name":"reference_annotation"}],"label":"Compute FPKM with cufflinks","name":"Cufflinks","outputs":[{"name":"genes_expression","type":"tabular"},{"name":"transcripts_expression","type":"tabular"},{"name":"assembled_isoforms","type":"gtf"},{"name":"total_map_mass","type":"txt"},{"name":"skipped","type":"gtf"}],"position":{"left":1998.2812719123385,"top":1150.343757982693},"post_job_actions":{"HideDatasetActionassembled_isoforms":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"assembled_isoforms"},"HideDatasetActionskipped":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"skipped"},"HideDatasetActiontotal_map_mass":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"total_map_mass"},"RenameDatasetActiongenes_expression":{"action_arguments":{"newname":"Genes Expression from Cufflinks"},"action_type":"RenameDatasetAction","output_name":"genes_expression"},"RenameDatasetActiontranscripts_expression":{"action_arguments":{"newname":"Transcripts Expression from Cufflinks"},"action_type":"RenameDatasetAction","output_name":"transcripts_expression"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4","tool_shed_repository":{"changeset_revision":"6926716bbfb5","name":"cufflinks","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_settings\\": {\\"use_advanced_settings\\": \\"Yes\\", \\"__current_case__\\": 1, \\"library_type\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mask_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inner_mean_dist\\": \\"200\\", \\"inner_dist_std_dev\\": \\"80\\", \\"max_mle_iterations\\": \\"5000\\", \\"junc_alpha\\": \\"0.001\\", \\"small_anchor_fraction\\": \\"0.09\\", \\"overhang_tolerance\\": \\"8\\", \\"max_bundle_length\\": \\"10000000\\", \\"max_bundle_frags\\": \\"1000000\\", \\"min_intron_length\\": \\"50\\", \\"trim_three_avgcov_thresh\\": \\"10\\", \\"trim_three_dropoff_frac\\": \\"0.1\\"}, \\"bias_correction\\": {\\"do_bias_correction\\": \\"Yes\\", \\"__current_case__\\": 0, \\"seq_source\\": {\\"index_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"global_model\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_correction\\": \\"--no-effective-length-correction\\", \\"max_intron_len\\": \\"300000\\", \\"min_isoform_fraction\\": \\"0.1\\", \\"multiread_correct\\": true, \\"pre_mrna_fraction\\": \\"0.15\\", \\"reference_annotation\\": {\\"use_ref\\": \\"Use reference annotation\\", \\"__current_case__\\": 1, \\"reference_annotation_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"compatible_hits_norm\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1.4","type":"tool","uuid":"af5419b7-ae6b-498c-8123-f609b8fcd8b4","when":"$(inputs.when)","workflow_outputs":[{"label":"Genes Expression from Cufflinks","output_name":"genes_expression","uuid":"3bd4fbda-f241-4ebd-b7a5-dea7271c521e"},{"label":"Transcripts Expression from Cufflinks","output_name":"transcripts_expression","uuid":"2c13ef79-948c-4f5f-8516-a769550128fd"}]},"24":{"annotation":"","id":24,"input_connections":{"RNA STAR count tables":{"id":18,"input_subworkflow_step_id":3,"output_name":"reads_per_gene"},"Strandness param":{"id":6,"input_subworkflow_step_id":2,"output_name":"output"},"featureCounts count table":{"id":21,"input_subworkflow_step_id":4,"output_name":"output_short"},"featureCounts summaries collection":{"id":21,"input_subworkflow_step_id":1,"output_name":"output_summary"}},"inputs":[],"label":null,"name":"Process Count files","outputs":[],"position":{"left":2161.105895746449,"top":665.1799212422835},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","name":"Process Count files","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"This file will be used in MultiQC if the user opts out of featureCounts to generate count files. ","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":0,"input_connections":{},"inputs":[],"label":"A dummy featureCounts summary file","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":0,"top":16},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": \\"Status\\\\tFeaturecounts_was_not_used\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}, {\\"__index__\\": 1, \\"line\\": \\"Assigned\\\\t0\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}, {\\"__index__\\": 2, \\"line\\": \\"Unassigned_Unmapped\\\\t0\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"be0268be-a3ec-4d75-854b-d8eecf4b4521","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection from featureCounts summaries","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection from featureCounts summaries","name":"featureCounts summaries collection"}],"label":"featureCounts summaries collection","name":"Input dataset collection","outputs":[],"position":{"left":2.3481338306177992,"top":215.26828327443155},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9ca62bd8-0272-41e2-9e4b-e8f45c4e0c19","when":null,"workflow_outputs":[]},"2":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"Strandness param"}],"label":"Strandness param","name":"Input parameter","outputs":[],"position":{"left":0.000017190473798223138,"top":433.85391433745605},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"180fc8bd-b8f4-486b-b300-e503bf379373","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d0e53d5e-1b5f-40d7-80e6-f15c5e2ffbc8"}]},"3":{"annotation":"Count tables collection from RNA STAR","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Count tables collection from RNA STAR","name":"RNA STAR count tables"}],"label":"RNA STAR count tables","name":"Input dataset collection","outputs":[],"position":{"left":5.368903743075304,"top":589.5341644488473},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"d94c5c03-7bad-493f-9f5d-c5b6f33157ae","when":null,"workflow_outputs":[]},"4":{"annotation":"Count tables collection from featureCounts","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Count tables collection from featureCounts","name":"featureCounts count table"}],"label":"featureCounts count table","name":"Input dataset collection","outputs":[],"position":{"left":7.4993932207943,"top":704.0194730841151},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"12f710b8-757e-4c25-90be-4cfca673f54b","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":0,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":279.9999490094584,"top":0},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/__tc__/\\\\\\\\t/g\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3588c02b-15b7-487b-8970-bc76ddf8348a","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":6,"input_connections":{"input_param_type|input_param":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"awk command from strand for STAR counts","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":279.9999833196541,"top":401.33828843669767},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$3}\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$4}\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$2}\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"2b9a5ca1-e974-450b-99b3-3fcce531154b","when":null,"workflow_outputs":[]},"7":{"annotation":"featureCounts summaries","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":7,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":1,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":5,"output_name":"output"}},"inputs":[],"label":"featureCounts summaries","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":840.0000391186242,"top":16},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"FeatureCounts Summary Table"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"26eca035-e403-4392-9ba9-08cd471154dd","when":null,"workflow_outputs":[{"label":"FeatureCounts Summary Table","output_name":"data_param","uuid":"8c15f8aa-f2eb-41bf-bbaa-947263f3bfec"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":8,"input_connections":{"code":{"id":6,"output_name":"output_param_text"},"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":"Extract gene counts","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":560.0000901091657,"top":386.17188642026633},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"HTS count like output"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"34922bd3-4676-4f7d-8e05-af0ef0c1efa4","when":null,"workflow_outputs":[]},"9":{"annotation":"Counts Table ","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":9,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":4,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":8,"output_name":"outfile"}},"inputs":[],"label":"Counts table","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":837.0001537846339,"top":413.1722363660909},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Counts Table"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"d87cdd15-8b7c-4498-a818-7369ee4e006a","when":null,"workflow_outputs":[{"label":"Counts Table","output_name":"data_param","uuid":"bb7b60bf-81b1-4d58-9724-9814e9284ab3"}]}},"tags":[],"uuid":"dc18851c-cee0-4300-ac1d-25d3104b9cd2"},"tool_id":null,"type":"subworkflow","uuid":"6957fcb0-4312-4764-9d80-c5306759f914","when":null,"workflow_outputs":[{"label":"Counts Table","output_name":"Counts Table","uuid":"dfe40af6-1361-4953-aac7-fbca6cfdb47a"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":25,"input_connections":{"results_0|software_cond|input":{"id":12,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":18,"output_name":"output_log"},"results_1|software_cond|output_1|type|input":{"id":18,"output_name":"reads_per_gene"},"results_2|software_cond|input":{"id":24,"output_name":"FeatureCounts Summary Table"},"when":{"id":13,"output_name":"output_param_boolean"}},"inputs":[],"label":"Combined MultiQC without additional QC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2405.2752524038633,"top":15.83131975019036},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"featureCounts\\", \\"__current_case__\\": 9, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"8ad489c5-92b6-4e73-99a4-543923bea427","when":"$(inputs.when)","workflow_outputs":[{"label":"Small MultiQC HTML report","output_name":"html_report","uuid":"91f6fe1c-d957-4e0e-83a7-757d879ffec3"},{"label":"Small MultiQC stats","output_name":"stats","uuid":"db8e150d-c55c-4554-a790-c031a35f8b45"}]},"26":{"annotation":"","id":26,"input_connections":{"FASTQ collection":{"id":11,"input_subworkflow_step_id":0,"output_name":"output"},"STAR gene counts":{"id":18,"input_subworkflow_step_id":3,"output_name":"reads_per_gene"},"STAR logs":{"id":18,"input_subworkflow_step_id":2,"output_name":"output_log"},"STAR paired-end BAM":{"id":18,"input_subworkflow_step_id":6,"output_name":"mapped_reads"},"fastp Reports":{"id":12,"input_subworkflow_step_id":1,"output_name":"report_json"},"featureCounts summaries":{"id":24,"input_subworkflow_step_id":4,"output_name":"FeatureCounts Summary Table"},"reference_annotation_gtf":{"id":5,"input_subworkflow_step_id":5,"output_name":"output"},"when":{"id":3,"output_name":"output"}},"inputs":[{"description":"","name":"Combined MultiQC Quality Report with additional QC"}],"label":"Combined MultiQC Quality Report with additional QC","name":"RNA-seq-Paired-QC","outputs":[],"position":{"left":2465.503860179348,"top":451.63585583919314},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","name":"RNA-seq-Paired-QC","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"RNA-seq paired FASTQ collection","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"RNA-seq paired FASTQ collection","name":"FASTQ collection"}],"label":"FASTQ collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":50.38988268780724},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"1bb12b7e-2ee1-4d8a-80d6-4259bea1efae","when":null,"workflow_outputs":[]},"1":{"annotation":"fastp Reports","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"fastp Reports","name":"fastp Reports"}],"label":"fastp Reports","name":"Input dataset collection","outputs":[],"position":{"left":134.80355171857803,"top":181.47189808543578},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"43c69a53-b1dc-4089-9801-18e980088a2e","when":null,"workflow_outputs":[]},"2":{"annotation":"STAR log files","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"STAR log files","name":"STAR logs"}],"label":"STAR logs","name":"Input dataset collection","outputs":[],"position":{"left":218.26764632413384,"top":291.7068715859467},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"5ae91949-4ffe-4aef-81fa-fa3cb4e86f5a","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"STAR gene counts"}],"label":"STAR gene counts","name":"Input dataset collection","outputs":[],"position":{"left":282.5174775510844,"top":388.3856046156558},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"65ca0df2-1764-4605-8485-56633366ed5d","when":null,"workflow_outputs":[]},"4":{"annotation":"featureCounts Summaries file","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"featureCounts Summaries file","name":"featureCounts summaries"}],"label":"featureCounts summaries","name":"Input dataset collection","outputs":[],"position":{"left":351.6280284762066,"top":524.3280615905143},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"5a5cb076-92cb-48bd-80e0-d7bfa19fe969","when":null,"workflow_outputs":[]},"5":{"annotation":"Reference annotation","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Reference annotation","name":"reference_annotation_gtf"}],"label":"reference_annotation_gtf","name":"Input dataset","outputs":[],"position":{"left":414.3587638558393,"top":727.1586541357461},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"9657bd5f-5510-412b-9128-0bb903e2f572","when":null,"workflow_outputs":[]},"6":{"annotation":"Mapped BAM files from STAR","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Mapped BAM files from STAR","name":"STAR paired-end BAM"}],"label":"STAR paired-end BAM","name":"Input dataset collection","outputs":[],"position":{"left":413.35877588481725,"top":876.1586721395935},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"b3a7345d-49fa-4b34-bd56-565f250f6d97","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":7,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":"FastQC check read qualities","name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":639.7276714482445,"top":0},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"d4f16966-836e-49cd-8987-18b6e68205b8","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357","errors":null,"id":8,"input_connections":{"gtf_file":{"id":5,"output_name":"output"}},"inputs":[],"label":"convert gtf to bed12","name":"Convert GTF to BED12","outputs":[{"name":"bed_file","type":"bed12"}],"position":{"left":693.3587755686708,"top":712.1586721395935},"post_job_actions":{"HideDatasetActionbed_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bed_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357","tool_shed_repository":{"changeset_revision":"b026dae67fba","name":"gtftobed12","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"advanced_options_selector\\": \\"default\\", \\"__current_case__\\": 0}, \\"gtf_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"357","type":"tool","uuid":"ed8685b7-7904-4f28-8ea9-2faffb05bacf","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":9,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":"Subsample 200k reads","name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":693.3585496045475,"top":864.5092441480267},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": null, \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"target\\", \\"__current_case__\\": 1, \\"target\\": \\"200000\\", \\"seed\\": \\"1\\"}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"fa918b19-c767-43dd-afb6-fcdb3cd20fa7","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","errors":null,"id":10,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":"Get reads number per chromosome","name":"Samtools idxstats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":693.3587931207321,"top":1071.1586721395934},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","tool_shed_repository":{"changeset_revision":"fa5d3e61e429","name":"samtools_idxstats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"0c0789f8-21d9-4fdc-80c6-488aded24611","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","errors":null,"id":11,"input_connections":{"inputFile":{"id":6,"output_name":"output"}},"inputs":[],"label":"Remove duplicates","name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":691.7090825073274,"top":1221.4970533629626},"post_job_actions":{"HideDatasetActionmetrics_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"metrics_file"},"HideDatasetActionoutFile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": false, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"c8d04f50-da21-4f5b-9d64-2dc652b71d77","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/5.0.3+galaxy0","errors":null,"id":12,"input_connections":{"input":{"id":6,"output_name":"output"},"refgene":{"id":8,"output_name":"bed_file"}},"inputs":[],"label":"Read distribution over genomic features","name":"Read Distribution","outputs":[{"name":"output","type":"txt"}],"position":{"left":963.3587758848173,"top":603.1586721395935},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/5.0.3+galaxy0","tool_shed_repository":{"changeset_revision":"27e16a30667a","name":"rseqc","owner":"nilesh","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"refgene\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.3+galaxy0","type":"tool","uuid":"4ecf1a7a-8648-4224-9c04-ab5d33614d63","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/5.0.3+galaxy0","errors":null,"id":13,"input_connections":{"batch_mode|input":{"id":9,"output_name":"outputsam"},"refgene":{"id":8,"output_name":"bed_file"}},"inputs":[{"description":"runtime parameter for tool Gene Body Coverage (BAM)","name":"batch_mode"}],"label":"Get gene body coverage","name":"Gene Body Coverage (BAM)","outputs":[{"name":"outputcurvespdf","type":"pdf"},{"name":"outputtxt","type":"txt"}],"position":{"left":962.3585802454679,"top":813.1586690267329},"post_job_actions":{"HideDatasetActionoutputcurvespdf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcurvespdf"},"HideDatasetActionoutputtxt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputtxt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/5.0.3+galaxy0","tool_shed_repository":{"changeset_revision":"27e16a30667a","name":"rseqc","owner":"nilesh","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_mode\\": {\\"batch_mode_selector\\": \\"batch\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"minimum_length\\": \\"100\\", \\"refgene\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rscript_output\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.3+galaxy0","type":"tool","uuid":"c996512e-789b-422c-833c-de854e30674f","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":14,"input_connections":{"results_0|software_cond|output_0|input":{"id":7,"output_name":"text_file"},"results_1|software_cond|input":{"id":1,"output_name":"output"},"results_2|software_cond|output_0|type|input":{"id":2,"output_name":"output"},"results_2|software_cond|output_1|type|input":{"id":3,"output_name":"output"},"results_3|software_cond|input":{"id":4,"output_name":"output"},"results_4|software_cond|output_0|type|input":{"id":12,"output_name":"output"},"results_4|software_cond|output_1|type|input":{"id":13,"output_name":"outputtxt"},"results_5|software_cond|output_0|type|input":{"id":10,"output_name":"output"},"results_6|software_cond|output_0|input":{"id":11,"output_name":"metrics_file"}},"inputs":[],"label":"Combined Quality Report","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1326.7698466852496,"top":452.11475989525854},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"MultiQC HTML report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"MultiQC Stats table"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 3, \\"software_cond\\": {\\"software\\": \\"featureCounts\\", \\"__current_case__\\": 9, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"software_cond\\": {\\"software\\": \\"rseqc\\", \\"__current_case__\\": 22, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"read_distribution\\", \\"__current_case__\\": 6, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"gene_body_coverage\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 5, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"idxstats\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 6, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"title\\": \\"Combined Quality Report \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"b5e9c0d2-62f4-41b2-a1c7-5677d8555fd4","when":null,"workflow_outputs":[{"label":"MultiQC stats","output_name":"stats","uuid":"cbf99516-9349-4f2a-ad1f-05c732cb5263"},{"label":"MultiQC HTML report","output_name":"html_report","uuid":"7306cd4b-1017-4542-bf57-81d2d83d6a97"}]}},"tags":[],"uuid":"71cd8557-bf26-44c0-8449-afed4b411651"},"tool_id":null,"type":"subworkflow","uuid":"c0aaa16c-8a9b-4fe1-8c3d-9e77f64da413","when":"$(inputs.when)","workflow_outputs":[{"label":"MultiQC HTML report","output_name":"MultiQC HTML report","uuid":"717627ed-188d-4cb5-9a03-b1eec15f7922"},{"label":"MultiQC stats","output_name":"MultiQC stats","uuid":"43ed1ffc-7df8-42a6-95f7-dc2b2060a910"}]}},"tags":["RNAseq","transcriptomics"],"uuid":"5ab1d724-01d4-4673-a3c3-ae2bc944458f","version":1},"readme":"# RNA-Seq Analysis: Paired-End Read Processing and Quantification\\n\\n## Inputs dataset\\n\\n- Collection paired FASTQ files: The workflow needs a list of dataset pairs of fastqsanger.\\n- GTF file of annotation: A gtf file with genes annotation.\\n- GTF with regions to exclude from FPKM normalization with Cufflinks: Optional, but recommended. A gtf file with regions to exclude from normalization in Cufflinks.\\n\\n  - For instance a gtf that masks chrM for the mm10 genome:\\n\\n```\\nchrM\\tchrM_gene\\texon\\t0\\t16299\\t.\\t+\\t.\\tgene_id \\"chrM_gene_plus\\"; transcript_id \\"chrM_tx_plus\\"; exon_id \\"chrM_ex_plus\\";\\nchrM\\tchrM_gene\\texon\\t0\\t16299\\t.\\t-\\t.\\tgene_id \\"chrM_gene_minus\\"; transcript_id \\"chrM_tx_minus\\"; exon_id \\"chrM_ex_minus\\";\\n```\\n\\n## Inputs values\\n\\n- Forward and Reverse adapter (optional): By default, fastp will try to overlap both reads and will only use these sequences if R1/R2 are found not overlapped. Their sequences depends on the library preparation. Usually classical Illumina RNA libraries is Truseq and ISML (relatively new Illumina library) is Nextera.\\n- Generate additional QC reports: whether to compute additional QC: FastQC, Picard, Read distribution on genomic features, gene body coverage, reads per chromosomes.\\n- Reference genome: this field will be adapted to the genomes available for STAR.\\n- Strandedness: For stranded RNA, reverse means that the first read in a pair is complementary to the coding sequence, forward means that the first read in a pair is in the same orientation as the coding sequence. This will only count alignments that are compatible with your library preparation strategy. This is also used for the stranded coverage and for FPKM computation with cufflinks/StringTie.\\n- Use featureCounts for generating count tables: Whether to use count tables from featureCounts instead of from STAR.\\n- Compute Cufflinks FPKM: Whether you want to get FPKM with Cufflinks (pretty long).\\n- Compute StringTie FPKM: Whether you want to get FPKM/TPM etc... with StringTie.\\n\\n## Processing\\n\\n- The workflow will remove adapters and low quality bases and filter out any read smaller than 15bp.\\n- The filtered reads are mapped with STAR with ENCODE parameters (for long RNA-seq but I use it for short also). STAR is also used to count reads per gene and generate strand-specific normalized coverage (on uniquely mapped reads).\\n- Optionally featureCounts is used to generate count files when this option enabled.\\n- Optionally FastQC, Picard, read_distribution, geneBody_coverage, samtools idxstats, Picard are run to get additional QC.\\n- A multiQC is run to have an overview of the QC. This can also be used to get the strandedness.\\n- FPKM values for genes and transcripts are computed with cufflinks using correction for multi-mapped reads (this step is optionnal).\\n- FPKM/TPM values for genes are computed with StringTie (this step is optional).\\n- The BAM is filtered to keep only uniquely mapped reads (tag NH:i:1).\\n- Unstranded coverage is computed with bedtools and normalized to the number of million uniquely mapped reads.\\n- The three coverage files are converted to bigwig.\\n\\n### Warning\\n\\n- The coverage stranded output depends on the strandedness of the library:\\n  - If you have an unstranded library, stranded coverages are useless\\n  - If you have a forward stranded library, the label matches the orientation of the first read in pairs.\\n  - If you have a reverse stranded library, the label matches the orientation of the second read in pairs.\\n","changelog":"# Changelog\\n\\n## [1.2] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [1.1] - 2024-11-18\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.6+galaxy0`\\n\\n## [1.0] - 2024-09-23\\n\\n### Changes in workflows\\n- Add an optional subworkflow with more QC: FastQC, Picard, Read distribution on genomic features, gene body coverage, reads per chromosomes.\\n- Add featureCounts as an alternative way to generate count files\\n- Use fastp instead of cutadapt which uses pair overlap and allows to have optional adapter sequences\\n\\n### Tool update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.3` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4`\\n\\n### Test dataset\\n- Using a new subsampled Yeast test data from Zenodo record https://zenodo.org/records/13987631\\n\\n## [0.9] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.8] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.7] - 2024-06-25\\n\\n### Automatic update (triggered manually)\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0`\\n\\n## [0.6] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0`\\n\\n## [0.5] - 2023-09-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4`\\n\\n### Manual update\\n- Use STAR to compute normalized strand-specific coverage\\n- Add an option to use StringTie to compute FPKM\\n- Make cufflinks step optional\\n\\n## [0.4.1] - 2023-09-14\\n- add author in dockstore file\\n\\n## [0.4] - 2023-01-16\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1`\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-02\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy1`\\n\\n## [0.1] - 2022-10-21\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## RNA-Seq Analysis: Paired-End Read Processing and Quantification\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Collection paired FASTQ files\\"]@{ shape: docs }\\n1[\\"ℹ️ Forward adapter\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Reverse adapter\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Generate additional QC reports\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Reference genome\\"]@{ shape: lean-l }\\n5[\\"ℹ️ GTF file of annotation\\"]@{ shape: doc }\\n6[\\"ℹ️ Strandedness\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Use featureCounts for generating count tables\\"]@{ shape: lean-l }\\n8[\\"ℹ️ Compute Cufflinks FPKM\\"]@{ shape: lean-l }\\n9[\\"ℹ️ GTF with regions to exclude from FPKM normalization with Cufflinks\\"]@{ shape: doc }\\n10[\\"ℹ️ Compute StringTie FPKM\\"]@{ shape: lean-l }\\n11[\\"Flatten collection\\"]@{ shape: process }\\n0 --> 11\\n12[\\"remove adapters + bad quality bases\\"]@{ shape: process }\\n1 --> 12\\n2 --> 12\\n0 --> 12\\n13[\\"no additional QC\\"]@{ shape: process }\\n3 --> 13\\n14[\\"get reference_genome as text parameter\\"]@{ shape: process }\\n4 --> 14\\n15[\\"Get featureCounts strandedness parameter\\"]@{ shape: process }\\n6 --> 15\\n16[\\"Get cufflinks strandedness parameter\\"]@{ shape: process }\\n6 --> 16\\n17[\\"Get Stringtie strandedness parameter\\"]@{ shape: process }\\n6 --> 17\\n18[\\"STAR: map and count and coverage splitted\\"]@{ shape: process }\\n4 --> 18\\n5 --> 18\\n12 --> 18\\n19[\\"🛠️ Generate Unstranded Coverage\\"]@{ shape: subprocess }\\n18 --> 19\\n18 --> 19\\n20[\\"🛠️ Generate Stranded Coverage\\"]@{ shape: subprocess }\\n18 --> 20\\n18 --> 20\\n6 --> 20\\n21[\\"featureCounts\\"]@{ shape: process }\\n18 --> 21\\n5 --> 21\\n15 --> 21\\n7 --> 21\\n22[\\"Compute FPKM with StringTie\\"]@{ shape: process }\\n5 --> 22\\n18 --> 22\\n17 --> 22\\n10 --> 22\\n23[\\"Compute FPKM with cufflinks\\"]@{ shape: process }\\n16 --> 23\\n9 --> 23\\n14 --> 23\\n18 --> 23\\n5 --> 23\\n8 --> 23\\n24[\\"🛠️ Process Count files\\"]@{ shape: subprocess }\\n18 --> 24\\n6 --> 24\\n21 --> 24\\n21 --> 24\\n25[\\"Combined MultiQC without additional QC\\"]@{ shape: process }\\n12 --> 25\\n18 --> 25\\n18 --> 25\\n24 --> 25\\n13 --> 25\\n26[\\"🛠️ Combined MultiQC Quality Report with additional QC\\"]@{ shape: subprocess }\\n11 --> 26\\n18 --> 26\\n18 --> 26\\n18 --> 26\\n12 --> 26\\n24 --> 26\\n5 --> 26\\n3 --> 26\\n```\\n\\n## RNA-seq-Paired-QC\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ FASTQ collection\\"]@{ shape: docs }\\n1[\\"ℹ️ fastp Reports\\"]@{ shape: docs }\\n2[\\"ℹ️ STAR logs\\"]@{ shape: docs }\\n3[\\"ℹ️ STAR gene counts\\"]@{ shape: docs }\\n4[\\"ℹ️ featureCounts summaries\\"]@{ shape: docs }\\n5[\\"ℹ️ reference_annotation_gtf\\"]@{ shape: doc }\\n6[\\"ℹ️ STAR paired-end BAM\\"]@{ shape: docs }\\n7[\\"FastQC check read qualities\\"]@{ shape: process }\\n0 --> 7\\n8[\\"convert gtf to bed12\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Subsample 200k reads\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Get reads number per chromosome\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Remove duplicates\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Read distribution over genomic features\\"]@{ shape: process }\\n6 --> 12\\n8 --> 12\\n13[\\"Get gene body coverage\\"]@{ shape: process }\\n9 --> 13\\n8 --> 13\\n14[\\"Combined Quality Report\\"]@{ shape: process }\\n7 --> 14\\n1 --> 14\\n2 --> 14\\n3 --> 14\\n4 --> 14\\n12 --> 14\\n13 --> 14\\n10 --> 14\\n11 --> 14\\n```\\n\\n## Process Count files\\n\\n```mermaid\\ngraph LR\\n0[\\"A dummy featureCounts summary file\\"]@{ shape: process }\\n1[\\"ℹ️ featureCounts summaries collection\\"]@{ shape: docs }\\n2[\\"ℹ️ Strandness param\\"]@{ shape: lean-l }\\n3[\\"ℹ️ RNA STAR count tables\\"]@{ shape: docs }\\n4[\\"ℹ️ featureCounts count table\\"]@{ shape: docs }\\n5[\\"Text transformation\\"]@{ shape: process }\\n0 --> 5\\n6[\\"awk command from strand for STAR counts\\"]@{ shape: process }\\n2 --> 6\\n7[\\"featureCounts summaries\\"]@{ shape: process }\\n1 --> 7\\n5 --> 7\\n8[\\"Extract gene counts\\"]@{ shape: process }\\n6 --> 8\\n3 --> 8\\n9[\\"Counts table\\"]@{ shape: process }\\n4 --> 9\\n8 --> 9\\n```\\n\\n## Re-arrange Stranded RNA-seq coverage\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ strandedness\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Bedgraph strand 1\\"]@{ shape: docs }\\n2[\\"ℹ️ Bedgraph strand 2\\"]@{ shape: docs }\\n3[\\"Get replacement for strand2\\"]@{ shape: process }\\n0 --> 3\\n4[\\"Get replacement for strand1\\"]@{ shape: process }\\n0 --> 4\\n5[\\"get identifiers\\"]@{ shape: process }\\n1 --> 5\\n6[\\"New labels strand 2\\"]@{ shape: process }\\n3 --> 6\\n5 --> 6\\n7[\\"New labels strand 1\\"]@{ shape: process }\\n4 --> 7\\n5 --> 7\\n8[\\"Relabelled strand 2\\"]@{ shape: process }\\n6 --> 8\\n2 --> 8\\n9[\\"Relabelled strand 1\\"]@{ shape: process }\\n7 --> 9\\n1 --> 9\\n10[\\"Merge collections\\"]@{ shape: process }\\n9 --> 10\\n8 --> 10\\n11[\\"convert to bigwig\\"]@{ shape: process }\\n10 --> 11\\n```\\n\\n## Get Uniquely mapped unstranded coverage\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ STAR log\\"]@{ shape: docs }\\n1[\\"ℹ️ STAR BAM\\"]@{ shape: docs }\\n2[\\"get scaling factor\\"]@{ shape: process }\\n0 --> 2\\n3[\\"keep uniquely mapped reads\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Scaled Coverage both strands combined\\"]@{ shape: process }\\n3 --> 5\\n4 --> 5\\n6[\\"convert both strands coverage to bigwig\\"]@{ shape: process }\\n5 --> 6\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/rnaseq-pe/main","dockstore_id":19869,"categories":[],"collections":["Transcriptomics"],"doi":"10.5281/zenodo.8354569","tests":[{"doc":"Test outline for RNAseq_PE","job":{"GTF file of annotation":{"class":"File","location":"https://zenodo.org/records/13987631/files/Saccharomyces_cerevisiae.R64-1-1.113.gtf","filetype":"gtf"},"Collection paired FASTQ files":{"class":"Collection","collection_type":"list:paired","elements":[{"class":"Collection","type":"paired","identifier":"SRR5085167","elements":[{"class":"File","identifier":"forward","location":"https://zenodo.org/records/13987631/files/SRR5085167_forward.fastqsanger.gz"},{"class":"File","identifier":"reverse","location":"https://zenodo.org/records/13987631/files/SRR5085167_reverse.fastqsanger.gz"}]}]},"Forward adapter":"AGATCGGAAGAG","Reverse adapter":"GATCGTCGGACT","Generate additional QC reports":true,"Reference genome":"sacCer3","Use featureCounts for generating count tables":true,"Strandedness":"stranded - forward","Compute Cufflinks FPKM":true,"GTF with regions to exclude from FPKM normalization with Cufflinks":null,"Compute StringTie FPKM":true},"outputs":{"MultiQC stats":{"asserts":[{"that":"has_text_matching","expression":"SRR5085167\\t0.10[0-9]*\\t16.45[0-9]*\\t43.38[0-9]*\\t0.32[0-9]*\\t0.30[0-9]*\\t93.75\\t0.11[0-9]*\\t34.29\\t0.19[0-9]*\\t17.6[0-9]*\\t91.8[0-9]*\\t30.0[0-9]*\\t0.64[0-9]*\\t43.5[0-9]*\\t81.0[0-9]*\\t72.6[0-9]*\\t"},{"that":"has_text_matching","expression":"SRR5085167_forward\\t*36.33[0-9]*\\t46.0\\t75.0\\t75\\t27.27[0-9]*\\t0.39[0-9]*"},{"that":"has_text_matching","expression":"SRR5085167_reverse\\t*35.31[0-9]*\\t46.0\\t75.0\\t75\\t45.45[0-9]*\\t0.39[0-9]*"}]},"Stranded Coverage":{"element_tests":{"SRR5085167_forward":{"asserts":{"has_size":{"value":635210,"delta":30000}}},"SRR5085167_reverse":{"asserts":{"has_size":{"value":618578,"delta":30000}}}}},"Gene Abundance Estimates from StringTie":{"element_tests":{"SRR5085167":{"asserts":{"has_text_matching":{"expression":"YAL038W\\tCDC19\\tchrI\\t\\\\+\\t71786\\t73288\\t92.5[0-9]*\\t3273.9[0-9]*\\t3900.[0-9]*"}}}}},"Unstranded Coverage":{"element_tests":{"SRR5085167":{"asserts":{"has_size":{"value":1140004,"delta":50000}}}}},"Mapped Reads":{"element_tests":{"SRR5085167":{"asserts":{"has_size":{"value":56913572,"delta":2500000}}}}},"Genes Expression from Cufflinks":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W\\t-\\t-\\tYAL038W\\tCDC19\\t-\\tchrI:71785-73288\\t-\\t-\\t3437.1\\t3211.44\\t3662.76\\tOK"}}}}},"Transcripts Expression from Cufflinks":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W_mRNA\\t-\\t-\\tYAL038W\\tCDC19\\t-\\tchrI:71785-73288\\t1503\\t102.837\\t3437.1\\t3211.44\\t3662.76\\tOK"}}}}},"Counts Table":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W\\t1591"}}}}}}}]}],"path":"./workflows/transcriptomics/rnaseq-pe"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/rnaseq-sr.ga","testParameterFiles":["/rnaseq-sr-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"},{"name":"Pavankumar Videm","orcid":"0000-0002-5192-126X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Complete RNA-Seq analysis for single-end data:  Processes raw FASTQ data through adapter and bad quality removal (fastp), alignment with STAR using ENCODE parameters, gene quantification via multiple methods (STAR and featureCounts), and expression calculation (FPKM with Cufflinks/StringTie, normalized coverage with bedtools). Produces count tables, normalized expression values, and genomic coverage tracks. Supports stranded and unstranded libraries, generating both HTSeq-compatible counts and normalized measures for downstream analysis.","comments":[{"child_steps":[12,20,21],"color":"yellow","data":{"title":"Additional quantification"},"id":2,"position":[1391.2086321940171,1073.6999999999998],"size":[860,716],"type":"frame"},{"child_steps":[17,18],"color":"red","data":{"title":"Coverage Files"},"id":1,"position":[1603.68749985966,0],"size":[375,538],"type":"frame"},{"child_steps":[13,14,15],"color":"lime","data":{"title":"Map Strandedness parameter"},"id":0,"position":[855.5086321940169,1252.5],"size":[251,558],"type":"frame"}],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","release":"1.2","name":"RNA-Seq Analysis: Single-End Read Processing and Quantification","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Should be a list of single-read RNA-seq fastqs","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be a list of single-read RNA-seq fastqs","name":"Collection of FASTQ files"}],"label":"Collection of FASTQ files","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":462.87500890145213},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"020f7580-f647-471f-b210-dbf9dfcf80bf","when":null,"workflow_outputs":[]},"1":{"annotation":"This is optional. If not provided, fastp will try to guess the sequence. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC, for TruSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"This is optional. If not provided, fastp will try to guess the sequence. If you want to specify, for Nextera use: CTGTCTCTTATACACATCTCCGAGCCCACGAGAC, for TruSeq: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC or AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC","name":"Forward adapter"}],"label":"Forward adapter","name":"Input parameter","outputs":[],"position":{"left":35.17186494140515,"top":567.328103383874},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"6d1de206-97c4-41ba-8702-a980f748a689","when":null,"workflow_outputs":[]},"2":{"annotation":"Whether to compute additional QC like fastQC, gene body coverage etc...","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Whether to compute additional QC like fastQC, gene body coverage etc...","name":"Generate additional QC reports"}],"label":"Generate additional QC reports","name":"Input parameter","outputs":[],"position":{"left":114.671875,"top":764.5781339014521},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"bea331ce-d6fa-4312-8dce-a8b13bc4e77b","when":null,"workflow_outputs":[]},"3":{"annotation":"Select the reference genome","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Select the reference genome","name":"Reference genome"}],"label":"Reference genome","name":"Input parameter","outputs":[],"position":{"left":157.109375,"top":870.218728383874},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"a72b29ae-65a7-4e4d-a996-2943ab5b02aa","when":null,"workflow_outputs":[]},"4":{"annotation":"GTF compatible with the reference genome. Mind the UCSC/Ensembl differences in chromosome naming.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"GTF compatible with the reference genome. Mind the UCSC/Ensembl differences in chromosome naming.","name":"GTF file of annotation"}],"label":"GTF file of annotation","name":"Input dataset","outputs":[],"position":{"left":199.671875,"top":977.3906033838739},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"4987fd01-b72c-4bc0-9d49-a840a0d04b8e","when":null,"workflow_outputs":[]},"5":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"Strandedness"}],"label":"Strandedness","name":"Input parameter","outputs":[],"position":{"left":251.890625,"top":1073.781228383874},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"2ccf2242-dddd-45b6-8cea-746d9b5d49bc","when":null,"workflow_outputs":[]},"6":{"annotation":"Use featureCounts tool instead of RNA STAR?","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Use featureCounts tool instead of RNA STAR?","name":"Use featureCounts for generating count tables"}],"label":"Use featureCounts for generating count tables","name":"Input parameter","outputs":[],"position":{"left":291.34375,"top":1173.124978383874},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f43d45e6-5692-41cb-8cfc-205dbeab1bd0","when":null,"workflow_outputs":[]},"7":{"annotation":"Whether FPKM values should be computed with Cufflinks","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"Whether FPKM values should be computed with Cufflinks","name":"Compute Cufflinks FPKM"}],"label":"Compute Cufflinks FPKM","name":"Input parameter","outputs":[],"position":{"left":346.83340121591186,"top":1309.3879390873915},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"78868cdb-baff-4388-a41a-a700259da1ed","when":null,"workflow_outputs":[]},"8":{"annotation":"It could be a GTF with for example one entry for the chrM forward and one entry for the chrM reverse","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"It could be a GTF with for example one entry for the chrM forward and one entry for the chrM reverse","name":"GTF with regions to exclude from FPKM normalization with Cufflinks"}],"label":"GTF with regions to exclude from FPKM normalization with Cufflinks","name":"Input dataset","outputs":[],"position":{"left":380.53912206977975,"top":1425.307227138327},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"0d4e1623-adc9-4ff3-8156-7e1322b1e278","when":null,"workflow_outputs":[]},"9":{"annotation":"Whether FPKM values should be computed with StringTie","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"Whether FPKM values should be computed with StringTie","name":"Compute StringTie FPKM"}],"label":"Compute StringTie FPKM","name":"Input parameter","outputs":[],"position":{"left":406.9259176995931,"top":1559.4856968335116},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"056f11f4-bdd9-4361-b037-2c0822320cc2","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","errors":null,"id":10,"input_connections":{"single_paired|adapter_trimming_options|adapter_sequence1":{"id":1,"output_name":"output"},"single_paired|in1":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":"remove adapters + bad quality bases","name":"fastp","outputs":[{"name":"out1","type":"input"},{"name":"report_json","type":"json"}],"position":{"left":377,"top":106.21400000000001},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"},"HideDatasetActionreport_json":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","tool_shed_repository":{"changeset_revision":"a626e8c0e1ba","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": \\"30\\", \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": \\"15\\", \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": false, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": null, \\"umi_len\\": null, \\"umi_prefix\\": null}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"in1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy3","type":"tool","uuid":"e31141cc-c8a5-4dde-882f-72c1b8aabe97","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":11,"input_connections":{"input_param_type|input_param":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"no additional QC","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":2139.879611894974,"top":184.34956182248365},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"boolean\\", \\"__current_case__\\": 3, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": false, \\"to\\": \\"true\\"}, {\\"__index__\\": 1, \\"from\\": true, \\"to\\": \\"false\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"input\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"68188ffa-f288-4da4-8d29-1f3d3b689937","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":12,"input_connections":{"components_0|param_type|component_value":{"id":3,"output_name":"output"}},"inputs":[],"label":"get reference_genome as text parameter","name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1497.7951400347756,"top":1173.104961480357},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"bcceb886-0592-4b3f-8b39-26fe79caeb62","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":13,"input_connections":{"input_param_type|input_param":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get featureCounts strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":880.1168847150709,"top":1303.1637555172579},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"1\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"2\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"0\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"72d7d4d1-d1be-481c-b2a1-1e85192b2661","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":14,"input_connections":{"input_param_type|input_param":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get cufflinks strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":880.334534933914,"top":1483.3515528952898},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"fr-secondstrand\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"fr-firststrand\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"fr-unstranded\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"f027f3d5-3b91-4061-b5cd-78553509ce5e","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":15,"input_connections":{"input_param_type|input_param":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get Stringtie strandedness parameter","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":875.5945006269037,"top":1653.5777444451405},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"--fr\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"--rf\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"63e6501b-e1de-4184-af42-720cc9a6dfb0","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1","errors":null,"id":16,"input_connections":{"refGenomeSource|GTFconditional|genomeDir":{"id":3,"output_name":"output"},"refGenomeSource|GTFconditional|sjdbGTFfile":{"id":4,"output_name":"output"},"singlePaired|input1":{"id":10,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool RNA STAR","name":"singlePaired"}],"label":"STAR: map and count and coverage splitted","name":"RNA STAR","outputs":[{"name":"output_log","type":"txt"},{"name":"splice_junctions","type":"interval"},{"name":"mapped_reads","type":"bam"},{"name":"reads_per_gene","type":"tabular"},{"name":"signal_unique_str1","type":"bedgraph"},{"name":"signal_uniquemultiple_str1","type":"bedgraph"},{"name":"signal_unique_str2","type":"bedgraph"},{"name":"signal_uniquemultiple_str2","type":"bedgraph"}],"position":{"left":732.1609477893844,"top":246.8799136582777},"post_job_actions":{"HideDatasetActionoutput_log":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_log"},"HideDatasetActionreads_per_gene":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"reads_per_gene"},"HideDatasetActionsignal_unique_str1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_unique_str1"},"HideDatasetActionsignal_unique_str2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_unique_str2"},"HideDatasetActionsignal_uniquemultiple_str1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_uniquemultiple_str1"},"HideDatasetActionsignal_uniquemultiple_str2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"signal_uniquemultiple_str2"},"HideDatasetActionsplice_junctions":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"splice_junctions"},"RenameDatasetActionmapped_reads":{"action_arguments":{"newname":"Mapped Reads"},"action_type":"RenameDatasetAction","output_name":"mapped_reads"},"RenameDatasetActionreads_per_gene":{"action_arguments":{"newname":"Reads per gene from STAR"},"action_type":"RenameDatasetAction","output_name":"reads_per_gene"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1","tool_shed_repository":{"changeset_revision":"53255f6eecfc","name":"rgrnastar","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"algo\\": {\\"params\\": {\\"settingsType\\": \\"full\\", \\"__current_case__\\": 3, \\"seed\\": {\\"seedSearchStartLmax\\": \\"50\\", \\"seedSearchStartLmaxOverLread\\": \\"1.0\\", \\"seedSearchLmax\\": \\"0\\", \\"seedMultimapNmax\\": \\"10000\\", \\"seedPerReadNmax\\": \\"1000\\", \\"seedPerWindowNmax\\": \\"50\\", \\"seedNoneLociPerWindow\\": \\"10\\"}, \\"align\\": {\\"alignIntronMin\\": \\"20\\", \\"alignIntronMax\\": \\"1000000\\", \\"alignMatesGapMax\\": \\"1000000\\", \\"alignSJoverhangMin\\": \\"8\\", \\"alignSJstitchMismatchNmax\\": {\\"alignSJstitchMismatchNmax1\\": \\"0\\", \\"alignSJstitchMismatchNmax2\\": \\"-1\\", \\"alignSJstitchMismatchNmax3\\": \\"0\\", \\"alignSJstitchMismatchNmax4\\": \\"0\\"}, \\"alignSJDBoverhangMin\\": \\"1\\", \\"alignSplicedMateMapLmin\\": \\"0\\", \\"alignSplicedMateMapLminOverLmate\\": \\"0.66\\", \\"alignWindowsPerReadNmax\\": \\"10000\\", \\"alignTranscriptsPerWindowNmax\\": \\"100\\", \\"alignTranscriptsPerReadNmax\\": \\"10000\\", \\"alignEndsType\\": \\"Local\\", \\"peOverlapNbasesMin\\": \\"0\\", \\"peOverlapMMp\\": \\"0.01\\"}, \\"chim_settings\\": {\\"chimSegmentMin\\": \\"12\\", \\"chimScoreMin\\": \\"0\\", \\"chimScoreDropMax\\": \\"20\\", \\"chimScoreSeparation\\": \\"10\\", \\"chimScoreJunctionNonGTAG\\": \\"-1\\", \\"chimJunctionOverhangMin\\": \\"20\\", \\"chimSegmentReadGapMax\\": \\"0\\", \\"chimFilter\\": true, \\"chimMainSegmentMultNmax\\": \\"10\\", \\"chimMultimapNmax\\": \\"1\\", \\"chimMultimapScoreRange\\": \\"1\\"}, \\"junction_limits\\": {\\"limitOutSJoneRead\\": \\"1000\\", \\"limitOutSJcollapsed\\": \\"1000000\\", \\"limitSjdbInsertNsj\\": \\"1000000\\"}}}, \\"chimOutType\\": \\"\\", \\"filter\\": {\\"basic_filters\\": [\\"exclude_unmapped\\"], \\"output_params2\\": {\\"output_select2\\": \\"yes\\", \\"__current_case__\\": 0, \\"outFilterType\\": true, \\"outFilterMultimapScoreRange\\": \\"1\\", \\"outFilterMultimapNmax\\": \\"20\\", \\"outFilterMismatchNmax\\": \\"999\\", \\"outFilterMismatchNoverLmax\\": \\"0.3\\", \\"outFilterMismatchNoverReadLmax\\": \\"0.04\\", \\"outFilterScoreMin\\": \\"0\\", \\"outFilterScoreMinOverLread\\": \\"0.66\\", \\"outFilterMatchNmin\\": \\"0\\", \\"outFilterMatchNminOverLread\\": \\"0.66\\", \\"outSAMmultNmax\\": \\"-1\\", \\"outSAMtlen\\": \\"1\\"}}, \\"oformat\\": {\\"outSAMattributes\\": [\\"NH\\", \\"HI\\", \\"AS\\", \\"nM\\"], \\"HI_offset\\": \\"1\\", \\"outSAMprimaryFlag\\": \\"OneBestScore\\", \\"outSAMmapqUnique\\": \\"255\\", \\"wasp_conditional\\": {\\"waspOutputMode\\": \\"\\", \\"__current_case__\\": 1}}, \\"outWig\\": {\\"outWigType\\": \\"bedGraph\\", \\"__current_case__\\": 1, \\"outWigTypeSecondWord\\": \\"\\", \\"outWigStrand\\": true, \\"outWigReferencesPrefix\\": \\"-\\", \\"outWigNorm\\": true}, \\"perf\\": {\\"outBAMsortingBinsN\\": \\"50\\", \\"winAnchorMultimapNmax\\": \\"50\\"}, \\"refGenomeSource\\": {\\"geneSource\\": \\"indexed\\", \\"__current_case__\\": 0, \\"GTFconditional\\": {\\"GTFselect\\": \\"without-gtf-with-gtf\\", \\"__current_case__\\": 1, \\"genomeDir\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sjdbGTFfeatureExon\\": \\"exon\\", \\"sjdbOverhang\\": \\"100\\", \\"quantmode_output\\": {\\"quantMode\\": \\"GeneCounts\\", \\"__current_case__\\": 1}}}, \\"singlePaired\\": {\\"sPaired\\": \\"single\\", \\"__current_case__\\": 0, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"twopass\\": {\\"twopassMode\\": \\"None\\", \\"__current_case__\\": 0, \\"twopass_read_subset\\": \\"\\", \\"sj_precalculated\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.11a+galaxy1","type":"tool","uuid":"be1a277d-6d0d-4aaf-acea-9735426932b0","when":null,"workflow_outputs":[{"label":"Mapped Reads","output_name":"mapped_reads","uuid":"f5722954-fd5c-4e66-aed0-4cd099dd9b1c"}]},"17":{"annotation":"","id":17,"input_connections":{"STAR BAM":{"id":16,"input_subworkflow_step_id":1,"output_name":"mapped_reads"},"STAR log":{"id":16,"input_subworkflow_step_id":0,"output_name":"output_log"}},"inputs":[{"description":"","name":"Generate Unstranded Coverage"}],"label":"Generate Unstranded Coverage","name":"Get Uniquely mapped unstranded coverage","outputs":[],"position":{"left":1683.2031849964808,"top":86.70584740647254},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Get Uniquely mapped unstranded coverage","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"RNA STAR log files","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"RNA STAR log files","name":"STAR log"}],"label":"STAR log","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":46},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"3e3f7463-a635-44a2-a1f4-7e12aae3d25b","when":null,"workflow_outputs":[]},"1":{"annotation":"Mapped BAM files generate from RNA STAR","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Mapped BAM files generate from RNA STAR","name":"STAR BAM"}],"label":"STAR BAM","name":"Input dataset collection","outputs":[],"position":{"left":280,"top":209},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"10dddc85-d3fc-4cba-a637-0c40153d284f","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":2,"input_connections":{"infile":{"id":0,"output_name":"output"}},"inputs":[],"label":"get scaling factor","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":280,"top":20},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"$0~\\\\\\"Uniquely mapped reads number\\\\\\"{print 1000000/$NF}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"77cf2304-36bf-4337-bc2d-bc4d8b831668","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","errors":null,"id":3,"input_connections":{"input_bam":{"id":1,"output_name":"output"}},"inputs":[],"label":"keep uniquely mapped reads","name":"Filter BAM","outputs":[{"name":"out_file2","type":"txt"},{"name":"out_file1","type":"bam"}],"position":{"left":560,"top":195},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"HideDatasetActionout_file2":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2","tool_shed_repository":{"changeset_revision":"993b19f20c76","name":"bamtools_filter","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"conditions\\": [{\\"__index__\\": 0, \\"filters\\": [{\\"__index__\\": 0, \\"bam_property\\": {\\"bam_property_selector\\": \\"tag\\", \\"__current_case__\\": 21, \\"bam_property_value\\": \\"NH=1\\"}}]}], \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rule_configuration\\": {\\"rules_selector\\": \\"false\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.2+galaxy2","type":"tool","uuid":"59db7088-85bf-42ba-9dcd-76836f9d6310","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":4,"input_connections":{"input1":{"id":2,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":560,"top":0},"post_job_actions":{"HideDatasetActionfloat_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"float_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"float\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"83e6d856-5809-419a-b4bc-639662a95751","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","errors":null,"id":5,"input_connections":{"input_type|input":{"id":3,"output_name":"out_file1"},"report|scale":{"id":4,"output_name":"float_param"}},"inputs":[{"description":"runtime parameter for tool bedtools Genome Coverage","name":"input_type"},{"description":"runtime parameter for tool bedtools Genome Coverage","name":"report"}],"label":"Scaled Coverage both strands combined","name":"bedtools Genome Coverage","outputs":[{"name":"output","type":"bedgraph"}],"position":{"left":840,"top":12},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1","tool_shed_repository":{"changeset_revision":"64e2edfe7a2c","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"d\\": false, \\"dz\\": false, \\"five\\": false, \\"input_type\\": {\\"input_type_select\\": \\"bam\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"report\\": {\\"report_select\\": \\"bg\\", \\"__current_case__\\": 0, \\"zero_regions\\": false, \\"scale\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split\\": true, \\"strand\\": \\"\\", \\"three\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.1","type":"tool","uuid":"df53e789-be2d-4b09-be8d-a07a35c14913","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":6,"input_connections":{"input1":{"id":5,"output_name":"output"}},"inputs":[],"label":"convert both strands coverage to bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1120,"top":67},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Both Strands Coverage"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"62d76728-f614-48c0-9560-3abd05b8366d","when":null,"workflow_outputs":[{"label":"Both Strands Coverage","output_name":"out_file1","uuid":"73763ad9-c132-4764-ada6-eece0c30e74f"}]}},"tags":[],"uuid":"025c5a41-6e1e-4480-8165-c01f8c828a9f"},"tool_id":null,"type":"subworkflow","uuid":"6ce6314b-d339-40bf-9f0e-53ca525204be","when":null,"workflow_outputs":[{"label":"Unstranded Coverage","output_name":"Both Strands Coverage","uuid":"f90cc885-438b-49ee-b1d1-8ed45f7dea5c"}]},"18":{"annotation":"","id":18,"input_connections":{"Bedgraph strand 1":{"id":16,"input_subworkflow_step_id":1,"output_name":"signal_unique_str1"},"Bedgraph strand 2":{"id":16,"input_subworkflow_step_id":2,"output_name":"signal_unique_str2"},"strandedness":{"id":5,"input_subworkflow_step_id":0,"output_name":"output"}},"inputs":[{"description":"","name":"Generate Stranded Coverage"}],"label":"Generate Stranded Coverage","name":"Re-arrange Stranded RNA-seq coverage","outputs":[],"position":{"left":1692.210489743303,"top":282.0471550654421},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"MIT","name":"Re-arrange Stranded RNA-seq coverage","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"strandedness"}],"label":"strandedness","name":"Input parameter","outputs":[],"position":{"left":0,"top":370},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"56955f25-7e9e-4201-a056-b61f47359c84","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d363c181-7f37-45ef-9732-86aaf879478a"}]},"1":{"annotation":"From STAR strand 1","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"From STAR strand 1","name":"Bedgraph strand 1"}],"label":"Bedgraph strand 1","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":472},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bedgraph\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"7919f2ed-554a-4e08-93eb-6896a2195293","when":null,"workflow_outputs":[]},"2":{"annotation":"From STAR strand 2","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"From STAR strand 2","name":"Bedgraph strand 2"}],"label":"Bedgraph strand 2","name":"Input dataset collection","outputs":[],"position":{"left":569.999952609169,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bedgraph\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9318223c-4443-4341-8448-11df606ccfdf","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":3,"input_connections":{"input_param_type|input_param":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get replacement for strand2","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":289.9999763045845,"top":51},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"$1_reverse\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"$1_forward\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"$1_reverse\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"2ec50fce-ef76-4f56-9fc6-5c6b34c6593e","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":4,"input_connections":{"input_param_type|input_param":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Get replacement for strand1","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":289.9999763045845,"top":359},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"},"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"$1_forward\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"$1_reverse\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"$1_forward\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"b7e2cedf-e846-4553-9e48-4aa9a93d40b2","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":5,"input_connections":{"input_collection":{"id":1,"output_name":"output"}},"inputs":[],"label":"get identifiers","name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":289.9999763045845,"top":225},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"7d866aad-dd6c-4d17-bf6d-8f0026652265","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":6,"input_connections":{"find_and_replace_0|replace_pattern":{"id":3,"output_name":"output_param_text"},"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"New labels strand 2","name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":569.999952609169,"top":123},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"ec10a98b-adb9-4037-9137-1717068d7157","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","errors":null,"id":7,"input_connections":{"find_and_replace_0|replace_pattern":{"id":4,"output_name":"output_param_text"},"infile":{"id":5,"output_name":"output"}},"inputs":[],"label":"New labels strand 1","name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":569.999952609169,"top":328},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"line\\", \\"__current_case__\\": 0}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"4eb0d67c-f53c-4f84-90fc-3db41e4f0aa7","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":8,"input_connections":{"how|labels":{"id":6,"output_name":"outfile"},"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabelled strand 2","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":849.9998437853558,"top":106},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"147ccbb0-b951-4851-9bf5-2c6b439ac1b6","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"__RELABEL_FROM_FILE__","errors":null,"id":9,"input_connections":{"how|labels":{"id":7,"output_name":"outfile"},"input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Relabel identifiers","name":"how"}],"label":"Relabelled strand 1","name":"Relabel identifiers","outputs":[{"name":"output","type":"input"}],"position":{"left":849.9999005376209,"top":365},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__RELABEL_FROM_FILE__","tool_state":"{\\"how\\": {\\"how_select\\": \\"txt\\", \\"__current_case__\\": 0, \\"labels\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"strict\\": false}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"e9dcd2df-2834-4e9a-bef2-7666de012b8a","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__MERGE_COLLECTION__","errors":null,"id":10,"input_connections":{"inputs_0|input":{"id":9,"output_name":"output"},"inputs_1|input":{"id":8,"output_name":"output"}},"inputs":[],"label":null,"name":"Merge collections","outputs":[{"name":"output","type":"input"}],"position":{"left":1129.9999903467356,"top":120},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__MERGE_COLLECTION__","tool_state":"{\\"advanced\\": {\\"conflict\\": {\\"duplicate_options\\": \\"keep_first\\", \\"__current_case__\\": 3}}, \\"inputs\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"f47d7381-13b2-4615-b947-17a0ac2ad500","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"wig_to_bigWig","errors":null,"id":11,"input_connections":{"input1":{"id":10,"output_name":"output"}},"inputs":[],"label":"convert to bigwig","name":"Wig/BedGraph-to-bigWig","outputs":[{"name":"out_file1","type":"bigwig"}],"position":{"left":1410.000136908115,"top":171},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"uniquely mapped stranded coverage"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"wig_to_bigWig","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"settingsType\\": \\"preset\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"852f2d23-b48e-44d0-aaf2-aed33532d98a","when":null,"workflow_outputs":[{"label":"Uniquely Mapped Stranded Coverage","output_name":"out_file1","uuid":"0a31ff80-b3d1-4959-869f-1a6a3c627dde"}]}},"tags":[],"uuid":"8e66ff06-6d90-4091-a7bc-801711c5b59b"},"tool_id":null,"type":"subworkflow","uuid":"bf990b24-1ec8-4e67-acec-fecf26a3d49f","when":null,"workflow_outputs":[{"label":"Stranded Coverage","output_name":"Uniquely Mapped Stranded Coverage","uuid":"9161f772-6904-4479-aff4-566295f5b3b2"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0","errors":null,"id":19,"input_connections":{"alignment":{"id":16,"output_name":"mapped_reads"},"anno|reference_gene_sets":{"id":4,"output_name":"output"},"strand_specificity":{"id":13,"output_name":"output_param_text"},"when":{"id":6,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool featureCounts","name":"anno"}],"label":null,"name":"featureCounts","outputs":[{"name":"output_short","type":"tabular"},{"name":"output_summary","type":"tabular"}],"position":{"left":1713.913195680125,"top":672.9994811085061},"post_job_actions":{"HideDatasetActionoutput_short":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_short"},"HideDatasetActionoutput_summary":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_summary"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0","tool_shed_repository":{"changeset_revision":"5dbff3a5845a","name":"featurecounts","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"anno\\": {\\"anno_select\\": \\"history\\", \\"__current_case__\\": 2, \\"reference_gene_sets\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gff_feature_type\\": \\"exon\\", \\"gff_feature_attribute\\": \\"gene_id\\", \\"summarization_level\\": false}, \\"extended_parameters\\": {\\"multifeatures\\": {\\"multifeat\\": \\"\\", \\"__current_case__\\": 0}, \\"exon_exon_junction_read_counting_enabled\\": {\\"count_exon_exon_junction_reads\\": \\"\\", \\"__current_case__\\": 1}, \\"long_reads\\": false, \\"by_read_group\\": false, \\"largest_overlap\\": false, \\"min_overlap\\": \\"1\\", \\"frac_overlap\\": \\"0\\", \\"frac_overlap_feature\\": \\"0\\", \\"read_extension_5p\\": \\"0\\", \\"read_extension_3p\\": \\"0\\", \\"read_reduction\\": \\"\\", \\"R\\": false}, \\"format\\": \\"tabdel_short\\", \\"include_feature_length_file\\": false, \\"pe_parameters\\": {\\"paired_end_status\\": \\"single_end\\", \\"__current_case__\\": 0}, \\"read_filtering_parameters\\": {\\"mapping_quality\\": \\"0\\", \\"splitonly\\": \\"\\", \\"primary\\": false, \\"ignore_dup\\": false}, \\"strand_specificity\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.8+galaxy0","type":"tool","uuid":"5a03b546-eb3f-4e99-b845-24c8f54ed183","when":"$(inputs.when)","workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","errors":null,"id":20,"input_connections":{"guide|guide_source|ref_hist":{"id":4,"output_name":"output"},"input_options|input_bam":{"id":16,"output_name":"mapped_reads"},"rna_strandness":{"id":15,"output_name":"output_param_text"},"when":{"id":9,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"input_options"}],"label":"Compute FPKM with StringTie","name":"StringTie","outputs":[{"name":"output_gtf","type":"gtf"},{"name":"gene_abundance_estimation","type":"tabular"}],"position":{"left":1731.7359886400766,"top":1424.2142641508385},"post_job_actions":{"HideDatasetActionoutput_gtf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_gtf"},"RenameDatasetActiongene_abundance_estimation":{"action_arguments":{"newname":"Gene abundance estimates from StringTie"},"action_type":"RenameDatasetAction","output_name":"gene_abundance_estimation"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","tool_shed_repository":{"changeset_revision":"cbf488da3b2c","name":"stringtie","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"abundance_estimation\\": true, \\"omit_sequences\\": \\"\\", \\"name_prefix\\": null, \\"fraction\\": \\"0.01\\", \\"min_tlen\\": \\"200\\", \\"min_anchor_len\\": \\"10\\", \\"min_anchor_cov\\": \\"1\\", \\"min_bundle_cov\\": \\"1\\", \\"bdist\\": \\"50\\", \\"bundle_fraction\\": \\"1.0\\", \\"disable_trimming\\": false, \\"multi_mapping\\": false, \\"point_features\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"guide\\": {\\"use_guide\\": \\"yes\\", \\"__current_case__\\": 1, \\"guide_source\\": {\\"guide_gff_select\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_hist\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_estimation\\": true, \\"special_outputs\\": {\\"special_outputs_select\\": \\"no\\", \\"__current_case__\\": 2}, \\"coverage_file\\": false}, \\"input_options\\": {\\"input_mode\\": \\"short_reads\\", \\"__current_case__\\": 0, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rna_strandness\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.3+galaxy0","type":"tool","uuid":"5a796064-9c52-4b5c-b5da-02c43f6c742f","when":"$(inputs.when)","workflow_outputs":[{"label":"Gene Abundance Estimates from StringTie","output_name":"gene_abundance_estimation","uuid":"9bcb277a-bb4d-4629-8c71-4cac7cee4c63"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4","errors":null,"id":21,"input_connections":{"advanced_settings|library_type":{"id":14,"output_name":"output_param_text"},"advanced_settings|mask_file":{"id":8,"output_name":"output"},"bias_correction|seq_source|index":{"id":12,"output_name":"out1"},"input":{"id":16,"output_name":"mapped_reads"},"reference_annotation|reference_annotation_file":{"id":4,"output_name":"output"},"when":{"id":7,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Cufflinks","name":"advanced_settings"},{"description":"runtime parameter for tool Cufflinks","name":"advanced_settings"},{"description":"runtime parameter for tool Cufflinks","name":"reference_annotation"}],"label":"Compute FPKM with cufflinks","name":"Cufflinks","outputs":[{"name":"genes_expression","type":"tabular"},{"name":"transcripts_expression","type":"tabular"},{"name":"assembled_isoforms","type":"gtf"},{"name":"total_map_mass","type":"txt"},{"name":"skipped","type":"gtf"}],"position":{"left":1998.2812719123385,"top":1150.343757982693},"post_job_actions":{"HideDatasetActionassembled_isoforms":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"assembled_isoforms"},"HideDatasetActionskipped":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"skipped"},"HideDatasetActiontotal_map_mass":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"total_map_mass"},"RenameDatasetActiongenes_expression":{"action_arguments":{"newname":"Genes Expression from Cufflinks"},"action_type":"RenameDatasetAction","output_name":"genes_expression"},"RenameDatasetActiontranscripts_expression":{"action_arguments":{"newname":"Transcripts Expression from Cufflinks"},"action_type":"RenameDatasetAction","output_name":"transcripts_expression"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4","tool_shed_repository":{"changeset_revision":"6926716bbfb5","name":"cufflinks","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_settings\\": {\\"use_advanced_settings\\": \\"Yes\\", \\"__current_case__\\": 1, \\"library_type\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mask_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inner_mean_dist\\": \\"200\\", \\"inner_dist_std_dev\\": \\"80\\", \\"max_mle_iterations\\": \\"5000\\", \\"junc_alpha\\": \\"0.001\\", \\"small_anchor_fraction\\": \\"0.09\\", \\"overhang_tolerance\\": \\"8\\", \\"max_bundle_length\\": \\"10000000\\", \\"max_bundle_frags\\": \\"1000000\\", \\"min_intron_length\\": \\"50\\", \\"trim_three_avgcov_thresh\\": \\"10\\", \\"trim_three_dropoff_frac\\": \\"0.1\\"}, \\"bias_correction\\": {\\"do_bias_correction\\": \\"Yes\\", \\"__current_case__\\": 0, \\"seq_source\\": {\\"index_source\\": \\"cached\\", \\"__current_case__\\": 0, \\"index\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"global_model\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_correction\\": \\"--no-effective-length-correction\\", \\"max_intron_len\\": \\"300000\\", \\"min_isoform_fraction\\": \\"0.1\\", \\"multiread_correct\\": true, \\"pre_mrna_fraction\\": \\"0.15\\", \\"reference_annotation\\": {\\"use_ref\\": \\"Use reference annotation\\", \\"__current_case__\\": 1, \\"reference_annotation_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"compatible_hits_norm\\": \\"\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1.4","type":"tool","uuid":"af5419b7-ae6b-498c-8123-f609b8fcd8b4","when":"$(inputs.when)","workflow_outputs":[{"label":"Transcripts Expression from Cufflinks","output_name":"transcripts_expression","uuid":"2c13ef79-948c-4f5f-8516-a769550128fd"},{"label":"Genes Expression from Cufflinks","output_name":"genes_expression","uuid":"3bd4fbda-f241-4ebd-b7a5-dea7271c521e"}]},"22":{"annotation":"","id":22,"input_connections":{"RNA STAR count tables":{"id":16,"input_subworkflow_step_id":3,"output_name":"reads_per_gene"},"Strandness param":{"id":5,"input_subworkflow_step_id":2,"output_name":"output"},"featureCounts count table":{"id":19,"input_subworkflow_step_id":4,"output_name":"output_short"},"featureCounts summaries collection":{"id":19,"input_subworkflow_step_id":1,"output_name":"output_summary"}},"inputs":[],"label":null,"name":"Process Count files","outputs":[],"position":{"left":2161.105895746449,"top":665.1799212422835},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"},{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","name":"Process Count files","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"This file will be used in MultiQC if the user opts out of featureCounts to generate count files. ","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":0,"input_connections":{},"inputs":[],"label":"A dummy featureCounts summary file","name":"Create text file","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":0,"top":16},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": \\"Status\\\\tFeaturecounts_was_not_used\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}, {\\"__index__\\": 1, \\"line\\": \\"Assigned\\\\t0\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}, {\\"__index__\\": 2, \\"line\\": \\"Unassigned_Unmapped\\\\t0\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"be0268be-a3ec-4d75-854b-d8eecf4b4521","when":null,"workflow_outputs":[]},"1":{"annotation":"Collection from featureCounts summaries","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Collection from featureCounts summaries","name":"featureCounts summaries collection"}],"label":"featureCounts summaries collection","name":"Input dataset collection","outputs":[],"position":{"left":2.3481338306177992,"top":215.26828327443155},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"9ca62bd8-0272-41e2-9e4b-e8f45c4e0c19","when":null,"workflow_outputs":[]},"2":{"annotation":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence","name":"Strandness param"}],"label":"Strandness param","name":"Input parameter","outputs":[],"position":{"left":0.000017190473798223138,"top":433.85391433745605},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"180fc8bd-b8f4-486b-b300-e503bf379373","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"d0e53d5e-1b5f-40d7-80e6-f15c5e2ffbc8"}]},"3":{"annotation":"Count tables collection from RNA STAR","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Count tables collection from RNA STAR","name":"RNA STAR count tables"}],"label":"RNA STAR count tables","name":"Input dataset collection","outputs":[],"position":{"left":5.368903743075304,"top":589.5341644488473},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"d94c5c03-7bad-493f-9f5d-c5b6f33157ae","when":null,"workflow_outputs":[]},"4":{"annotation":"Count tables collection from featureCounts","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Count tables collection from featureCounts","name":"featureCounts count table"}],"label":"featureCounts count table","name":"Input dataset collection","outputs":[],"position":{"left":7.4993932207943,"top":704.0194730841151},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"12f710b8-757e-4c25-90be-4cfca673f54b","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","errors":null,"id":5,"input_connections":{"infile":{"id":0,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":279.9999490094584,"top":0},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/__tc__/\\\\\\\\t/g\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"3588c02b-15b7-487b-8970-bc76ddf8348a","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":6,"input_connections":{"input_param_type|input_param":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"awk command from strand for STAR counts","name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":279.9999833196541,"top":401.33828843669767},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$3}\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$4}\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"NR>4{print $1\\\\\\"\\\\\\\\t\\\\\\"$2}\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"2b9a5ca1-e974-450b-99b3-3fcce531154b","when":null,"workflow_outputs":[]},"7":{"annotation":"featureCounts summaries","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":7,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":1,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":5,"output_name":"output"}},"inputs":[],"label":"featureCounts summaries","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":840.0000391186242,"top":16},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"FeatureCounts Summary Table"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"26eca035-e403-4392-9ba9-08cd471154dd","when":null,"workflow_outputs":[{"label":"FeatureCounts Summary Table","output_name":"data_param","uuid":"8c15f8aa-f2eb-41bf-bbaa-947263f3bfec"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":8,"input_connections":{"code":{"id":6,"output_name":"output_param_text"},"infile":{"id":3,"output_name":"output"}},"inputs":[],"label":"Extract gene counts","name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":560.0000901091657,"top":386.17188642026633},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"},"RenameDatasetActionoutfile":{"action_arguments":{"newname":"HTS count like output"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"34922bd3-4676-4f7d-8e05-af0ef0c1efa4","when":null,"workflow_outputs":[]},"9":{"annotation":"Counts Table ","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":9,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":4,"output_name":"output"},"style_cond|type_cond|pick_from_1|value":{"id":8,"output_name":"outfile"}},"inputs":[],"label":"Counts table","name":"Pick parameter value","outputs":[{"name":"data_param","type":"input"}],"position":{"left":837.0001537846339,"top":413.1722363660909},"post_job_actions":{"RenameDatasetActiondata_param":{"action_arguments":{"newname":"Counts Table"},"action_type":"RenameDatasetAction","output_name":"data_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first\\", \\"__current_case__\\": 0, \\"type_cond\\": {\\"param_type\\": \\"data\\", \\"__current_case__\\": 4, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"d87cdd15-8b7c-4498-a818-7369ee4e006a","when":null,"workflow_outputs":[{"label":"Counts Table","output_name":"data_param","uuid":"bb7b60bf-81b1-4d58-9724-9814e9284ab3"}]}},"tags":[],"uuid":"e58ded35-46a9-4fdd-8524-cfda6b2e7c39"},"tool_id":null,"type":"subworkflow","uuid":"6957fcb0-4312-4764-9d80-c5306759f914","when":null,"workflow_outputs":[{"label":"Counts Table","output_name":"Counts Table","uuid":"dfe40af6-1361-4953-aac7-fbca6cfdb47a"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":23,"input_connections":{"results_0|software_cond|input":{"id":10,"output_name":"report_json"},"results_1|software_cond|output_0|type|input":{"id":16,"output_name":"output_log"},"results_1|software_cond|output_1|type|input":{"id":16,"output_name":"reads_per_gene"},"results_2|software_cond|input":{"id":22,"output_name":"FeatureCounts Summary Table"},"when":{"id":11,"output_name":"output_param_boolean"}},"inputs":[],"label":"Combined MultiQC without additional QC","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":2423.9455325643053,"top":32.99591513431855},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"featureCounts\\", \\"__current_case__\\": 9, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"title\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"e5335a24-fec9-422c-b93c-c172ad7e1b8c","when":"$(inputs.when)","workflow_outputs":[{"label":"Small MultiQC stats","output_name":"stats","uuid":"f9eecf9b-d360-4d97-962f-24efc44d5b99"},{"label":"Small MultiQC HTML report","output_name":"html_report","uuid":"24cf626f-6539-4123-85d6-214b88e2c4b7"}]},"24":{"annotation":"","id":24,"input_connections":{"FASTQ collection":{"id":0,"input_subworkflow_step_id":0,"output_name":"output"},"STAR BAM":{"id":16,"input_subworkflow_step_id":6,"output_name":"mapped_reads"},"STAR counts":{"id":16,"input_subworkflow_step_id":3,"output_name":"reads_per_gene"},"STAR logs":{"id":16,"input_subworkflow_step_id":2,"output_name":"output_log"},"fastp Reports":{"id":10,"input_subworkflow_step_id":1,"output_name":"report_json"},"featureCounts summaries":{"id":22,"input_subworkflow_step_id":4,"output_name":"FeatureCounts Summary Table"},"reference_annotation_gtf":{"id":4,"input_subworkflow_step_id":5,"output_name":"output"},"when":{"id":2,"output_name":"output"}},"inputs":[{"description":"","name":"Combined MultiQC Quality Report"}],"label":"Combined MultiQC Quality Report","name":"RNA-seq-QC","outputs":[],"position":{"left":2465.503860179348,"top":451.63585583919314},"subworkflow":{"a_galaxy_workflow":"true","annotation":"","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","name":"RNA-seq-QC","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"RNA-seq paired FASTQ collection","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"RNA-seq paired FASTQ collection","name":"FASTQ collection"}],"label":"FASTQ collection","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":13.596214101155162},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"1bb12b7e-2ee1-4d8a-80d6-4259bea1efae","when":null,"workflow_outputs":[]},"1":{"annotation":"fastp Reports","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"fastp Reports","name":"fastp Reports"}],"label":"fastp Reports","name":"Input dataset collection","outputs":[],"position":{"left":61.11669921875,"top":134.59621410115517},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"43c69a53-b1dc-4089-9801-18e980088a2e","when":null,"workflow_outputs":[]},"2":{"annotation":"STAR log files","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"STAR log files","name":"STAR logs"}],"label":"STAR logs","name":"Input dataset collection","outputs":[],"position":{"left":103.183349609375,"top":232.67952708943642},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"5ae91949-4ffe-4aef-81fa-fa3cb4e86f5a","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"STAR counts"}],"label":"STAR counts","name":"Input dataset collection","outputs":[],"position":{"left":135,"top":311.09621410115517},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"57b8e456-2987-42aa-8ee3-3d25a0d5fecf","when":null,"workflow_outputs":[]},"4":{"annotation":"featureCounts Summaries file","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"featureCounts Summaries file","name":"featureCounts summaries"}],"label":"featureCounts summaries","name":"Input dataset collection","outputs":[],"position":{"left":168.6500244140625,"top":397.7962263081864},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"5a5cb076-92cb-48bd-80e0-d7bfa19fe969","when":null,"workflow_outputs":[]},"5":{"annotation":"Reference annotation","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Reference annotation","name":"reference_annotation_gtf"}],"label":"reference_annotation_gtf","name":"Input dataset","outputs":[],"position":{"left":236.05178147043327,"top":521.1329885056177},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"9657bd5f-5510-412b-9128-0bb903e2f572","when":null,"workflow_outputs":[]},"6":{"annotation":"Mapped BAM files from STAR","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"Mapped BAM files from STAR","name":"STAR BAM"}],"label":"STAR BAM","name":"Input dataset collection","outputs":[],"position":{"left":287.0148354755526,"top":645.9144573009889},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"b3a7345d-49fa-4b34-bd56-565f250f6d97","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","errors":null,"id":7,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":"FastQC check read qualities","name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":709.0902620065212,"top":0},"post_job_actions":{"HideDatasetActionhtml_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"html_file"},"HideDatasetActiontext_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy1","tool_shed_repository":{"changeset_revision":"2c64fded1286","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy1","type":"tool","uuid":"d4f16966-836e-49cd-8987-18b6e68205b8","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357","errors":null,"id":8,"input_connections":{"gtf_file":{"id":5,"output_name":"output"}},"inputs":[],"label":"convert gtf to bed12","name":"Convert GTF to BED12","outputs":[{"name":"bed_file","type":"bed12"}],"position":{"left":650.1095703232029,"top":767.5962172140157},"post_job_actions":{"HideDatasetActionbed_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"bed_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357","tool_shed_repository":{"changeset_revision":"b026dae67fba","name":"gtftobed12","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"advanced_options_selector\\": \\"default\\", \\"__current_case__\\": 0}, \\"gtf_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"357","type":"tool","uuid":"ed8685b7-7904-4f28-8ea9-2faffb05bacf","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","errors":null,"id":9,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":"Subsample 200k reads","name":"Samtools view","outputs":[{"name":"outputsam","type":"input"}],"position":{"left":650.1093443590796,"top":919.946789222449},"post_job_actions":{"HideDatasetActionoutputsam":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputsam"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.20+galaxy3","tool_shed_repository":{"changeset_revision":"32dc5f781059","name":"samtools_view","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"addref_cond\\": {\\"addref_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": {\\"outtype\\": \\"selected_reads\\", \\"__current_case__\\": 1, \\"filter_config\\": {\\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_rg\\": {\\"select_rg\\": \\"no\\", \\"__current_case__\\": 0}, \\"cond_expr\\": {\\"select_expr\\": \\"no\\", \\"__current_case__\\": 0}, \\"quality\\": null, \\"library\\": \\"\\", \\"cigarcons\\": null, \\"inclusive_filter\\": null, \\"exclusive_filter\\": null, \\"exclusive_filter_all\\": null, \\"tag\\": \\"\\", \\"qname_file\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"subsample_config\\": {\\"subsampling_mode\\": {\\"select_subsample\\": \\"target\\", \\"__current_case__\\": 1, \\"target\\": \\"200000\\", \\"seed\\": \\"1\\"}}, \\"output_options\\": {\\"reads_report_type\\": \\"retained\\", \\"__current_case__\\": 0, \\"complementary_output\\": false, \\"adv_output\\": {\\"readtags\\": [], \\"collapsecigar\\": false}, \\"output_format\\": {\\"oformat\\": \\"bam\\", \\"__current_case__\\": 2}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.20+galaxy3","type":"tool","uuid":"fa918b19-c767-43dd-afb6-fcdb3cd20fa7","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","errors":null,"id":10,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":"Get reads number per chromosome","name":"Samtools idxstats","outputs":[{"name":"output","type":"tabular"}],"position":{"left":650.1095878752642,"top":1126.5962172140155},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.5","tool_shed_repository":{"changeset_revision":"fa5d3e61e429","name":"samtools_idxstats","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5","type":"tool","uuid":"0c0789f8-21d9-4fdc-80c6-488aded24611","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","errors":null,"id":11,"input_connections":{"inputFile":{"id":6,"output_name":"output"}},"inputs":[],"label":"Remove duplicates","name":"MarkDuplicates","outputs":[{"name":"metrics_file","type":"txt"},{"name":"outFile","type":"bam"}],"position":{"left":648.4598772618594,"top":1276.9345984373845},"post_job_actions":{"HideDatasetActionmetrics_file":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"metrics_file"},"HideDatasetActionoutFile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outFile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/3.1.1.0","tool_shed_repository":{"changeset_revision":"3f254c5ced1d","name":"picard","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"assume_sorted\\": true, \\"barcode_tag\\": \\"\\", \\"comments\\": [], \\"duplicate_scoring_strategy\\": \\"SUM_OF_BASE_QUALITIES\\", \\"inputFile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"optical_duplicate_pixel_distance\\": \\"100\\", \\"read_name_regex\\": \\"\\", \\"remove_duplicates\\": false, \\"validation_stringency\\": \\"LENIENT\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1.1.0","type":"tool","uuid":"c8d04f50-da21-4f5b-9d64-2dc652b71d77","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/5.0.3+galaxy0","errors":null,"id":12,"input_connections":{"input":{"id":6,"output_name":"output"},"refgene":{"id":8,"output_name":"bed_file"}},"inputs":[],"label":"Read distribution over genomic features","name":"Read Distribution","outputs":[{"name":"output","type":"txt"}],"position":{"left":920.1095706393493,"top":658.5962172140157},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/5.0.3+galaxy0","tool_shed_repository":{"changeset_revision":"27e16a30667a","name":"rseqc","owner":"nilesh","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"refgene\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.3+galaxy0","type":"tool","uuid":"4ecf1a7a-8648-4224-9c04-ab5d33614d63","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/5.0.3+galaxy0","errors":null,"id":13,"input_connections":{"batch_mode|input":{"id":9,"output_name":"outputsam"},"refgene":{"id":8,"output_name":"bed_file"}},"inputs":[{"description":"runtime parameter for tool Gene Body Coverage (BAM)","name":"batch_mode"}],"label":"Get gene body coverage","name":"Gene Body Coverage (BAM)","outputs":[{"name":"outputcurvespdf","type":"pdf"},{"name":"outputtxt","type":"txt"}],"position":{"left":919.109375,"top":868.5962141011552},"post_job_actions":{"HideDatasetActionoutputcurvespdf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputcurvespdf"},"HideDatasetActionoutputtxt":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outputtxt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/5.0.3+galaxy0","tool_shed_repository":{"changeset_revision":"27e16a30667a","name":"rseqc","owner":"nilesh","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"batch_mode\\": {\\"batch_mode_selector\\": \\"batch\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"minimum_length\\": \\"100\\", \\"refgene\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rscript_output\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.0.3+galaxy0","type":"tool","uuid":"c996512e-789b-422c-833c-de854e30674f","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","errors":null,"id":14,"input_connections":{"results_0|software_cond|output_0|input":{"id":7,"output_name":"text_file"},"results_1|software_cond|input":{"id":1,"output_name":"output"},"results_2|software_cond|output_0|type|input":{"id":2,"output_name":"output"},"results_2|software_cond|output_1|type|input":{"id":3,"output_name":"output"},"results_3|software_cond|input":{"id":4,"output_name":"output"},"results_4|software_cond|output_0|type|input":{"id":12,"output_name":"output"},"results_4|software_cond|output_1|type|input":{"id":13,"output_name":"outputtxt"},"results_5|software_cond|output_0|type|input":{"id":10,"output_name":"output"},"results_6|software_cond|output_0|input":{"id":11,"output_name":"metrics_file"}},"inputs":[],"label":"Combined Quality Report","name":"MultiQC","outputs":[{"name":"html_report","type":"html"},{"name":"stats","type":"tabular"}],"position":{"left":1240.1096398991554,"top":428.5962172140157},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"MultiQC HTML report"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"MultiQC Stats table"},"action_type":"RenameDatasetAction","output_name":"stats"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0","tool_shed_repository":{"changeset_revision":"a7e081ceb76a","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"fastp\\", \\"__current_case__\\": 7, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"software_cond\\": {\\"software\\": \\"star\\", \\"__current_case__\\": 28, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"log\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"genecounts\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 3, \\"software_cond\\": {\\"software\\": \\"featureCounts\\", \\"__current_case__\\": 9, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 4, \\"software_cond\\": {\\"software\\": \\"rseqc\\", \\"__current_case__\\": 22, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"read_distribution\\", \\"__current_case__\\": 6, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"type\\": {\\"type\\": \\"gene_body_coverage\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 5, \\"software_cond\\": {\\"software\\": \\"samtools\\", \\"__current_case__\\": 24, \\"output\\": [{\\"__index__\\": 0, \\"type\\": {\\"type\\": \\"idxstats\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}}, {\\"__index__\\": 6, \\"software_cond\\": {\\"software\\": \\"picard\\", \\"__current_case__\\": 17, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"markdups\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"title\\": \\"Combined Quality Report \\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.27+galaxy0","type":"tool","uuid":"b5e9c0d2-62f4-41b2-a1c7-5677d8555fd4","when":null,"workflow_outputs":[{"label":"MultiQC stats","output_name":"stats","uuid":"cbf99516-9349-4f2a-ad1f-05c732cb5263"},{"label":"MultiQC HTML report","output_name":"html_report","uuid":"7306cd4b-1017-4542-bf57-81d2d83d6a97"}]}},"tags":[],"uuid":"81396b8d-2758-4979-94b2-66c34bf47d67"},"tool_id":null,"type":"subworkflow","uuid":"c0aaa16c-8a9b-4fe1-8c3d-9e77f64da413","when":"$(inputs.when)","workflow_outputs":[{"label":"MultiQC HTML report","output_name":"MultiQC HTML report","uuid":"717627ed-188d-4cb5-9a03-b1eec15f7922"},{"label":"MultiQC stats","output_name":"MultiQC stats","uuid":"43ed1ffc-7df8-42a6-95f7-dc2b2060a910"}]}},"tags":["RNAseq","transcriptomics"],"uuid":"885ec907-9f61-4a11-9d23-2243389e7335","version":1},"readme":"# RNA-Seq Analysis: Single-End Read Processing and Quantification\\n\\n## Inputs dataset\\n\\n- Collection of FASTQ files: The workflow needs a list of datasets of fastqsanger.\\n- GTF file of annotation: A gtf file with genes annotation.\\n- GTF with regions to exclude from FPKM normalization with Cufflinks: Optional, but recommended. A gtf file with regions to exclude from normalization in Cufflinks.\\n\\n  - For instance a gtf that masks chrM for the mm10 genome:\\n\\n```\\nchrM\\tchrM_gene\\texon\\t0\\t16299\\t.\\t+\\t.\\tgene_id \\"chrM_gene_plus\\"; transcript_id \\"chrM_tx_plus\\"; exon_id \\"chrM_ex_plus\\";\\nchrM\\tchrM_gene\\texon\\t0\\t16299\\t.\\t-\\t.\\tgene_id \\"chrM_gene_minus\\"; transcript_id \\"chrM_tx_minus\\"; exon_id \\"chrM_ex_minus\\";\\n```\\n\\n## Inputs values\\n\\n- Forward adapter (optional): If not provided, fastp will try to guess the adapter sequence from the data. Its sequences  depends on the library preparation. Usually classical Illumina RNA libraries are Truseq and ISML (relatively new Illumina library) is Nextera. If you don\'t know, use FastQC to determine if it is Truseq or Nextera. If the read length is relatively short (50bp), there is probably no adapter so it will not impact your results.\\n- Generate additional QC reports: whether to compute additional QC: FastQC, Picard, Read distribution on genomic features, gene body coverage, reads per chromosomes.\\n- Reference genome: this field will be adapted to the genomes available for STAR.\\n- Strandedness: For stranded RNA, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence. This will only count alignments that are compatible with your library preparation strategy. This is also used for the stranded coverage and for FPKM computation with cufflinks/StringTie.\\n- Use featureCounts for generating count tables: Whether to use count tables from featureCounts instead of from STAR.\\n- Compute Cufflinks FPKM: Whether you want to get FPKM with Cufflinks (pretty long).\\n- Compute StringTie FPKM: Whether you want to get FPKM/TPM etc... with StringTie.\\n\\n## Processing\\n\\n- The workflow will remove adapters and low quality bases and filter out any read smaller than 15bp.\\n- The filtered reads are mapped with STAR with ENCODE parameters (for long RNA-seq but I use it for short also). STAR is also used to count reads per gene and generate strand-specific normalized coverage (on uniquely mapped reads).\\n- A multiQC is run to have an overview of the QC. This can also be used to get the strandedness.\\n- FPKM values for genes and transcripts are computed with cufflinks using correction for multi-mapped reads (this step is optionnal).\\n- FPKM/TPM values for genes are computed with StringTie (this step is optional).\\n- The BAM is filtered to keep only uniquely mapped reads (tag NH:i:1).\\n- Unstranded coverage is computed with bedtools and normalized to the number of million uniquely mapped reads.\\n- The three coverage files are converted to bigwig.\\n\\n### Warning\\n\\n- The coverage stranded output depends on the strandedness of the library:\\n  - If you have an unstranded library, stranded coverages are useless\\n  - If you have a forward stranded library, the label matches the orientation of reads.\\n  - If you have a reverse stranded library, `forward` should correspond to genes on the forward strand and uses the reads mapped on the reverse strand. `reverse` should correspond to genes on the reverse strand and uses the reads mapped on the forward strand.\\n\\n## Contribution\\n\\n### Version 0.1\\n\\n@lldelisle wrote the workflow and the tests.\\n\\n@nagoue updated the tools, made it work in usegalaxy.org, fixed some best practices.\\n\\n### Version 1.0\\n\\n@pavanvidem added the new features (featurecount + additional QC) and found a smaller test dataset.\\n","changelog":"# Changelog\\n\\n## [1.2] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.6+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.27+galaxy0`\\n\\n## [1.1] - 2024-11-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.6+galaxy0`\\n\\n## [1.0] - 2024-10-22\\n\\n### Changes in workflows\\n- Add an optional subworkflow with more QC: FastQC, Picard, Read distribution on genomic features, gene body coverage, reads per chromosomes.\\n- Add featureCounts as an alternative way to generate count files\\n- Use fastp instead of cutadapt which uses pair overlap and allows to have optional adapter sequences\\n\\n### Tool update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.3` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/2.2.1.4`\\n\\n### Test dataset\\n- Using a new subsampled Yeast test data from Zenodo record https://zenodo.org/records/13987631\\n\\n## [0.9] - 2024-09-23\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.24.1+galaxy0`\\n\\n## [0.8] - 2024-07-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.9+galaxy0`\\n\\n## [0.7] - 2024-06-25\\n\\n### Automatic update (triggered manually)\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.30.0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.31.1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0`\\n\\n## [0.6] - 2024-02-05\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/4.6+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.1.1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.11a+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.1.2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/9.3+galaxy0`\\n\\n## [0.5] - 2023-09-15\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.10b+galaxy4`\\n\\n### Manual update\\n- Use STAR to compute normalized strand-specific coverage\\n- Add an option to use StringTie to compute FPKM\\n- Make cufflinks step optional\\n\\n## [0.4.1] - 2023-09-14\\n- add author in dockstore file\\n\\n## [0.4] - 2023-01-16\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.5.2+galaxy1`\\n\\n## [0.3] - 2022-12-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1`\\n\\n## [0.2] - 2022-11-02\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy1`\\n\\n## [0.1] - 2022-10-12\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## RNA-Seq Analysis: Single-End Read Processing and Quantification\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Collection of FASTQ files\\"]@{ shape: docs }\\n1[\\"ℹ️ Forward adapter\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Generate additional QC reports\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Reference genome\\"]@{ shape: lean-l }\\n4[\\"ℹ️ GTF file of annotation\\"]@{ shape: doc }\\n5[\\"ℹ️ Strandedness\\"]@{ shape: lean-l }\\n6[\\"ℹ️ Use featureCounts for generating count tables\\"]@{ shape: lean-l }\\n7[\\"ℹ️ Compute Cufflinks FPKM\\"]@{ shape: lean-l }\\n8[\\"ℹ️ GTF with regions to exclude from FPKM normalization with Cufflinks\\"]@{ shape: doc }\\n9[\\"ℹ️ Compute StringTie FPKM\\"]@{ shape: lean-l }\\n10[\\"remove adapters + bad quality bases\\"]@{ shape: process }\\n1 --> 10\\n0 --> 10\\n11[\\"no additional QC\\"]@{ shape: process }\\n2 --> 11\\n12[\\"get reference_genome as text parameter\\"]@{ shape: process }\\n3 --> 12\\n13[\\"Get featureCounts strandedness parameter\\"]@{ shape: process }\\n5 --> 13\\n14[\\"Get cufflinks strandedness parameter\\"]@{ shape: process }\\n5 --> 14\\n15[\\"Get Stringtie strandedness parameter\\"]@{ shape: process }\\n5 --> 15\\n16[\\"STAR: map and count and coverage splitted\\"]@{ shape: process }\\n3 --> 16\\n4 --> 16\\n10 --> 16\\n17[\\"🛠️ Generate Unstranded Coverage\\"]@{ shape: subprocess }\\n16 --> 17\\n16 --> 17\\n18[\\"🛠️ Generate Stranded Coverage\\"]@{ shape: subprocess }\\n16 --> 18\\n16 --> 18\\n5 --> 18\\n19[\\"featureCounts\\"]@{ shape: process }\\n16 --> 19\\n4 --> 19\\n13 --> 19\\n6 --> 19\\n20[\\"Compute FPKM with StringTie\\"]@{ shape: process }\\n4 --> 20\\n16 --> 20\\n15 --> 20\\n9 --> 20\\n21[\\"Compute FPKM with cufflinks\\"]@{ shape: process }\\n14 --> 21\\n8 --> 21\\n12 --> 21\\n16 --> 21\\n4 --> 21\\n7 --> 21\\n22[\\"🛠️ Process Count files\\"]@{ shape: subprocess }\\n16 --> 22\\n5 --> 22\\n19 --> 22\\n19 --> 22\\n23[\\"Combined MultiQC without additional QC\\"]@{ shape: process }\\n10 --> 23\\n16 --> 23\\n16 --> 23\\n22 --> 23\\n11 --> 23\\n24[\\"🛠️ Combined MultiQC Quality Report\\"]@{ shape: subprocess }\\n0 --> 24\\n16 --> 24\\n16 --> 24\\n16 --> 24\\n10 --> 24\\n22 --> 24\\n4 --> 24\\n2 --> 24\\n```\\n\\n## RNA-seq-QC\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ FASTQ collection\\"]@{ shape: docs }\\n1[\\"ℹ️ fastp Reports\\"]@{ shape: docs }\\n2[\\"ℹ️ STAR logs\\"]@{ shape: docs }\\n3[\\"ℹ️ STAR counts\\"]@{ shape: docs }\\n4[\\"ℹ️ featureCounts summaries\\"]@{ shape: docs }\\n5[\\"ℹ️ reference_annotation_gtf\\"]@{ shape: doc }\\n6[\\"ℹ️ STAR BAM\\"]@{ shape: docs }\\n7[\\"FastQC check read qualities\\"]@{ shape: process }\\n0 --> 7\\n8[\\"convert gtf to bed12\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Subsample 200k reads\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Get reads number per chromosome\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Remove duplicates\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Read distribution over genomic features\\"]@{ shape: process }\\n6 --> 12\\n8 --> 12\\n13[\\"Get gene body coverage\\"]@{ shape: process }\\n9 --> 13\\n8 --> 13\\n14[\\"Combined Quality Report\\"]@{ shape: process }\\n7 --> 14\\n1 --> 14\\n2 --> 14\\n3 --> 14\\n4 --> 14\\n12 --> 14\\n13 --> 14\\n10 --> 14\\n11 --> 14\\n```\\n\\n## Process Count files\\n\\n```mermaid\\ngraph LR\\n0[\\"A dummy featureCounts summary file\\"]@{ shape: process }\\n1[\\"ℹ️ featureCounts summaries collection\\"]@{ shape: docs }\\n2[\\"ℹ️ Strandness param\\"]@{ shape: lean-l }\\n3[\\"ℹ️ RNA STAR count tables\\"]@{ shape: docs }\\n4[\\"ℹ️ featureCounts count table\\"]@{ shape: docs }\\n5[\\"Text transformation\\"]@{ shape: process }\\n0 --> 5\\n6[\\"awk command from strand for STAR counts\\"]@{ shape: process }\\n2 --> 6\\n7[\\"featureCounts summaries\\"]@{ shape: process }\\n1 --> 7\\n5 --> 7\\n8[\\"Extract gene counts\\"]@{ shape: process }\\n6 --> 8\\n3 --> 8\\n9[\\"Counts table\\"]@{ shape: process }\\n4 --> 9\\n8 --> 9\\n```\\n\\n## Re-arrange Stranded RNA-seq coverage\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ strandedness\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Bedgraph strand 1\\"]@{ shape: docs }\\n2[\\"ℹ️ Bedgraph strand 2\\"]@{ shape: docs }\\n3[\\"Get replacement for strand2\\"]@{ shape: process }\\n0 --> 3\\n4[\\"Get replacement for strand1\\"]@{ shape: process }\\n0 --> 4\\n5[\\"get identifiers\\"]@{ shape: process }\\n1 --> 5\\n6[\\"New labels strand 2\\"]@{ shape: process }\\n3 --> 6\\n5 --> 6\\n7[\\"New labels strand 1\\"]@{ shape: process }\\n4 --> 7\\n5 --> 7\\n8[\\"Relabelled strand 2\\"]@{ shape: process }\\n6 --> 8\\n2 --> 8\\n9[\\"Relabelled strand 1\\"]@{ shape: process }\\n7 --> 9\\n1 --> 9\\n10[\\"Merge collections\\"]@{ shape: process }\\n9 --> 10\\n8 --> 10\\n11[\\"convert to bigwig\\"]@{ shape: process }\\n10 --> 11\\n```\\n\\n## Get Uniquely mapped unstranded coverage\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ STAR log\\"]@{ shape: docs }\\n1[\\"ℹ️ STAR BAM\\"]@{ shape: docs }\\n2[\\"get scaling factor\\"]@{ shape: process }\\n0 --> 2\\n3[\\"keep uniquely mapped reads\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Parse parameter value\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Scaled Coverage both strands combined\\"]@{ shape: process }\\n3 --> 5\\n4 --> 5\\n6[\\"convert both strands coverage to bigwig\\"]@{ shape: process }\\n5 --> 6\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/rnaseq-sr/main","dockstore_id":19841,"categories":[],"collections":["Transcriptomics"],"doi":"10.5281/zenodo.8354479","tests":[{"doc":"Test outline for RNAseq_SR","job":{"GTF file of annotation":{"class":"File","location":"https://zenodo.org/records/13987631/files/Saccharomyces_cerevisiae.R64-1-1.113.gtf","filetype":"gtf"},"Collection of FASTQ files":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"SRR5085167","location":"https://zenodo.org/records/13987631/files/SRR5085167_forward.fastqsanger.gz"}]},"Forward adapter":"AGATCGGAAGAG","Generate additional QC reports":true,"Reference genome":"sacCer3","Strandedness":"stranded - forward","Use featureCounts for generating count tables":true,"Compute Cufflinks FPKM":true,"GTF with regions to exclude from FPKM normalization with Cufflinks":null,"Compute StringTie FPKM":true},"outputs":{"MultiQC stats":{"asserts":{"has_text_matching":{"expression":"SRR5085167\\t0.11[0-9]*\\t18.3[0-9]*\\t69.6[0-9]*\\t0.3[0-9]*\\t0.3[0-9]*\\t94.62\\t0.12[0-9]*\\t34.43\\t0.2[0-9]*\\t28.[0-9]*\\t90.[0-9]*\\t16.[0-9]*\\t0.36[0-9]*\\t43.[0-9]*\\t91.[0-9]*\\t70.[0-9]*\\t36.[0-9]*\\t46.0\\t75.0\\t75\\t27.27[0-9]*\\t0.39[0-9]*"}}},"Counts Table":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W\\t1775"}}}}},"Mapped Reads":{"element_tests":{"SRR5085167":{"asserts":{"has_size":{"value":31570787,"delta":3000000}}}}},"Gene Abundance Estimates from StringTie":{"element_tests":{"SRR5085167":{"asserts":{"has_text_matching":{"expression":"YAL038W\\tCDC19\\tchrI\\t\\\\+\\t71786\\t73288\\t57.[0-9]*\\t3575.[0-9]*\\t3084.[0-9]*"}}}}},"Genes Expression from Cufflinks":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W\\t-\\t-\\tYAL038W\\tCDC19\\t-\\tchrI:71785-73288\\t-\\t-\\t3375.85\\t3161.36\\t3590.33\\tOK"}}}}},"Transcripts Expression from Cufflinks":{"element_tests":{"SRR5085167":{"asserts":{"has_line":{"line":"YAL038W_mRNA\\t-\\t-\\tYAL038W\\tCDC19\\t-\\tchrI:71785-73288\\t1503\\t57.5601\\t3375.85\\t3161.36\\t3590.33\\tOK"}}}}},"Stranded Coverage":{"element_tests":{"SRR5085167_forward":{"asserts":{"has_size":{"value":555489,"delta":50000}}},"SRR5085167_reverse":{"asserts":{"has_size":{"value":526952,"delta":50000}}}}},"Unstranded Coverage":{"element_tests":{"SRR5085167":{"asserts":{"has_size":{"value":978542,"delta":90000}}}}}}}]}],"path":"./workflows/transcriptomics/rnaseq-sr"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/BREW3R.ga","testParameterFiles":["/BREW3R-tests.yml"],"authors":[{"name":"Lucille Delisle","orcid":"0000-0002-1964-4960"}],"definition":{"a_galaxy_workflow":"true","annotation":"Extends 3\' ends of gene annotations using BAM files (from STAR alignments) and a reference GTF. Specifically designed for 3\'-biased sequencing techniques like 10X scRNA-seq or BRB-seq that primarily capture transcript 3\' ends. The BREW3R tool enhances annotations by using evidence from RNA-seq data to improve 3\' UTR definitions, which is particularly important for accurate quantification in single-cell and bulk RNA-seq experiments.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-1964-4960","name":"Lucille Delisle"}],"format-version":"0.1","license":"GPL-3.0-or-later","name":"BREW3R","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Input gtf"}],"label":"Input gtf","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gtf\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c59ff317-5e58-4823-bd62-6474555f40a7","when":null,"workflow_outputs":[]},"1":{"annotation":"BAM collection for example output of STAR","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"BAM collection for example output of STAR","name":"BAM collection"}],"label":"BAM collection","name":"Input dataset collection","outputs":[],"position":{"left":52,"top":92.84911999999173},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"14322b26-f7e5-415d-ab99-74b11036d9d8","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"strandedness"}],"label":"strandedness","name":"Input parameter","outputs":[],"position":{"left":122,"top":186.84911999999173},"tool_id":null,"tool_state":"{\\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ee38bb19-c158-4a54-82b1-a2eeb195e5f0","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"","name":"minimum coverage"}],"label":"minimum coverage","name":"Input parameter","outputs":[],"position":{"left":229,"top":269.84911999999173},"tool_id":null,"tool_state":"{\\"default\\": 10, \\"parameter_type\\": \\"integer\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"912d8e34-2782-4375-a683-1f71ec0cabb7","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"","name":"minimum FPKM for merge"}],"label":"minimum FPKM for merge","name":"Input parameter","outputs":[],"position":{"left":278,"top":383.34911999999173},"tool_id":null,"tool_state":"{\\"default\\": 1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"b71eaaf4-31b8-47e9-a5dd-27c48faaa0c6","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":5,"input_connections":{"input_param_type|input_param":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":null,"name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":437,"top":150.5},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"--fr\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"--rf\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"41e690b0-597b-4a3a-9289-5fc22fc6644b","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":6,"input_connections":{"input_param_type|input_param":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":"Unstranded","name":"Map parameter value","outputs":[{"name":"output_param_boolean","type":"expression.json"}],"position":{"left":882,"top":394.3499755859375},"post_job_actions":{"HideDatasetActionoutput_param_boolean":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_boolean"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"unstranded\\", \\"to\\": \\"true\\"}]}, \\"output_param_type\\": \\"boolean\\", \\"unmapped\\": {\\"on_unmapped\\": \\"default\\", \\"__current_case__\\": 2, \\"default_value\\": \\"false\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"62529fa0-d1b5-40c2-a1c4-69976ace92ce","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","errors":null,"id":7,"input_connections":{"adv|min_anchor_cov":{"id":3,"output_name":"output"},"adv|min_bundle_cov":{"id":3,"output_name":"output"},"input_options|input_bam":{"id":1,"output_name":"output"},"rna_strandness":{"id":5,"output_name":"output_param_text"}},"inputs":[{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"input_options"}],"label":"assembl with StringTie","name":"StringTie","outputs":[{"name":"output_gtf","type":"gtf"}],"position":{"left":692,"top":38.850006103515625},"post_job_actions":{"HideDatasetActionoutput_gtf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_gtf"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","tool_shed_repository":{"changeset_revision":"cbf488da3b2c","name":"stringtie","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"abundance_estimation\\": false, \\"omit_sequences\\": \\"\\", \\"name_prefix\\": \\"\\", \\"fraction\\": \\"0.01\\", \\"min_tlen\\": \\"200\\", \\"min_anchor_len\\": \\"10\\", \\"min_anchor_cov\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_bundle_cov\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bdist\\": \\"50\\", \\"bundle_fraction\\": \\"1.0\\", \\"disable_trimming\\": false, \\"multi_mapping\\": false, \\"point_features\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"guide\\": {\\"use_guide\\": \\"no\\", \\"__current_case__\\": 0}, \\"input_options\\": {\\"input_mode\\": \\"short_reads\\", \\"__current_case__\\": 0, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rna_strandness\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.3+galaxy0","type":"tool","uuid":"dd55ed02-93cf-4341-927b-d68879ac94c2","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie_merge/2.2.3+galaxy0","errors":null,"id":8,"input_connections":{"input_gtf":{"id":7,"output_name":"output_gtf"},"min_fpkm":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool StringTie merge","name":"guide_gff"}],"label":"merge assembled transcripts","name":"StringTie merge","outputs":[{"name":"out_gtf","type":"gtf"}],"position":{"left":936,"top":108.85000610351562},"post_job_actions":{"HideDatasetActionout_gtf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_gtf"},"RenameDatasetActionout_gtf":{"action_arguments":{"newname":"merged StringTie gtf"},"action_type":"RenameDatasetAction","output_name":"out_gtf"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie_merge/2.2.3+galaxy0","tool_shed_repository":{"changeset_revision":"cbf488da3b2c","name":"stringtie","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"gap_len\\": \\"0\\", \\"guide_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_gtf\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_introns\\": false, \\"min_cov\\": \\"0\\", \\"min_fpkm\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_iso\\": \\"0.01\\", \\"min_len\\": \\"50\\", \\"min_tpm\\": \\"1.0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.3+galaxy0","type":"tool","uuid":"d93e48b1-da4c-4eaa-bd1c-c12aca790c05","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/brew3r_r/brew3r_r/1.0.2+galaxy1","errors":null,"id":9,"input_connections":{"filter_unstranded":{"id":6,"output_name":"output_param_boolean"},"gtf_to_extend":{"id":0,"output_name":"output"},"gtf_to_overlap":{"id":8,"output_name":"out_gtf"}},"inputs":[],"label":"BREW3R.r","name":"BREW3R.r","outputs":[{"name":"output","type":"gtf"}],"position":{"left":1243,"top":115.3499755859375},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"#{gtf_to_extend} BREW3R extended"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/brew3r_r/brew3r_r/1.0.2+galaxy1","tool_shed_repository":{"changeset_revision":"d3b0390f325f","name":"brew3r_r","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exclude_pattern\\": \\"\\", \\"filter_unstranded\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gtf_to_extend\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gtf_to_overlap\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"no_add\\": false, \\"sup_output\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2+galaxy1","type":"tool","uuid":"c6a1e810-55f4-4c0e-9f19-d162503edbf0","when":null,"workflow_outputs":[{"label":"extended_gtf","output_name":"output","uuid":"fddb2278-64af-422b-88da-1c8889a175be"}]}},"tags":[],"uuid":"ac8ef6cb-1621-4b73-9362-8a460ef3c3a2","version":1},"readme":"# BREW3R\\n\\n![BREW3R logo](https://raw.githubusercontent.com/lldelisle/BREW3R/main/images/logo.png)\\n\\nBREW3R stands for **B**ulk **R**NA-seq **E**vidence-based **W**orkflow for **3**\' UTR **R**eannotation. \\n\\nThis workflow enables extending an existing gtf downloaded on a public website, like Ensembl, Genecode or UCSC, using *de novo* gene annotation with StringTie on full length bulk RNA-seq.\\n\\nBREW3R highly relies on a R package called BREW3R.r available on [bioconductor](https://bioconductor.org/packages/release/bioc/html/BREW3R.r.html).\\n\\n## Input datasets\\n\\n- The workflow requires an input gtf file which will be extended.\\n- As well as a collection of BAM files.\\n\\n## Input values\\n\\n- strandedness: Must be one of `stranded - forward`, `stranded - reverse` and `unstranded`. For stranded libraries, reverse means that the read is complementary to the coding sequence, forward means that the read is in the same orientation as the coding sequence.\\n- minimum coverage: Minimum reads per bp coverage to consider for assembly in each de novo assembly (for each BAM file). Default: 10\\n- minimum FPKM for merge: Minimum FPKM value for a transcript to be included into the merged de novo assembly.\\n\\n## Processing\\n\\n- StringTie is called once per input BAM file to compute de novo assembly.\\n- StringTie is called to merge all outputs of previous steps.\\n- BREW3R.r is run with the default parameters on the input gtf to extend and the output of StringTie. If the library was unstranded all merged transcripts without orientation that overlaps exons of both strands are not used for the extension.\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-10-07\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/brew3r_r/brew3r_r/1.0.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/brew3r_r/brew3r_r/1.0.2+galaxy1`\\n\\n## [0.1] - 2024-06-17\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## BREW3R\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input gtf\\"]@{ shape: doc }\\n1[\\"ℹ️ BAM collection\\"]@{ shape: docs }\\n2[\\"ℹ️ strandedness\\"]@{ shape: lean-l }\\n3[\\"ℹ️ minimum coverage\\"]@{ shape: lean-l }\\n4[\\"ℹ️ minimum FPKM for merge\\"]@{ shape: lean-l }\\n5[\\"Map parameter value\\"]@{ shape: process }\\n2 --> 5\\n6[\\"Unstranded\\"]@{ shape: process }\\n2 --> 6\\n7[\\"assembl with StringTie\\"]@{ shape: process }\\n3 --> 7\\n3 --> 7\\n1 --> 7\\n5 --> 7\\n8[\\"merge assembled transcripts\\"]@{ shape: process }\\n7 --> 8\\n4 --> 8\\n9[\\"BREW3R.r\\"]@{ shape: process }\\n6 --> 9\\n0 --> 9\\n8 --> 9\\n```\\n","updated":"2024-10-07T00:00:00","trsID":"#workflow/github.com/iwc-workflows/brew3r/main","dockstore_id":27294,"categories":[],"collections":["Transcriptomics"],"doi":"10.5281/zenodo.15078471","tests":[{"doc":"Test outline for BREW3R","job":{"Input gtf":{"class":"File","path":"test-data/input.gtf","filetype":"gtf"},"BAM collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"GSM4196550","path":"test-data/GSM4196550_subset.bam"},{"class":"File","identifier":"GSM4196551","path":"test-data/GSM4196551_subset.bam"},{"class":"File","identifier":"GSM4196552","path":"test-data/GSM4196552_subset.bam"}]},"strandedness":"stranded - reverse","minimum coverage":10,"minimum FPKM for merge":1},"outputs":{"extended_gtf":{"asserts":{"has_line":{"line":"chr12\\tHAVANA\\texon\\t11206402\\t11208972\\t.\\t-\\t.\\tgene_id \\"ENSMUSG00000047002.2\\"; gene_type \\"protein_coding\\"; gene_name \\"Msgn1\\"; level \\"2\\"; mgi_id \\"MGI:1860483\\"; havana_gene \\"OTTMUSG00000065713.1\\"; transcript_id \\"ENSMUST00000049877.2\\"; transcript_type \\"protein_coding\\"; transcript_name \\"Msgn1-201\\"; protein_id \\"ENSMUSP00000055001.1\\"; transcript_support_level \\"NA\\"; tag \\"CCDS\\"; ccdsid \\"CCDS25814.1\\"; havana_transcript \\"OTTMUST00000159929.1\\"; exon_number \\"1\\"; exon_id \\"ENSMUSE00000372682.2.ext\\""},"has_text":{"text":"CDS"}}}}}]}],"path":"./workflows/transcriptomics/brew3r"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/rnaseq-de-filtering-plotting.ga","testParameterFiles":["/rnaseq-de-filtering-plotting-tests.yml"],"authors":[{"name":"Pavankumar Videm","orcid":"0000-0002-5192-126X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Identifies differentially expressed genes between exactly two experimental conditions from count tables. The workflow performs statistical testing, applies filters based on adjusted p-value and log2 fold change thresholds, and generates publication-quality visualizations including volcano plots, MA plots, and heatmaps. Takes two collections of count tables as input and produces filtered gene lists and interactive plots for interpreting expression differences. Optimal for simple two-condition experimental designs.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-5192-126X","name":"Pavankumar Videm"}],"format-version":"0.1","license":"MIT","release":"0.4","name":"RNA-Seq Differential Expression Analysis with Visualization","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Counts from experimental condition or changed condition. For eg. counts from treatment or knockdown samples.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Counts from experimental condition or changed condition. For eg. counts from treatment or knockdown samples.","name":"Counts from changed condition"}],"label":"Counts from changed condition","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":231.6810606846659},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"7e7436a6-e408-4941-81c2-7a86c0c9235a","when":null,"workflow_outputs":[]},"1":{"annotation":"Counts from reference condition or base condition. For eg. counts from untreated or wildtype samples.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Counts from reference condition or base condition. For eg. counts from untreated or wildtype samples.","name":"Counts from reference condition"}],"label":"Counts from reference condition","name":"Input dataset collection","outputs":[],"position":{"left":47.54331804667683,"top":344.2691910956221},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"2f811a41-c4af-4661-b17b-76ee244f3b27","when":null,"workflow_outputs":[]},"2":{"annotation":"Usually, count files generated from featureCounts tool have a header line whereas count files from RNA-STAR do not have.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Usually, count files generated from featureCounts tool have a header line whereas count files from RNA-STAR do not have.","name":"Count files have header"}],"label":"Count files have header","name":"Input parameter","outputs":[],"position":{"left":110.4753923836675,"top":456.18541400369907},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"boolean\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"7eeb1fbf-cacd-4395-a22f-cceaeab27079","when":null,"workflow_outputs":[]},"3":{"annotation":"The same annotation GTF used for mapping and counting","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"The same annotation GTF used for mapping and counting","name":"Gene Annotaton"}],"label":"Gene Annotaton","name":"Input dataset","outputs":[],"position":{"left":153.82812224613164,"top":585.5486767979796},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"4f6df6ae-39ac-4a5e-98df-a0cb082a44fe","when":null,"workflow_outputs":[]},"4":{"annotation":"Adjusted p-value threshold to call genes as differentially expressed. If not set, default 0.05 is used.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Adjusted p-value threshold to call genes as differentially expressed. If not set, default 0.05 is used.","name":"Adjusted p-value threshold"}],"label":"Adjusted p-value threshold","name":"Input parameter","outputs":[],"position":{"left":184.15025210412472,"top":735.8477989364526},"tool_id":null,"tool_state":"{\\"default\\": 0.05, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"43754055-ab65-475e-ae75-b84be92f9a06","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","errors":null,"id":5,"input_connections":{},"inputs":[],"label":null,"name":"DE table header","outputs":[{"name":"outfile","type":"txt"}],"position":{"left":812.5163140349712,"top":0},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"token_set\\": [{\\"__index__\\": 0, \\"line\\": \\"GeneID\\\\tBase mean\\\\tlog2(FC)\\\\tStdErr\\\\tWald-Stats\\\\tP-value\\\\tP-adj\\\\tChromosome\\\\tStart\\\\tEnd\\\\tStrand\\\\tFeature\\\\tGene name\\", \\"repeat_select\\": {\\"repeat_select_opts\\": \\"user\\", \\"__current_case__\\": 0, \\"times\\": \\"1\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"afef7b5d-f329-4792-bc41-a8bb705e6ccc","when":null,"workflow_outputs":[]},"6":{"annotation":"log2 fold change threshold to filter for highly regulated genes. If not set, default 1.0 is used.","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"log2 fold change threshold to filter for highly regulated genes. If not set, default 1.0 is used.","name":"log2 fold change threshold"}],"label":"log2 fold change threshold","name":"Input parameter","outputs":[],"position":{"left":211.7099474957992,"top":880.615777567409},"tool_id":null,"tool_state":"{\\"default\\": 1.0, \\"parameter_type\\": \\"float\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"bb8bb582-3f12-4dfd-b973-2e24bf6aea07","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":7,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":450.98839299501634,"top":659.5669461166157},"post_job_actions":{"HideDatasetActionfloat_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"float_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"0.05\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"82ade695-5064-4c41-8f17-113a91571438","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","errors":null,"id":8,"input_connections":{"infile":{"id":5,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Text transformation","outputs":[{"name":"output","type":"input"}],"position":{"left":1065.0285936858018,"top":27.545761768432413},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"output"},"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"code\\": \\"s/__tc__/\\\\\\\\t/g\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"c06fbbc6-6875-439c-ad0a-823c41339bdd","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","errors":null,"id":9,"input_connections":{"style_cond|type_cond|pick_from_0|value":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Pick parameter value","outputs":[{"name":"float_param","type":"expression.json"}],"position":{"left":497.8587064818947,"top":824.4797977763942},"post_job_actions":{"HideDatasetActionfloat_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"float_param"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/pick_value/pick_value/0.2.0","tool_shed_repository":{"changeset_revision":"b19e21af9c52","name":"pick_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"style_cond\\": {\\"pick_style\\": \\"first_or_default\\", \\"__current_case__\\": 1, \\"type_cond\\": {\\"param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"pick_from\\": [{\\"__index__\\": 0, \\"value\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"default_value\\": \\"1.0\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"9fb45a2f-63d2-48c2-b2cb-295bc323a14c","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.8+galaxy0","errors":null,"id":10,"input_connections":{"header":{"id":2,"output_name":"output"},"output_options|alpha_ma":{"id":7,"output_name":"float_param"},"select_data|rep_factorName_0|rep_factorLevel_0|countsFile":{"id":0,"output_name":"output"},"select_data|rep_factorName_0|rep_factorLevel_1|countsFile":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool DESeq2","name":"batch_factors"},{"description":"runtime parameter for tool DESeq2","name":"output_options"}],"label":"Differential Analysis","name":"DESeq2","outputs":[{"name":"deseq_out","type":"tabular"},{"name":"plots","type":"pdf"},{"name":"counts_out","type":"tabular"}],"position":{"left":741.3738148942634,"top":199.67185694924424},"post_job_actions":{"HideDatasetActiondeseq_out":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"deseq_out"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.8+galaxy0","tool_shed_repository":{"changeset_revision":"9a882d108833","name":"deseq2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"esf_cond\\": {\\"esf\\": \\"\\", \\"__current_case__\\": 0}, \\"fit_type\\": \\"1\\", \\"outlier_replace_off\\": false, \\"outlier_filter_off\\": false, \\"auto_mean_filter_off\\": false, \\"prefilter_conditional\\": {\\"prefilter\\": \\"\\", \\"__current_case__\\": 1}, \\"use_beta_priors\\": false}, \\"batch_factors\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"header\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_options\\": {\\"output_selector\\": [\\"pdf\\", \\"normCounts\\"], \\"alpha_ma\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"select_data\\": {\\"how\\": \\"datasets_per_level\\", \\"__current_case__\\": 1, \\"rep_factorName\\": [{\\"__index__\\": 0, \\"factorName\\": \\"DEFactor\\", \\"rep_factorLevel\\": [{\\"__index__\\": 0, \\"factorLevel\\": \\"MainFactor\\", \\"countsFile\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"factorLevel\\": \\"BaseFactor\\", \\"countsFile\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}]}, \\"tximport\\": {\\"tximport_selector\\": \\"count\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.11.40.8+galaxy0","type":"tool","uuid":"45883538-b692-4e35-8dee-d08974682a9c","when":null,"workflow_outputs":[{"label":"DESeq2 Plots","output_name":"plots","uuid":"82fc30c6-1af6-415c-98dd-4ac28671dc63"},{"label":"DESeq2 Normalized Counts","output_name":"counts_out","uuid":"d7768a86-18fe-4752-946f-1b54e468d11e"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":11,"input_connections":{"components_1|param_type|component_value":{"id":7,"output_name":"float_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1030.3831032381008,"top":646.2395043126207},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c7<\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"346c524e-98f0-4b7e-8531-eb9dc6d697c8","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":12,"input_connections":{"components_1|param_type|component_value":{"id":9,"output_name":"float_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1037.5832128440811,"top":834.4804304279605},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"abs(c3)>\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"float\\", \\"__current_case__\\": 2, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"b802f520-01d7-4a42-9138-eaccb948d624","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/deg_annotate/deg_annotate/1.1.0","errors":null,"id":13,"input_connections":{"annotation":{"id":3,"output_name":"output"},"input_table":{"id":10,"output_name":"deseq_out"}},"inputs":[],"label":null,"name":"Annotate DESeq2/DEXSeq output tables","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1053.2118356859382,"top":253.53181990473982},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/deg_annotate/deg_annotate/1.1.0","tool_shed_repository":{"changeset_revision":"e98d4ab5b5bc","name":"deg_annotate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"advanced_parameters\\": {\\"gff_feature_type\\": \\"exon\\", \\"gff_feature_attribute\\": \\"gene_id\\", \\"gff_transcript_attribute\\": \\"transcript_id\\", \\"gff_attributes\\": \\"gene_biotype, gene_name\\"}, \\"annotation\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/dm6.len\\", \\"input_table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mode\\": \\"degseq\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"8c9667ec-283d-40d8-9fe0-a2c36d875d2b","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","errors":null,"id":14,"input_connections":{"infile":{"id":10,"output_name":"counts_out"}},"inputs":[],"label":null,"name":"Text reformatting","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1444.0049932961922,"top":643.7692280168025},"post_job_actions":{"HideDatasetActionoutfile":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"code\\": \\"END{print NF}\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"e99d5d73-f738-4977-9740-1159a3291a5c","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1","errors":null,"id":15,"input_connections":{"inputs":{"id":8,"output_name":"output"},"queries_0|inputs2":{"id":13,"output_name":"output"}},"inputs":[],"label":"Annotate DESeq2 table","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1353.5294001633204,"top":147.05412728617347},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"tabular"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Annotated DESeq2 results"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"inputs2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"5aca8071-b7e2-48b6-ba2e-ef258852578c","when":null,"workflow_outputs":[{"label":"Annotated DESeq2 results table","output_name":"out_file1","uuid":"15fc04aa-406f-40a7-8b85-f99467590be5"}]},"16":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":16,"input_connections":{"input1":{"id":14,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":1795.8106940526452,"top":744.1580154160025},"post_job_actions":{"HideDatasetActiontext_param":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"55b39be6-635d-491a-afd9-c8341f9932b4","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"Filter1","errors":null,"id":17,"input_connections":{"cond":{"id":11,"output_name":"out1"},"input":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":"Filter with p-adj threshold","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1605.546743310479,"top":371.3592826475832},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Genes filtered with adj p-value threshold"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"9cf4c012-7a52-4b47-bbec-4ad8275542ea","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.7","errors":null,"id":18,"input_connections":{"input":{"id":15,"output_name":"out_file1"},"lfc_thresh":{"id":9,"output_name":"float_param"},"signif_thresh":{"id":7,"output_name":"float_param"}},"inputs":[],"label":"Generate Valcanot plot of DE genes","name":"Volcano Plot","outputs":[{"name":"plot","type":"pdf"}],"position":{"left":2610.2086912391,"top":957.2308087191843},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.7","tool_shed_repository":{"changeset_revision":"5e08a1e22dbc","name":"volcanoplot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fdr_col\\": \\"7\\", \\"header\\": \\"yes\\", \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"label_col\\": \\"13\\", \\"labels\\": {\\"label_select\\": \\"signif\\", \\"__current_case__\\": 0, \\"top_num\\": \\"10\\"}, \\"lfc_col\\": \\"3\\", \\"lfc_thresh\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out_options\\": {\\"rscript_out\\": false}, \\"plot_options\\": {\\"boxes\\": false, \\"title\\": null, \\"xlab\\": null, \\"ylab\\": null, \\"xmin\\": null, \\"xmax\\": null, \\"ymax\\": null, \\"legend\\": null, \\"legend_labs\\": \\"Down,Not Sig,Up\\"}, \\"pval_col\\": \\"6\\", \\"shape_col\\": \\"\\", \\"shape_or_facet\\": null, \\"signif_thresh\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.7","type":"tool","uuid":"cb314a3d-0b05-4d45-a961-591d59b3852b","when":null,"workflow_outputs":[{"label":"Volcano Plot of DE genes","output_name":"plot","uuid":"9c1cd777-caa5-4477-8a24-9708480bd693"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":19,"input_connections":{"components_1|param_type|component_value":{"id":16,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":2144.7794821932416,"top":725.294415795629},"post_job_actions":{"HideDatasetActionout1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"c1-c\\"}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"34f60b08-6d52-4cc0-8a91-dedaff1ca499","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"Filter1","errors":null,"id":20,"input_connections":{"cond":{"id":12,"output_name":"out1"},"input":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":"Filter with log2 FC threshold","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1910.07823962931,"top":474.9743920452724},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Genes filtered with adj p-value and log2(FC) thresholds"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"1f787252-7a94-43b6-bb66-9624923ed05c","when":null,"workflow_outputs":[{"label":"Significantly differentially expressed genes","output_name":"out_file1","uuid":"ca9ba83d-00e7-4778-a670-9a5605606cfc"}]},"21":{"annotation":"","content_id":"join1","errors":null,"id":21,"input_connections":{"input1":{"id":10,"output_name":"counts_out"},"input2":{"id":20,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2271.404780182028,"top":406.30668475687355},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/dm6.len\\", \\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"no_fill\\", \\"__current_case__\\": 0}, \\"header\\": \\"-H\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"\\", \\"unmatched\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"672e150d-ae77-41ae-9b66-2c3bf55bc701","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"Cut1","errors":null,"id":22,"input_connections":{"columnList":{"id":19,"output_name":"out1"},"input":{"id":21,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2560.320455929024,"top":649.1552708097212},"post_job_actions":{"HideDatasetActionout_file1":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Normalized counts for the most differentially expressed genes"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"699a7dd8-e3d0-42b9-bf2d-02e92d208c14","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.2.0+galaxy1","errors":null,"id":23,"input_connections":{"input1":{"id":22,"output_name":"out_file1"}},"inputs":[],"label":"Generate Heatmap of counts","name":"heatmap2","outputs":[{"name":"output1","type":"pdf"}],"position":{"left":2874.905386062883,"top":454.7381454908998},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Normalized counts Heatmap"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.2.0+galaxy1","tool_shed_repository":{"changeset_revision":"48b42ee61bc5","name":"ggplot2_heatmap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/dm6.len\\", \\"cluster_cond\\": {\\"cluster\\": \\"yes\\", \\"__current_case__\\": 0, \\"cluster_cols_rows\\": \\"both\\", \\"distance\\": \\"euclidean\\", \\"clustering\\": \\"complete\\"}, \\"colorchoice\\": {\\"type\\": \\"two\\", \\"__current_case__\\": 1, \\"color1\\": \\"#ffffff\\", \\"color2\\": \\"#ff0000\\"}, \\"image_file_format\\": \\"pdf\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"key\\": \\"\\", \\"labels\\": \\"columns\\", \\"title\\": \\"\\", \\"transform\\": \\"log2plus1\\", \\"zscore_cond\\": {\\"zscore\\": \\"none\\", \\"__current_case__\\": 0, \\"scale\\": \\"none\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.2.0+galaxy1","type":"tool","uuid":"b8b1d4c3-d8bf-41e1-93ab-00c021d2896f","when":null,"workflow_outputs":[{"label":"Heatmap of log transformed normalized counts","output_name":"output1","uuid":"7361cd0a-4247-4274-b10e-d208c3465fdb"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.2.0+galaxy1","errors":null,"id":24,"input_connections":{"input1":{"id":22,"output_name":"out_file1"}},"inputs":[],"label":"Generate Heatmap of Z-scores","name":"heatmap2","outputs":[{"name":"output1","type":"pdf"}],"position":{"left":2899.706231952903,"top":852.4572467969889},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"Z-scores Heatmap"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.2.0+galaxy1","tool_shed_repository":{"changeset_revision":"48b42ee61bc5","name":"ggplot2_heatmap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/dm6.len\\", \\"cluster_cond\\": {\\"cluster\\": \\"yes\\", \\"__current_case__\\": 0, \\"cluster_cols_rows\\": \\"both\\", \\"distance\\": \\"euclidean\\", \\"clustering\\": \\"complete\\"}, \\"colorchoice\\": {\\"type\\": \\"three\\", \\"__current_case__\\": 2, \\"color1\\": \\"#0000ff\\", \\"color2\\": \\"#ffffff\\", \\"color3\\": \\"#ff0000\\"}, \\"image_file_format\\": \\"pdf\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"key\\": \\"\\", \\"labels\\": \\"columns\\", \\"title\\": \\"\\", \\"transform\\": \\"none\\", \\"zscore_cond\\": {\\"zscore\\": \\"rows\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.2.0+galaxy1","type":"tool","uuid":"f4f25636-3963-4624-8047-cc5aa286ca0e","when":null,"workflow_outputs":[{"label":"Heatmap of Z-scores","output_name":"output1","uuid":"03b9a2ff-1520-41c1-8a53-0ada7b883e3f"}]}},"tags":["transcriptomics","RNAseq"],"uuid":"15d188a7-b028-4cf7-bb5d-13858e1f6e78","version":15},"readme":"# RNA-Seq Differential Expression Analysis with Visualization\\n\\nThis workflow works only with an experimental setup containing exactly 2 conditions with at least 2 replicates per condition.\\n\\n## Inputs dataset\\n\\n- Counts from changed condition: Counts from experimental condition or changed condition. For eg. counts from treatment or knockdown samples.\\n- Counts from reference condition: Counts from reference condition or base condition. For eg. counts from untreated or wildtype samples.\\n- Gene Annotaton: The same GTF file used for mapping and quantification. It is used to annotate the DESeq2 results table. Ideally, the GTF file should contain `gene_id`, `gene_biotype` and `gene_name` attributes.\\n\\n## Inputs values\\n\\n- Count files have header: Indicate whether your input count files have a header line. Usually, count files generated from featureCounts tool have a header line whereas count files from RNA-STAR do not have.\\n- Adjusted p-value threshold: All the genes with an adjusted p-value less than this value are considered as differentially expressed. With a value of 0.05, expect 5% of false positives in the differentially expressed genes list. If empty, a default value of 0.05 is used.\\n- log2 fold change threshold: All the genes with an absolute fold change (regarless of up or down regulation) more than this value are selected. A log2 FC of 3 equals to an absolute fold change of 8 (2³). If empty, a default value of 1.0 is used.\\n\\n## Processing\\n\\n- The workflow uses DESeq2 for performing differential expression analysis. In addition to the results table, it also produces normalized counts table.\\n- The results table is annotated with gene positions, biotypes, gene symbols.\\n- The annotated results table is further filtered with the input adjusted p-value and log2 fold change thresholds.\\n- A valcano plot is generated with top 10 significantly differentially expressed genes.\\n- A heatmap of log trasformed normalized counts and another heatmap of Z-scores is generated.\\n","changelog":"# Changelog\\n\\n## [0.4] - 2025-03-24\\n- Change the **Text transformation** tool output to tabular format so that the header file format matches with the DESeq2 output format\\n\\n## [0.3] - 2025-02-03\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.6` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/volcanoplot/volcanoplot/0.0.7`\\n\\n## [0.2] - 2024-11-18\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.7+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.8+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.1.3.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.2.0+galaxy1`\\n\\n## [0.1] - 2024-10-25\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## RNA-Seq Differential Expression Analysis with Visualization\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Counts from changed condition\\"]@{ shape: docs }\\n1[\\"ℹ️ Counts from reference condition\\"]@{ shape: docs }\\n2[\\"ℹ️ Count files have header\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Gene Annotaton\\"]@{ shape: doc }\\n4[\\"ℹ️ Adjusted p-value threshold\\"]@{ shape: lean-l }\\n5[\\"DE table header\\"]@{ shape: process }\\n6[\\"ℹ️ log2 fold change threshold\\"]@{ shape: lean-l }\\n7[\\"Pick parameter value\\"]@{ shape: process }\\n4 --> 7\\n8[\\"Text transformation\\"]@{ shape: process }\\n5 --> 8\\n9[\\"Pick parameter value\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Differential Analysis\\"]@{ shape: process }\\n2 --> 10\\n7 --> 10\\n0 --> 10\\n1 --> 10\\n11[\\"Compose text parameter value\\"]@{ shape: process }\\n7 --> 11\\n12[\\"Compose text parameter value\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Annotate DESeq2/DEXSeq output tables\\"]@{ shape: process }\\n3 --> 13\\n10 --> 13\\n14[\\"Text reformatting\\"]@{ shape: process }\\n10 --> 14\\n15[\\"Annotate DESeq2 table\\"]@{ shape: process }\\n8 --> 15\\n13 --> 15\\n16[\\"Parse parameter value\\"]@{ shape: process }\\n14 --> 16\\n17[\\"Filter with p-adj threshold\\"]@{ shape: process }\\n11 --> 17\\n15 --> 17\\n18[\\"Generate Valcanot plot of DE genes\\"]@{ shape: process }\\n15 --> 18\\n9 --> 18\\n7 --> 18\\n19[\\"Compose text parameter value\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Filter with log2 FC threshold\\"]@{ shape: process }\\n12 --> 20\\n17 --> 20\\n21[\\"Join two Datasets\\"]@{ shape: process }\\n10 --> 21\\n20 --> 21\\n22[\\"Cut\\"]@{ shape: process }\\n19 --> 22\\n21 --> 22\\n23[\\"Generate Heatmap of counts\\"]@{ shape: process }\\n22 --> 23\\n24[\\"Generate Heatmap of Z-scores\\"]@{ shape: process }\\n22 --> 24\\n```\\n","updated":"2025-03-24T00:00:00","trsID":"#workflow/github.com/iwc-workflows/rnaseq-de/main","dockstore_id":29329,"categories":[],"collections":["Transcriptomics"],"doi":"10.5281/zenodo.15078905","tests":[{"doc":"Test outline for RNAseq_DE_filtering_plotting","job":{"Gene Annotaton":{"class":"File","location":"https://zenodo.org/records/14056162/files/Saccharomyces_cerevisiae.R64-1-1.113.gtf","filetype":"gtf"},"Counts from changed condition":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"SRR5085169 Counts Table","location":"https://zenodo.org/records/14056162/files/SRR5085169.tabular"},{"class":"File","identifier":"SRR5085170 Counts Table","location":"https://zenodo.org/records/14056162/files/SRR5085170.tabular"}]},"Counts from reference condition":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"SRR5085167 Counts Table","location":"https://zenodo.org/records/14056162/files/SRR5085167.tabular"},{"class":"File","identifier":"SRR5085168 Counts Table","location":"https://zenodo.org/records/14056162/files/SRR5085168.tabular"}]},"Count files have header":true,"Adjusted p-value threshold":0.1,"log2 fold change threshold":0.5},"outputs":{"Annotated DESeq2 results table":{"has_text_matching":{"expression":"YKL081W\\t264.71[0-9]*\\t-0.54[0-9]*\\t0.15[0-9]*\\t-3.46[0-9]*\\t0.00[0-9]*\\t0.09[0-9]*\\tchrXI\\t282890\\t284455\\t+\\tprotein_coding\\tTEF4"}},"Heatmap of Z-scores":{"has_size":{"value":19510,"delta":1000}},"DESeq2 Normalized Counts":{"has_text_matching":{"expression":"YKL081W\\t313.76[0-9]*\\t322.37[0-9]*\\t223.48[0-9]*\\t199.24[0-9]*"}},"DESeq2 Plots":{"has_size":{"value":1193021,"delta":60000}},"Volcano Plot of DE genes":{"has_size":{"value":301346,"delta":15000}},"Heatmap of log transformed normalized counts":{"has_size":{"value":19501,"delta":1000}}}}]}],"path":"./workflows/transcriptomics/rnaseq-de"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/goseq-go-kegg-enrichment-analsis.ga","testParameterFiles":["/goseq-go-kegg-enrichment-analsis-tests.yml"],"authors":[{"name":"Amirhossein Naghsh Nilchi","orcid":"0000-0003-1329-3044"}],"definition":{"a_galaxy_workflow":"true","annotation":"Performs functional enrichment analysis of gene sets using GOseq, identifying over-represented Gene Ontology terms and KEGG pathways. This workflow accounts for gene length bias in RNA-seq data when calculating enrichment statistics. Input requires differentially expressed genes and gene lengths. Generates comprehensive tables and visualizations of enriched GO terms across all three ontologies (Biological Process, Molecular Function, Cellular Component) as well as KEGG pathway enrichment results.","comments":[],"creator":[{"class":"Person","identifier":"https://orcid.org/0000-0003-1329-3044","name":"Amirhossein Naghsh Nilchi"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Gene Ontology and KEGG Pathway Enrichment Analysis","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Select which genome your data is from","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Select which genome your data is from","name":"Select genome to use"}],"label":"Select genome to use","name":"Input parameter","outputs":[],"position":{"left":1.3275521636144472,"top":485.4318730139212},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"621e1d7b-7884-4adb-911e-cdd249855486","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"c5c1a856-ae08-4281-816b-d0e70a3d3373"}]},"1":{"annotation":"input1: table with two columns. 1 gene 2 T/F based on Differential Expression Analysis","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"input1: table with two columns. 1 gene 2 T/F based on Differential Expression Analysis","name":"Differential expression result"}],"label":"Differential expression result","name":"Input dataset","outputs":[],"position":{"left":337.51186934168175,"top":496.3229330319098},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"52bf21e0-e1d7-41dd-87c2-ba4808a77b5d","when":null,"workflow_outputs":[]},"2":{"annotation":"Select the format of your genes","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Select the format of your genes","name":"Select gene ID format"}],"label":"Select gene ID format","name":"Input parameter","outputs":[],"position":{"left":0,"top":603.4685835542876},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"03c15189-4a0c-44fa-b4bc-717db6650797","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"61021ac9-ae04-486a-bb69-583314324551"}]},"3":{"annotation":"input2: table with two columns. 1 gene 2 gene length","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"input2: table with two columns. 1 gene 2 gene length","name":"gene length"}],"label":"gene length","name":"Input dataset","outputs":[],"position":{"left":346.07461300595065,"top":583.8602444483994},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"def1d40a-71dc-4c0b-bd8d-cb0cd00b1ffa","when":null,"workflow_outputs":[]},"4":{"annotation":"a table with two columns:\\n1. Pathway ID\\n2. Pathway name\\nlike:\\nID            Name\\n01100\\tMetabolic pathways - mmus\\n01200\\tCarbon metabolism - mmus\\n\\nyou can get it from KEGG\\nhttps://rest.kegg.jp/list/pathway/mmu","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"a table with two columns:\\n1. Pathway ID\\n2. Pathway name\\nlike:\\nID            Name\\n01100\\tMetabolic pathways - mmus\\n01200\\tCarbon metabolism - mmus\\n\\nyou can get it from KEGG\\nhttps://rest.kegg.jp/list/pathway/mmu","name":"KEGG pathways"}],"label":"KEGG pathways","name":"Input dataset","outputs":[],"position":{"left":1041.1943582885483,"top":1074.3811050057798},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ce8469f0-065c-4c4d-a186-bea4ed476868","when":null,"workflow_outputs":[]},"5":{"annotation":"input1: table with two columns. 1 gene 2 T/F based on DEA\\n\\ninput2: table with two columns. 1 gene 2 gene length","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","errors":null,"id":5,"input_connections":{"categorySource|gene_id":{"id":2,"output_name":"output"},"categorySource|genome":{"id":0,"output_name":"output"},"dge_file":{"id":1,"output_name":"output"},"length_file":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool goseq","name":"categorySource"},{"description":"runtime parameter for tool goseq","name":"categorySource"}],"label":"goseq - Cellular Component","name":"goseq","outputs":[{"name":"wallenius_tab","type":"tabular"},{"name":"top_plot","type":"pdf"},{"name":"cat_genes_tab","type":"tabular"}],"position":{"left":749.3281584980234,"top":0},"post_job_actions":{"RenameDatasetActioncat_genes_tab":{"action_arguments":{"newname":"goseq Differential Genes in each Category (Cellular Component)"},"action_type":"RenameDatasetAction","output_name":"cat_genes_tab"},"RenameDatasetActiontop_plot":{"action_arguments":{"newname":"goseq Top Categories plot (Cellular Component)"},"action_type":"RenameDatasetAction","output_name":"top_plot"},"RenameDatasetActionwallenius_tab":{"action_arguments":{"newname":"goseq Enrichment Table (Cellular Component)"},"action_type":"RenameDatasetAction","output_name":"wallenius_tab"},"TagDatasetActioncat_genes_tab":{"action_arguments":{"tags":"GO-CC"},"action_type":"TagDatasetAction","output_name":"cat_genes_tab"},"TagDatasetActiontop_plot":{"action_arguments":{"tags":"GO-CC"},"action_type":"TagDatasetAction","output_name":"top_plot"},"TagDatasetActionwallenius_tab":{"action_arguments":{"tags":"GO-CC"},"action_type":"TagDatasetAction","output_name":"wallenius_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","tool_shed_repository":{"changeset_revision":"602de62d995b","name":"goseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"p_adj_method\\": \\"BH\\", \\"use_genes_without_cat\\": false}, \\"categorySource\\": {\\"catSource\\": \\"getgo\\", \\"__current_case__\\": 0, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_id\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fetchcats\\": [\\"GO:CC\\"]}, \\"dge_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"wallenius\\": true, \\"hypergeometric\\": false, \\"repcnt\\": \\"0\\"}, \\"out\\": {\\"topgo_plot\\": true, \\"make_plots\\": false, \\"cat_genes\\": true, \\"rdata_out\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.50.0+galaxy0","type":"tool","uuid":"82335d35-73d1-466a-b060-67dbc7cf5c98","when":null,"workflow_outputs":[{"label":"goseq Enrichment Table (Cellular Component)","output_name":"wallenius_tab","uuid":"d54a7b06-aed2-4af0-87ed-a5e3cbaf19f7"},{"label":"goseq Top Categories plot (Cellular Component)","output_name":"top_plot","uuid":"aae04e11-b7b4-49c7-91cb-1f873e437619"},{"label":"goseq Differential Genes in each Category (Cellular Component)","output_name":"cat_genes_tab","uuid":"bff6fd1b-d67b-4ca1-a108-853e0ab61377"}]},"6":{"annotation":"input1: table with two columns. 1 gene 2 T/F based on DEA (is it DEG?)\\n\\ninput2: table with two columns. 1 gene 2 gene length","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","errors":null,"id":6,"input_connections":{"categorySource|gene_id":{"id":2,"output_name":"output"},"categorySource|genome":{"id":0,"output_name":"output"},"dge_file":{"id":1,"output_name":"output"},"length_file":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool goseq","name":"categorySource"},{"description":"runtime parameter for tool goseq","name":"categorySource"}],"label":"goseq - Biological Process","name":"goseq","outputs":[{"name":"wallenius_tab","type":"tabular"},{"name":"top_plot","type":"pdf"},{"name":"cat_genes_tab","type":"tabular"}],"position":{"left":748.4594510204071,"top":402.6739640536188},"post_job_actions":{"RenameDatasetActioncat_genes_tab":{"action_arguments":{"newname":"goseq Differential Genes in each Category (Biological Process)"},"action_type":"RenameDatasetAction","output_name":"cat_genes_tab"},"RenameDatasetActiontop_plot":{"action_arguments":{"newname":"goseq Top Categories plot (Biological Process)"},"action_type":"RenameDatasetAction","output_name":"top_plot"},"RenameDatasetActionwallenius_tab":{"action_arguments":{"newname":"goseq Enrichment Table (Biological Process)"},"action_type":"RenameDatasetAction","output_name":"wallenius_tab"},"TagDatasetActioncat_genes_tab":{"action_arguments":{"tags":"GO-BP"},"action_type":"TagDatasetAction","output_name":"cat_genes_tab"},"TagDatasetActiontop_plot":{"action_arguments":{"tags":"GO-BP"},"action_type":"TagDatasetAction","output_name":"top_plot"},"TagDatasetActionwallenius_tab":{"action_arguments":{"tags":"GO-BP"},"action_type":"TagDatasetAction","output_name":"wallenius_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","tool_shed_repository":{"changeset_revision":"602de62d995b","name":"goseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"p_adj_method\\": \\"BH\\", \\"use_genes_without_cat\\": false}, \\"categorySource\\": {\\"catSource\\": \\"getgo\\", \\"__current_case__\\": 0, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_id\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fetchcats\\": [\\"GO:BP\\"]}, \\"dge_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"wallenius\\": true, \\"hypergeometric\\": false, \\"repcnt\\": \\"0\\"}, \\"out\\": {\\"topgo_plot\\": true, \\"make_plots\\": false, \\"cat_genes\\": true, \\"rdata_out\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.50.0+galaxy0","type":"tool","uuid":"41f3b5f7-6510-4b49-a95f-2a7b438e4351","when":null,"workflow_outputs":[{"label":"goseq Enrichment Table (Biological Process)","output_name":"wallenius_tab","uuid":"6922dc30-d93d-4d77-ad26-b7fbd9c6bf13"},{"label":"goseq Differential Genes in each Category (Biological Process)","output_name":"cat_genes_tab","uuid":"d5ce0788-ec4d-48ab-934f-14c3fc98515e"},{"label":"goseq Top Categories plot (Biological Process)","output_name":"top_plot","uuid":"4cb0d747-f917-4cd6-82b2-7d41180e9333"}]},"7":{"annotation":"input1: table with two columns. 1 gene 2 T/F based on DEA (is it DEG?)\\n\\ninput2: table with two columns. 1 gene 2 gene length","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","errors":null,"id":7,"input_connections":{"categorySource|gene_id":{"id":2,"output_name":"output"},"categorySource|genome":{"id":0,"output_name":"output"},"dge_file":{"id":1,"output_name":"output"},"length_file":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool goseq","name":"categorySource"},{"description":"runtime parameter for tool goseq","name":"categorySource"}],"label":"goseq - Molecular Function","name":"goseq","outputs":[{"name":"wallenius_tab","type":"tabular"},{"name":"top_plot","type":"pdf"},{"name":"cat_genes_tab","type":"tabular"}],"position":{"left":749.0118758398745,"top":797.1539759402796},"post_job_actions":{"RenameDatasetActioncat_genes_tab":{"action_arguments":{"newname":"goseq Differential Genes in each Category (Molecular Function)"},"action_type":"RenameDatasetAction","output_name":"cat_genes_tab"},"RenameDatasetActiontop_plot":{"action_arguments":{"newname":"goseq Top Categories plot (Molecular Function)"},"action_type":"RenameDatasetAction","output_name":"top_plot"},"RenameDatasetActionwallenius_tab":{"action_arguments":{"newname":"goseq Enrichment Table (Molecular Function)"},"action_type":"RenameDatasetAction","output_name":"wallenius_tab"},"TagDatasetActioncat_genes_tab":{"action_arguments":{"tags":"GO-MF"},"action_type":"TagDatasetAction","output_name":"cat_genes_tab"},"TagDatasetActiontop_plot":{"action_arguments":{"tags":"GO-MF"},"action_type":"TagDatasetAction","output_name":"top_plot"},"TagDatasetActionwallenius_tab":{"action_arguments":{"tags":"GO-MF"},"action_type":"TagDatasetAction","output_name":"wallenius_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","tool_shed_repository":{"changeset_revision":"602de62d995b","name":"goseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"p_adj_method\\": \\"BH\\", \\"use_genes_without_cat\\": false}, \\"categorySource\\": {\\"catSource\\": \\"getgo\\", \\"__current_case__\\": 0, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_id\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fetchcats\\": [\\"GO:MF\\"]}, \\"dge_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"wallenius\\": true, \\"hypergeometric\\": false, \\"repcnt\\": \\"0\\"}, \\"out\\": {\\"topgo_plot\\": true, \\"make_plots\\": false, \\"cat_genes\\": true, \\"rdata_out\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.50.0+galaxy0","type":"tool","uuid":"f7b4ad99-12fe-47a8-9ec0-aa9ff82fdf04","when":null,"workflow_outputs":[{"label":"goseq Top Categories plot (Molecular Function)","output_name":"top_plot","uuid":"80736ba0-9ee6-4e54-bb4f-1af639e2a766"},{"label":"goseq Differential Genes in each Category (Molecular Function)","output_name":"cat_genes_tab","uuid":"1de602fd-b5ea-4181-bf5d-c920b344aa72"},{"label":"goseq Enrichment Table (Molecular Function)","output_name":"wallenius_tab","uuid":"ced68bff-2dea-4af9-9135-e32f9efa5c7a"}]},"8":{"annotation":"input1: table with two columns. 1 gene 2 T/F based on DEA (is it DEG?)\\n\\ninput2: table with two columns. 1 gene 2 gene length","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","errors":null,"id":8,"input_connections":{"categorySource|gene_id":{"id":2,"output_name":"output"},"categorySource|genome":{"id":0,"output_name":"output"},"dge_file":{"id":1,"output_name":"output"},"length_file":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool goseq","name":"categorySource"},{"description":"runtime parameter for tool goseq","name":"categorySource"}],"label":"goseq - KEGG","name":"goseq","outputs":[{"name":"wallenius_tab","type":"tabular"},{"name":"cat_genes_tab","type":"tabular"}],"position":{"left":749.7009351947241,"top":1212.027572682219},"post_job_actions":{"RenameDatasetActioncat_genes_tab":{"action_arguments":{"newname":"goseq Differential Genes in each Category (KEGG)"},"action_type":"RenameDatasetAction","output_name":"cat_genes_tab"},"TagDatasetActioncat_genes_tab":{"action_arguments":{"tags":"KEGG"},"action_type":"TagDatasetAction","output_name":"cat_genes_tab"},"TagDatasetActionwallenius_tab":{"action_arguments":{"tags":"KEGG"},"action_type":"TagDatasetAction","output_name":"wallenius_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.50.0+galaxy0","tool_shed_repository":{"changeset_revision":"602de62d995b","name":"goseq","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"p_adj_method\\": \\"BH\\", \\"use_genes_without_cat\\": false}, \\"categorySource\\": {\\"catSource\\": \\"getgo\\", \\"__current_case__\\": 0, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gene_id\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fetchcats\\": [\\"KEGG\\"]}, \\"dge_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"length_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"wallenius\\": true, \\"hypergeometric\\": false, \\"repcnt\\": \\"0\\"}, \\"out\\": {\\"topgo_plot\\": false, \\"make_plots\\": false, \\"cat_genes\\": true, \\"rdata_out\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.50.0+galaxy0","type":"tool","uuid":"4b153ed9-7cca-4f19-9555-d7c8f41a0745","when":null,"workflow_outputs":[{"label":"goseq Differential Genes in each Category (KEGG)","output_name":"cat_genes_tab","uuid":"cc754b2d-28a9-42d9-8474-63e47fe5a319"}]},"9":{"annotation":"","content_id":"join1","errors":null,"id":9,"input_connections":{"input1":{"id":8,"output_name":"wallenius_tab"},"input2":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Join two Datasets","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1340.377060477933,"top":1170.345367972186},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"goseq Enrichment Table (KEGG)"},"action_type":"RenameDatasetAction","output_name":"out_file1"},"TagDatasetActionout_file1":{"action_arguments":{"tags":"KEGG"},"action_type":"TagDatasetAction","output_name":"out_file1"}},"tool_id":"join1","tool_state":"{\\"field1\\": \\"1\\", \\"field2\\": \\"1\\", \\"fill_empty_columns\\": {\\"fill_empty_columns_switch\\": \\"no_fill\\", \\"__current_case__\\": 0}, \\"header\\": \\"-H\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"partial\\": \\"-p\\", \\"unmatched\\": \\"-u\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3","type":"tool","uuid":"035755f9-8e41-450e-a813-a3be2bae6377","when":null,"workflow_outputs":[{"label":"goseq Enrichment Table (KEGG)","output_name":"out_file1","uuid":"5bb823d3-9fab-417a-82c8-65a6b74913a2"}]}},"tags":[],"uuid":"84a63bc6-a96a-4263-a1e5-bc140f412e42","version":5},"readme":"# Gene Ontology and KEGG Pathway Enrichment Analysis\\n\\n## Inputs dataset\\n\\nThe workflow need the following inputs:\\n- **The DEG file:**\\n    - A tabular file with first column the gene symbol and second column a boolean value whether the gene is a differentially expressed gene or not. \\n- **The gene length file:**\\n    - A tabular file with first column the gene symbol and second column the gene length of the genes.\\n    - You can create this file with **[Gene length and GC content](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Flength_and_gc_content%2Flength_and_gc_content%2F0.1.2&version=latest)** tool. You will need a GTF file as input.\\n    - If you are using **[featureCounts](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Ffeaturecounts%2Ffeaturecounts%2F2.0.6%2Bgalaxy0&version=latest)** you can set `Create gene-length file` to yes and get gene length as separate output.\\n- **The KEGG file:**\\n    - A tabular file with first column the Pathway ID and second column the Pathway name like: \\n        -   ```\\n            ID  Name\\n            01100   Metabolic pathways - mmus\\n            01200   Carbon metabolism - mmus \\n            ```\\n    - You can get this information from the KEGG database. For example:\\n        - mouse: https://rest.kegg.jp/list/pathway/mmu\\n        - human: https://rest.kegg.jp/list/pathway/hsa\\n- **Genome:** Select one of the available genomes\\n- **Gene ID format:** Select the format of your input genes (Ensembl, Entrez, or Symbol)\\n## Processing\\n\\n- The workflow will do a simple enrichment analysis with taking into account the gene length\\n- The output will be 3 files `GO table`, `Top ontology plot` and `DE genes in each category` for Cellular Component, Biological Processes, and Molecular Function ontologies and `KEGG table` and `DE genes in each KEGG Pathways`\\n\\n## Contribution\\n\\n@nilchia wrote the workflow and the tests.","changelog":"# Changelog\\n\\nAll notable changes to this workflow will be documented in this file.\\n\\n\\n## [0.1] - 2024-11-03\\n\\n### Added\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Gene Ontology and KEGG Pathway Enrichment Analysis\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Select genome to use\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Differential expression result\\"]@{ shape: doc }\\n2[\\"ℹ️ Select gene ID format\\"]@{ shape: lean-l }\\n3[\\"ℹ️ gene length\\"]@{ shape: doc }\\n4[\\"ℹ️ KEGG pathways\\"]@{ shape: doc }\\n5[\\"goseq - Cellular Component\\"]@{ shape: process }\\n2 --> 5\\n0 --> 5\\n1 --> 5\\n3 --> 5\\n6[\\"goseq - Biological Process\\"]@{ shape: process }\\n2 --> 6\\n0 --> 6\\n1 --> 6\\n3 --> 6\\n7[\\"goseq - Molecular Function\\"]@{ shape: process }\\n2 --> 7\\n0 --> 7\\n1 --> 7\\n3 --> 7\\n8[\\"goseq - KEGG\\"]@{ shape: process }\\n2 --> 8\\n0 --> 8\\n1 --> 8\\n3 --> 8\\n9[\\"Join two Datasets\\"]@{ shape: process }\\n8 --> 9\\n4 --> 9\\n```\\n","updated":"2024-11-03T00:00:00","trsID":"#workflow/github.com/iwc-workflows/goseq/main","dockstore_id":29249,"categories":[],"collections":["Transcriptomics"],"doi":"10.5281/zenodo.15078481","tests":[{"doc":"Test outline for goseq-go-kegg-enrichment-analsis","job":{"Differential expression result":{"class":"File","location":"https://zenodo.org/records/14033254/files/de-result.tabular","filetype":"tabular"},"gene length":{"class":"File","location":"https://zenodo.org/records/14033254/files/gene-length.tabular","filetype":"tabular"},"KEGG pathways":{"class":"File","location":"https://zenodo.org/records/14033254/files/kegg-pathways.tabular","filetype":"tabular"},"Select genome to use":"mm10","Select gene ID format":"ensGene"},"outputs":{"goseq Enrichment Table (Cellular Component)":{"asserts":[{"that":"has_text_matching","expression":"category\\tover_represented_pvalue\\tunder_represented_pvalue\\tnumDEInCat\\tnumInCat\\tterm\\tontology\\tp_adjust_over_represented\\tp_adjust_under_represented"},{"that":"has_text_matching","expression":"GO:0005737\\t4.79[0-9]+e-08\\t1\\t4810\\t8312\\tcytoplasm\\tCC\\t9.31[0-9]+e-05\\t1"},{"that":"has_text_matching","expression":"over_represented_pvalue"},{"that":"has_text_matching","expression":"under_represented_pvalue"}]},"goseq Top Categories plot (Cellular Component)":{"asserts":[{"that":"has_size","size":7078}]},"goseq Differential Genes in each Category (Cellular Component)":{"asserts":[{"that":"has_text_matching","expression":"category\\tde_genes"},{"that":"has_text_matching","expression":"GO:0005737\\tENSMUSG00000045545,ENSMUSG00000029802,ENSMUSG00000058056"}]},"goseq Differential Genes in each Category (Biological Process)":{"asserts":[{"that":"has_text_matching","expression":"category\\tde_genes"},{"that":"has_text_matching","expression":"GO:0043254\\tENSMUSG00000023951,ENSMUSG00000032060,ENSMUSG00000031284,"}]},"goseq Top Categories plot (Biological Process)":{"asserts":[{"that":"has_size","size":7092}]},"goseq Enrichment Table (Biological Process)":{"asserts":[{"that":"has_text_matching","expression":"category\\tover_represented_pvalue\\tunder_represented_pvalue\\tnumDEInCat\\tnumInCat\\tterm\\tontology\\tp_adjust_over_represented\\tp_adjust_under_represented"},{"that":"has_text_matching","expression":"GO:0043254\\t3.41[0-9]+e-07\\t1\\t239\\t345\\tregulation of protein-containing complex assembly\\tBP\\t0.0034[0-9]+\\t1"},{"that":"has_text_matching","expression":"over_represented_pvalue"},{"that":"has_text_matching","expression":"under_represented_pvalue"}]},"goseq Differential Genes in each Category (KEGG)":{"asserts":[{"that":"has_text_matching","expression":"category\\tde_genes"},{"that":"has_text_matching","expression":"04540\\tENSMUSG00000063358,ENSMUSG00000045136,ENSMUSG00000025856,ENSMUSG00000028019"}]},"goseq Top Categories plot (Molecular Function)":{"asserts":[{"that":"has_size","size":7166}]},"goseq Enrichment Table (Molecular Function)":{"asserts":[{"that":"has_text_matching","expression":"category\\tover_represented_pvalue\\tunder_represented_pvalue\\tnumDEInCat\\tnumInCat\\tterm\\tontology\\tp_adjust_over_represented\\tp_adjust_under_represented"},{"that":"has_text_matching","expression":"GO:0005515\\t1.4094[0-9]+e-07\\t1\\t3758\\t6415\\tprotein binding\\tMF\\t0.0006[0-9]+\\t1"},{"that":"has_text_matching","expression":"over_represented_pvalue"},{"that":"has_text_matching","expression":"under_represented_pvalue"}]},"goseq Differential Genes in each Category (Molecular Function)":{"asserts":[{"that":"has_text_matching","expression":"category\\tde_genes"},{"that":"has_text_matching","expression":"GO:0005515\\tENSMUSG00000045545,ENSMUSG00000029802,ENSMUSG00000058056,ENSMUSG00000004044,"}]},"goseq Enrichment Table (KEGG)":{"asserts":[{"that":"has_text_matching","expression":"category\\tover_represented_pvalue\\tunder_represented_pvalue\\tnumDEInCat\\tnumInCat\\tp_adjust_over_represented\\tp_adjust_under_represented\\tID\\tName"},{"that":"has_text_matching","expression":"04540\\t0.0031[0-9]+\\t0.99[0-9]+\\t47\\t61\\t0.46[0-9]+\\t1\\t04540\\tGap junction - mmus"}]}}}]}],"path":"./workflows/transcriptomics/goseq"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Genome_annotation_with_maker_short.ga","testParameterFiles":["/Genome_annotation_with_maker_short_tests.yml"],"authors":[{"name":"Romane Libouban","email":"romane.libouban@irisa.fr","orcid":"0009-0001-4920-9951"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow allows for genome annotation using Maker and evaluates the quality of the annotation.","comments":[{"child_steps":[13],"color":"pink","data":{"title":"Visualization"},"id":6,"position":[1620,820],"size":[240,244.8],"type":"frame"},{"child_steps":[1,4,5,2,3],"color":"yellow","data":{"title":"Inputs"},"id":0,"position":[10,440],"size":[240,682.6],"type":"frame"},{"child_steps":[6,7],"color":"green","data":{"title":"Genome quality evaluation"},"id":1,"position":[540,0],"size":[470,478],"type":"frame"},{"child_steps":[8],"color":"red","data":{"title":"Annotation with Maker"},"id":2,"position":[450,530],"size":[240,742],"type":"frame"},{"child_steps":[9,12],"color":"turquoise","data":{"title":"Evaluation - Predicted proteins from annotation"},"id":3,"position":[1080,220],"size":[490,580.4],"type":"frame"},{"child_steps":[10],"color":"lime","data":{"title":"Annotation statistics"},"id":4,"position":[930,830],"size":[240,254.8],"type":"frame"},{"child_steps":[11],"color":"black","data":{"title":"Improving gene naming"},"id":5,"position":[1300,1020],"size":[240,224.4],"type":"frame"}],"creator":[{"class":"Person","identifier":"0009-0001-4920-9951","name":"Romane Libouban"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Genome annotation with Maker","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"BUSCO lineage","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"BUSCO lineage","name":"BUSCO lineage"}],"label":"BUSCO lineage","name":"Input parameter","outputs":[],"position":{"left":0,"top":140},"tool_id":null,"tool_state":"{\\"multiple\\": false, \\"validators\\": [], \\"restrictions\\": [\\"aconoidasida_odb10\\", \\"acidobacteria_odb10\\", \\"actinobacteria_class_odb10\\", \\"alphaproteobacteria_odb10\\", \\"amoebozoa_odb10\\", \\"annelida_odb10\\", \\"apicomplexa_odb10\\", \\"arthropoda_odb10\\", \\"ascomycota_odb10\\", \\"bacillariophyta_odb10\\", \\"bacilli_odb10\\", \\"bacteroidia_odb10\\", \\"bacteroidetes_odb10\\", \\"bacteroidetes-chlorobi_group_odb10\\", \\"bacteroidales_odb10\\", \\"basidiomycota_odb10\\", \\"betaproteobacteria_odb10\\", \\"boletales_odb10\\", \\"bryophyta_odb10\\", \\"burkholderiales_odb10\\", \\"campylobacterales_odb10\\", \\"capnodiales_odb10\\", \\"carnivora_odb10\\", \\"cellvibrionales_odb10\\", \\"cetartiodactyla_odb10\\", \\"chaetothyriales_odb10\\", \\"chlamydiae_odb10\\", \\"chlorobi_odb10\\", \\"chloroflexi_odb10\\", \\"chondrichthyes_odb10\\", \\"chordata_odb10\\", \\"chromatiales_odb10\\", \\"chroococcales_odb10\\", \\"clostridia_odb10\\", \\"clostridiales_odb10\\", \\"coccidia_odb10\\", \\"coriobacteriales_odb10\\", \\"coriobacteriia_odb10\\", \\"corynebacteriales_odb10\\", \\"corynebacteriia_odb10\\", \\"cyanobacteria_odb10\\", \\"cytophagia_odb10\\", \\"cytophagales_odb10\\", \\"delta-epsilon-subdivisions_odb10\\", \\"deltaproteobacteria_odb10\\", \\"desulfobacterales_odb10\\", \\"desulfurococcales_odb10\\", \\"desulfuromonadales_odb10\\", \\"desulfovibrionales_odb10\\", \\"diptera_odb10\\", \\"dinoflagellata_odb10\\", \\"dothideomycetes_odb10\\", \\"embryophyta_odb10\\", \\"endopterygota_odb10\\", \\"enterobacterales_odb10\\", \\"entomoplasmatales_odb10\\", \\"epsilonproteobacteria_odb10\\", \\"euarchontoglires_odb10\\", \\"eudicots_odb10\\", \\"eukaryota_odb10\\", \\"euryarchaeota_odb10\\", \\"eurotiomycetes_odb10\\", \\"eurotiales_odb10\\", \\"eutheria_odb10\\", \\"fabales_odb10\\", \\"flavobacteriales_odb10\\", \\"flavobacteriia_odb10\\", \\"fungi_odb10\\", \\"fusobacteria_odb10\\", \\"fusobacteriales_odb10\\", \\"gammaproteobacteria_odb10\\", \\"glomerellales_odb10\\", \\"glomeromycota_odb10\\", \\"gnathostomata_odb10\\", \\"haloferacales_odb10\\", \\"halobacteriales_odb10\\", \\"halobacteria_odb10\\", \\"helotiales_odb10\\", \\"hemiptera_odb10\\", \\"hexapoda_odb10\\", \\"hymenoptera_odb10\\", \\"hypocreales_odb10\\", \\"lactobacillales_odb10\\", \\"laurasiatheria_odb10\\", \\"legionellales_odb10\\", \\"leotiomycetes_odb10\\", \\"liliopsida_odb10\\", \\"mammalia_odb10\\", \\"metazoa_odb10\\", \\"methanobacteria_odb10\\", \\"methanococcales_odb10\\", \\"methanomicrobiales_odb10\\", \\"micrococcales_odb10\\", \\"microsporidia_odb10\\", \\"mollicutes_odb10\\", \\"mucorales_odb10\\", \\"mucoromycota_odb10\\", \\"mycoplasmatales_odb10\\", \\"natrialbales_odb10\\", \\"neisseriales_odb10\\", \\"nitrosomonadales_odb10\\", \\"nostocales_odb10\\", \\"oceanospirillales_odb10\\", \\"onygenales_odb10\\", \\"oscillatoriales_odb10\\", \\"passeriformes_odb10\\", \\"pasteurellales_odb10\\", \\"planctomycetes_odb10\\", \\"plantae_odb10\\", \\"plasmodium_odb10\\", \\"pleosporales_odb10\\", \\"poales_odb10\\", \\"polyporales_odb10\\", \\"primates_odb10\\", \\"propionibacteriales_odb10\\", \\"pseudomonadales_odb10\\", \\"pseudomonadota_odb10\\", \\"reptilia_odb10\\", \\"rhodobacterales_odb10\\", \\"rhizaria_odb10\\", \\"rhizobiales_odb10\\", \\"rhizobium-agrobacterium_group_odb10\\", \\"rhodospirillales_odb10\\", \\"rickettsiales_odb10\\", \\"saccharomycetes_odb10\\", \\"sauropsida_odb10\\", \\"selenomonadales_odb10\\", \\"solanales_odb10\\", \\"sphingobacteriia_odb10\\", \\"sphingomonadales_odb10\\", \\"spirochaetes_odb10\\", \\"spirochaetales_odb10\\", \\"spirochaetia_odb10\\", \\"streptomycetales_odb10\\", \\"streptosporangiales_odb10\\", \\"stramenopiles_odb10\\", \\"sulfolobales_odb10\\", \\"synechococcales_odb10\\", \\"synergistetes_odb10\\", \\"tardigrada_odb10\\", \\"tenericutes_odb10\\", \\"tetrapoda_odb10\\", \\"thaumarchaeota_odb10\\", \\"thermoanaerobacterales_odb10\\", \\"thermoplasmata_odb10\\", \\"thermoproteales_odb10\\", \\"thermoprotei_odb10\\", \\"thermotogae_odb10\\", \\"thiotrichales_odb10\\", \\"tissierellales_odb10\\", \\"tissierellia_odb10\\", \\"tremellomycetes_odb10\\", \\"vertebrata_odb10\\", \\"verrucomicrobia_odb10\\", \\"viridiplantae_odb10\\", \\"xanthomonadales_odb10\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"e715ce34-6147-48a4-b784-099d10e698ab","when":null,"workflow_outputs":[]},"1":{"annotation":"Genome sequence","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Genome sequence","name":"Genome sequence"}],"label":"Genome sequence","name":"Input dataset","outputs":[],"position":{"left":30,"top":480},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"6b8b7209-c3a4-4113-b570-32978b84a5e8","when":null,"workflow_outputs":[]},"2":{"annotation":"Genome assembly","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Genome assembly","name":"Genome assembly"}],"label":"Genome assembly","name":"Input dataset","outputs":[],"position":{"left":30,"top":610},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"0c3209a5-1a70-4ab5-b2d5-340748d77590","when":null,"workflow_outputs":[]},"3":{"annotation":"Protein sequences","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Protein sequences","name":"Protein sequences"}],"label":"Protein sequences","name":"Input dataset","outputs":[],"position":{"left":30,"top":740},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"772fbd62-3182-4fc1-bfa2-5bf1395ae0e4","when":null,"workflow_outputs":[]},"4":{"annotation":"Augustus training","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Augustus training","name":"Augustus training"}],"label":"Augustus training","name":"Input dataset","outputs":[],"position":{"left":30,"top":860},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"0e91370f-3709-42d5-b869-bfd9eb213631","when":null,"workflow_outputs":[]},"5":{"annotation":"SNAP training","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"SNAP training","name":"SNAP training"}],"label":"SNAP training","name":"Input dataset","outputs":[],"position":{"left":30,"top":1000},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"1d3dc619-b0df-4cdd-89de-7e16d6afbaa1","when":null,"workflow_outputs":[]},"6":{"annotation":"Fasta Statistics on the genome","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasta_stats/fasta-stats/2.0","errors":null,"id":6,"input_connections":{"fasta":{"id":1,"output_name":"output"}},"inputs":[],"label":" Fasta Statistics ","name":"Fasta Statistics","outputs":[{"name":"stats_output","type":"tabular"}],"position":{"left":580,"top":270},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasta_stats/fasta-stats/2.0","tool_shed_repository":{"changeset_revision":"0dbb995c7d35","name":"fasta_stats","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gaps_option\\": false, \\"genome_size\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0","type":"tool","uuid":"1b11b2a5-4085-4bae-8049-7d75f04620cc","when":null,"workflow_outputs":[{"label":"fasta stats genome","output_name":"stats_output","uuid":"4c11da0f-57a5-4751-84f9-c6d4e1325602"}]},"7":{"annotation":"Completeness assessment of the genome using the Busco tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0","errors":null,"id":7,"input_connections":{"input":{"id":1,"output_name":"output"},"lineage|lineage_dataset":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"lineage"}],"label":"BUSCO on the genome sequences","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":790,"top":40.02932102857932},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"2babe6d5c561","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"augustus\\", \\"__current_case__\\": 1}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": \\"all+2024-03-21-114020\\"}, \\"outputs\\": [\\"short_summary\\", \\"image\\", \\"gff\\", \\"missing\\"], \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.7.1+galaxy0","type":"tool","uuid":"b3def683-4967-454a-96c5-e95d609cb83c","when":null,"workflow_outputs":[{"label":"busco image genome","output_name":"summary_image","uuid":"c2f275c1-dc94-4914-a3b2-4683f5f742a3"},{"label":"busco sum genome","output_name":"busco_sum","uuid":"8d39ad7f-b6c7-4b29-b225-c0c9e842f528"},{"label":"busco table genome","output_name":"busco_table","uuid":"88d4ce85-b45d-40a2-85a0-ee4af20b70aa"},{"label":"busco gff genome","output_name":"busco_gff","uuid":"e44bfb06-31df-434e-bd12-b3ff7aef73ae"},{"label":"busco missing genome","output_name":"busco_missing","uuid":"facba50a-b7d0-426a-9fe5-62871940f5c9"}]},"8":{"annotation":"Annotation with Maker","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/maker/maker/2.31.11+galaxy2","errors":null,"id":8,"input_connections":{"abinitio_gene_prediction|aug_prediction|augustus_model":{"id":4,"output_name":"output"},"abinitio_gene_prediction|snaphmm":{"id":5,"output_name":"output"},"est_evidences|est":{"id":2,"output_name":"output"},"genome":{"id":1,"output_name":"output"},"protein_evidences|protein":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Maker","name":"abinitio_gene_prediction"},{"description":"runtime parameter for tool Maker","name":"advanced"},{"description":"runtime parameter for tool Maker","name":"est_evidences"},{"description":"runtime parameter for tool Maker","name":"est_evidences"},{"description":"runtime parameter for tool Maker","name":"est_evidences"},{"description":"runtime parameter for tool Maker","name":"est_evidences"},{"description":"runtime parameter for tool Maker","name":"gene_prediction"},{"description":"runtime parameter for tool Maker","name":"gene_prediction"},{"description":"runtime parameter for tool Maker","name":"gene_prediction"},{"description":"runtime parameter for tool Maker","name":"protein_evidences"},{"description":"runtime parameter for tool Maker","name":"protein_evidences"}],"label":"Maker","name":"Maker","outputs":[{"name":"output_gff","type":"gff3"},{"name":"output_evidences","type":"gff3"},{"name":"output_full","type":"gff3"}],"position":{"left":470,"top":570},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/maker/maker/2.31.11+galaxy2","tool_shed_repository":{"changeset_revision":"370c210d9541","name":"maker","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"abinitio_gene_prediction\\": {\\"snaphmm\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"aug_prediction\\": {\\"augustus_mode\\": \\"history\\", \\"__current_case__\\": 1, \\"augustus_model\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"unmask\\": false}, \\"advanced\\": {\\"fix_nucleotides\\": false, \\"other_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"alt_peptide\\": \\"C\\", \\"max_dna_len\\": \\"100000\\", \\"min_contig\\": \\"1\\", \\"pred_flank\\": \\"200\\", \\"pred_stats\\": false, \\"AED_threshold\\": \\"1.0\\", \\"min_protein\\": \\"0\\", \\"alt_splice\\": false, \\"always_complete\\": false, \\"map_forward\\": false, \\"keep_preds\\": \\"0.0\\", \\"split_hit\\": \\"10000\\", \\"correct_est_fusion\\": false, \\"single_exon\\": {\\"single_exon\\": \\"0\\", \\"__current_case__\\": 0}}, \\"est_evidences\\": {\\"est2genome\\": false, \\"est\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"altest\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"est_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"altest_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"gene_prediction\\": {\\"pred_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"model_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"trna\\": false, \\"snoscan_rrna\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"license_agreement\\": true, \\"organism_type\\": \\"eukaryotic\\", \\"protein_evidences\\": {\\"protein2genome\\": false, \\"protein\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"protein_gff\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"reannotation\\": {\\"reannotate\\": \\"no\\", \\"__current_case__\\": 0}, \\"repeat_masking\\": {\\"repeat_source\\": {\\"source_type\\": \\"dfam\\", \\"__current_case__\\": 0, \\"species_source\\": {\\"species_from_list\\": \\"yes\\", \\"__current_case__\\": 0, \\"species_list\\": \\"human\\"}, \\"softmask\\": true}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.11+galaxy2","type":"tool","uuid":"9f03ad16-7f18-434d-9e5a-deb504074908","when":null,"workflow_outputs":[{"label":"maker full","output_name":"output_full","uuid":"4c64666f-1d19-4e47-86f6-4cebbb9a701d"},{"label":"maker gff","output_name":"output_gff","uuid":"dd274ed8-5879-4dce-b37f-16b53ec642a9"},{"label":"maker evidences","output_name":"output_evidences","uuid":"168d9962-c7d8-4c97-a8fb-e76b63ff3cf3"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","errors":null,"id":9,"input_connections":{"input":{"id":8,"output_name":"output_gff"},"reference_genome|genome_fasta":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gffread","name":"chr_replace"},{"description":"runtime parameter for tool gffread","name":"reference_genome"}],"label":"GFFRead","name":"gffread","outputs":[{"name":"output_exons","type":"fasta"}],"position":{"left":1100,"top":440},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","tool_shed_repository":{"changeset_revision":"3e436657dcd0","name":"gffread","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chr_replace\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"decode_url\\": false, \\"expose\\": false, \\"filtering\\": null, \\"full_gff_attribute_preservation\\": false, \\"gffs\\": {\\"gff_fmt\\": \\"none\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maxintron\\": null, \\"merging\\": {\\"merge_sel\\": \\"none\\", \\"__current_case__\\": 0}, \\"reference_genome\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 2, \\"genome_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ref_filtering\\": null, \\"fa_outputs\\": [\\"-w exons.fa\\"]}, \\"region\\": {\\"region_filter\\": \\"none\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1.4+galaxy0","type":"tool","uuid":"71e53a4c-8fcd-4965-9959-1667c6e1fb91","when":null,"workflow_outputs":[{"label":"gffread exons","output_name":"output_exons","uuid":"0de0bce5-9d34-4251-8536-c03fa830d184"}]},"10":{"annotation":"Genome annotation statistics on the maker\'s annotation","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/jcvi_gff_stats/jcvi_gff_stats/0.8.4","errors":null,"id":10,"input_connections":{"gff":{"id":8,"output_name":"output_gff"},"ref_genome|genome":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Genome annotation statistics","name":"ref_genome"}],"label":"Genome annotation statistics","name":"Genome annotation statistics","outputs":[{"name":"summary","type":"txt"},{"name":"graphs","type":"pdf"}],"position":{"left":950,"top":870},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/jcvi_gff_stats/jcvi_gff_stats/0.8.4","tool_shed_repository":{"changeset_revision":"8cffbd184762","name":"jcvi_gff_stats","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"gff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ref_genome\\": {\\"genome_type_select\\": \\"history\\", \\"__current_case__\\": 1, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.8.4","type":"tool","uuid":"d159e179-469d-4236-8fa5-02ecc45f6cd3","when":null,"workflow_outputs":[{"label":"graphs genome","output_name":"graphs","uuid":"f68d4236-0e55-4e00-8f44-57b66e7e8b04"},{"label":"summary genome","output_name":"summary","uuid":"28c75ded-a74c-40c8-8735-80c569735f4b"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/maker_map_ids/maker_map_ids/2.31.11","errors":null,"id":11,"input_connections":{"maker_gff":{"id":8,"output_name":"output_gff"}},"inputs":[],"label":"Map annotation ids","name":"Map annotation ids","outputs":[{"name":"renamed","type":"gff"},{"name":"id_map","type":"tabular"}],"position":{"left":1320,"top":1060},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/maker_map_ids/maker_map_ids/2.31.11","tool_shed_repository":{"changeset_revision":"e906fa778440","name":"maker_map_ids","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"justify\\": \\"6\\", \\"maker_gff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"prefix\\": null, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.31.11","type":"tool","uuid":"66ffbb05-19e7-48ae-8d08-272d34595ddb","when":null,"workflow_outputs":[{"label":"renamed gff3","output_name":"renamed","uuid":"895adcb0-059e-48d7-adb2-283d52f193cf"},{"label":"id map","output_name":"id_map","uuid":"c415d422-c636-48b3-bd56-ea209880c5f9"}]},"12":{"annotation":"Completeness assessment of the genome using the Busco tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0","errors":null,"id":12,"input_connections":{"input":{"id":9,"output_name":"output_exons"},"lineage|lineage_dataset":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Busco","name":"input"},{"description":"runtime parameter for tool Busco","name":"lineage"}],"label":"BUSCO on the predicted protein sequences","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":1350,"top":260},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"2babe6d5c561","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"tran\\", \\"__current_case__\\": 1}, \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": \\"all+2024-03-21-114020\\"}, \\"outputs\\": [\\"short_summary\\", \\"image\\", \\"gff\\", \\"missing\\"], \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.7.1+galaxy0","type":"tool","uuid":"0cf163fb-6f82-468d-8ad0-442a688d0d05","when":null,"workflow_outputs":[{"label":"busco missing predicted proteins","output_name":"busco_missing","uuid":"54997a8e-0555-42d7-938c-fd8a3fd0caf0"},{"label":"busco table predicted proteins","output_name":"busco_table","uuid":"aa85c4a1-7d89-48f1-900a-77508af8d600"},{"label":"busco gff predicted proteins","output_name":"busco_gff","uuid":"7b588cd3-1d62-487d-979a-3da07e2441c7"},{"label":"busco sum predicted proteins","output_name":"busco_sum","uuid":"73725e32-7d72-42a6-b298-2e8c18ecc870"},{"label":"busco image predicted proteins","output_name":"summary_image","uuid":"cb6b286c-ba11-48e9-b4dd-96a0e6da011b"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1","errors":null,"id":13,"input_connections":{"reference_genome|genome":{"id":1,"output_name":"output"},"track_groups_0|data_tracks_0|data_format|annotation":{"id":11,"output_name":"renamed"}},"inputs":[{"description":"runtime parameter for tool JBrowse","name":"reference_genome"}],"label":"JBrowse","name":"JBrowse","outputs":[{"name":"output","type":"html"}],"position":{"left":1640,"top":860},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1","tool_shed_repository":{"changeset_revision":"a6e57ff585c0","name":"jbrowse","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"action\\": {\\"action_select\\": \\"create\\", \\"__current_case__\\": 0}, \\"gencode\\": \\"1\\", \\"jbgen\\": {\\"defaultLocation\\": \\"\\", \\"trackPadding\\": \\"20\\", \\"shareLink\\": true, \\"aboutDescription\\": \\"\\", \\"show_tracklist\\": true, \\"show_nav\\": true, \\"show_overview\\": true, \\"show_menu\\": true, \\"hideGenomeOptions\\": false}, \\"plugins\\": {\\"BlastView\\": true, \\"ComboTrackSelector\\": false, \\"GCContent\\": false}, \\"reference_genome\\": {\\"genome_type_select\\": \\"history\\", \\"__current_case__\\": 1, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"standalone\\": \\"minimal\\", \\"track_groups\\": [{\\"__index__\\": 0, \\"category\\": \\"Maker annotation\\", \\"data_tracks\\": [{\\"__index__\\": 0, \\"data_format\\": {\\"data_format_select\\": \\"gene_calls\\", \\"__current_case__\\": 2, \\"annotation\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"match_part\\": {\\"match_part_select\\": false, \\"__current_case__\\": 1}, \\"index\\": false, \\"track_config\\": {\\"track_class\\": \\"NeatHTMLFeatures/View/Track/NeatFeatures\\", \\"__current_case__\\": 3, \\"html_options\\": {\\"topLevelFeatures\\": null}}, \\"jbstyle\\": {\\"style_classname\\": \\"feature\\", \\"style_label\\": \\"product,name,id\\", \\"style_description\\": \\"note,description\\", \\"style_height\\": \\"10px\\", \\"max_height\\": \\"600\\"}, \\"jbcolor_scale\\": {\\"color_score\\": {\\"color_score_select\\": \\"none\\", \\"__current_case__\\": 0, \\"color\\": {\\"color_select\\": \\"automatic\\", \\"__current_case__\\": 0}}}, \\"jb_custom_config\\": {\\"option\\": []}, \\"jbmenu\\": {\\"track_menu\\": []}, \\"track_visibility\\": \\"default_off\\", \\"override_apollo_plugins\\": \\"False\\", \\"override_apollo_drag\\": \\"False\\"}}]}], \\"uglyTestingHack\\": \\"\\", \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.16.11+galaxy1","type":"tool","uuid":"ddb9f594-c5b4-4460-b52e-4ea8fb9777cb","when":null,"workflow_outputs":[{"label":"jbrowse","output_name":"output","uuid":"c25c5443-10b5-478f-a04b-26c98383090a"}]}},"tags":["maker","genomeannotation","busco","jbrowse"],"uuid":"87be9418-f744-42b6-8ac8-01dae7998e4f","version":3},"readme":"# Genome annotation workflow with Maker\\n\\nThis workflow allows for genome annotation using Maker and evaluates the quality of the annotation with BUSCO and genome annotation statistics. The annotation can then be improved, standardized, and visualized with additional tools.\\n\\n**Maker** is a genome model prediction software that uses ab initio predictors (SANP and Augustus) to improve its predictions. Maker is capable of annotating both prokaryotes and eukaryotes. It works by aligning as much evidence as possible along the genome sequence, then reconciling all these signals to determine likely genetic structures.\\n\\n## Workflow Steps\\n\\n- Annotation with Maker: Maker uses the genome sequence, protein evidence, ab-initio predictions, and ESTs to produce the annotation.\\n- Quality Evaluation:\\n    - Run Fasta Statistics to assess genome assembly quality.\\n    - Use BUSCO to evaluate annotation completeness.\\n- Annotation Statistics: Analyze the annotation using Genome Annotation Statistics, producing graphical and textual summaries.\\n- Sequence Extraction: Extract predicted protein sequences using GFFRead for downstream analysis.\\n- Improve Gene Names: Standardize gene names using Map annotation ids for better readability.\\n- Visualization: Load the genome sequence and annotation into JBrowse for interactive browsing.\\n\\n## Input data\\nThe following input files are required for the workflow:\\n- Genome sequence (FASTA format): The genome to be annotated. Used by Maker, Fasta Statistics, and BUSCO.\\n- Protein sequences (FASTA format): Evidence to assist annotation in Maker.\\n- EST evidences (FASTA format): Alignments used as evidence by Maker.\\n- Ab-initio gene predictions: Supplementary data for Maker to refine annotations.\\n\\n\\n## Output Data\\nThe workflow generates the following outputs:\\n- Annotation file (GFF3): Contains the final consensus gene models produced by Maker.\\n- Genome statistics: A tabular file summarizing contig sizes and base content, produced by Fasta Statistics.\\n- BUSCO results: Assess the completeness of the annotation and include:\\n    - A summary of results.\\n    - A table of all searched BUSCO genes with their status.\\n    - A table of missing BUSCO genes.\\n- Annotation statistics: Summary and graphical analyses of the annotation, produced by Genome Annotation Statistics.\\n- Protein sequences (FASTA): Predicted from the annotation using GFFRead.\\n- Renamed GFF annotation file: Contains standardized gene names, produced by Map annotation ids.\\n- Genome browser visualization (HTML): An interactive genome view produced by JBrowse.","changelog":"# Changelog\\n\\n## [0.1] - 2025-03-06\\n\\nInitial version of the genome annotation workflow with maker.","diagrams":"# Workflow diagrams\\n\\n## Genome annotation with Maker\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ BUSCO lineage\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Genome sequence\\"]@{ shape: doc }\\n2[\\"ℹ️ Genome assembly\\"]@{ shape: doc }\\n3[\\"ℹ️ Protein sequences\\"]@{ shape: doc }\\n4[\\"ℹ️ Augustus training\\"]@{ shape: doc }\\n5[\\"ℹ️ SNAP training\\"]@{ shape: doc }\\n6[\\" Fasta Statistics \\"]@{ shape: process }\\n1 --> 6\\n7[\\"BUSCO on the genome sequences\\"]@{ shape: process }\\n1 --> 7\\n0 --> 7\\n8[\\"Maker\\"]@{ shape: process }\\n4 --> 8\\n5 --> 8\\n2 --> 8\\n1 --> 8\\n3 --> 8\\n9[\\"GFFRead\\"]@{ shape: process }\\n8 --> 9\\n1 --> 9\\n10[\\"Genome annotation statistics\\"]@{ shape: process }\\n8 --> 10\\n1 --> 10\\n11[\\"Map annotation ids\\"]@{ shape: process }\\n8 --> 11\\n12[\\"BUSCO on the predicted protein sequences\\"]@{ shape: process }\\n9 --> 12\\n0 --> 12\\n13[\\"JBrowse\\"]@{ shape: process }\\n1 --> 13\\n11 --> 13\\n```\\n","updated":"2025-03-06T00:00:00","trsID":"#workflow/github.com/iwc-workflows/annotation-maker/main","dockstore_id":31169,"categories":[],"collections":[],"doi":"10.5281/zenodo.15078841"}],"path":"./workflows/genome_annotation/annotation-maker"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Galaxy-Workflow-lncRNAs_annotation_workflow.ga","testParameterFiles":["/Galaxy-Workflow-lncRNAs_annotation_workflow-tests.yml"],"authors":[{"name":"Romane Libouban","email":"romane.libouban@irisa.fr","orcid":"0009-0001-4920-9951"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow runs the FEELnc tool to annotate long non-coding RNAs. Before annotating these long non-coding RNAs, StringTie will be used to assemble the RNA-seq alignments into potential trancriptions. The gffread tool provides a genome annotation file in GTF format.","comments":[{"child_steps":[8],"color":"blue","data":{"title":"Annotation with mRNA and lncRNA"},"id":4,"position":[1134.3,654.4],"size":[268,262],"type":"frame"},{"child_steps":[7],"color":"red","data":{"title":"lncRNAs annotation with FEELnc"},"id":3,"position":[810,380],"size":[240,380],"type":"frame"},{"child_steps":[5],"color":"pink","data":{"title":"Conversion from GFF3 to GTF format"},"id":2,"position":[440,650],"size":[240,230],"type":"frame"},{"child_steps":[2,1],"color":"yellow","data":{"title":"Inputs"},"id":1,"position":[0,426.4],"size":[243,326.2],"type":"frame"},{"child_steps":[4,6,0],"color":"green","data":{"title":"Transcripts assembly with StringTie"},"id":0,"position":[130,0],"size":[580,375.2],"type":"frame"}],"creator":[{"class":"Person","email":"mailto:romane.libouban@irisa.fr","identifier":"https://orcid.org/0009-0001-4920-9951","name":"Romane Libouban"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"lncRNAs annotation workflow","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"strandednes","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"strandednes","name":"strandedness"}],"label":"strandedness","name":"Input parameter","outputs":[],"position":{"left":150,"top":230},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictions\\": [\\"stranded - forward\\", \\"stranded - reverse\\", \\"unstranded\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ee38bb19-c158-4a54-82b1-a2eeb195e5f0","when":null,"workflow_outputs":[]},"1":{"annotation":"Genome assembly","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Genome assembly","name":"Genome assembly"}],"label":"Genome assembly","name":"Input dataset","outputs":[],"position":{"left":20,"top":460},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"6c129c58-d982-445f-b41c-c2c7387c5e81","when":null,"workflow_outputs":[]},"2":{"annotation":"Genome annotation","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Genome annotation","name":"Genome annotation"}],"label":"Genome annotation","name":"Input dataset","outputs":[],"position":{"left":20,"top":570},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"gff3\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c221afda-d336-4416-bfa0-f41a19c50097","when":null,"workflow_outputs":[]},"3":{"annotation":"Alignments from RNA-seq","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Alignments from RNA-seq","name":"Alignments from RNA-seq"}],"label":"Alignments from RNA-seq","name":"Input dataset","outputs":[],"position":{"left":20,"top":680},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"bam\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"1eccf2fb-d243-4be7-a952-a9d6d1375b89","when":null,"workflow_outputs":[]},"4":{"annotation":"Map parameter value","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","errors":null,"id":4,"input_connections":{"input_param_type|input_param":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map parameter value","name":"input_param_type"}],"label":null,"name":"Map parameter value","outputs":[{"name":"output_param_text","type":"expression.json"}],"position":{"left":270,"top":40},"post_job_actions":{"HideDatasetActionoutput_param_text":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output_param_text"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/map_param_value/map_param_value/0.2.0","tool_shed_repository":{"changeset_revision":"5ac8a4bf7a8d","name":"map_param_value","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_param_type\\": {\\"type\\": \\"text\\", \\"__current_case__\\": 0, \\"input_param\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mappings\\": [{\\"__index__\\": 0, \\"from\\": \\"stranded - forward\\", \\"to\\": \\"--fr\\"}, {\\"__index__\\": 1, \\"from\\": \\"stranded - reverse\\", \\"to\\": \\"--rf\\"}, {\\"__index__\\": 2, \\"from\\": \\"unstranded\\", \\"to\\": \\"\\"}]}, \\"output_param_type\\": \\"text\\", \\"unmapped\\": {\\"on_unmapped\\": \\"fail\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.0","type":"tool","uuid":"41e690b0-597b-4a3a-9289-5fc22fc6644b","when":null,"workflow_outputs":[]},"5":{"annotation":"Protein sequences extracted from genomic annotation","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","errors":null,"id":5,"input_connections":{"input":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gffread","name":"chr_replace"},{"description":"runtime parameter for tool gffread","name":"input"}],"label":"gffread","name":"gffread","outputs":[{"name":"output_gtf","type":"gtf"}],"position":{"left":460,"top":710},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","tool_shed_repository":{"changeset_revision":"3e436657dcd0","name":"gffread","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chr_replace\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"decode_url\\": false, \\"expose\\": false, \\"filtering\\": null, \\"full_gff_attribute_preservation\\": false, \\"gffs\\": {\\"gff_fmt\\": \\"gtf\\", \\"__current_case__\\": 2, \\"tname\\": \\"\\"}, \\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"maxintron\\": null, \\"merging\\": {\\"merge_sel\\": \\"none\\", \\"__current_case__\\": 0}, \\"reference_genome\\": {\\"source\\": \\"none\\", \\"__current_case__\\": 0}, \\"region\\": {\\"region_filter\\": \\"none\\", \\"__current_case__\\": 0}, \\"__page__\\": 0, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1.4+galaxy0","type":"tool","uuid":"1a2ed039-986a-47c0-bc76-7bb1509fad87","when":null,"workflow_outputs":[{"label":"annotation","output_name":"output_gtf","uuid":"51d0370b-a8d5-4eee-a780-58fea5d9e1a1"}]},"6":{"annotation":"StringTie is a fast, highly efficient assembler of RNA-Seq alignments into potential transcripts. ","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","errors":null,"id":6,"input_connections":{"guide|guide_source|ref_hist":{"id":2,"output_name":"output"},"input_options|input_bam":{"id":3,"output_name":"output"},"rna_strandness":{"id":4,"output_name":"output_param_text"}},"inputs":[{"description":"runtime parameter for tool StringTie","name":"adv"},{"description":"runtime parameter for tool StringTie","name":"input_options"}],"label":"Assembly step with StringTie","name":"StringTie","outputs":[{"name":"output_gtf","type":"gtf"}],"position":{"left":500,"top":60},"post_job_actions":{"HideDatasetActionoutput gtf":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output gtf"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/stringtie/stringtie/2.2.3+galaxy0","tool_shed_repository":{"changeset_revision":"cbf488da3b2c","name":"stringtie","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"abundance_estimation\\": false, \\"omit_sequences\\": \\"\\", \\"name_prefix\\": null, \\"fraction\\": \\"0.01\\", \\"min_tlen\\": \\"200\\", \\"min_anchor_len\\": \\"10\\", \\"min_anchor_cov\\": \\"1\\", \\"min_bundle_cov\\": \\"1\\", \\"bdist\\": \\"50\\", \\"bundle_fraction\\": \\"1.0\\", \\"disable_trimming\\": false, \\"multi_mapping\\": false, \\"point_features\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"guide\\": {\\"use_guide\\": \\"yes\\", \\"__current_case__\\": 1, \\"guide_source\\": {\\"guide_gff_select\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_hist\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input_estimation\\": false, \\"special_outputs\\": {\\"special_outputs_select\\": \\"no\\", \\"__current_case__\\": 2}, \\"coverage_file\\": false}, \\"input_options\\": {\\"input_mode\\": \\"short_reads\\", \\"__current_case__\\": 0, \\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rna_strandness\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.3+galaxy0","type":"tool","uuid":"632f7928-d838-4bbf-b0f9-6a94ea9eb427","when":null,"workflow_outputs":[{"label":"stringtie gtf","output_name":"output_gtf","uuid":"6edddada-3f7f-4a62-bbb7-8f412234342f"}]},"7":{"annotation":"The FEELnc pipeline identifies and classifies long non-coding RNAs (lncRNAs) through three main steps: filter, codpot and classifier.","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/feelnc/feelnc/0.2.1+galaxy0","errors":null,"id":7,"input_connections":{"candidate":{"id":6,"output_name":"output_gtf"},"genome":{"id":1,"output_name":"output"},"reference":{"id":5,"output_name":"output_gtf"}},"inputs":[],"label":"lncRNA annotation step with FEELnc","name":"FEELnc","outputs":[{"name":"lcnRNA annotation","type":"gtf"},{"name":"lcnRNA annotation","type":"gtf"},{"name":"classifier","type":"txt"}],"position":{"left":830,"top":440},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/feelnc/feelnc/0.2.1+galaxy0","tool_shed_repository":{"changeset_revision":"55daa4712413","name":"feelnc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"candidate\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.1+galaxy0","type":"tool","uuid":"e7d02c0b-09a9-46e0-836a-048a3a5fdc33","when":null,"workflow_outputs":[{"label":"candidate mRNA","output_name":"candidate_mRNA","uuid":"1bdec63f-0ed2-48fb-bf46-0550961565fe"},{"label":"classification","output_name":"classifier","uuid":"a1e803b3-e991-4c7f-969c-7e73b0d19b62"},{"label":"candidate lncRNA","output_name":"candidate_lncRNA","uuid":"4a8e3b1c-7547-4c71-bf51-4dbf64d6f412"}]},"8":{"annotation":" The lncRNA annotation is merged with the reference annotation to create a unified genome annotation containing both mRNAs and lncRNAs.","content_id":"cat1","errors":null,"id":8,"input_connections":{"input1":{"id":5,"output_name":"output_gtf"},"queries_0|input2":{"id":7,"output_name":"candidate_lncRNA"}},"inputs":[],"label":"Concatenation step","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1160,"top":720},"post_job_actions":{},"tool_id":"cat1","tool_state":"{\\"__input_ext\\": \\"gtf\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [{\\"__index__\\": 0, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"27355e91-6006-443f-af12-33b801ec371b","when":null,"workflow_outputs":[{"label":"annotation final","output_name":"out_file1","uuid":"fd8e526f-df5e-4130-81ec-d7700968d58e"}]}},"tags":[],"uuid":"f6d6b63f-6fcc-42bb-bafb-12d9dd98983e","version":1},"readme":"# lncRNAs annotation workflow\\n\\nThis workflow uses the FEELnc tool to annotate long non-coding RNAs. Before annotating these long non-coding RNAs, StringTie will be used to assemble the RNA-seq alignments into potential trancriptions. The gffread tool provides a genome annotation file in GTF format.\\n\\nFor future analyses, it would be interesting to use an updated annotation containing messenger RNA and long non-coding RNA. The concatenante tool merges the reference annotation with the long non-coding RNA annotation obtained with FEELnc.\\n\\nThis workflow is taken from the tutorial “Long non-coding RNAs (lncRNAs) annotation with FEELnc” on the GTN.\\n\\n## Workflows steps\\n- Transcript Assembly with StringTie: RNA-seq alignments are assembled into potential transcripts to provide a comprehensive view of expressed regions.\\n- Genome Annotation Conversion with GFFRead: Genome annotations are converted into a standardized format (GTF) to ensure compatibility with downstream tools.\\n- lncRNA Annotation with FEELnc: The FEELnc pipeline identifies and classifies long non-coding RNAs (lncRNAs) through three main steps:\\n    - Filter: Removes unwanted transcripts and those overlapping reference exons.\\n    - Codpot: Evaluates coding potential to differentiate lncRNAs from coding RNAs.\\n    - Classifier: Assigns lncRNAs to categories based on their genomic location and transcriptional direction.\\n- Annotation Merging with Concatenate: The lncRNA annotation is merged with the reference annotation to create a unified genome annotation containing both mRNAs and lncRNAs.\\n\\n## Input data\\nThe following input files are required for the workflow:\\n- RNA-seq alignments (BAM format): Required by StringTie for transcript assembly.\\n- Genome annotation (GFF3 format): Used by StringTie and GFFRead for processing.\\n- Genome sequence (FASTA format): Required by FEELnc for lncRNA identification.\\n- Reference annotation (GTF format): Provided by GFFRead for FEELnc analysis.\\n\\n## Output data\\nThe workflow produces the following outputs:\\n- Transcript annotation (GTF format): Generated by StringTie, containing assembled transcripts from RNA-seq data.\\n- Converted genome annotation (GTF format): Produced by GFFRead, used as input for FEELnc.\\n- lncRNA annotation (GTF format): Generated by FEELnc, containing identified lncRNAs.\\n- mRNA annotation (GTF format): Produced by FEELnc for downstream use.\\n- lncRNA classification table: Produced by FEELnc, detailing genomic relationships of lncRNAs.\\n- Comprehensive genome annotation (GTF format): Generated by Concatenate, combining mRNA and lncRNA annotations.\\n","changelog":"# Changelog\\n\\n## [0.1] - 2025-03-05\\n\\nInitial version of the lncRNAs annotation workflow.","diagrams":"# Workflow diagrams\\n\\n## lncRNAs annotation workflow\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ strandedness\\"]@{ shape: lean-l }\\n1[\\"ℹ️ Genome assembly\\"]@{ shape: doc }\\n2[\\"ℹ️ Genome annotation\\"]@{ shape: doc }\\n3[\\"ℹ️ Alignments from RNA-seq\\"]@{ shape: doc }\\n4[\\"Map parameter value\\"]@{ shape: process }\\n0 --> 4\\n5[\\"gffread\\"]@{ shape: process }\\n2 --> 5\\n6[\\"Assembly step with StringTie\\"]@{ shape: process }\\n2 --> 6\\n3 --> 6\\n4 --> 6\\n7[\\"lncRNA annotation step with FEELnc\\"]@{ shape: process }\\n6 --> 7\\n1 --> 7\\n5 --> 7\\n8[\\"Concatenation step\\"]@{ shape: process }\\n5 --> 8\\n7 --> 8\\n```\\n","updated":"2025-03-05T00:00:00","trsID":"#workflow/github.com/iwc-workflows/lncRNAs-annotation/main","dockstore_id":30909,"categories":[],"collections":["Genome Annotation"],"doi":"10.5281/zenodo.15078797","tests":[{"doc":"Test outline for Galaxy-Workflow-lncRNAs_annotation_workflow.ga","job":{"Genome assembly":{"class":"File","location":"https://zenodo.org/records/11367439/files/genome_assembly.fasta","filetype":"fasta"},"Genome annotation":{"class":"File","location":"https://zenodo.org/records/11367439/files/genome_annotation.gff3","filetype":"gff3"},"Alignments from RNA-seq":{"class":"File","location":"https://zenodo.org/records/14726111/files/SRR8534859_RNASeq_mapped_small_version.bam","filetype":"bam"},"strandedness":"unstranded"},"outputs":{"annotation":{"asserts":[{"has_n_lines":{"n":137448}},{"has_text":{"text":"transcript"}}]},"stringtie gtf":{"asserts":[{"has_n_lines":{"n":77365}},{"has_text":{"text":"transcript"}}]},"candidate lncRNA":{"asserts":[{"has_n_lines":{"n":233}},{"has_text":{"text":"scaffold_21"}}]},"classification":{"asserts":[{"has_n_lines":{"n":696}},{"has_text":{"text":"antisense"}}]},"annotation final":{"asserts":[{"has_n_lines":{"n":137681}},{"has_text":{"text":"StringTie"}}]}}}]}],"path":"./workflows/genome_annotation/lncRNAs-annotation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Galaxy-Workflow-annotation_helixer.ga","testParameterFiles":["/Galaxy-Workflow-annotation_helixer-tests.yml"],"authors":[{"name":"Romane Libouban","email":"romane.libouban@irisa.fr","orcid":"0009-0001-4920-9951"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. ","comments":[{"child_steps":[4,2],"color":"lime","data":{"title":"Evaluation - Genome annotation"},"id":2,"position":[468.3,902.5],"size":[496.5,356.1],"type":"frame"},{"child_steps":[3],"color":"orange","data":{"title":"Protein prediction with Helixer annotation"},"id":1,"position":[628.9,255.2],"size":[258,275],"type":"frame"},{"child_steps":[1],"color":"blue","data":{"title":"Annotation step"},"id":0,"position":[238.5,458.79999999999995],"size":[240,183],"type":"frame"},{"child_steps":[5],"color":"pink","data":{"title":"Visualization"},"id":4,"position":[1045.3,680],"size":[240,244.5],"type":"frame"},{"child_steps":[6,7],"color":"turquoise","data":{"title":"Evaluation - Predicted protein from annotation"},"id":3,"position":[1104.4,0],"size":[312,563],"type":"frame"}],"creator":[{"class":"Person","email":"mailto:romane.libouban@irisa.fr","identifier":"https://orcid.org/0009-0001-4920-9951","name":"Romane Libouban"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"Genome annotation with Helixer","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Input dataset containing genomic sequences in FASTA format","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Input dataset containing genomic sequences in FASTA format","name":"Genome sequence"}],"label":"Genome sequence","name":"Input dataset","outputs":[],"position":{"left":0,"top":812.5362146249206},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"e267e1df-03ae-4b70-98ce-65ea177a172e","when":null,"workflow_outputs":[]},"1":{"annotation":"Helixer tool for genomic annotation","content_id":"toolshed.g2.bx.psu.edu/repos/genouest/helixer/helixer/0.3.3+galaxy1","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Helixer","name":"input_model"}],"label":"Helixer","name":"Helixer","outputs":[{"name":"output","type":"gff3"}],"position":{"left":258.5333251953125,"top":498.8339807128906},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/genouest/helixer/helixer/0.3.3+galaxy1","tool_shed_repository":{"changeset_revision":"c2fc4ac35199","name":"helixer","owner":"genouest","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_model\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"lineages\\": \\"land_plant\\", \\"option_overlap\\": {\\"use_overlap\\": \\"true\\", \\"__current_case__\\": 0, \\"overlap_offset\\": null, \\"overlap_core_length\\": null}, \\"post_processing\\": {\\"window_size\\": \\"100\\", \\"edge_threshold\\": \\"0.1\\", \\"peak_threshold\\": \\"0.8\\", \\"min_coding_length\\": \\"100\\"}, \\"size\\": \\"8\\", \\"species\\": null, \\"subsequence_length\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.3+galaxy1","type":"tool","uuid":"f60cf54d-31f2-4395-bb55-4916828cd211","when":null,"workflow_outputs":[{"label":"helixer output","output_name":"output","uuid":"fe43bcd6-5f99-4fd3-b184-2d6bfb340030"}]},"2":{"annotation":"Completeness assessment of the genome using the Busco tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"Busco on genome","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":744.7633406324078,"top":942.4706486763349},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"geno\\", \\"__current_case__\\": 0, \\"use_augustus\\": {\\"use_augustus_selector\\": \\"augustus\\", \\"__current_case__\\": 2, \\"aug_prediction\\": {\\"augustus_mode\\": \\"no\\", \\"__current_case__\\": 0}, \\"long\\": false}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": \\"mucorales_odb10\\"}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": \\"all+2024-03-21-114020\\"}, \\"outputs\\": [\\"short_summary\\", \\"image\\", \\"gff\\", \\"missing\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"c0e4cca7-0bc3-4ef2-81b2-c990b1b77d87","when":null,"workflow_outputs":[{"label":"busco gff genome","output_name":"busco_gff","uuid":"961890cc-7a33-422a-ab09-b787e3592dd1"},{"label":"busco sum genome","output_name":"busco_sum","uuid":"bf09f09a-b403-4517-9a1a-acece8f36735"},{"label":"summary image genome","output_name":"summary_image","uuid":"3232c386-3c31-4989-ac76-02722ea2d79b"},{"label":"busco table genome","output_name":"busco_table","uuid":"5cbbd77a-f521-4ee6-b990-a494b7671534"},{"label":"busco missing genome","output_name":"busco_missing","uuid":"d039ef78-640f-4f7d-b449-69fac1a25130"}]},"3":{"annotation":"Converts GFF files to other formats, such as FASTA","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","errors":null,"id":3,"input_connections":{"input":{"id":1,"output_name":"output"},"reference_genome|genome_fasta":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool gffread","name":"chr_replace"},{"description":"runtime parameter for tool gffread","name":"reference_genome"}],"label":"Gffread","name":"gffread","outputs":[{"name":"output_pep","type":"fasta"}],"position":{"left":658.9081573207637,"top":316.7812237670679},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/gffread/gffread/2.2.1.4+galaxy0","tool_shed_repository":{"changeset_revision":"3e436657dcd0","name":"gffread","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"chr_replace\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"decode_url\\": false, \\"expose\\": false, \\"filtering\\": null, \\"full_gff_attribute_preservation\\": false, \\"gffs\\": {\\"gff_fmt\\": \\"none\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maxintron\\": null, \\"merging\\": {\\"merge_sel\\": \\"none\\", \\"__current_case__\\": 0}, \\"reference_genome\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 2, \\"genome_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ref_filtering\\": null, \\"fa_outputs\\": [\\"-y pep.fa\\"]}, \\"region\\": {\\"region_filter\\": \\"none\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.1.4+galaxy0","type":"tool","uuid":"00d60c74-1ed5-4529-aa82-8745b50205b7","when":null,"workflow_outputs":[{"label":"gffread peptides","output_name":"output_pep","uuid":"aa178118-cd37-495b-9e81-e2e53ebf27fd"}]},"4":{"annotation":"Generates statistics and graphs for genome annotation","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/jcvi_gff_stats/jcvi_gff_stats/0.8.4","errors":null,"id":4,"input_connections":{"gff":{"id":1,"output_name":"output"},"ref_genome|genome":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Genome annotation statistics","name":"ref_genome"}],"label":"Genome annotation statistics","name":"Genome annotation statistics","outputs":[{"name":"summary","type":"txt"},{"name":"graphs","type":"pdf"}],"position":{"left":488.25061259116643,"top":991.5198240353345},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/jcvi_gff_stats/jcvi_gff_stats/0.8.4","tool_shed_repository":{"changeset_revision":"8cffbd184762","name":"jcvi_gff_stats","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"gff\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ref_genome\\": {\\"genome_type_select\\": \\"history\\", \\"__current_case__\\": 1, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.8.4","type":"tool","uuid":"f47f89eb-23f4-4a16-b0a8-49d8e62c9f3d","when":null,"workflow_outputs":[{"label":"summary","output_name":"summary","uuid":"fb8ed4c9-4b55-4547-880d-1916a91f8a6e"},{"label":"graphs","output_name":"graphs","uuid":"4638cc23-fdb6-4e82-9cdf-c9fe38e76bd7"}]},"5":{"annotation":"JBrowse","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1","errors":null,"id":5,"input_connections":{"reference_genome|genome":{"id":0,"output_name":"output"},"track_groups_0|data_tracks_0|data_format|annotation":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool JBrowse","name":"reference_genome"}],"label":"JBrowse","name":"JBrowse","outputs":[{"name":"output","type":"html"}],"position":{"left":1065.313344724818,"top":719.9967480789329},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1","tool_shed_repository":{"changeset_revision":"a6e57ff585c0","name":"jbrowse","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"action\\": {\\"action_select\\": \\"create\\", \\"__current_case__\\": 0}, \\"gencode\\": \\"1\\", \\"jbgen\\": {\\"defaultLocation\\": \\"\\", \\"trackPadding\\": \\"20\\", \\"shareLink\\": true, \\"aboutDescription\\": \\"\\", \\"show_tracklist\\": true, \\"show_nav\\": true, \\"show_overview\\": true, \\"show_menu\\": true, \\"hideGenomeOptions\\": false}, \\"plugins\\": {\\"BlastView\\": true, \\"ComboTrackSelector\\": false, \\"GCContent\\": false}, \\"reference_genome\\": {\\"genome_type_select\\": \\"history\\", \\"__current_case__\\": 1, \\"genome\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"standalone\\": \\"minimal\\", \\"track_groups\\": [{\\"__index__\\": 0, \\"category\\": \\"Annotation\\", \\"data_tracks\\": [{\\"__index__\\": 0, \\"data_format\\": {\\"data_format_select\\": \\"gene_calls\\", \\"__current_case__\\": 2, \\"annotation\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"match_part\\": {\\"match_part_select\\": false, \\"__current_case__\\": 1}, \\"index\\": false, \\"track_config\\": {\\"track_class\\": \\"NeatHTMLFeatures/View/Track/NeatFeatures\\", \\"__current_case__\\": 3, \\"html_options\\": {\\"topLevelFeatures\\": null}}, \\"jbstyle\\": {\\"style_classname\\": \\"feature\\", \\"style_label\\": \\"product,name,id\\", \\"style_description\\": \\"note,description\\", \\"style_height\\": \\"10px\\", \\"max_height\\": \\"600\\"}, \\"jbcolor_scale\\": {\\"color_score\\": {\\"color_score_select\\": \\"none\\", \\"__current_case__\\": 0, \\"color\\": {\\"color_select\\": \\"automatic\\", \\"__current_case__\\": 0}}}, \\"jb_custom_config\\": {\\"option\\": []}, \\"jbmenu\\": {\\"track_menu\\": []}, \\"track_visibility\\": \\"default_off\\", \\"override_apollo_plugins\\": \\"False\\", \\"override_apollo_drag\\": \\"False\\"}}]}], \\"uglyTestingHack\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.16.11+galaxy1","type":"tool","uuid":"04807fae-95f6-49c1-893e-76932a79cdf9","when":null,"workflow_outputs":[{"label":"output","output_name":"output","uuid":"19976896-9df1-45e4-9c96-89e24ae6e596"}]},"6":{"annotation":"Completeness assessment of the genome using the Busco tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","errors":null,"id":6,"input_connections":{"input":{"id":3,"output_name":"output_pep"}},"inputs":[],"label":"Busco on protein","name":"Busco","outputs":[{"name":"busco_sum","type":"txt"},{"name":"busco_table","type":"tabular"},{"name":"busco_missing","type":"tabular"},{"name":"summary_image","type":"png"},{"name":"busco_gff","type":"gff3"}],"position":{"left":1166.6977253236494,"top":58.61198039869754},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0","tool_shed_repository":{"changeset_revision":"e5c372c91e46","name":"busco","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"evalue\\": \\"0.001\\", \\"limit\\": \\"3\\", \\"contig_break\\": \\"10\\"}, \\"busco_mode\\": {\\"mode\\": \\"prot\\", \\"__current_case__\\": 2}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lineage\\": {\\"lineage_mode\\": \\"select_lineage\\", \\"__current_case__\\": 1, \\"lineage_dataset\\": \\"mucorales_odb10\\"}, \\"lineage_conditional\\": {\\"selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"cached_db\\": \\"all+2024-03-21-114020\\"}, \\"outputs\\": [\\"short_summary\\", \\"image\\", \\"gff\\", \\"missing\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.8.0+galaxy0","type":"tool","uuid":"51dcc6a4-ff87-4a98-98fa-de00ce54325f","when":null,"workflow_outputs":[{"label":"busco table peptides","output_name":"busco_table","uuid":"1a113d6c-a167-432b-8200-dfb3aedc4ba1"},{"label":"busco missing peptides","output_name":"busco_missing","uuid":"dc2d4533-d9c2-4cb0-a144-184e90fd4e01"},{"label":"busco gff peptides","output_name":"busco_gff","uuid":"1db166fb-10c2-4823-a80c-9f22c7c15576"},{"label":"summary image peptides","output_name":"summary_image","uuid":"13c6bee4-824c-4533-bc78-c99ddf0b190d"},{"label":"busco sum peptides","output_name":"busco_sum","uuid":"f44047d9-e713-41d9-a9f9-5543f0371d9d"}]},"7":{"annotation":"OMArk","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/omark/omark/0.3.0+galaxy2","errors":null,"id":7,"input_connections":{"input":{"id":3,"output_name":"output_pep"}},"inputs":[{"description":"runtime parameter for tool OMArk","name":"input_iso"}],"label":"OMArk","name":"OMArk","outputs":[{"name":"omark_detail_sum","type":"txt"},{"name":"omark_sum","type":"sum"}],"position":{"left":1339.8464912941502,"top":460.49653123113796},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/omark/omark/0.3.0+galaxy2","tool_shed_repository":{"changeset_revision":"6f570ba54b41","name":"omark","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"database\\": \\"LUCA-v2.0.0.h5\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_iso\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"omark_mode\\": false, \\"outputs\\": \\"detail_sum\\", \\"r\\": null, \\"t\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.0+galaxy2","type":"tool","uuid":"75e1dde7-5d60-4092-af57-cd7b065145e2","when":null,"workflow_outputs":[{"label":"omark sum","output_name":"omark_sum","uuid":"d1eb0367-c708-4794-a256-81dac775776a"},{"label":"omark detail sum","output_name":"omark_detail_sum","uuid":"90f0b094-b667-4e05-a1e4-e6388770a107"}]}},"tags":[],"uuid":"24ab5ff5-6b5b-4caf-8cf0-c2f2ccc139e8","version":1},"readme":"# Genome annotation workflow with Helixer\\n\\nThis workflow allows you to annotate a genome with Helixer and evaluate the quality of the annotation using BUSCO and Genome Annotation statistics. GFFRead is also used to predict protein sequences derived from this annotation, and BUSCO and OMArk are used to assess proteome quality. \\n\\n\\nHelixer is an annotation software with a new and different approach: it performs evidence-free predictions (no need for RNASeq data or sequence aligments), using Graphics Processing Unit (GPU), with a much faster execution time. The annotation is based on the development and use of a cross-species deep learning model. The software is used to configure and train models for ab initio prediction of gene structure. In other words, it identifies the base pairs in a genome that belong to the UTR/CDS/Intron genes.\\n\\nTo assess the quality of the proteome, we will use the GFFRead tool to extract the predicted protein sequences from the annotation (i.e. the Helixer annotation).\\n\\nTo assess the quality of the annotation, we will use different tools:\\n- Genome Annotation Statistics: is a program designed to analyze and provide statistics on genomic annotations. This software performs its analyses from a GFF3 file.\\n- BUSCO (Benchmarking Universal Single-Copy Orthologs):  is a tool allowing to evaluate the quality of a genome assembly or of a genome annotation. By comparing genomes from various more or less related species, the authors determined sets of ortholog genes that are present in single copy in (almost) all the species of a clade (Bacteria, Fungi, Plants, Insects, Mammals, …). Most of these genes are essential for the organism to live, and are expected to be found in any newly sequenced and annotated genome from the corresponding clade. Using this data, BUSCO is able to evaluate the proportion of these essential genes (also named BUSCOs) found in a set of (predicted) transcript or protein sequences. This is a good evaluation of the “completeness” of the annotation.\\n- OMArk: is proteome quality assessment software. It provides measures of proteome completeness, characterises the consistency of all protein-coding genes with their homologues and identifies the presence of contamination by other species. OMArk is based on the OMA orthology database, from which it exploits orthology relationships, and on the OMAmer software for rapid placement of all proteins in gene families.\\n\\nThe final step is to view the generated annotation using a genome browser such as JBrowse. This browser allows you to navigate along the chromosomes of the genome and view the structure of each predicted gene.\\n\\n## Input dataset for Helixer\\nHelixer requires the genome sequence to be annotated, in fasta format.\\n\\n## Output dataset for Helixer\\nHelixer produces a single output dataset: a GFF3 file. \\n\\n\\n## Input dataset for Genome Annotation Statistics\\nThis software requires a GFF3 file. In this workflow, the output generated is Helixer.\\n\\n## Output dataset for Genome Annotation Statistics\\nTwo output files are generated:\\n- a file containing graphs in pdf format\\n- a summary in txt format\\n\\n## Input dataset for GFFRead\\nIn this workflow, GFFRead requires two inputs:\\n- an annotation file in GFF3 format (the Helixer format)\\n- the genome sequence in fasta format\\n\\n## Output dataset for GFFRead\\nIn this workflow, a unique output will be generated. This file, in fasta format, contains the protein sequences predicted from the annotation.\\n\\n## Input dataset for BUSCO\\nBUSCO requires a fasta file.\\nBUSCO will be used twice for this workflow. Firstly on the predicted protein sequences and secondly on the genome sequence. \\n\\n## Output dataset for BUSCO\\nWith BUSCO, we can obtain different output files:\\n- short summary : statistical summary of the quality of genomic assembly or annotation, including total number of genes evaluated, percentage of complete genes, percentage of partial genes, etc.\\n- full table : list of universal orthologs found in the assembled or annotated genome, with information on their completeness, location in the genome, quality score, etc.\\n- missing BUSCOs : list of orthologs not found in the genome, which may indicate gaps in assembly or annotation.\\n- summary image : graphics and visualizations to visually represent the results of the evaluation, such as bar charts showing the proportion of complete, partial and missing genes.\\n- GFF : contain information on gene locations, exons, introns, etc.\\n\\n## Input dataset for OMArk\\nOMAk requires the fasta file produced by GFFRead, containing the predicted protein sequences. \\n\\n## Output dataset for OMArk\\nIn this tutorial, a single output file will be generated: a file detailing the assessment of completeness, consistency and species composition. \\n\\n## Input dataset for JBrowse\\nJBrowse requires two inputs:\\n- the genome sequence in fasta format\\n- the annotation file in gff3 format, generated by Helixer\\n\\n## Output dataset for JBrowse\\nAn html file is generated for browsing the genome.","changelog":"# Changelog\\n\\n## [0.2] - 2025-01-27\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.7.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.8.0+galaxy0`\\n\\n## [0.1]\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Genome annotation with Helixer\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Genome sequence\\"]@{ shape: doc }\\n1[\\"Helixer\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Busco on genome\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Gffread\\"]@{ shape: process }\\n1 --> 3\\n0 --> 3\\n4[\\"Genome annotation statistics\\"]@{ shape: process }\\n1 --> 4\\n0 --> 4\\n5[\\"JBrowse\\"]@{ shape: process }\\n0 --> 5\\n1 --> 5\\n6[\\"Busco on protein\\"]@{ shape: process }\\n3 --> 6\\n7[\\"OMArk\\"]@{ shape: process }\\n3 --> 7\\n```\\n","updated":"2025-01-27T00:00:00","trsID":"#workflow/github.com/iwc-workflows/annotation-helixer/main","dockstore_id":30334,"categories":[],"collections":["Genome Annotation"],"doi":"10.5281/zenodo.15078836","tests":[{"doc":"Test outline for Helixer Workflow","job":{"Genome sequence":{"class":"File","location":"https://zenodo.org/records/13890774/files/genome_masked.fa?download=1","filetype":"fasta"}},"outputs":{"helixer output":{"location":"https://zenodo.org/records/13890774/files/Helixer.gff3?download=1","compare":"sim_size","delta":300000},"busco sum genome":{"location":"https://zenodo.org/records/13890774/files/Busco_short_summary_genome.txt?download=1","compare":"sim_size","delta":30000},"busco gff genome":{"location":"https://zenodo.org/records/13890774/files/Busco_GFF_genome.gff3?download=1","compare":"sim_size","delta":30000},"summary image genome":{"location":"https://zenodo.org/records/13890774/files/Busco_summary_image_genome.png?download=1","compare":"sim_size","delta":30000},"busco missing genome":{"location":"https://zenodo.org/records/13890774/files/Busco_missing_buscos_genome.tabular?download=1","compare":"sim_size","delta":30000},"busco table genome":{"location":"https://zenodo.org/records/13890774/files/Busco_full_table_genome.tabular?download=1","compare":"sim_size","delta":500000},"gffread peptides":{"location":"https://zenodo.org/records/13890774/files/gffread_pep.fasta?download=1","compare":"sim_size","delta":30000},"summary":{"location":"https://zenodo.org/records/13890774/files/genome_annotation_statistics_summary.txt?download=1","compare":"sim_size","delta":30000},"graphs":{"location":"https://zenodo.org/records/13890774/files/genome_annotation_statistics_graphs.pdf?download=1","compare":"sim_size","delta":30000},"summary image peptides":{"location":"https://zenodo.org/records/13902305/files/Busco_pep_summary_image.png?download=1","compare":"sim_size","delta":30000},"busco table peptides":{"location":"https://zenodo.org/records/13890774/files/Busco_full_table_pep.tabular?download=1","compare":"sim_size","delta":500000},"busco sum peptides":{"location":"https://zenodo.org/records/13890774/files/Busco_short_summary_pep.txt?download=1","compare":"sim_size","delta":30000},"busco gff peptides":{"location":"https://zenodo.org/records/13890774/files/Busco_GFF_pep.gff3?download=1","compare":"sim_size","delta":30000},"busco missing peptides":{"location":"https://zenodo.org/records/13890774/files/Busco_missing_buscos_pep.tabular?download=1","compare":"sim_size","delta":30000},"omark detail sum":{"location":"https://zenodo.org/records/14260288/files/OMArk_Detailed_summary_LUCA.txt?download=1","compare":"sim_size","delta":30000}}}]}],"path":"./workflows/genome_annotation/annotation-helixer"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Functional_annotation_of_protein_sequences.ga","testParameterFiles":["/Functional_annotation_of_protein_sequences-tests.yml"],"authors":[{"name":"Romane Libouban","email":"romane.libouban@irisa.fr","orcid":"0009-0001-4920-9951"},{"name":"Anthony Bretaudeau","email":"anthony.bretaudeau@irisa.fr","orcid":"0000-0003-0914-2470"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow uses eggNOG mapper and InterProScan for functional annotation of protein sequences.","comments":[{"child_steps":[1,2],"color":"green","data":{"title":"Functional annotation"},"id":0,"position":[300,0],"size":[240,496],"type":"frame"}],"creator":[{"class":"Person","email":"mailto:romane.libouban@irisa.fr","identifier":"https://orcid.org/0009-0001-4920-9951","name":"Romane Libouban"},{"class":"Person","email":"mailto:anthony.bretaudeau@irisa.fr","identifier":"https://orcid.org/0000-0003-0914-2470","name":"Anthony Bretaudeau"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Functional annotation of protein sequences","steps":{"0":{"annotation":"This workflow uses eggNOG mapper and Interproscan for functional annotation of protein sequences.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"This workflow uses eggNOG mapper and Interproscan for functional annotation of protein sequences.","name":"input"}],"label":"input","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"fb78bb38-ab6a-4676-98c5-5d3be83e7474","when":null,"workflow_outputs":[]},"1":{"annotation":"InterProScan is a tool that analyses each protein sequence from our annotation to determine if they contain one or several of the signatures from InterPro.","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/interproscan/interproscan/5.59-91.0+galaxy3","errors":null,"id":1,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":"InterProScan","name":"InterProScan","outputs":[{"name":"outfile_tsv","type":"tabular"},{"name":"outfile_xml","type":"xml"}],"position":{"left":162.5,"top":279.5},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/interproscan/interproscan/5.59-91.0+galaxy3","tool_shed_repository":{"changeset_revision":"74810db257cc","name":"interproscan","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"applications\\": [\\"TIGRFAM\\", \\"FunFam\\", \\"SFLD\\", \\"SUPERFAMILY\\", \\"PANTHER\\", \\"Gene3D\\", \\"Hamap\\", \\"PrositeProfiles\\", \\"Coils\\", \\"SMART\\", \\"CDD\\", \\"PRINTS\\", \\"PIRSR\\", \\"PrositePatterns\\", \\"AntiFam\\", \\"Pfam\\", \\"MobiDBLite\\", \\"PIRSF\\"], \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"database\\": \\"5.59-91.0\\", \\"goterms\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iprlookup\\": false, \\"licensed\\": {\\"use\\": \\"false\\", \\"__current_case__\\": 1, \\"applications_licensed\\": [\\"Phobius\\", \\"SignalP_EUK\\", \\"TMHMM\\"]}, \\"oformat\\": [\\"TSV\\", \\"XML\\"], \\"pathways\\": true, \\"seqtype\\": \\"p\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.59-91.0+galaxy3","type":"tool","uuid":"36d72511-ef8c-42ab-8944-b7aef340a9bc","when":null,"workflow_outputs":[{"label":"interproscan xml","output_name":"outfile_xml","uuid":"bef32b2c-0065-4854-9e5a-898e689d559c"},{"label":"interproscan tabular","output_name":"outfile_tsv","uuid":"d58da4b2-4c05-491f-8509-609184241715"}]},"2":{"annotation":"EggNOG Mapper compares each protein sequence of the annotation to a huge set of ortholog groups from the EggNOG database.","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/eggnog_mapper/eggnog_mapper/2.1.8+galaxy4","errors":null,"id":2,"input_connections":{"ortho_method|input":{"id":0,"output_name":"output"}},"inputs":[],"label":"eggNOG Mapper","name":"eggNOG Mapper","outputs":[{"name":"seed_orthologs","type":"tabular"},{"name":"annotations","type":"tabular"}],"position":{"left":340,"top":52},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/eggnog_mapper/eggnog_mapper/2.1.8+galaxy4","tool_shed_repository":{"changeset_revision":"d9c3016f7283","name":"eggnog_mapper","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"annotation_options\\": {\\"no_annot\\": \\"\\", \\"__current_case__\\": 0, \\"seed_ortholog_evalue\\": \\"0.001\\", \\"seed_ortholog_score\\": null, \\"tax_scope\\": null, \\"target_orthologs\\": \\"all\\", \\"go_evidence\\": \\"non-electronic\\"}, \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"eggnog_data\\": \\"5.0.2\\", \\"ortho_method\\": {\\"m\\": \\"diamond\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_trans\\": {\\"itype\\": \\"proteins\\", \\"__current_case__\\": 0}, \\"matrix_gapcosts\\": {\\"matrix\\": \\"BLOSUM62\\", \\"__current_case__\\": 2, \\"gap_costs\\": \\"--gapopen 11 --gapextend 1\\"}, \\"sensmode\\": \\"sensitive\\", \\"dmnd_iterate\\": false, \\"dmnd_ignore_warnings\\": false, \\"query_cover\\": null, \\"subject_cover\\": null, \\"pident\\": null, \\"evalue\\": null, \\"score\\": \\"0.001\\"}, \\"output_options\\": {\\"no_file_comments\\": false, \\"report_orthologs\\": false, \\"md5\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.8+galaxy4","type":"tool","uuid":"76b27114-c49f-41b8-9333-91953729dee3","when":null,"workflow_outputs":[{"label":"eggNOG Mapper annotations","output_name":"annotations","uuid":"25a1e387-baa5-48b6-b142-198bec95463e"},{"label":"eggNOG Mapper seed_orthologs","output_name":"seed_orthologs","uuid":"47ac8226-b800-479f-a6f0-f882bafc33d7"}]}},"tags":[],"uuid":"4cbba315-c9bc-4895-aeeb-57dadef3542a","version":2},"readme":"# Functional annotation of protein sequences Workflow\\n\\nThis workflow uses eggNOG mapper and Interproscan for functional annotation of protein sequences.\\nIt can be used on proteins from any organism.\\n\\nEggNOG Mapper compares each protein sequence of the annotation to a huge set of ortholog groups from the EggNOG database. In this database, each ortholog group is associated with functional annotation like Gene Ontology (GO) terms or KEGG pathways. When the protein sequence of a new gene is found to be very similar to one of these ortholog groups, the corresponding functional annotation is transfered to this new gene.\\n\\nInterProScan is a tool that analyses each protein sequence from our annotation to determine if they contain one or several of the signatures from InterPro. When a protein contains a known signature, the corresponding functional annotation will be assigned to it by InterProScan.\\n\\n## Input dataset\\nThis workflow requires only a input file: a protein sequences file in fasta format.\\n\\n\\n## Outputs for eggNOG Mapper\\nThe output of this tool is a tabular file, where each line represents a gene from our annotation, with the functional annotation that was found by EggNOG-mapper. It includes a predicted protein name, GO terms, EC numbers, KEGG identifiers, etc.\\n\\n## Outputs for Interproscan\\nThe output of this tool is both a tabular file and an XML file. Both contain the same information, but the tabular one is more readable for a Human: each line represents a gene from our annotation, with the different domains and motifs that were found by InterProScan.\\n\\nEach line correspond to a motif found in one of the annotated proteins. The most interesting columns are:\\n- Column 1: the protein identifier\\n- Column 5: the identifier of the signature that was found in the protein sequence\\n- Column 4: the databank where this signature comes from (InterProScan regroups several motifs databanks)\\n- Column 6: the human readable description of the motif\\n- Columns 7 and 8: the position where the motif was found\\n- Column 9: a score for the match (if available)\\n- Column 12 and 13: identifier of the signature integrated in InterPro (if available). Have a look an example webpage for IPR036859 on InterPro.\\n- The following columns contains various identifiers that were assigned to the protein based on the match with the signature (Gene ontology term, Reactome, …)\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-12-04\\n\\nInitial version of the Functional annotation of protein sequence Workflow.\\n","diagrams":"# Workflow diagrams\\n\\n## Functional annotation of protein sequences\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ input\\"]@{ shape: doc }\\n1[\\"InterProScan\\"]@{ shape: process }\\n0 --> 1\\n2[\\"eggNOG Mapper\\"]@{ shape: process }\\n0 --> 2\\n```\\n","updated":"2024-12-04T00:00:00","trsID":"#workflow/github.com/iwc-workflows/functional-annotation-protein-sequences/main","dockstore_id":30414,"categories":[],"collections":["Genome Annotation"],"doi":"10.5281/zenodo.15078818","tests":[{"doc":"Test outline for Functional_annotation_of_protein_sequences.ga","job":{"input":{"class":"File","location":"https://zenodo.org/record/8414802/files/protein_sequences.fasta?download=1","filetype":"fasta"}},"outputs":{"eggNOG Mapper seed_orthologs":{"location":"https://zenodo.org/records/13951790/files/eggNOG_Mapper_seed_orthologs.tabular?download=1&preview=1","compare":"sim_size","delta":50000},"eggNOG Mapper annotations":{"location":"https://zenodo.org/records/13951790/files/eggNOG_Mapper_annot.tabular?download=1&preview=1","compare":"sim_size","delta":100000},"interproscan xml":{"location":"https://zenodo.org/records/13951790/files/interProScan.xml?download=1&preview=1","compare":"sim_size","delta":7000000},"interproscan tabular":{"location":"https://zenodo.org/records/13951790/files/interProScan.tabular?download=1&preview=1","compare":"sim_size","delta":2000000}}}]}],"path":"./workflows/genome_annotation/functional-annotation/functional-annotation-protein-sequences"},{"version":1.2,"workflows":[{"name":"main","primaryDescriptorPath":"/metaprosip.ga","subclass":"Galaxy","publish":true,"testParameterFiles":["/metaprosip-tests.yml"],"authors":[{"name":"Matthias Bernt","orcid":"0000-0003-3763-0797"}],"definition":{"a_galaxy_workflow":"true","annotation":"Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics ","creator":[{"class":"Person","identifier":"0000-0003-3763-0797","name":"Matthias Bernt"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"MetaProSIP OpenMS 2.8","steps":{"0":{"annotation":"mgf, ms2, mzml or mzxml datasets","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"mgf, ms2, mzml or mzxml datasets","name":"Centroided LC-MS datasets"}],"label":"Centroided LC-MS datasets","name":"Input dataset collection","outputs":[],"position":{"left":7.076599011515327,"top":129.1847085591696},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"7676403b-b391-41fe-b07b-a35a13ff318f","when":null,"workflow_outputs":[]},"1":{"annotation":"A protein sequence database in FASTA format","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"A protein sequence database in FASTA format","name":"Fasta Database"}],"label":"Fasta Database","name":"Input dataset","outputs":[],"position":{"left":5.760014806230345,"top":246.36005905656776},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"27f8d0f6-773b-4373-ad0a-b3b1ddb0b6ec","when":null,"workflow_outputs":[]},"2":{"annotation":"passed to\\n\\n- MSGF+ -precursor_mass_tolerance\\n- MetaProSIP -mz_tolerance_ppm","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"passed to\\n\\n- MSGF+ -precursor_mass_tolerance\\n- MetaProSIP -mz_tolerance_ppm","name":"Precursor monoisotopic mass tolerance (ppm)"}],"label":"Precursor monoisotopic mass tolerance (ppm)","name":"Input parameter","outputs":[],"position":{"left":9.126262562065165,"top":661.3959437836211},"tool_id":null,"tool_state":"{\\"parameter_type\\": \\"float\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f5dd354c-a6af-490a-9613-93f269eefebe","when":null,"workflow_outputs":[]},"3":{"annotation":"used for the MSGF+ search","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"used for the MSGF+ search","name":"Fixed modifications"}],"label":"Fixed modifications","name":"Input parameter","outputs":[],"position":{"left":10.008971181763343,"top":802.6437324919638},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"f1d7a7ce-4007-456b-aa64-e673391ca563","when":null,"workflow_outputs":[]},"4":{"annotation":"used for the MSGF+ search","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"used for the MSGF+ search","name":"Variable modifications"}],"label":"Variable modifications","name":"Input parameter","outputs":[],"position":{"left":6.208847840341662,"top":928.9752741457733},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"b3ef07df-9879-4b2d-860a-6395606f7c25","when":null,"workflow_outputs":[]},"5":{"annotation":"Used for MetaProSIP (single letter code)","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Used for MetaProSIP (single letter code)","name":"Labeled element"}],"label":"Labeled element","name":"Input parameter","outputs":[],"position":{"left":0,"top":1053.3020245946063},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"65dc4bdd-aff3-4e04-8420-3f49d44fbcb9","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"__SORTLIST__","errors":null,"id":6,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Sort collection","outputs":[{"name":"output","type":"input"}],"position":{"left":283.10891498997137,"top":113.17633490829857},"post_job_actions":{"HideDatasetActionoutput":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"output"}},"tool_id":"__SORTLIST__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sort_type\\": {\\"sort_type\\": \\"alpha\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"428ef215-857a-4525-a650-240c1b1a4d85","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_decoydatabase/DecoyDatabase/3.1+galaxy0","errors":null,"id":7,"input_connections":{"in_cond|in":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool DecoyDatabase","name":"in_cond"}],"label":null,"name":"DecoyDatabase","outputs":[{"name":"out","type":"fasta"}],"position":{"left":259.8666687011719,"top":444.29214961038565},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_decoydatabase/DecoyDatabase/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"06cca574e337","name":"openms_decoydatabase","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"Decoy\\": {\\"non_shuffle_pattern\\": \\"\\", \\"keepPeptideNTerm\\": true, \\"keepPeptideCTerm\\": true}, \\"OPTIONAL_OUTPUTS\\": null, \\"adv_opts\\": {\\"shuffle_max_attempts\\": \\"30\\", \\"shuffle_sequence_identity_threshold\\": \\"0.5\\", \\"seed\\": \\"1\\", \\"force\\": false, \\"test\\": \\"False\\"}, \\"decoy_string\\": \\"DECOY_\\", \\"decoy_string_position\\": \\"prefix\\", \\"enzyme\\": \\"Trypsin\\", \\"in_cond\\": {\\"in_select\\": \\"no\\", \\"__current_case__\\": 0, \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"method\\": \\"reverse\\", \\"only_decoy\\": false, \\"type\\": \\"protein\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"dbe2ad39-3591-46c8-abf1-2fcbe284f78e","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex/3.1+galaxy0","errors":null,"id":8,"input_connections":{"in":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"FeatureFinderMultiplex","outputs":[{"name":"out","type":"featurexml"}],"position":{"left":889.1042176446133,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"c9b2a5022d9e","name":"openms_featurefindermultiplex","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"OPTIONAL_OUTPUTS\\": [\\"out_FLAG\\"], \\"adv_opts\\": {\\"force\\": false, \\"test\\": \\"False\\"}, \\"algorithm\\": {\\"labels\\": \\"\\", \\"charge\\": \\"1:4\\", \\"isotopes_per_peptide\\": \\"3:6\\", \\"rt_typical\\": \\"40.0\\", \\"rt_band\\": \\"0.0\\", \\"rt_min\\": \\"2.0\\", \\"mz_tolerance\\": \\"6.0\\", \\"mz_unit\\": \\"ppm\\", \\"intensity_cutoff\\": \\"1000.0\\", \\"peptide_similarity\\": \\"0.5\\", \\"averagine_similarity\\": \\"0.4\\", \\"averagine_similarity_scaling\\": \\"0.95\\", \\"missed_cleavages\\": \\"0\\", \\"spectrum_type\\": \\"automatic\\", \\"averagine_type\\": \\"peptide\\", \\"knock_out\\": false}, \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"labels\\": {\\"Arg6\\": \\"6.0201290268\\", \\"Arg10\\": \\"10.0082686\\", \\"Lys4\\": \\"4.0251069836\\", \\"Lys6\\": \\"6.0201290268\\", \\"Lys8\\": \\"8.0141988132\\", \\"Leu3\\": \\"3.01883\\", \\"Dimethyl0\\": \\"28.0313\\", \\"Dimethyl4\\": \\"32.056407\\", \\"Dimethyl6\\": \\"34.063117\\", \\"Dimethyl8\\": \\"36.07567\\", \\"ICPL0\\": \\"105.021464\\", \\"ICPL4\\": \\"109.046571\\", \\"ICPL6\\": \\"111.041593\\", \\"ICPL10\\": \\"115.0667\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"a98005bb-1181-4632-939c-520962481b41","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_msgfplusadapter/MSGFPlusAdapter/3.1+galaxy0","errors":null,"id":9,"input_connections":{"database":{"id":7,"output_name":"out"},"fixed_modifications":{"id":3,"output_name":"output"},"in":{"id":6,"output_name":"output"},"precursor_mass_tolerance":{"id":2,"output_name":"output"},"variable_modifications":{"id":4,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MSGFPlusAdapter","name":"conf"}],"label":null,"name":"MSGFPlusAdapter","outputs":[{"name":"out","type":"idxml"}],"position":{"left":563.0759987411635,"top":291.24336246769883},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_msgfplusadapter/MSGFPlusAdapter/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"810779f0cf49","name":"openms_msgfplusadapter","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"OPTIONAL_OUTPUTS\\": [\\"out_FLAG\\"], \\"PeptideIndexing\\": {\\"decoy_string\\": \\"\\", \\"decoy_string_position\\": \\"prefix\\", \\"missing_decoy_action\\": \\"warn\\", \\"write_protein_sequence\\": false, \\"write_protein_description\\": false, \\"keep_unreferenced_proteins\\": false, \\"unmatched_action\\": \\"error\\", \\"aaa_max\\": \\"3\\", \\"mismatches_max\\": \\"0\\", \\"IL_equivalent\\": false, \\"allow_nterm_protein_cleavage\\": true, \\"enzyme\\": {\\"name\\": \\"auto\\", \\"specificity\\": \\"auto\\"}}, \\"add_features\\": true, \\"adv_opts\\": {\\"legacy_conversion\\": false, \\"force\\": false, \\"test\\": \\"False\\"}, \\"conf\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"enzyme\\": \\"Trypsin/P\\", \\"fixed_modifications\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fragment_method\\": \\"from_spectrum\\", \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"instrument\\": \\"low_res\\", \\"isotope_error_range\\": \\"0,1\\", \\"matches_per_spec\\": \\"1\\", \\"max_missed_cleavages\\": \\"-1\\", \\"max_mods\\": \\"2\\", \\"max_peptide_length\\": \\"40\\", \\"max_precursor_charge\\": \\"3\\", \\"min_peptide_length\\": \\"6\\", \\"min_precursor_charge\\": \\"2\\", \\"precursor_error_units\\": \\"ppm\\", \\"precursor_mass_tolerance\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"protocol\\": \\"automatic\\", \\"reindex\\": true, \\"tasks\\": \\"0\\", \\"tryptic\\": \\"fully\\", \\"variable_modifications\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"4edeb1a3-bc6e-44d0-8633-1bc439ca988e","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_peptideindexer/PeptideIndexer/3.1+galaxy0","errors":null,"id":10,"input_connections":{"fasta":{"id":7,"output_name":"out"},"in":{"id":9,"output_name":"out"}},"inputs":[],"label":null,"name":"PeptideIndexer","outputs":[{"name":"out","type":"idxml"}],"position":{"left":856.0592678761781,"top":466.70716390817233},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_peptideindexer/PeptideIndexer/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"0693b186f39e","name":"openms_peptideindexer","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"IL_equivalent\\": false, \\"OPTIONAL_OUTPUTS\\": null, \\"aaa_max\\": \\"3\\", \\"adv_opts\\": {\\"force\\": false, \\"test\\": \\"False\\"}, \\"allow_nterm_protein_cleavage\\": true, \\"decoy_string\\": \\"\\", \\"decoy_string_position\\": \\"prefix\\", \\"enzyme\\": {\\"name\\": \\"auto\\", \\"specificity\\": \\"auto\\"}, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_unreferenced_proteins\\": false, \\"mismatches_max\\": \\"0\\", \\"missing_decoy_action\\": \\"error\\", \\"unmatched_action\\": \\"error\\", \\"write_protein_description\\": false, \\"write_protein_sequence\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"87a572fe-0b20-478e-bbc3-58b1dff8bc7d","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate/3.1+galaxy0","errors":null,"id":11,"input_connections":{"in":{"id":10,"output_name":"out"}},"inputs":[],"label":null,"name":"FalseDiscoveryRate","outputs":[{"name":"out","type":"idxml"}],"position":{"left":1145.676234466169,"top":737.5180702002503},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"4138acdeedfb","name":"openms_falsediscoveryrate","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"FDR\\": {\\"PSM\\": \\"1.0\\", \\"protein\\": \\"1.0\\", \\"cleanup\\": {\\"remove_proteins_without_psms\\": true, \\"remove_psms_without_proteins\\": true, \\"remove_spectra_without_psms\\": true}}, \\"OPTIONAL_OUTPUTS\\": null, \\"PSM\\": true, \\"adv_opts\\": {\\"force\\": false, \\"test\\": \\"False\\"}, \\"algorithm\\": {\\"no_qvalues\\": false, \\"use_all_hits\\": false, \\"split_charge_variants\\": false, \\"treat_runs_separately\\": false, \\"add_decoy_peptides\\": false, \\"add_decoy_proteins\\": false, \\"conservative\\": true}, \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"peptide\\": false, \\"protein\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"21e5233c-0395-44aa-814a-e690c9a759db","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idmapper/IDMapper/3.1+galaxy0","errors":null,"id":12,"input_connections":{"id":{"id":11,"output_name":"out"},"in":{"id":8,"output_name":"out"}},"inputs":[{"description":"runtime parameter for tool IDMapper","name":"spectra"}],"label":null,"name":"IDMapper","outputs":[{"name":"out","type":"input"}],"position":{"left":1393.1927501192213,"top":333.20925115710634},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idmapper/IDMapper/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"ea02403ff738","name":"openms_idmapper","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"OPTIONAL_OUTPUTS\\": null, \\"adv_opts\\": {\\"ignore_charge\\": false, \\"force\\": false, \\"test\\": \\"False\\"}, \\"consensus\\": {\\"use_subelements\\": false, \\"annotate_ids_with_subelements\\": false}, \\"feature\\": {\\"use_centroid_rt\\": false, \\"use_centroid_mz\\": true}, \\"id\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mz_measure\\": \\"ppm\\", \\"mz_reference\\": \\"peptide\\", \\"mz_tolerance\\": \\"20.0\\", \\"rt_tolerance\\": \\"5.0\\", \\"spectra\\": {\\"_in\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"d59740a0-a78e-44d1-9481-b9249e3489b6","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_metaprosip/MetaProSIP/3.1+galaxy0","errors":null,"id":13,"input_connections":{"in_fasta":{"id":7,"output_name":"out"},"in_featureXML":{"id":12,"output_name":"out"},"in_mzML":{"id":6,"output_name":"output"},"labeling_element":{"id":5,"output_name":"output"},"mz_tolerance_ppm":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"MetaProSIP","outputs":[{"name":"out_csv","type":"csv"},{"name":"out_peptide_centric_csv","type":"csv"}],"position":{"left":1666.1430646529957,"top":602.8082753835706},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/openms_metaprosip/MetaProSIP/3.1+galaxy0","tool_shed_repository":{"changeset_revision":"5f72a8fa935d","name":"openms_metaprosip","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"OPTIONAL_OUTPUTS\\": null, \\"adv_opts\\": {\\"min_correlation_distance_to_averagine\\": \\"-1.0\\", \\"pattern_15N_TIC_threshold\\": \\"0.95\\", \\"pattern_13C_TIC_threshold\\": \\"0.95\\", \\"pattern_2H_TIC_threshold\\": \\"0.95\\", \\"pattern_18O_TIC_threshold\\": \\"0.95\\", \\"heatmap_bins\\": \\"20\\", \\"observed_peak_fraction\\": \\"0.5\\", \\"min_consecutive_isotopes\\": \\"2\\", \\"score_plot_yaxis_min\\": \\"0.0\\", \\"collect_method\\": \\"correlation_maximum\\", \\"lowRIA_correlation_threshold\\": \\"-1.0\\", \\"force\\": false, \\"test\\": \\"False\\"}, \\"cluster\\": false, \\"correlation_threshold\\": \\"0.7\\", \\"decomposition_threshold\\": \\"0.7\\", \\"filter_monoisotopic\\": false, \\"in_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"in_featureXML\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"in_mzML\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"intensity_threshold\\": \\"10.0\\", \\"labeling_element\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mz_tolerance_ppm\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"plot_extension\\": \\"png\\", \\"qc_output_directory\\": \\"\\", \\"report_natural_peptides\\": false, \\"rt_tolerance_s\\": \\"30.0\\", \\"use_averagine_ids\\": false, \\"use_unassigned_ids\\": false, \\"weight_merge_window\\": \\"5.0\\", \\"xic_threshold\\": \\"0.7\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.1+galaxy0","type":"tool","uuid":"224b7012-ffd4-4286-9338-97ac411bb313","when":null,"workflow_outputs":[{"label":"Feature fitting result","output_name":"out_csv","uuid":"b2370639-8c23-46a6-be8a-b2136d0b8de0"},{"label":"Peptide centric result","output_name":"out_peptide_centric_csv","uuid":"1985589f-e690-48e4-a41f-7dc20eb7446f"}]}},"tags":[],"uuid":"b03da1a7-fc6f-419f-aa2c-431a670f9a96","version":1},"readme":"# MetaProSIP: automated inference of elemental fluxes in microbial communities\\n\\n## Inputs dataset\\n\\n- `Centroided LC-MS datasets` in mzML (MetaProSIP is mainly tested on data generated by orbitrap instruments)\\n- `Fasta Database` in Fasta (aminoacid sequences)\\n\\n## Inputs values\\n\\n- `Precursor monoisotopic mass tolerance` (ppm): This value is passed to\\n  - MSGFPlusAdapter parameter `Precursor monoisotopic mass tolerance` (-precursor_mass_tolerance)\\n  - MetaProSIP parameter `Tolerance in ppm` (-mz_tolerance_ppm)\\n- Fixed modifications\\n- Variable modifications\\n- Labeled element\\n\\n## Processing\\n\\n- DecoyDatabase: Add decoy sequences to the Fasta database (for FDR calculation)\\n- FeatureFinderCentroided: identify eluting peptides that correspond to isotopologues with natural isotopic distributions \\n- MSGFPlusAdapter: identify peptides through peptide fragment fingerprinting (database search)\\n- FeatureFinderMultiplex: detect elution profiles of unlabeled peptides\\n- PeptideIndexer: annotate protein association to identified peptides\\n- FalseDiscoveryRate: Calculate FDR\\n- IDMapper: map identified spectra to elution profiles\\n- MetaProSIP: calculate the protein-SIP features, to perform functional grouping, and for protein inference\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-06-17\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_decoydatabase/DecoyDatabase/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_decoydatabase/DecoyDatabase/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_msgfplusadapter/MSGFPlusAdapter/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_msgfplusadapter/MSGFPlusAdapter/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_peptideindexer/PeptideIndexer/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_peptideindexer/PeptideIndexer/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_falsediscoveryrate/FalseDiscoveryRate/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idmapper/IDMapper/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idmapper/IDMapper/3.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_metaprosip/MetaProSIP/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/openms_metaprosip/MetaProSIP/3.1+galaxy0`\\n\\n## [0.1] - 2023-04-13\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## MetaProSIP OpenMS 2.8\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Centroided LC-MS datasets\\"]@{ shape: docs }\\n1[\\"ℹ️ Fasta Database\\"]@{ shape: doc }\\n2[\\"ℹ️ Precursor monoisotopic mass tolerance (ppm)\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Fixed modifications\\"]@{ shape: lean-l }\\n4[\\"ℹ️ Variable modifications\\"]@{ shape: lean-l }\\n5[\\"ℹ️ Labeled element\\"]@{ shape: lean-l }\\n6[\\"Sort collection\\"]@{ shape: process }\\n0 --> 6\\n7[\\"DecoyDatabase\\"]@{ shape: process }\\n1 --> 7\\n8[\\"FeatureFinderMultiplex\\"]@{ shape: process }\\n6 --> 8\\n9[\\"MSGFPlusAdapter\\"]@{ shape: process }\\n7 --> 9\\n3 --> 9\\n6 --> 9\\n2 --> 9\\n4 --> 9\\n10[\\"PeptideIndexer\\"]@{ shape: process }\\n7 --> 10\\n9 --> 10\\n11[\\"FalseDiscoveryRate\\"]@{ shape: process }\\n10 --> 11\\n12[\\"IDMapper\\"]@{ shape: process }\\n11 --> 12\\n8 --> 12\\n13[\\"MetaProSIP\\"]@{ shape: process }\\n7 --> 13\\n12 --> 13\\n6 --> 13\\n5 --> 13\\n2 --> 13\\n```\\n","updated":"2024-06-17T00:00:00","trsID":"#workflow/github.com/iwc-workflows/openms-metaprosip/main","dockstore_id":24039,"categories":[],"collections":["Proteomics"],"doi":"10.5281/zenodo.15078850","tests":[{"doc":"Test outline for MetaProSIP","job":{"Fasta Database":{"class":"File","location":"https://abibuilder.cs.uni-tuebingen.de/archive/openms/galaxy-testdata/metaprosip/Zeitz_1and2_454AllContigs_HEX.fasta","filetype":"fasta"},"Centroided LC-MS datasets":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"Zeitz_SIP_13-II_020_picked.mzML","location":"https://abibuilder.cs.uni-tuebingen.de/archive/openms/galaxy-testdata/metaprosip/Zeitz_SIP_13-II_020_picked.mzML","filetype":"mzml"}]},"Precursor monoisotopic mass tolerance (ppm)":10,"Fixed modifications":"Carbamidomethyl (C)","Variable modifications":"Oxidation (M)","Labeled element":"C"},"outputs":{"Peptide centric result":{"element_tests":{"Zeitz_SIP_13-II_020_picked.mzML":{"asserts":[{"that":"has_n_lines","n":85},{"that":"has_text","text":"Peptide Sequence"},{"that":"has_text","text":"AATGTPSPAGSPPPIVPAPK"}]}}},"Feature fitting result":{"element_tests":{"Zeitz_SIP_13-II_020_picked.mzML":{"asserts":[{"that":"has_n_lines","n":4091},{"that":"has_text","text":"Group 1"},{"that":"has_text","text":"LYSFHTLHQTYMK"}]}}}}}]}],"path":"./workflows/proteomics/openms-metaprosip"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/iwc-clinicalmp-discovery-workflow.ga","testParameterFiles":["/iwc-clinicalmp-discovery-workflow-tests.yml"],"authors":[{"name":"Subina Mehta","orcid":"0000-0001-9818-0537"}],"definition":{"a_galaxy_workflow":"true","annotation":"Workflow for clinical metaproteomics database searching","comments":[],"creator":[{"class":"Organization","identifier":"0000-0001-9818-0537","name":"Subina Mehta"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Clinical Metaproteomics Discovery Workflow","release":"0.1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Adding decoys to MetaNovo Human SwissProt cRAP to improve FDR","name":"Human UniProt Microbial Proteins from MetaNovo and cRAP"}],"label":"Human UniProt Microbial Proteins from MetaNovo and cRAP","name":"Input dataset","outputs":[],"position":{"left":249.51593017578125,"top":272.60801696777344},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c0c1c1cf-e899-4b8e-b1b1-6a4f24d87ad1","when":null,"workflow_outputs":[]},"1":{"annotation":"identification parameters for SGPS","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/ident_params/4.0.41+galaxy1","errors":null,"id":1,"input_connections":{},"inputs":[],"label":"IdentificationParameters:PAR file","name":"Identification Parameters","outputs":[{"name":"Identification_Parameters_File","type":"json"}],"position":{"left":814.9930419921875,"top":0},"post_job_actions":{"RenameDatasetActionIdentification_Parameters_File":{"action_arguments":{"newname":"Identification Parameters (PAR)"},"action_type":"RenameDatasetAction","output_name":"Identification_Parameters_File"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/ident_params/4.0.41+galaxy1","tool_shed_repository":{"changeset_revision":"3ec27b4cee7c","name":"peptideshaker","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_options\\": {\\"spectrum_annotation_options\\": {\\"annotation_level\\": \\"0.75\\", \\"annotation_mz_tolerance\\": \\"0.5\\", \\"annotation_high_resolution\\": true}, \\"sequence_matching_options\\": {\\"sequence_matching_type\\": \\"2\\", \\"sequence_matching_x\\": \\"0.25\\", \\"sequence_matching_enzymatic_tags\\": false, \\"sequence_matching_max_ptms_per_tag\\": \\"3\\", \\"sequence_matching_min_amino_acid_score\\": \\"30\\", \\"sequence_matching_min_tag_length\\": \\"3\\"}, \\"import_filters_options\\": {\\"min_peptide_length\\": \\"8\\", \\"max_peptide_length\\": \\"30\\", \\"max_precursor_error\\": \\"10.0\\", \\"max_precursor_error_type\\": \\"1\\", \\"missed_cleavages_min\\": null, \\"missed_cleavages_max\\": null, \\"exclude_unknown_ptms\\": true}, \\"ptm_localization_options\\": {\\"ptm_score\\": {\\"ptm_score_selector\\": \\"1\\", \\"__current_case__\\": 0, \\"score_neutral_losses\\": false, \\"ptm_threshold\\": \\"95.0\\"}, \\"ptm_sequence_matching_type\\": \\"1\\", \\"ptm_alignment\\": true}, \\"gene_annotation_options\\": {\\"use_gene_mapping\\": false, \\"update_gene_mapping\\": false}, \\"protein_inference_options\\": {\\"simplify_protein_groups_conditional\\": {\\"simplify_protein_groups_selector\\": \\"yes\\", \\"__current_case__\\": 1, \\"simplify_protein_groups_evidence\\": true, \\"simplify_protein_groups_confidence\\": true, \\"simplify_protein_groups_confidence_threshold\\": \\"0.05\\", \\"simplify_protein_groups_enzymaticity\\": true, \\"simplify_protein_groups_variant\\": true}, \\"simplify_protein_pi_modifications\\": true}, \\"validation_levels_options\\": {\\"protein_fdr\\": \\"1.0\\", \\"peptide_fdr\\": \\"1.0\\", \\"psm_fdr\\": \\"1.0\\"}, \\"fraction_analysis_options\\": {\\"protein_fraction_mw_confidence\\": \\"95.0\\"}, \\"database_processing_options\\": {\\"decoy_conditional\\": {\\"create_decoy_selector\\": \\"yes\\", \\"__current_case__\\": 1, \\"decoy_tag\\": \\"_REVERSED\\", \\"decoy_type\\": \\"1\\"}, \\"decoy_file_tag\\": \\"_concatenated_target_decoy\\"}}, \\"searchengines_options\\": {\\"xtandem\\": {\\"xtandem_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"myrimatch\\": {\\"myrimatch_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"msgf\\": {\\"msgf_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"ms_amanda\\": {\\"ms_amanda_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"omssa\\": {\\"omssa_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"comet\\": {\\"comet_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"tide\\": {\\"tide_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"directtag\\": {\\"directtag_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"metamorpheus\\": {\\"metamorpheus_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"novor\\": {\\"novor_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"pepnovo\\": {\\"pepnovo_advanced\\": \\"no\\", \\"__current_case__\\": 0}, \\"pnovo\\": {\\"pnovo_advanced\\": \\"no\\", \\"__current_case__\\": 0}}, \\"spectrum_matching_options\\": {\\"fixed_modifications\\": [\\"Carbamidomethylation of C\\", \\"TMT 11-plex of K+4\\", \\"TMT 11-plex of peptide N-term\\"], \\"variable_modifications\\": [\\"Oxidation of M\\"], \\"digestion\\": {\\"cleavage\\": \\"0\\", \\"__current_case__\\": 0, \\"digests\\": [{\\"__index__\\": 0, \\"enzyme\\": \\"Trypsin\\", \\"missed_cleavages\\": \\"2\\", \\"specificity\\": \\"0\\"}]}, \\"precursor_ion_tol_units\\": \\"1\\", \\"precursor_ion_tol\\": \\"10.0\\", \\"fragment_tol_units\\": \\"0\\", \\"fragment_tol\\": \\"0.6\\", \\"min_charge\\": \\"2\\", \\"max_charge\\": \\"6\\", \\"forward_ion\\": \\"b\\", \\"reverse_ion\\": \\"y\\", \\"min_isotope\\": \\"0\\", \\"max_isotope\\": \\"1\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.0.41+galaxy1","type":"tool","uuid":"abaf5892-7976-4526-9cf6-a0c96b9222ea","when":null,"workflow_outputs":[]},"2":{"annotation":"Input Raw files","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Input Raw files","name":"Tandem Mass Spectrometry MSMS files"}],"label":"Tandem Mass Spectrometry MSMS files","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":833.7075958251953},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"thermo.raw\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"c0f85358-244a-4254-ac94-e920a6cdbe39","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","errors":null,"id":3,"input_connections":{},"inputs":[],"label":"Human SwissProt","name":"Protein Database Downloader","outputs":[{"name":"output_database","type":"fasta"}],"position":{"left":634.640625488066,"top":607.9116168878928},"post_job_actions":{"RenameDatasetActionoutput_database":{"action_arguments":{"newname":"Human SwissProt"},"action_type":"RenameDatasetAction","output_name":"output_database"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","tool_shed_repository":{"changeset_revision":"983bf725dfc2","name":"dbbuilder","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"source\\": {\\"from\\": \\"uniprot\\", \\"__current_case__\\": 0, \\"taxon\\": \\"9606\\", \\"taxon_id\\": null, \\"reviewed\\": \\"reviewed:true\\", \\"set\\": \\"keyword:KW-1185\\", \\"include_isoform\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.4","type":"tool","uuid":"5b0de886-9f18-4dd2-8e81-4b29467e727a","when":null,"workflow_outputs":[{"label":"Human SwissProt","output_name":"output_database","uuid":"335f3fe6-c7fb-4fd6-b04e-9f5afb3fe5a6"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","errors":null,"id":4,"input_connections":{},"inputs":[],"label":"Contaminants","name":"Protein Database Downloader","outputs":[{"name":"output_database","type":"fasta"}],"position":{"left":631.7343839163847,"top":789.1078163439809},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","tool_shed_repository":{"changeset_revision":"983bf725dfc2","name":"dbbuilder","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"source\\": {\\"from\\": \\"cRAP\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.4","type":"tool","uuid":"6f0be889-015d-4be0-a997-dd63e18344eb","when":null,"workflow_outputs":[]},"5":{"annotation":"File for experimental design containing Name, Fraction, Experiment, and PTM, where the name refers to the dataset filename, with each dataset listed on a separate line.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Experimental Design manually created for MaxQuant","name":"Experimental Design Discovery MaxQuant"}],"label":"Experimental Design Discovery MaxQuant","name":"Input dataset","outputs":[],"position":{"left":872.66015625,"top":1088.484375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"9652a70a-de8c-4966-b7cf-121f1b9f0b54","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/fasta_cli/4.0.41+galaxy1","errors":null,"id":6,"input_connections":{"input_database":{"id":0,"output_name":"output"}},"inputs":[],"label":"FastaCLI_ MetaNovo Human SwissProt cRAP with decoys","name":"FastaCLI","outputs":[{"name":"input_database_concatenated_target_decoy","type":"fasta"}],"position":{"left":873.1780375305545,"top":286.53022010497557},"post_job_actions":{"RenameDatasetActioninput_database_concatenated_target_decoy":{"action_arguments":{"newname":"FastaCLI_ MetaNovo Human SwissProt cRAP with decoys"},"action_type":"RenameDatasetAction","output_name":"input_database_concatenated_target_decoy"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/fasta_cli/4.0.41+galaxy1","tool_shed_repository":{"changeset_revision":"3ec27b4cee7c","name":"peptideshaker","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"database_processing_options\\": {\\"decoy_tag\\": \\"_REVERSED\\", \\"decoy_type\\": \\"2\\", \\"decoy_file_tag\\": \\"_concatenated_target_decoy\\"}, \\"input_database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.0.41+galaxy1","type":"tool","uuid":"7a0729f9-8827-4202-a20e-48e103efc502","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.20287.2","errors":null,"id":7,"input_connections":{"input":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"msconvert","outputs":[{"name":"output","type":"mzml"}],"position":{"left":213.25472010132836,"top":454.38506518621944},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.20287.2","tool_shed_repository":{"changeset_revision":"6153e8ada1ee","name":"msconvert","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"data_processing\\": {\\"peak_picking\\": {\\"pick_peaks\\": true, \\"__current_case__\\": 1, \\"pick_peaks_ms_levels\\": \\"1-\\", \\"pick_peaks_algorithm\\": \\"vendor\\"}, \\"precursor_refinement\\": {\\"use_mzrefinement\\": false, \\"__current_case__\\": 0}, \\"charge_state_calculation\\": {\\"charge_state_calculation_method\\": \\"false\\", \\"__current_case__\\": 0}, \\"thresholds\\": [], \\"filter_mz_windows\\": {\\"do_mzwindow_filter\\": false, \\"__current_case__\\": 0}, \\"etd_filtering\\": {\\"do_etd_filtering\\": false, \\"__current_case__\\": 0}, \\"ms2denoise\\": {\\"denoise\\": false, \\"__current_case__\\": 1}, \\"ms2deisotope\\": false, \\"demultiplex\\": {\\"demultiplex_on\\": false, \\"__current_case__\\": 1}}, \\"filtering\\": {\\"activation\\": \\"false\\", \\"indices\\": [], \\"scan_numbers\\": [], \\"strip_it\\": false, \\"filter_ms_levels\\": {\\"do_ms_level_filter\\": false, \\"__current_case__\\": 0}, \\"polarity\\": \\"false\\", \\"analyzer\\": \\"false\\"}, \\"general_options\\": {\\"combineIonMobilitySpectra\\": false, \\"scan_summing\\": {\\"do_scan_summing\\": false, \\"__current_case__\\": 0}, \\"simAsSpectra\\": false, \\"srmAsSpectra\\": false, \\"acceptZeroLengthSpectra\\": false, \\"ignoreUnknownInstrumentError\\": false, \\"multi_run_output\\": {\\"do_multi_run_output\\": false, \\"__current_case__\\": 0, \\"runIndexSet\\": \\"0\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"license_agreement\\": true, \\"output_type\\": \\"mgf\\", \\"settings\\": {\\"mz_encoding\\": \\"64\\", \\"intensity_encoding\\": \\"32\\", \\"binary_compression\\": \\"zlib\\", \\"gzip_compression\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.0.20287.2","type":"tool","uuid":"83018642-2850-4df8-b328-ee1e8bdc8fcb","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":8,"input_connections":{"batchmode|input_fastas_0|input_fasta":{"id":3,"output_name":"output_database"},"batchmode|input_fastas_1|input_fasta":{"id":4,"output_name":"output_database"}},"inputs":[],"label":"Human SwissProt+cRAP","name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":902.4973253039605,"top":577.0886700391638},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Human SwissProt+cRAP"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"individual\\", \\"__current_case__\\": 0, \\"input_fastas\\": [{\\"__index__\\": 0, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"a35c2efa-70f2-41e8-bc46-cb2a245fdf57","when":null,"workflow_outputs":[{"label":"Human SwissProt+cRAP","output_name":"output","uuid":"2f14902f-b005-4c08-b6d1-6d7ee0b4b199"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/maxquant/2.0.3.0+galaxy0","errors":null,"id":9,"input_connections":{"input_opts|fasta_files":{"id":0,"output_name":"output"},"paramGroups_0|files":{"id":2,"output_name":"output"},"search_opts|template":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MaxQuant","name":"input_opts"},{"description":"runtime parameter for tool MaxQuant","name":"search_opts"}],"label":null,"name":"MaxQuant","outputs":[{"name":"proteinGroups","type":"tabular"},{"name":"mqpar","type":"xml"},{"name":"peptides","type":"tabular"},{"name":"msms","type":"tabular"},{"name":"msmsScans","type":"tabular"},{"name":"summary","type":"tabular"},{"name":"log","type":"txt"},{"name":"config","type":"txt"}],"position":{"left":1389.0771241398825,"top":778.3294605307798},"post_job_actions":{"TagDatasetActionpeptides":{"action_arguments":{"tags":"#MQ_Peptides"},"action_type":"TagDatasetAction","output_name":"peptides"},"TagDatasetActionproteinGroups":{"action_arguments":{"tags":"#MQ_ProteinGroups"},"action_type":"TagDatasetAction","output_name":"proteinGroups"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/maxquant/2.0.3.0+galaxy0","tool_shed_repository":{"changeset_revision":"9c52362ae7bb","name":"maxquant","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_opts\\": {\\"ftype\\": \\".thermo.raw\\", \\"fasta_files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"identifier_parse_rule\\": \\">([^\\\\\\\\s]*)\\", \\"description_parse_rule\\": \\">(.*)\\"}, \\"output_opts\\": {\\"dry_run\\": false, \\"output\\": [\\"proteinGroups\\", \\"mqpar\\", \\"peptides\\", \\"msms\\", \\"msmsScans\\", \\"summary\\", \\"log\\", \\"config\\"]}, \\"paramGroups\\": [{\\"__index__\\": 0, \\"files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maxMissedCleavages\\": \\"2\\", \\"fixedModifications\\": [\\"Carbamidomethyl (C)\\"], \\"variableModifications\\": [\\"Oxidation (M)\\"], \\"enzymes\\": [\\"Trypsin/P\\"], \\"digestion_mode\\": \\"0\\", \\"quant_method\\": {\\"select_quant_method\\": \\"reporter_ion_ms2\\", \\"__current_case__\\": 3, \\"iso_labels\\": {\\"labeling\\": \\"tmt11plex\\", \\"__current_case__\\": 4}, \\"filter_by_pif\\": {\\"filter_pif\\": \\"True\\", \\"__current_case__\\": 0, \\"reporter_pif\\": \\"0.75\\"}}}], \\"protein_quant\\": {\\"peptides_for_quantification\\": \\"1\\", \\"only_unmod_prot\\": {\\"unmod_prot\\": \\"True\\", \\"__current_case__\\": 0, \\"mods_used_prot_quant\\": [\\"Oxidation (M)\\"], \\"discard_unmod_cpart_peptides\\": true}, \\"lfq_opts\\": {\\"separateLfq\\": false, \\"lfqStabilizeLargeRatios\\": true, \\"lfqRequireMsms\\": true, \\"do_ibaq\\": {\\"ibaq\\": \\"False\\", \\"__current_case__\\": 1}, \\"advancedSiteIntensities\\": true}}, \\"qc\\": {\\"do_it\\": \\"false\\", \\"__current_case__\\": 1}, \\"search_opts\\": {\\"template\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_peptide_len\\": \\"8\\", \\"max_peptide_mass\\": \\"4600\\", \\"min_unique_pep\\": \\"0\\", \\"calc_peak_properties\\": false, \\"mbr\\": {\\"match_between_runs\\": \\"True\\", \\"__current_case__\\": 1, \\"matching_time_window\\": \\"0.7\\", \\"matching_ion_mobility_window\\": \\"0.05\\", \\"alignment_time_window\\": \\"20\\", \\"alignment_ion_mobility_window\\": \\"1\\", \\"match_unidentified_features\\": false}, \\"incl_contaminants\\": false, \\"decoy_mode\\": \\"revert\\", \\"psm_fdr\\": \\"0.01\\", \\"protein_fdr\\": \\"0.01\\", \\"min_pep_length\\": \\"8\\", \\"max_pep_length\\": \\"50\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.3.0+galaxy0","type":"tool","uuid":"46c61a69-355c-44ac-9945-0e4d43f0b2ec","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/4.0.41+galaxy1","errors":null,"id":10,"input_connections":{"input_fasta_file":{"id":6,"output_name":"input_database_concatenated_target_decoy"},"input_parameters_file":{"id":1,"output_name":"Identification_Parameters_File"},"peak_lists_files":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Search GUI","outputs":[{"name":"searchgui_results","type":"searchgui_archive"}],"position":{"left":1136.7332639974097,"top":127.37631486278212},"post_job_actions":{"TagDatasetActionsearchgui_results":{"action_arguments":{"tags":"#SGPS"},"action_type":"TagDatasetAction","output_name":"searchgui_results"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/4.0.41+galaxy1","tool_shed_repository":{"changeset_revision":"3ec27b4cee7c","name":"peptideshaker","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_fasta_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_parameters_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"peak_lists_files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"search_engines_options\\": {\\"engines\\": [\\"X!Tandem\\", \\"MSGF\\"]}, \\"searchgui_advanced\\": {\\"searchgui_advanced_selector\\": \\"basic\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.0.41+galaxy1","type":"tool","uuid":"58cf1b78-91c0-4726-8697-df2504d14611","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.0","errors":null,"id":11,"input_connections":{"input":{"id":8,"output_name":"output"}},"inputs":[],"label":"FASTA to Tabular","name":"FASTA-to-Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1164.015625,"top":614.1015625},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.0","tool_shed_repository":{"changeset_revision":"7e801ab2b70e","name":"fasta_to_tabular","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"descr_columns\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_first\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0","type":"tool","uuid":"9a63693a-1b5b-4160-99da-4636a09f3fed","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"Grep1","errors":null,"id":12,"input_connections":{"input":{"id":9,"output_name":"peptides"}},"inputs":[],"label":"Select microbial peptides MQ","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1644.109353292642,"top":809.7786537420372},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Select microbial peptides MQ"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": true, \\"pattern\\": \\"(_HUMAN)|(_REVERSED)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"cb9b81ce-c86e-4b07-ba18-3d1deddeded5","when":null,"workflow_outputs":[{"label":"Select microbial peptides MQ","output_name":"out_file1","uuid":"aff2fb3d-58d3-4d1d-bfa2-8b68f14ec0a0"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/2.0.33+galaxy1","errors":null,"id":13,"input_connections":{"searchgui_input":{"id":10,"output_name":"searchgui_results"}},"inputs":[{"description":"runtime parameter for tool Peptide Shaker","name":"optional_main_parameters"},{"description":"runtime parameter for tool Peptide Shaker","name":"optional_main_parameters"},{"description":"runtime parameter for tool Peptide Shaker","name":"optional_main_parameters"}],"label":"Peptide Shaker","name":"Peptide Shaker","outputs":[{"name":"mzidentML","type":"mzid"},{"name":"output_certificate","type":"txt"},{"name":"output_psm","type":"tabular"},{"name":"output_peptides","type":"tabular"},{"name":"output_proteins","type":"tabular"}],"position":{"left":1376.546875,"top":136.69921875},"post_job_actions":{"TagDatasetActionoutput_peptides":{"action_arguments":{"tags":"#PS_Peptides"},"action_type":"TagDatasetAction","output_name":"output_peptides"},"TagDatasetActionoutput_psm":{"action_arguments":{"tags":"#PS_PSM"},"action_type":"TagDatasetAction","output_name":"output_psm"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/2.0.33+galaxy1","tool_shed_repository":{"changeset_revision":"a6ff76e057fd","name":"peptideshaker","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exporting_options\\": {\\"zip_conditional\\": {\\"zip_output_boolean\\": false, \\"__current_case__\\": 0, \\"export_psdb\\": false}, \\"mzidentml_conditional\\": {\\"mzidentml_creation\\": true, \\"__current_case__\\": 0, \\"include_sequences\\": false, \\"contact_options\\": {\\"contact_options_selector\\": \\"no\\", \\"__current_case__\\": 0}}, \\"followup_conditional\\": {\\"followup_conditional_selector\\": \\"no\\", \\"__current_case__\\": 0}, \\"output_reports\\": [\\"3\\", \\"6\\", \\"9\\", \\"0\\"]}, \\"optional_main_parameters\\": {\\"input_optional_identification_parameters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_fasta_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_optional_spectrum_files\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"searchgui_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.33+galaxy1","type":"tool","uuid":"d8293305-e003-4c82-9085-a6a6e27563f3","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.0","errors":null,"id":14,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":"Filtering Accesions","name":"Filter Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1909.4926854592713,"top":567.042966564121},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Filtering Accessions"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/filter_tabular/filter_tabular/3.3.0","tool_shed_repository":{"changeset_revision":"557ec8d7087d","name":"filter_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment_char\\": true, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"select_columns\\", \\"__current_case__\\": 9, \\"columns\\": \\"1\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"replace\\", \\"__current_case__\\": 11, \\"column\\": \\"1\\", \\"regex_pattern\\": \\"^[^|]+[|]([^| ]+).*$\\", \\"regex_replace\\": \\"\\\\\\\\1\\", \\"add\\": null}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.0","type":"tool","uuid":"e42dd88c-5e82-400e-810f-cb34746ce438","when":null,"workflow_outputs":[{"label":"Filtering Accessions","output_name":"output","uuid":"94f1a6b8-0f60-4ffc-ab90-31ba68d37cdc"}]},"15":{"annotation":"","content_id":"Cut1","errors":null,"id":15,"input_connections":{"input":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":"MQ Peptide Sequences","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1902.578125,"top":824.57421875},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"MQ Peptide Sequences"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"d182a597-2b4f-4f60-bd76-e6cea6f9f610","when":null,"workflow_outputs":[{"label":"MQ Peptide Sequences","output_name":"out_file1","uuid":"26146545-0adf-43cd-a407-3981a64e2b8a"}]},"16":{"annotation":"","content_id":"Grep1","errors":null,"id":16,"input_connections":{"input":{"id":13,"output_name":"output_peptides"}},"inputs":[],"label":"Select microbial peptides SGPS","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1624.6171875,"top":133.078125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Select microbial peptides SGPS"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": true, \\"pattern\\": \\"(_HUMAN)|(_REVERSED)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"088c67e2-5680-45ac-8fc6-58107304378b","when":null,"workflow_outputs":[{"label":"Select microbial peptides SGPS","output_name":"out_file1","uuid":"98fa521e-5f10-41f8-bbb0-0c37e36c7db8"}]},"17":{"annotation":"","content_id":"Grep1","errors":null,"id":17,"input_connections":{"input":{"id":13,"output_name":"output_psm"}},"inputs":[],"label":"Select microbial PSMs SGPS","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1633.46875,"top":337.29296875},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Select microbial PSMs SGPS"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": true, \\"pattern\\": \\"(_HUMAN)|(_REVERSED)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"77835fcd-c645-48fa-bfe2-2519cebdf93c","when":null,"workflow_outputs":[{"label":"Select microbial PSMs SGPS","output_name":"out_file1","uuid":"2d8e7875-144c-4cb8-8278-37156df85c6e"}]},"18":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":18,"input_connections":{"input":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":"Remove beginning from MQ Peptides","name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2433.9375,"top":727.6328125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Header removed MQ Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3eb21bbd-483c-4685-b3e8-1feec0354c47","when":null,"workflow_outputs":[{"label":"Header removed MQ Peptides","output_name":"out_file1","uuid":"a31d9c7f-5c8a-462a-a114-f2d9201d7f78"}]},"19":{"annotation":"","content_id":"Filter1","errors":null,"id":19,"input_connections":{"input":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":"Filter confident microbial Peptides","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1882.4063928456353,"top":138.52877496551775},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Filter confident microbial Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"c17==\'Confident\'\\", \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"a2763f6c-4bb4-4637-b9a5-4319b8176e57","when":null,"workflow_outputs":[{"label":"Filter confident microbial Peptides","output_name":"out_file1","uuid":"bd070980-580f-4e1c-ba87-0e7bf097c4b9"}]},"20":{"annotation":"","content_id":"Filter1","errors":null,"id":20,"input_connections":{"input":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":"Filter confident microbial PSMs","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1892.8281925994486,"top":347.30488486357063},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Filter confident microbial PSMs"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Filter1","tool_state":"{\\"cond\\": \\"c24==\'Confident\'\\", \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"2986ffd8-f797-4bdd-ba33-e3383bb3cbd1","when":null,"workflow_outputs":[{"label":"Filter confident microbial PSMs","output_name":"out_file1","uuid":"902234ac-5870-40fb-8b9b-2baf6e7a5786"}]},"21":{"annotation":"","content_id":"Grouping1","errors":null,"id":21,"input_connections":{"input1":{"id":18,"output_name":"out_file1"}},"inputs":[],"label":"MQ Distinct Peptides","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2692.1307728764114,"top":524.4947117996797},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"MQ Distinct Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"eadd5e67-4653-4895-bd99-d7ec8e8514f7","when":null,"workflow_outputs":[{"label":"MQ Distinct Peptides","output_name":"out_file1","uuid":"635bd0b8-42da-4605-a7b5-c914f8628917"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.0","errors":null,"id":22,"input_connections":{"tables_0|table":{"id":20,"output_name":"out_file1"},"tables_1|table":{"id":20,"output_name":"out_file1"},"tables_2|table":{"id":14,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Query Tabular","name":"add_to_database"}],"label":"Extracting peptides SGPS from PSM","name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2171.984360939529,"top":230.6947040667832},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Extracting peptides SGPS from PSM"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.0","tool_shed_repository":{"changeset_revision":"cf34c344508d","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"add_to_database\\": {\\"withdb\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"addqueries\\": {\\"queries\\": []}, \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"yes\\", \\"__current_case__\\": 0, \\"header_prefix\\": null}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT id,Proteins,Sequence\\\\nFROM psms \\\\nWHERE psms.ln NOT IN\\\\n(SELECT distinct prots.ln\\\\nFROM prots JOIN Uniprot ON prots.prot = Uniprot.prot)\\\\nORDER BY psms.ln\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"skip\\", \\"__current_case__\\": 0, \\"skip_lines\\": \\"1\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_line_num\\", \\"__current_case__\\": 2}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"psms\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"ln,id,Proteins,Sequence\\", \\"load_named_columns\\": true, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": [{\\"__index__\\": 0, \\"unique\\": false, \\"index_columns\\": \\"ln\\"}]}}, {\\"__index__\\": 1, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"skip\\", \\"__current_case__\\": 0, \\"skip_lines\\": \\"1\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"prepend_line_num\\", \\"__current_case__\\": 2}}, {\\"__index__\\": 2, \\"filter\\": {\\"filter_type\\": \\"normalize\\", \\"__current_case__\\": 12, \\"columns\\": \\"3\\", \\"separator\\": \\",\\"}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"prots\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"ln,id,prot\\", \\"load_named_columns\\": true, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": [{\\"__index__\\": 0, \\"unique\\": true, \\"index_columns\\": \\"prot,ln\\"}]}}, {\\"__index__\\": 2, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"Uniprot\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"prot\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": [{\\"__index__\\": 0, \\"unique\\": false, \\"index_columns\\": \\"prot\\"}]}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.0","type":"tool","uuid":"6727a48b-5576-4fd9-a8d6-b998c2f25e65","when":null,"workflow_outputs":[{"label":"Extracting peptides SGPS from PSM","output_name":"output","uuid":"5ea51c53-b7f1-447c-8b15-6ae3b89c2246"}]},"23":{"annotation":"","content_id":"Cut1","errors":null,"id":23,"input_connections":{"input":{"id":22,"output_name":"output"}},"inputs":[],"label":"Cut Peptide column","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2429,"top":320.03125},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"SGPS Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"079cee20-8cf1-417e-9ede-9e8d3e4157e1","when":null,"workflow_outputs":[{"label":"SGPS Peptides","output_name":"out_file1","uuid":"6d45c71b-64ef-4786-a89d-ee43c27ee969"}]},"24":{"annotation":"","content_id":"Grouping1","errors":null,"id":24,"input_connections":{"input1":{"id":23,"output_name":"out_file1"}},"inputs":[],"label":"SGPS Distinct Peptides","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2692.21992119554,"top":329.70161173161625},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"SGPS Distinct Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"17433c88-5d56-453f-9371-dc2943eecac2","when":null,"workflow_outputs":[{"label":"SGPS Distinct Peptides","output_name":"out_file1","uuid":"1eea0b81-8c35-4351-bbf3-2941561e4fcf"}]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","errors":null,"id":25,"input_connections":{"inputs":[{"id":21,"output_name":"out_file1"},{"id":24,"output_name":"out_file1"}]},"inputs":[],"label":"SGPS MQ Peptides","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":3005.427930624518,"top":411.55050647320707},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"SGPS MQ Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.1","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"967a29c5-8ef9-45de-918a-f99e3f9a058f","when":null,"workflow_outputs":[{"label":"SGPS MQ Peptides","output_name":"out_file1","uuid":"df2db225-d108-4337-a684-dbca973da64e"}]},"26":{"annotation":"","content_id":"Grouping1","errors":null,"id":26,"input_connections":{"input1":{"id":25,"output_name":"out_file1"}},"inputs":[],"label":"Distinct Peptides","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":3283.015625,"top":419.21875},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Distinct Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"b5462dd0-3af9-41d4-a011-65d4244525a9","when":null,"workflow_outputs":[{"label":"Distinct Peptides","output_name":"out_file1","uuid":"c283b145-a685-49c4-8787-6e2e70f262b4"}]}},"tags":["name:clinicalMP"],"uuid":"94138387-134e-4622-96a9-fdaee428ee1b","version":2},"readme":"# Clinical Metaproteomics 2: Discovery\\n\\nDiscovery in clinical metaproteomics is greatly enhanced by using a well-curated database, particularly one generated with the **MetaNovo tool**. This tool creates a manageable and streamlined database by identifying proteins relevant to the dataset, reducing the complexity of downstream analysis. For optimal results, the MetaNovo-generated database can be merged with reviewed proteins from **Human SwissProt** and known contaminants from the **cRAP (common Repository of Adventitious Proteins)** database, resulting in a compact yet comprehensive database of approximately 21,200 protein sequences. This refined database serves as the foundation for peptide identification, where mass spectrometry (MS) data is matched against the database to identify relevant peptides efficiently and accurately. By reducing redundancy and focusing on clinically relevant sequences, this approach improves the discovery of biomarkers and key protein insights, allowing researchers to extract meaningful biological information with reduced noise and false positives. This streamlined process is particularly valuable in clinical studies, where precision and relevance are critical for advancing diagnostics and therapeutic research.\\n\\nIn this current workflow, we perform Discovery using the SearchGUI and MaxQuant tools. A GTN has been developed for this workflow.\\n[https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html](https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html)\\n\\n## Inputs dataset\\n\\n- `MSMS datasets` in RAW dataset collection\\n- `Databases for discovery` in Fasta (protein sequences for database searching)\\n- `Experimental-Design Discovery MaxQuant` in Tabular Format \\n\\n## Inputs values\\n\\nFor MaxQuant and SearchGUI/PeptideShaker \\n- Peptide Length\\n- Variable modifications\\n- Labeled element\\n\\n\\n## Processing\\n\\n- extract microbial proteins and peptides using text formating tools\\n- Grouping duplicates using the Group tool\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-11-18\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Clinical Metaproteomics Discovery Workflow\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Human UniProt Microbial Proteins from MetaNovo and cRAP\\"]@{ shape: doc }\\n1[\\"IdentificationParameters:PAR file\\"]@{ shape: process }\\n2[\\"ℹ️ Tandem Mass Spectrometry MSMS files\\"]@{ shape: docs }\\n3[\\"Human SwissProt\\"]@{ shape: process }\\n4[\\"Contaminants\\"]@{ shape: process }\\n5[\\"ℹ️ Experimental Design Discovery MaxQuant\\"]@{ shape: doc }\\n6[\\"FastaCLI_ MetaNovo Human SwissProt cRAP with decoys\\"]@{ shape: process }\\n0 --> 6\\n7[\\"msconvert\\"]@{ shape: process }\\n2 --> 7\\n8[\\"Human SwissProt+cRAP\\"]@{ shape: process }\\n3 --> 8\\n4 --> 8\\n9[\\"MaxQuant\\"]@{ shape: process }\\n0 --> 9\\n2 --> 9\\n5 --> 9\\n10[\\"Search GUI\\"]@{ shape: process }\\n6 --> 10\\n1 --> 10\\n7 --> 10\\n11[\\"FASTA to Tabular\\"]@{ shape: process }\\n8 --> 11\\n12[\\"Select microbial peptides MQ\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Peptide Shaker\\"]@{ shape: process }\\n10 --> 13\\n14[\\"Filtering Accesions\\"]@{ shape: process }\\n11 --> 14\\n15[\\"MQ Peptide Sequences\\"]@{ shape: process }\\n12 --> 15\\n16[\\"Select microbial peptides SGPS\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Select microbial PSMs SGPS\\"]@{ shape: process }\\n13 --> 17\\n18[\\"Remove beginning from MQ Peptides\\"]@{ shape: process }\\n15 --> 18\\n19[\\"Filter confident microbial Peptides\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Filter confident microbial PSMs\\"]@{ shape: process }\\n17 --> 20\\n21[\\"MQ Distinct Peptides\\"]@{ shape: process }\\n18 --> 21\\n22[\\"Extracting peptides SGPS from PSM\\"]@{ shape: process }\\n20 --> 22\\n20 --> 22\\n14 --> 22\\n23[\\"Cut Peptide column\\"]@{ shape: process }\\n22 --> 23\\n24[\\"SGPS Distinct Peptides\\"]@{ shape: process }\\n23 --> 24\\n25[\\"SGPS MQ Peptides\\"]@{ shape: process }\\n21 --> 25\\n24 --> 25\\n26[\\"Distinct Peptides\\"]@{ shape: process }\\n25 --> 26\\n```\\n","updated":"2024-11-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/clinicalmp-discovery/main","dockstore_id":29726,"categories":[],"collections":["Proteomics","Metaproteomics"],"doi":"10.5281/zenodo.15078834","tests":[{"doc":"Test outline for iwc-clinicalmp-discovery-workflow","job":{"Human UniProt Microbial Proteins from MetaNovo and cRAP":{"class":"File","location":"https://zenodo.org/records/10720030/files/Human_UniProt_Microbial_Proteins_(from_MetaNovo)_and_cRAP.fasta","filetype":"fasta"},"Experimental Design Discovery MaxQuant":{"class":"File","path":"test-data/Experimental Design Discovery MaxQuant.tabular","filetype":"tabular"},"Tandem Mass Spectrometry MSMS files":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.raw","location":"https://zenodo.org/records/14182981/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.raw"}]}},"outputs":{"SGPS MQ Peptides":{"asserts":[{"has_n_columns":{"n":1}},{"has_text":{"text":"AAFPNVTAMNITTNNGK"}}]}}}]}],"path":"./workflows/proteomics/clinicalmp/clinicalmp-discovery"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/iwc-clinicalmp-database-generation.ga","testParameterFiles":["/iwc-clinicalmp-database-generation-tests.yml"],"authors":[{"name":"Subina Mehta","orcid":"0000-0001-9818-0537"}],"definition":{"a_galaxy_workflow":"true","annotation":"The workflow begins with the Database Generation process. The Galaxy-P team has developed a workflow that collects protein sequences from known disease-causing microorganisms to build a comprehensive database. This extensive database is then refined into a smaller, more relevant dataset using the Metanovo tool.","comments":[],"creator":[{"class":"Organization","identifier":"0000-0001-9818-0537","name":"Subina Mehta"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Generate a Clinical Metaproteomics Database","release":"0.1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Human SwissProt Protein Database","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Human SwissProt Protein Database","name":"Human SwissProt Protein Database"}],"label":"Human SwissProt Protein Database","name":"Input dataset","outputs":[],"position":{"left":0,"top":207.8446261703461},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"bb037df4-ef56-42ad-8b3e-93425bc7bdfa","when":null,"workflow_outputs":[]},"1":{"annotation":"Tandem Mass Spectrometry (MS/MS) datasets","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Tandem Mass Spectrometry (MS/MS) datasets","name":"Tandem Mass Spectrometry (MS/MS) datasets"}],"label":"Tandem Mass Spectrometry (MS/MS) datasets","name":"Input dataset collection","outputs":[],"position":{"left":260.96875,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"d6bb6e3e-ebf6-4c65-bcf7-ea9527cf8653","when":null,"workflow_outputs":[]},"2":{"annotation":"Species UniProt Protein Database","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Species UniProt Protein Database","name":"Species UniProt Protein Database"}],"label":"Species UniProt Protein Database","name":"Input dataset","outputs":[],"position":{"left":2.4375,"top":373.22882080078125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"b4b90dcd-9228-4aad-b51d-2f08b2d4cc0d","when":null,"workflow_outputs":[]},"3":{"annotation":"Contaminants cRAP Protein Database","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Contaminants cRAP Protein Database","name":"Contaminants cRAP Protein Database"}],"label":"Contaminants cRAP Protein Database","name":"Input dataset","outputs":[],"position":{"left":14.40625,"top":471.45147705078125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ca7de659-6e60-4bd5-aa4c-a88f2860e8c9","when":null,"workflow_outputs":[]},"4":{"annotation":"Merging FASTA files","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":4,"input_connections":{"batchmode|input_fastas_0|input_fasta":{"id":0,"output_name":"output"},"batchmode|input_fastas_1|input_fasta":{"id":2,"output_name":"output"},"batchmode|input_fastas_2|input_fasta":{"id":3,"output_name":"output"}},"inputs":[],"label":"Human UniProt Microbial Proteins cRAP for MetaNovo","name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":369.32421875,"top":137.68359736327898},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Human UniProt Microbial Proteins cRAP for MetaNovo"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"individual\\", \\"__current_case__\\": 0, \\"input_fastas\\": [{\\"__index__\\": 0, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"4c1b21ac-9ceb-49be-9c48-771a3fda8db4","when":null,"workflow_outputs":[{"label":"Human UniProt Microbial Proteins cRAP for MetaNovo","output_name":"output","uuid":"fbc778a5-f931-4b0c-b641-7782abfb4351"}]},"5":{"annotation":"Generating Customized Database","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/metanovo/metanovo/1.9.4+galaxy4","errors":null,"id":5,"input_connections":{"input_fasta":{"id":4,"output_name":"output"},"input_type|input_mgf_collection":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MetaNovo","name":"input_type"}],"label":"Metanovo","name":"MetaNovo","outputs":[{"name":"output_fasta","type":"fasta"},{"name":"output_csv","type":"csv"}],"position":{"left":651.57421875,"top":90.94140986327898},"post_job_actions":{"RenameDatasetActionoutput_csv":{"action_arguments":{"newname":"Metanovo Compact CSV database"},"action_type":"RenameDatasetAction","output_name":"output_csv"},"RenameDatasetActionoutput_fasta":{"action_arguments":{"newname":"Metanovo Compact database"},"action_type":"RenameDatasetAction","output_name":"output_fasta"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/metanovo/metanovo/1.9.4+galaxy4","tool_shed_repository":{"changeset_revision":"d6dcd3173bdf","name":"metanovo","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"directag\\": {\\"directag_tic_cutoff\\": \\"85\\", \\"directag_max_peak_count\\": \\"400\\", \\"directag_intensity_classes\\": \\"3\\", \\"directag_adjust_precursor\\": false, \\"directag_min_adjustment\\": \\"-2.5\\", \\"directag_max_adjustment\\": \\"2.5\\", \\"directag_adjustment_step\\": \\"0.1\\", \\"directag_charge_states\\": \\"3\\", \\"directag_ms_charge_state\\": false, \\"directag_duplicate_spectra\\": true, \\"directag_deisotoping\\": \\"0\\", \\"directag_isotope_tolerance\\": \\"0.25\\", \\"directag_complement_tolerance\\": \\"0.5\\", \\"directag_tag_length\\": \\"4\\", \\"directag_max_var_mods\\": \\"2\\", \\"directag_max_tag_count\\": \\"5\\", \\"directag_intensity_weight\\": \\"1.0\\", \\"directag_fidelity_weight\\": \\"1.0\\", \\"directag_complement_weight\\": \\"1.0\\"}, \\"fraction_analysis\\": {\\"protein_fraction_mw_confidence\\": \\"95.0\\"}, \\"gene_annotation\\": {\\"useGeneMapping\\": true, \\"updateGeneMapping\\": true}, \\"import_filters\\": {\\"import_peptide_length_min\\": \\"8\\", \\"import_peptide_length_max\\": \\"50\\", \\"import_precursor_mz_ppm\\": \\"0\\", \\"exclude_unknown_ptms\\": true}, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_type\\": {\\"type\\": \\"collection\\", \\"__current_case__\\": 1, \\"input_mgf_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"metanovo_parameters\\": {\\"mn_specificity\\": \\"specific\\", \\"mn_enzymes\\": \\"Trypsin, no P rule\\", \\"mn_max_missed_cleavages\\": \\"2\\"}, \\"processing_control\\": {\\"CHUNKSIZE\\": \\"100000\\"}, \\"protein_inference\\": {\\"simplify_groups\\": true, \\"simplify_score\\": true, \\"simplify_enzymaticity\\": true, \\"simplify_evidence\\": true, \\"simplify_uncharacterized\\": true}, \\"ptm_localization\\": {\\"ptm_score\\": \\"1\\", \\"score_neutral_losses\\": false, \\"ptm_sequence_matching_type\\": \\"1\\", \\"ptm_alignment\\": true}, \\"sequence_matching\\": {\\"sequence_index_type\\": \\"0\\", \\"sequence_matching_type\\": \\"2\\", \\"sequence_matching_x\\": \\"0.25\\"}, \\"spectrum_annotation\\": {\\"annotation_level\\": \\"0.75\\", \\"annotation_high_resolution\\": true}, \\"spectrum_matching_parameters\\": {\\"prec_tol\\": \\"10.0\\", \\"prec_ppm\\": \\"1\\", \\"frag_tol\\": \\"0.01\\", \\"frag_ppm\\": \\"0\\", \\"digestion\\": \\"0\\", \\"enzyme\\": [\\"Trypsin (no P rule)\\"], \\"specificity\\": \\"0\\", \\"mc\\": \\"2\\", \\"fixed_mods\\": [\\"Carbamidomethylation of C\\", \\"TMT 10-plex of K\\", \\"TMT 10-plex of peptide N-term\\"], \\"variable_mods\\": [\\"Oxidation of M\\"], \\"min_charge\\": \\"2\\", \\"max_charge\\": \\"5\\", \\"fi\\": \\"b\\", \\"ri\\": \\"y\\", \\"min_isotope\\": \\"0\\", \\"max_isotope\\": \\"1\\"}, \\"validation_levels\\": {\\"psm_fdr\\": \\"1\\", \\"peptide_fdr\\": \\"1\\", \\"protein_fdr\\": \\"1\\", \\"group_psms\\": true, \\"group_peptides\\": true, \\"merge_subgroups\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.9.4+galaxy4","type":"tool","uuid":"07d51088-eb68-4ffd-b505-7228a21de93c","when":null,"workflow_outputs":[{"label":"Metanovo Compact database","output_name":"output_fasta","uuid":"02940034-f0fe-4151-acd0-c6982e77b717"},{"label":"Metanovo Compact CSV database","output_name":"output_csv","uuid":"d7208cb6-a907-4a8b-b8b7-cf44c103d14a"}]},"6":{"annotation":"Merge all FASTA","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":6,"input_connections":{"batchmode|input_fastas_0|input_fasta":{"id":0,"output_name":"output"},"batchmode|input_fastas_1|input_fasta":{"id":5,"output_name":"output_fasta"},"batchmode|input_fastas_2|input_fasta":{"id":3,"output_name":"output"}},"inputs":[],"label":"Merge all FASTA","name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":924.9069441945012,"top":322.62533209947645},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Human UniProt Microbial Proteins from MetaNovo cRAP"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"fasta\\", \\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"individual\\", \\"__current_case__\\": 0, \\"input_fastas\\": [{\\"__index__\\": 0, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"3880c712-4f3c-4edc-bda0-ee348ab1a2c4","when":null,"workflow_outputs":[{"label":"Human UniProt Microbial Proteins from MetaNovo cRAP","output_name":"output","uuid":"e3f490ef-1307-4892-bab1-6085cf8194ae"}]}},"tags":["name:clinicalMP"],"uuid":"758f86d1-b81e-438d-9666-bc7dc31b4627","version":10},"readme":"# Clinical Metaproteomics 1: Database Generation\\nMetaproteomics involves the large-scale identification and analysis of all proteins expressed by microbiota. However, analyzing clinical samples using metaproteomics is complicated by the presence of abundant human (host) proteins, which can obscure the detection of less abundant microbial proteins.\\n\\nTo overcome this challenge, we developed a metaproteomics workflow using tandem mass spectrometry (MS/MS) and bioinformatics tools on the Galaxy platform. This workflow enables the characterization of metaproteomes in clinical samples.\\n\\nThe first step in this workflow is the Database Generation process. The Galaxy-P team has created a workflow that compiles a large database by downloading protein sequences of known disease-causing microorganisms. From this extensive database, a compact, relevant database is then created using the Metanovo tool.\\nA GTN has been developed for this workflow. [https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html](https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html)\\n\\n## Inputs dataset\\n\\n### Search Databases (FASTA) from [Zenodo](https://zenodo.org/records/14181725)\\n- `HUMAN SwissProt Protein_Database.fasta`\\n- `Species UniProt Protein Database FASTA.fasta`\\n- `Contaminants (cRAP) Protein Database.fasta`\\n\\n### MSMS files download from [Zenodo](https://zenodo.org/records/14181725)\\n- `PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf`\\n- `PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf`\\n- `PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf`\\n- `PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf`\\n\\n## Input Values\\nFor Metanovo \\n- Peptide Length\\n- Variable modifications\\n- Labeled element\\n\\n## Processing\\n- Merge all the resultant FASTA files\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-11-18\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Generate a Clinical Metaproteomics Database\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Human SwissProt Protein Database\\"]@{ shape: doc }\\n1[\\"ℹ️ Tandem Mass Spectrometry (MS/MS) datasets\\"]@{ shape: docs }\\n2[\\"ℹ️ Species UniProt Protein Database\\"]@{ shape: doc }\\n3[\\"ℹ️ Contaminants cRAP Protein Database\\"]@{ shape: doc }\\n4[\\"Human UniProt Microbial Proteins cRAP for MetaNovo\\"]@{ shape: process }\\n0 --> 4\\n2 --> 4\\n3 --> 4\\n5[\\"Metanovo\\"]@{ shape: process }\\n4 --> 5\\n1 --> 5\\n6[\\"Merge all FASTA\\"]@{ shape: process }\\n0 --> 6\\n5 --> 6\\n3 --> 6\\n```\\n","updated":"2024-11-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/clinicalmp-database-generation/main","dockstore_id":29491,"categories":[],"collections":["Proteomics","Metaproteomics"],"doi":"10.5281/zenodo.15078858","tests":[{"doc":"Test outline for iwc-clinicalmp-database-generation","job":{"Human SwissProt Protein Database":{"class":"File","location":"https://zenodo.org/records/14181725/files/HUMAN-SwissProt-Protein-Database.fasta?download=1","filetype":"fasta"},"Species UniProt Protein Database":{"class":"File","location":"https://zenodo.org/records/14181725/files/Species_UniProt_FASTA.fasta?download=1","filetype":"fasta"},"Contaminants cRAP Protein Database":{"class":"File","location":"https://zenodo.org/records/14181725/files/Contaminants(cRAP)-Protein-Database.fasta?download=1","filetype":"fasta"},"Tandem Mass Spectrometry (MS/MS) datasets":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf?download=1"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf?download=1"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf?download=1"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf?download=1"}]}},"outputs":{"Human UniProt Microbial Proteins cRAP for MetaNovo":{"asserts":[{"that":"has_text","text":">sp|"}]},"Metanovo Compact database":{"asserts":[{"that":"has_text","text":">sp|"}]},"Metanovo Compact CSV database":{"asserts":[{"that":"has_text","text":"index"}]},"Human UniProt Microbial Proteins from MetaNovo cRAP":{"asserts":[{"that":"has_text","text":">sp|"}]}}}]}],"path":"./workflows/proteomics/clinicalmp/clinicalmp-database-generation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/clinicalmp-verification.ga","testParameterFiles":["/clinicalmp-verification-tests.yml"],"authors":[{"name":"Pratik Jagtap","orcid":"0000-0003-0984-0973"}],"definition":{"a_galaxy_workflow":"true","annotation":"In proteomics research, verifying detected peptides is essential for ensuring data accuracy and biological relevance. This tutorial continues from the clinical metaproteomics discovery workflow, focusing on verifying identified microbial peptides using the PepQuery tool.","comments":[],"creator":[{"class":"Organization","identifier":"0000-0003-0984-0973","name":"Pratik Jagtap"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Clinical Metaproteomics Verification Workflow","release":"0.2","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","errors":null,"id":0,"input_connections":{},"inputs":[],"label":"Human UniProt Isoforms FASTA","name":"Protein Database Downloader","outputs":[{"name":"output_database","type":"fasta"}],"position":{"left":0,"top":0},"post_job_actions":{"RenameDatasetActionoutput_database":{"action_arguments":{"newname":"Human UniProt+Isoforms FASTA"},"action_type":"RenameDatasetAction","output_name":"output_database"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","tool_shed_repository":{"changeset_revision":"983bf725dfc2","name":"dbbuilder","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"data\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"source\\": {\\"from\\": \\"uniprot\\", \\"__current_case__\\": 0, \\"taxon\\": \\"9606\\", \\"taxon_id\\": null, \\"reviewed\\": \\"\\", \\"set\\": \\"keyword:KW-1185\\", \\"include_isoform\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.4","type":"tool","uuid":"5fd79188-e100-4341-bd0f-866038115166","when":null,"workflow_outputs":[{"label":"Human UniProt+Isoforms FASTA","output_name":"output_database","uuid":"782c5789-e1ac-4dcf-8f97-05325930ea0d"}]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","errors":null,"id":1,"input_connections":{},"inputs":[],"label":"cRAP","name":"Protein Database Downloader","outputs":[{"name":"output_database","type":"fasta"}],"position":{"left":20.775440205938253,"top":196.89138890513698},"post_job_actions":{"RenameDatasetActionoutput_database":{"action_arguments":{"newname":"cRAP"},"action_type":"RenameDatasetAction","output_name":"output_database"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4","tool_shed_repository":{"changeset_revision":"983bf725dfc2","name":"dbbuilder","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"data\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"source\\": {\\"from\\": \\"cRAP\\", \\"__current_case__\\": 1}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.3.4","type":"tool","uuid":"27b84e7a-ad80-4852-82b5-3f35edeeff53","when":null,"workflow_outputs":[{"label":"cRAP","output_name":"output_database","uuid":"2ba03427-1c51-4293-b05e-e0265f3cc813"}]},"2":{"annotation":"The MGF files from mass spectrometry data.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"The MGF files from mass spectrometry data.","name":"Tandem Mass Spectrometry (MS/MS) datasets"}],"label":"Tandem Mass Spectrometry (MS/MS) datasets","name":"Input dataset collection","outputs":[],"position":{"left":26.467106653090468,"top":441.64657624309564},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"651e8fe7-c4dc-48f0-8a8b-cb3b3e26f152","when":null,"workflow_outputs":[]},"3":{"annotation":"The peptide report from SearchGUI/PeptideShaker.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"The peptide report from SearchGUI/PeptideShaker.","name":"SGPS peptide report"}],"label":"SGPS peptide report","name":"Input dataset","outputs":[],"position":{"left":49.219598040314054,"top":775.0615167366277},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"95e78101-a2bf-4730-bc04-275471ea0b36","when":null,"workflow_outputs":[]},"4":{"annotation":"Microbial peptides that will be used for PepQuery2 analysis.","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Microbial peptides that will be used for PepQuery2 analysis.","name":"Distinct Peptides for PepQuery"}],"label":"Distinct Peptides for PepQuery","name":"Input dataset","outputs":[],"position":{"left":974.964669791108,"top":210.36937340005647},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"2211e22a-293b-4bed-8f3d-1ad06f4b5bd6","when":null,"workflow_outputs":[]},"5":{"annotation":"Peptide report from MaxQuant search.","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"Peptide report from MaxQuant search.","name":"MaxQuant peptide report"}],"label":"MaxQuant peptide report","name":"Input dataset","outputs":[],"position":{"left":29.12351833088178,"top":1007.203580787368},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e58dd2a2-d2b7-4d4f-b41e-8f74cafa41ad","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":6,"input_connections":{"batchmode|input_fastas_0|input_fasta":{"id":0,"output_name":"output_database"},"batchmode|input_fastas_1|input_fasta":{"id":1,"output_name":"output_database"}},"inputs":[],"label":"Human UniProt Isoforms cRAP FASTA","name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":497.09213191562515,"top":177.90819833483664},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Human UniProt+Isoforms+cRAP FASTA"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"fasta\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"individual\\", \\"__current_case__\\": 0, \\"input_fastas\\": [{\\"__index__\\": 0, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"a56471d8-efad-4acc-b4b0-f29104b03670","when":null,"workflow_outputs":[{"label":"Human UniProt+Isoforms+cRAP FASTA","output_name":"output","uuid":"da7f5f25-ed5e-4349-a385-34ac99e6acd8"}]},"7":{"annotation":"","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":3,"output_name":"output"}},"inputs":[],"label":"SGPS Cut","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":436.9729921443104,"top":787.605861962579},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c6,c2\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"2bf85070-1a5f-49c8-a8ce-f273720f53dc","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"Cut1","errors":null,"id":8,"input_connections":{"input":{"id":5,"output_name":"output"}},"inputs":[],"label":"MQ Cut","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":448.2414801728456,"top":1099.886206236094},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c1,c35\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"3287732b-c14d-471b-ba5a-0cfd95c9aff6","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/pepquery2/pepquery2/2.0.2+galaxy2","errors":null,"id":9,"input_connections":{"req_inputs|db_type|db_file":{"id":6,"output_name":"output"},"req_inputs|input_type|multiple|input":{"id":4,"output_name":"output"},"req_inputs|ms_dataset|spectrum_files":{"id":2,"output_name":"output"}},"inputs":[],"label":"PepQuery2","name":"PepQuery2","outputs":[{"name":"log_txt","type":"txt"},{"name":"psm_rank_txt","type":"tabular"}],"position":{"left":1200.1863890085322,"top":479.9317148105498},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/pepquery2/pepquery2/2.0.2+galaxy2","tool_shed_repository":{"changeset_revision":"c32806a80862","name":"pepquery2","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"digestion\\": {\\"enzyme\\": \\"1\\", \\"max_missed_cleavages\\": \\"2\\"}, \\"fast\\": false, \\"modifications\\": {\\"fixed_mod\\": [\\"1\\", \\"13\\", \\"14\\"], \\"var_mod\\": [\\"2\\"], \\"max_mods\\": null, \\"unmodified\\": true, \\"aa\\": true}, \\"ms_params\\": {\\"tolerance_params\\": {\\"precursor_tolerance\\": \\"10\\", \\"precursor_unit\\": \\"ppm\\", \\"tolerance\\": \\"0.6\\"}, \\"search\\": {\\"frag_method\\": \\"1\\", \\"scoring_method\\": \\"1\\", \\"extra_score_validation\\": false, \\"min_charge\\": \\"2\\", \\"max_charge\\": \\"6\\", \\"min_peaks\\": null, \\"isotope_error\\": \\"\\", \\"min_score\\": null, \\"min_length\\": null, \\"max_length\\": null, \\"num_random_peptides\\": null}}, \\"outputs_selected\\": \\"psm_rank.txt\\", \\"parameter_set\\": \\"\\", \\"req_inputs\\": {\\"input_type\\": {\\"input_type_selector\\": \\"peptide\\", \\"__current_case__\\": 0, \\"multiple\\": {\\"peptide_input_selector\\": \\"multiple\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"db_type\\": {\\"db_type_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"db_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"ms_dataset\\": {\\"ms_dataset_type\\": \\"history\\", \\"__current_case__\\": 0, \\"spectrum_files\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexType\\": \\"2\\"}, \\"validation\\": {\\"task_type\\": \\"novel\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.2+galaxy2","type":"tool","uuid":"aa965340-309e-4876-bb38-1f65fe045e7c","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":10,"input_connections":{"input":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"SGPS Remove Beginner","name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":801.3730129932684,"top":779.6493516759031},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"9edd4d67-68bb-46e8-a5a1-dc41caa2a798","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":11,"input_connections":{"input":{"id":8,"output_name":"out_file1"}},"inputs":[],"label":"MQ Remove Beginner","name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":851.7817140618107,"top":920.7620488798409},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"dd2e96ad-4713-41de-ac42-7fe39d782f60","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":12,"input_connections":{"input_list":{"id":9,"output_name":"psm_rank_txt"}},"inputs":[],"label":"Collapse Collection","name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1488.346878408912,"top":364.29121937931995},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"0aacd94f-278b-45d1-a674-fbd42e5f7eb0","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1","errors":null,"id":13,"input_connections":{"inputs":[{"id":10,"output_name":"out_file1"},{"id":11,"output_name":"out_file1"}]},"inputs":[],"label":"Concatenate datasets","name":"Concatenate datasets","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1271.4098936773632,"top":1059.406265472264},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"86755160afbf","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"queries\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"b6d41190-2aef-40cc-9764-324a5307aec6","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"Filter1","errors":null,"id":14,"input_connections":{"input":{"id":12,"output_name":"output"}},"inputs":[],"label":"Filter","name":"Filter","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1559.7817684575534,"top":623.6693261138875},"post_job_actions":{},"tool_id":"Filter1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"cond\\": \\"c20==\'Yes\'\\", \\"header_lines\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"e4b875ee-a9ff-4c72-8684-0bac2232045d","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":15,"input_connections":{"input":{"id":14,"output_name":"out_file1"}},"inputs":[],"label":"Filter Remove beginning","name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1779.7816859216491,"top":624.5489443634883},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"68cb09b4-f29a-480b-adc4-f3b9a957180b","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"Cut1","errors":null,"id":16,"input_connections":{"input":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":"FRB Cut","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1999.7817115852936,"top":625.4286563925251},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ce9461a1-41e4-42f8-a4b5-82ab53ba0280","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":17,"input_connections":{"tables_0|table":{"id":16,"output_name":"out_file1"},"tables_1|table":{"id":13,"output_name":"out_file1"}},"inputs":[],"label":"Peptide and Protein from Peptide Reports","name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2219.7817115852936,"top":625.4286563925251},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Peptide and Protein from Peptide Reports"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"add_to_database\\": {\\"withdb\\": null}, \\"addqueries\\": {\\"queries\\": []}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"yes\\", \\"__current_case__\\": 0, \\"header_prefix\\": \\"\\"}, \\"save_db\\": false, \\"sqlquery\\": \\"select pep.mpep, prot.prot\\\\nFROM pep\\\\nINNER JOIN  prot on pep.mpep=prot.pep \\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"pep\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"mpep\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}, {\\"__index__\\": 1, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": []}, \\"tbl_opts\\": {\\"table_name\\": \\"prot\\", \\"column_names_from_first_line\\": false, \\"col_names\\": \\"pep,prot\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"d62c05c8-883c-4b12-b057-79aa1956713e","when":null,"workflow_outputs":[{"label":"Peptide and Protein from Peptide Reports","output_name":"output","uuid":"aaca61f9-da59-4bbd-a2e9-24e3f5abab07"}]},"18":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":18,"input_connections":{"input":{"id":17,"output_name":"output"}},"inputs":[],"label":"PPPR Remove beginning","name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2439.7817115852936,"top":625.4286563925251},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"78d58065-2258-40d6-9b5f-7452a4c087d2","when":null,"workflow_outputs":[]},"19":{"annotation":"","content_id":"Grouping1","errors":null,"id":19,"input_connections":{"input1":{"id":18,"output_name":"out_file1"}},"inputs":[],"label":"Group","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2659.7818389962604,"top":624.5489443634883},"post_job_actions":{},"tool_id":"Grouping1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"cat_uniq\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"3597c830-1bf0-4e8a-92c3-2a8ec5a6e704","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","errors":null,"id":20,"input_connections":{"tables_0|table":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":"Uniprot ID from verified Peptides","name":"Query Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2545.114966014936,"top":878.0953618681222},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Uniprot ID from verified Peptides"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.3.2","tool_shed_repository":{"changeset_revision":"cf4397560712","name":"query_tabular","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"add_to_database\\": {\\"withdb\\": null}, \\"addqueries\\": {\\"queries\\": []}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"modify_database\\": {\\"sql_stmts\\": []}, \\"query_result\\": {\\"header\\": \\"no\\", \\"__current_case__\\": 1}, \\"save_db\\": false, \\"sqlquery\\": \\"SELECT distinct(prot) AS Accession \\\\nfrom t1\\", \\"tables\\": [{\\"__index__\\": 0, \\"table\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_opts\\": {\\"linefilters\\": [{\\"__index__\\": 0, \\"filter\\": {\\"filter_type\\": \\"normalize\\", \\"__current_case__\\": 12, \\"columns\\": \\"2\\", \\"separator\\": \\";\\"}}, {\\"__index__\\": 1, \\"filter\\": {\\"filter_type\\": \\"replace\\", \\"__current_case__\\": 11, \\"column\\": \\"2\\", \\"regex_pattern\\": \\"(tr|sp)[|]\\", \\"regex_replace\\": \\"\\", \\"add\\": null}}, {\\"__index__\\": 2, \\"filter\\": {\\"filter_type\\": \\"replace\\", \\"__current_case__\\": 11, \\"column\\": \\"2\\", \\"regex_pattern\\": \\"[ ]+\\", \\"regex_replace\\": \\"\\", \\"add\\": null}}, {\\"__index__\\": 3, \\"filter\\": {\\"filter_type\\": \\"replace\\", \\"__current_case__\\": 11, \\"column\\": \\"2\\", \\"regex_pattern\\": \\"[|].*$\\", \\"regex_replace\\": \\"\\", \\"add\\": null}}, {\\"__index__\\": 4, \\"filter\\": {\\"filter_type\\": \\"normalize\\", \\"__current_case__\\": 12, \\"columns\\": \\"2\\", \\"separator\\": \\",\\"}}]}, \\"tbl_opts\\": {\\"table_name\\": \\"t1\\", \\"column_names_from_first_line\\": true, \\"col_names\\": \\"pep,prot\\", \\"load_named_columns\\": false, \\"pkey_autoincr\\": \\"\\", \\"indexes\\": []}}], \\"workdb\\": \\"workdb.sqlite\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.3.2","type":"tool","uuid":"8e2944b5-7681-443f-a8fd-b7da28aed0bc","when":null,"workflow_outputs":[{"label":"Uniprot ID from verified Peptides","output_name":"output","uuid":"3f94bf2c-64a1-4064-96a6-906c8495afe7"}]},"21":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.5.0","errors":null,"id":21,"input_connections":{"input_method|id_file":{"id":20,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool UniProt","name":"input_method"}],"label":"UniProt","name":"UniProt","outputs":[{"name":"proteome","type":"uniprotxml"}],"position":{"left":2657.781714061811,"top":1098.0954229032784},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.5.0","tool_shed_repository":{"changeset_revision":"4ddc8da62671","name":"uniprotxml_downloader","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"format\\": \\"fasta\\", \\"format_cond\\": {\\"format\\": \\"fasta\\", \\"__current_case__\\": 0}, \\"input_method\\": {\\"input_choice\\": \\"history\\", \\"__current_case__\\": 2, \\"id_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": \\"1\\", \\"field\\": \\"accession\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.0","type":"tool","uuid":"93ad396e-e41b-4ea8-8d90-c28e0ca1d0c7","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":22,"input_connections":{"batchmode|input_fastas_0|input_fasta":{"id":0,"output_name":"output_database"},"batchmode|input_fastas_1|input_fasta":{"id":1,"output_name":"output_database"},"batchmode|input_fastas_2|input_fasta":{"id":21,"output_name":"proteome"}},"inputs":[],"label":"Quantitation Database for MaxQuant","name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":3070.223999018992,"top":1289.3686346167783},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"Quantitation Database for MaxQuant"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"fasta\\", \\"__workflow_invocation_uuid__\\": \\"08a702aa736d11ee91a5001e67d2ec02\\", \\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"individual\\", \\"__current_case__\\": 0, \\"input_fastas\\": [{\\"__index__\\": 0, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 1, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, {\\"__index__\\": 2, \\"input_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}, \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"07901f80-c71e-431e-9b18-21fb781afa29","when":null,"workflow_outputs":[{"label":"Quantitation Database for MaxQuant","output_name":"output","uuid":"a7ee74bc-8d8b-4f0f-a00c-01197f92ab01"}]}},"tags":["name:clinicalMP"],"uuid":"22302833-4954-4414-8725-d8bf74d6ffd0","version":2},"readme":"# Clinical Metaproteomics 3: Verification\\nThis workflow uses the PepQuery tool to verify peptides discovered with the clinical metaproteomics discovery workflow.\\nThe PepQuery tool outputs verified peptides that can be used to generate a verified protein database that can be used for the clinical metaproteomics quantitation workflow.\\n\\nMore background on this workflow can be found in the [Clinical Metaproteomics 3: Verification tutorial](https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html)\\n\\n## Inputs datasets\\n\\n### Tabular Input Files\\n- SGPS peptide report\\n- MaxQuant Peptide report\\n- Distinct Peptides for PepQuery\\n\\n### Search Databases (FASTA)\\n- Uniprot HUMAN database\\n- cRAP\\n\\n### MSMS files\\nThe tandem MS/MS files can be downloaded via Zenodo. These MS/MS input files are pilot datasets from Papanicolaou test samples from healthy, benign and ovarian cancer patients. \\n\\n## Input Values\\nFor PepQuery:\\n- Search Tolerances\\n- Digestion Enzyme\\n- Peptide Length\\n- Modifications\\n\\n## Processing\\nExtract protein sequences for the verified peptides.\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-12-16\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.4.0` was updated to `toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.5.0`\\n\\n## [0.1] - 2024-11-19\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Clinical Metaproteomics Verification Workflow\\n\\n```mermaid\\ngraph LR\\n0[\\"Human UniProt Isoforms FASTA\\"]@{ shape: process }\\n1[\\"cRAP\\"]@{ shape: process }\\n2[\\"ℹ️ Tandem Mass Spectrometry (MS/MS) datasets\\"]@{ shape: docs }\\n3[\\"ℹ️ SGPS peptide report\\"]@{ shape: doc }\\n4[\\"ℹ️ Distinct Peptides for PepQuery\\"]@{ shape: doc }\\n5[\\"ℹ️ MaxQuant peptide report\\"]@{ shape: doc }\\n6[\\"Human UniProt Isoforms cRAP FASTA\\"]@{ shape: process }\\n0 --> 6\\n1 --> 6\\n7[\\"SGPS Cut\\"]@{ shape: process }\\n3 --> 7\\n8[\\"MQ Cut\\"]@{ shape: process }\\n5 --> 8\\n9[\\"PepQuery2\\"]@{ shape: process }\\n6 --> 9\\n4 --> 9\\n2 --> 9\\n10[\\"SGPS Remove Beginner\\"]@{ shape: process }\\n7 --> 10\\n11[\\"MQ Remove Beginner\\"]@{ shape: process }\\n8 --> 11\\n12[\\"Collapse Collection\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Concatenate datasets\\"]@{ shape: process }\\n10 --> 13\\n11 --> 13\\n14[\\"Filter\\"]@{ shape: process }\\n12 --> 14\\n15[\\"Filter Remove beginning\\"]@{ shape: process }\\n14 --> 15\\n16[\\"FRB Cut\\"]@{ shape: process }\\n15 --> 16\\n17[\\"Peptide and Protein from Peptide Reports\\"]@{ shape: process }\\n16 --> 17\\n13 --> 17\\n18[\\"PPPR Remove beginning\\"]@{ shape: process }\\n17 --> 18\\n19[\\"Group\\"]@{ shape: process }\\n18 --> 19\\n20[\\"Uniprot ID from verified Peptides\\"]@{ shape: process }\\n19 --> 20\\n21[\\"UniProt\\"]@{ shape: process }\\n20 --> 21\\n22[\\"Quantitation Database for MaxQuant\\"]@{ shape: process }\\n0 --> 22\\n1 --> 22\\n21 --> 22\\n```\\n","updated":"2024-12-16T00:00:00","trsID":"#workflow/github.com/iwc-workflows/clinicalmp-verification/main","dockstore_id":29521,"categories":[],"collections":["Proteomics","Metaproteomics"],"doi":"10.5281/zenodo.15078843","tests":[{"doc":"Test outline for Clinical Metaproteomics Verification Workflow","job":{"SGPS peptide report":{"class":"File","path":"test-data/SGPS_peptide-report.tabular","filetype":"tabular"},"Distinct Peptides for PepQuery":{"class":"File","path":"test-data/Distinct_Peptides_for_PepQuery.tabular","filetype":"tabular"},"MaxQuant peptide report":{"class":"File","path":"test-data/MaxQuant-peptide-report.tabular","filetype":"tabular"},"Tandem Mass Spectrometry (MS/MS) datasets":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf","location":"https://zenodo.org/records/14181725/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf"}]}},"outputs":{"Human UniProt+Isoforms FASTA":{"asserts":{"has_text":{"text":">sp"}}},"cRAP":{"path":"test-data/cRAP.fasta"},"Human UniProt+Isoforms+cRAP FASTA":{"asserts":{"has_text":{"text":">sp"}}},"Peptide and Protein from Peptide Reports":{"path":"test-data/Peptide_and_Protein_from_Peptide_Reports.tabular"},"Uniprot ID from verified Peptides":{"path":"test-data/Uniprot-ID_from_verified_Peptides.tabular"},"Quantitation Database for MaxQuant":{"asserts":{"has_text":{"text":">tr"}}}}}]}],"path":"./workflows/proteomics/clinicalmp/clinicalmp-verification"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/WF5_Data_Interpretation_Worklow.ga","testParameterFiles":["/WF5_Data_Interpretation_Worklow_tests.yml"],"authors":[{"name":"GalaxyP","url":"https://galaxyp.org/"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow will perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT.","comments":[],"creator":[{"class":"Organization","name":"GalaxyP"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Clinical Metaproteomics Data Interpretation","release":"0.1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Quantified peptides from MaxQuant","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Quantified peptides from MaxQuant","name":"Quantified Peptides"}],"label":"Quantified Peptides","name":"Input dataset","outputs":[],"position":{"left":189.871337890625,"top":255.26802202427945},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"cd669a99-182a-48db-a2c6-f8d24323e353","when":null,"workflow_outputs":[]},"1":{"annotation":"MaxQuant Protein Groups","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"MaxQuant Protein Groups","name":"MaxQuant Protein Groups"}],"label":"MaxQuant Protein Groups","name":"Input dataset","outputs":[],"position":{"left":0,"top":544.9199843045528},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"2ecd2d61-5e94-4dbd-a0cc-c30c343b38b4","when":null,"workflow_outputs":[]},"2":{"annotation":"Evidence file from MaxQuant","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Evidence file from MaxQuant","name":"MaxQuant Evidence"}],"label":"MaxQuant Evidence","name":"Input dataset","outputs":[],"position":{"left":221.97045898437503,"top":823.9836556138064},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"11afae12-7cc4-402b-a0e3-cf6e53276674","when":null,"workflow_outputs":[]},"3":{"annotation":"Annotation file for MSstatsTMT","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Annotation file for MSstatsTMT","name":"Annotation"}],"label":"Annotation","name":"Input dataset","outputs":[],"position":{"left":293.3940978424075,"top":1107.2505582800675},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"4ed9cd3d-2473-4b69-8ac8-3404e5628f85","when":null,"workflow_outputs":[]},"4":{"annotation":"Comparison matrix for MSstatsTMT","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"Comparison matrix for MSstatsTMT","name":"Comparison Matrix"}],"label":"Comparison Matrix","name":"Input dataset","outputs":[],"position":{"left":307.5031433105469,"top":1229.2167641343142},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"6c47825f-c68c-4f0e-a99b-a961f27c7719","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.5.1","errors":null,"id":5,"input_connections":{"peptide_src|input_tsv":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Unipept","name":"peptide_src"}],"label":"Unipept","name":"Unipept","outputs":[{"name":"output_json","type":"d3_hierarchy"},{"name":"output_ec_json","type":"d3_hierarchy"},{"name":"output_tsv","type":"tabular"},{"name":"output_ec_tsv","type":"tabular"},{"name":"output_go_tsv","type":"tabular"},{"name":"output_ipr_tsv","type":"tabular"}],"position":{"left":603.5423879409451,"top":0},"post_job_actions":{"RenameDatasetActionoutput_ec_json":{"action_arguments":{"newname":"Microbial EC Proteins Tree"},"action_type":"RenameDatasetAction","output_name":"output_ec_json"},"RenameDatasetActionoutput_ec_tsv":{"action_arguments":{"newname":"Microbial EC Terms Output"},"action_type":"RenameDatasetAction","output_name":"output_ec_tsv"},"RenameDatasetActionoutput_go_tsv":{"action_arguments":{"newname":"Microbial GO Terms Output"},"action_type":"RenameDatasetAction","output_name":"output_go_tsv"},"RenameDatasetActionoutput_ipr_tsv":{"action_arguments":{"newname":"Microbial InterPro Outputs"},"action_type":"RenameDatasetAction","output_name":"output_ipr_tsv"},"RenameDatasetActionoutput_json":{"action_arguments":{"newname":"Microbial Taxonomy Tree"},"action_type":"RenameDatasetAction","output_name":"output_json"},"RenameDatasetActionoutput_tsv":{"action_arguments":{"newname":"Unipept peptinfo"},"action_type":"RenameDatasetAction","output_name":"output_tsv"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.5.1","tool_shed_repository":{"changeset_revision":"21a560af5913","name":"unipept","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"peptide_match\\": \\"full\\", \\"peptide_src\\": {\\"fmt\\": \\"tabular\\", \\"__current_case__\\": 1, \\"input_tsv\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"column\\": \\"1\\"}, \\"selected_outputs\\": [\\"tsv\\", \\"json\\", \\"ipr_tsv\\", \\"go_tsv\\", \\"ec_tsv\\", \\"ec_json\\"], \\"strict\\": false, \\"unipept\\": {\\"api\\": \\"peptinfo\\", \\"__current_case__\\": 7, \\"equate_il\\": true, \\"extra\\": true, \\"domains\\": false, \\"names\\": true, \\"allfields\\": false}, \\"unmatched_aa\\": \\"default\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.5.1","type":"tool","uuid":"f0385dbb-0248-4c41-a3be-29915ec0ec2b","when":null,"workflow_outputs":[{"label":"Microbial InterPro Outputs","output_name":"output_ipr_tsv","uuid":"83e9749c-d5b1-4ca6-84ed-a71c871be90b"},{"label":"Microbial Taxonomy Tree","output_name":"output_json","uuid":"513a089a-614b-4557-9e40-30de6481b754"},{"label":"Microbial EC Proteins Tree","output_name":"output_ec_json","uuid":"bed390f7-ec36-4d3e-a84e-b36572aad229"},{"label":"Microbial EC Terms Output","output_name":"output_ec_tsv","uuid":"88bc0672-de66-4060-b84a-25e2f1554f49"},{"label":"Microbial GO Terms Output","output_name":"output_go_tsv","uuid":"03b83b33-3936-46fe-a437-4f0acf3c604f"},{"label":"Unipept peptinfo","output_name":"output_tsv","uuid":"2c471afe-49d6-4aae-87be-3594d7716049"}]},"6":{"annotation":"","content_id":"Grep1","errors":null,"id":6,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":"Microbial Proteins","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":302.63226318359375,"top":456.09629961217007},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Microbial-Proteins"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": true, \\"pattern\\": \\"(HUMAN)|(REV)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"1b00f695-2038-46be-891f-dcc645c6c768","when":null,"workflow_outputs":[{"label":"Microbial Proteins","output_name":"out_file1","uuid":"b23a9caa-826c-4dfc-8739-3a4511e00fad"}]},"7":{"annotation":"","content_id":"Grep1","errors":null,"id":7,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":211.46368408203125,"top":662.4912123318966},"post_job_actions":{},"tool_id":"Grep1","tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": true, \\"pattern\\": \\"(HUMAN)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"b7d94fff-d2f7-4678-90ef-91730e0b7fb6","when":null,"workflow_outputs":[]},"8":{"annotation":"MSstatsTMT analysis for microbial proteins","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msstatstmt/msstatstmt/2.0.0+galaxy1","errors":null,"id":8,"input_connections":{"group|use_comp_matrix|comparison_matrix":{"id":4,"output_name":"output"},"input|annotation":{"id":3,"output_name":"output"},"input|evidence":{"id":2,"output_name":"output"},"input|proteinGroups":{"id":6,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool MSstatsTMT","name":"input"},{"description":"runtime parameter for tool MSstatsTMT","name":"input"},{"description":"runtime parameter for tool MSstatsTMT","name":"input"}],"label":"MSstatsTMT_for_microbial_proteins","name":"MSstatsTMT","outputs":[{"name":"out_msstatstmt_sum_log","type":"txt"},{"name":"out_msstatstmt_sum_ms","type":"txt"},{"name":"out_quant","type":"tabular"},{"name":"out_group_comp","type":"tabular"},{"name":"out_group_volcano_plot","type":"pdf"},{"name":"out_group_comp_plot","type":"pdf"}],"position":{"left":986.6640958581056,"top":433.7097936276964},"post_job_actions":{"RenameDatasetActionout_group_comp_plot":{"action_arguments":{"newname":"MICROBIAL_PROTEINS_COMPARISON"},"action_type":"RenameDatasetAction","output_name":"out_group_comp_plot"},"RenameDatasetActionout_group_volcano_plot":{"action_arguments":{"newname":"MICROBIAL_PROTEINS_VOLCANO_PLOT"},"action_type":"RenameDatasetAction","output_name":"out_group_volcano_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msstatstmt/msstatstmt/2.0.0+galaxy1","tool_shed_repository":{"changeset_revision":"bd1535813cfe","name":"msstatstmt","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"group\\": {\\"group_comparison\\": \\"true\\", \\"__current_case__\\": 1, \\"use_comp_matrix\\": {\\"select\\": \\"true\\", \\"__current_case__\\": 1, \\"comparison_matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"moderated\\": false, \\"adj_method\\": \\"none\\", \\"remove_norm_channel\\": true, \\"remove_empty_channel\\": true, \\"selected_group_outputs\\": [\\"comparison_result\\", \\"VolcanoPlot\\", \\"ComparisonPlot\\"], \\"comparison_plots_opt\\": {\\"sig\\": \\"0.05\\", \\"FCcutoff\\": null, \\"logBase_pvalue\\": \\"10\\", \\"ylimUp\\": null, \\"ylimDown\\": null, \\"xlimUp\\": null, \\"axis_size\\": \\"10\\", \\"x_axis_size\\": \\"10\\", \\"y_axis_size\\": \\"10\\", \\"dot_size\\": \\"3\\", \\"text_size\\": \\"4\\", \\"text_angle\\": \\"90\\", \\"legend_size\\": \\"7\\", \\"ProteinName\\": true, \\"colorkey\\": true, \\"numProtein\\": \\"100\\", \\"clustering\\": \\"protein\\", \\"width\\": \\"8\\", \\"height\\": \\"5\\", \\"which_Protein\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}, \\"which_Comparison\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}}}, \\"input\\": {\\"input_src\\": \\"MaxQuant\\", \\"__current_case__\\": 1, \\"evidence\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"proteinGroups\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"proteinID\\": \\"Proteins\\", \\"input_options\\": {\\"rmProt_Onlyidentifiedbysite\\": false, \\"useUniquePeptide\\": true, \\"rmPSM_withMissing_withinRun\\": false, \\"rmPSM_withfewMea_withinRun\\": true, \\"rmProtein_with1Feature\\": false, \\"summaryforMultipleRows\\": \\"sum\\"}}, \\"out_plots_opt\\": {\\"which_Protein\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}, \\"width\\": \\"10\\", \\"height\\": \\"10\\", \\"adv\\": {\\"ylimUp\\": false, \\"ylimDown\\": false, \\"x_axis_size\\": \\"10\\", \\"y_axis_size\\": \\"10\\", \\"text_size\\": \\"4\\", \\"text_angle\\": \\"90\\", \\"legend_size\\": \\"7\\", \\"dot_size_profile\\": \\"2\\", \\"ncol_guide\\": \\"5\\", \\"originalPlot\\": true, \\"summaryPlot\\": true}}, \\"proteinSummarization\\": {\\"method\\": \\"msstats\\", \\"global_norm\\": true, \\"reference_norm\\": true, \\"remove_norm_channel\\": true, \\"remove_empty_channel\\": true, \\"MBimpute\\": true, \\"maxQuantileforCensored\\": null}, \\"selected_outputs\\": [\\"msstatstmt_sum_log\\", \\"msstatstmt_sum_ms\\", \\"quant\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.0+galaxy1","type":"tool","uuid":"e58242d9-4047-49fa-9cc5-b383c507a70f","when":null,"workflow_outputs":[{"label":"Microbial Proteins Comparison Plot","output_name":"out_group_comp_plot","uuid":"c82b343d-b865-4a66-b4dd-e1f26bd8869c"},{"label":"Microbial Proteins Volcano Plot","output_name":"out_group_volcano_plot","uuid":"e66969ae-d8da-43df-a88c-57f6f284f0b0"}]},"9":{"annotation":"","content_id":"Grep1","errors":null,"id":9,"input_connections":{"input":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Human Proteins","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":484.9521484375,"top":632.1494154568966},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Human_Proteins"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": true, \\"pattern\\": \\"(REV)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"898a2203-5746-465a-aa31-95837eb97a04","when":null,"workflow_outputs":[{"label":"Human Proteins","output_name":"out_file1","uuid":"cf77265d-e114-4d57-8e07-56378da1bc62"}]},"10":{"annotation":"MSstatsTMT analysis for human proteins","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msstatstmt/msstatstmt/2.0.0+galaxy1","errors":null,"id":10,"input_connections":{"group|use_comp_matrix|comparison_matrix":{"id":4,"output_name":"output"},"input|annotation":{"id":3,"output_name":"output"},"input|evidence":{"id":2,"output_name":"output"},"input|proteinGroups":{"id":9,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool MSstatsTMT","name":"input"},{"description":"runtime parameter for tool MSstatsTMT","name":"input"},{"description":"runtime parameter for tool MSstatsTMT","name":"input"}],"label":"MSstatsTMT_for_human_proteins","name":"MSstatsTMT","outputs":[{"name":"out_msstatstmt_sum_log","type":"txt"},{"name":"out_msstatstmt_sum_ms","type":"txt"},{"name":"out_quant","type":"tabular"},{"name":"out_group_comp","type":"tabular"},{"name":"out_group_volcano_plot","type":"pdf"},{"name":"out_group_comp_plot","type":"pdf"}],"position":{"left":955.3109130859375,"top":1163.6116306568422},"post_job_actions":{"RenameDatasetActionout_group_comp_plot":{"action_arguments":{"newname":"HUMAN_PROTEINS_COMPARISON"},"action_type":"RenameDatasetAction","output_name":"out_group_comp_plot"},"RenameDatasetActionout_group_volcano_plot":{"action_arguments":{"newname":"HUMAN_PROTEINS_VOLCANO_PLOT"},"action_type":"RenameDatasetAction","output_name":"out_group_volcano_plot"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/msstatstmt/msstatstmt/2.0.0+galaxy1","tool_shed_repository":{"changeset_revision":"bd1535813cfe","name":"msstatstmt","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"group\\": {\\"group_comparison\\": \\"true\\", \\"__current_case__\\": 1, \\"use_comp_matrix\\": {\\"select\\": \\"true\\", \\"__current_case__\\": 1, \\"comparison_matrix\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"moderated\\": false, \\"adj_method\\": \\"none\\", \\"remove_norm_channel\\": true, \\"remove_empty_channel\\": true, \\"selected_group_outputs\\": [\\"comparison_result\\", \\"VolcanoPlot\\", \\"ComparisonPlot\\"], \\"comparison_plots_opt\\": {\\"sig\\": \\"0.05\\", \\"FCcutoff\\": null, \\"logBase_pvalue\\": \\"10\\", \\"ylimUp\\": null, \\"ylimDown\\": null, \\"xlimUp\\": null, \\"axis_size\\": \\"10\\", \\"x_axis_size\\": \\"10\\", \\"y_axis_size\\": \\"10\\", \\"dot_size\\": \\"3\\", \\"text_size\\": \\"4\\", \\"text_angle\\": \\"90\\", \\"legend_size\\": \\"7\\", \\"ProteinName\\": true, \\"colorkey\\": true, \\"numProtein\\": \\"100\\", \\"clustering\\": \\"protein\\", \\"width\\": \\"8\\", \\"height\\": \\"5\\", \\"which_Protein\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}, \\"which_Comparison\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}}}, \\"input\\": {\\"input_src\\": \\"MaxQuant\\", \\"__current_case__\\": 1, \\"evidence\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"proteinGroups\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"proteinID\\": \\"Proteins\\", \\"input_options\\": {\\"rmProt_Onlyidentifiedbysite\\": false, \\"useUniquePeptide\\": true, \\"rmPSM_withMissing_withinRun\\": false, \\"rmPSM_withfewMea_withinRun\\": true, \\"rmProtein_with1Feature\\": false, \\"summaryforMultipleRows\\": \\"sum\\"}}, \\"out_plots_opt\\": {\\"which_Protein\\": {\\"select\\": \\"all\\", \\"__current_case__\\": 0}, \\"width\\": \\"10\\", \\"height\\": \\"10\\", \\"adv\\": {\\"ylimUp\\": false, \\"ylimDown\\": false, \\"x_axis_size\\": \\"10\\", \\"y_axis_size\\": \\"10\\", \\"text_size\\": \\"4\\", \\"text_angle\\": \\"90\\", \\"legend_size\\": \\"7\\", \\"dot_size_profile\\": \\"2\\", \\"ncol_guide\\": \\"5\\", \\"originalPlot\\": true, \\"summaryPlot\\": true}}, \\"proteinSummarization\\": {\\"method\\": \\"msstats\\", \\"global_norm\\": true, \\"reference_norm\\": true, \\"remove_norm_channel\\": true, \\"remove_empty_channel\\": true, \\"MBimpute\\": true, \\"maxQuantileforCensored\\": null}, \\"selected_outputs\\": [\\"msstatstmt_sum_log\\", \\"msstatstmt_sum_ms\\", \\"quant\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.0+galaxy1","type":"tool","uuid":"c1f844d8-487d-4060-beec-1c283abebd2a","when":null,"workflow_outputs":[{"label":"Human Proteins Volcano Plot","output_name":"out_group_volcano_plot","uuid":"3eb72696-b370-4407-a221-f35a74ab6aff"},{"label":"Human Proteins Comparison Plot","output_name":"out_group_comp_plot","uuid":"88fd3bd3-70ce-486e-8cee-361cb9a61bda"}]}},"tags":["name:clinicalMP"],"uuid":"3a5d60ec-2b30-47cf-a351-7b85df8af9e3","version":4},"readme":"# Clinical Metaproteomics 5: Data Interpretation\\nData interpretation can provide meaningful insights about quantitative proteomic data. Here, MaxQuant data will be interpreted using MSstats by applying a rigorous statistical framework to understand data distribution and variability. Systematic variations are accounted for using subsequent normalization. Users can define the experimental design used by MSstats, such as sample groups and conditions, to perform statistical analysis. The MSstats output provides valuable information about differential protein expression across various conditions, estimates of fold changes, and associated p-values, which aids in the identification of biologically significant proteins. Furthermore, MSstats enables quality control and data visualization, enhancing our ability to draw meaningful conclusions from large-scale proteomic datasets and expands our understanding of complex biological processes.\\n\\nIn this current workflow, we perform taxonomic and functional annotations using Unipept and statistical analysis using MSstatsTMT. A GTN has been developed for this workflow.\\n[https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html](https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html).\\n\\n## Input datasets\\n\\n### Input tabular files\\n- `Annotation.tabular`\\n- `Comparison_Matrix.tabular`\\n- `MaxQuant_Evidence.tabular`\\n- `MaxQuant_Protein_Groups.tabular`\\n- `Quantified-Peptides.tabular`\\n\\n## Input Values\\n\\nFor Unipept:\\n- Unipept application: peptinfo: Tryptic peptides and associated EC and GO terms and lowest common ancestor taxonomy\\n- Match input peptides by: Match to the full input peptide\\n\\nFor MSstatsTMT:\\n- Generate separate plots for microbial and human proteins\\n\\n## Processing\\n- Perform taxonomic and functional annotations for quantified microbial peptides using Unipept.\\n- Select microbial and human protein groups to perform statistical analysis using MSstatsTMT.\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-11-19\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Clinical Metaproteomics Data Interpretation\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Quantified Peptides\\"]@{ shape: doc }\\n1[\\"ℹ️ MaxQuant Protein Groups\\"]@{ shape: doc }\\n2[\\"ℹ️ MaxQuant Evidence\\"]@{ shape: doc }\\n3[\\"ℹ️ Annotation\\"]@{ shape: doc }\\n4[\\"ℹ️ Comparison Matrix\\"]@{ shape: doc }\\n5[\\"Unipept\\"]@{ shape: process }\\n0 --> 5\\n6[\\"Microbial Proteins\\"]@{ shape: process }\\n1 --> 6\\n7[\\"Select\\"]@{ shape: process }\\n1 --> 7\\n8[\\"MSstatsTMT_for_microbial_proteins\\"]@{ shape: process }\\n4 --> 8\\n3 --> 8\\n2 --> 8\\n6 --> 8\\n9[\\"Human Proteins\\"]@{ shape: process }\\n7 --> 9\\n10[\\"MSstatsTMT_for_human_proteins\\"]@{ shape: process }\\n4 --> 10\\n3 --> 10\\n2 --> 10\\n9 --> 10\\n```\\n","updated":"2024-11-19T00:00:00","trsID":"#workflow/github.com/iwc-workflows/clinicalmp-data-interpretation/main","dockstore_id":29519,"categories":[],"collections":["Proteomics","Metaproteomics"],"doi":"10.5281/zenodo.15078879"}],"path":"./workflows/proteomics/clinicalmp/clinicalmp-data-interpretation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/iwc-clinicalmp-quantitation.ga","testParameterFiles":["/iwc-clinicalmp-quantitation-tests.yml"],"authors":[{"name":"GalaxyP","url":"https://galaxyp.org/"}],"definition":{"a_galaxy_workflow":"true","annotation":"Clinical Metaproteomics 4: Quantitation ","comments":[],"creator":[{"class":"Organization","name":"GalaxyP"}],"format-version":"0.1","license":"CC-BY-4.0","name":"Clinical Metaproteomics Quantitation","release":"0.1","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n"},"steps":{"0":{"annotation":"Input protein database","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Input protein database","name":"Quantitation_Database-For-MaxQuant"}],"label":"Quantitation_Database-For-MaxQuant","name":"Input dataset","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c2311b55-b0c7-4237-bed1-4b25bf1a1655","when":null,"workflow_outputs":[]},"1":{"annotation":"design file ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"design file ","name":"Experimental-Design Discovery MaxQuant"}],"label":"Experimental-Design Discovery MaxQuant","name":"Input dataset","outputs":[],"position":{"left":65.796875,"top":168.203125},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"d957f00d-79a9-4e2e-8e15-088444ca5605","when":null,"workflow_outputs":[]},"2":{"annotation":"MSMS data","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"MSMS data","name":"Input Raw-files"}],"label":"Input Raw-files","name":"Input dataset collection","outputs":[],"position":{"left":41.671875,"top":532.3671875},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"2864b493-6d12-4313-ad29-f3b1e2564aba","when":null,"workflow_outputs":[]},"3":{"annotation":"Database searching","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/maxquant/2.0.3.0+galaxy0","errors":null,"id":3,"input_connections":{"input_opts|fasta_files":{"id":0,"output_name":"output"},"paramGroups_0|files":{"id":2,"output_name":"output"},"search_opts|template":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MaxQuant","name":"input_opts"},{"description":"runtime parameter for tool MaxQuant","name":"search_opts"}],"label":"MaxQuant","name":"MaxQuant","outputs":[{"name":"proteinGroups","type":"tabular"},{"name":"mqpar","type":"xml"},{"name":"peptides","type":"tabular"},{"name":"evidence","type":"tabular"},{"name":"msms","type":"tabular"}],"position":{"left":427.9609375,"top":170.17578125},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/maxquant/2.0.3.0+galaxy0","tool_shed_repository":{"changeset_revision":"9c52362ae7bb","name":"maxquant","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_opts\\": {\\"ftype\\": \\".thermo.raw\\", \\"fasta_files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"identifier_parse_rule\\": \\">([^\\\\\\\\s]*)\\", \\"description_parse_rule\\": \\">(.*)\\"}, \\"output_opts\\": {\\"dry_run\\": false, \\"output\\": [\\"proteinGroups\\", \\"mqpar\\", \\"peptides\\", \\"msms\\", \\"evidence\\"]}, \\"paramGroups\\": [{\\"__index__\\": 0, \\"files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"maxMissedCleavages\\": \\"2\\", \\"fixedModifications\\": [\\"Carbamidomethyl (C)\\"], \\"variableModifications\\": [\\"Oxidation (M)\\"], \\"enzymes\\": [\\"Trypsin/P\\"], \\"digestion_mode\\": \\"0\\", \\"quant_method\\": {\\"select_quant_method\\": \\"reporter_ion_ms2\\", \\"__current_case__\\": 3, \\"iso_labels\\": {\\"labeling\\": \\"tmt11plex\\", \\"__current_case__\\": 4}, \\"filter_by_pif\\": {\\"filter_pif\\": \\"False\\", \\"__current_case__\\": 1}}}], \\"protein_quant\\": {\\"peptides_for_quantification\\": \\"1\\", \\"only_unmod_prot\\": {\\"unmod_prot\\": \\"True\\", \\"__current_case__\\": 0, \\"mods_used_prot_quant\\": [\\"Oxidation (M)\\"], \\"discard_unmod_cpart_peptides\\": true}, \\"lfq_opts\\": {\\"separateLfq\\": false, \\"lfqStabilizeLargeRatios\\": true, \\"lfqRequireMsms\\": true, \\"do_ibaq\\": {\\"ibaq\\": \\"False\\", \\"__current_case__\\": 1}, \\"advancedSiteIntensities\\": true}}, \\"qc\\": {\\"do_it\\": \\"false\\", \\"__current_case__\\": 1}, \\"search_opts\\": {\\"template\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_peptide_len\\": \\"8\\", \\"max_peptide_mass\\": \\"4600\\", \\"min_unique_pep\\": \\"0\\", \\"calc_peak_properties\\": false, \\"mbr\\": {\\"match_between_runs\\": \\"True\\", \\"__current_case__\\": 1, \\"matching_time_window\\": \\"0.7\\", \\"matching_ion_mobility_window\\": \\"0.05\\", \\"alignment_time_window\\": \\"20\\", \\"alignment_ion_mobility_window\\": \\"1\\", \\"match_unidentified_features\\": false}, \\"incl_contaminants\\": false, \\"decoy_mode\\": \\"revert\\", \\"psm_fdr\\": \\"0.01\\", \\"protein_fdr\\": \\"0.01\\", \\"min_pep_length\\": \\"8\\", \\"max_pep_length\\": \\"50\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.3.0+galaxy0","type":"tool","uuid":"7e8ec7d9-8fe8-4cf0-a2b3-6358b96c7af5","when":null,"workflow_outputs":[]},"4":{"annotation":"extracting microbial Proteins","content_id":"Grep1","errors":null,"id":4,"input_connections":{"input":{"id":3,"output_name":"proteinGroups"}},"inputs":[],"label":"extracting microbial Proteins","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":701.3374260878941,"top":226.36456298828125},"post_job_actions":{},"tool_id":"Grep1","tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": false, \\"pattern\\": \\"(_HUMAN)|(_REVERSED)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"ee371998-9bf4-4462-abad-63d6a73b995e","when":null,"workflow_outputs":[]},"5":{"annotation":"extracting microbial Peptides","content_id":"Grep1","errors":null,"id":5,"input_connections":{"input":{"id":3,"output_name":"peptides"}},"inputs":[],"label":"extracting microbial Peptides","name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":700.39453125,"top":350.8984375},"post_job_actions":{},"tool_id":"Grep1","tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"-v\\", \\"keep_header\\": false, \\"pattern\\": \\"(_HUMAN)|(_REVERSED)|(CON)|(con)\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"72ea8f91-5de4-43bc-a2c9-264491627eb2","when":null,"workflow_outputs":[]},"6":{"annotation":"extract proteins","content_id":"Cut1","errors":null,"id":6,"input_connections":{"input":{"id":4,"output_name":"out_file1"}},"inputs":[],"label":"extract proteins","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":966.0546875,"top":189.30078125},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"44654f5d-e388-4604-bd61-8eda7bd71152","when":null,"workflow_outputs":[]},"7":{"annotation":"extract peptides","content_id":"Cut1","errors":null,"id":7,"input_connections":{"input":{"id":5,"output_name":"out_file1"}},"inputs":[],"label":"extract peptides","name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":936.2249993300816,"top":350.8452453613281},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"c8cc87cd-13ec-418a-8f95-d0d23dd37e6c","when":null,"workflow_outputs":[]},"8":{"annotation":"Quantified-Proteins","content_id":"Grouping1","errors":null,"id":8,"input_connections":{"input1":{"id":6,"output_name":"out_file1"}},"inputs":[],"label":"Quantified-Proteins","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1221.822265625,"top":177.025390625},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Quantified-Proteins"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"d76c6369-3878-45de-9b28-45ae510ba481","when":null,"workflow_outputs":[{"label":"Quantified-Proteins","output_name":"out_file1","uuid":"57a2ff00-7ad2-4fbb-aafb-6b8323bd23c3"}]},"9":{"annotation":"Quantified-Peptides","content_id":"Grouping1","errors":null,"id":9,"input_connections":{"input1":{"id":7,"output_name":"out_file1"}},"inputs":[],"label":"Quantified-Peptides","name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1213.8046875,"top":342.296875},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"Quantified-Peptides"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grouping1","tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/opt/galaxy/tool-data/shared/ucsc/chrom/?.len\\", \\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"019fe847-4fb9-441f-9450-505f855434e7","when":null,"workflow_outputs":[{"label":"Quantified-Peptides","output_name":"out_file1","uuid":"41b20ef9-acef-4b1d-9a93-1df0c1447a03"}]}},"tags":["name:clinicalMP"],"uuid":"e21d954c-bb46-46f3-8081-0095b6b0de1f","version":1},"readme":"# Clinical Metaproteomics 4: Quantitation\\n\\nQuantitative proteomics is crucial for many important purposes. It allows researchers to measure and compare the levels of proteins or peptides in biological samples. This provides valuable insights into biomarker discovery, comparative analysis, and differential expression studies. Quantitative proteomics also helps in understanding the functional roles of proteins, the composition of protein complexes, and the effects of drugs on protein expression in pharmacological studies. Additionally, it serves as a quality control measure by validating initial protein identifications and providing data normalization for increased accuracy. The quantitative data are essential for hypothesis testing, and systems biology, and have clinical relevance in areas such as disease diagnosis, prognosis, and therapeutic decision-making. In summary, the quantitation workflow in proteomics is essential for understanding the complexities of protein expression and regulation, and it facilitates a wide range of biological and clinical applications.\\n\\nIn this current workflow, we perform Quantification using the MaxQuant tool. A GTN has been developed for this workflow.\\nhttps://training.galaxyproject.org/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html\\n\\n## Inputs dataset\\n\\n- `RAW MSMS datasets` in RAW dataset collection \\n- `Quantitation_Database_for_MaxQuant` in Fasta (protein sequences for database searching)\\n- `Experimental-Design Discovery MaxQuant` in Tabular Format \\n\\n## Inputs values\\n\\nFor MaxQuant \\n- Peptide Length\\n- Variable modifications\\n- Labeled element\\n\\n\\n## Processing\\n\\n- extract microbial proteins and peptides using Select and Cut\\n- Grouping duplicates using the Group tool\\n","changelog":"# Changelog\\n\\n\\n## [0.1] - 2024-08-14\\n\\n- Initial version of Clinical Metaproteomics 4: Quantitation\\n","diagrams":"# Workflow diagrams\\n\\n## Clinical Metaproteomics Quantitation\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Quantitation_Database-For-MaxQuant\\"]@{ shape: doc }\\n1[\\"ℹ️ Experimental-Design Discovery MaxQuant\\"]@{ shape: doc }\\n2[\\"ℹ️ Input Raw-files\\"]@{ shape: docs }\\n3[\\"MaxQuant\\"]@{ shape: process }\\n0 --> 3\\n2 --> 3\\n1 --> 3\\n4[\\"extracting microbial Proteins\\"]@{ shape: process }\\n3 --> 4\\n5[\\"extracting microbial Peptides\\"]@{ shape: process }\\n3 --> 5\\n6[\\"extract proteins\\"]@{ shape: process }\\n4 --> 6\\n7[\\"extract peptides\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Quantified-Proteins\\"]@{ shape: process }\\n6 --> 8\\n9[\\"Quantified-Peptides\\"]@{ shape: process }\\n7 --> 9\\n```\\n","updated":"2024-08-14T00:00:00","trsID":"#workflow/github.com/iwc-workflows/clinicalmp-quantitation/main","dockstore_id":28835,"categories":[],"collections":["Proteomics","Metaproteomics"],"doi":"10.5281/zenodo.15078920","tests":[{"doc":"Test outline for iwc-clinicalmp-quantitation","job":{"Quantitation_Database-For-MaxQuant":{"class":"File","location":"https://zenodo.org/records/10720030/files/Quantitation_Database_for_MaxQuant.fasta","filetype":"fasta"},"Experimental-Design Discovery MaxQuant":{"class":"File","location":"https://zenodo.org/records/10720030/files/Experimental-Design_Discovery_MaxQuant.tabular","filetype":"tabular"},"Input Raw-files":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.raw","location":"https://zenodo.org/records/10720030/files/PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.raw"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.raw","location":"https://zenodo.org/records/10720030/files/PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.raw"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.raw","location":"https://zenodo.org/records/10720030/files/PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.raw"},{"class":"File","identifier":"PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.raw","location":"https://zenodo.org/records/10720030/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.raw"}]}},"outputs":{"Quantified-Proteins":{"path":"test-data/Quantified-Proteins.tabular"},"Quantified-Peptides":{"path":"test-data/Quantified-Peptides.tabular"}}}]}],"path":"./workflows/proteomics/clinicalmp/clinicalmp-quantitation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/amr_gene_detection.ga","testParameterFiles":["/amr_gene_detection-tests.yml"],"authors":[{"name":"ABRomics","email":"abromics-support@groupes.france-bioinformatique.fr"},{"name":"abromics-consortium","url":"https://www.abromics.fr/"},{"name":"Pierre Marin","alternateName":"pimarin","orcid":"0000-0002-8304-138X"},{"name":"Clea Siguret","alternateName":"clsiguret","orcid":"0009-0005-6140-0379"}],"definition":{"a_galaxy_workflow":"true","annotation":"Antimicrobial resistance gene detection from assembled bacterial genomes","creator":[{"class":"Person","email":"mailto:abromics-support@groupes.france-bioinformatique.fr","name":"ABRomics"},{"class":"Organization","name":"abromics-consortium","url":"https://www.abromics.fr/"},{"alternateName":"pimarin","class":"Person","identifier":"https://orcid.org/0000-0002-8304-138X","name":"Pierre Marin"},{"alternateName":"clsiguret","class":"Person","identifier":"https://orcid.org/0000-0002-8304-138X","name":"Clea Siguret"}],"format-version":"0.1","license":"GPL-3.0-or-later","release":"1.1.5","name":"amr_gene_detection","steps":{"0":{"annotation":"Can be any fasta file.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Can be any fasta file.","name":"Input sequence fasta"}],"label":"Input sequence fasta","name":"Input dataset","outputs":[],"position":{"left":2.4727149882562167,"top":677.1068397845053},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"67722a09-41de-4af4-9e60-378b2738f910","when":null,"workflow_outputs":[]},"1":{"annotation":"Select the database to identify resistance-associated point mutations with AMRFinderPlus.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Select the database to identify resistance-associated point mutations with AMRFinderPlus.","name":"Select a taxonomy group point mutation"}],"label":"Select a taxonomy group point mutation","name":"Input parameter","outputs":[],"position":{"left":6.398714988256147,"top":785.1080080587546},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"6092b401-9042-4ec2-9970-d614afd77441","when":null,"workflow_outputs":[]},"2":{"annotation":"Select the database to identify AMR genes with AMRFinderPlus.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Select the database to identify AMR genes with AMRFinderPlus.","name":"Select a AMR genes detection database"}],"label":"Select a AMR genes detection database","name":"Input parameter","outputs":[],"position":{"left":0,"top":1031.8311633489975},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"3c9163c4-9393-49ca-8271-b98a89abcfb9","when":null,"workflow_outputs":[]},"3":{"annotation":"Select the database to identify virulence genes with ABRicate..","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Select the database to identify virulence genes with ABRicate..","name":"Select a virulence genes detection database"}],"label":"Select a virulence genes detection database","name":"Input parameter","outputs":[],"position":{"left":2.0333251953125,"top":1166.9166870117188},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"eba8ebb2-0072-41dd-a9b2-878d05f04ff6","when":null,"workflow_outputs":[]},"4":{"annotation":"staramr_amr_genes","content_id":"toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.10.0+galaxy1","errors":null,"id":4,"input_connections":{"genomes":{"id":0,"output_name":"output"}},"inputs":[],"label":"staramr_amr_genes","name":"staramr","outputs":[{"name":"blast_hits","type":"input"},{"name":"mlst","type":"tabular"},{"name":"summary","type":"tabular"},{"name":"detailed_summary","type":"tabular"},{"name":"resfinder","type":"tabular"},{"name":"plasmidfinder","type":"tabular"},{"name":"settings","type":"txt"},{"name":"excel","type":"xlsx"}],"position":{"left":496.9350972460573,"top":0},"post_job_actions":{"RenameDatasetActionblast_hits":{"action_arguments":{"newname":"staramr_blast_hits"},"action_type":"RenameDatasetAction","output_name":"blast_hits"},"RenameDatasetActiondetailed_summary":{"action_arguments":{"newname":"staramr_detailed_summary"},"action_type":"RenameDatasetAction","output_name":"detailed_summary"},"RenameDatasetActionexcel":{"action_arguments":{"newname":"staramr_excel_report"},"action_type":"RenameDatasetAction","output_name":"excel"},"RenameDatasetActionmlst":{"action_arguments":{"newname":"staramr_mlst_report"},"action_type":"RenameDatasetAction","output_name":"mlst"},"RenameDatasetActionplasmidfinder":{"action_arguments":{"newname":"staramr_plasmidfinder_report"},"action_type":"RenameDatasetAction","output_name":"plasmidfinder"},"RenameDatasetActionresfinder":{"action_arguments":{"newname":"staramr_resfinder_report"},"action_type":"RenameDatasetAction","output_name":"resfinder"},"RenameDatasetActionsettings":{"action_arguments":{"newname":"staramr_settings"},"action_type":"RenameDatasetAction","output_name":"settings"},"RenameDatasetActionsummary":{"action_arguments":{"newname":"staramr_summary"},"action_type":"RenameDatasetAction","output_name":"summary"},"TagDatasetActionblast_hits":{"action_arguments":{"tags":"staramr_blast_hits"},"action_type":"TagDatasetAction","output_name":"blast_hits"},"TagDatasetActiondetailed_summary":{"action_arguments":{"tags":"staramr_detailed_summary"},"action_type":"TagDatasetAction","output_name":"detailed_summary"},"TagDatasetActionexcel":{"action_arguments":{"tags":"staramr_excel_report"},"action_type":"TagDatasetAction","output_name":"excel"},"TagDatasetActionmlst":{"action_arguments":{"tags":"staramr_mlst_report"},"action_type":"TagDatasetAction","output_name":"mlst"},"TagDatasetActionplasmidfinder":{"action_arguments":{"tags":"staramr_plasmidfinder_report"},"action_type":"TagDatasetAction","output_name":"plasmidfinder"},"TagDatasetActionresfinder":{"action_arguments":{"tags":"staramr_resfinder_report"},"action_type":"TagDatasetAction","output_name":"resfinder"},"TagDatasetActionsettings":{"action_arguments":{"tags":"staramr_settings"},"action_type":"TagDatasetAction","output_name":"settings"},"TagDatasetActionsummary":{"action_arguments":{"tags":"staramr_summary"},"action_type":"TagDatasetAction","output_name":"summary"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.10.0+galaxy1","tool_shed_repository":{"changeset_revision":"3b22de20bd4b","name":"staramr","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced\\": {\\"pid_threshold\\": \\"90.0\\", \\"plength_resfinder\\": \\"60.0\\", \\"plength_pointfinder\\": \\"95.0\\", \\"plength_plasmidfinder\\": \\"60.0\\", \\"genome_size_lower_bound\\": \\"4000000\\", \\"genome_size_upper_bound\\": \\"6000000\\", \\"minimum_N50_value\\": \\"10000\\", \\"minimum_contig_length\\": \\"300\\", \\"unacceptable_number_contigs\\": \\"1000\\", \\"report_all_blast\\": false, \\"exclude_negatives\\": false, \\"exclude_resistance_phenotypes\\": false, \\"mlst_scheme\\": \\"auto\\", \\"exclude_genes\\": {\\"exclude_genes_condition\\": \\"default\\", \\"__current_case__\\": 0}, \\"complex_mutations\\": {\\"complex_mutations_condition\\": \\"default\\", \\"__current_case__\\": 0}, \\"plasmidfinder_type\\": \\"include_all\\"}, \\"genomes\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"pointfinder_db\\": {\\"use_pointfinder\\": \\"disabled\\", \\"__current_case__\\": 0}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.10.0+galaxy1","type":"tool","uuid":"10eed503-9e5b-4acf-8cd8-29e81d7668a5","when":null,"workflow_outputs":[{"label":"staramr_detailed_summary","output_name":"detailed_summary","uuid":"7a78b7cd-f61a-4c1a-9dea-68714afe61ca"},{"label":"staramr_plasmidfinder_report","output_name":"plasmidfinder","uuid":"cfe793e3-ff25-45f6-9f5e-cb4564cfa3ce"},{"label":"staramr_excel_report","output_name":"excel","uuid":"a273d6e1-9935-4760-acee-a9d43c06216d"},{"label":"staramr_mlst_report","output_name":"mlst","uuid":"95872b33-b343-4225-b150-874b4a10fcb6"},{"label":"staramr_resfinder_report","output_name":"resfinder","uuid":"605f5f68-027d-4435-94db-f9d148143994"},{"label":"staramr_settings","output_name":"settings","uuid":"12fe0304-ebd1-4ec9-935e-f41a21872caf"},{"label":"staramr_blast_hits","output_name":"blast_hits","uuid":"a919060d-f83c-4da6-93ef-6d71fa564e8a"},{"label":"staramr_summary","output_name":"summary","uuid":"bf2d3ff0-7116-46a8-91fb-027609667b83"}]},"5":{"annotation":"amrfinderplus_point_mutation","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/amrfinderplus/amrfinderplus/3.12.8+galaxy0","errors":null,"id":5,"input_connections":{"input_option|amrfinder_db_select":{"id":2,"output_name":"output"},"input_option|input_mode|nucleotide_input":{"id":0,"output_name":"output"},"organism_options|organism_conditionnal|organism":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool AMRFinderPlus","name":"input_option"}],"label":"amrfinderplus_point_mutation","name":"AMRFinderPlus","outputs":[{"name":"amrfinderplus_report","type":"tabular"},{"name":"mutation_all_report","type":"tabular"},{"name":"nucleotide_output","type":"fasta"}],"position":{"left":641.3405290986273,"top":837.2197586407639},"post_job_actions":{"TagDatasetActionamrfinderplus_report":{"action_arguments":{"tags":"amrfinderplus_report"},"action_type":"TagDatasetAction","output_name":"amrfinderplus_report"},"TagDatasetActionmutation_all_report":{"action_arguments":{"tags":"amrfinderplus_mutation"},"action_type":"TagDatasetAction","output_name":"mutation_all_report"},"TagDatasetActionnucleotide_output":{"action_arguments":{"tags":"amrfinderplus_nucleotide"},"action_type":"TagDatasetAction","output_name":"nucleotide_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/amrfinderplus/amrfinderplus/3.12.8+galaxy0","tool_shed_repository":{"changeset_revision":"7844dbe4f8e2","name":"amrfinderplus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_option\\": {\\"amrfinder_db_select\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_mode\\": {\\"input_select\\": \\"nucleotide\\", \\"__current_case__\\": 0, \\"nucleotide_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nucleotide_flank5_size\\": \\"0\\"}}, \\"options\\": {\\"ident_min\\": \\"-1.0\\", \\"coverage_min\\": \\"0.5\\", \\"translation_table\\": \\"11\\", \\"plus\\": true, \\"report_all_equal\\": true, \\"print_node\\": true, \\"name\\": null}, \\"organism_options\\": {\\"organism_conditionnal\\": {\\"organism_select\\": \\"add_organism\\", \\"__current_case__\\": 0, \\"organism\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mutation_all\\": true, \\"report_common\\": true}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.8+galaxy0","type":"tool","uuid":"3b79182a-17e6-46ec-9623-99dd5838f7cc","when":null,"workflow_outputs":[{"label":"amrfinderplus_nucleotide","output_name":"nucleotide_output","uuid":"7c952fb9-0d93-4c24-9bac-3c5373d2af07"},{"label":"amrfinderplus_mutation","output_name":"mutation_all_report","uuid":"5b415cff-3d62-4e3b-997c-b01a940c1938"},{"label":"amrfinderplus_report","output_name":"amrfinderplus_report","uuid":"6a5038ba-93fd-4e89-a00c-2b458578898b"}]},"6":{"annotation":"abricate_virulence","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","errors":null,"id":6,"input_connections":{"adv|db":{"id":3,"output_name":"output"},"file_input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool ABRicate","name":"adv"}],"label":"abricate_virulence","name":"ABRicate","outputs":[{"name":"report","type":"tabular"}],"position":{"left":885.033177397046,"top":342.096138386568},"post_job_actions":{"RenameDatasetActionreport":{"action_arguments":{"newname":"abricate_virulence_report"},"action_type":"RenameDatasetAction","output_name":"report"},"TagDatasetActionreport":{"action_arguments":{"tags":"abricate_virulence_report"},"action_type":"TagDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","tool_shed_repository":{"changeset_revision":"c2ef298da409","name":"abricate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"db\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"no_header\\": false, \\"min_dna_id\\": \\"80.0\\", \\"min_cov\\": \\"80.0\\"}, \\"file_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"f51a45e6-f937-4286-a2b7-f0ff71febb67","when":null,"workflow_outputs":[{"label":"abricate_virulence_report","output_name":"report","uuid":"a1347ac2-372c-42a4-953e-e9b5422b9b5d"}]},"7":{"annotation":"ToolDistillator extracts results from tools and creates a JSON file for each tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","errors":null,"id":7,"input_connections":{"tool_section|tools_0|select_tool|input":{"id":6,"output_name":"report"},"tool_section|tools_0|select_tool|reference_database_version":{"id":3,"output_name":"output"},"tool_section|tools_1|select_tool|input":{"id":4,"output_name":"resfinder"},"tool_section|tools_1|select_tool|mlst_file_path":{"id":4,"output_name":"mlst"},"tool_section|tools_1|select_tool|setting_file_path":{"id":4,"output_name":"settings"},"tool_section|tools_2|select_tool|input":{"id":5,"output_name":"amrfinderplus_report"},"tool_section|tools_2|select_tool|nucleotide_sequence_path":{"id":5,"output_name":"nucleotide_output"},"tool_section|tools_2|select_tool|point_mutation_report_path":{"id":5,"output_name":"mutation_all_report"},"tool_section|tools_2|select_tool|reference_database_version":{"id":2,"output_name":"output"}},"inputs":[],"label":"ToolDistillator","name":"ToolDistillator","outputs":[{"name":"output_json","type":"input"}],"position":{"left":1138.6692404030955,"top":584.9586737297199},"post_job_actions":{"RenameDatasetActionoutput_json":{"action_arguments":{"newname":"tooldistillator_results"},"action_type":"RenameDatasetAction","output_name":"output_json"},"TagDatasetActionoutput_json":{"action_arguments":{"tags":"tooldistillator_results"},"action_type":"TagDatasetAction","output_name":"output_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"ea93df4b3df2","name":"tooldistillator","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"log\\": false, \\"tool_section\\": {\\"tools\\": [{\\"__index__\\": 0, \\"select_tool\\": {\\"tool_list\\": \\"abricate\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"select_tool\\": {\\"tool_list\\": \\"staramr\\", \\"__current_case__\\": 20, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mlst_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"plasmidfinder_file_path\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"pointfinder_file_path\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"setting_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 2, \\"select_tool\\": {\\"tool_list\\": \\"amrfinderplus\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"point_mutation_report_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nucleotide_sequence_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"1589f8a1-65b6-43f0-9e75-cad9b1752e5c","when":null,"workflow_outputs":[{"label":"tooldistillator_results","output_name":"output_json","uuid":"fd122fdd-0907-416d-85a3-30521674a50b"}]},"8":{"annotation":"ToolDistillator summarize groups all JSON file into a unique JSON file","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","errors":null,"id":8,"input_connections":{"summarize_data":{"id":7,"output_name":"output_json"}},"inputs":[],"label":"ToolDistillator summarize","name":"ToolDistillator Summarize","outputs":[{"name":"summary_json","type":"json"}],"position":{"left":1510.406873337906,"top":835.1044439315301},"post_job_actions":{"RenameDatasetActionsummary_json":{"action_arguments":{"newname":"tooldistillator_summarize"},"action_type":"RenameDatasetAction","output_name":"summary_json"},"TagDatasetActionsummary_json":{"action_arguments":{"tags":"tooldistillator_summarize"},"action_type":"TagDatasetAction","output_name":"summary_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"24eec94e6dfc","name":"tooldistillator_summarize","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"summarize_data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"a88f6b82-502c-46cf-a174-2c8361e8e8dd","when":null,"workflow_outputs":[{"label":"tooldistillator_summarize","output_name":"summary_json","uuid":"d5921f67-84d7-4265-926b-86cb3e2977b0"}]}},"tags":["fasta","Genomics","ABRomics","antibiotic-resistance","antimicrobial-resistance-genes","antimicrobial resistance","bacterial-genomics","AMR","AMR-detection"],"uuid":"dfcedd6a-9dc1-4dcd-8b15-dbd21482f463","version":1},"readme":"# AMR gene detection workflow in an assembled bacterial genome (v1.0)\\n\\nThis workflow uses assembled bacterial genome fasta files (but can be any fasta file) and executes the following steps:\\n1. Genomic detection\\n    - Antimicrobial resistance gene identification:\\n        - **staramr** to blast against ResFinder and PlasmidFinder database\\n        - **AMRFinderPlus** to find antimicrobial resistance genes and point mutations \\n    - Virulence gene identification:\\n        - **ABRicate** with VFDB_A database\\n2. Aggregating outputs into a single JSON file\\n    - **ToolDistillator** to extract and aggregate information from different tool outputs to JSON parsable files\\n\\n## Inputs\\n\\n1. Assembled bacterial genome in fasta format.\\n\\n## Outputs\\n\\n1. Genomic detection\\n    - Antimicrobial resistance gene identification:\\n        - AMR gene list\\n        - MLST typing\\n        - Plasmid gene identification\\n        - Blast hits\\n        - AMR gene fasta (assembled nucleotide sequences)\\n        - Point mutation list\\n    - Virulence gene identification:\\n        - Gene identification in tabular format\\n2. Aggregating outputs:\\n    - JSON file with information about the outputs of **staramr**, **AMRFinderPlus**, **ABRicate**","changelog":"# Changelog\\n\\n## [1.1.5] - 2024-10-21\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1`\\n\\n## [1.1.4] - 2024-10-11\\n\\n### Manual update\\n\\n- Changed input parameters to select databases in WFs. Use \\"Attempt restriction based on connections\\" instead of \\"Provide list of suggested values\\".\\n\\n## [1.1.3] - 2024-09-30\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0`\\n\\n## [1.1.2] - 2024-09-19\\n\\n### Manual update\\n\\n- Change percent identity threshold for Resfinder blast results in StarAMR from 98.0% to 90.0%\\n\\n## [1.1.1] - 2024-07-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0`\\n\\n## [1.1] - 2024-06-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.10.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.10.0+galaxy1`\\n\\n## [1.0] - 05-06-2024\\n\\n","diagrams":"# Workflow diagrams\\n\\n## amr_gene_detection\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input sequence fasta\\"]@{ shape: doc }\\n1[\\"ℹ️ Select a taxonomy group point mutation\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Select a AMR genes detection database\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Select a virulence genes detection database\\"]@{ shape: lean-l }\\n4[\\"staramr_amr_genes\\"]@{ shape: process }\\n0 --> 4\\n5[\\"amrfinderplus_point_mutation\\"]@{ shape: process }\\n2 --> 5\\n0 --> 5\\n1 --> 5\\n6[\\"abricate_virulence\\"]@{ shape: process }\\n3 --> 6\\n0 --> 6\\n7[\\"ToolDistillator\\"]@{ shape: process }\\n6 --> 7\\n3 --> 7\\n4 --> 7\\n4 --> 7\\n4 --> 7\\n5 --> 7\\n5 --> 7\\n5 --> 7\\n2 --> 7\\n8[\\"ToolDistillator summarize\\"]@{ shape: process }\\n7 --> 8\\n```\\n","updated":"2024-10-21T00:00:00","trsID":"#workflow/github.com/iwc-workflows/amr_gene_detection/main","dockstore_id":27290,"categories":[],"collections":["Genome Annotation"],"doi":"10.5281/zenodo.15078806","tests":[{"doc":"Test outline for amr_gene_detection.ga","job":{"Input sequence fasta":{"class":"File","path":"https://zenodo.org/records/11488310/files/shovill_contigs_fasta"},"Select a taxonomy group point mutation":"Enterococcus_faecalis","Select a AMR genes detection database":"amrfinderplus_V3.12_2024-05-02.2","Select a virulence genes detection database":"vfdb"},"outputs":{"staramr_detailed_summary":{"asserts":{"has_text":{"text":"Resistance"},"has_n_columns":{"n":12}}},"staramr_resfinder_report":{"asserts":{"has_text":{"text":"tetracycline"},"has_n_columns":{"n":13}}},"staramr_mlst_report":{"asserts":{"has_text":{"text":"efaecalis"},"has_n_columns":{"n":10}}},"staramr_plasmidfinder_report":{"asserts":{"has_text":{"text":"CP002494"},"has_n_columns":{"n":9}}},"staramr_summary":{"asserts":{"has_text":{"text":"streptomycin"},"has_n_columns":{"n":12}}},"amrfinderplus_report":{"asserts":{"has_text":{"text":"Enterococcus faecalis quinolone resistant ParC"},"has_n_columns":{"n":23}}},"amrfinderplus_mutation":{"asserts":{"has_text":{"text":"cardiolipin synthase Cls"},"has_n_columns":{"n":23}}},"abricate_virulence_report":{"asserts":{"has_text":{"text":"endocarditis specific antigen"},"has_n_columns":{"n":15}}},"tooldistillator_summarize":{"asserts":[{"that":"has_text","text":"% Identity to reference sequence"},{"that":"has_text","text":"Enterococcus faecalis V583"},{"that":"has_text","text":"Fibrinogen binding protein"}]}}}]}],"path":"./workflows/bacterial_genomics/amr_gene_detection"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/bacterial_genome_annotation.ga","testParameterFiles":["/bacterial_genome_annotation-tests.yml"],"authors":[{"name":"ABRomics","email":"abromics-support@groupes.france-bioinformatique.fr"},{"name":"abromics-consortium","url":"https://www.abromics.fr/"},{"name":"Pierre Marin","alternateName":"pimarin","orcid":"0000-0002-8304-138X"},{"name":"Clea Siguret","alternateName":"clsiguret","orcid":"0009-0005-6140-0379"}],"definition":{"a_galaxy_workflow":"true","annotation":"Annotation of an assembled bacterial genomes to detect genes, potential plasmids, integrons and Insertion sequence (IS) elements.","creator":[{"class":"Person","email":"mailto:abromics-support@groupes.france-bioinformatique.fr","name":"ABRomics"},{"class":"Organization","name":"abromics-consortium","url":"https://www.abromics.fr/"},{"alternateName":"pimarin","class":"Person","identifier":"https://orcid.org/0000-0002-8304-138X","name":"Pierre Marin"},{"alternateName":"clsiguret","class":"Person","identifier":"https://orcid.org/0009-0005-6140-0379","name":"Clea Siguret"}],"format-version":"0.1","license":"GPL-3.0-or-later","release":"1.1.7","name":"bacterial_genome_annotation","steps":{"0":{"annotation":"Can be any fasta file.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Can be any fasta file.","name":"Input sequence fasta"}],"label":"Input sequence fasta","name":"Input dataset","outputs":[],"position":{"left":0,"top":559.2197431502472},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fasta\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"8afa880e-a04c-4831-b37b-d0f317f767b0","when":null,"workflow_outputs":[]},"1":{"annotation":"Select a database to identify plasmids with PlasmidFinder.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Select a database to identify plasmids with PlasmidFinder.","name":"Select a plasmid detection database"}],"label":"Select a plasmid detection database","name":"Input parameter","outputs":[],"position":{"left":9.602014342350742,"top":734.3939210886799},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"66c11ad1-7cd5-40b7-b97f-8d15cc097dd9","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"a2d4e531-c070-45c1-b56a-be4356aabc79"}]},"2":{"annotation":"Select a database to predict CDS and small proteins (sORF) with Bakta.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Select a database to predict CDS and small proteins (sORF) with Bakta.","name":"Select a bacterial genome annotation database"}],"label":"Select a bacterial genome annotation database","name":"Input parameter","outputs":[],"position":{"left":21.882601498378413,"top":859.5166625976562},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"4de1f52c-078c-4112-86ad-e4e8f3b62e3a","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"724dc1d5-6f90-4c1c-915c-99dded697740"}]},"3":{"annotation":"Select a database to annotate AMR genes with Bakta.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Select a database to annotate AMR genes with Bakta.","name":"Select a AMRFinderPlus database"}],"label":"Select a AMRFinderPlus database","name":"Input parameter","outputs":[],"position":{"left":10.643158620281753,"top":995.2631409484753},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"aec02a2c-6bf7-476d-91aa-3f1ce49083c7","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"768e962a-f016-41c6-96c5-4b32b26b4856"}]},"4":{"annotation":"genomic_annotation_insertionelement_isescan","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/isescan/isescan/1.7.2.3+galaxy1","errors":null,"id":4,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[],"label":"genomic_annotation_insertionelement_isescan","name":"ISEScan","outputs":[{"name":"logfile","type":"txt"},{"name":"summary","type":"tabular"},{"name":"all_results","type":"tabular"},{"name":"annotation","type":"gff"},{"name":"is_fasta","type":"fasta"},{"name":"orf_fna","type":"fasta"},{"name":"orf_faa","type":"fasta"}],"position":{"left":887.7889628790149,"top":105.14125185095409},"post_job_actions":{"RenameDatasetActionall_results":{"action_arguments":{"newname":"isescan_results_tabular"},"action_type":"RenameDatasetAction","output_name":"all_results"},"RenameDatasetActionannotation":{"action_arguments":{"newname":"isescan_annotation_gff3"},"action_type":"RenameDatasetAction","output_name":"annotation"},"RenameDatasetActionis_fasta":{"action_arguments":{"newname":"isescan_is_fasta"},"action_type":"RenameDatasetAction","output_name":"is_fasta"},"RenameDatasetActionlogfile":{"action_arguments":{"newname":"isescan_logfile_text"},"action_type":"RenameDatasetAction","output_name":"logfile"},"RenameDatasetActionorf_faa":{"action_arguments":{"newname":"isescan_orf_faa"},"action_type":"RenameDatasetAction","output_name":"orf_faa"},"RenameDatasetActionorf_fna":{"action_arguments":{"newname":"isescan_orf_fasta"},"action_type":"RenameDatasetAction","output_name":"orf_fna"},"RenameDatasetActionsummary":{"action_arguments":{"newname":"isescan_summary_tabular"},"action_type":"RenameDatasetAction","output_name":"summary"},"TagDatasetActionall_results":{"action_arguments":{"tags":"isescan_results_tabular"},"action_type":"TagDatasetAction","output_name":"all_results"},"TagDatasetActionannotation":{"action_arguments":{"tags":"isescan_annotation_gff3"},"action_type":"TagDatasetAction","output_name":"annotation"},"TagDatasetActionis_fasta":{"action_arguments":{"tags":"isescan_is_fasta"},"action_type":"TagDatasetAction","output_name":"is_fasta"},"TagDatasetActionlogfile":{"action_arguments":{"tags":"isescan,logfile"},"action_type":"TagDatasetAction","output_name":"logfile"},"TagDatasetActionorf_faa":{"action_arguments":{"tags":"isescan_orf_faa"},"action_type":"TagDatasetAction","output_name":"orf_faa"},"TagDatasetActionorf_fna":{"action_arguments":{"tags":"isescan_orf_fasta"},"action_type":"TagDatasetAction","output_name":"orf_fna"},"TagDatasetActionsummary":{"action_arguments":{"tags":"isescan_summary_tabular"},"action_type":"TagDatasetAction","output_name":"summary"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/isescan/isescan/1.7.2.3+galaxy1","tool_shed_repository":{"changeset_revision":"9e776e7fab4f","name":"isescan","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"log_activate\\": true, \\"remove_short_is\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.7.2.3+galaxy1","type":"tool","uuid":"f589af2f-9dd9-41c0-b7a2-a499c875cb54","when":null,"workflow_outputs":[{"label":"isescan_summary_tabular","output_name":"summary","uuid":"a09e2ed7-5b31-4beb-9923-d5ec6910da16"},{"label":"isescan_annotation_gff3","output_name":"annotation","uuid":"de82533b-90f8-470d-80a2-e1559be93f0d"},{"label":"isescan_logfile_text","output_name":"logfile","uuid":"c0539e24-831e-406f-a5d2-3dd52845a5d7"},{"label":"isescan_results_tabular","output_name":"all_results","uuid":"24f7dee0-f50a-4444-83f4-b09623a4a993"},{"label":"isescan_orf_faa","output_name":"orf_faa","uuid":"c74754f5-e011-4f55-801f-feeab8dd777c"},{"label":"isescan_is_fasta","output_name":"is_fasta","uuid":"a89b0159-ccef-4c10-9b17-d5da62312bda"},{"label":"isescan_orf_fasta","output_name":"orf_fna","uuid":"1a91afd7-104d-4807-bffa-61c5df646b85"}]},"5":{"annotation":"genomic_annotation_integron","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.5+galaxy0","errors":null,"id":5,"input_connections":{"sequence":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Integron Finder","name":"topology_file"}],"label":"genomic_annotation_integron","name":"Integron Finder","outputs":[{"name":"integron_log","type":"txt"},{"name":"integrons_table","type":"tsv"},{"name":"summary","type":"tsv"}],"position":{"left":581.3510244512896,"top":719.4853608141518},"post_job_actions":{"RenameDatasetActionintegron_log":{"action_arguments":{"newname":"integronfinder2_logfile_text"},"action_type":"RenameDatasetAction","output_name":"integron_log"},"RenameDatasetActionintegrons_table":{"action_arguments":{"newname":"integronfinder2_results_tabular"},"action_type":"RenameDatasetAction","output_name":"integrons_table"},"RenameDatasetActionsummary":{"action_arguments":{"newname":"integronfinder2_summary"},"action_type":"RenameDatasetAction","output_name":"summary"},"TagDatasetActionintegron_log":{"action_arguments":{"tags":"integronfinder,logfile"},"action_type":"TagDatasetAction","output_name":"integron_log"},"TagDatasetActionintegrons_table":{"action_arguments":{"tags":"integronfinder2_results_tabular"},"action_type":"TagDatasetAction","output_name":"integrons_table"},"TagDatasetActionsummary":{"action_arguments":{"tags":"integronfinder2_summary"},"action_type":"TagDatasetAction","output_name":"summary"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.5+galaxy0","tool_shed_repository":{"changeset_revision":"bfd290fe1588","name":"integron_finder","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"gbk\\": false, \\"local_max\\": true, \\"no_logfile\\": false, \\"pdf\\": false, \\"promoter_attI\\": true, \\"sequence\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"settings\\": {\\"attc_settings\\": {\\"dist_thresh\\": \\"4000\\", \\"calin_threshold\\": \\"2\\", \\"max_attc_size\\": \\"200\\", \\"min_attc_size\\": \\"40\\", \\"keep_palindromes\\": false, \\"covar_matrix\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"protein_settings\\": {\\"no_proteins\\": false, \\"union_integrases\\": false, \\"func_annot\\": false}}, \\"topology_file\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"type_replicon\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.5+galaxy0","type":"tool","uuid":"0f39c064-cac7-4ee7-9f37-94b7717a033e","when":null,"workflow_outputs":[{"label":"integronfinder2_logfile_text","output_name":"integron_log","uuid":"dca60f87-9f7a-4631-b197-62ca54f7bafe"},{"label":"integronfinder2_summary","output_name":"summary","uuid":"b96d90c3-7678-41b8-9b01-61d5c9da80bc"},{"label":"integronfinder2_results_tabular","output_name":"integrons_table","uuid":"cb38191b-3a0e-43d3-823a-0c9563e39d66"}]},"6":{"annotation":"genomic_annotation_plasmid_plasmidfinder","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/plasmidfinder/plasmidfinder/2.1.6+galaxy1","errors":null,"id":6,"input_connections":{"input|database_name":{"id":1,"output_name":"output"},"input|input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool PlasmidFinder","name":"input"},{"description":"runtime parameter for tool PlasmidFinder","name":"input"}],"label":"genomic_annotation_plasmid_plasmidfinder","name":"PlasmidFinder","outputs":[{"name":"json_file","type":"json"},{"name":"hit_file","type":"fasta"},{"name":"plasmid_file","type":"fasta"},{"name":"result_file","type":"tabular"},{"name":"raw_file","type":"txt"}],"position":{"left":469.0425884161387,"top":0},"post_job_actions":{"RenameDatasetActionhit_file":{"action_arguments":{"newname":"plasmidfinder_hit_genome_fasta"},"action_type":"RenameDatasetAction","output_name":"hit_file"},"RenameDatasetActionjson_file":{"action_arguments":{"newname":"plasmidfinder_result_json"},"action_type":"RenameDatasetAction","output_name":"json_file"},"RenameDatasetActionplasmid_file":{"action_arguments":{"newname":"plasmidfinder_hit_reference_fasta"},"action_type":"RenameDatasetAction","output_name":"plasmid_file"},"RenameDatasetActionraw_file":{"action_arguments":{"newname":"plasmidfinder_results_raw"},"action_type":"RenameDatasetAction","output_name":"raw_file"},"RenameDatasetActionresult_file":{"action_arguments":{"newname":"plasmidfinder_results_tabular"},"action_type":"RenameDatasetAction","output_name":"result_file"},"TagDatasetActionhit_file":{"action_arguments":{"tags":"plasmidfinder_hit_genome_fasta"},"action_type":"TagDatasetAction","output_name":"hit_file"},"TagDatasetActionjson_file":{"action_arguments":{"tags":"plasmidfinder_result_json"},"action_type":"TagDatasetAction","output_name":"json_file"},"TagDatasetActionplasmid_file":{"action_arguments":{"tags":"plasmidfinder_hit_reference_fasta"},"action_type":"TagDatasetAction","output_name":"plasmid_file"},"TagDatasetActionraw_file":{"action_arguments":{"tags":"plasmidfinder_results_raw"},"action_type":"TagDatasetAction","output_name":"raw_file"},"TagDatasetActionresult_file":{"action_arguments":{"tags":"plasmidfinder_results_tabular"},"action_type":"TagDatasetAction","output_name":"result_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/plasmidfinder/plasmidfinder/2.1.6+galaxy1","tool_shed_repository":{"changeset_revision":"7075b7a5441b","name":"plasmidfinder","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"database_name\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"options\\": {\\"min_cov\\": \\"0.6\\", \\"threshold\\": \\"0.95\\"}, \\"output_files\\": {\\"output_selection\\": [\\"hit_fasta\\", \\"plasmid_fasta\\", \\"result_tsv\\", \\"result_txt\\", \\"data_json\\"]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.6+galaxy1","type":"tool","uuid":"0cbd7d74-9525-4fc4-9518-836c99f6d397","when":null,"workflow_outputs":[{"label":"plasmidfinder_results_tabular","output_name":"result_file","uuid":"696867f7-0700-4ca5-bee4-bf405f44f6fa"},{"label":"plasmidfinder_result_json","output_name":"json_file","uuid":"1449c8c3-9621-4ba2-b32e-29a597a63f2d"},{"label":"plasmidfinder_results_raw","output_name":"raw_file","uuid":"e4c42f08-1e32-40e0-a0f0-09037299c4a7"},{"label":"plasmidfinder_hit_genome_fasta","output_name":"hit_file","uuid":"2924588e-064d-4c31-9011-ef9cfc646c6e"},{"label":"plasmidfinder_hit_reference_fasta","output_name":"plasmid_file","uuid":"b70ea191-97da-4e59-b14c-3988284e552e"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.4+galaxy0","errors":null,"id":7,"input_connections":{"input_option|amrfinder_db_select":{"id":3,"output_name":"output"},"input_option|bakta_db_select":{"id":2,"output_name":"output"},"input_option|input_file":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Bakta","name":"annotation"},{"description":"runtime parameter for tool Bakta","name":"annotation"},{"description":"runtime parameter for tool Bakta","name":"annotation"},{"description":"runtime parameter for tool Bakta","name":"annotation"},{"description":"runtime parameter for tool Bakta","name":"input_option"},{"description":"runtime parameter for tool Bakta","name":"input_option"},{"description":"runtime parameter for tool Bakta","name":"input_option"}],"label":null,"name":"Bakta","outputs":[{"name":"annotation_tsv","type":"tabular"},{"name":"annotation_gff3","type":"gff3"},{"name":"annotation_gbff","type":"tabular"},{"name":"annotation_embl","type":"tabular"},{"name":"annotation_fna","type":"fasta"},{"name":"annotation_ffn","type":"fasta"},{"name":"annotation_faa","type":"fasta"},{"name":"hypotheticals_tsv","type":"tabular"},{"name":"hypotheticals_faa","type":"fasta"},{"name":"summary_txt","type":"txt"},{"name":"annotation_json","type":"json"},{"name":"annotation_plot","type":"svg"}],"position":{"left":1035.0012031530584,"top":927.9172491992002},"post_job_actions":{"RenameDatasetActionannotation_embl":{"action_arguments":{"newname":"bakta_annotation_embl"},"action_type":"RenameDatasetAction","output_name":"annotation_embl"},"RenameDatasetActionannotation_faa":{"action_arguments":{"newname":"bakta_aminoacid_sequence_faa"},"action_type":"RenameDatasetAction","output_name":"annotation_faa"},"RenameDatasetActionannotation_ffn":{"action_arguments":{"newname":"bakta_nucleotide_sequence_fasta"},"action_type":"RenameDatasetAction","output_name":"annotation_ffn"},"RenameDatasetActionannotation_fna":{"action_arguments":{"newname":"bakta_assembly_fasta"},"action_type":"RenameDatasetAction","output_name":"annotation_fna"},"RenameDatasetActionannotation_gbff":{"action_arguments":{"newname":"bakta_annotation_gbff"},"action_type":"RenameDatasetAction","output_name":"annotation_gbff"},"RenameDatasetActionannotation_gff3":{"action_arguments":{"newname":"bakta_annotation_gff3"},"action_type":"RenameDatasetAction","output_name":"annotation_gff3"},"RenameDatasetActionannotation_json":{"action_arguments":{"newname":"bakta_annotation_json"},"action_type":"RenameDatasetAction","output_name":"annotation_json"},"RenameDatasetActionannotation_plot":{"action_arguments":{"newname":"bakta_annotation_plot"},"action_type":"RenameDatasetAction","output_name":"annotation_plot"},"RenameDatasetActionannotation_tsv":{"action_arguments":{"newname":"bakta_annotation_tabular"},"action_type":"RenameDatasetAction","output_name":"annotation_tsv"},"RenameDatasetActionhypotheticals_faa":{"action_arguments":{"newname":"bakta_hypothetical_faa"},"action_type":"RenameDatasetAction","output_name":"hypotheticals_faa"},"RenameDatasetActionhypotheticals_tsv":{"action_arguments":{"newname":"bakta_hypothetical_tabular"},"action_type":"RenameDatasetAction","output_name":"hypotheticals_tsv"},"RenameDatasetActionsummary_txt":{"action_arguments":{"newname":"bakta_summary_text"},"action_type":"RenameDatasetAction","output_name":"summary_txt"},"TagDatasetActionannotation_embl":{"action_arguments":{"tags":"bakta_annotation_embl"},"action_type":"TagDatasetAction","output_name":"annotation_embl"},"TagDatasetActionannotation_faa":{"action_arguments":{"tags":"bakta_aminoacid_sequence_faa"},"action_type":"TagDatasetAction","output_name":"annotation_faa"},"TagDatasetActionannotation_ffn":{"action_arguments":{"tags":"bakta_nucleotide_sequence_fasta"},"action_type":"TagDatasetAction","output_name":"annotation_ffn"},"TagDatasetActionannotation_fna":{"action_arguments":{"tags":"bakta_assembly_fasta"},"action_type":"TagDatasetAction","output_name":"annotation_fna"},"TagDatasetActionannotation_gbff":{"action_arguments":{"tags":"bakta_annotation_gbff"},"action_type":"TagDatasetAction","output_name":"annotation_gbff"},"TagDatasetActionannotation_gff3":{"action_arguments":{"tags":"bakta_annotation_gff3"},"action_type":"TagDatasetAction","output_name":"annotation_gff3"},"TagDatasetActionannotation_json":{"action_arguments":{"tags":"bakta_annotation_json"},"action_type":"TagDatasetAction","output_name":"annotation_json"},"TagDatasetActionannotation_plot":{"action_arguments":{"tags":"bakta_annotation_plot"},"action_type":"TagDatasetAction","output_name":"annotation_plot"},"TagDatasetActionannotation_tsv":{"action_arguments":{"tags":"bakta_annotation_tabular"},"action_type":"TagDatasetAction","output_name":"annotation_tsv"},"TagDatasetActionhypotheticals_faa":{"action_arguments":{"tags":"bakta_hypothetical_faa"},"action_type":"TagDatasetAction","output_name":"hypotheticals_faa"},"TagDatasetActionhypotheticals_tsv":{"action_arguments":{"tags":"bakta_hypothetical_tabular"},"action_type":"TagDatasetAction","output_name":"hypotheticals_tsv"},"TagDatasetActionsummary_txt":{"action_arguments":{"tags":"bakta_summary_text"},"action_type":"TagDatasetAction","output_name":"summary_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.4+galaxy0","tool_shed_repository":{"changeset_revision":"d77802fe76f7","name":"bakta","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"annotation\\": {\\"complete\\": false, \\"prodigal\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"translation_table\\": \\"11\\", \\"keep_contig_headers\\": false, \\"replicons\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"compliant\\": false, \\"proteins\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"meta\\": false, \\"regions\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"input_option\\": {\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_contig_length\\": null, \\"bakta_db_select\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"amrfinder_db_select\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"organism\\": {\\"genus\\": null, \\"species\\": null, \\"strain\\": null, \\"plasmid\\": null}, \\"output_files\\": {\\"output_selection\\": [\\"file_tsv\\", \\"file_gff3\\", \\"file_ffn\\", \\"file_plot\\", \\"file_gbff\\", \\"file_embl\\", \\"file_fna\\", \\"file_faa\\", \\"hypo_tsv\\", \\"hypo_fa\\", \\"file_json\\", \\"sum_txt\\"]}, \\"workflow\\": {\\"skip_analysis\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.9.4+galaxy0","type":"tool","uuid":"0a32dcad-f456-460e-9e6f-9e306a8fd515","when":null,"workflow_outputs":[{"label":"bakta_nucleotide_sequence_fasta","output_name":"annotation_ffn","uuid":"31d239a4-5065-49fb-9989-8b83f3b53642"},{"label":"bakta_aminoacid_sequence_faa","output_name":"annotation_faa","uuid":"3827a853-ad2a-48ad-89dd-648883467abd"},{"label":"bakta_hypothetical_tabular","output_name":"hypotheticals_tsv","uuid":"db4f4056-01ba-42ed-b053-e3c6086011ab"},{"label":"bakta_assembly_fasta","output_name":"annotation_fna","uuid":"7d88510c-813b-40f7-8fd6-5b079ff11ab2"},{"label":"bakta_annotation_tabular","output_name":"annotation_tsv","uuid":"b84d6b52-d921-42d5-a7df-4abbd523bdc7"},{"label":"bakta_annotation_gff3","output_name":"annotation_gff3","uuid":"3af64163-439d-4241-b950-f540e6f18134"},{"label":"bakta_annotation_gbff","output_name":"annotation_gbff","uuid":"5a6c32eb-42ca-45c0-afc9-a12cc777247d"},{"label":"bakta_annotation_embl","output_name":"annotation_embl","uuid":"e3b70765-90ee-47e9-8870-411e6a263963"},{"label":"bakta_hypothetical_faa","output_name":"hypotheticals_faa","uuid":"4aa10d6e-0577-45b3-8759-63f1e0712b0e"},{"label":"bakta_annotation_json","output_name":"annotation_json","uuid":"4ab292e3-0eb7-4746-b2aa-e03c2679aa2b"},{"label":"bakta_annotation_plot","output_name":"annotation_plot","uuid":"c5b8dc03-6ee6-4d0b-8703-f92b15165d77"},{"label":"bakta_summary_text","output_name":"summary_txt","uuid":"12e3ec34-9d84-4924-a8a6-0bc41a9f23f1"}]},"8":{"annotation":"ToolDistillator extracts results from tools and creates a JSON file for each tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","errors":null,"id":8,"input_connections":{"tool_section|tools_0|select_tool|genome_hit_path":{"id":6,"output_name":"hit_file"},"tool_section|tools_0|select_tool|input":{"id":6,"output_name":"json_file"},"tool_section|tools_0|select_tool|plasmid_hit_path":{"id":6,"output_name":"plasmid_file"},"tool_section|tools_0|select_tool|plasmid_result_tabular_path":{"id":6,"output_name":"result_file"},"tool_section|tools_0|select_tool|reference_database_version":{"id":1,"output_name":"output"},"tool_section|tools_1|select_tool|annotation_path":{"id":4,"output_name":"annotation"},"tool_section|tools_1|select_tool|input":{"id":4,"output_name":"all_results"},"tool_section|tools_1|select_tool|is_fna_path":{"id":4,"output_name":"is_fasta"},"tool_section|tools_1|select_tool|orf_faa_path":{"id":4,"output_name":"orf_faa"},"tool_section|tools_1|select_tool|orf_fna_path":{"id":4,"output_name":"orf_fna"},"tool_section|tools_1|select_tool|summary_path":{"id":4,"output_name":"summary"},"tool_section|tools_2|select_tool|input":{"id":5,"output_name":"integrons_table"},"tool_section|tools_2|select_tool|summary_file_path":{"id":5,"output_name":"summary"},"tool_section|tools_3|select_tool|amino_acid_annotation_path":{"id":7,"output_name":"annotation_faa"},"tool_section|tools_3|select_tool|annotation_embl_path":{"id":7,"output_name":"annotation_embl"},"tool_section|tools_3|select_tool|annotation_genbank_path":{"id":7,"output_name":"annotation_gbff"},"tool_section|tools_3|select_tool|annotation_tabular_path":{"id":7,"output_name":"annotation_tsv"},"tool_section|tools_3|select_tool|contig_sequences_path":{"id":7,"output_name":"annotation_fna"},"tool_section|tools_3|select_tool|gff_file_path":{"id":7,"output_name":"annotation_gff3"},"tool_section|tools_3|select_tool|hypothetical_protein_path":{"id":7,"output_name":"hypotheticals_faa"},"tool_section|tools_3|select_tool|hypothetical_tabular_path":{"id":7,"output_name":"hypotheticals_tsv"},"tool_section|tools_3|select_tool|input":{"id":7,"output_name":"annotation_json"},"tool_section|tools_3|select_tool|nucleotide_annotation_path":{"id":7,"output_name":"annotation_ffn"},"tool_section|tools_3|select_tool|plot_file_path":{"id":7,"output_name":"annotation_plot"},"tool_section|tools_3|select_tool|reference_database_version":{"id":2,"output_name":"output"},"tool_section|tools_3|select_tool|summary_result_path":{"id":7,"output_name":"summary_txt"}},"inputs":[],"label":"ToolDistillator","name":"ToolDistillator","outputs":[{"name":"output_json","type":"input"}],"position":{"left":1521.7059634117236,"top":515.5903476661389},"post_job_actions":{"RenameDatasetActionoutput_json":{"action_arguments":{"newname":"tooldistillator_results"},"action_type":"RenameDatasetAction","output_name":"output_json"},"TagDatasetActionoutput_json":{"action_arguments":{"tags":"tooldistillator_results"},"action_type":"TagDatasetAction","output_name":"output_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"ea93df4b3df2","name":"tooldistillator","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"log\\": false, \\"tool_section\\": {\\"tools\\": [{\\"__index__\\": 0, \\"select_tool\\": {\\"tool_list\\": \\"plasmidfinder\\", \\"__current_case__\\": 14, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"plasmid_result_tabular_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"genome_hit_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"plasmid_hit_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"select_tool\\": {\\"tool_list\\": \\"isescan\\", \\"__current_case__\\": 11, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"summary_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"orf_fna_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"orf_faa_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_fna_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 2, \\"select_tool\\": {\\"tool_list\\": \\"integronfinder2\\", \\"__current_case__\\": 10, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"summary_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 3, \\"select_tool\\": {\\"tool_list\\": \\"bakta\\", \\"__current_case__\\": 2, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation_tabular_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation_genbank_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"annotation_embl_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"contig_sequences_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hypothetical_protein_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"hypothetical_tabular_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"plot_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"summary_result_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"nucleotide_annotation_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"amino_acid_annotation_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gff_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"678bdbfe-7a3a-4a91-8ea7-1cb17f53daa6","when":null,"workflow_outputs":[{"label":"tooldistillator_results","output_name":"output_json","uuid":"a8db4b4d-8573-45bf-990c-304868ad5150"}]},"9":{"annotation":"ToolDistillator summarize groups all JSON file into a unique JSON file","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","errors":null,"id":9,"input_connections":{"summarize_data":{"id":8,"output_name":"output_json"}},"inputs":[],"label":"ToolDistillator summarize","name":"ToolDistillator Summarize","outputs":[{"name":"summary_json","type":"json"}],"position":{"left":1999.7866424448703,"top":1179.174084225585},"post_job_actions":{"RenameDatasetActionsummary_json":{"action_arguments":{"newname":"tooldistillator_summarize"},"action_type":"RenameDatasetAction","output_name":"summary_json"},"TagDatasetActionsummary_json":{"action_arguments":{"tags":"tooldistillator_summarize"},"action_type":"TagDatasetAction","output_name":"summary_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"24eec94e6dfc","name":"tooldistillator_summarize","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"summarize_data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"3427b1f1-79ce-4ade-a350-baf15ac250c2","when":null,"workflow_outputs":[{"label":"tooldistillator_summarize","output_name":"summary_json","uuid":"96d63721-4cb5-4879-8984-4eeba8dec36b"}]}},"tags":["Genomics","fasta","ABRomics","bacterial-genomics","Annotation","genome-annotation"],"uuid":"d55a9e46-a205-4143-b92f-3783499063ed","version":1},"readme":"# Bacterial genome annotation workflow (v1.0)\\n\\nThis workflow uses assembled bacterial genome fasta files (but can be any fasta file) and executes the following steps:\\n1. Genomic annotation\\n    - **Bakta** to predict CDS and small proteins (sORF)\\n2. Integron identification\\n    - **IntegronFinder2** to identify CALIN elements, In0 elements, and complete integrons\\n3. Plasmid gene identification\\n    - **Plasmidfinder** to identify and typing plasmid sequences\\n4. Inserted sequence (IS) detection\\n    - **ISEScan** to detect IS elements\\n5. Aggregating outputs into a single JSON file\\n    - **ToolDistillator** to extract and aggregate information from different tool outputs to JSON parsable files\\n\\n## Inputs\\n\\n1. Assembled bacterial genome in fasta format.\\n\\n## Outputs\\n\\n1. Genomic annotation:\\n    - genome annotation in tabular, gff and several other formats\\n    - annotation plot\\n    - nucleotide and protein sequences identified\\n    - summary of genomic identified elements\\n2. Integron identification:\\n    - integron identification in tabular format and a summary\\n3. Plasmid gene identification:\\n    - plasmid gene identified and associated blast hits\\n4. Inserted Element (IS) detection:\\n    - IS element list in tabular format\\n    - is hits in fasta format\\n    - ORF hits in protein and nucleotide fasta format\\n    - IS annotation gff format\\n5. Aggregating outputs:\\n    - JSON file with information about the outputs of **Bakta**, **IntegronFinder2**, **Plasmidfinder**, **ISEScan**","changelog":"# Changelog\\n\\n## [1.1.7] - 2024-10-21\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1`\\n\\n## [1.1.6] - 2024-10-11\\n\\n### Manual update\\n\\n- Changed input parameters to select databases in WFs. Use \\"Attempt restriction based on connections\\" instead of \\"Provide list of suggested values\\".\\n\\n## [1.1.5] - 2024-09-30\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0`\\n\\n## [1.1.4] - 2024-09-19\\n\\n### Manual update\\n\\n- Updated the tool versions manually in the ToolDistillator parameters\\n\\n## [1.1.3] - 2024-09-09\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.4+galaxy0`\\n\\n## [1.1.2] - 2024-07-19\\n\\n- PlasmidFinder database correction in .ga\\n\\n## [1.1.1] - 2024-07-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0`\\n\\n## [1.1] - 2024-06-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.3+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.5+galaxy0`\\n\\n## [1.0] - 14-06-2024\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## bacterial_genome_annotation\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input sequence fasta\\"]@{ shape: doc }\\n1[\\"ℹ️ Select a plasmid detection database\\"]@{ shape: lean-l }\\n2[\\"ℹ️ Select a bacterial genome annotation database\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Select a AMRFinderPlus database\\"]@{ shape: lean-l }\\n4[\\"genomic_annotation_insertionelement_isescan\\"]@{ shape: process }\\n0 --> 4\\n5[\\"genomic_annotation_integron\\"]@{ shape: process }\\n0 --> 5\\n6[\\"genomic_annotation_plasmid_plasmidfinder\\"]@{ shape: process }\\n1 --> 6\\n0 --> 6\\n7[\\"Bakta\\"]@{ shape: process }\\n3 --> 7\\n2 --> 7\\n0 --> 7\\n8[\\"ToolDistillator\\"]@{ shape: process }\\n6 --> 8\\n6 --> 8\\n6 --> 8\\n6 --> 8\\n1 --> 8\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n5 --> 8\\n5 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n2 --> 8\\n7 --> 8\\n9[\\"ToolDistillator summarize\\"]@{ shape: process }\\n8 --> 9\\n```\\n","updated":"2024-10-21T00:00:00","trsID":"#workflow/github.com/iwc-workflows/bacterial_genome_annotation/main","dockstore_id":27296,"categories":[],"collections":["Genome Annotation"],"doi":"10.5281/zenodo.15078495","tests":[{"doc":"Test outline for bacterial_genome_annotation.ga","job":{"Input sequence fasta":{"class":"File","path":"https://zenodo.org/records/11488310/files/shovill_contigs_fasta"},"Select a plasmid detection database":"plasmidfinder_81c11f4_2023_12_04","Select a bacterial genome annotation database":"V5.0_2023-02-20","Select a AMRFinderPlus database":"amrfinderplus_V3.12_2024-05-02.2"},"outputs":{"integronfinder2_logfile_text":{"asserts":{"has_text":{"text":"Writing out results for replicon"}}},"integronfinder2_summary":{"asserts":{"has_text":{"text":"contig00001"}}},"integronfinder2_results_tabular":{"asserts":{"has_text":{"text":"contig00009_42"}}},"bakta_hypothetical_tabular":{"asserts":{"has_text":{"text":"DHJLLP_04750"}}},"bakta_annotation_json":{"asserts":{"has_text":{"text":"aa_hexdigest"}}},"bakta_annotation_tabular":{"asserts":{"has_text":{"text":"Phosphotransferase system cellobiose-specific component IIC"}}},"isescan_results_tabular":{"asserts":{"has_text":{"text":"IS256_162"}}},"isescan_summary_tabular":{"asserts":{"has_text":{"text":"nIS"}}},"isescan_logfile_text":{"asserts":{"has_text":{"text":"Both complete and partial IS elements are reported."}}},"plasmidfinder_result_json":{"asserts":{"has_text":{"text":"positions_in_contig"}}},"plasmidfinder_results_tabular":{"asserts":{"has_n_columns":{"n":8}}},"tooldistillator_summarize":{"asserts":[{"that":"has_text","text":"CDS12738(DOp1)"},{"that":"has_text","text":"PFAM"}]}}}]}],"path":"./workflows/bacterial_genomics/bacterial_genome_annotation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/RepeatMasking-Workflow.ga","testParameterFiles":["/RepeatMasking-Workflow-tests.yml"],"authors":[{"name":"Romane Libouban","email":"romane.libouban@irisa.fr"}],"definition":{"a_galaxy_workflow":"true","annotation":"","format-version":"0.1","license":"MIT","release":"0.1","name":"Repeat masking with RepeatModeler and RepeatMasker","creator":[{"class":"Person","email":"mailto:romane.libouban@irisa.fr","name":"Romane Libouban"}],"steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Apply repeat masking to this fasta file","name":"input"}],"label":"input","name":"Input dataset","outputs":[],"position":{"left":10,"top":10},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"ab5e19b0-ce35-4e54-a55e-f75243c86e3d","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/csbl/repeatmodeler/repeatmodeler/2.0.4+galaxy1","errors":null,"id":1,"input_connections":{"input_file":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"RepeatModeler","outputs":[{"name":"sequences","type":"fasta"},{"name":"seeds","type":"stockholm"}],"position":{"left":230,"top":10},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/csbl/repeatmodeler/repeatmodeler/2.0.4+galaxy1","tool_shed_repository":{"changeset_revision":"8661b2607b7e","name":"repeatmodeler","owner":"csbl","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"input_file\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0.4+galaxy1","type":"tool","uuid":"9312ba36-4275-4d40-8ba6-95eea1b23b11","when":null,"workflow_outputs":[{"output_name":"sequences","label":"RepeatModeler consensus sequences"},{"output_name":"seeds","label":"RepeatModeler seeds alignments"}]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/4.1.5+galaxy0","errors":null,"id":2,"input_connections":{"input_fasta":{"id":1,"output_name":"sequences"}},"inputs":[],"label":null,"name":"RepeatMasker","outputs":[{"name":"output_masked_genome","type":"fasta"},{"name":"output_log","type":"tabular"},{"name":"output_table","type":"txt"},{"name":"output_repeat_catalog","type":"txt"},{"name":"output_gff","type":"gff"}],"position":{"left":450,"top":10},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/repeat_masker/repeatmasker_wrapper/4.1.5+galaxy0","tool_shed_repository":{"changeset_revision":"ba6d2c32f797","name":"repeat_masker","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"input\\", \\"advanced\\": {\\"is_only\\": false, \\"is_clip\\": false, \\"no_is\\": false, \\"rodspec\\": false, \\"primspec\\": false, \\"nolow\\": false, \\"noint\\": false, \\"norna\\": false, \\"alu\\": false, \\"div\\": false, \\"search_speed\\": \\"\\", \\"frag\\": \\"40000\\", \\"gc\\": null, \\"gccalc\\": false, \\"nocut\\": false, \\"xout\\": false, \\"keep_alignments\\": false, \\"invert_alignments\\": false, \\"poly\\": false}, \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"excln\\": true, \\"gff\\": true, \\"input_fasta\\": null, \\"repeat_source\\": {\\"source_type\\": \\"dfam\\", \\"__current_case__\\": 0, \\"species_source\\": {\\"species_from_list\\": \\"no\\", \\"__current_case__\\": 1, \\"species_name\\": \\"\\"}}, \\"xsmall\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.1.5+galaxy0","type":"tool","uuid":"e6c8e6a1-efe8-4291-b12b-5fdb3795b6ca","when":null,"workflow_outputs":[{"output_name":"output_masked_genome","label":"RepeatMasker masked genome"},{"output_name":"output_log","label":"RepeatMasker output log"},{"output_name":"output_table","label":"RepeatMasker repeat statistics"},{"output_name":"output_repeat_catalog","label":"RepeatMasker repeat catalog"},{"output_name":"output_gff","label":"RepeatMasker repeat annotation"}]}},"tags":[],"uuid":"f25be8fa-7823-456f-9707-a497703f48d7","version":0},"readme":"# RepeatMasking Workflow\\n\\nThis workflow uses RepeatModeler and RepeatMasker for genome analysis.\\n\\n- RepeatModeler is a software package for identifying and modeling de novo families of transposable elements (TEs). At the heart of RepeatModeler are three de novo repeat search programs (RECON, RepeatScout and LtrHarvest/Ltr_retriever) which use complementary computational methods to identify repeat element boundaries and family relationships from sequence data.\\n\\n- RepeatMasker is a program that analyzes DNA sequences for *interleaved repeats* and *low-complexity* DNA sequences. The result of the program is a detailed annotation of the repeats present in the query sequence, as well as a modified version of the query sequence in which all annotated repeats are present.\\n\\n## Input dataset for RepeatModeler\\n- RepeatModeler requires a single input file, a genome in fasta format.\\n\\n\\n## Outputs dataset for RepeatModeler\\n- Two output files are generated:\\n    - summary file (.tbl)\\n    - fasta file containing alignments in order of appearance in the query sequence\\n\\n\\n## Input dataset for RepeatMasker\\n- ReapatMasker requires the fasta file generated by RepeatModeler\\n\\n## Outputs datasets for RepeatMasker\\n- Five output files are generated:\\n    - a fasta file\\n    - .gff3 file\\n    - a table summarizing the repeated content of the sequence analyzed\\n    - a file with statistics related to the repeated content of the sequence analyzed\\n    - a summary of the mutation sites found and the order of grouping\\n    \\n","changelog":"# Changelog\\n\\n## [0.1] - 2023-09-21\\n\\nInitial version of the RepeatMasking workflow for genomic sequencing data.\\n","diagrams":"# Workflow diagrams\\n\\n## Repeat masking with RepeatModeler and RepeatMasker\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ input\\"]@{ shape: doc }\\n1[\\"RepeatModeler\\"]@{ shape: process }\\n0 --> 1\\n2[\\"RepeatMasker\\"]@{ shape: process }\\n1 --> 2\\n```\\n","updated":"2023-09-21T00:00:00","trsID":"#workflow/github.com/iwc-workflows/repeatmasking/main","dockstore_id":23726,"categories":[],"collections":["Genome Annotation"],"doi":null,"tests":[{"doc":"Test outline for RepeatMasking Workflow","job":{"input":{"class":"File","location":"https://zenodo.org/record/8364146/files/eco.fasta?download=1","filetype":"fasta"}},"outputs":{"RepeatModeler consensus sequences":{"location":"https://zenodo.org/record/8364146/files/repeatmodeler_output_sequences.fasta?download=1","compare":"sim_size","delta":30000},"RepeatModeler seeds alignments":{"location":"https://zenodo.org/record/8364146/files/repeatmodeler_output_seeds.stockholm?download=1","compare":"sim_size","delta":90000000},"RepeatMasker masked genome":{"location":"https://zenodo.org/record/8364146/files/repeatmasker_output_masked_genome.fasta?download=1","compare":"sim_size","delta":30000},"RepeatMasker output log":{"location":"https://zenodo.org/record/8364146/files/repeatmasker_output_log.tabular?download=1","compare":"sim_size","delta":30000},"RepeatMasker repeat statistics":{"location":"https://zenodo.org/record/8364146/files/repeatmasker_output_table.txt?download=1","compare":"sim_size","delta":30000},"RepeatMasker repeat catalog":{"location":"https://zenodo.org/record/8364146/files/repeatmasker_output_repeat_catalog.txt?download=1","compare":"sim_size","delta":30000},"RepeatMasker repeat annotation":{"location":"https://zenodo.org/record/8364146/files/repeatmasker_output_gff.gff?download=1","compare":"sim_size","delta":30000}}}]}],"path":"./workflows/repeatmasking"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Mass-spectrometry__GCMS-with-metaMS.ga","testParameterFiles":["/Mass-spectrometry__GCMS-with-metaMS-tests.yml"],"authors":[{"name":"workflow4metabolomics","image":"https://raw.githubusercontent.com/workflow4metabolomics/workflow4metabolomics/master/images/logo/logo_w4m-0.1-black-orange.png","url":"https://workflow4metabolomics.org/"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract and the metaMS R package (Wehrens, R 2014) for the field of untargeted metabolomics. \\n\\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html","creator":[{"class":"Organization","image":"https://raw.githubusercontent.com/workflow4metabolomics/workflow4metabolomics/master/images/logo/logo_w4m-0.1-black-orange.png","name":"workflow4metabolomics","url":"https://workflow4metabolomics.org/"}],"format-version":"0.1","release":"0.1","license":"MIT","name":"Mass spectrometry: GCMS with metaMS","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Mass-spectrometry Dataset Collection"}],"label":"Mass-spectrometry Dataset Collection","name":"Input dataset collection","outputs":[],"position":{"left":15,"top":101},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"27aea272-13ef-4ab2-81ef-329830985fe0","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"sampleMetadata"}],"label":"sampleMetadata","name":"Input dataset","outputs":[],"position":{"left":0,"top":314},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"a3d47eb1-e815-4219-86c0-23ce34b662d7","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy2","errors":null,"id":2,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool MSnbase readMSData","name":"input"}],"label":null,"name":"MSnbase readMSData","outputs":[{"name":"xsetRData","type":"rdata.msnbase.raw"}],"position":{"left":295,"top":80},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy2","tool_shed_repository":{"changeset_revision":"12def6edac2f","name":"msnbase_readmsdata","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.16.1+galaxy2","type":"tool","uuid":"60192f76-c4a8-4505-a24e-e730fe87694e","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_xcmsset/abims_xcms_xcmsSet/3.12.0+galaxy0","errors":null,"id":3,"input_connections":{"image":{"id":2,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms findChromPeaks (xcmsSet)","outputs":[{"name":"xsetRData","type":"rdata.xcms.findchrompeaks"},{"name":"log","type":"txt"}],"position":{"left":575,"top":311},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_xcmsset/abims_xcms_xcmsSet/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"b02d1992a43a","name":"xcms_xcmsset","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"filterSection\\": {\\"filterAcquisitionNum\\": \\"\\", \\"filterRt\\": \\"\\", \\"filterMz\\": \\"\\"}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"method\\": \\"MatchedFilter\\", \\"__current_case__\\": 1, \\"fwhm\\": \\"5\\", \\"binSize\\": \\"0.5\\", \\"MatchedFilterAdv\\": {\\"impute_cond\\": {\\"impute\\": \\"none\\", \\"__current_case__\\": 0}, \\"sigma\\": null, \\"max\\": \\"500\\", \\"snthresh\\": \\"2\\", \\"steps\\": \\"2\\", \\"mzdiff\\": \\"0.5\\", \\"peaklist\\": false}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"117244c7-a9b0-4143-bfba-48c6cca56ee5","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","errors":null,"id":4,"input_connections":{"images":{"id":3,"output_name":"xsetRData"},"sampleMetadata":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"xcms plot chromatogram","outputs":[{"name":"ticsPdf","type":"pdf"},{"name":"bpisPdf","type":"pdf"}],"position":{"left":915,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"024974037c4e","name":"xcms_plot_chromatogram","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"rdata.xcms.findchrompeaks\\", \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"images\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampleMetadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"a5e5e89e-4521-49b4-9554-1004cefb3842","when":null,"workflow_outputs":[{"label":null,"output_name":"bpisPdf","uuid":"98f84d78-fb87-4ba8-9781-d4935013f052"},{"label":null,"output_name":"ticsPdf","uuid":"3a9785ad-afad-410f-8f17-6f7b9644f37b"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_merge/xcms_merge/3.12.0+galaxy0","errors":null,"id":5,"input_connections":{"images":{"id":3,"output_name":"xsetRData"},"sampleMetadata":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"xcms findChromPeaks Merger","outputs":[{"name":"xsetRData","type":"rdata.xcms.findchrompeaks"}],"position":{"left":861,"top":521},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_merge/xcms_merge/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"5bd125a3f3b0","name":"xcms_merge","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"rdata.xcms.findchrompeaks\\", \\"chromInfo\\": \\"/shared/ifbstor1/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"images\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampleMetadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"cb2b0664-f992-4e0b-8a40-58c83a0e47f9","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/yguitton/metams_rungc/metams_runGC/3.0.0+metaMS1.24.0-galaxy0","errors":null,"id":6,"input_connections":{"input":{"id":5,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"metaMS.runGC","outputs":[{"name":"peaktable","type":"tabular"},{"name":"sampleMetadata","type":"tabular"},{"name":"variableMetadata","type":"tabular"},{"name":"dataMatrix","type":"tabular"},{"name":"peakspectra","type":"txt"},{"name":"rungcRData","type":"rdata"}],"position":{"left":1140,"top":202},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/yguitton/metams_rungc/metams_runGC/3.0.0+metaMS1.24.0-galaxy0","tool_shed_repository":{"changeset_revision":"93508ea69eb5","name":"metams_rungc","owner":"yguitton","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"options_rifilter\\": {\\"option\\": \\"false\\", \\"__current_case__\\": 1}, \\"settings\\": {\\"setting\\": \\"usersettings\\", \\"__current_case__\\": 1, \\"options_rtrange\\": {\\"option\\": \\"hide\\", \\"__current_case__\\": 1}, \\"rtdiff\\": \\"0.05\\", \\"minfeat\\": \\"5\\", \\"simthreshold\\": \\"0.7\\", \\"minclassfraction\\": \\"0.5\\", \\"minclasssize\\": \\"3\\", \\"options_db\\": {\\"option\\": \\"hide\\", \\"__current_case__\\": 1}, \\"options_ri\\": {\\"option\\": \\"hide\\", \\"__current_case__\\": 1}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.0.0+metaMS1.24.0-galaxy0","type":"tool","uuid":"4f636be8-c31f-474e-8c47-8fd644710863","when":null,"workflow_outputs":[{"label":"metaMS.runGC dataMatrix","output_name":"dataMatrix","uuid":"2ad0157b-8587-47ba-8481-3e8b6c2fe7f2"},{"label":null,"output_name":"peaktable","uuid":"2f467902-d943-431b-89d5-731911e06f2d"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/ethevenot/checkformat/checkFormat/3.0.0","errors":null,"id":7,"input_connections":{"dataMatrix_in":{"id":6,"output_name":"dataMatrix"},"sampleMetadata_in":{"id":1,"output_name":"output"},"variableMetadata_in":{"id":6,"output_name":"variableMetadata"}},"inputs":[],"label":null,"name":"Check Format","outputs":[{"name":"dataMatrix_out","type":"tabular"},{"name":"sampleMetadata_out","type":"tabular"},{"name":"variableMetadata_out","type":"tabular"},{"name":"information","type":"txt"}],"position":{"left":1496,"top":281},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/ethevenot/checkformat/checkFormat/3.0.0","tool_shed_repository":{"changeset_revision":"e7c5811ec12f","name":"checkformat","owner":"ethevenot","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"dataMatrix_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"makeNameL\\": \\"FALSE\\", \\"sampleMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"variableMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.0.0","type":"tool","uuid":"a3701788-b7fd-4eb9-adf1-d3834a935567","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/ethevenot/multivariate/Multivariate/2.3.10","errors":null,"id":8,"input_connections":{"dataMatrix_in":{"id":7,"output_name":"dataMatrix_out"},"sampleMetadata_in":{"id":7,"output_name":"sampleMetadata_out"},"variableMetadata_in":{"id":7,"output_name":"variableMetadata_out"}},"inputs":[],"label":null,"name":"Multivariate","outputs":[{"name":"sampleMetadata_out","type":"tabular"},{"name":"variableMetadata_out","type":"tabular"},{"name":"figure","type":"pdf"},{"name":"information","type":"txt"}],"position":{"left":1827,"top":293},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/ethevenot/multivariate/Multivariate/2.3.10","tool_shed_repository":{"changeset_revision":"5526f8258e8a","name":"multivariate","owner":"ethevenot","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advCpt\\": {\\"opcC\\": \\"full\\", \\"__current_case__\\": 1, \\"scaleC\\": \\"standard\\", \\"permI\\": \\"20\\", \\"log10L\\": \\"FALSE\\", \\"algoC\\": \\"default\\", \\"crossvalI\\": \\"3\\"}, \\"advGph\\": {\\"opgC\\": \\"full\\", \\"__current_case__\\": 1, \\"typeC\\": \\"summary\\", \\"parMahalC\\": \\"NA\\", \\"parAsColC\\": \\"class\\", \\"parLabVc\\": \\"none\\", \\"parPc1I\\": \\"1\\", \\"parPc2I\\": \\"2\\", \\"parCexN\\": \\"0.8\\"}, \\"dataMatrix_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"orthoI\\": \\"0\\", \\"predI\\": \\"NA\\", \\"respC\\": \\"none\\", \\"sampleMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"save_rdata\\": false, \\"testL\\": \\"FALSE\\", \\"variableMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.3.10","type":"tool","uuid":"52e89249-a543-441c-8da6-60b7e40178b2","when":null,"workflow_outputs":[{"label":"Multivariate variableMetadata","output_name":"variableMetadata_out","uuid":"376507dd-94ff-4bf1-8481-5fe5ed61918e"},{"label":"Multivariate sampleMetadata","output_name":"sampleMetadata_out","uuid":"d574dc78-9310-44e9-a6c3-b02b104e598c"}]}},"tags":["metabolomics","MS","workflow4metabolomics","GC-MS","GTN","metaMS"],"uuid":"a2b0d4f7-164c-40d3-ac34-f248db18bbef","version":2},"readme":"# Mass spectrometry: GCMS with metaMS \\n\\nThis workflow uses the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) to extract, filter, align and fill gaps, and uses the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html) to annotate isotopes, adducts and fragments.\\n\\nThis workflow is composed with the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) to extract and the metaMS R package [(Wehrens, R 2014)](https://bioconductor.org/packages/release/bioc/html/metaMS.html) for the field of untargeted metabolomics. \\n\\n🎓 For more information see the [Galaxy Training Network tutorial: Mass spectrometry: GC-MS analysis with metaMS package](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gcms/tutorial.html)\\n\\n## Inputs\\n### sampleMetadata\\nThe sampleMetadata tabular file corresponds to a table containing information about your samples\\n\\nA sample metadata file contains various information for each of your raw files:\\n- Classes which will be used during the preprocessing steps\\n- Analytical batches which will be useful for a batch correction step, along with sample types (pool/sample) and injection order\\n- Different experimental conditions which can be used for statistics\\n- Any information about samples that you want to keep, in a column format\\n\\nThe content of your sample metadata file has to be filled by you, since it is not contained in your raw data. Note that you can either:\\n- Upload an existing metadata file\\n- Use a template to create one (because it can be painful to get the sample list without misspelling or omission)\\n  - Generate a template with the `xcms get a sampleMetadata file` tool available in Galaxy\\n  - Fill it using your favorite table editor (Excel, LibreOffice)\\n  - Upload it within Galaxy\\n\\n**Formats:** tab-separated values as tsv, tab, txt, ...\\n\\n### Mass-spectrometry Dataset Collection\\nMass-spectrometry data files gathered in a Galaxy Dataser Collection\\n\\n**Formats:** open format as mzXML, mzMl, mzData and netCDF\\n\\n## Main steps\\n1. MSnbase readMSData: read the mzXML and prepare for xcms\\n2. XCMS findChromPeaks: peak picking\\n3. metaMS.runGC: definition of pseudo-spectra\\n","changelog":"# Changelog\\n\\nAll notable changes to this project will be documented in this file.\\n\\n## [0.1] - 2023-11-22\\n\\nFirst release\\n","diagrams":"# Workflow diagrams\\n\\n## Mass spectrometry: GCMS with metaMS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Mass-spectrometry Dataset Collection\\"]@{ shape: docs }\\n1[\\"ℹ️ sampleMetadata\\"]@{ shape: doc }\\n2[\\"MSnbase readMSData\\"]@{ shape: process }\\n0 --> 2\\n3[\\"xcms findChromPeaks (xcmsSet)\\"]@{ shape: process }\\n2 --> 3\\n4[\\"xcms plot chromatogram\\"]@{ shape: process }\\n3 --> 4\\n1 --> 4\\n5[\\"xcms findChromPeaks Merger\\"]@{ shape: process }\\n3 --> 5\\n1 --> 5\\n6[\\"metaMS.runGC\\"]@{ shape: process }\\n5 --> 6\\n7[\\"Check Format\\"]@{ shape: process }\\n6 --> 7\\n1 --> 7\\n6 --> 7\\n8[\\"Multivariate\\"]@{ shape: process }\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n```\\n","updated":"2023-11-22T00:00:00","trsID":"#workflow/github.com/iwc-workflows/gcms-metams/main","dockstore_id":24516,"categories":[],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078864","tests":[{"doc":"Test outline for Mass-spectrometry__GCMS-with-metaMS.ga","job":{"sampleMetadata":{"class":"File","location":"https://zenodo.org/record/3631074/files/sampleMetadata.tsv","filetype":"tabular"},"Mass-spectrometry Dataset Collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"alg11.mzData","location":"https://zenodo.org/record/3631074/files/alg11.mzData","filetype":"mzdata"},{"class":"File","identifier":"alg2.mzData","location":"https://zenodo.org/record/3631074/files/alg2.mzData","filetype":"mzdata"},{"class":"File","identifier":"alg3.mzData","location":"https://zenodo.org/record/3631074/files/alg3.mzData","filetype":"mzdata"},{"class":"File","identifier":"alg7.mzData","location":"https://zenodo.org/record/3631074/files/alg7.mzData","filetype":"mzdata"},{"class":"File","identifier":"alg8.mzData","location":"https://zenodo.org/record/3631074/files/alg8.mzData","filetype":"mzdata"},{"class":"File","identifier":"alg9.mzData","location":"https://zenodo.org/record/3631074/files/alg9.mzData","filetype":"mzdata"}]}},"outputs":{"metaMS.runGC dataMatrix":{"path":"test-data/metaMS.runGC_dataMatrix.tabular","lines_diff":10},"Multivariate variableMetadata":{"asserts":[{"that":"has_n_lines","n":42},{"that":"has_text","text":"Unknown 1\\tUnknown\\t0.0046\\t19.499\\t-0.17"},{"that":"has_text","text":"Unknown 41\\tUnknown\\t0.0039\\t14.857\\t-0.08"}]},"Multivariate sampleMetadata":{"asserts":[{"that":"has_n_lines","n":7},{"that":"has_text","text":"alg11\\tFWS_100perNaCl\\t-6.2"},{"that":"has_text","text":"alg8\\tFWS_5percNaCL\\t-0.6"}]}}}]}],"path":"./workflows/metabomics/gcms-metams"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Mass_spectrometry__LC-MS_preprocessing_with_XCMS.ga","testParameterFiles":["/Mass_spectrometry__LC-MS_preprocessing_with_XCMS-tests.yml"],"authors":[{"name":"workflow4metabolomics","image":"https://raw.githubusercontent.com/workflow4metabolomics/workflow4metabolomics/master/images/logo/logo_w4m-0.1-black-orange.png","url":"https://workflow4metabolomics.org"}],"definition":{"a_galaxy_workflow":"true","annotation":"This workflow is composed with the XCMS tool R package (Smith, C.A. 2006) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package (Kuhl, C 2012).\\n\\n\\nhttps://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html ","creator":[{"class":"Organization","image":"https://raw.githubusercontent.com/workflow4metabolomics/workflow4metabolomics/master/images/logo/logo_w4m-0.1-black-orange.png","name":"workflow4metabolomics","url":"https://workflow4metabolomics.org"}],"format-version":"0.1","license":"MIT","release":"1.0","name":"Mass spectrometry: LC-MS preprocessing with XCMS","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"SampleMetadata"}],"label":"SampleMetadata","name":"Input dataset","outputs":[],"position":{"left":0,"top":151.0066585286458},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"tabular\\"], \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"bf49de4d-c959-476c-86e1-6ae621bc915c","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"Mass-spectrometry Dataset Collection"}],"label":"Mass-spectrometry Dataset Collection","name":"Input dataset collection","outputs":[],"position":{"left":9,"top":300.0066585286458},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"mzml\\", \\"mzxml\\", \\"mzdata\\", \\"netcdf\\"], \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"93611663-3e2a-41d1-a659-52217d5be454","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy2","errors":null,"id":2,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"MSnbase readMSData","outputs":[{"name":"xsetRData","type":"rdata.msnbase.raw"}],"position":{"left":289,"top":279.0066585286458},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/msnbase_readmsdata/msnbase_readmsdata/2.16.1+galaxy2","tool_shed_repository":{"changeset_revision":"12def6edac2f","name":"msnbase_readmsdata","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.16.1+galaxy2","type":"tool","uuid":"44237c6c-b5fa-42de-b3db-f0be6b59ddf6","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","errors":null,"id":3,"input_connections":{"images":{"id":2,"output_name":"xsetRData"},"sampleMetadata":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"xcms plot chromatogram","outputs":[{"name":"ticsPdf","type":"pdf"},{"name":"bpisPdf","type":"pdf"}],"position":{"left":574.9933374023435,"top":3},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"024974037c4e","name":"xcms_plot_chromatogram","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"images\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampleMetadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"b8f69b68-dd61-41c4-9339-d821b14b50dd","when":null,"workflow_outputs":[{"label":null,"output_name":"ticsPdf","uuid":"4a8d8533-5b39-49e5-9eeb-b6fb29ecfad4"},{"label":null,"output_name":"bpisPdf","uuid":"a483b83e-9fc8-4d06-a5d4-a5d49840016d"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_xcmsset/abims_xcms_xcmsSet/3.12.0+galaxy0","errors":null,"id":4,"input_connections":{"image":{"id":2,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms findChromPeaks (xcmsSet)","outputs":[{"name":"xsetRData","type":"rdata.xcms.findchrompeaks"},{"name":"log","type":"txt"}],"position":{"left":572,"top":264.0066585286458},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_xcmsset/abims_xcms_xcmsSet/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"b02d1992a43a","name":"xcms_xcmsset","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"filterSection\\": {\\"filterAcquisitionNum\\": \\"\\", \\"filterRt\\": \\"\\", \\"filterMz\\": \\"\\"}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"method\\": \\"CentWave\\", \\"__current_case__\\": 0, \\"ppm\\": \\"3.0\\", \\"peakwidth\\": \\"5,20\\", \\"CentWaveAdv\\": {\\"snthresh\\": \\"10\\", \\"prefilter\\": \\"3,5000\\", \\"mzCenterFun\\": \\"wMean\\", \\"integrate\\": \\"1\\", \\"mzdiff\\": \\"-0.001\\", \\"fitgauss\\": false, \\"noise\\": \\"1000\\", \\"verboseColumns\\": false, \\"peaklist\\": false, \\"CentWaveAdvROI\\": {\\"roiList\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"firstBaselineCheck\\": true, \\"roiScales\\": \\"\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"73ed7f35-a7a4-49d8-b2c8-bd1e06f705c3","when":null,"workflow_outputs":[]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_merge/xcms_merge/3.12.0+galaxy0","errors":null,"id":5,"input_connections":{"images":{"id":4,"output_name":"xsetRData"},"sampleMetadata":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"xcms findChromPeaks Merger","outputs":[{"name":"xsetRData","type":"rdata.xcms.findchrompeaks"}],"position":{"left":849,"top":210.0066585286458},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_merge/xcms_merge/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"5bd125a3f3b0","name":"xcms_merge","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"images\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampleMetadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"686849c4-b729-4998-8560-44d34b820ba6","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_group/abims_xcms_group/3.12.0+galaxy0","errors":null,"id":6,"input_connections":{"image":{"id":5,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms groupChromPeaks (group)","outputs":[{"name":"xsetRData","type":"rdata.xcms.group"},{"name":"plotChromPeakDensity","type":"pdf"},{"name":"variableMetadata","type":"tabular"},{"name":"dataMatrix","type":"tabular"}],"position":{"left":1129,"top":176},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_group/abims_xcms_group/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"d45a786cbc40","name":"xcms_group","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"method\\": \\"PeakDensity\\", \\"__current_case__\\": 0, \\"bw\\": \\"5.0\\", \\"minFraction\\": \\"0.9\\", \\"minSamples\\": \\"1\\", \\"binSize\\": \\"0.01\\", \\"PeakDensityAdv\\": {\\"maxFeatures\\": \\"50\\"}}, \\"peaklist\\": {\\"peaklistBool\\": true, \\"__current_case__\\": 0, \\"convertRTMinute\\": false, \\"numDigitsMZ\\": \\"4\\", \\"numDigitsRT\\": \\"0\\", \\"intval\\": \\"into\\", \\"naTOzero\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"7d43b889-0799-4ec1-9b0f-1bff1c3a189b","when":null,"workflow_outputs":[{"label":null,"output_name":"plotChromPeakDensity","uuid":"28f8a44d-db88-4bd8-89b3-bde0bc4948b8"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_retcor/abims_xcms_retcor/3.12.0+galaxy0","errors":null,"id":7,"input_connections":{"image":{"id":6,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms adjustRtime (retcor)","outputs":[{"name":"xsetRData","type":"rdata.xcms.retcor"},{"name":"rawVSadjustedPdf","type":"pdf"}],"position":{"left":1412,"top":232.0066585286458},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_retcor/abims_xcms_retcor/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"aa252eec9229","name":"xcms_retcor","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"method\\": \\"PeakGroups\\", \\"__current_case__\\": 0, \\"minFraction\\": \\"0.7\\", \\"extraPeaks\\": \\"1\\", \\"smooth_cond\\": {\\"smooth\\": \\"loess\\", \\"__current_case__\\": 0, \\"PeakGroupsSmoothLoessAdv\\": {\\"span\\": \\"0.2\\", \\"family\\": \\"gaussian\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"656670ac-dfae-41c8-a551-f3ff1f1bb61f","when":null,"workflow_outputs":[{"label":null,"output_name":"rawVSadjustedPdf","uuid":"0bf0db72-a5f7-40cc-96a8-c200393d558d"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/melpetera/intensity_checks/intens_check/1.2.8","errors":null,"id":8,"input_connections":{"dataMatrix_in":{"id":6,"output_name":"dataMatrix"},"sampleMetadata_in":{"id":0,"output_name":"output"},"variableMetadata_in":{"id":6,"output_name":"variableMetadata"}},"inputs":[],"label":null,"name":"Intensity Check","outputs":[{"name":"variableMetadata_out","type":"tabular"},{"name":"graphs_out","type":"pdf"}],"position":{"left":1366,"top":495},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/melpetera/intensity_checks/intens_check/1.2.8","tool_shed_repository":{"changeset_revision":"a31f3f802b2b","name":"intensity_checks","owner":"melpetera","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__input_ext\\": \\"tabular\\", \\"chromInfo\\": \\"/shared/mfs/data/galaxy/mutable-config/tool-data/shared/ucsc/chrom/?.len\\", \\"dataMatrix_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"method_cond\\": {\\"method\\": \\"each_class\\", \\"__current_case__\\": 1, \\"class_col\\": \\"2\\", \\"chosen_stat\\": [\\"NA\\"], \\"test_fold_cond\\": {\\"test_fold\\": \\"No\\", \\"__current_case__\\": 1}}, \\"sampleMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"variableMetadata_in\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.8","type":"tool","uuid":"fcbec34b-b0d1-4d09-9873-a3c9c95399dc","when":null,"workflow_outputs":[{"label":null,"output_name":"graphs_out","uuid":"80858377-f286-48d8-8b8e-edc70f4fbb95"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","errors":null,"id":9,"input_connections":{"images":{"id":7,"output_name":"xsetRData"},"sampleMetadata":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"xcms plot chromatogram","outputs":[{"name":"ticsPdf","type":"pdf"},{"name":"bpisPdf","type":"pdf"}],"position":{"left":1716.9933374023435,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_plot_chromatogram/xcms_plot_chromatogram/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"024974037c4e","name":"xcms_plot_chromatogram","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"images\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sampleMetadata\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"95ef16c0-4b3f-44de-99b3-2c4ace994d5f","when":null,"workflow_outputs":[{"label":null,"output_name":"ticsPdf","uuid":"72185083-b1c0-4b45-8a91-875873d4fdcb"},{"label":null,"output_name":"bpisPdf","uuid":"15c97a18-a8af-46da-9c2b-38e9126affa4"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_group/abims_xcms_group/3.12.0+galaxy0","errors":null,"id":10,"input_connections":{"image":{"id":7,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms groupChromPeaks (group)","outputs":[{"name":"xsetRData","type":"rdata.xcms.group"},{"name":"plotChromPeakDensity","type":"pdf"},{"name":"variableMetadata","type":"tabular"},{"name":"dataMatrix","type":"tabular"}],"position":{"left":1689,"top":518.0066585286459},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_group/abims_xcms_group/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"d45a786cbc40","name":"xcms_group","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"methods\\": {\\"method\\": \\"PeakDensity\\", \\"__current_case__\\": 0, \\"bw\\": \\"5.0\\", \\"minFraction\\": \\"0.9\\", \\"minSamples\\": \\"1\\", \\"binSize\\": \\"0.01\\", \\"PeakDensityAdv\\": {\\"maxFeatures\\": \\"50\\"}}, \\"peaklist\\": {\\"peaklistBool\\": true, \\"__current_case__\\": 0, \\"convertRTMinute\\": true, \\"numDigitsMZ\\": \\"4\\", \\"numDigitsRT\\": \\"2\\", \\"intval\\": \\"into\\", \\"naTOzero\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"a6477b12-d89e-48b1-8572-9a86490211a4","when":null,"workflow_outputs":[{"label":null,"output_name":"plotChromPeakDensity","uuid":"6547dcea-fd0c-4029-bb47-a71dc6432e70"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_fillpeaks/abims_xcms_fillPeaks/3.12.0+galaxy0","errors":null,"id":11,"input_connections":{"image":{"id":10,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"xcms fillChromPeaks (fillPeaks)","outputs":[{"name":"xsetRData","type":"rdata.xcms.fillpeaks"},{"name":"variableMetadata","type":"tabular"},{"name":"dataMatrix","type":"tabular"}],"position":{"left":1969,"top":602.0066585286459},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/xcms_fillpeaks/abims_xcms_fillPeaks/3.12.0+galaxy0","tool_shed_repository":{"changeset_revision":"26d77e9ceb49","name":"xcms_fillpeaks","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"Adv\\": {\\"expandMz\\": \\"0\\", \\"expandRt\\": \\"0\\", \\"ppm\\": \\"0\\"}, \\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"peaklist\\": {\\"peaklistBool\\": \\"True\\", \\"convertRTMinute\\": true, \\"numDigitsMZ\\": \\"4\\", \\"numDigitsRT\\": \\"2\\", \\"intval\\": \\"into\\", \\"naTOzero\\": true}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.12.0+galaxy0","type":"tool","uuid":"41073bd0-81ec-410c-a2bc-127be2044740","when":null,"workflow_outputs":[{"label":"xcms fillChromPeaks dataMatrix","output_name":"dataMatrix","uuid":"b716b601-92fa-472c-993b-0769078d2b01"},{"label":null,"output_name":"xsetRData","uuid":"967392c7-016e-4d7e-8bf6-ce3628113345"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/camera_annotate/abims_CAMERA_annotateDiffreport/2.2.6+camera1.48.0-galaxy1","errors":null,"id":12,"input_connections":{"image":{"id":11,"output_name":"xsetRData"}},"inputs":[],"label":null,"name":"CAMERA.annotate","outputs":[{"name":"variableMetadata","type":"tabular"},{"name":"rdata_quick_true","type":"rdata.camera.quick"}],"position":{"left":2249,"top":669.0066585286459},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/lecorguille/camera_annotate/abims_CAMERA_annotateDiffreport/2.2.6+camera1.48.0-galaxy1","tool_shed_repository":{"changeset_revision":"4b9ab71be05e","name":"camera_annotate","owner":"lecorguille","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"diffreport\\": {\\"options\\": {\\"option\\": \\"hide\\", \\"__current_case__\\": 1}}, \\"export\\": {\\"convertRTMinute\\": true, \\"numDigitsMZ\\": \\"4\\", \\"numDigitsRT\\": \\"2\\", \\"intval\\": \\"into\\"}, \\"file_load_section\\": {\\"file_load_conditional\\": {\\"file_load_select\\": \\"no\\", \\"__current_case__\\": 0}}, \\"findgeneral\\": {\\"ppm\\": \\"5\\", \\"mzabs\\": \\"0.015\\"}, \\"findisotopes\\": {\\"maxcharge\\": \\"2\\", \\"maxiso\\": \\"4\\", \\"minfrac\\": \\"0.5\\"}, \\"groupfwhm\\": {\\"sigma\\": \\"6\\", \\"perfwhm\\": \\"0.6\\"}, \\"image\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"quick_block\\": {\\"quick\\": \\"TRUE\\", \\"__current_case__\\": 1, \\"polarity\\": \\"quick\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.2.6+camera1.48.0-galaxy1","type":"tool","uuid":"7f7f0568-40cf-445b-8f4e-78a3af20109b","when":null,"workflow_outputs":[{"label":"CAMERA.annotate variableMetadata","output_name":"variableMetadata","uuid":"9b321589-0f82-4f17-bf9b-faae8c353079"},{"label":null,"output_name":"rdata_quick_true","uuid":"5fd93902-b994-4124-8646-41b8f2c47b7d"}]}},"tags":["metabolomics","MS","LC-MS","workflow4metabolomics","xcms","GTN"],"uuid":"4b4dda98-7422-4207-b9bd-62c11ce8e0f0","version":5},"readme":"# Mass spectrometry: LC-MS preprocessing with XCMS \\n\\nThis workflow uses the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) to extract, filter, align and fill gaps, and uses the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html) to annotate isotopes, adducts and fragments.\\n\\n🎓 For more information see the [Galaxy Training Network tutorial: Mass spectrometry: LC-MS preprocessing with XCMS](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html)\\n\\n## Inputs\\n### sampleMetadata\\nThe sampleMetadata tabular file corresponds to a table containing information about your samples\\n\\nA sample metadata file contains various information for each of your raw files:\\n- Classes which will be used during the preprocessing steps\\n- Analytical batches which will be useful for a batch correction step, along with sample types (pool/sample) and injection order\\n- Different experimental conditions which can be used for statistics\\n- Any information about samples that you want to keep, in a column format\\n\\nThe content of your sample metadata file has to be filled by you, since it is not contained in your raw data. Note that you can either:\\n- Upload an existing metadata file\\n- Use a template to create one (because it can be painful to get the sample list without misspelling or omission)\\n  - Generate a template with the `xcms get a sampleMetadata file` tool available in Galaxy\\n  - Fill it using your favorite table editor (Excel, LibreOffice)\\n  - Upload it within Galaxy\\n\\n**Formats:** tab-separated values as tsv, tab, txt, ...\\n\\n### Mass-spectrometry Dataset Collection\\nMass-spectrometry data files gathered in a Galaxy Dataser Collection\\n\\n**Formats:** open format as mzXML, mzMl, mzData and netCDF\\n\\n## Main steps\\n1. MSnbase readMSData: read the mzXML and prepare for xcms\\n2. XCMS findChromPeaks: peak picking\\n3. XCMS groupChromPeaks: determining shared ions across samples\\n4. XCMS adjustRtime: retention time correction\\n5. XCMS fillChromPeaks: integrating areas of missing peaks\\n6. CAMERA.annotate: annotation\\n","changelog":"# Changelog\\n\\nAll notable changes to this project will be documented in this file.\\n\\n## [1.0] - 2023-11-22\\n\\nFirst release\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Mass spectrometry: LC-MS preprocessing with XCMS\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ SampleMetadata\\"]@{ shape: doc }\\n1[\\"ℹ️ Mass-spectrometry Dataset Collection\\"]@{ shape: docs }\\n2[\\"MSnbase readMSData\\"]@{ shape: process }\\n1 --> 2\\n3[\\"xcms plot chromatogram\\"]@{ shape: process }\\n2 --> 3\\n0 --> 3\\n4[\\"xcms findChromPeaks (xcmsSet)\\"]@{ shape: process }\\n2 --> 4\\n5[\\"xcms findChromPeaks Merger\\"]@{ shape: process }\\n4 --> 5\\n0 --> 5\\n6[\\"xcms groupChromPeaks (group)\\"]@{ shape: process }\\n5 --> 6\\n7[\\"xcms adjustRtime (retcor)\\"]@{ shape: process }\\n6 --> 7\\n8[\\"Intensity Check\\"]@{ shape: process }\\n6 --> 8\\n0 --> 8\\n6 --> 8\\n9[\\"xcms plot chromatogram\\"]@{ shape: process }\\n7 --> 9\\n0 --> 9\\n10[\\"xcms groupChromPeaks (group)\\"]@{ shape: process }\\n7 --> 10\\n11[\\"xcms fillChromPeaks (fillPeaks)\\"]@{ shape: process }\\n10 --> 11\\n12[\\"CAMERA.annotate\\"]@{ shape: process }\\n11 --> 12\\n```\\n","updated":"2023-11-22T00:00:00","trsID":"#workflow/github.com/iwc-workflows/lcms-preprocessing/main","dockstore_id":24508,"categories":[],"collections":["Computational Chemistry"],"doi":"10.5281/zenodo.15078810","tests":[{"doc":"Test outline for Mass_spectrometry__LC-MS_preprocessing_with_XCMS","job":{"SampleMetadata":{"class":"File","identifier":"sampleMetadata.tsv","location":"https://zenodo.org/records/10130758/files/sampleMetadata_12samp_completed.tsv","filetype":"tabular"},"Mass-spectrometry Dataset Collection":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"Blanc05.mzML","location":"https://zenodo.org/record/10130758/files/Blanc05.mzML","filetype":"mzml"},{"class":"File","identifier":"Blanc10.mzML","location":"https://zenodo.org/record/10130758/files/Blanc10.mzML","filetype":"mzml"},{"class":"File","identifier":"Blanc16.mzML","location":"https://zenodo.org/record/10130758/files/Blanc16.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_048.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_048.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_090.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_090.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_123.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_123.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_157.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_157.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_173.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_173.mzML","filetype":"mzml"},{"class":"File","identifier":"HU_neg_192.mzML","location":"https://zenodo.org/record/10130758/files/HU_neg_192.mzML","filetype":"mzml"},{"class":"File","identifier":"QC1_002.mzML","location":"https://zenodo.org/record/10130758/files/QC1_002.mzML","filetype":"mzml"},{"class":"File","identifier":"QC1_008.mzML","location":"https://zenodo.org/record/10130758/files/QC1_008.mzML","filetype":"mzml"},{"class":"File","identifier":"QC1_014.mzML","location":"https://zenodo.org/record/10130758/files/QC1_014.mzML","filetype":"mzml"}]}},"outputs":{"xcms fillChromPeaks dataMatrix":{"file":"test-data/xset.merged.groupChromPeaks.adjustRtime.groupChromPeaks.fillpeaks.dataMatrix.tsv"},"CAMERA.annotate variableMetadata":{"file":"test-data/xset.merged.groupChromPeaks.adjustRtime.groupChromPeaks.fillChromPeaks.annotate.variableMetadata.tsv"}}}]}],"path":"./workflows/metabomics/lcms-preprocessing"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/bacterial_genome_assembly.ga","testParameterFiles":["/bacterial_genome_assembly-tests.yml"],"authors":[{"name":"abromics-consortium","url":"https://www.abromics.fr/"},{"name":"ABRomics","email":"abromics-support@groupes.france-bioinformatique.fr"},{"name":"Pierre Marin","alternateName":"pimarin","orcid":"0000-0002-8304-138X"},{"name":"Clea Siguret","alternateName":"clsiguret","orcid":"0009-0005-6140-0379"}],"definition":{"a_galaxy_workflow":"true","annotation":"Assembly of bacterial paired-end short read data with generation of quality metrics and reports","creator":[{"class":"Organization","name":"abromics-consortium","url":"https://www.abromics.fr/"},{"class":"Person","email":"mailto:abromics-support@groupes.france-bioinformatique.fr","name":"ABRomics"},{"alternateName":"pimarin","class":"Person","identifier":"https://orcid.org/0000-0002-8304-138X","name":"Pierre Marin"},{"alternateName":"clsiguret","class":"Person","identifier":"https://orcid.org/0009-0005-6140-0379","name":"Clea Siguret"}],"format-version":"0.1","release":"1.1.5","license":"GPL-3.0-or-later","name":"Bacterial Genome Assembly using Shovill","steps":{"0":{"annotation":"Should be the adapter trimmed forward strand of the paired end reads.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be the adapter trimmed forward strand of the paired end reads.","name":"Input adapter trimmed sequence reads (forward)"}],"label":"Input adapter trimmed sequence reads (forward)","name":"Input dataset","outputs":[],"position":{"left":0,"top":26.85905752718789},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\", \\"fastq.gz\\", \\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"d223414b-29dc-4ffa-9851-a366cc4a3a80","when":null,"workflow_outputs":[]},"1":{"annotation":"Should be the adapter trimmed reverse strand of the paired end reads.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Should be the adapter trimmed reverse strand of the paired end reads.","name":"Input adapter trimmed sequence reads (reverse)"}],"label":"Input adapter trimmed sequence reads (reverse)","name":"Input dataset","outputs":[],"position":{"left":1.5420743426057233,"top":179.85816302215434},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\", \\"fastq.gz\\", \\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"33a6db62-4324-4469-a903-33b5ef505a17","when":null,"workflow_outputs":[]},"2":{"annotation":"genome assembly with shovill tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy2","errors":null,"id":2,"input_connections":{"library|R1":{"id":0,"output_name":"output"},"library|R2":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Shovill","name":"library"},{"description":"runtime parameter for tool Shovill","name":"library"}],"label":"shovill_genome_assembly","name":"Shovill","outputs":[{"name":"shovill_std_log","type":"txt"},{"name":"contigs","type":"fasta"},{"name":"contigs_graph","type":"txt"},{"name":"bamfiles","type":"unsorted.bam"}],"position":{"left":308.4719663659007,"top":131.00139799356907},"post_job_actions":{"ChangeDatatypeActionbamfiles":{"action_arguments":{"newtype":"bam"},"action_type":"ChangeDatatypeAction","output_name":"bamfiles"},"RenameDatasetActionbamfiles":{"action_arguments":{"newname":"shovill_alignment_bam"},"action_type":"RenameDatasetAction","output_name":"bamfiles"},"RenameDatasetActioncontigs":{"action_arguments":{"newname":"shovill_contigs_fasta"},"action_type":"RenameDatasetAction","output_name":"contigs"},"RenameDatasetActioncontigs_graph":{"action_arguments":{"newname":"shovill_contigs_graph"},"action_type":"RenameDatasetAction","output_name":"contigs_graph"},"RenameDatasetActionshovill_std_log":{"action_arguments":{"newname":"shovill_logfile"},"action_type":"RenameDatasetAction","output_name":"shovill_std_log"},"TagDatasetActionbamfiles":{"action_arguments":{"tags":"shovill,alignment,bam"},"action_type":"TagDatasetAction","output_name":"bamfiles"},"TagDatasetActioncontigs":{"action_arguments":{"tags":"shovill,fasta,contigs"},"action_type":"TagDatasetAction","output_name":"contigs"},"TagDatasetActioncontigs_graph":{"action_arguments":{"tags":"shovill,graph"},"action_type":"TagDatasetAction","output_name":"contigs_graph"},"TagDatasetActionshovill_std_log":{"action_arguments":{"tags":"logfile"},"action_type":"TagDatasetAction","output_name":"shovill_std_log"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy2","tool_shed_repository":{"changeset_revision":"ee17a294d3a3","name":"shovill","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"namefmt\\": \\"contig%05d\\", \\"depth\\": \\"100\\", \\"gsize\\": \\"\\", \\"kmers\\": \\"\\", \\"opts\\": \\"\\", \\"keep_files\\": {\\"nocorr\\": \\"yes_correction\\", \\"__current_case__\\": 1, \\"keepfiles\\": true}, \\"minlen\\": \\"0\\", \\"mincov\\": \\"2\\"}, \\"assembler\\": \\"spades\\", \\"library\\": {\\"lib_type\\": \\"paired\\", \\"__current_case__\\": 0, \\"R1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"R2\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"log\\": true, \\"trim\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.0+galaxy2","type":"tool","uuid":"316380fb-2944-4d85-8832-8c8b70fe3623","when":null,"workflow_outputs":[{"label":"shovill_contigs_fasta","output_name":"contigs","uuid":"c74a1bd3-1a9e-45c2-aa00-4c42a6269eee"},{"label":"shovill_contigs_graph","output_name":"contigs_graph","uuid":"d368ce7d-6123-436e-a566-a7ae0f56234d"},{"label":"shovill_alignment_bam","output_name":"bamfiles","uuid":"8cbb3659-4de3-4c4f-b15d-c44dd5b5ccfa"},{"label":"shovill_logfile","output_name":"shovill_std_log","uuid":"3b25ebd4-8348-43ac-afdc-870e292d420e"}]},"3":{"annotation":"quast_quality","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.3.0+galaxy0","errors":null,"id":3,"input_connections":{"mode|in|inputs":{"id":2,"output_name":"contigs"},"mode|reads|input_1":{"id":0,"output_name":"output"},"mode|reads|input_2":{"id":1,"output_name":"output"}},"inputs":[],"label":"quast_quality","name":"Quast","outputs":[{"name":"report_tabular","type":"tabular"},{"name":"report_html","type":"html"},{"name":"report_pdf","type":"pdf"},{"name":"log","type":"txt"}],"position":{"left":583.5485996614345,"top":0},"post_job_actions":{"RenameDatasetActionlog":{"action_arguments":{"newname":"quast_log"},"action_type":"RenameDatasetAction","output_name":"log"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"quast_report_html"},"action_type":"RenameDatasetAction","output_name":"report_html"},"RenameDatasetActionreport_pdf":{"action_arguments":{"newname":"quast_report_pdf"},"action_type":"RenameDatasetAction","output_name":"report_pdf"},"RenameDatasetActionreport_tabular":{"action_arguments":{"newname":"quast_report_tabular"},"action_type":"RenameDatasetAction","output_name":"report_tabular"},"TagDatasetActionlog":{"action_arguments":{"tags":"quast_log"},"action_type":"TagDatasetAction","output_name":"log"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"quast_report_html"},"action_type":"TagDatasetAction","output_name":"report_html"},"TagDatasetActionreport_pdf":{"action_arguments":{"tags":"quast_report_pdf"},"action_type":"TagDatasetAction","output_name":"report_pdf"},"TagDatasetActionreport_tabular":{"action_arguments":{"tags":"quast_report_tabular"},"action_type":"TagDatasetAction","output_name":"report_tabular"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.3.0+galaxy0","tool_shed_repository":{"changeset_revision":"a3b35edea53a","name":"quast","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced\\": {\\"contig_thresholds\\": \\"0,1000\\", \\"strict_NA\\": false, \\"extensive_mis_size\\": \\"1000\\", \\"scaffold_gap_max_size\\": \\"1000\\", \\"unaligned_part_size\\": \\"500\\", \\"skip_unaligned_mis_contigs\\": true, \\"fragmented_max_indent\\": null, \\"report_all_metrics\\": false, \\"x_for_Nx\\": \\"90\\"}, \\"alignments\\": {\\"use_all_alignments\\": false, \\"min_alignment\\": \\"65\\", \\"ambiguity_usage\\": \\"one\\", \\"ambiguity_score\\": \\"0.99\\", \\"fragmented\\": false, \\"upper_bound_assembly\\": false, \\"upper_bound_min_con\\": null, \\"local_mis_size\\": \\"200\\"}, \\"assembly\\": {\\"type\\": \\"genome\\", \\"__current_case__\\": 0, \\"ref\\": {\\"use_ref\\": \\"false\\", \\"__current_case__\\": 1, \\"est_ref_size\\": null}, \\"orga_type\\": \\"\\", \\"min_identity\\": \\"95.0\\"}, \\"genes\\": {\\"gene_finding\\": {\\"tool\\": \\"none\\", \\"__current_case__\\": 0}, \\"rna_finding\\": false, \\"conserved_genes_finding\\": true}, \\"large\\": false, \\"min_contig\\": \\"500\\", \\"mode\\": {\\"mode\\": \\"individual\\", \\"__current_case__\\": 0, \\"in\\": {\\"custom\\": \\"false\\", \\"__current_case__\\": 1, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"reads\\": {\\"reads_option\\": \\"paired\\", \\"__current_case__\\": 2, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input_2\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"output_files\\": [\\"html\\", \\"pdf\\", \\"tabular\\", \\"log\\", \\"summary\\", \\"krona\\"], \\"split_scaffolds\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.3.0+galaxy0","type":"tool","uuid":"94a29697-b779-4cf4-a25c-c59f7f88325c","when":null,"workflow_outputs":[{"label":"quast_log","output_name":"log","uuid":"eadbce9e-d43b-43b1-9704-cae70da20521"},{"label":"quast_report_pdf","output_name":"report_pdf","uuid":"7477d62e-ff9f-4fca-8318-dbd876805655"},{"label":"quast_report_tabular","output_name":"report_tabular","uuid":"e8269834-4b1d-40c7-9ccc-efaa011ad255"},{"label":"quast_report_html","output_name":"report_html","uuid":"4169fc6b-c47e-4994-b44e-ca62e804f902"}]},"4":{"annotation":"refseqmasher_genome, search close reference genome in refseq database","content_id":"toolshed.g2.bx.psu.edu/repos/nml/refseq_masher/refseq_masher_matches/0.1.2","errors":null,"id":4,"input_connections":{"input|fasta":{"id":2,"output_name":"contigs"}},"inputs":[{"description":"runtime parameter for tool RefSeq Masher Matches","name":"input"}],"label":"refseqmasher_genome","name":"RefSeq Masher Matches","outputs":[{"name":"output_path_tab","type":"tabular"}],"position":{"left":793.6819614778408,"top":329.4499969482422},"post_job_actions":{"RenameDatasetActionoutput_path_tab":{"action_arguments":{"newname":"refseqmasher_report_tabular"},"action_type":"RenameDatasetAction","output_name":"output_path_tab"},"TagDatasetActionoutput_path_tab":{"action_arguments":{"tags":"refseqmasher,report"},"action_type":"TagDatasetAction","output_name":"output_path_tab"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/refseq_masher/refseq_masher_matches/0.1.2","tool_shed_repository":{"changeset_revision":"1ec42f033bb4","name":"refseq_masher","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"min_kmer_threshold\\": \\"8\\", \\"output_type\\": \\"tab\\", \\"verbosity\\": \\"-vv\\"}, \\"input\\": {\\"type\\": \\"fasta\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"top_n_results\\": \\"3\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.2","type":"tool","uuid":"55213145-5284-462b-98ab-e8871ff9f056","when":null,"workflow_outputs":[{"label":"refseqmasher_report_tabular","output_name":"output_path_tab","uuid":"538ade11-0fcb-4bb3-be5d-8dc164c91602"}]},"5":{"annotation":"bandage_contig_graph_stats","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/2022.09+galaxy2","errors":null,"id":5,"input_connections":{"input_file":{"id":2,"output_name":"contigs_graph"}},"inputs":[],"label":"bandage_contig_graph_stats","name":"Bandage Info","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":771.3861542615132,"top":516.142295707301},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"bandage_contig_graph_stats"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"bandage_contig_graph_stats"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/2022.09+galaxy2","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"tsv\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy2","type":"tool","uuid":"e84778db-553a-424d-b71e-547994be4050","when":null,"workflow_outputs":[{"label":"bandage_contig_graph_stats","output_name":"outfile","uuid":"c075d076-3c99-48e2-94e9-ba9330c6c5f8"}]},"6":{"annotation":"bandage_contig_graph_plot","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":6,"input_connections":{"input_file":{"id":2,"output_name":"contigs_graph"}},"inputs":[],"label":"bandage_contig_graph_plot","name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":769.9687076931637,"top":740.2606081887517},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"bandage_contig_graph_plot"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"bandage_contig_graph_plot"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"1000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": null, \\"output_format\\": \\"svg\\", \\"width\\": \\"1000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"7a357116-66a1-46d7-9af2-14e7415ef19e","when":null,"workflow_outputs":[{"label":"bandage_contig_graph_plot","output_name":"outfile","uuid":"eb9d5175-0589-4c3e-8373-ef9cafc72e1e"}]},"7":{"annotation":"ToolDistillator extracts results from tools and creates a JSON file for each tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","errors":null,"id":7,"input_connections":{"tool_section|tools_0|select_tool|bam_file_path":{"id":2,"output_name":"bamfiles"},"tool_section|tools_0|select_tool|contig_graph_path":{"id":2,"output_name":"contigs_graph"},"tool_section|tools_0|select_tool|input":{"id":2,"output_name":"contigs"},"tool_section|tools_1|select_tool|input":{"id":3,"output_name":"report_tabular"},"tool_section|tools_1|select_tool|quast_html_path":{"id":3,"output_name":"report_html"},"tool_section|tools_2|select_tool|input":{"id":4,"output_name":"output_path_tab"},"tool_section|tools_3|select_tool|bandage_plot_path":{"id":6,"output_name":"outfile"},"tool_section|tools_3|select_tool|input":{"id":5,"output_name":"outfile"}},"inputs":[],"label":"ToolDistillator","name":"ToolDistillator","outputs":[{"name":"output_json","type":"input"}],"position":{"left":1299.389187169496,"top":264.0158282844628},"post_job_actions":{"RenameDatasetActionoutput_json":{"action_arguments":{"newname":"tooldistillator_results"},"action_type":"RenameDatasetAction","output_name":"output_json"},"TagDatasetActionoutput_json":{"action_arguments":{"tags":"tooldistillator_results"},"action_type":"TagDatasetAction","output_name":"output_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"ea93df4b3df2","name":"tooldistillator","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"log\\": false, \\"tool_section\\": {\\"tools\\": [{\\"__index__\\": 0, \\"select_tool\\": {\\"tool_list\\": \\"shovill\\", \\"__current_case__\\": 19, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"contig_graph_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bam_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 1, \\"select_tool\\": {\\"tool_list\\": \\"quast\\", \\"__current_case__\\": 16, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"quast_html_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 2, \\"select_tool\\": {\\"tool_list\\": \\"refseqmasher\\", \\"__current_case__\\": 18, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 3, \\"select_tool\\": {\\"tool_list\\": \\"bandage\\", \\"__current_case__\\": 3, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bandage_plot_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"b6156c8e-5048-41e3-90d9-f9302734c3d1","when":null,"workflow_outputs":[{"label":"tooldistillator_results","output_name":"output_json","uuid":"6f9bffbd-5b92-47ee-9d52-5846a7db2a4a"}]},"8":{"annotation":"ToolDistillator summarize groups all JSON file into a unique JSON file","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","errors":null,"id":8,"input_connections":{"summarize_data":{"id":7,"output_name":"output_json"}},"inputs":[],"label":"ToolDistillator summarize","name":"ToolDistillator Summarize","outputs":[{"name":"summary_json","type":"json"}],"position":{"left":1646.4377091829172,"top":358.6654042662252},"post_job_actions":{"RenameDatasetActionsummary_json":{"action_arguments":{"newname":"tooldistillator_summarize"},"action_type":"RenameDatasetAction","output_name":"summary_json"},"TagDatasetActionsummary_json":{"action_arguments":{"tags":"tooldistillator_summarize"},"action_type":"TagDatasetAction","output_name":"summary_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"24eec94e6dfc","name":"tooldistillator_summarize","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"summarize_data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"3bbe0eba-b6bb-4d43-ab57-2b899b8112ae","when":null,"workflow_outputs":[{"label":"tooldistillator_summarize","output_name":"summary_json","uuid":"9097e16d-ace5-47f5-9af5-7eb3b13b91d2"}]}},"tags":["fastq","Genomics","bacterial-genomics","paired-end","assembly","quality","ABRomics"],"uuid":"bfe7f341-07ef-4bfe-9b50-2bc7628d6c55","version":1},"readme":"# Bacterial genome assembly workflow for paired end data (v1.0)\\n\\nThis workflow uses paired-end illumina trimmed reads fastq(.gz) files and executes the following steps:\\n1. Assembly raw reads to a final contig fasta file \\n    - **Shovill**\\n2. Quality control of the assembly\\n    - **Quast**\\n    - **Bandage** to plot assembly graph\\n    - **Refseqmasher** to identify the closed reference genome\\n3. Aggregating outputs into a single JSON file\\n    - **ToolDistillator** to extract and aggregate information from different tool outputs to JSON parsable files\\n\\n## Inputs\\n\\n1. Paired-end illumina trimmed reads in fastq(.gz) format.\\n\\n## Outputs\\n\\n1. Assembly:\\n    - Assembly with contig in fasta\\n    - Mapped read on assembly in bam format\\n    - Graph assembly in gfa format\\n2. Quality of Assembly:\\n    - Assembly report\\n    - Assembly Graph\\n    - Tabular result of closed reference genome\\n3. Aggregating outputs\\n    - JSON file with information about the outputs of **Shovill**, **Quast**, **Bandage**, **Refseqmasher**","changelog":"# Changelog\\n\\n## [1.1.5] - 2024-11-18\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.3.0+galaxy0`\\n\\n## [1.1.4] - 2024-10-21\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1`\\n\\n## [1.1.3] - 2024-09-30\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0`\\n\\n## [1.1.2] - 2024-09-19\\n\\n### Manual update\\n\\n- Updated the tool versions manually in the ToolDistillator parameters\\n\\n## [1.1.1] - 2024-07-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0`\\n\\n## [1.1] - 2024-06-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/0.8.1+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/2022.09+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy4` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4`\\n\\n## [1.0] - 05-06-2024\\n\\n","diagrams":"# Workflow diagrams\\n\\n## Bacterial Genome Assembly using Shovill\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input adapter trimmed sequence reads (forward)\\"]@{ shape: doc }\\n1[\\"ℹ️ Input adapter trimmed sequence reads (reverse)\\"]@{ shape: doc }\\n2[\\"shovill_genome_assembly\\"]@{ shape: process }\\n0 --> 2\\n1 --> 2\\n3[\\"quast_quality\\"]@{ shape: process }\\n2 --> 3\\n0 --> 3\\n1 --> 3\\n4[\\"refseqmasher_genome\\"]@{ shape: process }\\n2 --> 4\\n5[\\"bandage_contig_graph_stats\\"]@{ shape: process }\\n2 --> 5\\n6[\\"bandage_contig_graph_plot\\"]@{ shape: process }\\n2 --> 6\\n7[\\"ToolDistillator\\"]@{ shape: process }\\n2 --> 7\\n2 --> 7\\n2 --> 7\\n3 --> 7\\n3 --> 7\\n4 --> 7\\n6 --> 7\\n5 --> 7\\n8[\\"ToolDistillator summarize\\"]@{ shape: process }\\n7 --> 8\\n```\\n","updated":"2024-11-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/bacterial-genome-assembly/main","dockstore_id":27291,"categories":[],"collections":["Genome assembly"],"doi":"10.5281/zenodo.15078808","tests":[{"doc":"Test outline for bacterial_genome_assembly.ga","job":{"Input adapter trimmed sequence reads (forward)":{"class":"File","path":"https://zenodo.org/records/11485346/files/fastp_trimmed_R1.fastqsanger.gz","filetype":"fastqsanger.gz"},"Input adapter trimmed sequence reads (reverse)":{"class":"File","path":"https://zenodo.org/records/11485346/files/fastp_trimmed_R2.fastqsanger.gz","filetype":"fastqsanger.gz"}},"outputs":{"shovill_contigs_graph":{"asserts":{"has_text":{"text":"KC:i"}}},"shovill_logfile":{"asserts":{"has_text":{"text":"[shovill] Done."}}},"shovill_contigs_fasta":{"asserts":{"has_text":{"text":">contig00001"}}},"quast_report_tabular":{"asserts":{"has_n_columns":{"n":2},"has_text":{"text":"shovill_contigs_fasta"}}},"quast_log":{"asserts":{"has_text":{"text":"Thank you for using QUAST!"}}},"refseqmasher_report_tabular":{"asserts":{"has_text":{"text":"Enterococcus faecalis"},"has_n_columns":{"n":21}}},"bandage_contig_graph_stats":{"asserts":{"has_text":{"text":"Total length orphaned nodes (bp):"},"has_n_columns":{"n":2}}},"tooldistillator_summarize":{"asserts":{"has_text":{"text":"bandage_plot_path"}}}}}]}],"path":"./workflows/genome-assembly/bacterial-genome-assembly"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Assembly-polishing-with-long-reads.ga","testParameterFiles":["/Assembly-polishing-with-long-reads-tests.yml"],"authors":[{"name":"Anna Syme","orcid":"0000-0002-9906-0673"}],"definition":{"a_galaxy_workflow":"true","annotation":"Racon polish with long reads, x4","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9906-0673","name":"Anna Syme"}],"format-version":"0.1","release":"0.1","license":"MIT","name":"Assembly polishing with long reads","steps":{"0":{"annotation":"","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"","name":"Assembly to be polished"}],"label":"Assembly to be polished","name":"Input dataset","outputs":[],"position":{"left":0,"top":216.625},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"c7796c36-adf8-49e4-a8e4-66e606d70311","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"","name":"long reads"}],"label":"long reads","name":"Input dataset","outputs":[],"position":{"left":4.5,"top":360.234375},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"e2e4a70b-6e36-4b8b-bdf6-12b97189a6bd","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"","name":"minimap setting (for long reads) "}],"label":"minimap setting (for long reads) ","name":"Input parameter","outputs":[],"position":{"left":0.578125,"top":496.10504150390625},"tool_id":null,"tool_state":"{\\"suggestions\\": [\\"map-ont\\", \\"map-pb\\", \\"map-hifi\\"], \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"fb80c05c-08d4-4ac2-b4bf-81ed906e4d72","when":null,"workflow_outputs":[]},"3":{"annotation":"Map raw reads to assembly; output paf","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","errors":null,"id":3,"input_connections":{"fastq_input|analysis_type_selector":{"id":2,"output_name":"output"},"fastq_input|fastq_input1":{"id":1,"output_name":"output"},"reference_source|ref_file":{"id":0,"output_name":"output"}},"inputs":[],"label":"Minimap2: map long reads to assembly","name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":377.640625,"top":0},"post_job_actions":{"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","tool_shed_repository":{"changeset_revision":"be1d967337e4","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.26+galaxy0","type":"tool","uuid":"0c06144f-d1ac-4b45-9669-1ddf11fcb06b","when":null,"workflow_outputs":[]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","errors":null,"id":4,"input_connections":{"corrected_reads":{"id":0,"output_name":"output"},"overlaps":{"id":3,"output_name":"alignment_output"},"reads":{"id":1,"output_name":"output"}},"inputs":[],"label":"Racon: polish 1","name":"Racon","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":440.06805419921875,"top":598.93798828125},"post_job_actions":{"HideDatasetActionconsensus":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","tool_shed_repository":{"changeset_revision":"e100a765370e","name":"racon","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"corrected_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"e\\": \\"0.3\\", \\"error_threshold\\": \\"0.3\\", \\"f\\": \\"false\\", \\"fragment_correction\\": false, \\"g\\": \\"-8\\", \\"gap\\": \\"-4\\", \\"include_unpolished\\": false, \\"m\\": \\"5\\", \\"match\\": \\"3\\", \\"mismatch\\": \\"-5\\", \\"no_trimming\\": false, \\"overlaps\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"q\\": \\"10.0\\", \\"quality_threshold\\": \\"10.0\\", \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"u\\": \\"true\\", \\"w\\": \\"500\\", \\"window_length\\": \\"500\\", \\"x\\": \\"-4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.5.0+galaxy1","type":"tool","uuid":"6c144354-91f2-49dd-a1f5-7c4119859730","when":null,"workflow_outputs":[]},"5":{"annotation":"Map raw reads to assembly; output paf","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","errors":null,"id":5,"input_connections":{"fastq_input|analysis_type_selector":{"id":2,"output_name":"output"},"fastq_input|fastq_input1":{"id":1,"output_name":"output"},"reference_source|ref_file":{"id":4,"output_name":"consensus"}},"inputs":[],"label":"Minimap2: map long reads to polished assembly 1","name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":721.2265625,"top":0.6822357177734375},"post_job_actions":{"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","tool_shed_repository":{"changeset_revision":"be1d967337e4","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.26+galaxy0","type":"tool","uuid":"c1b83dad-a8ed-42cb-951f-73f59f2dfa92","when":null,"workflow_outputs":[]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","errors":null,"id":6,"input_connections":{"corrected_reads":{"id":4,"output_name":"consensus"},"overlaps":{"id":5,"output_name":"alignment_output"},"reads":{"id":1,"output_name":"output"}},"inputs":[],"label":"Racon: polish 2","name":"Racon","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":759.9375,"top":485.71875},"post_job_actions":{"HideDatasetActionconsensus":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","tool_shed_repository":{"changeset_revision":"e100a765370e","name":"racon","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"corrected_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"e\\": \\"0.3\\", \\"error_threshold\\": \\"0.3\\", \\"f\\": \\"false\\", \\"fragment_correction\\": false, \\"g\\": \\"-8\\", \\"gap\\": \\"-4\\", \\"include_unpolished\\": false, \\"m\\": \\"5\\", \\"match\\": \\"3\\", \\"mismatch\\": \\"-5\\", \\"no_trimming\\": false, \\"overlaps\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"q\\": \\"10.0\\", \\"quality_threshold\\": \\"10.0\\", \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"u\\": \\"true\\", \\"w\\": \\"500\\", \\"window_length\\": \\"500\\", \\"x\\": \\"-4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.5.0+galaxy1","type":"tool","uuid":"3208491e-b831-4238-957e-5b7388d7ff8d","when":null,"workflow_outputs":[]},"7":{"annotation":"Map raw reads to assembly; output paf","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","errors":null,"id":7,"input_connections":{"fastq_input|analysis_type_selector":{"id":2,"output_name":"output"},"fastq_input|fastq_input1":{"id":1,"output_name":"output"},"reference_source|ref_file":{"id":6,"output_name":"consensus"}},"inputs":[],"label":"Minimap2: map long reads to polished assembly 2","name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":998.75,"top":27.9375},"post_job_actions":{"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","tool_shed_repository":{"changeset_revision":"be1d967337e4","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.26+galaxy0","type":"tool","uuid":"42af524b-7f74-4c75-8d78-f6888078dd2e","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","errors":null,"id":8,"input_connections":{"corrected_reads":{"id":6,"output_name":"consensus"},"overlaps":{"id":7,"output_name":"alignment_output"},"reads":{"id":1,"output_name":"output"}},"inputs":[],"label":"Racon: polish 3","name":"Racon","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":1089.71875,"top":455.53125},"post_job_actions":{"HideDatasetActionconsensus":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","tool_shed_repository":{"changeset_revision":"e100a765370e","name":"racon","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"corrected_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"e\\": \\"0.3\\", \\"error_threshold\\": \\"0.3\\", \\"f\\": \\"false\\", \\"fragment_correction\\": false, \\"g\\": \\"-8\\", \\"gap\\": \\"-4\\", \\"include_unpolished\\": false, \\"m\\": \\"5\\", \\"match\\": \\"3\\", \\"mismatch\\": \\"-5\\", \\"no_trimming\\": false, \\"overlaps\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"q\\": \\"10.0\\", \\"quality_threshold\\": \\"10.0\\", \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"u\\": \\"true\\", \\"w\\": \\"500\\", \\"window_length\\": \\"500\\", \\"x\\": \\"-4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.5.0+galaxy1","type":"tool","uuid":"21f339bd-ebc1-4c73-a976-5d0b9ab2127a","when":null,"workflow_outputs":[]},"9":{"annotation":"Map raw reads to assembly; output paf","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","errors":null,"id":9,"input_connections":{"fastq_input|analysis_type_selector":{"id":2,"output_name":"output"},"fastq_input|fastq_input1":{"id":1,"output_name":"output"},"reference_source|ref_file":{"id":8,"output_name":"consensus"}},"inputs":[],"label":"Minimap2: map long reads to polished assembly 3","name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":1316.53125,"top":38.046875},"post_job_actions":{"HideDatasetActionalignment_output":{"action_arguments":{},"action_type":"HideDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.26+galaxy0","tool_shed_repository":{"changeset_revision":"be1d967337e4","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"paf\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.26+galaxy0","type":"tool","uuid":"7d36291e-aa76-4312-9943-b5e56d0e90c5","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","errors":null,"id":10,"input_connections":{"corrected_reads":{"id":8,"output_name":"consensus"},"overlaps":{"id":9,"output_name":"alignment_output"},"reads":{"id":1,"output_name":"output"}},"inputs":[],"label":"Racon: polish 4","name":"Racon","outputs":[{"name":"consensus","type":"fasta"}],"position":{"left":1496.078125,"top":431.1875},"post_job_actions":{"RenameDatasetActionconsensus":{"action_arguments":{"newname":"Racon_long_reads_polished_assembly.fasta"},"action_type":"RenameDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.5.0+galaxy1","tool_shed_repository":{"changeset_revision":"e100a765370e","name":"racon","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"corrected_reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"e\\": \\"0.3\\", \\"error_threshold\\": \\"0.3\\", \\"f\\": \\"false\\", \\"fragment_correction\\": false, \\"g\\": \\"-8\\", \\"gap\\": \\"-4\\", \\"include_unpolished\\": false, \\"m\\": \\"5\\", \\"match\\": \\"3\\", \\"mismatch\\": \\"-5\\", \\"no_trimming\\": false, \\"overlaps\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"q\\": \\"10.0\\", \\"quality_threshold\\": \\"10.0\\", \\"reads\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"u\\": \\"true\\", \\"w\\": \\"500\\", \\"window_length\\": \\"500\\", \\"x\\": \\"-4\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.5.0+galaxy1","type":"tool","uuid":"53f4572e-3a4a-4585-89ab-9459f8a61720","when":null,"workflow_outputs":[{"label":"Assembly polished by long reads using Racon","output_name":"consensus","uuid":"bcf0f03c-5951-46a7-aa38-545aed9bc183"}]}},"tags":[],"uuid":"68d6270a-bfd1-452a-abc9-4a314e13af85","version":9},"readme":"# Assembly polishing with Racon workflow\\n\\n## Inputs\\n\\n- Sequencing reads in format: fastq, fastq.gz, fastqsanger.gz or fastqsanger\\n- Genome assembly to be polished, in fasta format\\n\\n## What does the workflow do\\n\\n- After long reads have been assembled into a genome (contigs), this can be polished with the same long reads. \\n- This workflow uses the tool minimap2 to map the long reads back to the assembly, and then uses Racon to make polishes. \\n- This is repeated a further 3 times. \\n\\nIn more detail:\\n\\n- minimap2 : long reads are mapped to assembly => overlaps.paf.\\n- overaps, long reads, assembly => Racon => polished assembly 1\\n- using polished assembly 1 as input; repeat minimap2 + racon => polished assembly 2\\n- using polished assembly 2 as input, repeat minimap2 + racon => polished assembly 3\\n- using polished assembly 3 as input, repeat minimap2 + racon => polished assembly 4\\n\\n## Settings\\n\\n- Run as-is or change parameters at runtime.\\n- For the input at \\"minimap settings for long reads\\", enter (map-pb) for PacBio reads, (map-hifi) for PacBio HiFi reads, or (map-ont) for Oxford Nanopore reads.\\n\\n## Outputs\\n\\nThere is one output: the polished assembly in fasta format. \\n\\n","changelog":"# Changelog\\n\\n## [0.1] - 2023-07-15\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Assembly polishing with long reads\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Assembly to be polished\\"]@{ shape: doc }\\n1[\\"ℹ️ long reads\\"]@{ shape: doc }\\n2[\\"ℹ️ minimap setting (for long reads) \\"]@{ shape: lean-l }\\n3[\\"Minimap2: map long reads to assembly\\"]@{ shape: process }\\n2 --> 3\\n1 --> 3\\n0 --> 3\\n4[\\"Racon: polish 1\\"]@{ shape: process }\\n0 --> 4\\n3 --> 4\\n1 --> 4\\n5[\\"Minimap2: map long reads to polished assembly 1\\"]@{ shape: process }\\n2 --> 5\\n1 --> 5\\n4 --> 5\\n6[\\"Racon: polish 2\\"]@{ shape: process }\\n4 --> 6\\n5 --> 6\\n1 --> 6\\n7[\\"Minimap2: map long reads to polished assembly 2\\"]@{ shape: process }\\n2 --> 7\\n1 --> 7\\n6 --> 7\\n8[\\"Racon: polish 3\\"]@{ shape: process }\\n6 --> 8\\n7 --> 8\\n1 --> 8\\n9[\\"Minimap2: map long reads to polished assembly 3\\"]@{ shape: process }\\n2 --> 9\\n1 --> 9\\n8 --> 9\\n10[\\"Racon: polish 4\\"]@{ shape: process }\\n8 --> 10\\n9 --> 10\\n1 --> 10\\n```\\n","updated":"2023-07-15T00:00:00","trsID":"#workflow/github.com/iwc-workflows/polish-with-long-reads/main","dockstore_id":23536,"categories":[],"collections":["Genome assembly"],"doi":"10.5281/zenodo.15078869","tests":[{"doc":"Test outline for Assembly-polishing-with-long-reads","job":{"Assembly to be polished":{"class":"File","path":"test-data/assembly.fasta","filetype":"fasta"},"long reads":{"class":"File","path":"test-data/long_reads.fastqsanger.gz","filetype":"fastqsanger.gz"},"minimap setting (for long reads) ":"map-ont"},"outputs":{"Assembly polished by long reads using Racon":{"path":"test-data/assembly_polished_by_long_reads.fasta","compare":"sim_size","delta_frac":0.2}}}]}],"path":"./workflows/genome-assembly/polish-with-long-reads"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Genome-assembly-with-Flye.ga","testParameterFiles":["/Genome-assembly-with-Flye-tests.yml"],"authors":[{"name":"Anna Syme","orcid":"0000-0002-9906-0673"}],"definition":{"a_galaxy_workflow":"true","annotation":"Assemble long reads with Flye, then view assembly statistics and assembly graph","creator":[{"class":"Person","identifier":"https://orcid.org/0000-0002-9906-0673","name":"Anna Syme"}],"format-version":"0.1","license":"MIT","release":"0.2","name":"Genome assembly with Flye","report":{"markdown":"\\n# Workflow Execution Report\\n\\n## Workflow Inputs\\n```galaxy\\ninvocation_inputs()\\n```\\n\\n## Workflow Outputs\\n```galaxy\\ninvocation_outputs()\\n```\\n\\n## Workflow\\n```galaxy\\nworkflow_display()\\n```\\n\\n"},"steps":{"0":{"annotation":"Sequence reads e.g. nanopore","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Sequence reads e.g. nanopore","name":"Input sequence reads"}],"label":"Input sequence reads","name":"Input dataset","outputs":[],"position":{"left":0,"top":263.5110244255367},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"501d7387-3bed-4510-891f-a4571a48d9ab","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.3+galaxy0","errors":null,"id":1,"input_connections":{"inputs":{"id":0,"output_name":"output"}},"inputs":[],"label":"Flye: assembly","name":"Flye","outputs":[{"name":"consensus","type":"fasta"},{"name":"assembly_graph","type":"graph_dot"},{"name":"assembly_gfa","type":"txt"},{"name":"assembly_info","type":"tabular"}],"position":{"left":264.3488280182951,"top":139.5081025362239},"post_job_actions":{"RenameDatasetActionassembly_gfa":{"action_arguments":{"newname":"Flye output as GFA for bandage"},"action_type":"RenameDatasetAction","output_name":"assembly_gfa"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.3+galaxy0","tool_shed_repository":{"changeset_revision":"291923e6f276","name":"flye","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"asm\\": {\\"asm_select\\": \\"false\\", \\"__current_case__\\": 1}, \\"generate_log\\": false, \\"i\\": \\"1\\", \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterations\\": \\"1\\", \\"keep_haplotypes\\": false, \\"m\\": null, \\"meta\\": false, \\"min_overlap\\": null, \\"mode\\": \\"--nano-raw\\", \\"mode_conditional\\": {\\"mode\\": \\"--nano-raw\\", \\"__current_case__\\": 0}, \\"no_alt_contigs\\": false, \\"no_trestle\\": \\"false\\", \\"plasmids\\": \\"false\\", \\"scaffold\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.9.3+galaxy0","type":"tool","uuid":"8bed3591-266a-41f1-91d3-11b3456e8609","when":null,"workflow_outputs":[{"label":"Flye assembly (consensus)","output_name":"consensus","uuid":"3960e31d-7a9e-400c-bb21-f6e47b75e649"},{"label":"Flye assembly (assembly_graph)","output_name":"assembly_graph","uuid":"e524f295-a957-4c91-838c-f8e98e809b6c"},{"label":"Flye assembly (Graphical Fragment Assembly)","output_name":"assembly_gfa","uuid":"48b854e2-dd6e-4345-8d05-09abca6659da"},{"label":"Flye assembly (assembly_info)","output_name":"assembly_info","uuid":"8672c172-71a7-432c-9679-a8e37f36cf53"}]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+galaxy1","errors":null,"id":2,"input_connections":{"mode|in|inputs":{"id":1,"output_name":"consensus"}},"inputs":[],"label":"Quast genome report","name":"Quast","outputs":[{"name":"report_html","type":"html"}],"position":{"left":562.9918212890625,"top":0},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+galaxy1","tool_shed_repository":{"changeset_revision":"72472698a2df","name":"quast","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"__job_resource\\": {\\"__current_case__\\": 0, \\"__job_resource__select\\": \\"no\\"}, \\"advanced\\": {\\"contig_thresholds\\": \\"0,1000\\", \\"strict_NA\\": false, \\"extensive_mis_size\\": \\"1000\\", \\"scaffold_gap_max_size\\": \\"1000\\", \\"unaligned_part_size\\": \\"500\\", \\"skip_unaligned_mis_contigs\\": true, \\"fragmented_max_indent\\": null, \\"report_all_metrics\\": false, \\"x_for_Nx\\": \\"90\\"}, \\"al\\": {\\"ambiguity_score\\": \\"0.99\\", \\"ambiguity_usage\\": \\"one\\", \\"fragmented\\": false, \\"fragmented_max_indent\\": \\"50\\", \\"min_alignment\\": \\"65\\", \\"min_identity\\": \\"95.0\\", \\"upper_bound_assembly\\": false, \\"upper_bound_min_con\\": \\"2\\", \\"use_all_alignments\\": false}, \\"alignments\\": {\\"use_all_alignments\\": false, \\"min_alignment\\": \\"65\\", \\"ambiguity_usage\\": \\"one\\", \\"ambiguity_score\\": \\"0.99\\", \\"fragmented\\": false, \\"upper_bound_assembly\\": false, \\"upper_bound_min_con\\": null, \\"local_mis_size\\": \\"200\\"}, \\"assembly\\": {\\"type\\": \\"genome\\", \\"__current_case__\\": 0, \\"ref\\": {\\"use_ref\\": \\"false\\", \\"__current_case__\\": 1, \\"est_ref_size\\": null}, \\"orga_type\\": \\"\\", \\"min_identity\\": \\"95.0\\"}, \\"circos\\": \\"false\\", \\"contig_thresholds\\": \\"0,1000\\", \\"extensive_mis_size\\": \\"1000\\", \\"genes\\": {\\"gene_finding\\": {\\"tool\\": \\"none\\", \\"__current_case__\\": 0}, \\"rna_finding\\": false, \\"conserved_genes_finding\\": false}, \\"in\\": {\\"__current_case__\\": 1, \\"custom\\": \\"false\\", \\"inputs\\": {\\"__class__\\": \\"RuntimeValue\\"}}, \\"k_mer\\": {\\"__current_case__\\": 1, \\"k_mer_stats\\": \\"\\"}, \\"large\\": true, \\"min_contig\\": \\"500\\", \\"mode\\": {\\"mode\\": \\"co\\", \\"__current_case__\\": 1, \\"in\\": {\\"custom\\": \\"false\\", \\"__current_case__\\": 1, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"reads\\": {\\"reads_option\\": \\"disabled\\", \\"__current_case__\\": 0}}, \\"output_files\\": \\"html\\", \\"scaffold_gap_max_size\\": \\"1000\\", \\"skip_unaligned_mis_contigs\\": \\"false\\", \\"split_scaffolds\\": false, \\"strict_NA\\": \\"false\\", \\"unaligned_part_size\\": \\"500\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.2.0+galaxy1","type":"tool","uuid":"5bc29646-ec4c-4eb3-af01-73d82fc690a2","when":null,"workflow_outputs":[{"label":"Quast: HTML report","output_name":"report_html","uuid":"17cdf8e0-8ad4-4570-afae-1861934fc678"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasta_stats/fasta-stats/2.0","errors":null,"id":3,"input_connections":{"fasta":{"id":1,"output_name":"consensus"}},"inputs":[],"label":"Fasta statistics","name":"Fasta Statistics","outputs":[{"name":"stats_output","type":"tabular"}],"position":{"left":569.7271118164062,"top":189.37097199902308},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasta_stats/fasta-stats/2.0","tool_shed_repository":{"changeset_revision":"0dbb995c7d35","name":"fasta_stats","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"dataset\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"gaps_option\\": false, \\"genome_size\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0","type":"tool","uuid":"811b5756-e24e-4de3-a3d6-2ee4bf347c1b","when":null,"workflow_outputs":[{"label":"Flye assembly statistics","output_name":"stats_output","uuid":"49f24151-e76c-4d94-8045-4d52bcd9aae5"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":4,"input_connections":{"input_file":{"id":1,"output_name":"assembly_gfa"}},"inputs":[],"label":"Bandage image: Flye assembly","name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":571.0624389648438,"top":311.80291748046875},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"1000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": null, \\"output_format\\": \\"jpg\\", \\"width\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"395f2dfe-3fa5-4684-bf97-7dd12e970f78","when":null,"workflow_outputs":[{"label":"Bandage Image: Assembly Graph Image","output_name":"outfile","uuid":"e66bd129-146f-48dc-95b8-39a2a1ffb68d"}]}},"tags":[],"uuid":"83ef6fdc-b8a9-41af-a824-0225f0199e63","version":21},"readme":"# Genome assembly with Flye workflow\\n\\n\\n## Why use this workflow?\\n\\n- This is a fairly simple workflow that assembles a genome from long sequencing reads.\\n- It takes in sequencing reads from PacBio (Hifi or non-Hifi), or Oxford Nanopore.\\n- If you have PacBio Hifi reads, you may prefer to use a workflow with the assembly tool Hifiasm, such as the those in the suite of VGP workflows. \\n\\n## Inputs\\n\\nRaw sequencing reads from PacBio or Oxford Nanopore in format:\\nfasta, fasta.gz, fastq, fastq.gz, fastqsanger.gz or fastqsanger\\n\\n## What does the workflow do\\n\\n- Assembles the reads with the tool Flye\\n- Summarizes the statistics with the tool Fasta statistics\\n- Report with the tool Quast\\n- Renders the assembly graph with the tool Bandage\\n\\n## Settings\\n\\nRun as-is or change parameters at runtime\\n\\nFor example:\\n- change the Flye option of \\"mode\\" to the correct sequencing type\\n- change the Quast option for \\"Type of organism\\" to correct taxon\\n \\n## Outputs\\n\\n- Flye assembly output - four files: fasta, gfa for bandage, graph_dot file, assembly info\\n- Fasta statistics\\n- Bandage image\\n- Quast report\\n","changelog":"# Changelog\\n\\n## [0.2] - 2024-03-25\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.3+galaxy0`\\n\\nAll notable changes to this project will be documented in this file.\\n\\nThe format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),\\nand this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).\\n\\n## [0.1] - 2023-07-14\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Genome assembly with Flye\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input sequence reads\\"]@{ shape: doc }\\n1[\\"Flye: assembly\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Quast genome report\\"]@{ shape: process }\\n1 --> 2\\n3[\\"Fasta statistics\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Bandage image: Flye assembly\\"]@{ shape: process }\\n1 --> 4\\n```\\n","updated":"2024-03-25T00:00:00","trsID":"#workflow/github.com/iwc-workflows/assembly-with-flye/main","dockstore_id":25508,"categories":[],"collections":["Genome assembly"],"doi":"10.5281/zenodo.15078846","tests":[{"doc":"Test outline for Genome-assembly-with-Flye","job":{"Input sequence reads":{"class":"File","path":"test-data/Input_reads.fastqsanger.gz","filetype":"fastqsanger.gz"}},"outputs":{"Flye assembly (consensus)":{"asserts":{"has_text":{"text":">contig_1"},"has_size":{"min":"80k","max":"86k"}}},"Flye assembly (assembly_graph)":{"asserts":{"has_text":{"text":"digraph {"},"has_n_lines":{"min":20,"max":24}}},"Flye assembly (Graphical Fragment Assembly)":{"asserts":{"has_text":{"text":"edge_1"},"has_n_lines":{"min":6,"max":8}}},"Flye assembly (assembly_info)":{"asserts":{"has_text":{"text":"seq_name"}}},"Quast: HTML report":{"asserts":{"has_size":{"min":"250k","max":"400k"}}},"Flye assembly statistics":{"asserts":{"has_text":{"text":"Scaffold L50"}}},"Bandage Image: Assembly Graph Image":{"asserts":{"has_size":{"min":"30k","max":"50k"}}}}}]}],"path":"./workflows/genome-assembly/assembly-with-flye"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/quality_and_contamination_control.ga","testParameterFiles":["/quality_and_contamination_control-tests.yml"],"authors":[{"name":"ABRomics","email":"abromics-support@groupes.france-bioinformatique.fr"},{"name":"abromics-consortium","url":"https://www.abromics.fr/"},{"name":"Pierre Marin","alternateName":"pimarin","orcid":"0000-0002-8304-138X"},{"name":"Clea Siguret","alternateName":"clsiguret","orcid":"0009-0005-6140-0379"}],"definition":{"a_galaxy_workflow":"true","annotation":"Short paired-end read analysis to provide quality analysis, read cleaning and taxonomy assignation","creator":[{"class":"Person","email":"mailto:abromics-support@groupes.france-bioinformatique.fr","name":"ABRomics"},{"class":"Organization","name":"abromics-consortium","url":"https://www.abromics.fr/"},{"alternateName":"pimarin","class":"Person","identifier":"https://orcid.org/0000-0002-8304-138X","name":"Pierre Marin"},{"alternateName":"clsiguret","class":"Person","identifier":"https://orcid.org/0009-0005-6140-0379","name":"Clea Siguret"}],"format-version":"0.1","license":"GPL-3.0-or-later","release":"1.1.6","name":"Quality and Contamination Control For Genome Assembly","steps":{"0":{"annotation":"Should be the forward strand of the paired end reads.","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Should be the forward strand of the paired end reads.","name":"Input sequence reads (forward)"}],"label":"Input sequence reads (forward)","name":"Input dataset","outputs":[],"position":{"left":21.02195459971947,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\", \\"fastq.gz\\", \\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"05125e59-489b-4896-8e58-8bf49a8dde78","when":null,"workflow_outputs":[]},"1":{"annotation":"Should be the reverse strand of the paired end reads.","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Should be the reverse strand of the paired end reads.","name":"Input sequence reads (reverse)"}],"label":"Input sequence reads (reverse)","name":"Input dataset","outputs":[],"position":{"left":17.95353354143969,"top":108.04692569668924},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"format\\": [\\"fastq\\", \\"fastq.gz\\", \\"fastqsanger\\", \\"fastqsanger.gz\\"], \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"f0763317-fde5-4f81-9297-0163663b1742","when":null,"workflow_outputs":[]},"2":{"annotation":"Select the database to use for assigning taxonomies for Kraken2 and Bracken.","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Select the database to use for assigning taxonomies for Kraken2 and Bracken.","name":"Select a taxonomy database"}],"label":"Select a taxonomy database","name":"Input parameter","outputs":[],"position":{"left":0,"top":307.433349609375},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"1a56b57e-8893-4cbc-a984-329e0b815cbd","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"62ded62a-d0c4-49ba-b3ef-c97ff39c6c5e"}]},"3":{"annotation":"Select the ncbi taxonomy database for Recentrifuge.","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Select the ncbi taxonomy database for Recentrifuge.","name":"Select a NCBI taxonomy database"}],"label":"Select a NCBI taxonomy database","name":"Input parameter","outputs":[],"position":{"left":5.9666748046875,"top":454.91066885428484},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"daac0634-bcf4-4fdd-bcd2-67247f5afdd0","when":null,"workflow_outputs":[{"label":null,"output_name":"output","uuid":"9b99164e-0a87-4bba-8838-ec4f428cf4d5"}]},"4":{"annotation":"fastp_triming, quality analaysis and read cleaning","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","errors":null,"id":4,"input_connections":{"single_paired|in1":{"id":0,"output_name":"output"},"single_paired|in2":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"},{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":"fastp_trimming_step","name":"fastp","outputs":[{"name":"out1","type":"input"},{"name":"out2","type":"input"},{"name":"report_html","type":"html"},{"name":"report_json","type":"json"}],"position":{"left":256.3279681321078,"top":11.29864083194758},"post_job_actions":{"ChangeDatatypeActionout1":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"out1"},"ChangeDatatypeActionout2":{"action_arguments":{"newtype":"fastqsanger.gz"},"action_type":"ChangeDatatypeAction","output_name":"out2"},"RenameDatasetActionout1":{"action_arguments":{"newname":"fastp_trimmed_R1"},"action_type":"RenameDatasetAction","output_name":"out1"},"RenameDatasetActionout2":{"action_arguments":{"newname":"fastp_trimmed_R2"},"action_type":"RenameDatasetAction","output_name":"out2"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"fastp_report_html"},"action_type":"RenameDatasetAction","output_name":"report_html"},"RenameDatasetActionreport_json":{"action_arguments":{"newname":"fastp_report_json"},"action_type":"RenameDatasetAction","output_name":"report_json"},"TagDatasetActionout1":{"action_arguments":{"tags":"R1"},"action_type":"TagDatasetAction","output_name":"out1"},"TagDatasetActionout2":{"action_arguments":{"tags":"R2"},"action_type":"TagDatasetAction","output_name":"out2"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"fastp,html"},"action_type":"TagDatasetAction","output_name":"report_html"},"TagDatasetActionreport_json":{"action_arguments":{"tags":"fastp,report,json"},"action_type":"TagDatasetAction","output_name":"report_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3","tool_shed_repository":{"changeset_revision":"a626e8c0e1ba","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": true}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": null, \\"umi_len\\": null, \\"umi_prefix\\": null}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"paired\\", \\"__current_case__\\": 1, \\"in1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"in2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"merge_reads\\": {\\"merge\\": \\"\\", \\"__current_case__\\": 1}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": null, \\"adapter_sequence2\\": null, \\"detect_adapter_for_pe\\": false}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null, \\"trim_front2\\": null, \\"trim_tail2\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.24.0+galaxy3","type":"tool","uuid":"7f405009-2491-4633-b4e1-ca0fdbd8360d","when":null,"workflow_outputs":[{"label":"fastp_trimmed_R1","output_name":"out1","uuid":"d596e01c-9180-4482-9326-cc8289427fa1"},{"label":"fastp_trimmed_R2","output_name":"out2","uuid":"44c14c4e-40af-4045-856c-f2429af36b98"},{"label":"fastp_report_html","output_name":"report_html","uuid":"f6b7db2c-fdd8-4d7c-81c0-6d4a9dd76ff7"},{"label":"fastp_report_json","output_name":"report_json","uuid":"4ce40cda-0c59-4128-b8fe-d2fe67fe0ba9"}]},"5":{"annotation":"kraken_taxonomy_assignation","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1","errors":null,"id":5,"input_connections":{"kraken2_database":{"id":2,"output_name":"output"},"single_paired|forward_input":{"id":4,"output_name":"out1"},"single_paired|reverse_input":{"id":4,"output_name":"out2"}},"inputs":[{"description":"runtime parameter for tool Kraken2","name":"single_paired"},{"description":"runtime parameter for tool Kraken2","name":"single_paired"}],"label":"kraken_taxonomy_assignation","name":"Kraken2","outputs":[{"name":"report_output","type":"tabular"},{"name":"output","type":"tabular"}],"position":{"left":563.3447180137721,"top":148.60919325423987},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"kraken_report_reads"},"action_type":"RenameDatasetAction","output_name":"output"},"RenameDatasetActionreport_output":{"action_arguments":{"newname":"kraken_report_tabular"},"action_type":"RenameDatasetAction","output_name":"report_output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"kraken2,report,reads"},"action_type":"TagDatasetAction","output_name":"output"},"TagDatasetActionreport_output":{"action_arguments":{"tags":"kraken2,report,tabular"},"action_type":"TagDatasetAction","output_name":"report_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1","tool_shed_repository":{"changeset_revision":"cdee7158adf3","name":"kraken2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"confidence\\": \\"0.0\\", \\"kraken2_database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_base_quality\\": \\"10\\", \\"minimum_hit_groups\\": \\"2\\", \\"quick\\": false, \\"report\\": {\\"create_report\\": true, \\"use_mpa_style\\": false, \\"report_zero_counts\\": false, \\"report_minimizer_data\\": false}, \\"single_paired\\": {\\"single_paired_selector\\": \\"yes\\", \\"__current_case__\\": 1, \\"forward_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reverse_input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split_reads\\": false, \\"use_names\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.3+galaxy1","type":"tool","uuid":"e414f0c1-5e79-472c-bf06-a3984aa3a46b","when":null,"workflow_outputs":[{"label":"kraken_report_tabular","output_name":"report_output","uuid":"68f9f0c0-375c-43ba-8687-9e2a1e602e8e"},{"label":"kraken_report_reads","output_name":"output","uuid":"267ff8a0-b6bc-4fcf-97b9-bbf5a3d0c05f"}]},"6":{"annotation":"bracken_abundance_estimation","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/3.0+galaxy0","errors":null,"id":6,"input_connections":{"input":{"id":5,"output_name":"report_output"},"kmer_distr":{"id":2,"output_name":"output"}},"inputs":[],"label":"bracken_abundance_estimation","name":"Bracken","outputs":[{"name":"report","type":"tabular"},{"name":"kraken_report","type":"tabular"}],"position":{"left":810.7535213344084,"top":77.49065954589844},"post_job_actions":{"RenameDatasetActionkraken_report":{"action_arguments":{"newname":"bracken_kraken_report"},"action_type":"RenameDatasetAction","output_name":"kraken_report"},"RenameDatasetActionreport":{"action_arguments":{"newname":"bracken_report_tsv"},"action_type":"RenameDatasetAction","output_name":"report"},"TagDatasetActionkraken_report":{"action_arguments":{"tags":"bracken_kraken_report"},"action_type":"TagDatasetAction","output_name":"kraken_report"},"TagDatasetActionreport":{"action_arguments":{"tags":"bracken,report"},"action_type":"TagDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/3.0+galaxy0","tool_shed_repository":{"changeset_revision":"0c7b6eb7e752","name":"bracken","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmer_distr\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"level\\": \\"S\\", \\"logfile_output\\": false, \\"out_report\\": true, \\"threshold\\": \\"10\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.0+galaxy0","type":"tool","uuid":"e088cdc6-d79b-453c-8787-6d4b7004312a","when":null,"workflow_outputs":[{"label":"bracken_kraken_report","output_name":"kraken_report","uuid":"63933cf1-8f43-4481-98ff-3a7c36fbaf00"},{"label":"bracken_report_tsv","output_name":"report","uuid":"2d962685-0630-4cf8-a943-800eab2d4f67"}]},"7":{"annotation":"recentrifuge_taxonomy_visualization","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.15.0+galaxy0","errors":null,"id":7,"input_connections":{"database|database_name":{"id":3,"output_name":"output"},"input_file":{"id":5,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Recentrifuge","name":"database"}],"label":"recentrifuge_taxonomy_visualization","name":"Recentrifuge","outputs":[{"name":"html_report","type":"html"},{"name":"logfile","type":"txt"},{"name":"data_tsv","type":"tabular"},{"name":"stat_tsv","type":"tabular"}],"position":{"left":809.9535335414397,"top":405.274},"post_job_actions":{"RenameDatasetActiondata_tsv":{"action_arguments":{"newname":"recentrifuge_data_tabular"},"action_type":"RenameDatasetAction","output_name":"data_tsv"},"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"recentrifuge_report_html"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionlogfile":{"action_arguments":{"newname":"recentrifuge_logfile"},"action_type":"RenameDatasetAction","output_name":"logfile"},"RenameDatasetActionstat_tsv":{"action_arguments":{"newname":"recentrifuge_stats_tabular"},"action_type":"RenameDatasetAction","output_name":"stat_tsv"},"TagDatasetActiondata_tsv":{"action_arguments":{"tags":"recentrifuge,data"},"action_type":"TagDatasetAction","output_name":"data_tsv"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"recentrifuge_report_html"},"action_type":"TagDatasetAction","output_name":"html_report"},"TagDatasetActionlogfile":{"action_arguments":{"tags":"logfile"},"action_type":"TagDatasetAction","output_name":"logfile"},"TagDatasetActionstat_tsv":{"action_arguments":{"tags":"recentrifuge,stats"},"action_type":"TagDatasetAction","output_name":"stat_tsv"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.15.0+galaxy0","tool_shed_repository":{"changeset_revision":"591175c0f051","name":"recentrifuge","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_option\\": {\\"controls\\": \\"0\\", \\"scoring\\": null, \\"minscore_value\\": \\"0\\", \\"mintaxa\\": \\"0\\", \\"exclude_taxa_name\\": null, \\"include_taxa_name\\": null, \\"avoidcross\\": false}, \\"database\\": {\\"database_name\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"file_type\\": {\\"filetype\\": \\"kraken\\", \\"__current_case__\\": 3}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"more_advanced_option\\": {\\"ctrlminscore\\": \\"0\\", \\"ctrlmintaxa\\": \\"0\\", \\"summary\\": \\"ADD\\", \\"takeoutroot\\": false, \\"nokollapse\\": false, \\"strain\\": true, \\"sequential\\": false}, \\"output_option\\": {\\"extra\\": \\"TSV\\", \\"nohtml\\": false, \\"no_logfile\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.0+galaxy0","type":"tool","uuid":"2db5d20e-62c7-43ad-b21f-e9992c77faa1","when":null,"workflow_outputs":[{"label":"recentrifuge_report_html","output_name":"html_report","uuid":"1cb280ed-a5dc-4d26-8ce4-833b3c3ffe47"},{"label":"recentrifuge_stats_tabular","output_name":"stat_tsv","uuid":"62543f1b-765b-4022-bf57-ce2a7f6d3845"},{"label":"recentrifuge_data_tabular","output_name":"data_tsv","uuid":"2c9ca7c0-6fd8-4324-8684-94e4d1fa24ba"},{"label":"recentrifuge_logfile","output_name":"logfile","uuid":"f09996c8-a1be-484c-9b0d-3d96a68ae8c7"}]},"8":{"annotation":"ToolDistillator extracts results from tools and creates a JSON file for each tool","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","errors":null,"id":8,"input_connections":{"tool_section|tools_0|select_tool|html_report_path":{"id":4,"output_name":"report_html"},"tool_section|tools_0|select_tool|input":{"id":4,"output_name":"report_json"},"tool_section|tools_0|select_tool|trimmed_forward_R1_path":{"id":4,"output_name":"out1"},"tool_section|tools_0|select_tool|trimmed_reverse_R2_path":{"id":4,"output_name":"out2"},"tool_section|tools_1|select_tool|input":{"id":5,"output_name":"report_output"},"tool_section|tools_1|select_tool|reference_database_version":{"id":2,"output_name":"output"},"tool_section|tools_1|select_tool|seq_classification_file_path":{"id":5,"output_name":"output"},"tool_section|tools_2|select_tool|input":{"id":6,"output_name":"report"},"tool_section|tools_2|select_tool|kraken_report_path":{"id":6,"output_name":"kraken_report"},"tool_section|tools_2|select_tool|reference_database_version":{"id":2,"output_name":"output"},"tool_section|tools_3|select_tool|input":{"id":7,"output_name":"data_tsv"},"tool_section|tools_3|select_tool|rcf_html_path":{"id":7,"output_name":"html_report"},"tool_section|tools_3|select_tool|rcf_stat_path":{"id":7,"output_name":"stat_tsv"},"tool_section|tools_3|select_tool|reference_database_version":{"id":3,"output_name":"output"}},"inputs":[],"label":"ToolDistillator","name":"ToolDistillator","outputs":[{"name":"output_json","type":"input"}],"position":{"left":1208.036907564877,"top":47.093340454101565},"post_job_actions":{"RenameDatasetActionoutput_json":{"action_arguments":{"newname":"tooldistillator_results"},"action_type":"RenameDatasetAction","output_name":"output_json"},"TagDatasetActionoutput_json":{"action_arguments":{"tags":"tooldistillator_results"},"action_type":"TagDatasetAction","output_name":"output_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"ea93df4b3df2","name":"tooldistillator","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"log\\": false, \\"tool_section\\": {\\"tools\\": [{\\"__index__\\": 0, \\"select_tool\\": {\\"tool_list\\": \\"fastp\\", \\"__current_case__\\": 6, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trimmed_forward_R1_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"trimmed_reverse_R2_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"html_report_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": null}}, {\\"__index__\\": 1, \\"select_tool\\": {\\"tool_list\\": \\"kraken2\\", \\"__current_case__\\": 12, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seq_classification_file_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 2, \\"select_tool\\": {\\"tool_list\\": \\"bracken\\", \\"__current_case__\\": 4, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kraken_report_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"threshold\\": null, \\"read_len\\": \\"100\\", \\"level\\": \\"S\\"}}, {\\"__index__\\": 3, \\"select_tool\\": {\\"tool_list\\": \\"recentrifuge\\", \\"__current_case__\\": 17, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rcf_stat_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"rcf_html_path\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"origin\\": {\\"origin\\": \\"true\\", \\"__current_case__\\": 0, \\"analysis_software_version\\": null}, \\"reference_database_version\\": {\\"__class__\\": \\"ConnectedValue\\"}}}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"b88b0ffe-f660-43ea-b00a-40ca0087f83f","when":null,"workflow_outputs":[{"label":"tooldistillator_results","output_name":"output_json","uuid":"d0ad8d88-a27f-439a-8523-6e0626401c58"}]},"9":{"annotation":"ToolDistillator summarize groups all JSON file into a unique JSON file","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","errors":null,"id":9,"input_connections":{"summarize_data":{"id":8,"output_name":"output_json"}},"inputs":[],"label":"ToolDistillator summarize","name":"ToolDistillator Summarize","outputs":[{"name":"summary_json","type":"json"}],"position":{"left":1509.353557955502,"top":342.8073251953125},"post_job_actions":{"RenameDatasetActionsummary_json":{"action_arguments":{"newname":"tooldistillator_summarize"},"action_type":"RenameDatasetAction","output_name":"summary_json"},"TagDatasetActionsummary_json":{"action_arguments":{"tags":"tooldistillator_summarize"},"action_type":"TagDatasetAction","output_name":"summary_json"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1","tool_shed_repository":{"changeset_revision":"24eec94e6dfc","name":"tooldistillator_summarize","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"summarize_data\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.9.1+galaxy1","type":"tool","uuid":"fbc8d552-c0b1-426c-b26f-8bfe8371b05e","when":null,"workflow_outputs":[{"label":"tooldistillator_summarize","output_name":"summary_json","uuid":"2db8147f-05b9-4ddc-a3db-bd31fb57c690"}]}},"tags":["Genomics","fastq","bacterial-genomics","taxonomy-assignment","paired-end","quality","ABRomics","trimming"],"uuid":"eb1c667d-639d-4403-a2b4-1cb6683e0fa5","version":1},"readme":"# Quality and Contamination control workflow for paired end data (v1.0)\\n\\nThis workflow uses paired-end illumina fastq(.gz) files and executes the following steps:\\n1. Quality control and trimming\\n    - **fastp** QC control and trimming\\n2. Taxonomic assignation on trimmed data\\n    - **Kraken2** assignation\\n    - **Bracken** to re-estimate abundance to the species level\\n    - **Recentrifuge** to make a krona chart\\n3. Aggregating outputs into a single JSON file\\n    - **ToolDistillator** to extract and aggregate information from different tool outputs to JSON parsable files\\n\\n## Inputs\\n\\n1. Paired-end illumina raw reads in fastq(.gz) format.\\n\\n## Outputs\\n\\n1. Quality control:\\n    - quality report\\n    - trimmed raw reads\\n2. Taxonomic assignation:\\n    - Tabular report of identified species\\n    - Tabular file with assigned read to a taxonomic level\\n    - Krona chart to illustrate species diversity of the sample\\n3. Aggregating outputs:\\n    - JSON file with information about the outputs of **fastp**, **Kraken2**, **Bracken**, **Recentrifuge** \\n","changelog":"# Changelog\\n\\n## [1.1.6] - 2024-11-18\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.24.0+galaxy3`\\n\\n## [1.1.5] - 2024-10-21\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy1`\\n\\n## [1.1.4] - 2024-10-10\\n\\n### Manual update\\n\\n- Changed input parameters to select databases in WFs. Use \\"Attempt restriction based on connections\\" instead of \\"Provide list of suggested values\\".\\n\\n## [1.1.3] - 2024-09-30\\n\\n### Manual update\\n\\n- Updated Recentrifuge version manually in the ToolDistillator parameters\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.14.1+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.15.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9.1+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9.1+galaxy0`\\n\\n## [1.1.2] - 2024-08-05\\n\\n### Manual update\\n\\n- Updated the tool versions manually in the ToolDistillator parameters\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/2.9+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/3.0+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.14.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/recentrifuge/recentrifuge/1.14.1+galaxy0`\\n\\n## [1.1.1] - 2024-07-08\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator/tooldistillator/0.9+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.8.5.0+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.9+galaxy0`\\n\\n## [1.1] - 2024-06-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.3+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/2.8+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bracken/est_abundance/2.9+galaxy0`\\n\\n## [1.0] - 04-06-2024\\n\\n- First release\\n","diagrams":"# Workflow diagrams\\n\\n## Quality and Contamination Control For Genome Assembly\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ Input sequence reads (forward)\\"]@{ shape: doc }\\n1[\\"ℹ️ Input sequence reads (reverse)\\"]@{ shape: doc }\\n2[\\"ℹ️ Select a taxonomy database\\"]@{ shape: lean-l }\\n3[\\"ℹ️ Select a NCBI taxonomy database\\"]@{ shape: lean-l }\\n4[\\"fastp_trimming_step\\"]@{ shape: process }\\n0 --> 4\\n1 --> 4\\n5[\\"kraken_taxonomy_assignation\\"]@{ shape: process }\\n2 --> 5\\n4 --> 5\\n4 --> 5\\n6[\\"bracken_abundance_estimation\\"]@{ shape: process }\\n5 --> 6\\n2 --> 6\\n7[\\"recentrifuge_taxonomy_visualization\\"]@{ shape: process }\\n3 --> 7\\n5 --> 7\\n8[\\"ToolDistillator\\"]@{ shape: process }\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n4 --> 8\\n5 --> 8\\n2 --> 8\\n5 --> 8\\n6 --> 8\\n6 --> 8\\n2 --> 8\\n7 --> 8\\n7 --> 8\\n7 --> 8\\n3 --> 8\\n9[\\"ToolDistillator summarize\\"]@{ shape: process }\\n8 --> 9\\n```\\n","updated":"2024-11-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/quality-and-contamination-control/main","dockstore_id":27292,"categories":[],"collections":["Genome assembly"],"doi":"10.5281/zenodo.15078503","tests":[{"doc":"Test outline for quality_and_contamination_control.ga","job":{"Input sequence reads (forward)":{"class":"File","path":"https://zenodo.org/records/11484215/files/paired_r1.fastq.gz","filetype":"fastqsanger.gz"},"Input sequence reads (reverse)":{"class":"File","path":"https://zenodo.org/records/11484215/files/paired_r2.fastq.gz","filetype":"fastqsanger.gz"},"Select a taxonomy database":"k2_minusb_20210517","Select a NCBI taxonomy database":"2022-03-08"},"outputs":{"fastp_report_json":{"asserts":{"has_text":{"text":"read2_before_filtering"}}},"kraken_report_tabular":{"asserts":{"has_text":{"text":"Enterococcus avium"},"has_n_columns":{"n":6}}},"kraken_report_reads":{"asserts":{"has_text":{"text":"M07044:90:000000000-JRJWP:1:1119:23974:4461"},"has_n_columns":{"n":5}}},"bracken_kraken_report":{"asserts":{"has_text":{"text":"Enterococcus gallinarum"},"has_n_columns":{"n":6}}},"bracken_report_tsv":{"asserts":{"has_text":{"text":"Escherichia coli"},"has_n_columns":{"n":7}}},"recentrifuge_data_tabular":{"asserts":{"has_text":{"text":"Enterococcus faecalis"},"has_n_columns":{"n":6}}},"recentrifuge_stats_tabular":{"asserts":{"has_text":{"text":"input_dir/kraken_report_reads"},"has_n_columns":{"n":2}}},"tooldistillator_summarize":{"asserts":[{"that":"has_text","text":"fastp_report"},{"that":"has_text","text":"ncbi_taxonomic_id"},{"that":"has_text","text":"kraken2_report"}]}}}]}],"path":"./workflows/genome-assembly/quality-and-contamination-control"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Gene-based-Pathogen-Identification.ga","testParameterFiles":["/Gene-based-Pathogen-Identification-tests.yml"],"authors":[{"name":"Engy Nasr","orcid":"0000-0001-9047-4215","url":"https://orcid.org/0000-0001-9047-4215"},{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987","url":"https://orcid.org/0000-0001-9852-1987"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X","url":"https://orcid.org/0000-0003-2982-388X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Nanopore datasets analysis - Phylogenetic Identification - antibiotic resistance genes detection and contigs building","comments":[{"child_steps":[10,13],"color":"turquoise","data":{"title":"Identifying VF"},"id":2,"position":[1821.7,0.6000000000000014],"size":[461.7,234.8],"type":"frame"},{"child_steps":[11,14],"color":"turquoise","data":{"title":"Identifying AMR"},"id":3,"position":[1830.2,640.3],"size":[521.5,234.8],"type":"frame"},{"child_steps":[2,4,6,7,9,12,15],"color":"turquoise","data":{"title":"Extracting Contigs from our Samples"},"id":1,"position":[310,490.20000000000005],"size":[1469,676],"type":"frame"},{"child_steps":[1,3,5,8],"color":"turquoise","data":{"title":"Extract Samples IDs"},"id":0,"position":[302.4,0],"size":[1060.5,234.8],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0001-9047-4215","name":"Engy Nasr","url":"https://orcid.org/0000-0001-9047-4215"},{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut","url":"https://orcid.org/0000-0001-9852-1987"},{"class":"Person","identifier":"0000-0003-2982-388X","name":"Paul Zierep"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Gene-based Pathogen Identification","report":{"markdown":"# Gene-based Pathogen Identification Workflow Report\\n\\nBelow are the results for the Gene based Pathogen Identification Workflow\\n\\nThis workflow was run on:\\n\\n```galaxy\\ngenerate_time()\\n```\\n\\nWith Galaxy version:\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Workflow Inputs\\nThe Perprocessing workflow main output (Collection of all samples reads after quality retaining and hosts filtering)\\n\\n## Workflow Output\\n\\n### Identified contigs per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"contigs\\")\\n```\\n\\n### Identified pathogenic genes per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"vfs\\")\\n```\\n\\n### Identified Antimicrobial Resistance genes per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"amrs\\")\\n```\\n"},"steps":{"0":{"annotation":"Output collection from the Nanopore Preprocessing workflow","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Output collection from the Nanopore Preprocessing workflow","name":"collection_of_preprocessed_samples"}],"label":"collection_of_preprocessed_samples","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":316.35},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"fb7f7c4b-70a7-4d6d-8b20-a3bfdc4380ff","when":null,"workflow_outputs":[]},"1":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","errors":null,"id":1,"input_connections":{"input_collection":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Extract element identifiers","outputs":[{"name":"output","type":"txt"}],"position":{"left":322.37152236646415,"top":59.69265140673434},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"extracted_samples_IDs"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2","tool_shed_repository":{"changeset_revision":"d3c07d270a50","name":"collection_element_identifiers","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_collection\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.2","type":"tool","uuid":"80a87744-c9fa-498c-9412-b59346fb7904","when":null,"workflow_outputs":[{"label":"extracted_samples_IDs","output_name":"output","uuid":"d82a93c6-23a9-4f85-879c-ecd759a31087"}]},"2":{"annotation":"","content_id":"__BUILD_LIST__","errors":null,"id":2,"input_connections":{"datasets_0|input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Build list","outputs":[{"name":"output","type":"input"}],"position":{"left":330,"top":597.5999771118164},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"list_of_lists_of_preprocessed_samples"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__BUILD_LIST__","tool_state":"{\\"datasets\\": [{\\"__index__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"id_cond\\": {\\"id_select\\": \\"idx\\", \\"__current_case__\\": 0}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"38bc82a3-b563-44cf-b0db-9e3311fe99bb","when":null,"workflow_outputs":[{"label":"list_of_lists_of_preprocessed_samples","output_name":"output","uuid":"f5b5b256-8ddf-4da8-8111-b1f6d3025a0d"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","errors":null,"id":3,"input_connections":{"split_parms|input":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Split file","name":"split_parms"}],"label":null,"name":"Split file","outputs":[{"name":"list_output_txt","type":"input"}],"position":{"left":608.6215223664642,"top":47.19265140673434},"post_job_actions":{"RenameDatasetActionlist_output_txt":{"action_arguments":{"newname":"splitted_extracted_samples_IDs"},"action_type":"RenameDatasetAction","output_name":"list_output_txt"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0","tool_shed_repository":{"changeset_revision":"6cbe2f30c2d7","name":"split_file_to_collection","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"split_parms\\": {\\"select_ftype\\": \\"txt\\", \\"__current_case__\\": 5, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"select_mode\\": {\\"mode\\": \\"chunk\\", \\"__current_case__\\": 0, \\"chunksize\\": \\"1\\"}, \\"newfilenames\\": \\"split_file\\", \\"select_allocate\\": {\\"allocate\\": \\"byrow\\", \\"__current_case__\\": 2}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.5.0","type":"tool","uuid":"f1e00cb8-a264-4acb-aff6-8c8b5aa0b48b","when":null,"workflow_outputs":[{"label":"splitted_extracted_samples_IDs","output_name":"list_output_txt","uuid":"eb5317bd-4bad-4cad-9219-3ac379221d6e"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.1+galaxy0","errors":null,"id":4,"input_connections":{"inputs":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Flye","outputs":[{"name":"consensus","type":"fasta"},{"name":"assembly_graph","type":"graph_dot"},{"name":"assembly_gfa","type":"txt"},{"name":"assembly_info","type":"tabular"}],"position":{"left":576.4000244140625,"top":602.433349609375},"post_job_actions":{"RenameDatasetActionassembly_gfa":{"action_arguments":{"newname":"flye_assembly_gfa"},"action_type":"RenameDatasetAction","output_name":"assembly_gfa"},"RenameDatasetActionassembly_graph":{"action_arguments":{"newname":"flye_assembly_graph"},"action_type":"RenameDatasetAction","output_name":"assembly_graph"},"RenameDatasetActionassembly_info":{"action_arguments":{"newname":"flye_assembly_info_tabular"},"action_type":"RenameDatasetAction","output_name":"assembly_info"},"RenameDatasetActionconsensus":{"action_arguments":{"newname":"flye_consensus_fasta"},"action_type":"RenameDatasetAction","output_name":"consensus"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9.1+galaxy0","tool_shed_repository":{"changeset_revision":"cb8dfd28c16f","name":"flye","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"asm\\": {\\"asm_select\\": \\"false\\", \\"__current_case__\\": 1}, \\"generate_log\\": false, \\"inputs\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"iterations\\": \\"1\\", \\"keep_haplotypes\\": false, \\"meta\\": true, \\"min_overlap\\": null, \\"mode_conditional\\": {\\"mode\\": \\"--nano-hq\\", \\"__current_case__\\": 2}, \\"no_alt_contigs\\": false, \\"scaffold\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.9.1+galaxy0","type":"tool","uuid":"a343c482-44b1-4cb3-a46e-2e714f785960","when":null,"workflow_outputs":[{"label":"flye_consensus_fasta","output_name":"consensus","uuid":"ff0d8c7b-2ac7-4c6d-a110-f71418dad938"},{"label":"flye_assembly_graph","output_name":"assembly_graph","uuid":"2385436f-fbe8-4d77-a40b-27a097d02941"},{"label":"flye_assembly_gfa","output_name":"assembly_gfa","uuid":"0e42ce51-c46b-45d8-baa4-45c0e98ac712"},{"label":"flye_assembly_info_tabular","output_name":"assembly_info","uuid":"3f297aed-7cee-4999-bbf4-69d84de6b64f"}]},"5":{"annotation":"","content_id":"param_value_from_file","errors":null,"id":5,"input_connections":{"input1":{"id":3,"output_name":"list_output_txt"}},"inputs":[],"label":null,"name":"Parse parameter value","outputs":[{"name":"text_param","type":"expression.json"}],"position":{"left":891.1214460725189,"top":39.942666665523404},"post_job_actions":{"RenameDatasetActiontext_param":{"action_arguments":{"newname":"parsed_extracted_samples_IDs_to_text"},"action_type":"RenameDatasetAction","output_name":"text_param"}},"tool_id":"param_value_from_file","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"param_type\\": \\"text\\", \\"remove_newlines\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.0","type":"tool","uuid":"ee2062e3-4a3e-4bd0-98a1-264704be0009","when":null,"workflow_outputs":[{"label":"parsed_extracted_samples_IDs_to_text","output_name":"text_param","uuid":"3608e170-c462-42a2-8003-f3f65baa3834"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline/1.7.2+galaxy0","errors":null,"id":6,"input_connections":{"d":{"id":4,"output_name":"consensus"},"i":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"medaka consensus pipeline","outputs":[{"name":"out_consensus","type":"fasta"},{"name":"out_probs","type":"h5"},{"name":"out_calls","type":"bam"},{"name":"out_gaps","type":"bed"},{"name":"out_log","type":"txt"}],"position":{"left":923.816650390625,"top":543.25},"post_job_actions":{"RenameDatasetActionout_calls":{"action_arguments":{"newname":"medaka_calls_of_draft_bam_file"},"action_type":"RenameDatasetAction","output_name":"out_calls"},"RenameDatasetActionout_consensus":{"action_arguments":{"newname":"sample_all_contigs"},"action_type":"RenameDatasetAction","output_name":"out_consensus"},"RenameDatasetActionout_gaps":{"action_arguments":{"newname":"medaka_gaps_in_draft_bed_file"},"action_type":"RenameDatasetAction","output_name":"out_gaps"},"RenameDatasetActionout_log":{"action_arguments":{"newname":"medaka_log_file"},"action_type":"RenameDatasetAction","output_name":"out_log"},"RenameDatasetActionout_probs":{"action_arguments":{"newname":"medaka_propability_h5_file"},"action_type":"RenameDatasetAction","output_name":"out_probs"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline/1.7.2+galaxy0","tool_shed_repository":{"changeset_revision":"79e9a869f40e","name":"medaka_consensus_pipeline","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"b\\": \\"100\\", \\"d\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"g\\": false, \\"i\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"m\\": \\"r941_min_hac_g507\\", \\"out\\": [\\"consensus\\", \\"probs\\", \\"calls\\", \\"log\\", \\"gaps\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.7.2+galaxy0","type":"tool","uuid":"9e88fc55-f41e-4e87-852e-044359dcea53","when":null,"workflow_outputs":[{"label":"medaka_gaps_in_draft_bed_file","output_name":"out_gaps","uuid":"9e5ad6ec-b408-4132-ba07-dec9fa626923"},{"label":"medaka_log_file","output_name":"out_log","uuid":"fcbd3e3f-2e93-4798-b696-dad7db9f2efd"},{"label":"medaka_propability_h5_file","output_name":"out_probs","uuid":"60656aac-ad2a-4c9b-9a68-b9fb18ae5595"},{"label":"medaka_calls_of_draft_bam_file","output_name":"out_calls","uuid":"c790d434-8e78-4df0-a0d4-8f9da0692158"},{"label":"sample_all_contigs","output_name":"out_consensus","uuid":"df361e19-b6d1-405b-96cc-b48c1ab7c604"}]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","errors":null,"id":7,"input_connections":{"input_file":{"id":4,"output_name":"assembly_gfa"}},"inputs":[],"label":null,"name":"Bandage Image","outputs":[{"name":"outfile","type":"jpg"}],"position":{"left":927.4666748046875,"top":965.5333251953125},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"bandage_assembly_graph_image"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"bandage_contigs_image"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/2022.09+galaxy4","tool_shed_repository":{"changeset_revision":"ddddce450736","name":"bandage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fontsize\\": null, \\"height\\": \\"1000\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"lengths\\": false, \\"names\\": false, \\"nodewidth\\": null, \\"output_format\\": \\"jpg\\", \\"width\\": null, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2022.09+galaxy4","type":"tool","uuid":"ab317904-7296-4549-966f-f2a27ad9990c","when":null,"workflow_outputs":[{"label":"bandage_assembly_graph_image","output_name":"outfile","uuid":"9612a851-1f94-4d54-b001-5d082bcc9055"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","errors":null,"id":8,"input_connections":{"components_0|param_type|component_value":{"id":5,"output_name":"text_param"}},"inputs":[],"label":null,"name":"Compose text parameter value","outputs":[{"name":"out1","type":"expression.json"}],"position":{"left":1147.2882164496511,"top":50.54090483396629},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1","tool_shed_repository":{"changeset_revision":"e188c9826e0f","name":"compose_text_param","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"components\\": [{\\"__index__\\": 0, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, {\\"__index__\\": 1, \\"param_type\\": {\\"select_param_type\\": \\"text\\", \\"__current_case__\\": 0, \\"component_value\\": \\"_$1\\"}}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.1.1","type":"tool","uuid":"5b617e69-8366-49cb-8d8b-393c8c2f78cb","when":null,"workflow_outputs":[]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","errors":null,"id":9,"input_connections":{"input":{"id":6,"output_name":"out_consensus"}},"inputs":[],"label":null,"name":"FASTA-to-Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1210.4666748046875,"top":757.7333374023438},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"sample_specific_contigs_tabular_file_preparation"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","tool_shed_repository":{"changeset_revision":"e7ed3c310b74","name":"fasta_to_tabular","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"descr_columns\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_first\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"c55f2b32-2214-4c07-90e7-ce425388dc01","when":null,"workflow_outputs":[{"label":"sample_specific_contigs_tabular_file_preparation","output_name":"output","uuid":"de06c362-5b0a-4c29-956e-fbf02539789a"}]},"10":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","errors":null,"id":10,"input_connections":{"file_input":{"id":6,"output_name":"out_consensus"}},"inputs":[],"label":null,"name":"ABRicate","outputs":[{"name":"report","type":"tabular"}],"position":{"left":1831.7666625976562,"top":31.583328247070312},"post_job_actions":{"RenameDatasetActionreport":{"action_arguments":{"newname":"vfs_of_genes_identified_by_vfdb"},"action_type":"RenameDatasetAction","output_name":"report"},"TagDatasetActionreport":{"action_arguments":{"tags":"vfs_of_genes_identified_by_vfdb"},"action_type":"TagDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","tool_shed_repository":{"changeset_revision":"3f3e247c053d","name":"abricate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"db\\": \\"vfdb\\", \\"no_header\\": false, \\"min_dna_id\\": \\"50.0\\", \\"min_cov\\": \\"50.0\\"}, \\"file_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"1de6612c-0191-4e11-8b12-e321e460c4bf","when":null,"workflow_outputs":[{"label":"abricate_with_vfdb_to_identify_genes_with_VFs","output_name":"report","uuid":"8f227fc7-2d92-4c6f-af64-e841c1315b4f"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","errors":null,"id":11,"input_connections":{"file_input":{"id":6,"output_name":"out_consensus"}},"inputs":[],"label":null,"name":"ABRicate","outputs":[{"name":"report","type":"tabular"}],"position":{"left":1847.25,"top":671.316650390625},"post_job_actions":{"RenameDatasetActionreport":{"action_arguments":{"newname":"amr_identified_by_ncbi"},"action_type":"RenameDatasetAction","output_name":"report"},"TagDatasetActionreport":{"action_arguments":{"tags":"amr_identified_by_ncbi"},"action_type":"TagDatasetAction","output_name":"report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1","tool_shed_repository":{"changeset_revision":"3f3e247c053d","name":"abricate","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"db\\": \\"ncbi\\", \\"no_header\\": false, \\"min_dna_id\\": \\"50.0\\", \\"min_cov\\": \\"50.0\\"}, \\"file_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.1","type":"tool","uuid":"92d285c6-2c41-49e2-9ddd-9ca07c317b7a","when":null,"workflow_outputs":[{"label":"abricate_report_using_ncbi_database_to_indentify_amr","output_name":"report","uuid":"fa021d98-d885-4834-ac62-f30d5792260e"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","errors":null,"id":12,"input_connections":{"find_and_replace_0|replace_pattern":{"id":8,"output_name":"out1"},"infile":{"id":9,"output_name":"output"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1528.3333740234375,"top":802.5333251953125},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"sample_specific_contigs_tabular_file"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"column\\", \\"__current_case__\\": 1, \\"column\\": \\"1\\"}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.4","type":"tool","uuid":"4de0cdf1-967a-4c47-9c95-f1f129ad5810","when":null,"workflow_outputs":[{"label":"sample_specific_contigs_tabular_file","output_name":"outfile","uuid":"174cb2c3-ed28-453c-8afb-85150e0b51ad"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","errors":null,"id":13,"input_connections":{"find_and_replace_1|replace_pattern":{"id":8,"output_name":"out1"},"infile":{"id":10,"output_name":"report"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2063.4155768063843,"top":43.25688097234852},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"vfs"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"vfs"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#FILE\\", \\"replace_pattern\\": \\"SampleID\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"column\\", \\"__current_case__\\": 1, \\"column\\": \\"1\\"}}, {\\"__index__\\": 1, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": true, \\"searchwhere\\": {\\"searchwhere_select\\": \\"column\\", \\"__current_case__\\": 1, \\"column\\": \\"2\\"}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.4","type":"tool","uuid":"4e5b87e2-4743-44d3-b57d-b5cab2887e46","when":null,"workflow_outputs":[{"label":"vfs","output_name":"outfile","uuid":"19e56e00-6eef-4a4e-9cc6-d93dbf9420ad"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","errors":null,"id":14,"input_connections":{"find_and_replace_1|replace_pattern":{"id":8,"output_name":"out1"},"infile":{"id":11,"output_name":"report"}},"inputs":[],"label":null,"name":"Replace","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2131.751242708332,"top":692.620361328125},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"amrs"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"amrs"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4","tool_shed_repository":{"changeset_revision":"d698c222f354","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"find_and_replace\\": [{\\"__index__\\": 0, \\"find_pattern\\": \\"#FILE\\", \\"replace_pattern\\": \\"SampleID\\", \\"is_regex\\": false, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": false, \\"searchwhere\\": {\\"searchwhere_select\\": \\"column\\", \\"__current_case__\\": 1, \\"column\\": \\"1\\"}}, {\\"__index__\\": 1, \\"find_pattern\\": \\"^(.+)$\\", \\"replace_pattern\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_regex\\": true, \\"global\\": true, \\"caseinsensitive\\": false, \\"wholewords\\": false, \\"skip_first_line\\": true, \\"searchwhere\\": {\\"searchwhere_select\\": \\"column\\", \\"__current_case__\\": 1, \\"column\\": \\"2\\"}}], \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.4","type":"tool","uuid":"9ebff4c8-3b3c-43a1-a008-490b4745f407","when":null,"workflow_outputs":[{"label":"amrs","output_name":"outfile","uuid":"a8eb74a5-362b-4f0f-a944-672e27981a41"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","errors":null,"id":15,"input_connections":{"input":{"id":12,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Tabular-to-FASTA","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1542.88330078125,"top":1009.6666870117188},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"contigs"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"contigs"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","tool_shed_repository":{"changeset_revision":"0a7799698fe5","name":"tabular_to_fasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seq_col\\": \\"2\\", \\"title_col\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"52af9a9c-0c9c-4605-a2a6-caefde8efa89","when":null,"workflow_outputs":[{"label":"contigs","output_name":"output","uuid":"a5bb1cbe-1bde-49a1-ad86-6d57b1319fbe"}]}},"tags":["name:Collection","name:PathoGFAIR","name:IWC","name:microGalaxy"],"uuid":"d6795753-1562-4ccc-8f9c-f598acd5404e","version":142},"readme":"# Gene-based Pathogen Identification\\n\\nIn this workflow, we determine whether the samples are pathogenic or not, by looking for genes known to be linked to pathogenicity or to the pathogenecity character of the organism.\\n\\n- Virulence Factor (VF): gene products, usually proteins, involved in pathogenicity. By identifying them, we can call a pathogen and its severity level\\n\\n- Antimicrobial Resistance genes (AMR).\\n\\n    These type of genes have three fundamental mechanisms of antimicrobial resistance that are enzymatic degradation of antibacterial drugs, alteration of bacterial proteins that are antimicrobial targets, and changes in membrane permeability to antibiotics, which will lead to not altering the target site and spread throughput the pathogenic bacteria decreasing the overall fitness of the host.\\n\\nIn this workflow we:\\n\\n1. Perform genome assembly to get contigs, i.e. longer sequences, using metaflye (Kolmogorov et al. 2020) then assembly polishing using medaka consensus pipeline and visualizing the assembly graph using Bandage Image (Wick et al. 2015)\\n2. Generate reports with AMR genes and VF using ABRicate\\n\\n## Input Datasets\\n- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format, the output of **Nanopore Preprocessing** workflow.\\n\\n## Output Datasets\\n- FASTA and Tabular files to track genes and visualise our pathogenic identification through out all samples.\\n\\nIf you\'re unsure how to use this workflows, or if you want to see it in action with test datasets, it is included in our detailed training material for foodborne pathogen detection and tracking. You can find step-by-step instructions and practical examples in the following [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-04-18\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Gene-based Pathogen Identification\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ collection_of_preprocessed_samples\\"]@{ shape: docs }\\n1[\\"Extract element identifiers\\"]@{ shape: process }\\n0 --> 1\\n2[\\"Build list\\"]@{ shape: process }\\n0 --> 2\\n3[\\"Split file\\"]@{ shape: process }\\n1 --> 3\\n4[\\"Flye\\"]@{ shape: process }\\n2 --> 4\\n5[\\"Parse parameter value\\"]@{ shape: process }\\n3 --> 5\\n6[\\"medaka consensus pipeline\\"]@{ shape: process }\\n4 --> 6\\n0 --> 6\\n7[\\"Bandage Image\\"]@{ shape: process }\\n4 --> 7\\n8[\\"Compose text parameter value\\"]@{ shape: process }\\n5 --> 8\\n9[\\"FASTA-to-Tabular\\"]@{ shape: process }\\n6 --> 9\\n10[\\"ABRicate\\"]@{ shape: process }\\n6 --> 10\\n11[\\"ABRicate\\"]@{ shape: process }\\n6 --> 11\\n12[\\"Replace\\"]@{ shape: process }\\n8 --> 12\\n9 --> 12\\n13[\\"Replace\\"]@{ shape: process }\\n8 --> 13\\n10 --> 13\\n14[\\"Replace\\"]@{ shape: process }\\n8 --> 14\\n11 --> 14\\n15[\\"Tabular-to-FASTA\\"]@{ shape: process }\\n12 --> 15\\n```\\n","updated":"2024-04-18T00:00:00","trsID":"#workflow/github.com/iwc-workflows/gene-based-pathogen-identification/main","dockstore_id":27328,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.12744314","tests":[{"doc":"Test outline for Gene-based-Pathogen-Identification","job":{"collection_of_preprocessed_samples":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10.fastq.gz","filetype":"fastqsanger.gz"},{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12.fastq.gz","filetype":"fastqsanger.gz"}]}},"outputs":{"extracted_samples_IDs":{"path":"test-data/extracted_samples_IDs.txt"}}}]}],"path":"./workflows/microbiome/pathogen-identification/gene-based-pathogen-identification"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Nanopore-Pre-Processing.ga","testParameterFiles":["/Nanopore-Pre-Processing-tests.yml"],"authors":[{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987","url":"https://orcid.org/0000-0001-9852-1987"},{"name":"Engy Nasr","orcid":"0000-0001-9047-4215","url":"https://orcid.org/0000-0001-9047-4215"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X","url":"https://orcid.org/0000-0003-2982-388X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Microbiome - QC and Contamination Filtering","comments":[{"color":"lime","data":{"text":"In this workflow, we recommend nanopore samples since we are using some tools that are specified for nanopore such as: Minimap2, Nanoplot porechp and fastp.\\n\\nIn order to switch to illumina please remove the nanoplot step, replace porechop and fastp with cutadapt or trimmomatic, and finally replace Minimap2 with  Bowtie2.\\n\\nThis workflow also take single-end reads to switch to paired-end, you have to check this option in most of the tools used. "},"id":0,"position":[295.3,262.9999938964844],"size":[460,228],"type":"markdown"},{"child_steps":[12,18,26,15,24,25],"color":"green","data":{"title":"Creating a table for samples reads count before and after hosts sequences removal and the percentage of found hosts sequences"},"id":7,"position":[2025.8999999999999,831.0999938964843],"size":[1403,223],"type":"frame"},{"child_steps":[14,17,20],"color":"green","data":{"title":"Hosts Sequences Identification and Removal (Kraken Technique)"},"id":6,"position":[1407.3,442.8999938964844],"size":[709.1,300],"type":"frame"},{"child_steps":[9,10,27],"color":"green","data":{"title":"Samples QC After Trimming"},"id":5,"position":[892.1,943.5999938964844],"size":[569.3,815],"type":"frame"},{"child_steps":[4,5,7],"color":"green","data":{"title":"Samples Reads QC"},"id":4,"position":[293,947.7999938964844],"size":[520,810],"type":"frame"},{"child_steps":[2],"color":"green","data":{"title":"Input Samples Collection"},"id":3,"position":[0,973.7999938964844],"size":[240,142.2],"type":"frame"},{"child_steps":[23,8,11,13,19,21,16,22],"color":"green","data":{"title":"Host Sequences Detection and Removal (Mapping Technique)"},"id":2,"position":[886,32.399993896484375],"size":[1772,293],"type":"frame"},{"child_steps":[3,6],"color":"green","data":{"title":"Trimming to retain samples reads quality"},"id":1,"position":[294,506.3999938964844],"size":[498,346.8],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut","url":"https://orcid.org/0000-0001-9852-1987"},{"class":"Person","identifier":"0000-0001-9047-4215","name":"Engy Nasr","url":"https://orcid.org/0000-0001-9047-4215"},{"class":"Person","identifier":"0000-0003-2982-388X","name":"Paul Zierep"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Nanopore Preprocessing","report":{"markdown":"# Nanopore - Preprocessing Workflow Report\\nBelow are the results of the Preprocessing  Workflow\\n\\nThis workflow was run on:\\n\\n```galaxy\\ngenerate_time()\\n```\\n\\nWith Galaxy version:\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Workflow Inputs\\nA collection of all sample reads (all in the same sequence file format, e.g.: Fastq, Fastq.gz, Fastqsanger, etc.)\\n\\n## Workflow Outputs\\n\\n### Multi QC Report of all samples before reads quality retaining\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"multiQC_html_report_before_preprocessing\\")\\n```\\n### Multi QC Report of all samples after reads quality retaining\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"multiQC_html_report_after_preprocessing\\")\\n```\\n\\n### Table of the total number of reads and the number of host reads\\n\\n```galaxy\\nhistory_dataset_display(output=\\"removed_hosts_percentage_tabular\\")\\n```\\n\\n"},"steps":{"0":{"annotation":"based on the lab preparation of the samples during sequencing, there should be a sample profile better than the other, to be chosen as an optional input to Minimap2. e.g. PacBio/Oxford Nanpore\\n\\nFor more details check: https://github.com/lh3/minimap2?tab=readme-ov-file#use-cases","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"based on the lab preparation of the samples during sequencing, there should be a sample profile better than the other, to be chosen as an optional input to Minimap2. e.g. PacBio/Oxford Nanpore\\n\\nFor more details check: https://github.com/lh3/minimap2?tab=readme-ov-file#use-cases","name":"samples_profile"}],"label":"samples_profile","name":"Input parameter","outputs":[],"position":{"left":593,"top":0},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"1435b9fc-84c0-4a1a-af49-007da98f4d5f","when":null,"workflow_outputs":[]},"1":{"annotation":"Where the samples are coming from, Human, Chicken, Cow, etc. ","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"Where the samples are coming from, Human, Chicken, Cow, etc. ","name":"host_reference_genome"}],"label":"host_reference_genome","name":"Input parameter","outputs":[],"position":{"left":594,"top":111.39999389648438},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"ec61e040-966c-4785-913c-1cd3b1866595","when":null,"workflow_outputs":[]},"2":{"annotation":"Nanopore reads of each sample are all in a single fastq or fastq.gz file","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Nanopore reads of each sample are all in a single fastq or fastq.gz file","name":"collection_of_all_samples"}],"label":"collection_of_all_samples","name":"Input dataset collection","outputs":[],"position":{"left":20,"top":1013.7999938964844},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"6a0dea4c-3d2c-4ab9-bb87-82cff75ac6d0","when":null,"workflow_outputs":[]},"3":{"annotation":"Preprocessing (Trimming)","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.4+galaxy0","errors":null,"id":3,"input_connections":{"input_file":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Porechop","outputs":[{"name":"outfile","type":"fasta"}],"position":{"left":314,"top":550.4441938961835},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"porechop_output_trimmed_reads"},"action_type":"RenameDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.4+galaxy0","tool_shed_repository":{"changeset_revision":"543cbeef3949","name":"porechop","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapter_search_settings\\": {\\"adapter_threshold\\": \\"90.0\\", \\"check_reads\\": \\"10000\\", \\"scoring_scheme\\": \\"3,-6,-5,-2\\"}, \\"barcode_binning_settings\\": {\\"barcode_threshold\\": \\"75.0\\", \\"barcode_diff\\": \\"5.0\\", \\"require_two_barcodes\\": false, \\"discard_unassigned\\": false}, \\"end_adapter_settings\\": {\\"end_size\\": \\"150\\", \\"min_trim_size\\": \\"4\\", \\"extra_end_trim\\": \\"2\\", \\"end_threshold\\": \\"75.0\\"}, \\"format\\": \\"fastq.gz\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"middle_adapter_settings\\": {\\"no_split\\": false, \\"discard_middle\\": false, \\"middle_threshold\\": \\"85.0\\", \\"extra_middle_trim_good_side\\": \\"10\\", \\"extra_middle_trim_bad_side\\": \\"100\\", \\"min_split_read_size\\": \\"1000\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.2.4+galaxy0","type":"tool","uuid":"c57309b3-3d32-42be-b1c1-509ed12d6f4a","when":null,"workflow_outputs":[{"label":"porechop_output_trimmed_reads","output_name":"outfile","uuid":"34ea26db-11cb-41ee-85c3-75af8a53a2c0"}]},"4":{"annotation":"Quality Check Before Preprocessing","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.42.0+galaxy1","errors":null,"id":4,"input_connections":{"mode|reads|files":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"NanoPlot","outputs":[{"name":"output_html","type":"html"},{"name":"nanostats","type":"tabular"},{"name":"nanostats_post_filtering","type":"tabular"}],"position":{"left":314,"top":987.8031200000532},"post_job_actions":{"RenameDatasetActionlog_read_length":{"action_arguments":{"newname":"nanoplot_on_reads_before_preprocessing_log_transformed_histogram_read_length"},"action_type":"RenameDatasetAction","output_name":"log_read_length"},"RenameDatasetActionnanostats":{"action_arguments":{"newname":"nanoplot_qc_on_reads_before_preprocessing_nanostats"},"action_type":"RenameDatasetAction","output_name":"nanostats"},"RenameDatasetActionnanostats_post_filtering":{"action_arguments":{"newname":"nanoplot_on_reads_before_preprocessing_nanostats_post_filtering"},"action_type":"RenameDatasetAction","output_name":"nanostats_post_filtering"},"RenameDatasetActionoutput_html":{"action_arguments":{"newname":"nanoplot_qc_on_reads_before_preprocessing_html_report"},"action_type":"RenameDatasetAction","output_name":"output_html"},"RenameDatasetActionread_length":{"action_arguments":{"newname":"nanoplot_on_reads_before_preprocessing_histogram_read_length"},"action_type":"RenameDatasetAction","output_name":"read_length"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.42.0+galaxy1","tool_shed_repository":{"changeset_revision":"a02b8b3f5a0c","name":"nanoplot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"customization\\": {\\"color\\": null, \\"plots\\": null, \\"N50\\": false}, \\"filter\\": {\\"maxlength\\": null, \\"minlength\\": null, \\"drop_outliers\\": false, \\"downsample\\": null, \\"loglength\\": false, \\"percentqual\\": false, \\"alength\\": false, \\"minqual\\": null, \\"runtime_until\\": null, \\"readtype\\": null, \\"barcoded\\": false, \\"no_supplementary\\": false}, \\"mode\\": {\\"choice\\": \\"batch\\", \\"__current_case__\\": 0, \\"reads\\": {\\"type\\": \\"fastq\\", \\"__current_case__\\": 0, \\"files\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.42.0+galaxy1","type":"tool","uuid":"60c022d4-2cc5-4caf-a9f0-333cf8a3edc7","when":null,"workflow_outputs":[{"label":"nanoplot_qc_on_reads_before_preprocessing_html_report","output_name":"output_html","uuid":"f2bd0a1f-cd60-4a36-a7d0-8025cc19ea2e"},{"label":"nanoplot_qc_on_reads_before_preprocessing_nanostats","output_name":"nanostats","uuid":"15ecf5b1-e0eb-405a-ac3a-359feb66d4cd"},{"label":"nanoplot_on_reads_before_preprocessing_nanostats_post_filtering","output_name":"nanostats_post_filtering","uuid":"304110f9-60d0-4ba2-8b3b-fae0e2a49554"}]},"5":{"annotation":"Quality Check Before Preprocessing","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","errors":null,"id":5,"input_connections":{"input_file":{"id":2,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":313,"top":1337.802741689313},"post_job_actions":{"RenameDatasetActionhtml_file":{"action_arguments":{"newname":"fastqc_quality_check_before_preprocessing_html_file"},"action_type":"RenameDatasetAction","output_name":"html_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"fastqc_quality_check_before_preprocessing_text_file"},"action_type":"RenameDatasetAction","output_name":"text_file"},"TagDatasetActionhtml_file":{"action_arguments":{"tags":"FastQC_Before_Preprocessing_HTM"},"action_type":"TagDatasetAction","output_name":"html_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","tool_shed_repository":{"changeset_revision":"5ec9f6bceaee","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy0","type":"tool","uuid":"b393d71b-0708-4825-97f2-d43a3f312f73","when":null,"workflow_outputs":[{"label":"fastqc_quality_check_before_preprocessing_html_file","output_name":"html_file","uuid":"d0a64624-05d0-4068-835b-a025fc011760"},{"label":"fastqc_quality_check_before_preprocessing_text_file","output_name":"text_file","uuid":"e61fef5d-1bc8-4c8e-be6a-f74e210e9920"}]},"6":{"annotation":"Preprocessing","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0","errors":null,"id":6,"input_connections":{"single_paired|in1":{"id":3,"output_name":"outfile"}},"inputs":[{"description":"runtime parameter for tool fastp","name":"single_paired"}],"label":null,"name":"fastp","outputs":[{"name":"out1","type":"input"},{"name":"report_html","type":"html"}],"position":{"left":575,"top":538.3698009082564},"post_job_actions":{"RenameDatasetActionout1":{"action_arguments":{"newname":"nanopore_sequenced_reads_processed_with_fastp_after_host_removal"},"action_type":"RenameDatasetAction","output_name":"out1"},"RenameDatasetActionreport_html":{"action_arguments":{"newname":"nanopore_sequenced_reads_processed_with_fastp_after_host_removal_html_report"},"action_type":"RenameDatasetAction","output_name":"report_html"},"TagDatasetActionreport_html":{"action_arguments":{"tags":"fastp_HTML"},"action_type":"TagDatasetAction","output_name":"report_html"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.4+galaxy0","tool_shed_repository":{"changeset_revision":"c59d48774d03","name":"fastp","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_options\\": {\\"quality_filtering_options\\": {\\"disable_quality_filtering\\": false, \\"qualified_quality_phred\\": null, \\"unqualified_percent_limit\\": null, \\"n_base_limit\\": null}, \\"length_filtering_options\\": {\\"disable_length_filtering\\": false, \\"length_required\\": null, \\"length_limit\\": null}, \\"low_complexity_filter\\": {\\"enable_low_complexity_filter\\": false, \\"complexity_threshold\\": null}}, \\"output_options\\": {\\"report_html\\": true, \\"report_json\\": false}, \\"overrepresented_sequence_analysis\\": {\\"overrepresentation_analysis\\": false, \\"overrepresentation_sampling\\": null}, \\"read_mod_options\\": {\\"polyg_tail_trimming\\": {\\"trimming_select\\": \\"\\", \\"__current_case__\\": 1, \\"poly_g_min_len\\": null}, \\"polyx_tail_trimming\\": {\\"polyx_trimming_select\\": \\"\\", \\"__current_case__\\": 1}, \\"umi_processing\\": {\\"umi\\": false, \\"umi_loc\\": \\"\\", \\"umi_len\\": null, \\"umi_prefix\\": \\"\\"}, \\"cutting_by_quality_options\\": {\\"cut_by_quality5\\": false, \\"cut_by_quality3\\": false, \\"cut_window_size\\": null, \\"cut_mean_quality\\": null}, \\"base_correction_options\\": {\\"correction\\": false}}, \\"single_paired\\": {\\"single_paired_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"in1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"adapter_trimming_options\\": {\\"disable_adapter_trimming\\": false, \\"adapter_sequence1\\": \\"\\"}, \\"global_trimming_options\\": {\\"trim_front1\\": null, \\"trim_tail1\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.23.4+galaxy0","type":"tool","uuid":"ca35835e-9381-49ee-9248-cbb8809113bf","when":null,"workflow_outputs":[{"label":"nanopore_sequenced_reads_processed_with_fastp_after_host_removal_html_report","output_name":"report_html","uuid":"2a9a8b4d-458b-40e7-9a21-fb7108d5bbe4"},{"label":"nanopore_sequenced_reads_processed_with_fastp_after_host_removal","output_name":"out1","uuid":"a2219483-50b7-4aed-98dd-333ad2e12eb8"}]},"7":{"annotation":"Quality Control Before and After Preprocessing\\n\\n","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","errors":null,"id":7,"input_connections":{"results_0|software_cond|output_0|input":{"id":5,"output_name":"text_file"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"left":593,"top":1475.802741689313},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"multiQC_html_report_before_preprocessing"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"multiQC_stats_before_preprocessing"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"MultiQC_Before_Preprocessing_HTML"},"action_type":"TagDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","tool_shed_repository":{"changeset_revision":"abfd8a6544d7","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}], \\"saveLog\\": false, \\"title\\": \\"MultiQC Report Before PreProcessing\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy1","type":"tool","uuid":"cf00d759-c11e-4e7c-bc39-2735ada49668","when":null,"workflow_outputs":[{"label":"multiQC_html_report_before_preprocessing","output_name":"html_report","uuid":"0f92196d-047d-4918-819d-c0ff7cd3ae85"},{"label":"multiQC_stats_before_preprocessing","output_name":"stats","uuid":"ebffe782-a56c-431f-8af4-c0cb8d7a02fc"}]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0","errors":null,"id":8,"input_connections":{"fastq_input|analysis_type_selector":{"id":0,"output_name":"output"},"fastq_input|fastq_input1":{"id":6,"output_name":"out1"},"reference_source|ref_file":{"id":1,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":895.0333251953125,"top":88.26976428716262},"post_job_actions":{"RenameDatasetActionalignment_output":{"action_arguments":{"newname":"bam_map_to_host"},"action_type":"RenameDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0","tool_shed_repository":{"changeset_revision":"5cc34c3f440d","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"BAM\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"cached\\", \\"__current_case__\\": 0, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy0","type":"tool","uuid":"d3ae590a-f40f-40f8-a6c4-d9d07be650ca","when":null,"workflow_outputs":[{"label":"bam_map_to_host","output_name":"alignment_output","uuid":"9d7bb3b7-09a1-401f-a132-bb35a53375ea"}]},"9":{"annotation":"Nanoplot Quality Check After Preprocessing","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.42.0+galaxy1","errors":null,"id":9,"input_connections":{"mode|reads|files":{"id":6,"output_name":"out1"}},"inputs":[],"label":null,"name":"NanoPlot","outputs":[{"name":"output_html","type":"html"},{"name":"nanostats","type":"tabular"},{"name":"nanostats_post_filtering","type":"tabular"}],"position":{"left":914.2000122070312,"top":983.603107793022},"post_job_actions":{"RenameDatasetActionlog_read_length":{"action_arguments":{"newname":"nanoplot_on_reads_after_preprocessing_log_transformed_histogram_read_length"},"action_type":"RenameDatasetAction","output_name":"log_read_length"},"RenameDatasetActionnanostats":{"action_arguments":{"newname":"nanoplot_qc_on_reads_after_preprocessing_nanostats"},"action_type":"RenameDatasetAction","output_name":"nanostats"},"RenameDatasetActionnanostats_post_filtering":{"action_arguments":{"newname":"nanoplot_on_reads_after_preprocessing_nanostats_post_filtering"},"action_type":"RenameDatasetAction","output_name":"nanostats_post_filtering"},"RenameDatasetActionoutput_html":{"action_arguments":{"newname":"nanoplot_qc_on_reads_after_preprocessing_html_report"},"action_type":"RenameDatasetAction","output_name":"output_html"},"RenameDatasetActionread_length":{"action_arguments":{"newname":"nanoplot_on_reads_after_preprocessing_histogram_read_length"},"action_type":"RenameDatasetAction","output_name":"read_length"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.42.0+galaxy1","tool_shed_repository":{"changeset_revision":"a02b8b3f5a0c","name":"nanoplot","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"customization\\": {\\"color\\": null, \\"plots\\": null, \\"N50\\": false}, \\"filter\\": {\\"maxlength\\": null, \\"minlength\\": null, \\"drop_outliers\\": false, \\"downsample\\": null, \\"loglength\\": false, \\"percentqual\\": false, \\"alength\\": false, \\"minqual\\": null, \\"runtime_until\\": null, \\"readtype\\": null, \\"barcoded\\": false, \\"no_supplementary\\": false}, \\"mode\\": {\\"choice\\": \\"batch\\", \\"__current_case__\\": 0, \\"reads\\": {\\"type\\": \\"fastq\\", \\"__current_case__\\": 0, \\"files\\": {\\"__class__\\": \\"ConnectedValue\\"}}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.42.0+galaxy1","type":"tool","uuid":"33511a93-8e27-4936-91e7-451ba3c1bb54","when":null,"workflow_outputs":[{"label":"nanoplot_qc_on_reads_after_preprocessing_html_report","output_name":"output_html","uuid":"b5899290-4c57-4662-ad22-860654652ade"},{"label":"nanoplot_qc_on_reads_after_preprocessing_nanostats","output_name":"nanostats","uuid":"42db7f93-919e-4bbb-81a1-06411a9da410"},{"label":"nanoplot_on_reads_after_preprocessing_nanostats_post_filtering","output_name":"nanostats_post_filtering","uuid":"949bfdf5-3d79-4dad-bdd8-c3a25e6af4cf"}]},"10":{"annotation":"Quality Check After Preprocessing","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","errors":null,"id":10,"input_connections":{"input_file":{"id":6,"output_name":"out1"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":912.1138463352464,"top":1338.5819672179302},"post_job_actions":{"RenameDatasetActionhtml_file":{"action_arguments":{"newname":"fastqc_quality_check_after_preprocessing_html_file"},"action_type":"RenameDatasetAction","output_name":"html_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"fastqc_quality_check_after_preprocessing_text_file"},"action_type":"RenameDatasetAction","output_name":"text_file"},"TagDatasetActionhtml_file":{"action_arguments":{"tags":"FastQC_After_Preprocessing_HTML"},"action_type":"TagDatasetAction","output_name":"html_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","tool_shed_repository":{"changeset_revision":"5ec9f6bceaee","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy0","type":"tool","uuid":"2862f65a-7df0-4d6a-8aaf-caa677dd8bcf","when":null,"workflow_outputs":[{"label":"fastqc_quality_check_after_preprocessing_text_file","output_name":"text_file","uuid":"09306471-afa0-4106-9cc7-259b93dfc862"},{"label":"fastqc_quality_check_after_preprocessing_html_file","output_name":"html_file","uuid":"084f982f-20f1-457e-8012-91ebbb85633d"}]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_mapped/bamtools_split_mapped/2.5.2+galaxy2","errors":null,"id":11,"input_connections":{"input_bam":{"id":8,"output_name":"alignment_output"}},"inputs":[],"label":null,"name":"Split BAM by reads mapping status","outputs":[{"name":"mapped","type":"bam"},{"name":"unmapped","type":"bam"}],"position":{"left":1109.933349609375,"top":92.33644519536574},"post_job_actions":{"RenameDatasetActionmapped":{"action_arguments":{"newname":"host_sequences_bam"},"action_type":"RenameDatasetAction","output_name":"mapped"},"RenameDatasetActionunmapped":{"action_arguments":{"newname":"non_host_sequences_bam"},"action_type":"RenameDatasetAction","output_name":"unmapped"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_mapped/bamtools_split_mapped/2.5.2+galaxy2","tool_shed_repository":{"changeset_revision":"fa7b5520ae53","name":"bamtools_split_mapped","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input_bam\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.5.2+galaxy2","type":"tool","uuid":"f4882cef-bfad-45ca-8274-d5b6b18fa314","when":null,"workflow_outputs":[{"label":"host_sequences_bam","output_name":"mapped","uuid":"3b1e626f-6bc1-484c-be01-366534361b73"},{"label":"non_host_sequences_bam","output_name":"unmapped","uuid":"14a53fe2-f296-43aa-86b7-243278c1050c"}]},"12":{"annotation":"","content_id":"Grep1","errors":null,"id":12,"input_connections":{"input":{"id":10,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2045.9000244140625,"top":877.6833190917969},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"total_sequences_before_hosts_sequences_removal"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"Total Sequences\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"fce6adca-1caa-4032-9247-e884fa992348","when":null,"workflow_outputs":[{"label":"total_sequences_before_hosts_sequences_removal","output_name":"out_file1","uuid":"a809853b-119f-44d2-986b-8d2006439fbe"}]},"13":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_fastx/samtools_fastx/1.15.1+galaxy2","errors":null,"id":13,"input_connections":{"input":{"id":11,"output_name":"mapped"}},"inputs":[],"label":null,"name":"Samtools fastx","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1331.5833129882812,"top":76.78642688481887},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"fastq.gz"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"host_sequences_fastq"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_fastx/samtools_fastx/1.15.1+galaxy2","tool_shed_repository":{"changeset_revision":"66e23b11feae","name":"samtools_fastx","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"copy_arb_tags\\": null, \\"copy_tags\\": false, \\"exclusive_filter\\": [\\"256\\", \\"2048\\"], \\"exclusive_filter_all\\": null, \\"idxout_cond\\": {\\"idxout_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"inclusive_filter\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_fmt_cond\\": {\\"output_fmt_select\\": \\"fastqsanger.gz\\", \\"__current_case__\\": 2, \\"default_quality\\": null, \\"output_quality\\": false, \\"ilumina_casava\\": false}, \\"outputs\\": [\\"other\\"], \\"read_numbering\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy2","type":"tool","uuid":"f79f5480-af0e-42e8-8a66-37dc436fedd9","when":null,"workflow_outputs":[{"label":"host_sequences_fastq","output_name":"output","uuid":"10d4eaec-81d8-444e-8075-7b77a1fb6870"}]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_fastx/samtools_fastx/1.15.1+galaxy2","errors":null,"id":14,"input_connections":{"input":{"id":11,"output_name":"unmapped"}},"inputs":[],"label":null,"name":"Samtools fastx","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1427.3131395036098,"top":489.4217518640123},"post_job_actions":{"ChangeDatatypeActionoutput":{"action_arguments":{"newtype":"fastq.gz"},"action_type":"ChangeDatatypeAction","output_name":"output"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"non_host_sequences_fastq"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_fastx/samtools_fastx/1.15.1+galaxy2","tool_shed_repository":{"changeset_revision":"66e23b11feae","name":"samtools_fastx","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"copy_arb_tags\\": null, \\"copy_tags\\": false, \\"exclusive_filter\\": [\\"256\\", \\"2048\\"], \\"exclusive_filter_all\\": null, \\"idxout_cond\\": {\\"idxout_select\\": \\"no\\", \\"__current_case__\\": 0}, \\"inclusive_filter\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"output_fmt_cond\\": {\\"output_fmt_select\\": \\"fastqsanger.gz\\", \\"__current_case__\\": 2, \\"default_quality\\": null, \\"output_quality\\": false, \\"ilumina_casava\\": false}, \\"outputs\\": [\\"other\\"], \\"read_numbering\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy2","type":"tool","uuid":"520e829b-f118-4655-8c0b-f996d01d67c3","when":null,"workflow_outputs":[{"label":"non_host_sequences_fastq","output_name":"output","uuid":"0c2dd74d-ac4f-45cf-839c-50386a7ece28"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":15,"input_connections":{"input_list":{"id":12,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2280.3802527630014,"top":879.056357216495},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"bf581820-873a-4e1a-84bc-a622dd7ece50","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":16,"input_connections":{"input":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1516,"top":208.40312610356887},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"18f79c89-cdcf-4ee7-a85f-b06e862f2e07","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1","errors":null,"id":17,"input_connections":{"single_paired|input_sequences":{"id":14,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Kraken2","name":"single_paired"}],"label":null,"name":"Kraken2","outputs":[{"name":"report_output","type":"tabular"},{"name":"output","type":"tabular"}],"position":{"left":1655.9777899859946,"top":482.9011443435088},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"kraken2_with_kalamri_database_output"},"action_type":"RenameDatasetAction","output_name":"output"},"RenameDatasetActionreport_output":{"action_arguments":{"newname":"kraken2_with_kalamri_database_report"},"action_type":"RenameDatasetAction","output_name":"report_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1","tool_shed_repository":{"changeset_revision":"20e2f64aa1fe","name":"kraken2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"confidence\\": \\"0.0\\", \\"kraken2_database\\": \\"kalamari\\", \\"min_base_quality\\": \\"0\\", \\"minimum_hit_groups\\": \\"2\\", \\"quick\\": false, \\"report\\": {\\"create_report\\": true, \\"use_mpa_style\\": false, \\"report_zero_counts\\": true, \\"report_minimizer_data\\": true}, \\"single_paired\\": {\\"single_paired_selector\\": \\"no\\", \\"__current_case__\\": 2, \\"input_sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split_reads\\": false, \\"use_names\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy1","type":"tool","uuid":"2c6bbcf9-178e-4cc2-8340-385961c4bdb1","when":null,"workflow_outputs":[{"label":"kraken2_with_kalamri_database_output","output_name":"output","uuid":"203d303e-8f3a-4242-971f-b345842ebdb8"},{"label":"kraken2_with_kalamri_database_report","output_name":"report_output","uuid":"843afd4d-23a8-46e7-b945-8b67dd7ae341"}]},"18":{"annotation":"","content_id":"Cut1","errors":null,"id":18,"input_connections":{"input":{"id":15,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2500.9833374023438,"top":883.3333435058594},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"quality_retained_all_reads"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1,c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"be1b0c09-4ce8-480d-9116-3077c82ea684","when":null,"workflow_outputs":[{"label":"quality_retained_all_reads","output_name":"out_file1","uuid":"d07be9f1-d250-4008-91ee-59a68521eb56"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","errors":null,"id":19,"input_connections":{"input_file":{"id":16,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FastQC","name":"adapters"},{"description":"runtime parameter for tool FastQC","name":"contaminants"},{"description":"runtime parameter for tool FastQC","name":"limits"}],"label":null,"name":"FastQC","outputs":[{"name":"html_file","type":"html"},{"name":"text_file","type":"txt"}],"position":{"left":1766.433349609375,"top":65.81978259770949},"post_job_actions":{"RenameDatasetActionhtml_file":{"action_arguments":{"newname":"hosts_qc_html"},"action_type":"RenameDatasetAction","output_name":"html_file"},"RenameDatasetActiontext_file":{"action_arguments":{"newname":"hosts_qc_text_file"},"action_type":"RenameDatasetAction","output_name":"text_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0","tool_shed_repository":{"changeset_revision":"5ec9f6bceaee","name":"fastqc","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adapters\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"contaminants\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"kmers\\": \\"7\\", \\"limits\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"min_length\\": null, \\"nogroup\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.74+galaxy0","type":"tool","uuid":"7140e440-b547-4cae-baa0-89e29a3f2bda","when":null,"workflow_outputs":[{"label":"hosts_qc_html","output_name":"html_file","uuid":"b72ff57b-0921-43bf-a817-6cd444c8f3cb"},{"label":"hosts_qc_text_file","output_name":"text_file","uuid":"b0ee6e31-0eb1-437d-8c04-fc3640b9a0b7"}]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads/1.2+galaxy1","errors":null,"id":20,"input_connections":{"library|input_1":{"id":6,"output_name":"out1"},"report":{"id":17,"output_name":"report_output"},"results":{"id":17,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Krakentools: Extract Kraken Reads By ID","name":"library"}],"label":null,"name":"Krakentools: Extract Kraken Reads By ID","outputs":[{"name":"output_1","type":"fasta.gz"}],"position":{"left":1897.4166870117188,"top":475.30312000005324},"post_job_actions":{"RenameDatasetActionoutput_1":{"action_arguments":{"newname":"collection_of_preprocessed_samples"},"action_type":"RenameDatasetAction","output_name":"output_1"},"TagDatasetActionoutput_1":{"action_arguments":{"tags":"collection_of_preprocessed_samples"},"action_type":"TagDatasetAction","output_name":"output_1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads/1.2+galaxy1","tool_shed_repository":{"changeset_revision":"f329328da134","name":"krakentools_extract_kraken_reads","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"exclude\\": true, \\"fastq_output\\": true, \\"include_children\\": true, \\"include_parents\\": true, \\"library\\": {\\"type\\": \\"single\\", \\"__current_case__\\": 0, \\"input_1\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"max\\": \\"100000000\\", \\"report\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"results\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"taxid\\": \\"9031 9606 9913\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2+galaxy1","type":"tool","uuid":"9d22fc4d-9b85-4f6b-b9df-ac9486de7acd","when":null,"workflow_outputs":[{"label":"collection_of_preprocessed_samples","output_name":"output_1","uuid":"57e3b725-8e13-40b2-9acc-31fd56ebc80a"}]},"21":{"annotation":"","content_id":"Grep1","errors":null,"id":21,"input_connections":{"input":{"id":19,"output_name":"text_file"}},"inputs":[],"label":null,"name":"Select","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1985.4666748046875,"top":65.26976428716262},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"total_sequences_after_hosts_sequences_removal"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Grep1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"invert\\": \\"\\", \\"keep_header\\": false, \\"pattern\\": \\"Total Sequences\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.4","type":"tool","uuid":"a2a29388-186c-41f3-b9a9-c4a6a9d3ffc3","when":null,"workflow_outputs":[{"label":"total_sequences_after_hosts_sequences_removal","output_name":"out_file1","uuid":"3ba35c71-32f0-4741-98d4-ea8522e27500"}]},"22":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":22,"input_connections":{"input_list":{"id":21,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2199.9000244140625,"top":70.41978870122512},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"877f3aa0-e22e-496e-8fff-742d035f1ed6","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"Cut1","errors":null,"id":23,"input_connections":{"input":{"id":22,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2434.11669921875,"top":72.58644519536574},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"quality_retained_hosts_reads"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1,c3\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"10bee624-ebbf-4173-9444-ae45062bc401","when":null,"workflow_outputs":[{"label":"quality_retained_hosts_reads","output_name":"out_file1","uuid":"cef36c68-4549-4fd6-b7c8-71fb21df012f"}]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":24,"input_connections":{"input_tabular":[{"id":18,"output_name":"out_file1"},{"id":23,"output_name":"out_file1"}]},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":2746.9152825405945,"top":883.0100512350168},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"0\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"a5159607-6ab6-40b3-8c7f-e0d3aee7a304","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0","errors":null,"id":25,"input_connections":{"input":{"id":24,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Compute","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2972.0099280935715,"top":881.2397437058577},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0","tool_shed_repository":{"changeset_revision":"6595517c2dd8","name":"column_maker","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"avoid_scientific_notation\\": false, \\"error_handling\\": {\\"auto_col_types\\": true, \\"fail_on_non_existent_columns\\": true, \\"non_computable\\": {\\"action\\": \\"--fail-on-non-computable\\", \\"__current_case__\\": 0}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ops\\": {\\"header_lines_select\\": \\"yes\\", \\"__current_case__\\": 1, \\"expressions\\": [{\\"__index__\\": 0, \\"cond\\": \\"(c3/c2)*100\\", \\"add_column\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0, \\"pos\\": \\"\\"}, \\"new_column_name\\": \\"removed_hosts_percentage\\"}]}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.0","type":"tool","uuid":"1456a078-b970-493d-9fdd-6a7590aa2972","when":null,"workflow_outputs":[]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","errors":null,"id":26,"input_connections":{"input":{"id":25,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Column Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":3208.8475559627727,"top":871.060475413193},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"removed_hosts_percentage_tabular"},"action_type":"RenameDatasetAction","output_name":"out_file1"},"TagDatasetActionout_file1":{"action_arguments":{"tags":"removed_hosts_percentage_tabular"},"action_type":"TagDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"#KEY\\", \\"replacement\\": \\"key\\"}], \\"field\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"5f41f84c-e27e-4fb7-8128-42ae7f143965","when":null,"workflow_outputs":[{"label":"removed_hosts_percentage_tabular","output_name":"out_file1","uuid":"470892ee-dab9-48d7-ad97-45dbd52afaa7"}]},"27":{"annotation":"Quality Control Before and After Preprocessing\\n\\n","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","errors":null,"id":27,"input_connections":{"results_0|software_cond|output_0|input":{"id":10,"output_name":"text_file"},"results_1|software_cond|input":{"id":26,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"MultiQC","outputs":[{"name":"stats","type":"input"},{"name":"html_report","type":"html"}],"position":{"left":1241.3867206791317,"top":1007.7576821710621},"post_job_actions":{"RenameDatasetActionhtml_report":{"action_arguments":{"newname":"multiQC_html_report_after_preprocessing"},"action_type":"RenameDatasetAction","output_name":"html_report"},"RenameDatasetActionstats":{"action_arguments":{"newname":"multiQC_stats_after_preprocessing"},"action_type":"RenameDatasetAction","output_name":"stats"},"TagDatasetActionhtml_report":{"action_arguments":{"tags":"MultiQC_After_Preprocessing_HTML"},"action_type":"TagDatasetAction","output_name":"html_report"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1","tool_shed_repository":{"changeset_revision":"abfd8a6544d7","name":"multiqc","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"comment\\": \\"\\", \\"export\\": false, \\"flat\\": false, \\"results\\": [{\\"__index__\\": 0, \\"software_cond\\": {\\"software\\": \\"fastqc\\", \\"__current_case__\\": 8, \\"output\\": [{\\"__index__\\": 0, \\"type\\": \\"data\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}]}}, {\\"__index__\\": 1, \\"software_cond\\": {\\"software\\": \\"custom_content\\", \\"__current_case__\\": 32, \\"plot_type\\": \\"table\\", \\"section_name\\": \\"Host Sequences Removal\\", \\"title\\": \\"Host Sequences\\", \\"description\\": null, \\"xlab\\": null, \\"ylab\\": null, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}}], \\"saveLog\\": false, \\"title\\": \\"MultiQC Report After PreProcessing\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.11+galaxy1","type":"tool","uuid":"df53a320-802c-459a-a4cb-fd41aac4af92","when":null,"workflow_outputs":[{"label":"multiQC_stats_after_preprocessing","output_name":"stats","uuid":"13cbf6c7-6954-4458-aa66-a5b020c63822"},{"label":"multiQC_html_report_after_preprocessing","output_name":"html_report","uuid":"0b1b5a73-36ee-42a2-a220-1ced6ec7378b"}]}},"tags":["name:Collection","name:microGalaxy","name:PathoGFAIR","name:Nanopore","name:IWC"],"uuid":"0b976704-e1e0-4cc3-9e75-38bda67c01ff","version":190},"readme":"# Nanopore Preprocessing\\n\\nBefore starting any analysis, it is always a good idea to assess the quality of your input data and to discard poor-quality base content by trimming and filtering reads.\\n\\nGenerally, we are not interested in the host sequences, but rather only those originating from the pathogen itself. It is important to get rid of all host sequences and to only retain sequences that might include a pathogen, both in order to speed up further steps and to avoid host sequences compromising the analysis.\\n\\n## Input Datasets\\n\\n- Collection of sequenced Nanopore reads of all samples to be analysed in a `fastqsanger` or `fastqsanger.gz` format.\\n\\n## Output Datasets\\n\\n- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format.\\n\\n- Tables indicating total number of reads before and after host sequences trimming, and the host sequences percentages found in each sample.\\n\\nIf you\'re unsure how to use this workflows, or if you want to see it in action with test datasets, it is included in our detailed training material for foodborne pathogen detection and tracking. You can find step-by-step instructions and practical examples in the following [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-04-25\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Nanopore Preprocessing\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ samples_profile\\"]@{ shape: lean-l }\\n1[\\"ℹ️ host_reference_genome\\"]@{ shape: lean-l }\\n2[\\"ℹ️ collection_of_all_samples\\"]@{ shape: docs }\\n3[\\"Porechop\\"]@{ shape: process }\\n2 --> 3\\n4[\\"NanoPlot\\"]@{ shape: process }\\n2 --> 4\\n5[\\"FastQC\\"]@{ shape: process }\\n2 --> 5\\n6[\\"fastp\\"]@{ shape: process }\\n3 --> 6\\n7[\\"MultiQC\\"]@{ shape: process }\\n5 --> 7\\n8[\\"Map with minimap2\\"]@{ shape: process }\\n0 --> 8\\n6 --> 8\\n1 --> 8\\n9[\\"NanoPlot\\"]@{ shape: process }\\n6 --> 9\\n10[\\"FastQC\\"]@{ shape: process }\\n6 --> 10\\n11[\\"Split BAM by reads mapping status\\"]@{ shape: process }\\n8 --> 11\\n12[\\"Select\\"]@{ shape: process }\\n10 --> 12\\n13[\\"Samtools fastx\\"]@{ shape: process }\\n11 --> 13\\n14[\\"Samtools fastx\\"]@{ shape: process }\\n11 --> 14\\n15[\\"Collapse Collection\\"]@{ shape: process }\\n12 --> 15\\n16[\\"Filter failed datasets\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Kraken2\\"]@{ shape: process }\\n14 --> 17\\n18[\\"Cut\\"]@{ shape: process }\\n15 --> 18\\n19[\\"FastQC\\"]@{ shape: process }\\n16 --> 19\\n20[\\"Krakentools: Extract Kraken Reads By ID\\"]@{ shape: process }\\n6 --> 20\\n17 --> 20\\n17 --> 20\\n21[\\"Select\\"]@{ shape: process }\\n19 --> 21\\n22[\\"Collapse Collection\\"]@{ shape: process }\\n21 --> 22\\n23[\\"Cut\\"]@{ shape: process }\\n22 --> 23\\n24[\\"Column join\\"]@{ shape: process }\\n18 --> 24\\n23 --> 24\\n25[\\"Compute\\"]@{ shape: process }\\n24 --> 25\\n26[\\"Column Regex Find And Replace\\"]@{ shape: process }\\n25 --> 26\\n27[\\"MultiQC\\"]@{ shape: process }\\n10 --> 27\\n26 --> 27\\n```\\n","updated":"2024-04-25T00:00:00","trsID":"#workflow/github.com/iwc-workflows/nanopore-pre-processing/main","dockstore_id":27330,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.12744321","tests":[{"doc":"Test outline for Nanopore-Pre-Processing","job":{"collection_of_all_samples":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode10","location":"https://zenodo.org/record/12190648/files/collection_of_all_samples_Spike3bBarcode10.fastq.gz","filetype":"fastqsanger.gz"},{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode12","location":"https://zenodo.org/record/12190648/files/collection_of_all_samples_Spike3bBarcode12.fastq.gz","filetype":"fastqsanger.gz"}]},"samples_profile":"map-pb","host_reference_genome":"apiMel3"},"outputs":{"nanoplot_qc_on_reads_after_preprocessing_nanostats":{"attributes":{},"element_tests":{"collection_of_all_samples_Spike3bBarcode12":{"asserts":{"has_n_lines":{"n":24}},"path":"test-data/nanoplot_qc_on_reads_after_preprocessing_nanostats_collection_of_all_samples_Spike3bBarcode12.tabular"}}}}}]}],"path":"./workflows/microbiome/pathogen-identification/nanopore-pre-processing"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Allele-based-Pathogen-Identification.ga","testParameterFiles":["/Allele-based-Pathogen-Identification-tests.yml"],"authors":[{"name":"Engy Nasr","orcid":"0000-0001-9047-4215","url":"https://orcid.org/0000-0001-9047-4215"},{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987","url":"https://orcid.org/0000-0001-9852-1987"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X","url":"https://orcid.org/0000-0003-2982-388X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Microbiome - Variant calling and Consensus Building","comments":[{"child_steps":[18,7,10,13,16,20],"color":"pink","data":{"title":"Mapping coverage"},"id":2,"position":[910,909.9],"size":[1400,237.8],"type":"frame"},{"child_steps":[4,5,8,11,14,15,25,17,19,21,23,24],"color":"pink","data":{"title":"Mapping, Variant and SNP calling"},"id":0,"position":[553,0],"size":[2628.6,462.8],"type":"frame"},{"child_steps":[22,6,9,12],"color":"pink","data":{"title":"Mapping mean depth"},"id":1,"position":[888,577.9],"size":[1586,239.4],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0001-9047-4215","name":"Engy Nasr","url":"https://orcid.org/0000-0001-9047-4215"},{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut","url":"https://orcid.org/0000-0001-9852-1987"},{"class":"Person","identifier":"0000-0003-2982-388X","name":"Paul Zierep"}],"format-version":"0.1","license":"MIT","release":"0.1.4","name":"Allele-based Pathogen Identification","report":{"markdown":"# Allele-based Pathogen Identification Workflow Report\\nBelow are the results for the Allele based Pathogenic Identification Workflow\\n\\nThis workflow was run on:\\n\\n```galaxy\\ngenerate_time()\\n```\\n\\nWith Galaxy version:\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Workflow Inputs\\nThe Perprocessing workflow main output (Collection of all samples reads after quality retaining and hosts filtering), and a FASTA file of the reference genome of the main Pathogen identified in the Gene based Pathogen Identification workflow, or per-known to the user.\\n\\n## Workflow Output: \\n\\n### All variants found per sample against the reference genome\\n\\n```galaxy\\nhistory_dataset_display(output=\\"extracted_fields_from_the_vcf_output\\")\\n```\\n\\n### Number of variants per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"number_of_variants_per_sample\\")\\n```\\n\\n### Mapping mean depth of coverage per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"mapping_mean_depth_per_sample\\")\\n```\\n\\n### Mapping breadth of coverage percentage per sample\\n\\n```galaxy\\nhistory_dataset_display(output=\\"mapping_coverage_percentage_per_sample\\")\\n```\\n\\n"},"steps":{"0":{"annotation":"Output collection from the Nanopore Preprocessing workflow","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Output collection from the Nanopore Preprocessing workflow","name":"collection_of_preprocessed_samples"}],"label":"collection_of_preprocessed_samples","name":"Input dataset collection","outputs":[],"position":{"left":6,"top":205},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"7aaa03a4-2647-411b-863d-90f28ed01f32","when":null,"workflow_outputs":[]},"1":{"annotation":"based on the lab preparation of the samples during sequencing, there should be a sample profile better than the other, to be chosen as an optional input to Minimap2. e.g. PacBio/Oxford Nanpore\\n\\nFor more details check: https://github.com/lh3/minimap2?tab=readme-ov-file#use-cases","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"based on the lab preparation of the samples during sequencing, there should be a sample profile better than the other, to be chosen as an optional input to Minimap2. e.g. PacBio/Oxford Nanpore\\n\\nFor more details check: https://github.com/lh3/minimap2?tab=readme-ov-file#use-cases","name":"samples_profile"}],"label":"samples_profile","name":"Input parameter","outputs":[],"position":{"left":260,"top":35},"tool_id":null,"tool_state":"{\\"validators\\": [], \\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": true}","tool_version":null,"type":"parameter_input","uuid":"d28b9bed-c674-406c-b58b-165bcc8eec17","when":null,"workflow_outputs":[]},"2":{"annotation":"Can be built in the tool later","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"Can be built in the tool later","name":"reference_genome_of_tested_strain"}],"label":"reference_genome_of_tested_strain","name":"Input dataset","outputs":[],"position":{"left":0,"top":374},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\"}","tool_version":null,"type":"data_input","uuid":"773b7592-d66c-4a85-92ba-06c0ab0f843f","when":null,"workflow_outputs":[]},"3":{"annotation":"","content_id":"CONVERTER_gz_to_uncompressed","errors":null,"id":3,"input_connections":{"input1":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Convert compressed file to uncompressed.","outputs":[{"name":"output1","type":"auto"}],"position":{"left":233,"top":346},"post_job_actions":{"ChangeDatatypeActionoutput1":{"action_arguments":{"newtype":"fasta"},"action_type":"ChangeDatatypeAction","output_name":"output1"},"RenameDatasetActionoutput1":{"action_arguments":{"newname":"decompressed_rg_file"},"action_type":"RenameDatasetAction","output_name":"output1"}},"tool_id":"CONVERTER_gz_to_uncompressed","tool_state":"{\\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"1ce51556-5526-4c7e-9b94-9634e8e25ed0","when":null,"workflow_outputs":[{"label":"decompressed_rg_file","output_name":"output1","uuid":"9d6bde4a-7180-4097-9ffd-6992111a617c"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","errors":null,"id":4,"input_connections":{"fastq_input|analysis_type_selector":{"id":1,"output_name":"output"},"fastq_input|fastq_input1":{"id":0,"output_name":"output"},"reference_source|ref_file":{"id":3,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"fastq_input"},{"description":"runtime parameter for tool Map with minimap2","name":"reference_source"}],"label":null,"name":"Map with minimap2","outputs":[{"name":"alignment_output","type":"bam"}],"position":{"left":573,"top":40},"post_job_actions":{"RenameDatasetActionalignment_output":{"action_arguments":{"newname":"map_with_minimap2"},"action_type":"RenameDatasetAction","output_name":"alignment_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1","tool_shed_repository":{"changeset_revision":"66367287b4e6","name":"minimap2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"alignment_options\\": {\\"splicing\\": {\\"splice_mode\\": \\"preset\\", \\"__current_case__\\": 0}, \\"A\\": null, \\"B\\": null, \\"O\\": null, \\"O2\\": null, \\"E\\": null, \\"E2\\": null, \\"z\\": null, \\"z2\\": null, \\"s\\": null, \\"no_end_flt\\": true}, \\"fastq_input\\": {\\"fastq_input_selector\\": \\"single\\", \\"__current_case__\\": 0, \\"fastq_input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"analysis_type_selector\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"indexing_options\\": {\\"H\\": false, \\"k\\": null, \\"w\\": null, \\"I\\": null}, \\"io_options\\": {\\"output_format\\": \\"BAM\\", \\"Q\\": false, \\"L\\": false, \\"K\\": null, \\"cs\\": null, \\"c\\": false, \\"eqx\\": false, \\"Y\\": false}, \\"mapping_options\\": {\\"N\\": null, \\"F\\": null, \\"f\\": null, \\"kmer_ocurrence_interval\\": {\\"interval\\": \\"\\", \\"__current_case__\\": 1}, \\"min_occ_floor\\": null, \\"q_occ_frac\\": \\"0.01\\", \\"g\\": null, \\"r\\": null, \\"n\\": null, \\"m\\": null, \\"max_chain_skip\\": null, \\"max_chain_iter\\": null, \\"X\\": false, \\"p\\": null, \\"mask_len\\": null}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"ref_file\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.28+galaxy1","type":"tool","uuid":"6da37636-966d-4975-bce3-d650feaeb17c","when":null,"workflow_outputs":[{"label":"map_with_minimap2","output_name":"alignment_output","uuid":"31779c23-4f26-418e-9418-2551e883dbe7"}]},"5":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/1.0.10+galaxy1","errors":null,"id":5,"input_connections":{"bam_input":{"id":4,"output_name":"alignment_output"},"ref_source|ref_fasta":{"id":3,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool Clair3","name":"ref_source"}],"label":null,"name":"Clair3","outputs":[{"name":"merge_output","type":"vcf_bgzip"},{"name":"pileup","type":"vcf_bgzip"},{"name":"full_alignment","type":"vcf_bgzip"}],"position":{"left":853,"top":46},"post_job_actions":{"RenameDatasetActionfull_alignment":{"action_arguments":{"newname":"clair3_full_alignment_vcf"},"action_type":"RenameDatasetAction","output_name":"full_alignment"},"RenameDatasetActionmerge_output":{"action_arguments":{"newname":"clair3_merged_output"},"action_type":"RenameDatasetAction","output_name":"merge_output"},"RenameDatasetActionpileup":{"action_arguments":{"newname":"clair3_pileup_vcf"},"action_type":"RenameDatasetAction","output_name":"pileup"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/1.0.10+galaxy1","tool_shed_repository":{"changeset_revision":"c9218b9dc7f0","name":"clair3","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"qual\\": \\"0\\", \\"snp_min_af\\": \\"0.08\\", \\"indel_min_af\\": \\"0.15\\", \\"enable_phasing\\": false, \\"no_phasing_for_fa\\": false, \\"print_ref_calls\\": false, \\"ploidity_model\\": \\"--haploid_precise\\", \\"chunk_size\\": \\"5000000\\"}, \\"bam_input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"bed_or_vcf\\": {\\"bed_or_vcf_selector\\": \\"unrestricted\\", \\"__current_case__\\": 2, \\"include_all_ctgs\\": true}, \\"gvcf\\": false, \\"model_source\\": {\\"source\\": \\"built-in\\", \\"__current_case__\\": 0, \\"select_built_in\\": \\"r941_prom_hac_g360+g422\\"}, \\"ont_license_agree\\": false, \\"output_files\\": [\\"full_alignment\\", \\"pileup\\"], \\"ref_source\\": {\\"source\\": \\"history\\", \\"__current_case__\\": 0, \\"ref_fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.10+galaxy1","type":"tool","uuid":"ffbad354-41de-488d-bb37-929926bbfb31","when":null,"workflow_outputs":[{"label":"clair3_full_alignment_vcf","output_name":"full_alignment","uuid":"987f9145-83ab-44f1-801a-b0d527ecbce8"},{"label":"clair3_pileup_vcf","output_name":"pileup","uuid":"1cf1ee6c-4cb7-45e3-9c7b-88a1b678dd53"},{"label":"clair3_merged_output","output_name":"merge_output","uuid":"fab82215-f59a-43b1-92d7-37629a1fdb65"}]},"6":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_depth/samtools_depth/1.15.1+galaxy2","errors":null,"id":6,"input_connections":{"input_bams":{"id":4,"output_name":"alignment_output"}},"inputs":[],"label":null,"name":"Samtools depth","outputs":[{"name":"output","type":"tabular"}],"position":{"left":907.9936348039837,"top":625.9951261037309},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_depth/samtools_depth/1.15.1+galaxy2","tool_shed_repository":{"changeset_revision":"971dc85e9441","name":"samtools_depth","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"additional_options\\": {\\"required_flags\\": null, \\"skipped_flags\\": [\\"4\\", \\"256\\", \\"512\\", \\"1024\\"], \\"deletions\\": false, \\"single_read\\": false}, \\"all\\": \\"\\", \\"basequality\\": null, \\"cond_region\\": {\\"select_region\\": \\"no\\", \\"__current_case__\\": 0}, \\"input_bams\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"mapquality\\": null, \\"maxdepth\\": null, \\"minlength\\": null, \\"output_options\\": {\\"header\\": false}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy2","type":"tool","uuid":"d18e3c23-1d6c-4001-9a77-5167c069585b","when":null,"workflow_outputs":[]},"7":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_coverage/samtools_coverage/1.15.1+galaxy2","errors":null,"id":7,"input_connections":{"condition_input|input":{"id":4,"output_name":"alignment_output"}},"inputs":[{"description":"runtime parameter for tool Samtools coverage","name":"condition_input"}],"label":null,"name":"Samtools coverage","outputs":[{"name":"output","type":"tabular"}],"position":{"left":929.9945251946597,"top":959.0024861037228},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/samtools_coverage/samtools_coverage/1.15.1+galaxy2","tool_shed_repository":{"changeset_revision":"c4ff13d2aab3","name":"samtools_coverage","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"additional_options\\": {\\"min_read_length\\": \\"0\\", \\"min_mq\\": \\"0\\", \\"min_bq\\": \\"0\\", \\"required_flags\\": null, \\"skipped_flags\\": [\\"4\\", \\"256\\", \\"512\\", \\"1024\\"], \\"region\\": \\"\\"}, \\"condition_histogram\\": {\\"histogram_select\\": \\"no\\", \\"__current_case__\\": 1}, \\"condition_input\\": {\\"input_pooling\\": \\"No\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy2","type":"tool","uuid":"c7c149a8-ec6b-4cd3-80d0-bf2badca7e9d","when":null,"workflow_outputs":[]},"8":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy4","errors":null,"id":8,"input_connections":{"input_file":{"id":5,"output_name":"merge_output"},"reference_source|fasta_ref":{"id":3,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool bcftools norm","name":"reference_source"}],"label":null,"name":"bcftools norm","outputs":[{"name":"output_file","type":"vcf"}],"position":{"left":1089,"top":48},"post_job_actions":{"RenameDatasetActionoutput_file":{"action_arguments":{"newname":"normalized_vcf_output"},"action_type":"RenameDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy4","tool_shed_repository":{"changeset_revision":"271965b55437","name":"bcftools_norm","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"atomize\\": false, \\"check_ref\\": \\"w\\", \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"multiallelics\\": {\\"mode\\": \\"\\", \\"__current_case__\\": 0}, \\"normalize_indels\\": false, \\"old_rec_tag\\": \\"\\", \\"output_type\\": \\"v\\", \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"fasta_ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rm_dup\\": \\"\\", \\"sec_default\\": {\\"site_win\\": \\"1000\\"}, \\"sec_restrict\\": {\\"regions\\": {\\"regions_src\\": \\"__none__\\", \\"__current_case__\\": 0}, \\"regions_overlap\\": null, \\"targets\\": {\\"targets_src\\": \\"__none__\\", \\"__current_case__\\": 0}, \\"targets_overlap\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy4","type":"tool","uuid":"9a34ec91-725e-49b2-973b-1c508e4bdc5d","when":null,"workflow_outputs":[{"label":"normalized_vcf_output","output_name":"output_file","uuid":"22553aa1-f5db-4d37-87c4-1164dbb3d2d5"}]},"9":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":9,"input_connections":{"input":{"id":6,"output_name":"output"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1161.0269599992962,"top":623.8951200002152},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"--complement\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": \\"1,2\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"f6f94754-2660-4fe4-b1dc-937728fc898f","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":10,"input_connections":{"input":{"id":7,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1158.944537401691,"top":952.9024800002072},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"d728f856-b2dd-4261-9b25-84c36bdaced7","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","errors":null,"id":11,"input_connections":{"input":{"id":8,"output_name":"output_file"}},"inputs":[],"label":null,"name":"SnpSift Filter","outputs":[{"name":"output","type":"vcf"}],"position":{"left":1317,"top":46},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"quality_filtered_vcf_output"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filter_expression\\": {\\"type\\": \\"simple\\", \\"__current_case__\\": 0, \\"expr\\": \\"(QUAL > 2)\\"}, \\"filtering\\": {\\"mode\\": \\"entries\\", \\"__current_case__\\": 0}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inverse\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy1","type":"tool","uuid":"9a9a0e61-4966-496a-85df-c3a137abe8d3","when":null,"workflow_outputs":[{"label":"quality_filtered_vcf_output","output_name":"output","uuid":"3e7981ec-2205-4571-93eb-10c0dd14b288"}]},"12":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy1","errors":null,"id":12,"input_connections":{"singtabop|input":{"id":9,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Table Compute","name":"singtabop"}],"label":null,"name":"Table Compute","outputs":[{"name":"table","type":"tabular"}],"position":{"left":1430.0269599992962,"top":617.8951200002152},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy1","tool_shed_repository":{"changeset_revision":"56a8da1f2243","name":"table_compute","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"out_opts\\": [\\"ignore_nas\\"], \\"precision\\": \\"2\\", \\"singtabop\\": {\\"use_type\\": \\"single\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"col_row_names\\": null, \\"adv\\": {\\"header\\": null, \\"nrows\\": null, \\"skipfooter\\": null, \\"skip_blank_lines\\": true}, \\"user\\": {\\"mode\\": \\"matrixapply\\", \\"__current_case__\\": 3, \\"matrixapply_func\\": {\\"vector_op\\": \\"mean\\", \\"__current_case__\\": 6}, \\"dimension\\": \\"0\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.4+galaxy1","type":"tool","uuid":"b20b66f5-e975-4947-8935-8c628d016514","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"Cut1","errors":null,"id":13,"input_connections":{"input":{"id":10,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1392.9445984368472,"top":958.9024800002072},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c6\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"6712cd94-ab45-4052-a2a2-507bbf0652e8","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","errors":null,"id":14,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"SnpSift Extract Fields","outputs":[{"name":"output","type":"tabular"}],"position":{"left":1559,"top":45},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"extracted_fields_from_the_vcf_output"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"extracted_fields_from_the_vcf_output"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0","tool_shed_repository":{"changeset_revision":"5fab4f81391d","name":"snpsift","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"empty_text\\": \\"\\", \\"extract\\": \\"CHROM POS ID REF ALT FILTER\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_effect_per_line\\": false, \\"separator\\": \\"\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"4.3+t.galaxy0","type":"tool","uuid":"c9d51e45-9c09-438e-ab4b-c280a91dd2bc","when":null,"workflow_outputs":[{"label":"extracted_fields_from_the_vcf_output","output_name":"output","uuid":"1b0f4f2c-4717-45be-b580-fe10cba78c35"}]},"15":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4","errors":null,"id":15,"input_connections":{"input_file":{"id":11,"output_name":"output"},"reference_source|fasta_ref":{"id":3,"output_name":"output1"}},"inputs":[{"description":"runtime parameter for tool bcftools consensus","name":"reference_source"},{"description":"runtime parameter for tool bcftools consensus","name":"sec_default"}],"label":null,"name":"bcftools consensus","outputs":[{"name":"output_file","type":"fasta"}],"position":{"left":1565,"top":248},"post_job_actions":{"RenameDatasetActionoutput_file":{"action_arguments":{"newname":"bcftools_consensus"},"action_type":"RenameDatasetAction","output_name":"output_file"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4","tool_shed_repository":{"changeset_revision":"8aa352093d73","name":"bcftools_consensus","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"absent\\": \\"\\", \\"chain\\": false, \\"input_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"reference_source\\": {\\"reference_source_selector\\": \\"history\\", \\"__current_case__\\": 1, \\"fasta_ref\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"rename\\": false, \\"sec_default\\": {\\"mask\\": {\\"__class__\\": \\"RuntimeValue\\"}, \\"iupac_codes\\": false, \\"sample\\": \\"\\", \\"select_haplotype\\": null, \\"mark_del\\": \\"\\", \\"mark_ins\\": null, \\"mark_snv\\": null, \\"conditional_mask\\": {\\"selector\\": \\"disabled\\", \\"__current_case__\\": 0}}, \\"sec_restrict\\": {\\"include\\": null, \\"exclude\\": null}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.15.1+galaxy4","type":"tool","uuid":"872ba641-1909-43ea-b3c2-62e6773583a6","when":null,"workflow_outputs":[{"label":"bcftools_consensus","output_name":"output_file","uuid":"49497f3c-9332-4924-bcfd-cb71788ad2c2"}]},"16":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.5+galaxy0","errors":null,"id":16,"input_connections":{"infile":{"id":13,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Select first","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1623.6611511712222,"top":952.9024800002072},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"\\", \\"count\\": \\"1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"de8ceb06-c76a-4de5-ad01-4e86edd32b84","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":17,"input_connections":{"input":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1789.4166259765625,"top":47.899993896484375},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"b7da4892-be3a-4cc5-8cdf-5205b7435220","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":18,"input_connections":{"input_list":{"id":16,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1861.2076226643767,"top":940.9587935676732},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"mapping_coverage_percentage_per_sample"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"mapping_coverage_percentage_per_sample"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"bc05695c-433b-44df-9c94-d345f66be3cd","when":null,"workflow_outputs":[{"label":"mapping_coverage_percentage_per_sample","output_name":"output","uuid":"5ee1c158-1a85-45ca-9abd-b30632b3092c"}]},"19":{"annotation":"","content_id":"Count1","errors":null,"id":19,"input_connections":{"input":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2020.2333984375,"top":51.899993896484375},"post_job_actions":{},"tool_id":"Count1","tool_state":"{\\"column\\": \\"1\\", \\"delim\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sorting\\": \\"value\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"1827b4be-3a5e-45d8-b08f-9b90871ecd6c","when":null,"workflow_outputs":[]},"20":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","errors":null,"id":20,"input_connections":{"input":{"id":18,"output_name":"output"}},"inputs":[],"label":null,"name":"Advanced Cut","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2089.961199999347,"top":949.9024800002072},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"\\", \\"cut_type_options\\": {\\"cut_element\\": \\"-f\\", \\"__current_case__\\": 0, \\"delimiter\\": \\"\\", \\"colnames_option\\": {\\"header\\": \\"N\\", \\"__current_case__\\": 1, \\"list\\": \\"1\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"74036007-7df4-4e3f-86c3-e73f511bdbbe","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"Cut1","errors":null,"id":21,"input_connections":{"input":{"id":19,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2250.1666259765625,"top":52.899993896484375},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"aff32212-caf5-4e5c-86b3-fd25097dc051","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"Paste1","errors":null,"id":22,"input_connections":{"input1":{"id":20,"output_name":"output"},"input2":{"id":12,"output_name":"table"}},"inputs":[],"label":null,"name":"Paste","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2249.0333251953125,"top":620.8999938964844},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"mapping_mean_depth_per_sample"},"action_type":"RenameDatasetAction","output_name":"out_file1"},"TagDatasetActionout_file1":{"action_arguments":{"tags":"mapping_mean_depth_per_sample"},"action_type":"TagDatasetAction","output_name":"out_file1"}},"tool_id":"Paste1","tool_state":"{\\"delimiter\\": \\"T\\", \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"input2\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"38616748-2327-49f3-a53e-c204a356d97b","when":null,"workflow_outputs":[{"label":"mapping_mean_depth_per_sample","output_name":"out_file1","uuid":"25b7b87b-5ef1-487f-90cc-81351a2f81ce"}]},"23":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.5+galaxy0","errors":null,"id":23,"input_connections":{"infile":{"id":21,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Select first","outputs":[{"name":"outfile","type":"input"}],"position":{"left":2493.75,"top":44.899993896484375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.5+galaxy0","tool_shed_repository":{"changeset_revision":"3dc70b59608c","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"complement\\": \\"\\", \\"count\\": \\"1\\", \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.5+galaxy0","type":"tool","uuid":"49b1771c-f107-4dff-8683-65aa58c1c0ad","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":24,"input_connections":{"input_list":{"id":23,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":2735.0166015625,"top":49.899993896484375},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"282932f1-bbc9-4081-b154-ddb3a2ecb779","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","errors":null,"id":25,"input_connections":{"input":{"id":24,"output_name":"output"}},"inputs":[],"label":null,"name":"Column Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":2961.5528311718776,"top":47.899993896484375},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"number_of_variants_per_sample"},"action_type":"RenameDatasetAction","output_name":"out_file1"},"TagDatasetActionout_file1":{"action_arguments":{"tags":"number_of_variants_per_sample"},"action_type":"TagDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"^$\\", \\"replacement\\": \\"0\\"}], \\"field\\": \\"2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"dfa6d815-9c8a-4a04-8862-00a6f83789dc","when":null,"workflow_outputs":[{"label":"number_of_variants_per_sample","output_name":"out_file1","uuid":"2852c4ee-24ea-4df7-b59e-54eb6e2f470b"}]}},"tags":["name:Collection","name:microGalaxy","name:PathoGFAIR","name:IWC"],"uuid":"435875a1-72c4-4fda-8943-3a78f25c6574","version":72},"readme":"# Allele-based Pathogen Identification\\n\\nThis workflow identifies pathogens using an allelic approach, detecting Single Nucleotide Polymorphisms (SNPs) to track emerging variants, i.e. markers showing evolutionary histories of homogeneous strains. This process includes SNP calling, aimed at identifying novel pathogen strains and elucidating discrepancies compared to reference sequences, thereby facilitating the tracking of emerging variants. Within this workflow, both variants and SNPs are discerned, serving as crucial elements for subsequent pathogen identification and variant tracking purposes.\\n\\n## Input Datasets\\n- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger or fastqsanger.gz` format, the output of **Nanopore Preprocessing** workflow.\\n- A reference genome to the tested pathogen.\\n\\n## Output Datasets\\n- VCF files indicating identified variants and SNPs, BAM files with mapping results, and Tabular files with mapping depth and coverage calculations.\\n\\nIf you\'re unsure how to use this workflows, or if you want to see it in action with test datasets, it is included in our detailed training material for foodborne pathogen detection and tracking. You can find step-by-step instructions and practical examples in the following [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)\\n","changelog":"# Changelog\\n\\n## [0.1.4] - 2025-03-10\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.5+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/9.5+galaxy0`\\n\\n## [0.1.3] - 2025-02-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/1.0.8+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/1.0.10+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy1`\\n\\n## [0.1.2] - 2024-09-24\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/0.1.12+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/1.0.8+galaxy1`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy4`\\n- `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy2`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy4`\\n\\n## [0.1.1] - 2024-06-28\\n\\n### Automatic update\\n- `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.24+galaxy0` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy3`\\n- `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.9+galaxy1` was updated to `toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3`\\n\\n## [0.1] - 2024-06-19\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Allele-based Pathogen Identification\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ collection_of_preprocessed_samples\\"]@{ shape: docs }\\n1[\\"ℹ️ samples_profile\\"]@{ shape: lean-l }\\n2[\\"ℹ️ reference_genome_of_tested_strain\\"]@{ shape: doc }\\n3[\\"Convert compressed file to uncompressed.\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Map with minimap2\\"]@{ shape: process }\\n1 --> 4\\n0 --> 4\\n3 --> 4\\n5[\\"Clair3\\"]@{ shape: process }\\n4 --> 5\\n3 --> 5\\n6[\\"Samtools depth\\"]@{ shape: process }\\n4 --> 6\\n7[\\"Samtools coverage\\"]@{ shape: process }\\n4 --> 7\\n8[\\"bcftools norm\\"]@{ shape: process }\\n5 --> 8\\n3 --> 8\\n9[\\"Advanced Cut\\"]@{ shape: process }\\n6 --> 9\\n10[\\"Remove beginning\\"]@{ shape: process }\\n7 --> 10\\n11[\\"SnpSift Filter\\"]@{ shape: process }\\n8 --> 11\\n12[\\"Table Compute\\"]@{ shape: process }\\n9 --> 12\\n13[\\"Cut\\"]@{ shape: process }\\n10 --> 13\\n14[\\"SnpSift Extract Fields\\"]@{ shape: process }\\n11 --> 14\\n15[\\"bcftools consensus\\"]@{ shape: process }\\n11 --> 15\\n3 --> 15\\n16[\\"Select first\\"]@{ shape: process }\\n13 --> 16\\n17[\\"Remove beginning\\"]@{ shape: process }\\n14 --> 17\\n18[\\"Collapse Collection\\"]@{ shape: process }\\n16 --> 18\\n19[\\"Count\\"]@{ shape: process }\\n17 --> 19\\n20[\\"Advanced Cut\\"]@{ shape: process }\\n18 --> 20\\n21[\\"Cut\\"]@{ shape: process }\\n19 --> 21\\n22[\\"Paste\\"]@{ shape: process }\\n20 --> 22\\n12 --> 22\\n23[\\"Select first\\"]@{ shape: process }\\n21 --> 23\\n24[\\"Collapse Collection\\"]@{ shape: process }\\n23 --> 24\\n25[\\"Column Regex Find And Replace\\"]@{ shape: process }\\n24 --> 25\\n```\\n","updated":"2025-03-10T00:00:00","trsID":"#workflow/github.com/iwc-workflows/allele-based-pathogen-identification/main","dockstore_id":27326,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.12744316","tests":[{"doc":"Test outline for Allele-based-Pathogen-Identification","job":{"reference_genome_of_tested_strain":{"class":"File","location":"https://zenodo.org/record/12190648/files/reference_genome_of_tested_strain.fasta.gz","filetype":"fasta.gz"},"collection_of_preprocessed_samples":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10.fastq.gz","filetype":"fastqsanger.gz"},{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12.fastq.gz","filetype":"fastqsanger.gz"}]},"samples_profile":null},"outputs":{"mapping_mean_depth_per_sample":{"path":"test-data/mapping_mean_depth_per_sample.tabular"}}}]}],"path":"./workflows/microbiome/pathogen-identification/allele-based-pathogen-identification"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Pathogen-Detection-PathoGFAIR-Samples-Aggregation-and-Visualisation.ga","testParameterFiles":["/Pathogen-Detection-PathoGFAIR-Samples-Aggregation-and-Visualisation-tests.yml"],"authors":[{"name":"Engy Nasr","orcid":"0000-0001-9047-4215","url":"https://orcid.org/0000-0001-9047-4215"},{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987","url":"https://orcid.org/0000-0001-9852-1987"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X","url":"https://orcid.org/0000-0003-2982-388X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Pathogens of all samples report generation and visualization","comments":[{"child_steps":[39,0,33,27,21,15,10,45],"color":"yellow","data":{"title":"AMR Count table"},"id":0,"position":[0,186],"size":[1859.9,219.3],"type":"frame"},{"child_steps":[41,46,16,11,1,23,29,35],"color":"yellow","data":{"title":"VF count table and heatmap"},"id":1,"position":[15,646.1000000000001],"size":[1888,285],"type":"frame"},{"child_steps":[40,50,34,28,22],"color":"yellow","data":{"title":"Both VF and AMR count table"},"id":2,"position":[609.7,418.20000000000005],"size":[1438.3,199.5],"type":"frame"},{"child_steps":[17,30,68,12,2,24,36,42,47,51,54,57,60,62,64,66],"color":"yellow","data":{"title":"AMR Phylogenetic tree for all samples"},"id":3,"position":[31,1088.0000000000002],"size":[4170,273],"type":"frame"},{"child_steps":[18,19,20,25,31,32,37,43,48,52,55,58,69,13,3,14,4,26,38,44,49,53,56,59,61,63,65,67],"color":"yellow","data":{"title":"VFs Phylogenetic Tree for all samples and Phylogenetic trees for each VF genes across all samples"},"id":4,"position":[27,1415.0000000000002],"size":[4170,620],"type":"frame"},{"child_steps":[5,6,7,8,9],"color":"yellow","data":{"title":"Tabulars for Bar chart Visualisations &amp; Metadata for extending with Jupytools"},"id":5,"position":[2104,0],"size":[650,614],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0001-9047-4215","name":"Engy Nasr","url":"https://orcid.org/0000-0001-9047-4215"},{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut","url":"https://orcid.org/0000-0001-9852-1987"},{"class":"Person","identifier":"0000-0003-2982-388X","name":"Paul Zierep"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation","report":{"markdown":"# Pathogen Detection - PathoGFAIR Samples Aggregation and Visualisation Workflow Report\\nBelow are the results for the PathoGFAIR Samples Aggregation and Visualisation Workflow\\n\\nThis workflow was run on:\\n\\n```galaxy\\ngenerate_time()\\n```\\n\\nWith Galaxy version:\\n\\n```galaxy\\ngenerate_galaxy_version()\\n```\\n\\n## Workflow Inputs\\nTabular files and a FASTA file from the Gene-based Pathogen Identification workflow, four other tabular files from Nanopore Preprocessing and Nanopore - Allele-based Pathogen Identification workflow, and an optional Metadata tabular file with more sample information:\\n\\nFrom Gene-based Pathogenic Identification workflow: \\n- contigs, FASTA file\\n- VFs, Tabular file\\n- vfs_of_genes_identified_by_vfdb, Tabular file\\n- AMRs, Tabular file\\n- amr_identified_by_ncbi, Tabular file\\n\\nFrom Nanopore - Allele bases Pathogen Identification workflow: \\n- number_of_variants_per_sample, Tabular file\\n- mapping_mean_depth_per_sample, Tabular file\\n- mapping_coverage_percentage_per_sample, Tabular file\\n\\nFrom Nanopore Preprocessing: \\n- removed_hosts_percentage_tabular, Tabular file\\n\\n## Some of the Workflow Outputs\\n\\n1- All Samples VFs Heatmap\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"heatmap_png\\")\\n```\\n\\n2- All samples phylogenetic tree VFs based\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"all_samples_phylogenetic_tree_based_vfs\\")\\n```\\n\\n3- All samples Phylogenetic tree AMR based \\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"all_samples_phylogenetic_tree_based_amrs\\")\\n```\\n\\n4- Bar-plot for the Number of reads before host sequences removal and Number of found host reads per sample, performed in the Nanopore - Preprocessing workflow\\n\\n\\n5- Barplot for the total number of removed host sequences per sample\\n\\n \\n6- Barplot for the Mapping mean depth of coverage per sample\\n\\n\\n6- Barplot for the Mapping breadth of coverage percentage per sample\\n\\n\\n7- Barplot for the total number of complex variants and SNPs identified per sample\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n"},"steps":{"0":{"annotation":"output_of_abricate_with_amrfinderncbi","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"output_of_abricate_with_amrfinderncbi","name":"amr_identified_by_ncbi"}],"label":"amr_identified_by_ncbi","name":"Input dataset collection","outputs":[],"position":{"left":19.98333740234375,"top":226.0443929882739},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"a7f1c4a9-5d07-49df-82cd-1bbcf3fd628f","when":null,"workflow_outputs":[]},"1":{"annotation":"output_of_abricate_with_vfdb","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"output_of_abricate_with_vfdb","name":"vfs_of_genes_identified_by_vfdb"}],"label":"vfs_of_genes_identified_by_vfdb","name":"Input dataset collection","outputs":[],"position":{"left":34.98484848484827,"top":728.141282170953},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"2908a402-5edf-4bcc-a970-1db21c474d09","when":null,"workflow_outputs":[]},"2":{"annotation":"amrs","content_id":null,"errors":null,"id":2,"input_connections":{},"inputs":[{"description":"amrs","name":"amrs"}],"label":"amrs","name":"Input dataset collection","outputs":[],"position":{"left":50.9848484848481,"top":1153.9745890557185},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"297bede6-bef8-44f1-bcbb-134464821c44","when":null,"workflow_outputs":[]},"3":{"annotation":"Contigs","content_id":null,"errors":null,"id":3,"input_connections":{},"inputs":[{"description":"Contigs","name":"contigs"}],"label":"contigs","name":"Input dataset collection","outputs":[],"position":{"left":600.9833984375,"top":1485.9999877929688},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"ce948faa-f897-47e3-ac8c-0227afdd8203","when":null,"workflow_outputs":[]},"4":{"annotation":"VFs","content_id":null,"errors":null,"id":4,"input_connections":{},"inputs":[{"description":"VFs","name":"vfs"}],"label":"vfs","name":"Input dataset collection","outputs":[],"position":{"left":46.984848484848044,"top":1707.0079752861873},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": \\"\\", \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"8ff72508-664e-4af1-b0e6-ca2643626295","when":null,"workflow_outputs":[]},"5":{"annotation":"removed_hosts_percentage_tabular","content_id":null,"errors":null,"id":5,"input_connections":{},"inputs":[{"description":"removed_hosts_percentage_tabular","name":"removed_hosts_percentage_tabular"}],"label":"removed_hosts_percentage_tabular","name":"Input dataset","outputs":[],"position":{"left":2118,"top":45},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"f3419e21-1bc4-4d48-adb3-551186a59cbf","when":null,"workflow_outputs":[]},"6":{"annotation":"mapping_mean_depth_per_sample","content_id":null,"errors":null,"id":6,"input_connections":{},"inputs":[{"description":"mapping_mean_depth_per_sample","name":"mapping_mean_depth_per_sample"}],"label":"mapping_mean_depth_per_sample","name":"Input dataset","outputs":[],"position":{"left":2117,"top":191},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"45b393c2-ba47-421c-a97c-be576c3b5b9d","when":null,"workflow_outputs":[]},"7":{"annotation":"mapping_coverage_percentage_per_sample","content_id":null,"errors":null,"id":7,"input_connections":{},"inputs":[{"description":"mapping_coverage_percentage_per_sample","name":"mapping_coverage_percentage_per_sample"}],"label":"mapping_coverage_percentage_per_sample","name":"Input dataset","outputs":[],"position":{"left":2122,"top":317},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"3b907a00-b207-49e4-984e-30e95ba33247","when":null,"workflow_outputs":[]},"8":{"annotation":"number_of_variants_per_sample","content_id":null,"errors":null,"id":8,"input_connections":{},"inputs":[{"description":"number_of_variants_per_sample","name":"number_of_variants_per_sample"}],"label":"number_of_variants_per_sample","name":"Input dataset","outputs":[],"position":{"left":2124,"top":456},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"f159a3af-e63e-46fa-a5ac-e01d2f6858cf","when":null,"workflow_outputs":[]},"9":{"annotation":"samples_metadata","content_id":null,"errors":null,"id":9,"input_connections":{},"inputs":[{"description":"samples_metadata","name":"metadata"}],"label":"metadata","name":"Input dataset","outputs":[],"position":{"left":2407,"top":50},"tool_id":null,"tool_state":"{\\"optional\\": true, \\"tag\\": null}","tool_version":null,"type":"data_input","uuid":"50dca89f-20f6-46ea-975e-1b727e5a01a9","when":null,"workflow_outputs":[]},"10":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":10,"input_connections":{"input":{"id":0,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":241,"top":229.70000000000005},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"43b72f45-df6b-487a-94bb-b8c296796a54","when":null,"workflow_outputs":[]},"11":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":11,"input_connections":{"input":{"id":1,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":249,"top":722.7},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"96c9024b-3f8e-4d1d-8467-e467c92a7974","when":null,"workflow_outputs":[]},"12":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":12,"input_connections":{"input":{"id":2,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":330.9848484848481,"top":1127.9745890557185},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3531bc7a-3179-40a6-9b37-7532e8706da5","when":null,"workflow_outputs":[]},"13":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":13,"input_connections":{"input":{"id":3,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":887,"top":1463.7},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"9f80a828-111f-448a-af3f-9ec8203f7e57","when":null,"workflow_outputs":[]},"14":{"annotation":"","content_id":"__FILTER_FAILED_DATASETS__","errors":null,"id":14,"input_connections":{"input":{"id":4,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter failed datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":326.98484848484804,"top":1681.0079752861873},"post_job_actions":{},"tool_id":"__FILTER_FAILED_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"273dcefc-45a3-4217-a126-3f1cd1c548e4","when":null,"workflow_outputs":[]},"15":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":15,"input_connections":{"input":{"id":10,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":458.316650390625,"top":227.95000000000005},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3f11deed-67b9-48d5-a56d-59203e12ef78","when":null,"workflow_outputs":[]},"16":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":16,"input_connections":{"input":{"id":11,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":476.98333740234375,"top":685.133349609375},"post_job_actions":{},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"4e7d838b-f28d-419d-99b7-14e81ff93513","when":null,"workflow_outputs":[]},"17":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":17,"input_connections":{"input":{"id":12,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":610.984848484848,"top":1127.9745890557185},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"}},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"db9fb0c0-5ba9-4d44-8817-ddb3470f746f","when":null,"workflow_outputs":[]},"18":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":18,"input_connections":{"input_list":{"id":13,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1166.984848484848,"top":1455.0079752861868},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"all_samples_contigs_in_one_fasta_file"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"fb9327fe-11d1-4e20-95e9-3630958f9aed","when":null,"workflow_outputs":[{"label":"all_samples_contigs_in_one_fasta_file","output_name":"output","uuid":"86910e39-57bf-4a76-ac4c-739340fd2387"}]},"19":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":19,"input_connections":{"input_list":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":606.984848484848,"top":1661.0079752861873},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"all_vfs_in_one_tabular"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": false, \\"__current_case__\\": 1}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": true, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"29c0dd1a-0cc1-45c2-b9ac-4787d5c6d003","when":null,"workflow_outputs":[{"label":"all_vfs_in_one_tabular","output_name":"output","uuid":"02b996c6-a912-4e4e-b3ec-49601faaa452"}]},"20":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":20,"input_connections":{"input":{"id":14,"output_name":"output"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":606.984848484848,"top":1856.0079752861873},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"}},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"1547d2fd-18d9-49b9-a701-14eb4bccbfc8","when":null,"workflow_outputs":[]},"21":{"annotation":"","content_id":"Count1","errors":null,"id":21,"input_connections":{"input":{"id":15,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":675.316650390625,"top":236.38549220705966},"post_job_actions":{},"tool_id":"Count1","tool_state":"{\\"column\\": \\"1\\", \\"delim\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sorting\\": \\"value\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"6cb8fe59-f07e-431c-a56b-d5fc14b79148","when":null,"workflow_outputs":[]},"22":{"annotation":"","content_id":"Count1","errors":null,"id":22,"input_connections":{"input":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Count","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":629.6210737748572,"top":487.0620233773345},"post_job_actions":{},"tool_id":"Count1","tool_state":"{\\"column\\": \\"1\\", \\"delim\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"sorting\\": \\"value\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"36afdf86-010a-4d4b-9088-a5c2aaccef14","when":null,"workflow_outputs":[]},"23":{"annotation":"","content_id":"Grouping1","errors":null,"id":23,"input_connections":{"input1":{"id":16,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":652.9833984375,"top":822.133349609375},"post_job_actions":{},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"6\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"length\\", \\"opcol\\": \\"6\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"123ef067-7859-4bf4-bab3-384ee4b68cec","when":null,"workflow_outputs":[]},"24":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","errors":null,"id":24,"input_connections":{"infile":{"id":17,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Unique","outputs":[{"name":"outfile","type":"input"}],"position":{"left":890.984848484848,"top":1127.9745890557185},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"advanced\\", \\"__current_case__\\": 1, \\"column_start\\": \\"12\\", \\"column_end\\": \\"13\\"}, \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_numeric\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"9e349e46-366d-49f5-9dc1-e2a09641d810","when":null,"workflow_outputs":[]},"25":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6","errors":null,"id":25,"input_connections":{"infile":{"id":19,"output_name":"output"}},"inputs":[],"label":null,"name":"Split by group","outputs":[{"name":"split_output","type":"input"}],"position":{"left":886.984848484848,"top":1661.0079752861873},"post_job_actions":{"RenameDatasetActionsplit_output":{"action_arguments":{"newname":"split_by_group_collection"},"action_type":"RenameDatasetAction","output_name":"split_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6","tool_shed_repository":{"changeset_revision":"ff2a81aa3f08","name":"split_file_on_column","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"column\\": \\"13\\", \\"include_header\\": true, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.6","type":"tool","uuid":"eabf505c-a46f-4180-86c5-60c096f78b35","when":null,"workflow_outputs":[{"label":"split_by_group_collection","output_name":"split_output","uuid":"59f8cb09-424b-47b1-b94b-a612c2610cab"}]},"26":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","errors":null,"id":26,"input_connections":{"infile":{"id":20,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Unique","outputs":[{"name":"outfile","type":"input"}],"position":{"left":886.984848484848,"top":1856.0079752861873},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv_opts\\": {\\"adv_opts_selector\\": \\"advanced\\", \\"__current_case__\\": 1, \\"column_start\\": \\"12\\", \\"column_end\\": \\"13\\"}, \\"ignore_case\\": false, \\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"is_numeric\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"518f5f76-6822-426c-b665-18b9d010f4ca","when":null,"workflow_outputs":[]},"27":{"annotation":"","content_id":"Cut1","errors":null,"id":27,"input_connections":{"input":{"id":21,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":893.316650390625,"top":233.7188051953409},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"45f77e8c-377c-4c5b-9956-df628491623c","when":null,"workflow_outputs":[]},"28":{"annotation":"","content_id":"Cut1","errors":null,"id":28,"input_connections":{"input":{"id":22,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":882.2878218217322,"top":486.395397400772},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"2f18d7ed-c6b8-4d6c-ba32-c00f6b954119","when":null,"workflow_outputs":[]},"29":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":29,"input_connections":{"input":{"id":23,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":873.9848484848483,"top":686.1356351818547},"post_job_actions":{},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"66d6667f-1d20-44d8-899b-107e15d6735f","when":null,"workflow_outputs":[]},"30":{"annotation":"","content_id":"Cut1","errors":null,"id":30,"input_connections":{"input":{"id":24,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1170.984848484848,"top":1137.9745890557185},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c3,c4\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ad35c661-a2b4-4c92-b7cc-6c6bec3d361d","when":null,"workflow_outputs":[]},"31":{"annotation":"","content_id":"Cut1","errors":null,"id":31,"input_connections":{"input":{"id":25,"output_name":"split_output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1166.984848484848,"top":1671.0079752861873},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"adjusted_abricate_vfs_tabular_part1"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c3,c4\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"f69bdaed-4620-4ac1-9ca5-8e5f6d1ac3f9","when":null,"workflow_outputs":[{"label":"adjusted_abricate_vfs_tabular_part1","output_name":"out_file1","uuid":"eeb25a51-ea21-4a19-a196-55d5bd919b10"}]},"32":{"annotation":"","content_id":"Cut1","errors":null,"id":32,"input_connections":{"input":{"id":26,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":1166.984848484848,"top":1866.0079752861873},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"}},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c2,c3,c4\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ab999fa6-3720-432c-a9ee-7e0b747aee19","when":null,"workflow_outputs":[]},"33":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":33,"input_connections":{"input_list":{"id":27,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1136.2999877929688,"top":233.38549220705966},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"46304b64-6d89-4822-b2a0-8b36e84852c9","when":null,"workflow_outputs":[]},"34":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","errors":null,"id":34,"input_connections":{"input_list":{"id":28,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Collapse Collection","outputs":[{"name":"output","type":"input"}],"position":{"left":1130.2878218217322,"top":463.7287714242095},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0","tool_shed_repository":{"changeset_revision":"90981f86000f","name":"collapse_collections","owner":"nml","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"filename\\": {\\"add_name\\": true, \\"__current_case__\\": 0, \\"place_name\\": \\"same_once\\"}, \\"input_list\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"one_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"5.1.0","type":"tool","uuid":"16bf0bb8-b1c3-474e-9c82-a7c3702b6c68","when":null,"workflow_outputs":[]},"35":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","errors":null,"id":35,"input_connections":{"input_tabular":{"id":29,"output_name":"output"}},"inputs":[],"label":null,"name":"Column join","outputs":[{"name":"tabular_output","type":"tabular"}],"position":{"left":1154.9848484848483,"top":712.141282170953},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3","tool_shed_repository":{"changeset_revision":"3ddd99c7efee","name":"collection_column_join","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fill_char\\": \\"0\\", \\"has_header\\": \\"0\\", \\"identifier_column\\": \\"1\\", \\"include_outputs\\": null, \\"input_tabular\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"old_col_in_header\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"0.0.3","type":"tool","uuid":"05642ad5-2d5c-455e-919a-91ddf623729a","when":null,"workflow_outputs":[]},"36":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","errors":null,"id":36,"input_connections":{"fasta_source|fasta":{"id":18,"output_name":"output"},"input":{"id":30,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1460.984848484848,"top":1132.9745890557185},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","tool_shed_repository":{"changeset_revision":"a1a923cd89e8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.30.0+galaxy1","type":"tool","uuid":"8018f07e-551c-4fef-b7ef-42589ff5b59e","when":null,"workflow_outputs":[]},"37":{"annotation":"","content_id":"Remove beginning1","errors":null,"id":37,"input_connections":{"input":{"id":31,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Remove beginning","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1456.984848484848,"top":1605.0079752861868},"post_job_actions":{"ChangeDatatypeActionout_file1":{"action_arguments":{"newtype":"bed"},"action_type":"ChangeDatatypeAction","output_name":"out_file1"},"RenameDatasetActionout_file1":{"action_arguments":{"newname":"adjusted_abricate_vfs_tabular_part2"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"Remove beginning1","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"num_lines\\": \\"1\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"3eab5012-f658-4d5b-9919-83621281470c","when":null,"workflow_outputs":[{"label":"adjusted_abricate_vfs_tabular_part2","output_name":"out_file1","uuid":"aaaa4446-0817-4e5c-aa1b-9ec384f2a363"}]},"38":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","errors":null,"id":38,"input_connections":{"fasta_source|fasta":{"id":18,"output_name":"output"},"input":{"id":32,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1456.984848484848,"top":1800.0079752861873},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","tool_shed_repository":{"changeset_revision":"a1a923cd89e8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.30.0+galaxy1","type":"tool","uuid":"d9d6c096-58e0-4a33-af38-e94709c9af3b","when":null,"workflow_outputs":[]},"39":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","errors":null,"id":39,"input_connections":{"input":{"id":33,"output_name":"output"}},"inputs":[],"label":null,"name":"Column Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1387.63330078125,"top":233.05214870120028},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"amrs_count"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"^$\\", \\"replacement\\": \\"0\\"}], \\"field\\": \\"2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"a8a0cf10-6f23-436b-b7ee-28cd8f7eee61","when":null,"workflow_outputs":[{"label":"amrs_count","output_name":"out_file1","uuid":"4809c36b-31ef-4664-8e4e-47f0f72152de"}]},"40":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","errors":null,"id":40,"input_connections":{"input":{"id":34,"output_name":"output"}},"inputs":[],"label":null,"name":"Column Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1500.5544233842322,"top":470.395397400772},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"vfs_count"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"^$\\", \\"replacement\\": \\"0\\"}], \\"field\\": \\"2\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"8557f83f-3010-4d1d-976f-d0552086f06b","when":null,"workflow_outputs":[{"label":"vfs_count","output_name":"out_file1","uuid":"87efc81d-4d84-4af3-831f-dfe033c59f78"}]},"41":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","errors":null,"id":41,"input_connections":{"input":{"id":35,"output_name":"tabular_output"}},"inputs":[],"label":null,"name":"Column Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1434.9848484848483,"top":722.141282170953},"post_job_actions":{"RenameDatasetActionout_file1":{"action_arguments":{"newname":"heatmap_table"},"action_type":"RenameDatasetAction","output_name":"out_file1"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"#KEY\\", \\"replacement\\": \\"key\\"}], \\"field\\": \\"1\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"ea2bfbb1-3cfb-4a10-9443-d83ee964271f","when":null,"workflow_outputs":[{"label":"heatmap_table","output_name":"out_file1","uuid":"f5c221e3-00ef-4834-9a5f-a94c97fd6764"}]},"42":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","errors":null,"id":42,"input_connections":{"input":{"id":36,"output_name":"output"}},"inputs":[],"label":null,"name":"Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1705.9833374023438,"top":1146.9666625976563},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"_contig\\", \\"replacement\\": \\" contig\\"}], \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"6cf10249-819d-4279-9820-d556d6d20537","when":null,"workflow_outputs":[]},"43":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","errors":null,"id":43,"input_connections":{"fasta_source|fasta":{"id":18,"output_name":"output"},"input":{"id":37,"output_name":"out_file1"}},"inputs":[{"description":"runtime parameter for tool bedtools getfasta","name":"fasta_source"}],"label":null,"name":"bedtools getfasta","outputs":[{"name":"output","type":"fasta"}],"position":{"left":1736.984848484848,"top":1496.0079752861868},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"filtered_sequences_with_vfs_fasta"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1","tool_shed_repository":{"changeset_revision":"a1a923cd89e8","name":"bedtools","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"fasta_source\\": {\\"fasta_source_selector\\": \\"history\\", \\"__current_case__\\": 0, \\"fasta\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"name\\": false, \\"nameOnly\\": false, \\"split\\": false, \\"strand\\": false, \\"tab\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.30.0+galaxy1","type":"tool","uuid":"7283594d-0455-4f60-98f5-d9c315e28ee4","when":null,"workflow_outputs":[{"label":"filtered_sequences_with_vfs_fasta","output_name":"output","uuid":"82ce2107-89a3-438c-95bb-dc871b5258b7"}]},"44":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","errors":null,"id":44,"input_connections":{"input":{"id":38,"output_name":"output"}},"inputs":[],"label":null,"name":"Regex Find And Replace","outputs":[{"name":"out_file1","type":"input"}],"position":{"left":1676.9833374023438,"top":1806.9999877929688},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3","tool_shed_repository":{"changeset_revision":"503bcd6ebe4b","name":"regex_find_replace","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"checks\\": [{\\"__index__\\": 0, \\"pattern\\": \\"_contig\\", \\"replacement\\": \\" contig\\"}], \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.3","type":"tool","uuid":"519ad123-dedd-4116-8d89-99c4b5e6f612","when":null,"workflow_outputs":[]},"45":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/9.3+galaxy1","errors":null,"id":45,"input_connections":{"files":{"id":40,"output_name":"out_file1"},"first_file":{"id":39,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Multi-Join","outputs":[{"name":"outfile","type":"tabular"}],"position":{"left":1639.86669921875,"top":237.2521303906534},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"files\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"filler\\": \\"0\\", \\"first_file\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"ignore_dups\\": false, \\"input_header\\": false, \\"key_column\\": \\"1\\", \\"output_header\\": true, \\"value_columns\\": \\"2\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"e7d50f58-1fef-4bd3-8858-045b52c3c676","when":null,"workflow_outputs":[]},"46":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0","errors":null,"id":46,"input_connections":{"input1":{"id":41,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Heatmap w ggplot","outputs":[{"name":"output1","type":"png"},{"name":"output2","type":"pdf"}],"position":{"left":1680.9666748046875,"top":691.133349609375},"post_job_actions":{"RenameDatasetActionoutput1":{"action_arguments":{"newname":"heatmap_png"},"action_type":"RenameDatasetAction","output_name":"output1"},"RenameDatasetActionoutput2":{"action_arguments":{"newname":"heatmap_pdf"},"action_type":"RenameDatasetAction","output_name":"output2"},"TagDatasetActionoutput1":{"action_arguments":{"tags":"vfs_heatmap_png"},"action_type":"TagDatasetAction","output_name":"output1"},"TagDatasetActionoutput2":{"action_arguments":{"tags":"vfs_heatmap_pdf"},"action_type":"TagDatasetAction","output_name":"output2"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0","tool_shed_repository":{"changeset_revision":"10515715c940","name":"ggplot2_heatmap","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"adv\\": {\\"transform\\": \\"none\\", \\"cluster\\": true, \\"colorscheme\\": \\"whrd\\"}, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"inputdata\\": {\\"input_type\\": \\"with_header_rownames\\", \\"__current_case__\\": 2, \\"header\\": \\"TRUE\\", \\"row_names_index\\": \\"1\\", \\"sample_name_orientation\\": \\"FALSE\\"}, \\"out\\": {\\"unit_output_dim\\": \\"cm\\", \\"width_output_dim\\": \\"70.0\\", \\"height_output_dim\\": \\"70.0\\", \\"dpi_output_dim\\": \\"1000.0\\", \\"additional_output_format\\": \\"pdf\\"}, \\"title\\": \\"Pathogeneic Genes Per Samples\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"3.4.0+galaxy0","type":"tool","uuid":"14c05908-0f84-44a6-bc2e-1c4cd9d73fae","when":null,"workflow_outputs":[{"label":"heatmap_png","output_name":"output1","uuid":"97816bc2-fd0c-4077-a721-8dd1470879d1"},{"label":"heatmap_pdf","output_name":"output2","uuid":"c0417c91-a513-4c6a-9a62-3aac2f1f8e85"}]},"47":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":47,"input_connections":{"input":{"id":42,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1946,"top":1144.7},"post_job_actions":{},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"301c0fe7-4908-4cf7-ab99-a6b6fc9b41b9","when":null,"workflow_outputs":[]},"48":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","errors":null,"id":48,"input_connections":{"input":{"id":43,"output_name":"output"}},"inputs":[],"label":null,"name":"ClustalW","outputs":[{"name":"output","type":"clustal"},{"name":"dnd","type":"nhx"}],"position":{"left":2016.984848484848,"top":1541.0079752861868},"post_job_actions":{"RenameDatasetActiondnd":{"action_arguments":{"newname":"clustalw_on_input_dnd"},"action_type":"RenameDatasetAction","output_name":"dnd"},"RenameDatasetActionoutput":{"action_arguments":{"newname":"clustalw_on_input_clustal"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","tool_shed_repository":{"changeset_revision":"5b097fde95dd","name":"clustalw","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out_order\\": \\"ALIGNED\\", \\"outcontrol\\": {\\"outform\\": \\"fasta\\", \\"__current_case__\\": 0}, \\"range\\": {\\"mode\\": \\"complete\\", \\"__current_case__\\": 0}, \\"tree_calculation\\": {\\"OUTPUTTREE\\": \\"PHYLIP\\", \\"KIMURA\\": false, \\"TOSSGAPS\\": false}, \\"type_conditional\\": {\\"dnarna\\": \\"DNA\\", \\"__current_case__\\": 0, \\"algorithm_conditional\\": {\\"selector\\": \\"\\", \\"__current_case__\\": 0, \\"slow_pairwise_alignments\\": {\\"PWDNAMATRIX\\": \\"IUB\\", \\"PWGAPOPEN\\": null, \\"PWGAPEXT\\": null}}, \\"multiple_alignments\\": {\\"DNAMATRIX\\": \\"IUB\\", \\"GAPOPEN\\": null, \\"GAPEXT\\": null, \\"ENDGAPS\\": false, \\"GAPDIST\\": null, \\"NOPGAP\\": false, \\"NOHGAP\\": false, \\"MAXDIV\\": null, \\"NEGATIVE\\": false, \\"TRANSWEIGHT\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1+galaxy1","type":"tool","uuid":"7904af41-77da-4959-adae-01a374d16af6","when":null,"workflow_outputs":[{"label":"clustalw_on_input_dnd","output_name":"dnd","uuid":"9b7bd78c-f112-480b-a7df-c10711af254c"},{"label":"clustalw_on_input_clustal","output_name":"output","uuid":"6af20322-24da-4036-80de-37bed2d25848"}]},"49":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":49,"input_connections":{"input":{"id":44,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":1891,"top":1792.7},"post_job_actions":{},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"6384dc2f-3d4a-47c0-8789-0513635fbc16","when":null,"workflow_outputs":[]},"50":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1","errors":null,"id":50,"input_connections":{"infile":{"id":45,"output_name":"outfile"}},"inputs":[],"label":null,"name":"Replace Text","outputs":[{"name":"outfile","type":"input"}],"position":{"left":1827.9212446732947,"top":458.1953241585845},"post_job_actions":{"RenameDatasetActionoutfile":{"action_arguments":{"newname":"vfs_amrs_count_table"},"action_type":"RenameDatasetAction","output_name":"outfile"},"TagDatasetActionoutfile":{"action_arguments":{"tags":"vfs_amrs_count_table"},"action_type":"TagDatasetAction","output_name":"outfile"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1","tool_shed_repository":{"changeset_revision":"fbf99087e067","name":"text_processing","owner":"bgruening","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"infile\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"replacements\\": [{\\"__index__\\": 0, \\"column\\": \\"2\\", \\"find_pattern\\": \\".*V2\\", \\"replace_pattern\\": \\"AMRs count\\"}, {\\"__index__\\": 1, \\"column\\": \\"3\\", \\"find_pattern\\": \\".*V2\\", \\"replace_pattern\\": \\"VFs count\\"}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"9.3+galaxy1","type":"tool","uuid":"9c858c45-8750-4738-8d6c-7fd8b2fcb94b","when":null,"workflow_outputs":[{"label":"vfs_amrs_count_table","output_name":"outfile","uuid":"91745da0-0b8d-4a7a-b927-b36107f17ec5"}]},"51":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","errors":null,"id":51,"input_connections":{"input":{"id":47,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA-to-Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2175.9833374023438,"top":1144.066668701172},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","tool_shed_repository":{"changeset_revision":"e7ed3c310b74","name":"fasta_to_tabular","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"descr_columns\\": \\"3\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_first\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"b4a26739-32ba-4964-bc2c-a39a945820ee","when":null,"workflow_outputs":[]},"52":{"annotation":"","content_id":"__FILTER_EMPTY_DATASETS__","errors":null,"id":52,"input_connections":{"input":{"id":48,"output_name":"output"}},"inputs":[],"label":null,"name":"Filter empty datasets","outputs":[{"name":"output","type":"input"}],"position":{"left":2296.984848484848,"top":1557.0079752861868},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"filtered_empty_datasets"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"__FILTER_EMPTY_DATASETS__","tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.0","type":"tool","uuid":"5a0c3914-d149-4ccd-82fe-5065f9eb8995","when":null,"workflow_outputs":[{"label":"filtered_empty_datasets","output_name":"output","uuid":"32d00c1d-68c4-4069-8d5d-023aabdfadbe"}]},"53":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","errors":null,"id":53,"input_connections":{"input":{"id":49,"output_name":"output"}},"inputs":[],"label":null,"name":"FASTA-to-Tabular","outputs":[{"name":"output","type":"tabular"}],"position":{"left":2124.9833374023438,"top":1835.0833312988282},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1","tool_shed_repository":{"changeset_revision":"e7ed3c310b74","name":"fasta_to_tabular","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"descr_columns\\": \\"3\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"keep_first\\": \\"0\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"3f5f5558-90cc-4b90-af8b-a55aa84bb32c","when":null,"workflow_outputs":[]},"54":{"annotation":"","content_id":"Cut1","errors":null,"id":54,"input_connections":{"input":{"id":51,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2392.9833374023438,"top":1142.9666625976563},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1,c4\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"ad001843-0ee2-4bc5-83d6-71d98e5b3748","when":null,"workflow_outputs":[]},"55":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","errors":null,"id":55,"input_connections":{"input_selector|input":{"id":52,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FASTTREE","name":"input_selector"}],"label":null,"name":"FASTTREE","outputs":[{"name":"output","type":"nhx"}],"position":{"left":2576.984848484848,"top":1577.0079752861868},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"fasttree_nhx"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","tool_shed_repository":{"changeset_revision":"e005e659ae21","name":"fasttree","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_selector\\": {\\"maximize\\": \\"min\\", \\"__current_case__\\": 0}, \\"input_selector\\": {\\"select_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"quote\\": false, \\"intree_selector\\": {\\"intree_format\\": \\"none\\", \\"__current_case__\\": 0}}, \\"model_selector\\": {\\"format\\": \\"-nt\\", \\"__current_case__\\": 0, \\"model\\": \\"-gtr\\"}, \\"save_logfile\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.10+galaxy1","type":"tool","uuid":"4faa1735-bea7-4e71-9961-9928bbba13d3","when":null,"workflow_outputs":[{"label":"fasttree_nhx","output_name":"output","uuid":"aacdfe45-eb0c-4f6e-a479-eeb170774757"}]},"56":{"annotation":"","content_id":"Cut1","errors":null,"id":56,"input_connections":{"input":{"id":53,"output_name":"output"}},"inputs":[],"label":null,"name":"Cut","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2379.9833374023438,"top":1837.9999877929688},"post_job_actions":{},"tool_id":"Cut1","tool_state":"{\\"columnList\\": \\"c1,c4\\", \\"delimiter\\": \\"T\\", \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.0.2","type":"tool","uuid":"fa9b41dc-1809-48ff-88fe-c9d2b1ee69bd","when":null,"workflow_outputs":[]},"57":{"annotation":"","content_id":"Grouping1","errors":null,"id":57,"input_connections":{"input1":{"id":54,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2616.9833984375,"top":1154.9666625976563},"post_job_actions":{},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"cat\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"3acff0ad-353a-470e-af62-fe0d51417965","when":null,"workflow_outputs":[]},"58":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","errors":null,"id":58,"input_connections":{"fileNewick":{"id":55,"output_name":"output"}},"inputs":[],"label":null,"name":"Newick Display","outputs":[{"name":"output","type":"svg"}],"position":{"left":2856.984848484848,"top":1586.0079752861868},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"newick_genes_tree_graphs_collection"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"vfs_newick_genes_tree_graphs_collection"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","tool_shed_repository":{"changeset_revision":"b4163d2f64ab","name":"newick_utils","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"condbranchlength\\": {\\"branchlength\\": \\"false\\", \\"__current_case__\\": 1}, \\"condbranchsupport\\": {\\"branchsupport\\": \\"false\\", \\"__current_case__\\": 1}, \\"fileNewick\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"leafFont\\": \\"sans-serif\\", \\"leafSize\\": \\"10px\\", \\"legend\\": \\"\\", \\"outformat\\": \\"svg\\", \\"radial\\": false, \\"scalebar\\": true, \\"width\\": \\"2000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6+galaxy1","type":"tool","uuid":"29e9c814-e0e6-4e1b-ab29-3e9aa2110eac","when":null,"workflow_outputs":[{"label":"newick_genes_tree_graphs_collection","output_name":"output","uuid":"0c22178c-dc85-4137-80e2-f3040b92bd20"}]},"59":{"annotation":"","content_id":"Grouping1","errors":null,"id":59,"input_connections":{"input1":{"id":56,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Group","outputs":[{"name":"out_file1","type":"tabular"}],"position":{"left":2624.9833374023438,"top":1829.9999877929688},"post_job_actions":{},"tool_id":"Grouping1","tool_state":"{\\"groupcol\\": \\"1\\", \\"ignorecase\\": false, \\"ignorelines\\": null, \\"input1\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"operations\\": [{\\"__index__\\": 0, \\"optype\\": \\"cat\\", \\"opcol\\": \\"2\\", \\"opround\\": \\"no\\", \\"opdefault\\": null}], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.4","type":"tool","uuid":"c0762ad5-02d1-4a91-91a9-b730e17e77d5","when":null,"workflow_outputs":[]},"60":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","errors":null,"id":60,"input_connections":{"input":{"id":57,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Tabular-to-FASTA","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2855.984848484848,"top":1152.0761716921845},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","tool_shed_repository":{"changeset_revision":"0a7799698fe5","name":"tabular_to_fasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seq_col\\": \\"2\\", \\"title_col\\": [\\"1\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"81bca5e0-4dfb-4a0f-9f4b-2862813bb013","when":null,"workflow_outputs":[]},"61":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","errors":null,"id":61,"input_connections":{"input":{"id":59,"output_name":"out_file1"}},"inputs":[],"label":null,"name":"Tabular-to-FASTA","outputs":[{"name":"output","type":"fasta"}],"position":{"left":2862.984848484848,"top":1820.0771579226227},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1","tool_shed_repository":{"changeset_revision":"0a7799698fe5","name":"tabular_to_fasta","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"seq_col\\": \\"2\\", \\"title_col\\": [\\"1\\"], \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.1.1","type":"tool","uuid":"0b59476f-e5a7-4a3e-890d-86b5383150c5","when":null,"workflow_outputs":[]},"62":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":62,"input_connections":{"batchmode|input_fastas":{"id":60,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FASTA Merge Files and Filter Unique Sequences","name":"batchmode"}],"label":null,"name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":3140.984848484848,"top":1147.9745890557185},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"merge\\", \\"__current_case__\\": 1, \\"input_fastas\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"34e1170c-e3e2-4555-ac89-90dbbddf352b","when":null,"workflow_outputs":[]},"63":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","errors":null,"id":63,"input_connections":{"batchmode|input_fastas":{"id":61,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FASTA Merge Files and Filter Unique Sequences","name":"batchmode"}],"label":null,"name":"FASTA Merge Files and Filter Unique Sequences","outputs":[{"name":"output","type":"fasta"}],"position":{"left":3136.984848484848,"top":1815.0079752861873},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0","tool_shed_repository":{"changeset_revision":"f546e7278f04","name":"fasta_merge_files_and_filter_unique_sequences","owner":"galaxyp","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"accession_parser\\": \\"^>([^ ]+).*$\\", \\"batchmode\\": {\\"processmode\\": \\"merge\\", \\"__current_case__\\": 1, \\"input_fastas\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"uniqueness_criterion\\": \\"sequence\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2.0","type":"tool","uuid":"995a70a0-bb00-49a7-a347-89e409bac04d","when":null,"workflow_outputs":[]},"64":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","errors":null,"id":64,"input_connections":{"input":{"id":62,"output_name":"output"}},"inputs":[],"label":null,"name":"ClustalW","outputs":[{"name":"output","type":"clustal"},{"name":"dnd","type":"nhx"}],"position":{"left":3420.984848484848,"top":1169.9745890557185},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","tool_shed_repository":{"changeset_revision":"5b097fde95dd","name":"clustalw","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out_order\\": \\"ALIGNED\\", \\"outcontrol\\": {\\"outform\\": \\"fasta\\", \\"__current_case__\\": 0}, \\"range\\": {\\"mode\\": \\"complete\\", \\"__current_case__\\": 0}, \\"tree_calculation\\": {\\"OUTPUTTREE\\": \\"PHYLIP\\", \\"KIMURA\\": false, \\"TOSSGAPS\\": false}, \\"type_conditional\\": {\\"dnarna\\": \\"DNA\\", \\"__current_case__\\": 0, \\"algorithm_conditional\\": {\\"selector\\": \\"\\", \\"__current_case__\\": 0, \\"slow_pairwise_alignments\\": {\\"PWDNAMATRIX\\": \\"IUB\\", \\"PWGAPOPEN\\": null, \\"PWGAPEXT\\": null}}, \\"multiple_alignments\\": {\\"DNAMATRIX\\": \\"IUB\\", \\"GAPOPEN\\": null, \\"GAPEXT\\": null, \\"ENDGAPS\\": false, \\"GAPDIST\\": null, \\"NOPGAP\\": false, \\"NOHGAP\\": false, \\"MAXDIV\\": null, \\"NEGATIVE\\": false, \\"TRANSWEIGHT\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1+galaxy1","type":"tool","uuid":"4d5bef4a-6153-49b7-825a-a75cb2bf0ba0","when":null,"workflow_outputs":[]},"65":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","errors":null,"id":65,"input_connections":{"input":{"id":63,"output_name":"output"}},"inputs":[],"label":null,"name":"ClustalW","outputs":[{"name":"output","type":"clustal"},{"name":"dnd","type":"nhx"}],"position":{"left":3416.984848484848,"top":1837.0079752861873},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1","tool_shed_repository":{"changeset_revision":"5b097fde95dd","name":"clustalw","owner":"devteam","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"out_order\\": \\"ALIGNED\\", \\"outcontrol\\": {\\"outform\\": \\"fasta\\", \\"__current_case__\\": 0}, \\"range\\": {\\"mode\\": \\"complete\\", \\"__current_case__\\": 0}, \\"tree_calculation\\": {\\"OUTPUTTREE\\": \\"PHYLIP\\", \\"KIMURA\\": false, \\"TOSSGAPS\\": false}, \\"type_conditional\\": {\\"dnarna\\": \\"DNA\\", \\"__current_case__\\": 0, \\"algorithm_conditional\\": {\\"selector\\": \\"\\", \\"__current_case__\\": 0, \\"slow_pairwise_alignments\\": {\\"PWDNAMATRIX\\": \\"IUB\\", \\"PWGAPOPEN\\": null, \\"PWGAPEXT\\": null}}, \\"multiple_alignments\\": {\\"DNAMATRIX\\": \\"IUB\\", \\"GAPOPEN\\": null, \\"GAPEXT\\": null, \\"ENDGAPS\\": false, \\"GAPDIST\\": null, \\"NOPGAP\\": false, \\"NOHGAP\\": false, \\"MAXDIV\\": null, \\"NEGATIVE\\": false, \\"TRANSWEIGHT\\": null}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1+galaxy1","type":"tool","uuid":"b188b571-9da6-4d28-a17e-cc0d8f546700","when":null,"workflow_outputs":[]},"66":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","errors":null,"id":66,"input_connections":{"input_selector|input":{"id":64,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FASTTREE","name":"input_selector"}],"label":null,"name":"FASTTREE","outputs":[{"name":"output","type":"nhx"}],"position":{"left":3700.984848484848,"top":1184.9745890557185},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","tool_shed_repository":{"changeset_revision":"e005e659ae21","name":"fasttree","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_selector\\": {\\"maximize\\": \\"min\\", \\"__current_case__\\": 0}, \\"input_selector\\": {\\"select_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"quote\\": false, \\"intree_selector\\": {\\"intree_format\\": \\"none\\", \\"__current_case__\\": 0}}, \\"model_selector\\": {\\"format\\": \\"-nt\\", \\"__current_case__\\": 0, \\"model\\": \\"-gtr\\"}, \\"save_logfile\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.10+galaxy1","type":"tool","uuid":"38119133-c15d-472e-9df2-38ea8519f795","when":null,"workflow_outputs":[]},"67":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","errors":null,"id":67,"input_connections":{"input_selector|input":{"id":65,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool FASTTREE","name":"input_selector"}],"label":null,"name":"FASTTREE","outputs":[{"name":"output","type":"nhx"}],"position":{"left":3696.984848484848,"top":1852.0079752861873},"post_job_actions":{},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1","tool_shed_repository":{"changeset_revision":"e005e659ae21","name":"fasttree","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"advanced_selector\\": {\\"maximize\\": \\"min\\", \\"__current_case__\\": 0}, \\"input_selector\\": {\\"select_format\\": \\"fasta\\", \\"__current_case__\\": 0, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"quote\\": false, \\"intree_selector\\": {\\"intree_format\\": \\"none\\", \\"__current_case__\\": 0}}, \\"model_selector\\": {\\"format\\": \\"-nt\\", \\"__current_case__\\": 0, \\"model\\": \\"-gtr\\"}, \\"save_logfile\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.10+galaxy1","type":"tool","uuid":"02345d58-0ef8-49dd-ab98-ee74a5e69adc","when":null,"workflow_outputs":[]},"68":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","errors":null,"id":68,"input_connections":{"fileNewick":{"id":66,"output_name":"output"}},"inputs":[],"label":null,"name":"Newick Display","outputs":[{"name":"output","type":"svg"}],"position":{"left":3976.9833374023438,"top":1162.9666625976563},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"all_samples_phylogenetic_tree_based_amrs"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"all_samples_phylogenetic_tree_based_amrs"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","tool_shed_repository":{"changeset_revision":"b4163d2f64ab","name":"newick_utils","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"condbranchlength\\": {\\"branchlength\\": \\"false\\", \\"__current_case__\\": 1}, \\"condbranchsupport\\": {\\"branchsupport\\": \\"false\\", \\"__current_case__\\": 1}, \\"fileNewick\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"leafFont\\": \\"sans-serif\\", \\"leafSize\\": \\"10px\\", \\"legend\\": \\"\\", \\"outformat\\": \\"svg\\", \\"radial\\": false, \\"scalebar\\": true, \\"width\\": \\"2000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6+galaxy1","type":"tool","uuid":"da6052c7-ec6e-4105-96d6-4c3d1204eb99","when":null,"workflow_outputs":[{"label":"all_samples_phylogenetic_tree_based_amrs","output_name":"output","uuid":"1f9cb2cf-219f-48de-8058-d6d45f3b3158"}]},"69":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","errors":null,"id":69,"input_connections":{"fileNewick":{"id":67,"output_name":"output"}},"inputs":[],"label":null,"name":"Newick Display","outputs":[{"name":"output","type":"svg"}],"position":{"left":3972.9833374023438,"top":1792.9999877929688},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"all_samples_phylogenetic_tree_based_vfs"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"all_samples_phylogenetic_tree_based_vfs"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1","tool_shed_repository":{"changeset_revision":"b4163d2f64ab","name":"newick_utils","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"condbranchlength\\": {\\"branchlength\\": \\"false\\", \\"__current_case__\\": 1}, \\"condbranchsupport\\": {\\"branchsupport\\": \\"false\\", \\"__current_case__\\": 1}, \\"fileNewick\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"leafFont\\": \\"sans-serif\\", \\"leafSize\\": \\"10px\\", \\"legend\\": \\"\\", \\"outformat\\": \\"svg\\", \\"radial\\": false, \\"scalebar\\": true, \\"width\\": \\"2000\\", \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.6+galaxy1","type":"tool","uuid":"8af67049-73c6-4121-a6b0-daa8db404a24","when":null,"workflow_outputs":[{"label":"all_samples_phylogenetic_tree_based_vfs","output_name":"output","uuid":"6bb4b32b-7cca-4e04-8120-be9f64ccba39"}]}},"tags":["name:Collection","name:microGalaxy","name:PathoGFAIR","name:IWC"],"uuid":"cb26018e-9d2f-4513-a372-122474df703e","version":58},"readme":"# Pathogen Detection: PathoGFAIR Samples Aggregation and Visualisation\\n\\nIn this workflow, we will aggregate results and use the results from 3 workflows (**Nanopore Preprocessing**, **Gene-based Pathogen Identification** and **Nanopore Allele-based Pathogen Identification**) to help track pathogens among samples and visualise all performed analysis by:\\n\\n1. Drawing a presence-absence heatmap of the identified VF genes within all samples to visualise in which samples these genes can be found.\\n2. Drawing a phylogenetic tree for each pathogenic genes detected, where we will relate the contigs of the samples together where this gene is found.\\n3. Plotting QC reads, host reads, mapping coverage and depth, and SNP analysis.\\n\\nWith these types of visualisations, we can have an overview of all samples and the genes, but also how samples are related to each other, which common pathogenic genes they share. Given the time of the sampling and the location one can easily identify using these graphs, where and when the contamination has occurred among the different samples.\\n\\nIf you\'re unsure how to use this workflows, or if you want to see it in action with test datasets, it is included in our detailed training material for foodborne pathogen detection and tracking. You can find step-by-step instructions and practical examples in the following [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-04-24\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ amr_identified_by_ncbi\\"]@{ shape: docs }\\n1[\\"ℹ️ vfs_of_genes_identified_by_vfdb\\"]@{ shape: docs }\\n2[\\"ℹ️ amrs\\"]@{ shape: docs }\\n3[\\"ℹ️ contigs\\"]@{ shape: docs }\\n4[\\"ℹ️ vfs\\"]@{ shape: docs }\\n5[\\"ℹ️ removed_hosts_percentage_tabular\\"]@{ shape: doc }\\n6[\\"ℹ️ mapping_mean_depth_per_sample\\"]@{ shape: doc }\\n7[\\"ℹ️ mapping_coverage_percentage_per_sample\\"]@{ shape: doc }\\n8[\\"ℹ️ number_of_variants_per_sample\\"]@{ shape: doc }\\n9[\\"ℹ️ metadata\\"]@{ shape: doc }\\n10[\\"Filter failed datasets\\"]@{ shape: process }\\n0 --> 10\\n11[\\"Filter failed datasets\\"]@{ shape: process }\\n1 --> 11\\n12[\\"Filter failed datasets\\"]@{ shape: process }\\n2 --> 12\\n13[\\"Filter failed datasets\\"]@{ shape: process }\\n3 --> 13\\n14[\\"Filter failed datasets\\"]@{ shape: process }\\n4 --> 14\\n15[\\"Remove beginning\\"]@{ shape: process }\\n10 --> 15\\n16[\\"Remove beginning\\"]@{ shape: process }\\n11 --> 16\\n17[\\"Remove beginning\\"]@{ shape: process }\\n12 --> 17\\n18[\\"Collapse Collection\\"]@{ shape: process }\\n13 --> 18\\n19[\\"Collapse Collection\\"]@{ shape: process }\\n14 --> 19\\n20[\\"Remove beginning\\"]@{ shape: process }\\n14 --> 20\\n21[\\"Count\\"]@{ shape: process }\\n15 --> 21\\n22[\\"Count\\"]@{ shape: process }\\n16 --> 22\\n23[\\"Group\\"]@{ shape: process }\\n16 --> 23\\n24[\\"Unique\\"]@{ shape: process }\\n17 --> 24\\n25[\\"Split by group\\"]@{ shape: process }\\n19 --> 25\\n26[\\"Unique\\"]@{ shape: process }\\n20 --> 26\\n27[\\"Cut\\"]@{ shape: process }\\n21 --> 27\\n28[\\"Cut\\"]@{ shape: process }\\n22 --> 28\\n29[\\"Filter empty datasets\\"]@{ shape: process }\\n23 --> 29\\n30[\\"Cut\\"]@{ shape: process }\\n24 --> 30\\n31[\\"Cut\\"]@{ shape: process }\\n25 --> 31\\n32[\\"Cut\\"]@{ shape: process }\\n26 --> 32\\n33[\\"Collapse Collection\\"]@{ shape: process }\\n27 --> 33\\n34[\\"Collapse Collection\\"]@{ shape: process }\\n28 --> 34\\n35[\\"Column join\\"]@{ shape: process }\\n29 --> 35\\n36[\\"bedtools getfasta\\"]@{ shape: process }\\n18 --> 36\\n30 --> 36\\n37[\\"Remove beginning\\"]@{ shape: process }\\n31 --> 37\\n38[\\"bedtools getfasta\\"]@{ shape: process }\\n18 --> 38\\n32 --> 38\\n39[\\"Column Regex Find And Replace\\"]@{ shape: process }\\n33 --> 39\\n40[\\"Column Regex Find And Replace\\"]@{ shape: process }\\n34 --> 40\\n41[\\"Column Regex Find And Replace\\"]@{ shape: process }\\n35 --> 41\\n42[\\"Regex Find And Replace\\"]@{ shape: process }\\n36 --> 42\\n43[\\"bedtools getfasta\\"]@{ shape: process }\\n18 --> 43\\n37 --> 43\\n44[\\"Regex Find And Replace\\"]@{ shape: process }\\n38 --> 44\\n45[\\"Multi-Join\\"]@{ shape: process }\\n40 --> 45\\n39 --> 45\\n46[\\"Heatmap w ggplot\\"]@{ shape: process }\\n41 --> 46\\n47[\\"Filter empty datasets\\"]@{ shape: process }\\n42 --> 47\\n48[\\"ClustalW\\"]@{ shape: process }\\n43 --> 48\\n49[\\"Filter empty datasets\\"]@{ shape: process }\\n44 --> 49\\n50[\\"Replace Text\\"]@{ shape: process }\\n45 --> 50\\n51[\\"FASTA-to-Tabular\\"]@{ shape: process }\\n47 --> 51\\n52[\\"Filter empty datasets\\"]@{ shape: process }\\n48 --> 52\\n53[\\"FASTA-to-Tabular\\"]@{ shape: process }\\n49 --> 53\\n54[\\"Cut\\"]@{ shape: process }\\n51 --> 54\\n55[\\"FASTTREE\\"]@{ shape: process }\\n52 --> 55\\n56[\\"Cut\\"]@{ shape: process }\\n53 --> 56\\n57[\\"Group\\"]@{ shape: process }\\n54 --> 57\\n58[\\"Newick Display\\"]@{ shape: process }\\n55 --> 58\\n59[\\"Group\\"]@{ shape: process }\\n56 --> 59\\n60[\\"Tabular-to-FASTA\\"]@{ shape: process }\\n57 --> 60\\n61[\\"Tabular-to-FASTA\\"]@{ shape: process }\\n59 --> 61\\n62[\\"FASTA Merge Files and Filter Unique Sequences\\"]@{ shape: process }\\n60 --> 62\\n63[\\"FASTA Merge Files and Filter Unique Sequences\\"]@{ shape: process }\\n61 --> 63\\n64[\\"ClustalW\\"]@{ shape: process }\\n62 --> 64\\n65[\\"ClustalW\\"]@{ shape: process }\\n63 --> 65\\n66[\\"FASTTREE\\"]@{ shape: process }\\n64 --> 66\\n67[\\"FASTTREE\\"]@{ shape: process }\\n65 --> 67\\n68[\\"Newick Display\\"]@{ shape: process }\\n66 --> 68\\n69[\\"Newick Display\\"]@{ shape: process }\\n67 --> 69\\n```\\n","updated":"2024-04-24T00:00:00","trsID":"#workflow/github.com/iwc-workflows/pathogen-detection-pathogfair-samples-aggregation-and-visualisation/main","dockstore_id":27331,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.12744324","tests":[{"doc":"Test outline for Pathogen-Detection-PathoGFAIR-Samples-Aggregation-and-Visualisation","job":{"removed_hosts_percentage_tabular":{"class":"File","path":"test-data/removed_hosts_percentage_tabular.tabular","filetype":"tabular"},"mapping_mean_depth_per_sample":{"class":"File","path":"test-data/mapping_mean_depth_per_sample.tabular","filetype":"tabular"},"mapping_coverage_percentage_per_sample":{"class":"File","path":"test-data/mapping_coverage_percentage_per_sample.tabular","filetype":"tabular"},"number_of_variants_per_sample":{"class":"File","path":"test-data/number_of_variants_per_sample.tabular","filetype":"tabular"},"amr_identified_by_ncbi":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode10.fastq.gz","path":"test-data/amr_identified_by_ncbi_collection_of_all_samples_Spike3bBarcode10.fastq.gz.tabular"},{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode12.fastq.gz","path":"test-data/amr_identified_by_ncbi_collection_of_all_samples_Spike3bBarcode12.fastq.gz.tabular"}]},"vfs_of_genes_identified_by_vfdb":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode10.fastq.gz","path":"test-data/vfs_of_genes_identified_by_vfdb_collection_of_all_samples_Spike3bBarcode10.fastq.gz.tabular"},{"class":"File","identifier":"collection_of_all_samples_Spike3bBarcode12.fastq.gz","path":"test-data/vfs_of_genes_identified_by_vfdb_collection_of_all_samples_Spike3bBarcode12.fastq.gz.tabular"}]},"amrs":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"split_file_000000.txt","path":"test-data/amrs_split_file_000000.txt.tabular"},{"class":"File","identifier":"split_file_000001.txt","path":"test-data/amrs_split_file_000001.txt.tabular"}]},"vfs":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"split_file_000000.txt","path":"test-data/vfs_split_file_000000.txt.tabular"},{"class":"File","identifier":"split_file_000001.txt","path":"test-data/vfs_split_file_000001.txt.tabular"}]},"contigs":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"split_file_000000.txt","path":"test-data/contigs_split_file_000000.txt.fasta"},{"class":"File","identifier":"split_file_000001.txt","path":"test-data/contigs_split_file_000001.txt.fasta"}]},"metadata":null},"outputs":{"adjusted_abricate_vfs_tabular_part1":{"attributes":{},"element_tests":{"YP_001006764":{"path":"test-data/adjusted_abricate_vfs_tabular_part1_YP_001006764.tabular"}}}}}]}],"path":"./workflows/microbiome/pathogen-identification/pathogen-detection-pathogfair-samples-aggregation-and-visualisation"},{"version":1.2,"workflows":[{"name":"main","subclass":"Galaxy","publish":true,"primaryDescriptorPath":"/Taxonomy-Profiling-and-Visualization-with-Krona.ga","testParameterFiles":["/Taxonomy-Profiling-and-Visualization-with-Krona-tests.yml"],"authors":[{"name":"Engy Nasr","orcid":"0000-0001-9047-4215","url":"https://orcid.org/0000-0001-9047-4215"},{"name":"Bérénice Batut","orcid":"0000-0001-9852-1987","url":"https://orcid.org/0000-0001-9852-1987"},{"name":"Paul Zierep","orcid":"0000-0003-2982-388X","url":"https://orcid.org/0000-0003-2982-388X"}],"definition":{"a_galaxy_workflow":"true","annotation":"Microbiome - Taxonomy Profiling","comments":[{"child_steps":[4],"color":"red","data":{"title":"Taxonomy Visualisation"},"id":1,"position":[685,167.31666564941406],"size":[236,191],"type":"frame"},{"child_steps":[2,3,1],"color":"red","data":{"title":"Taxonomy Profiling"},"id":0,"position":[4.800000000000011,165.01666564941405],"size":[663.2,295.6],"type":"frame"}],"creator":[{"class":"Person","identifier":"0000-0001-9047-4215","name":"Engy Nasr","url":"https://orcid.org/0000-0001-9047-4215"},{"class":"Person","identifier":"0000-0001-9852-1987","name":"Bérénice Batut","url":"https://orcid.org/0000-0001-9852-1987"},{"class":"Person","identifier":"0000-0003-2982-388X","name":"Paul Zierep"}],"format-version":"0.1","license":"MIT","release":"0.1","name":"Taxonomy Profiling and Visualization with Krona","report":{"markdown":"# Taxonomy Profiling and Visualisation with Krona Workflow Report\\n\\nIn this workflow you can identify your microbial community, but identifying all possible Taxon found in your samples, from the kingdom level down to the species level. In the application of the pathogen detection, this workflow gives an overview of the possible bacterial found in samples and what might be a possible suspect for pathogenicity.\\n\\nThe workflow uses Kraken2 tool and allow the user to choose any Kraken2 database or other added ones depending on the user application or the input samples. All Kraken2 tool databases are available in Galaxy. More description about these databases can be found [here](https://benlangmead.github.io/aws-indexes/k2). Other databases can be also added to the tool, just contact us anytime for that. We recommend using StandardPF database.\\n\\nVisualization is done using Krona Pie Chart, and you can replace it with any other tool. e.g. Phinch interactive tool, which is also available in Galaxy\\n\\n## Workflow Inputs:\\n\\nThe per-processing workflow main output (Collection of all samples reads after quality retaining and hosts filtering)\\n\\n## Workflow Outputs are:\\n\\n### Taxonomy Tabular Report\\n\\nThe report shows the abundance of every taxon from the kingdom level down to the species level\\n\\n```galaxy\\nhistory_dataset_display(output=\\"converted_kraken_report\\")\\n```\\n\\n### Taxonomy Profiling Visualization with Krona\\nAn interactive Pie chart showing the different identified taxons along with their percentages, which is one of the ways to visualize the previous taxonomy tabular\\n\\n```galaxy\\nhistory_dataset_as_image(output=\\"krona_pie_chart\\")\\n```\\n"},"steps":{"0":{"annotation":"Output collection from the Nanopore Preprocessing workflow","content_id":null,"errors":null,"id":0,"input_connections":{},"inputs":[{"description":"Output collection from the Nanopore Preprocessing workflow","name":"collection_of_preprocessed_samples"}],"label":"collection_of_preprocessed_samples","name":"Input dataset collection","outputs":[],"position":{"left":0,"top":0},"tool_id":null,"tool_state":"{\\"optional\\": false, \\"tag\\": null, \\"collection_type\\": \\"list\\"}","tool_version":null,"type":"data_collection_input","uuid":"c4a14ff1-1f0c-4cb5-aefe-b56f0a84404e","when":null,"workflow_outputs":[]},"1":{"annotation":"kraken_database_for_user_to_choose","content_id":null,"errors":null,"id":1,"input_connections":{},"inputs":[{"description":"kraken_database_for_user_to_choose","name":"kraken_database"}],"label":"kraken_database","name":"Input parameter","outputs":[],"position":{"left":24.79997558593749,"top":276.7833526611328},"tool_id":null,"tool_state":"{\\"restrictOnConnections\\": true, \\"parameter_type\\": \\"text\\", \\"optional\\": false}","tool_version":null,"type":"parameter_input","uuid":"cc59191c-1bc9-4007-b88c-8765ff2dfefd","when":null,"workflow_outputs":[]},"2":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1","errors":null,"id":2,"input_connections":{"kraken2_database":{"id":1,"output_name":"output"},"single_paired|input_sequences":{"id":0,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Kraken2","name":"single_paired"}],"label":null,"name":"Kraken2","outputs":[{"name":"report_output","type":"tabular"},{"name":"output","type":"tabular"}],"position":{"left":231.99998779296874,"top":205.00002746582032},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"kraken2_with_pluspf_database_output"},"action_type":"RenameDatasetAction","output_name":"output"},"RenameDatasetActionreport_output":{"action_arguments":{"newname":"kraken2_with_pluspf_database_output_report"},"action_type":"RenameDatasetAction","output_name":"report_output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1","tool_shed_repository":{"changeset_revision":"20e2f64aa1fe","name":"kraken2","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"confidence\\": \\"0.1\\", \\"kraken2_database\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"min_base_quality\\": \\"0\\", \\"minimum_hit_groups\\": \\"2\\", \\"quick\\": false, \\"report\\": {\\"create_report\\": true, \\"use_mpa_style\\": false, \\"report_zero_counts\\": false, \\"report_minimizer_data\\": false}, \\"single_paired\\": {\\"single_paired_selector\\": \\"no\\", \\"__current_case__\\": 2, \\"input_sequences\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"split_reads\\": false, \\"use_names\\": false, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.1.1+galaxy1","type":"tool","uuid":"ddb4e42b-6381-411e-9328-1c7959ecda47","when":null,"workflow_outputs":[{"label":"kraken2_with_pluspf_database_output","output_name":"output","uuid":"26f9041f-d8a4-4f64-8a53-404d5c7e2f61"},{"label":"kraken2_with_pluspf_database_output_report","output_name":"report_output","uuid":"f2ddd05b-66a0-4cea-9da6-ed107eecbe1b"}]},"3":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/iuc/krakentools_kreport2krona/krakentools_kreport2krona/1.2+galaxy1","errors":null,"id":3,"input_connections":{"report":{"id":2,"output_name":"report_output"}},"inputs":[],"label":null,"name":"Krakentools: Convert kraken report file","outputs":[{"name":"output","type":"tabular"}],"position":{"left":447.9999877929688,"top":205.31667785644532},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"converted_kraken_report"},"action_type":"RenameDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/iuc/krakentools_kreport2krona/krakentools_kreport2krona/1.2+galaxy1","tool_shed_repository":{"changeset_revision":"2deaada69b14","name":"krakentools_kreport2krona","owner":"iuc","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"intermediate_ranks\\": false, \\"report\\": {\\"__class__\\": \\"ConnectedValue\\"}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"1.2+galaxy1","type":"tool","uuid":"7464e3a0-2253-4fe5-832f-ea155a660b08","when":null,"workflow_outputs":[{"label":"converted_kraken_report","output_name":"output","uuid":"c490ab8d-1b1d-4914-aa51-a62a967a4018"}]},"4":{"annotation":"","content_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","errors":null,"id":4,"input_connections":{"type_of_data|input":{"id":3,"output_name":"output"}},"inputs":[{"description":"runtime parameter for tool Krona pie chart","name":"type_of_data"}],"label":null,"name":"Krona pie chart","outputs":[{"name":"output","type":"html"}],"position":{"left":701,"top":212.31666564941406},"post_job_actions":{"RenameDatasetActionoutput":{"action_arguments":{"newname":"krona_pie_chart"},"action_type":"RenameDatasetAction","output_name":"output"},"TagDatasetActionoutput":{"action_arguments":{"tags":"taxonomy_profiling_visualisation_krona_pie_chart"},"action_type":"TagDatasetAction","output_name":"output"}},"tool_id":"toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0","tool_shed_repository":{"changeset_revision":"e9005d1f3cfd","name":"taxonomy_krona_chart","owner":"crs4","tool_shed":"toolshed.g2.bx.psu.edu"},"tool_state":"{\\"combine_inputs\\": false, \\"root_name\\": \\"Root\\", \\"type_of_data\\": {\\"type_of_data_selector\\": \\"text\\", \\"__current_case__\\": 1, \\"input\\": {\\"__class__\\": \\"ConnectedValue\\"}}, \\"__page__\\": null, \\"__rerun_remap_job_id__\\": null}","tool_version":"2.7.1+galaxy0","type":"tool","uuid":"8b130a6a-6fb3-4498-ac19-0d2df97c09e1","when":null,"workflow_outputs":[{"label":"krona_pie_chart","output_name":"output","uuid":"dcc857c6-f579-4c74-84ec-bb6ce3794a56"}]}},"tags":["name:Collection","name:microGalaxy","name:PathoGFAIR","name:IWC"],"uuid":"a55428c8-9934-4941-93c4-3920b4ba3366","version":58},"readme":"# Taxonomy Profiling and Visualisation with Krona\\n\\nIn this workflow, we identify the different organisms found in our samples by assigning taxonomy levels to the reads starting from the kingdom level down to the species level and visualise the result.\\n\\nIt’s important to check what might be the species of a possible pathogen to be found, it gets us closer to the investigation as well as discovering possible multiple pathogenetic infections if any existed.\\n\\nFor taxonomy profiling Kraken2 tool is used along with one of its standard databases available on Galaxy, you can freely choose between Kraken2 different databases based on your input datasets. For visualisation multiple tools can be used, Krona pie chart (as default in this workflow), Phinch interactive tool, Pavian, etc.\\n\\n## Input Datasets\\n- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format, the output of **Nanopore Preprocessing** workflow.\\n\\n## Output Datasets\\n- Taxonomy profiling Tabular file, visualisation figures and interactive pie charts.\\n\\nIf you\'re unsure how to use this workflows, or if you want to see it in action with test datasets, it is included in our detailed training material for foodborne pathogen detection and tracking. You can find step-by-step instructions and practical examples in the following [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)\\n","changelog":"# Changelog\\n\\n## [0.1] - 2024-04-25\\n\\nFirst release.\\n","diagrams":"# Workflow diagrams\\n\\n## Taxonomy Profiling and Visualization with Krona\\n\\n```mermaid\\ngraph LR\\n0[\\"ℹ️ collection_of_preprocessed_samples\\"]@{ shape: docs }\\n1[\\"ℹ️ kraken_database\\"]@{ shape: lean-l }\\n2[\\"Kraken2\\"]@{ shape: process }\\n1 --> 2\\n0 --> 2\\n3[\\"Krakentools: Convert kraken report file\\"]@{ shape: process }\\n2 --> 3\\n4[\\"Krona pie chart\\"]@{ shape: process }\\n3 --> 4\\n```\\n","updated":"2024-04-25T00:00:00","trsID":"#workflow/github.com/iwc-workflows/taxonomy-profiling-and-visualization-with-krona/main","dockstore_id":27333,"categories":[],"collections":["Microbiome"],"doi":"10.5281/zenodo.12744308","tests":[{"doc":"Test outline for Taxonomy-Profiling-and-Visualization-with-Krona","job":{"collection_of_preprocessed_samples":{"class":"Collection","collection_type":"list","elements":[{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode10.fastq.gz","filetype":"fastqsanger.gz"},{"class":"File","identifier":"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12","location":"https://zenodo.org/record/12190648/files/nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12.fastq.gz","filetype":"fastqsanger.gz"}]},"kraken_database":"k2_minusb_20210517"},"outputs":{"converted_kraken_report":{"attributes":{},"element_tests":{"nanopore_preprocessed_collection_of_all_samples_Spike3bBarcode12":{"path":"test-data/converted_kraken_report_collection_of_all_samples_Spike3bBarcode12.tabular"}}}}}]}],"path":"./workflows/microbiome/pathogen-identification/taxonomy-profiling-and-visualization-with-krona"}]'),ke=Ae,Fe=ie("workflow",()=>{const t=N(),s=b(null),a=p(()=>ke.flatMap(_=>_.workflows)),o=_=>a.value.find(m=>m.trsID===_),e=p(()=>Array.from(new Set(a.value.flatMap(_=>_.categories)))),n=p(()=>Array.from(new Set(a.value.flatMap(_=>_.collections)))),i=p(()=>Array.from(new Set(a.value.flatMap(_=>_.definition.tags)))),r=b([]),l=p(()=>{const _=a.value.flatMap(m=>m.collections);return Array.from(new Set(_)).sort()}),u=p(()=>l.value);function c(_){r.value.includes(_)?r.value=[]:r.value=[_]}return{workflow:s,allCategories:e,allCollections:n,allTags:i,allWorkflows:a,getWorkflowByTrsId:o,setWorkflow:()=>{s.value=a.value.find(_=>_.trsID===t.params.id)},selectedFilters:r,allFilters:l,validFilters:u,invalidFilters:p(()=>l.value.filter(_=>!u.value.includes(_))),toggleFilter:c}}),Ce=f({name:"LayoutLoader",inheritAttrs:!1,props:{name:String,layoutProps:Object},setup(t,s){return()=>y(h[t.name],t.layoutProps,s.slots)}}),He=f({name:"NuxtLayout",inheritAttrs:!1,props:{name:{type:[String,Boolean,Object],default:null},fallback:{type:[String,Object],default:null}},setup(t,s){const a=_e(),o=C(G),e=o===le()?N():o,n=p(()=>{let l=j(t.name)??e.meta.layout??"default";return l&&!(l in h)&&t.fallback&&(l=j(t.fallback)),l}),i=b();s.expose({layoutRef:i});const r=a.deferHydration();if(a.isHydrating){const l=a.hooks.hookOnce("app:error",r);ue().beforeEach(l)}return()=>{const l=n.value&&n.value in h,u=e.meta.layoutTransition??re;return pe(l&&u,{default:()=>y(de,{suspensible:!0,onResolve:()=>{ce(r)}},{default:()=>y(De,{layoutProps:g(s.attrs,{ref:i}),key:n.value||void 0,name:n.value,shouldProvide:!t.name,hasTransition:!!u},s.slots)})}).default()}}}),De=f({name:"NuxtLayoutProvider",inheritAttrs:!1,props:{name:{type:[String,Boolean]},layoutProps:{type:Object},hasTransition:{type:Boolean},shouldProvide:{type:Boolean}},setup(t,s){const a=t.name;return t.shouldProvide&&T(me,{isCurrent:o=>a===(o.meta.layout??"default")}),()=>{var o,e;return!a||typeof a=="string"&&!(a in h)?(e=(o=s.slots).default)==null?void 0:e.call(o):y(Ce,{key:a,layoutProps:t.layoutProps,name:a},s.slots)}}});export{Se as _,Ie as a,Re as b,Fe as c,He as d,Me as e,Ve as f,Te as g,P as i,U as n,Pe as s,je as u};
